--- distdir_deps.bzl.orig	2023-09-05 16:10:13.000617742 +0900
+++ distdir_deps.bzl	2023-09-05 16:10:42.471484181 +0900
@@ -163,6 +163,10 @@
     },
     "com_google_absl": {
         "archive": "20211102.0.tar.gz",
+	"patch_args": ["-p1"],
+        "patches": [
+            "//:abseil-missing-stdint.patch"
+        ],
         "sha256": "dcf71b9cba8dc0ca9940c4b316a0c796be8fab42b070bb6b7cab62b48f0e66c4",
         "urls": [
             "https://mirror.bazel.build/github.com/abseil/abseil-cpp/archive/refs/tags/20211102.0.tar.gz",
