diff -uar docker-machine-kvm-0.10.0.orig/kvm.go docker-machine-kvm-0.10.0/kvm.go
--- docker-machine-kvm-0.10.0.orig/kvm.go	2017-04-18 00:06:36.000000000 +0300
+++ docker-machine-kvm-0.10.0/kvm.go	2017-11-03 19:24:27.219136476 +0200
@@ -368,9 +368,9 @@
 			return err
 		}
 		mode := info.Mode()
-		if mode&0001 != 1 {
+		if mode&0011 != 0011 {
 			log.Debugf("Setting executable bit set on %s", dir)
-			mode |= 0001
+			mode |= 0011
 			os.Chmod(dir, mode)
 		}
 	}
