--- a/Makefile.OCaml	2011-04-16 17:35:38.000000000 -0300
+++ b/Makefile.OCaml	2014-09-25 13:18:43.239151396 -0300
@@ -71,8 +71,8 @@ OCAMLLIBDIR=$(shell ocamlc -v | tail -1
 #
 # This should be set to an appropriate value automatically, depending
 # on whether the lablgtk library is available
-LABLGTKLIB=$(OCAMLLIBDIR)/lablgtk
-LABLGTK2LIB=$(OCAMLLIBDIR)/lablgtk2
+LABLGTKLIB=$(OCAMLLIBDIR)/site-lib/lablgtk
+LABLGTK2LIB=$(OCAMLLIBDIR)/site-lib/lablgtk2
 ##BCP [3/2007]: Removed temporarily, since the OSX UI is not working well 
 ## at the moment and we don't want to confuse people by building it by default
 ifeq ($(OSARCH),osx)
@@ -92,7 +92,7 @@ buildexecutable::
 
 INCLFLAGS=-I lwt -I ubase -I system
 CAMLFLAGS+=$(INCLFLAGS)
-CAMLFLAGS+=-I system/$(SYSTEM) -I lwt/$(SYSTEM)
+CAMLFLAGS+=-I system/$(SYSTEM) -I lwt/$(SYSTEM) -I $(LABLGTK2LIB)
 
 ifeq ($(OSARCH),win32)
   # Win32 system
