From fe121ce70ad5e99bd3b3b896dfcbe439dd22716c Mon Sep 17 00:00:00 2001
From: dgod <dgod.osa@gmail.com>
Date: Sun, 21 Feb 2016 11:45:35 +0800
Subject: [PATCH] fix not honor ctl+alt+backspacer for restarting

---
 src/lxdm.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/lxdm.c b/src/lxdm.c
index d81b587..e00d219 100644
--- a/src/lxdm.c
+++ b/src/lxdm.c
@@ -962,6 +962,11 @@ static void on_xserver_stop(void *data,int pid, int status)
 		s->dpy=NULL;
 		ui_drop();
 		lxdm_startx(s);
+		#ifndef DISABLE_XAUTH
+		char temp[256];
+		sprintf(temp,"/var/run/lxdm/lxdm-:%d.auth",s->display);
+		setenv("XAUTHORITY",temp,1);
+		#endif
 		ui_prepare();
 		lxsession_set_active(s);
 	}
-- 
2.1.4

