diff -Nur lxdm-0.3.0.orig//src/lxdm.c lxdm-0.3.0/src/lxdm.c
--- lxdm-0.3.0.orig//src/lxdm.c	2010-09-24 10:43:49.000000000 -0500
+++ lxdm-0.3.0/src/lxdm.c	2010-10-10 23:01:24.711716765 -0500
@@ -150,6 +150,7 @@
     while( waitpid(-1, 0, WNOHANG) > 0 ) ;
 }
 
+/* Sorry, -ENOPAM here
 static void close_pam_session(pam_handle_t *pamh)
 {
     int err;
@@ -159,6 +160,7 @@
     pam_end(pamh, err);
     pamh = NULL;
 }
+*/
 
 static LXSession *lxsession_find_greeter(void)
 {
