diff -Naur lxdm-20120419_2ce143c.orig/src/lxdm.c lxdm-20120419_2ce143c/src/lxdm.c
--- lxdm-20120419_2ce143c.orig/src/lxdm.c	2012-04-19 16:31:16.000000000 +0200
+++ lxdm-20120419_2ce143c/src/lxdm.c	2012-04-28 18:17:25.186023147 +0200
@@ -1490,6 +1490,10 @@
 							  NULL))
 		{
 			setenv("XDG_SESSION_COOKIE", ck_connector_get_cookie(s->ckc), 1);
+			/* I really don't like doing this, but since both kdm and gdm
+			   set this in the environment, and our xinitrc scripts depend
+			   on it, here we go with a minor patch... */
+			setenv("DESKTOP_SESSION", session_name, 1);
 		}
 		else
 		{
