diff -Naur lxdm-0.4.0.orig/src/lxdm.c lxdm-0.4.0/src/lxdm.c
--- lxdm-0.4.0.orig/src/lxdm.c	2011-07-13 14:00:34.000000000 +0200
+++ lxdm-0.4.0/src/lxdm.c	2011-07-22 17:27:34.000000000 +0200
@@ -153,6 +153,7 @@
     while( waitpid(-1, 0, WNOHANG) > 0 ) ;
 }
 
+/* Sorry, -ENOPAM here
 #if HAVE_LIBPAM
 static void close_pam_session(pam_handle_t *pamh)
 {
@@ -164,6 +165,7 @@
     pamh = NULL;
 }
 #endif
+*/
 
 static LXSession *lxsession_find_greeter(void)
 {
