diff -Nur lxdm-0.3.0.orig//data/Xsession lxdm-0.3.0/data/Xsession
--- lxdm-0.3.0.orig//data/Xsession	2010-10-12 20:13:01.316302089 -0500
+++ lxdm-0.3.0/data/Xsession	2010-10-12 20:12:40.989504750 -0500
@@ -31,10 +31,6 @@
 elif [ -x /etc/X11/Xsession ]; then
 # mandriva, debian, ubuntu
 	exec /etc/X11/Xsession $LXSESSION
-elif [ -x /etc/X11/xinit/xinitrc ]; then
-#suse
-	export WINDOWMANAGER=$LXSESSION
-	exec -l $SHELL -c /etc/X11/xinit/xinitrc
 else
 # unknown, user should custom /etc/lxdm/xinitrc self
 	if [ -x /etc/lxdm/xinitrc ]; then
