diff -Naur lxdm-0.4.0.orig/data/Xsession lxdm-0.4.0/data/Xsession
--- lxdm-0.4.0.orig/data/Xsession	2011-03-17 16:35:47.000000000 +0100
+++ lxdm-0.4.0/data/Xsession	2011-07-22 17:23:49.000000000 +0200
@@ -27,10 +27,6 @@
 elif [ -x /etc/X11/Xsession ]; then
 # mandriva, debian, ubuntu
 	exec /etc/X11/Xsession "$LXSESSION"
-elif [ -x /etc/X11/xinit/xinitrc ]; then
-#suse
-	export WINDOWMANAGER=$LXSESSION
-	exec -l $SHELL -c /etc/X11/xinit/xinitrc
 else
 # unknown, user should custom /etc/lxdm/xinitrc self
 	if [ -x /etc/lxdm/xinitrc ]; then
