diff -Naur neverball-1.6.0.orig/Makefile neverball-1.6.0/Makefile
--- neverball-1.6.0.orig/Makefile	2014-05-21 15:21:43.000000000 +0200
+++ neverball-1.6.0/Makefile	2015-04-23 04:30:07.093366557 +0200
@@ -21,7 +21,7 @@
 # Paths (packagers might want to set DATADIR and LOCALEDIR)
 
 USERDIR   := .neverball
-DATADIR   := ./data
+DATADIR   := /usr/share/neverball/data
 LOCALEDIR := ./locale
 
 ifeq ($(PLATFORM),mingw)
