
Index: opencascade/ros/adm/make/BinLPlugin/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/BinLPlugin/Makefile.am
+++ opencascade/ros/adm/make/BinLPlugin/Makefile.am
@@ -23,3 +23,4 @@
 ../TKernel/libTKernel.la \
 ../TKG3d/libTKG3d.la \
  $(STLPort_LIB) $(CSF_SOCKETLibs_LIB) $(CSF_FPELibs_LIB) $(CSF_ThreadLibs_LIB) $(CSF_advapi32_LIB) $(CSF_gdi32_LIB) $(CSF_user32_LIB) $(CSF_kernel32_LIB)  
+libBinLPlugin_la_LDFLAGS = -release 6.3.0
Index: opencascade/ros/adm/make/BinPlugin/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/BinPlugin/Makefile.am
+++ opencascade/ros/adm/make/BinPlugin/Makefile.am
@@ -30,3 +30,4 @@
 ../TKMath/libTKMath.la \
 ../TKG3d/libTKG3d.la \
  $(STLPort_LIB) $(CSF_user32_LIB) $(CSF_gdi32_LIB) $(CSF_SOCKETLibs_LIB) $(CSF_FPELibs_LIB) $(CSF_ThreadLibs_LIB) $(CSF_advapi32_LIB) $(CSF_kernel32_LIB) $(CSF_XwLibs_LIB) $(CSF_dpsLibs_LIB) $(CSF_XmuLibs_LIB)  
+libBinPlugin_la_LDFLAGS = -release 6.3.0
Index: opencascade/ros/adm/make/BinTObjPlugin/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/BinTObjPlugin/Makefile.am
+++ opencascade/ros/adm/make/BinTObjPlugin/Makefile.am
@@ -25,3 +25,4 @@
 ../TKernel/libTKernel.la \
 ../TKG3d/libTKG3d.la \
  $(STLPort_LIB) $(CSF_SOCKETLibs_LIB) $(CSF_FPELibs_LIB) $(CSF_ThreadLibs_LIB) $(CSF_advapi32_LIB) $(CSF_gdi32_LIB) $(CSF_user32_LIB) $(CSF_kernel32_LIB)  
+libBinTObjPlugin_la_LDFLAGS = -release 6.3.0
Index: opencascade/ros/adm/make/BinXCAFPlugin/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/BinXCAFPlugin/Makefile.am
+++ opencascade/ros/adm/make/BinXCAFPlugin/Makefile.am
@@ -32,3 +32,4 @@
 ../TKMath/libTKMath.la \
 ../TKG3d/libTKG3d.la \
  $(STLPort_LIB) $(CSF_user32_LIB) $(CSF_gdi32_LIB) $(CSF_SOCKETLibs_LIB) $(CSF_FPELibs_LIB) $(CSF_ThreadLibs_LIB) $(CSF_advapi32_LIB) $(CSF_kernel32_LIB) $(CSF_XwLibs_LIB) $(CSF_dpsLibs_LIB) $(CSF_XmuLibs_LIB)  
+libBinXCAFPlugin_la_LDFLAGS = -release 6.3.0
Index: opencascade/ros/adm/make/FWOSPlugin/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/FWOSPlugin/Makefile.am
+++ opencascade/ros/adm/make/FWOSPlugin/Makefile.am
@@ -22,6 +22,7 @@
 ../TKernel/libTKernel.la \
 ../TKCDF/libTKCDF.la \
 $(STLPort_LIB) 
+libFWOSPlugin_la_LDFLAGS = -release 6.3.0
 
 libFWOSPlugin_la_SOURCES = \
 @top_srcdir@/src/FWOSDriver/FWOSDriver.cxx \
Index: opencascade/ros/adm/make/PTKernel/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/PTKernel/Makefile.am
+++ opencascade/ros/adm/make/PTKernel/Makefile.am
@@ -35,6 +35,7 @@
 libPTKernel_la_LIBADD = \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) 
+libPTKernel_la_LDFLAGS = -release 6.3.0
 
 libPTKernel_la_SOURCES = \
 @top_srcdir@/drv/DBC/DBC_VArrayNodeOfVArrayOfCharacter_0.cxx \
Index: opencascade/ros/adm/make/StdLPlugin/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/StdLPlugin/Makefile.am
+++ opencascade/ros/adm/make/StdLPlugin/Makefile.am
@@ -27,3 +27,4 @@
 ../TKernel/libTKernel.la \
 ../TKG3d/libTKG3d.la \
  $(STLPort_LIB) $(CSF_SOCKETLibs_LIB) $(CSF_FPELibs_LIB) $(CSF_ThreadLibs_LIB) $(CSF_advapi32_LIB) $(CSF_gdi32_LIB) $(CSF_user32_LIB) $(CSF_kernel32_LIB)  
+libStdLPlugin_la_LDFLAGS = -release 6.3.0
Index: opencascade/ros/adm/make/StdPlugin/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/StdPlugin/Makefile.am
+++ opencascade/ros/adm/make/StdPlugin/Makefile.am
@@ -35,3 +35,4 @@
 ../TKMath/libTKMath.la \
 ../TKG3d/libTKG3d.la \
  $(STLPort_LIB) $(CSF_user32_LIB) $(CSF_gdi32_LIB) $(CSF_SOCKETLibs_LIB) $(CSF_FPELibs_LIB) $(CSF_ThreadLibs_LIB) $(CSF_advapi32_LIB) $(CSF_kernel32_LIB) $(CSF_XwLibs_LIB) $(CSF_dpsLibs_LIB) $(CSF_XmuLibs_LIB)  
+libStdPlugin_la_LDFLAGS = -release 6.3.0
Index: opencascade/ros/adm/make/TKAdvTools/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKAdvTools/Makefile.am
+++ opencascade/ros/adm/make/TKAdvTools/Makefile.am
@@ -31,6 +31,7 @@
 libTKAdvTools_la_LIBADD = \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) 
+libTKAdvTools_la_LDFLAGS = -release 6.3.0
 
 libTKAdvTools_la_SOURCES = \
 @top_srcdir@/drv/Dynamic/Dynamic_ParameterNode_0.cxx \
Index: opencascade/ros/adm/make/TKBO/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKBO/Makefile.am
+++ opencascade/ros/adm/make/TKBO/Makefile.am
@@ -38,6 +38,7 @@
 ../TKG2d/libTKG2d.la \
 ../TKGeomAlgo/libTKGeomAlgo.la \
 $(STLPort_LIB) 
+libTKBO_la_LDFLAGS = -release 6.3.0
 
 libTKBO_la_SOURCES = \
 @top_srcdir@/drv/BOP/BOP_IndexedDataMapNodeOfIndexedDataMapOfEdgeListFaceInfo_0.cxx \
Index: opencascade/ros/adm/make/TKBRep/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKBRep/Makefile.am
+++ opencascade/ros/adm/make/TKBRep/Makefile.am
@@ -37,6 +37,7 @@
 ../TKG2d/libTKG2d.la \
 ../TKG3d/libTKG3d.la \
 $(STLPort_LIB) 
+libTKBRep_la_LDFLAGS = -release 6.3.0
 
 libTKBRep_la_SOURCES = \
 @top_srcdir@/drv/BRep/BRep_ListIteratorOfListOfCurveRepresentation_0.cxx \
Index: opencascade/ros/adm/make/TKBin/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKBin/Makefile.am
+++ opencascade/ros/adm/make/TKBin/Makefile.am
@@ -31,6 +31,7 @@
 ../TKCAF/libTKCAF.la \
 ../TKCDF/libTKCDF.la \
 $(STLPort_LIB) 
+libTKBin_la_LDFLAGS = -release 6.3.0
 
 libTKBin_la_SOURCES = \
 @top_srcdir@/src/BinDrivers/BinDrivers.cxx \
Index: opencascade/ros/adm/make/TKBinL/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKBinL/Makefile.am
+++ opencascade/ros/adm/make/TKBinL/Makefile.am
@@ -39,6 +39,7 @@
 ../TKG3d/libTKG3d.la \
 ../TKCDF/libTKCDF.la \
 $(STLPort_LIB) 
+libTKBinL_la_LDFLAGS = -release 6.3.0
 
 libTKBinL_la_SOURCES = \
 @top_srcdir@/drv/BinMDF/BinMDF_DataMapIteratorOfStringIdMap_0.cxx \
Index: opencascade/ros/adm/make/TKBinTObj/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKBinTObj/Makefile.am
+++ opencascade/ros/adm/make/TKBinTObj/Makefile.am
@@ -26,6 +26,7 @@
 ../TKBinL/libTKBinL.la \
 ../TKCDF/libTKCDF.la \
 $(STLPort_LIB) 
+libTKBinTObj_la_LDFLAGS = -release 6.3.0
 
 libTKBinTObj_la_SOURCES = \
 @top_srcdir@/src/BinTObjDrivers/BinTObjDrivers.cxx \
Index: opencascade/ros/adm/make/TKBinXCAF/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKBinXCAF/Makefile.am
+++ opencascade/ros/adm/make/TKBinXCAF/Makefile.am
@@ -29,6 +29,7 @@
 ../TKXCAF/libTKXCAF.la \
 ../TKCDF/libTKCDF.la \
 $(STLPort_LIB) 
+libTKBinXCAF_la_LDFLAGS = -release 6.3.0
 
 libTKBinXCAF_la_SOURCES = \
 @top_srcdir@/src/BinMXCAFDoc/BinMXCAFDoc.cxx \
Index: opencascade/ros/adm/make/TKBool/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKBool/Makefile.am
+++ opencascade/ros/adm/make/TKBool/Makefile.am
@@ -42,6 +42,7 @@
 ../TKBRep/libTKBRep.la \
 ../TKPrim/libTKPrim.la \
 $(STLPort_LIB) 
+libTKBool_la_LDFLAGS = -release 6.3.0
 
 libTKBool_la_SOURCES = \
 @top_srcdir@/drv/BRepAlgo/BRepAlgo_DataMapIteratorOfDataMapOfShapeBoolean_0.cxx \
Index: opencascade/ros/adm/make/TKCAF/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKCAF/Makefile.am
+++ opencascade/ros/adm/make/TKCAF/Makefile.am
@@ -36,6 +36,7 @@
 ../TKTopAlgo/libTKTopAlgo.la \
 ../TKV2d/libTKV2d.la \
 $(STLPort_LIB) 
+libTKCAF_la_LDFLAGS = -release 6.3.0
 
 libTKCAF_la_SOURCES = \
 @top_srcdir@/drv/TFunction/TFunction_Array1OfDataMapOfGUIDDriver_0.cxx \
Index: opencascade/ros/adm/make/TKCDF/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKCDF/Makefile.am
+++ opencascade/ros/adm/make/TKCDF/Makefile.am
@@ -29,6 +29,7 @@
 libTKCDF_la_LIBADD = \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) 
+libTKCDF_la_LDFLAGS = -release 6.3.0
 
 libTKCDF_la_SOURCES = \
 @top_srcdir@/drv/CDF/CDF_MetaDataDriverError_0.cxx \
Index: opencascade/ros/adm/make/TKCDLFront/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKCDLFront/Makefile.am
+++ opencascade/ros/adm/make/TKCDLFront/Makefile.am
@@ -22,6 +22,7 @@
 ../TKWOK/libTKWOK.la \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) 
+libTKCDLFront_la_LDFLAGS = -release 6.3.0
 
 libTKCDLFront_la_SOURCES = \
 @top_srcdir@/drv/CDLFront/CDL.tab.c \
Index: opencascade/ros/adm/make/TKCPPClient/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKCPPClient/Makefile.am
+++ opencascade/ros/adm/make/TKCPPClient/Makefile.am
@@ -22,6 +22,7 @@
 ../TKWOK/libTKWOK.la \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) 
+libTKCPPClient_la_LDFLAGS = -release 6.3.0
 
 libTKCPPClient_la_SOURCES = \
 @top_srcdir@/src/CPPClient/CPPClient.cxx \
Index: opencascade/ros/adm/make/TKCPPExt/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKCPPExt/Makefile.am
+++ opencascade/ros/adm/make/TKCPPExt/Makefile.am
@@ -22,6 +22,7 @@
 ../TKWOK/libTKWOK.la \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) 
+libTKCPPExt_la_LDFLAGS = -release 6.3.0
 
 libTKCPPExt_la_SOURCES = \
 @top_srcdir@/src/CPPExt/CPPExt.cxx \
Index: opencascade/ros/adm/make/TKCPPIntExt/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKCPPIntExt/Makefile.am
+++ opencascade/ros/adm/make/TKCPPIntExt/Makefile.am
@@ -22,6 +22,7 @@
 ../TKWOK/libTKWOK.la \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) 
+libTKCPPIntExt_la_LDFLAGS = -release 6.3.0
 
 libTKCPPIntExt_la_SOURCES = \
 @top_srcdir@/src/CPPIntExt/CPPIntExt.cxx \
Index: opencascade/ros/adm/make/TKCPPJini/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKCPPJini/Makefile.am
+++ opencascade/ros/adm/make/TKCPPJini/Makefile.am
@@ -22,6 +22,7 @@
 ../TKWOK/libTKWOK.la \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) 
+libTKCPPJini_la_LDFLAGS = -release 6.3.0
 
 libTKCPPJini_la_SOURCES = \
 @top_srcdir@/drv/CPPJini/CPPJini_DataMapIteratorOfDataMapOfAsciiStringInteger_0.cxx \
Index: opencascade/ros/adm/make/TKCSFDBSchema/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKCSFDBSchema/Makefile.am
+++ opencascade/ros/adm/make/TKCSFDBSchema/Makefile.am
@@ -22,6 +22,7 @@
 ../TKWOK/libTKWOK.la \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) 
+libTKCSFDBSchema_la_LDFLAGS = -release 6.3.0
 
 libTKCSFDBSchema_la_SOURCES = \
 @top_srcdir@/src/CSFDBSchema/CSFDBSchema.cxx
Index: opencascade/ros/adm/make/TKDCAF/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKDCAF/Makefile.am
+++ opencascade/ros/adm/make/TKDCAF/Makefile.am
@@ -46,6 +46,7 @@
 ../TKNIS/libTKNIS.la \
 ../TKV2d/libTKV2d.la \
 $(STLPort_LIB) 
+libTKDCAF_la_LDFLAGS = -release 6.3.0
 
 libTKDCAF_la_SOURCES = \
 @top_srcdir@/drv/DDF/DDF_StackIteratorOfTransactionStack_0.cxx \
Index: opencascade/ros/adm/make/TKDraw/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKDraw/Makefile.am
+++ opencascade/ros/adm/make/TKDraw/Makefile.am
@@ -34,6 +34,7 @@
 ../TKHLR/libTKHLR.la \
 ../TKGeomAlgo/libTKGeomAlgo.la \
 $(STLPort_LIB) $(CSF_TclLibs_LIB) $(CSF_TclTkLibs_LIB) $(CSF_gdi32_LIB) $(CSF_advapi32_LIB) $(CSF_user32_LIB) 
+libTKDraw_la_LDFLAGS = -release 6.3.0
 
 libTKDraw_la_SOURCES = \
 @top_srcdir@/drv/DBRep/DBRep_ListIteratorOfListOfEdge_0.cxx \
Index: opencascade/ros/adm/make/TKFeat/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKFeat/Makefile.am
+++ opencascade/ros/adm/make/TKFeat/Makefile.am
@@ -33,6 +33,7 @@
 ../TKBRep/libTKBRep.la \
 ../TKPrim/libTKPrim.la \
 $(STLPort_LIB) 
+libTKFeat_la_LDFLAGS = -release 6.3.0
 
 libTKFeat_la_SOURCES = \
 @top_srcdir@/drv/LocOpe/LocOpe_DataMapIteratorOfDataMapOfShapePnt_0.cxx \
Index: opencascade/ros/adm/make/TKFillet/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKFillet/Makefile.am
+++ opencascade/ros/adm/make/TKFillet/Makefile.am
@@ -45,6 +45,7 @@
 ../TKTopAlgo/libTKTopAlgo.la \
 ../TKBRep/libTKBRep.la \
 $(STLPort_LIB) 
+libTKFillet_la_LDFLAGS = -release 6.3.0
 
 libTKFillet_la_SOURCES = \
 @top_srcdir@/drv/BRepBlend/BRepBlend_AppSurf_0.cxx \
Index: opencascade/ros/adm/make/TKG2d/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKG2d/Makefile.am
+++ opencascade/ros/adm/make/TKG2d/Makefile.am
@@ -36,6 +36,7 @@
 ../TKMath/libTKMath.la \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) 
+libTKG2d_la_LDFLAGS = -release 6.3.0
 
 libTKG2d_la_SOURCES = \
 @top_srcdir@/drv/Adaptor2d/Adaptor2d_HLine2d_0.cxx \
Index: opencascade/ros/adm/make/TKG3d/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKG3d/Makefile.am
+++ opencascade/ros/adm/make/TKG3d/Makefile.am
@@ -35,6 +35,7 @@
 ../TKernel/libTKernel.la \
 ../TKG2d/libTKG2d.la \
 $(STLPort_LIB) 
+libTKG3d_la_LDFLAGS = -release 6.3.0
 
 libTKG3d_la_SOURCES = \
 @top_srcdir@/drv/Adaptor3d/Adaptor3d_HCurveOnSurface_0.cxx \
Index: opencascade/ros/adm/make/TKGeomAlgo/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKGeomAlgo/Makefile.am
+++ opencascade/ros/adm/make/TKGeomAlgo/Makefile.am
@@ -94,6 +94,7 @@
 ../TKG2d/libTKG2d.la \
 ../TKBRep/libTKBRep.la \
 $(STLPort_LIB) 
+libTKGeomAlgo_la_LDFLAGS = -release 6.3.0
 
 libTKGeomAlgo_la_SOURCES = \
 @top_srcdir@/drv/AppBlend/AppBlend_SequenceNodeOfSequenceOfArray1OfPnt2d_0.cxx \
Index: opencascade/ros/adm/make/TKGeomBase/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKGeomBase/Makefile.am
+++ opencascade/ros/adm/make/TKGeomBase/Makefile.am
@@ -66,6 +66,7 @@
 ../TKMath/libTKMath.la \
 ../TKG2d/libTKG2d.la \
 $(STLPort_LIB) 
+libTKGeomBase_la_LDFLAGS = -release 6.3.0
 
 libTKGeomBase_la_SOURCES = \
 @top_srcdir@/drv/AdvApp2Var/AdvApp2Var_SequenceNodeOfSequenceOfNode_0.cxx \
Index: opencascade/ros/adm/make/TKHLR/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKHLR/Makefile.am
+++ opencascade/ros/adm/make/TKHLR/Makefile.am
@@ -40,6 +40,7 @@
 ../TKBRep/libTKBRep.la \
 ../TKG2d/libTKG2d.la \
 $(STLPort_LIB) 
+libTKHLR_la_LDFLAGS = -release 6.3.0
 
 libTKHLR_la_SOURCES = \
 @top_srcdir@/drv/Contap/Contap_Contour_0.cxx \
Index: opencascade/ros/adm/make/TKIDLFront/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKIDLFront/Makefile.am
+++ opencascade/ros/adm/make/TKIDLFront/Makefile.am
@@ -22,6 +22,7 @@
 ../TKWOK/libTKWOK.la \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) 
+libTKIDLFront_la_LDFLAGS = -release 6.3.0
 
 libTKIDLFront_la_SOURCES = \
 @top_srcdir@/drv/IDLFront/IDL.tab.c \
Index: opencascade/ros/adm/make/TKIGES/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKIGES/Makefile.am
+++ opencascade/ros/adm/make/TKIGES/Makefile.am
@@ -67,6 +67,7 @@
 ../TKBool/libTKBool.la \
 ../TKPrim/libTKPrim.la \
 $(STLPort_LIB) 
+libTKIGES_la_LDFLAGS = -release 6.3.0
 
 libTKIGES_la_SOURCES = \
 @top_srcdir@/drv/IGESAppli/IGESAppli_Array1OfFiniteElement_0.cxx \
Index: opencascade/ros/adm/make/TKLCAF/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKLCAF/Makefile.am
+++ opencascade/ros/adm/make/TKLCAF/Makefile.am
@@ -36,6 +36,7 @@
 ../TKTopAlgo/libTKTopAlgo.la \
 ../TKGeomBase/libTKGeomBase.la \
 $(STLPort_LIB) 
+libTKLCAF_la_LDFLAGS = -release 6.3.0
 
 libTKLCAF_la_SOURCES = \
 @top_srcdir@/drv/TDF/TDF_AttributeArray1_0.cxx \
Index: opencascade/ros/adm/make/TKMath/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKMath/Makefile.am
+++ opencascade/ros/adm/make/TKMath/Makefile.am
@@ -49,6 +49,7 @@
 libTKMath_la_LIBADD = \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) 
+libTKMath_la_LDFLAGS = -release 6.3.0
 
 libTKMath_la_SOURCES = \
 @top_srcdir@/drv/Bnd/Bnd_Array1OfBox2d_0.cxx \
Index: opencascade/ros/adm/make/TKMesh/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKMesh/Makefile.am
+++ opencascade/ros/adm/make/TKMesh/Makefile.am
@@ -39,6 +39,7 @@
 ../TKGeomBase/libTKGeomBase.la \
 ../TKGeomAlgo/libTKGeomAlgo.la \
 $(STLPort_LIB) 
+libTKMesh_la_LDFLAGS = -release 6.3.0
 
 libTKMesh_la_SOURCES = \
 @top_srcdir@/drv/BRepMesh/BRepMesh_Array1OfBiPoint_0.cxx \
Index: opencascade/ros/adm/make/TKMeshVS/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKMeshVS/Makefile.am
+++ opencascade/ros/adm/make/TKMeshVS/Makefile.am
@@ -27,6 +27,7 @@
 ../TKG2d/libTKG2d.la \
 ../TKG3d/libTKG3d.la \
 $(STLPort_LIB) 
+libTKMeshVS_la_LDFLAGS = -release 6.3.0
 
 libTKMeshVS_la_SOURCES = \
 @top_srcdir@/drv/MeshVS/MeshVS_Array1OfSequenceOfInteger_0.cxx \
Index: opencascade/ros/adm/make/TKNIS/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKNIS/Makefile.am
+++ opencascade/ros/adm/make/TKNIS/Makefile.am
@@ -27,6 +27,7 @@
 ../TKG3d/libTKG3d.la \
 ../TKV3d/libTKV3d.la \
 $(STLPort_LIB) 
+libTKNIS_la_LDFLAGS = -release 6.3.0
 
 libTKNIS_la_SOURCES = \
 @top_srcdir@/src/NIS/NIS_DrawList.cxx \
Index: opencascade/ros/adm/make/TKOffset/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKOffset/Makefile.am
+++ opencascade/ros/adm/make/TKOffset/Makefile.am
@@ -39,6 +39,7 @@
 ../TKFillet/libTKFillet.la \
 ../TKPrim/libTKPrim.la \
 $(STLPort_LIB) 
+libTKOffset_la_LDFLAGS = -release 6.3.0
 
 libTKOffset_la_SOURCES = \
 @top_srcdir@/drv/BRepOffset/BRepOffset_DataMapIteratorOfDataMapOfShapeListOfInterval_0.cxx \
Index: opencascade/ros/adm/make/TKOpenGl/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKOpenGl/Makefile.am
+++ opencascade/ros/adm/make/TKOpenGl/Makefile.am
@@ -24,6 +24,7 @@
 ../TKernel/libTKernel.la \
 ../TKV3d/libTKV3d.la \
 $(STLPort_LIB) $(CSF_OpenGlLibs_LIB) $(CSF_user32_LIB) $(CSF_gdi32_LIB) 
+libTKOpenGl_la_LDFLAGS = -release 6.3.0
 
 libTKOpenGl_la_SOURCES = \
 @top_srcdir@/drv/OpenGl/OpenGl_Error_0.cxx \
Index: opencascade/ros/adm/make/TKPCAF/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKPCAF/Makefile.am
+++ opencascade/ros/adm/make/TKPCAF/Makefile.am
@@ -38,6 +38,7 @@
 ../TKG3d/libTKG3d.la \
 ../TKCDF/libTKCDF.la \
 $(STLPort_LIB) 
+libTKPCAF_la_LDFLAGS = -release 6.3.0
 
 libTKPCAF_la_SOURCES = \
 @top_srcdir@/src/MFunction/MFunction.cxx \
Index: opencascade/ros/adm/make/TKPLCAF/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKPLCAF/Makefile.am
+++ opencascade/ros/adm/make/TKPLCAF/Makefile.am
@@ -43,6 +43,7 @@
 ../TKBRep/libTKBRep.la \
 ../TKG3d/libTKG3d.la \
 $(STLPort_LIB) 
+libTKPLCAF_la_LDFLAGS = -release 6.3.0
 
 libTKPLCAF_la_SOURCES = \
 @top_srcdir@/drv/MDF/MDF_ARDriverHSequence_0.cxx \
Index: opencascade/ros/adm/make/TKPShape/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKPShape/Makefile.am
+++ opencascade/ros/adm/make/TKPShape/Makefile.am
@@ -54,6 +54,7 @@
 ../TKBRep/libTKBRep.la \
 ../TKG3d/libTKG3d.la \
 $(STLPort_LIB) 
+libTKPShape_la_LDFLAGS = -release 6.3.0
 
 libTKPShape_la_SOURCES = \
 @top_srcdir@/drv/PColPGeom/PColPGeom_FieldOfHArray1OfBSplineCurve_0.cxx \
Index: opencascade/ros/adm/make/TKPrim/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKPrim/Makefile.am
+++ opencascade/ros/adm/make/TKPrim/Makefile.am
@@ -35,6 +35,7 @@
 ../TKG3d/libTKG3d.la \
 ../TKTopAlgo/libTKTopAlgo.la \
 $(STLPort_LIB) 
+libTKPrim_la_LDFLAGS = -release 6.3.0
 
 libTKPrim_la_SOURCES = \
 @top_srcdir@/drv/BRepPrim/BRepPrim_GWedge_0.cxx \
Index: opencascade/ros/adm/make/TKSTEP/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKSTEP/Makefile.am
+++ opencascade/ros/adm/make/TKSTEP/Makefile.am
@@ -57,6 +57,7 @@
 ../TKGeomAlgo/libTKGeomAlgo.la \
 ../TKTopAlgo/libTKTopAlgo.la \
 $(STLPort_LIB) $(CSF_wsock32_LIB) 
+libTKSTEP_la_LDFLAGS = -release 6.3.0
 
 libTKSTEP_la_SOURCES = \
 @top_srcdir@/drv/STEPConstruct/STEPConstruct_DataMapIteratorOfDataMapOfAsciiStringTransient_0.cxx \
Index: opencascade/ros/adm/make/TKSTEP209/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKSTEP209/Makefile.am
+++ opencascade/ros/adm/make/TKSTEP209/Makefile.am
@@ -29,6 +29,7 @@
 ../TKernel/libTKernel.la \
 ../TKXSBase/libTKXSBase.la \
 $(STLPort_LIB) 
+libTKSTEP209_la_LDFLAGS = -release 6.3.0
 
 libTKSTEP209_la_SOURCES = \
 @top_srcdir@/drv/StepElement/StepElement_Array1OfCurveElementEndReleasePacket_0.cxx \
Index: opencascade/ros/adm/make/TKSTEPAttr/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKSTEPAttr/Makefile.am
+++ opencascade/ros/adm/make/TKSTEPAttr/Makefile.am
@@ -29,6 +29,7 @@
 ../TKernel/libTKernel.la \
 ../TKXSBase/libTKXSBase.la \
 $(STLPort_LIB) 
+libTKSTEPAttr_la_LDFLAGS = -release 6.3.0
 
 libTKSTEPAttr_la_SOURCES = \
 @top_srcdir@/drv/StepDimTol/StepDimTol_Array1OfDatumReference_0.cxx \
Index: opencascade/ros/adm/make/TKSTEPBase/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKSTEPBase/Makefile.am
+++ opencascade/ros/adm/make/TKSTEPBase/Makefile.am
@@ -37,6 +37,7 @@
 ../TKXSBase/libTKXSBase.la \
 ../TKMath/libTKMath.la \
 $(STLPort_LIB) 
+libTKSTEPBase_la_LDFLAGS = -release 6.3.0
 
 libTKSTEPBase_la_SOURCES = \
 @top_srcdir@/drv/StepBasic/StepBasic_Array1OfApproval_0.cxx \
Index: opencascade/ros/adm/make/TKSTL/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKSTL/Makefile.am
+++ opencascade/ros/adm/make/TKSTL/Makefile.am
@@ -33,6 +33,7 @@
 ../TKG3d/libTKG3d.la \
 ../TKTopAlgo/libTKTopAlgo.la \
 $(STLPort_LIB) 
+libTKSTL_la_LDFLAGS = -release 6.3.0
 
 libTKSTL_la_SOURCES = \
 @top_srcdir@/drv/StlMesh/StlMesh_SequenceNodeOfSequenceOfMeshDomain_0.cxx \
Index: opencascade/ros/adm/make/TKService/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKService/Makefile.am
+++ opencascade/ros/adm/make/TKService/Makefile.am
@@ -46,6 +46,7 @@
 ../TKernel/libTKernel.la \
 ../TKMath/libTKMath.la \
 $(STLPort_LIB) $(CSF_XwLibs_LIB) $(CSF_dpsLibs_LIB) $(CSF_XmuLibs_LIB) 
+libTKService_la_LDFLAGS = -release 6.3.0
 
 libTKService_la_SOURCES = \
 @top_srcdir@/drv/Aspect/Aspect_Array1OfEdge_0.cxx \
Index: opencascade/ros/adm/make/TKShHealing/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKShHealing/Makefile.am
+++ opencascade/ros/adm/make/TKShHealing/Makefile.am
@@ -46,6 +46,7 @@
 ../TKBRep/libTKBRep.la \
 ../TKGeomAlgo/libTKGeomAlgo.la \
 $(STLPort_LIB) $(CSF_wsock32_LIB) 
+libTKShHealing_la_LDFLAGS = -release 6.3.0
 
 libTKShHealing_la_SOURCES = \
 @top_srcdir@/drv/ShapeAnalysis/ShapeAnalysis_DataMapIteratorOfDataMapOfShapeListOfReal_0.cxx \
Index: opencascade/ros/adm/make/TKShapeSchema/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKShapeSchema/Makefile.am
+++ opencascade/ros/adm/make/TKShapeSchema/Makefile.am
@@ -28,6 +28,7 @@
 ../TKG2d/libTKG2d.la \
 ../TKCDF/libTKCDF.la \
 $(STLPort_LIB) 
+libTKShapeSchema_la_LDFLAGS = -release 6.3.0
 
 libTKShapeSchema_la_SOURCES = \
 @top_srcdir@/drv/ShapeSchema/ShapeSchema.cxx \
Index: opencascade/ros/adm/make/TKStdLSchema/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKStdLSchema/Makefile.am
+++ opencascade/ros/adm/make/TKStdLSchema/Makefile.am
@@ -32,6 +32,7 @@
 ../TKCDF/libTKCDF.la \
 ../TKShapeSchema/libTKShapeSchema.la \
 $(STLPort_LIB) 
+libTKStdLSchema_la_LDFLAGS = -release 6.3.0
 
 libTKStdLSchema_la_SOURCES = \
 @top_srcdir@/drv/StdLSchema/StdLSchema.cxx \
Index: opencascade/ros/adm/make/TKStdSchema/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKStdSchema/Makefile.am
+++ opencascade/ros/adm/make/TKStdSchema/Makefile.am
@@ -32,6 +32,7 @@
 ../TKShapeSchema/libTKShapeSchema.la \
 ../TKStdLSchema/libTKStdLSchema.la \
 $(STLPort_LIB) 
+libTKStdSchema_la_LDFLAGS = -release 6.3.0
 
 libTKStdSchema_la_SOURCES = \
 @top_srcdir@/drv/StdSchema/StdSchema.cxx \
Index: opencascade/ros/adm/make/TKTCPPExt/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKTCPPExt/Makefile.am
+++ opencascade/ros/adm/make/TKTCPPExt/Makefile.am
@@ -22,6 +22,7 @@
 ../TKWOK/libTKWOK.la \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) 
+libTKTCPPExt_la_LDFLAGS = -release 6.3.0
 
 libTKTCPPExt_la_SOURCES = \
 @top_srcdir@/src/TCPPExt/TCPPExt.cxx
Index: opencascade/ros/adm/make/TKTObj/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKTObj/Makefile.am
+++ opencascade/ros/adm/make/TKTObj/Makefile.am
@@ -24,6 +24,7 @@
 ../TKLCAF/libTKLCAF.la \
 ../TKCDF/libTKCDF.la \
 $(STLPort_LIB) 
+libTKTObj_la_LDFLAGS = -release 6.3.0
 
 libTKTObj_la_SOURCES = \
 @top_srcdir@/src/TObj/TObj_Application.cxx \
Index: opencascade/ros/adm/make/TKTObjDRAW/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKTObjDRAW/Makefile.am
+++ opencascade/ros/adm/make/TKTObjDRAW/Makefile.am
@@ -27,6 +27,7 @@
 ../TKernel/libTKernel.la \
 ../TKDraw/libTKDraw.la \
 $(STLPort_LIB) 
+libTKTObjDRAW_la_LDFLAGS = -release 6.3.0
 
 libTKTObjDRAW_la_SOURCES = \
 @top_srcdir@/src/TObjDRAW/TObjDRAW.cxx
Index: opencascade/ros/adm/make/TKTopAlgo/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKTopAlgo/Makefile.am
+++ opencascade/ros/adm/make/TKTopAlgo/Makefile.am
@@ -57,6 +57,7 @@
 ../TKG3d/libTKG3d.la \
 ../TKBRep/libTKBRep.la \
 $(STLPort_LIB) 
+libTKTopAlgo_la_LDFLAGS = -release 6.3.0
 
 libTKTopAlgo_la_SOURCES = \
 @top_srcdir@/drv/BRepApprox/BRepApprox_ApproxLine_0.cxx \
Index: opencascade/ros/adm/make/TKTopTest/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKTopTest/Makefile.am
+++ opencascade/ros/adm/make/TKTopTest/Makefile.am
@@ -58,6 +58,7 @@
 ../TKPrim/libTKPrim.la \
 ../TKV2d/libTKV2d.la \
 $(STLPort_LIB) 
+libTKTopTest_la_LDFLAGS = -release 6.3.0
 
 libTKTopTest_la_SOURCES = \
 @top_srcdir@/drv/TestTopOpeDraw/TestTopOpeDraw_Array1OfDrawableMesure_0.cxx \
Index: opencascade/ros/adm/make/TKV2d/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKV2d/Makefile.am
+++ opencascade/ros/adm/make/TKV2d/Makefile.am
@@ -40,6 +40,7 @@
 ../TKHLR/libTKHLR.la \
 ../TKGeomAlgo/libTKGeomAlgo.la \
 $(STLPort_LIB) 
+libTKV2d_la_LDFLAGS = -release 6.3.0
 
 libTKV2d_la_SOURCES = \
 @top_srcdir@/drv/AIS2D/AIS2D_DataMapIteratorOfDataMapOfIOStatus_0.cxx \
Index: opencascade/ros/adm/make/TKV3d/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKV3d/Makefile.am
+++ opencascade/ros/adm/make/TKV3d/Makefile.am
@@ -54,6 +54,7 @@
 ../TKG2d/libTKG2d.la \
 ../TKV2d/libTKV2d.la \
 $(STLPort_LIB) $(CSF_user32_LIB) $(CSF_gdi32_LIB) 
+libTKV3d_la_LDFLAGS = -release 6.3.0
 
 libTKV3d_la_SOURCES = \
 @top_srcdir@/drv/AIS/AIS_DataMapIteratorOfDataMapOfILC_0.cxx \
Index: opencascade/ros/adm/make/TKVRML/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKVRML/Makefile.am
+++ opencascade/ros/adm/make/TKVRML/Makefile.am
@@ -39,6 +39,7 @@
 ../TKService/libTKService.la \
 ../TKV3d/libTKV3d.la \
 $(STLPort_LIB) 
+libTKVRML_la_LDFLAGS = -release 6.3.0
 
 libTKVRML_la_SOURCES = \
 @top_srcdir@/src/Vrml/Vrml.cxx \
Index: opencascade/ros/adm/make/TKViewerTest/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKViewerTest/Makefile.am
+++ opencascade/ros/adm/make/TKViewerTest/Makefile.am
@@ -41,6 +41,7 @@
 ../TKDraw/libTKDraw.la \
 ../TKNIS/libTKNIS.la \
 $(STLPort_LIB) $(CSF_TclLibs_LIB) $(CSF_TclTkLibs_LIB) $(CSF_user32_LIB) $(CSF_gdi32_LIB) 
+libTKViewerTest_la_LDFLAGS = -release 6.3.0
 
 libTKViewerTest_la_SOURCES = \
 @top_srcdir@/drv/Viewer2dTest/Viewer2dTest_DataMapIteratorOfDataMapOfText_0.cxx \
Index: opencascade/ros/adm/make/TKWOK/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKWOK/Makefile.am
+++ opencascade/ros/adm/make/TKWOK/Makefile.am
@@ -48,6 +48,7 @@
 ../TKernel/libTKernel.la \
 ../TKAdvTools/libTKAdvTools.la \
 $(STLPort_LIB) 
+libTKWOK_la_LDFLAGS = -release 6.3.0
 
 libTKWOK_la_SOURCES = \
 @top_srcdir@/drv/EDL/EDL.tab.c \
Index: opencascade/ros/adm/make/TKWOKTcl/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKWOKTcl/Makefile.am
+++ opencascade/ros/adm/make/TKWOKTcl/Makefile.am
@@ -28,6 +28,7 @@
 ../TKWOK/libTKWOK.la \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) $(CSF_TclLibs_LIB) $(CSF_TclTkLibs_LIB) 
+libTKWOKTcl_la_LDFLAGS = -release 6.3.0
 
 libTKWOKTcl_la_SOURCES = \
 @top_srcdir@/src/MSAPI/MSAPI_Class.cxx \
Index: opencascade/ros/adm/make/TKXCAF/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKXCAF/Makefile.am
+++ opencascade/ros/adm/make/TKXCAF/Makefile.am
@@ -36,6 +36,7 @@
 ../TKG3d/libTKG3d.la \
 ../TKTopAlgo/libTKTopAlgo.la \
 $(STLPort_LIB) 
+libTKXCAF_la_LDFLAGS = -release 6.3.0
 
 libTKXCAF_la_SOURCES = \
 @top_srcdir@/drv/XCAFDoc/XCAFDoc_DataMapIteratorOfDataMapOfShapeLabel_0.cxx \
Index: opencascade/ros/adm/make/TKXCAFSchema/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKXCAFSchema/Makefile.am
+++ opencascade/ros/adm/make/TKXCAFSchema/Makefile.am
@@ -38,6 +38,7 @@
 ../TKShapeSchema/libTKShapeSchema.la \
 ../TKStdLSchema/libTKStdLSchema.la \
 $(STLPort_LIB) 
+libTKXCAFSchema_la_LDFLAGS = -release 6.3.0
 
 libTKXCAFSchema_la_SOURCES = \
 @top_srcdir@/drv/PXCAFDoc/PXCAFDoc_GraphNodeSequence_0.cxx \
Index: opencascade/ros/adm/make/TKXDEDRAW/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKXDEDRAW/Makefile.am
+++ opencascade/ros/adm/make/TKXDEDRAW/Makefile.am
@@ -44,6 +44,7 @@
 ../TKCAF/libTKCAF.la \
 ../TKXDEIGES/libTKXDEIGES.la \
 $(STLPort_LIB) 
+libTKXDEDRAW_la_LDFLAGS = -release 6.3.0
 
 libTKXDEDRAW_la_SOURCES = \
 @top_srcdir@/src/XDEDRAW/XDEDRAW.cxx \
Index: opencascade/ros/adm/make/TKXDEIGES/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKXDEIGES/Makefile.am
+++ opencascade/ros/adm/make/TKXDEIGES/Makefile.am
@@ -29,6 +29,7 @@
 ../TKMath/libTKMath.la \
 ../TKCDF/libTKCDF.la \
 $(STLPort_LIB) 
+libTKXDEIGES_la_LDFLAGS = -release 6.3.0
 
 libTKXDEIGES_la_SOURCES = \
 @top_srcdir@/src/IGESCAFControl/IGESCAFControl.cxx \
Index: opencascade/ros/adm/make/TKXDESTEP/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKXDESTEP/Makefile.am
+++ opencascade/ros/adm/make/TKXDESTEP/Makefile.am
@@ -32,6 +32,7 @@
 ../TKSTEPAttr/libTKSTEPAttr.la \
 ../TKCDF/libTKCDF.la \
 $(STLPort_LIB) 
+libTKXDESTEP_la_LDFLAGS = -release 6.3.0
 
 libTKXDESTEP_la_SOURCES = \
 @top_srcdir@/drv/STEPCAFControl/STEPCAFControl_DataMapIteratorOfDataMapOfLabelExternFile_0.cxx \
Index: opencascade/ros/adm/make/TKXMesh/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKXMesh/Makefile.am
+++ opencascade/ros/adm/make/TKXMesh/Makefile.am
@@ -25,6 +25,7 @@
 ../TKMath/libTKMath.la \
 ../TKBRep/libTKBRep.la \
 $(STLPort_LIB) 
+libTKXMesh_la_LDFLAGS = -release 6.3.0
 
 libTKXMesh_la_SOURCES = \
 @top_srcdir@/src/XBRepMesh/XBRepMesh.cxx
Index: opencascade/ros/adm/make/TKXSBase/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKXSBase/Makefile.am
+++ opencascade/ros/adm/make/TKXSBase/Makefile.am
@@ -58,6 +58,7 @@
 ../TKGeomBase/libTKGeomBase.la \
 ../TKTopAlgo/libTKTopAlgo.la \
 $(STLPort_LIB) 
+libTKXSBase_la_LDFLAGS = -release 6.3.0
 
 libTKXSBase_la_SOURCES = \
 @top_srcdir@/drv/IFSelect/IFSelect_HSeqOfSelection_0.cxx \
Index: opencascade/ros/adm/make/TKXSDRAW/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKXSDRAW/Makefile.am
+++ opencascade/ros/adm/make/TKXSDRAW/Makefile.am
@@ -51,6 +51,7 @@
 ../TKNIS/libTKNIS.la \
 ../TKV2d/libTKV2d.la \
 $(STLPort_LIB) 
+libTKXSDRAW_la_LDFLAGS = -release 6.3.0
 
 libTKXSDRAW_la_SOURCES = \
 @top_srcdir@/drv/XSDRAWSTLVRML/XSDRAWSTLVRML_CoordsMap_0.cxx \
Index: opencascade/ros/adm/make/TKXml/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKXml/Makefile.am
+++ opencascade/ros/adm/make/TKXml/Makefile.am
@@ -31,6 +31,7 @@
 ../TKCAF/libTKCAF.la \
 ../TKCDF/libTKCDF.la \
 $(STLPort_LIB) 
+libTKXml_la_LDFLAGS = -release 6.3.0
 
 libTKXml_la_SOURCES = \
 @top_srcdir@/src/XmlDrivers/XmlDrivers.cxx \
Index: opencascade/ros/adm/make/TKXmlL/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKXmlL/Makefile.am
+++ opencascade/ros/adm/make/TKXmlL/Makefile.am
@@ -38,6 +38,7 @@
 ../TKCDF/libTKCDF.la \
 ../TKGeomBase/libTKGeomBase.la \
 $(STLPort_LIB) 
+libTKXmlL_la_LDFLAGS = -release 6.3.0
 
 libTKXmlL_la_SOURCES = \
 @top_srcdir@/drv/XmlLDrivers/XmlLDrivers_SequenceNodeOfSequenceOfNamespaceDef_0.cxx \
Index: opencascade/ros/adm/make/TKXmlTObj/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKXmlTObj/Makefile.am
+++ opencascade/ros/adm/make/TKXmlTObj/Makefile.am
@@ -26,6 +26,7 @@
 ../TKXmlL/libTKXmlL.la \
 ../TKCDF/libTKCDF.la \
 $(STLPort_LIB) 
+libTKXmlTObj_la_LDFLAGS = -release 6.3.0
 
 libTKXmlTObj_la_SOURCES = \
 @top_srcdir@/src/XmlTObjDrivers/XmlTObjDrivers.cxx \
Index: opencascade/ros/adm/make/TKXmlXCAF/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKXmlXCAF/Makefile.am
+++ opencascade/ros/adm/make/TKXmlXCAF/Makefile.am
@@ -29,6 +29,7 @@
 ../TKCDF/libTKCDF.la \
 ../TKXml/libTKXml.la \
 $(STLPort_LIB) 
+libTKXmlXCAF_la_LDFLAGS = -release 6.3.0
 
 libTKXmlXCAF_la_SOURCES = \
 @top_srcdir@/src/XmlMXCAFDoc/XmlMXCAFDoc.cxx \
Index: opencascade/ros/adm/make/TKernel/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKernel/Makefile.am
+++ opencascade/ros/adm/make/TKernel/Makefile.am
@@ -56,6 +56,7 @@
 
 libTKernel_la_LIBADD = \
 $(STLPort_LIB) $(CSF_SOCKETLibs_LIB) $(CSF_FPELibs_LIB) $(CSF_ThreadLibs_LIB) $(CSF_advapi32_LIB) $(CSF_gdi32_LIB) $(CSF_user32_LIB) $(CSF_kernel32_LIB) 
+libTKernel_la_LDFLAGS = -release 6.3.0
 
 libTKernel_la_SOURCES = \
 @top_srcdir@/drv/Dico/Dico_DictionaryOfInteger_0.cxx \
Index: opencascade/ros/adm/make/TKjcas/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/TKjcas/Makefile.am
+++ opencascade/ros/adm/make/TKjcas/Makefile.am
@@ -21,6 +21,7 @@
 libTKjcas_la_LIBADD = \
 ../TKernel/libTKernel.la \
 $(STLPort_LIB) 
+libTKjcas_la_LDFLAGS = -release 6.3.0
 
 libTKjcas_la_SOURCES = \
 @top_srcdir@/src/jcas/jcas.cxx \
Index: opencascade/ros/adm/make/WOKLibs/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/WOKLibs/Makefile.am
+++ opencascade/ros/adm/make/WOKLibs/Makefile.am
@@ -64,3 +64,12 @@
 ../TKAdvTools/libTKAdvTools.la \
 ../TKernel/libTKernel.la \
  $(STLPort_LIB) $(CSF_TclLibs_LIB) $(CSF_TclTkLibs_LIB) $(CSF_SOCKETLibs_LIB) $(CSF_FPELibs_LIB) $(CSF_ThreadLibs_LIB) $(CSF_advapi32_LIB) $(CSF_gdi32_LIB) $(CSF_user32_LIB) $(CSF_kernel32_LIB)  
+ libwokcmd_la_LDFLAGS = -release 6.3.0
+ libwoktoolscmd_la_LDFLAGS = -release 6.3.0
+ libwokutilscmd_la_LDFLAGS = -release 6.3.0
+ libwoksteps_la_LDFLAGS = -release 6.3.0
+ libwokobjssteps_la_LDFLAGS = -release 6.3.0
+ libwokdfltsteps_la_LDFLAGS = -release 6.3.0
+ libwokdeliverysteps_la_LDFLAGS = -release 6.3.0
+ libwokorbixsteps_la_LDFLAGS = -release 6.3.0
+ libmscmd_la_LDFLAGS = -release 6.3.0
Index: opencascade/ros/adm/make/XCAFPlugin/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/XCAFPlugin/Makefile.am
+++ opencascade/ros/adm/make/XCAFPlugin/Makefile.am
@@ -37,3 +37,4 @@
 ../TKMath/libTKMath.la \
 ../TKG3d/libTKG3d.la \
  $(STLPort_LIB) $(CSF_user32_LIB) $(CSF_gdi32_LIB) $(CSF_SOCKETLibs_LIB) $(CSF_FPELibs_LIB) $(CSF_ThreadLibs_LIB) $(CSF_advapi32_LIB) $(CSF_kernel32_LIB) $(CSF_XwLibs_LIB) $(CSF_dpsLibs_LIB) $(CSF_XmuLibs_LIB)  
+libXCAFPlugin_la_LDFLAGS = -release 6.3.0
Index: opencascade/ros/adm/make/XmlLPlugin/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/XmlLPlugin/Makefile.am
+++ opencascade/ros/adm/make/XmlLPlugin/Makefile.am
@@ -23,3 +23,4 @@
 ../TKernel/libTKernel.la \
 ../TKG3d/libTKG3d.la \
  $(STLPort_LIB) $(CSF_SOCKETLibs_LIB) $(CSF_FPELibs_LIB) $(CSF_ThreadLibs_LIB) $(CSF_advapi32_LIB) $(CSF_gdi32_LIB) $(CSF_user32_LIB) $(CSF_kernel32_LIB)  
+libXmlLPlugin_la_LDFLAGS = -release 6.3.0
Index: opencascade/ros/adm/make/XmlPlugin/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/XmlPlugin/Makefile.am
+++ opencascade/ros/adm/make/XmlPlugin/Makefile.am
@@ -30,3 +30,4 @@
 ../TKMath/libTKMath.la \
 ../TKG3d/libTKG3d.la \
  $(STLPort_LIB) $(CSF_user32_LIB) $(CSF_gdi32_LIB) $(CSF_SOCKETLibs_LIB) $(CSF_FPELibs_LIB) $(CSF_ThreadLibs_LIB) $(CSF_advapi32_LIB) $(CSF_kernel32_LIB) $(CSF_XwLibs_LIB) $(CSF_dpsLibs_LIB) $(CSF_XmuLibs_LIB)  
+libXmlPlugin_la_LDFLAGS = -release 6.3.0
Index: opencascade/ros/adm/make/XmlTObjPlugin/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/XmlTObjPlugin/Makefile.am
+++ opencascade/ros/adm/make/XmlTObjPlugin/Makefile.am
@@ -25,3 +25,4 @@
 ../TKernel/libTKernel.la \
 ../TKG3d/libTKG3d.la \
  $(STLPort_LIB) $(CSF_SOCKETLibs_LIB) $(CSF_FPELibs_LIB) $(CSF_ThreadLibs_LIB) $(CSF_advapi32_LIB) $(CSF_gdi32_LIB) $(CSF_user32_LIB) $(CSF_kernel32_LIB)  
+libXmlTObjPlugin_la_LDFLAGS = -release 6.3.0
Index: opencascade/ros/adm/make/XmlXCAFPlugin/Makefile.am
===================================================================
--- opencascade.orig/ros/adm/make/XmlXCAFPlugin/Makefile.am
+++ opencascade/ros/adm/make/XmlXCAFPlugin/Makefile.am
@@ -32,3 +32,4 @@
 ../TKMath/libTKMath.la \
 ../TKG3d/libTKG3d.la \
  $(STLPort_LIB) $(CSF_user32_LIB) $(CSF_gdi32_LIB) $(CSF_SOCKETLibs_LIB) $(CSF_FPELibs_LIB) $(CSF_ThreadLibs_LIB) $(CSF_advapi32_LIB) $(CSF_kernel32_LIB) $(CSF_XwLibs_LIB) $(CSF_dpsLibs_LIB) $(CSF_XmuLibs_LIB)  
+libXmlXCAFPlugin_la_LDFLAGS = -release 6.3.0

