diff -Naur transcode-1.1.7.orig/configure.in transcode-1.1.7/configure.in
--- transcode-1.1.7.orig/configure.in	2011-11-19 17:50:27.000000000 +0100
+++ transcode-1.1.7/configure.in	2026-08-06 12:59:45.634000000 +0200
@@ -807,11 +807,6 @@
   CFLAGS="$ac_save_CFLAGS"
   LIBS="$ac_save_LIBS"
 
-  if test 393 -ge "$lame_version" ; then
-    have_lame="no"
-    TC_PKG_ERROR(lame, lame.h, yes, lame, [http://www.mp3dev.org/],
-      [lame version < 3.93])
-  fi
 fi
 TC_PKG_HAVE(lame, LAME)
 
