diff -urpN vice-1.22/configure vice-1.22-64bit/configure
--- vice-1.22/configure	2007-08-05 21:53:33.000000000 +0200
+++ vice-1.22-64bit/configure	2008-02-16 14:22:15.000000000 +0100
@@ -345,7 +345,7 @@ datadir='${prefix}/share'
 sysconfdir='${prefix}/etc'
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
-libdir='${exec_prefix}/lib'
+libdir='${exec_prefix}/lib64'
 includedir='${prefix}/include'
 oldincludedir='/usr/include'
 infodir='${prefix}/info'
@@ -821,7 +821,7 @@ Installation directories:
 			  [PREFIX]
 
 By default, \`make install' will install all the files in
-\`$ac_default_prefix/bin', \`$ac_default_prefix/lib' etc.  You can specify
+\`$ac_default_prefix/bin', \`$ac_default_prefix/lib64' etc.  You can specify
 an installation prefix other than \`$ac_default_prefix' using \`--prefix',
 for instance \`--prefix=\$HOME'.
 
@@ -835,7 +835,7 @@ Fine tuning of the installation director
   --sysconfdir=DIR       read-only single-machine data [PREFIX/etc]
   --sharedstatedir=DIR   modifiable architecture-independent data [PREFIX/com]
   --localstatedir=DIR    modifiable single-machine data [PREFIX/var]
-  --libdir=DIR           object code libraries [EPREFIX/lib]
+  --libdir=DIR           object code libraries [EPREFIX/lib64]
   --includedir=DIR       C header files [PREFIX/include]
   --oldincludedir=DIR    C header files for non-gcc [/usr/include]
   --infodir=DIR          info documentation [PREFIX/info]
@@ -7988,7 +7988,7 @@ subdirs="$subdirs src/resid"
   else
         no_recursion=yes
 
-    LDFLAGS="$LDFLAGS -L$with_resid/lib"
+    LDFLAGS="$LDFLAGS -L$with_resid/lib64"
     CPPFLAGS="$CPPFLAGS -I$with_resid/include"
 
 
@@ -22745,7 +22745,7 @@ _ACEOF
 	*) test -f "$ac_im_incroot/X11/Xos.h" && ac_x_includes=$ac_im_incroot;;
     esac
     case $ac_im_usrlibdir in
-	/usr/lib | /lib) ;;
+	/usr/lib64 | /lib) ;;
 	*) test -d "$ac_im_usrlibdir" && ac_x_libraries=$ac_im_usrlibdir ;;
     esac
   fi
@@ -22886,7 +22886,7 @@ else
 sed 's/^/| /' conftest.$ac_ext >&5
 
 LIBS=$ac_save_LIBS
-for ac_dir in `echo "$ac_x_includes $ac_x_header_dirs" | sed s/include/lib/g`
+for ac_dir in `echo "$ac_x_includes $ac_x_header_dirs" | sed s/include/lib64/g`
 do
   # Don't even attempt the hair of trying to link an X program!
   for ac_extension in a so sl; do
@@ -27845,7 +27845,7 @@ fi
 
     ;;
   freebsd*)
-    VICEDIR="$PREFIX/lib/vice";
+    VICEDIR="$PREFIX/lib64/vice";
 
 
 if true; then
@@ -27858,7 +27858,7 @@ fi
 
     ;;
   *)
-    VICEDIR="$PREFIX/lib/vice";
+    VICEDIR="$PREFIX/lib64/vice";
 
 
 if false; then
diff -urpN vice-1.22/src/arch/unix/archdep.h vice-1.22-64bit/src/arch/unix/archdep.h
--- vice-1.22/src/arch/unix/archdep.h	2007-04-29 20:50:17.000000000 +0200
+++ vice-1.22-64bit/src/arch/unix/archdep.h	2008-02-16 14:16:32.000000000 +0100
@@ -109,7 +109,7 @@ extern const char *archdep_home_path(voi
 #ifdef __NetBSD__
 #define LIBDIR          PREFIX "/share/vice"
 #else
-#define LIBDIR          PREFIX "/lib/vice"
+#define LIBDIR          PREFIX "/lib64/vice"
 #endif
 
 #if defined(__FreeBSD__) || defined(__NetBSD__)
