Index: bless/src/buffers/ByteBuffer.cs
===================================================================
--- bless.orig/src/buffers/ByteBuffer.cs
+++ bless/src/buffers/ByteBuffer.cs
@@ -26,6 +26,8 @@ using System.Threading;
 using Bless.Util;
 using Bless.Tools;
 
+using Range=Bless.Util.Range;
+
 namespace Bless.Buffers {
 
 ///<summary>
Index: bless/src/tools/find/IFindStrategy.cs
===================================================================
--- bless.orig/src/tools/find/IFindStrategy.cs
+++ bless/src/tools/find/IFindStrategy.cs
@@ -23,6 +23,8 @@ using System;
 using Bless.Util;
 using Bless.Buffers;
 
+using Range=Bless.Util.Range;
+
 namespace Bless.Tools.Find {
 
 public interface IFindStrategy
@@ -40,4 +42,4 @@ public interface IFindStrategy
 	bool Cancelled { get; set; }
 		}
 
-} // end namespace
\ No newline at end of file
+} // end namespace
Index: bless/src/tools/find/IFinder.cs
===================================================================
--- bless.orig/src/tools/find/IFinder.cs
+++ bless/src/tools/find/IFinder.cs
@@ -23,6 +23,8 @@ using System;
 using Bless.Util;
 using Bless.Buffers;
 
+using Range=Bless.Util.Range;
+
 namespace Bless.Tools.Find {
 
 public delegate void FirstFindHandler();
@@ -41,4 +43,4 @@ public interface IFinder
 	IAsyncResult ReplaceAll(byte[] ba, AsyncCallback ac);
 }
 
-} // end namespace
\ No newline at end of file
+} // end namespace
Index: bless/src/tools/find/FindOperations.cs
===================================================================
--- bless.orig/src/tools/find/FindOperations.cs
+++ bless/src/tools/find/FindOperations.cs
@@ -24,6 +24,8 @@ using System.IO;
 using System.Threading;
 using Bless.Util;
 
+using Range=Bless.Util.Range;
+
 namespace Bless.Tools.Find {
 
 ///<summary>
Index: bless/src/gui/areas/AreaGroup.cs
===================================================================
--- bless.orig/src/gui/areas/AreaGroup.cs
+++ bless/src/gui/areas/AreaGroup.cs
@@ -25,6 +25,8 @@ using Bless.Buffers;
 using Bless.Util;
 using Bless.Gui.Drawers;
 
+using Range=Bless.Util.Range;
+
 namespace Bless.Gui.Areas
 {
 
Index: bless/src/gui/DataBookFinder.cs
===================================================================
--- bless.orig/src/gui/DataBookFinder.cs
+++ bless/src/gui/DataBookFinder.cs
@@ -26,6 +26,8 @@ using Bless.Util;
 using Bless.Buffers;
 using Bless.Tools.Find;
 
+using Range=Bless.Util.Range;
+
 namespace Bless.Gui {
 
 public class DataBookFinder : IFinder
