diff -Naur bless-0.6.0/src/buffers/ByteBuffer.cs bless-0.6.0-fixed/src/buffers/ByteBuffer.cs
--- bless-0.6.0/src/buffers/ByteBuffer.cs	2008-06-07 21:59:25.000000000 +0700
+++ bless-0.6.0-fixed/src/buffers/ByteBuffer.cs	2009-12-11 20:07:42.000000000 +0600
@@ -957,7 +957,7 @@
 	///</summary>
 	public string TempDir {
 		get { return tempDir; }
-		set { tempDir = value;}
+		set { if(value != "") tempDir = value; }
 	}
 	
 	///<summary>
