diff -urNad bless-0.5.2.orig/src/gui/plugins/BitwiseOperationsPlugin.cs bless-0.5.2/src/gui/plugins/BitwiseOperationsPlugin.cs
--- bless-0.5.2.orig/src/gui/plugins/BitwiseOperationsPlugin.cs	2008-04-10 03:50:32.000000000 +0100
+++ bless-0.5.2/src/gui/plugins/BitwiseOperationsPlugin.cs	2008-04-10 03:51:45.000000000 +0100
@@ -30,6 +30,8 @@
 using Bless.Buffers;
 using Mono.Unix;
 
+using Action = Gtk.Action;
+
 namespace Bless.Gui.Plugins {
 
 ///<summary>
diff -urNad bless-0.5.2.orig/src/gui/plugins/EditOperationsPlugin.cs bless-0.5.2/src/gui/plugins/EditOperationsPlugin.cs
--- bless-0.5.2.orig/src/gui/plugins/EditOperationsPlugin.cs	2008-04-10 03:50:32.000000000 +0100
+++ bless-0.5.2/src/gui/plugins/EditOperationsPlugin.cs	2008-04-10 03:52:06.000000000 +0100
@@ -27,6 +27,8 @@
 using Bless.Gui.Dialogs;
 using Bless.Tools;
 
+using Action = Gtk.Action;
+
 namespace Bless.Gui.Plugins {
 
 public class EditOperationsPlugin : GuiPlugin
diff -urNad bless-0.5.2.orig/src/gui/plugins/FileOperationsPlugin.cs bless-0.5.2/src/gui/plugins/FileOperationsPlugin.cs
--- bless-0.5.2.orig/src/gui/plugins/FileOperationsPlugin.cs	2008-04-10 03:50:32.000000000 +0100
+++ bless-0.5.2/src/gui/plugins/FileOperationsPlugin.cs	2008-04-10 03:52:26.000000000 +0100
@@ -29,6 +29,8 @@
 using Gtk;
 using Mono.Unix;
 
+using Action = Gtk.Action;
+
 namespace Bless.Gui.Plugins {
 
 public class FileOperationsPlugin : GuiPlugin
diff -urNad bless-0.5.2.orig/src/gui/plugins/FindReplacePlugin.cs bless-0.5.2/src/gui/plugins/FindReplacePlugin.cs
--- bless-0.5.2.orig/src/gui/plugins/FindReplacePlugin.cs	2008-04-10 03:50:32.000000000 +0100
+++ bless-0.5.2/src/gui/plugins/FindReplacePlugin.cs	2008-04-10 03:51:21.000000000 +0100
@@ -31,6 +31,8 @@
 using Bless.Plugins;
 using Mono.Unix;
 
+using Action = Gtk.Action;
+
 namespace Bless.Gui.Plugins {
 
 public class FindReplacePlugin : GuiPlugin
diff -urNad bless-0.5.2.orig/src/gui/plugins/HistoryPlugin.cs bless-0.5.2/src/gui/plugins/HistoryPlugin.cs
--- bless-0.5.2.orig/src/gui/plugins/HistoryPlugin.cs	2008-04-10 03:50:32.000000000 +0100
+++ bless-0.5.2/src/gui/plugins/HistoryPlugin.cs	2008-04-10 03:53:22.000000000 +0100
@@ -26,6 +26,8 @@
 using Bless.Gui;
 using Bless.Tools;
 
+using Action = Gtk.Action;
+
 namespace Bless.Gui.Plugins {
 
 public class HistoryPlugin : GuiPlugin
diff -urNad bless-0.5.2.orig/src/gui/plugins/InfobarPlugin.cs bless-0.5.2/src/gui/plugins/InfobarPlugin.cs
--- bless-0.5.2.orig/src/gui/plugins/InfobarPlugin.cs	2008-04-10 03:50:32.000000000 +0100
+++ bless-0.5.2/src/gui/plugins/InfobarPlugin.cs	2008-04-10 03:52:48.000000000 +0100
@@ -28,6 +28,8 @@
 using Bless.Tools;
 using Mono.Unix;
 
+using Action = Gtk.Action;
+
 namespace Bless.Gui.Plugins {
 
 public class InfobarPlugin : GuiPlugin
