--- sbsigntools-0.7/lib/ccan.git/Makefile	2015-02-07 12:56:54.457781357 +0000
+++ sbsigntools-0.7/lib/ccan.git/Makefile	2015-02-07 13:00:25.482175092 +0000
@@ -15,7 +15,7 @@
 EXCLUDE=wwviaudio ogg_to_pcm jmap jset nfs
 
 # Where make scores puts the results
-SCOREDIR=scores/$(shell whoami)/$(shell uname -s)-$(shell uname -m)-$(CC)-$(shell git describe --always --dirty)
+#SCOREDIR=scores/$(shell whoami)/$(shell uname -s)-$(shell uname -m)-$(CC)-$(shell git describe --always --dirty)
 
 ALL=$(filter-out $(EXCLUDE), $(REALLY_ALL))
 
@@ -77,7 +77,7 @@ distclean: clean
 scores: $(SCOREDIR)/SUMMARY
 
 $(SCOREDIR)/SUMMARY: $(patsubst ccan/%/_info, $(SCOREDIR)/score-%, $(wildcard ccan/*/_info))
-	git describe --always > $@
+#	git describe --always > $@
 	uname -a >> $@
 	$(CC) -v >> $@
 	cat $^ | grep 'Total score:' >> $@
