diff --git a/libgcab/cabinet.c b/libgcab/cabinet.c
index 6241904..275d5f5 100644
--- a/libgcab/cabinet.c
+++ b/libgcab/cabinet.c
@@ -26,7 +26,7 @@
 static voidpf
 zalloc (voidpf opaque, uInt items, uInt size)
 {
-    return g_malloc (items *size);
+    return g_malloc_n (items, size);
 }
 static void
 zfree (voidpf opaque, voidpf address)
diff --git a/meson.build b/meson.build
index 15fd5cd..05e162c 100644
--- a/meson.build
+++ b/meson.build
@@ -6,10 +6,6 @@ project('gcab', 'c',
 )
 
 git_version = []
-git = find_program('git', required: false)
-if git.found()
-  git_version = run_command(git, 'describe', '--abbrev=4', '--dirty', check: true).stdout().strip().split('-')
-endif
 
 # libtool versioning
 lt_current = 2
