diff -Naur mscore.orig/share/locale/CMakeLists.txt mscore/share/locale/CMakeLists.txt
--- mscore.orig/share/locale/CMakeLists.txt	2009-02-06 10:34:11.000000000 +0000
+++ mscore/share/locale/CMakeLists.txt	2009-04-02 18:04:07.000000000 +0000
@@ -20,30 +20,17 @@
 
 install(FILES
       mscore_de.qm
-      mscore_de.ts
       mscore_it.qm
-      mscore_it.ts
       mscore_pt.qm
-      mscore_pt.ts
       mscore_ru.qm
-      mscore_ru.ts
       mscore_es.qm
-      mscore_es.ts
       mscore_fr.qm
-      mscore_fr.ts
       mscore_nl.qm
-      mscore_nl.ts
       mscore_cz.qm
-      mscore_cz.ts
-      mscore_gl.ts
       mscore_gl.qm
-      mscore_tr.ts
       mscore_tr.qm
-      mscore_hi_IN.ts
       mscore_hi_IN.qm
-      mscore_pt_BR.ts
       mscore_pt_BR.qm
-      mscore_en_GB.ts
       mscore_en_GB.qm
       DESTINATION ${Mscore_SHARE_NAME}${Mscore_INSTALL_NAME}locale
       )
