From 10d4e479c5f6d91bf86e15be3a12c6b09c9808be Mon Sep 17 00:00:00 2001
From: Markus Teich <markus.teich@stusta.mhn.de>
Date: Fri, 8 May 2015 16:43:13 +0200
Subject: [PATCH 5/9] consistently use () with sizeof

---
 slock.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/slock.c b/slock.c
index 1551a9e..68e813f 100644
--- a/slock.c
+++ b/slock.c
@@ -133,7 +133,7 @@ readpw(Display *dpy, const char *pws)
 	while (running && !XNextEvent(dpy, &ev)) {
 		if (ev.type == KeyPress) {
 			buf[0] = 0;
-			num = XLookupString(&ev.xkey, buf, sizeof buf, &ksym, 0);
+			num = XLookupString(&ev.xkey, buf, sizeof(buf), &ksym, 0);
 			if (IsKeypadKey(ksym)) {
 				if (ksym == XK_KP_Enter)
 					ksym = XK_Return;
@@ -168,7 +168,7 @@ readpw(Display *dpy, const char *pws)
 					--len;
 				break;
 			default:
-				if (num && !iscntrl((int) buf[0]) && (len + num < sizeof passwd)) {
+				if (num && !iscntrl((int) buf[0]) && (len + num < sizeof(passwd))) {
 					memcpy(passwd + len, buf, num);
 					len += num;
 				}
-- 
2.6.4

