From b95ee111c7625375716e848ec81af2f57ca35b02 Mon Sep 17 00:00:00 2001
From: David Phillips <dbphillipsnz@gmail.com>
Date: Thu, 27 Aug 2015 06:16:25 +0200
Subject: [PATCH 8/9] Don't change to failure colour on success

---
 slock.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/slock.c b/slock.c
index b3bee92..6be8f22 100644
--- a/slock.c
+++ b/slock.c
@@ -187,7 +187,7 @@ readpw(Display *dpy, const char *pws)
 				break;
 			}
 			color = len ? INPUT : (failure || failonclear ? FAILED : INIT);
-			if (oldc != color) {
+			if (running && oldc != color) {
 				for (screen = 0; screen < nscreens; screen++) {
 					XSetWindowBackground(dpy, locks[screen]->win, locks[screen]->colors[color]);
 					XClearWindow(dpy, locks[screen]->win);
-- 
2.6.4

