diff -Nur slock-1.4.orig/config.mk slock-1.4/config.mk
--- slock-1.4.orig/config.mk	2016-11-19 18:31:23.000000000 -0600
+++ slock-1.4/config.mk	2017-05-14 20:23:33.114633934 -0500
@@ -16,7 +16,7 @@
 
 # flags
 CPPFLAGS = -DVERSION=\"${VERSION}\" -D_DEFAULT_SOURCE -DHAVE_SHADOW_H
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
+CFLAGS = @SLACKCFLAGS@ -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
 LDFLAGS = -s ${LIBS}
 COMPATSRC = explicit_bzero.c
 
