diff -Nur f1spirit-0.rc9.1615/build/linux/Makefile f1spirit-0.rc9.1615.new/build/linux/Makefile
--- f1spirit-0.rc9.1615/build/linux/Makefile	2009-02-06 10:12:51.000000000 +0000
+++ f1spirit-0.rc9.1615.new/build/linux/Makefile	2013-06-12 18:21:52.435681841 +0000
@@ -41,7 +41,7 @@
 CC = gcc
 #CFLAGS = -g3 -O3 -Wall `sdl-config --cflags` `curl-config --cflags` -I/usr/X11R6/include
 CFLAGS = -g3 -O3 -Wno-write-strings `sdl-config --cflags` `curl-config --cflags` -I/usr/X11R6/include
-LDFLAGS = `sdl-config --libs` `curl-config --libs` -L/usr/X11R6/lib/ -lSDL_image -lSDL_mixer -lSDL_net -lGL -lGLU
+LDFLAGS = `sdl-config --libs` `curl-config --libs` -L/usr/X11R6/lib/ -lSDL_image -lSDL_mixer -lSDL_net -lGL -lGLU -lm -lstdc++
 RM = rm -f
 CP = cp -r
 MD = mkdir -p
