--- lxde-common-0.5.0/lxpanel/panel.in.orig	2010-08-18 23:38:58.000000000 +0200
+++ lxde-common-0.5.0/lxpanel/panel.in	2010-08-18 23:44:34.000000000 +0200
@@ -3,10 +3,10 @@
 
 Global {
     edge=bottom
-    allign=left
+    allign=center
     margin=0
     widthtype=percent
-    width=100
+    width=95
     height=26
     transparent=0
     tintcolor=#000000
@@ -29,7 +29,7 @@
 Plugin {
     type = menu
     Config {
-        image=@prefix@/share/lxde/images/lxde-icon.png
+        image=@prefix@/share/lxpanel/images/my-computer.png
         system {
         }
         separator {
@@ -53,7 +53,13 @@
             id=pcmanfm.desktop
         }
         Button {
-            id=firefox.desktop
+            id=mozilla-firefox.desktop
+        }
+        Button {
+            id=mozilla-thunderbird.desktop
+        }
+        Button {
+            id=lxterminal.desktop
         }
     }
 }
@@ -110,7 +116,29 @@
 }
 
 Plugin {
-    type = netstat
+    type = volumealsa
+}
+
+Plugin {
+    type = space
+    Config {
+        Size=4
+    }
+}
+
+Plugin {
+    type = netstatus
+    Config {
+        iface=eth0
+        configtool=network-admin --configure %i
+    }
+}
+
+Plugin {
+    type = space
+    Config {
+        Size=4
+    }
 }
 
 Plugin {
@@ -118,7 +146,14 @@
 }
 
 Plugin {
-    type = tray
+    type = space
+    Config {
+        Size=4
+    }
+}
+
+Plugin {
+    type = cpufreq
 }
 
 Plugin {
@@ -127,6 +162,14 @@
         ClockFmt=%R
         TooltipFmt=%A %x
         BoldFont=0
+        IconOnly=0
+    }
+}
+
+Plugin {
+    type = space
+    Config {
+        Size=4
     }
 }
 
