diff -up quesoglc-0.7.2/include/internal.h~ quesoglc-0.7.2/include/internal.h
--- quesoglc-0.7.2/include/internal.h~	2008-10-26 19:21:23.000000000 +0100
+++ quesoglc-0.7.2/include/internal.h	2017-02-02 21:51:35.932747442 +0100
@@ -347,9 +347,6 @@ extern void __glcSaveGLState(__GLCglStat
 extern void __glcRestoreGLState(__GLCglState* inGLState,
 				__GLCcontext* inContext, GLboolean inAll);
 
-/* Function for GLEW so that it can get a context */
-GLEWAPI GLEWContext* glewGetContext(void);
-
 #ifndef HAVE_TLS
 /* This function initializes the thread management of QuesoGLC when TLS is not
  * available. It must be called once (see the macro GLC_INIT_THREAD)
diff -up quesoglc-0.7.2/src/misc.c~ quesoglc-0.7.2/src/misc.c
--- quesoglc-0.7.2/src/misc.c~	2009-03-24 14:29:01.000000000 +0100
+++ quesoglc-0.7.2/src/misc.c	2017-02-02 21:56:24.323417683 +0100
@@ -522,23 +522,6 @@ void __glcRestoreGLState(__GLCglState* i
     glBindBufferARB(GL_ARRAY_BUFFER_ARB, inGLState->bufferObjectID);
 }
 
-
-
-/* Function for GLEW so that it can get a context */
-GLEWAPI GLEWContext* glewGetContext(void)
-{
-  __GLCcontext* ctx = GLC_GET_CURRENT_CONTEXT();
-
-  if (!ctx) {
-    __glcRaiseError(GLC_STATE_ERROR);
-    return NULL;
-  }
-
-  return &ctx->glewContext;
-}
-
-
-
 /* This function initializes the thread management of QuesoGLC when TLS is not
  * available. It must be called once (see the macro GLC_INIT_THREAD)
  */
diff -up quesoglc-0.7.2/src/ocontext.h~ quesoglc-0.7.2/src/ocontext.h
--- quesoglc-0.7.2/src/ocontext.h~	2009-01-18 13:29:04.000000000 +0100
+++ quesoglc-0.7.2/src/ocontext.h	2017-02-02 21:54:41.812429255 +0100
@@ -131,7 +131,6 @@ struct __GLCcontextRec {
   __GLCarray* vertexIndices;	/* Array of vertex indices */
   __GLCarray* geomBatches;	/* Array of geometric batches */
 
-  GLEWContext glewContext;	/* GLEW context for OpenGL extensions */
   __GLCtexture texture;		/* Texture for immediate mode rendering */
 
   __GLCtexture atlas;
diff -up quesoglc-0.7.2/configure~ quesoglc-0.7.2/configure
--- quesoglc-0.7.2/configure~	2009-03-31 22:26:09.000000000 +0200
+++ quesoglc-0.7.2/configure	2017-02-02 22:19:23.999156843 +0100
@@ -21170,9 +21170,9 @@ fi
 
 
 
-{ echo "$as_me:$LINENO: checking for glewContextInit in -lGLEW" >&5
-echo $ECHO_N "checking for glewContextInit in -lGLEW... $ECHO_C" >&6; }
-if test "${ac_cv_lib_GLEW_glewContextInit+set}" = set; then
+{ echo "$as_me:$LINENO: checking for glewInit in -lGLEW" >&5
+echo $ECHO_N "checking for glewInit in -lGLEW... $ECHO_C" >&6; }
+if test "${ac_cv_lib_GLEW_glewInit+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
@@ -21190,11 +21190,11 @@ cat >>conftest.$ac_ext <<_ACEOF
 #ifdef __cplusplus
 extern "C"
 #endif
-char glewContextInit ();
+char glewInit ();
 int
 main ()
 {
-return glewContextInit ();
+return glewInit ();
   ;
   return 0;
 }
@@ -21217,21 +21217,21 @@ eval "echo \"\$as_me:$LINENO: $ac_try_ec
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext &&
        $as_test_x conftest$ac_exeext; then
-  ac_cv_lib_GLEW_glewContextInit=yes
+  ac_cv_lib_GLEW_glewInit=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-	ac_cv_lib_GLEW_glewContextInit=no
+	ac_cv_lib_GLEW_glewInit=no
 fi
 
 rm -f core conftest.err conftest.$ac_objext conftest_ipa8_conftest.oo \
       conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-{ echo "$as_me:$LINENO: result: $ac_cv_lib_GLEW_glewContextInit" >&5
-echo "${ECHO_T}$ac_cv_lib_GLEW_glewContextInit" >&6; }
-if test $ac_cv_lib_GLEW_glewContextInit = yes; then
+{ echo "$as_me:$LINENO: result: $ac_cv_lib_GLEW_glewInit" >&5
+echo "${ECHO_T}$ac_cv_lib_GLEW_glewInit" >&6; }
+if test $ac_cv_lib_GLEW_glewInit = yes; then
   cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBGLEW 1
 _ACEOF
@@ -21259,7 +21259,7 @@ cat >>conftest.$ac_ext <<_ACEOF
 int
 main ()
 {
-glewContextInit()
+glewInit()
   ;
   return 0;
 }
diff -up quesoglc-0.7.2/tests/test6.c~ quesoglc-0.7.2/tests/test6.c
--- quesoglc-0.7.2/tests/test6.c~	2008-06-05 00:14:51.000000000 +0200
+++ quesoglc-0.7.2/tests/test6.c	2017-02-02 22:21:13.684056957 +0100
@@ -44,8 +44,6 @@
 #define QUESOGLC_MAJOR 0
 #define QUESOGLC_MINOR 2
 
-GLEWAPI GLEWContext* glewGetContext(void);
-
 static GLCchar* __glcExtensions1 = (GLCchar*) "GLC_QSO_attrib_stack"
   " GLC_QSO_extrude GLC_QSO_hinting GLC_QSO_kerning GLC_QSO_matrix_stack"
   " GLC_QSO_utf8 GLC_SGI_full_name";
