diff -up feh-3.10.3/src/imlib.c.orig feh-3.10.3/src/imlib.c
--- feh-3.10.3/src/imlib.c.orig	2024-09-08 22:51:00.354957296 -0500
+++ feh-3.10.3/src/imlib.c	2024-09-08 22:56:21.891617710 -0500
@@ -880,11 +880,7 @@ static char *feh_http_load_image(char *u
 
 	D(("sfn is %s\n", sfn))
 
-#ifdef HAVE_MKSTEMPS
-	fd = mkstemps(sfn, strlen(basename) + 1);
-#else
-	fd = mkstemp(sfn);
-#endif
+fd = mkstemp(sfn);
 
 	if (fd != -1) {
 		sfp = fdopen(fd, "w+");
