diff -Naur fbdump-0.4.2/src/surface.c fbdump-0.4.2.patched/src/surface.c
--- fbdump-0.4.2/src/surface.c	2007-03-02 22:13:50.000000000 -0500
+++ fbdump-0.4.2.patched/src/surface.c	2017-03-24 16:54:08.538269897 -0400
@@ -49,6 +49,7 @@
 
       /* Write out the converted pixels to the stream. */
       fwrite( buf, width * 3, 1, stream );
+      fflush(stream);
 
       /* And get a ptr to the next row. */
       rowptr += s->modulo;
