From 9186b8c49e54cf59ace4e5ddf52aa10b1a386fa5 Mon Sep 17 00:00:00 2001
From: Lubomir Host <lubomir.host@gmail.com>
Date: Sun, 13 Nov 2016 18:25:44 +0100
Subject: [PATCH] Remove 'I<something>' markers from configuration example.

---
 sendxmpp | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/sendxmpp b/sendxmpp
index 8a9758e..4d06ab9 100755
--- a/sendxmpp
+++ b/sendxmpp
@@ -695,19 +695,19 @@ Show debugging info while running. B<WARNING>: This will include passwords etc.
 You may define a 'F<~/.sendxmpprc>' file with the necessary data for your
 xmpp-account. Since version 1.24 the following format is supported:
 
-    username: I<your_username>
-    jserver: I<jabber_server>
-    port: I<jabber_port>
-    password: I<your_jabber_password>
-    component: I<optional_component_name>
+    username: <your_username>
+    jserver: <jabber_server>
+    port: <jabber_port>
+    password: <your_jabber_password>
+    component: <optional_component_name>
 
 
 Example for Google Talk servers:
 
-    username: I<lubomir.host>
-    jserver: I<talk.google.com>
-    password: I<my-secure-password>
-    component: I<gmail.com>
+    username: <lubomir.host>
+    jserver: <talk.google.com>
+    password: <my-secure-password>
+    component: <gmail.com>
 
 With version 1.23 and older only one-line format is supported:
 
