diff -Naur ydotool-0.1.9/CMakeLists.txt ydotool-0.1.9.patched/CMakeLists.txt
--- ydotool-0.1.9/CMakeLists.txt	2021-01-03 01:04:26.000000000 -0500
+++ ydotool-0.1.9.patched/CMakeLists.txt	2022-01-31 23:47:50.830095182 -0500
@@ -2,11 +2,11 @@
 project(ydotool)
 
 set(CMAKE_CXX_STANDARD 14)
-set(PROJECT_VERSION "0.1.5")
+set(PROJECT_VERSION "0.1.9")
 set(CPACK_PACKAGE_VERSION ${PROJECT_VERSION})
 set(CPACK_PACKAGE_VERSION_MAJOR "0")
 set(CPACK_PACKAGE_VERSION_MINOR "1")
-set(CPACK_PACKAGE_VERSION_PATCH "5")
+set(CPACK_PACKAGE_VERSION_PATCH "9")
 set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Generic Linux command-line automation tool (no X!)")
 set(CPACK_PACKAGE_CONTACT "Reimu NotMoe <https://github.com/ReimuNotMoe>")
 set(CMAKE_PROJECT_HOMEPAGE_URL "https://github.com/ReimuNotMoe/ydotool")
