diff -Nur nvclock0.8b4.orig/Makefile.in nvclock0.8b4/Makefile.in
--- nvclock0.8b4.orig/Makefile.in	2008-05-11 05:25:44.000000000 -0500
+++ nvclock0.8b4/Makefile.in	2009-01-07 16:15:50.088688188 -0600
@@ -1,8 +1,9 @@
 INSTALL=@INSTALL@
+DESTDIR =
+prefix=@prefix@
+docdir=@docdir@
+mandir=@mandir@
 
-prefix=$(DESTDIR)@prefix@
-docdir=${prefix}/share/doc/nvclock
-mandir=${prefix}/man
 DOCS= ABOUT AUTHORS ChangeLog FAQ README
 
 all:
@@ -21,7 +22,7 @@
 install-docs:
 	@documentations='$(DOCS)'; \
 	for doc in $$documentations; do \
-	destdir=$(docdir); \
+	destdir=$(DESTDIR)$(docdir); \
 	name=`echo $$doc` \
 	dir=$$destdir; \
 	mkdir -p $$dir; \
@@ -30,14 +31,14 @@
 	done
 
 install-desktop-shortcut:
-	mkdir -p $(prefix)/share/applications
-	mkdir -p $(prefix)/share/icons/hicolor/48x48/apps
-	cp nvclock.desktop $(prefix)/share/applications
-	cp nvclock.png $(prefix)/share/icons/hicolor/48x48/apps
+	mkdir -p $(DESTDIR)$(prefix)/share/applications
+	mkdir -p $(DESTDIR)$(prefix)/share/icons/hicolor/48x48/apps
+	cp nvclock.desktop $(DESTDIR)$(prefix)/share/applications
+	cp nvclock.png $(DESTDIR)$(prefix)/share/icons/hicolor/48x48/apps
 
 install-man:
-	mkdir -p $(mandir)/man1
-	cp nvclock.1 $(mandir)/man1
+	mkdir -p $(DESTDIR)$(mandir)/man1
+	cp nvclock.1 $(DESTDIR)$(mandir)/man1
 
 uninstall: uninstall-docs uninstall-man uninstall-desktop-shortcut
 	$(MAKE) -C src uninstall
@@ -45,7 +46,7 @@
 uninstall-docs:
 	@documentations='$(DOCS)'; \
 	for doc in $$documentations; do \
-	destdir=$(docdir); \
+	destdir=$(DESTDIR)$(docdir); \
 	name=`echo $$doc` \
 	dir=$$destdir; \
 	rm -f $$dir/$$name; \
@@ -53,11 +54,12 @@
 	done
 
 uninstall-desktop-shortcut:
-	rm -f $(prefix)/share/applications/nvclock.desktop
-	echo "removing $$prefix/share/applications/nvclock.desktop" ; \
-	rm -f $(prefix)/share/icons/hicolor/48x48/apps/nvclock.png
-	echo "removing $$prefix/share/icons/hicolor/48x48/apps/nvclock.png" ; \
+	rm -f $(DESTDIR)$(prefix)/share/applications/nvclock.desktop
+	echo "removing $$(DESTDIR)$$prefix/share/applications/nvclock.desktop" ; \
+	rm -f $(DESTDIR)$(prefix)/share/icons/hicolor/48x48/apps/nvclock.png
+	echo "removing $$(DESTDIR)$$prefix/share/icons/hicolor/48x48/apps/nvclock.png" ; \
 
 uninstall-man:
-	rm -f $(mandir)/man1/nvclock.1
-	echo "removing $$mandir/man1/nvclock.1" ; \
+	rm -f $$(DESTDIR)$(mandir)/man1/nvclock.1
+	echo "removing $$(DESTDIR)$$mandir/man1/nvclock.1" ; \
+
diff -Nur nvclock0.8b4.orig/src/Makefile.in nvclock0.8b4/src/Makefile.in
--- nvclock0.8b4.orig/src/Makefile.in	2009-01-03 11:02:54.000000000 -0600
+++ nvclock0.8b4/src/Makefile.in	2009-01-07 16:13:06.839689315 -0600
@@ -8,9 +8,9 @@
 libbackend=./backend/libbackend.a
 libnvcontrol=./nvcontrol/libnvcontrol.a
 
-prefix=$(DESTDIR)@prefix@
-exec_prefix=$(DESTDIR)@exec_prefix@
-bindir=$(DESTDIR)@bindir@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+bindir=@bindir@
 
 ifeq ($(HAVE_NVCONTROL), yes)
     DEPS=$(libbackend) $(libnvcontrol)
@@ -56,7 +56,7 @@
 
 uninstall: uninstall-recursive uninstall-app
 uninstall-app:
-	rm -f $(bindir)/smartdimmer
+	rm -f $(DESTDIR)$(bindir)/smartdimmer
 
 
 # Walk through the sub directories and do what is requested
diff -Nur nvclock0.8b4.orig/src/gtk/Makefile.in nvclock0.8b4/src/gtk/Makefile.in
--- nvclock0.8b4.orig/src/gtk/Makefile.in	2008-05-11 05:25:46.000000000 -0500
+++ nvclock0.8b4/src/gtk/Makefile.in	2009-01-07 16:13:54.808688249 -0600
@@ -7,14 +7,14 @@
 libbackend=../backend/libbackend.a
 libnvcontrol=../nvcontrol/libnvcontrol.a
 
-prefix=$(DESTDIR)@prefix@
-exec_prefix=$(DESTDIR)@exec_prefix@
-bindir=$(DESTDIR)@bindir@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+bindir=@bindir@
 
 ifeq ($(HAVE_GTK2), yes)
     PROGRAM=nvclock_gtk
-    INSTALL_APP=mkdir -p $(bindir); $(INSTALL) -c $(PROGRAM) $(bindir)/$(PROGRAM)
-    UNINSTALL_APP=rm -f $(bindir)/$(PROGRAM)
+    INSTALL_APP=mkdir -p $(DESTDIR)$(bindir); $(INSTALL) -c $(PROGRAM) $(DESTDIR)$(bindir)/$(PROGRAM)
+    UNINSTALL_APP=rm -f $(DESTDIR)$(bindir)/$(PROGRAM)
 endif
 
 ifeq ($(HAVE_NVCONTROL), yes)
diff -Nur nvclock0.8b4.orig/src/qt/Makefile.in nvclock0.8b4/src/qt/Makefile.in
--- nvclock0.8b4.orig/src/qt/Makefile.in	2008-05-11 05:25:46.000000000 -0500
+++ nvclock0.8b4/src/qt/Makefile.in	2009-01-07 16:15:03.885688805 -0600
@@ -8,14 +8,14 @@
 HAVE_QT=@HAVE_QT@
 libbackend=../backend/libbackend.a
 
-prefix=$(DESTDIR)@prefix@
-exec_prefix=$(DESTDIR)@exec_prefix@
-bindir=$(DESTDIR)@bindir@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+bindir=@bindir@
 
 ifeq ($(HAVE_QT), yes)
     PROGRAM=nvclock_qt
-    INSTALL_APP=mkdir -p $(bindir); $(INSTALL) -c $(PROGRAM) $(bindir)/$(PROGRAM)
-    UNINSTALL_APP=rm -f $(bindir)/$(PROGRAM)
+    INSTALL_APP=mkdir -p $(DESTDIR)$(bindir); $(INSTALL) -c $(PROGRAM) $(DESTDIR)$(bindir)/$(PROGRAM)
+    UNINSTALL_APP=rm -f $(DESTDIR)$(bindir)/$(PROGRAM)
 endif
 
 ifeq ($(HAVE_NVCONTROL), yes)
