--- a/pjlib/include/pj/config.h
+++ b/pjlib/include/pj/config.h
@@ -293,6 +293,21 @@
  */
 #include <pj/config_site.h>
 
+#undef PJ_ANDROID
+#define PJ_ANDROID            0
+#define PJ_JNI_HAS_JNI_ONLOAD 0
+#undef PJ_HAS_FLOATING_POINT
+#define PJ_HAS_FLOATING_POINT       0
+#define PJMEDIA_AUDIO_DEV_HAS_PORTAUDIO 0
+#define PJMEDIA_AUDIO_DEV_HAS_WMME      0
+#define PJMEDIA_AUDIO_DEV_HAS_OPENSL        0
+#define PJMEDIA_AUDIO_DEV_HAS_ANDROID_JNI   0
+#define PJMEDIA_HAS_L16_CODEC       0
+#define PJMEDIA_CODEC_SPEEX_DEFAULT_QUALITY 5
+#define PJMEDIA_VID_DEV_INFO_FMT_CNT    128
+#define PJSIP_MAX_TSX_COUNT         31
+#define PJSIP_MAX_DIALOG_COUNT      31
+
 /********************************************************************
  * PJLIB Features.
  */
--- a/aconfigure
+++ b/aconfigure
@@ -5809,6 +5809,3 @@
 	case $target in
-	  *android*)
-	  	ac_os_objs="$ac_os_objs guid_android.o"
-	  	;;
 	  *)
 		if test "$ac_has_uuid_lib" = "1" -a "$ac_has_uuid_h" = "1"; then
--- a/aconfigure.ac
+++ b/aconfigure.ac
@@ -457,6 +457,3 @@
 	case $target in
-	  *android*)
-	  	ac_os_objs="$ac_os_objs guid_android.o"
-	  	;;
 	  *)
 		if test "$ac_has_uuid_lib" = "1" -a "$ac_has_uuid_h" = "1"; then
--- a/pjlib/src/pj/os_timestamp_posix.c
+++ b/pjlib/src/pj/os_timestamp_posix.c
@@ -163,3 +163,3 @@
 
-#elif defined(__ANDROID__)
+#elif defined(PJ_ANDROID) && PJ_ANDROID
 