--- pjproject/pjsip/src/pjsua2/endpoint.cpp.orig	2014-09-05 16:39:29.708512865 -0400
+++ pjproject/pjsip/src/pjsua2/endpoint.cpp	2014-09-05 16:39:00.084513427 -0400
@@ -489,7 +489,7 @@
     LogEntry entry;
     entry.level = level;
     entry.msg = string(data, len);
-    entry.threadId = (long)pj_thread_this();
+    entry.threadId = (intptr_t)pj_thread_this();
     entry.threadName = string(pj_thread_get_name(pj_thread_this()));

     ep.utilLogWrite(entry);
