--- a/pjnath/include/pjnath/ice_strans.h
+++ b/pjnath/include/pjnath/ice_strans.h
@@ -930,6 +930,7 @@
 					  int dst_addr_len);
 
 
+PJ_DECL(pj_ice_sess *) pj_ice_strans_get_ice_sess(pj_ice_strans *ice_st);
 /**
  * @}
  */
--- a/pjnath/src/pjnath/ice_strans.c
+++ b/pjnath/src/pjnath/ice_strans.c
@@ -1465,6 +1468,11 @@
 	return PJ_EINVALIDOP;
 }
 
+PJ_DECL(pj_ice_sess *) pj_ice_strans_get_ice_sess( pj_ice_strans *ice_st )
+{
+	return ice_st->ice;
+}
+
 /*
  * Callback called by ICE session when ICE processing is complete, either
  * successfully or with failure.
  