--- nateon-1.2/nateon/knateon.cpp.orig	2012-09-23 21:32:04.881311086 +0700
+++ nateon-1.2/nateon/knateon.cpp	2012-09-23 21:32:16.785153870 +0700
@@ -2490,20 +2490,20 @@
 
         int nTID = m_pDPcon->putADSB(sCommand);
 
-        Buddy *pBuddy = new Buddy();
-        pBuddy->setFL( true );
-        pBuddy->setUID( sID );
-        pBuddy->setName( sID );
-        pBuddy->setNick( sID );
+        Buddy *pBuddy2 = new Buddy();
+        pBuddy2->setFL( true );
+        pBuddy2->setUID( sID );
+        pBuddy2->setName( sID );
+        pBuddy2->setNick( sID );
 
         /// TID를 저장하기 위해 임시로 사용됨.
-        pBuddy->setHandle( "TID:" + QString::number(nTID) );
+        pBuddy2->setHandle( "TID:" + QString::number(nTID) );
 
         /// 수락을 받기 전에 버디에 추가 시킴.
-        m_pBuddyList->addBuddy( pBuddy );
+        m_pBuddyList->addBuddy( pBuddy2 );
 
         /*! 0 그룹에 등록하기. */
-        m_pDPcon->setGroup0( pBuddy );
+        m_pDPcon->setGroup0( pBuddy2 );
 
         /*!
          * 목록 refresh
--- nateon-1.2/nateon/knateonmainview.cpp.orig	2012-09-23 21:41:02.870430208 +0700
+++ nateon-1.2/nateon/knateonmainview.cpp	2012-09-23 21:39:53.259320655 +0700
@@ -1171,18 +1171,18 @@
             slotEmoticonList( stConfig.viewemoticonlist );
         }
     } else {
-        Buddy *pBuddy = new Buddy();
-        pBuddy->setFL( true );
-        pBuddy->setHandle( slCommand[3] );
-        pBuddy->setUID( slCommand[7] );
-        pBuddy->setName( slCommand[4] );
-        pBuddy->setNick( slCommand[7] );
+        Buddy *pBuddy2 = new Buddy();
+        pBuddy2->setFL( true );
+        pBuddy2->setHandle( slCommand[3] );
+        pBuddy2->setUID( slCommand[7] );
+        pBuddy2->setName( slCommand[4] );
+        pBuddy2->setNick( slCommand[7] );
 		
 		/*! 그룹에 등록하기. */
 		Group *pGroup = pGroupList->getGroupByID( slCommand[8] );
-		pGroup->addBuddy( pBuddy );
+		pGroup->addBuddy( pBuddy2 );
 		
-		pBuddyList->addBuddy( pBuddy );
+		pBuddyList->addBuddy( pBuddy2 );
 
         /*!
          * 목록 refresh
