diff -ur double-conversion-1.1.5.old/CMakeLists.txt double-conversion-1.1.5/CMakeLists.txt
--- double-conversion-1.1.5.old/CMakeLists.txt	2016-12-12 20:35:31.150277793 -0200
+++ double-conversion-1.1.5/CMakeLists.txt	2016-12-12 20:55:00.341337905 -0200
@@ -10,12 +10,6 @@
 set(INSTALL_BIN_DIR CACHE PATH "Installation directory for libraries")
 set(INSTALL_LIB_DIR CACHE PATH "Installation directory for libraries")
 set(INSTALL_INCLUDE_DIR CACHE PATH "Installation directory for include")
-# set suffix for CMake files used for packaging
-if(WIN32 AND NOT CYGWIN)
-  set(INSTALL_CMAKE_DIR CMake)
-else()
-  set(INSTALL_CMAKE_DIR lib/CMake/double-conversion)
-endif()
 
 # Make relative paths absolute (needed later)
 foreach(p LIB BIN INCLUDE CMAKE)
diff -ur double-conversion-1.1.5.old/src/CMakeLists.txt double-conversion-1.1.5/src/CMakeLists.txt
--- double-conversion-1.1.5.old/src/CMakeLists.txt	2016-12-12 20:35:31.151277793 -0200
+++ double-conversion-1.1.5/src/CMakeLists.txt	2016-12-12 21:13:30.683045433 -0200
@@ -36,7 +36,7 @@
 install(TARGETS double-conversion
   EXPORT double-conversionLibraryDepends
   RUNTIME DESTINATION "${INSTALL_BIN_DIR}" COMPONENT bin
-  LIBRARY DESTINATION "${INSTALL_LIB_DIR}/lib" COMPONENT shlib
-  ARCHIVE DESTINATION "${INSTALL_LIB_DIR}/lib" COMPONENT lib
+  LIBRARY DESTINATION "${INSTALL_LIB_DIR}/lib${LIB_SUFFIX}" COMPONENT shlib
+  ARCHIVE DESTINATION "${INSTALL_LIB_DIR}/lib${LIB_SUFFIX}" COMPONENT lib
   PUBLIC_HEADER DESTINATION "${INSTALL_INCLUDE_DIR}/include/double-conversion"
   COMPONENT dev)
