diff -Naur zathura-djvu-0.2.0/djvu.c zathura-djvu-0.2.0.patched/djvu.c
--- zathura-djvu-0.2.0/djvu.c	2012-06-09 17:37:36.000000000 -0400
+++ zathura-djvu-0.2.0.patched/djvu.c	2012-07-24 00:00:01.000000000 -0400
@@ -3,6 +3,7 @@
 #include <stdlib.h>
 #include <girara/datastructures.h>
 #include <string.h>
+#include <stddef.h>
 #include <libdjvu/miniexp.h>
 #include <glib.h>
 
diff -Naur zathura-djvu-0.2.0/djvu.h zathura-djvu-0.2.0.patched/djvu.h
--- zathura-djvu-0.2.0/djvu.h	2012-06-09 17:37:36.000000000 -0400
+++ zathura-djvu-0.2.0.patched/djvu.h	2012-07-24 00:00:09.000000000 -0400
@@ -5,6 +5,7 @@
 
 #include <stdbool.h>
 #include <zathura/plugin-api.h>
+#include <stddef.h>
 #include <libdjvu/ddjvuapi.h>
 #ifdef HAVE_CAIRO
 #include <cairo.h>
diff -Naur zathura-djvu-0.2.0/page-text.c zathura-djvu-0.2.0.patched/page-text.c
--- zathura-djvu-0.2.0/page-text.c	2012-06-09 17:37:36.000000000 -0400
+++ zathura-djvu-0.2.0.patched/page-text.c	2012-07-24 00:00:18.000000000 -0400
@@ -1,5 +1,6 @@
 /* See LICENSE file for license and copyright information */
 
+#include <stddef.h>
 #include <libdjvu/miniexp.h>
 #include <string.h>
 #include <sys/types.h>
diff -Naur zathura-djvu-0.2.0/page-text.h zathura-djvu-0.2.0.patched/page-text.h
--- zathura-djvu-0.2.0/page-text.h	2012-06-09 17:37:36.000000000 -0400
+++ zathura-djvu-0.2.0.patched/page-text.h	2012-07-24 00:00:26.000000000 -0400
@@ -6,6 +6,7 @@
 #include <girara/datastructures.h>
 #include <girara/macros.h>
 #include <zathura/document.h>
+#include <stddef.h>
 #include <libdjvu/miniexp.h>
 
 #include "djvu.h"
