--- arptables-v0.0.3-2/Makefile.orig	2006-08-22 10:40:51.826701081 -0500
+++ arptables-v0.0.3-2/Makefile	2006-08-22 10:41:38.379149449 -0500
@@ -50,11 +50,11 @@
 
 $(MANDIR)/man8/arptables.8: arptables.8
 	mkdir -p $(@D)
-	install -m 0644 -o root -g root $< $@
+	install -m 0644 $< $@
 
 .PHONY: exec
 exec: arptables
-	install -m 0755 -o root -g root $< $(BINFILE)
+	install -m 0755 $< $(BINFILE)
 
 .PHONY: install
 install: $(MANDIR)/man8/arptables.8 exec
@@ -71,14 +71,14 @@
 .PHONY: release
 release:
 	mkdir -p include/linux/netfilter_arp
-	install -m 0644 -o root -g root \
+	install -m 0644 \
 		$(KERNEL_DIR)/include/linux/netfilter_arp.h include/linux/
-	install -m 0644 -o root -g root \
+	install -m 0644 \
 		$(KERNEL_DIR)/include/linux/netfilter_arp/*.h \
 		include/linux/netfilter_arp/
-	install -m 0644 -o root -g root \
+	install -m 0644 \
 		include/netfilter_arp.h include/linux/netfilter_arp.h
-	install -m 0644 -o root -g root \
+	install -m 0644 \
 		include/arp_tables.h include/linux/netfilter_arp/arp_tables.h
 	make clean
 	cd ..;tar -c $(DIR) | gzip >$(DIR).tar.gz
