From 37765d46ef9ebfc0ce07a9a6d125649c07196d8b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?I=C3=B1aki=20=C3=9Acar?= <iucar@fedoraproject.org>
Date: Thu, 23 Mar 2023 10:13:27 +0100
Subject: [PATCH] curl v8

---
 configure | 2 +-
 m4/R.m4   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index f30284e..fbd8168 100755
--- a/configure
+++ b/configure
@@ -47606,7 +47606,7 @@ int main(int argc, const char * argv[])
 {
 #ifdef LIBCURL_VERSION_MAJOR
 #if LIBCURL_VERSION_MAJOR > 7
-  exit(1);
+  exit(0);
 #elif LIBCURL_VERSION_MAJOR == 7 && LIBCURL_VERSION_MINOR >= 28
   exit(0);
 #else
diff --git a/m4/R.m4 b/m4/R.m4
index 55ace6a..9d37330 100644
--- a/m4/R.m4
+++ b/m4/R.m4
@@ -4308,7 +4308,7 @@ int main(int argc, const char * argv[])
 {
 #ifdef LIBCURL_VERSION_MAJOR
 #if LIBCURL_VERSION_MAJOR > 7
-  exit(1);
+  exit(0);
 #elif LIBCURL_VERSION_MAJOR == 7 && LIBCURL_VERSION_MINOR >= 28
   exit(0);
 #else
-- 
2.39.2

