--- a/config.def.h	2022-04-10 23:38:48.000000000 -0300
+++ b/config.def.h	2022-05-12 00:58:25.427926600 -0300
@@ -52,7 +52,7 @@
 static const int natural_scrolling = 0;
 
 /* If you want to use the windows key change this to WLR_MODIFIER_LOGO */
-#define MODKEY WLR_MODIFIER_ALT
+#define MODKEY WLR_MODIFIER_LOGO 
 #define TAGKEYS(KEY,SKEY,TAG) \
 	{ MODKEY,                    KEY,            view,            {.ui = 1 << TAG} }, \
 	{ MODKEY|WLR_MODIFIER_CTRL,  KEY,            toggleview,      {.ui = 1 << TAG} }, \
