--- a/config.def.h	2022-05-12 00:58:25.427926600 -0300
+++ b/config.def.h	2022-05-13 23:31:21.236290183 -0300
@@ -64,7 +64,7 @@
 
 /* commands */
 static const char *termcmd[] = { "alacritty", NULL };
-static const char *menucmd[] = { "bemenu-run", NULL };
+static const char *menucmd[] = { "fuzzel", NULL };
 
 static const Key keys[] = {
 	/* Note that Shift changes certain key codes: c -> C, 2 -> at, etc. */
