diff --git a/razorqt-config/src/CMakeLists.txt b/razorqt-config/src/CMakeLists.txt
index 193809c..76df89d 100644
--- a/razorqt-config/src/CMakeLists.txt
+++ b/razorqt-config/src/CMakeLists.txt
@@ -52,4 +52,4 @@ target_link_libraries( razor-config qcategorizedview)
 
 INSTALL(TARGETS razor-config RUNTIME DESTINATION bin)
 install(FILES ${DESKTOP_FILES} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/applications)
-install(FILES razor-config.menu DESTINATION /etc/xdg/menus/)
+install(FILES razor-config.menu DESTINATION ${RAZOR_ETC_XDG_DIRECTORY}/menus/)
diff --git a/razorqt-resources/application-menu/CMakeLists.txt b/razorqt-resources/application-menu/CMakeLists.txt
index cfb1a64..356d7e5 100644
--- a/razorqt-resources/application-menu/CMakeLists.txt
+++ b/razorqt-resources/application-menu/CMakeLists.txt
@@ -26,5 +26,5 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/razor-applications.menu.in
                ${CMAKE_CURRENT_BINARY_DIR}/razor-applications.menu
                @ONLY)
 
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/razor-applications.menu DESTINATION /etc/xdg/menus/)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/razor-applications.menu DESTINATION ${RAZOR_ETC_XDG_DIRECTORY}/menus/)
 install(DIRECTORY desktop-directories DESTINATION share/ PATTERN ".desktop")
diff --git a/razorqt-resources/autostart/CMakeLists.txt b/razorqt-resources/autostart/CMakeLists.txt
index aa9dece..a00c119 100644
--- a/razorqt-resources/autostart/CMakeLists.txt
+++ b/razorqt-resources/autostart/CMakeLists.txt
@@ -11,4 +11,4 @@ razor_translate_desktop(DESKTOP_FILES
 add_custom_target(autostart_desktop_files ALL DEPENDS ${DESKTOP_FILES})
 #************************************************
 
-install(FILES ${DESKTOP_FILES} DESTINATION /etc/xdg/autostart)
+install(FILES ${DESKTOP_FILES} DESTINATION ${RAZOR_ETC_XDG_DIRECTORY}/autostart)
