diff -Nur razorqt-0.4.1.orig/razorqt-session/src/main.cpp razorqt-0.4.1/razorqt-session/src/main.cpp
--- razorqt-0.4.1.orig/razorqt-session/src/main.cpp	2012-02-13 04:24:51.000000000 -0600
+++ razorqt-0.4.1/razorqt-session/src/main.cpp	2012-04-22 22:33:05.094043520 -0500
@@ -28,6 +28,8 @@
 #include <QApplication>
 #include <QIcon>
 
+#include <unistd.h>
+
 #include "razormodman.h"
 #include "sessiondbusadaptor.h"
 #include "razortranslate.h"
diff -Nur razorqt-0.4.1.orig/razorqt-session/src/razormodman.cpp razorqt-0.4.1/razorqt-session/src/razormodman.cpp
--- razorqt-0.4.1.orig/razorqt-session/src/razormodman.cpp	2012-02-13 04:24:51.000000000 -0600
+++ razorqt-0.4.1/razorqt-session/src/razormodman.cpp	2012-04-22 22:38:53.867441049 -0500
@@ -25,6 +25,8 @@
  *
  * END_COMMON_COPYRIGHT_HEADER */
 
+#include <unistd.h>
+
 #include "razormodman.h"
 #include <razorqt/razorsettings.h>
 #include <qtxdg/xdgautostart.h>
