diff -Naur qt-everywhere-opensource-src-5.9.8.orig/qtserialbus/src/plugins/canbus/socketcan/socketcanbackend.cpp qt-everywhere-opensource-src-5.9.8/qtserialbus/src/plugins/canbus/socketcan/socketcanbackend.cpp
--- qt-everywhere-opensource-src-5.9.8.orig/qtserialbus/src/plugins/canbus/socketcan/socketcanbackend.cpp	2019-03-27 12:53:53.000000000 +0100
+++ qt-everywhere-opensource-src-5.9.8/qtserialbus/src/plugins/canbus/socketcan/socketcanbackend.cpp	2019-11-18 18:33:43.632000000 +0100
@@ -44,6 +44,7 @@
 
 #include <linux/can/error.h>
 #include <linux/can/raw.h>
+#include <linux/sockios.h>
 #include <errno.h>
 #include <unistd.h>
 #include <net/if.h>
