diff -Naur qt-everywhere-opensource-src-5.5.1.orig/qtwebengine/src/3rdparty/chromium/net/quic/crypto/chacha20_poly1305_decrypter_nss.cc qt-everywhere-opensource-src-5.5.1/qtwebengine/src/3rdparty/chromium/net/quic/crypto/chacha20_poly1305_decrypter_nss.cc
--- qt-everywhere-opensource-src-5.5.1.orig/qtwebengine/src/3rdparty/chromium/net/quic/crypto/chacha20_poly1305_decrypter_nss.cc	2015-10-13 04:36:54.000000000 +0000
+++ qt-everywhere-opensource-src-5.5.1/qtwebengine/src/3rdparty/chromium/net/quic/crypto/chacha20_poly1305_decrypter_nss.cc	2016-03-15 11:57:53.898988200 +0000
@@ -66,9 +66,9 @@
                                                AeadParams* aead_params) const {
   aead_params->len = sizeof(aead_params->data.nss_aead_params);
   CK_NSS_AEAD_PARAMS* nss_aead_params = &aead_params->data.nss_aead_params;
-  nss_aead_params->pIv =
+  nss_aead_params->pNonce =
       reinterpret_cast<CK_BYTE*>(const_cast<char*>(nonce.data()));
-  nss_aead_params->ulIvLen = nonce.size();
+  nss_aead_params->ulNonceLen = nonce.size();
   nss_aead_params->pAAD =
       reinterpret_cast<CK_BYTE*>(const_cast<char*>(associated_data.data()));
   nss_aead_params->ulAADLen = associated_data.size();
diff -Naur qt-everywhere-opensource-src-5.5.1.orig/qtwebengine/src/3rdparty/chromium/net/quic/crypto/chacha20_poly1305_encrypter_nss.cc qt-everywhere-opensource-src-5.5.1/qtwebengine/src/3rdparty/chromium/net/quic/crypto/chacha20_poly1305_encrypter_nss.cc
--- qt-everywhere-opensource-src-5.5.1.orig/qtwebengine/src/3rdparty/chromium/net/quic/crypto/chacha20_poly1305_encrypter_nss.cc	2015-10-13 04:36:54.000000000 +0000
+++ qt-everywhere-opensource-src-5.5.1/qtwebengine/src/3rdparty/chromium/net/quic/crypto/chacha20_poly1305_encrypter_nss.cc	2016-03-15 11:57:53.902988200 +0000
@@ -66,9 +66,9 @@
                                                AeadParams* aead_params) const {
   aead_params->len = sizeof(aead_params->data.nss_aead_params);
   CK_NSS_AEAD_PARAMS* nss_aead_params = &aead_params->data.nss_aead_params;
-  nss_aead_params->pIv =
+  nss_aead_params->pNonce =
       reinterpret_cast<CK_BYTE*>(const_cast<char*>(nonce.data()));
-  nss_aead_params->ulIvLen = nonce.size();
+  nss_aead_params->ulNonceLen = nonce.size();
   nss_aead_params->pAAD =
       reinterpret_cast<CK_BYTE*>(const_cast<char*>(associated_data.data()));
   nss_aead_params->ulAADLen = associated_data.size();
diff -Naur qt-everywhere-opensource-src-5.5.1.orig/qtwebengine/src/3rdparty/chromium/net/third_party/nss/ssl/ssl3con.c qt-everywhere-opensource-src-5.5.1/qtwebengine/src/3rdparty/chromium/net/third_party/nss/ssl/ssl3con.c
--- qt-everywhere-opensource-src-5.5.1.orig/qtwebengine/src/3rdparty/chromium/net/third_party/nss/ssl/ssl3con.c	2015-10-13 04:36:52.000000000 +0000
+++ qt-everywhere-opensource-src-5.5.1/qtwebengine/src/3rdparty/chromium/net/third_party/nss/ssl/ssl3con.c	2016-03-15 12:00:01.376986796 +0000
@@ -2088,8 +2088,8 @@
     param.len = sizeof(aeadParams);
     param.data = (unsigned char *) &aeadParams;
     memset(&aeadParams, 0, sizeof(aeadParams));
-    aeadParams.pIv = (unsigned char *) additionalData;
-    aeadParams.ulIvLen = 8;
+    aeadParams.pNonce = (unsigned char *) additionalData;
+    aeadParams.ulNonceLen = 8;
     aeadParams.pAAD = (unsigned char *) additionalData;
     aeadParams.ulAADLen = additionalDataLen;
     aeadParams.ulTagLen = tagSize;
