diff -Naur qt-everywhere-opensource-src-5.0.2.orig/qtwebkit/Source/WebKit/qt/examples/platformplugin/platformplugin.pro qt-everywhere-opensource-src-5.0.2/qtwebkit/Source/WebKit/qt/examples/platformplugin/platformplugin.pro
--- qt-everywhere-opensource-src-5.0.2.orig/qtwebkit/Source/WebKit/qt/examples/platformplugin/platformplugin.pro	2013-04-09 00:11:38.000000000 +0000
+++ qt-everywhere-opensource-src-5.0.2/qtwebkit/Source/WebKit/qt/examples/platformplugin/platformplugin.pro	2013-05-31 01:18:29.157175082 +0000
@@ -20,7 +20,8 @@
     }
 }
 
-DESTDIR = $$[QT_INSTALL_PLUGINS]/webkit
+target.path = $$[QT_INSTALL_PLUGINS]/webkit
+INSTALLS += target
 
 SOURCES += \
     WebPlugin.cpp \
