--- qt-everywhere-opensource-src-5.4.0/qtwebengine/src/3rdparty/chromium/media/media.gyp.orig     2014-12-05 17:26:44.000000000 +0100
+++ qt-everywhere-opensource-src-5.4.0/qtwebengine/src/3rdparty/chromium/media/media.gyp  2014-12-12 10:27:20.393735405 +0100
@@ -29,7 +29,7 @@
           ['use_cras==1', {
             'use_pulseaudio%': 0,
           }, {
-            'use_pulseaudio%': 1,
+            'use_pulseaudio%': 0,
           }],
         ],
       }, {

