diff -up qtbase-opensource-src-5.2.0-alpha/src/gui/text/qtextengine.cpp.harfbuzz qtbase-opensource-src-5.2.0-alpha/src/gui/text/qtextengine.cpp
--- qtbase-opensource-src-5.2.0-alpha/src/gui/text/qtextengine.cpp.harfbuzz	2013-09-27 06:05:10.000000000 -0500
+++ qtbase-opensource-src-5.2.0-alpha/src/gui/text/qtextengine.cpp	2013-10-01 12:20:23.089054279 -0500
@@ -1098,7 +1098,7 @@ int QTextEngine::shapeTextWithHarfbuzzNG
         hb_buffer_set_segment_properties(buffer, &props);
         hb_buffer_guess_segment_properties(buffer);
 
-        uint buffer_flags = HB_BUFFER_FLAG_DEFAULT;
+        uint buffer_flags = HB_BUFFER_FLAGS_DEFAULT;
         // Symbol encoding used to encode various crap in the 32..255 character code range,
         // and thus might override U+00AD [SHY]; avoid hiding default ignorables
         if (actualFontEngine->symbol)
