Index: doc/foot.html
===================================================================
--- doc/foot.html.orig	2016-09-24 11:03:56.000000000 +0200
+++ doc/foot.html	2017-04-10 15:51:55.951931300 +0200
@@ -10,7 +10,6 @@
     <table><tr>
       <td></td>
       <td class="copyright">Copyright &copy;$year DBus-C++ Project</td>
-      <td class="generated">Docs generated $datetime</td>
     </tr></table>
   </div>
 
Index: doc/Doxyfile.in
===================================================================
--- doc/Doxyfile.in.orig	2016-09-24 11:03:56.000000000 +0200
+++ doc/Doxyfile.in	2017-04-10 15:54:18.748279756 +0200
@@ -836,7 +836,7 @@ HTML_STYLESHEET        = @srcdir@/dbus-c
 # page will contain the date and time when the page was generated. Setting
 # this to NO can help when comparing the output of multiple runs.
 
-HTML_TIMESTAMP         = YES
+HTML_TIMESTAMP         = NO
 
 # If the HTML_ALIGN_MEMBERS tag is set to YES, the members of classes,
 # files or namespaces will be aligned in HTML using tables. If set to
