Index: include/dbus-c++/util.h
===================================================================
--- include/dbus-c++/util.h.orig	2011-11-28 17:03:19.000000000 +0200
+++ include/dbus-c++/util.h	2012-03-11 13:21:18.851849466 +0200
@@ -243,6 +243,7 @@ public:
     // TODO: think about return type in this case
     // this assert should help me to find the use case where it's needed...
     //assert (false);
+      return _cb->call(param);
   }
 
   R call(P param) const
@@ -255,6 +256,8 @@ public:
     // TODO: think about return type in this case
     // this assert should help me to find the use case where it's needed...
     //assert (false);
+      return _cb->call(param);
+    
   }
 
   bool empty() const
