Index: dbus-c++-1.pc.in
===================================================================
--- dbus-c++-1.pc.in.orig	2011-11-28 22:52:01.000000000 +0200
+++ dbus-c++-1.pc.in	2012-03-11 15:49:36.452603560 +0200
@@ -7,5 +7,5 @@ Name: @PACKAGE@
 Description: Native C++ bindings for D-Bus
 Requires: dbus-1
 Version: @VERSION@
-Libs: -L${libdir} -ldbus-c++-1
+Libs: -L${libdir} -ldbus-c++-1 -lpthread
 Cflags: -I${includedir}/dbus-c++-1 -DDBUS_API_SUBJECT_TO_CHANGE
