Index: src/Makefile.am
===================================================================
--- src/Makefile.am.orig	2016-09-24 11:03:56.000000000 +0200
+++ src/Makefile.am	2017-03-28 14:53:22.941031645 +0200
@@ -28,6 +28,9 @@ libdbus_c___1_la_CXXFLAGS = \
 	-I$(top_srcdir)/include \
 	-Wno-unused-parameter
 
+libdbus_c___1_la_LDFLAGS= \
+	-version-info "$(SOVER)"
+
 libdbus_c___1_la_LIBADD = \
 	$(dbus_LIBS)
 
Index: src/integration/glib/Makefile.am
===================================================================
--- src/integration/glib/Makefile.am.orig	2016-09-24 11:03:56.000000000 +0200
+++ src/integration/glib/Makefile.am	2017-03-28 15:04:24.695822590 +0200
@@ -16,6 +16,9 @@ libdbus_c___glib_1_la_LIBADD = \
 	$(dbus_LIBS) \
 	$(glib_LIBS)
 
+libdbus_c___glib_1_la_LDFLAGS= \
+	-version-info "$(SOVER)"
+
 AM_CPPFLAGS = \
 	$(dbus_CFLAGS) \
 	$(glib_CFLAGS)
