diff -ru thunar-media-tags-plugin-0.2.0.orig/configure thunar-media-tags-plugin-0.2.0/configure
--- thunar-media-tags-plugin-0.2.0.orig/configure	2011-11-30 20:34:12.000000000 +0100
+++ thunar-media-tags-plugin-0.2.0/configure	2012-11-27 12:35:58.432844202 +0100
@@ -14396,48 +14396,10 @@
 
 TAGLIBC_LIBS=""
 TAGLIBC_CFLAGS=""
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for taglib_file_new in -ltag_c" >&5
-$as_echo_n "checking for taglib_file_new in -ltag_c... " >&6; }
-if ${ac_cv_lib_tag_c_taglib_file_new+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  ac_check_lib_save_LIBS=$LIBS
-LIBS="-ltag_c  $LIBS"
-cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.  */
-#ifdef __cplusplus
-extern "C"
-#endif
-char taglib_file_new ();
-int
-main ()
-{
-return taglib_file_new ();
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_tag_c_taglib_file_new=yes
-else
-  ac_cv_lib_tag_c_taglib_file_new=no
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-LIBS=$ac_check_lib_save_LIBS
-fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_tag_c_taglib_file_new" >&5
-$as_echo "$ac_cv_lib_tag_c_taglib_file_new" >&6; }
-if test "x$ac_cv_lib_tag_c_taglib_file_new" = xyes; then :
-
-  ac_fn_c_check_header_mongrel "$LINENO" "taglib/tag_c.h" "ac_cv_header_taglib_tag_c_h" "$ac_includes_default"
+ac_fn_c_check_header_mongrel "$LINENO" "taglib/tag_c.h" "ac_cv_header_taglib_tag_c_h" "$ac_includes_default"
 if test "x$ac_cv_header_taglib_tag_c_h" = xyes; then :
 
-    TAGLIBC_LIBS="-ltag_c"
+TAGLIBC_LIBS="-ltag_c"
 
 $as_echo "#define HAVE_TAGLIBC 1" >>confdefs.h
 
@@ -14446,11 +14408,6 @@
 
 
 
-else
-  -lm
-fi
-
-
 
 
 
diff -ru thunar-media-tags-plugin-0.2.0.orig/configure.in thunar-media-tags-plugin-0.2.0/configure.in
--- thunar-media-tags-plugin-0.2.0.orig/configure.in	2011-11-30 20:34:07.000000000 +0100
+++ thunar-media-tags-plugin-0.2.0/configure.in	2012-11-27 12:37:09.712953077 +0100
@@ -81,14 +81,11 @@
 dnl ***********************************
 TAGLIBC_LIBS=""
 TAGLIBC_CFLAGS=""
-AC_CHECK_LIB([tag_c], [taglib_file_new],
+AC_CHECK_HEADER([taglib/tag_c.h],
 [
-  AC_CHECK_HEADER([taglib/tag_c.h],
-  [
-    TAGLIBC_LIBS="-ltag_c"
-    AC_DEFINE([HAVE_TAGLIBC], [1], [Define to 1 if tag_c is found])
-  ])
-], [-lm])
+  TAGLIBC_LIBS="-ltag_c"
+  AC_DEFINE([HAVE_TAGLIBC], [1], [Define to 1 if tag_c is found])
+])
 AC_SUBST([TAGLIBC_CFLAGS])
 AC_SUBST([TAGLIBC_LIBS])
 
