diff -Naur thunar-media-tags-plugin-0.3.0.orig/configure thunar-media-tags-plugin-0.3.0/configure
--- thunar-media-tags-plugin-0.3.0.orig/configure	2018-02-14 21:50:48.000000000 +0100
+++ thunar-media-tags-plugin-0.3.0/configure	2019-08-29 09:11:23.390608000 +0200
@@ -15292,45 +15292,8 @@
 
 TAGLIBC_LIBS=""
 TAGLIBC_CFLAGS=""
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for taglib_file_new in -ltag_c" >&5
-$as_echo_n "checking for taglib_file_new in -ltag_c... " >&6; }
-if ${ac_cv_lib_tag_c_taglib_file_new+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  ac_check_lib_save_LIBS=$LIBS
-LIBS="-ltag_c  $LIBS"
-cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.  */
-#ifdef __cplusplus
-extern "C"
-#endif
-char taglib_file_new ();
-int
-main ()
-{
-return taglib_file_new ();
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_tag_c_taglib_file_new=yes
-else
-  ac_cv_lib_tag_c_taglib_file_new=no
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-LIBS=$ac_check_lib_save_LIBS
-fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_tag_c_taglib_file_new" >&5
-$as_echo "$ac_cv_lib_tag_c_taglib_file_new" >&6; }
-if test "x$ac_cv_lib_tag_c_taglib_file_new" = xyes; then :
 
-  ac_fn_c_check_header_mongrel "$LINENO" "taglib/tag_c.h" "ac_cv_header_taglib_tag_c_h" "$ac_includes_default"
+ac_fn_c_check_header_mongrel "$LINENO" "taglib/tag_c.h" "ac_cv_header_taglib_tag_c_h" "$ac_includes_default"
 if test "x$ac_cv_header_taglib_tag_c_h" = xyes; then :
 
     TAGLIBC_LIBS="-ltag_c"
@@ -15342,10 +15305,6 @@
 
 
 
-else
-  -lm
-fi
-
 
 
 
