diff -Naur Pixie.orig/src/ri/rib.l Pixie/src/ri/rib.l
--- Pixie.orig/src/ri/rib.l	2008-11-12 02:00:24.000000000 -0500
+++ Pixie/src/ri/rib.l	2012-07-25 21:01:07.357154937 -0400
@@ -17,7 +17,7 @@
 
 // Overwrite the YYinput so that it uses libz
 #undef YY_INPUT
-#define YY_INPUT(buf, retval, maxlen)	if ( (retval = gzread(ribin,buf,maxlen)) < 0) 			\
+#define YY_INPUT(buf, retval, maxlen)	if ( (retval = gzread((gzFile) ribin,buf,maxlen)) < 0) 			\
 											YY_FATAL_ERROR( "input in flex scanner failed" );
 
 #endif
@@ -223,7 +223,7 @@
 															TRibFile	*nextFile	=	ribStack->next;
 															rib_delete_buffer( YY_CURRENT_BUFFER );
 #ifdef HAVE_ZLIB
-															gzclose(ribin);
+															gzclose((gzFile) ribin);
 #else
 															fclose(ribin);
 #endif
@@ -243,7 +243,7 @@
 															rib_delete_buffer( YY_CURRENT_BUFFER );
 															
 #ifdef HAVE_ZLIB
-															gzclose(ribin);
+															gzclose((gzFile) ribin);
 #else
 															fclose(ribin);
 #endif
diff -Naur Pixie.orig/src/ri/rib.y Pixie/src/ri/rib.y
--- Pixie.orig/src/ri/rib.y	2009-01-26 19:32:50.000000000 -0500
+++ Pixie/src/ri/rib.y	2012-07-25 21:00:33.716950771 -0400
@@ -2940,7 +2940,7 @@
 		
 		if (ribin != NULL) {
 #ifdef HAVE_ZLIB
-			gzclose(ribin);
+			gzclose((gzFile) ribin);
 #else
 			fclose(ribin);
 #endif
diff -Naur Pixie.orig/src/ri/ribOut.cpp Pixie/src/ri/ribOut.cpp
--- Pixie.orig/src/ri/ribOut.cpp	2009-02-23 00:48:03.000000000 -0500
+++ Pixie/src/ri/ribOut.cpp	2012-07-25 21:00:43.191008332 -0400
@@ -162,7 +162,7 @@
 
 #ifdef HAVE_ZLIB
 			if (outputCompressed) {
-				gzclose(outFile);
+				gzclose((gzFile) outFile);
 			} else {
 				fclose(outFile);
 			}
diff -Naur Pixie.orig/src/ri/ribOut.h Pixie/src/ri/ribOut.h
--- Pixie.orig/src/ri/ribOut.h	2009-02-23 00:48:03.000000000 -0500
+++ Pixie/src/ri/ribOut.h	2012-07-25 20:54:33.101761642 -0400
@@ -237,7 +237,7 @@
 												const int	l	=	vsprintf(scratch,mes,args);
 
 												#ifdef HAVE_ZLIB
-													if (outputCompressed)	gzwrite(outFile,scratch,l);
+													if (outputCompressed)	gzwrite((gzFile) outFile,scratch,l);
 													else					fwrite(scratch,1,l,outFile);
 												#else
 													fwrite(scratch,1,l,outFile);
@@ -258,7 +258,7 @@
 												const int l	=	vsprintf(scratch,mes,args);
 
 												#ifdef HAVE_ZLIB
-													if (outputCompressed)	gzwrite(outFile,scratch,l);
+													if (outputCompressed)	gzwrite((gzFile) outFile,scratch,l);
 													else					fwrite(scratch,1,l,outFile);
 												#else
 													fwrite(scratch,1,l,outFile);
