--- Pixie/src/common/mathSpec.h.orig	2017-07-05 02:13:02.301256099 +0700
+++ Pixie/src/common/mathSpec.h	2017-07-05 02:12:40.349255850 +0700
@@ -7,6 +7,9 @@
 
 
 
+#define min(a,b) ((a)<(b)?(a):(b))
+#define max(a,b) ((a)>(b)?(a):(b))
+
 inline	void	initv(SCALAR_TYPE *r,const SCALAR_TYPE x) {
 	r[0]	=	x;
 	r[1]	=	x;
--- Pixie/src/common/containers.h.orig	2017-07-05 02:14:23.367257020 +0700
+++ Pixie/src/common/containers.h	2017-07-05 02:15:14.528257601 +0700
@@ -50,7 +50,8 @@
 
 
 
-
+#define min(a,b) ((a)<(b)?(a):(b))
+#define max(a,b) ((a)>(b)?(a):(b))
 
 ///////////////////////////////////////////////////////////////////////
 // Class				:	CDictionary
--- Pixie/src/ri/memory.h.orig	2017-07-05 02:18:00.590259487 +0700
+++ Pixie/src/ri/memory.h	2017-07-05 02:18:11.143259607 +0700
@@ -36,6 +36,9 @@
 
 #include <string.h>
 
+#define min(a,b) ((a)<(b)?(a):(b))
+#define max(a,b) ((a)>(b)?(a):(b))
+
 ///////////////////////////////////////////////////////////////////////
 // Class				:	CMemPage
 // Description			:	This class coltains memory that's allocated on the fly
--- Pixie/src/ri/occlusion.h.orig	2017-07-05 02:23:25.745263181 +0700
+++ Pixie/src/ri/occlusion.h	2017-07-05 02:23:35.920263297 +0700
@@ -34,6 +34,9 @@
 #include "common/global.h"
 #include "common/os.h"
 
+#define min(a,b) ((a)<(b)?(a):(b))
+#define max(a,b) ((a)>(b)?(a):(b))
+
 ///////////////////////////////////////////////////////////////////////
 // Class				:	COcclusionCuller
 // Description			:	The occlusion culler class
