--- FreeFileSync-5.6/Makefile.orig	2012-08-06 10:04:22.606837678 +0200
+++ FreeFileSync-5.6/Makefile	2012-08-06 10:06:56.758716307 +0200
@@ -5,7 +5,7 @@
 APPSHAREDIR = $(SHAREDIR)/$(APPNAME)
 DOCSHAREDIR = $(SHAREDIR)/doc/$(APPNAME)
 
-COMMON_COMPILE_FLAGS = -Wall -pipe -O3 -pthread -std=gnu++0x -DNDEBUG -DwxUSE_UNICODE -DFFS_LINUX -DZEN_PLATFORM_OTHER -DWXINTL_NO_GETTEXT_MACRO -I. -include "zen/i18n.h"
+COMMON_COMPILE_FLAGS = -Wall -pipe $(OPTS) -pthread -std=gnu++0x -DNDEBUG -DwxUSE_UNICODE -DFFS_LINUX -DZEN_PLATFORM_OTHER -DWXINTL_NO_GETTEXT_MACRO -I. -include "zen/i18n.h"
 COMMON_LINK_FLAGS    = -pthread -lrt -lz
 
 #default build
