diff -Nur orig/Makefile new/Makefile
--- orig/Makefile	2011-04-19 14:46:30.000000000 -0500
+++ new/Makefile	2011-06-20 12:12:45.252826394 -0500
@@ -4,8 +4,8 @@
 SHAREDIR    = $(DESTDIR)$(prefix)/share
 APPSHAREDIR = $(SHAREDIR)/$(APPNAME)
 
-COMMON_COMPILE_FLAGS = -Wall -pipe `pkg-config --cflags gtk+-2.0` -O3 -pthread -DNDEBUG -DwxUSE_UNICODE -DFFS_LINUX -DTIXML_USE_STL -DWXINTL_NO_GETTEXT_MACRO
-COMMON_LINK_FLAGS  = -O3 -pthread
+COMMON_COMPILE_FLAGS = -Wall -pipe `pkg-config --cflags gtk+-2.0` $(OPTS) -pthread -DNDEBUG -DwxUSE_UNICODE -DFFS_LINUX -DTIXML_USE_STL -DWXINTL_NO_GETTEXT_MACRO
+COMMON_LINK_FLAGS  = $(OPTS) -pthread
 
 #default build
 FFS_CPPFLAGS = $(COMMON_COMPILE_FLAGS) `wx-config --cxxflags --debug=no --unicode=yes`
