--- /tmp/ttt/bzr-1.5/tools/rst2html.py	2008-05-16 22:21:24.000000000 +0000
+++ rst2html.py	2008-08-06 16:46:29.000000000 +0000
@@ -20,7 +20,8 @@
 from docutils.core import publish_cmdline, default_description
 
 
-if docutils.__version__ <= '0.4.1':
+if True:
+#if docutils.__version__ <= '0.4.1':
     from docutils.parsers.rst.states import Body
     # we have some option names that contain dot; which is not allowed by
     # python-docutils 0.4-4 -- so monkeypatch in a better pattern
