diff -Naur SOAPpy-0.11.6.orig/SOAPpy/Client.py SOAPpy-0.11.6/SOAPpy/Client.py
--- SOAPpy-0.11.6.orig/SOAPpy/Client.py	2004-04-27 17:45:58.000000000 +0200
+++ SOAPpy-0.11.6/SOAPpy/Client.py	2009-03-13 13:15:11.000000000 +0100
@@ -40,10 +40,10 @@
 ################################################################################
 """
 
-ident = '$Id: Client.py,v 1.20 2004/04/10 04:22:52 irjudson Exp $'
+from __future__ import nested_scopes
 from version import __version__
 
-from __future__ import nested_scopes
+ident = '$Id: Client.py,v 1.20 2004/04/10 04:22:52 irjudson Exp $'
 
 #import xml.sax
 import urllib
diff -Naur SOAPpy-0.11.6.orig/SOAPpy/Server.py SOAPpy-0.11.6/SOAPpy/Server.py
--- SOAPpy-0.11.6.orig/SOAPpy/Server.py	2004-04-28 23:47:10.000000000 +0200
+++ SOAPpy-0.11.6/SOAPpy/Server.py	2009-03-13 13:15:11.000000000 +0100
@@ -40,10 +40,10 @@
 ################################################################################
 """
 
-ident = '$Id: Server.py,v 1.20 2004/04/28 21:47:10 warnes Exp $'
+from __future__ import nested_scopes
 from version import __version__
 
-from __future__ import nested_scopes
+ident = '$Id: Server.py,v 1.20 2004/04/28 21:47:10 warnes Exp $'
 
 #import xml.sax
 import re
diff -Naur SOAPpy-0.11.6.orig/SOAPpy/Types.py SOAPpy-0.11.6/SOAPpy/Types.py
--- SOAPpy-0.11.6.orig/SOAPpy/Types.py	2004-09-11 05:03:33.000000000 +0200
+++ SOAPpy-0.11.6/SOAPpy/Types.py	2009-03-13 13:15:11.000000000 +0100
@@ -33,10 +33,10 @@
 ################################################################################
 """
 
-ident = '$Id: Types.py,v 1.17 2004/09/11 03:03:33 warnes Exp $'
+from __future__ import nested_scopes
 from version import __version__
 
-from __future__ import nested_scopes
+ident = '$Id: Types.py,v 1.17 2004/09/11 03:03:33 warnes Exp $'
 
 import UserList
 import base64
