diff -Nur gdm-2.20.11.orig//config/Xsession.in gdm-2.20.11/config/Xsession.in
--- gdm-2.20.11.orig//config/Xsession.in	2010-06-02 20:06:25.000000000 -0500
+++ gdm-2.20.11/config/Xsession.in	2010-12-20 00:02:58.480225271 -0600
@@ -213,14 +213,6 @@
   fi
 fi
 
-# add ssh-agent if found
-sshagent="`gdmwhich ssh-agent`"
-if [ -n "$sshagent" ] && [ -x "$sshagent" ] && [ -z "$SSH_AUTH_SOCK" ]; then
-    command="$sshagent -- $command"
-elif [ -z "$sshagent" ] ; then
-    echo "$0: ssh-agent not found!"
-fi
-
 echo "$0: Setup done, will execute: $command"
 
 eval exec $command
