diff -Naur gtkimageview-1.6.4.orig/configure.in gtkimageview-1.6.4/configure.in
--- gtkimageview-1.6.4.orig/configure.in	2009-04-05 22:02:43.000000000 +0200
+++ gtkimageview-1.6.4/configure.in	2015-09-02 10:43:04.455960995 +0200
@@ -20,7 +20,7 @@
 # Initialize GNOME environment
 GNOME_COMMON_INIT
 GNOME_COMPILE_WARNINGS(error)
-DEPRECATED_FLAGS="-DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGDK_PIXBUF_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED"
+DEPRECATED_FLAGS="-DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED"
 AC_SUBST(DEPRECATED_FLAGS)
 
 PKG_CHECK_MODULES(DEP,
@@ -39,7 +39,7 @@
 ######################################################################
 ##### Twiddle CFLAGS #################################################
 ######################################################################
-CFLAGS="${CFLAGS} -Wall -Werror -std=c99 -Wmissing-prototypes"
+CFLAGS="${CFLAGS} -Wall -std=c99 -Wmissing-prototypes"
 
 
 ######################################################################
