diff -Naur IO-Compress-Brotli-0.019/Makefile.PL IO-Compress-Brotli-0.019.patched/Makefile.PL
--- IO-Compress-Brotli-0.019/Makefile.PL	2024-11-30 11:56:32.000000000 -0500
+++ IO-Compress-Brotli-0.019.patched/Makefile.PL	2025-02-06 01:39:08.368620641 -0500
@@ -1,6 +1,5 @@
 use 5.008000;
 use ExtUtils::MakeMaker;
-use Alien::cmake3;
 use File::Spec::Functions qw/catfile/;
 use Config;
 
@@ -33,12 +32,6 @@
 		'Getopt::Long'  => '0',
 		'Time::HiRes'   => '0',
 	},
-	CONFIGURE_REQUIRES   => {
-		'Alien::cmake3' => '0',
-	},
-	BUILD_REQUIRES   => {
-		'Alien::cmake3' => '0',
-	},
 	INC              => '-Ibrotli/c/include',
 	MYEXTLIB         => $myextlib,
 	EXTRALIBS        => ' brotli/libbrotlienc$(LIB_EXT) brotli/libbrotlidec$(LIB_EXT) brotli/libbrotlicommon$(LIB_EXT) ',
@@ -52,10 +45,8 @@
 );
 
 sub MY::postamble {
-    my @dirs = Alien::cmake3->bin_dir;
-    my $cmake = defined $dirs[0] ? catfile($dirs[0] , Alien::cmake3->exe) : Alien::cmake3->exe;
 '
 $(MYEXTLIB): brotli/CMakeLists.txt
-	cd brotli && "' . $cmake . '"' . $cmake_options . ' -DCMAKE_MAKE_PROGRAM=$(MAKE) -DBUILD_SHARED_LIBS=OFF -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=./installed . && $(MAKE)
+	cd brotli && cmake ' . $cmake_options . ' -DCMAKE_MAKE_PROGRAM=$(MAKE) -DBUILD_SHARED_LIBS=OFF -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=./installed . && $(MAKE)
 '
 }
