--- CMakeLists.txt.orig	2020-11-03 14:54:48.786455348 -0800
+++ CMakeLists.txt	2020-11-03 15:00:16.067445503 -0800
@@ -197,7 +197,7 @@
 check_symbol_exists(strlcpy "string.h" HAVE_STRLCPY)
 check_symbol_exists(program_invocation_name "errno.h" HAVE_DECL_PROGRAM_INVOCATION_SHORT_NAME)
 if (HAVE_DECL_PROGRAM_INVOCATION_SHORT_NAME)
-  add_compile_definitions($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:_GNU_SOURCE=1>)
+  add_definitions(-D_GNU_SOURCE=1)
 endif()
 check_symbol_exists(__printflike "bsd/sys/cdefs.h" HAVE_PRINTFLIKE)
 
@@ -217,9 +217,9 @@
 endif()
 
 if(dtrace_EXECUTABLE)
-  add_compile_definitions($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:DISPATCH_USE_DTRACE=1>)
+  add_definitions(-DDISPATCH_USE_DTRACE=1)
 else()
-  add_compile_definitions($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:DISPATCH_USE_DTRACE=0>)
+  add_definitions(-DDISPATCH_USE_DTRACE=0)
 endif()
 
 find_program(leaks_EXECUTABLE leaks)
@@ -252,7 +252,7 @@
 
 configure_file("${PROJECT_SOURCE_DIR}/cmake/config.h.in"
                "${PROJECT_BINARY_DIR}/config/config_ac.h")
-add_compile_definitions($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:HAVE_CONFIG_H>)
+add_definitions(-DHAVE_CONFIG_H)
 
 
 if(ENABLE_SWIFT)
