Thanks to hotchili on LQ!

diff -Naur python-daemon-3.0.1.orig/daemon/_metadata.py python-daemon-3.0.1/daemon/_metadata.py
--- python-daemon-3.0.1.orig/daemon/_metadata.py	2023-03-04 01:04:02.000000000 +0100
+++ python-daemon-3.0.1/daemon/_metadata.py	2026-08-28 08:51:09.953107000 +0200
@@ -10,7 +10,7 @@
 import datetime
 import json
 
-import pkg_resources
+from importlib import metadata
 
 
 distribution_name = "python-daemon"
@@ -26,8 +26,8 @@
         """
     distribution = None
     try:
-        distribution = pkg_resources.get_distribution(name)
-    except pkg_resources.DistributionNotFound:
+        distribution = metadata.distribution(name)
+    except metadata.PackageNotFoundError:
         pass
 
     return distribution
@@ -56,8 +56,8 @@
             }
 
     if distribution is not None:
-        if distribution.has_metadata(filename):
-            content = distribution.get_metadata(filename)
+        content = distribution.read_text(filename)
+        if content:
             version_info = json.loads(content)
 
     return version_info
