--- fail2banreader.py.orig	2014-09-27 23:40:05.512547198 -0500
+++ fail2banreader.py	2014-09-27 23:41:27.294412338 -0500
@@ -45,7 +45,7 @@
 		return ConfigReader.getOptions(self, "Definition", opts)
 	
 	def getOptions(self):
-		opts = [["int", "loglevel", "INFO" ],
+		opts = [["string", "loglevel", "INFO" ],
 				["string", "logtarget", "STDERR"],
 				["string", "dbfile", "/var/lib/fail2ban/fail2ban.sqlite3"],
 				["int", "dbpurgeage", 86400]]
