--- jail.conf.new.orig	2014-11-01 18:05:39.298887793 -0500
+++ jail.conf.new	2014-11-01 18:06:07.763868656 -0500
@@ -33,7 +33,7 @@
 [INCLUDES]
 
 #before = paths-distro.conf
-before = paths-debian.conf
+before = paths-slackware.conf
 
 # The DEFAULT allows a global definition of the options. They can be overridden
 # in each jail afterwards.
@@ -723,4 +723,4 @@
 [portsentry]
 enabled  = false
 logpath  = /var/lib/portsentry/portsentry.history
-maxretry = 1
\ No newline at end of file
+maxretry = 1
