diff --git a/docs/CMakeLists.txt b/docs/CMakeLists.txt
index f0e7b4e..5bdc6c2 100644
--- a/docs/CMakeLists.txt
+++ b/docs/CMakeLists.txt
@@ -1,4 +1,4 @@
-find_program(PANDOC_EXECUTABLE pandoc)
+find_program(PANDOC_EXECUTABLE md2roff)
 
 if (NOT PANDOC_EXECUTABLE)
 	message(FATAL_ERROR "Pandoc not found, can not build man-pages. "
@@ -7,13 +7,11 @@ if (NOT PANDOC_EXECUTABLE)
 endif()
 
 if (BUILD_TOOLS)
-	add_custom_command(OUTPUT nuspell.1
-		COMMAND
-		${PANDOC_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/nuspell.1.md
-		--from=markdown --to=man --standalone --output=nuspell.1
-		"--metadata=footer:Nuspell ${PROJECT_VERSION}"
-		MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/nuspell.1.md
-		COMMENT "Building manpage nuspell.1")
+  add_custom_command(OUTPUT nuspell.1
+			COMMAND
+			 ${PANDOC_EXECUTABLE} --man ${CMAKE_CURRENT_SOURCE_DIR}/nuspell.1.md > nuspell.1
+			MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/nuspell.1.md
+			COMMENT "Building manpage nuspell.1")
 	add_custom_target(nuspell-man-pages ALL DEPENDS nuspell.1)
 	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/nuspell.1
 		DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
diff --git a/docs/nuspell.1.md b/docs/nuspell.1.md
index c3a40e1..5bd7e5c 100644
--- a/docs/nuspell.1.md
+++ b/docs/nuspell.1.md
@@ -1,13 +1,3 @@
----
-title: NUSPELL
-section: 1
-header: User Commands
-footer: Nuspell vX.Y  # override this on the command line in CMake
-author: Dimitrij Mijoski
-date: 2024-07-03 # This date should be changed when signifcant changes in this
-                 # document are made. It is not release date or build date.
----
-
 # NAME
 
 nuspell - Command-line tool for spellchecking.
@@ -77,10 +67,10 @@ is not strictly defined and may change, thus it is not machine-readable. Other
 programs should use the C++ library directly which has stable API.
 
 # EXAMPLES
-
+```
     nuspell -d en_US file.txt
     nuspell -d ../../subdir/di_CT.aff
-
+```
 # REPORTING BUGS
 
 Bug reports: <https://github.com/nuspell/nuspell/issues>
@@ -89,6 +79,10 @@ Bug reports: <https://github.com/nuspell/nuspell/issues>
 
 Copyright 2016-2024 Nuspell authors.
 
+# AUTHOR Dimitrij Mijoski
+
+# DATE 2024-07-03
+
 # SEE ALSO
 
 Full documentation: <https://github.com/nuspell/nuspell/wiki>
