--- nagvis-1.8rc3/share/server/core/defines/global.php.orig	2015-02-18 01:38:00.696662555 +0100
+++ nagvis-1.8rc3/share/server/core/defines/global.php	2015-02-18 01:40:33.052563023 +0100
@@ -79,11 +79,11 @@
 // The last value wins.
 //
 // Path to the main configuration file
-define('CONST_MAINCFG', '../../../etc/nagvis.ini.php');
-define('CONST_MAINCFG_CACHE', '../../../var/nagvis-conf');
+define('CONST_MAINCFG', '/etc/nagvis/nagvis.ini.php');
+define('CONST_MAINCFG_CACHE', '/var/cache/nagvis/nagvis-conf');
 
 // Path to the main configuration conf.d directory
-define('CONST_MAINCFG_DIR', '../../../etc/conf.d');
+define('CONST_MAINCFG_DIR', '/etc/nagvis/conf.d');
 
 // The directory below the NagVis root which is shared by the webserver
 define('HTDOCS_DIR', 'share');
--- nagvis-1.8rc3/share/server/core/classes/GlobalMainCfg.php.orig	2015-01-03 18:19:57.000000000 +0100
+++ nagvis-1.8rc3/share/server/core/classes/GlobalMainCfg.php	2015-02-18 01:43:31.837753573 +0100
@@ -1516,11 +1516,11 @@
      * @author 	Lars Michelsen <lars@vertical-visions.de>
      */
     private function setPathsByBase($base, $htmlBase) {
-        $this->validConfig['paths']['cfg']['default']                = $base.'etc/';
-        $this->validConfig['paths']['mapcfg']['default']             = $base.'etc/maps/';
-        $this->validConfig['paths']['geomap']['default']             = $base.'etc/geomap';
-        $this->validConfig['paths']['profiles']['default']           = $base.'etc/profiles';
-        $this->validConfig['global']['authorisation_group_perms_file']['default'] = $base.'etc/perms.db';
+        $this->validConfig['paths']['cfg']['default']                = '/etc/nagvis/';
+        $this->validConfig['paths']['mapcfg']['default']             = '/etc/nagvis/maps/';
+        $this->validConfig['paths']['geomap']['default']             = '/etc/nagvis/geomap';
+        $this->validConfig['paths']['profiles']['default']           = '/etc/nagvis/profiles';
+        $this->validConfig['global']['authorisation_group_perms_file']['default'] = '/etc/nagvis/perms.db';
 
         $this->validConfig['paths']['var']['default']                = $base.'var/';
         $this->validConfig['paths']['sharedvar']['default']          = $base.HTDOCS_DIR.'/var/';
