--- src/tiff.imageio/tiffoutput.cpp.orig	2015-02-12 15:30:01.000000000 +1000
+++ src/tiff.imageio/tiffoutput.cpp	2015-03-12 17:46:38.306214366 +1000
@@ -518,7 +518,7 @@
 bool
 TIFFOutput::write_exif_data ()
 {
-#if TIFFLIB_VERSION >= 20120922
+#if TIFFLIB_VERSION > 20120922
     // Older versions of libtiff do not support writing Exif directories
 
     // First, see if we have any Exif data at all
