--- ./src/libutil/filesystem.cpp	2022-07-02 00:43:12.000000000 +0200
+++ ./src/libutil/filesystem.cpp.fix	2024-06-15 19:20:43.444592937 +0200
@@ -227,7 +227,7 @@
         const filesystem::path d(u8path(d_utf8));
         filesystem::path f = d / filename;
         error_code ec;
-        if (filesystem::is_regular(f, ec)) {
+        if (filesystem::is_regular_file(f, ec)) {
             return pathstr(f);
         }
 
@@ -922,7 +922,7 @@
              !ec && it != end_it; ++it) {
             std::string itpath = Filesystem::generic_filepath(
                 it->path().string());
-            if (filesystem::is_regular(itpath, ec)) {
+            if (filesystem::is_regular_file(itpath, ec)) {
                 const std::string f = pathstr(itpath);
                 match_results<std::string::const_iterator> frame_match;
                 if (regex_match(f, frame_match, pattern_re)) {
