diff -Naur xpra-4.4.3/setup.py xpra-4.4.3.patched/setup.py
--- xpra-4.4.3/setup.py	2022-11-27 09:21:57.000000000 -0500
+++ xpra-4.4.3.patched/setup.py	2023-01-12 13:02:42.255843773 -0500
@@ -2045,8 +2045,8 @@
 
 nvidia_ENABLED = nvenc_ENABLED or nvfbc_ENABLED or nvjpeg_encoder_ENABLED or nvjpeg_decoder_ENABLED
 toggle_packages(nvidia_ENABLED, "xpra.codecs.nvidia")
+CUDA_BIN = f"{share_xpra}/cuda"
 if nvidia_ENABLED:
-    CUDA_BIN = f"{share_xpra}/cuda"
     #find nvcc:
     from xpra.util import sorted_nicely  # pylint: disable=import-outside-toplevel
     path_options = os.environ.get("PATH", "").split(os.path.pathsep)
