From 374245ec1418e7e1e57120fcaf0a12ec695f5f6d Mon Sep 17 00:00:00 2001
From: Michal Sekletar <msekleta@redhat.com>
Date: Wed, 24 Oct 2018 15:22:19 +0000
Subject: [PATCH] avahi-client: fix resource leak

---
 avahi-client/browser.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/avahi-client/browser.c b/avahi-client/browser.c
index c978d942..fa4a9a80 100644
--- a/avahi-client/browser.c
+++ b/avahi-client/browser.c
@@ -72,6 +72,8 @@ static void parse_domain_file(AvahiDomainBrowser *b) {
         if (avahi_normalize_name(buf, domain, sizeof(domain)))
             b->static_browse_domains = avahi_string_list_add(b->static_browse_domains, domain);
     }
+
+    fclose(f);
 }
 
 static void domain_browser_ref(AvahiDomainBrowser *db) {
