From 4b48927e8e2c721d103018b4ce39a164b6c2898f Mon Sep 17 00:00:00 2001
From: Michal Sekletar <msekleta@redhat.com>
Date: Wed, 24 Oct 2018 15:38:48 +0000
Subject: [PATCH] chroot: fix bogus assignments in assertions

---
 avahi-daemon/chroot.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/avahi-daemon/chroot.c b/avahi-daemon/chroot.c
index ccd56be0..871b3b34 100644
--- a/avahi-daemon/chroot.c
+++ b/avahi-daemon/chroot.c
@@ -188,8 +188,8 @@ static int recv_fd(int fd) {
             return -1;
         }
 
-        assert(h->cmsg_len = CMSG_LEN(sizeof(int)));
-        assert(h->cmsg_level = SOL_SOCKET);
+        assert(h->cmsg_len == CMSG_LEN(sizeof(int)));
+        assert(h->cmsg_level == SOL_SOCKET);
         assert(h->cmsg_type == SCM_RIGHTS);
 
         return *((int*)CMSG_DATA(h));
