diff --exclude-from=/home/dang/.diffrc -up -ruN evince-0.7.1.orig/data/evince.desktop.in.in evince-0.7.1/data/evince.desktop.in.in
--- evince-0.7.1.orig/data/evince.desktop.in.in	2007-01-10 11:04:20.000000000 -0500
+++ evince-0.7.1/data/evince.desktop.in.in	2007-01-18 17:30:52.000000000 -0500
@@ -9,7 +9,6 @@ StartupNotify=true
 Terminal=false
 Type=Application
 Icon=evince
-NoDisplay=true
 X-GNOME-DocPath=
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=evince
