--- options.h.old	2008-01-29 11:54:45.000000000 -0200
+++ options.h	2008-01-29 11:54:52.000000000 -0200
@@ -197,7 +197,7 @@
 /* The command to invoke for xauth when using X11 forwarding.
  * "-q" for quiet */
 #ifndef XAUTH_COMMAND
-#define XAUTH_COMMAND "/usr/X11R6/bin/xauth -q"
+#define XAUTH_COMMAND "/usr/bin/xauth -q"
 #endif
 
 /* if you want to enable running an sftp server (such as the one included with
