diff -Naur gdal-2.2.4.orig/frmts/pdf/pdfobject.cpp gdal-2.2.4/frmts/pdf/pdfobject.cpp
--- gdal-2.2.4.orig/frmts/pdf/pdfobject.cpp	2018-03-19 14:26:54.000000000 +0100
+++ gdal-2.2.4/frmts/pdf/pdfobject.cpp	2019-10-15 06:58:41.248215000 +0200
@@ -1055,7 +1055,7 @@
 {
     if (GetType() == PDFObjectType_String)
     {
-        GooString* gooString = m_po->getString();
+        const GooString* gooString = m_po->getString();
         return (osStr = GDALPDFGetUTF8StringFromBytes(reinterpret_cast<const GByte*>(gooString->getCString()),
                                                       static_cast<int>(gooString->getLength())));
     }
