diff -ur lppf-0.1-rc1/src/libppf.cc lppf-0.1-rc1/src/libppf.cc
--- lppf-0.1-rc1/src/libppf.cc	2008-01-11 09:14:56.000000000 -0800
+++ lppf-0.1-rc1/src/libppf.cc	2016-12-15 21:40:03.440589358 -0800
@@ -122,11 +122,12 @@
          // Read and add chunk data. If chunkSize is 0, the first byte is the
          // data and the second is number of repetitions. This is only valid
          // for PPFv1 patches
-         if (chunkSize == 0)
+         if (chunkSize == 0) {
             if (version == 1)
                return ERROR_PPF_FORMAT;
             else if (version == 3)
                chunkSize = 2;
+         }
 
          if ((signed)fread(buf, 1, chunkSize, file) != chunkSize)
             return ERROR_PPF_READ;
