diff -ur sqlpp11-connector-sqlite3-0.23.old/CMakeLists.txt sqlpp11-connector-sqlite3-0.23/CMakeLists.txt
--- sqlpp11-connector-sqlite3-0.23.old/CMakeLists.txt	2016-09-03 08:37:59.000000000 -0300
+++ sqlpp11-connector-sqlite3-0.23/CMakeLists.txt	2016-12-15 17:13:47.022040304 -0200
@@ -75,7 +75,6 @@
 include_directories("${DATE_INCLUDE_DIR}")
 include_directories("${SQLITE3_INCLUDE_DIRS}")
 set(include_dir "${PROJECT_SOURCE_DIR}/include")
-file(GLOB_RECURSE sqlpp_headers ${include_dir}/*.h ${SQLPP11_INCLUDE_DIR}/*.h)
 include_directories(${include_dir})
 
 add_subdirectory(src)
Only in sqlpp11-connector-sqlite3-0.23/src: .CMakeLists.txt.un~
diff -ur sqlpp11-connector-sqlite3-0.23.old/src/CMakeLists.txt sqlpp11-connector-sqlite3-0.23/src/CMakeLists.txt
--- sqlpp11-connector-sqlite3-0.23.old/src/CMakeLists.txt	2016-09-03 08:37:59.000000000 -0300
+++ sqlpp11-connector-sqlite3-0.23/src/CMakeLists.txt	2016-12-15 17:15:24.270043748 -0200
@@ -4,4 +4,4 @@
 		prepared_statement.cpp
 		detail/connection_handle.cpp)
 
-install(TARGETS sqlpp11-connector-sqlite3 DESTINATION lib)
+install(TARGETS sqlpp11-connector-sqlite3 DESTINATION lib${LIB_SUFFIX})
