diff -ru sqlpp11-connector-sqlite3-0.25.old/CMakeLists.txt sqlpp11-connector-sqlite3-0.25/CMakeLists.txt
--- sqlpp11-connector-sqlite3-0.25.old/CMakeLists.txt	2017-09-02 06:15:02.000000000 -0300
+++ sqlpp11-connector-sqlite3-0.25/CMakeLists.txt	2017-09-26 07:55:27.798021515 -0300
@@ -82,7 +82,7 @@
 include_directories("${DATE_INCLUDE_DIR}")
 include_directories("${SQLITE3_INCLUDE_DIRS}")
 set(include_dir "${PROJECT_SOURCE_DIR}/include")
-file(GLOB_RECURSE sqlpp_headers ${include_dir}/*.h ${SQLPP11_INCLUDE_DIR}/*.h)
+#file(GLOB_RECURSE sqlpp_headers ${include_dir}/*.h ${SQLPP11_INCLUDE_DIR}/*.h)
 include_directories(${include_dir})
 
 add_subdirectory(src)
diff -ru sqlpp11-connector-sqlite3-0.25.old/src/CMakeLists.txt sqlpp11-connector-sqlite3-0.25/src/CMakeLists.txt
--- sqlpp11-connector-sqlite3-0.25.old/src/CMakeLists.txt	2017-09-02 06:15:02.000000000 -0300
+++ sqlpp11-connector-sqlite3-0.25/src/CMakeLists.txt	2017-09-26 07:52:43.037017830 -0300
@@ -17,5 +17,5 @@
 target_compile_options(sqlpp11-connector-sqlite3-dynamic INTERFACE -DSQLPP_DYNAMIC_LOADING)
 target_compile_options(sqlpp11-connector-sqlite3-dynamic PRIVATE -DSQLPP_DYNAMIC_LOADING -DSQLPP_DYNAMIC_LOADING_FILENAME=${SQLITE3_LIB_FILE})
 
-install(TARGETS sqlpp11-connector-sqlite3 DESTINATION lib)
-install(TARGETS sqlpp11-connector-sqlite3-dynamic DESTINATION lib)
+install(TARGETS sqlpp11-connector-sqlite3 DESTINATION lib${LIB_SUFFIX})
+install(TARGETS sqlpp11-connector-sqlite3-dynamic DESTINATION lib${LIB_SUFFIX})
