diff -urN ghostpdl-9.54.0/tesseract/src/ccutil/strngs.h ghostpdl/tesseract/src/ccutil/strngs.h
--- ghostpdl-9.54.0/tesseract/src/ccutil/strngs.h	2021-03-22 13:05:59.000000000 +0200
+++ ghostpdl/tesseract/src/ccutil/strngs.h	2021-04-01 17:22:19.653876318 +0300
@@ -35,6 +35,7 @@
 class STRING : public std::string {
  public:
   using std::string::string;
+  STRING() : std::string() {}
   STRING(const std::string &s) : std::string(s) {}
   STRING(const char *s) : std::string(s ? s : "") {}
 
