diff -Naur wmwebcam-0.3.5-fixed/Makefile wmwebcam-0.3.5-customize/Makefile
--- wmwebcam-0.3.5-fixed/Makefile	2015-06-01 11:40:40.000000000 -0700
+++ wmwebcam-0.3.5-customize/Makefile	2015-11-24 02:23:25.695647261 -0700
@@ -1,5 +1,5 @@
 CC	= gcc
-CFLAGS  = -O2 -Wall -I/usr/X11R6/include -DVIDEO_DEV=\"/dev/video0\"
+CFLAGS  = -O2 -Wall -I/usr/X11R6/include -DVIDEO_DEV=\"/dev/video0\"
 LDFLAGS = 
 LIBS	= -L/usr/X11R6/lib -lX11 -lXext -lm -ljpeg -lv4l1
 
diff -Naur wmwebcam-0.3.5-fixed/wmwebcam.c wmwebcam-0.3.5-customize/wmwebcam.c
--- wmwebcam-0.3.5-fixed/wmwebcam.c	2015-06-01 11:37:49.000000000 -0700
+++ wmwebcam-0.3.5-customize/wmwebcam.c	2015-11-24 02:22:29.172647747 -0700
@@ -46,15 +46,15 @@
 
 //////////////////////// CHANGE THESE IF NECESSARY ///////////////////////////
 
-#define DEF_WIDTH	352         // changing these requires adjustements
-#define DEF_HEIGHT	288         // to the source, use default if possible
+#define DEF_WIDTH	352         // changing these requires adjustements
+#define DEF_HEIGHT	288         // to the source, use default if possible
 
-#define SENDINGDELAY	60          // default delay between saving jpeg
+#define SENDINGDELAY	60          // default delay between saving jpeg
                                     // images and scriptrunning (in seconds)
 
-#define OUTPUTFILE	"/tmp/wmwebcam.jpg"     // default output file
-#define CUSTOMSCRIPT    "wmwebcam.pl"           // default custom script
-#define QUAL_DEFAULT	100                     // default jpeg outputquality
+#define OUTPUTFILE	"/tmp/wmwebcam.jpg"     // default output file
+#define CUSTOMSCRIPT    "wmwebcam.pl"           // default custom script
+#define QUAL_DEFAULT	100                     // default jpeg outputquality
 
 //////////////////////////////////////////////////////////////////////////////
 
diff -Naur wmwebcam-0.3.5-fixed/wmwebcam.pl wmwebcam-0.3.5-customize/wmwebcam.pl
--- wmwebcam-0.3.5-fixed/wmwebcam.pl	2001-01-27 09:24:16.000000000 -0700
+++ wmwebcam-0.3.5-customize/wmwebcam.pl	2015-11-24 02:23:01.323647470 -0700
@@ -5,12 +5,12 @@
 
 # check if the ppp is on before stamping&sending, otherwise don't do
 # anything
-$areweonline = `grep "ppp" /proc/net/route`;
+$areweonline = `grep "ppp" /proc/net/route`;
 
-if (length $areweonline != 0) { # THIS IS JUST MY CONFIGURATION,
+if (length $areweonline != 0) { # THIS IS JUST MY CONFIGURATION,
                                 # CHANGE TO YOUR OWN
 
-system "stamp";    # stamp reads /tmp/wmwebcam.jpg and outputs
+system "stamp";    # stamp reads /tmp/wmwebcam.jpg and outputs
                    # /tmp/webcam.jpg with some info (has to be
                    # configured to do so)
 # get stamp from:
@@ -19,7 +19,7 @@
 # NOTE: stamp has the ability to send the image to server via ftp, but I
 # prefer using scp.
 
-system "scp -q /tmp/webcam.jpg SOME_HOST:public_html/webcam.jpg";
+system "scp -q /tmp/webcam.jpg SOME_HOST:public_html/webcam.jpg";
                    # replace the previous line to suit your needs
                    # or disable if you use stamp's own ftpsend
 } else {
