diff -Nur spl-1.0pre6.orig//GNUmakefile spl-1.0pre6/GNUmakefile
--- spl-1.0pre6.orig//GNUmakefile	2008-09-18 07:34:23.000000000 -0700
+++ spl-1.0pre6/GNUmakefile	2012-05-18 19:04:38.769984565 -0700
@@ -102,7 +102,7 @@
 # Set this to '1' if you want to build an unoptimized SPL
 # library with debug symbols.
 #
-BUILD_WITH_DEBUG = 1
+BUILD_WITH_DEBUG = 0
 
 # Set this to '1' if you are doing SPL core development
 #
@@ -225,7 +225,7 @@
 ifeq ($(BUILD_WITH_DEBUG),1)
   CFLAGS += -O1 -ggdb
 else
-  CFLAGS += -Os -fomit-frame-pointer
+  CFLAGS += -fomit-frame-pointer $(SLCKFLAGS)
 endif
 
 ifeq ($(BUILD_WITH_WERROR),1)
