diff -Naur OLD/opt/openoffice.org2.2/share/xdg/base.desktop NEW/opt/openoffice.org2.2/share/xdg/base.desktop
--- OLD/opt/openoffice.org2.2/share/xdg/base.desktop	2007-02-16 08:59:53.000000000 -0600
+++ NEW/opt/openoffice.org2.2/share/xdg/base.desktop	2007-02-21 17:22:21.000000000 -0600
@@ -2,11 +2,11 @@
 Version=1.0
 Encoding=UTF-8
 Terminal=false
-Icon=openofficeorg22-base
+Icon=/usr/share/pixmaps/openofficeorg22-base.png
 Type=Application
 InitialPreference=2
 Categories=Application;Office;X-Red-Hat-Base;X-SuSE-Core-Office;
-Exec=openoffice.org2.2 -base %U
+Exec=sbase %U
 MimeType=application/vnd.oasis.opendocument.database;application/vnd.sun.xml.base
 Name=OpenOffice.org 2.2 Base
 Comment=Manage databases, create queries and reports to track and manage your information by using Base.
diff -Naur OLD/opt/openoffice.org2.2/share/xdg/calc.desktop NEW/opt/openoffice.org2.2/share/xdg/calc.desktop
--- OLD/opt/openoffice.org2.2/share/xdg/calc.desktop	2007-02-16 08:58:50.000000000 -0600
+++ NEW/opt/openoffice.org2.2/share/xdg/calc.desktop	2007-02-21 17:22:28.000000000 -0600
@@ -2,11 +2,11 @@
 Version=1.0
 Encoding=UTF-8
 Terminal=false
-Icon=openofficeorg22-calc
+Icon=/usr/share/pixmaps/openofficeorg22-calc.png
 Type=Application
 InitialPreference=2
 Categories=Application;Office;X-Red-Hat-Base;X-SuSE-Core-Office;
-Exec=openoffice.org2.2 -calc %U
+Exec=scalc %U
 MimeType=application/vnd.oasis.opendocument.spreadsheet;application/vnd.oasis.opendocument.spreadsheet-template;application/vnd.sun.xml.calc;application/vnd.sun.xml.calc.template;application/vnd.stardivision.calc;application/vnd.stardivision.chart;application/msexcel;application/vnd.ms-excel
 Name=OpenOffice.org 2.2 Calc
 Comment=Perform calculation, analyze information and manage lists in spreadsheets by using Calc.
diff -Naur OLD/opt/openoffice.org2.2/share/xdg/draw.desktop NEW/opt/openoffice.org2.2/share/xdg/draw.desktop
--- OLD/opt/openoffice.org2.2/share/xdg/draw.desktop	2007-02-16 08:59:31.000000000 -0600
+++ NEW/opt/openoffice.org2.2/share/xdg/draw.desktop	2007-02-21 17:22:33.000000000 -0600
@@ -2,11 +2,11 @@
 Version=1.0
 Encoding=UTF-8
 Terminal=false
-Icon=openofficeorg22-draw
+Icon=/usr/share/pixmaps/openofficeorg22-draw.png
 Type=Application
 InitialPreference=2
 Categories=Application;Office;X-Red-Hat-Base;X-SuSE-Core-Office;
-Exec=openoffice.org2.2 -draw %U
+Exec=sdraw %U
 MimeType=application/vnd.oasis.opendocument.graphics;application/vnd.oasis.opendocument.graphics-template;application/vnd.sun.xml.draw;application/vnd.sun.xml.draw.template;application/vnd.stardivision.draw
 Name=OpenOffice.org 2.2 Draw
 Comment=Create and edit drawings, flow charts, and logos by using Draw.
diff -Naur OLD/opt/openoffice.org2.2/share/xdg/impress.desktop NEW/opt/openoffice.org2.2/share/xdg/impress.desktop
--- OLD/opt/openoffice.org2.2/share/xdg/impress.desktop	2007-02-16 08:59:40.000000000 -0600
+++ NEW/opt/openoffice.org2.2/share/xdg/impress.desktop	2007-02-21 17:23:56.000000000 -0600
@@ -2,11 +2,11 @@
 Version=1.0
 Encoding=UTF-8
 Terminal=false
-Icon=openofficeorg22-impress
+Icon=/usr/share/pixmaps/openofficeorg22-impress.png
 Type=Application
 InitialPreference=2
 Categories=Application;Office;X-Red-Hat-Base;X-SuSE-Core-Office;
-Exec=openoffice.org2.2 -impress %U
+Exec=simpress %U
 MimeType=application/vnd.oasis.opendocument.presentation;application/vnd.oasis.opendocument.presentation-template;application/vnd.sun.xml.impress;application/vnd.sun.xml.impress.template;application/vnd.stardivision.impress;application/mspowerpoint;application/vnd.ms-powerpoint
 Name=OpenOffice.org 2.2 Impress
 Comment=Create and edit presentations for slideshows, meeting and Web pages by using Impress.
diff -Naur OLD/opt/openoffice.org2.2/share/xdg/math.desktop NEW/opt/openoffice.org2.2/share/xdg/math.desktop
--- OLD/opt/openoffice.org2.2/share/xdg/math.desktop	2007-02-16 09:00:12.000000000 -0600
+++ NEW/opt/openoffice.org2.2/share/xdg/math.desktop	2007-02-21 17:24:01.000000000 -0600
@@ -2,11 +2,11 @@
 Version=1.0
 Encoding=UTF-8
 Terminal=false
-Icon=openofficeorg22-math
+Icon=/usr/share/pixmaps/openofficeorg22-math.png
 Type=Application
 InitialPreference=2
 Categories=Application;Office;X-Red-Hat-Base;X-SuSE-Core-Office;
-Exec=openoffice.org2.2 -math %U
+Exec=smath %U
 MimeType=application/vnd.oasis.opendocument.formula;application/vnd.sun.xml.math;application/vnd.stardivision.math
 Name=OpenOffice.org 2.2 Math
 Comment=Create and edit scientific formulas and equations by using Math.
diff -Naur OLD/opt/openoffice.org2.2/share/xdg/printeradmin.desktop NEW/opt/openoffice.org2.2/share/xdg/printeradmin.desktop
--- OLD/opt/openoffice.org2.2/share/xdg/printeradmin.desktop	2007-02-16 09:18:11.000000000 -0600
+++ NEW/opt/openoffice.org2.2/share/xdg/printeradmin.desktop	2007-02-21 17:21:50.000000000 -0600
@@ -2,7 +2,7 @@
 Version=1.0
 Encoding=UTF-8
 Terminal=false
-Icon=openofficeorg22-printeradmin.png
+Icon=/usr/share/pixmaps/openofficeorg22-printeradmin.png
 Type=Application
 Categories=Application;Office;X-Red-Hat-Base;X-SuSE-Core-Office;
 Exec=openoffice.org2.2-printeradmin
diff -Naur OLD/opt/openoffice.org2.2/share/xdg/writer.desktop NEW/opt/openoffice.org2.2/share/xdg/writer.desktop
--- OLD/opt/openoffice.org2.2/share/xdg/writer.desktop	2007-02-16 08:58:36.000000000 -0600
+++ NEW/opt/openoffice.org2.2/share/xdg/writer.desktop	2007-02-21 17:24:38.000000000 -0600
@@ -2,11 +2,11 @@
 Version=1.0
 Encoding=UTF-8
 Terminal=false
-Icon=openofficeorg22-writer
+Icon=/usr/share/pixmaps/openofficeorg22-writer.png
 Type=Application
 InitialPreference=2
 Categories=Application;Office;X-Red-Hat-Base;X-SuSE-Core-Office;
-Exec=openoffice.org2.2 -writer %U
+Exec=swriter %U
 MimeType=application/vnd.oasis.opendocument.text;application/vnd.oasis.opendocument.text-template;application/vnd.oasis.opendocument.text-web;application/vnd.oasis.opendocument.text-master;application/vnd.sun.xml.writer;application/vnd.sun.xml.writer.template;application/vnd.sun.xml.writer.global;application/vnd.stardivision.writer;application/msword;application/vnd.ms-word;application/x-doc;application/rtf;application/vnd.wordperfect;application/wordperfect
 Name=OpenOffice.org 2.2 Writer
 Comment=Create and edit text and graphics in letters, reports, documents and Web pages by using Writer.
