diff -Nur orig/opt/openoffice.org2.4/share/xdg/base.desktop fixed/opt/openoffice.org2.4/share/xdg/base.desktop
--- orig/opt/openoffice.org2.4/share/xdg/base.desktop	2008-03-14 15:42:17.000000000 -0500
+++ fixed/opt/openoffice.org2.4/share/xdg/base.desktop	2008-03-23 16:14:17.331439021 -0500
@@ -6,7 +6,7 @@
 Type=Application
 InitialPreference=2
 Categories=Application;Office;X-Red-Hat-Base;X-SuSE-Core-Office;X-MandrivaLinux-MoreApplications-Databases;
-Exec=openoffice.org2.4 -base %U
+Exec=sbase %U
 MimeType=application/vnd.oasis.opendocument.database;application/vnd.sun.xml.base
 Name=OpenOffice.org 2.4 Base
 GenericName=Database Development
diff -Nur orig/opt/openoffice.org2.4/share/xdg/calc.desktop fixed/opt/openoffice.org2.4/share/xdg/calc.desktop
--- orig/opt/openoffice.org2.4/share/xdg/calc.desktop	2008-03-14 15:42:40.000000000 -0500
+++ fixed/opt/openoffice.org2.4/share/xdg/calc.desktop	2008-03-23 16:14:24.724377432 -0500
@@ -6,7 +6,7 @@
 Type=Application
 InitialPreference=2
 Categories=Application;Office;X-Red-Hat-Base;X-SuSE-Core-Office;X-MandrivaLinux-Office-Spreadsheets;
-Exec=openoffice.org2.4 -calc %U
+Exec=scalc %U
 MimeType=application/vnd.oasis.opendocument.spreadsheet;application/vnd.oasis.opendocument.spreadsheet-template;application/vnd.sun.xml.calc;application/vnd.sun.xml.calc.template;application/vnd.stardivision.calc;application/vnd.stardivision.chart;application/msexcel;application/vnd.ms-excel
 Name=OpenOffice.org 2.4 Calc
 GenericName=Spreadsheet
diff -Nur orig/opt/openoffice.org2.4/share/xdg/draw.desktop fixed/opt/openoffice.org2.4/share/xdg/draw.desktop
--- orig/opt/openoffice.org2.4/share/xdg/draw.desktop	2008-03-14 15:42:10.000000000 -0500
+++ fixed/opt/openoffice.org2.4/share/xdg/draw.desktop	2008-03-23 16:14:45.627028307 -0500
@@ -6,7 +6,7 @@
 Type=Application
 InitialPreference=2
 Categories=Application;Office;X-Red-Hat-Base;X-SuSE-Core-Office;X-MandrivaLinux-Office-Drawing;
-Exec=openoffice.org2.4 -draw %U
+Exec=sdraw %U
 MimeType=application/vnd.oasis.opendocument.graphics;application/vnd.oasis.opendocument.graphics-template;application/vnd.sun.xml.draw;application/vnd.sun.xml.draw.template;application/vnd.stardivision.draw
 Name=OpenOffice.org 2.4 Draw
 Comment=Create and edit drawings, flow charts, and logos by using Draw.
diff -Nur orig/opt/openoffice.org2.4/share/xdg/impress.desktop fixed/opt/openoffice.org2.4/share/xdg/impress.desktop
--- orig/opt/openoffice.org2.4/share/xdg/impress.desktop	2008-03-14 15:42:01.000000000 -0500
+++ fixed/opt/openoffice.org2.4/share/xdg/impress.desktop	2008-03-23 16:15:35.953410981 -0500
@@ -6,7 +6,7 @@
 Type=Application
 InitialPreference=2
 Categories=Application;Office;X-Red-Hat-Base;X-SuSE-Core-Office;X-MandrivaLinux-Office-Presentations;
-Exec=openoffice.org2.4 -impress %U
+Exec=simpress %U
 MimeType=application/vnd.oasis.opendocument.presentation;application/vnd.oasis.opendocument.presentation-template;application/vnd.sun.xml.impress;application/vnd.sun.xml.impress.template;application/vnd.stardivision.impress;application/mspowerpoint;application/vnd.ms-powerpoint
 Name=OpenOffice.org 2.4 Impress
 GenericName=Presentation
diff -Nur orig/opt/openoffice.org2.4/share/xdg/math.desktop fixed/opt/openoffice.org2.4/share/xdg/math.desktop
--- orig/opt/openoffice.org2.4/share/xdg/math.desktop	2008-03-14 15:42:29.000000000 -0500
+++ fixed/opt/openoffice.org2.4/share/xdg/math.desktop	2008-03-23 16:15:44.998558367 -0500
@@ -6,7 +6,7 @@
 Type=Application
 InitialPreference=2
 Categories=Application;Office;X-Red-Hat-Base;X-SuSE-Core-Office;X-MandrivaLinux-Office-Other;
-Exec=openoffice.org2.4 -math %U
+Exec=smath %U
 MimeType=application/vnd.oasis.opendocument.formula;application/vnd.sun.xml.math;application/vnd.stardivision.math
 Name=OpenOffice.org 2.4 Math
 Comment=Create and edit scientific formulas and equations by using Math.
diff -Nur orig/opt/openoffice.org2.4/share/xdg/printeradmin.desktop fixed/opt/openoffice.org2.4/share/xdg/printeradmin.desktop
--- orig/opt/openoffice.org2.4/share/xdg/printeradmin.desktop	2008-03-14 15:36:23.000000000 -0500
+++ fixed/opt/openoffice.org2.4/share/xdg/printeradmin.desktop	2008-03-23 16:13:25.876915512 -0500
@@ -2,7 +2,7 @@
 Version=1.0
 Encoding=UTF-8
 Terminal=false
-Icon=openofficeorg24-printeradmin.png
+Icon=openofficeorg24-printeradmin
 Type=Application
 Categories=Application;Office;X-Red-Hat-Base;X-SuSE-Core-Office;
 Exec=openoffice.org2.4-printeradmin
diff -Nur orig/opt/openoffice.org2.4/share/xdg/writer.desktop fixed/opt/openoffice.org2.4/share/xdg/writer.desktop
--- orig/opt/openoffice.org2.4/share/xdg/writer.desktop	2008-03-14 15:41:47.000000000 -0500
+++ fixed/opt/openoffice.org2.4/share/xdg/writer.desktop	2008-03-23 16:16:04.965089685 -0500
@@ -6,7 +6,7 @@
 Type=Application
 InitialPreference=2
 Categories=Application;Office;X-Red-Hat-Base;X-SuSE-Core-Office;X-MandrivaLinux-Office-Wordprocessors;
-Exec=openoffice.org2.4 -writer %U
+Exec=swriter %U
 MimeType=application/vnd.oasis.opendocument.text;application/vnd.oasis.opendocument.text-template;application/vnd.oasis.opendocument.text-web;application/vnd.oasis.opendocument.text-master;application/vnd.sun.xml.writer;application/vnd.sun.xml.writer.template;application/vnd.sun.xml.writer.global;application/vnd.stardivision.writer;application/msword;application/vnd.ms-word;application/x-doc;application/rtf;application/vnd.wordperfect;application/wordperfect
 Name=OpenOffice.org 2.4 Writer
 GenericName=Word Processor
