diff -Nur a/opt/openoffice.org3/share/xdg/base.desktop b/opt/openoffice.org3/share/xdg/base.desktop
--- a/opt/openoffice.org3/share/xdg/base.desktop	2008-09-19 20:04:46.000000000 -0500
+++ b/opt/openoffice.org3/share/xdg/base.desktop	2008-09-30 14:05:11.639524235 -0500
@@ -4,7 +4,7 @@
 Icon=openofficeorg3-base
 Type=Application
 Categories=Office;X-Red-Hat-Base;X-SuSE-Core-Office;X-MandrivaLinux-MoreApplications-Databases;
-Exec=openoffice.org3 -base %U
+Exec=sbase %U
 MimeType=application/vnd.oasis.opendocument.database;application/vnd.sun.xml.base;
 Name=OpenOffice.org 3.0 Base
 GenericName=Database Development
diff -Nur a/opt/openoffice.org3/share/xdg/calc.desktop b/opt/openoffice.org3/share/xdg/calc.desktop
--- a/opt/openoffice.org3/share/xdg/calc.desktop	2008-09-19 20:04:46.000000000 -0500
+++ b/opt/openoffice.org3/share/xdg/calc.desktop	2008-09-30 14:05:19.224052521 -0500
@@ -4,7 +4,7 @@
 Icon=openofficeorg3-calc
 Type=Application
 Categories=Office;X-Red-Hat-Base;X-SuSE-Core-Office;X-MandrivaLinux-Office-Spreadsheets;
-Exec=openoffice.org3 -calc %U
+Exec=scalc %U
 MimeType=application/vnd.oasis.opendocument.spreadsheet;application/vnd.oasis.opendocument.spreadsheet-template;application/vnd.sun.xml.calc;application/vnd.sun.xml.calc.template;application/vnd.stardivision.calc;application/vnd.stardivision.chart;application/msexcel;application/vnd.ms-excel;application/vnd.openxmlformats-officedocument.spreadsheetml.sheet;application/vnd.ms-excel.sheet.macroenabled.12;application/vnd.openxmlformats-officedocument.spreadsheetml.template;application/vnd.ms-excel.template.macroenabled.12;application/vnd.ms-excel.sheet.binary.macroenabled.12;
 Name=OpenOffice.org 3.0 Calc
 GenericName=Spreadsheet
diff -Nur a/opt/openoffice.org3/share/xdg/draw.desktop b/opt/openoffice.org3/share/xdg/draw.desktop
--- a/opt/openoffice.org3/share/xdg/draw.desktop	2008-09-19 20:04:46.000000000 -0500
+++ b/opt/openoffice.org3/share/xdg/draw.desktop	2008-09-30 14:05:26.016528565 -0500
@@ -4,7 +4,7 @@
 Icon=openofficeorg3-draw
 Type=Application
 Categories=Office;X-Red-Hat-Base;X-SuSE-Core-Office;X-MandrivaLinux-Office-Drawing;
-Exec=openoffice.org3 -draw %U
+Exec=sdraw %U
 MimeType=application/vnd.oasis.opendocument.graphics;application/vnd.oasis.opendocument.graphics-template;application/vnd.sun.xml.draw;application/vnd.sun.xml.draw.template;application/vnd.stardivision.draw;
 Name=OpenOffice.org 3.0 Draw
 Comment=Create and edit drawings, flow charts, and logos by using Draw.
diff -Nur a/opt/openoffice.org3/share/xdg/impress.desktop b/opt/openoffice.org3/share/xdg/impress.desktop
--- a/opt/openoffice.org3/share/xdg/impress.desktop	2008-09-19 20:04:46.000000000 -0500
+++ b/opt/openoffice.org3/share/xdg/impress.desktop	2008-09-30 14:05:32.997015784 -0500
@@ -4,7 +4,7 @@
 Icon=openofficeorg3-impress
 Type=Application
 Categories=Office;X-Red-Hat-Base;X-SuSE-Core-Office;X-MandrivaLinux-Office-Presentations;
-Exec=openoffice.org3 -impress %U
+Exec=simpress %U
 MimeType=application/vnd.oasis.opendocument.presentation;application/vnd.oasis.opendocument.presentation-template;application/vnd.sun.xml.impress;application/vnd.sun.xml.impress.template;application/vnd.stardivision.impress;application/mspowerpoint;application/vnd.ms-powerpoint;application/vnd.openxmlformats-officedocument.presentationml.presentation;application/vnd.ms-powerpoint.presentation.macroenabled.12;application/vnd.openxmlformats-officedocument.presentationml.template;application/vnd.ms-powerpoint.template.macroenabled.12;
 Name=OpenOffice.org 3.0 Impress
 GenericName=Presentation
diff -Nur a/opt/openoffice.org3/share/xdg/math.desktop b/opt/openoffice.org3/share/xdg/math.desktop
--- a/opt/openoffice.org3/share/xdg/math.desktop	2008-09-19 20:04:46.000000000 -0500
+++ b/opt/openoffice.org3/share/xdg/math.desktop	2008-09-30 14:05:40.449534571 -0500
@@ -4,7 +4,7 @@
 Icon=openofficeorg3-math
 Type=Application
 Categories=Office;X-Red-Hat-Base;X-SuSE-Core-Office;X-MandrivaLinux-Office-Other;
-Exec=openoffice.org3 -math %U
+Exec=smath %U
 MimeType=application/vnd.oasis.opendocument.formula;application/vnd.sun.xml.math;application/vnd.stardivision.math;
 Name=OpenOffice.org 3.0 Math
 Comment=Create and edit scientific formulas and equations by using Math.
diff -Nur a/opt/openoffice.org3/share/xdg/writer.desktop b/opt/openoffice.org3/share/xdg/writer.desktop
--- a/opt/openoffice.org3/share/xdg/writer.desktop	2008-09-19 20:04:46.000000000 -0500
+++ b/opt/openoffice.org3/share/xdg/writer.desktop	2008-09-30 14:06:00.938967174 -0500
@@ -4,7 +4,7 @@
 Icon=openofficeorg3-writer
 Type=Application
 Categories=Office;X-Red-Hat-Base;X-SuSE-Core-Office;X-MandrivaLinux-Office-Wordprocessors;
-Exec=openoffice.org3 -writer %U
+Exec=swriter %U
 MimeType=application/vnd.oasis.opendocument.text;application/vnd.oasis.opendocument.text-template;application/vnd.oasis.opendocument.text-web;application/vnd.oasis.opendocument.text-master;application/vnd.sun.xml.writer;application/vnd.sun.xml.writer.template;application/vnd.sun.xml.writer.global;application/vnd.stardivision.writer;application/msword;application/vnd.ms-word;application/x-doc;application/rtf;text/rtf;application/vnd.wordperfect;application/wordperfect;application/vnd.openxmlformats-officedocument.wordprocessingml.document;application/vnd.ms-word.document.macroenabled.12;application/vnd.openxmlformats-officedocument.wordprocessingml.template;application/vnd.ms-word.template.macroenabled.12;
 Name=OpenOffice.org 3.0 Writer
 GenericName=Word Processor
