--- x11/Makefile.in.orig	2018-01-26 17:56:15.000000000 -0800
+++ x11/Makefile.in	2018-06-14 13:42:43.989390736 -0700
@@ -13,9 +13,9 @@
 
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
-libdir		= @libdir@
-bindir		= @bindir@
-includedir	= @includedir@
+libdir		= $(DESTDIR)@libdir@
+bindir		= $(DESTDIR)@bindir@
+includedir	= $(DESTDIR)@includedir@
 
 PDCURSES_CONFIG_H	=$(osdir)/config.h
 PDCURSES_CURSES_H	=$(PDCURSES_SRCDIR)/curses.h
@@ -62,8 +62,12 @@
 	$(INSTALL) -d -m 755 $(includedir)
 	$(INSTALL) -d -m 755 $(includedir)/xcurses
 	$(INSTALL) -c -m 644 $(PDCURSES_CURSES_H) \
+		$(includedir)/xcurses.h
+	$(INSTALL) -c -m 644 $(PDCURSES_CURSES_H) \
 		$(includedir)/xcurses/curses.h
 	$(INSTALL) -c -m 644 $(PDCURSES_SRCDIR)/panel.h \
+		$(includedir)/xpanel.h
+	$(INSTALL) -c -m 644 $(PDCURSES_SRCDIR)/panel.h \
 		$(includedir)/xcurses/panel.h
 	$(INSTALL) -c -m 644 $(PDCURSES_SRCDIR)/term.h \
 		$(includedir)/xcurses/term.h
