--- PDCursesMod-4.3.2a/x11/Makefile.in.orig	2022-02-06 08:35:35.000000000 -0800
+++ PDCursesMod-4.3.2a/x11/Makefile.in	2022-04-07 09:18:27.712387150 -0700
@@ -13,9 +13,9 @@
 
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
-libdir		= @libdir@
-bindir		= @bindir@
-includedir	= @includedir@
+libdir		= $(DESTDIR)@libdir@
+bindir		= $(DESTDIR)@bindir@
+includedir	= $(DESTDIR)@includedir@
 
 PDCURSES_CONFIG_H	=$(osdir)/config.h
 PDCURSES_CURSES_H	=$(PDCURSES_SRCDIR)/curses.h
