--- PDCurses-3.9/x11/xcurses-config.in.orig	2019-09-04 13:08:02.000000000 -0700
+++ PDCurses-3.9/x11/xcurses-config.in	2022-04-08 17:06:28.631395983 -0700
@@ -21,9 +21,9 @@
     echo "  --cflags       pre-processor and compiler flags"
     echo "                 [-I$includedir @PDC_WIDE@]"
     echo "  --libs         library linking information"
-    echo "                 [-L$libdir -lXCurses @LDFLAGS@ @MH_EXTRA_LIBS@"
+    echo "                 [-L$libdir -lXCurses @LDFLAGS@ @MH_EXTRA_LIBS@]"
     echo "  --libs-static  static library linking information (libXCurses only)"
-    echo "                 [$libdir/libXCurses.a @LDFLAGS@ @MH_XLIBS@ @MH_EXTRA_LIBS@"
+    echo "                 [-l:libXCurses.a @LDFLAGS@ @MH_XLIBS@ @MH_EXTRA_LIBS@]"
     echo "  --prefix       PDCurses install prefix"
     echo "                 [$prefix]"
     echo "  --version      output version information"
@@ -67,7 +67,7 @@
        ;;
 
     --libs-static)
-       echo $libdir/libXCurses.a @LDFLAGS@ @MH_XLIBS@ @MH_EXTRA_LIBS@
+       echo -l:libXCurses.a @LDFLAGS@ @MH_XLIBS@ @MH_EXTRA_LIBS@
        ;;
 
     *)
