diff -up ./src/coolkey/slot.cpp.coolkey-gcc43 ./src/coolkey/slot.cpp
--- ./src/coolkey/slot.cpp.coolkey-gcc43	2008-02-13 18:01:45.000000000 -0800
+++ ./src/coolkey/slot.cpp	2008-02-13 18:03:05.000000000 -0800
@@ -25,7 +25,6 @@
 #include "PKCS11Exception.h"
 #include <winscard.h>
 #include "slot.h"
-#include <memory.h>
 #include "zlib.h"
 #include "params.h"
 
@@ -33,7 +32,6 @@
 
 #define MIN(x, y) ((x) < (y) ? (x) : (y))
 
-using std::auto_ptr;
 
 
 #ifdef DEBUG
diff -up ./src/coolkey/machdep.cpp.coolkey-gcc43 ./src/coolkey/machdep.cpp
--- ./src/coolkey/machdep.cpp.coolkey-gcc43	2008-02-13 18:02:06.000000000 -0800
+++ ./src/coolkey/machdep.cpp	2008-02-13 18:04:04.000000000 -0800
@@ -33,6 +33,8 @@
 #include <sys/stat.h>
 #include <sys/mman.h>
 #include <pthread.h>
+#include <string.h>
+#include <stdlib.h>
 #endif
 
 #ifdef _WIN32
diff -up ./src/coolkey/log.cpp.coolkey-gcc43 ./src/coolkey/log.cpp
--- ./src/coolkey/log.cpp.coolkey-gcc43	2008-02-13 18:01:55.000000000 -0800
+++ ./src/coolkey/log.cpp	2008-02-13 18:03:37.000000000 -0800
@@ -18,6 +18,8 @@
  * ***** END COPYRIGHT BLOCK *****/
 
 #include <string>
+#include <string.h>
+#include <stdlib.h>
 #include "mypkcs11.h"
 #include <assert.h>
 #include <stdio.h>
diff -up ./src/coolkey/object.cpp.coolkey-gcc43 ./src/coolkey/object.cpp
--- ./src/coolkey/object.cpp.coolkey-gcc43	2008-02-13 18:02:20.000000000 -0800
+++ ./src/coolkey/object.cpp	2008-02-13 18:04:22.000000000 -0800
@@ -21,6 +21,7 @@
 #include "PKCS11Exception.h"
 #include "object.h"
 #include <algorithm>
+#include <string.h>
 
 using std::find_if;
 
