From cc15d1c67144200f070e74d80b6d840df0a16ee3 Mon Sep 17 00:00:00 2001
From: Thomas Heller <thomas.heller@cs.fau.de>
Date: Thu, 4 May 2017 08:15:13 +0200
Subject: [PATCH] Fixing uninitialized usage warning with gcc 7

---
 src/runtime/parcelset/parcel.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/runtime/parcelset/parcel.cpp b/src/runtime/parcelset/parcel.cpp
index 0f86c4043a1..56714af4c7b 100644
--- a/src/runtime/parcelset/parcel.cpp
+++ b/src/runtime/parcelset/parcel.cpp
@@ -47,7 +47,8 @@ namespace hpx { namespace parcelset
             creation_time_(util::high_resolution_timer::now()),
 #endif
             source_id_(naming::invalid_gid),
-            dest_(naming::invalid_gid)
+            dest_(naming::invalid_gid),
+            has_continuation_(false)
         {}
 
         parcel_data::parcel_data(naming::gid_type&& dest, naming::address&& addr,
