diff -Naur obcpl-0.9.8/src/cg.c obcpl-0.9.8.patcged/src/cg.c
--- obcpl-0.9.8/src/cg.c	2013-03-25 17:41:43.000000000 -0400
+++ obcpl-0.9.8.patcged/src/cg.c	2021-09-06 04:43:29.444894247 -0400
@@ -374,7 +374,7 @@
                     emit(".equ G%d,%s", x, label(rdn()));
                 }
             }
-            return;
+            break;
         case S_FINISH:
             emit("jmp finish");
             break;
