diff -Nur xfce4-quicklauncher-plugin-1.9.4.orig//panel-plugin/quicklauncher.desktop.in.in xfce4-quicklauncher-plugin-1.9.4/panel-plugin/quicklauncher.desktop.in.in
--- xfce4-quicklauncher-plugin-1.9.4.orig//panel-plugin/quicklauncher.desktop.in.in	2005-09-25 14:08:43.000000000 -0500
+++ xfce4-quicklauncher-plugin-1.9.4/panel-plugin/quicklauncher.desktop.in.in	2011-01-01 15:41:49.621453224 -0600
@@ -4,4 +4,5 @@
 _Name=Quicklauncher
 _Comment=Program with several launchers
 Icon=gnome-fs-executable
-X-XFCE-Module=@INTERNAL_PLUGIN_PATH@/libquicklauncher.so
+X-XFCE-Module=quicklauncher
+X-XFCE-Module-Path=@INTERNAL_PLUGIN_PATH@
