--- wormblast-2.0.2/src/game/game.original.cc	2014-04-17 04:44:16.025841490 +0300
+++ wormblast-2.0.2/src/game/game.cc	2014-04-17 04:45:16.415086521 +0300
@@ -54,6 +54,7 @@
 // **** C++ standard includes ****
 
 #include <cmath>
+#include <unistd.h>
 
 using std::tan;
 
--- wormblast-2.0.2/src/gui_new/about_dlg.original.hh	2014-04-17 05:19:09.918664228 +0300
+++ wormblast-2.0.2/src/gui_new/about_dlg.hh	2014-04-17 05:58:16.569327057 +0300
@@ -24,7 +24,7 @@
 #include <QWidget>
 #include <QResizeEvent>
 
-#include "ui_about.h"
+#include "../../build/src/gui_new/ui_about.h"
 
 //! The about box
 class AboutDlg : public QDialog
--- wormblast-2.0.2/src/gui_new/highscores_dlg.original.hh	2014-04-17 06:01:56.538577063 +0300
+++ wormblast-2.0.2/src/gui_new/highscores_dlg.hh	2014-04-17 06:03:11.096644959 +0300
@@ -21,7 +21,7 @@
 #define HIGHSCORES_DLG_HH
 
 #include <QDialog>
-#include "ui_highscores.h"
+#include "../../build/src/gui_new/ui_highscores.h"
 #include "config.hh"
 
 //! Dialog that shows high scores for one-player and two-player games
--- wormblast-2.0.2/src/gui_new/settings_dlg.original.hh	2014-04-17 06:06:55.647837683 +0300
+++ wormblast-2.0.2/src/gui_new/settings_dlg.hh	2014-04-17 06:07:21.624512930 +0300
@@ -23,7 +23,7 @@
 #include <QDialog>
 #include <QWidget>
 
-#include "ui_settings.h"
+#include "../../build/src/gui_new/ui_settings.h"
 #include "config.hh"
 
 //! The settings dialog
--- wormblast-2.0.2/CMakeLists.original.txt	2014-04-17 06:39:16.678571460 +0300
+++ wormblast-2.0.2/CMakeLists.txt	2014-04-17 07:09:17.741055080 +0300
@@ -17,7 +17,7 @@
 
 # Set install dirs
 set(CMAKE_INSTALL_PREFIX /usr)
-set(DEST_PATH /opt/WormBlast)
+set(DEST_PATH /usr/share/games/WormBlast)
 set(DATA_PATH ${DEST_PATH}/data)
 set(BIN_PATH ${DEST_PATH}/bin)
 add_definitions(-DGAME_BINARY_PATH="${BIN_PATH}/wormblast-game.x")
@@ -57,8 +57,8 @@
 add_subdirectory(src/gui_new)
 
 # Install binaries & data
-install(PROGRAMS bin/game/wormblast-game.x DESTINATION ${BIN_PATH})
-install(PROGRAMS bin/gui/wormblast DESTINATION ${BIN_PATH})
+install(PROGRAMS build/bin/game/wormblast-game.x DESTINATION ${BIN_PATH})
+install(PROGRAMS build/bin/gui/wormblast DESTINATION ${BIN_PATH})
 install(FILES data/sounds.map data/textures.map data/verasans.map DESTINATION ${DATA_PATH})
 install(DIRECTORY data/images DESTINATION ${DATA_PATH} FILES_MATCHING PATTERN "*.jpg")
 install(DIRECTORY data/images DESTINATION ${DATA_PATH} FILES_MATCHING PATTERN "*.png")
--- wormblast-2.0.2/data/wormblast.desktop.original	2014-04-17 07:14:57.288810149 +0300
+++ wormblast-2.0.2/data/wormblast.desktop	2014-04-17 07:16:06.055950441 +0300
@@ -2,7 +2,7 @@
 Name=WormBlast
 Comment[fi]=Action matopeli 1-2 pelaajalle
 Comment=Worm action game for 1-2 players
-Exec=/opt/WormBlast/bin/wormblast
+Exec=wormblast
 Icon=wormblast-32
 Terminal=false
 Type=Application
