diff -Naur bsdiff-4.3.orig/Makefile bsdiff-4.3/Makefile
--- bsdiff-4.3.orig/Makefile	2005-08-17 00:14:10.000000000 +0200
+++ bsdiff-4.3/Makefile	2010-02-19 13:58:05.568018203 +0100
@@ -1,15 +1,16 @@
 CFLAGS		+=	-O3 -lbz2
 
 PREFIX		?=	/usr/local
-INSTALL_PROGRAM	?=	${INSTALL} -c -s -m 555
-INSTALL_MAN	?=	${INSTALL} -c -m 444
+INSTALL_PROGRAM	?=	${INSTALL} -m 0755
+INSTALL_DIR	?=	${INSTALL} -d
+INSTALL_MAN	?=	${INSTALL} -m 0644
 
 all:		bsdiff bspatch
 bsdiff:		bsdiff.c
 bspatch:	bspatch.c
 
 install:
-	${INSTALL_PROGRAM} bsdiff bspatch ${PREFIX}/bin
-.ifndef WITHOUT_MAN
-	${INSTALL_MAN} bsdiff.1 bspatch.1 ${PREFIX}/man/man1
-.endif
+	${INSTALL_DIR} $(DESTDIR)/${PREFIX}/bin
+	${INSTALL_DIR} $(DESTDIR)/${PREFIX}/man/man1
+	${INSTALL_PROGRAM} bsdiff bspatch $(DESTDIR)/${PREFIX}/bin
+	${INSTALL_MAN} bsdiff.1 bspatch.1 $(DESTDIR)/${PREFIX}/man/man1
