diff -Naur chocolate-doom-1.6.0/configure.in chocolate-doom-1.6.0.patched/configure.in
--- chocolate-doom-1.6.0/configure.in	2011-05-17 19:01:50.000000000 -0400
+++ chocolate-doom-1.6.0.patched/configure.in	2012-09-20 01:32:26.000000000 -0400
@@ -103,7 +103,7 @@
 esac
 
 AM_CONDITIONAL(WINDOWS_CE, $WINDOWS_CE)
-AM_CONDITIONAL(HAVE_WINDRES, test "$WINDRES" != "")
+AM_CONDITIONAL(HAVE_WINDRES, false)
 AM_CONDITIONAL(HAVE_PYTHON, $HAVE_PYTHON)
 
 dnl Automake v1.8.0 is required, please upgrade!
