diff -Naur chocolate-doom-2.0.0/configure.ac chocolate-doom-2.0.0.patched/configure.ac
--- chocolate-doom-2.0.0/configure.ac	2013-12-09 00:40:53.000000000 -0500
+++ chocolate-doom-2.0.0.patched/configure.ac	2014-03-20 14:54:11.000000000 -0400
@@ -90,7 +90,7 @@
 AC_CHECK_TOOL(WINDRES, windres, )
 AC_CHECK_TOOL(STRIP, strip, )
 
-AM_CONDITIONAL(HAVE_WINDRES, test "$WINDRES" != "")
+AM_CONDITIONAL(HAVE_WINDRES, false)
 AM_CONDITIONAL(HAVE_PYTHON, $HAVE_PYTHON)
 
 dnl Automake v1.8.0 is required, please upgrade!
