--- setup.py	2017-06-17 09:13:43.000000000 -0600
+++ setup.py	2017-08-14 19:56:12.899404271 -0600
@@ -9,15 +9,14 @@
 	description='Integrated circuit reverse engineering research and development',
 	long_description = long_description,
 	package_dir={
-			'pr0ntools': 'lib/pr0ntools',
-			'pr0ntools.image': 'lib/pr0ntools/image',
-			'pr0ntools.jssim': 'lib/pr0ntools/jssim',
-			'pr0ntools.jssim.cif': 'lib/pr0ntools/jssim/cif',
-			'pr0ntools.jssim.files': 'lib/pr0ntools/jssim/files',
-			'pr0ntools.stitch': 'lib/pr0ntools/stitch',
-			'pr0ntools.stitch.pto': 'lib/pr0ntools/stitch/pto',
-			'pr0ntools.tile': 'lib/pr0ntools/tile',
-			'pr0ntools.util': 'lib/pr0ntools/util',
+			'pr0ntools': 'pr0ntools',
+			'pr0ntools.image': 'pr0ntools/image',
+			'pr0ntools.jssim': 'pr0ntools/jssim',
+			'pr0ntools.jssim.cif': 'pr0ntools/jssim/cif',
+			'pr0ntools.jssim.files': 'pr0ntools/jssim/files',
+			'pr0ntools.stitch': 'pr0ntools/stitch',
+			'pr0ntools.stitch.pto': 'pr0ntools/stitch/pto',
+			'pr0ntools.tile': 'pr0ntools/tile',
 			},
 	packages=[
 			'pr0ntools',
@@ -28,7 +27,6 @@
 			'pr0ntools.stitch',
 			'pr0ntools.stitch.pto',
 			'pr0ntools.tile',
-			'pr0ntools.util',
 			],
 	scripts=[
 			'stitch/pr0nhugin.py',
