--- wwwroot/cgi-bin/awstats.pl.orig	2016-07-21 17:42:26.196363651 +0700
+++ wwwroot/cgi-bin/awstats.pl	2016-07-21 17:44:55.882578219 +0700
@@ -2223,7 +2223,7 @@
 # Check lib files in common possible directories :
 # Windows and standard package:        		"$DIR/lib" (lib in same dir than awstats.pl)
 # Debian package:                    		"/usr/share/awstats/lib"
-	my @PossibleLibDir = ( "$DIR/lib", "/usr/share/awstats/lib" );
+	my @PossibleLibDir = ( "$DIR/lib", "/usr/share/awstats/lib", "@DOCROOT@/awstats/wwwroot/cgi-bin/lib" );
 	my %FilePath       = ();
 	my %DirAddedInINC  = ();
 	my @FileListToLoad = ();
@@ -3114,7 +3114,8 @@
 	my @PossiblePluginsDir = (
 		"$DIR/plugins",
 		"/usr/local/awstats/wwwroot/cgi-bin/plugins",
-		"/usr/share/awstats/plugins"
+		"/usr/share/awstats/plugins",
+                "@DOCROOT@/awstats/wwwroot/cgi-bin/plugins"
 	);
 	my %DirAddedInINC = ();
 
