diff -Nur awstats-7.4.orig/wwwroot/cgi-bin/awstats.pl awstats-7.4/wwwroot/cgi-bin/awstats.pl
--- awstats-7.4.orig/wwwroot/cgi-bin/awstats.pl	2015-07-14 07:52:31.000000000 -0500
+++ awstats-7.4/wwwroot/cgi-bin/awstats.pl	2016-02-17 16:51:41.341139338 -0600
@@ -9040,12 +9040,12 @@
 		$LogFormatString =~ s/%>s/%code/g;
 		$LogFormatString =~ s/%b(\s)/%bytesd$1/g;
 		$LogFormatString =~ s/%b$/%bytesd/g;
-		$LogFormatString =~ s/\"%{Referer}i\"/%refererquot/g;
-		$LogFormatString =~ s/\"%{User-Agent}i\"/%uaquot/g;
-		$LogFormatString =~ s/%{mod_gzip_input_size}n/%gzipin/g;
-		$LogFormatString =~ s/%{mod_gzip_output_size}n/%gzipout/g;
-		$LogFormatString =~ s/%{mod_gzip_compression_ratio}n/%gzipratio/g;
-		$LogFormatString =~ s/\(%{ratio}n\)/%deflateratio/g;
+		$LogFormatString =~ s/\"%\{Referer}i\"/%refererquot/g;
+		$LogFormatString =~ s/\"%\{User-Agent}i\"/%uaquot/g;
+		$LogFormatString =~ s/%\{mod_gzip_input_size}n/%gzipin/g;
+		$LogFormatString =~ s/%\{mod_gzip_output_size}n/%gzipout/g;
+		$LogFormatString =~ s/%\{mod_gzip_compression_ratio}n/%gzipratio/g;
+		$LogFormatString =~ s/\(%\{ratio}n\)/%deflateratio/g;
 
 		# Replacement for a IIS and ISA format string
 		$LogFormatString =~ s/cs-uri-query/%query/g;    # Must be before cs-uri
