diff -Nur id3v2-0.1.11.orig/Makefile id3v2-0.1.11/Makefile
--- id3v2-0.1.11.orig/Makefile	2004-05-04 13:33:53.000000000 -0500
+++ id3v2-0.1.11/Makefile	2009-07-29 00:36:39.438041437 -0500
@@ -2,8 +2,10 @@
 
 VERSION=0.1.11
 
+OPT=	-O2
+
 PREFIX=	/usr/local
-CXXFLAGS+=	-I${PREFIX}/include/ -DVERSION="\"${VERSION}\"" #-DSORT_RUNTIME
+CXXFLAGS+=	$(OPT) -I${PREFIX}/include/ -DVERSION="\"${VERSION}\"" #-DSORT_RUNTIME
 LDFLAGS+=	-L${PREFIX}/lib/ 
 
 id3v2:	convert.o list.o id3v2.o genre.o
