--- pyftpdlib-1.5.1/setup.orig	2016-07-09 15:38:35.715719570 +0300
+++ pyftpdlib-1.5.1/setup.py	2016-07-09 15:40:26.852330171 +0300
@@ -134,9 +134,7 @@
         except ImportError:
             msg = textwrap.dedent("""
                 'pysendfile' third-party module is not installed. This is not
-                essential but it considerably speeds up file transfers.
-                You can install it with 'pip install pysendfile'.
-                More at: https://github.com/giampaolo/pysendfile""")
+                essential but it considerably speeds up file transfers'.""")
             print(hilite(msg, ok=False), file=sys.stderr)
 
     try:
@@ -144,8 +142,7 @@
     except ImportError:
         msg = textwrap.dedent("""
             'pyopenssl' third-party module is not installed. This means
-            FTPS support will be disabled. You can install it with:
-            'pip install pyopenssl'.""")
+            FTPS support will be disabled'.""")
         print(hilite(msg, ok=False), file=sys.stderr)
 
 
