diff -Naur pdfarranger-1.8.2/data/com.github.jeromerobert.pdfarranger.desktop pdfarranger-1.8.2_icon/data/com.github.jeromerobert.pdfarranger.desktop
--- pdfarranger-1.8.2/data/com.github.jeromerobert.pdfarranger.desktop	2021-12-25 16:53:29.000000000 +0000
+++ pdfarranger-1.8.2_icon/data/com.github.jeromerobert.pdfarranger.desktop	2022-06-28 00:05:36.510916505 +0100
@@ -17,7 +17,7 @@
 Comment[ko]=PDF 합치기, 재정렬, 나누기, 회전 및 잘라내기
 Type=Application
 Exec=pdfarranger %U
-Icon=com.github.jeromerobert.pdfarranger
+Icon=pdfarranger
 MimeType=application/pdf;application/postscript;application/x-ext-pdf;image/x-bmp;image/bmp;image/gif;image/tiff;image/jp2;image/jpeg;image/jpg;image/png;image/x-icon;image/x-portable-pixmap;image/x-gray;image/x-ico;image/x-png;image/vnd.zbrush.pcx;image/x-pcx;
 Categories=Office;
 Terminal=false
diff -Naur pdfarranger-1.8.2/pdfarranger/pdfarranger.py pdfarranger-1.8.2_icon/pdfarranger/pdfarranger.py
--- pdfarranger-1.8.2/pdfarranger/pdfarranger.py	2021-12-25 16:53:29.000000000 +0000
+++ pdfarranger-1.8.2_icon/pdfarranger/pdfarranger.py	2022-06-28 00:04:05.856916851 +0100
@@ -54,7 +54,7 @@
     pass  # Gtk already prints a warning
 
 DOMAIN = 'pdfarranger'
-ICON_ID = 'com.github.jeromerobert.' + DOMAIN
+ICON_ID = DOMAIN
 if hasattr(locale, 'bindtextdomain'):
     # glibc
     locale.bindtextdomain(DOMAIN, localedir)
