diff -Naur kmldonkey-0.10.1.orig/src/scripts/submit.cpp kmldonkey-0.10.1/src/scripts/submit.cpp
--- kmldonkey-0.10.1.orig/src/scripts/submit.cpp	2005-07-26 18:40:40.000000000 +0200
+++ kmldonkey-0.10.1/src/scripts/submit.cpp	2007-09-02 23:36:00.000000000 +0200
@@ -48,7 +48,7 @@
 {
     kdDebug() << "Callback cmd: " << cmd << endl;
     kdDebug() << "Callback result: " << res << endl;
-    if (res.lower() != "done" && !res.contains("parsing http url", false)) {
+    if (!res.contains("Added link", false)) {
 	/*
 	if (res.contains("force_download")) {
 	    if (KMessageBox::warningYesNo(0, i18n("Failed to submit URL %1\n\nForce download?")
