commit eae955f0525724526fa602fd126623b9d84599fa
Author: orbea <orbea@riseup.net>
Date:   Thu Nov 5 22:08:24 2020 -0800

    cmake: Install cmake files to CMAKE_INSTALL_DATADIR

diff --git a/CMakeLists.txt b/CMakeLists.txt
index eb46947..6f01ef0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,7 +79,7 @@ target_include_directories(${PROJECT_NAME} INTERFACE
 if (SPIRV_HEADERS_ENABLE_INSTALL)
     message(STATUS "Installing SPIRV-Header")
 
-    set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+    set(config_install_dir "${CMAKE_INSTALL_DATADIR}/cmake/${PROJECT_NAME}")
 
     set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
 
