diff -Naur a/scripts/src-to-paps b/scripts/src-to-paps
--- a/scripts/src-to-paps	2026-10-03 17:11:44.000000000 -0400
+++ b/scripts/src-to-paps	2026-10-06 02:30:35.027172952 -0400
@@ -49,7 +49,7 @@
 # Defaults
 
 # TBD - Make this a configuration variable
-pango_outlang_path = '/usr/local/share/paps/pango_markup.outlang'
+pango_outlang_path = '/usr/share/paps/pango_markup.outlang'
 
 parser = argparse.ArgumentParser(description='Process a file')
 parser.add_argument('-o', '--output',
@@ -114,7 +114,11 @@
                   '.cpp' : 'C',
                   '.cxx' : 'C',
                   '.cc'  : 'C',
-                  '.c'   : 'C'
+                  '.c'   : 'C',
+                  '.pl'  : 'perl',
+                  '.SlackBuild'  : 'sh',
+                  '.cs'  : 'csharp',
+                  '.s'   : 'asm'
                   }.get(Path(fn).suffix)
                  if args.source_lang is None
                  else args.source_lang)
