diff -Naur paps-0.8.0/scripts/src-to-paps paps-0.8.0.patched/scripts/src-to-paps
--- paps-0.8.0/scripts/src-to-paps	2023-02-09 02:27:38.000000000 -0500
+++ paps-0.8.0.patched/scripts/src-to-paps	2023-05-03 16:53:01.568348300 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 ######################################################################
 #  Use GNU source-hightlight to turn source code into pango markup
@@ -49,7 +49,7 @@
 # Defaults
 
 # TBD - Make this a configuration variable
-pango_outlang_path = '/usr/local/share/paps/pango_markup.outlang'
+pango_outlang_path = '/usr/share/paps/pango_markup.outlang'
 
 parser = argparse.ArgumentParser(description='Process a file')
 parser.add_argument('-o', '--output',
@@ -114,7 +114,11 @@
                   '.cpp' : 'C',
                   '.cxx' : 'C',
                   '.cc'  : 'C',
-                  '.c'   : 'C'
+                  '.c'   : 'C',
+                  '.pl'  : 'perl',
+                  '.SlackBuild' : 'sh',
+                  '.cs'  : 'csharp',
+                  '.s' : 'asm'
                   }.get(Path(fn).suffix)
                  if args.source_lang is None
                  else args.source_lang)
