--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,6 +36,7 @@
 find_package(Qt5X11Extras ${QT_MINIMUM_VERSION} REQUIRED)
 find_package(Qt5Xml ${QT_MINIMUM_VERSION} REQUIRED)
 find_package(lxqt ${LXQT_MINIMUM_VERSION} REQUIRED)
+find_package(lxqt-menu-data 1.4.1 REQUIRED)
 
 include(LXQtPreventInSourceBuilds)
 include(LXQtCompilerSettings NO_POLICY_SCOPE)
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -65,7 +65,3 @@
     DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/applications"
     COMPONENT Runtime
 )
-install(FILES lxqt-config.menu
-    DESTINATION "${LXQT_ETC_XDG_DIR}/menus"
-    COMPONENT Runtime
-)
--- a/src/menuname/CMakeLists.txt
+++ b/src/menuname/CMakeLists.txt
@@ -8,9 +8,3 @@
 )
 add_custom_target(desktop_directories_files ALL DEPENDS ${DIRECTORY_FILES})
 #************************************************
-
-install(FILES
-    ${DIRECTORY_FILES}
-    DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/desktop-directories"
-    COMPONENT Runtime
-)
