diff -aur xmms2-0.6DrMattDestruction/src/include/xmms/wscript xmms2-0.6DrMattDestruction.e/src/include/xmms/wscript
--- xmms2-0.6DrMattDestruction/src/include/xmms/wscript	2009-04-21 19:51:11.000000000 +0200
+++ xmms2-0.6DrMattDestruction.e/src/include/xmms/wscript	2009-11-01 19:39:25.226520223 +0100
@@ -15,8 +15,7 @@
 
     defs = {}
 
-    defs['PKGLIBDIR'] = os.path.join(conf.env['PREFIX'],
-                                     'lib', 'xmms2')
+    defs['PKGLIBDIR'] = os.path.join(conf.env['LIBDIR'], 'xmms2')
     defs['BINDIR']    = conf.env['BINDIR']
     defs['SHAREDDIR'] = os.path.join(conf.env['PREFIX'],
                                      'share', 'xmms2')
diff -aur xmms2-0.6DrMattDestruction/wafadmin/Tools/ccroot.py xmms2-0.6DrMattDestruction.e/wafadmin/Tools/ccroot.py
--- xmms2-0.6DrMattDestruction/wafadmin/Tools/ccroot.py	2009-04-21 19:51:11.000000000 +0200
+++ xmms2-0.6DrMattDestruction.e/wafadmin/Tools/ccroot.py	2009-11-01 19:27:40.099784900 +0100
@@ -121,13 +121,13 @@
 @feature('cprogram', 'dprogram')
 @before('apply_core')
 def vars_target_cprogram(self):
-	self.default_install_path = '${PREFIX}/bin'
+	self.default_install_path = '${BINDIR}'
 	self.default_chmod = O755
 
 @feature('cstaticlib', 'dstaticlib', 'cshlib', 'dshlib')
 @before('apply_core')
 def vars_target_cstaticlib(self):
-	self.default_install_path = '${PREFIX}/lib'
+	self.default_install_path = '${LIBDIR}'
 	if sys.platform in ['win32', 'cygwin']:
 		# on win32, libraries need the execute bit, else we
 		# get 'permission denied' when using them (issue 283)
diff -aur xmms2-0.6DrMattDestruction/waftools/tool.py xmms2-0.6DrMattDestruction.e/waftools/tool.py
--- xmms2-0.6DrMattDestruction/waftools/tool.py	2009-04-21 19:51:11.000000000 +0200
+++ xmms2-0.6DrMattDestruction.e/waftools/tool.py	2009-11-01 19:39:25.227770032 +0100
@@ -6,5 +6,5 @@
 
     if env['explicit_install_name']:
         libname = obj.env["shlib_PATTERN"] % obj.target
-        insname = os.path.join(obj.env["PREFIX"], 'lib', libname)
+        insname = os.path.join(obj.env["LIBDIR"], libname)
         obj.env.append_unique("LINKFLAGS", '-install_name ' + insname)
diff -aur xmms2-0.6DrMattDestruction/wscript xmms2-0.6DrMattDestruction.e/wscript
--- xmms2-0.6DrMattDestruction/wscript	2009-04-21 19:51:11.000000000 +0200
+++ xmms2-0.6DrMattDestruction.e/wscript	2009-11-01 19:39:44.857519630 +0100
@@ -103,7 +103,7 @@
                 'LIB': lib,
              'PREFIX': bld.env['PREFIX'],
              'BINDIR': os.path.join("${prefix}", "bin"),
-             'LIBDIR': os.path.join("${prefix}", "lib"),
+             'LIBDIR': bld.env['LIBDIR'],
          'INCLUDEDIR': os.path.join("${prefix}", "include", "xmms2"),
             'VERSION': bld.env["VERSION"],
         }

