--- a/qwtpolarbuild.pri	2014-09-19 20:28:24.314002267 +1000
+++ b/qwtpolarbuild.pri	2015-09-25 16:41:20.445026649 +1000
@@ -13,7 +13,7 @@
 CONFIG           += qt
 CONFIG           += warn_on
 CONFIG           += no_keywords
-CONFIG           += silent
+#CONFIG           += silent
 
 CONFIG           += qwt
 
--- a/qwtpolarconfig.pri	2014-09-19 20:28:24.318002267 +1000
+++ b/qwtpolarconfig.pri	2015-09-25 16:45:43.466042709 +1000
@@ -16,23 +16,19 @@
 QWT_POLAR_VERSION      = $${QWT_POLAR_VER_MAJ}.$${QWT_POLAR_VER_MIN}.$${QWT_POLAR_VER_PAT}
 
 unix {
-    QWT_POLAR_INSTALL_PREFIX    = /usr/local/qwtpolar-$$QWT_POLAR_VERSION
+    QWT_POLAR_INSTALL_PREFIX    = /usr
 }
 
-win32 {
-    QWT_POLAR_INSTALL_PREFIX    = C:/QwtPolar-$$QWT_POLAR_VERSION
-}
-
-QWT_POLAR_INSTALL_DOCS      = $${QWT_POLAR_INSTALL_PREFIX}/doc
-QWT_POLAR_INSTALL_HEADERS   = $${QWT_POLAR_INSTALL_PREFIX}/include
-QWT_POLAR_INSTALL_LIBS      = $${QWT_POLAR_INSTALL_PREFIX}/lib
+QWT_POLAR_INSTALL_DOCS      = $${QWT_POLAR_INSTALL_PREFIX}/doc/qwtpolar-$${QWT_POLAR_VERSION}
+QWT_POLAR_INSTALL_HEADERS   = $${QWT_POLAR_INSTALL_PREFIX}/include/qwt
+QWT_POLAR_INSTALL_LIBS      = $${QWT_POLAR_INSTALL_PREFIX}/lib$${LIBDIRSUFFIX}
 
 ######################################################################
 # Designer plugin
 ######################################################################
 
-QWT_POLAR_INSTALL_PLUGINS   = $${QWT_POLAR_INSTALL_PREFIX}/plugins/designer
-# QWT_POLAR_INSTALL_PLUGINS   = $${QT_INSTALL_PREFIX}/plugins/designer
+#QWT_POLAR_INSTALL_PLUGINS   = $${QWT_POLAR_INSTALL_PREFIX}/plugins/designer
+QWT_POLAR_INSTALL_PLUGINS   = $$[QT_INSTALL_PLUGINS]/designer
 
 ######################################################################
 # Build the static/shared libraries.
@@ -61,8 +57,8 @@
 # with every Qt upgrade. 
 ######################################################################
 
-QWT_POLAR_INSTALL_FEATURES  = $${QWT_POLAR_INSTALL_PREFIX}/features
-# QWT_INSTALL_FEATURES  = $${QT_INSTALL_PREFIX}/features
+#QWT_POLAR_INSTALL_FEATURES  = $${QWT_POLAR_INSTALL_PREFIX}/features
+QWT_POLAR_INSTALL_FEATURES  = $$[QMAKE_MKSPECS]/features
 
 ######################################################################
 # If you want to build the QwtPolar designer plugin,
