--- ../src/src.pro.orig	2018-09-17 17:15:25.091000000 +1000
+++ ../src/src.pro	2018-09-17 17:27:50.284000000 +1000
@@ -14,7 +14,12 @@
 include( $${QWT_POLAR_ROOT}/qwtpolarfunctions.pri )
 
 TEMPLATE          = lib
-TARGET            = $$qwtPolarLibraryTarget(qwtpolar)
+greaterThan(QT_MAJOR_VERSION, 4) {
+    TARGET            = $$qwtPolarLibraryTarget(qwtpolar-qt5)
+}
+else {
+    TARGET            = $$qwtPolarLibraryTarget(qwtpolar)
+}
 
 # pkgconfig support
 CONFIG     += create_pc create_prl no_install_prl
--- ../designer/designer.pro.orig	2014-09-19 20:28:24.313002267 +1000
+++ ../designer/designer.pro	2018-09-17 17:32:26.175000000 +1000
@@ -60,7 +60,12 @@
         LIBS      += -L$${QWT_POLAR_ROOT}/lib
     }
 
-    qwtPolarAddLibrary(qwtpolar)
+    greaterThan(QT_MAJOR_VERSION, 4) {
+        qwtPolarAddLibrary(qwtpolar-qt5)
+    }
+    else {
+        qwtPolarAddLibrary(qwtpolar)
+    }
 
     contains(CONFIG, QwtPolarDll) {
 
