commit 223b5436294642c8d1e2150783f3703b28a93430
Author: Volker Fröhlich <volker27@gmx.at>
Date:   Tue Sep 23 21:06:37 2014 +0200

    Add chkconfig support

diff --git a/src/src.pro b/src/src.pro
index 86e6a57..d9c7567 100644
--- a/src/src.pro	2014-09-19 20:28:24.308002267 +1000
+++ b/src/src.pro	2015-10-05 10:29:40.869985742 +1000
@@ -16,6 +16,16 @@
 TEMPLATE          = lib
 TARGET            = $$qwtPolarLibraryTarget(qwtpolar)
 
+# pkgconfig support
+CONFIG     += create_pc create_prl no_install_prl
+QMAKE_PKGCONFIG_NAME = qwtpolar
+QMAKE_PKGCONFIG_DESCRIPTION = Qt/Qwt Polar Plot Widget
+QMAKE_PKGCONFIG_PREFIX = $${QWT_POLAR_INSTALL_PREFIX}
+QMAKE_PKGCONFIG_LIBDIR = $${QWT_POLAR_INSTALL_LIBS}
+QMAKE_PKGCONFIG_INCDIR = $${QWT_POLAR_INSTALL_HEADERS}
+QMAKE_PKGCONFIG_REQUIRES = qwt QtCore
+QMAKE_PKGCONFIG_DESTDIR = pkgconfig
+
 DESTDIR           = $${QWT_POLAR_ROOT}/lib
 
 contains(QWT_POLAR_CONFIG, QwtPolarDll ) {
