diff -u -r jfsw-orig/jfbuild/src/kplib.c jfsw/jfbuild/src/kplib.c
--- jfsw-orig/jfbuild/src/kplib.c       2016-02-20 06:31:24.000000000 +0000
+++ jfsw/jfbuild/src/kplib.c    2016-06-08 00:12:21.495980101 +0100
@@ -181,13 +181,13 @@
 static int gslidew = 0, gslider = 0, xm, xmn[4], xr0, xr1, xplc, yplc;
 static INT_PTR nfplace;
 static int clen[320], cclen[19], bitpos, filt, xsiz, ysiz;
-static int xsizbpl, ixsiz, ixoff, iyoff, ixstp, iystp, intlac, nbpl, trnsrgb ASMNAME("trnsrgb");
+static int xsizbpl, ixsiz, ixoff, iyoff, ixstp, iystp, intlac, nbpl, __attribute__((used)) trnsrgb ASMNAME("trnsrgb");
 static int ccind[19] = {16,17,18,0,8,7,9,6,10,5,11,4,12,3,13,2,14,1,15};
 static int hxbit[59][2], ibuf0[288], nbuf0[32], ibuf1[32], nbuf1[32];
 static const unsigned char *filptr;
 static unsigned char slidebuf[32768], opixbuf0[4], opixbuf1[4];
 static unsigned char pnginited = 0, olinbuf[65536] ASMNAME("olinbuf"); //WARNING:max xres is: 65536/bpp-1
-static int gotcmov = -2, abstab10[1024] ASMNAME("abstab10");
+static int gotcmov = -2, __attribute__((used)) abstab10[1024] ASMNAME("abstab10");
 
        //Variables to speed up dynamic Huffman decoding:
 #define LOGQHUFSIZ0 9
