diff -Naur jfsw-8fc2d549d4ad840021da40e9021c6a2b891f41c2/src/grpscan.c jfsw-8fc2d549d4ad840021da40e9021c6a2b891f41c2.patched/src/grpscan.c
--- jfsw-8fc2d549d4ad840021da40e9021c6a2b891f41c2/src/grpscan.c	2017-03-24 23:22:17.000000000 -0400
+++ jfsw-8fc2d549d4ad840021da40e9021c6a2b891f41c2.patched/src/grpscan.c	2020-04-28 20:48:40.091365360 -0400
@@ -34,6 +34,7 @@
 	{ "Registered Version",		0x7545319F, 47536148, NULL },
 	{ "Shareware Version",		0x08A7FA1F, 26056769, NULL },
 	{ "Wanton Destruction (Addon)", 0xA9AAA7B7, 48698128, NULL },
+	{ "Twin Dragon (Addon)", 0xe06cb6b4, 53480783 },
 };
 struct grpfile *foundgrps = NULL;
 
diff -Naur jfsw-8fc2d549d4ad840021da40e9021c6a2b891f41c2/src/grpscan.h jfsw-8fc2d549d4ad840021da40e9021c6a2b891f41c2.patched/src/grpscan.h
--- jfsw-8fc2d549d4ad840021da40e9021c6a2b891f41c2/src/grpscan.h	2017-03-24 23:22:17.000000000 -0400
+++ jfsw-8fc2d549d4ad840021da40e9021c6a2b891f41c2.patched/src/grpscan.h	2020-04-28 20:52:36.787343843 -0400
@@ -25,7 +25,7 @@
 #define __grpscan_h__
 
 // List of internally-known GRP files
-#define numgrpfiles 3
+#define numgrpfiles 4
 struct grpfile {
 	const char *name;
 	int crcval;
