--- libmirage-1.3.0_orig/src/parsers/image-iso/image-iso-parser.c	2010-05-28 14:43:59.000000000 -0300
+++ libmirage-1.3.0/src/parsers/image-iso/image-iso-parser.c	2010-10-31 09:36:45.000000000 -0200
@@ -312,7 +312,7 @@
         "PARSER-ISO",
         "ISO Image Parser",
         "ISO images",
-        "application/libmirage-iso"
+        "application/x-cd-image"
     );
     
     return;
--- libmirage-1.3.0_orig/src/parsers/image-toc/image-toc-parser.c	2010-05-28 14:43:58.000000000 -0300
+++ libmirage-1.3.0/src/parsers/image-toc/image-toc-parser.c	2010-10-31 09:56:38.000000000 -0200
@@ -1337,7 +1337,7 @@
         "PARSER-TOC",
         "TOC Image Parser",
         "Cdrdao TOC files",
-        "application/libmirage-toc"
+        "application/x-cdrdao-toc"
     );
     
     __mirage_parser_toc_init_regex_parser(MIRAGE_PARSER(instance));
--- libmirage-1.3.0_orig/src/parsers/image-cue/image-cue-parser.c	2010-05-28 14:43:55.000000000 -0300
+++ libmirage-1.3.0/src/parsers/image-cue/image-cue-parser.c	2010-10-31 09:58:34.000000000 -0200
@@ -1030,7 +1030,7 @@
         "PARSER-CUE",
         "CUE Image Parser",
         "CUE images",
-        "application/libmirage-cue"
+        "application/x-cue"
     );
     
     __mirage_parser_cue_init_regex_parser(MIRAGE_PARSER(instance));
