diff -Nur lxdm-0.2.0.orig//data/Xsession lxdm-0.2.0/data/Xsession
--- lxdm-0.2.0.orig//data/Xsession	2010-04-12 11:49:44.000000000 -0500
+++ lxdm-0.2.0/data/Xsession	2010-06-26 04:55:43.301983179 -0500
@@ -2,6 +2,13 @@
 
 # use bash for "exec -l", howto run login shell by /bin/sh ?
 
+# First read /etc/profile and .profile
+test -f /etc/profile && . /etc/profile
+test -f "$HOME/.profile" && . "$HOME/.profile"
+# Second read /etc/xprofile and .xprofile for X specific setup
+test -f /etc/xprofile && . /etc/xprofile
+test -f "$HOME/.xprofile" && . "$HOME/.xprofile"
+
 if [ $# -eq 1 -a -n "$1" ]; then
 	LXSESSION=$1
 else
