Index: rss-glx-0.9.1/src/Makefile.am
===================================================================
--- rss-glx-0.9.1.orig/src/Makefile.am
+++ rss-glx-0.9.1/src/Makefile.am
@@ -38,7 +38,9 @@ matrixview_CFLAGS = @Wand_CFLAGS@
 matrixview_LDADD = @BZIP2@ @PTHREAD_LIBS@ @Wand_LIBS@
 plasma_SOURCES = driver.c plasma.cpp rsDefines.h rsRand.h
 pixelcity_SOURCES = driver.c
-pixelcity_LDADD = @quesoglc_LIBS@ -LPixelCity -lPixelCity
+pixelcity_LDADD = -LPixelCity -lPixelCity @quesoglc_LIBS@
+# Dummy file, not built.  Forces g++ linking
+nodist_EXTRA_pixelcity_SOURCES = dummy.cpp
 skyrocket_SOURCES = driver.c skyrocket.cpp skyrocket_flare.cpp skyrocket_particle.cpp skyrocket_shockwave.cpp skyrocket_smoke.cpp skyrocket_sound.cpp skyrocket_world.cpp skyrocket_textures.cpp rsDefines.h rsRand.h
 solarwinds_SOURCES = driver.c solarwinds.cpp rsDefines.h rsRand.h
 spirographx_SOURCES = driver.c spirographx.c
