diff -Naur pdfshuffler-0.6.0/pdfshuffler/pdfshuffler.py pdfshuffler-0.6.0-patched/pdfshuffler/pdfshuffler.py
--- pdfshuffler-0.6.0/pdfshuffler/pdfshuffler.py	2012-04-28 15:15:17.000000000 +0100
+++ pdfshuffler-0.6.0-patched/pdfshuffler/pdfshuffler.py	2015-11-24 16:27:34.381182962 +0000
@@ -71,7 +71,7 @@
 import cairo
 
 import poppler      #for the rendering of pdf pages
-from pyPdf import PdfFileWriter, PdfFileReader
+from PyPDF2 import PdfFileWriter, PdfFileReader
 
 from pdfshuffler_iconview import CellRendererImage
 gobject.type_register(CellRendererImage)
