diff -Naur tint-0.07/tint.c tint-0.07.patched/tint.c
--- tint-0.07/tint.c	2022-02-06 11:57:02.000000000 -0500
+++ tint-0.07.patched/tint.c	2024-10-08 21:48:52.809771728 -0400
@@ -607,14 +607,18 @@
 			 switch (ch)
 			   {
 				case 'j':
+				case 'J':
 				case KEY_LEFT:
 				  engine_move (&engine,ACTION_LEFT);
 				  break;
 				case 'k':
+				case 'K':
 				case '\n':
+				case KEY_UP:
 				  engine_move (&engine,ACTION_ROTATE);
 				  break;
 				case 'l':
+				case 'L':
 				case KEY_RIGHT:
 				  engine_move (&engine,ACTION_RIGHT);
 				  break;
@@ -624,15 +628,17 @@
 				  break;
 				  /* show next piece */
 				case 's':
+				case 'S':
 				  shownext = TRUE;
 				  break;
 				  /* toggle dotted lines */
 				case 'd':
+				case 'D':
 				  dottedlines = !dottedlines;
 				  break;
 				  /* next level */
 				case 'a':
-				case KEY_UP:
+				case 'A':
 				  if (level < MAXLEVEL)
 					{
 					   level++;
@@ -642,10 +648,12 @@
 				  break;
 				  /* quit */
 				case 'q':
+				case 'Q':
 				  finished = TRUE;
 				  break;
 				  /* pause */
 				case 'p':
+				case 'P':
 				  out_setcolor (COLOR_WHITE,COLOR_BLACK);
 				  out_gotoxy ((out_width () - 34) / 2,out_height () - 2);
 				  out_printf ("Paused - Press any key to continue");
