--- rxtx-2.2pre2_original/Makefile.in	2008-12-31 02:29:12.000000000 +0000
+++ rxtx-2.2pre2_patched/Makefile.in	2009-12-21 12:34:37.099017471 +0000
@@ -619,13 +619,15 @@
 #  install librxtxSerial.so into the proper directory and copy $(JARTARGET) to its
 #  proper location
 install: all
+	$(INSTALL_PROGRAM) -d $(DESTDIR)$(RXTX_PATH)
 	@$(LIBTOOL_INST) \
 	`for i in $(TARGETLIB);do \
 		if [ -f $$i ];then \
 			echo $$i; \
 		fi; \
-	done` $(RXTX_PATH)
-	$(INSTALL_PROGRAM) $(JARTARGET) $(JHOME)/
+	done` $(DESTDIR)$(RXTX_PATH)
+	$(INSTALL_PROGRAM) -d $(DESTDIR)$(JHOME)
+	$(INSTALL_DATA) $(JARTARGET) $(DESTDIR)$(JHOME)/
 
 clean-generic:
 	rm -rf $(DEST) $(TOP)/$(CLASSTOP)
