diff --git a/backend/libinput/switch.c b/backend/libinput/switch.c
index abeec86d..f0a8dece 100644
--- a/backend/libinput/switch.c
+++ b/backend/libinput/switch.c
@@ -36,6 +36,8 @@ void handle_switch_toggle(struct libinput_event *event,
 	case LIBINPUT_SWITCH_TABLET_MODE:
 		wlr_event.switch_type = WLR_SWITCH_TYPE_TABLET_MODE;
 		break;
+	case LIBINPUT_SWITCH_KEYPAD_SLIDE:
+		break;
 	}
 	switch (libinput_event_switch_get_switch_state(sevent)) {
 	case LIBINPUT_SWITCH_STATE_OFF:
