diff -Naur XDecorations-0.1.7.orig/xdecorations.cpp XDecorations-0.1.7/xdecorations.cpp
--- XDecorations-0.1.7.orig/xdecorations.cpp	2014-12-21 16:32:53.000000000 +0100
+++ XDecorations-0.1.7/xdecorations.cpp	2022-02-08 10:43:41.415924000 +0100
@@ -1,4 +1,4 @@
-#if 0
+/*
 ©Keith Hedger Mon 20 Oct 14:21:11 BST 2014 kdhedger68713@gmail.com
 *   This program is free software;
 you can redistribute it and/or modify
@@ -19,7 +19,7 @@
 *   Foundation,Inc.,59 Temple Place,Suite 330,Boston,MA  02111-1307  USA
 
 *   toon_root.c Copyright (C) 1999-2001  Robin Hogan
-#endif
+*/
 
 #include <X11/Intrinsic.h>
 #include <X11/Xlib.h>
