--- Makefile-	2014-12-21 20:31:41.775025980 +0200
+++ Makefile	2014-12-21 20:33:13.430143913 +0200
@@ -12,10 +12,10 @@
 
 install:
 	g++ -o $(PROGRAM) -DDATADIR=\"${INSTALLDATADIR}\" $(SRC) $(LIBS)
-	mkdir -vp $(DESTDIR)$(PREFIX)/share/$(PROGRAM)/pixmaps $(DESTDIR)$(PREFIX)/bin $(DESTDIR)$(PREFIX)/share/doc/$(APPNAME) ||true
+	mkdir -vp $(DESTDIR)$(PREFIX)/share/$(PROGRAM)/pixmaps $(DESTDIR)$(PREFIX)/bin $(DESTDIR)$(PREFIX)/doc/$(APPNAME) ||true
 	cp -r pixmaps/* $(DESTDIR)$(PREFIX)/share/$(PROGRAM)/pixmaps||true
 	cp $(PROGRAM) $(DESTDIR)$(PREFIX)/bin||true
-	cp INSTALL xdecorations.rc $(DESTDIR)$(PREFIX)/share/doc/$(APPNAME) ||true
+	cp INSTALL xdecorations.rc $(DESTDIR)$(PREFIX)/doc/$(APPNAME) ||true
 
 uninstall:
 	rm -r $(DESTDIR)$(PREFIX)/bin/$(PROGRAM) $(DESTDIR)$(PREFIX)/share/$(PROGRAM)
@@ -27,4 +27,5 @@
 	cp $(PROGRAM).rc ~/.config
 
 makeball:
-	gcc -o ball double.c  -L/usr/lib64 -lXext -lX11 -lm -lXpm
\ No newline at end of file
+	gcc -o ball double.c  -L/usr/lib64 -lXext -lX11 -lm -lXpm
+
