diff -urN Transmission-0.6.1/mk/common.mk Transmission-0.6.1-patched/mk/common.mk
--- Transmission-0.6.1/mk/common.mk	2006-06-26 00:34:31.000000000 +0300
+++ Transmission-0.6.1-patched/mk/common.mk	2007-03-30 16:00:50.000000000 +0300
@@ -62,20 +62,20 @@
 
 define INSTALL_BIN_RULE
        @echo "Install $<"
-       @$(MKDIR) $(PREFIX)/bin
-       @$(CP) $< $(PREFIX)/bin/
+       @$(MKDIR) $(DESTDIR)$(PREFIX)/bin
+       @$(CP) $< $(DESTDIR)$(PREFIX)/bin/
 endef
 
 define INSTALL_LOCALE_RULE
        @echo "Install $<"
-       @$(MKDIR) $(LOCALEDIR)/$*/LC_MESSAGES
-       @$(CP) $< $(LOCALEDIR)/$*/LC_MESSAGES/transmission-gtk.mo
+       @$(MKDIR) $(DESTDIR)$(LOCALEDIR)/$*/LC_MESSAGES
+       @$(CP) $< $(DESTDIR)$(LOCALEDIR)/$*/LC_MESSAGES/transmission-gtk.mo
 endef
 
 define INSTALL_MAN_RULE
 	@echo "Install $<"
-	@$(MKDIR) $(PREFIX)/man/man1
-	@$(CP) $< $(PREFIX)/man/man1/
+	@$(MKDIR) $(DESTDIR)$(PREFIX)/man/man1
+	@$(CP) $< $(DESTDIR)$(PREFIX)/man/man1/
 endef
 
 RM    = rm -Rf
