diff --ignore-space-change --unified --recursive --no-dereference a/nvidia-drm/nvidia-drm-drv.c b/nvidia-drm/nvidia-drm-drv.c
--- a/nvidia-drm/nvidia-drm-drv.c  
+++ b/nvidia-drm/nvidia-drm-drv.c  
@@ -700,8 +700,11 @@
 
     .ioctls                 = nv_drm_ioctls,
     .num_ioctls             = ARRAY_SIZE(nv_drm_ioctls),
-
+// Rel. commit "drm/prime: Unexport helpers for fd/handle conversion" (Thomas Zimmermann, 20 Jun 2023)
+// This function is no longer exported, but leaving it to NULL is equivalent
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 6, 0)
     .prime_handle_to_fd     = drm_gem_prime_handle_to_fd,
+#endif
 
 #if defined(NV_DRM_DRIVER_HAS_GEM_PRIME_CALLBACKS)
     .gem_prime_export       = nv_drm_gem_prime_export,
