diff --git a/kernel/nvidia-uvm/uvm8_pmm_gpu.c b/kernel/nvidia-uvm/uvm8_pmm_gpu.c
index 602a286..d6eb597 100644
--- a/kernel/nvidia-uvm/uvm8_pmm_gpu.c
+++ b/kernel/nvidia-uvm/uvm8_pmm_gpu.c
@@ -2794,7 +2794,12 @@ NV_STATUS init_chunk_split_cache_level(uvm_pmm_gpu_t *pmm, size_t level)
             size_t size;
             size_t align;
             if (level == 0) {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 2, 0)
+                // Rel. commit "string: Remove strncpy() from the kernel" (Kees Cook, 18 Jun 2026)
+                strscpy(chunk_split_cache[level].name, "uvm_gpu_chunk_t", sizeof(chunk_split_cache[level].name));
+#else
                 strncpy(chunk_split_cache[level].name, "uvm_gpu_chunk_t", sizeof(chunk_split_cache[level].name) - 1);
+#endif
                 size = sizeof(uvm_gpu_chunk_t);
                 align = __alignof__(uvm_gpu_chunk_t);
             } else {
