--- certbot-0.21.0/certbot-apache/certbot_apache/configurator.py.old	2018-01-19 05:16:48.965843221 +0700
+++ certbot-0.21.0/certbot-apache/certbot_apache/configurator.py	2018-01-19 05:35:24.916445040 +0700
@@ -89,20 +89,20 @@
     description = "Apache Web Server plugin - Beta"
 
     OS_DEFAULTS = dict(
-        server_root="/etc/apache2",
-        vhost_root="/etc/apache2/sites-available",
-        vhost_files="*",
-        logs_root="/var/log/apache2",
-        version_cmd=['apache2ctl', '-v'],
-        apache_cmd="apache2ctl",
-        restart_cmd=['apache2ctl', 'graceful'],
-        conftest_cmd=['apache2ctl', 'configtest'],
+        server_root="/etc/httpd",
+        vhost_root="/etc/httpd/extra",
+        vhost_files="httpd-vhosts.conf",
+        logs_root="/var/log/httpd",
+        version_cmd=['apachectl', '-v'],
+        apache_cmd="apachectl",
+        restart_cmd=['apachectl', 'graceful'],
+        conftest_cmd=['apachectl', 'configtest'],
         enmod=None,
         dismod=None,
-        le_vhost_ext="-le-ssl.conf",
+        le_vhost_ext="",
         handle_mods=False,
         handle_sites=False,
-        challenge_location="/etc/apache2",
+        challenge_location="/etc/httpd",
         MOD_SSL_CONF_SRC=pkg_resources.resource_filename(
             "certbot_apache", "options-ssl-apache.conf")
     )
