diff --git a/antiwordxp.rb b/antiwordxp.rb
index 70e1689..2590b11 100755
--- a/antiwordxp.rb
+++ b/antiwordxp.rb
@@ -312,8 +312,8 @@ if(process_xml)
 	
 	#Some UTF-8 characters don't print
 	#This translates from utf-8 to ascii
-	require "iconv"
-	document = Iconv.conv("ascii//translit", "UTF-8", document)	
+#	require "iconv"
+#	document = Iconv.conv("ascii//translit", "UTF-8", document)	
 end
 
 begin
