diff -Naur grip-2.93.orig/grip.c grip-2.93/grip.c
--- grip-2.93.orig/grip.c	2007-04-19 01:15:23.000000000 -0400
+++ grip-2.93/grip.c	2007-04-19 01:34:39.000000000 -0400
@@ -1637,7 +1637,7 @@
 #if defined(SOLARIS) || defined(__FreeBSD__)
   pthread_exit(&status);
 #else
-    pthread_kill_other_threads_np();
+    /* pthread_kill_other_threads_np(); */
 #endif
     Debug("Aborted\n");
     looking_up=FALSE;
