--- b/release.makefile	2023-04-13 18:46:36.368966456 +0200
+++ a/release.makefile	2023-04-13 18:55:38.068985747 +0200
@@ -1,6 +1,6 @@
 
 BUILD_DIR := build
-INSTALL_DIR := ${PREFIX}/opt/stremio
+INSTALL_DIR := ${PREFIX}/usr/share/stremio
 
 ICON_BIN := smartcode-stremio.svg
 
@@ -8,12 +8,13 @@
 
 STREMIO_BIN := ${BUILD_DIR}/stremio
 
-ALL: ${STREMIO_BIN} ${SERVER_JS} icons
+ALL: ${STREMIO_BIN} icons
 
 install:
 	make -C ${BUILD_DIR} install
 	install -Dm 644 ${SERVER_JS} "${INSTALL_DIR}/server.js"
-	install -Dm 644 smartcode-stremio.desktop "${INSTALL_DIR}/smartcode-stremio.desktop"
+	install -Dm 644 stremio.asar "${INSTALL_DIR}/stremio.asar"
+	install -Dm 644 smartcode-stremio.desktop "${INSTALL_DIR}/../applications/com.stremio.stremio.desktop"
 	cp -r icons "${INSTALL_DIR}/"
-	ln -s "${shell which node}" "${INSTALL_DIR}/node"
+
 ifneq ("$(wildcard ../mpv-build/mpv/build)","")
@@ -28,14 +29,11 @@
 	mkdir -p "$@"
 	cd "$@" && printf 16,22,24,32,64,128 | xargs -I^ -d, sh -c 'rsvg-convert ../images/stremio.svg -w ^ -o smartcode-stremio_^.png && rsvg-convert ../images/stremio_tray_white.svg -w ^ -o smartcode-stremio-tray_^.png'
 
-${SERVER_JS}: 
-	wget "${shell cat server-url.txt}" -qO ${SERVER_JS} || rm ${SERVER_JS}
-
 ${STREMIO_BIN}:
 	mkdir -p ${BUILD_DIR}
 	cd ${BUILD_DIR} && cmake -G"Unix Makefiles" -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX="${PREFIX}" ..
 	make -j -C ${BUILD_DIR}
 
 clean:
-	rm -rf ${BUILD_DIR} ${SERVER_JS} icons
+	rm -rf ${BUILD_DIR} icons
 
