diff -Nur pgrouting-1.05/CMakeLists.txt pgrouting-1.05-flags/CMakeLists.txt
--- pgrouting-1.05/CMakeLists.txt	2010-11-16 18:42:21.000000000 -0800
+++ pgrouting-1.05-flags/CMakeLists.txt	2013-03-06 22:05:59.934147381 -0800
@@ -85,8 +85,8 @@
   LINK_LIBRARIES(postgres)
 ENDIF(WIN32)
 
-SET(CMAKE_C_FLAGS "-O2 -g -frounding-math")
-SET(CMAKE_CXX_FLAGS "-O2 -g -frounding-math")
+SET(CMAKE_C_FLAGS "${SLKCFLAGS} -frounding-math")
+SET(CMAKE_CXX_FLAGS "${SLKCFLAGS} -frounding-math")
 
 # Recurse into the subdirectories.  This does not actually
 # cause another cmake executable to run.  The same process will walk through
