diff -Nur pgrouting-1.05/core/src/astar_boost_wrapper.cpp pgrouting-1.05-weightmap/core/src/astar_boost_wrapper.cpp
--- pgrouting-1.05/core/src/astar_boost_wrapper.cpp	2010-11-16 18:42:21.000000000 -0800
+++ pgrouting-1.05-weightmap/core/src/astar_boost_wrapper.cpp	2013-03-06 22:06:43.438193285 -0800
@@ -139,9 +139,6 @@
 
   graph_t graph(num_nodes);
 
-  property_map<graph_t, edge_weight_t>::type weightmap = get(edge_weight, 
-							     graph);
-
   for (std::size_t j = 0; j < count; ++j)
   {
 
diff -Nur pgrouting-1.05/core/src/boost_wrapper.cpp pgrouting-1.05-weightmap/core/src/boost_wrapper.cpp
--- pgrouting-1.05/core/src/boost_wrapper.cpp	2010-11-16 18:42:21.000000000 -0800
+++ pgrouting-1.05-weightmap/core/src/boost_wrapper.cpp	2013-03-06 22:06:54.689205167 -0800
@@ -73,8 +73,6 @@
 
     graph_t graph(num_nodes);
 
-    property_map<graph_t, edge_weight_t>::type weightmap = get(edge_weight, graph);
-
     for (std::size_t j = 0; j < count; ++j)
     {
 	edge_descriptor e; bool inserted;
diff -Nur pgrouting-1.05/extra/driving_distance/src/boost_drivedist.cpp pgrouting-1.05-weightmap/extra/driving_distance/src/boost_drivedist.cpp
--- pgrouting-1.05/extra/driving_distance/src/boost_drivedist.cpp	2010-11-16 18:42:21.000000000 -0800
+++ pgrouting-1.05-weightmap/extra/driving_distance/src/boost_drivedist.cpp	2013-03-06 22:07:23.983236104 -0800
@@ -84,9 +84,6 @@
   
   graph_t graph( num_nodes );
   
-  property_map<graph_t, edge_weight_t>::type weightmap = 
-    get(edge_weight, graph);
-  
   for (std::size_t j = 0; j < count; ++j)
     {
       graph_add_edge<graph_t, edge_descriptor>
