diff -Nur pgrouting-1.05/core/sql/routing_core.sql pgrouting-1.05-sql/core/sql/routing_core.sql
--- pgrouting-1.05/core/sql/routing_core.sql	2010-11-16 18:42:21.000000000 -0800
+++ pgrouting-1.05-sql/core/sql/routing_core.sql	2013-03-06 23:07:39.797615201 -0800
@@ -14,7 +14,7 @@
 --
 -- You should have received a copy of the GNU General Public License
 -- along with this program; if not, write to the Free Software
--- Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+-- Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 --
 
 
@@ -29,7 +29,7 @@
         target_id integer, directed boolean, has_reverse_cost boolean)
         RETURNS SETOF path_result
         AS '$libdir/librouting'
-        LANGUAGE 'C' IMMUTABLE STRICT;
+        LANGUAGE c IMMUTABLE STRICT;
 
 -----------------------------------------------------------------------
 -- Core function for shortest_path_astar computation
@@ -40,7 +40,7 @@
         target_id integer,directed boolean, has_reverse_cost boolean)
          RETURNS SETOF path_result
          AS '$libdir/librouting'
-         LANGUAGE 'C' IMMUTABLE STRICT; 
+         LANGUAGE c IMMUTABLE STRICT; 
 
 -----------------------------------------------------------------------
 -- Core function for shortest_path_astar computation
@@ -50,7 +50,7 @@
         target_id integer,directed boolean, has_reverse_cost boolean)
          RETURNS SETOF path_result
          AS '$libdir/librouting'
-         LANGUAGE 'C' IMMUTABLE STRICT; 
+         LANGUAGE c IMMUTABLE STRICT; 
 
 -----------------------------------------------------------------------
 -- This function should not be used directly. Use create_graph_tables instead
