--- hw-probe.pl.orig	2026-05-04 09:44:33.102884711 +0100
+++ hw-probe.pl	2026-05-04 09:45:01.126315978 +0100
@@ -15945,16 +15945,6 @@ sub writeLogs()
     
     if(enabledLog("pkglist"))
     {
-        if(checkCmd("pacman"))
-        { # Arch / Manjaro
-            listProbe("logs", "pkglist");
-            my $Pkglist = runCmd("pacman -Q 2>/dev/null");
-            
-            if($Pkglist) {
-                writeLog($LOG_DIR."/pkglist", $Pkglist);
-            }
-        }
-        
         if($Sys{"System"}=~/solus/i and checkCmd("eopkg"))
         {
             listProbe("logs", "pkglist");
