--- a/src/BFiles.cpp	2009-11-13 06:42:20.000000000 +0300
+++ b/src/BFiles.cpp	2009-12-09 21:08:16.000000000 +0300
@@ -294,6 +294,8 @@
 	while ((ByteCount <= FileSize)&&(Type == kUnknown)) {
 		if (LocateKeyWord("GAMESS VERSION", 14, -1))
 			Type = kGAMESSlogType;
+		else if (LocateKeyWord("Firefly (PC GAMESS) version", 27, -1))
+			Type = kGAMESSlogType;
 		else if (LocateKeyWord("===== IRC DATA PACKET", 21, -1))
 			Type = kGAMESSTRJType;
 		else if (LocateKeyWord("===== DRC DATA PACKET", 21, -1))
