--- plugins/FakeMiniaturesPlugin/src/DkFakeMiniaturesDialog.h.orig	2018-06-05 18:21:41.000000000 +1000
+++ plugins/FakeMiniaturesPlugin/src/DkFakeMiniaturesDialog.h	2018-11-28 23:01:53.711000000 +1000
@@ -44,8 +44,9 @@
 	#pragma warning(disable: 4996)
 #endif
 
-#include "opencv2/core/core.hpp"
-#include "opencv2/imgproc/imgproc.hpp"
+#include "opencv4/opencv2/core/core.hpp"
+#include "opencv4/opencv2/imgproc/imgproc.hpp"
+#include "opencv4/opencv2/imgproc/types_c.h"
 
 using namespace cv;
 #endif
@@ -263,4 +264,4 @@
 		~DkSaturation();
 };
 
-};
\ No newline at end of file
+};
--- plugins/PageExtractionPlugin/src/DkPageSegmentationUtils.h.orig	2018-11-28 23:15:54.600000000 +1000
+++ plugins/PageExtractionPlugin/src/DkPageSegmentationUtils.h	2018-11-28 23:20:57.093000000 +1000
@@ -27,7 +27,8 @@
 #include "DkMath.h"
 
 #pragma warning(push, 0)	// no warnings from includes - begin
-#include <opencv2/core/core.hpp>
+#include <opencv4/opencv2/core/core.hpp>
+#include <opencv4/opencv2/imgproc/imgproc_c.h>
 #include <QString>
 #pragma warning(pop)		// no warnings from includes - end
 
