diff -Naur glestae-src-0.3.93.orig/source/shared_lib/sources/physfs/MemFileOps.cpp glestae-src-0.3.93/source/shared_lib/sources/physfs/MemFileOps.cpp
--- glestae-src-0.3.93.orig/source/shared_lib/sources/physfs/MemFileOps.cpp	2011-10-05 17:48:30.000000000 +0000
+++ glestae-src-0.3.93/source/shared_lib/sources/physfs/MemFileOps.cpp	2012-07-04 03:02:22.835956987 +0000
@@ -16,6 +16,7 @@
 #include <stdexcept>
 #include "leak_dumper.h"
 #include <string.h>
+#include <stdio.h> // for SEEK_SET, SEEK_...
 
 namespace Shared { namespace PhysFS {
 
