diff -Naur glestae-src-0.3.93.orig/source/shared_lib/sources/graphics/pixmap.cpp glestae-src-0.3.93/source/shared_lib/sources/graphics/pixmap.cpp
--- glestae-src-0.3.93.orig/source/shared_lib/sources/graphics/pixmap.cpp	2011-10-05 17:48:29.000000000 +0000
+++ glestae-src-0.3.93/source/shared_lib/sources/graphics/pixmap.cpp	2012-07-04 02:54:14.252629579 +0000
@@ -12,6 +12,10 @@
 #include "pch.h"
 #include "pixmap.h"
 
+#ifndef Z_DEFAULT_COMPRESSION
+#   include "zlib.h"
+#endif
+
 #include <stdexcept>
 #include <cstdio>
 #include <cassert>
