--- 2011_0719_RT3070_RT3370_RT5370_RT5372_Linux_STA_V2.5.0.3_DPO.orig/os/linux/Makefile.6.netif	2011-07-19 10:56:36.000000000 +0200
+++ 2011_0719_RT3070_RT3370_RT5370_RT5372_Linux_STA_V2.5.0.3_DPO/os/linux/Makefile.6.netif	2012-06-20 14:41:11.367409249 +0200
@@ -371,9 +371,8 @@
 endif
 
 install:
-	install -d $(LINUX_SRC_MODULE)
-	install -m 644 -c $(addsuffix .ko,$(MOD_NAME)) $(LINUX_SRC_MODULE)
-	/sbin/depmod -a ${shell uname -r}
+	install -d $(DESTDIR)/$(LINUX_SRC_MODULE)
+	install -m 644 -c $(addsuffix .ko,$(MOD_NAME)) $(DESTDIR)/$(LINUX_SRC_MODULE)
 
 uninstall:
 	rm -rf $(addprefix $(LINUX_SRC_MODULE),$(addsuffix .ko,$(MOD_NAME)))
