diff -Naur xmms_ladspa-1.1/ladspa.c xmms_ladspa-1.1.patched/ladspa.c
--- xmms_ladspa-1.1/ladspa.c	2005-03-15 21:37:11.000000000 -0500
+++ xmms_ladspa-1.1.patched/ladspa.c	2017-03-15 00:28:15.262723243 -0400
@@ -500,6 +500,7 @@
   if (dir == NULL) return;
 
   while ((dirent= readdir(dir))) {
+    if(strstr(dirent->d_name, "guitarix") != NULL) continue;
     snprintf(lib_name, PATH_MAX, "%s/%s", path_entry, dirent->d_name);
     library = dlopen(lib_name, RTLD_LAZY);
     if (library == NULL) {
