diff -ruN ncmpc-0.54.orig/src/net/IPv4Address.hxx ncmpc-0.54/src/net/IPv4Address.hxx
--- ncmpc-0.54.orig/src/net/IPv4Address.hxx	2026-08-13 17:55:41.000000000 +0200
+++ ncmpc-0.54/src/net/IPv4Address.hxx	2026-09-23 19:58:44.715766999 +0200
@@ -217,7 +217,7 @@
 	 * much for anything else.  Returns nullptr if the address is
 	 * not supported.
 	 */
-	constexpr std::span<const std::byte> GetSteadyPart() const noexcept {
+	std::span<const std::byte> GetSteadyPart() const noexcept {
 		return {
 			reinterpret_cast<const std::byte *>(&address.sin_addr),
 			sizeof(address.sin_addr),
diff -ruN ncmpc-0.54.orig/src/net/IPv6Address.hxx ncmpc-0.54/src/net/IPv6Address.hxx
--- ncmpc-0.54.orig/src/net/IPv6Address.hxx	2026-08-13 17:55:41.000000000 +0200
+++ ncmpc-0.54/src/net/IPv6Address.hxx	2026-09-23 19:58:56.207041843 +0200
@@ -176,7 +176,7 @@
 	 * much for anything else.  Returns nullptr if the address is
 	 * not supported.
 	 */
-	constexpr std::span<const std::byte> GetSteadyPart() const noexcept {
+	std::span<const std::byte> GetSteadyPart() const noexcept {
 		return {
 			reinterpret_cast<const std::byte *>(&address.sin6_addr),
 			sizeof(address.sin6_addr),
