--- a/src/wl/sys/wl_linux.c 2009-06-19 12:30:52.000000000 -0500
+++ b/src/wl/sys/wl_linux.c 2011-01-11 22:21:10.623044411 -0600

332c332
< 	.ndo_set_multicast_list = wl_set_multicast_list,
---
> 	.ndo_set_rx_mode = wl_set_multicast_list,


--- a/src/wl/sys/wl_linux.c
+++ b/src/wl/sys/wl_linux.c
@@ -40,7 +40,9 @@
 #include <linux/pci_ids.h>
 #define WLC_MAXBSSCFG		1	
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 4, 0)
 #include <asm/system.h>
+#endif
 #include <asm/io.h>
 #include <asm/irq.h>
 #include <asm/pgtable.h>
