--- zxing.cmake.orig	2023-11-10 19:38:55.601450169 +0000
+++ zxing.cmake	2023-11-10 19:39:15.889449494 +0000
@@ -12,11 +12,8 @@
 
     if (NOT STB_FOUND)
         include(FetchContent)
-        FetchContent_Declare (stb
-            GIT_REPOSITORY https://github.com/nothings/stb.git)
-        FetchContent_MakeAvailable (stb)
         add_library(stb::stb INTERFACE IMPORTED)
-        target_include_directories(stb::stb INTERFACE ${stb_SOURCE_DIR})
+        target_include_directories(stb::stb INTERFACE ../stb-src)
     else()
         add_library(stb::stb ALIAS PkgConfig::STB)
     endif()
