--- paranoia.c.old	2024-12-08 20:23:44.227674984 +0900
+++ paranoia.c	2024-12-08 20:27:09.336066414 +0900
@@ -24,0 +25,3 @@
+void character();
+void instructions();
+
@@ -33 +36 @@
-more()
+void more()
@@ -47 +50 @@
-new_clone(resume)
+int new_clone(resume)
@@ -67 +70 @@
-dice_roll(number,faces)
+int dice_roll(number,faces)
@@ -75 +78 @@
-instructions()
+void instructions()
@@ -96 +99 @@
-character()
+void character()
@@ -122 +125 @@
-choose(a,aptr,b,bptr)
+int choose(a,aptr,b,bptr)
@@ -132 +135 @@
-page1()
+int page1()
@@ -140 +143 @@
-page2()
+int page2()
@@ -158 +161 @@
-page3()
+int page3()
@@ -166 +169 @@
-page4()
+int page4()
@@ -174 +177 @@
-page5()
+int page5()
@@ -185 +188 @@
-page6()
+int page6()
@@ -197 +200 @@
-page7()
+int page7()
@@ -224 +227 @@
-page8()
+int page8()
@@ -241 +244 @@
-page9()
+int page9()
@@ -254 +257 @@
-page10()
+int page10()
@@ -282 +285 @@
-page11()
+int page11()
@@ -313 +316 @@
-page12()
+int page12()
@@ -325 +328 @@
-page13()
+int page13()
@@ -338 +341 @@
-page14()
+int page14()
@@ -352 +355 @@
-page15()
+int page15()
@@ -376 +379 @@
-page16()
+int page16()
@@ -387 +390 @@
-page17()
+int page17()
@@ -424 +427 @@
-page18()
+int page18()
@@ -440 +443 @@
-page19()
+int page19()
@@ -453 +456 @@
-page20()
+int page20()
@@ -465 +468 @@
-page21()
+int page21()
@@ -474 +477 @@
-page22()
+int page22()
@@ -486 +489 @@
-page23()
+int page23()
@@ -495 +498 @@
-page24()
+int page24()
@@ -503 +506 @@
-page25()
+int page25()
@@ -511 +514 @@
-page26()
+int page26()
@@ -521 +524 @@
-page27()
+int page27()
@@ -527 +530 @@
-page28()
+int page28()
@@ -533 +536 @@
-page29()
+int page29()
@@ -551 +554 @@
-page30()
+int page30()
@@ -578 +581 @@
-page31()
+int page31()
@@ -591 +594 @@
-page32()
+int page32()
@@ -601 +604 @@
-page33()
+int page33()
@@ -614 +617 @@
-page34()
+int page34()
@@ -642 +645 @@
-page35()
+int page35()
@@ -661 +664 @@
-page36()
+int page36()
@@ -684 +687 @@
-page37()
+int page37()
@@ -700 +703 @@
-page38()
+int page38()
@@ -717 +720 @@
-page39()
+int page39()
@@ -735 +738 @@
-page40()
+int page40()
@@ -767 +770 @@
-page41()
+int page41()
@@ -781 +784 @@
-page42()
+int page42()
@@ -791 +794 @@
-page43()
+int page43()
@@ -807 +810 @@
-page44()
+int page44()
@@ -825 +828 @@
-page45()
+int page45()
@@ -839 +842 @@
-page46()
+int page46()
@@ -847 +850 @@
-page47()
+int page47()
@@ -859 +862 @@
-page48()
+int page48()
@@ -869 +872 @@
-page49()
+int page49()
@@ -876 +879 @@
-page50()
+int page50()
@@ -885 +888 @@
-page51()
+int page51()
@@ -892 +895 @@
-page52()
+int page52()
@@ -899 +902 @@
-page53()
+int page53()
@@ -905 +908 @@
-page54()
+int page54()
@@ -919 +922 @@
-page55()
+int page55()
@@ -947 +950 @@
-page56()
+int page56()
@@ -955 +958 @@
-page57()
+int page57()
@@ -962 +965 @@
-next_page(this_page)
+int next_page(this_page)
@@ -1030 +1033 @@
-main()
+void main()
