--- i3-4.18.2/include/libi3.h	2020-07-26 10:35:49.000000000 +0200
+++ i3-4.18.2/include/libi3.h.new	2020-09-21 09:53:05.502446300 +0200
@@ -341,7 +341,7 @@
  */
 uint32_t get_colorpixel(const char *hex) __attribute__((const));
 
-#ifndef HAVE_strndup
+#ifndef HAVE_STRNDUP
 /**
  * Taken from FreeBSD
  * Returns a pointer to a new string which is a duplicate of the
--- i3-4.18.2/libi3/strndup.c	2020-07-26 10:35:49.000000000 +0200
+++ i3-4.18.2/libi3/strndup.c.new	2020-09-21 09:51:37.792446300 +0200
@@ -10,7 +10,7 @@
 #include <sys/types.h>
 #include <string.h>
 
-#ifndef HAVE_strndup
+#ifndef HAVE_STRNDUP
 /*
  * Taken from FreeBSD
  * Returns a pointer to a new string which is a duplicate of the
