diff -u -r qgis-1.7.4/src/core/spatialindex/storagemanager/DiskStorageManager.cc qgis-1.7.4_patched/src/core/spatialindex/storagemanager/DiskStorageManager.cc
--- qgis-1.7.4/src/core/spatialindex/storagemanager/DiskStorageManager.cc	2012-02-19 20:41:41.000000000 +0000
+++ qgis-1.7.4_patched/src/core/spatialindex/storagemanager/DiskStorageManager.cc	2012-04-10 11:59:14.835027450 +0100
@@ -19,6 +19,7 @@
 //  Email:
 //    mhadji@gmail.com
 
+#include <unistd.h>
 #include "../spatialindex/SpatialIndexImpl.h"
 
 #include <fcntl.h>
diff -u -r qgis-1.7.4/src/core/spatialindex/tools/TemporaryFile.cc qgis-1.7.4_patched/src/core/spatialindex/tools/TemporaryFile.cc
--- qgis-1.7.4/src/core/spatialindex/tools/TemporaryFile.cc	2012-02-19 20:41:41.000000000 +0000
+++ qgis-1.7.4_patched/src/core/spatialindex/tools/TemporaryFile.cc	2012-04-10 11:26:05.611021572 +0100
@@ -20,6 +20,7 @@
 //    mhadji@gmail.com
 
 #include <stdio.h>
+#include <unistd.h>
 
 #include <Tools.h>
 
