--- qgis-1.4.0-original/cmake/Python.cmake	2009-08-19 20:09:10.000000000 +0100
+++ qgis-1.4.0-patched/cmake/Python.cmake	2010-06-29 09:54:01.043992226 +0100
@@ -150,9 +150,7 @@
       # minimal version is 4.7 (to support universal builds)
       SET (SIP_MIN_VERSION 040700)
       TRY_RUN_PYTHON (RES "import sip\nprint '%x' % sip.SIP_VERSION" SIP_VERSION)
-      IF (SIP_VERSION EQUAL "${SIP_MIN_VERSION}" OR SIP_VERSION GREATER "${SIP_MIN_VERSION}")
         SET (SIP_IS_GOOD TRUE)
-      ENDIF (SIP_VERSION EQUAL "${SIP_MIN_VERSION}" OR SIP_VERSION GREATER "${SIP_MIN_VERSION}")
     
       IF (NOT SIP_IS_GOOD)
         MESSAGE (STATUS "SIP is required in version 4.7 or later!")
