diff -Naur tvheadend-4.2.8/Makefile.hdhomerun tvheadend-4.2.8.patched/Makefile.hdhomerun
--- tvheadend-4.2.8/Makefile.hdhomerun	2019-01-12 03:58:47.000000000 -0500
+++ tvheadend-4.2.8.patched/Makefile.hdhomerun	2022-02-16 13:55:18.391084069 -0500
@@ -32,10 +32,10 @@
 # Upstream Packages
 # ###########################################################################
 
-LIBHDHR         = libhdhomerun_20171221
+LIBHDHR         = libhdhomerun_20210624
 LIBHDHR_TB      = $(LIBHDHR).tgz
 LIBHDHR_URL     = http://download.silicondust.com/hdhomerun/$(LIBHDHR_TB)
-LIBHDHR_SHA1    = 6b019728eadea3af7a5686ed5ba44e970bca7365
+LIBHDHR_SHA1    = 9db6089b956de4e58084b623937f5fc3231fff0b
 
 # ###########################################################################
 # Library Config
