diff -Naur beav-140/symbol.c beav-140.patched/symbol.c
--- beav-140/symbol.c	2018-08-07 16:32:10.287386812 -0400
+++ beav-140.patched/symbol.c	2018-08-07 18:01:00.110213510 -0400
@@ -280,16 +280,20 @@
 {
     { KCTRL | 'A', dispshift, MSG_byte_shift, 0 },
     { KCTRL | 'B', backchar, MSG_back_char, SSRCH | SRPLC },
+    { KMETA | KCTRL | KCTLX | 'D', backchar, MSG_back_char, SSRCH | SRPLC },
     { KCTRL | 'C', quit, MSG_quit, 0 },	/* pvr */
     { KCTRL | 'D', forwdel, MSG_forw_del_char, SMOD | SSIZE | SSRCH | SRPLC },
     { KCTRL | 'E', dispswapbyte, MSG_toggle_swap, SSRCH | SRPLC },	/* pvr */
     { KCTRL | 'F', forwchar, MSG_forw_char, SSRCH | SRPLC },
+    { KMETA | KCTRL | KCTLX | 'C', forwchar, MSG_forw_char, SSRCH | SRPLC },
     { KCTRL | 'G', ctrlg, MSG_abort, SSRCH | SRPLC },
     { KCTRL | 'I', selfinsert, MSG_ins_self, SMOD | SSRCH | SRPLC },
     { KCTRL | 'H', backdel, MSG_back_del_char, SMOD | SSIZE | SSRCH | SRPLC },
     { KCTRL | 'L', refresh, MSG_refresh, SSRCH | SRPLC },
     { KCTRL | 'N', forwline, MSG_forw_line, SSRCH | SRPLC },
+    { KMETA | KCTRL | KCTLX | 'B', forwline, MSG_forw_line, SSRCH | SRPLC },
     { KCTRL | 'P', backline, MSG_back_line, SSRCH | SRPLC },
+    { KMETA | KCTRL | KCTLX | 'A', backline, MSG_back_line, SSRCH | SRPLC },
     { KCTRL | 'Q', quote, MSG_quote, 0 },
     { KCTRL | 'R', recall, MSG_recall, SSRCH | SRPLC },
     { KCTRL | 'T', twiddle, MSG_twiddle, SMOD | SSRCH | SRPLC },
