diff -Naur beav-140/Makefile beav-140.patched/Makefile
--- beav-140/Makefile	2018-08-07 16:09:02.441431939 -0400
+++ beav-140.patched/Makefile	2018-08-07 16:20:08.928410267 -0400
@@ -1,6 +1,6 @@
 # This is the makefile for BSD UNIX
-#CFLAGS=     -g -DUNIX
-CFLAGS=     -g -DUNIX -Wall
+CFLAGS=     -DUNIX $(SLKCFLAGS)
+#CFLAGS=     -g -DUNIX -Wall
 CC=gcc
 
 OFILES=		basic.o ebcdic.o fileio.o region.o text.o wangpc.o \
diff -Naur beav-140/basic.c beav-140.patched/basic.c
--- beav-140/basic.c	1994-11-30 12:43:34.000000000 -0500
+++ beav-140.patched/basic.c	2018-08-07 16:19:18.515411907 -0400
@@ -9,7 +9,6 @@
 
 #include    "def.h"
 
-bool move_ptr ();
 bool forwchar ();
 bool wind_on_dot ();
 bool backline ();
diff -Naur beav-140/def.h beav-140.patched/def.h
--- beav-140/def.h	2018-08-07 16:09:02.443431938 -0400
+++ beav-140.patched/def.h	2018-08-07 16:19:18.515411907 -0400
@@ -16,6 +16,7 @@
 
 #ifdef UNIX
 #include        <sys/types.h>
+#include        <string.h>
 #endif /* UNIX */
 
 #define BACKUP  1		/* Make backup file.            */
diff -Naur beav-140/file.c beav-140.patched/file.c
--- beav-140/file.c	2018-08-07 16:09:02.448431938 -0400
+++ beav-140.patched/file.c	2018-08-07 16:19:18.515411907 -0400
@@ -10,7 +10,6 @@
 char load_file ();
 char readin ();
 void makename ();
-bool writeout ();
 bool parse_f_name ();
 A32 ffseek ();
 A32 file_len ();
diff -Naur beav-140/random.c beav-140.patched/random.c
--- beav-140/random.c	2018-08-07 16:09:02.455431938 -0400
+++ beav-140.patched/random.c	2018-08-07 16:19:18.516411907 -0400
@@ -73,7 +73,6 @@
 extern bool dont_repeat;
 extern BUFFER sav_buf;
 
-char dec_chr_ok ();
 ulong get_long ();
 void wind_on_dot_all ();
 
diff -Naur beav-140/symbol.c beav-140.patched/symbol.c
--- beav-140/symbol.c	2018-08-07 16:09:02.459431938 -0400
+++ beav-140.patched/symbol.c	2018-08-07 16:19:18.516411907 -0400
@@ -8,7 +8,6 @@
 #include		<string.h>
 #include		"def.h"
 
-void keyadd ();
 void keydup ();
 
 
