From b0d9f0c79b1dbcb42a5661e96a364519ac5faded Mon Sep 17 00:00:00 2001
From: Andrey Utkin <andrey.krieger.utkin@gmail.com>
Date: Tue, 8 Oct 2013 23:09:52 +0300
Subject: [PATCH 5/5] Fix compilation on pre-3.9

Provide file_inode().
---
 driver/proc.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/driver/proc.c b/driver/proc.c
index 0a28e45..367ff97 100644
--- a/driver/proc.c
+++ b/driver/proc.c
@@ -25,6 +25,12 @@
 
 #define MAX_PROC_STR_LEN 32
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,9,0)
+static inline struct inode *file_inode(struct file *f)
+{
+    return f->f_path.dentry->d_inode;
+}
+#endif
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3,10,0)
 static inline void *PDE_DATA(const struct inode *inode)
-- 
1.8.4

