diff -Naur kmscon-9.0.0/meson.build kmscon-9.0.0.patched/meson.build
--- kmscon-9.0.0/meson.build	2022-06-15 03:48:52.000000000 -0400
+++ kmscon-9.0.0.patched/meson.build	2023-07-22 03:32:29.965571375 -0400
@@ -7,7 +7,7 @@
   license: 'MIT',
   # meson 0.58: f-string
   # meson 0.62: dependency libdl
-  meson_version: '>=0.62.0',
+  meson_version: '>=0.59.4',
   default_options: [
     'warning_level=1',
     'werror=true',
@@ -48,7 +48,6 @@
 libtsm_deps = dependency('libtsm', version: '>=4.0.0')
 libudev_deps = dependency('libudev', version: '>=172')
 libdrm_deps = dependency('libdrm')
-dl_deps = dependency('dl')
 threads_deps = dependency('threads')
 
 python = find_program('python3')
diff -Naur kmscon-9.0.0/src/meson.build kmscon-9.0.0.patched/src/meson.build
--- kmscon-9.0.0/src/meson.build	2022-06-15 03:48:52.000000000 -0400
+++ kmscon-9.0.0.patched/src/meson.build	2023-07-22 03:33:42.300564567 -0400
@@ -257,7 +257,7 @@
   kmscon_srcs += 'kmscon_terminal.c'
 endif
 kmscon = executable('kmscon', kmscon_srcs,
-  dependencies: [xkbcommon_deps, libtsm_deps, threads_deps, dl_deps, conf_deps, shl_deps, eloop_deps, uterm_deps],
+  dependencies: [xkbcommon_deps, libtsm_deps, threads_deps, conf_deps, shl_deps, eloop_deps, uterm_deps],
   export_dynamic: true,
   install: true,
   install_dir: libexecdir,
