diff -Naur kmscon-8/src/uterm_drm3d_video.c kmscon-8.patched/src/uterm_drm3d_video.c
--- kmscon-8/src/uterm_drm3d_video.c	2013-10-18 12:32:27.698000000 -0400
+++ kmscon-8.patched/src/uterm_drm3d_video.c	2021-08-25 00:16:43.117558748 -0400
@@ -30,6 +30,7 @@
 #define EGL_EGLEXT_PROTOTYPES
 #define GL_GLEXT_PROTOTYPES
 
+#include <sys/sysmacros.h>
 #include <EGL/egl.h>
 #include <EGL/eglext.h>
 #include <errno.h>
diff -Naur kmscon-8/src/uterm_vt.c kmscon-8.patched/src/uterm_vt.c
--- kmscon-8/src/uterm_vt.c	2013-10-23 17:15:18.865000000 -0400
+++ kmscon-8.patched/src/uterm_vt.c	2021-08-25 00:16:51.038558028 -0400
@@ -27,6 +27,7 @@
  * Virtual Terminals
  */
 
+#include <sys/sysmacros.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <linux/kd.h>
