diff -Naur ZoneMinder-r3649.orig/src/zm_logger.cpp ZoneMinder-r3649/src/zm_logger.cpp
--- ZoneMinder-r3649.orig/src/zm_logger.cpp	2011-06-28 13:07:35.197263000 +0200
+++ ZoneMinder-r3649/src/zm_logger.cpp	2012-06-20 15:00:53.054661523 +0200
@@ -22,6 +22,7 @@
 #include "zm_config.h"
 #include "zm_utils.h"
 
+#include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -Naur ZoneMinder-r3649.orig/src/zm_rtp_data.cpp ZoneMinder-r3649/src/zm_rtp_data.cpp
--- ZoneMinder-r3649.orig/src/zm_rtp_data.cpp	2010-11-11 13:22:35.884567000 +0100
+++ ZoneMinder-r3649/src/zm_rtp_data.cpp	2012-06-20 15:05:36.466118392 +0200
@@ -17,6 +17,7 @@
 // Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 // 
 
+#include <unistd.h>
 #include "zm.h"
 
 #if HAVE_LIBAVFORMAT
diff -Naur ZoneMinder-r3649.orig/src/zm_rtp_source.cpp ZoneMinder-r3649/src/zm_rtp_source.cpp
--- ZoneMinder-r3649.orig/src/zm_rtp_source.cpp	2010-07-04 20:00:35.480475000 +0200
+++ ZoneMinder-r3649/src/zm_rtp_source.cpp	2012-06-20 15:05:26.907237894 +0200
@@ -17,6 +17,7 @@
 // Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 // 
 
+#include <unistd.h>
 #include "zm_rtp_source.h"
 
 #include "zm_time.h"
diff -Naur ZoneMinder-r3649.orig/src/zm_thread.cpp ZoneMinder-r3649/src/zm_thread.cpp
--- ZoneMinder-r3649.orig/src/zm_thread.cpp	2011-06-21 11:19:10.272416000 +0200
+++ ZoneMinder-r3649/src/zm_thread.cpp	2012-06-20 15:06:50.786189262 +0200
@@ -17,6 +17,7 @@
 // Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 // 
 
+#include <unistd.h>
 #include "zm_thread.h"
 
 #include "zm_logger.h"
diff -Naur ZoneMinder-r3649.orig/src/zm_timer.cpp ZoneMinder-r3649/src/zm_timer.cpp
--- ZoneMinder-r3649.orig/src/zm_timer.cpp	2011-06-21 11:19:10.272416000 +0200
+++ ZoneMinder-r3649/src/zm_timer.cpp	2012-06-20 15:07:44.437518529 +0200
@@ -17,6 +17,7 @@
 // Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 // 
 
+#include <unistd.h>
 #include "zm_timer.h"
 
 #include "zm_logger.h"
