--- ZoneMinder-1.24.2/Makefile.am.orig	2006-12-25 01:21:07.000000000 +0000
+++ ZoneMinder-1.24.2/Makefile.am	2006-12-25 01:22:04.000000000 +0000
@@ -23,9 +23,9 @@
 # Yes, you are correct. This is a HACK!
 install-data-hook:
 	( cd $(DESTDIR)$(sysconfdir); chown $(webuser):$(webgroup) $(sysconf_DATA); chmod 600 $(sysconf_DATA) )
-	-( if ! test -e $(RUNDIR); then mkdir $(RUNDIR); fi )
-	( if ! test -e $(ZM_RUNDIR); then mkdir $(ZM_RUNDIR); fi; chown $(webuser):$(webgroup) $(ZM_RUNDIR); chmod u+w $(ZM_RUNDIR) )
+	-( if ! test -e $(DESTDIR)$(RUNDIR); then @mkdir_p@ $(DESTDIR)$(RUNDIR); fi )
+	( if ! test -e $(DESTDIR)$(ZM_RUNDIR); then @mkdir_p@ $(DESTDIR)$(ZM_RUNDIR); fi; chown $(webuser):$(webgroup) $(DESTDIR)$(ZM_RUNDIR); chmod u+w $(DESTDIR)$(ZM_RUNDIR) )
 
 uninstall-hook:
 	@-( cd $(DESTDIR)$(webdir); rm -rf events graphics images sounds temp )
-	@-( rm -rf $(ZM_RUNDIR) )
+	@-( rm -rf $(DESTDIR)$(ZM_RUNDIR) )
