--- original/web/includes/functions.php   2009-05-08 12:17:10.000000000 +0300
+++ patched/web/includes/functions.php   2010-04-13 22:44:27.000000000 +0300
@@ -820,7 +820,7 @@
                 $version = $logVersion[1];
                 $browser = 'konqueror';
         }
-    elseif (ereg( 'Opera ([0-9].[0-9]{1,2})',$_SERVER['HTTP_USER_AGENT'],$logVersion))
+    elseif (ereg( 'Opera/([0-9].[0-9]{1,2})',$_SERVER['HTTP_USER_AGENT'],$logVersion))
     {
         $version = $logVersion[1];
         $browser = 'opera';
@@ -844,6 +844,18 @@
     return( $browser == "mozilla" );
 }
 
+function isOpera()
+{
+    getBrowser( $browser, $version );
+    return( $browser == "opera" );
+}
+
+function isSafari()
+{
+    getBrowser( $browser, $version );
+    return( $browser == "safari" );
+}
+
 function isKonqueror()
 {
    getBrowser( $browser, $version );
@@ -875,7 +887,7 @@
 
 function canStreamNative()
 {
-   return( ZM_CAN_STREAM == "yes" || ( ZM_CAN_STREAM == "auto" && (isNetscape() || isKonqueror()) ) );
+   return( ZM_CAN_STREAM == "yes" || ( ZM_CAN_STREAM == "auto" && (isNetscape() || isKonqueror() || isOpera() || isSafari()) ) );
 }
 
 function canStreamApplet()
