diff -u -ru ZoneMinder-1.24.2-old//src/zm_utils.cpp ZoneMinder-1.24.2/src/zm_utils.cpp
--- ZoneMinder-1.24.2-old//src/zm_utils.cpp	2010-06-03 01:34:27.981235559 +0300
+++ ZoneMinder-1.24.2/src/zm_utils.cpp	2010-06-03 01:44:52.108291327 +0300
@@ -20,6 +20,7 @@
 //#include "zm_debug.h"
 #include "zm_utils.h"
 
+#include <cstdio>
 #include <stdarg.h>
 
 const std::string stringtf( const char *format, ... )
