diff -Naur minitube-20120813_338dac6.orig/src/qtsingleapplication/qtlocalpeer.cpp minitube-20120813_338dac6/src/qtsingleapplication/qtlocalpeer.cpp
--- minitube-20120813_338dac6.orig/src/qtsingleapplication/qtlocalpeer.cpp	2012-08-13 17:09:04.000000000 +0200
+++ minitube-20120813_338dac6/src/qtsingleapplication/qtlocalpeer.cpp	2012-09-12 13:00:44.330889268 +0200
@@ -58,6 +58,7 @@
 #endif
 #if defined(Q_OS_UNIX)
 #include <time.h>
+#include <unistd.h>
 #endif
 
 namespace QtLP_Private {
