diff -Naur Tk-804.036/Event/Event.xs Tk-804.036.new/Event/Event.xs
--- Tk-804.036/Event/Event.xs	2021-02-07 17:34:03.000000000 +0100
+++ Tk-804.036.new/Event/Event.xs	2024-06-14 21:23:55.743817569 +0200
@@ -1532,7 +1532,7 @@
 BOOT:
  {
 #ifdef pWARN_NONE
-  SV *old_warn = PL_curcop->cop_warnings;
+  void *old_warn = PL_curcop->cop_warnings;
   PL_curcop->cop_warnings = pWARN_NONE;
 #endif
   newXS("Tk::Event::INIT", XS_Tk__Event_INIT, file);
diff -Naur Tk-804.036/tkGlue.c Tk-804.036.new/tkGlue.c
--- Tk-804.036/tkGlue.c	2021-02-07 17:34:04.000000000 +0100
+++ Tk-804.036.new/tkGlue.c	2024-06-14 21:25:08.742817569 +0200
@@ -5543,13 +5543,8 @@
  char *XEventMethods = "abcdfhkmopstvwxyABDEKNRSTWXY#";
  char buf[128];
  CV *cv;
-#if PERL_REVISION > 5 || (PERL_REVISION == 5 && PERL_VERSION >= 9)
-#define COP_WARNINGS_TYPE STRLEN*
-#else
-#define COP_WARNINGS_TYPE SV*
-#endif
 #ifdef pWARN_NONE
- COP_WARNINGS_TYPE old_warn = PL_curcop->cop_warnings;
+ void *old_warn = PL_curcop->cop_warnings;
  PL_curcop->cop_warnings = pWARN_NONE;
 #endif
 
