diff -Nur glw-master/Makefile.am glw-master.new/Makefile.am
--- glw-master/Makefile.am	2018-07-18 19:05:24.000000000 +0200
+++ glw-master.new/Makefile.am	2022-10-13 22:38:37.276257964 +0200
@@ -35,7 +35,7 @@
 libGLw_la_CFLAGS += $(MOTIF_CFLAGS)
 libGLw_la_LIBADD += $(MOTIF_LIBS)
 endif
-library_includedir=$(includedir)/GL/
+library_includedir=$(includedir)
 library_include_HEADERS=GLwDrawA.h GLwDrawAP.h GLwMDrawA.h GLwMDrawAP.h
 
 clean:
