diff -Nur nant-0.90.orig//NAnt.build nant-0.90//NAnt.build
--- nant-0.90.orig//NAnt.build	2010-05-08 16:12:26.000000000 -0500
+++ nant-0.90//NAnt.build	2010-07-07 03:25:12.707823923 -0500
@@ -868,7 +868,7 @@
             <property name="install.destdir" value="${destdir}" />
         </if>
         <property name="install.bindir" value="${install.destdir + path::combine(install.prefix, 'bin')}" />
-        <property name="install.pkgconfigdir" value="${install.destdir + path::combine(install.prefix, 'lib/pkgconfig')}" />
+        <property name="install.pkgconfigdir" value="${install.destdir + path::combine(install.prefix, 'lib64/pkgconfig')}" />
         <property name="install.copylocation" value="${install.destdir + install.nant}" />
 
         <!-- perform (staged) install -->
