diff -ur libcap-1.97/Make.Rules libcap-1.97.new/Make.Rules
--- libcap-1.97/Make.Rules	2007-08-14 08:21:26.000000000 +0200
+++ libcap-1.97.new/Make.Rules	2007-10-29 12:35:31.000000000 +0100
@@ -48,7 +48,7 @@
 
 CC=gcc
 COPTFLAGS=-O2
-DEBUG=-O2 -g #-DDEBUG
+DEBUG=-O2 -march=i686 -mtune=i686 #-g #-DDEBUG
 WARNINGS=-fPIC -D_POSIX_SOURCE -Wall -Wwrite-strings \
         -Wpointer-arith -Wcast-qual -Wcast-align \
         -Wstrict-prototypes -Wmissing-prototypes \
