--- SDL_Pango-0.1.2/src/SDL_Pango.c.warn	2004-12-10 18:06:33.000000000 +0900
+++ SDL_Pango-0.1.2/src/SDL_Pango.c	2006-09-23 22:49:42.000000000 +0900
@@ -437,6 +437,10 @@
     PangoRectangle ink_rect;
     int x_off = 0;
 
+    /* only suppress warning */
+    fg_color.red = fg_color.green = fg_color.blue = 0;
+    bg_color.red = bg_color.green = bg_color.blue = 0;
+
     while (tmp_list) {
 	SDLPango_Matrix color_matrix = context->color_matrix;
 	PangoUnderline uline = PANGO_UNDERLINE_NONE;
