--- makefile.orig	2009-09-23 11:17:50.000000000 +0400
+++ makefile	2009-10-31 19:21:26.000000000 +0300
@@ -244,6 +244,7 @@
 biewlib/tw_class.o:           biewlib/tw_class.c
 biewlib/sysdep/$(MACHINE)/aclib.o:            biewlib/sysdep/$(MACHINE)/aclib.c
 biewlib/sysdep/$(MACHINE)/cpu_info.o:         biewlib/sysdep/$(MACHINE)/cpu_info.c
+	$(CC) $(CFLAGS) $(INCS) -msse -c $< -o $@
 biewlib/sysdep/$(MACHINE)/$(HOST)/fileio.o:   biewlib/sysdep/$(MACHINE)/$(HOST)/fileio.c
 biewlib/sysdep/$(MACHINE)/$(HOST)/keyboard.o: biewlib/sysdep/$(MACHINE)/$(HOST)/keyboard.c
 biewlib/sysdep/$(MACHINE)/$(HOST)/mmfio.o:    biewlib/sysdep/$(MACHINE)/$(HOST)/mmfio.c
