--- pcapy-0.11.3/setup.py.orig	2018-04-09 14:01:47.209882717 -0600
+++ pcapy-0.11.3/setup.py	2018-04-09 14:06:58.308869383 -0600
@@ -88,7 +88,4 @@
           library_dirs=library_dirs,
           libraries=libraries)],
       #scripts=['tests/pcapytests.py', 'tests/96pings.pcap'],
-      data_files=[
-          (os.path.join('share', 'doc', PACKAGE_NAME), ['README', 'LICENSE', 'pcapy.html']),
-          (os.path.join('share', 'doc', PACKAGE_NAME, 'tests'), glob.glob('tests/*'))]
       )
