--- pcapy-0.11.1.orig/setup.py	2017-01-16 09:38:07.000000000 -0700
+++ pcapy-0.11.1/setup.py	2017-04-06 17:01:20.736516673 -0600
@@ -79,8 +79,8 @@
           library_dirs=library_dirs,
           libraries=libraries)],
       # scripts=['tests/pcapytests.py', 'tests/96pings.pcap'],
-      data_files=[
-          (os.path.join('share', 'doc', PACKAGE_NAME),
-              ['README', 'LICENSE', 'pcapy.html']),
-          ('tests', ['tests/pcapytests.py', 'tests/96pings.pcap'])]
+      #data_files=[
+      #    (os.path.join('share', 'doc', PACKAGE_NAME),
+      #        ['README', 'LICENSE', 'pcapy.html']),
+      #    ('tests', ['tests/pcapytests.py', 'tests/96pings.pcap'])]
       )
