diff -u -r flashqard-0.15.0/src/PrettyButton.cpp flashqard-0.15.0-patched/src/PrettyButton.cpp
--- flashqard-0.15.0/src/PrettyButton.cpp	2009-09-08 22:56:54.000000000 +0100
+++ flashqard-0.15.0-patched/src/PrettyButton.cpp	2015-07-28 11:49:56.506872447 +0100
@@ -25,6 +25,7 @@
 #include "PrettyButton.h"
 #include "TextEditor.h"
 #include "MimeData.h"
+#include <QApplication>
 #include <QPainter>
 #include <QTimerEvent>
 #include <QDropEvent>
diff -u -r flashqard-0.15.0/src/Utilities.cpp flashqard-0.15.0-patched/src/Utilities.cpp
--- flashqard-0.15.0/src/Utilities.cpp	2009-09-08 22:56:54.000000000 +0100
+++ flashqard-0.15.0-patched/src/Utilities.cpp	2015-07-28 11:45:33.484870666 +0100
@@ -22,6 +22,7 @@
 // | Author:  Shahab Shirazi <shahab@flashqard-project.org>                  |
 // +-------------------------------------------------------------------------+
 
+#include <QDir>
 #include <QRegExp>
 #include <QToolBar>
 #include <QMessageBox>
