--- Makefile.ORIG	2008-05-10 22:59:21.000000000 -0500
+++ Makefile	2008-05-10 23:01:31.000000000 -0500
@@ -35,17 +36,17 @@
 wvdial wvdialconf papchaptest pppmon: wvdial.a
 
 install-bin: all
-	[ -d ${BINDIR}      ] || install -d ${BINDIR}
-	[ -d ${PPPDIR}      ] || install -d ${PPPDIR}
-	install -m 0755 wvdial wvdialconf ${BINDIR}
-	cp ppp.provider ${PPPDIR}/wvdial
-	cp ppp.provider-pipe ${PPPDIR}/wvdial-pipe
+	[ -d $(DESTDIR)/${BINDIR}      ] || install -d $(DESTDIR)/${BINDIR}
+	[ -d $(DESTDIR)/${PPPDIR}      ] || install -d $(DESTDIR)/${PPPDIR}
+	install -m 0755 wvdial wvdialconf $(DESTDIR)/${BINDIR}
+	cp ppp.provider $(DESTDIR)/${PPPDIR}/wvdial
+	cp ppp.provider-pipe $(DESTDIR)/${PPPDIR}/wvdial-pipe
 
 install-man:
-	[ -d ${MANDIR}/man1 ] || install -d ${MANDIR}/man1
-	[ -d ${MANDIR}/man5 ] || install -d ${MANDIR}/man5
-	install -m 0644 wvdial.1 wvdialconf.1 ${MANDIR}/man1
-	install -m 0644 wvdial.conf.5 ${MANDIR}/man5
+	[ -d $(DESTDIR)/${MANDIR}/man1 ] || install -d $(DESTDIR)/${MANDIR}/man1
+	[ -d $(DESTDIR)/${MANDIR}/man5 ] || install -d $(DESTDIR)/${MANDIR}/man5
+	install -m 0644 wvdial.1 wvdialconf.1 $(DESTDIR)/${MANDIR}/man1
+	install -m 0644 wvdial.conf.5 $(DESTDIR)/${MANDIR}/man5
 
 install: install-bin install-man
 
