diff -Naur GPXSee-5.13.orig/src/GUI/units.h GPXSee-5.13/src/GUI/units.h
--- GPXSee-5.13.orig/src/GUI/units.h	2018-03-20 00:40:49.000000000 +0300
+++ GPXSee-5.13/src/GUI/units.h	2018-04-08 20:17:44.008003642 +0300
@@ -30,6 +30,6 @@
 #define C2FS    1.8  // Celsius to Farenheit - scale
 #define C2FO    32.0 // Celsius to Farenheit - offset
 
-#define UNIT_SPACE     QString::fromUtf8("\xE2\x80\x89")
+#define UNIT_SPACE     QLatin1Char{' '}
 
 #endif // UNITS_H
