diff -Naur GPXSee-7.2.orig/src/GUI/units.h GPXSee-7.2/src/GUI/units.h
--- GPXSee-7.3.orig/src/GUI/units.h	2018-12-08 21:09:23.510215182 +0300
+++ GPXSee-7.3/src/GUI/units.h	2018-12-08 21:10:17.907437326 +0300
@@ -30,6 +30,6 @@
 #define C2FS    1.8  // Celsius to Farenheit - scale
 #define C2FO    32.0 // Celsius to Farenheit - offset
 
-#define UNIT_SPACE     QString::fromUtf8("\xE2\x80\x89")
+#define UNIT_SPACE     QLatin1Char{' '}
 
 #endif // UNITS_H
