diff -Naur graphviz-2.28.0.orig/tclpkg/gv/gv_php_init.c graphviz-2.28.0/tclpkg/gv/gv_php_init.c
--- graphviz-2.28.0.orig/tclpkg/gv/gv_php_init.c	2011-01-25 17:30:51.000000000 +0100
+++ graphviz-2.28.0/tclpkg/gv/gv_php_init.c	2012-05-30 04:10:40.657221055 +0200
@@ -19,11 +19,13 @@
 
 static size_t gv_string_writer (GVJ_t *job, const char *s, size_t len)
 {
+    TSRMLS_FETCH();
     return PHPWRITE(s, len);
 }
 
 static size_t gv_channel_writer (GVJ_t *job, const char *s, size_t len)
 {
+    TSRMLS_FETCH();
     return PHPWRITE(s, len);
 }
 
