--- ksh-openbsd_cvs/main.c	2010-02-23 12:49:19.000000000 +0100
+++ ksh-openbsd_cvs.new/main.c	2010-04-01 21:01:24.134232647 +0200
@@ -9,7 +9,12 @@
 #include "sh.h"
 #include <sys/stat.h>
 #include <pwd.h>
+#include "strlcpy.h"
 
+/* on Linux we do not have _PW_NAME_LEN, but libc6 allows us 32 character usernames */
+#ifndef _PW_NAME_LEN 
+# define _PW_NAME_LEN 31
+#endif
 extern char **environ;
 
 /*
