diff -Naur rplay-3.3.2-orig/Makefile.in rplay-3.3.2/Makefile.in
--- rplay-3.3.2-orig/Makefile.in	1998-07-14 18:35:22.000000000 -0400
+++ rplay-3.3.2/Makefile.in	2011-12-25 22:49:48.496492799 -0500
@@ -7,6 +7,8 @@
 
 TARGET= @BUILD_TARGETS@
 
+DESTDIR=
+
 all:
 		@for i in $(TARGET); \
 		do \
@@ -18,13 +20,13 @@
 install:
 		@for i in $(TARGET); \
 		do \
-			(echo make install in $$i...; cd $$i; $(MAKE) $(MFLAGS) install); \
+			(echo make install in $$i...; cd $$i; $(MAKE) $(MFLAGS) install $(DESTDIR)); \
 		done;
 
 uninstall:
 		@for i in $(TARGET); \
 		do \
-			(echo make uninstall in $$i...; cd $$i; $(MAKE) $(MFLAGS) uninstall); \
+			(echo make uninstall in $$i...; cd $$i; $(MAKE) $(MFLAGS) uninstall $(DESTDIR)); \
 		done;
 
 clean:
diff -Naur rplay-3.3.2-orig/devrplay/Makefile.in rplay-3.3.2/devrplay/Makefile.in
--- rplay-3.3.2-orig/devrplay/Makefile.in	1999-03-22 21:36:30.000000000 -0500
+++ rplay-3.3.2/devrplay/Makefile.in	2011-12-25 22:56:17.270462280 -0500
@@ -8,6 +8,8 @@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS= @srcdir@/../mkinstalldirs
 
+DESTDIR=
+
 CPPFLAGS=	$(CC_OPTIONS) -fPIC -I. -I../include -I@srcdir@/../include @DEFS@
 
 .c.o:
@@ -31,11 +33,11 @@
 		(cd ../librplay; $(MAKE) $(MFLAGS) librplay.so)
 
 install:	all
-		$(MKINSTALLDIRS) $(libdir)
-		$(INSTALL_DATA) $(TARGET) $(libdir)
+		$(MKINSTALLDIRS) $(DESTDIR)/$(libdir)
+		$(INSTALL_DATA) $(TARGET) $(DESTDIR)/$(libdir)
 
 uninstall:
-		$(RM) $(libdir)/$(TARGET)
+		$(RM) $(DESTDIR)/$(libdir)/$(TARGET)
 
 clean:
 		$(RM) $(OBJS) $(TARGET) a.out core *~ *.bak *.orig TAGS
diff -Naur rplay-3.3.2-orig/doc/Makefile.in rplay-3.3.2/doc/Makefile.in
--- rplay-3.3.2-orig/doc/Makefile.in	1998-07-14 18:35:23.000000000 -0400
+++ rplay-3.3.2/doc/Makefile.in	2011-12-25 22:59:07.183495856 -0500
@@ -8,6 +8,8 @@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS= @srcdir@/../mkinstalldirs
 
+DESTDIR=
+
 .SUFFIXES:
 .SUFFIXES: .texi .info .dvi .ps .html
 
@@ -39,31 +41,31 @@
 	./genman rplayd.8.in
 
 install: all
-	-$(MKINSTALLDIRS) $(mandir)/man1 $(mandir)/man5 $(mandir)/man8 $(infodir)
-	-$(INSTALL_DATA) rplay.1 $(mandir)/man1
-	-$(INSTALL_DATA) rplayd.8 $(mandir)/man8
-	-$(INSTALL_DATA) rptp.1 $(mandir)/man1
-	-$(INSTALL_DATA) rplay.conf.5 $(mandir)/man5
-	-$(INSTALL_DATA) rplay.helpers.5 $(mandir)/man5
-	-$(INSTALL_DATA) rplay.hosts.5 $(mandir)/man5
-	-$(INSTALL_DATA) rplay.servers.5 $(mandir)/man5
-	-$(INSTALL_DATA) RPLAY.info $(infodir)/RPLAY.info
-	-$(INSTALL_DATA) RPTP.info $(infodir)/RPTP.info
-	-$(INSTALL_DATA) librplay.info $(infodir)/librplay.info
-	-$(INSTALL_DATA) rplay.info $(infodir)/rplay.info
+	-$(MKINSTALLDIRS) $(DESTDIR)/$(mandir)/man1 $(DESTDIR)/$(mandir)/man5 $(DESTDIR)/$(mandir)/man8 $(DESTDIR)/$(infodir)
+	-$(INSTALL_DATA) rplay.1 $(DESTDIR)/$(mandir)/man1
+	-$(INSTALL_DATA) rplayd.8 $(DESTDIR)/$(mandir)/man8
+	-$(INSTALL_DATA) rptp.1 $(DESTDIR)/$(mandir)/man1
+	-$(INSTALL_DATA) rplay.conf.5 $(DESTDIR)/$(mandir)/man5
+	-$(INSTALL_DATA) rplay.helpers.5 $(DESTDIR)/$(mandir)/man5
+	-$(INSTALL_DATA) rplay.hosts.5 $(DESTDIR)/$(mandir)/man5
+	-$(INSTALL_DATA) rplay.servers.5 $(DESTDIR)/$(mandir)/man5
+	-$(INSTALL_DATA) RPLAY.info $(DESTDIR)/$(infodir)/RPLAY.info
+	-$(INSTALL_DATA) RPTP.info $(DESTDIR)/$(infodir)/RPTP.info
+	-$(INSTALL_DATA) librplay.info $(DESTDIR)/$(infodir)/librplay.info
+	-$(INSTALL_DATA) rplay.info $(DESTDIR)/$(infodir)/rplay.info
 
 uninstall:
-	-$(RM) $(mandir)/man1/rplay.1
-	-$(RM) $(mandir)/man8/rplayd.8
-	-$(RM) $(mandir)/man1/rptp.1
-	-$(RM) $(mandir)/man5/rplay.conf.5
-	-$(RM) $(mandir)/man5/rplay.helpers.5
-	-$(RM) $(mandir)/man5/rplay.hosts.5
-	-$(RM) $(mandir)/man5/rplay.servers.5
-	-$(RM) $(infodir)/RPLAY.info
-	-$(RM) $(infodir)/RPTP.info
-	-$(RM) $(infodir)/librplay.info
-	-$(RM) $(infodir)/rplay.info
+	-$(RM) $(DESTDIR)/$(mandir)/man1/rplay.1
+	-$(RM) $(DESTDIR)/$(mandir)/man8/rplayd.8
+	-$(RM) $(DESTDIR)/$(mandir)/man1/rptp.1
+	-$(RM) $(DESTDIR)/$(mandir)/man5/rplay.conf.5
+	-$(RM) $(DESTDIR)/$(mandir)/man5/rplay.helpers.5
+	-$(RM) $(DESTDIR)/$(mandir)/man5/rplay.hosts.5
+	-$(RM) $(DESTDIR)/$(mandir)/man5/rplay.servers.5
+	-$(RM) $(DESTDIR)/$(infodir)/RPLAY.info
+	-$(RM) $(DESTDIR)/$(infodir)/RPTP.info
+	-$(RM) $(DESTDIR)/$(infodir)/librplay.info
+	-$(RM) $(DESTDIR)/$(infodir)/rplay.info
 
 clean:
 	$(RM) *~ *.bak *.orig
diff -Naur rplay-3.3.2-orig/include/Makefile.in rplay-3.3.2/include/Makefile.in
--- rplay-3.3.2-orig/include/Makefile.in	1998-07-14 18:35:22.000000000 -0400
+++ rplay-3.3.2/include/Makefile.in	2011-12-25 23:01:02.926501049 -0500
@@ -8,6 +8,8 @@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS= @srcdir@/../mkinstalldirs
 
+DESTDIR=
+
 TARGET=		rplay.h
 
 all:
@@ -15,8 +17,8 @@
 $(TARGET):
 
 install:	$(TARGET)
-		$(MKINSTALLDIRS) $(includedir)
-		$(INSTALL_DATA) @srcdir@/$(TARGET) $(includedir)/$(TARGET)
+		$(MKINSTALLDIRS) $(DESTDIR)/$(includedir)
+		$(INSTALL_DATA) @srcdir@/$(TARGET) $(DESTDIR)/$(includedir)/$(TARGET)
 
 uninstall:
 		$(RM) $(includedir)/$(TARGET)
diff -Naur rplay-3.3.2-orig/librplay/Makefile.in rplay-3.3.2/librplay/Makefile.in
--- rplay-3.3.2-orig/librplay/Makefile.in	1999-03-22 21:38:38.000000000 -0500
+++ rplay-3.3.2/librplay/Makefile.in	2011-12-25 22:52:41.361565058 -0500
@@ -11,6 +11,8 @@
 CPPFLAGS=	$(CC_OPTIONS) -I. -I../include -I@srcdir@/../include @DEFS@
 LDFLAGS=	$(LD_OPTIONS) -shared @LDFLAGS@
 
+DESTDIR=
+
 .c.o:
 		$(CC) -c $(CPPFLAGS) $(CFLAGS) $<
 
@@ -48,13 +50,13 @@
 #
 
 install:	all
-		$(MKINSTALLDIRS) $(libdir)
-		$(INSTALL_DATA) $(TARGET) $(libdir)
-		$(INSTALL_DATA) librplay.so $(libdir)
+		$(MKINSTALLDIRS) $(DESTDIR)/$(libdir)
+		$(INSTALL_DATA) $(TARGET) $(DESTDIR)/$(libdir)
+		$(INSTALL_DATA) librplay.so $(DESTDIR)/$(libdir)
 
 uninstall:
-		$(RM) $(libdir)/$(TARGET)
-		$(RM) $(libdir)/librplay.so
+		$(RM) $(DESTDIR)/$(libdir)/$(TARGET)
+		$(RM) $(DESTDIR)/$(libdir)/librplay.so
 
 clean:
 		$(RM) $(OBJS) $(SHAREDOBJS) $(TARGET) librplay.so a.out core *~ *.bak *.orig TAGS
diff -Naur rplay-3.3.2-orig/rplay/Makefile.in rplay-3.3.2/rplay/Makefile.in
--- rplay-3.3.2-orig/rplay/Makefile.in	1999-03-20 19:44:57.000000000 -0500
+++ rplay-3.3.2/rplay/Makefile.in	2011-12-25 22:54:52.564503922 -0500
@@ -8,6 +8,8 @@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS= @srcdir@/../mkinstalldirs
 
+DESTDIR=
+
 CPPFLAGS=	$(CC_OPTIONS) -I. -I../include -I@srcdir@/../include -I@srcdir@/../lib @DEFS@
 
 .c.o:
@@ -31,11 +33,11 @@
 		(cd ../lib; $(MAKE) $(MFLAGS))
 
 install:	all
-		$(MKINSTALLDIRS) $(bindir)
-		$(INSTALL_PROGRAM) $(TARGET) $(bindir)/$(TARGET)
+		$(MKINSTALLDIRS) $(DESTDIR)/$(bindir)
+		$(INSTALL_PROGRAM) $(TARGET) $(DESTDIR)/$(bindir)/$(TARGET)
 
 uninstall:
-		$(RM) $(bindir)/$(TARGET)
+		$(RM) $(DESTDIR)/$(bindir)/$(TARGET)
 
 clean:
 		$(RM) $(OBJS) $(TARGET) a.out core *~ *.bak *.orig TAGS
diff -Naur rplay-3.3.2-orig/rplayd/Makefile.in rplay-3.3.2/rplayd/Makefile.in
--- rplay-3.3.2-orig/rplayd/Makefile.in	1999-06-09 02:27:43.000000000 -0400
+++ rplay-3.3.2/rplayd/Makefile.in	2011-12-25 22:51:37.270484658 -0500
@@ -9,6 +9,8 @@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS= @srcdir@/../mkinstalldirs
 
+DESTDIR=
+
 CPPFLAGS = $(CC_OPTIONS) -I. -I../include -I${srcdir} -I${srcdir}/../include \
 	-I${srcdir}/../lib -I${srcdir}/../adpcm \
 	@GSM_INCS@ \
@@ -52,12 +54,12 @@
 		(cd ../adpcm; $(MAKE) $(MFLAGS))
 
 install:	all
-		$(MKINSTALLDIRS) $(sbindir)
-		$(INSTALL_PROGRAM) $(TARGET) $(sbindir)/$(TARGET)
-		@test -x $(bindir)/$(TARGET) && echo "" && echo "    Note: Consider removing $(bindir)/$(TARGET)" && echo ""
+		$(MKINSTALLDIRS) $(DESTDIR)/$(sbindir)
+		$(INSTALL_PROGRAM) $(TARGET) $(DESTDIR)/$(sbindir)/$(TARGET)
+		@test -x $(DESTDIR)/$(bindir)/$(TARGET) && echo "" && echo "    Note: Consider removing $(DESTDIR)/$(bindir)/$(TARGET)" && echo ""
 
 uninstall:
-		$(RM) $(bindir)/$(TARGET)
+		$(RM) $(DESTDIR)/$(bindir)/$(TARGET)
 
 clean:
 		$(RM) $(OBJS) $(TARGET) a.out core *~ *.bak *.orig TAGS
diff -Naur rplay-3.3.2-orig/rplayd/rplayd.h rplay-3.3.2/rplayd/rplayd.h
--- rplay-3.3.2-orig/rplayd/rplayd.h	1999-03-10 02:58:04.000000000 -0500
+++ rplay-3.3.2/rplayd/rplayd.h	2011-12-25 22:49:10.967652476 -0500
@@ -28,6 +28,7 @@
 #include "config.h"
 #endif
 #include <sys/types.h>
+#include <errno.h>
 #include "rplay.h"
 #include "misc.h"
 #include "audio.h"
@@ -73,7 +74,6 @@
 extern fd_set read_mask;
 extern fd_set write_mask;
 extern int debug;
-extern int inetd;
 extern int rptp_timeout;
 extern char hostname[];
 extern char *hostaddr;
diff -Naur rplay-3.3.2-orig/rptp/Makefile.in rplay-3.3.2/rptp/Makefile.in
--- rplay-3.3.2-orig/rptp/Makefile.in	1999-06-09 02:27:48.000000000 -0400
+++ rplay-3.3.2/rptp/Makefile.in	2011-12-25 22:54:11.616505077 -0500
@@ -8,6 +8,8 @@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS= @srcdir@/../mkinstalldirs
 
+DESTDIR=
+
 CPPFLAGS=	$(CC_OPTIONS) -I. -I../include -I@srcdir@/../include -I@srcdir@/../lib @DEFS@
 
 .c.o:
@@ -33,11 +35,11 @@
 		(cd ../lib; $(MAKE) $(MFLAGS))
 
 install:	all
-		$(MKINSTALLDIRS) $(bindir)
-		$(INSTALL_PROGRAM) $(TARGET) $(bindir)/$(TARGET)
+		$(MKINSTALLDIRS) $(DESTDIR)/$(bindir)
+		$(INSTALL_PROGRAM) $(TARGET) $(DESTDIR)/$(bindir)/$(TARGET)
 
 uninstall:
-		$(RM) $(bindir)/$(TARGET)
+		$(RM) $(DESTDIR)/$(bindir)/$(TARGET)
 
 clean:
 		$(RM) $(OBJS) $(TARGET) a.out core *~ *.bak *.orig TAGS
diff -Naur rplay-3.3.2-orig/rx/Makefile.in rplay-3.3.2/rx/Makefile.in
--- rplay-3.3.2-orig/rx/Makefile.in	1998-07-14 18:35:25.000000000 -0400
+++ rplay-3.3.2/rx/Makefile.in	2011-12-25 23:00:33.819517890 -0500
@@ -29,6 +29,8 @@
 includedir = $(prefix)/include
 infodir = $(prefix)/info
 
+DESTDIR=
+
 #### Start of system configuration section. ####
 
 CC = @CC@
@@ -64,16 +66,16 @@
 all: librx.a
 
 install: all
-	test -d $(libdir) || mkdir $(libdir)
-	test -d $(includedir) || mkdir $(includedir)
-	$(INSTALL) librx.a $(libdir)/librx.a
-	$(RANLIB) $(libdir)/librx.a
-	$(INSTALL) $(srcdir)/inst-rxposix.h $(includedir)/rxposix.h
+	test -d $(DESTDIR)/$(libdir) || mkdir $(DESTDIR)/$(libdir)
+	test -d $(DESTDIR)/$(includedir) || mkdir $(DESTDIR)/$(includedir)
+	$(INSTALL) librx.a $(DESTDIR)/$(libdir)/librx.a
+	$(RANLIB) $(DESTDIR)/$(libdir)/librx.a
+	$(INSTALL) $(srcdir)/inst-rxposix.h $(DESTDIR)/$(includedir)/rxposix.h
 
 
 uninstall:
-	-rm -f $(libdir)/librx.a
-	-rm -f $(includedir)/rxposix.h
+	-rm -f $(DESTDIR)/$(libdir)/librx.a
+	-rm -f $(DESTDIR)/$(includedir)/rxposix.h
 
 clean:
 	-rm -f $(libobjs) librx.a  $(opt_objs) 
