#######################################################################
# Fix remote directory links.
# dave@slackbuilds.org
#######################################################################
diff -Naur -x .git bb/src/protocol/gopher/gopher.c cc/src/protocol/gopher/gopher.c
--- bb/src/protocol/gopher/gopher.c	2020-01-13 23:22:18.918292198 +0000
+++ cc/src/protocol/gopher/gopher.c	2020-01-13 23:21:00.631960039 +0000
@@ -321,6 +321,18 @@
 		selectorlen++;
 	}
 
+	if (entity_info->type == '1') {
+		if (strchr(selector, '1') == (char *) selector) {
+			selector++;
+			selectorlen--;
+		}
+		else {
+			if ((char *) selector  == "/1") {
+				selectorlen--;
+			}
+		}
+	}
+
 	state = add_gopher_command(conn, &command, entity, selector, selectorlen);
 	if (!is_in_state(state, S_CONN))
 		return state;
