diff -Nur culmus-0.130.orig/culmus.conf culmus-0.130/culmus.conf
--- culmus-0.130.orig/culmus.conf	2012-11-02 17:59:16.000000000 -0500
+++ culmus-0.130/culmus.conf	2014-12-27 12:44:44.674610931 -0600
@@ -17,6 +17,8 @@
 	<match target="pattern" >
 		<test name="family" >
 			<string>Guttman Adii</string>
+		</test>
+		<test name="family" >
 			<string>Guttman Adii-Light</string>
 		</test>
 		<edit mode="append" binding="same" name="family" >
@@ -39,6 +41,8 @@
         <match target="pattern" >
                 <test name="family" >
                         <string>Guttman Stam</string>
+                </test>
+                <test name="family" >
                         <string>Guttman Stam1</string>
                 </test>
                 <edit mode="append" binding="same" name="family" >
@@ -49,6 +53,8 @@
         <match target="pattern" >
                 <test name="family" >
                         <string>Guttman Hodes</string>
+                </test>
+                <test name="family" >
                         <string>Monotype Hadassah</string>
                 </test>
                 <edit mode="append" binding="same" name="family" >
@@ -59,7 +65,11 @@
         <match target="pattern" >
                 <test name="family" >
                         <string>Aharoni</string>
+                </test>
+                <test name="family" >
                         <string>Guttman Aharoni</string>
+                </test>
+                <test name="family" >
                         <string>Guttman-Aharoni</string>
                 </test>
                 <edit mode="append" binding="same" name="family" >
@@ -70,7 +80,11 @@
         <match target="pattern" >
                 <test name="family" >
                         <string>FrankRuehl</string>
+                </test>
+                <test name="family" >
                         <string>Guttman Frank</string>
+                </test>
+                <test name="family" >
                         <string>Guttman Frnew</string>
                 </test>
                 <edit mode="append" binding="same" name="family" >
@@ -81,7 +95,11 @@
         <match target="pattern" >
                 <test name="family" >
                         <string>David</string>
+                </test>
+                <test name="family" >
                         <string>David Transparent</string>
+                </test>
+                <test name="family" >
                         <string>Guttman David</string>
                 </test>
                 <edit mode="append" binding="same" name="family" >
@@ -92,8 +110,14 @@
         <match target="pattern" >
                 <test name="family" >
                         <string>Fixed Miriam Transparent</string>
+                </test>
+                <test name="family" >
                         <string>Guttman Myamfix</string>
+                </test>
+                <test name="family" >
                         <string>Guttman-CourMir</string>
+                </test>
+                <test name="family" >
                         <string>Miriam Fixed</string>
                 </test>
                 <edit mode="append" binding="same" name="family" >
@@ -104,8 +128,14 @@
         <match target="pattern" >
                 <test name="family" >
                         <string>Guttman Miryam</string>
+                </test>
+                <test name="family" >
                         <string>Levenim MT</string>
+                </test>
+                <test name="family" >
                         <string>Miriam</string>
+                </test>
+                <test name="family" >
                         <string>Miriam Transparent</string>
                 </test>
                 <edit mode="append" binding="same" name="family" >
@@ -116,6 +146,8 @@
         <match target="pattern" >
                 <test name="family" >
                         <string>Guttman Drogolin</string>
+                </test>
+                <test name="family" >
                         <string>Guttman Vilna</string>
                 </test>
                 <edit mode="append" binding="same" name="family" >
