diff -Nur SuperCollider-Source.orig//common/SConstruct SuperCollider-Source/common/SConstruct
--- SuperCollider-Source.orig//common/SConstruct	2010-03-11 14:11:21.000000000 -0600
+++ SuperCollider-Source/common/SConstruct	2010-09-12 03:10:12.839854106 -0500
@@ -153,7 +153,7 @@
     out.writelines([
         'prefix=%s\n' % env['PREFIX'],
         'exec_prefix=${prefix}\n',
-        'libdir=${exec_prefix}/lib\n',
+        'libdir=${exec_prefix}/lib64\n',
         'includedir=${prefix}/include/%s\n' % env['PACKAGE'],
         '\n'])
     write_field(out, 'PKGCONFIG_NAME', 'Name: ')
@@ -211,7 +211,7 @@
 def bin_dir(prefix):
     return os.path.join(prefix, 'bin')
 def lib_dir(prefix):
-    return os.path.join(prefix, 'lib')
+    return os.path.join(prefix, 'lib64')
 
 def pkg_data_dir(prefix, *args):
     if PLATFORM == 'darwin':
@@ -369,13 +369,13 @@
 
 # checks for DISTCC and CCACHE as used in modern linux-distros:
 
-if os.path.exists('/usr/lib/distcc/bin'):
-    os.environ['PATH']         = '/usr/lib/distcc/bin:' + os.environ['PATH']
+if os.path.exists('/usr/lib64/distcc/bin'):
+    os.environ['PATH']         = '/usr/lib64/distcc/bin:' + os.environ['PATH']
     #env['ENV']['DISTCC_HOSTS'] = os.environ['DISTCC_HOSTS']
     env['ENV']['DISTCC_HOSTS'] = os.environ.get('DISTCC_HOSTS')
 
-if os.path.exists('/usr/lib/ccache/bin'):
-    os.environ['PATH']         = '/usr/lib/ccache/bin:' + os.environ['PATH']
+if os.path.exists('/usr/lib64/ccache/bin'):
+    os.environ['PATH']         = '/usr/lib64/ccache/bin:' + os.environ['PATH']
     #env['ENV']['CCACHE_DIR']   = os.environ['CCACHE_DIR']
     env['ENV']['CCACHE_DIR']   = os.environ.get('CCACHE_DIR')
 
@@ -1011,7 +1011,7 @@
                 CPPPATH = ['#include/icu/unicode'])
 elif PLATFORM == 'linux':
     langEnv.Append(
-        LINKFLAGS = ['-Wl,-rpath,build', '-Wl,-rpath,' + FINAL_PREFIX + '/lib'])
+        LINKFLAGS = ['-Wl,-rpath,build', '-Wl,-rpath,' + FINAL_PREFIX + '/lib64'])
 
 
 elif PLATFORM == 'freebsd':
