diff -Nur SuperCollider-Source.orig/cmake_modules/FindFFTW3f.cmake SuperCollider-Source/cmake_modules/FindFFTW3f.cmake
--- SuperCollider-Source.orig/cmake_modules/FindFFTW3f.cmake	2012-12-07 10:38:09.000000000 -0600
+++ SuperCollider-Source/cmake_modules/FindFFTW3f.cmake	2013-02-17 19:25:40.179025178 -0600
@@ -11,9 +11,9 @@
 FIND_LIBRARY(
     FFTW3F_LIBRARY
     NAMES fftw3f libfftw3f fftw3f-3
-    HINTS $ENV{FFTW3_DIR}/lib
+    HINTS $ENV{FFTW3_DIR}/lib64
     PATHS /usr/local/lib
-          /usr/lib
+          /usr/lib64
 )
 
 SET(FFTW3F_FOUND "NO")
diff -Nur SuperCollider-Source.orig/cmake_modules/FindJack.cmake SuperCollider-Source/cmake_modules/FindJack.cmake
--- SuperCollider-Source.orig/cmake_modules/FindJack.cmake	2011-05-11 04:52:03.000000000 -0500
+++ SuperCollider-Source/cmake_modules/FindJack.cmake	2013-02-17 19:26:09.253703065 -0600
@@ -45,7 +45,7 @@
       jack
     PATHS
       ${_JACK_LIBDIR}
-      /usr/lib
+      /usr/lib64
       /usr/local/lib
       /opt/local/lib
       /sw/lib
diff -Nur SuperCollider-Source.orig/cmake_modules/FindPortaudio.cmake SuperCollider-Source/cmake_modules/FindPortaudio.cmake
--- SuperCollider-Source.orig/cmake_modules/FindPortaudio.cmake	2011-05-11 04:52:03.000000000 -0500
+++ SuperCollider-Source/cmake_modules/FindPortaudio.cmake	2013-02-17 19:26:45.942296599 -0600
@@ -53,7 +53,7 @@
       NAMES
         portaudio
       PATHS
-        /usr/lib
+        /usr/lib64
         /usr/local/lib
         /opt/local/lib
         /sw/lib
@@ -63,7 +63,7 @@
       NAMES
         portaudio
       PATHS
-        /usr/lib
+        /usr/lib64
         /usr/local/lib
         /opt/local/lib
         /sw/lib
diff -Nur SuperCollider-Source.orig/editors/sced/CMakeLists.txt SuperCollider-Source/editors/sced/CMakeLists.txt
--- SuperCollider-Source.orig/editors/sced/CMakeLists.txt	2012-12-07 10:38:09.000000000 -0600
+++ SuperCollider-Source/editors/sced/CMakeLists.txt	2013-02-17 19:27:46.569624922 -0600
@@ -14,10 +14,10 @@
             DESTINATION share/gtksourceview-2.0/language-specs)
 
     install(FILES data/sced.gedit-plugin
-            DESTINATION lib/gedit-2/plugins)
+            DESTINATION lib64/gedit-2/plugins)
 
     install(DIRECTORY sced
-            DESTINATION lib/gedit-2/plugins)
+            DESTINATION lib64/gedit-2/plugins)
 
 elseif(NOT WIN32 AND SC_ED EQUAL 3)
 
@@ -29,7 +29,7 @@
             DESTINATION share/gtksourceview-3.0/language-specs)
 
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/sced3/supercollider.plugin sced3/supercollider.py
-            DESTINATION lib/gedit/plugins)
+            DESTINATION lib64/gedit/plugins)
 
 elseif(WIN32)
 
diff -Nur SuperCollider-Source.orig/external_libraries/oscpack/Makefile SuperCollider-Source/external_libraries/oscpack/Makefile
--- SuperCollider-Source.orig/external_libraries/oscpack/Makefile	2011-05-11 04:52:03.000000000 -0500
+++ SuperCollider-Source/external_libraries/oscpack/Makefile	2013-02-17 19:29:13.509661735 -0600
@@ -71,12 +71,12 @@
 
 #Installs the library on a system global location
 install: lib
-	@$(INSTALL) -m 755 $(LIBFILENAME) $(PREFIX)/lib/$(LIBFILENAME)
-	@ln -s -f $(PREFIX)/lib/$(LIBFILENAME) $(PREFIX)/lib/$(LIBSONAME) 
+	@$(INSTALL) -m 755 $(LIBFILENAME) $(PREFIX)/lib64/$(LIBFILENAME)
+	@ln -s -f $(PREFIX)/lib64/$(LIBFILENAME) $(PREFIX)/lib64/$(LIBSONAME) 
 	@mkdir  -p $(PREFIX)/include/oscpack/ip $(PREFIX)/include/oscpack/osc
 	@$(INSTALL) -m 644 ip/*.h $(PREFIX)/include/oscpack/ip
 	@$(INSTALL) -m 644 osc/*.h $(PREFIX)/include/oscpack/osc
-	@echo "SUCCESS! oscpack has been installed in $(PREFIX)/lib and $(PREFIX)/include/ospack/"
+	@echo "SUCCESS! oscpack has been installed in $(PREFIX)/lib64 and $(PREFIX)/include/ospack/"
 	@echo "now doing ldconfig..."
 	@ldconfig
 
diff -Nur SuperCollider-Source.orig/server/plugins/CMakeLists.txt SuperCollider-Source/server/plugins/CMakeLists.txt
--- SuperCollider-Source.orig/server/plugins/CMakeLists.txt	2012-12-07 10:38:09.000000000 -0600
+++ SuperCollider-Source/server/plugins/CMakeLists.txt	2013-02-17 19:31:54.557877532 -0600
@@ -263,6 +263,6 @@
 			PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 else()
 	install(TARGETS ${plugins} ${supernova_plugins}
-			DESTINATION "lib/SuperCollider/plugins"
+			DESTINATION "lib64/SuperCollider/plugins"
 			PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 endif()
diff -Nur SuperCollider-Source.orig/server/scsynth/CMakeLists.txt SuperCollider-Source/server/scsynth/CMakeLists.txt
--- SuperCollider-Source.orig/server/scsynth/CMakeLists.txt	2012-12-18 06:03:55.000000000 -0600
+++ SuperCollider-Source/server/scsynth/CMakeLists.txt	2013-02-17 19:32:38.951385712 -0600
@@ -169,7 +169,7 @@
 endif(SNDFILE_FOUND)
 
 if(CMAKE_SYSTEM_NAME MATCHES "Linux" OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
-	add_definitions(-DSC_PLUGIN_DIR="${CMAKE_INSTALL_PREFIX}/lib/SuperCollider/plugins")
+	add_definitions(-DSC_PLUGIN_DIR="${CMAKE_INSTALL_PREFIX}/lib64/SuperCollider/plugins")
 endif()
 
 if (AUDIOAPI STREQUAL jack)
@@ -247,6 +247,6 @@
 else()
 	install(TARGETS ${INSTALL_TARGETS}
 			RUNTIME DESTINATION "bin"
-			LIBRARY DESTINATION "lib"
+			LIBRARY DESTINATION "lib64"
 			PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 endif()
diff -Nur SuperCollider-Source.orig/server/supernova/server/main.cpp SuperCollider-Source/server/supernova/server/main.cpp
--- SuperCollider-Source.orig/server/supernova/server/main.cpp	2012-12-18 06:03:55.000000000 -0600
+++ SuperCollider-Source/server/supernova/server/main.cpp	2013-02-17 19:33:08.666056514 -0600
@@ -196,7 +196,7 @@
 #ifdef __linux__
         path home = resolve_home();
         sc_factory->load_plugin_folder("/usr/local/lib/SuperCollider/plugins");
-        sc_factory->load_plugin_folder("/usr/lib/SuperCollider/plugins");
+        sc_factory->load_plugin_folder("/usr/lib64/SuperCollider/plugins");
         sc_factory->load_plugin_folder(home / "/.local/share/SuperCollider/plugins");
         sc_factory->load_plugin_folder(home / "share/SuperCollider/plugins");
 #else
