diff -Naur xmoto-0.5.0.orig/po/Makefile.in.in xmoto-0.5.0/po/Makefile.in.in
--- xmoto-0.5.0.orig/po/Makefile.in.in	2008-11-23 09:32:28.000000000 +0000
+++ xmoto-0.5.0/po/Makefile.in.in	2009-03-30 20:44:01.000000000 +0000
@@ -26,7 +26,7 @@
 datarootdir = @datarootdir@
 datadir = @datadir@
 localedir = @localedir@
-gettextsrcdir = $(datadir)/gettext/po
+gettextsrcdir = @localedir@/gettext/po
 
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
