diff -Nur qemu-kvm-1.0.orig/kvm/scripts/65-kvm.rules qemu-kvm-1.0/kvm/scripts/65-kvm.rules
--- qemu-kvm-1.0.orig/kvm/scripts/65-kvm.rules	2011-12-04 04:38:06.000000000 -0600
+++ qemu-kvm-1.0/kvm/scripts/65-kvm.rules	2012-02-21 19:22:48.449351958 -0600
@@ -1 +1,2 @@
-KERNEL=="kvm", NAME="%k", GROUP="kvm", MODE="0660"
+KERNEL=="kvm", GROUP="@GROUP@", MODE="0660"
+KERNEL=="vhost-net", GROUP="@GROUP@", MODE="0660"
