From 1aa9ea92e20142a1ce2f2e1ac40ed1fa781fac19 Mon Sep 17 00:00:00 2001
From: gabime <gmelman1@gmail.com>
Date: Tue, 3 Dec 2019 00:35:28 +0200
Subject: [PATCH] Fix #1340

---
 include/spdlog/details/circular_q.h            | 1 +
 1 files changed, 1 insertions(+)

diff --git a/include/spdlog/details/circular_q.h b/include/spdlog/details/circular_q.h
index 325dbfea..1f2712e7 100644
--- a/include/spdlog/details/circular_q.h
+++ b/include/spdlog/details/circular_q.h
@@ -5,6 +5,7 @@
 #pragma once
 
 #include <vector>
+#include <cassert>
 
 namespace spdlog {
 namespace details {
