diff -Naur darcs-2.16.5.orig/darcs.cabal darcs-2.16.5/darcs.cabal
--- darcs-2.16.5.orig/darcs.cabal	2001-09-09 13:46:40.000000000 +1200
+++ darcs-2.16.5/darcs.cabal	2022-02-27 14:54:23.056244490 +1300
@@ -436,10 +436,10 @@
                       vector            >= 0.11 && < 0.13,
                       tar               >= 0.5 && < 0.6,
                       data-ordlist      == 0.4.*,
-                      attoparsec        >= 0.13.0.1 && < 0.14,
+                      attoparsec        >= 0.13.0.1 && < 0.15,
                       zip-archive       >= 0.3 && < 0.5,
                       async             >= 2.0.2 && < 2.3,
-                      constraints       >= 0.11 && < 0.13,
+                      constraints       >= 0.11 && < 0.14,
                       unix-compat       >= 0.5 && < 0.6,
                       bytestring        >= 0.10.6 && < 0.11,
                       old-time          >= 1.1.0.3 && < 1.2,
@@ -449,7 +449,7 @@
                       temporary         >= 1.2.1 && < 1.4,
                       process           >= 1.2.3.0 && < 1.7,
                       array             >= 0.5.1.0 && < 0.6,
-                      hashable          >= 1.2.3.3 && < 1.4,
+                      hashable          >= 1.2.3.3 && < 1.5,
                       mmap              >= 0.5.9 && < 0.6,
                       zlib              >= 0.6.1.2 && < 0.7.0.0,
                       network-uri       >= 2.6 && < 2.8,
