--- darcs.cabal.orig	2011-02-11 23:36:22.000000000 +0200
+++ darcs.cabal	2011-02-11 23:50:05.000000000 +0200
@@ -84,6 +84,10 @@
   description: Build static binary
   default:     False
 
+flag dynamic
+  description: Link darcs executable dynamically
+  default:     False
+
 flag terminfo
   description: Use the terminfo package for enhanced console support.
 
@@ -499,6 +503,8 @@
 
   if flag(static)
     ghc-options: -static -optl-static -optl-pthread
+  if flag(dynamic)
+    ghc-options: -dynamic
 
   if flag(hpc)
     ghc-prof-options: -fhpc
