--- darcs.cabal.orig	2011-05-31 11:46:05.441999585 +0300
+++ darcs.cabal	2011-05-31 11:47:59.589999593 +0300
@@ -170,11 +170,11 @@
       cpp-options:    -DWIN32
       c-sources:      src/win32/send_email.c
       build-depends:  unix-compat >= 0.1.2,
-                      regex-posix >= 0.94.4 && < 0.95
+                      regex-posix >= 0.94.4 && <= 0.95.1
 
     build-depends:   base          < 5,
                      extensible-exceptions >= 0.1 && < 0.2,
-                     regex-compat >= 0.71 && < 0.94,
+                     regex-compat >= 0.71 && <= 0.95.1,
                      mtl          >= 1.0 && < 2.1,
                      parsec       >= 2.0 && < 3.2,
                      html         == 1.0.*,
@@ -372,14 +372,14 @@
       cpp-options:    -DWIN32
       c-sources:      src/win32/send_email.c
       build-depends:  unix-compat >= 0.1.2,
-                      regex-posix >= 0.94.4 && < 0.95
+                      regex-posix >= 0.94.4 && <= 0.95.1
 
     if os(solaris)
       cc-options:     -DHAVE_SIGINFO_H
 
     build-depends:   base          < 5,
                      extensible-exceptions >= 0.1 && < 0.2,
-                     regex-compat >= 0.71 && < 0.94,
+                     regex-compat >= 0.71 && <= 0.95.1,
                      mtl          >= 1.0 && < 2.1,
                      parsec       >= 2.0 && < 3.2,
                      html         == 1.0.*,
@@ -522,14 +522,14 @@
     cpp-options:    -DWIN32
     c-sources:      src/win32/send_email.c
     build-depends:  unix-compat >= 0.1.2,
-                    regex-posix >= 0.94.4 && < 0.95
+                    regex-posix >= 0.94.4 && <= 0.95.1
 
   if os(solaris)
     cc-options:     -DHAVE_SIGINFO_H
 
   build-depends:   base          < 5,
                    extensible-exceptions >= 0.1 && < 0.2,
-                   regex-compat >= 0.71 && < 0.94,
+                   regex-compat >= 0.71 && <= 0.95.1,
                    mtl          >= 1.0 && < 2.1,
                    parsec       >= 2.0 && < 3.2,
                    html         == 1.0.*,
@@ -625,7 +625,7 @@
 
     build-depends:   base          < 5,
                      extensible-exceptions >= 0.1 && < 0.2,
-                     regex-compat >= 0.71 && < 0.94,
+                     regex-compat >= 0.71 && <= 0.95.1,
                      mtl          >= 1.0 && < 2.1,
                      parsec       >= 2.0 && < 3.2,
                      html         == 1.0.*,
@@ -678,7 +678,7 @@
       cpp-options:    -DWIN32
       c-sources:      src/win32/send_email.c
       build-depends:  unix-compat >= 0.1.2,
-                      regex-posix >= 0.94.4 && < 0.95
+                      regex-posix >= 0.94.4 && <= 0.95.1
 
     if os(solaris)
       cc-options:     -DHAVE_SIGINFO_H
