--- config.h.in
+++ config.h.in
@@ -50,1 +50,1 @@
-#define DFM_IMG_CHAFA_CMD "chafa", "--format=sixel", "-s"
+#define DFM_IMG_CHAFA_CMD "chafa", "--format=sixels", "--animate=off", "--colors=256", "--dither=ordered", "--margin-bottom=1", "--margin-right=1", "-s"
--- dfm.c
+++ dfm.c
@@ -3091,1 +3091,1 @@
-  str_push_u32(&p->io, p->col);
+  str_push_u32(&p->io, MIN(p->col, 100));
@@ -3093,1 +3093,1 @@
-  str_push_u32(&p->io, p->row - 1);
+  str_push_u32(&p->io, MIN(p->row - 1, 20));
