diff --git a/icons/meson.build b/icons/meson.build
index 370428a5f1d1035a336c83cad1d161281f1be61a..8d588865bdcf8f32fca12d2b918559e42b5e2d8e 100644
--- a/icons/meson.build
+++ b/icons/meson.build
@@ -15,7 +15,7 @@ foreach icon : public_icons
     file = join_paths('maki' , icon)
     install_dir = join_paths(get_option('datadir'),
                              'icons',
-                             'gnome',
+                             'hicolor',
                              'scalable',
                              'places')
 
