diff -Naur dmenu-4.8-orig/config.mk dmenu-4.8/config.mk
--- dmenu-4.8-orig/config.mk	2018-03-14 13:26:40.000000000 -0600
+++ dmenu-4.8/config.mk	2018-04-01 18:03:50.431134881 -0600
@@ -4,6 +4,7 @@
 # paths
 PREFIX = /usr/local
 MANPREFIX = ${PREFIX}/share/man
+LIBDIR = /usr/local/lib
 
 X11INC = /usr/X11R6/include
 X11LIB = /usr/X11R6/lib
@@ -20,11 +21,12 @@
 
 # includes and libs
 INCS = -I${X11INC} -I${FREETYPEINC}
-LIBS = -L${X11LIB} -lX11 ${XINERAMALIBS} ${FREETYPELIBS}
+LIBS = -L${LIBDIR} -L${X11LIB} -lX11 ${XINERAMALIBS} ${FREETYPELIBS}
 
 # flags
+OPTS ?= -Os
 CPPFLAGS = -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE=700 -D_POSIX_C_SOURCE=200809L -DVERSION=\"${VERSION}\" ${XINERAMAFLAGS}
-CFLAGS   = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
+CFLAGS   = -std=c99 -pedantic -Wall ${OPTS} ${INCS} ${CPPFLAGS}
 LDFLAGS  = -s ${LIBS}
 
 # compiler and linker
