diff -Nur dmenu-4.0.orig/config.mk dmenu-4.0/config.mk
--- dmenu-4.0.orig/config.mk	2009-04-18 06:50:04.000000000 -0500
+++ dmenu-4.0/config.mk	2009-08-03 00:32:57.258455322 -0500
@@ -6,6 +6,7 @@
 # paths
 PREFIX = /usr/local
 MANPREFIX = ${PREFIX}/share/man
+LIBDIR = /usr/local/lib
 
 X11INC = /usr/X11R6/include
 X11LIB = /usr/X11R6/lib
@@ -16,11 +17,12 @@
 
 # includes and libs
 INCS = -I. -I/usr/include -I${X11INC}
-LIBS = -L/usr/lib -lc -L${X11LIB} -lX11 ${XINERAMALIBS}
+LIBS = -L${LIBDIR} -lc -L${X11LIB} -lX11 ${XINERAMALIBS}
 
 # flags
+OPTS ?= -Os
 CPPFLAGS = -D_BSD_SOURCE -DVERSION=\"${VERSION}\" ${XINERAMAFLAGS}
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
+CFLAGS = -std=c99 -pedantic -Wall ${OPTS} ${INCS} ${CPPFLAGS}
 LDFLAGS = -s ${LIBS}
 
 # Solaris
