diff -Naur dmenu-4.5-orig/config.mk dmenu-4.5/config.mk
--- dmenu-4.5-orig/config.mk	2012-01-08 07:18:43.000000000 -0500
+++ dmenu-4.5/config.mk	2012-03-03 13:13:44.000000000 -0500
@@ -4,6 +4,7 @@
 # paths
 PREFIX = /usr/local
 MANPREFIX = ${PREFIX}/share/man
+LIBDIR = /usr/local/lib
 
 X11INC = /usr/X11R6/include
 X11LIB = /usr/X11R6/lib
@@ -14,12 +15,13 @@
 
 # includes and libs
 INCS = -I${X11INC}
-LIBS = -L${X11LIB} -lX11 ${XINERAMALIBS}
+LIBS = -L${LIBDIR} -L${X11LIB} -lX11 ${XINERAMALIBS}
 
 # flags
+OPTS ?= -Os
 CPPFLAGS = -D_BSD_SOURCE -D_POSIX_C_SOURCE=2 -DVERSION=\"${VERSION}\" ${XINERAMAFLAGS}
 #CFLAGS   = -g -std=c99 -pedantic -Wall -O0 ${INCS} ${CPPFLAGS}
-CFLAGS   = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
+CFLAGS   = -std=c99 -pedantic -Wall ${OPTS} ${INCS} ${CPPFLAGS}
 LDFLAGS  = -s ${LIBS}
 
 # compiler and linker
