diff -Naur syncterm-1.1/src/syncterm/syncterm.man.in syncterm-1.1.patched/src/syncterm/syncterm.man.in
--- syncterm-1.1/src/syncterm/syncterm.man.in	2020-06-26 19:40:21.000000000 -0400
+++ syncterm-1.1.patched/src/syncterm/syncterm.man.in	2023-07-27 16:48:25.575523339 -0400
@@ -1,11 +1,14 @@
-.TH SYNCTERM 1 "27 May 2007"
+.TH SYNCTERM 1 "27 Jul 2023"
 .SH NAME
 syncterm \- BBS terminal program
 .SH SYNOPSIS
 \fBsyncterm\fP [\-e\fIdly\fP] [\-i\fIifc\fP] [\-l\fIlns\fP] [\-t|\-r|\-h] [\-s] [\fIURL\fP]
 .SH DESCRIPTION
 \fBSyncTERM\fP is a terminal program designed specifically for connecting to
-a Bulletin Board System (\fBBBS\fP)
+a Bulletin Board System (\fBBBS\fP).
+.LP
+[\fIURL\fP] may be an actual URL, a hostname, or the name of
+an entry from the dialing list.
 .SS Options
 .IP \fB\-e\fIdly\fR
 Specify the "escape delay" in milliseconds for curses(3) mode. Since curses
@@ -96,7 +99,7 @@
 Specifies the default scheme for \fIURL\fP to be telnet://
 .IP \fB\-r\fP
 Specifies the default scheme for \fIURL\fP to be rlogin://
-.IP \fB\-r\fP
+.IP \fB\-h\fP
 Specifies the default scheme for \fIURL\fP to be ssh://
 .IP \fB\-4\fP
 Only resolve IPv4 addresses
@@ -112,7 +115,7 @@
 run-time linking is employed using dlopen(). Using this, it is possible to
 build a version which will use SDL, and X11 if they are present, but
 not fail to run if they are not.
-.SH TEXT MODE CAVAETS
+.SH TEXT MODE CAVEATS
 When running in one of the text modes such as curses or ANSI, many of the
 control functions are modified since there is no reliable way to use extra
 modifier keys. As a result, CTRL-S and CTRL-Q can not be sent to the remote
@@ -130,12 +133,12 @@
 a traditional screen mode. As such, filler will be added around the terminal
 screen. For some display sizes, unpredictable results may occur.
 .SH FILES
-.IP \fC%%DESTDIR%%%%PREFIX%%/etc/syncterm.lst\fR
-The system wide dialing directory.
+.IP /etc/syncterm.lst\fR
+The system wide dialing directory [note: doesn't seem to be used in SyncTERM 1.1].
 .IP \fC~/.syncterm/syncterm.lst\fR
-The users dialing directory
+The user's dialing directory
 .IP \fC~/.syncterm/syncterm.ini\fR
-The users SyncTERM configuration file
+The user's SyncTERM configuration file
 .IP \fClibSDL.so\fR
 When this file exists in the library path, SyncTERM will support the use of SDL
 for input and output. This file is part of the Simple Directmedia Layer
