diff -Naur syncterm-1.3/src/syncterm/syncterm.man.in syncterm-1.3.patched/src/syncterm/syncterm.man.in
--- syncterm-1.3/src/syncterm/syncterm.man.in	2024-11-25 00:31:08.000000000 -0500
+++ syncterm-1.3.patched/src/syncterm/syncterm.man.in	2024-12-03 16:59:55.248149405 -0500
@@ -1,4 +1,4 @@
-.TH SYNCTERM 1 2024-09-24
+.TH SYNCTERM 1 2024-12-03
 .SH NAME
 syncterm \- BBS terminal program
 .SH SYNOPSIS
@@ -10,7 +10,10 @@
 \fBsyncterm\fP -v\fP
 .SH DESCRIPTION
 \fBSyncTERM\fP is a terminal program designed specifically for connecting to
-a Bulletin Board System (\fBBBS\fP)
+a Bulletin Board System (\fBBBS\fP).
+.LP
+[\fIURL\fP] may be an actual URL, a hostname, or the name of
+an entry from the dialing list.
 .SS Options
 .IP \fB\-4\fP
 Only resolve IPv4 addresses
@@ -101,7 +104,7 @@
 run-time linking is employed using dlopen().
 Using this, it is possible to build a version which will use SDL,
 and X11 if they are present, but not fail to run if they are not.
-.SH TEXT MODE CAVAETS
+.SH TEXT MODE CAVEATS
 When running in one of the text modes such as curses or ANSI, many of the
 control functions are modified since there is no reliable way to use extra
 modifier keys. As a result, CTRL-S and CTRL-Q can not be sent to the remote
@@ -119,14 +122,14 @@
 a traditional screen mode. As such, filler will be added around the terminal
 screen. For some display sizes, unpredictable results may occur.
 .SH FILES
-.IP \fI%%DESTDIR%%%%PREFIX%%/etc/syncterm.lst\fR
+.IP \fI/etc/syncterm.lst\fR
 The system wide dialing directory.
 .IP \fIsyncterm.lst\fR
-The users dialing directory.
+The user's dialing directory.
 The location of this file varies by OS and can be viewed in SyncTERM in the
 "File Locations" option of the "SyncTERM Settings" menu.
 .IP \fIsyncterm.ini\fR
-The users SyncTERM configuration file.
+The user's SyncTERM configuration file.
 The location of this file varies by OS and can be viewed in SyncTERM in the
 "File Locations" option of the "SyncTERM Settings" menu.
 .SH AUTHOR
