diff -Naur lumina-pdf/src-qt5/Renderer-poppler.cpp lumina-pdf-patched/src-qt5/Renderer-poppler.cpp
--- lumina-pdf/src-qt5/Renderer-poppler.cpp	2019-08-21 11:46:01.653515619 +0000
+++ lumina-pdf-patched/src-qt5/Renderer-poppler.cpp	2019-08-23 10:54:36.334600721 +0000
@@ -70,7 +70,7 @@
     } // invalid password
   }
   // qDebug() << "Opening File:" << path;
-  doctitle = DOC->title();
+  // doctitle = DOC->title();
   if (doctitle.isEmpty()) {
     doctitle = path.section("/", -1);
   }
diff -Naur lumina-pdf/src-qt5/drawablepage.h lumina-pdf-patched/src-qt5/drawablepage.h
--- lumina-pdf/src-qt5/drawablepage.h	2019-08-21 11:46:01.653515619 +0000
+++ lumina-pdf-patched/src-qt5/drawablepage.h	2019-08-23 10:48:00.544733374 +0000
@@ -4,6 +4,7 @@
 #include <QImage>
 #include <QSize>
 #include <poppler/qt5/poppler-qt5.h>
+#include <memory>
 
 namespace LuminaPDF {
 
