diff -ur sqlpp11-connector-mysql-0.20.old/CMakeLists.txt sqlpp11-connector-mysql-0.20/CMakeLists.txt
--- sqlpp11-connector-mysql-0.20.old/CMakeLists.txt	2016-09-03 08:51:52.000000000 -0300
+++ sqlpp11-connector-mysql-0.20/CMakeLists.txt	2016-12-15 20:50:56.302051737 -0200
@@ -72,7 +72,6 @@
 include_directories("${DATE_INCLUDE_DIR}")
 include_directories("${MYSQL_INCLUDE_DIRS}")
 set(include_dir "${PROJECT_SOURCE_DIR}/include")
-file(GLOB_RECURSE sqlpp_headers ${include_dir}/*.h ${SQLPP11_INCLUDE_DIR}/*.h)
 include_directories(${include_dir})
 
 add_subdirectory(src)
Only in sqlpp11-connector-mysql-0.20: build
diff -ur sqlpp11-connector-mysql-0.20.old/src/CMakeLists.txt sqlpp11-connector-mysql-0.20/src/CMakeLists.txt
--- sqlpp11-connector-mysql-0.20.old/src/CMakeLists.txt	2016-09-03 08:51:52.000000000 -0300
+++ sqlpp11-connector-mysql-0.20/src/CMakeLists.txt	2016-12-15 20:52:25.970056347 -0200
@@ -9,4 +9,4 @@
 	target_link_libraries(sqlpp-mysql "${Boost_THREAD_LIBRARY}")
 endif()
 
-install(TARGETS sqlpp-mysql DESTINATION lib)
+install(TARGETS sqlpp-mysql DESTINATION lib${LIB_SUFFIX})
