diff -Naur basilisk-2021.11.14-source_orig/platform/gfx/2d/BaseRect.h basilisk-2021.11.14-source/platform/gfx/2d/BaseRect.h
--- basilisk-2021.11.14-source_orig/platform/gfx/2d/BaseRect.h	2021-11-12 17:10:16.000000000 -0500
+++ basilisk-2021.11.14-source/platform/gfx/2d/BaseRect.h	2021-11-15 17:32:59.411746189 -0500
@@ -9,6 +9,7 @@
 #include <algorithm>
 #include <cmath>
 #include <ostream>
+#include <limits>
 
 #include "mozilla/Assertions.h"
 #include "mozilla/FloatingPoint.h"
diff -Naur basilisk-2021.11.14-source_orig/platform/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/opus/audio_encoder_opus.cc basilisk-2021.11.14-source/platform/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/opus/audio_encoder_opus.cc
--- basilisk-2021.11.14-source_orig/platform/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/opus/audio_encoder_opus.cc	2021-11-12 17:10:36.000000000 -0500
+++ basilisk-2021.11.14-source/platform/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/opus/audio_encoder_opus.cc	2021-11-15 17:35:40.787737049 -0500
@@ -7,7 +7,7 @@
  *  in the file PATENTS.  All contributing project authors may
  *  be found in the AUTHORS file in the root of the source tree.
  */
-
+#include <limits>
 #include "webrtc/modules/audio_coding/codecs/opus/interface/audio_encoder_opus.h"
 
 #include "webrtc/base/checks.h"
diff -Naur basilisk-2021.11.14-source_orig/platform/netwerk/base/nsURLParsers.cpp basilisk-2021.11.14-source/platform/netwerk/base/nsURLParsers.cpp
--- basilisk-2021.11.14-source_orig/platform/netwerk/base/nsURLParsers.cpp	2021-11-12 17:10:38.000000000 -0500
+++ basilisk-2021.11.14-source/platform/netwerk/base/nsURLParsers.cpp	2021-11-15 17:33:50.427743299 -0500
@@ -4,6 +4,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include <string.h>
+#include <limits>
 
 #include "mozilla/RangedPtr.h"
 
