--- a/palemoon-31.3.0.1_Release/platform/ipc/chromium/src/third_party/libevent/arc4random.c	2022-09-27 20:12:40.000000000 -0000
+++ b/palemoon-31.3.0.1_Release/platform/ipc/chromium/src/third_party/libevent/arc4random.c	2022-10-03 10:09:16.859171354 -0000
@@ -481,7 +481,8 @@
 }
 #endif
 
-#ifndef HAVE_ARC4RANDOM_BUF
+#if defined(_we_have_arc4random_buf) || !defined(XP_SOLARIS)
+#if (__GLIBC__ <= 2 && __GLIBC_MINOR__ < 36) || !defined(XP_LINUX)
 ARC4RANDOM_EXPORT void
 arc4random_buf(void *_buf, size_t n)
 {
@@ -496,6 +497,7 @@
 	_ARC4_UNLOCK();
 }
 #endif
+#endif
 
 #ifndef ARC4RANDOM_NOUNIFORM
 /*
