--- grun.c.orig	2026-07-28 11:53:02.379230427 -0400
+++ grun.c	2026-07-28 11:55:11.414133311 -0400
@@ -1036,7 +1036,7 @@
 	gtk_window_set_policy(GTK_WINDOW (win), FALSE, FALSE, FALSE);
 	gtk_widget_realize(win);
 	
-	gtk_window_set_icon(GTK_WINDOW(win), gdk_pixbuf_new_from_xpm_data(grun2));
+	gtk_window_set_icon(GTK_WINDOW(win), gdk_pixbuf_new_from_xpm_data((const char **) grun2));
 
 	gtk_widget_show_all(win);
 	gtk_main();
