diff -Naur uget-1.6.4/uglib/UgPlugin-curl.c uget-1.6.4.patched//uglib/UgPlugin-curl.c
--- uget-1.6.4/uglib/UgPlugin-curl.c	2011-03-27 22:10:11.000000000 -0400
+++ uget-1.6.4.patched//uglib/UgPlugin-curl.c	2011-05-21 16:45:24.000000000 -0400
@@ -280,6 +280,7 @@
 	curl_code = CURLE_FAILED_INIT;
 
 	// Common option ----------------------------------------------------------
+	curl_easy_setopt (curl, CURLOPT_NOSIGNAL , 1);
 	curl_easy_setopt (curl, CURLOPT_URL, common->url);
 	// setup scheme and it's related data (user & password)
 	ug_plugin_curl_set_scheme (plugin, curl);
