diff -Naur uget-1.8.0/uglib/UgXmlrpc.c uget-1.8.0.patched//uglib/UgXmlrpc.c
--- uget-1.8.0/uglib/UgXmlrpc.c	2011-04-28 21:34:25.000000000 -0400
+++ uget-1.8.0.patched//uglib/UgXmlrpc.c	2012-01-02 15:00:39.000000000 -0500
@@ -111,6 +111,7 @@
 	xmlrpc->user_agent = g_strdup (user_agent ? user_agent : "uGet/1.7");
 
 	curl_easy_setopt (xmlrpc->curl, CURLOPT_URL, xmlrpc->uri);
+	curl_easy_setopt (xmlrpc->curl, CURLOPT_NOSIGNAL , 1);
 	curl_easy_setopt (xmlrpc->curl, CURLOPT_USERAGENT, xmlrpc->user_agent);
 }
 
