diff -Naur acme-20250126_r434/src/platform.c acme-20250126_r434.patched/src/platform.c
--- acme-20250126_r434/src/platform.c	2025-02-17 18:03:46.000000000 -0500
+++ acme-20250126_r434.patched/src/platform.c	2025-02-17 18:26:16.166315740 -0500
@@ -45,6 +45,7 @@
 
 	// Find out the path of ACME's library
 	env_var = getenv("ACME");
+	if(!env_var) env_var = "/usr/share/acme/ACME_Lib";
 	// if environment variable was found, make a copy
 	if (env_var) {
 		dynabuf_clear(GlobalDynaBuf);
