--- Makefile	2006-12-16 11:13:34.000000000 -0600
+++ Makefile.no-gvfs	2007-01-01 20:02:01.000000000 -0600
@@ -15,9 +15,9 @@
 archive=atol-$(VERSION)_src
 
 #define plugin install path
-plugin_dir=$(DESTDIR)$(prefix)/local/lib/atol
+plugin_dir=$(DESTDIR)$(prefix)/lib/atol
 #comment this to remove gnome-vfs support
-HAVE_GNOME_VFS=1
+#HAVE_GNOME_VFS=1
 
 # To profile Atol:
 # 1. uncomment "PROFILE=..." line below 
@@ -32,7 +32,7 @@
 #DEBUG=-g -D_DEBUG
 
 CXX=g++ 
-CFLAGS=-Wall 
+CFLAGS+= -Wall 
 FLAGS=$(CFLAGS)
 BIN=./bin
 OBJ=./bin
