--- openclonk-5.5.1/CMakeLists.orig	2015-02-20 12:19:12.760304874 +0200
+++ openclonk-5.5.1/CMakeLists.txt	2015-02-20 12:20:39.503220438 +0200
@@ -1347,7 +1347,7 @@
 endif()
 
 if(USE_BOOST_REGEX)
-	SET(Boost_USE_STATIC_LIBS ${USE_STATIC_BOOST})
+	SET(Boost_USE_STATIC_LIBS/usr/include/boost ${USE_STATIC_BOOST})
 	find_package(Boost 1.40.0 REQUIRED COMPONENTS regex)
 	# Disable automatic linking, we'll do it ourselves
 	add_definitions(-DBOOST_REGEX_NO_LIB)
