--- openclonk-5.4.1-src/CMakeLists.txt.original	2014-03-17 19:40:34.633722226 +0200
+++ openclonk-5.4.1-src/CMakeLists.txt	2014-03-17 19:41:33.135990847 +0200
@@ -1425,7 +1425,7 @@
 endif()
 
 if(USE_BOOST_REGEX)
-	SET(Boost_USE_STATIC_LIBS ${USE_STATIC_BOOST})
+	SET(Boost_USE_STATIC_LIBS/usr/include/boost ${USE_STATIC_BOOST})
 	find_package(Boost 1.40.0 REQUIRED COMPONENTS regex)
 	# Disable automatic linking, we'll do it ourselves
 	add_definitions(-DBOOST_REGEX_NO_LIB)
