--- dao-2.0-devel-2014-01-19/CMakeLists.original.txt	2014-04-03 12:51:50.845397447 +0300
+++ dao-2.0-devel-2014-01-19/CMakeLists.txt	2014-04-03 12:59:56.966320100 +0300
@@ -89,12 +89,12 @@
 endif(DAO_HAS_FENV)
 
 # Check whether the system has readline.h or not.
-find_path(DAO_HAS_READLINE NAMES "readline/readline.h" "readline/history.h")
-if(DAO_HAS_READLINE)
-    include_directories("${DAO_HAS_READLINE}")
-    add_definitions(-DDAO_USE_READLINE)
-    list(APPEND DAO_SYSTEM_LIBS "readline")
-endif(DAO_HAS_READLINE)
+# find_path(DAO_HAS_READLINE NAMES "readline/readline.h" "readline/history.h")
+# if(DAO_HAS_READLINE)
+#    include_directories("${DAO_HAS_READLINE}")
+#    add_definitions(-DDAO_USE_READLINE)
+#    list(APPEND DAO_SYSTEM_LIBS "readline")
+# endif(DAO_HAS_READLINE)
 
 # Setup for Minix:
 if( "${CMAKE_SYSTEM_NAME}" STREQUAL "Minix" )
@@ -194,13 +194,13 @@
 set_target_properties(daoexe PROPERTIES OUTPUT_NAME ${DAO_EXE_NAME})
 
 # Installation rules for the interpreter and dao libs
-set(DAO_MODULES_INSTALL_DIR "lib/dao/modules")
+set(DAO_MODULES_INSTALL_DIR "lib64/dao/modules")
 install(TARGETS daoexe daolib daolibs
         RUNTIME DESTINATION "bin"
-        LIBRARY DESTINATION "lib"
-        ARCHIVE DESTINATION "lib"
-        PRIVATE_HEADER DESTINATION "include"
-        PUBLIC_HEADER  DESTINATION "include"
+        LIBRARY DESTINATION "lib64"
+        ARCHIVE DESTINATION "lib64"
+        PRIVATE_HEADER DESTINATION "include/dao"
+        PUBLIC_HEADER  DESTINATION "include/dao"
        )
 #autobind.dao is obsolete:
 #install(FILES "${PROJECT_SOURCE_DIR}/tools/autobind.dao" DESTINATION "bin")
