--- configure.orig	2015-07-12 10:56:29.415500175 -0700
+++ configure	2015-07-12 10:56:57.546148494 -0700
@@ -110,10 +110,10 @@
 	
 	puts "Testing support for the graphical frontend..."
 	begin
-		require 'gtk2'
-		puts "ruby-gtk2 bindings found"
+		require 'gtk2/base'
+		puts "rubygem-gtk2 bindings found"
 	rescue LoadError
-		puts "ruby-gtk2 is not found. The graphical frontend won't work!"
+		puts "rubygem-gtk2 is not found. The graphical frontend won't work!"
 	end
 	
 	puts "\nTesting support for freedb metadata fetching..."
