*** timeshift-gtk.desktop	2022-11-22 10:11:02.000000000 +0000
--- timeshift-gtk.desktop.new	2022-12-15 21:45:34.173993274 +0000
***************
*** 1,6 ****
  [Desktop Entry]
  Name=Timeshift
- MimeType=
  Exec=timeshift-launcher
  Type=Application
  GenericName=System Restore Utility
--- 1,6 ----
  [Desktop Entry]
+ Type=Application
  Name=Timeshift
  Exec=timeshift-launcher
  Type=Application
  GenericName=System Restore Utility
*************** GenericName[nl]=Hulpmiddel voor systeemh
*** 16,32 ****
  GenericName[ru]=Программа для восстановления системы
  Terminal=false
  Icon=timeshift
! Comment=System Restore Utility
! Comment[ca]=Utilitat de restauració del sistema
! Comment[cs]=Nástroj pro obnovení systému
! Comment[da]=Værktøj til systemgendannelse
! Comment[fr]=Utilitaire de restauration système
! Comment[hr]=Alat obnove sustava
! Comment[ja]=システムの復元ユーティリティ
! Comment[ko]=시스템 복원 유틸리티
! Comment[lt]=Sistemos atkūrimo paslaugų programa
! Comment[nl]=Hulpmiddel voor systeemherstel
! Comment[ru]=Программа для восстановления системы
  X-KDE-StartupNotify=false
  Categories=System;
  X-GNOME-UsesNotifications=true
--- 16,32 ----
  GenericName[ru]=Программа для восстановления системы
  Terminal=false
  Icon=timeshift
! Comment=Timeshift restores your system to an earlier backup
! Comment[ca]=Timeshift restaura el sistema a una còpia de seguretat anterior
! Comment[cs]=Timeshift obnoví váš systém do dřívější zálohy
! Comment[da]=Timeshift gendanner dit system til en tidligere sikkerhedskopi
! Comment[fr]=Timeshift restaure votre système à une sauvegarde antérieure
! Comment[hr]=Timeshift vraća vaš sustav na raniju sigurnosnu kopiju
! Comment[ja]=Timeshift はシステムを以前のバックアップに復元します
! Comment[ko]=Timeshift는 시스템을 이전 백업으로 복원합니다.
! Comment[lt]=Timeshift atkuria jūsų sistemos ankstesnę atsarginę kopiją
! Comment[nl]=Timeshift herstelt uw systeem naar een eerdere back-up
! Comment[ru]=Timeshift восстанавливает вашу систему до более ранней резервной копии
  X-KDE-StartupNotify=false
  Categories=System;
  X-GNOME-UsesNotifications=true
