--- data/crack-attack.desktop.orig   2007-06-19 14:09:14.000000000 -0700
+++ data/crack-attack.desktop        2007-06-19 14:10:47.000000000 -0700
@@ -1,9 +1,10 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Crack Attack!
+GenericName=Fast-paced Puzzle Game
 Exec=crack-attack
 Icon=crack-attack.xpm
 Terminal=false
 Type=Application
-Categories=Application;Games;
+Categories=Application;Game;StrategyGame
 StartupNotify=false
