diff --git a/src/ActionRecorder.cxx b/src/ActionRecorder.cxx
index f68e674..3312523 100644
--- a/src/ActionRecorder.cxx
+++ b/src/ActionRecorder.cxx
@@ -21,6 +21,7 @@
  */
 
 #include <fstream>
+#include <cstring>
 #include "TextureLoader.h"
 
 using namespace std;
diff --git a/src/Attack.cxx b/src/Attack.cxx
index 594ef83..1b58f6a 100644
--- a/src/Attack.cxx
+++ b/src/Attack.cxx
@@ -27,6 +27,7 @@
 #include <cstring>
 #include <cctype>
 #include <sys/stat.h>
+#include <unistd.h>
 
 #ifndef _WIN32
 #  include <pwd.h>
diff --git a/src/Game.cxx b/src/Game.cxx
index be77c38..a239e65 100644
--- a/src/Game.cxx
+++ b/src/Game.cxx
@@ -27,6 +27,7 @@
  */
 
 #include <GL/glut.h>
+#include <unistd.h>
 
 #include "glext.h"
 
diff --git a/src/TextureLoader.h b/src/TextureLoader.h
index ae3bb8d..024fb0a 100644
--- a/src/TextureLoader.h
+++ b/src/TextureLoader.h
@@ -28,6 +28,7 @@
 #define TEXTURELOADER_H
 
 #include <GL/glut.h>
+#include <cstring>
 
 #include "glext.h"
 #include "sstream.h"
