diff -urpN vice-2.1/configure vice-2.1-64/configure
--- vice-2.1/configure	2008-12-15 21:18:47.000000000 +0100
+++ vice-2.1-64/configure	2009-10-11 00:02:00.763205494 +0200
@@ -960,7 +960,7 @@ htmldir='${docdir}'
 dvidir='${docdir}'
 pdfdir='${docdir}'
 psdir='${docdir}'
-libdir='${exec_prefix}/lib'
+libdir='${exec_prefix}/lib64'
 localedir='${datarootdir}/locale'
 mandir='${datarootdir}/man'
 
@@ -1445,7 +1445,7 @@ Installation directories:
 			  [PREFIX]
 
 By default, \`make install' will install all the files in
-\`$ac_default_prefix/bin', \`$ac_default_prefix/lib' etc.  You can specify
+\`$ac_default_prefix/bin', \`$ac_default_prefix/lib64' etc.  You can specify
 an installation prefix other than \`$ac_default_prefix' using \`--prefix',
 for instance \`--prefix=\$HOME'.
 
@@ -1458,7 +1458,7 @@ Fine tuning of the installation director
   --sysconfdir=DIR       read-only single-machine data [PREFIX/etc]
   --sharedstatedir=DIR   modifiable architecture-independent data [PREFIX/com]
   --localstatedir=DIR    modifiable single-machine data [PREFIX/var]
-  --libdir=DIR           object code libraries [EPREFIX/lib]
+  --libdir=DIR           object code libraries [EPREFIX/lib64]
   --includedir=DIR       C header files [PREFIX/include]
   --oldincludedir=DIR    C header files for non-gcc [/usr/include]
   --datarootdir=DIR      read-only arch.-independent data root [PREFIX/share]
@@ -5583,7 +5583,7 @@ if test -z "$CPP"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
       # Double quotes because CPP needs to be expanded
-    for CPP in "$CC -E" "$CC -E -traditional-cpp" "/lib/cpp"
+    for CPP in "$CC -E" "$CC -E -traditional-cpp" "/lib64/cpp"
     do
       ac_preproc_ok=false
 for ac_c_preproc_warn_flag in '' yes
@@ -6369,7 +6369,7 @@ if test -z "$CPP"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
       # Double quotes because CPP needs to be expanded
-    for CPP in "$CC -E" "$CC -E -traditional-cpp" "/lib/cpp"
+    for CPP in "$CC -E" "$CC -E -traditional-cpp" "/lib64/cpp"
     do
       ac_preproc_ok=false
 for ac_c_preproc_warn_flag in '' yes
@@ -8652,7 +8652,7 @@ if test -z "$CXXCPP"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
       # Double quotes because CXXCPP needs to be expanded
-    for CXXCPP in "$CXX -E" "/lib/cpp"
+    for CXXCPP in "$CXX -E" "/lib64/cpp"
     do
       ac_preproc_ok=false
 for ac_cxx_preproc_warn_flag in '' yes
@@ -8911,7 +8911,7 @@ _ACEOF
   else
         no_recursion=yes
 
-    LDFLAGS="$LDFLAGS -L$with_resid/lib"
+    LDFLAGS="$LDFLAGS -L$with_resid/lib64"
     CPPFLAGS="$CPPFLAGS -I$with_resid/include"
 
 
@@ -9131,7 +9131,7 @@ cat >>confdefs.h <<\_ACEOF
 _ACEOF
 
     RESID_DIR=
-    RESID_LIBS="$with_resid/lib/libresid.a"
+    RESID_LIBS="$with_resid/lib64/libresid.a"
     RESID_INCLUDES="-I$with_resid/include"
   fi
 
@@ -23284,7 +23284,7 @@ _ACEOF
 	*) test -f "$ac_im_incroot/X11/Xos.h" && ac_x_includes=$ac_im_incroot;;
     esac
     case $ac_im_usrlibdir in
-	/usr/lib | /lib) ;;
+	/usr/lib64 | /lib64) ;;
 	*) test -d "$ac_im_usrlibdir" && ac_x_libraries=$ac_im_usrlibdir ;;
     esac
   fi
@@ -28704,7 +28704,7 @@ fi
 
     ;;
   *)
-    VICEDIR="$PREFIX/lib/vice";
+    VICEDIR="$PREFIX/lib64/vice";
 
 
 if false; then
diff -urpN vice-2.1/src/arch/unix/archdep.h vice-2.1-64/src/arch/unix/archdep.h
--- vice-2.1/src/arch/unix/archdep.h	2008-10-10 21:08:31.000000000 +0200
+++ vice-2.1-64/src/arch/unix/archdep.h	2009-10-11 00:03:04.366722924 +0200
@@ -116,7 +116,7 @@ extern const char *archdep_pref_path;
 #ifdef __NetBSD__
 #define LIBDIR          PREFIX "/share/vice"
 #else
-#define LIBDIR          PREFIX "/lib/vice"
+#define LIBDIR          PREFIX "/lib64/vice"
 #endif
 
 #if defined(__FreeBSD__) || defined(__NetBSD__)
