diff -urpN vice-1.22/man/vice.1 vice-1.22-fixdoc/man/vice.1
--- vice-1.22/man/vice.1	2005-08-21 21:15:04.000000000 +0200
+++ vice-1.22-fixdoc/man/vice.1	2008-02-16 19:40:11.000000000 +0100
@@ -40,7 +40,7 @@ a Commodore BASIC de-tokenizer;
 .PP
 The whole documentation for these programs is available in HTML
 format; the main file should be installed on your system as
-/usr/local/lib/VICE/doc/vice_toc.html.
+/usr/doc/vice/html/vice_toc.html.
 .P
 For up to date news about VICE, have a look at the official home page
 at
diff -urpN vice-1.22/src/arch/unix/gui/uicommands.c vice-1.22-fixdoc/src/arch/unix/gui/uicommands.c
--- vice-1.22/src/arch/unix/gui/uicommands.c	2007-04-01 17:02:44.000000000 +0200
+++ vice-1.22-fixdoc/src/arch/unix/gui/uicommands.c	2008-02-16 19:31:10.000000000 +0100
@@ -160,7 +160,7 @@ static UI_CALLBACK(browse_manual)
         char *manual_path;
         manual_path = util_concat(boot_path,"/../doc/vice_toc.html",NULL);
 #else
-        static const char manual_path[] = DOCDIR "/vice_toc.html";
+        static const char manual_path[] = DOCDIR "/html/vice_toc.html";
 #endif
         
         char *res_ptr;
