diff -Naur vice-2.4.1.orig/man/vice.1 vice-2.4.1/man/vice.1
--- vice-2.4.1.orig/man/vice.1	2012-11-29 12:44:39.000000000 +0100
+++ vice-2.4.1/man/vice.1	2013-01-23 19:48:47.065865099 +0100
@@ -43,7 +43,7 @@
 .PP
 The whole documentation for these programs is available in HTML
 format; the main file should be installed on your system as
-/usr/local/lib/vice/doc/vice_toc.html.
+/usr/doc/vice-@VERSION@/html/vice_toc.html.
 .P
 For up to date news about VICE, have a look at the official home page
 at
diff -Naur vice-2.4.1.orig/src/arch/unix/gui/uicommands.c vice-2.4.1/src/arch/unix/gui/uicommands.c
--- vice-2.4.1.orig/src/arch/unix/gui/uicommands.c	2012-12-30 15:12:52.000000000 +0100
+++ vice-2.4.1/src/arch/unix/gui/uicommands.c	2013-01-23 19:50:01.276832152 +0100
@@ -162,7 +162,7 @@
         char *manual_path;
         manual_path = util_concat(boot_path, "/../doc/vice_toc.html", NULL);
 #else
-        static const char manual_path[] = DOCDIR "/vice_toc.html";
+        static const char manual_path[] = DOCDIR "/html/vice_toc.html";
 #endif
         
         char *res_ptr;
