diff -Naur vice-2.3.21.orig/configure.in vice-2.3.21/configure.in
--- vice-2.3.21.orig/configure.in	2012-08-06 02:23:21.000000000 +0200
+++ vice-2.3.21/configure.in	2012-08-06 11:35:40.509403194 +0200
@@ -288,13 +288,7 @@
 
 AC_CHECK_PROGS(MAKEINFO, makeinfo, no)
 
-AC_CHECK_PROGS(TEXI2DVI, texi2dvi, no)
-AC_CHECK_PROGS(PDFTEX, pdftex, no)
-if test x"$TEXI2DVI" = "xno" -o x"$PDFTEX" = "xno"; then
-  AM_CONDITIONAL(BUILD_PDF, false)
-else
-  AM_CONDITIONAL(BUILD_PDF, true)
-fi
+AM_CONDITIONAL(BUILD_PDF, false)
 
 AC_CHECK_PROGS(DOS2UNIX, dos2unix, dos2unix)
 
diff -Naur vice-2.3.21.orig/doc/html/Makefile.am vice-2.3.21/doc/html/Makefile.am
--- vice-2.3.21.orig/doc/html/Makefile.am	2012-06-30 05:31:11.000000000 +0200
+++ vice-2.3.21/doc/html/Makefile.am	2012-08-06 11:32:11.953905072 +0200
@@ -45,10 +45,6 @@
 	vice_14.html \
 	vice_15.html \
 	vice_16.html \
-	vice_17.html \
-	vice_18.html \
-	vice_19.html \
-	vice_20.html \
 	vice_toc.html
 
 EXTRA_DIST = $(doc_DATA) \
