--- Makefile.am	2014-10-28 02:56:56.000000000 +0800
+++ Makefile.am.new	2014-11-01 16:41:05.534958325 +0800
@@ -34,6 +34,11 @@
 pixmap128dir=$(datadir)/icons/hicolor/128x128/apps
 pixmap128_DATA=claws-mail-128x128.png
 
+# when changing the name or location of the .desktop file,
+# also update src/common/Makefile.am
+gnomapdir = $(datadir)/applications
+gnomap_DATA=claws-mail.desktop
+
 rename-icons:
 	@cd $(DESTDIR)$(datadir)/icons/hicolor/64x64/apps && \
             mv claws-mail-64x64.png claws-mail.png
--- Makefile.in	2014-10-28 02:59:07.000000000 +0800
+++ Makefile.in.new	2014-11-01 16:57:58.497854587 +0800
@@ -165,11 +165,12 @@
     || { echo " ( cd '$$dir' && rm -f" $$files ")"; \
          $(am__cd) "$$dir" && rm -f $$files; }; \
   }
-am__installdirs = "$(DESTDIR)$(pixmapdir)" "$(DESTDIR)$(pixmap128dir)" \
-	"$(DESTDIR)$(pixmap64dir)" "$(DESTDIR)$(pkgconfigdir)" \
-	"$(DESTDIR)$(relnotesdir)" "$(DESTDIR)$(clawsincludedir)"
-DATA = $(pixmap_DATA) $(pixmap128_DATA) $(pixmap64_DATA) \
-	$(pkgconfig_DATA) $(relnotes_DATA)
+am__installdirs = "$(DESTDIR)$(gnomapdir)" "$(DESTDIR)$(pixmapdir)" \
+	"$(DESTDIR)$(pixmap128dir)" "$(DESTDIR)$(pixmap64dir)" \
+	"$(DESTDIR)$(pkgconfigdir)" "$(DESTDIR)$(relnotesdir)" \
+	"$(DESTDIR)$(clawsincludedir)"
+DATA = $(gnomap_DATA) $(pixmap_DATA) $(pixmap128_DATA) \
+	$(pixmap64_DATA) $(pkgconfig_DATA) $(relnotes_DATA)
 HEADERS = $(clawsinclude_HEADERS)
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
@@ -512,6 +513,11 @@
 pixmap64_DATA = claws-mail-64x64.png
 pixmap128dir = $(datadir)/icons/hicolor/128x128/apps
 pixmap128_DATA = claws-mail-128x128.png
+
+# when changing the name or location of the .desktop file,
+# also update src/common/Makefile.am
+gnomapdir = $(datadir)/applications
+gnomap_DATA = claws-mail.desktop
 @UPDATE_GTK_ICON_CACHE_TRUE@gtk_update_icon_cache = gtk-update-icon-cache -f -t $(datadir)/icons/hicolor
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = claws-mail.pc
@@ -593,6 +599,27 @@
 
 distclean-libtool:
 	-rm -f libtool config.lt
+install-gnomapDATA: $(gnomap_DATA)
+	@$(NORMAL_INSTALL)
+	@list='$(gnomap_DATA)'; test -n "$(gnomapdir)" || list=; \
+	if test -n "$$list"; then \
+	  echo " $(MKDIR_P) '$(DESTDIR)$(gnomapdir)'"; \
+	  $(MKDIR_P) "$(DESTDIR)$(gnomapdir)" || exit 1; \
+	fi; \
+	for p in $$list; do \
+	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
+	  echo "$$d$$p"; \
+	done | $(am__base_list) | \
+	while read files; do \
+	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(gnomapdir)'"; \
+	  $(INSTALL_DATA) $$files "$(DESTDIR)$(gnomapdir)" || exit $$?; \
+	done
+
+uninstall-gnomapDATA:
+	@$(NORMAL_UNINSTALL)
+	@list='$(gnomap_DATA)'; test -n "$(gnomapdir)" || list=; \
+	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
+	dir='$(DESTDIR)$(gnomapdir)'; $(am__uninstall_files_from_dir)
 install-pixmapDATA: $(pixmap_DATA)
 	@$(NORMAL_INSTALL)
 	@list='$(pixmap_DATA)'; test -n "$(pixmapdir)" || list=; \
@@ -1017,7 +1044,7 @@
 all-am: Makefile $(DATA) $(HEADERS) config.h claws-features.h
 installdirs: installdirs-recursive
 installdirs-am:
-	for dir in "$(DESTDIR)$(pixmapdir)" "$(DESTDIR)$(pixmap128dir)" "$(DESTDIR)$(pixmap64dir)" "$(DESTDIR)$(pkgconfigdir)" "$(DESTDIR)$(relnotesdir)" "$(DESTDIR)$(clawsincludedir)"; do \
+	for dir in "$(DESTDIR)$(gnomapdir)" "$(DESTDIR)$(pixmapdir)" "$(DESTDIR)$(pixmap128dir)" "$(DESTDIR)$(pixmap64dir)" "$(DESTDIR)$(pkgconfigdir)" "$(DESTDIR)$(relnotesdir)" "$(DESTDIR)$(clawsincludedir)"; do \
 	  test -z "$$dir" || $(MKDIR_P) "$$dir"; \
 	done
 install: install-recursive
@@ -1072,9 +1099,9 @@
 
 info-am:
 
-install-data-am: install-clawsincludeHEADERS install-pixmap128DATA \
-	install-pixmap64DATA install-pixmapDATA install-pkgconfigDATA \
-	install-relnotesDATA
+install-data-am: install-clawsincludeHEADERS install-gnomapDATA \
+	install-pixmap128DATA install-pixmap64DATA \
+	install-pixmapDATA install-pkgconfigDATA install-relnotesDATA
 	@$(NORMAL_INSTALL)
 	$(MAKE) $(AM_MAKEFLAGS) install-data-hook
 install-dvi: install-dvi-recursive
@@ -1122,7 +1149,8 @@
 
 ps-am:
 
-uninstall-am: uninstall-clawsincludeHEADERS uninstall-pixmap128DATA \
+uninstall-am: uninstall-clawsincludeHEADERS \
+	uninstall-gnomapDATA uninstall-pixmap128DATA \
 	uninstall-pixmap64DATA uninstall-pixmapDATA \
 	uninstall-pkgconfigDATA uninstall-relnotesDATA
 	@$(NORMAL_INSTALL)
@@ -1138,8 +1166,9 @@
 	distclean-hdr distclean-libtool distclean-tags distcleancheck \
 	distdir distuninstallcheck dvi dvi-am html html-am info \
 	info-am install install-am install-clawsincludeHEADERS \
-	install-data install-data-am install-data-hook install-dvi \
-	install-dvi-am install-exec install-exec-am install-html \
+	install-data install-data-am install-data-hook \
+	install-dvi install-dvi-am install-exec \
+	install-exec-am install-gnomapDATA install-html \
 	install-html-am install-info install-info-am install-man \
 	install-pdf install-pdf-am install-pixmap128DATA \
 	install-pixmap64DATA install-pixmapDATA install-pkgconfigDATA \
@@ -1148,8 +1177,8 @@
 	maintainer-clean maintainer-clean-generic \
 	maintainer-clean-local mostlyclean mostlyclean-generic \
 	mostlyclean-libtool pdf pdf-am ps ps-am tags tags-am uninstall \
-	uninstall-am uninstall-clawsincludeHEADERS uninstall-hook \
-	uninstall-pixmap128DATA uninstall-pixmap64DATA \
+	uninstall-am uninstall-clawsincludeHEADERS uninstall-gnomapDATA \
+	uninstall-hook uninstall-pixmap128DATA uninstall-pixmap64DATA \
 	uninstall-pixmapDATA uninstall-pkgconfigDATA \
 	uninstall-relnotesDATA
 
