diff --git i/libgnucash/engine/test/utest-Account.cpp w/libgnucash/engine/test/utest-Account.cpp
index fc52032ba1..4c5ea1ca61 100644
--- i/libgnucash/engine/test/utest-Account.cpp
+++ w/libgnucash/engine/test/utest-Account.cpp
@@ -44,11 +44,13 @@ extern "C" void test_suite_account (void);
 #include <qofinstance-p.h>
 #include <kvp-frame.hpp>
 
+namespace {
 typedef struct
 {
     Account *acct;
     AccountTestFunctions *func;
 } Fixture;
+} // anonymous namespace
 
 typedef struct
 {
diff --git i/libgnucash/engine/test/utest-Split.cpp w/libgnucash/engine/test/utest-Split.cpp
index b9b177b6b0..4077ec0ecd 100644
--- i/libgnucash/engine/test/utest-Split.cpp
+++ w/libgnucash/engine/test/utest-Split.cpp
@@ -45,6 +45,7 @@ extern "C" void test_suite_split ( void );
 #include <qofinstance-p.h>
 #include <kvp-frame.hpp>
 
+namespace {
 typedef struct
 {
     Split *split;
@@ -53,6 +54,7 @@ typedef struct
     gnc_commodity *comm;
     GSList *hdlrs;
 } Fixture;
+} // anonymous namespace
 
 static void
 setup (Fixture *fixture, gconstpointer pData)
