From 58528a9e983ae8ba53e4e5c10f0565ac0b12c2b3 Mon Sep 17 00:00:00 2001
From: Eugen Wissner <belka@caraus.de>
Date: Sun, 25 Oct 2020 10:18:08 +0100
Subject: [PATCH] Link static libiberty

---
 hphp/hhvm/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hphp/hhvm/CMakeLists.txt b/hphp/hhvm/CMakeLists.txt
index f70bd1f542..ab41ec1e5d 100644
--- a/hphp/hhvm/CMakeLists.txt
+++ b/hphp/hhvm/CMakeLists.txt
@@ -28,7 +28,7 @@ endif()
 
 get_object_libraries_objects(additionalObjects ${HHVM_WHOLE_ARCHIVE_LIBRARIES})
 add_executable(hhvm ${CXX_SOURCES} ${additionalObjects})
-target_link_libraries(hhvm ${HHVM_LINK_LIBRARIES} ${EZC_LINK_LIBRARIES} ${HRE_LINK_LIBRARIES})
+target_link_libraries(hhvm /usr/lib64/libiberty.a ${HHVM_LINK_LIBRARIES} ${EZC_LINK_LIBRARIES} ${HRE_LINK_LIBRARIES})
 link_object_libraries(hhvm ${HHVM_WHOLE_ARCHIVE_LIBRARIES})
 if (GOLD_FOUND AND ENABLE_LD_GOLD)
   if (CMAKE_CONFIGURATION_TYPES)
-- 
2.17.5

