diff -Nur scribus_orig/CMakeLists.txt scribus_new/CMakeLists.txt
--- scribus_orig/CMakeLists.txt	2024-01-08 00:06:51.000000000 +0700
+++ scribus_new/CMakeLists.txt	2024-05-06 14:06:27.783226671 +0700
@@ -417,6 +417,9 @@
 elseif(WANT_CPP17)
 		message(STATUS "Enabling C++17 compiler features")
 		set(CMAKE_CXX_STANDARD 17)
+elseif(WANT_CPP20)
+		message(STATUS "Enabling C++20 compiler features")
+		set(CMAKE_CXX_STANDARD 20)
 endif()
 set (CMAKE_CXX_STANDARD_REQUIRED ON)
 set (CMAKE_CXX_EXTENSIONS OFF)
diff -Nur scribus_orig/scribus/plugins/import/pdf/importpdf.cpp scribus_new/scribus/plugins/import/pdf/importpdf.cpp
--- scribus_orig/scribus/plugins/import/pdf/importpdf.cpp	2024-01-08 00:07:00.000000000 +0700
+++ scribus_new/scribus/plugins/import/pdf/importpdf.cpp	2024-05-05 13:48:19.683235339 +0700
@@ -28,6 +28,7 @@
 #include <poppler/cpp/poppler-version.h>
 #include <poppler/SplashOutputDev.h>
 #include <poppler/splash/SplashBitmap.h>
+#include <poppler/UTF.h>
 
 #include "importpdf.h"
 #include "importpdfconfig.h"
diff -Nur scribus_orig/scribus/plugins/import/pdf/pdftextrecognition.cpp scribus_new/scribus/plugins/import/pdf/pdftextrecognition.cpp
--- scribus_orig/scribus/plugins/import/pdf/pdftextrecognition.cpp	2024-01-08 00:07:00.000000000 +0700
+++ scribus_new/scribus/plugins/import/pdf/pdftextrecognition.cpp	2024-05-06 09:59:28.809026072 +0700
@@ -6,6 +6,7 @@
 */
 
 #include "pdftextrecognition.h"
+#include <poppler/UTF.h>
 
 constexpr double PdfTextRegion::boundingBoxShape[32];
 
diff -Nur scribus_orig/scribus/plugins/import/pdf/slaoutput.cpp scribus_new/scribus/plugins/import/pdf/slaoutput.cpp
--- scribus_orig/scribus/plugins/import/pdf/slaoutput.cpp	2024-01-08 00:07:00.000000000 +0700
+++ scribus_new/scribus/plugins/import/pdf/slaoutput.cpp	2024-05-05 12:56:49.313450485 +0700
@@ -16,6 +16,7 @@
 #include <poppler/poppler-config.h>
 #include <poppler/FileSpec.h>
 #include <poppler/fofi/FoFiTrueType.h>
+#include <poppler/UTF.h>
 
 #include <QApplication>
 #include <QFile>
diff -Nur scribus_orig/scribus/third_party/lib2geom/CMakeLists.txt scribus_new/scribus/third_party/lib2geom/CMakeLists.txt
--- scribus_orig/scribus/third_party/lib2geom/CMakeLists.txt	2024-01-08 00:07:04.000000000 +0700
+++ scribus_new/scribus/third_party/lib2geom/CMakeLists.txt	2024-05-06 14:07:06.099229776 +0700
@@ -6,6 +6,7 @@
 set(2GEOM_VERSION 0.1.0)
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_FPIC}")
+set(CMAKE_CXX_STANDARD 14)
 
 option(2GEOM_BUILD_SHARED
   "Build lib2geom and libtoy as shared libraries."
