--- Makefile.org	2009-05-31 10:17:52.000000000 -0400
+++ Makefile	2009-05-31 16:52:00.959195506 -0400
@@ -185,8 +185,6 @@
   	$(INSTALL) -m 644 -D po/$$tr.mo -T $(INSTALLDIR)/share/locale/$$tr/LC_MESSAGES/gogglesmm.mo ; \
 	done;
 endif
-	@echo "    Updating the icon cache ..."
-	@gtk-update-icon-cache -t $(INSTALLDIR)/share/icons/hicolor
 
 # Clean
 #----------------------------------------------------------
