diff -Naur mount-zip-1.0.9/lib/data_node.cc mount-zip-1.0.9.patched/lib/data_node.cc
--- mount-zip-1.0.9/lib/data_node.cc	2023-08-17 01:41:40.000000000 -0400
+++ mount-zip-1.0.9.patched/lib/data_node.cc	2023-08-29 02:59:18.342998941 -0400
@@ -269,7 +269,7 @@
   ZipFile file = Reader::Open(zip, id);
   assert(file);
 
-  const bool seekable = zip_file_is_seekable(file.get()) > 0;
+  const bool seekable = false;
   if (seekable) {
     Log(LOG_DEBUG, "No need to cache ", file_node, ": File is seekable");
     return false;
@@ -293,7 +293,7 @@
   ZipFile file = Reader::Open(zip, id);
   assert(file);
 
-  const bool seekable = zip_file_is_seekable(file.get()) > 0;
+  const bool seekable = false;
   Reader::Ptr reader(seekable ? new UnbufferedReader(std::move(file), id, size)
                               : new BufferedReader(zip, std::move(file), id,
                                                    size, &cached_reader));
