Author: Daniel Baumann <daniel@debian.org>
Description: Explicitly linking against libgcrypt.

diff -Naurp gnomint.orig/src/Makefile.in gnomint/src/Makefile.in
--- gnomint.orig/src/Makefile.in	2011-05-17 19:57:50.013883380 +0200
+++ gnomint/src/Makefile.in	2011-05-17 19:57:37.209981808 +0200
@@ -347,6 +347,7 @@ gnomint_cli_SOURCES = \
 
 gnomint_cli_LDADD = \
 	$(GNOMINTCLI_LIBS) \
+	$(LIBGCRYPT_LIBS) \
 	$(LIBGNUTLS_LIBS) \
 	$(SQLITE_LIBS) \
 	$(READLINE_LIBS) \
