Author: Daniel Baumann <daniel@debian.org>
Description:
 Don't hardcode CFLAGS and respect the environemnt (Closes: #634395).

diff -Naurp gnomint.orig/configure gnomint/configure
--- gnomint.orig/configure	2011-09-08 07:58:09.324752523 +0000
+++ gnomint/configure	2011-09-08 07:59:20.113103539 +0000
@@ -14338,7 +14338,7 @@ fi
 
 
 if test "x$GCC" = "xyes"; then
-	CFLAGS="-Wall -Werror"
+	CFLAGS="${CFLAGS:--Wall -Werror}"
 	if test "x$use_debug" = "xYes"; then
    	   CFLAGS="$CFLAGS -g -O0"
 	fi
diff -Naurp gnomint.orig/configure.in gnomint/configure.in
--- gnomint.orig/configure.in	2011-09-08 07:58:09.328752536 +0000
+++ gnomint/configure.in	2011-09-08 07:59:20.113103539 +0000
@@ -181,7 +181,7 @@ dnl
 
 dnl
 if test "x$GCC" = "xyes"; then
-	CFLAGS="-Wall -Werror"
+	CFLAGS="${CFLAGS:--Wall -Werror}"
 	if test "x$use_debug" = "xYes"; then
    	   CFLAGS="$CFLAGS -g -O0"
 	fi
