Description: Fix double free in tls_creation_data_free.
Author: Sebastian Ramacher <s.ramacher@gmx.at>
Bug-Debian: http://bugs.debian.org/682432
Last-Update: 2012-08-04

--- gnomint-1.3.0.orig/src/tls.c
+++ gnomint-1.3.0/src/tls.c
@@ -1665,8 +1665,6 @@
 		g_free (cd->emailAddress);
 	if (cd->password)
 		g_free (cd->password);
-        if (cd->crl_distribution_point)
-                g_free (cd->crl_distribution_point);
         if (cd->parent_ca_id_str)
                 g_free (cd->parent_ca_id_str);
 	if (cd->crl_distribution_point)
