Description: Fix some spelling errors.
Author: Yavor Doganov <yavor@gnu.org>
Forwarded: gnomint-devel@lists.sourceforge.net
Last-Update: 2019-10-12
---

--- gnomint-1.3.0.orig/src/ca.c
+++ gnomint-1.3.0/src/ca.c
@@ -978,7 +978,7 @@
 
 	widget = gtk_builder_get_object (main_window_gtkb, "main_window1");
 	
-	dialog = GTK_DIALOG (gtk_file_chooser_dialog_new (_("Export uncrypted private key"),
+	dialog = GTK_DIALOG (gtk_file_chooser_dialog_new (_("Export unencrypted private key"),
 							  GTK_WINDOW(widget),
 							  GTK_FILE_CHOOSER_ACTION_SAVE,
 							  GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL,
--- gnomint-1.3.0.orig/src/country_table.c
+++ gnomint-1.3.0/src/country_table.c
@@ -520,7 +520,7 @@
 	country_table[i++].code = "VU";
 	country_table[i].name = D_("Venezuela");
 	country_table[i++].code = "VE";
-	country_table[i].name = D_("Viet Nam");
+	country_table[i].name = D_("Vietnam");
 	country_table[i++].code = "VN";
 	country_table[i].name = D_("Virgin Islands, British");
 	country_table[i++].code = "VG";
--- gnomint-1.3.0.orig/src/import.c
+++ gnomint-1.3.0/src/import.c
@@ -82,7 +82,7 @@
 	printf (_("The whole selected file, or some of its elements, seems to\n"
 		  "be cyphered using a password or passphrase. For importing\n"
 		  "the file into gnoMint database, you must provide an \n"
-		  "appropiate password.\n"));
+		  "appropriate password.\n"));
 
 	prompt = g_strdup_printf (_("Please introduce password for `%s'"), crypted_part_description);
 	password = dialog_ask_for_password (prompt);
