--- ldglite_orig/makefile.linux	2009-11-06 16:50:32.000000000 -0200
+++ ldglite/makefile.linux	2010-01-27 16:55:47.000000000 -0200
@@ -42,7 +42,7 @@
     LIBS = libmui.a
 endif
 
-CFLAGS=-g -DUNIX -DUSE_OPENGL -DUSE_L3_PARSER -DUSE_BMP8 -DNEED_MIN_MAX $(PNG_FLAGS) $(TR_FLAGS) $(OFFSCREEN_FLAGS) $(GUI_FLAGS)
+CFLAGS=-g -DUNIX -DUSE_OPENGL -DUSE_L3_PARSER -DUSE_BMP8 -DNEED_MIN_MAX $(PNG_FLAGS) $(TR_FLAGS) $(OFFSCREEN_FLAGS) $(GUI_FLAGS) $(SBO_FLAGS)
 CXXFLAGS=$(CFLAGS)
 
 SRCS = ldliteVR_main.c platform.c dirscan.c gleps.c camera.c f00QuatC.c quant.c stub.c lcolors.c y.tab.c lex.yy.c qbuf.c main.c ldglpr.c L3Edit.c L3Math.cpp L3Input.cpp L3View.cpp hoser.c ldglmenu.c plugstub.c $(TR_SRC) $(GUI_SRC)
