--- Makefile.in	2014-02-22 14:09:49.000000000 +0100
+++ Makefile-patched.in	2021-08-14 00:15:51.668161476 +0200
@@ -364,10 +364,6 @@
 
 $(VM): c/main.o $(LIBSCHEME48) c/scheme48.exp c/scheme48.def
 	$(CC) $(LDFLAGS) $(LDFLAGS_VM) $(CFLAGS) $(CPPFLAGS) -o $@ c/main.o	$(LIBSCHEME48) $(LIBS)
-	# Cygwin
-	if test -x /usr/bin/dlltool; then \
-		dlltool --dllname $(VM).exe --output-lib $(VM).a --def $(srcdir)/c/scheme48.def; \
-	fi
 
 $(LIBSCHEME48): $(OBJS) $(GC_OBJS) $(UNIX_OBJS) $(LIBOBJS) $(EXTERNAL_OBJECTS)
 	rm -f $@;							\
