diff -Naur tribler-6.4.0/Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py tribler-6.4.0.patched/Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py
--- tribler-6.4.0/Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py	2014-12-21 02:24:05.000000000 -0500
+++ tribler-6.4.0.patched/Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py	2014-12-21 03:45:45.000000000 -0500
@@ -845,7 +845,7 @@
         public = self.tdef and not isinstance(self.tdef, TorrentDefNoMetainfo) and not self.tdef.is_private()
 
         result = self.tracker_status.copy()
-        result['[DHT]'] = [dht_peers, 'Working' if ltsession.is_dht_running() and public else 'Disabled']
+        result['[DHT]'] = [dht_peers, 'Working' if dht_peers > 0 else 'Not Working']
         result['[PeX]'] = [pex_peers, 'Working' if not self.get_anon_mode() else 'Disabled']
         return result
 
