--- /library/slackware/SBo_repository/SBo/libraries/poco/poco.SlackBuild	2022-12-01 16:48:10.265823314 -0800
+++ poco.SlackBuild	2022-12-28 20:11:48.439588528 -0800
@@ -47,6 +47,14 @@
   exit 0
 fi
 
+# Make optionally detected components really optional as they are in 
+# CMAKE build configuration upstream.
+if [ ${POSTGRESQL:-no} = "no" ]; then
+  NOPOSTGRESQL="--omit=Data/PostgreSQL"
+else
+  NOPOSTGRESQL=""
+fi
+
 TMP=${TMP:-/tmp/SBo}
 PKG=$TMP/package-$PRGNAM
 OUTPUT=${OUTPUT:-/tmp}
@@ -84,10 +92,14 @@
     patch -p1 < $CWD/poco_x64.patch
 fi
 
+# Slackware ships with MariaDB not MySQL, patch from Arch Linux
+patch -p1 < $CWD/poco_mariadb.patch
+
 ./configure \
   --prefix=/usr \
   --no-tests \
   --no-samples \
+  $NOPOSTGRESQL \
   --shared
 
 make
