From 436f659bdcef3726ece7928e221ba7ea7805381e Mon Sep 17 00:00:00 2001
From: Fabrice Bellet <fabrice@bellet.info>
Date: Sun, 22 Sep 2013 12:55:33 +0200
Subject: [PATCH 5/5] Update required SimGear version

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index aa95c59..4b6fea5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -131,7 +131,7 @@ find_package(Boost REQUIRED)
 find_package(ZLIB REQUIRED)
 find_package(FLTK REQUIRED)
 find_package(OpenSceneGraph 2.8.1 REQUIRED osgText osgDB osgParticle osgUtil osgViewer osgGA)
-find_package(SimGear 2.8.0 REQUIRED)
+find_package(SimGear 2.12.0 REQUIRED)
 if (ENABLE_NLS)
     find_package(Gettext)
 endif (ENABLE_NLS)
-- 
1.8.3.1

