--- fgrun-1.6.1/src/wizard_funcs.cxx.orig	2012-02-27 22:56:08.000000000 +0100
+++ fgrun-1.6.1/src/wizard_funcs.cxx	2012-03-21 21:00:52.072609457 +0100
@@ -1200,7 +1200,6 @@ Wizard::~Wizard()
     aircraft->clear();
 
     delete logwin;
-    delete win;
     delete adv;
 }
 
@@ -2265,7 +2264,6 @@ Wizard::reset_settings()
     prefs.set( "time_of_day_value", "noon" );
 
     aircraft->value( 0 );
-    preview->make_current();
     preview->clear();
     preview->redraw();
 
