diff -ur linuxconsoletools-1.4rc1.orig//docs/Makefile linuxconsoletools-1.4rc1/docs/Makefile
--- linuxconsoletools-1.4rc1.orig//docs/Makefile	2011-04-04 07:20:32.000000000 +0200
+++ linuxconsoletools-1.4rc1/docs/Makefile	2011-04-06 09:09:39.269000004 +0200
@@ -26,7 +26,7 @@
 PREFIX          ?= /usr/local
 
 install:
-	install -d $(DESTDIR)$(PREFIX)/share/man/man1
-	install $(MANPAGES) $(DESTDIR)$(PREFIX)/share/man/man1
+	install -d $(DESTDIR)$(PREFIX)/man/man1
+	install $(MANPAGES) $(DESTDIR)$(PREFIX)/man/man1
 
 .PHONY: install
