diff -ur linuxconsoletools-1.4.2.orig//docs/Makefile linuxconsoletools-1.4.2/docs/Makefile
--- linuxconsoletools-1.4.2.orig//docs/Makefile	2011-04-05 09:10:57.000000000 +0200
+++ linuxconsoletools-1.4.2/docs/Makefile	2011-11-04 10:42:19.097000004 +0100
@@ -26,7 +26,7 @@
 PREFIX          ?= /usr/local
 
 install:
-	install -d $(DESTDIR)$(PREFIX)/share/man/man1
-	install $(MANPAGES) $(DESTDIR)$(PREFIX)/share/man/man1
+	install -d $(DESTDIR)$(PREFIX)/man/man1
+	install $(MANPAGES) $(DESTDIR)$(PREFIX)/man/man1
 
 .PHONY: install
