diff -ru linuxconsoletools-1.4.4.orig/docs/Makefile linuxconsoletools-1.4.4/docs/Makefile
--- linuxconsoletools-1.4.4.orig/docs/Makefile	2011-12-15 16:49:01.000000000 +0100
+++ linuxconsoletools-1.4.4/docs/Makefile	2012-12-11 08:46:08.276507553 +0100
@@ -26,7 +26,7 @@
 PREFIX          ?= /usr/local
 
 install:
-	install -d $(DESTDIR)$(PREFIX)/share/man/man1
-	install $(MANPAGES) $(DESTDIR)$(PREFIX)/share/man/man1
+	install -d $(DESTDIR)$(PREFIX)/man/man1
+	install $(MANPAGES) $(DESTDIR)$(PREFIX)/man/man1
 
 .PHONY: install
