diff -uaN command-list.c command-list.c.new                 
--- command-list.c	2018-01-24 20:59:24.000000000 -0800
+++ command-list.c.new	2018-02-18 00:51:13.218202297 -0800
@@ -903,6 +903,8 @@
     // Allocate enough space for the whole file list, plus two extra files and
     // a NULL at the end.
     cfList = nvalloc((p->num_entries + 3) * sizeof(ConflictingFileInfo));
+/* let's just return an empty list!  */
+    return cfList;
 
     for (i = 0; i < p->num_entries; i++) {
         PackageEntry *entry = &p->entries[i];
	 
diff -uaN misc.c misc.c.new      
--- misc.c	2018-02-18 00:52:42.822203145 -0800
+++ misc.c.new	2018-02-18 00:53:28.926203581 -0800
@@ -2289,7 +2289,10 @@
                  "For further details, please see the appendix SUPPORTED "
                  "NVIDIA GRAPHICS CHIPS in the README available on the Linux "
                  "driver download page at www.nvidia.com.", p->version);
-        return FALSE;
+/*        return FALSE;
+ * We're just going to keep the message but skip the result
+ * so we can run this in a VM.
+ */
     }
 
     if (!found_vga_device)

