diff -Naur sbagen-1.4.5/sbagen.c sbagen-1.4.5.patched/sbagen.c
--- sbagen-1.4.5/sbagen.c	2011-08-07 20:12:21.000000000 -0400
+++ sbagen-1.4.5.patched/sbagen.c	2022-03-22 22:02:03.585606715 -0400
@@ -181,7 +181,7 @@
 void warn(char *fmt, ...) ;
 void * Alloc(size_t len) ;
 char * StrDup(char *str) ;
-inline int calcNow() ;
+int calcNow() ;
 void loop() ;
 void outChunk() ;
 void corrVal(int ) ;
@@ -1207,7 +1207,7 @@
   time_ref_ms= 1000*tt->tm_sec + 60000*tt->tm_min + 3600000*tt->tm_hour;
 }  
 
-inline int  
+int  
 calcNow() {
   struct timeval tv;
   if (0 != gettimeofday(&tv, 0)) error("Can't get current time");
@@ -1217,7 +1217,7 @@
 #endif
 
 #ifdef WIN_TIME
-inline int  
+int  
 calcNow() {
   SYSTEMTIME st;
   GetLocalTime(&st);
