From 0498f71e811ff3068ba491929c09aa5137ea0cca Mon Sep 17 00:00:00 2001
From: Garrett Potts <gpotts@radiantblue.com>
Date: Thu, 31 Oct 2019 07:08:03 -0400
Subject: [PATCH] Added missing headers

---
 src/init/ossimInit.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/init/ossimInit.cpp b/src/init/ossimInit.cpp
index 9aeec9de..2c2ec2c1 100644
--- a/src/init/ossimInit.cpp
+++ b/src/init/ossimInit.cpp
@@ -66,7 +66,9 @@
 #include <algorithm>
 #include <mutex>
 #include <geos_c.h>
-
+#include <cstdio>
+#include <cstdarg>
+#include <cstddef>
 using namespace std;
 
 static ossimTrace traceExec = ossimTrace("ossimInit:exec");
