--- sixad/sixad.init.orig	2013-03-30 18:09:41.713986598 +0100
+++ sixad/sixad.init	2013-03-30 18:11:19.787352424 +0100
@@ -19,40 +19,37 @@
 ps -e | grep sixad-bin > /dev/null
 }

-. /lib/lsb/init-functions
-
 case "$1" in
   start)
 if (sixad_already_running_check "$1"); then
-  log_warning_msg "sixad is already running"
+  echo "sixad is already running"
+  exit 1
 else
 {
-  log_daemon_msg "Starting sixad"
+  echo "Starting sixad"
   $DAEMON --start &>>/var/log/sixad &
-  log_end_msg 0
+  exit 0
 }
 fi
     ;;
   stop)
     if (sixad_already_running_check "$1"); then
     {
-      log_daemon_msg "Stopping sixad"
+      echo "Stopping sixad"
       $DAEMON --stop || true
-      log_end_msg 0
+      exit 0
     }
     else
-      log_warning_msg "sixad is not running"
+      echo "sixad is not running"
+      exit 1
     fi
     ;;
   restart)
     $0 stop
     $0 start
     ;;
-  status)
-    status_of_proc "sixad-bin" "sixad" && exit 0 || exit $?
-    ;;
   *)
-    echo "Usage: /etc/init.d/sixad {start|stop|restart|status}" >&2
+    echo "Usage: /etc/rc.d/rc.sixad {start|stop|restart}" >&2
     exit 1
     ;;
 esac
