diff -Naur innotop-1.8.0-orig/t/InnoDBParser.t innotop-1.8.0/t/InnoDBParser.t
--- innotop-1.8.0-orig/t/InnoDBParser.t	2010-11-06 17:32:11.000000000 +0200
+++ innotop-1.8.0/t/InnoDBParser.t	2010-12-09 15:22:01.000000000 +0200
@@ -5,7 +5,7 @@
 use Test::More tests => 1;
 use Data::Dumper;
 
-require "../innotop";
+require "innotop";
 my $innodb_parser = new InnoDBParser;
 
 sub read_file {
@@ -18,7 +18,7 @@
 }
 
 my %tests = (
-   'innodb-status-001' => {
+   't/innodb-status-001' => {
       IB_bp_add_pool_alloc              => '0',
       IB_bp_awe_mem_alloc               => 0,
       IB_bp_buf_free                    => '8172',
