--- texmaker-1.9/texmaker.cpp.orig	2009-05-21 12:56:32.047389802 -0300
+++ texmaker-1.9/texmaker.cpp	2009-05-21 12:56:33.936554284 -0300
@@ -402,6 +402,8 @@
 CentralLayout->setMargin(0);
 CentralLayout->addWidget(centralToolBar);
 EditorView=new QTabWidget(centralFrame);
+EditorView->setFocusPolicy(Qt::ClickFocus);
+EditorView->setFocus();
 EditorView->setTabsClosable(true);
 connect(EditorView, SIGNAL( currentChanged( int ) ), this, SLOT(UpdateCaption()) );
 CentralLayout->addWidget(EditorView);
