diff -Naur texmaker-4.4.1.orig/singleapp/qtlocalpeer.cpp texmaker-4.4.1/singleapp/qtlocalpeer.cpp
--- texmaker-4.4.1.orig/singleapp/qtlocalpeer.cpp	2013-01-07 11:55:45.000000000 +0000
+++ texmaker-4.4.1/singleapp/qtlocalpeer.cpp	2015-09-13 01:53:52.722192775 +0000
@@ -42,6 +42,7 @@
 #include "qtlocalpeer.h"
 #include <QCoreApplication>
 #include <QTime>
+#include <QDataStream>
 
 #if defined(Q_OS_WIN)
 #include <QLibrary>
