diff -ruN monkey-1.5.6/configure monkey-1.5.6.new/configure
--- monkey-1.5.6/configure	2015-03-03 00:52:16.000000000 +0300
+++ monkey-1.5.6.new/configure	2015-03-13 11:51:55.984045498 +0300
@@ -903,17 +903,17 @@
 
 create_banana_script()
 {
-	cat > bin/banana << EOF
+	cat > bin/rc.monkey << EOF
 #!/bin/sh
 #
-# Monkey HTTP Daemon - Banana Script
+# Monkey HTTP Daemon - rc.monkey Script
 # -----------------------------------
 # This script allow you to control monkey. Written by Eduardo Silva
 # ----------------------------
 # Date		: 2002/09/01.
 # ----------------------------
 #
-# Use: ./banana OPTION
+# Use: ./rc.monkey OPTION
 #
 # Options available to banana:
 #
@@ -1000,16 +1000,16 @@
 			exit 0
 			;;
 		*)
-			echo "Use : banana [start|stop|restart|status|help]"
+			echo "Use : \$0 [start|stop|restart|status|help]"
 			exit 1
 		;;
 	esac
 done
-echo "Use : banana [start|stop|restart|status|help]"
+echo "Use : \$0 [start|stop|restart|status|help]"
 
 exit 0
 EOF
-	chmod 755 bin/banana
+	chmod 755 bin/rc.monkey
 }
 
 #---------------------------#
diff -ruN monkey-1.5.6/lang/en/mconf monkey-1.5.6.new/lang/en/mconf
--- monkey-1.5.6/lang/en/mconf	2015-03-03 00:52:16.000000000 +0300
+++ monkey-1.5.6.new/lang/en/mconf	2015-03-13 11:52:36.464045763 +0300
@@ -66,7 +66,7 @@
     # For security reasons, sometimes people want to hide the version of his
     # own webserver to clients (values on/off).
 
-    HideVersion off
+    HideVersion on
 
 
     # Resume:
@@ -124,7 +124,7 @@
     # --------
     # Allow request to symbolic link files.
 
-    SymLink Off
+    SymLink On
 
     # TransportLayer:
     # ---------------
diff -ruN monkey-1.5.6/plugins/cgi/conf/cgi.conf monkey-1.5.6.new/plugins/cgi/conf/cgi.conf
--- monkey-1.5.6/plugins/cgi/conf/cgi.conf	2015-03-03 00:52:16.000000000 +0300
+++ monkey-1.5.6.new/plugins/cgi/conf/cgi.conf	2015-03-13 11:41:34.654041430 +0300
@@ -19,4 +19,4 @@
     Match /cgi-bin/.*\.cgi
 
     # Execute PHP scripts through php-cgi interpreter:
-    Match /.*\.php /usr/bin/php-cgi application/x-httpd-php
+    #Match /.*\.php /usr/bin/php-cgi application/x-httpd-php
