diff -ur evince-2.32.0-orig/libdocument/ev-document-info.h evince-2.32.0/libdocument/ev-document-info.h
--- evince-2.32.0-orig/libdocument/ev-document-info.h	2010-07-14 14:54:39.000000000 +0700
+++ evince-2.32.0/libdocument/ev-document-info.h	2025-09-08 19:58:13.899490513 +0700
@@ -113,8 +113,8 @@
 	char *keywords;
 	char *creator;
 	char *producer;
-	char *linearized;
-        char *security;
+	gboolean linearized;
+	char *security;
 	GTime creation_date;
 	GTime modified_date;
 	EvDocumentLayout layout;
diff -ur evince-2.32.0-orig/libdocument/ev-document.c evince-2.32.0/libdocument/ev-document.c
--- evince-2.32.0-orig/libdocument/ev-document.c	2010-09-27 23:54:34.000000000 +0700
+++ evince-2.32.0/libdocument/ev-document.c	2025-09-08 19:58:13.899490513 +0700
@@ -724,7 +724,9 @@
 	copy->security = g_strdup (info->security);
 	copy->creator = g_strdup (info->creator);
 	copy->producer = g_strdup (info->producer);
-	copy->linearized = g_strdup (info->linearized);
+
+	// "info->linearized" is a boolean
+	copy->linearized = info->linearized;
 	
 	copy->creation_date = info->creation_date;
 	copy->modified_date = info->modified_date;
@@ -753,7 +755,6 @@
 	g_free (info->keywords);
 	g_free (info->creator);
 	g_free (info->producer);
-	g_free (info->linearized);
 	g_free (info->security);
 	ev_document_license_free (info->license);
 
diff -ur evince-2.32.0-orig/properties/ev-properties-view.c evince-2.32.0/properties/ev-properties-view.c
--- evince-2.32.0-orig/properties/ev-properties-view.c	2010-07-14 14:54:40.000000000 +0700
+++ evince-2.32.0/properties/ev-properties-view.c	2025-09-08 20:01:55.236662565 +0700
@@ -357,7 +357,13 @@
 		g_free (text);
 	}
 	if (info->fields_mask & EV_DOCUMENT_INFO_LINEARIZED) {
-		set_property (properties, GTK_TABLE (table), LINEARIZED_PROPERTY, info->linearized, &row);
+		if (info->linearized) {
+			text = g_strdup("Yes");
+		} else {
+			text = g_strdup("No");
+		}
+		set_property (properties, GTK_TABLE (table), LINEARIZED_PROPERTY, text, &row);
+		g_free (text);
 	}
 	if (info->fields_mask & EV_DOCUMENT_INFO_SECURITY) {
 		set_property (properties, GTK_TABLE (table), SECURITY_PROPERTY, info->security, &row);
