diff -Naur evince-2.32.0/backend/comics/comics-document.c evince-2.32.0.patched/backend/comics/comics-document.c
--- evince-2.32.0/backend/comics/comics-document.c	2010-09-27 12:54:34.000000000 -0400
+++ evince-2.32.0.patched/backend/comics/comics-document.c	2024-08-19 23:35:22.539062511 -0400
@@ -320,7 +320,8 @@
 	 * included in shared-mime-info */
 	
 	if (!strcmp (mime_type, "application/x-cbr") ||
-	    !strcmp (mime_type, "application/x-rar")) {
+	    !strcmp (mime_type, "application/x-rar") ||
+	    !strcmp (mime_type, "application/vnd.rar")) {
 	        /* The RARLAB provides a no-charge proprietary (freeware) 
 	        * decompress-only client for Linux called unrar. Another 
 		* option is a GPLv2-licensed command-line tool developed by 
@@ -405,7 +406,8 @@
 				return TRUE;
 			}
 	} else if (!strcmp (mime_type, "application/x-cbt") ||
-		   !strcmp (mime_type, "application/x-tar")) {
+		   !strcmp (mime_type, "application/x-tar") ||
+		   !strcmp (mime_type, "application/gzip")) {
 		/* tar utility (Tape ARchive) */
 		comics_document->selected_command =
 				g_find_program_in_path ("tar");
diff -Naur evince-2.32.0/backend/comics/comicsdocument.evince-backend.in evince-2.32.0.patched/backend/comics/comicsdocument.evince-backend.in
--- evince-2.32.0/backend/comics/comicsdocument.evince-backend.in	2010-07-14 03:54:39.000000000 -0400
+++ evince-2.32.0.patched/backend/comics/comicsdocument.evince-backend.in	2024-08-19 23:33:18.180074214 -0400
@@ -1,4 +1,4 @@
 [Evince Backend]
 Module=comicsdocument
 _TypeDescription=Comic Books
-MimeType=application/x-cbr;application/x-cbz;application/x-cb7;application/x-cbt;
+MimeType=application/x-cbr;application/x-cbz;application/x-cb7;application/x-cbt;application/zip;application/vnd.rar;application/x-7z-compressed;application/x-tar;application/gzip;
diff -Naur evince-2.32.0/configure evince-2.32.0.patched/configure
--- evince-2.32.0/configure	2010-09-27 15:02:20.000000000 -0400
+++ evince-2.32.0.patched/configure	2024-08-19 23:33:23.710073694 -0400
@@ -21439,7 +21439,7 @@
 	EVINCE_MIME_TYPES="${EVINCE_MIME_TYPES}image/tiff;"
 fi
 if test "x$enable_comics" = "xyes"; then
-	EVINCE_MIME_TYPES="${EVINCE_MIME_TYPES}application/x-cbr;application/x-cbz;application/x-cb7;application/x-cbt;"
+	EVINCE_MIME_TYPES="${EVINCE_MIME_TYPES}application/x-cbr;application/x-cbz;application/x-cb7;application/x-cbt;application/zip;application/vnd.rar;application/x-7z-compressed;application/x-tar;application/gzip;"
 fi
 if test "x$enable_pixbuf" = "xyes"; then
 	EVINCE_MIME_TYPES="${EVINCE_MIME_TYPES}image/*;"
