--- src/main.c.orig	2017-05-21 08:52:45.000000000 +0200
+++ src/main.c	2017-11-24 15:39:26.653090218 +0100
@@ -349,8 +349,15 @@
     g_free (cal);
     g_free (appGUI);
     g_free(cmd_cfg_path);
+
+#ifdef CONTACTS_ENABLED
     g_free(cmd_mutt_query);
+#endif  /* CONTACTS_ENABLED */
+
+#ifdef HAVE_LIBWEBKIT
     g_free(cmd_stylesheet_file);
+#endif  /* HAVE_LIBWEBKIT */
+
 #ifndef WIN32
     g_free (s_lock);
 #endif /* WIN32 */
