
Index: notes_items.c
===================================================================
--- notes_items.c	(revision 692)
+++ notes_items.c	(working copy)
@@ -1038,7 +1037,7 @@ gchar *current_filename;
 		if (check_if_encrypted (current_filename, appGUI) == TRUE) {
 
 			g_snprintf (tmpbuf, BUFFER_SIZE, "%s\n\n(%s)", _("Cannot open the note."), _("encryption support is disabled"));
-			gui_create_dialog (GTK_MESSAGE_ERROR, tmpbuf, GTK_WINDOW(appGUI->main_window));
+			utl_gui_create_dialog (GTK_MESSAGE_ERROR, tmpbuf, GTK_WINDOW(appGUI->main_window));
 			g_free (current_filename);
 			return;
 
