diff -Naur scummvm-2.1.2/audio/softsynth/fluidsynth.cpp scummvm-2.1.2.patched/audio/softsynth/fluidsynth.cpp
--- scummvm-2.1.2/audio/softsynth/fluidsynth.cpp	2020-03-30 18:32:51.000000000 -0400
+++ scummvm-2.1.2.patched/audio/softsynth/fluidsynth.cpp	2020-04-11 17:40:39.247407731 -0400
@@ -20,10 +20,13 @@
  *
  */
 
-#include "common/scummsys.h"
+#include "config.h"
 
 #ifdef USE_FLUIDSYNTH
 
+#include <fluidsynth.h>
+
+#include "common/scummsys.h"
 #include "common/config-manager.h"
 #include "common/error.h"
 #include "common/system.h"
@@ -35,8 +38,6 @@
 #include "backends/platform/ios7/ios7_common.h"
 #endif
 
-#include <fluidsynth.h>
-
 class MidiDriver_FluidSynth : public MidiDriver_Emulated {
 private:
 	MidiChannel_MPU401 _midiChannels[16];
