diff -Nur libmimedir-0.5.1.orig/Makefile.in libmimedir-0.5.1/Makefile.in
--- libmimedir-0.5.1.orig/Makefile.in	2008-06-03 11:58:26.000000000 -0500
+++ libmimedir-0.5.1/Makefile.in	2009-03-22 23:30:41.953080041 -0500
@@ -55,9 +55,11 @@
 	./test
 
 install: all
-	${INSTALL} -m 444 libmimedir.h ${includedir}
-	${INSTALL} -m 444 libmimedir.a ${libdir}
-	$(LIBTOOL) --mode=install ${INSTALL} -c -m 444 libmimedir.la ${libdir}/libmimedir.la
+	${INSTALL} -d 755 $(DESTDIR)${includedir}
+	${INSTALL} -m 644 libmimedir.h $(DESTDIR)${includedir}
+	${INSTALL} -d 755 $(DESTDIR)${libdir}
+	${INSTALL} -m 644 libmimedir.a $(DESTDIR)${libdir}
+	$(LIBTOOL) --mode=install ${INSTALL} -c -m 644 libmimedir.la $(DESTDIR)${libdir}/libmimedir.la
 
 clean:
 	rm -f $(OBJS) $(LOBJS) test.o *.a *.la *.core
