diff -Naur a/build b/build
--- a/build	2006-09-10 16:02:58.000000000 +0100
+++ b/build	2017-05-01 06:05:08.309903881 +0100
@@ -61,7 +61,7 @@
 
 echo "Compiling AtomicParsley.cpp ..."
 
-g++ -g -O2 -Wall -Wno-unused $EXTRA_CPP_FLAGS -MT obj_files/AtomicParsley.o -MD -MP -MF "./obj_files/AtomicParsley.Tpo" -c -o obj_files/AtomicParsley.o AtomicParsley.cpp
+g++ -fpermissive -g -O2 -Wall -Wno-unused $EXTRA_CPP_FLAGS -MT obj_files/AtomicParsley.o -MD -MP -MF "./obj_files/AtomicParsley.Tpo" -c -o obj_files/AtomicParsley.o AtomicParsley.cpp
 
 echo "Compiling AP_AtomExtracts.cpp ..."
 
