diff -up cde-2.5.2/programs/dtksh/ksh93/src/cmd/ksh93/sh/xec.c.orig cde-2.5.2/programs/dtksh/ksh93/src/cmd/ksh93/sh/xec.c
--- cde-2.5.2/programs/dtksh/ksh93/src/cmd/ksh93/sh/xec.c.orig	2025-12-07 14:29:18.583782183 -0600
+++ cde-2.5.2/programs/dtksh/ksh93/src/cmd/ksh93/sh/xec.c	2025-12-07 14:33:26.382717822 -0600
@@ -1963,7 +1963,7 @@ int sh_exec(register const Shnode_t *t,
 			int flag = errorflg|OPTIMIZE_FLAG;
 			struct dolnod	*argsav=0;
 			struct comnod	*tp;
-			char *cp, *trap, *nullptr = 0;
+			char *cp, *trap, *nulptr = 0;
 			int nameref, refresh=1;
 			char *av[5];
 #if SHOPT_OPTIMIZE
@@ -2008,7 +2008,7 @@ int sh_exec(register const Shnode_t *t,
 					save_prompt = sh.nextprompt;
 					sh.nextprompt = 3;
 					sh.timeout = 0;
-					sh.exitval=sh_readline(&nullptr,0,1,(size_t)0,1000*sh.st.tmout);
+					sh.exitval=sh_readline(&nulptr,0,1,(size_t)0,1000*sh.st.tmout);
 					sh.nextprompt = save_prompt;
 					if(sh.exitval||sfeof(sfstdin)||sferror(sfstdin)
 					{
