--- install.sh.orig	2007-02-25 15:03:06.000000000 -0600
+++ install.sh	2007-02-25 15:05:54.000000000 -0600
@@ -10,9 +10,10 @@
 fi
 
 # Find kernel install path
-kernel_path="/lib/modules/`uname -r`"
+kernel_path="/lib/modules/$KERNELVERSION"
 
-mkdir -p "$kernel_path/misc"
-cp "$module" "$kernel_path/misc"
+mkdir -p "$DESTDIR/$kernel_path/misc"
+cp "$module" "$DESTDIR/$kernel_path/misc"
+
+# /sbin/depmod -a	# Done in doinst.sh
 
-/sbin/depmod -a
