--- a/src/main.py
+++ b/src/main.py
@@ -45,7 +45,7 @@
         Priority = Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION
         with open(CSS_PATH+'style.css', 'r') as css_file:
             css = bytes(css_file.read(), 'utf-8')
-        css_provider.load_from_data(css,len(css))
+        css_provider.load_from_data(css)
         Gtk.StyleContext.add_provider_for_display(Gdk.Display.get_default(), css_provider, Priority)
 
         launch_maximized = self.settings.get_boolean("launch-maximized")
