--- lbench-1.9/zfuncs.cc	2012-07-29 08:48:24.000000000 -0300
+++ lbench-1.9_patched/zfuncs.cc	2012-09-13 19:00:01.000000000 -0300
@@ -3232,6 +3232,7 @@
    using namespace zfuncs;
 
    char           work[200];
+   char           doc[200];
    char           logfile[200], oldlog[200];
    cchar          *appfile;
    int            secs, err;
@@ -3254,7 +3255,8 @@
    strncatv(zdatadir,199,work,"/share/",zappname,"/data",null);            //  /prefix/share/appname/data
    strncatv(zicondir,199,work,"/share/",zappname,"/icons",null);           //  /prefix/share/appname/icons
    strncatv(zlocalesdir,199,work,"/share/",zappname,"/locales",null);      //  /prefix/share/appname/locales
-   strncatv(zdocdir,199,work,"/share/doc/",zappname,null);                 //  /prefix/share/doc/appname
+   strncpy0(doc,DOCDIR,199);
+   strcpy(zdocdir,doc);                                                //  as defined in Makefile
    
    snprintf(zuserdir,199,"%s/.%s",getenv("HOME"),zappname);                //  /home/user/.appname/
    err = stat(zuserdir,&statdat);                                          //  does it exist already?
--- lbench-1.9/Makefile	2012-07-29 08:48:24.000000000 -0300
+++ lbench-1.9_patched/Makefile	2012-09-13 18:54:12.000000000 -0300
@@ -28,7 +28,7 @@
 	$(CXX) $(CFLAGS) -o $(PROGRAM).o $(SOURCE)
 
 zfuncs.o: zfuncs.cc
-	$(CXX) $(CFLAGS) zfuncs.cc -D PREFIX=\"$(PREFIX)\"
+	$(CXX) $(CFLAGS) zfuncs.cc -D PREFIX=\"$(PREFIX)\" -D DOCDIR=\"$(DOCDIR)\"
 
 install: $(PROGRAM)
 	mkdir -p  $(DESTDIR)$(BINDIR)
