Description: Enable anti-aliasing by default
Author: Stephen Kitt <skitt@debian.org>
Bug-Debian: https://bugs.debian.org/813715

--- a/src/zoomrc
+++ b/src/zoomrc
@@ -25,7 +25,7 @@
   font 9 "font3"                                      symbolic
   
   size 80,30
-  antialias no
+  antialias yes
 
   colours (0,0,0), (255,0,0), (0,255,0), (255,255,0), (0,0,255), (255,0,255),
           (0,255,255), (255,255,204),
