--- setup.py.orig	2015-07-24 15:00:04.000000000 +1000
+++ setup.py	2019-06-24 21:34:06.883000000 +1000
@@ -35,6 +35,9 @@
         'Programming Language :: Python :: 3.2',
         'Programming Language :: Python :: 3.3',
         'Programming Language :: Python :: 3.4',
+        'Programming Language :: Python :: 3.5',
+        'Programming Language :: Python :: 3.6',
+        'Programming Language :: Python :: 3.7',
         'Topic :: Software Development :: Libraries :: Python Modules',
         'Topic :: Text Processing :: Markup :: XML',
     ],
--- feedparser/feedparser.py.orig	2015-07-24 15:00:04.000000000 +1000
+++ feedparser/feedparser.py	2019-06-24 22:15:09.636000000 +1000
@@ -130,6 +130,8 @@
 #ACCEPTABLE_URI_SCHEMES = ()
 
 # ---------- required modules (should come with any Python distribution) ----------
+from __future__ import absolute_import, unicode_literals
+
 import cgi
 import codecs
 import copy
