diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -19,6 +19,8 @@
     build/ltmain.sh \
     build/missing
 
+pkgconfig_DATA = udis86.pc
+
 include_ladir = ${includedir}
 include_la_HEADERS = udis86.h
 
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -68,7 +68,10 @@
 
 AC_CHECK_HEADERS([assert.h stdio.h])
 
+m4_ifdef([PKG_INSTALLDIR], [PKG_INSTALLDIR], [AC_SUBST([pkgconfigdir], [${libdir}/pkgconfig])])
+
 AC_CONFIG_FILES([
+    udis86.pc
     Makefile
     scripts/Makefile
     libudis86/Makefile
diff --git a/udis86.pc.in b/udis86.pc.in
new file mode 100644
--- /dev/null
+++ b/udis86.pc.in
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: @PACKAGE_NAME@
+Description: Disassembler library for x86 and x86-64
+Version: @PACKAGE_VERSION@
+Libs: -L${libdir} -ludis86
+Cflags: -I${includedir}
