diff -Naur rman-3.2.orig/rman.c rman-3.2/rman.c
--- rman-3.2.orig/rman.c	2003-07-26 21:15:00.000000000 +0200
+++ rman-3.2/rman.c	2025-04-04 08:19:27.736288000 +0200
@@ -268,7 +268,7 @@
 	else return 1;
 }
 
-int lcexceptionscmp(const char **a, const char **b) { return stricmp(*a, *b); }
+int lcexceptionscmp(const void *a, const void  *b) { return stricmp(*(char**)a, *(char**)b); }
 
 int
 strincmp(const char *s1, const char *s2, size_t n) {
@@ -4171,7 +4171,7 @@
   } else if (fTclTk && checkcmd("OP")) {
     source_struct(BEGINBODY);
     for (i=0; i<3; i++) {
-	 if (fcharout) { source_out(tcltkOP[i]); source_out(": "); }
+	 if (fcharout) { source_out((char *)tcltkOP[i]); source_out(": "); }
 	 stagadd(BEGINBOLD); p=source_out_word(p); stagadd(ENDBOLD); 
 	 source_struct(SHORTLINE);
     }
