diff -Naur gnucash-2.4.11.orig/src/core-utils/gnc-filepath-utils.c gnucash-2.4.11/src/core-utils/gnc-filepath-utils.c
--- gnucash-2.4.11.orig/src/core-utils/gnc-filepath-utils.c	2012-01-10 00:58:04.000000000 +0100
+++ gnucash-2.4.11/src/core-utils/gnc-filepath-utils.c	2012-10-06 18:05:10.853171945 +0200
@@ -47,7 +47,6 @@
 #include "gnc-filepath-utils.h"
 
 #ifdef _MSC_VER
-#include <glib/gwin32.h>
 #define PATH_MAX MAXPATHLEN
 #endif
 
diff -Naur gnucash-2.4.11.orig/src/libqof/qof/gnc-date.c gnucash-2.4.11/src/libqof/qof/gnc-date.c
--- gnucash-2.4.11.orig/src/libqof/qof/gnc-date.c	2012-01-10 00:58:15.000000000 +0100
+++ gnucash-2.4.11/src/libqof/qof/gnc-date.c	2012-10-06 18:06:22.286171945 +0200
@@ -40,7 +40,6 @@
 #include <string.h>
 #include <time.h>
 
-#include <glib.h>
 #include <glib/gi18n.h>
 
 #include "gnc-date-p.h"
