Index: gnucash-2.4.5/src/core-utils/gnc-filepath-utils.c
===================================================================
--- gnucash-2.4.5.orig/src/core-utils/gnc-filepath-utils.c
+++ gnucash-2.4.5/src/core-utils/gnc-filepath-utils.c
@@ -47,7 +47,6 @@
 #include "gnc-filepath-utils.h"
 
 #ifdef _MSC_VER
-#include <glib/gwin32.h>
 #define PATH_MAX MAXPATHLEN
 #endif
 
Index: gnucash-2.4.5/src/gnome-utils/gnc-frequency.c
===================================================================
--- gnucash-2.4.5.orig/src/gnome-utils/gnc-frequency.c
+++ gnucash-2.4.5/src/gnome-utils/gnc-frequency.c
@@ -26,7 +26,7 @@
 #include "config.h"
 
 #include <gtk/gtk.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <math.h>
 #include <time.h>
 
Index: gnucash-2.4.5/src/libqof/qof/gnc-date.c
===================================================================
--- gnucash-2.4.5.orig/src/libqof/qof/gnc-date.c
+++ gnucash-2.4.5/src/libqof/qof/gnc-date.c
@@ -40,7 +40,6 @@
 #include <string.h>
 #include <time.h>
 
-#include <glib.h>
 #include <glib/gi18n.h>
 
 #include "gnc-date-p.h"
