diff -Nur stfl-0.21.orig//Makefile stfl-0.21/Makefile
--- stfl-0.21.orig//Makefile	2009-06-19 12:30:52.000000000 -0500
+++ stfl-0.21/Makefile	2011-01-11 22:21:10.623044411 -0600
@@ -21,7 +21,8 @@
 include Makefile.cfg
 
 export CC = gcc -pthread
-export CFLAGS += -I. -Wall -Os -ggdb -D_GNU_SOURCE -fPIC
+OPTS = -Os
+export CFLAGS += -I. -Wall $(OPTS) -ggdb -D_GNU_SOURCE -fPIC
 export LDLIBS += -lncursesw
 
 SONAME  := libstfl.so.0
