diff -Nur stfl-0.22.a/python/Makefile.snippet stfl-0.22.b/python/Makefile.snippet
--- stfl-0.22.a/python/Makefile.snippet	2009-06-18 01:27:16.000000000 -0700
+++ stfl-0.22.b/python/Makefile.snippet	2012-10-05 18:24:00.912721777 -0700
@@ -33,8 +33,7 @@
 
 install_python: python/_stfl.so python/stfl.py python/stfl.pyc
 	mkdir -p $(DESTDIR)$(PYTHON_SITEARCH)/
-	mkdir -p $(DESTDIR)$(PYTHON_SITEARCH)/lib-dynload/
-	cp python/_stfl.so $(DESTDIR)$(PYTHON_SITEARCH)/lib-dynload/
+	cp python/_stfl.so $(DESTDIR)$(PYTHON_SITEARCH)/
 	cp python/stfl.pyc $(DESTDIR)$(PYTHON_SITEARCH)/
 	cp python/stfl.py $(DESTDIR)$(PYTHON_SITEARCH)/
 
