--- src/Makefile.am.orig	2014-02-14 23:45:45.000000000 +1000
+++ src/Makefile.am	2014-10-26 21:59:33.220234317 +1000
@@ -19,6 +19,8 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
 # USA.
 
+.NOTPARALLEL:
+
 ###################################
 #       Extra directories         #
 ###################################
--- Makefile.am.orig	2014-01-24 04:57:55.000000000 +1000
+++ Makefile.am	2014-10-26 21:58:47.090622189 +1000
@@ -15,6 +15,8 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
 # USA.
 
+.NOTPARALLEL:
+
 ACLOCAL_AMFLAGS = -I m4
 
 EXTRA_DIST = \
