diff -Naur pulseaudio-0.9.23-a/configure pulseaudio-0.9.23-b/configure
--- pulseaudio-0.9.23-a/configure	2011-06-24 05:56:32.036894840 +0900
+++ pulseaudio-0.9.23-b/configure	2011-07-19 03:08:08.234282954 +0900
@@ -20360,7 +20360,7 @@
 
 LIBWRAP_LIBS=
 saved_LIBS="$LIBS"
-LIBS="$LIBS -lwrap"
+LIBS="$LIBS -lnsl -lwrap"
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for tcpwrap library and headers" >&5
 $as_echo_n "checking for tcpwrap library and headers... " >&6; }
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
@@ -20382,7 +20382,7 @@
 
 $as_echo "#define HAVE_LIBWRAP /**/" >>confdefs.h
 
-LIBWRAP_LIBS="-lwrap"
+LIBWRAP_LIBS="-lnsl -lwrap"
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 else
