diff -r 6670ea4caf96 -r 5d8d64f95c49 Rakefile
--- a/Rakefile
+++ b/Rakefile
@@ -65,7 +65,7 @@
   "revision"   => "3224", #< Latest stable
   "cflags"     => "-Wall -Werror -Wpointer-arith -Wstrict-prototypes -Wunused -Wshadow -std=gnu99",
   "cpppath"    => "-I. -I$(builddir) -Isrc -Isrc/shared -Isrc/subtle -idirafter$(hdrdir) -idirafter$(archdir)",
-  "ldflags"    => "-L$(libdir) $(rpath) $(LIBS) -l$(RUBY_SO_NAME)",
+  "ldflags"    => "$(rpath) -L$(libdir) $(LIBS) -l$(RUBY_SO_NAME)",
   "extflags"   => "$(LDFLAGS) $(rpath) $(LIBS) -l$(RUBY_SO_NAME)",
   "rpath"      => "-L$(libdir) -Wl,-rpath=$(libdir)",
   "checksums"  => []
@@ -258,7 +258,7 @@
     make_config unless checksums
   else
     # Check version
-    if 1 != RbConfig::CONFIG["MAJOR"].to_i or 9 != RbConfig::CONFIG["MINOR"].to_i
+    if 1 == RbConfig::CONFIG["MAJOR"].to_i and 9 > RbConfig::CONFIG["MINOR"].to_i
       fail("Ruby 1.9.0 or higher required")
     end
 
