diff -r 43d1d0478c46 -r 31e0a9716e58 Rakefile
--- a/Rakefile
+++ b/Rakefile
@@ -397,6 +397,12 @@
         ret = false
 
         cflags, ldflags, libs = pkg_config("xft")
+
+        # Fix a bug in ruby 2.2.0 (https://bugs.ruby-lang.org/issues/10651)
+        if cflags.empty?
+          cflags << `#{$PKGCONFIG} --cflags xft`.chomp
+        end
+
         unless libs.nil?
           # Update flags
           @options["cpppath"] << " %s" % [ cflags ]
