--- fpm2-0.79orig/src/fpm.c	2011-01-17 13:51:17.000000000 +0100
+++ fpm2-0.79new/src/fpm.c	2015-01-21 20:45:05.777833506 +0100
@@ -1197,9 +1197,12 @@
 
 	columns = gtk_tree_view_get_columns(gui->main_clist);
 	while (columns != NULL) {
-	    ini->columns_width[i] = gtk_tree_view_column_get_width (columns->data);
-	    columns = g_list_next(columns);
-	    i++;
+		gint width;
+
+		width = gtk_tree_view_column_get_width (columns->data);
+		if (width != 0) ini->columns_width[i] = width;
+		columns = g_list_next(columns);
+		i++;
 	}
 	g_list_free(columns);
 
