diff -ru ../farsight2-0.0.21.orig/common/gtk-doc.mak ./common/gtk-doc.mak
--- ../farsight2-0.0.21.orig/common/gtk-doc.mak	2010-06-24 00:20:40.000000000 +0200
+++ ./common/gtk-doc.mak	2010-09-08 01:37:27.036523000 +0200
@@ -232,6 +232,6 @@
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
diff -ru ../farsight2-0.0.21.orig/python/Makefile.am ./python/Makefile.am
--- ../farsight2-0.0.21.orig/python/Makefile.am	2009-05-08 23:34:14.000000000 +0200
+++ ./python/Makefile.am	2010-09-08 01:40:08.948271000 +0200
@@ -1,5 +1,5 @@
-PYDEFS=`pkg-config --variable=defsdir pygobject-2.0`
-[ -z "$PYDEFS" && PYDEFS=`pkg-config --variable=defsdir pygtk-2.0`]
+PYDEFS=`pkg-config --variable=defsdir pygobject-2.0 || \
+			pkg-config --variable=defsdir pygtk-2.0`
 GSTPYDEFS=`pkg-config --variable=defsdir gst-python-0.10`
 
 AM_CPPFLAGS =						\
