From: Emanuele Rocca <ema@debian.org>
Date: Thu, 16 Sep 2021 12:33:54 +0200
Subject: configure.in: invoke AM_INIT_AUTOMAKE with "foreign"

---
 configure.in | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/configure.in b/configure.in
index ddd8043..45e4e89 100644
--- a/configure.in
+++ b/configure.in
@@ -1,5 +1,6 @@
-AC_INIT(urlview.c)
-AM_INIT_AUTOMAKE(urlview, 0.9)
+AC_INIT([urlview], [0.9])
+AC_CONFIG_SRCDIR([urlview.c])
+AM_INIT_AUTOMAKE([foreign])
 dnl AM_CONFIG_HEADER(config.h)
 AM_PROG_CC_STDC
 CFLAGS="$CFLAGS -DURLVIEW -Dunix"
