--- sample.urlview~	2007-06-08 14:11:09.000000000 +0200
+++ sample.urlview	2007-06-08 14:11:33.000000000 +0200
@@ -19,10 +19,10 @@
 #REGEXP (((http|https|ftp|gopher)|mailto):(//)?[^ >"\t]*|www\.[-a-z0-9.]+)[^ .,;\t>">\):]
 REGEXP (((http|https|ftp|gopher)|mailto)[.:][^ >"\t]*|www\.[-a-z0-9.]+)[^ .,;\t>">\):]
 
-# Command to invoke for selected URL.  Use lynx, netscape, or url_handler.sh
+# Command to invoke for selected URL.  Use lynx, firefox, or url_handler.sh
 # shell script.
 
 #COMMAND lynx %s
-#COMMAND netscape -remote 'openURL(%s)'
+#COMMAND firefox -remote 'openURL(%s)'
 COMMAND url_handler.sh
 
