--- url_handler.sh~	2007-06-08 14:12:20.000000000 +0200
+++ url_handler.sh	2007-06-08 14:14:06.000000000 +0200
@@ -28,8 +28,10 @@
 # VT: Launch in the same terminal
 
 # The lists of programs to be executed are
-https_prgs="/usr/X11R6/bin/netscape:XW /usr/bin/lynx:XT"
-http_prgs="/usr/bin/lynx:XT /usr/X11R6/bin/netscape:XW"
+https_prgs="/usr/bin/firefox:XW /usr/bin/lynx:XT"
+https_prgs="/usr/bin/firefox:XW /usr/bin/lynx:XT"
+http_prgs="/usr/bin/lynx:XT /usr/bin/firefox:XW"
+http_prgs="/usr/bin/lynx:XT /usr/bin/firefox:XW"
 mailto_prgs="/usr/bin/mutt:VT /usr/bin/elm:VT /usr/bin/pine:VT /usr/bin/mail:VT"
 gopher_prgs="/usr/bin/lynx:XT /usr/bin/gopher:XT"
 ftp_prgs="/usr/bin/lynx:XT /usr/bin/ncftp:XT"
