diff --git a/ax_blas.m4 b/ax_blas.m4
--- a/ax_blas.m4
+++ b/ax_blas.m4
@@ -117,12 +117,16 @@
 # BLAS in ATLAS library? (http://math-atlas.sourceforge.net/)
 if test $ax_blas_ok = no; then
 	AC_CHECK_LIB(atlas, ATL_xerbla,
-		[AC_CHECK_LIB(f77blas, $sgemm,
-		[AC_CHECK_LIB(cblas, cblas_dgemm,
-			[ax_blas_ok=yes
-			 BLAS_LIBS="-lcblas -lf77blas -latlas"],
-			[], [-lf77blas -latlas])],
-			[], [-latlas])])
+		[AC_CHECK_LIB(ptf77blas, $sgemm,
+			[AC_CHECK_LIB(ptcblas, cblas_dgemm,
+				[ax_blas_ok=yes; BLAS_LIBS="-lptcblas -lptf77blas -latlas"],,
+				[-lptf77blas -latlas])],
+			[AC_CHECK_LIB(f77blas, $sgemm,
+				[AC_CHECK_LIB(cblas, cblas_dgemm,
+					[ax_blas_ok=yes; BLAS_LIBS="-lcblas -lf77blas -latlas"],,
+					[-lf77blas -latlas])],,
+				[-latlas])],
+			[-latlas])])
 fi
 
 # BLAS in PhiPACK libraries? (requires generic BLAS lib, too)
