diff --git a/check/pdf-check.sh b/check/pdf-check.sh
--- a/check/pdf-check.sh
+++ b/check/pdf-check.sh
@@ -72,11 +72,11 @@
     echo "GS does not understand PDF; skipping this test" >& 2
     exit 77
 fi 
-actiondiff "$TMP2" "$REFDATA" 1000
+actiondiff "$TMP2" "$REFDATA" 1200
 
 action $POTRACE -r50 -p -L 0 -B 0 --opaque -b pdf -o "$TMP1" "$DATA"
 action "$GS" -q -dNOPAUSE -sDEVICE=pbmraw -g460x394 -r100x100 -sOutputFile="$TMP2" -- "$TMP1"
-actiondiff "$TMP2" "$REFDATA" 1200
+actiondiff "$TMP2" "$REFDATA" 1600
 
 action $POTRACE -r50 -p -L 0 -B 0 -A 160 -b pdf -o "$TMP1" "$DATA"
 action "$GS" -q -dNOPAUSE -sDEVICE=pbmraw -g568x528 -r100x100 -sOutputFile="$TMP2" -- "$TMP1"
diff --git a/check/postscript-check.sh b/check/postscript-check.sh
--- a/check/postscript-check.sh
+++ b/check/postscript-check.sh
@@ -70,11 +70,11 @@
     echo "Something is wrong with $GS; skipping this test" >& 2
     exit 77
 fi 
-actiondiff "$TMP2" "$REFDATA" 1000
+actiondiff "$TMP2" "$REFDATA" 1200
 
 action $POTRACE -r50 -p -L 0 -B 0 --opaque -o "$TMP1" "$DATA"
 action "$GS" -q -dNOPAUSE -sDEVICE=pbmraw -g460x394 -r100x100 -sOutputFile="$TMP2" -- "$TMP1"
-actiondiff "$TMP2" "$REFDATA" 1200
+actiondiff "$TMP2" "$REFDATA" 1600
 
 action $POTRACE -r50 -p -L 0 -B 0 -A 160 -o "$TMP1" "$DATA"
 action "$GS" -q -dNOPAUSE -sDEVICE=pbmraw -g568x528 -r100x100 -sOutputFile="$TMP2" -- "$TMP1"
