Description: Make makefile compatible with Debian's standards.
Author: Mateusz Łukasik <mati75@linuxmint.pl>
Last-Update: 2013-11-30

--- a/Makefile
+++ b/Makefile
@@ -7,7 +7,7 @@ CC=gcc
 CFLAGS+=-DWITH_ICONS
 # Uncomment this line if Openbox can display SVG icons 
 # Check SVG support with '$ ldd /usr/bin/openbox | grep svg', librsvg must appear..
-# CFLAGS+=-DWITH_SVG
+CFLAGS+=-DWITH_SVG
 
 prefix= /usr/local
 DESTDIR ?= $(prefix)
