diff --git a/yt_dlp/plugins.py b/yt_dlp/plugins.py
index 6eecdb4d0..85ff5f19b 100644
--- a/yt_dlp/plugins.py
+++ b/yt_dlp/plugins.py
@@ -88,7 +88,7 @@ def _get_package_paths(*root_paths, containing_folder='plugins'):
             candidate = path / parts
             if candidate.is_dir():
                 yield candidate
-            elif path.suffix in ('.zip', '.egg', '.whl'):
+            elif path.suffix in ('.zip', '.egg', '.whl') and not path.is_dir():
                 if parts in dirs_in_zip(path):
                     yield candidate
