diff -up libpst-0.6.76/src/libpst.c.incompatible-pointer-i686 libpst-0.6.76/src/libpst.c
--- libpst-0.6.76/src/libpst.c.incompatible-pointer-i686	2024-01-22 17:39:52.792216627 +0100
+++ libpst-0.6.76/src/libpst.c	2024-01-22 17:40:35.788761273 +0100
@@ -3828,7 +3828,7 @@ static size_t pst_read_block_size(pst_fi
         return -1;
     }
     *buf = (char *) pst_malloc(inflated_size);
-    size_t result_size = inflated_size;
+    unsigned long result_size = inflated_size;
     if (uncompress((Bytef *) *buf, &result_size, (Bytef *) zbuf, size) != Z_OK || result_size != inflated_size) {
         DEBUG_WARN(("Failed to uncompress %i bytes to %i bytes, got %i\n", size, inflated_size, result_size));
         if (zbuf) free(zbuf);
