diff -Naur yaml-cpp-release-0.5.3.bak/include/yaml-cpp/node/detail/iterator.h yaml-cpp-release-0.5.3/include/yaml-cpp/node/detail/iterator.h
--- yaml-cpp-release-0.5.3.bak/include/yaml-cpp/node/detail/iterator.h	2016-01-11 07:11:40.000000000 +1300
+++ yaml-cpp-release-0.5.3/include/yaml-cpp/node/detail/iterator.h	2018-07-22 16:58:24.156667873 +1200
@@ -11,6 +11,7 @@
 #include "yaml-cpp/node/ptr.h"
 #include "yaml-cpp/node/detail/node_iterator.h"
 #include <boost/iterator/iterator_adaptor.hpp>
+#include <boost/next_prior.hpp>
 #include <boost/utility.hpp>
 
 namespace YAML {
