From 519cb78d38be04a3af79784b5326e21651ced6e1 Mon Sep 17 00:00:00 2001
From: Sude <lgogdownloader@gmail.com>
Date: Fri, 25 Mar 2016 23:09:17 +0200
Subject: [PATCH] Fix compiling with JsonCpp 1.7.1

---
 src/util.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/util.cpp b/src/util.cpp
index 4b98c45..3d9dfed 100644
--- a/src/util.cpp
+++ b/src/util.cpp
@@ -388,11 +388,11 @@ int Util::getTerminalWidth()
 void Util::getDownloaderUrlsFromJSON(const Json::Value &root, std::vector<std::string> &urls)
 {
     if(root.size() > 0) {
-        for(Json::ValueIterator it = root.begin() ; it != root.end() ; ++it)
+        for(Json::ValueConstIterator it = root.begin() ; it != root.end() ; ++it)
         {
             if (it.key() == "downloaderUrl")
             {
-                Json::Value& url = *it;
+                Json::Value url = *it;
                 urls.push_back(url.asString());
             }
             else
