diff -Naur devedeng-4.16.0.orig/src/devedeng/configuration_data.py devedeng-4.16.0/src/devedeng/configuration_data.py
--- devedeng-4.16.0.orig/src/devedeng/configuration_data.py	2019-10-15 00:26:37.000000000 +0200
+++ devedeng-4.16.0/src/devedeng/configuration_data.py	2026-02-16 08:22:01.731160416 +0100
@@ -18,7 +18,7 @@
 
 from gi.repository import GObject
 import os
-import pkg_resources
+import importlib_metadata
 
 
 class configuration(GObject.GObject):
@@ -36,7 +36,7 @@
 
     def __init__(self):
         GObject.GObject.__init__(self)
-        self.version = str(pkg_resources.require("devedeng")[0].version)
+        self.version = str(importlib_metadata.version("devedeng"))
         print("Version: " + self.version)
 
     def _get_config_folder(self):
