diff -Naur calf-0.0.19kx/configure.ac calf-0.0.19kx.patched/configure.ac
--- calf-0.0.19kx/configure.ac	2014-03-08 16:26:41.000000000 -0500
+++ calf-0.0.19kx.patched/configure.ac	2017-07-06 04:56:04.737642086 -0400
@@ -62,24 +62,9 @@
 AC_CHECK_LIB(fftw3, true, AC_MSG_ERROR([FFTW3 library (libfftw3) not found]))
 AC_CHECK_LIB(fftw3f, true, AC_MSG_ERROR([Single-precision FFTW3 library (libfftw3f) not found]))
 
-# JACK
-AC_CHECK_HEADERS([jack/jack.h], have_jack_header=yes, have_jack_header=no)
-
 # GLib
 PKG_CHECK_MODULES(GLIB_DEPS, glib-2.0 >= 2.0.0 gthread-2.0 >= 2.0.0, true, AC_MSG_ERROR([Calf requires glib2 and gthread2]))
 
-# Fluidsynth
-PKG_CHECK_MODULES(FLUIDSYNTH_DEPS, fluidsynth >= 1.0.7, true, AC_MSG_ERROR([fluidsynth library not found]))
-
-if test "$have_jack_header" = "yes"; then
-  PKG_CHECK_MODULES(JACK_DEPS, jack >= 0.103.0,
-    AC_CHECK_LIB([jack], [jack_port_register], JACK_FOUND="yes", ),)
-fi
-
-if test "$JACK_FOUND" = "yes"; then
-  PKG_CHECK_MODULES(JACK_MIDI_DEPS, jack >= 0.105.0, OLD_JACK="no", OLD_JACK="yes")
-fi
-
 PKG_CHECK_MODULES(LV2_DEPS, lv2 >= 1, LV2_FOUND="yes", LV2_FOUND="no")
 
 PKG_CHECK_MODULES(LASH_DEPS, lash-1.0 >= 0.6.0,
