--- OutputFormats.cpp.orig	2023-10-15 11:47:37.000000000 +0200
+++ OutputFormats.cpp	2025-09-12 15:48:12.584775148 +0200
@@ -219,6 +219,16 @@
   std::stringstream ss;
 
   for(auto c:UniqueChannels(List)) {
+     ss << "LCN.";
+     if(c.LCN != -1) {
+        int t = ss.width(4); ss.fill('0');
+        ss << c.LCN;
+        if(c.LCN_minor != -1)
+           ss << "." << c.LCN_minor;
+        ss.width(t), ss.fill(' ');
+     } else ss << "----";
+     ss << "-";
+
      if (c.Name.empty())
         ss << "???";
      else
