diff -Naur gnome-shell-extensions-41.2/data/meson.build gnome-shell-extensions-41.2-patched/data/meson.build
--- gnome-shell-extensions-41.2/data/meson.build	2022-02-15 19:54:18.000000000 +0000
+++ gnome-shell-extensions-41.2-patched/data/meson.build	2023-02-05 16:42:11.238927940 +0000
@@ -1,5 +1,5 @@
 session_desktop = 'gnome-classic.desktop'
-i18n.merge_file('',
+i18n.merge_file(
   input:  session_desktop + '.in',
   output: session_desktop,
   po_dir: '../po',
