From e1f9005cf5ebafbed6be4b0cae4bee59112ea780 Mon Sep 17 00:00:00 2001
From: Ale Martinez <amtriathlon@gmail.com>
Date: Thu, 11 Jun 2020 11:36:54 -0300
Subject: Fix Qwt incompatibilities with Qt 5.15.0


diff --git a/qwt/src/qwt_compass_rose.cpp b/qwt/src/qwt_compass_rose.cpp
index 21a35f244..d67175940 100644
--- a/qwt/src/qwt_compass_rose.cpp
+++ b/qwt/src/qwt_compass_rose.cpp
@@ -11,6 +11,7 @@
 #include "qwt_point_polar.h"
 #include "qwt_painter.h"
 #include <qpainter.h>
+#include <qpainterpath.h>
 
 static QPointF qwtIntersection( 
     QPointF p11, QPointF p12, QPointF p21, QPointF p22 )
diff --git a/qwt/src/qwt_dial_needle.cpp b/qwt/src/qwt_dial_needle.cpp
index 1b53a3d5b..694f7f2bb 100644
--- a/qwt/src/qwt_dial_needle.cpp
+++ b/qwt/src/qwt_dial_needle.cpp
@@ -13,6 +13,7 @@
 #include "qwt_painter.h"
 #include <qapplication.h>
 #include <qpainter.h>
+#include <qpainterpath.h>
 
 #if QT_VERSION < 0x040601
 #define qFastSin(x) qSin(x)
diff --git a/qwt/src/qwt_null_paintdevice.cpp b/qwt/src/qwt_null_paintdevice.cpp
index db1611da2..b9b5dafb6 100644
--- a/qwt/src/qwt_null_paintdevice.cpp
+++ b/qwt/src/qwt_null_paintdevice.cpp
@@ -9,6 +9,7 @@
 
 #include "qwt_null_paintdevice.h"
 #include <qpaintengine.h>
+#include <qpainterpath.h>
 #include <qpixmap.h>
 
 class QwtNullPaintDevice::PrivateData
diff --git a/qwt/src/qwt_painter.cpp b/qwt/src/qwt_painter.cpp
index 0bbf258c5..07b217033 100644
--- a/qwt/src/qwt_painter.cpp
+++ b/qwt/src/qwt_painter.cpp
@@ -19,6 +19,7 @@
 #include <qpainter.h>
 #include <qpalette.h>
 #include <qpaintdevice.h>
+#include <qpainterpath.h>
 #include <qpixmap.h>
 #include <qstyle.h>
 #include <qtextdocument.h>
diff --git a/qwt/src/qwt_painter_command.h b/qwt/src/qwt_painter_command.h
index 2da597a7f..4fafd555c 100644
--- a/qwt/src/qwt_painter_command.h
+++ b/qwt/src/qwt_painter_command.h
@@ -15,6 +15,7 @@
 #include <qpixmap.h>
 #include <qimage.h>
 #include <qpolygon.h>
+#include <qpainterpath.h>
 
 class QPainterPath;
 
diff --git a/qwt/src/qwt_plot_glcanvas.h b/qwt/src/qwt_plot_glcanvas.h
index 2ff1cf2e3..89462bba9 100644
--- a/qwt/src/qwt_plot_glcanvas.h
+++ b/qwt/src/qwt_plot_glcanvas.h
@@ -13,6 +13,7 @@
 #include "qwt_global.h"
 #include <qframe.h>
 #include <qgl.h>
+#include <qpainterpath.h>
 
 class QwtPlot;
 
diff --git a/qwt/src/qwt_plot_panner.cpp b/qwt/src/qwt_plot_panner.cpp
index 8ed3dbee9..44de4cbef 100644
--- a/qwt/src/qwt_plot_panner.cpp
+++ b/qwt/src/qwt_plot_panner.cpp
@@ -15,6 +15,7 @@
 #include <qbitmap.h>
 #include <qstyle.h>
 #include <qstyleoption.h>
+#include <qpainterpath.h>
 
 static QBitmap qwtBorderMask( const QWidget *canvas, const QSize &size )
 {
diff --git a/qwt/src/qwt_plot_renderer.cpp b/qwt/src/qwt_plot_renderer.cpp
index 3cdcd8c0e..09a5c0a10 100644
--- a/qwt/src/qwt_plot_renderer.cpp
+++ b/qwt/src/qwt_plot_renderer.cpp
@@ -19,6 +19,7 @@
 #include "qwt_math.h"
 #include <qpainter.h>
 #include <qpaintengine.h>
+#include <qpainterpath.h>
 #include <qtransform.h>
 #include <qprinter.h>
 #include <qprintdialog.h>
diff --git a/qwt/src/qwt_widget_overlay.cpp b/qwt/src/qwt_widget_overlay.cpp
index 07c6272e1..9a458277b 100644
--- a/qwt/src/qwt_widget_overlay.cpp
+++ b/qwt/src/qwt_widget_overlay.cpp
@@ -11,6 +11,7 @@
 #include "qwt_painter.h"
 #include <qpainter.h>
 #include <qpaintengine.h>
+#include <qpainterpath.h>
 #include <qimage.h>
 #include <qevent.h>
 
