From 372dd5c14415885c8808c4ad4729cb39ae717db0 Mon Sep 17 00:00:00 2001
From: Mark Liversedge <liversedge@gmail.com>
Date: Mon, 13 Jan 2020 20:21:00 +0000
Subject: SEGV on Overview Chart

.. when no rides available on new user.

Fixes #3295

diff --git a/src/Charts/OverviewWindow.cpp b/src/Charts/OverviewWindow.cpp
index bfbd31432..7aa0792df 100644
--- a/src/Charts/OverviewWindow.cpp
+++ b/src/Charts/OverviewWindow.cpp
@@ -558,6 +558,8 @@ static const QStringList timeInZonesWBAL = QStringList()
 void
 Card::setData(RideItem *item)
 {
+    if (item == NULL || item->ride() == NULL) return;
+
     // use ride colors in painting?
     ridecolor = item->color;
 
