--- sage-6.10/configure.old	2015-12-19 08:15:37.404497471 +0700
+++ sage-6.10/configure	2015-12-19 08:15:55.691461292 +0700
@@ -2965,21 +2965,6 @@
 
 #---------------------------------------------------------
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for root user" >&5
-$as_echo_n "checking for root user... " >&6; }
-
-uid=`id -u`
-if test "x$uid" == "x0"; then
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-    as_fn_error $? "You cannot build Sage as root, switch to an unpriviledged user" "$LINENO" 5
-else
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-
-fi;
-
-
 # Check whether we are on a supported platform
 # Make sure we can run config.sub.
 $SHELL "$ac_aux_dir/config.sub" sun4 >/dev/null 2>&1 ||
