--- sage-6.5/configure.old	2015-02-19 08:11:31.873711070 +0700
+++ sage-6.5/configure	2015-02-19 08:12:00.982710447 +0700
@@ -2942,19 +2942,19 @@
 
 
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for root user" >&5
-$as_echo_n "checking for root user... " >&6; }
+#{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for root user" >&5
+#$as_echo_n "checking for root user... " >&6; }
 
-uid=`id -u`
-if test "x$uid" == "x0"; then
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-    as_fn_error $? "You cannot build Sage as root, switch to a unpriviledged user" "$LINENO" 5
-else
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
+#uid=`id -u`
+#if test "x$uid" == "x0"; then
+#    { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
+#$as_echo "yes" >&6; }
+#    as_fn_error $? "You cannot build Sage as root, switch to a unpriviledged user" "$LINENO" 5
+#else
+#    { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
+#$as_echo "no" >&6; }
 
-fi;
+#fi;
 
 
 #---------------------------------------------------------
