diff --git a/libsxclient/src/openssl.c b/libsxclient/src/openssl.c
index 62bfc01..666430a 100644
--- a/libsxclient/src/openssl.c
+++ b/libsxclient/src/openssl.c
@@ -98,6 +98,8 @@ int sxi_sslctxfun(sxc_client_t *sx, curlev_t *ev, const struct curl_tlssessionin
         return -1;
     }
     SSL_CTX *ctx = (SSL_CTX*)info->internals;
+    if (!ctx)
+        return -EAGAIN;
     SSL_CTX_set_cert_verify_callback(ctx, ssl_verify_hostname, ev);
     return 0;
 }
