diff -Naur wesnoth-1.18.6.orig/SConstruct wesnoth-1.18.6/SConstruct
--- wesnoth-1.18.6.orig/SConstruct	2025-12-22 04:34:31.000000000 +0100
+++ wesnoth-1.18.6/SConstruct	2026-06-27 18:46:25.629239806 +0200
@@ -355,7 +355,6 @@
         else:
             have_libpthread = conf.CheckLib("pthread")
         return have_libpthread & \
-            conf.CheckBoost("system") & \
             conf.CheckBoost("asio", header_only = True) & \
             conf.CheckBoost("context") & \
             conf.CheckBoost("coroutine")
