--- src/terrain_translation.cpp.orig	2012-04-13 06:55:53.256767111 +0700
+++ src/terrain_translation.cpp	2012-04-13 06:56:38.718095606 +0700
@@ -800,7 +800,12 @@
 	tcode[1] = ((terrain.base & 0x00FF0000) >> 16);
 	tcode[2] = ((terrain.base & 0x0000FF00) >> 8);
 	tcode[3] =  (terrain.base & 0x000000FF);
-
+	tcode[4] = 0;
+	tcode[5] = 0;
+	tcode[6] = 0;
+	tcode[7] = 0;
+	tcode[8] = 0;
+	
 	if(terrain.overlay != NO_LAYER) {
 		tcode[4] = '^'; //the layer separator
 		tcode[5] = ((terrain.overlay & 0xFF000000) >> 24);
