diff -Naur a/untex.c b/untex.c
--- a/untex.c	1995-11-22 10:07:51.000000000 -0500
+++ b/untex.c	2017-06-26 17:18:46.919412040 -0400
@@ -43,6 +43,8 @@
 */
 
 
+#include <stdlib.h>
+#include <string.h>
 #include <stddef.h>
 #include <stdio.h>
 #include <ctype.h>
@@ -75,7 +77,7 @@
 char *umlaut(int c);
 int parsecmd(FILE *inf);
 int skipcomment(FILE *inf, int *c);
-int skip(FILE *inf, int c);
+void skip(FILE *inf, int c);
 
 int main(int argc, char *argv[])
 {
@@ -292,7 +294,7 @@
     return((*c != EOF));
 }
 
-int skip(FILE *inf, int c)
+void skip(FILE *inf, int c)
 {
     int cc, ec, openb, closeb;
     switch (c) {
@@ -304,7 +306,7 @@
     cc = getc(inf);
     if (cc != c) {
 	ungetc(cc, inf);
-	return(cc != EOF);
+	return;
     }
     openb  = 1;
     closeb = 0;
@@ -342,7 +344,6 @@
     int c, ce;
     char cmd[MAXCMDLEN], env[MAXCMDLEN], envtst[MAXCMDLEN];
     int fnc, cmdc, envc, i, openb, closeb;
-    int proceed;
     char fname[MAXFILENAMELEN];
     
     fnc = envc = cmdc = 0;
