diff -Naur td_lib-20150704/configure td_lib-20150704.patched/configure
--- td_lib-20150704/configure	2015-07-04 11:20:27.000000000 -0400
+++ td_lib-20150704.patched/configure	2017-03-05 11:29:40.888486013 -0500
@@ -5031,7 +5031,7 @@
 else
   ac_tmpdirs='/tmp /var/tmp /usr/tmp'
 fi
-for ac_dir in  . $ac_tmpdirs `eval echo $prefix/lib $exec_prefix/lib` ; do
+for ac_dir in  . $ac_tmpdirs  ; do
   test -d $ac_dir || continue
   test -w $ac_dir || continue # It is less confusing to not echo anything here.
   ac_xdir=$ac_dir/cf$$
