--- xmonad.cabal.orig	2012-05-16 01:04:06.426000004 +0300
+++ xmonad.cabal	2012-05-16 01:04:39.336999960 +0300
@@ -32,6 +32,10 @@
     description: Testing mode, only build minimal components
     default: False
 
+flag threaded
+    description: Use threaded runtime for xmonad binary
+    default: False
+
 library
     exposed-modules:    XMonad
                         XMonad.Main
@@ -88,3 +96,5 @@
         ghc-options:    -Werror
     if flag(testing) && flag(small_base)
         build-depends:  filepath, process, directory, mtl, unix, X11, base, containers, random, extensible-exceptions
+    if flag(threaded)
+        ghc-options:    -threaded
