--- xmonad.cabal.orig	2010-06-24 04:07:24.000000000 +0300
+++ xmonad.cabal	2010-06-24 04:09:57.000000000 +0300
@@ -32,6 +32,14 @@
     description: Testing mode, only build minimal components
     default: False
 
+flag threaded
+    description: Use threaded runtime for xmonad binary
+    default: False
+
+flag dynamic
+    description: Link xmonad binary dynamically
+    default: False
+
 library
     exposed-modules:    XMonad
                         XMonad.Main
@@ -77,3 +85,7 @@
         ghc-options:    -Werror
     if flag(testing) && flag(small_base)
         build-depends:  random
+    if flag(threaded)
+        ghc-options:    -threaded
+    if flag(dynamic)
+        ghc-options:    -dynamic
