--- xmonad.cabal.orig	2011-11-19 01:14:36.000000000 +0200
+++ xmonad.cabal	2012-01-05 14:57:48.000000000 +0200
@@ -32,6 +32,14 @@
     description: Testing mode, only build minimal components
     default: False
 
+flag threaded
+    description: Use threaded runtime for xmonad binary
+    default: False
+
+flag dynamic
+    description: Link xmonad binary dynamically
+    default: False
+
 library
     exposed-modules:    XMonad
                         XMonad.Main
@@ -88,3 +96,7 @@
         ghc-options:    -Werror
     if flag(testing) && flag(small_base)
         build-depends:  filepath, process, directory, mtl, unix, X11, base, containers, random, extensible-exceptions
+    if flag(threaded)
+        ghc-options:    -threaded
+    if flag(dynamic)
+        ghc-options:    -dynamic
