
--- src/dump-mplayer.c	2024-01-02 01:28:41.000000000 -0000
+++ src/dump-mplayer.c	2024-01-15 16:46:22.194419459 -0000
@@ -35,7 +35,7 @@
 {
 	int i;
 
-	fprintf(f, "%s:", s->service_name);
+	fprintf(f, "[LCN-%.4d]%s:", s->logical_channel_number, s->service_name);
 	xine_dump_dvb_parameters(f, t, flags);
 	fprintf(f, ":%i", s->pmt_pid);
     if (s->video_pid) {

--- src/dump-vdr.c	2024-01-02 01:28:41.000000000 -0000
+++ src/dump-vdr.c	2024-01-15 16:46:19.737419311 -0000
@@ -558,7 +558,7 @@
 
 	if (!flags->ca_select && s->scrambled)
 		return;
-	fprintf(f, "%s", s->service_name);
+	fprintf(f, "[LCN-%.4d]%s", s->logical_channel_number, s->service_name);
 
 	if (flags->dump_provider)
 		fprintf(f, ";%s", s->provider_name);

--- src/dump-xine.c	2024-01-02 01:28:41.000000000 -0000
+++ src/dump-xine.c	2024-01-15 16:46:17.076419151 -0000
@@ -113,10 +113,10 @@
 {
 	if (s->video_pid || s->audio_pid[0]) {
 		if (s->provider_name)
-			fprintf(f, "%s(%s):", s->service_name,
+			fprintf(f, "[LCN-%.4d]%s(%s):", s->logical_channel_number, s->service_name,
 				s->provider_name);
 		else
-			fprintf(f, "%s:", s->service_name);
+			fprintf(f, "[LCN-%.4d]%s:", s->logical_channel_number, s->service_name);
 		xine_dump_dvb_parameters(f, t, flags);
 		fprintf(f, ":%i:%i:%i", s->video_pid,
 			s->ac3_pid[0] ? s->ac3_pid[0] : s->audio_pid[0],
