--- Network/CGI/Protocol.hs.orig	2015-02-22 23:49:07.811998631 +0200
+++ Network/CGI/Protocol.hs	2015-02-22 23:49:47.854998351 +0200
@@ -44,7 +44,7 @@
 import qualified Data.ByteString.Lazy.Char8 as BS
 import Data.ByteString.Lazy.Char8 (ByteString)
 
-import Data.Typeable (Typeable(..), mkTyConApp, mkTyCon)
+import Data.OldTypeable (Typeable(..), mkTyConApp, mkTyCon)
 
 import Network.CGI.Header
 import Network.CGI.Multipart
--- Network/CGI/Monad.hs.orig	2015-02-22 23:49:22.836998526 +0200
+++ Network/CGI/Monad.hs	2015-02-22 23:50:04.933998232 +0200
@@ -40,7 +40,7 @@
 import Control.Monad.Reader (ReaderT(..), asks)
 import Control.Monad.Writer (WriterT(..), tell)
 import Control.Monad.Trans (MonadTrans, MonadIO, liftIO, lift)
-import Data.Typeable (Typeable(..), Typeable1(..), 
+import Data.OldTypeable (Typeable(..), Typeable1(..), 
                       mkTyConApp, mkTyCon)
 
 import Network.CGI.Protocol
