diff -Naur grip-2.96.orig/Makefile grip-2.96/Makefile
--- grip-2.96.orig/Makefile	2001-07-16 12:15:32.000000000 -0400
+++ grip-2.96/Makefile	2009-02-05 00:38:28.000000000 -0500
@@ -28,8 +28,8 @@
 ifeq ($(OS), FreeBSD)
 LIBS+=	-pthread
 endif
-PARLIBS=	cdparanoia/interface/libcdda_interface.a \
-		cdparanoia/paranoia/libcdda_paranoia.a
+PARLIBS=	-lcdda_interface \
+		-lcdda_paranoia
 
 # This is needed for "make install"
 OWNER  = root
diff -Naur grip-2.96.orig/cdpar.c grip-2.96/cdpar.c
--- grip-2.96.orig/cdpar.c	2001-07-16 12:15:32.000000000 -0400
+++ grip-2.96/cdpar.c	2009-02-05 00:39:02.000000000 -0500
@@ -36,9 +36,9 @@
 #define size16 short
 #define size32 int
 
-#include "cdparanoia/interface/cdda_interface.h"
-#include "cdparanoia/paranoia/cdda_paranoia.h"
-#include "cdparanoia/utils.h"
+#include <cdda_interface.h>
+#include <cdda_paranoia.h>
+#include <utils.h>
 
 static void PutNum(long num,int f,int endianness,int bytes);
 static void WriteWav(int f,long bytes);
diff -Naur grip-2.96.orig/grip.c grip-2.96/grip.c
--- grip-2.96.orig/grip.c	2001-07-16 12:15:32.000000000 -0400
+++ grip-2.96/grip.c	2009-02-05 00:42:26.000000000 -0500
@@ -57,8 +57,8 @@
 #ifdef CDPAR
 #define size16 short
 #define size32 int
-#include "cdparanoia/interface/cdda_interface.h"
-#include "cdparanoia/paranoia/cdda_paranoia.h"
+#include <cdda_interface.h>
+#include <cdda_paranoia.h>
 #endif
 
 void ShutDownCB(void);
