This is a simple patch to compile on 64 bit systems.
--- quickjs-2021-03-27/qjsc.c	2021-03-27 07:00:32.000000000 -0300
+++ quickjs-2021-03-27-modified/qjsc.c	2022-07-31 19:54:12.464023282 -0300
@@ -420,7 +420,7 @@
         pstrcpy(lib_dir, sizeof(lib_dir), exe_dir);
     } else {
         snprintf(inc_dir, sizeof(inc_dir), "%s/include/quickjs", CONFIG_PREFIX);
-        snprintf(lib_dir, sizeof(lib_dir), "%s/lib/quickjs", CONFIG_PREFIX);
+        snprintf(lib_dir, sizeof(lib_dir), "%s/quickjs", CONFIG_LIBDIR);
     }
     
     lto_suffix = "";
