This is a simple patch to compile on 64 bit systems.
--- quickjs-2024-01-13/qjsc.c	2024-01-13 07:20:39.000000000 -0300
+++ quickjs-2024-01-13m/qjsc.c	2024-02-04 01:43:19.516094810 -0300
@@ -419,7 +419,7 @@
         pstrcpy(lib_dir, sizeof(lib_dir), exe_dir);
     } else {
         snprintf(inc_dir, sizeof(inc_dir), "%s/include/quickjs", CONFIG_PREFIX);
-        snprintf(lib_dir, sizeof(lib_dir), "%s/lib/quickjs", CONFIG_PREFIX);
+        snprintf(lib_dir, sizeof(lib_dir), "%s/quickjs", CONFIG_LIBDIR);
     }
     
     lto_suffix = "";
