--- a/data/lightdm.conf
+++ b/data/lightdm.conf
@@ -22,7 +22,7 @@
 #start-default-seat=true
 #greeter-user=lightdm
 #minimum-display-number=0
-#minimum-vt=7
+minimum-vt=7
 #lock-memory=true
 #user-authority-in-system-dir=false
 #guest-account-script=guest-account
@@ -36,6 +36,8 @@
 #backup-logs=true
 #dbus-service=true
 
+sessions-directory=/usr/share/xsessions
+
 #
 # Seat configuration
 #
--- a/data/users.conf
+++ b/data/users.conf
@@ -9,6 +9,6 @@
 # hidden-shells = Shells that indicate a user cannot login
 #
 [UserList]
-minimum-uid=500
+minimum-uid=1000
 hidden-users=nobody nobody4 noaccess
 hidden-shells=/bin/false /usr/sbin/nologin /sbin/nologin
