diff -Naur oggvideotools-0.8a.orig/src/fileRepository.h oggvideotools-0.8a/src/fileRepository.h
--- oggvideotools-0.8a.orig/src/fileRepository.h	2010-05-30 21:11:10.000000000 +0200
+++ oggvideotools-0.8a/src/fileRepository.h	2012-09-13 16:42:11.550012490 +0200
@@ -23,6 +23,7 @@
 #define FILEREPOSITORY_H_
 
 #include <stdio.h>
+#include <unistd.h>
 
 #include "definition.h"
 #include "mediaRepository.h"
