--- a/xclock.c
+++ b/xclock.c
@@ -298,5 +298,6 @@ int main(int argc, char *argv[])
     XGCValues   gcv;
     u_long      valueMask;
     XmFontList  fontList = (XmFontList)NULL;
+    XmFontListEntry fontListEntry;
 
     static XtResource    resources[] = {
@@ -507,6 +508,9 @@ int main(int argc, char *argv[])
         }
     }
-    fontList = XmFontListCreate(appData.font, XmSTRING_ISO8859_1);
+    fontListEntry = XmFontListEntryCreate(XmSTRING_ISO8859_1,
+                                          XmFONT_IS_FONT, appData.font);
+    fontList = XmFontListAppendEntry(NULL, fontListEntry);
+    XmFontListEntryFree(&fontListEntry);
     if (fontList == (XmFontList)NULL) {
         fprintf(stderr, "xclock : Unable to create font list -- exiting\n");
         exit(-1);
@@ -1941,7 +1944,7 @@ static Widget CreateToggle(char *label, Widget parent)
     Arg         args[1];
     XmString    tcs;
 
-    tcs = XmStringLtoRCreate(label, XmSTRING_DEFAULT_CHARSET);
+    tcs = XmStringCreate(label, XmSTRING_DEFAULT_CHARSET);
 
     n = 0;
     XtSetArg(args[n], XmNlabelString, tcs);  n++;
@@ -1960,7 +1963,7 @@ static Widget CreatePushButton(char *label, Widget parent)
     Arg      args[1];
     XmString tcs;
 
-    tcs = XmStringLtoRCreate(label, XmSTRING_DEFAULT_CHARSET);
+    tcs = XmStringCreate(label, XmSTRING_DEFAULT_CHARSET);
 
     n = 0;
     XtSetArg(args[n], XmNlabelString, tcs);  n++;
