--- gcolor2-0.4.orig/src/main.c
+++ gcolor2-0.4/src/main.c
@@ -4,11 +4,15 @@
 
 #include <gtk/gtk.h>
 #include <stdio.h>
-
 #include "interface.h"
 #include "support.h"
 #include "callbacks.h"
 
+/* missing headers on original file */
+#include <stdlib.h>
+#include <string.h>
+#include <glib/gprintf.h>
+
 GtkListStore *liststore;
 GdkWindow    *gdkwin;
 gchar        *user_filename;
@@ -48,8 +52,9 @@
 
 void add_list_color (gchar *spec, gchar *name, gchar *type, gboolean is_new_color)
 {
+/* pixmap and mask don't used
 	GdkPixmap   *pixmap;
-	GdkBitmap   *mask;
+	GdkBitmap   *mask; */
 	GdkPixbuf   *buf;
 	GtkTreeIter  iter;
 	
@@ -103,7 +108,7 @@
 	gchar *name;
 	gchar  buffer[512];
 	gchar  spec[8];
-	gint   r, g, b, lr, lg, lb;
+	gint   r, g, b, lr=0, lg=0, lb=0;
 	
 	fp = fopen (filename, "r");
 	if (!fp)
@@ -140,7 +145,8 @@
 	GtkTreeViewColumn *column;
 	GtkWidget         *treeview;
 	GtkTreeSelection  *select;
-	gchar             *buf;
+/* dont used
+	gchar             *buf; */
 	
 #ifdef ENABLE_NLS
 	bindtextdomain (GETTEXT_PACKAGE, PACKAGE_LOCALE_DIR);
