--- AstroMenace/CMakeLists.orig.txt	2015-09-25 05:48:58.014083691 +0300
+++ AstroMenace/CMakeLists.txt	2015-09-25 05:59:11.322416282 +0300
@@ -192,7 +192,7 @@
 ELSE(FREETYPE_LIBRARY)
 	MESSAGE(FATAL_ERROR "freetype not found")
 ENDIF(FREETYPE_LIBRARY)
-FIND_PATH(FREETYPE_INCLUDE_DIR_freetype2 freetype/config/ftheader.h
+FIND_PATH(FREETYPE_INCLUDE_DIR_freetype2 /usr/include/freetype2/config/ftheader.h
 	HINTS
 	$ENV{FREETYPE_DIR}/include/freetype2
 	PATHS
