--- build.sh	2022-10-10 12:08:46.456483133 +0200
+++ build.sh.new	2022-10-10 12:08:01.796222720 +0200
@@ -512,12 +512,12 @@
     ;;
   *)
     if test -n "$private"; then
-      conf_rpath="$idir/lib/build2"
+      conf_rpath="$idir/lib64/build2"
     else
-      conf_rpath="$idir/lib"
+      conf_rpath="$idir/lib64"
     fi
 
-    conf_rpath_stage="$idir/lib"
+    conf_rpath_stage="$idir/lib64"
 
     if test -n "$sudo"; then
       conf_sudo="$sudo"
@@ -602,6 +602,7 @@
 config.bin.rpath="$conf_rpath" \
 config.install.root="$idir" \
 config.install.sudo="$conf_sudo" \
+config.install.lib="$conf_rpath" \
 $conf_exe_affixes \
 $private \
 $system_imports
@@ -666,6 +667,7 @@
 config.install.root="$idir" \
 config.install.data_root=root/stage \
 config.install.sudo="$conf_sudo" \
+config.install.lib="$conf_rpath" \
 $system_imports
 
 run build2/build2/b-boot $verbose $jobs install: build2/ bpkg/
@@ -692,6 +694,7 @@
 config.bin.rpath="$conf_rpath" \
 config.install.root="$idir" \
 config.install.sudo="$conf_sudo" \
+config.install.lib="$conf_rpath" \
 $conf_exe_affixes \
 $private
 
