diff -Nur redshift-1.12.orig/src/redshift-gtk/statusicon.py redshift-1.12/src/redshift-gtk/statusicon.py
--- redshift-1.12.orig/src/redshift-gtk/statusicon.py	2017-10-19 17:51:59.000000000 -0500
+++ redshift-1.12/src/redshift-gtk/statusicon.py	2021-01-03 21:48:00.721741441 -0600
@@ -32,11 +32,7 @@
 
 from gi.repository import Gtk, GLib
 
-try:
-    gi.require_version('AppIndicator3', '0.1')
-    from gi.repository import AppIndicator3 as appindicator
-except (ImportError, ValueError):
-    appindicator = None
+appindicator = None
 
 from .controller import RedshiftController
 from . import defs
