--- guacamole-server-0.9.14/src/terminal/terminal/typescript.h	2018-01-01 13:23:37.000000000 -0800
+++ guacamole-server-0.9.14-new/src/terminal/terminal/typescript.h	2018-01-26 17:11:22.180187667 -0800
@@ -97,7 +97,7 @@
      * The full path to the file which will contain the timing information for
      * this typescript.
      */
-    char timing_filename[GUAC_TERMINAL_TYPESCRIPT_MAX_NAME_LENGTH];
+    char timing_filename[GUAC_TERMINAL_TYPESCRIPT_MAX_NAME_LENGTH + sizeof(GUAC_TERMINAL_TYPESCRIPT_TIMING_SUFFIX) + 1];
 
     /**
      * The file descriptor of the file into which raw terminal output should be
