diff -Naur rubberband-1.6.0.orig/rubberband/RubberBandStretcher.h rubberband-1.6.0/rubberband/RubberBandStretcher.h
--- rubberband-1.6.0.orig/rubberband/RubberBandStretcher.h	2011-03-19 16:01:07.000000000 +0100
+++ rubberband-1.6.0/rubberband/RubberBandStretcher.h	2012-09-12 17:58:27.260000741 +0200
@@ -19,6 +19,7 @@
 #define RUBBERBAND_API_MAJOR_VERSION 2
 #define RUBBERBAND_API_MINOR_VERSION 4
 
+#include <unistd.h>
 #include <vector>
 #include <map>
 
