--- rubberband-1.6.0/Makefile.in.orig	2011-05-13 02:57:54.630362189 -0300
+++ rubberband-1.6.0/Makefile.in	2011-05-13 03:00:15.695391589 -0300
@@ -28,11 +28,11 @@
 
 INSTALL_BINDIR		:= @prefix@/bin
 INSTALL_INCDIR		:= @prefix@/include/rubberband
-INSTALL_LIBDIR		:= @prefix@/lib
-INSTALL_VAMPDIR		:= @prefix@/lib/vamp
-INSTALL_LADSPADIR	:= @prefix@/lib/ladspa
+INSTALL_LIBDIR		:= @prefix@/lib64
+INSTALL_VAMPDIR		:= @prefix@/lib64/vamp
+INSTALL_LADSPADIR	:= @prefix@/lib64/ladspa
 INSTALL_LRDFDIR		:= @prefix@/share/ladspa/rdf
-INSTALL_PKGDIR		:= @prefix@/lib/pkgconfig
+INSTALL_PKGDIR		:= @prefix@/lib64/pkgconfig
 
 all:	bin lib $(PROGRAM_TARGET) $(STATIC_TARGET) $(DYNAMIC_TARGET) $(VAMP_TARGET) $(LADSPA_TARGET)
 
