*** setup.py	2009-03-04 09:34:14.000000000 +0200
--- setup1.py	2009-03-04 09:36:03.000000000 +0200
***************
*** 1,5 ****
  #!/usr/bin/env python
! import stdeb
  __author__ = "Abhinav Sarkar <abhinav@abhinavsarkar.net>"
  __version__ = "0.1"
  __license__ = "GNU Lesser General Public License"
--- 1,5 ----
  #!/usr/bin/env python
! 
  __author__ = "Abhinav Sarkar <abhinav@abhinavsarkar.net>"
  __version__ = "0.1"
  __license__ = "GNU Lesser General Public License"
