--- af903x-drv.c.orig	2011-01-23 23:04:35.113333652 +0100
+++ af903x-drv.c	2011-01-23 22:47:28.676666674 +0100
@@ -1103,8 +1103,11 @@
 		PDC->StreamType = StreamType_DVBT_DATAGRAM;
 		PDC->UsbCtrlTimeOut = 1;
 
-		init_MUTEX(&PDC->powerLock);
-		init_MUTEX(&PDC->tunerLock);
+		//init_MUTEX(&PDC->powerLock);
+		//init_MUTEX(&PDC->tunerLock);
+		sema_init(&PDC->powerLock, 1);
+		sema_init(&PDC->tunerLock, 1);
+		
 		PDC->power_use_count = 0;
 
 		PDC->idVendor = udev->descriptor.idVendor;
@@ -1116,7 +1119,8 @@
 		PDC->fc[0].AVerFlags = 0x00;
 		PDC->fc[1].AVerFlags = 0x00;
 		
-		init_MUTEX(&PDC->regLock);
+		//init_MUTEX(&PDC->regLock);
+		sema_init(&PDC->regLock, 1);
 	}
 	else {
         	PDC->UsbCtrlTimeOut = 5;
@@ -1208,5 +1212,4 @@
 
 	return (error);
 }
-//EXPORT_SYMBOL(Device_init);
-
+//EXPORT_SYMBOL(Device_init);
\ Manca newline alla fine del file
