--- Makefile.orig	2010-11-10 04:46:22.000000000 +0100
+++ Makefile	2011-08-29 11:15:04.979000013 +0200
@@ -26,7 +26,7 @@
 #DEST = $(DESTDIR)/lib/modules/$(CURRENT)/kernel/$(MDIR)
 #KFIRMWARM = $(DESTDIR)/lib/firmware/
 INSTALL := $(shell which install)
-DEPMOD  := $(shell which depmod || which /sbin/depmod || echo "true You should now run depmod")
+#DEPMOD  := $(shell which depmod || which /sbin/depmod || echo "true You should now run depmod")
 
 # SYSTEM DVB setting
 DVB_FLAGS += -I$(KSRC)/drivers/media/dvb/dvb-core/
@@ -55,7 +55,7 @@
 install:
 	@install -d $(DESTDIR)/lib/modules/$(CURRENT)/kernel/drivers/media/dvb/a867
 	$(INSTALL) -c -D -m 644 -c a867.ko $(DESTDIR)/lib/modules/$(CURRENT)/kernel/drivers/media/dvb/a867/
-	/sbin/depmod -a
+#	/sbin/depmod -a
 
 remove:
 
