diff -Naur nted-1.10.18/voice.cpp nted-1.10.18.patched/voice.cpp
--- nted-1.10.18/voice.cpp	2011-01-27 08:09:44.000000000 -0500
+++ nted-1.10.18.patched/voice.cpp	2021-09-10 22:50:23.045105704 -0400
@@ -20,6 +20,7 @@
 /****************************************************************************************/
 
 #include <string.h>
+#include <stdint.h>
 #include "voice.h"
 #include "beaming.h"
 #include "staff.h"
@@ -1369,7 +1370,7 @@
 		return FALSE;
 	}
 	ref_duration = (method == 1) ? templ->getDuration() / 2 : templ->getDuration();
-	if ((min_pos_ptr = g_list_find(m_chord_or_rests, templ)) < 0) {
+	if ((intptr_t)(min_pos_ptr = g_list_find(m_chord_or_rests, templ)) < 0) {
 		NedResource::Abort("NedVoice::tryConvertToTuplet(1)");
 	}
 	if ((minpos = g_list_index(m_chord_or_rests, templ)) < 0) {
