diff --git a/utils/rbutilqt/rbutilqt.cpp b/utils/rbutilqt/rbutilqt.cpp
--- a/utils/rbutilqt/rbutilqt.cpp
+++ b/utils/rbutilqt/rbutilqt.cpp
@@ -303,21 +303,13 @@
 
     QString html = "<p>" + tr("Libraries used") + "</p>";
     html += "<ul>";
-    html += "<li>Speex: <a href='#speex'>Speex License</a></li>";
     html += "<li>bspatch: <a href='#bspatch'>bspatch License</a></li>";
-    html += "<li>bzip2: <a href='#bzip2'>bzip2 License</a></li>";
-    html += "<li>mspack: <a href='#lgpl2'>LGPL v2.1 License</a></li>";
-    html += "<li>quazip: <a href='#lgpl2'>LGPL v2.1 License</a></li>";
-    html += "<li>tomcrypt: <a href='#tomcrypt'>Tomcrypt License</a></li>";
     html += "<li>CuteLogger: <a href='#lgpl2'>LGPL v2.1 License</a></li>";
     html += "</ul>";
     about.browserLicenses->insertHtml(html);
 
     QMap<QString, QString> licenses;
-    licenses[":/docs/COPYING.SPEEX"] = "<a id='speex'>Speex License</a>";
     licenses[":/docs/lgpl-2.1.txt"] = "<a id='lgpl2'>LGPL v2.1</a>";
-    licenses[":/docs/LICENSE.TOMCRYPT"] = "<a id='tomcrypt'>Tomcrypt License</a>";
-    licenses[":/docs/LICENSE.BZIP2"] = "<a id='bzip2'>bzip2 License</a>";
     licenses[":/docs/LICENSE.BSPATCH"] = "<a id='bspatch'>bspatch License</a>";
 
     for (auto it = licenses.keyBegin(); it != licenses.keyEnd(); ++it) {
diff --git a/utils/rbutilqt/rbutilqt.qrc b/utils/rbutilqt/rbutilqt.qrc
--- a/utils/rbutilqt/rbutilqt.qrc
+++ b/utils/rbutilqt/rbutilqt.qrc
@@ -3,9 +3,6 @@
         <file alias="docs/CREDITS">../../docs/CREDITS</file>
         <file alias="docs/gpl-2.0.html">../../docs/gpl-2.0.html</file>
         <file alias="docs/lgpl-2.1.txt">logger/LICENSE.LGPL</file>
-        <file alias="docs/COPYING.SPEEX">../../lib/rbcodec/codecs/libspeex/COPYING</file>
-        <file alias="docs/LICENSE.TOMCRYPT">../../utils/tomcrypt/LICENSE</file>
-        <file alias="docs/LICENSE.BZIP2">../bzip2/LICENSE</file>
         <file alias="docs/LICENSE.BSPATCH">../bspatch/LICENSE</file>
         <file alias="docs/changelog.txt">changelog.txt</file>
     </qresource>
