diff -Naur qmapshack-V_1.15.0/src/qmapshack/helpers/CDraw.h qmapshack-V_1.15.0_X/src/qmapshack/helpers/CDraw.h
--- qmapshack-V_1.15.0/src/qmapshack/helpers/CDraw.h	2020-06-14 12:22:49.000000000 +0200
+++ qmapshack-V_1.15.0_X/src/qmapshack/helpers/CDraw.h	2020-06-14 20:55:16.675484111 +0200
@@ -21,6 +21,7 @@
 #define CPAINTER_H
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QPolygonF>
 #include <QRectF>
 
diff -Naur qmapshack-V_1.15.0/src/qmaptool/helpers/CDraw.h qmapshack-V_1.15.0_X/src/qmaptool/helpers/CDraw.h
--- qmapshack-V_1.15.0/src/qmaptool/helpers/CDraw.h	2020-06-14 12:22:49.000000000 +0200
+++ qmapshack-V_1.15.0_X/src/qmaptool/helpers/CDraw.h	2020-06-14 20:54:50.688816110 +0200
@@ -21,6 +21,7 @@
 #define CPAINTER_H
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QPolygonF>
 #include <QRectF>
 
