commit 76b4633f91b07d6319de5924d8ece4905f29646e
Author: Felipe Weckx <felipe@weckx.net>
Date:   Mon Jun 16 10:36:01 2014 -0300

    compatibility with autoconf 2.6.x

diff --git a/configure.ac b/configure.ac
index 7f5bb61..021b48f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -8,6 +8,7 @@ AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_SRCDIR([libtelnet.c])
 AM_INIT_AUTOMAKE([1.9 foreign -Wall -Werror subdir-objects])
+AM_PROG_AR
 #LT_INIT([win32-dll])
 AC_LIBTOOL_WIN32_DLL
 
