Index: /grass/trunk/lib/iostream/mm.cc
===================================================================
--- /grass/trunk/lib/iostream/mm.cc	(revision 48562)
+++ /grass/trunk/lib/iostream/mm.cc	(revision 50130)
@@ -275,5 +275,5 @@
  
 /* ************************************************************ */
-void* operator new[] (size_t sz) {
+void* operator new[] (size_t sz) throw(std::bad_alloc) {
   void *p;
   
@@ -326,5 +326,5 @@
  
 /* ************************************************************ */
-void* operator new (size_t sz) {
+void* operator new (size_t sz) throw(std::bad_alloc) {
   void *p;
   
@@ -378,5 +378,5 @@
 
 /* ---------------------------------------------------------------------- */
-void operator delete (void *ptr)  {
+void operator delete (void *ptr) throw() {
   size_t sz;
   void *p;
@@ -418,5 +418,5 @@
 
 /* ---------------------------------------------------------------------- */
-void operator delete[] (void *ptr) {
+void operator delete[] (void *ptr) throw() {
   size_t sz;
   void *p;
Index: /grass/trunk/include/iostream/mm.h
===================================================================
--- /grass/trunk/include/iostream/mm.h	(revision 32746)
+++ /grass/trunk/include/iostream/mm.h	(revision 50130)
@@ -129,8 +129,8 @@
 
   friend class mm_register_init;
-  friend void * operator new(size_t);
-  friend void * operator new[](size_t);
-  friend void operator delete(void *);
-  friend void operator delete[](void *);
+  friend void * operator new(size_t) throw(std::bad_alloc);
+  friend void * operator new[](size_t) throw(std::bad_alloc);
+  friend void operator delete(void *) throw();
+  friend void operator delete[](void *) throw();
 };
 
