diff -u -r grass-6.4.2.orig/configure grass-6.4.2.patched/configure
--- grass-6.4.2.orig/configure	2012-02-06 10:16:07.000000000 +0000
+++ grass-6.4.2.patched/configure	2013-02-25 20:13:08.277158035 +0000
@@ -10445,7 +10445,7 @@
 
 ac_save_cppflags="$CPPFLAGS"
 CPPFLAGS="$FFMPEGINCPATH $CPPFLAGS"
-for ac_hdr in avcodec.h
+for ac_hdr in libavcodec/avcodec.h
 do
 ac_safe=`echo "$ac_hdr" | sed 'y%./+-%__p_%'`
 echo $ac_n "checking for $ac_hdr""... $ac_c" 1>&6
@@ -10489,7 +10489,7 @@
 
 ac_save_cppflags="$CPPFLAGS"
 CPPFLAGS="$FFMPEGINCPATH $CPPFLAGS"
-for ac_hdr in avformat.h
+for ac_hdr in libavformat/avformat.h
 do
 ac_safe=`echo "$ac_hdr" | sed 'y%./+-%__p_%'`
 echo $ac_n "checking for $ac_hdr""... $ac_c" 1>&6
@@ -10533,7 +10533,7 @@
 
 ac_save_cppflags="$CPPFLAGS"
 CPPFLAGS="$FFMPEGINCPATH $CPPFLAGS"
-for ac_hdr in swscale.h
+for ac_hdr in libswscale/swscale.h
 do
 ac_safe=`echo "$ac_hdr" | sed 'y%./+-%__p_%'`
 echo $ac_n "checking for $ac_hdr""... $ac_c" 1>&6
@@ -10712,9 +10712,9 @@
 
 
 
-echo $ac_n "checking for avcodec_init in -lavcodec""... $ac_c" 1>&6
-echo "configure:10717: checking for avcodec_init in -lavcodec" >&5
-ac_lib_var=`echo avcodec'_'avcodec_init | sed 'y%./+-%__p_%'`
+echo $ac_n "checking for avcodec_open2 in -lavcodec""... $ac_c" 1>&6
+echo "configure:10717: checking for avcodec_open2 in -lavcodec" >&5
+ac_lib_var=`echo avcodec'_'avcodec_open2 | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
 LIBS="-lavcodec $FFMPEGLIB  $LIBS"
@@ -10724,10 +10724,10 @@
 /* Override any gcc2 internal prototype to avoid an error.  */
 /* We use char because int might match the return type of a gcc2
     builtin and then its argument prototype would still apply.  */
-char avcodec_init();
+char avcodec_open2();
 
 int main() {
-avcodec_init()
+avcodec_open2()
 ; return 0; }
 EOF
 if { (eval echo configure:10734: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
@@ -10748,9 +10748,9 @@
 else
   echo "$ac_t""no" 1>&6
 
-echo $ac_n "checking for avcodec_init in -lavcodec""... $ac_c" 1>&6
-echo "configure:10753: checking for avcodec_init in -lavcodec" >&5
-ac_lib_var=`echo avcodec'_'avcodec_init | sed 'y%./+-%__p_%'`
+echo $ac_n "checking for avcodec_open2 in -lavcodec""... $ac_c" 1>&6
+echo "configure:10753: checking for avcodec_open2 in -lavcodec" >&5
+ac_lib_var=`echo avcodec'_'avcodec_open2 | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
 LIBS="-lavcodec $FFMPEGLIB $MATHLIB $LIBS"
@@ -10760,10 +10760,10 @@
 /* Override any gcc2 internal prototype to avoid an error.  */
 /* We use char because int might match the return type of a gcc2
     builtin and then its argument prototype would still apply.  */
-char avcodec_init();
+char avcodec_open2();
 
 int main() {
-avcodec_init()
+avcodec_open2()
 ; return 0; }
 EOF
 if { (eval echo configure:10770: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
@@ -10810,9 +10810,9 @@
 
 
 
-echo $ac_n "checking for av_set_parameters in -lavformat""... $ac_c" 1>&6
-echo "configure:10815: checking for av_set_parameters in -lavformat" >&5
-ac_lib_var=`echo avformat'_'av_set_parameters | sed 'y%./+-%__p_%'`
+echo $ac_n "checking for av_write_frame in -lavformat""... $ac_c" 1>&6
+echo "configure:10815: checking for av_write_frame in -lavformat" >&5
+ac_lib_var=`echo avformat'_'av_write_frame | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
 LIBS="-lavformat $FFMPEGLIB  $LIBS"
@@ -10822,10 +10822,10 @@
 /* Override any gcc2 internal prototype to avoid an error.  */
 /* We use char because int might match the return type of a gcc2
     builtin and then its argument prototype would still apply.  */
-char av_set_parameters();
+char av_write_frame();
 
 int main() {
-av_set_parameters()
+av_write_frame()
 ; return 0; }
 EOF
 if { (eval echo configure:10832: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
@@ -10846,9 +10846,9 @@
 else
   echo "$ac_t""no" 1>&6
 
-echo $ac_n "checking for av_set_parameters in -lavformat""... $ac_c" 1>&6
-echo "configure:10851: checking for av_set_parameters in -lavformat" >&5
-ac_lib_var=`echo avformat'_'av_set_parameters | sed 'y%./+-%__p_%'`
+echo $ac_n "checking for av_write_frame in -lavformat""... $ac_c" 1>&6
+echo "configure:10851: checking for av_write_frame in -lavformat" >&5
+ac_lib_var=`echo avformat'_'av_write_frame | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
 LIBS="-lavformat $FFMPEGLIB $MATHLIB $LIBS"
@@ -10858,10 +10858,10 @@
 /* Override any gcc2 internal prototype to avoid an error.  */
 /* We use char because int might match the return type of a gcc2
     builtin and then its argument prototype would still apply.  */
-char av_set_parameters();
+char av_write_frame();
 
 int main() {
-av_set_parameters()
+av_write_frame()
 ; return 0; }
 EOF
 if { (eval echo configure:10868: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
diff -u -r grass-6.4.2.orig/configure.in grass-6.4.2.patched/configure.in
--- grass-6.4.2.orig/configure.in	2012-02-06 10:16:07.000000000 +0000
+++ grass-6.4.2.patched/configure.in	2013-02-25 20:45:48.025962482 +0000
@@ -1110,9 +1110,9 @@
 
 LOC_CHECK_INC_PATH(ffmpeg, FFMPEG, FFMPEGINCPATH)
 
-LOC_CHECK_INCLUDES(avcodec.h,FFMPEG,$FFMPEGINCPATH)
-LOC_CHECK_INCLUDES(avformat.h,FFMPEG,$FFMPEGINCPATH)
-LOC_CHECK_INCLUDES(swscale.h,FFMPEG,$FFMPEGINCPATH)
+LOC_CHECK_INCLUDES(libavcodec/avcodec.h,FFMPEG,$FFMPEGINCPATH)
+LOC_CHECK_INCLUDES(libavformat/avformat.h,FFMPEG,$FFMPEGINCPATH)
+LOC_CHECK_INCLUDES(libswscale/swscale.h,FFMPEG,$FFMPEGINCPATH)
 
 AC_DEFINE(HAVE_FFMPEG)
 
@@ -1125,8 +1125,8 @@
 LOC_CHECK_LIB_PATH(ffmpeg, FFMPEG, FFMPEGLIBPATH)
 
 LOC_CHECK_LIBS(avutil, av_free, FFMPEG, $FFMPEGLIBPATH, FFMPEGLIB,,,,$MATHLIB)
-LOC_CHECK_LIBS(avcodec, avcodec_init, FFMPEG, $FFMPEGLIBPATH, FFMPEGLIB, $FFMPEGLIB,,,$MATHLIB)
-LOC_CHECK_LIBS(avformat, av_set_parameters, FFMPEG, $FFMPEGLIBPATH, FFMPEGLIB, $FFMPEGLIB,,,$MATHLIB)
+LOC_CHECK_LIBS(avcodec, avcodec_open2, FFMPEG, $FFMPEGLIBPATH, FFMPEGLIB, $FFMPEGLIB,,,$MATHLIB)
+LOC_CHECK_LIBS(avformat, av_write_frame, FFMPEG, $FFMPEGLIBPATH, FFMPEGLIB, $FFMPEGLIB,,,$MATHLIB)
 
 fi # $USE_FFMPEG
 
