diff -uNr dahdi-linux-2.8.0.ORIG//Makefile dahdi-linux-2.8.0/Makefile
--- dahdi-linux-2.8.0.ORIG//Makefile	2010-09-09 14:41:11.000000000 +0100
+++ dahdi-linux-2.8.0/Makefile	2010-09-09 14:41:27.000000000 +0100
@@ -139,7 +139,7 @@
 	build_tools/uninstall-modules dahdi $(KVERS)
 endif
 	$(KMAKE) INSTALL_MOD_PATH=$(DESTDIR) INSTALL_MOD_DIR=dahdi modules_install
-	[ `id -u` = 0 ] && /sbin/depmod -a $(KVERS) || :
+#	[ `id -u` = 0 ] && /sbin/depmod -a $(KVERS) || :
 
 uninstall-modules:
 ifdef DESTDIR
