diff -uNr dahdi-linux-2.8.0.ORIG//Makefile dahdi-linux-2.8.0/Makefile
--- dahdi-linux-2.8.0/Makefile.ORIG	2013-08-07 13:48:28.909267134 +0100
+++ dahdi-linux-2.8.0/Makefile	2013-08-07 13:50:10.346268656 +0100
@@ -71,7 +71,7 @@
 	@echo "You do not appear to have the sources for the $(KVERS) kernel installed."
 	@exit 1
 endif
-	$(KMAKE) modules DAHDI_BUILD_ALL=$(DAHDI_BUILD_ALL)
+	+$(KMAKE) modules DAHDI_BUILD_ALL=$(DAHDI_BUILD_ALL)
 
 include/dahdi/version.h: FORCE
 	@DAHDIVERSION="${DAHDIVERSION}" build_tools/make_version_h > $@.tmp
@@ -138,7 +138,7 @@
 	fi
 	build_tools/uninstall-modules dahdi $(KVERS)
 endif
-	$(KMAKE) INSTALL_MOD_PATH=$(DESTDIR) INSTALL_MOD_DIR=dahdi modules_install
+	+$(KMAKE) INSTALL_MOD_PATH=$(DESTDIR) INSTALL_MOD_DIR=dahdi modules_install
 #	[ `id -u` = 0 ] && /sbin/depmod -a $(KVERS) || :
 
 uninstall-modules:
@@ -174,11 +174,11 @@
 
 clean:
 ifneq (no,$(HAS_KSRC))
-	$(KMAKE) clean
+	+$(KMAKE) clean
 endif
 	@rm -f $(GENERATED_DOCS)
	$(MAKE) -C drivers/dahdi/firmware clean
	$(MAKE) -C $(KSRC) M='$(PWD)/drivers/dahdi/oct612x' clean
 
 distclean: dist-clean
 
