--- dahdi-linux-3.1.0/drivers/dahdi/xpp/xpp_usb.c	2020-03-28 21:34:08.471611500 +0200
+++ dahdi-linux-3.1.0.o/drivers/dahdi/xpp/xpp_usb.c	2020-03-28 21:38:47.169161487 +0200
@@ -882,7 +882,7 @@
 		usec = 0; /* System clock jumped */
 	if (usec > xusb->max_tx_delay)
 		xusb->max_tx_delay = usec;
-	i = usec / USEC_BUCKET;
+	i = div_s64(usec, USEC_BUCKET);
 	if (i >= NUM_BUCKETS)
 		i = NUM_BUCKETS - 1;
 	xusb->usb_tx_delay[i]++;
