diff -Nur apcupsd-3.14.13.orig/platforms/slackware/apcupsd.in apcupsd-3.14.13/platforms/slackware/apcupsd.in
--- apcupsd-3.14.13.orig/platforms/slackware/apcupsd.in	2015-01-29 08:13:25.000000000 -0600
+++ apcupsd-3.14.13/platforms/slackware/apcupsd.in	2015-08-17 22:49:30.754638339 -0500
@@ -20,6 +20,7 @@
 	if [ -f ${APCPID} ]; then
 		return="  Already running."
 	else
+		mkdir -p @LOCKDIR@
 		@sbindir@/apcupsd && touch @LOCKDIR@/apcupsd \
 		|| return="  Failed."
 	fi
