diff -Nur apcupsd-3.14.12.orig/platforms/slackware/apcupsd.in apcupsd-3.14.12/platforms/slackware/apcupsd.in
--- apcupsd-3.14.12.orig/platforms/slackware/apcupsd.in	2005-03-03 11:18:09.000000000 -0600
+++ apcupsd-3.14.12/platforms/slackware/apcupsd.in	2014-04-01 13:53:23.113153628 -0500
@@ -20,6 +20,7 @@
 	if [ -f ${APCPID} ]; then
 		return="  Already running."
 	else
+		mkdir -p @LOCKDIR@/subsys
 		@sbindir@/apcupsd && touch @LOCKDIR@/subsys/apcupsd \
 		|| return="  Failed."
 	fi
