diff -Nur apcupsd-3.14.10.orig/platforms/slackware/apcupsd.in apcupsd-3.14.10/platforms/slackware/apcupsd.in
--- apcupsd-3.14.10.orig/platforms/slackware/apcupsd.in	2005-03-03 11:18:09.000000000 -0600
+++ apcupsd-3.14.10/platforms/slackware/apcupsd.in	2012-09-07 21:31:27.347857464 -0500
@@ -20,6 +20,7 @@
 	if [ -f ${APCPID} ]; then
 		return="  Already running."
 	else
+		mkdir -p @LOCKDIR@/subsys
 		@sbindir@/apcupsd && touch @LOCKDIR@/subsys/apcupsd \
 		|| return="  Failed."
 	fi
