diff -Nur apcupsd-3.14.10.orig/platforms/etc/apcupsd.conf.in apcupsd-3.14.10/platforms/etc/apcupsd.conf.in
--- apcupsd-3.14.10.orig/platforms/etc/apcupsd.conf.in	2011-05-19 17:55:25.000000000 -0500
+++ apcupsd-3.14.10/platforms/etc/apcupsd.conf.in	2013-05-15 12:44:25.497885953 -0500
@@ -90,7 +90,7 @@
 
 # LOCKFILE <path to lockfile>
 #   Path for device lock file. Not used on Win32.
-LOCKFILE @LOCKDIR@
+LOCKFILE @LOCKDIR@/subsys
 
 # SCRIPTDIR <path to script directory>
 #   Directory in which apccontrol and event scripts are located.
