diff -Nur apcupsd-3.14.12.orig/platforms/etc/apcupsd.conf.in apcupsd-3.14.12/platforms/etc/apcupsd.conf.in
--- apcupsd-3.14.12.orig/platforms/etc/apcupsd.conf.in	2013-08-05 09:05:17.000000000 -0500
+++ apcupsd-3.14.12/platforms/etc/apcupsd.conf.in	2014-04-01 13:54:06.786690735 -0500
@@ -93,7 +93,7 @@
 
 # LOCKFILE <path to lockfile>
 #   Path for device lock file. Not used on Win32.
-LOCKFILE @LOCKDIR@
+LOCKFILE @LOCKDIR@/subsys
 
 # SCRIPTDIR <path to script directory>
 #   Directory in which apccontrol and event scripts are located.
