diff -Nur apcupsd-3.14.13.orig/platforms/slackware/apcupsd.in apcupsd-3.14.13/platforms/slackware/apcupsd.in
--- apcupsd-3.14.13.orig/platforms/slackware/apcupsd.in	2015-08-17 22:51:39.595562965 -0500
+++ apcupsd-3.14.13/platforms/slackware/apcupsd.in	2015-08-17 22:51:34.106608743 -0500
@@ -9,14 +9,14 @@
 APCPID=@PIDDIR@/apcupsd.pid
 DISTVER="@DISTVER@"
 
-return="  Done."
+return=""
 
 
 case "$1" in
     start)
 	rm -f @PWRFAILDIR@/powerfail
 	rm -f @nologdir@/nologin
-	echo -n "Starting apcupsd power management.."
+	echo -n "Starting apcupsd power management:  /sbin/apcupsd"
 	if [ -f ${APCPID} ]; then
 		return="  Already running."
 	else
@@ -30,7 +30,7 @@
     stop)
 	echo -n "Stopping apcupsd power management.."
 	if [ -f ${APCPID} ]; then
-		THEPID=`cat ${APCPID}`
+		THEPID=$(cat ${APCPID})
 		kill ${THEPID} || return="  Failed."
 		rm -f ${APCPID}
 	else
