--- a/src/lib/quazip/unzip.c    2019-05-08 12:46:46.000000000 -0500
+++ b/src/lib/quazip/unzip.c    2025-04-05 17:49:37.273641245 -0500
@@ -1174,7 +1174,7 @@
     if (password != NULL)
     {
         int i;
-        s->pcrc_32_tab = get_crc_table();
+        s->pcrc_32_tab = (const long unsigned int *)get_crc_table();
         init_keys(password,s->keys,s->pcrc_32_tab);
         if (ZSEEK(s->z_filefunc, s->filestream,
                   s->pfile_in_zip_read->pos_in_zipfile +
--- a/src/lib/quazip/zip.c      2019-05-08 12:46:46.000000000 -0500
+++ b/src/lib/quazip/zip.c      2025-04-05 19:02:34.939833765 -0500
@@ -889,7 +889,7 @@
         unsigned char bufHead[RAND_HEAD_LEN];
         unsigned int sizeHead;
         zi->ci.encrypt = 1;
-        zi->ci.pcrc_32_tab = get_crc_table();
+        zi->ci.pcrc_32_tab = (const long unsigned int *)get_crc_table();
         /*init_keys(password,zi->ci.keys,zi->ci.pcrc_32_tab);*/
 
         sizeHead=crypthead(password,bufHead,RAND_HEAD_LEN,zi->ci.keys,zi->ci.pcrc_32_tab,crcForCrypting);
