From 472951243d70eeb40a53b1f7e16e6eab0588d079 Mon Sep 17 00:00:00 2001
From: PsikoBlock <rramsch@googlemail.com>
Date: Fri, 10 Apr 2020 21:41:52 +0200
Subject: [PATCH] Fix libgit2 version check to allow building with versions >=
 1.0

---
 src/version/partschecker.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/version/partschecker.cpp b/src/version/partschecker.cpp
index 65daf76e5..36300fe09 100644
--- a/src/version/partschecker.cpp
+++ b/src/version/partschecker.cpp
@@ -121,7 +121,7 @@
     /**
      * Connect to the remote.
      */
-#if LIBGIT2_VER_MINOR > 24
+#if LIBGIT2_VER_MAJOR > 0 || (LIBGIT2_VER_MAJOR == 0 && LIBGIT2_VER_MINOR > 24)
     error = git_remote_connect(remote, GIT_DIRECTION_FETCH, &callbacks, NULL, NULL);
 #elif LIBGIT2_VER_MINOR == 24
     error = git_remote_connect(remote, GIT_DIRECTION_FETCH, &callbacks, NULL);
