TODO: Fix those regexes

diff -Naur txt2tags-3.9.orig/docs/rules/rules.conf txt2tags-3.9/docs/rules/rules.conf
--- txt2tags-3.9.orig/docs/rules/rules.conf	2023-10-06 20:07:32.000000000 +0200
+++ txt2tags-3.9/docs/rules/rules.conf	2026-03-17 09:25:04.956421542 +0100
@@ -11,9 +11,7 @@
 %%%%% SPECIAL PROCESSING RULES
 %
 % Title section: remove title tests from TOC
-%!postproc: '^(?i) *<LI>.*[Tt]itle Level .*'  ''
 % Link section: one link per line
-%!postproc: '^(?i)((any text: )?<A HREF=.*)'   '\1<br>'
 % Inline section: special format for tests results
 %!preproc: '^(i\) .{82})'   '\1}{'
 %!preproc: '^(i\) .{69})'   '\1}{'
