--- gssdp-1.3.1/meson.build.orig	2021-08-14 23:12:19.168431148 +0700
+++ gssdp-1.3.1/meson.build	2021-08-14 23:12:46.260719024 +0700
@@ -88,12 +88,6 @@
   subdir('vala')
 endif
 
-
-gidocgen_dep = dependency('gi-docgen', version: '>= 2021.1',
-                          fallback: ['gi-docgen', 'dummy_dep'],
-                          required: get_option('gtk_doc') and get_option('introspection')
-                      )
-
 if get_option('gtk_doc')
   subdir('doc')
 endif
