--- common.mak.orig	2019-11-22 04:27:08.903418811 +0100
+++ common.mak	2019-11-22 04:29:32.031629463 +0100
@@ -2,6 +2,7 @@
 
 PREFIX ?= /usr/local
 OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only
+LIBDIR ?= lib
 ENABLE_CONVOLUTION ?= no
 INSTALL_EXTRA_LV2 ?= no
 FONTFILE?=/usr/share/fonts/truetype/ttf-bitstream-vera/VeraBd.ttf
@@ -20,7 +21,7 @@
 
 bindir = $(PREFIX)/bin
 sharedir = $(PREFIX)/share/setBfree
-lv2dir = $(PREFIX)/lib/lv2
+lv2dir = $(PREFIX)/$(LIBDIR)/lv2
 
 CFLAGS ?= $(OPTIMIZATIONS) -Wall
 ifeq ($(XWIN),)
