From: Debian Security Tools <team+pkg-security@tracker.debian.org>
Date: Mon, 22 Apr 2024 08:15:14 +0700
Subject: makefile_osdep_parallel_build_fix

Origin: upstream, https://github.com/aircrack-ng/mdk4/commit/a214f5cedc0e27060d10a359a1efeaed7d4ab2a3
---
 src/Makefile | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index 6ee36bc..1540755 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -27,8 +27,7 @@ att:
 osd:
 	$(MAKE) -C $(OSD)
 
-$(LIBOSD) $(OSD)/libosdep.a:
-	$(MAKE) -C $(OSD)
+$(LIBOSD) $(OSD)/libosdep.a: osd
 
 $(OBJ_ATT):
 	$(MAKE) -C $(ATTACKS)
