diff -pruN qtiplot-0.9.8.9.orig/qtiplot/qtiplot.pro qtiplot-0.9.8.9/qtiplot/qtiplot.pro
--- qtiplot-0.9.8.9.orig/qtiplot/qtiplot.pro	2011-11-02 18:55:25.000000000 +0200
+++ qtiplot-0.9.8.9/qtiplot/qtiplot.pro	2012-08-03 01:27:29.649544418 +0300
@@ -232,3 +232,14 @@ contains(CONFIG, BrowserPlugin){
 	RC_FILE	= qtiplot.rc
 	include(../3rdparty/QtSolutions/qtbrowserplugin/src/qtbrowserplugin.pri)
 }
+
+###############################################################
+##################### OriginPlugin ############################
+###############################################################
+
+TARGET		=	qtiplot
+DESTDIR		=	
+LIBS		+=	../tmp/qtiplot/libFreeSoftwareQtiPlotImportOPJ.a
+				
+
+include(src/plugins/OriginPlugin/OriginPlugin.pri)
