diff -Naur MuseScore-4.6.2.orig/src/framework/dockwindow/thirdparty/KDDockWidgets/src/CMakeLists.txt MuseScore-4.6.2/src/framework/dockwindow/thirdparty/KDDockWidgets/src/CMakeLists.txt
--- MuseScore-4.6.2.orig/src/framework/dockwindow/thirdparty/KDDockWidgets/src/CMakeLists.txt	2025-10-10 11:21:06.000000000 +0200
+++ MuseScore-4.6.2/src/framework/dockwindow/thirdparty/KDDockWidgets/src/CMakeLists.txt	2025-10-18 11:38:44.592843000 +0200
@@ -295,7 +295,7 @@
 endif()
 
 if(${PROJECT_NAME}_QTQUICK)
-  target_link_libraries(kddockwidgets PUBLIC Qt${Qt_VERSION_MAJOR}::Widgets Qt${Qt_VERSION_MAJOR}::Quick Qt${Qt_VERSION_MAJOR}::QuickControls2 PRIVATE Qt${Qt_VERSION_MAJOR}::GuiPrivate)
+  target_link_libraries(kddockwidgets PUBLIC Qt${Qt_VERSION_MAJOR}::Widgets Qt${Qt_VERSION_MAJOR}::Quick Qt${Qt_VERSION_MAJOR}::QuickControls2)
 else()
   target_link_libraries(kddockwidgets PUBLIC Qt${Qt_VERSION_MAJOR}::Widgets PRIVATE Qt${Qt_VERSION_MAJOR}::WidgetsPrivate)
 endif()
