diff -Naur listener-2.2/paudio.c listener-2.2.patched/paudio.c
--- listener-2.2/paudio.c	2013-01-21 04:23:54.000000000 -0500
+++ listener-2.2.patched/paudio.c	2021-12-24 15:07:16.791531492 -0500
@@ -15,7 +15,7 @@
 	err = Pa_Initialize();
 	error_check(err, "Error initializing audio");
 
-	pcm_param.device = 0; /* default device should be the first */
+	pcm_param.device = Pa_GetDefaultInputDevice();
 	pcm_param.channelCount = n_channels;
 	pcm_param.sampleFormat = paInt16;
 	pcm_param.suggestedLatency = 0.5; /* 0.5 seconds, in order to eliminate glitches */
