diff -Nur units-2.00.orig/Makefile.in units-2.00/Makefile.in
--- units-2.00.orig/Makefile.in	2012-05-02 15:37:55.000000000 -0500
+++ units-2.00/Makefile.in	2012-07-27 23:54:02.551369930 -0500
@@ -84,10 +84,10 @@
 	sed -e "s:outfile = 'currency.units':outfile='@UDAT@currency.units':" units_cur > units_cur_inst
 
 install-support: definitions.units currency.units units_cur_inst
-	$(srcdir)/mkinstalldirs @UDAT@  $(DESTDIR)$(bindir) 
-	$(INSTALL_DATA) $(srcdir)/definitions.units @UDAT@definitions.units
-	$(INSTALL_DATA) $(srcdir)/currency.units @UDAT@currency.units
-	$(INSTALL_DATA) $(srcdir)/locale.map @UDAT@locale.map
+	$(srcdir)/mkinstalldirs $(DESTDIR)@UDAT@  $(DESTDIR)$(bindir) 
+	$(INSTALL_DATA) $(srcdir)/definitions.units $(DESTDIR)@UDAT@definitions.units
+	$(INSTALL_DATA) $(srcdir)/currency.units $(DESTDIR)@UDAT@currency.units
+	$(INSTALL_DATA) $(srcdir)/locale.map $(DESTDIR)@UDAT@locale.map
 	$(INSTALL_PROGRAM) units_cur_inst $(DESTDIR)$(bindir)/`echo units_cur|sed '$(transform)'`
 
 install: units@EXEEXT@ install-doc install-support
@@ -138,8 +138,8 @@
 uninstall: 
 	-rm -f $(DESTDIR)$(bindir)/`echo units@EXEEXT@|sed '$(transform)'` \
 	       $(DESTDIR)$(mandir)/man1/`echo units|sed '$(transform)'`.1 \
-	       $(DESTDIR)$(infodir)/units.info @UDAT@definitions.units \
-	       @UDAT@currency.units \
+	       $(DESTDIR)$(infodir)/units.info $(DESTDIR)@UDAT@definitions.units \
+	       $(DESTDIR)@UDAT@currency.units \
 	@if test -f $(DESTDIR)$(infodir)/dir; then \
 	  echo You may need to remove units.info from $(DESTDIR)$(infodir)/dir; \
 	else true; fi
