diff -Nur device-tree-compiler-1.6.0.orig/pylibfdt/Makefile.pylibfdt device-tree-compiler-1.6.0/pylibfdt/Makefile.pylibfdt
--- device-tree-compiler-1.6.0.orig/pylibfdt/Makefile.pylibfdt	2020-03-04 00:14:58.000000000 -0600
+++ device-tree-compiler-1.6.0/pylibfdt/Makefile.pylibfdt	2021-11-06 12:38:30.133022247 -0500
@@ -22,7 +22,7 @@
 
 install_pylibfdt: $(PYMODULE)
 	@$(VECHO) INSTALL-PYLIB
-	$(PYTHON) $(SETUP) $(SETUPFLAGS) install --prefix=$(PREFIX)
+	$(PYTHON) $(SETUP) $(SETUPFLAGS) install --prefix=$(PREFIX) --root=$(DESTDIR)
 
 pylibfdt_clean:
 	@$(VECHO) CLEAN "(pylibfdt)"
