diff -urN xvnkb.cvs/Makefile.common xvnkb/Makefile.common
--- xvnkb.cvs/Makefile.common	2005-01-12 04:21:04.000000000 -0600
+++ xvnkb/Makefile.common	2009-07-09 10:37:22.491007609 -0500
@@ -6,7 +6,7 @@
 .PHONY: config debug core core_debug tools uninstall
 
 CORE=xvnkb.so.$(VERSION)
-CORE_OBJS=xvnkb.o visckey.o property.o
+CORE_OBJS=xvnkb.o visckey.o
 
 GUI=xvnkb
 GUI_OBJS=data.o flash.o main.o event.o mainwin.o menu.o hotkey.o systray.o \
diff -urN xvnkb.cvs/xvnkb.c xvnkb/xvnkb.c
--- xvnkb.cvs/xvnkb.c	2009-06-27 06:32:54.000000000 -0500
+++ xvnkb/xvnkb.c	2009-07-09 10:37:02.505973519 -0500
@@ -49,7 +49,7 @@
 	long sym;
 } vk_hotkey_info;
 /*----------------------------------------------------------------------------*/
-Window root = -1;
+static Window root = -1;
 Window focus = -1;
 /*----------------------------------------------------------------------------*/
 static KeyCode rk = 0;
@@ -422,3 +422,4 @@
 	return 0;
 }
 /*----------------------------------------------------------------------------*/
+#include "property.c"
