diff -urN xvnkb.cvs/Makefile.common xvnkb/Makefile.common
--- xvnkb.cvs/Makefile.common	2005-01-12 04:21:04.000000000 -0600
+++ xvnkb/Makefile.common	2009-07-09 12:44:28.789974861 -0500
@@ -19,7 +19,7 @@
 	$(CC) -fpic $(CFLAGS) $(FPTR) $(VK_OPT) -DVERSION=\"$(VERSION)\" -Wall -I$(XLIB_INC) -c $<
 
 $(CORE): $(CORE_OBJS)
-	$(CC) $(CFLAGS) $(DLFLAGS) $(CORE_OBJS) -o $@ -L$(XLIB_PATH) $(LIBDL)
+	$(CC) $(CFLAGS) $(DLFLAGS) $(CORE_OBJS) -o $@ -L$(XLIB_PATH) $(LIBDL) -lX11
 
 $(GUI): $(GUI_OBJS)
 	$(CC) $(CFLAGS) $(GUI_OBJS) -o $@ -L$(XLIB_PATH) $(LIBS) $(XFT_LIBS)
