diff -Naur AFLplusplus-3.14c/qemu_mode/build_qemu_support.sh AFLplusplus-3.14c.patched/qemu_mode/build_qemu_support.sh
--- AFLplusplus-3.14c/qemu_mode/build_qemu_support.sh	2021-07-19 04:50:09.000000000 -0400
+++ AFLplusplus-3.14c.patched/qemu_mode/build_qemu_support.sh	2021-08-08 13:20:14.656823570 -0400
@@ -68,36 +68,8 @@
 
 echo "[+] All checks passed!"
 
-echo "[*] Making sure qemuafl is checked out"
-
-git status 1>/dev/null 2>/dev/null
-if [ $? -eq 0 ]; then
-  echo "[*] initializing qemuafl submodule"
-  git submodule init || exit 1
-  git submodule update ./qemuafl 2>/dev/null # ignore errors
-else
-  echo "[*] cloning qemuafl"
-  test -d qemuafl || {
-    CNT=1
-    while [ '!' -d qemuafl -a "$CNT" -lt 4 ]; do
-      echo "Trying to clone qemuafl (attempt $CNT/3)"
-      git clone --depth 1 https://github.com/AFLplusplus/qemuafl
-      CNT=`expr "$CNT" + 1`
-    done
-  }
-fi
-
-test -d qemuafl || { echo "[-] Not checked out, please install git or check your internet connection." ; exit 1 ; }
-echo "[+] Got qemuafl."
-
 cd "qemuafl" || exit 1
-if [ -n "$NO_CHECKOUT" ]; then
-  echo "[*] Skipping checkout to $QEMUAFL_VERSION"
-else
-  echo "[*] Checking out $QEMUAFL_VERSION"
-  sh -c 'git stash' 1>/dev/null 2>/dev/null
-  git checkout "$QEMUAFL_VERSION" || echo Warning: could not check out to commit $QEMUAFL_VERSION
-fi
+sed -i '/hexagon/d' target/meson.build # broken!
 
 echo "[*] Making sure imported headers matches"
 cp "../../include/config.h" "./qemuafl/imported/" || exit 1
