From: Valentin Vidic <Valentin.Vidic@CARNet.hr>
Date: Wed, 25 Apr 2018 23:13:29 +0200
Subject: Check sscanf return value

---
 dlm_tool/main.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/dlm_tool/main.c b/dlm_tool/main.c
index 042caa9..1900236 100644
--- a/dlm_tool/main.c
+++ b/dlm_tool/main.c
@@ -692,6 +692,11 @@ static void print_lkb(char *line, struct rinfo *ri)
 		    (unsigned long long *)&lkb.timestamp,
 		    (unsigned long long *)&lkb.time_bast);
 
+	if (rv != 17) {
+		fprintf(stderr, "print_lkb error rv %d line \"%s\"\n", rv, line);
+		return;
+	}
+
 	ri->lkb_count++;
 
 	if (lkb.status == DLM_LKSTS_GRANTED) {
