From: =?utf-8?q?Ferenc_W=C3=A1gner?= <wferi@niif.hu>
Date: Sun, 30 Aug 2015 16:48:15 +0200
Subject: Debian path for the config file

---
 init/dlm.service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/init/dlm.service b/init/dlm.service
index 1db5925..5cd01c5 100644
--- a/init/dlm.service
+++ b/init/dlm.service
@@ -7,7 +7,7 @@ After=corosync.service sys-kernel-config.mount
 OOMScoreAdjust=-1000
 Type=notify
 NotifyAccess=main
-EnvironmentFile=/etc/sysconfig/dlm
+EnvironmentFile=-/etc/default/dlm
 ExecStartPre=/sbin/modprobe dlm 
 ExecStart=/usr/sbin/dlm_controld --foreground $DLM_CONTROLD_OPTS
 #ExecStopPost=/sbin/modprobe -r dlm
