From: =?utf-8?q?Ferenc_W=C3=A1gner?= <wferi@debian.org>
Date: Mon, 19 Dec 2016 22:57:22 +0100
Subject: Don't link dlm_stonith against libxml2

---
 fence/Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/fence/Makefile b/fence/Makefile
index d4a3c09..1382260 100644
--- a/fence/Makefile
+++ b/fence/Makefile
@@ -22,7 +22,6 @@ CFLAGS += `xml2-config --cflags`
 CFLAGS += -I../include
 
 LDFLAGS += -Wl,-z,relro -pie
-LDFLAGS += `xml2-config --libs`
 LDFLAGS += -ldl
 
 all: $(BIN_TARGET)
