From: =?utf-8?q?Ferenc_W=C3=A1gner?= <wferi@niif.hu>
Date: Wed, 27 Apr 2016 12:00:43 +0200
Subject: Enable systemctl help dlm to show relevant man pages

---
 init/dlm.service | 1 +
 1 file changed, 1 insertion(+)

diff --git a/init/dlm.service b/init/dlm.service
index 5cd01c5..5854738 100644
--- a/init/dlm.service
+++ b/init/dlm.service
@@ -2,6 +2,7 @@
 Description=dlm control daemon
 Requires=corosync.service sys-kernel-config.mount
 After=corosync.service sys-kernel-config.mount
+Documentation=man:dlm_controld man:dlm.conf man:dlm_stonith
 
 [Service]
 OOMScoreAdjust=-1000
