diff -Naur defendguin-0.0.11/Makefile defendguin-0.0.11.patched/Makefile
--- defendguin-0.0.11/Makefile	2006-01-29 04:23:31.000000000 -0500
+++ defendguin-0.0.11.patched/Makefile	2009-04-16 03:18:04.000000000 -0400
@@ -9,11 +9,12 @@
 
 # User-definable stuff:
 
-CFLAGS=-Wall -O2
+DESTDIR=
+COPT=
 #DATA_PREFIX=$(PWD)/data/
-PREFIX=/usr/local
+PREFIX=/usr
 MAN_PREFIX=$(PREFIX)
-BIN_PREFIX=$(PREFIX)/bin
+BIN_PREFIX=$(PREFIX)/games
 DATA_PREFIX=$(PREFIX)/share/defendguin/
 JOY=YES
 
@@ -24,7 +25,7 @@
 SDL_LDFLAGS := $(shell sdl-config --libs) -L/usr/X11R6/lib
 MIXER=-lSDL_mixer
 NOSOUNDFLAG=__SOUND
-CFLAGS=-Wall $(SDL_CFLAGS) -DDATA_PREFIX=\"$(DATA_PREFIX)\" -D$(NOSOUNDFLAG) \
+CFLAGS=-Wall $(COPT) $(SDL_CFLAGS) -DDATA_PREFIX=\"$(DATA_PREFIX)\" -D$(NOSOUNDFLAG) \
 	-DJOY_$(JOY)
 SDL_LIB=$(SDL_LDFLAGS) $(MIXER)
 
@@ -37,13 +38,17 @@
 	make defendguin MIXER= NOSOUNDFLAG=NOSOUND
 
 install:	defendguin
-	install -d $(DATA_PREFIX)
-	cp -R data/* $(DATA_PREFIX)
-	chmod -R a+rX,g-w,o-w $(DATA_PREFIX)
-	cp defendguin $(BIN_PREFIX)/
-	chmod a+rx,g-w,o-w $(BIN_PREFIX)/defendguin
-	cp src/defendguin.6 $(MAN_PREFIX)/man/man6/
-	chmod a+rx,g-w,o-w $(MAN_PREFIX)/man/man6/defendguin.6
+	install -d $(DESTDIR)$(DATA_PREFIX)
+	install -d $(DESTDIR)$(MAN_PREFIX)/man/man6/
+	install -d $(DESTDIR)$(BIN_PREFIX)
+	cp -R data/* $(DESTDIR)$(DATA_PREFIX)
+	chmod -R a+rX,g-w,o-w $(DESTDIR)$(DATA_PREFIX)
+	strip defendguin
+	cp defendguin $(DESTDIR)$(BIN_PREFIX)/
+	chmod a+rx,g-w,o-w $(DESTDIR)$(BIN_PREFIX)/defendguin
+	cp src/defendguin.6 $(DESTDIR)$(MAN_PREFIX)/man/man6/
+	chmod a+rx,g-w,o-w $(DESTDIR)$(MAN_PREFIX)/man/man6/defendguin.6
+	gzip -9 $(DESTDIR)$(MAN_PREFIX)/man/man6/defendguin.6
 
 uninstall:
 	-rm -r $(DATA_PREFIX)
