--- LDView.org/TRE/TREGLExtensions.cpp	2010-01-08 22:01:46.000000000 +0100
+++ LDView/TRE/TREGLExtensions.cpp	2015-02-04 11:49:42.774280878 +0100
@@ -204,7 +204,7 @@
 	// the function pointers won't be loaded.
 	if (haveVARExtension(true))
 	{
-#ifndef __APPLE__
+#ifdef WIN32
 		// NOTE: I should really change the __APPLE__ path to do their
 		// equivalent of wglGetProcAddress.  However, since this extension won't
 		// work anyway without a bunch of glX stuff that I'm not going to do,
