--- LDView-4.1/TRE/TREGLExtensions.cpp	2008-10-25 23:33:34.000000000 -0200
+++ LDView-4.1_patched/TRE/TREGLExtensions.cpp	2013-11-20 22:34:37.069129226 -0200
@@ -193,7 +193,7 @@
 	// the function pointers won't be loaded.
 	if (haveVARExtension(true))
 	{
-#ifndef __APPLE__
+#ifdef WIN32
 		// NOTE: I should really change the __APPLE__ path to do their
 		// equivalent of wglGetProcAddress.  However, since this extension won't
 		// work anyway without a bunch of glX stuff that I'm not going to do,
