diff -ru LDView.org/LDLib/LDLibraryUpdater.cpp LDView/LDLib/LDLibraryUpdater.cpp
--- LDView.org/LDLib/LDLibraryUpdater.cpp	2010-08-04 08:19:09.000000000 +0200
+++ LDView/LDLib/LDLibraryUpdater.cpp	2015-02-04 11:45:23.930291764 +0100
@@ -1205,7 +1205,7 @@
 				// it to signal.
 				boost::xtime xt;
 
-				boost::xtime_get(&xt, boost::TIME_UTC);
+				boost::xtime_get(&xt, boost::TIME_UTC_);
 				xt.nsec += 250 * 1000 * 1000;
 				m_threadFinish->timed_wait(lock, xt);
 			}
diff -ru LDView.org/TCFoundation/TCWebClient.cpp LDView/TCFoundation/TCWebClient.cpp
--- LDView.org/TCFoundation/TCWebClient.cpp	2011-02-12 05:24:14.000000000 +0100
+++ LDView/TCFoundation/TCWebClient.cpp	2015-02-04 11:45:39.781291097 +0100
@@ -118,7 +118,7 @@
 #else // _OSMESA
 	boost::xtime xt;
 
-	boost::xtime_get(&xt, boost::TIME_UTC);
+	boost::xtime_get(&xt, boost::TIME_UTC_);
 	xt.sec += sec;
 	boost::thread::sleep(xt);
 #endif // _OSMESA
diff -ru LDView.org/TRE/TREMainModel.cpp LDView/TRE/TREMainModel.cpp
--- LDView.org/TRE/TREMainModel.cpp	2011-02-12 05:24:14.000000000 +0100
+++ LDView/TRE/TREMainModel.cpp	2015-02-04 11:45:51.249290615 +0100
@@ -814,7 +814,7 @@
 #ifdef ANTI_DEADLOCK_HACK
 				boost::xtime xt;
 
-				boost::xtime_get(&xt, boost::TIME_UTC);
+				boost::xtime_get(&xt, boost::TIME_UTC_);
 				// 100,000,000 nsec == 100 msec
 				xt.nsec += 100 * 1000000;
 				// HACK: If any deadlocks are encountered during testing,
