--- LDView-4.1/LDLib/LDLibraryUpdater.cpp	2009-12-01 16:42:55.000000000 -0200
+++ LDView-4.1_patched/LDLib/LDLibraryUpdater.cpp	2013-11-20 20:39:36.003276476 -0200
@@ -1183,7 +1183,7 @@
 				// it to signal.
 				boost::xtime xt;
 
-				boost::xtime_get(&xt, boost::TIME_UTC);
+				boost::xtime_get(&xt, boost::TIME_UTC_);
 				xt.nsec += 250 * 1000 * 1000;
 				m_threadFinish->timed_wait(lock, xt);
 			}
--- LDView-4.1/TCFoundation/TCWebClient.cpp	2009-04-03 13:34:55.000000000 -0300
+++ LDView-4.1_patched/TCFoundation/TCWebClient.cpp	2013-11-20 20:39:13.900470975 -0200
@@ -118,7 +118,7 @@
 #else // _OSMESA
 	boost::xtime xt;
 
-	boost::xtime_get(&xt, boost::TIME_UTC);
+	boost::xtime_get(&xt, boost::TIME_UTC_);
 	xt.sec += sec;
 	boost::thread::sleep(xt);
 #endif // _OSMESA
--- LDView-4.1/TRE/TREMainModel.cpp	2009-11-09 22:10:29.000000000 -0200
+++ LDView-4.1_patched/TRE/TREMainModel.cpp	2013-11-20 20:39:50.523148711 -0200
@@ -779,7 +779,7 @@
 #ifdef ANTI_DEADLOCK_HACK
 				boost::xtime xt;
 
-				boost::xtime_get(&xt, boost::TIME_UTC);
+				boost::xtime_get(&xt, boost::TIME_UTC_);
 				// 100,000,000 nsec == 100 msec
 				xt.nsec += 100 * 1000000;
 				// HACK: If any deadlocks are encountered during testing,
