--- LDView-4.1/TCFoundation/TCArray.h	2008-05-18 18:16:21.000000000 -0300
+++ LDView-4.1_patched/TCFoundation/TCArray.h	2012-09-13 14:32:40.000000000 -0300
@@ -2,6 +2,7 @@
 #define __TCARRAY_H__
 
 #include <TCFoundation/TCObject.h>
+#include <TCFoundation/mystring.h>
 
 typedef int (*TCArraySortFunction)(const void*, const void*);
 
--- LDView-4.1/TCFoundation/mystring.h	2009-11-17 00:45:34.000000000 -0200
+++ LDView-4.1_patched/TCFoundation/mystring.h	2012-09-13 14:33:51.000000000 -0300
@@ -13,6 +13,10 @@
 #include <Qt/qstring.h>
 #endif // _QT
 
+#ifndef WIN32
+#include <unistd.h>
+#endif
+
 #ifdef _OSMESA
 #include <stdarg.h>
 #endif // _OSMESA
