--- LDView.org/TCFoundation/TCArray.h	2008-05-18 23:16:21.000000000 +0200
+++ LDView/TCFoundation/TCArray.h	2015-02-04 11:38:48.394308398 +0100
@@ -2,6 +2,7 @@
 #define __TCARRAY_H__
 
 #include <TCFoundation/TCObject.h>
+#include <TCFoundation/mystring.h>
 
 typedef int (*TCArraySortFunction)(const void*, const void*);
 
diff -ru LDView.org/TCFoundation/mystring.h TCFoundation/mystring.h
--- LDView/TCFoundation/mystring.h	2011-05-26 06:51:55.000000000 +0200
+++ TCFoundation/mystring.h	2015-02-04 11:39:16.529307215 +0100
@@ -13,6 +13,10 @@
 #include <Qt/qstring.h>
 #endif // _QT
 
+#ifndef WIN32
+#include <unistd.h>
+#endif
+
 #ifdef _OSMESA
 #include <stdarg.h>
 #endif // _OSMESA
