diff -Naur Maelstrom-3.0.6/screenlib/SDL_FrameBuf.cpp Maelstrom-3.0.6.patched/screenlib/SDL_FrameBuf.cpp
--- Maelstrom-3.0.6/screenlib/SDL_FrameBuf.cpp	2001-07-22 17:03:13.000000000 -0400
+++ Maelstrom-3.0.6.patched/screenlib/SDL_FrameBuf.cpp	2009-04-14 14:15:49.000000000 -0400
@@ -848,8 +848,8 @@
 			for ( i=0; i<dirtymaplen; ++i ) {
 				if ( dirtymap[i] != NULL ) {
 					dirtymap[i] = (SDL_Rect *)(
-					((int)dirtymap[i]-(int)updatelist) +
-								(int)newlist
+					((long)dirtymap[i]-(long)updatelist) +
+								(long)newlist
 					);
 				}
 			}
