diff -Naur Maelstrom-4.0.2/game/main.cpp Maelstrom-4.0.2.patched/game/main.cpp
--- Maelstrom-4.0.2/game/main.cpp	2026-06-25 00:23:18.000000000 -0400
+++ Maelstrom-4.0.2.patched/game/main.cpp	2026-07-30 01:52:08.462941592 -0400
@@ -178,7 +178,7 @@
 #ifdef SDL_PLATFORM_EMSCRIPTEN
 	window_flags |= SDL_WINDOW_FILL_DOCUMENT;
 #else
-	window_flags |= SDL_WINDOW_FULLSCREEN;
+	window_flags &= ~SDL_WINDOW_FULLSCREEN;
 #endif
 	window_flags |= SDL_WINDOW_HIGH_PIXEL_DENSITY;
 
