diff -urp aeskulap-0.2.2-beta1//configuration/aconfiguration-gconf.cpp aeskulap-0.2.2-beta1.new//configuration/aconfiguration-gconf.cpp
--- aeskulap-0.2.2-beta1//configuration/aconfiguration-gconf.cpp	2007-05-10 18:44:00.000000000 +0200
+++ aeskulap-0.2.2-beta1.new//configuration/aconfiguration-gconf.cpp	2011-08-08 19:58:17.986000024 +0200
@@ -28,6 +28,7 @@
 
 #include <gconfmm.h>
 #include <iostream>
+#include <cstdio>
 
 namespace Aeskulap {
 
diff -urp aeskulap-0.2.2-beta1//configure.in aeskulap-0.2.2-beta1.new//configure.in
--- aeskulap-0.2.2-beta1//configure.in	2007-05-24 20:59:09.000000000 +0200
+++ aeskulap-0.2.2-beta1.new//configure.in	2011-08-09 15:36:37.598000023 +0200
@@ -166,6 +166,6 @@ if test x"$MINGW32" = xno ; then
 cd $srcdir/dcmtk
 export CFLAGS="$CFLAGS"
 export CXXFLAGS="$CXXFLAGS"
-sh ./configure --without-openssl --without-zlib --without-libwrap --prefix=${prefix} --libdir=${prefix}/lib/aeskulap
+sh ./configure --without-openssl --without-zlib --without-libwrap --prefix=${prefix} --libdir=${libdir}/aeskulap --sysconfdir=${sysconfdir} --localstatedir=${localstatedir} --mandir=${mandir} 
 
 fi
