--- glob2-0.9.4.4/SConstruct.original	2014-04-09 06:25:12.710804438 +0300
+++ glob2-0.9.4.4/SConstruct	2014-04-09 06:47:08.010360930 +0300
@@ -109,6 +109,14 @@
         else:
             print "Could not find libz or zlib1.dll"
             missing.append("zlib")
+    
+    boost_system = ''
+    if conf.CheckLib("boost_system"):
+        boost_system="boost_system"
+    else:
+        print "Could not find libboost_system"
+        missing.append("libboost_system")
+    env.Append(LIBS=[boost_system])
 
     boost_thread = ''
     if conf.CheckLib("boost_thread") and conf.CheckCXXHeader("boost/thread/thread.hpp"):
