--- glob2-0.9.4.4/libgag/src/TextStream.original.cpp	2014-04-06 22:37:28.204008827 +0300
+++ glob2-0.9.4.4/libgag/src/TextStream.cpp	2014-04-06 22:37:59.509617453 +0300
@@ -17,11 +17,11 @@
   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 */
 
-#include <TextStream.h>
 #include <assert.h>
 #include <fstream>
 #include <iostream>
 #include <stack>
+#include <TextStream.h>
 #ifdef WIN32
 #define snprintf _snprintf
 #define vsnprintf _vsnprintf
