diff -Nur 3proxy-0.6.orig/scripts/add3proxyuser.sh 3proxy-0.6/scripts/add3proxyuser.sh
--- 3proxy-0.6.orig/scripts/add3proxyuser.sh	2007-06-11 03:18:12.000000000 -0500
+++ 3proxy-0.6/scripts/add3proxyuser.sh	2009-06-24 12:27:33.614202884 -0500
@@ -1,8 +1,8 @@
 #!/bin/sh
 if [ $4 ]; then  
-	echo $1:`/usr/local/etc/3proxy/bin/mycrypt $$ $2` >> /usr/local/etc/3proxy/passwd
-	echo countin \"`wc -l /usr/local/etc/3proxy/counters|awk '{print $1}'`/$1\" D $3 $1 >> /usr/local/etc/3proxy/counters
-	echo bandlimin $4 $1 >> /usr/local/etc/3proxy/bandlimiters
+	echo $1:`/usr/bin/mycrypt $$ $2` >> /etc/3proxy/passwd
+	echo countin \"$(wc -l /etc/3proxy/counters | awk '{print $1}')/$1\" D $3 $1 >> /etc/3proxy/counters
+	echo bandlimin $4 $1 >> /etc/3proxy/bandlimiters
 else
 	echo usage: $0 username password day_limit bandwidth
 	echo "	"day_limit - traffic limit in MB per day
