diff -Nur 3proxy-0.6.orig/src/stringtable.c 3proxy-0.6/src/stringtable.c
--- 3proxy-0.6.orig/src/stringtable.c	2008-02-03 05:28:24.000000000 -0600
+++ 3proxy-0.6/src/stringtable.c	2009-06-24 12:21:01.637499557 -0500
@@ -38,7 +38,7 @@
 /* 24 */	NULL,
 #ifndef TPROXY_CONF
 #ifndef _WIN32
-/* 25 */	(unsigned char *)"/usr/local/etc/3proxy/3proxy.cfg",
+/* 25 */	(unsigned char *)"/etc/3proxy/3proxy.cfg",
 #else
 /* 25 */	(unsigned char *)"3proxy.cfg",
 #endif
