--- meld-3.23.0/meld/actiongutter.py.orig	2025-05-15 19:50:55.488897719 +0700
+++ meld-3.23.0/meld/actiongutter.py	2025-05-15 19:52:15.430943358 +0700
@@ -215,8 +215,8 @@
     def motion_event(
         self,
         controller: Gtk.EventControllerMotion,
-        x: float | None = None,
-        y: float | None = None,
+        x: "float | None" = None,
+        y: "float | None" = None,
     ):
         if x is None or y is None:
             # Missing coordinates are leave events
--- meld-3.23.0/meld/chunkmap.py.orig	2025-05-15 19:52:22.395511767 +0700
+++ meld-3.23.0/meld/chunkmap.py	2025-05-15 19:52:40.767606420 +0700
@@ -258,8 +258,8 @@
     def motion_event(
         self,
         controller: Gtk.EventControllerMotion,
-        x: float | None = None,
-        y: float | None = None,
+        x: "float | None" = None,
+        y: "float | None" = None,
     ):
         if self._have_grab:
             self._scroll_fraction(y, animate=False)
