diff -Naur kdesvn-1.5.5/doc/nl/index.docbook kdesvn-1.5.5_patched/doc/nl/index.docbook
--- kdesvn-1.5.5/doc/nl/index.docbook	2008-05-31 10:02:54.000000000 +0000
+++ kdesvn-1.5.5_patched/doc/nl/index.docbook	2011-03-30 14:55:51.570351181 +0000
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.1.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
   <!ENTITY kdesvn "<application>kdesvn</application>">
   <!ENTITY kappname "&kdesvn;"><!-- Do *not* replace kappname-->
   <!ENTITY package "kde-module"><!-- kdebase, kdeadmin, etc -->
