--- config.def.h	2019-06-04 20:24:48.426235490 +0100
+++ config.h	2019-06-13 08:17:45.674540960 +0100
@@ -84,6 +84,16 @@
 	{ MODKEY,                       XK_period, focusmon,       {.i = +1 } },
 	{ MODKEY|ShiftMask,             XK_comma,  tagmon,         {.i = -1 } },
 	{ MODKEY|ShiftMask,             XK_period, tagmon,         {.i = +1 } },
+	TAGKEYS(                        XK_KP_End,                 0)
+	TAGKEYS(                        XK_KP_Down,                1)
+	TAGKEYS(                        XK_KP_Next,                2)
+	TAGKEYS(                        XK_KP_Left,                3)
+	TAGKEYS(                        XK_KP_Begin,               4)
+	TAGKEYS(                        XK_KP_Right,               5)
+	TAGKEYS(                        XK_KP_Home,                6)
+	TAGKEYS(                        XK_KP_Up,                  7)
+	TAGKEYS(                        XK_KP_Prior,               8)
+	TAGKEYS(                        XK_KP_Insert,              9)
 	TAGKEYS(                        XK_1,                      0)
 	TAGKEYS(                        XK_2,                      1)
 	TAGKEYS(                        XK_3,                      2)
