diff -Naur a/config.def.h b/config.def.h
--- a/config.def.h	2022-10-04 17:38:18.000000000 +0000
+++ b/config.def.h	2023-04-29 00:25:20.697595822 +0000
@@ -56,12 +56,12 @@
 #define SHCMD(cmd) { .v = (const char*[]){ "/bin/sh", "-c", cmd, NULL } }
 
 /* commands */
-static const char *dmenucmd[] = { "dmenu_run", "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
+static const char *roficmd[] = { "rofi", "-sort", "-show", "run", "-theme", "dwm", NULL };
 static const char *termcmd[]  = { "st", NULL };
 
 static const Key keys[] = {
 	/* modifier                     key        function        argument */
-	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
+	{ MODKEY,                       XK_p,      spawn,          {.v = roficmd } },
 	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
 	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
diff -Naur a/dwm.1 b/dwm.1
--- a/dwm.1	2022-10-04 17:38:18.000000000 +0000
+++ b/dwm.1	2023-04-29 00:26:08.906028725 +0000
@@ -62,7 +62,7 @@
 .TP
 .B Mod1\-p
 Spawn
-.BR dmenu(1)
+.BR rofi(1)
 for launching other programs.
 .TP
 .B Mod1\-,
@@ -156,7 +156,7 @@
 dwm is customized by creating a custom config.h and (re)compiling the source
 code. This keeps it fast, secure and simple.
 .SH SEE ALSO
-.BR dmenu (1),
+.BR rofi (1),
 .BR st (1)
 .SH ISSUES
 Java applications which use the XToolkit/XAWT backend may draw grey windows
