diff -Naur dwm-6.2/config.def.h b/config.def.h
--- dwm-6.2/config.def.h	2019-02-02 12:55:28.000000000 +0000
+++ b/config.def.h	2019-04-14 14:04:30.190258413 +0100
@@ -44,7 +44,7 @@
 };
 
 /* key definitions */
-#define MODKEY Mod1Mask
+#define MODKEY  Mod4Mask
 #define TAGKEYS(KEY,TAG) \
 	{ MODKEY,                       KEY,      view,           {.ui = 1 << TAG} }, \
 	{ MODKEY|ControlMask,           KEY,      toggleview,     {.ui = 1 << TAG} }, \
