diff --git a/Makefile b/Makefile
index ee7eba6..65f1426 100644
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,9 @@
 PREFIX  ?= /usr/local
 BINDIR  ?= $(PREFIX)/bin
 MANDIR  ?= $(PREFIX)/share/man/man1
+CC      ?= cc
 
-CFLAGS += -std=c99 -march=native -O3 -pipe
+CFLAGS += -std=c99 -O3 -pipe
 CFLAGS += -Wall
 CFLAGS += -Wconversion
 CFLAGS += -Wdouble-promotion
@@ -11,11 +12,10 @@ CFLAGS += -Wmissing-prototypes
 CFLAGS += -Wold-style-definition
 CFLAGS += -Wpedantic
 CFLAGS += -Wshadow
-
 all: xhidecursor
 
 xhidecursor: main.c Makefile
-	$(CC) $(CFLAGS) -o $@ $< -lX11 -lXfixes -lXi
+	$(CC) $(CFLAGS) -o $@ main.c -lX11 -lXfixes -lXi
 
 install: all
 	install -D xhidecursor $(DESTDIR)$(BINDIR)/xhidecursor
@@ -29,4 +29,3 @@ clean:
 	rm -f xhidecursor
 
 .PHONY: all install uninstall clean
-
