diff -Nurb libvmime-0.7.1.orig/src/mdn/MDNHelper.cpp libvmime-0.7.1/src/mdn/MDNHelper.cpp
--- libvmime-0.7.1.orig/src/mdn/MDNHelper.cpp	2006-11-28 09:43:47.995127250 +0100
+++ libvmime-0.7.1/src/mdn/MDNHelper.cpp	2006-11-28 09:43:27.505846750 +0100
@@ -246,6 +246,7 @@
 		(headerFieldFactory::getInstance()->create(vmime::fields::FINAL_RECIPIENT));
 
 	fr->setValue("rfc822; " + mdnInfos.getRecipient().getEmail());
+	fields.appendField(fr);
 
 	// -- Original-Message-ID
 	if (mdnInfos.getMessage()->getHeader()->hasField(vmime::fields::MESSAGE_ID))
