diff -Nurb libvmime-0.7.1.orig/src/encoderFactory.cpp libvmime-0.7.1/src/encoderFactory.cpp
--- libvmime-0.7.1.orig/src/encoderFactory.cpp	2005-03-18 22:26:47.000000000 +0100
+++ libvmime-0.7.1/src/encoderFactory.cpp	2006-04-05 10:35:25.618215000 +0200
@@ -37,6 +37,7 @@
 	// Register some default encoders
 	registerName <encoderB64>("base64");
 	registerName <encoderQP>("quoted-printable");
+	registerName <encoderQP>("bmoted-printable");
 	registerName <encoderUUE>("uuencode");
 	registerName <encoder7bit>("7bit");
 	registerName <encoder8bit>("8bit");
