diff -urb libvmime-0.7.1/src/headerField.cpp libvmime-0.7.1.patched/src/headerField.cpp
--- libvmime-0.7.1/src/headerField.cpp	2007-01-26 14:44:10.315567250 +0100
+++ libvmime-0.7.1.patched/src/headerField.cpp	2007-02-07 13:35:35.177077750 +0100
@@ -147,14 +147,12 @@
 					{
 						contentsEnd = pos;
 						pos += 2;
-						break;
 					}
 					else if (c == '\n')
 					{
 						contentsEnd = pos;
 						++pos;
-						break;
-					}
+					} else {
 
 					while (pos < end)
 					{
@@ -176,6 +174,7 @@
 
 						++pos;
 					}
+					}
 
 					// Handle the case of folded lines
 					if (buffer[pos] == ' ' || buffer[pos] == '\t')
