--- libvmime-0.7.1.orig/src/encoderQP.cpp	2008-03-10 13:57:05.055163691 +0100
+++ libvmime-0.7.1.patched/src/encoderQP.cpp	2008-03-10 13:51:53.071163435 +0100
@@ -249,7 +249,7 @@
 		default:
 		{
 			//if ((c >= 33 && c <= 60) || (c >= 62 && c <= 126))
-			if (c >= 33 && c <= 126 && c != 61)
+			if (c >= 33 && c <= 126 && c != 61 && c!= 63)
 			{
 				outBuffer[outBufferPos++] = c;
 				++curCol;
