diff -Nurb libvmime-0.7.1.orig/src/platforms/posix/posixHandler.cpp libvmime-0.7.1.edited/src/platforms/posix/posixHandler.cpp
--- libvmime-0.7.1.orig/src/platforms/posix/posixHandler.cpp	2005-04-30 09:53:21.000000000 +0200
+++ libvmime-0.7.1.edited/src/platforms/posix/posixHandler.cpp	2006-12-28 17:29:43.883410000 +0100
@@ -106,9 +106,9 @@
 
 const vmime::charset posixHandler::getLocaleCharset() const
 {
-	vmime::string prevLocale(::setlocale(LC_ALL, ""));
+	char *prevLocale = ::setlocale(LC_ALL, "");
 	vmime::charset ch(::nl_langinfo(CODESET));
-	::setlocale(LC_ALL, prevLocale.c_str());
+	::setlocale(LC_ALL, prevLocale);
 
 	return (ch);
 }
