--- libvmime-0.7.1-pristine/src/messaging/transport.cpp	2005-04-27 18:42:58.000000000 +0200
+++ libvmime-0.7.1/src/messaging/transport.cpp	2006-10-10 14:01:43.602621328 +0200
@@ -90,6 +90,14 @@
 	}
 	catch (exceptions::no_such_field&) { }
 
+	// Remove BCC headers from the message we're about to send
+	try {
+		headerField* bcc = msg->getHeader()->findField(fields::BCC);
+
+		msg->getHeader()->removeField(bcc);
+	}
+	catch (exceptions::no_such_field&) { }
+
 	// Generate the message, "stream" it and delegate the sending
 	// to the generic send() function.
 	std::ostringstream oss;
