--- libvmime-0.7.1.orig/src/body.cpp	2007-01-26 14:44:10.259563000 +0100
+++ libvmime-0.7.1.patched/src/body.cpp	2007-07-13 12:14:30.699846500 +0200
@@ -465,6 +465,9 @@
 		const contentEncodingField& cef = dynamic_cast<contentEncodingField&>
 			(*m_header->findField(fields::CONTENT_TRANSFER_ENCODING));
 
+		if (cef.getValue().getName().empty())
+			return (vmime::encoding(encodingTypes::SEVEN_BIT));
+
 		return (cef.getValue());
 	}
 	catch (exceptions::no_such_field&)
