diff -Nurb libvmime-0.7.1.orig/src/mdn/MDNHelper.cpp libvmime-0.7.1/src/mdn/MDNHelper.cpp
--- libvmime-0.7.1.orig/src/mdn/MDNHelper.cpp	2005-04-09 21:45:07.000000000 +0200
+++ libvmime-0.7.1/src/mdn/MDNHelper.cpp	2006-11-28 09:37:56.009129500 +0100
@@ -143,7 +143,7 @@
 
 	hdr->ContentType().setValue(mediaType(vmime::mediaTypes::MULTIPART,
 	                                      vmime::mediaTypes::MULTIPART_REPORT));
-	hdr->ContentType().setReportType("disosition-notification");
+	hdr->ContentType().setReportType("disposition-notification");
 
 	hdr->Disposition().setValue(dispo);
 
