diff -Nurb libvmime-0.7.1.orig/src/word.cpp libvmime-0.7.1/src/word.cpp
--- libvmime-0.7.1.orig/src/word.cpp	2005-06-13 12:56:52.000000000 +0200
+++ libvmime-0.7.1/src/word.cpp	2005-11-15 18:02:45.000000000 +0100
@@ -688,7 +688,13 @@
 {
 	string out;
 
+	try {
 	charset::convert(m_buffer, out, m_charset, dest);
+	}
+	catch (vmime::exception &e) {
+		// copy 'word' as text
+		out = m_buffer;
+	}
 
 	return (out);
 }
