diff -Nurb libvmime-0.7.1.orig/src/dateTime.cpp libvmime-0.7.1/src/dateTime.cpp
--- libvmime-0.7.1.orig/src/dateTime.cpp	2005-04-23 11:02:48.000000000 +0200
+++ libvmime-0.7.1/src/dateTime.cpp	2009-01-08 10:51:39.000000000 +0100
@@ -423,7 +423,7 @@
 				zone[zoneLength++] = *p;
 				++p;
 			}
-			while (zoneLength < 3 && p < pend && parserHelpers::isDigit(*p));
+			while (zoneLength < 3 && p < pend);
 
 			switch (zone[0])
 			{
@@ -487,6 +487,8 @@
 
 				break;
 			}
+			case 'g':
+			case 'G':
 			case 'u':
 			case 'U':
 			{
