--- doomretro-2.5.7/src/c_cmds.c.orig	2017-11-02 23:56:12.327149365 +0700
+++ doomretro-2.5.7/src/c_cmds.c	2017-11-02 23:56:42.013185704 +0700
@@ -2734,13 +2734,13 @@
             C_TabbedOutput(tabs, "Music format\t<b>MIDI</b>");
         else if (musictype == MUS_OGG)
             C_TabbedOutput(tabs, "Music format\t<b>Ogg Vorbis</b>");
-        else if (musictype == MUS_MP3 || musictype == MUS_MP3_MAD)
+        else if (musictype == MUS_MP3)
             C_TabbedOutput(tabs, "Music format\t<b>MP3</b>");
         else if (musictype == MUS_WAV)
             C_TabbedOutput(tabs, "Music format\t<b>WAV</b>");
         else if (musictype == MUS_FLAC)
             C_TabbedOutput(tabs, "Music format\t<b>FLAC</b>");
-        else if (musictype == MUS_MOD || musictype == MUS_MODPLUG)
+        else if (musictype == MUS_MOD)
             C_TabbedOutput(tabs, "Music format\t<b>MOD</b>");
     }
 }
