diff -Nur gpsd-2.39.orig/libgps.pc.in gpsd-2.39/libgps.pc.in
--- gpsd-2.39.orig/libgps.pc.in	2009-02-17 06:56:57.000000000 -0600
+++ gpsd-2.39/libgps.pc.in	2009-10-05 14:30:12.894417725 -0500
@@ -1,6 +1,6 @@
 prefix=@prefix@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=@libdir@
 includedir=${prefix}/include
 
 Name: GPSD
diff -Nur gpsd-2.39.orig/libgpsd.pc.in gpsd-2.39/libgpsd.pc.in
--- gpsd-2.39.orig/libgpsd.pc.in	2009-02-17 06:56:57.000000000 -0600
+++ gpsd-2.39/libgpsd.pc.in	2009-10-05 14:30:22.191417725 -0500
@@ -1,6 +1,6 @@
 prefix=@prefix@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=@libdir@
 includedir=${prefix}/include
 
 Name: libgpsd
