diff -Naur MediaInfoLib-25.04.orig/Source/MediaInfo/Export/Export_Graph.cpp MediaInfoLib-25.04/Source/MediaInfo/Export/Export_Graph.cpp
--- MediaInfoLib-25.04.orig/Source/MediaInfo/Export/Export_Graph.cpp	2025-04-30 13:43:55.000000000 +0200
+++ MediaInfoLib-25.04/Source/MediaInfo/Export/Export_Graph.cpp	2025-06-14 22:38:39.083581000 +0200
@@ -116,7 +116,7 @@
     GVC_t* Context=NULL;
     graph_t* Graph=NULL;
     char* Buffer=NULL;
-    unsigned int Size;
+    size_t Size;
     bool Cairo=false;
 
     if (!Export_Graph::Load())
diff -Naur MediaInfoLib-25.04.orig/Source/MediaInfo/Export/Export_Graph_gvc_Include.h MediaInfoLib-25.04/Source/MediaInfo/Export/Export_Graph_gvc_Include.h
--- MediaInfoLib-25.04.orig/Source/MediaInfo/Export/Export_Graph_gvc_Include.h	2025-04-30 13:43:55.000000000 +0200
+++ MediaInfoLib-25.04/Source/MediaInfo/Export/Export_Graph_gvc_Include.h	2025-06-14 22:39:21.435581000 +0200
@@ -102,7 +102,7 @@
 typedef char** (*gvc_Module_gvPluginList)(GVC_t*, const char*, int*); static gvc_Module_gvPluginList gvPluginList=NULL;
 typedef int (*gvc_Module_gvLayout)(GVC_t*, graph_t*, const char*); static gvc_Module_gvLayout gvLayout=NULL;
 typedef int (*gvc_Module_gvFreeLayout)(GVC_t*, graph_t*); static gvc_Module_gvFreeLayout gvFreeLayout=NULL;
-typedef int (*gvc_Module_gvRenderData)(GVC_t*, graph_t*, const char*, char**, unsigned int*); static gvc_Module_gvRenderData gvRenderData=NULL;
+typedef int (*gvc_Module_gvRenderData)(GVC_t*, graph_t*, const char*, char**, size_t*); static gvc_Module_gvRenderData gvRenderData=NULL;
 typedef void (*gvc_Module_gvFreeRenderData)(char*); static gvc_Module_gvFreeRenderData gvFreeRenderData=NULL;
 typedef void (*gvc_Module_gvFinalize)(GVC_t*); static gvc_Module_gvFinalize gvFinalize=NULL;
 typedef graph_t* (*cgraph_Module_agmemread)(const char*); static cgraph_Module_agmemread agmemread=NULL;
