--- deis-1.6.0/setup.orig.py	2015-05-09 03:43:01.776728194 +0300
+++ deis-1.6.0/setup.py	2015-05-09 03:44:20.037749797 +0300
@@ -2,6 +2,7 @@
 
 """Install the Deis command-line client."""
 
+import os
 
 try:
     from setuptools import setup
@@ -27,9 +28,11 @@
     KWARGS = {'scripts': ['deis']}
 
 
-with open('requirements.txt') as f:
-    required = f.read().splitlines()
-    required = [r for r in required if r.strip() and not r.startswith('#')]
+required = ""
+if os.path.isfile('requirements.txt'):
+    with open('requirements.txt') as f:
+        required = f.read().splitlines()
+        required = [r for r in required if r.strip() and not r.startswith('#')]
 
 
 setup(name='deis',
