--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -462,11 +462,15 @@
 #        Finding packages                  #
 ############################################
 
-set(BOOST_COMPONENTS date_time filesystem locale program_options system thread)
+set(BOOST_COMPONENTS date_time filesystem locale program_options thread)
 if(ENABLE_INNOEXTRACT)
 	list(APPEND BOOST_COMPONENTS iostreams)
 endif()
 find_package(Boost 1.48.0 REQUIRED COMPONENTS ${BOOST_COMPONENTS})
+if(Boost_MAJOR_VERSION EQUAL 1 AND Boost_MINOR_VERSION LESS 69)
+	list(APPEND BOOST_COMPONENTS system)
+	find_package(Boost COMPONENTS ${BOOST_COMPONENTS})
+endif()
 
 find_package(ZLIB REQUIRED)
 # Conan compatibility
