--- vcmi-0.95/CMakeLists.txt	2014-02-26 16:19:14.000000000 +0200
+++ vcmi-0.95/CMakeLists.original	2014-03-16 15:34:08.244278825 +0200
@@ -127,7 +127,7 @@
 		set(LIB_DIR "${CMAKE_INSTALL_LIBDIR}/vcmi" CACHE STRING "Where to install main library")
 	endif()
 	if (NOT DATA_DIR)
-		set(DATA_DIR "share/vcmi" CACHE STRING "Where to install data files")
+		set(DATA_DIR "share/games/vcmi" CACHE STRING "Where to install data files")
 	endif()
 endif()
 
