diff -Naur wmix-3.5/wmix.1x wmix-3.5.patched/wmix.1x
--- wmix-3.5/wmix.1x	2019-05-10 06:04:53.000000000 -0400
+++ wmix-3.5.patched/wmix.1x	2022-02-08 13:03:35.228542615 -0500
@@ -1,4 +1,4 @@
-.TH wmix 1x "3.2" "Fredrik Steen"
+.TH wmix 1x "3.5" "Fredrik Steen"
 .SH NAME
 wmix \- Dockapp mixer for OSS or ALSA
 .
@@ -24,6 +24,9 @@
 .
 .SH OPTIONS
 .TP
+\fB\-a\fR <\fIapi\fP>
+Use this sound API (\fIoss\fP or \fIalsa\fP, default is \fIalsa\fP).
+.TP
 \fB\-d\fR <\fIdisplay\fP>
 Connect to remote X display
 .TP
@@ -47,8 +50,8 @@
 below)
 .TP
 \fB\-m\fR <\fIdevice\fP>
-Use specified mixer device instead of
-.I /dev/mixer
+For oss, use specified mixer device instead of \fI/dev/mixer\fP.
+For alsa, use specified card name instead of \fIdefault\fP.
 .TP
 \fB\-v\fR
 Verbose start, report detected channels
@@ -58,8 +61,12 @@
 is read on startup.
 It may contains one configuration command per line:
 .TP
+api=<\fIapi\fP>
+Use this sound API (same as \fB\-a\fR option).
+.TP
 device=<\fImixer_device\fP>
-defines the device file to use instead of \fB/dev/mixer\fP
+For oss, defines the device file to use instead of \fB/dev/mixer\fP.
+For alsa, defines the audio card to use instead of \fBdefault\fP.
 .TP
 exclude=<\fIchannel_name\fP>
 exclude the specified channel from control, you can use it as many times as needed (see
