--- a/plugins/sorttab_display/normal_sorttab_page.c
+++ b/plugins/sorttab_display/normal_sorttab_page.c
@@ -41,7 +41,7 @@
 #include "normal_sorttab_page.h"
 #include "sorttab_display_context_menu.h"
 
-#define NO_IDLE_SELECTION_CALLBACK -1
+#define NO_IDLE_SELECTION_CALLBACK 0
 
 G_DEFINE_TYPE( NormalSortTabPage, normal_sort_tab_page, GTK_TYPE_TREE_VIEW);
 
@@ -570,7 +570,7 @@
 static void _st_selection_changed(GtkTreeSelection *selection, gpointer user_data) {
 
     NormalSortTabPage *self = NORMAL_SORT_TAB_PAGE(user_data);
-    gulong idleId = NO_IDLE_SELECTION_CALLBACK;
+    guint idleId = NO_IDLE_SELECTION_CALLBACK;
 
     if (self) {
         NormalSortTabPagePrivate *priv = NORMAL_SORT_TAB_PAGE_GET_PRIVATE(self);