Index: gtkpod-2.1.2/src/anjuta-app.c
===================================================================
--- gtkpod-2.1.2.orig/src/anjuta-app.c
+++ gtkpod-2.1.2/src/anjuta-app.c
@@ -38,6 +38,7 @@
 #include <libanjuta/resources.h>
 #include <libanjuta/anjuta-plugin-manager.h>
 #include <libanjuta/anjuta-debug.h>
+#include <libanjuta/anjuta-version.h>
 
 #include "anjuta-app.h"
 #include "anjuta-actions.h"
@@ -501,7 +502,11 @@ static void anjuta_app_instance_init(Anj
     g_list_free(plugins_dirs);
 
     /* Preferences */
+#if ANJUTA_CHECK_VERSION(3,5,3)
+    app->preferences = anjuta_preferences_new(app->plugin_manager, PREF_SCHEMA);
+#else
     app->preferences = anjuta_preferences_new(app->plugin_manager);
+#endif
     g_object_add_weak_pointer(G_OBJECT (app->preferences), (gpointer) &app->preferences);
 
     g_signal_connect(app->settings, "changed::" GDL_STYLE, G_CALLBACK(on_gdl_style_changed), app);
