diff -Naur KDDockWidgets-2.2.3.orig/CMakeLists.txt KDDockWidgets-2.2.3/CMakeLists.txt
--- KDDockWidgets-2.2.3.orig/CMakeLists.txt	2025-04-02 01:27:12.000000000 +0200
+++ KDDockWidgets-2.2.3/CMakeLists.txt	2026-03-12 15:34:34.165778782 +0100
@@ -177,11 +177,19 @@
 if(KDDockWidgets_FRONTENDS)
     if("qtwidgets" IN_LIST KDDockWidgets_FRONTENDS)
         find_package(Qt${QT_VERSION_MAJOR} ${QT_MIN_VERSION} NO_MODULE REQUIRED COMPONENTS Widgets)
+        if(Qt6Core_VERSION VERSION_GREATER_EQUAL "6.10.0")
+            set(QT_NO_PRIVATE_MODULE_WARNING ON)
+            find_package(Qt6 ${QT_MIN_VERSION} NO_MODULE REQUIRED COMPONENTS WidgetsPrivate)
+        endif()
         set(KDDW_FRONTEND_QTWIDGETS ON)
     endif()
 
     if("qtquick" IN_LIST KDDockWidgets_FRONTENDS)
         find_package(Qt${QT_VERSION_MAJOR} ${QT_MIN_VERSION} NO_MODULE REQUIRED COMPONENTS Quick QuickControls2)
+        if(Qt6Core_VERSION VERSION_GREATER_EQUAL "6.10.0")
+            set(QT_NO_PRIVATE_MODULE_WARNING ON)
+            find_package(Qt6 ${QT_MIN_VERSION} NO_MODULE REQUIRED COMPONENTS QuickPrivate)
+        endif()
         set(KDDW_FRONTEND_QTQUICK ON)
     endif()
 
@@ -203,12 +211,20 @@
     if(Qt${QT_VERSION_MAJOR}Widgets_FOUND)
         list(APPEND ENABLED_FRONTENDS "qtwidgets")
         set(KDDW_FRONTEND_QTWIDGETS ON)
+        if(Qt6Core_VERSION VERSION_GREATER_EQUAL "6.10.0")
+            set(QT_NO_PRIVATE_MODULE_WARNING ON)
+            find_package(Qt6 ${QT_MIN_VERSION} NO_MODULE REQUIRED COMPONENTS WidgetsPrivate)
+        endif()
     endif()
 
     # qtquick
     if(Qt${QT_VERSION_MAJOR}Quick_FOUND AND Qt${QT_VERSION_MAJOR}QuickControls2_FOUND)
         list(APPEND ENABLED_FRONTENDS "qtquick")
         set(KDDW_FRONTEND_QTQUICK ON)
+        if(Qt6Core_VERSION VERSION_GREATER_EQUAL "6.10.0")
+            set(QT_NO_PRIVATE_MODULE_WARNING ON)
+            find_package(Qt6 ${QT_MIN_VERSION} NO_MODULE REQUIRED COMPONENTS QuickPrivate)
+        endif()
     endif()
 
     if(NOT ENABLED_FRONTENDS)
