diff -Naur a/p_ac3.cpp b/p_ac3.cpp
--- a/p_ac3.cpp	2003-05-04 09:41:26.000000000 +0000
+++ b/p_ac3.cpp	2021-04-29 17:46:25.711376156 +0000
@@ -30,7 +30,7 @@
 ac3_packetizer_c::ac3_packetizer_c(unsigned long nsamples_per_sec,
                                    int nchannels, int nbitrate,
                                    audio_sync_t *nasync, range_t *nrange,
-                                   char **ncomments) throw (error_c) : q_c() {
+                                   char **ncomments) : q_c() {
   serialno = create_unique_serial();
   ogg_stream_init(&os, serialno);
   packetno = 0;
diff -Naur a/p_ac3.h b/p_ac3.h
--- a/p_ac3.h	2003-01-03 20:27:52.000000000 +0000
+++ b/p_ac3.h	2021-04-29 17:46:26.053379762 +0000
@@ -40,7 +40,7 @@
   public:
     ac3_packetizer_c(unsigned long nsamples_per_sec, int nchannels,
                      int nbitrate, audio_sync_t *nasync,
-                     range_t *nrange, char **ncomments) throw (error_c);
+                     range_t *nrange, char **ncomments);
     virtual ~ac3_packetizer_c();
     
     virtual int     process(char *buf, int size, int last_frame);
diff -Naur a/p_index.cpp b/p_index.cpp
--- a/p_index.cpp	2003-05-04 09:41:26.000000000 +0000
+++ b/p_index.cpp	2021-04-29 17:46:25.792377010 +0000
@@ -28,7 +28,7 @@
 #include "p_index.h"
 #include "vorbis_header_utils.h"
 
-index_packetizer_c::index_packetizer_c(int nserial) throw (error_c) : q_c() {
+index_packetizer_c::index_packetizer_c(int nserial) : q_c() {
   serialno = create_unique_serial();
   ogg_stream_init(&os, serialno);
   granulepos = 0;
diff -Naur a/p_index.h b/p_index.h
--- a/p_index.h	2003-05-04 09:41:26.000000000 +0000
+++ b/p_index.h	2021-04-29 17:46:25.976378950 +0000
@@ -33,7 +33,7 @@
     ogg_int64_t granulepos, packetno;
     int         serial;
   public:
-    index_packetizer_c(int nserial) throw (error_c);
+    index_packetizer_c(int nserial);
     virtual ~index_packetizer_c();
     
     virtual int     process(idx_entry *entries, int num);
diff -Naur a/p_mp3.cpp b/p_mp3.cpp
--- a/p_mp3.cpp	2003-05-20 07:08:37.000000000 +0000
+++ b/p_mp3.cpp	2021-04-29 17:46:25.863377759 +0000
@@ -30,7 +30,7 @@
 mp3_packetizer_c::mp3_packetizer_c(unsigned long nsamples_per_sec,
                                    int nchannels, int nmp3rate,
                                    audio_sync_t *nasync, range_t *nrange,
-                                   char **ncomments) throw (error_c) : q_c() {
+                                   char **ncomments) : q_c() {
   serialno = create_unique_serial();
   ogg_stream_init(&os, serialno);
   packetno = 0;
diff -Naur a/p_mp3.h b/p_mp3.h
--- a/p_mp3.h	2003-01-03 20:27:52.000000000 +0000
+++ b/p_mp3.h	2021-04-29 17:46:26.025379467 +0000
@@ -40,7 +40,7 @@
   public:
     mp3_packetizer_c(unsigned long nsamples_per_sec, int nchannels,
                      int nmp3rate, audio_sync_t *nasync,
-                     range_t *nrange, char **ncomments) throw (error_c);
+                     range_t *nrange, char **ncomments);
     virtual ~mp3_packetizer_c();
     
     virtual int     process(char *buf, int size, int last_frame);
diff -Naur a/p_pcm.cpp b/p_pcm.cpp
--- a/p_pcm.cpp	2003-05-04 09:41:26.000000000 +0000
+++ b/p_pcm.cpp	2021-04-29 17:46:25.726376314 +0000
@@ -29,7 +29,7 @@
 pcm_packetizer_c::pcm_packetizer_c(unsigned long nsamples_per_sec,
                                    int nchannels, int nbits_per_sample,
                                    audio_sync_t *nasync, range_t *nrange,
-                                   char **ncomments) throw (error_c) : q_c() {
+                                   char **ncomments) : q_c() {
   serialno = create_unique_serial();
   ogg_stream_init(&os, serialno);
   packetno = 0;
diff -Naur a/p_pcm.h b/p_pcm.h
--- a/p_pcm.h	2003-01-03 20:27:52.000000000 +0000
+++ b/p_pcm.h	2021-04-29 17:46:25.838377495 +0000
@@ -37,7 +37,7 @@
   public:
     pcm_packetizer_c(unsigned long nsamples_per_sec, int nchannels,
                      int nbits_per_sample, audio_sync_t *nasync,
-                     range_t *nrange, char **ncomments) throw (error_c);
+                     range_t *nrange, char **ncomments);
     virtual ~pcm_packetizer_c();
     
     virtual int     process(char *buf, int size, int last_frame);
diff -Naur a/p_textsubs.cpp b/p_textsubs.cpp
--- a/p_textsubs.cpp	2003-10-23 20:01:53.000000000 +0000
+++ b/p_textsubs.cpp	2021-04-29 17:46:25.802377115 +0000
@@ -29,7 +29,7 @@
 
 textsubs_packetizer_c::textsubs_packetizer_c(audio_sync_t *nasync,
                                              range_t *nrange, char **ncomments)
-                                             throw (error_c) : q_c() {
+                                             : q_c() {
   serialno = create_unique_serial();
   ogg_stream_init(&os, serialno);
   packetno = 0;
diff -Naur a/p_textsubs.h b/p_textsubs.h
--- a/p_textsubs.h	2003-01-03 20:27:52.000000000 +0000
+++ b/p_textsubs.h	2021-04-29 17:46:25.985379045 +0000
@@ -30,7 +30,7 @@
 
   public:
     textsubs_packetizer_c(audio_sync_t *nasync, range_t *nrange,
-                          char **ncomments) throw (error_c);
+                          char **ncomments);
     virtual ~textsubs_packetizer_c();
     
     virtual int     process(ogg_int64_t start, ogg_int64_t end, char *_subs,
diff -Naur a/p_video.cpp b/p_video.cpp
--- a/p_video.cpp	2003-05-04 09:41:26.000000000 +0000
+++ b/p_video.cpp	2021-04-29 17:46:25.948378655 +0000
@@ -31,7 +31,7 @@
                                        int nheight, int nbpp,
                                        int nmax_frame_size, audio_sync_t *as,
                                        range_t *nrange, char **ncomments)
-                                       throw (error_c) : q_c() {
+                                       : q_c() {
   serialno = create_unique_serial();
   ogg_stream_init(&os, serialno);
   packetno = 0;
diff -Naur a/p_video.h b/p_video.h
--- a/p_video.h	2003-01-03 20:27:52.000000000 +0000
+++ b/p_video.h	2021-04-29 17:46:25.813377231 +0000
@@ -35,7 +35,7 @@
     range_t         range;
   public:
     video_packetizer_c(char *, double, int, int, int, int, audio_sync_t *,
-                       range_t *nrange, char **ncomments) throw (error_c);
+                       range_t *nrange, char **ncomments);
     virtual ~video_packetizer_c();
     
     virtual int            process(char *buf, int size, int num_frames, int key,
diff -Naur a/p_vobsub.cpp b/p_vobsub.cpp
--- a/p_vobsub.cpp	2003-05-04 09:41:26.000000000 +0000
+++ b/p_vobsub.cpp	2021-04-29 17:46:26.065379889 +0000
@@ -34,7 +34,7 @@
                                          char *nid, int nindex,
                                          audio_sync_t *nasync,
                                          range_t *nrange, char **ncomments)
-                                         throw (error_c) : q_c() {
+                                         : q_c() {
   char buffer[50];
   serialno = create_unique_serial();
   ogg_stream_init(&os, serialno);
diff -Naur a/p_vobsub.h b/p_vobsub.h
--- a/p_vobsub.h	2003-03-01 17:53:41.000000000 +0000
+++ b/p_vobsub.h	2021-04-29 17:46:25.855377674 +0000
@@ -43,7 +43,7 @@
     vobsub_packetizer_c(int nwidth, int nheight, char *npalette,
                         int nlangidx, char *nid, int nindex,
                         audio_sync_t *nasync, range_t *nrange,
-                        char **ncomments) throw (error_c);
+                        char **ncomments);
     virtual ~vobsub_packetizer_c();
     
     virtual int     process(ogg_int64_t start, ogg_int64_t end, char *subs,
diff -Naur a/p_vorbis.cpp b/p_vorbis.cpp
--- a/p_vorbis.cpp	2003-10-23 19:46:32.000000000 +0000
+++ b/p_vorbis.cpp	2021-04-29 17:46:25.934378507 +0000
@@ -30,7 +30,7 @@
 #include "vorbis_header_utils.h"
 
 vorbis_packetizer_c::vorbis_packetizer_c(audio_sync_t *nasync, range_t *nrange,
-                                         char **ncomments) throw (error_c)
+                                         char **ncomments)
                                          : q_c() {
   packetno = 0;
   old_granulepos = 0;
diff -Naur a/p_vorbis.h b/p_vorbis.h
--- a/p_vorbis.h	2003-01-03 20:27:52.000000000 +0000
+++ b/p_vorbis.h	2021-04-29 17:46:25.776376841 +0000
@@ -37,7 +37,7 @@
 
   public:
     vorbis_packetizer_c(audio_sync_t *nasync, range_t *nrange, char **ncomments)
-                        throw (error_c);
+                       ;
     virtual ~vorbis_packetizer_c();
     
     virtual int     process(ogg_packet *op, ogg_int64_t gran);
diff -Naur a/queue.cpp b/queue.cpp
--- a/queue.cpp	2004-04-02 18:19:35.000000000 +0000
+++ b/queue.cpp	2021-04-29 17:46:26.011379319 +0000
@@ -21,7 +21,7 @@
 #include "ogmmerge.h"
 #include "queue.h"
 
-q_c::q_c() throw (error_c) : generic_packetizer_c() {
+q_c::q_c() : generic_packetizer_c() {
   first = NULL;
   current = NULL;
   next_is_key = -1;
diff -Naur a/queue.h b/queue.h
--- a/queue.h	2003-02-05 08:37:07.000000000 +0000
+++ b/queue.h	2021-04-29 17:46:26.079380037 +0000
@@ -36,7 +36,7 @@
     ogg_stream_state  os;
     
   public:
-    q_c() throw (error_c);
+    q_c();
     virtual ~q_c();
     
     virtual int              add_ogg_page(ogg_page *, int header_page,
diff -Naur a/r_ac3.cpp b/r_ac3.cpp
--- a/r_ac3.cpp	2003-11-12 08:48:17.000000000 +0000
+++ b/r_ac3.cpp	2021-04-29 17:46:26.095380205 +0000
@@ -55,7 +55,7 @@
 }
 
 ac3_reader_c::ac3_reader_c(char *fname, audio_sync_t *nasync,
-                           range_t *nrange, char **ncomments) throw (error_c) {
+                           range_t *nrange, char **ncomments) {
   int          pos;
   ac3_header_t ac3header;
   
diff -Naur a/r_ac3.h b/r_ac3.h
--- a/r_ac3.h	2003-05-04 09:41:26.000000000 +0000
+++ b/r_ac3.h	2021-04-29 17:46:25.700376040 +0000
@@ -36,7 +36,7 @@
      
   public:
     ac3_reader_c(char *fname, audio_sync_t *nasync, range_t *nrange,
-                 char **ncomments) throw (error_c);
+                 char **ncomments);
     virtual ~ac3_reader_c();
 
     virtual int              read();
diff -Naur a/r_avi.cpp b/r_avi.cpp
--- a/r_avi.cpp	2004-04-13 17:31:31.000000000 +0000
+++ b/r_avi.cpp	2021-04-29 17:46:25.846377579 +0000
@@ -55,7 +55,7 @@
 avi_reader_c::avi_reader_c(char *fname, unsigned char *astreams,
                            unsigned char *vstreams, audio_sync_t *nasync,
                            range_t *nrange, char **ncomments, char *nfourcc,
-                           char* nav_seek) throw (error_c) {
+                           char* nav_seek) {
   int            fsize, i;
   off_t          size;
   FILE          *f;
diff -Naur a/r_avi.h b/r_avi.h
--- a/r_avi.h	2004-04-01 16:02:52.000000000 +0000
+++ b/r_avi.h	2021-04-29 17:46:26.000379203 +0000
@@ -67,7 +67,7 @@
     avi_reader_c(char *fname, unsigned char *astreams,
                  unsigned char *vstreams, audio_sync_t *nasync,
                  range_t *nrange, char **ncomments, char *nfourcc,
-                 char* nav_seek) throw (error_c);
+                 char* nav_seek);
     virtual ~avi_reader_c();
 
     virtual int              read();
diff -Naur a/r_microdvd.cpp b/r_microdvd.cpp
--- a/r_microdvd.cpp	2003-05-04 09:41:26.000000000 +0000
+++ b/r_microdvd.cpp	2021-04-29 17:46:25.958378760 +0000
@@ -56,7 +56,7 @@
 
 microdvd_reader_c::microdvd_reader_c(char *fname, audio_sync_t *nasync,
                                      range_t *nrange, char **ncomments)
-                                     throw (error_c) {
+                                     {
   if ((file = fopen(fname, "r")) == NULL)
     throw error_c("microdvd_reader: Could not open source file.");
   if (!microdvd_reader_c::probe_file(file, 0))
diff -Naur a/r_microdvd.h b/r_microdvd.h
--- a/r_microdvd.h	2003-05-04 09:41:26.000000000 +0000
+++ b/r_microdvd.h	2021-04-29 17:46:25.741376472 +0000
@@ -35,7 +35,7 @@
      
   public:
     microdvd_reader_c(char *fname, audio_sync_t *nasync, range_t *nrange,
-                      char **ncomments) throw (error_c);
+                      char **ncomments);
     virtual ~microdvd_reader_c();
 
     virtual int              read();
diff -Naur a/r_mp3.cpp b/r_mp3.cpp
--- a/r_mp3.cpp	2004-07-06 15:42:16.000000000 +0000
+++ b/r_mp3.cpp	2021-04-29 17:46:25.907378223 +0000
@@ -60,7 +60,7 @@
 }
 
 mp3_reader_c::mp3_reader_c(char *fname, audio_sync_t *nasync,
-                           range_t *nrange, char **ncomments) throw (error_c) {
+                           range_t *nrange, char **ncomments) {
   int           pos;
   unsigned long header;
   mp3_header_t  mp3header;
diff -Naur a/r_mp3.h b/r_mp3.h
--- a/r_mp3.h	2003-05-04 09:41:26.000000000 +0000
+++ b/r_mp3.h	2021-04-29 17:46:26.036379583 +0000
@@ -40,7 +40,7 @@
      
   public:
     mp3_reader_c(char *fname, audio_sync_t *nasync, range_t *nrange,
-                 char **ncomments) throw (error_c);
+                 char **ncomments);
     virtual ~mp3_reader_c();
 
     virtual int              read();
diff -Naur a/r_ogm.cpp b/r_ogm.cpp
--- a/r_ogm.cpp	2004-02-10 11:17:17.000000000 +0000
+++ b/r_ogm.cpp	2021-04-29 17:46:25.967378855 +0000
@@ -60,7 +60,7 @@
 ogm_reader_c::ogm_reader_c(char *fname, unsigned char *astreams,
                            unsigned char *vstreams, unsigned char *tstreams,
                            audio_sync_t *nasync, range_t *nrange,
-                           char **ncomments, char *nfourcc) throw (error_c) {
+                           char **ncomments, char *nfourcc) {
   off_t size;
   
   if ((file = fopen(fname, "r")) == NULL)
diff -Naur a/r_ogm.h b/r_ogm.h
--- a/r_ogm.h	2004-02-10 11:17:17.000000000 +0000
+++ b/r_ogm.h	2021-04-29 17:46:25.880377938 +0000
@@ -62,7 +62,7 @@
     ogm_reader_c(char *fname, unsigned char *astreams,
                  unsigned char *vstreams, unsigned char *tstreams,
                  audio_sync_t *nasync, range_t *nrange, char **ncomments,
-                 char *nfourcc) throw (error_c);
+                 char *nfourcc);
     virtual ~ogm_reader_c();
 
     virtual int                   read();
diff -Naur a/r_srt.cpp b/r_srt.cpp
--- a/r_srt.cpp	2003-05-04 09:41:26.000000000 +0000
+++ b/r_srt.cpp	2021-04-29 17:46:25.925378412 +0000
@@ -62,7 +62,7 @@
 }
 
 srt_reader_c::srt_reader_c(char *fname, audio_sync_t *nasync,
-                           range_t *nrange, char **ncomments) throw (error_c) {
+                           range_t *nrange, char **ncomments) {
   if ((file = fopen(fname, "r")) == NULL)
     throw error_c("srt_reader: Could not open source file.");
   if (!srt_reader_c::probe_file(file, 0))
diff -Naur a/r_srt.h b/r_srt.h
--- a/r_srt.h	2003-05-04 09:41:26.000000000 +0000
+++ b/r_srt.h	2021-04-29 17:46:25.826377368 +0000
@@ -35,7 +35,7 @@
      
   public:
     srt_reader_c(char *fname, audio_sync_t *nasync, range_t *nrange,
-                 char **ncomments) throw (error_c);
+                 char **ncomments);
     virtual ~srt_reader_c();
 
     virtual int              read();
diff -Naur a/r_vobsub.cpp b/r_vobsub.cpp
--- a/r_vobsub.cpp	2004-04-13 17:31:31.000000000 +0000
+++ b/r_vobsub.cpp	2021-04-29 17:46:25.915378307 +0000
@@ -72,7 +72,7 @@
 
 vobsub_reader_c::vobsub_reader_c(char *fname, audio_sync_t *nasync,
                                  range_t *nrange, char **ncomments)
-                                 throw (error_c) {
+                                 {
   char *name;
   
   if ((file = fopen(fname, "r")) == NULL)
diff -Naur a/r_vobsub.h b/r_vobsub.h
--- a/r_vobsub.h	2003-05-04 09:41:26.000000000 +0000
+++ b/r_vobsub.h	2021-04-29 17:46:25.764376715 +0000
@@ -42,7 +42,7 @@
      
   public:
     vobsub_reader_c(char *fname, audio_sync_t *nasync, range_t *nrange,
-                    char **ncomments) throw (error_c);
+                    char **ncomments);
     virtual ~vobsub_reader_c();
 
     virtual int              read();
diff -Naur a/r_wav.cpp b/r_wav.cpp
--- a/r_wav.cpp	2003-05-04 09:41:26.000000000 +0000
+++ b/r_wav.cpp	2021-04-29 17:46:25.753376599 +0000
@@ -50,7 +50,7 @@
 }
 
 wav_reader_c::wav_reader_c(char *fname, audio_sync_t *nasync,
-                           range_t *nrange, char **ncomments) throw (error_c) {
+                           range_t *nrange, char **ncomments) {
   uint64_t size;
   uint32_t samplerate;
   uint16_t channels, bitdepth;
diff -Naur a/r_wav.h b/r_wav.h
--- a/r_wav.h	2003-05-04 09:41:26.000000000 +0000
+++ b/r_wav.h	2021-04-29 17:46:25.896378107 +0000
@@ -41,7 +41,7 @@
      
   public:
     wav_reader_c(char *fname, audio_sync_t *nasync, range_t *nrange,
-                 char **ncomments) throw (error_c);
+                 char **ncomments);
     virtual ~wav_reader_c();
 
     virtual int              read();
