diff -Naur a/src/app-indicator.c b/src/app-indicator.c
--- a/src/app-indicator.c	2017-10-23 07:03:29.000000000 +0000
+++ b/src/app-indicator.c	2017-10-23 13:08:25.530191393 +0000
@@ -2577,7 +2577,7 @@
 
 	g_return_if_fail(priv->shorties != NULL);
 
-	indicator_desktop_shortcuts_nick_exec_with_context(priv->shorties, nick, NULL);
+//	indicator_desktop_shortcuts_nick_exec_with_context(priv->shorties, nick, NULL);
 
 	return;
 }
