diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2017-10-23 14:05:48.291548000 +0000
+++ b/configure.ac	2017-10-23 15:37:41.434800880 +0000
@@ -104,7 +104,7 @@
 # Check for Mono support
 ###########################
 
-MONO_REQUIRED_VERSION=1.0
+MONO_REQUIRED_VERSION=9999
 PKG_CHECK_MODULES(MONO_DEPENDENCY, mono >= $MONO_REQUIRED_VERSION, has_mono=true, has_mono=false)
 AM_CONDITIONAL(HAS_MONO, [test "x$has_mono" = "xtrue"])
 
@@ -143,7 +143,7 @@
 AC_SUBST(GENERATED_SOURCES)
 
 GLIB_SHARP_REQ_VERSION=2.12.1
-GTK_SHARP_REQ_VERSION=2.12
+GTK_SHARP_REQ_VERSION=9999
 PKG_CHECK_MODULES(GTK_SHARP, gtk-sharp-2.0 >= $GTK_SHARP_REQ_VERSION, has_gtksharp=true, has_gtksharp=false)
 
 PKG_CHECK_MODULES(GAPI, gapi-2.0 >= $GLIB_SHARP_REQ_VERSION, has_gapi=true, has_gapi=false)
