diff -Naur a/src/app-indicator.c b/src/app-indicator.c
--- a/src/app-indicator.c	2017-10-23 07:03:29.000000000 +0000
+++ b/src/app-indicator.c	2017-10-23 13:25:03.332259298 +0000
@@ -1178,7 +1178,8 @@
 		delta = ABS(delta);
 		g_signal_emit(app, signals[SCROLL_EVENT], 0, delta, direction);
 
-	} else if (g_strcmp0(method, "SecondaryActivate") == 0 ||
+	} else if (g_strcmp0(method, "Activate") == 0 ||
+	           g_strcmp0(method, "SecondaryActivate") == 0 ||
 	           g_strcmp0(method, "XAyatanaSecondaryActivate") == 0) {
 		GtkWidget *menuitem = priv->sec_activate_target;
 		
diff -Naur a/src/notification-item.xml b/src/notification-item.xml
--- a/src/notification-item.xml	2017-10-23 07:03:29.000000000 +0000
+++ b/src/notification-item.xml	2017-10-23 13:24:48.252258271 +0000
@@ -20,6 +20,10 @@
 		<property name="XAyatanaOrderingIndex" type="u" access="read" />
 
 <!-- Methods -->
+		<method name="Activate">
+			<arg type="i" name="x" direction="in" />
+			<arg type="i" name="y" direction="in" />
+		</method>
 		<method name="Scroll">
 			<arg type="i" name="delta" direction="in" />
 			<arg type="s" name="orientation" direction="in" />
