diff -Nur qt-x11-opensource-src-4.3.4.orig/src/sql/drivers/mysql/qsql_mysql.h qt-x11-opensource-src-4.3.4/src/sql/drivers/mysql/qsql_mysql.h
--- qt-x11-opensource-src-4.3.4.orig/src/sql/drivers/mysql/qsql_mysql.h	2008-04-02 20:52:11.739343004 -0500
+++ qt-x11-opensource-src-4.3.4/src/sql/drivers/mysql/qsql_mysql.h	2008-04-02 20:52:05.970610574 -0500
@@ -51,7 +51,7 @@
 #include <QtCore/qt_windows.h>
 #endif
 
-#include <mysql.h>
+#include <mysql/mysql.h>
 
 #ifdef QT_PLUGIN
 #define Q_EXPORT_SQLDRIVER_MYSQL
