diff -NrU5 celestia-1.5.1.orig/src/celengine/overlay.h celestia-1.5.1/src/celengine/overlay.h
--- celestia-1.5.1.orig/src/celengine/overlay.h	2009-05-17 15:01:34.000000000 -0600
+++ celestia-1.5.1/src/celengine/overlay.h	2009-05-17 15:02:56.000000000 -0600
@@ -9,10 +9,11 @@
 
 #ifndef _OVERLAY_H_
 #define _OVERLAY_H_
 
 #include <string>
+#include <cstdio>
 #include <iostream>
 #include <celtxf/texturefont.h>
 
 
 class Overlay;
diff -NrU5 celestia-1.5.1.orig/src/celengine/star.cpp celestia-1.5.1/src/celengine/star.cpp
--- celestia-1.5.1.orig/src/celengine/star.cpp	2009-05-17 15:01:34.000000000 -0600
+++ celestia-1.5.1/src/celengine/star.cpp	2009-05-17 15:03:43.000000000 -0600
@@ -7,10 +7,11 @@
 // as published by the Free Software Foundation; either version 2
 // of the License, or (at your option) any later version.
 
 #include <celmath/mathlib.h>
 #include <cstring>
+#include <cstdio>
 #include <cassert>
 #include "celestia.h"
 #include "astro.h"
 #include "orbit.h"
 #include "star.h"
