--- Makefile.orig	2020-03-20 03:13:39.000000000 +1000
+++ Makefile	2020-04-23 18:31:43.521950531 +1000
@@ -14,10 +14,10 @@
 	ln -fns $(pkglibdir)/bin_steamdeps.py $(DESTDIR)$(bindir)/$(PACKAGE)deps
 
 install-docs:
-	install -d -m 755 $(DESTDIR)$(PREFIX)/share/doc/$(PACKAGE)/
-	install -p -m 644 README steam_subscriber_agreement.txt $(DESTDIR)$(PREFIX)/share/doc/$(PACKAGE)/
-	install -d -m 755 $(DESTDIR)$(PREFIX)/share/man/man6/
-	install -m 644 $(PACKAGE).6 $(DESTDIR)$(PREFIX)/share/man/man6/
+	install -d -m 755 $(DESTDIR)$(PREFIX)/doc/$(PACKAGE)-$(VERSION)/
+	install -p -m 644 README steam_subscriber_agreement.txt $(DESTDIR)$(PREFIX)/doc/$(PACKAGE)-$(VERSION)/
+	install -d -m 755 $(DESTDIR)$(PREFIX)/man/man6/
+	install -m 644 $(PACKAGE).6 $(DESTDIR)$(PREFIX)/man/man6/
 
 install-icons:
 	install -d -m 755 $(DESTDIR)$(PREFIX)/share/icons/hicolor/16x16/apps/
@@ -39,7 +39,7 @@
 	install -p -m 644 bootstraplinux_ubuntu12_32.tar.xz $(DESTDIR)$(PREFIX)/lib/$(PACKAGE)/
 
 install-desktop:
-	install -d -m 755 $(DESTDIR)$(PREFIX)/share/applications/
+	install -d -m 755 $(DESTDIR)$(datadir)/applications/
 	install -d -m 755 $(DESTDIR)$(pkglibdir)/
 	install -p -m 644 $(PACKAGE).desktop $(DESTDIR)$(pkglibdir)
 	ln -fns $(pkglibdir)/$(PACKAGE).desktop $(DESTDIR)$(datadir)/applications/
