--- a/aris.c	2021-02-10 17:51:33.532759576 +0400
+++ b/aris.c	2021-02-10 17:52:15.210759108 +0400
@@ -845,7 +845,7 @@
       if (the_app->working_dir)
 	{
 	  parent = g_file_get_parent (parent);
-	  sprintf (the_app->help_file, "file://%s/doc/aris/index.html", g_file_get_path (parent));
+	  sprintf (the_app->help_file, "file:///usr/doc/aris-2.2/html/index.html", g_file_get_path (parent));
 	}
 
       int ret;
