--- aris-2.0/configure	2012-09-25 05:06:22.000000000 -0300
+++ aris-2.0b/configure	2013-03-23 18:34:59.483719378 -0300
@@ -4610,3 +4610,1 @@
-cat >>confdefs.h <<_ACEOF
-#define HELP_FILE "file://`pwd`/doc/aris/index.html"
-_ACEOF
+echo "#define HELP_FILE \"file://"$htmldir"/index.html\"" >>confdefs.h
--- aris-2.0/src/aris.c	2012-09-25 05:06:22.000000000 -0300
+++ aris-2.0b/src/aris.c	2013-03-23 18:45:25.290749474 -0300
@@ -1064,0 +1065,6 @@
+#ifdef HELP_FILE /* Which should be defined in config.h */
+
+      strcpy (the_app->help_file, HELP_FILE);
+
+#else
+
@@ -1076,0 +1083,2 @@
+
+#endif /* HELP FILE */
