diff -Naur merkaartor-0.17.2.orig/3rdparty/qtsingleapplication-2.6_1-opensource/src/qtlocalpeer.cpp merkaartor-0.17.2/3rdparty/qtsingleapplication-2.6_1-opensource/src/qtlocalpeer.cpp
--- merkaartor-0.17.2.orig/3rdparty/qtsingleapplication-2.6_1-opensource/src/qtlocalpeer.cpp	2011-02-10 13:00:35.000000000 +0100
+++ merkaartor-0.17.2/3rdparty/qtsingleapplication-2.6_1-opensource/src/qtlocalpeer.cpp	2012-09-19 14:26:39.538277883 +0200
@@ -57,6 +57,7 @@
 #endif
 #if defined(Q_OS_UNIX)
 #include <time.h>
+#include <unistd.h>
 #endif
 
 namespace QtLP_Private {
