--- xemacs-21.4.15/Makefile.in.in.orig	2004-07-09 19:00:27.000000000 -0400
+++ xemacs-21.4.15/Makefile.in.in	2004-07-09 19:00:46.000000000 -0400
@@ -403,7 +403,7 @@
 ## all.  --cet
 
 check-features: all
-	${blddir}/src/${PROGNAME} -batch -l check-features.el
+	${blddir}/src/${PROGNAME} -batch -l ${blddir}/lisp/check-features.el
 
 install-only: ${MAKE_SUBDIR} check-features install-arch-dep install-arch-indep
 
