--- xemacs-21.5.34/lisp/find-paths.el.old	2016-01-04 18:49:46.227528964 +0700
+++ xemacs-21.5.34/lisp/find-paths.el	2016-01-04 18:50:32.334978047 +0700
@@ -226,7 +226,7 @@
 				  roots
 				  (file-name-as-directory
 				   (paths-construct-path (list
-							  (if arch-dependent-p "lib" "share")
+							  (if arch-dependent-p "lib64" "share")
 							  emacs-program-name)))
 				  bases
 				  envvar default))
@@ -276,7 +276,7 @@
 				  roots
 				  (file-name-as-directory
 				   (paths-construct-path
-				    (list (if arch-dependent-p "lib" "share")
+				    (list (if arch-dependent-p "lib64" "share")
 					  (construct-emacs-version-name))))
 				  bases
 				  envvar default))
