--- xemacs-21.4.22/lisp/find-paths.el.orig	2011-08-05 13:42:44.545564371 -0400
+++ xemacs-21.4.22/lisp/find-paths.el	2011-08-05 13:43:10.205459415 -0400
@@ -103,7 +103,7 @@
   (or
    ;; installed
    (paths-file-readable-directory-p (paths-construct-path (list directory
-								"lib"
+								"lib64"
 								emacs-program-name)))
    ;; in-place or windows-nt
    (and
@@ -216,7 +216,7 @@
   (paths-find-emacs-directory roots
 			      (file-name-as-directory
 			       (paths-construct-path (list
-						      "lib"
+						      "lib64"
 						      emacs-program-name)))
 			      base
 			      envvar default
@@ -230,7 +230,7 @@
   (paths-find-emacs-directory roots
 			      (file-name-as-directory
 			       (paths-construct-path
-				(list "lib"
+				(list "lib64"
 				      (construct-emacs-version-name))))
 			      base
 			      envvar default
