diff --git a/OSBindings/Qt/clksignal.pro b/OSBindings/Qt/clksignal.pro
index cb1e865b2..e26e3e77b 100644
--- a/OSBindings/Qt/clksignal.pro
+++ b/OSBindings/Qt/clksignal.pro
@@ -113,8 +113,6 @@ SOURCES += \
 	$$SRC/Processors/6502/Implementation/*.cpp \
 	$$SRC/Processors/6502/State/*.cpp \
 	$$SRC/Processors/65816/Implementation/*.cpp \
-	$$SRC/Processors/68000/Implementation/*.cpp \
-	$$SRC/Processors/68000/State/*.cpp \
 	$$SRC/Processors/Z80/Implementation/*.cpp \
 	$$SRC/Processors/Z80/State/*.cpp \
 \
@@ -259,7 +257,6 @@ HEADERS += \
 	$$SRC/Processors/65816/Implementation/*.hpp \
 	$$SRC/Processors/68000/*.hpp \
 	$$SRC/Processors/68000/Implementation/*.hpp \
-	$$SRC/Processors/68000/State/*.hpp \
 	$$SRC/Processors/Z80/*.hpp \
 	$$SRC/Processors/Z80/Implementation/*.hpp \
 	$$SRC/Processors/Z80/State/*.hpp \
diff --git a/OSBindings/SDL/SConstruct b/OSBindings/SDL/SConstruct
index 6b1e6e849..af2dd675c 100644
--- a/OSBindings/SDL/SConstruct
+++ b/OSBindings/SDL/SConstruct
@@ -100,8 +100,6 @@ SOURCES += glob.glob('../../Outputs/OpenGL/Primitives/*.cpp')
 SOURCES += glob.glob('../../Processors/6502/Implementation/*.cpp')
 SOURCES += glob.glob('../../Processors/6502/State/*.cpp')
 SOURCES += glob.glob('../../Processors/65816/Implementation/*.cpp')
-SOURCES += glob.glob('../../Processors/68000/Implementation/*.cpp')
-SOURCES += glob.glob('../../Processors/68000/State/*.cpp')
 SOURCES += glob.glob('../../Processors/Z80/Implementation/*.cpp')
 SOURCES += glob.glob('../../Processors/Z80/State/*.cpp')
 
