--- fbpanel-7.2/panel/misc.h	2020-06-12 00:00:00.000000000 +0000
+++ fbpanel-7.2/panel/misc.h	2026-07-18 00:00:00.000000000 +0000
@@ -48,1 +48,1 @@
-void configure();
+void configure(xconf *xc);
--- fbpanel-7.2/plugins/battery/power_supply.h	2020-06-12 00:00:00.000000000 +0000
+++ fbpanel-7.2/plugins/battery/power_supply.h	2026-07-18 00:00:00.000000000 +0000
@@ -41,1 +41,1 @@
-power_supply* power_supply_parse();
+power_supply* power_supply_parse(power_supply* ps);
