diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2019-12-16 21:07:03.000000000 +0000
+++ b/CMakeLists.txt	2019-12-18 17:18:01.270919658 +0000
@@ -99,9 +99,9 @@
 # The use of the word ZLIB_ROOT should still work prior to "3.12.0",
 # just it's been generalized for all packages now. Just set the policy
 # to new, so we use it, and it will be used prior to 3.12 anyway.
-if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.12")
-    cmake_policy(SET CMP0074 NEW)
-endif()
+#if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.12")
+#    cmake_policy(SET CMP0074 NEW)
+#endif()
 
 # OPTIONS
 # -------
