--- config.mk	2014-09-13 18:36:13.478755045 +0300
+++ a/config.mk	2014-09-13 18:36:18.552755303 +0300
@@ -1,9 +1,10 @@
 # dmenu2 version
-VERSION = 0.1
+VERSION = 0.2
 
 # paths
 PREFIX = /usr/local
 MANPREFIX = ${PREFIX}/share/man
+LIBDIR = /usr/local/lib
 
 X11INC = /usr/X11R6/include
 X11LIB = /usr/X11R6/lib
@@ -18,12 +19,13 @@
 
 # includes and libs
 INCS = -I${X11INC} ${XFTINC}
-LIBS = -L${X11LIB} -lX11 ${XINERAMALIBS} ${XFTLIBS}
+LIBS = -L${LIBDIR} -L${X11LIB} -lX11 ${XINERAMALIBS} ${XFTLIBS}
 
 # flags
+OPTS ?= 0s
 CPPFLAGS = -D_BSD_SOURCE -D_POSIX_C_SOURCE=2 -DVERSION=\"${VERSION}\" ${XINERAMAFLAGS}
 #CFLAGS   = -g -std=c99 -pedantic -Wall -O0 ${INCS} ${CPPFLAGS}
-CFLAGS   = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
+CFLAGS   = -std=c99 -pedantic -Wall ${OPTS} ${INCS} ${CPPFLAGS}
 LDFLAGS  = -s ${LIBS}
 
 # compiler and linker
