diff -Naur gambas-3.19.0.orig/gb.pdf/src/CPdfDocument.cpp gambas-3.19.0/gb.pdf/src/CPdfDocument.cpp
--- gambas-3.19.0.orig/gb.pdf/src/CPdfDocument.cpp	2024-02-16 00:33:51.000000000 +0100
+++ gambas-3.19.0/gb.pdf/src/CPdfDocument.cpp	2024-02-17 12:14:47.937408267 +0100
@@ -776,7 +776,8 @@
 BEGIN_PROPERTY(PDFINDEX_title)
 
 	OutlineItem *item = CPDF_index_get(THIS->currindex);
-	return_unicode_string(item->getTitle(), item->getTitleLength());
+	const std::vector<Unicode> &title = item->getTitle();
+	return_unicode_string(title.data(), title.size());
 
 END_PROPERTY
 
