diff --git a/bsnes/Makefile b/bsnes/Makefile
index 4af8708..2e95a38 100755
--- a/bsnes/Makefile
+++ b/bsnes/Makefile
@@ -10,7 +10,7 @@ options :=
 # compiler
 c       := $(compiler) -std=gnu99
 cpp     := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags   := -O3 -fomit-frame-pointer -I.
+flags   := $(CXXFLAGS) -fomit-frame-pointer -I.
 link    :=
 objects := libco
 
@@ -25,7 +25,7 @@ flags := $(flags) $(foreach o,$(call strupper,$(options)),-D$o)
 
 # platform
 ifeq ($(platform),x)
-  link += -s -ldl -lX11 -lXext
+  link += -ldl -lX11 -lXext
 else ifeq ($(platform),osx)
 else ifeq ($(platform),win)
   link += -mwindows
diff --git a/bsnes/ui/Makefile b/bsnes/ui/Makefile
index 7cf3420..dd3f89b 100755
--- a/bsnes/ui/Makefile
+++ b/bsnes/ui/Makefile
@@ -94,11 +94,8 @@ install:
 ifeq ($(platform),x)
 	install -D -m 755 out/bsnes $(DESTDIR)$(prefix)/bin/bsnes
 endif
-	install -D -m 644 data/bsnes.png $(DESTDIR)$(prefix)/share/pixmaps/bsnes.png
-	install -D -m 644 data/bsnes.desktop $(DESTDIR)$(prefix)/share/applications/bsnes.desktop
-	mkdir -p ~/.config/bsnes
-	cp data/cheats.xml ~/.config/bsnes/cheats.xml
-	chmod 777 ~/.config/bsnes ~/.config/bsnes/cheats.xml
+	install -D -m 644 data/bsnes.png $(DESTDIR)/usr/share/pixmaps/bsnes.png
+	install -D -m 644 data/bsnes.desktop $(DESTDIR)/usr/share/applications/bsnes.desktop
 
 uninstall:
 ifeq ($(platform),x)
