diff -crB bsnes_v091-source/bsnes/target-ethos/Makefile bsnes_v091-source-hh/bsnes/target-ethos/Makefile
*** bsnes_v091-source/bsnes/target-ethos/Makefile	2012-08-09 19:32:19.000000000 -0400
--- bsnes_v091-source-hh/bsnes/target-ethos/Makefile	2012-11-01 22:06:29.000000000 -0400
***************
*** 16,22 ****
  # platform
  ifeq ($(platform),x)
    ruby := video.glx video.xv video.sdl
!   ruby += audio.alsa audio.openal audio.oss audio.pulseaudio audio.pulseaudiosimple audio.ao
    ruby += input.sdl input.x
  else ifeq ($(platform),osx)
    ruby :=
--- 16,22 ----
  # platform
  ifeq ($(platform),x)
    ruby := video.glx video.xv video.sdl
!   ruby += audio.alsa audio.openal audio.oss audio.ao
    ruby += input.sdl input.x
  else ifeq ($(platform),osx)
    ruby :=
***************
*** 76,93 ****
  	sourcery $(ui)/resource/resource.xml $(ui)/resource/resource.cpp $(ui)/resource/resource.hpp
  
  install:
! ifeq ($(USER),root)
! 	@echo Please do not run make install as root.
! 	@echo The installer needs to know your home directory to install important files.
! else ifeq ($(platform),x)
! 	sudo install -D -m 755 out/$(name) $(DESTDIR)$(prefix)/bin/$(name)
! 	sudo install -D -m 644 data/$(name).png $(DESTDIR)$(prefix)/share/pixmaps/$(name).png
! 	sudo install -D -m 644 data/$(name).desktop $(DESTDIR)$(prefix)/share/applications/$(name).desktop
! 
! 	mkdir -p ~/.config/$(name)
! 	cp -R profile/* ~/.config/$(name)
! 	cp data/cheats.xml ~/.config/$(name)/cheats.xml
! 	chmod -R 777 ~/.config/$(name)
  endif
  
  uninstall:
--- 76,102 ----
  	sourcery $(ui)/resource/resource.xml $(ui)/resource/resource.cpp $(ui)/resource/resource.hpp
  
  install:
! # edit: hackedhead 2012-11-01: process installation as root for slackware, drop profile into /usr/share
! # 	we'll add a note in the readme about where to get the config files
! #ifeq ($(USER),root)
! #	@echo Please do not run make install as root.
! #	@echo The installer needs to know your home directory to install important files.
! #else ifeq ($(platform),x)
! ifeq ($(platform),x)
! 	install -D -m 755 out/$(name) $(DESTDIR)$(prefix)/bin/$(name)
! 	install -D -m 644 data/$(name).png $(DESTDIR)$(prefix)/share/pixmaps/$(name).png
! 	install -D -m 644 data/$(name).desktop $(DESTDIR)$(prefix)/share/applications/$(name).desktop
! 
! #	mkdir -p ~/.config/$(name)
! #	cp -R profile/* ~/.config/$(name)
! #	cp data/cheats.xml ~/.config/$(name)/cheats.xml
! #	chmod -R 777 ~/.config/$(name)
! 
! 	install -d -m 755 $(DESTDIR)$(prefix)/share/$(name)/profile
! 	cp -R profile/* $(DESTDIR)$(prefix)/share/$(name)/profile
! 	install -m 644 data/cheats.xml $(DESTDIR)$(prefix)/share/$(name)/profile/cheats.xml
! 	chmod -R 644 $(DESTDIR)$(prefix)/share/$(name)/profile/*/*
! 
  endif
  
  uninstall:
diff -crB bsnes_v091-source/purify/Makefile bsnes_v091-source-hh/purify/Makefile
*** bsnes_v091-source/purify/Makefile	2012-08-06 17:54:40.000000000 -0400
--- bsnes_v091-source-hh/purify/Makefile	2012-11-01 21:56:33.000000000 -0400
***************
*** 4,10 ****
  application := purify
  resource :=
  flags := -std=gnu++0x -I. -O3 -fomit-frame-pointer
! link := -s
  
  ifeq ($(platform),win)
    resource := resource.o
--- 4,10 ----
  application := purify
  resource :=
  flags := -std=gnu++0x -I. -O3 -fomit-frame-pointer
! link := -s -lX11
  
  ifeq ($(platform),win)
    resource := resource.o
