diff --git a/bsnes/ui/Makefile b/bsnes/ui/Makefile
index d5f72eb..604a0f7 100755
--- a/bsnes/ui/Makefile
+++ b/bsnes/ui/Makefile
@@ -7,11 +7,13 @@ ui_objects += $(if $(call streq,$(platform),win),resource)
 
 # platform
 ifeq ($(platform),x)
-# phoenix_compile = $(call compile,-DPHOENIX_GTK `pkg-config --cflags gtk+-2.0`)
-# link += `pkg-config --libs gtk+-2.0`
-
-  phoenix_compile = $(call compile,-DPHOENIX_QT `pkg-config --cflags QtCore QtGui`)
-  link += `pkg-config --libs QtCore QtGui`
+  ifeq ($(toolkit),gtk)
+    phoenix_compile = $(call compile,-DPHOENIX_GTK `pkg-config --cflags gtk+-2.0`)
+    link += `pkg-config --libs gtk+-2.0`
+  else ifeq ($(toolkit),qt)
+    phoenix_compile = $(call compile,-DPHOENIX_QT `pkg-config --cflags QtCore QtGui`)
+    link += `pkg-config --libs QtCore QtGui`
+  endif
 
   ruby := video.glx video.xv video.sdl
   ruby += audio.alsa audio.openal audio.oss audio.pulseaudio audio.pulseaudiosimple audio.ao
