diff -Naur Par-1.53.0/par.c Par-1.53.0.patched/par.c
--- Par-1.53.0/par.c	2020-03-14 21:01:03.000000000 -0400
+++ Par-1.53.0.patched/par.c	2020-06-22 19:46:25.353559989 -0400
@@ -762,6 +762,7 @@
   if (*errmsg) goto parcleanup;
 
   env = getenv("PARINIT");
+  if (!env) env = "rTbgqR B=.,?'_A_a_@ Q=_s>|";
   if (env) {
     parinit = malloc((strlen(env) + 1) * sizeof (char));
     if (!parinit) {
