--- pcgen	2020-11-14 22:10:16.000000000 +1300
+++ pcgen-slackware	2022-03-06 19:54:07.786181861 +1300
@@ -19,13 +19,13 @@
         PRG=`dirname "$PRG"`"/$link"
     fi
 done
-SAVED="`pwd`"
+SAVED="/opt/pcgen"
 cd "`dirname \"$PRG\"`/" >/dev/null
-APP_HOME="`pwd -P`"
+APP_HOME="/opt/pcgen"
 cd "$SAVED" >/dev/null
 
 APP_NAME="pcgen"
-APP_BASE_NAME=`basename "$0"`
+APP_BASE_NAME=pcgen
 
 # Add default JVM options here. You can also use JAVA_OPTS and PCGEN_OPTS to pass JVM options to this script.
 DEFAULT_JVM_OPTS='"-Dsun.java2d.d3d=false"'
