commit bfc4a6004d0d3984c8066289b8d8e563640c4ddd
Author: Gregg Townsend <gmt@cs.arizona.edu>
Date:   Mon Oct 29 10:07:31 2018 -0700

    Adapt to change in GLIBC v2.28 internals (thx Cheyenne Wills).

diff --git a/ipl/cfuncs/fpoll.c b/ipl/cfuncs/fpoll.c
--- a/ipl/cfuncs/fpoll.c
+++ b/ipl/cfuncs/fpoll.c
@@ -60,12 +64,9 @@
 
    /* check for data already in buffer */
    /* there's no legal way to do this in C; we cheat */
-#if defined(__GLIBC__) && defined(_STDIO_USES_IOSTREAM)	/* new GCC library */
+#if defined(__GLIBC__) 					/* new GCC library */
    if (f->_IO_read_ptr < f->_IO_read_end)
       RetArg(1);
-#elif defined(__GLIBC__)				/* old GCC library */
-   if (f->__bufp < f->__get_limit)
-      RetArg(1);
 #elif defined(_FSTDIO)					/* new BSD library */
    if (f->_r > 0)
       RetArg(1);
