commit 8cb4887b886ad0c9caf0be595e236d2a74a41a8d
Author: Gregg Townsend <gmt@cs.arizona.edu>
Date:   Mon Aug 28 09:10:00 2017 -0700

    Fix compilation warnings in xpm code (thx Sean Jensen).

diff --git a/src/xpm/create.c b/src/xpm/create.c
--- a/src/xpm/create.c
+++ b/src/xpm/create.c
@@ -344,6 +344,7 @@
 
 #endif
 
+int
 xpmCreateImage(display, attrib, image_return, shapeimage_return, attributes)
     Display *display;
     xpmInternAttrib *attrib;
diff --git a/src/xpm/data.c b/src/xpm/data.c
--- a/src/xpm/data.c
+++ b/src/xpm/data.c
@@ -110,6 +110,7 @@
 /*
  * skip to the end of the current string and the beginning of the next one
  */
+void
 xpmNextString(mdata)
     xpmData *mdata;
 {
@@ -277,6 +278,7 @@
 /*
  * get the current comment line
  */
+void
 xpmGetCmt(mdata, cmt)
     xpmData *mdata;
     char **cmt;
@@ -403,6 +405,7 @@
 /*
  * close the file related to the xpmData if any
  */
+void
 XpmDataClose(mdata)
     xpmData *mdata;
 {
diff --git a/src/xpm/misc.c b/src/xpm/misc.c
--- a/src/xpm/misc.c
+++ b/src/xpm/misc.c
@@ -14,6 +14,7 @@
  * Free the computed color table
  */
 
+void
 xpmFreeColorTable(colorTable, ncolors)
     char ***colorTable;
     int ncolors;
@@ -39,6 +40,7 @@
  * which ones must be freed later on.
  */
 
+void
 xpmInitInternAttrib(attrib)
     xpmInternAttrib *attrib;
 {
@@ -55,6 +57,7 @@
  * Free the xpmInternAttrib pointers which have been allocated
  */
 
+void
 xpmFreeInternAttrib(attrib)
     xpmInternAttrib *attrib;
 {
@@ -80,6 +83,7 @@
 /*
  * Free array of extensions
  */
+void
 XpmFreeExtensions(extensions, nextensions)
     XpmExtension *extensions;
     int          nextensions;
@@ -108,6 +112,7 @@
  * Return the XpmAttributes structure size
  */
 
+int
 XpmAttributesSize()
 {
     return sizeof(XpmAttributes);
@@ -119,6 +124,7 @@
  * but the structure itself
  */
 
+void
 XpmFreeAttributes(attributes)
     XpmAttributes *attributes;
 {
@@ -167,6 +173,7 @@
  * the xpmInternAttrib structure.
  */
 
+void
 xpmSetAttributes(attrib, attributes)
     xpmInternAttrib *attrib;
     XpmAttributes *attributes;
diff --git a/src/xpm/xpm.h b/src/xpm/xpm.h
--- a/src/xpm/xpm.h
+++ b/src/xpm/xpm.h
@@ -191,8 +191,8 @@
 				      XpmAttributes * attributes));
 
     FUNC(XpmAttributesSize, int, ());
-    FUNC(XpmFreeAttributes, int, (XpmAttributes * attributes));
-    FUNC(XpmFreeExtensions, int, (XpmExtension * extensions, int nextensions));
+    FUNC(XpmFreeAttributes, void, (XpmAttributes * attributes));
+    FUNC(XpmFreeExtensions, void, (XpmExtension * extensions, int nextensions));
 
 #ifdef __cplusplus
 }					/* for C++ V2.0 */
diff --git a/src/xpm/xpmP.h b/src/xpm/xpmP.h
--- a/src/xpm/xpmP.h
+++ b/src/xpm/xpmP.h
@@ -159,13 +159,13 @@
 			 XpmAttributes * attributes,
 			 xpmInternAttrib * attrib));
 
-FUNC(xpmFreeColorTable, int, (char ***colorTable, int ncolors));
+FUNC(xpmFreeColorTable, void, (char ***colorTable, int ncolors));
 
-FUNC(xpmInitInternAttrib, int, (xpmInternAttrib * xmpdata));
+FUNC(xpmInitInternAttrib, void, (xpmInternAttrib * xmpdata));
 
-FUNC(xpmFreeInternAttrib, int, (xpmInternAttrib * xmpdata));
+FUNC(xpmFreeInternAttrib, void, (xpmInternAttrib * xmpdata));
 
-FUNC(xpmSetAttributes, int, (xpmInternAttrib * attrib,
+FUNC(xpmSetAttributes, void, (xpmInternAttrib * attrib,
 			     XpmAttributes * attributes));
 
 FUNC(xpmGetAttributes, int, (XpmAttributes * attributes,
@@ -173,18 +173,20 @@
 
 /* I/O utility */
 
-FUNC(xpmNextString, int, (xpmData * mdata));
+FUNC(atoui, unsigned int, (char *p, unsigned int l, unsigned int *ui_return));
+FUNC(xpmGetString, int, (xpmData *mdata, char **sptr, unsigned int *l));
+FUNC(xpmNextString, void, (xpmData * mdata));
 FUNC(xpmNextUI, int, (xpmData * mdata, unsigned int *ui_return));
 
 #define xpmGetC(mdata) \
 	(mdata->type ? (getc(mdata->stream.file)) : (*mdata->cptr++))
 
 FUNC(xpmNextWord, unsigned int, (xpmData * mdata, char *buf));
-FUNC(xpmGetCmt, int, (xpmData * mdata, char **cmt));
+FUNC(xpmGetCmt, void, (xpmData * mdata, char **cmt));
 FUNC(xpmReadFile, int, (char *filename, xpmData * mdata));
 FUNC(xpmWriteFile, int, (char *filename, xpmData * mdata));
 FUNC(xpmOpenArray, void, (char **data, xpmData * mdata));
-FUNC(XpmDataClose, int, (xpmData * mdata));
+FUNC(XpmDataClose, void, (xpmData * mdata));
 
 /* RGB utility */
 
