commit bb7c6b5232d8cf0ad98e5ab8e8e3afd3a249751f
Author: Gregg Townsend <gmt@cs.arizona.edu>
Date:   Mon Aug 28 08:08:32 2017 -0700

    Fix ancient potential lexer bug (thx Sean Jensen).

diff --git a/src/common/yylex.h b/src/common/yylex.h
--- a/src/common/yylex.h
+++ b/src/common/yylex.h
@@ -263,7 +263,7 @@
    {
    register char *s1;
    s1 = lex_sbuf.strtimage;
-   while (s != '\0' && s1 < lex_sbuf.endimage && *s == *s1) {
+   while (*s != '\0' && s1 < lex_sbuf.endimage && *s == *s1) {
       ++s;
       ++s1;
       }
