commit e91f59f3affe69043586cb171ad1f8027e31d930
Author: Gregg Townsend <gmt@cs.arizona.edu>
Date:   Sat Mar 11 08:41:26 2017 -0700

    Fix innocuous MaxType definition bug (thx Don Ward).

diff --git a/src/h/rmacros.h b/src/h/rmacros.h
--- a/src/h/rmacros.h
+++ b/src/h/rmacros.h
@@ -301,7 +301,7 @@
 #define T_Kywdstr	24	/* string keyword */
 #define T_Kywdevent	25	/* keyword &eventsource, etc. */
 
-#define MaxType		26	/* maximum type number */
+#define MaxType		25	/* maximum type number */
 
 /*
  * Definitions for keywords.
