diff -Naur mupen64plus-bundle-src-2.5.orig/source/mupen64plus-core/src/main/workqueue.h mupen64plus-bundle-src-2.5/source/mupen64plus-core/src/main/workqueue.h
--- mupen64plus-bundle-src-2.5.orig/source/mupen64plus-core/src/main/workqueue.h	2015-04-27 00:42:49.000000000 +0200
+++ mupen64plus-bundle-src-2.5/source/mupen64plus-core/src/main/workqueue.h	2020-05-08 19:04:37.956818550 +0200
@@ -27,7 +27,6 @@
 
 struct work_struct;
 
-struct work_struct *work;
 typedef void (*work_func_t)(struct work_struct *work);
 struct work_struct {
     work_func_t func;
