diff -up ./glide64/Combine.cpp.00 ./glide64/Combine.cpp
--- ./glide64/Combine.cpp.00	2010-01-28 12:08:11.367269059 -0800
+++ ./glide64/Combine.cpp	2010-01-28 12:08:55.142019819 -0800
@@ -11873,7 +11873,7 @@ void InitCombine ()
   LOG ("InitCombine() ");
   memset(&cmb, 0, sizeof(cmb));
   const char *extensions = grGetString (GR_EXTENSION);
-  if (char * extstr = strstr(extensions, "COMBINE")) {
+  if (char * extstr = (char *) strstr(extensions, "COMBINE")) { 
     if (!strncmp(extstr, "COMBINE", 7)) {
       LOG ("extensions ");
       cmb.grColorCombineExt = (GRCOLORCOMBINEEXT) grGetProcAddress("grColorCombineExt");
diff -up ./glide64/Main.cpp.00 ./glide64/Main.cpp
--- ./glide64/Main.cpp.00	2010-01-28 11:55:52.981269116 -0800
+++ ./glide64/Main.cpp	2010-01-28 11:59:22.810019317 -0800
@@ -784,7 +784,7 @@ BOOL InitGfx (BOOL evoodoo_using_window)
   printf("bebefore2\n");
   if (settings.fb_hires)
   {
-    if (char * extstr = strstr(extensions, "TEXTUREBUFFER")) 
+    if (char * extstr = (char *) strstr(extensions, "TEXTUREBUFFER"))  
     {
       if (!strncmp(extstr, "TEXTUREBUFFER", 13)) 
       {
