diff -Nur rkhunter-1.3.4.orig/installer.sh rkhunter-1.3.4/installer.sh
--- rkhunter-1.3.4.orig/installer.sh	2008-12-30 15:23:00.000000000 -0600
+++ rkhunter-1.3.4/installer.sh	2009-07-14 00:36:40.755149686 -0500
@@ -171,7 +171,7 @@
 				esac
 				;;
 			RPM)	if [ -n "${RPM_BUILD_ROOT}" ]; then
-					PREFIX="${RPM_BUILD_ROOT}/usr/local"
+					PREFIX="${RPM_BUILD_ROOT}/usr"
 				else
 					echo "RPM prefix chosen but \$RPM_BUILD_ROOT variable not found, exiting."
 					exit 1
@@ -227,7 +227,7 @@
 					LIBDIR="${PREFIX}/lib"
 				fi
 				VARDIR="${RPM_BUILD_ROOT}/var"
-				SHAREDIR="${PREFIX}/share"; BINDIR="${PREFIX}/bin"
+				SHAREDIR="${PREFIX}"; BINDIR="${PREFIX}/bin"
 				;;
 			DEB)
 				LIBDIR="${PREFIX}/lib"
