All the patches are taken from Debian repo

--- tclvfs-1.3-20080503.orig/configure
+++ tclvfs-1.3-20080503/configure
@@ -7937,12 +7937,15 @@
 	    SHLIB_LD_LIBS='${LIBS}'
 	    SHLIB_SUFFIX=".so"
 
+	    # following line added by CW for Debian GNU/Hurd
+	    TCL_SHLIB_LD_EXTRAS="-Wl,-soname,\${TCL_LIB_FILE}.0"
+
 	    SHLIB_LD="${CC} -shared"
-	    DL_OBJS=""
+	    DL_OBJS="tclLoadDl.o"
 	    DL_LIBS="-ldl"
 	    LDFLAGS="$LDFLAGS -Wl,--export-dynamic"
 	    CC_SEARCH_FLAGS=""
-	    LD_SEARCH_FLAGS=""
+	    LD_SEARCH_FLAGS="${CC_SEARCH_FLAGS}"
 	    if test "`uname -m`" = "alpha" ; then
 		CFLAGS="$CFLAGS -mieee"
 	    fi
@@ -9803,7 +9806,6 @@
 # (actually we leave an empty line to preserve line numbers).
 if test "x$srcdir" = x.; then
   ac_vpsub='/^[	 ]*VPATH[	 ]*=/{
-s/:*\$(srcdir):*/:/;
 s/:*\${srcdir}:*/:/;
 s/:*@srcdir@:*/:/;
 s/^\([^=]*=[	 ]*\):*/\1/;
