
--- tclvfs-1.3-20080503.orig/tclconfig/tcl.m4
+++ tclvfs-1.3-20080503/tclconfig/tcl.m4
@@ -1486,12 +1486,15 @@
 	    SHLIB_LD_LIBS='${LIBS}'
 	    SHLIB_SUFFIX=".so"
 
+	    # following line added by CW for Debian GNU/Hurd
+	    TCL_SHLIB_LD_EXTRAS="-Wl,-soname,\${TCL_LIB_FILE}.0"
+
 	    SHLIB_LD="${CC} -shared"
-	    DL_OBJS=""
+	    DL_OBJS="tclLoadDl.o"
 	    DL_LIBS="-ldl"
 	    LDFLAGS="$LDFLAGS -Wl,--export-dynamic"
 	    CC_SEARCH_FLAGS=""
-	    LD_SEARCH_FLAGS=""
+	    LD_SEARCH_FLAGS="${CC_SEARCH_FLAGS}"
 	    if test "`uname -m`" = "alpha" ; then
 		CFLAGS="$CFLAGS -mieee"
 	    fi
