--- tclvfs-1.3-20080503.orig/doc/vfs.n
+++ tclvfs-1.3-20080503/doc/vfs.n
@@ -138,7 +138,7 @@
 \fImount.zip\fR is a zip archive which has been mounted (on top of
 itself) and contains \fIxxx/yyy\fR, and the current working directory
 is inside \fIxxx\fR, and we evaluate a command like \fIfile exists
-yyy\fR, then \fIroot\R will be \fIC:/foo/bar/mount.zip\fR,
+yyy\fR, then \fIroot\fR will be \fIC:/foo/bar/mount.zip\fR,
 \fIrelative\fR will be \fIxxx/yyy\fR, and \fIactualpath\fR will be
 \fIyyy\fR. The file separator between the \fIroot\fR and \fIrelative\fR
 is omitted.
@@ -218,7 +218,7 @@
 Delete the given directory.  \fIrecursive\fR is either 0 or 1. If
 it is 1 then even if the directory is non-empty, an attempt should
 be made to recursively delete it and its contents.  If it is 0 and
-the directory is non-empty, a posix error (EEXIST) should be
+the directory is non-empty, a posix error (ENOTEMPTY) should be
 thrown.
 .TP
 \fIcommand\fR \fIstat\fR \fIr-r-a\fR
--- tclvfs-1.3-20080503.orig/doc/vfslib.n
+++ tclvfs-1.3-20080503/doc/vfslib.n
@@ -58,8 +58,8 @@
 .PP
 The current supported types are ftp, tar, http, zip, mk4, ns, webdav.  In
 addition there is the ability to mount any 'urltype' as a new volume,
-provided an appropriate vfs is supported.  This means that you can treat
-'ftp://', 'http://' and 'file://' urls as files.  To do this, simply
+provided an appropriate vfs is supported.  This means that you can
+treat 'ftp://', 'http://' and 'file://' urls as files.  To do this, simply
 evaluate the command
 .PP
 \fIvfs::urltype::Mount ftp\fR
