--- tclvfs-1.3-20080503.orig/generic/vfs.c	2006/08/30 19:38:03	1.60
+++ tclvfs-1.3-20080503/generic/vfs.c	2008/10/10 21:57:36	1.61
@@ -39,6 +39,10 @@
 #define TCL_GLOB_TYPE_MOUNT		(1<<7)
 #endif
 
+#ifndef CONST86
+#define CONST86
+#endif
+
 /*
  * Only the _Init function is exported.
  */
@@ -1656,7 +1660,7 @@
     return returnVal;
 }
 
-static CONST char**
+static CONST char * CONST86 *
 VfsFileAttrStrings(pathPtr, objPtrRef)
     Tcl_Obj* pathPtr;
     Tcl_Obj** objPtrRef;
