--- libvmime-0.7.1/vmime/platforms/posix/posixChildProcess.hpp	2005-05-25 16:14:54.000000000 -0300
+++ libvmime-0.7.1_patched/vmime/platforms/posix/posixChildProcess.hpp	2011-03-09 01:09:18.000000000 -0300
@@ -20,6 +20,7 @@
 #ifndef VMIME_PLATFORMS_POSIX_POSIXCHILDPROCESS_HPP_INCLUDED
 #define VMIME_PLATFORMS_POSIX_POSIXCHILDPROCESS_HPP_INCLUDED
 
+#include <signal.h>
 
 #include "vmime/utility/childProcess.hpp"
 
