--- libvmime-0.7.1_orig/configure	2005-06-21 19:11:26.000000000 -0300
+++ libvmime-0.7.1/configure	2009-12-26 15:30:12.000000000 -0200
@@ -1557,7 +1557,7 @@
 GENERIC_VERSIONED_LIBRARY_NAME="vmime"
 
 
-LIBRARY_NAME="libvmime"
+LIBRARY_NAME="libvmime-zarafa"
 
 
 # Library version
@@ -25237,7 +25237,7 @@
 
 
 // Name of package
-#define VMIME_PACKAGE \"libvmime\"
+#define VMIME_PACKAGE \"libvmime-zarafa\"
 
 // Version number of package
 #define VMIME_VERSION \"0.7.1\"
--- libvmime-0.7.1_orig/src/Makefile.in	2005-06-21 19:11:35.000000000 -0300
+++ libvmime-0.7.1/src/Makefile.in	2009-12-26 17:57:50.000000000 -0200
@@ -357,7 +357,7 @@
 AUTOMAKE_OPTIONS = no-dependencies foreign
 INTERNALS = 
 INCLUDES = -I$(top_srcdir) -I$(srcdir) @PKGCONFIG_CFLAGS@ @EXTRA_CFLAGS@
-lib_LTLIBRARIES = libvmime.la
+lib_LTLIBRARIES = libvmime-zarafa.la
 libvmime_la_LDFLAGS = -export-dynamic -version-info @LIBRARY_VERSION@ @PKGCONFIG_LIBS@ @EXTRA_LIBS@
 libvmime_la_SOURCES = address.cpp \
 	addressList.cpp \
@@ -494,7 +494,7 @@
 	  echo "rm -f \"$${dir}/so_locations\""; \
 	  rm -f "$${dir}/so_locations"; \
 	done
-libvmime.la: $(libvmime_la_OBJECTS) $(libvmime_la_DEPENDENCIES) 
+libvmime-zarafa.la: $(libvmime_la_OBJECTS) $(libvmime_la_DEPENDENCIES) 
 	$(CXXLINK) -rpath $(libdir) $(libvmime_la_LDFLAGS) $(libvmime_la_OBJECTS) $(libvmime_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
