diff -Naur sdl-jstest-20150806git/CMakeLists.txt sdl-jstest-20150806git.patched/CMakeLists.txt
--- sdl-jstest-20150806git/CMakeLists.txt	2015-09-15 02:46:06.000000000 -0400
+++ sdl-jstest-20150806git.patched/CMakeLists.txt	2015-09-15 02:55:53.000000000 -0400
@@ -22,7 +22,6 @@
 find_package(SDL REQUIRED)
 
 find_package(PkgConfig REQUIRED)
-pkg_search_module(SDL2 REQUIRED sdl2)
 
 find_program(DOCBOOK2XMAN docbook2x-man)
 if(DOCBOOK2XMAN)
@@ -43,11 +42,6 @@
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
 endif()
 
-add_executable(sdl2-jstest sdl2-jstest.c)
-target_link_libraries(sdl2-jstest ${SDL2_LIBRARIES} ${CURSES_LIBRARIES})
-target_include_directories(sdl2-jstest PUBLIC ${SDL2_INCLUDE_DIRS} ${CURSES_INCLUDE_DIR})
-target_compile_definitions(sdl2-jstest PUBLIC ${SDL2_CFLAGS_OTHER})
-
 add_executable(sdl-jstest sdl-jstest.c)
 target_link_libraries(sdl-jstest ${SDL_LIBRARY} ${CURSES_LIBRARIES})
 target_include_directories(sdl-jstest PUBLIC ${SDL_INCLUDE_DIR} ${CURSES_INCLUDE_DIR})
@@ -59,22 +53,17 @@
     DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/sdl-jstest.xml
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
 
-  add_custom_command(
-    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/sdl2-jstest.1
-    COMMAND ${DOCBOOK2XMAN} ${CMAKE_CURRENT_SOURCE_DIR}/sdl2-jstest.xml
-    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/sdl2-jstest.xml
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
 else()
-  file(COPY sdl-jstest.1 sdl2-jstest.1
+  file(COPY sdl-jstest.1
     DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
 endif()
 
-add_custom_target(doc ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/sdl-jstest.1 ${CMAKE_CURRENT_BINARY_DIR}/sdl2-jstest.1)
+add_custom_target(doc ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/sdl-jstest.1)
 
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/sdl-jstest.1 ${CMAKE_CURRENT_BINARY_DIR}/sdl2-jstest.1
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/sdl-jstest.1
   DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
 
-install(TARGETS sdl-jstest sdl2-jstest
+install(TARGETS sdl-jstest
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 
 # EOF #
