diff -Naur mpgtx-1.3.1.orig/configure mpgtx-1.3.1/configure
--- mpgtx-1.3.1.orig/configure	2010-11-28 11:58:58.000000000 +0100
+++ mpgtx-1.3.1/configure	2010-11-28 12:00:00.000000000 +0100
@@ -11,7 +11,7 @@
 #compiler flags
 common_cflags="-Wall"
 devel_cflags="-g -Werror -fno-builtin"
-optimization_cflags="-O3"
+optimization_cflags="${CXXFLAGS}"
 release_cflags=$optimization_cflags
 OFLAGS="-ansi"
 cflags=$common_cflags
