--- a/gtk3proxies/CMakeLists.txt	2017-03-16 20:19:44.291236607 -0400
+++ b/gtk3proxies/CMakeLists.txt	2017-03-16 20:19:37.305236276 -0400
@@ -2,7 +2,7 @@
 
 include_directories(${GTK3_INCLUDE_DIRS})
 
-add_definitions(-DDATA_DIR="\\\"${DATA_INSTALL_DIR}/kcm-gtk-module/\\\"")
+add_definitions(-DDATA_DIR=\"${DATA_INSTALL_DIR}/kcm-gtk-module\")
 
 add_executable(gtk3_preview preview3.c)
 target_link_libraries(gtk3_preview ${GTK3_LIBRARY} ${GOBJECT2_LIBRARY} ${GLIB2_LIBRARY})
