Various fixes for libpcap0.8 compilation.

--- hping3-3.a2.ds1.orig/libpcap_stuff.c
+++ hping3-3.a2.ds1/libpcap_stuff.c
@@ -17,7 +17,6 @@
 #include <stdlib.h>
 #include <sys/ioctl.h>
 #include <pcap.h>
-#include <net/bpf.h>
 
 #include "globals.h"
 
--- hping3-3.a2.ds1.orig/script.c
+++ hping3-3.a2.ds1/script.c
@@ -24,7 +24,6 @@
 
 #include <sys/ioctl.h>
 #include <pcap.h>
-#include <net/bpf.h>
 
 #include "release.h"
 #include "hping2.h"
