--- filter/source/xsltfilter/makefile.mk	2009-03-27 10:14:39.000000000 +0300
+++ filter/source/xsltfilter/makefile.mk	2009-03-27 18:14:56.000000000 +0300
@@ -72,11 +72,11 @@
 CUSTOMMANIFESTFILE = Manifest
 
 JARCOMPRESS		= TRUE
-JARCLASSDIRS	= XSLTransformer*.class XSLTFilterOLEExtracter*.class
+JARCLASSDIRS	= XSLTransformer*.class XSLTFilterOLEExtracter*.class Base64*.class
 JARTARGET		= $(TARGET).jar
 
 # --- Files --------------------------------------------------------
-JAVACLASSFILES=$(CLASSDIR)$/XSLTransformer.class  $(CLASSDIR)$/XSLTFilterOLEExtracter.class
+JAVACLASSFILES=$(CLASSDIR)$/XSLTransformer.class  $(CLASSDIR)$/XSLTFilterOLEExtracter.class $(CLASSDIR)$/Base64.class
 .ENDIF
 
 # --- Targets ------------------------------------------------------
