--- sc/source/core/tool/makefile.mk-orig	2009-09-07 20:15:14.880679704 -0500
+++ sc/source/core/tool/makefile.mk	2009-09-07 20:14:38.941679215 -0500
@@ -132,7 +132,8 @@
         $(SLO)$/prnsave.obj \
 		$(SLO)$/reftokenhelper.obj \
 		$(SLO)$/stringutil.obj \
-		$(SLO)$/token.obj
+		$(SLO)$/token.obj \
+	$(SLO)$/detfunc.obj
 
 # [kh] POWERPC compiler problem
 .IF "$(OS)$(COM)$(CPUNAME)"=="LINUXGCCPOWERPC"
