diff -Naur miscom/fire.c miscom.patched/fire.c
--- miscom/fire.c	1997-03-15 15:42:10.000000000 -0500
+++ miscom.patched/fire.c	2016-10-03 19:58:35.332473407 -0400
@@ -66,7 +66,7 @@
 		}
 
 	if (found==0)
-		fprintf(stderr,"can't happen\n"),bombout();
+		fprintf(stderr,"can't happen\n"),bombout(0);
 
 	fireballs[a].x=x;
 	fireballs[a].y=y;
diff -Naur miscom/main.c miscom.patched/main.c
--- miscom/main.c	1998-01-22 11:30:32.000000000 -0500
+++ miscom.patched/main.c	2016-10-03 19:58:24.480472716 -0400
@@ -451,7 +451,7 @@
 	define_colours();
 	
 	if((keyptr=open("/dev/tty",O_RDONLY|O_NONBLOCK))<0)
-		fprintf(stderr,"Couldn't open /dev/tty."),bombout();
+		fprintf(stderr,"Couldn't open /dev/tty."),bombout(0);
 	
 	srandom(time(NULL));
 	signal(SIGINT,bombout);
diff -Naur miscom/sound.c miscom.patched/sound.c
--- miscom/sound.c	1997-03-15 12:17:45.000000000 -0500
+++ miscom.patched/sound.c	2016-10-03 19:54:26.923457583 -0400
@@ -5,6 +5,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <sys/soundcard.h>
+#include <sys/ioctl.h>
 #include "sound.h"
 
 int sound=1;
diff -Naur miscom/unix.c miscom.patched/unix.c
--- miscom/unix.c	1997-03-15 12:00:27.000000000 -0500
+++ miscom.patched/unix.c	2016-10-03 19:59:02.386475130 -0400
@@ -4,6 +4,8 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <errno.h>
+#include <stdlib.h>
+#include <string.h>
 
 #include "global.h"
 #include "draw.h"
@@ -78,7 +80,7 @@
 	}
 
 
-void bombout()
+void bombout(int unused)
 	{
 	queuesam(KILL_SNDSERV,0);
 	use_colour(COLOR_WHITE,A_NORMAL);
diff -Naur miscom/unix.h miscom.patched/unix.h
--- miscom/unix.h	1997-03-14 21:50:59.000000000 -0500
+++ miscom.patched/unix.h	2016-10-03 19:59:10.686475659 -0400
@@ -1,5 +1,5 @@
 extern unsigned char getnbkey(void);
-extern void bombout(void);
+extern void bombout(int unused);
 extern void key_pause(void);
 extern int numalive(void);
 extern void centre(char* buffer,int y);
