--- libclaw-1.7.4/CMakeLists.orig.txt	2015-05-14 17:54:05.443919651 +0300
+++ libclaw-1.7.4/CMakeLists.txt	2015-05-14 17:54:22.925701098 +0300
@@ -30,7 +30,7 @@
 set( CLAW_CODE_DIR claw/code/ )
 set( CLAW_INCLUDE_DIRECTORY "${CMAKE_INSTALL_PREFIX}/include/" )
 set( CLAW_INSTALLDIR_SOURCE include/claw/ )
-set( CLAW_INSTALLDIR_LIB lib/ )
+set( CLAW_INSTALLDIR_LIB lib64/ )
 set( CLAW_INSTALLDIR_DOC share/doc/libclaw${CLAW_VERSION_MAJOR}/ )
 set( CLAW_TRANSLATIONS_INSTALL_DIR "share/locale" )
 set( CLAW_EXECUTABLE_DIR bin/ )
