diff -Naur supertuxkart-0.6.1a.orig/src/Makefile.am supertuxkart-0.6.1a/src/Makefile.am
--- supertuxkart-0.6.1a.orig/src/Makefile.am	2009-02-16 01:23:45.000000000 +0000
+++ supertuxkart-0.6.1a/src/Makefile.am	2009-05-14 06:54:40.000000000 +0000
@@ -2,10 +2,10 @@
 
 SUBDIRS = ide
 
-bindir=$(prefix)/games
+bindir=$(prefix)/bin
 bin_PROGRAMS = supertuxkart
 
-AM_CPPFLAGS = -DSUPERTUXKART_DATADIR="\"$(datadir)/games/$(PACKAGE)/\""  \
+AM_CPPFLAGS = -DSUPERTUXKART_DATADIR="\"$(datadir)/$(PACKAGE)/\""  \
  -I$(srcdir)/bullet/src/ -I$(srcdir)/enet/include/
 
 supertuxkart_SOURCES = \
diff -Naur supertuxkart-0.6.1a.orig/src/Makefile.in supertuxkart-0.6.1a/src/Makefile.in
--- supertuxkart-0.6.1a.orig/src/Makefile.in	2009-02-18 23:30:42.000000000 +0000
+++ supertuxkart-0.6.1a/src/Makefile.in	2009-05-14 08:49:48.000000000 +0000
@@ -226,7 +226,7 @@
 am__quote = @am__quote@
 am__tar = @am__tar@
 am__untar = @am__untar@
-bindir = $(prefix)/games
+bindir = $(prefix)/bin
 build = @build@
 build_alias = @build_alias@
 build_cpu = @build_cpu@
@@ -271,7 +271,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 SUBDIRS = ide
-AM_CPPFLAGS = -DSUPERTUXKART_DATADIR="\"$(datadir)/games/$(PACKAGE)/\""  \
+AM_CPPFLAGS = -DSUPERTUXKART_DATADIR="\"$(datadir)/$(PACKAGE)/\""  \
  -I$(srcdir)/bullet/src/ -I$(srcdir)/enet/include/
 
 supertuxkart_SOURCES = \
diff -Naur supertuxkart-0.6.1a.orig/data/supertuxkart_desktop.template supertuxkart-0.6.1a/data/supertuxkart_desktop.template
--- supertuxkart-0.6.1a.orig/data/supertuxkart_desktop.template	2009-01-28 22:57:49.000000000 +0000
+++ supertuxkart-0.6.1a/data/supertuxkart_desktop.template	2009-05-14 09:01:44.000000000 +0000
@@ -8,10 +8,10 @@
 Comment[fr_FR]=Un jeu de karting
 Comment[ro_RO]=Un joc de curse cu carturi
 Comment[de_DE]=Ein Kart-Rennspiel
-Exec=PREFIX/games/supertuxkart --log=file
+Exec=PREFIX/bin/supertuxkart --log=file
 Terminal=false
 StartupNotify=false
 Type=Application
-TryExec=PREFIX/games/supertuxkart
+TryExec=PREFIX/bin/supertuxkart
 Categories=Game;ArcadeGame
 GenericName=
