diff -Naur supertuxkart-0.6.1a.orig/data/Makefile.in supertuxkart-0.6.1a/data/Makefile.in
--- supertuxkart-0.6.1a.orig/data/Makefile.in	2009-02-18 23:30:40.000000000 +0000
+++ supertuxkart-0.6.1a/data/Makefile.in	2009-05-14 08:20:11.000000000 +0000
@@ -75,7 +75,7 @@
 CTAGS = ctags
 DIST_SUBDIRS = $(SUBDIRS)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data
+pkgdatadir = $(datadir)/$(PACKAGE)/data
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur supertuxkart-0.6.1a.orig/data/fonts/Makefile.in supertuxkart-0.6.1a/data/fonts/Makefile.in
--- supertuxkart-0.6.1a.orig/data/fonts/Makefile.in	2009-02-18 23:30:40.000000000 +0000
+++ supertuxkart-0.6.1a/data/fonts/Makefile.in	2009-05-14 08:20:11.000000000 +0000
@@ -63,7 +63,7 @@
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/fonts
+pkgdatadir = $(datadir)/$(PACKAGE)/data/fonts
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur supertuxkart-0.6.1a.orig/data/grandprix/Makefile.in supertuxkart-0.6.1a/data/grandprix/Makefile.in
--- supertuxkart-0.6.1a.orig/data/grandprix/Makefile.in	2009-02-18 23:30:40.000000000 +0000
+++ supertuxkart-0.6.1a/data/grandprix/Makefile.in	2009-05-14 08:20:11.000000000 +0000
@@ -63,7 +63,7 @@
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/grandprix
+pkgdatadir = $(datadir)/$(PACKAGE)/data/grandprix
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur supertuxkart-0.6.1a.orig/data/items/Makefile.in supertuxkart-0.6.1a/data/items/Makefile.in
--- supertuxkart-0.6.1a.orig/data/items/Makefile.in	2009-02-18 23:30:40.000000000 +0000
+++ supertuxkart-0.6.1a/data/items/Makefile.in	2009-05-14 08:20:11.000000000 +0000
@@ -63,7 +63,7 @@
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/items
+pkgdatadir = $(datadir)/$(PACKAGE)/data/items
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur supertuxkart-0.6.1a.orig/data/karts/Makefile.in supertuxkart-0.6.1a/data/karts/Makefile.in
--- supertuxkart-0.6.1a.orig/data/karts/Makefile.in	2009-02-18 23:44:11.000000000 +0000
+++ supertuxkart-0.6.1a/data/karts/Makefile.in	2009-05-14 08:20:11.000000000 +0000
@@ -63,7 +63,7 @@
 nobase_dist_pkgdataDATA_INSTALL = $(install_sh_DATA)
 DATA = $(nobase_dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/karts
+pkgdatadir = $(datadir)/$(PACKAGE)/data/karts
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur supertuxkart-0.6.1a.orig/data/models/Makefile.in supertuxkart-0.6.1a/data/models/Makefile.in
--- supertuxkart-0.6.1a.orig/data/models/Makefile.in	2009-02-18 23:30:40.000000000 +0000
+++ supertuxkart-0.6.1a/data/models/Makefile.in	2009-05-14 08:20:11.000000000 +0000
@@ -63,7 +63,7 @@
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/models
+pkgdatadir = $(datadir)/$(PACKAGE)/data/models
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur supertuxkart-0.6.1a.orig/data/music/Makefile.in supertuxkart-0.6.1a/data/music/Makefile.in
--- supertuxkart-0.6.1a.orig/data/music/Makefile.in	2009-02-18 23:30:40.000000000 +0000
+++ supertuxkart-0.6.1a/data/music/Makefile.in	2009-05-14 08:20:11.000000000 +0000
@@ -63,7 +63,7 @@
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/music
+pkgdatadir = $(datadir)/$(PACKAGE)/data/music
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur supertuxkart-0.6.1a.orig/data/po/Makefile.in supertuxkart-0.6.1a/data/po/Makefile.in
--- supertuxkart-0.6.1a.orig/data/po/Makefile.in	2009-02-18 23:30:40.000000000 +0000
+++ supertuxkart-0.6.1a/data/po/Makefile.in	2009-05-14 08:20:11.000000000 +0000
@@ -62,7 +62,7 @@
 nobase_pkgdataDATA_INSTALL = $(install_sh_DATA)
 DATA = $(nobase_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/@PACKAGE@/data/po
+pkgdatadir = @localedir@
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur supertuxkart-0.6.1a.orig/data/sfx/Makefile.in supertuxkart-0.6.1a/data/sfx/Makefile.in
--- supertuxkart-0.6.1a.orig/data/sfx/Makefile.in	2009-02-18 23:30:41.000000000 +0000
+++ supertuxkart-0.6.1a/data/sfx/Makefile.in	2009-05-14 08:20:11.000000000 +0000
@@ -63,7 +63,7 @@
 nobase_dist_pkgdataDATA_INSTALL = $(install_sh_DATA)
 DATA = $(nobase_dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/sfx
+pkgdatadir = $(datadir)/$(PACKAGE)/data/sfx
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur supertuxkart-0.6.1a.orig/data/textures/Makefile.in supertuxkart-0.6.1a/data/textures/Makefile.in
--- supertuxkart-0.6.1a.orig/data/textures/Makefile.in	2009-02-18 23:30:41.000000000 +0000
+++ supertuxkart-0.6.1a/data/textures/Makefile.in	2009-05-14 08:20:11.000000000 +0000
@@ -63,7 +63,7 @@
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/textures
+pkgdatadir = $(datadir)/$(PACKAGE)/data/textures
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur supertuxkart-0.6.1a.orig/data/tracks/Makefile.in supertuxkart-0.6.1a/data/tracks/Makefile.in
--- supertuxkart-0.6.1a.orig/data/tracks/Makefile.in	2009-02-18 23:30:41.000000000 +0000
+++ supertuxkart-0.6.1a/data/tracks/Makefile.in	2009-05-14 08:20:11.000000000 +0000
@@ -63,7 +63,7 @@
 nobase_dist_pkgdataDATA_INSTALL = $(install_sh_DATA)
 DATA = $(nobase_dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/tracks
+pkgdatadir = $(datadir)/$(PACKAGE)/data/tracks
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
