diff -Naur supertuxkart-0.6.1a.orig/data/Makefile.am supertuxkart-0.6.1a/data/Makefile.am
--- supertuxkart-0.6.1a.orig/data/Makefile.am	2009-01-28 22:57:55.000000000 +0000
+++ supertuxkart-0.6.1a/data/Makefile.am	2009-05-14 06:53:59.000000000 +0000
@@ -1,7 +1,7 @@
 # data/
 
 SUBDIRS    = fonts items karts models music po sfx textures tracks grandprix
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data
+pkgdatadir = $(datadir)/$(PACKAGE)/data
 
 dist_pkgdata_DATA = \
  $(shell find $(srcdir) -name "*.data") \
diff -Naur supertuxkart-0.6.1a.orig/data/fonts/Makefile.am supertuxkart-0.6.1a/data/fonts/Makefile.am
--- supertuxkart-0.6.1a.orig/data/fonts/Makefile.am	2009-01-28 22:57:39.000000000 +0000
+++ supertuxkart-0.6.1a/data/fonts/Makefile.am	2009-05-14 06:53:59.000000000 +0000
@@ -1,5 +1,5 @@
 # data/fonts/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/fonts
+pkgdatadir = $(datadir)/$(PACKAGE)/data/fonts
 
 dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.txf")
diff -Naur supertuxkart-0.6.1a.orig/data/grandprix/Makefile.am supertuxkart-0.6.1a/data/grandprix/Makefile.am
--- supertuxkart-0.6.1a.orig/data/grandprix/Makefile.am	2009-01-28 22:57:55.000000000 +0000
+++ supertuxkart-0.6.1a/data/grandprix/Makefile.am	2009-05-14 06:53:59.000000000 +0000
@@ -1,6 +1,6 @@
 # data/music
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/grandprix
+pkgdatadir = $(datadir)/$(PACKAGE)/data/grandprix
 
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.grandprix")  \
                     $(shell find $(srcdir) -name "*.challenge")
diff -Naur supertuxkart-0.6.1a.orig/data/items/Makefile.am supertuxkart-0.6.1a/data/items/Makefile.am
--- supertuxkart-0.6.1a.orig/data/items/Makefile.am	2009-01-28 22:57:39.000000000 +0000
+++ supertuxkart-0.6.1a/data/items/Makefile.am	2009-05-14 06:53:59.000000000 +0000
@@ -1,6 +1,6 @@
 # data/herrings/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/items
+pkgdatadir = $(datadir)/$(PACKAGE)/data/items
 
 dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.ac") License.txt
 
diff -Naur supertuxkart-0.6.1a.orig/data/karts/Makefile.am supertuxkart-0.6.1a/data/karts/Makefile.am
--- supertuxkart-0.6.1a.orig/data/karts/Makefile.am	2009-02-18 23:43:38.000000000 +0000
+++ supertuxkart-0.6.1a/data/karts/Makefile.am	2009-05-14 06:53:59.000000000 +0000
@@ -1,6 +1,6 @@
 # data/karts
 
-pkgdatadir= $(datadir)/games/$(PACKAGE)/data/karts
+pkgdatadir= $(datadir)/$(PACKAGE)/data/karts
 
 nobase_dist_pkgdata_DATA = $(shell find $(srcdir) -name "materials.dat") \
                            $(shell find $(srcdir) -name "*.rgb") \
diff -Naur supertuxkart-0.6.1a.orig/data/models/Makefile.am supertuxkart-0.6.1a/data/models/Makefile.am
--- supertuxkart-0.6.1a.orig/data/models/Makefile.am	2009-01-28 22:57:55.000000000 +0000
+++ supertuxkart-0.6.1a/data/models/Makefile.am	2009-05-14 06:53:59.000000000 +0000
@@ -1,6 +1,6 @@
 # data/models/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/models
+pkgdatadir = $(datadir)/$(PACKAGE)/data/models
 
 dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.ac") License.txt
 
diff -Naur supertuxkart-0.6.1a.orig/data/music/Makefile.am supertuxkart-0.6.1a/data/music/Makefile.am
--- supertuxkart-0.6.1a.orig/data/music/Makefile.am	2009-01-28 22:57:47.000000000 +0000
+++ supertuxkart-0.6.1a/data/music/Makefile.am	2009-05-14 06:53:59.000000000 +0000
@@ -1,6 +1,6 @@
 # data/music
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/music
+pkgdatadir = $(datadir)/$(PACKAGE)/data/music
 
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.ogg")  \
                     $(shell find $(srcdir) -name "*.music") \
diff -Naur supertuxkart-0.6.1a.orig/data/po/Makefile.am supertuxkart-0.6.1a/data/po/Makefile.am
--- supertuxkart-0.6.1a.orig/data/po/Makefile.am	2009-01-28 22:57:38.000000000 +0000
+++ supertuxkart-0.6.1a/data/po/Makefile.am	2009-05-14 06:53:59.000000000 +0000
@@ -1,6 +1,6 @@
 # data/po/
 
-pkgdatadir = $(datadir)/games/@PACKAGE@/data/po
+pkgdatadir = @localedir@
 
 nobase_pkgdata_DATA = $(shell find $(srcdir) -name "*.po") \
                       $(shell find $(srcdir) -name "*.mo") \
diff -Naur supertuxkart-0.6.1a.orig/data/sfx/Makefile.am supertuxkart-0.6.1a/data/sfx/Makefile.am
--- supertuxkart-0.6.1a.orig/data/sfx/Makefile.am	2009-01-28 22:57:39.000000000 +0000
+++ supertuxkart-0.6.1a/data/sfx/Makefile.am	2009-05-14 06:53:59.000000000 +0000
@@ -1,6 +1,6 @@
 # sfx/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/sfx
+pkgdatadir = $(datadir)/$(PACKAGE)/data/sfx
 
 nobase_dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.wav") \
                             $(shell find $(srcdir) -name "README")\
diff -Naur supertuxkart-0.6.1a.orig/data/textures/Makefile.am supertuxkart-0.6.1a/data/textures/Makefile.am
--- supertuxkart-0.6.1a.orig/data/textures/Makefile.am	2009-01-28 22:57:54.000000000 +0000
+++ supertuxkart-0.6.1a/data/textures/Makefile.am	2009-05-14 06:53:59.000000000 +0000
@@ -1,6 +1,6 @@
 # data/textures
 
-pkgdatadir   = $(datadir)/games/$(PACKAGE)/data/textures
+pkgdatadir   = $(datadir)/$(PACKAGE)/data/textures
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.rgb") materials.dat License.txt
 
 
diff -Naur supertuxkart-0.6.1a.orig/data/tracks/Makefile.am supertuxkart-0.6.1a/data/tracks/Makefile.am
--- supertuxkart-0.6.1a.orig/data/tracks/Makefile.am	2009-01-28 22:58:20.000000000 +0000
+++ supertuxkart-0.6.1a/data/tracks/Makefile.am	2009-05-14 06:53:59.000000000 +0000
@@ -1,6 +1,6 @@
 # data/tracks
 
-pkgdatadir= $(datadir)/games/$(PACKAGE)/data/tracks
+pkgdatadir= $(datadir)/$(PACKAGE)/data/tracks
 nobase_dist_pkgdata_DATA = $(shell find $(srcdir) -name "readme.txt")      \
                            $(shell find $(srcdir) -name "*.track")         \
                            $(shell find $(srcdir) -name "*.drv?")          \
