diff -Naur qtwebkit-opensource-src-5.9.1.orig/Source/WebCore/dom/Document.cpp qtwebkit-opensource-src-5.9.1/Source/WebCore/dom/Document.cpp
--- qtwebkit-opensource-src-5.9.1.orig/Source/WebCore/dom/Document.cpp	2017-06-16 14:46:36.000000000 +0200
+++ qtwebkit-opensource-src-5.9.1/Source/WebCore/dom/Document.cpp	2019-10-08 06:49:07.806058000 +0200
@@ -3846,12 +3846,12 @@
     unsigned i = 0;
 
     UChar32 c;
-    U16_NEXT(characters, i, length, c)
+    U16_NEXT(characters, i, length, c);
     if (!isValidNameStart(c))
         return false;
 
     while (i < length) {
-        U16_NEXT(characters, i, length, c)
+        U16_NEXT(characters, i, length, c);
         if (!isValidNamePart(c))
             return false;
     }
@@ -3914,7 +3914,7 @@
     const UChar* s = qualifiedName.characters();
     for (unsigned i = 0; i < length;) {
         UChar32 c;
-        U16_NEXT(s, i, length, c)
+        U16_NEXT(s, i, length, c);
         if (c == ':') {
             if (sawColon) {
                 ec = NAMESPACE_ERR;
diff -Naur qtwebkit-opensource-src-5.9.1.orig/Source/WebCore/platform/graphics/SegmentedFontData.cpp qtwebkit-opensource-src-5.9.1/Source/WebCore/platform/graphics/SegmentedFontData.cpp
--- qtwebkit-opensource-src-5.9.1.orig/Source/WebCore/platform/graphics/SegmentedFontData.cpp	2017-06-16 14:46:36.000000000 +0200
+++ qtwebkit-opensource-src-5.9.1/Source/WebCore/platform/graphics/SegmentedFontData.cpp	2019-10-08 06:47:12.829058000 +0200
@@ -61,7 +61,7 @@
 {
     UChar32 c;
     for (int i = 0; i < length; ) {
-        U16_NEXT(characters, i, length, c)
+        U16_NEXT(characters, i, length, c);
         if (!containsCharacter(c))
             return false;
     }
diff -Naur qtwebkit-opensource-src-5.9.1.orig/Source/WebCore/platform/mac/WebCoreNSURLExtras.mm qtwebkit-opensource-src-5.9.1/Source/WebCore/platform/mac/WebCoreNSURLExtras.mm
--- qtwebkit-opensource-src-5.9.1.orig/Source/WebCore/platform/mac/WebCoreNSURLExtras.mm	2017-06-16 14:46:36.000000000 +0200
+++ qtwebkit-opensource-src-5.9.1/Source/WebCore/platform/mac/WebCoreNSURLExtras.mm	2019-10-08 06:48:15.130058000 +0200
@@ -175,7 +175,7 @@
     int32_t i = 0;
     while (i < length) {
         UChar32 c;
-        U16_NEXT(buffer, i, length, c)
+        U16_NEXT(buffer, i, length, c);
         UErrorCode error = U_ZERO_ERROR;
         UScriptCode script = uscript_getScript(c, &error);
         if (error != U_ZERO_ERROR) {
@@ -746,7 +746,7 @@
     CFIndex i = 0;
     while (i < length) {
         UChar32 c;
-        U16_NEXT(sourceBuffer, i, length, c)
+        U16_NEXT(sourceBuffer, i, length, c);
         
         if (isLookalikeCharacter(c)) {
             uint8_t utf8Buffer[4];
