diff -Naur yvonne-1.0.orig/color_provider.h yvonne-1.0/color_provider.h
--- yvonne-1.0.orig/color_provider.h	2006-12-28 16:37:54.000000000 -0500
+++ yvonne-1.0/color_provider.h	2022-12-27 22:38:08.296725828 -0500
@@ -39,7 +39,7 @@
 
 	CCPException(CPReason _eReason) throw():m_eReason(_eReason) {};
 	CPReason getReason(void) { return m_eReason; }
-	char* what(void)
+	const char* what(void)
 	{
 		switch(m_eReason)
 		{
diff -Naur yvonne-1.0.orig/image.h yvonne-1.0/image.h
--- yvonne-1.0.orig/image.h	2006-12-28 16:37:54.000000000 -0500
+++ yvonne-1.0/image.h	2022-12-27 22:38:24.529994904 -0500
@@ -56,7 +56,7 @@
 	CImageException(IEReason _eReason) throw():m_eReason(_eReason) {};
 	IEReason getReason(void)
 		{ return m_eReason; }
-	char* what(void)
+	const char* what(void)
 	{
 		switch(m_eReason)
 		{
diff -Naur yvonne-1.0.orig/rc_file.h yvonne-1.0/rc_file.h
--- yvonne-1.0.orig/rc_file.h	2006-12-28 16:37:54.000000000 -0500
+++ yvonne-1.0/rc_file.h	2022-12-27 22:38:47.368373460 -0500
@@ -41,7 +41,7 @@
 
 	CRCFileException(RCFReason _eReason) throw():m_eReason(_eReason) {};
 	RCFReason getReason(void) { return m_eReason; }
-	char* what(void)
+	const char* what(void)
 	{
 		switch(m_eReason)
 		{
