--- config.pri	2016-07-28 22:17:44.000000000 -0300
+++ config.pri.new	2016-12-12 16:43:50.000000000 -0200
@@ -280,12 +280,8 @@
 
 	exists(/usr/include/qwt5) {
 		INCLUDEPATH  += /usr/include/qwt5
-                exists(/usr/lib*/libqwt5.*) {
                       LIBS         += -lqwt5
                 } else {
-                      LIBS += -lqwt
-                }
-	} else {
 		INCLUDEPATH  += /usr/include/qwt$${qwtsuff}
 		LIBS         += -lqwt$${qwtsuff}
 	}
