diff --git a/libscidavis/src/ApplicationWindow.cpp b/libscidavis/src/ApplicationWindow.cpp
index 90d282a4..167debee 100644
--- libscidavis/src/ApplicationWindow.cpp
+++ libscidavis/src/ApplicationWindow.cpp
@@ -12521,8 +12521,11 @@ void ApplicationWindow::saveFolderAsProject(Folder *f)
 
 void ApplicationWindow::showFolderPopupMenu(const QPoint &p, bool fromFolders)
 {
-  if (fromFolders)
-    showFolderPopupMenuImpl(folders->itemAt(p), fromFolders)->exec(folders->mapToGlobal(p));
+  if (fromFolders) {
+    QMenu* cm = showFolderPopupMenuImpl(folders->itemAt(p), fromFolders);
+    if (cm)
+        cm->exec(folders->mapToGlobal(p));
+    }
   else
     showFolderPopupMenuImpl(lv->itemAt(p), fromFolders)->exec(lv->mapToGlobal(p));
 }
