--- scidavis/basic.pri	2017-07-14 04:54:53.000000000 -0300
+++ /home/fellype/compilando/scidavis-github/scidavis/basic.pri	2017-07-19 09:36:40.000000000 -0300
@@ -67,7 +67,7 @@
 # note the translation files are not writable during AEGIS
 # integration, so we don't want to perform an update then
         tstarget.commands = (! test -w translations/scidavis_de.ts || \
-$$LUPDATE_BIN src/*.cpp ../libscidavis/src/*.cpp -ts translations/*.ts) && $$LRELEASE_BIN translations/*.ts
+$$LUPDATE_BIN src ../libscidavis/src -ts translations/*.ts) && $$LRELEASE_BIN translations/*.ts
 
         QMAKE_EXTRA_TARGETS += tstarget
         QMAKE_CLEAN += $$translationfiles.files
