--- Fit.cpp	2016-07-28 22:17:44.000000000 -0300
+++ Fit.cpp.new	2016-11-28 10:46:40.000000000 -0200
@@ -127,9 +127,10 @@
 	gsl_multifit_covar (s->J, 0.0, covar);
 #else
         {
-          gsl_matrix J;
-          gsl_multifit_fdfsolver_jac(s,&J);
-          gsl_multifit_covar (&J, 0.0, covar);
+          gsl_matrix *J = gsl_matrix_alloc(d_n, d_p);
+          gsl_multifit_fdfsolver_jac(s,J);
+          gsl_multifit_covar (J, 0.0, covar);
+	  gsl_matrix_free (J);
         }
 #endif
 	if (d_y_error_source == UnknownErrors) {
