Index: libxdg-basedir-1.2.0/src/basedir.c
===================================================================
--- libxdg-basedir-1.2.0.orig/src/basedir.c	2014-02-11 19:56:41.702376488 +0100
+++ libxdg-basedir-1.2.0/src/basedir.c	2014-02-11 19:56:41.698376482 +0100
@@ -574,7 +574,7 @@
 		unsigned int homelen;
 		if (!(home = xdgGetEnv("HOME")))
 			return NULL;
-		if (!(relhome = (char*)malloc((homelen = strlen(home))+fallbacklength))) return NULL;
+		if (!(relhome = (char*)malloc((homelen = strlen(home))+fallbacklength+1))) return NULL;
 		memcpy(relhome, home, homelen);
 		memcpy(relhome+homelen, relativefallback, fallbacklength+1);
 	}
