diff -ur nim-2.2.0-orig/bin/nim-gdb nim-2.2.0/bin/nim-gdb
--- nim-2.2.0-orig/bin/nim-gdb	2024-10-04 13:48:24.730391583 +0200
+++ nim-2.2.0/bin/nim-gdb	2024-10-06 15:02:47.005375794 +0200
@@ -14,7 +14,7 @@
 fi
 
 # Find out where the pretty printer Python module is
-GDB_PYTHON_MODULE_PATH="$NIM_SYSROOT/tools/debug/nim-gdb.py"
+GDB_PYTHON_MODULE_PATH="/usr/lib/nim/tools/nim-gdb.py"
 
 # Run GDB with the additional arguments that load the pretty printers
 # Set the environment variable `NIM_GDB` to overwrite the call to a
