diff -ur nim-2.2.0-orig/install.sh nim-2.2.0/install.sh
--- nim-2.2.0-orig/install.sh	2024-10-04 13:48:24.730391583 +0200
+++ nim-2.2.0/install.sh	2024-10-06 15:57:45.877571032 +0200
@@ -27,7 +27,7 @@
     "/usr/bin")
       bindir=$1
       configdir="/etc/nim"
-      libdir="/usr/lib/nim"
+      libdir="/usr/lib/nim/lib"
       docdir="/usr/share/nim/doc"
       datadir="/usr/share/nim/data"
       nimbleDir="/opt/nimble/pkgs/nim-2.2.0"
