diff -Naur OpenSceneGraph-OpenSceneGraph-3.6.5/CMakeLists.txt OpenSceneGraph-OpenSceneGraph-3.6.5.patched/CMakeLists.txt
--- OpenSceneGraph-OpenSceneGraph-3.6.5/CMakeLists.txt	2020-01-31 06:03:07.000000000 -0500
+++ OpenSceneGraph-OpenSceneGraph-3.6.5.patched/CMakeLists.txt	2022-02-15 01:01:02.259927943 -0500
@@ -763,9 +763,6 @@
     FIND_PACKAGE(GtkGl)
     FIND_PACKAGE(DirectInput)
     FIND_PACKAGE(NVTT)
-    IF (NOT WIN32)
-        FIND_PACKAGE(Asio)
-    ENDIF()
     FIND_PACKAGE(ZeroConf)
 
     FIND_PACKAGE(LIBLAS)
