diff -ur CoinMP-1.8.3.orig/Cbc/Makefile.am CoinMP-1.8.3/Cbc/Makefile.am
--- CoinMP-1.8.3.orig/Cbc/Makefile.am	2015-02-07 11:56:19.000000000 -0800
+++ CoinMP-1.8.3/Cbc/Makefile.am	2018-06-25 08:31:55.590319068 -0700
@@ -158,4 +158,4 @@
 # Files that are generated and should be cleaned with make distclean
 DISTCLEANFILES =
 
-include BuildTools/Makemain.inc
+include ../BuildTools/Makemain.inc
diff -ur CoinMP-1.8.3.orig/Cbc/configure.ac CoinMP-1.8.3/Cbc/configure.ac
--- CoinMP-1.8.3.orig/Cbc/configure.ac	2015-09-22 09:20:00.000000000 -0700
+++ CoinMP-1.8.3/Cbc/configure.ac	2018-06-25 08:24:16.591742973 -0700
@@ -13,6 +13,7 @@
 AC_PREREQ(2.59)
 
 AC_INIT([Cbc],[2.9.6],[cbc@lists.coin-or.org])
+m4_include([../BuildTools/coin.m4])
 
 AC_COPYRIGHT([
 Copyright 2006 International Business Machines and others.
diff -ur CoinMP-1.8.3.orig/Cgl/Makefile.am CoinMP-1.8.3/Cgl/Makefile.am
--- CoinMP-1.8.3.orig/Cgl/Makefile.am	2015-02-07 11:56:21.000000000 -0800
+++ CoinMP-1.8.3/Cgl/Makefile.am	2018-06-25 08:32:03.454397372 -0700
@@ -96,4 +96,4 @@
 # Files that are generated and should be cleaned with make distclean
 DISTCLEANFILES =
 
-include BuildTools/Makemain.inc
+include ../BuildTools/Makemain.inc
diff -ur CoinMP-1.8.3.orig/Cgl/configure.ac CoinMP-1.8.3/Cgl/configure.ac
--- CoinMP-1.8.3.orig/Cgl/configure.ac	2015-09-22 08:01:33.000000000 -0700
+++ CoinMP-1.8.3/Cgl/configure.ac	2018-06-25 08:24:22.583802788 -0700
@@ -13,6 +13,7 @@
 AC_PREREQ(2.59)
 
 AC_INIT([Cgl],[0.59.7],[cgl@list.coin-or.org])
+m4_include([../BuildTools/coin.m4])
 
 AC_COPYRIGHT([
 Copyright 2006 International Business Machines and others.
diff -ur CoinMP-1.8.3.orig/Clp/Makefile.am CoinMP-1.8.3/Clp/Makefile.am
--- CoinMP-1.8.3.orig/Clp/Makefile.am	2015-03-09 08:16:55.000000000 -0700
+++ CoinMP-1.8.3/Clp/Makefile.am	2018-06-25 08:32:36.001721419 -0700
@@ -146,4 +146,4 @@
 # Files that are generated and should be cleaned with make distclean
 DISTCLEANFILES =
 
-include BuildTools/Makemain.inc
+include ../BuildTools/Makemain.inc
diff -ur CoinMP-1.8.3.orig/Clp/configure.ac CoinMP-1.8.3/Clp/configure.ac
--- CoinMP-1.8.3.orig/Clp/configure.ac	2015-09-21 11:52:13.000000000 -0700
+++ CoinMP-1.8.3/Clp/configure.ac	2018-06-25 08:24:33.811914865 -0700
@@ -13,6 +13,7 @@
 AC_PREREQ(2.59)
 
 AC_INIT([Clp],[1.16.8],[clp@list.coin-or.org])
+m4_include([../BuildTools/coin.m4])
 
 AC_COPYRIGHT([
 Copyright 2006 International Business Machines and others.
diff -ur CoinMP-1.8.3.orig/CoinMP/Makefile.am CoinMP-1.8.3/CoinMP/Makefile.am
--- CoinMP-1.8.3.orig/CoinMP/Makefile.am	2015-01-27 22:55:44.000000000 -0800
+++ CoinMP-1.8.3/CoinMP/Makefile.am	2018-06-25 08:32:53.339896167 -0700
@@ -88,4 +88,4 @@
 # Files that are generated and should be cleaned with make distclean
 DISTCLEANFILES =
 
-include BuildTools/Makemain.inc
+include ../BuildTools/Makemain.inc
diff -ur CoinMP-1.8.3.orig/CoinMP/configure.ac CoinMP-1.8.3/CoinMP/configure.ac
--- CoinMP-1.8.3.orig/CoinMP/configure.ac	2015-09-22 12:52:20.000000000 -0700
+++ CoinMP-1.8.3/CoinMP/configure.ac	2018-06-25 08:24:54.666123010 -0700
@@ -13,6 +13,7 @@
 AC_PREREQ(2.59)
 
 AC_INIT([CoinMP],[1.8.3],[coinmp@list.coin-or.org])
+m4_include([../BuildTools/coin.m4])
 
 AC_COPYRIGHT([
 Copyright 2006 International Business Machines and others.
diff -ur CoinMP-1.8.3.orig/CoinUtils/Makefile.am CoinMP-1.8.3/CoinUtils/Makefile.am
--- CoinMP-1.8.3.orig/CoinUtils/Makefile.am	2015-02-06 16:02:10.000000000 -0800
+++ CoinMP-1.8.3/CoinUtils/Makefile.am	2018-06-25 08:32:42.258783881 -0700
@@ -93,4 +93,4 @@
 # Files that are generated and should be cleaned with make distclean
 DISTCLEANFILES =
 
-include BuildTools/Makemain.inc
+include ../BuildTools/Makemain.inc
diff -ur CoinMP-1.8.3.orig/CoinUtils/configure.ac CoinMP-1.8.3/CoinUtils/configure.ac
--- CoinMP-1.8.3.orig/CoinUtils/configure.ac	2015-09-11 10:04:38.000000000 -0700
+++ CoinMP-1.8.3/CoinUtils/configure.ac	2018-06-25 08:24:39.825974894 -0700
@@ -13,6 +13,7 @@
 AC_PREREQ(2.59)
 
 AC_INIT([CoinUtils],[2.10.10],[http://projects.coin-or.org/CoinUtils])
+m4_include([../BuildTools/coin.m4])
 
 AC_COPYRIGHT([
 Copyright 2006 International Business Machines and others.
diff -ur CoinMP-1.8.3.orig/Data/Sample/Makefile.am CoinMP-1.8.3/Data/Sample/Makefile.am
--- CoinMP-1.8.3.orig/Data/Sample/Makefile.am	2012-11-21 10:57:40.000000000 -0800
+++ CoinMP-1.8.3/Data/Sample/Makefile.am	2018-06-25 08:31:42.277186499 -0700
@@ -22,7 +22,7 @@
 pkgconfiglibdir = $(libdir)/pkgconfig
 pkgconfiglib_DATA = coindatasample.pc
 
-include BuildTools/Makemain.inc
+include ../../BuildTools/Makemain.inc
 
 test:
 	@echo "No test available."
diff -ur CoinMP-1.8.3.orig/Data/Sample/configure.ac CoinMP-1.8.3/Data/Sample/configure.ac
--- CoinMP-1.8.3.orig/Data/Sample/configure.ac	2015-01-04 18:29:43.000000000 -0800
+++ CoinMP-1.8.3/Data/Sample/configure.ac	2018-06-25 08:24:01.621593527 -0700
@@ -13,6 +13,7 @@
 AC_PREREQ(2.59)
 
 AC_INIT([DataSample],[1.2.10],[https://projects.coin-or.org/BuildTools/])
+m4_include([../../BuildTools/coin.m4])
 
 AC_COPYRIGHT([
 Copyright 2006 International Business Machines and others.
diff -ur CoinMP-1.8.3.orig/Osi/Makefile.am CoinMP-1.8.3/Osi/Makefile.am
--- CoinMP-1.8.3.orig/Osi/Makefile.am	2015-02-07 11:56:33.000000000 -0800
+++ CoinMP-1.8.3/Osi/Makefile.am	2018-06-25 08:32:26.888630693 -0700
@@ -126,4 +126,4 @@
 # Files that are generated and should be cleaned with make distclean
 DISTCLEANFILES =
 
-include BuildTools/Makemain.inc
+include ../BuildTools/Makemain.inc
diff -ur CoinMP-1.8.3.orig/Osi/configure.ac CoinMP-1.8.3/Osi/configure.ac
--- CoinMP-1.8.3.orig/Osi/configure.ac	2015-09-21 15:51:25.000000000 -0700
+++ CoinMP-1.8.3/Osi/configure.ac	2018-06-25 08:24:28.184858698 -0700
@@ -13,6 +13,7 @@
 AC_PREREQ(2.59)
 
 AC_INIT([Osi],[0.107.6],[osi@list.coin-or.org])
+m4_include([../BuildTools/coin.m4])
 
 AC_COPYRIGHT([
 Copyright 2006 International Business Machines and others.
diff -ur CoinMP-1.8.3.orig/configure.ac CoinMP-1.8.3/configure.ac
--- CoinMP-1.8.3.orig/configure.ac	2015-09-22 12:52:20.000000000 -0700
+++ CoinMP-1.8.3/configure.ac	2018-06-25 08:24:48.602062487 -0700
@@ -13,6 +13,7 @@
 AC_PREREQ(2.59)
 
 AC_INIT([CoinMP],[1.8.3],[coinmp@list.coin-or.org])
+m4_include([BuildTools/coin.m4])
 
 AC_COPYRIGHT([
 Copyright 2006 International Business Machines and others.
