diff -ur CoinMP-1.8.3.orig/CoinMP/Makefile.am CoinMP-1.8.3/CoinMP/Makefile.am
--- CoinMP-1.8.3.orig/CoinMP/Makefile.am	2015-01-27 22:55:44.000000000 -0800
+++ CoinMP-1.8.3/CoinMP/Makefile.am	2018-06-25 09:44:47.242918098 -0700
@@ -57,9 +57,10 @@
 pkgconfiglibdir = $(libdir)/pkgconfig
 pkgconfiglib_DATA = coinmp.pc
 
-addlibsdir = $(DESTDIR)$(datadir)/coin/doc/CoinMP
+datacoindir = $(datadir)/coin/doc/CoinMP
+datacoin_DATA = examples/Makefile examples/example.c
 
-addlibs_DATA = examples/Makefile examples/example.c
+addlibsdir = $(DESTDIR)$(datadir)/coin/doc/CoinMP
 
 install-data-hook:
 	@$(mkdir_p) "$(addlibsdir)"
