diff -Naur ClientForm-0.2.10.orig/setup.py ClientForm-0.2.10/setup.py
--- ClientForm-0.2.10.orig/setup.py	2008-12-03 23:12:31.000000000 +0000
+++ ClientForm-0.2.10/setup.py	2009-03-16 15:48:45.000000000 +0000
@@ -89,8 +89,6 @@
             del kwargs["py_modules"]
         apply(_setup, (), kwargs)
 else:
-    import ez_setup
-    ez_setup.use_setuptools()
     from setuptools import setup
 
 setup(
