diff -Naur libadwaita-1.2.0/src/adw-message-dialog.ui libadwaita-1.2.0.new/src/adw-message-dialog.ui
--- libadwaita-1.2.0/src/adw-message-dialog.ui	2022-09-14 20:41:34.000000000 -0400
+++ libadwaita-1.2.0.new/src/adw-message-dialog.ui	2023-01-25 18:00:34.228428374 -0500
@@ -5,11 +5,6 @@
     <style>
       <class name="messagedialog"/>
     </style>
-    <property name="titlebar">
-      <object class="GtkBox">
-        <property name="visible">False</property>
-      </object>
-    </property>
     <property name="child">
       <object class="GtkWindowHandle">
         <property name="child">
