diff -Naur libadwaita-1.0.5/src/meson.build libadwaita-1.0.5.new/src/meson.build
--- libadwaita-1.0.5/src/meson.build	2022-06-01 06:48:47.000000000 -0400
+++ libadwaita-1.0.5.new/src/meson.build	2022-06-13 22:37:56.670269324 -0400
@@ -222,7 +222,7 @@
 ]
 
 glib_min_version = '>= 2.66'
-gtk_min_version = '>= 4.5.0'
+gtk_min_version = '>= 4.4.0'
 
 gio_dep = dependency('gio-2.0', version: glib_min_version)
 gtk_dep = dependency('gtk4', version: gtk_min_version)
