--- a/notebook/notebookapp.py    2023-12-20 18:04:52.352937385 -0800
+++ b/notebook/notebookapp.py    2023-12-20 18:06:22.564933788 -0800
@@ -1408,7 +1408,7 @@
             # and allow jupyter_server contents managers to pass
             # through. If jupyter_server is not installed, this class
             # will be ignored.
-            'jupyter_server.contents.services.managers.ContentsManager'
+            "jupyter_server.services.contents.managers.ContentsManager",
         ],
         config=True,
         help=_('The notebook manager class to use.')
--- a/notebook/traittypes.py    2023-12-20 18:08:44.904928112 -0800
+++ b/notebook/traittypes.py    2023-12-20 18:09:04.503927331 -0800
@@ -1,5 +1,6 @@
 import inspect
-from traitlets import ClassBasedTraitType, Undefined, warn
+from warnings import warn
+from traitlets import ClassBasedTraitType, Undefined
 
 # Traitlet's 5.x includes a set of utilities for building
 # description strings for objects. Traitlets 5.x does not
