diff -Naur wmudmount-2.2/make-chars.pl wmudmount-2.2.patched/make-chars.pl
--- wmudmount-2.2/make-chars.pl	2014-02-03 00:41:59.000000000 -0500
+++ wmudmount-2.2.patched/make-chars.pl	2018-02-19 16:09:56.660499062 -0500
@@ -52,7 +52,7 @@
 
 while(<IN>){
     $head.=$_;
-    if(/^static char \* ([a-zA-Z0-9_]+_xpm)\[\] = {/){
+    if(/^static char \* ([a-zA-Z0-9_]+_xpm)\[\] = \{/){
         $name=$1;
         last;
     }
