--- valkyrie-1.3.0/doc/Makefile.in.orig	2010-02-16 09:36:30.757843977 -0500
+++ valkyrie-1.3.0/doc/Makefile.in	2010-02-16 09:36:47.274614778 -0500
@@ -143,7 +143,7 @@
 build_alias = @build_alias@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(prefix)/doc/
+docdir = @docdir@
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host_alias = @host_alias@
--- valkyrie-1.3.0/doc/images/Makefile.in.orig	2010-02-16 09:40:41.232645924 -0500
+++ valkyrie-1.3.0/doc/images/Makefile.in	2010-02-16 09:40:58.332637251 -0500
@@ -134,7 +134,7 @@
 build_alias = @build_alias@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(prefix)/doc/images
+docdir = @docdir@/images
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host_alias = @host_alias@
