--- valkyrie-2.0.0/vk_config.pri.orig	2010-11-11 07:12:56.050000003 -0300
+++ valkyrie-2.0.0/vk_config.pri	2010-11-11 07:14:07.000000005 -0300
@@ -13,7 +13,7 @@
 # Set a default value for PREFIX (etc) in the generated makefiles
 # Note: ignore INSTALL_ROOT: it's a qt special, can't be set from here
 isEmpty(PREFIX) {
-  PREFIX = /usr/local
+  PREFIX = /usr
 }
 isEmpty(BINDIR) {
   BINDIR = $$PREFIX/bin
@@ -25,8 +25,8 @@
 ######################################################################
 # Install paths relative to $$PREFIX:
 target.path    = $$BINDIR
-doc.path       = $$DATADIR/$$PACKAGE/doc
-doc_imgs.path  = $$DATADIR/$$PACKAGE/doc/images
+doc.path       = $$DATADIR/doc/$$PACKAGE
+doc_imgs.path  = $$DATADIR/doc/$$PACKAGE/images
 
 ######################################################################
 # Qt modules
