diff -Naur bamf-0.5.6.orig/configure.ac bamf-0.5.6/configure.ac
--- bamf-0.5.6.orig/configure.ac	2022-02-17 19:16:55.346057503 +0100
+++ bamf-0.5.6/configure.ac	2023-02-04 19:24:42.506720000 +0100
@@ -106,18 +106,6 @@
 GTK_DOC_CHECK(1.0)
 
 ###########################
-# gtester2xunit checks    #
-###########################
-
-AC_PATH_PROG([PYTHON],[python3])
-AC_MSG_CHECKING(for gtester2xunit dependencies)
-if !($PYTHON -c "from lxml import etree" 2> /dev/null); then
-  AC_MSG_RESULT([no])
-  AC_MSG_ERROR([You need to install python3-lxml]);
-fi
-AC_MSG_RESULT([yes])
-
-###########################
 # Headless tests
 ###########################
 AC_ARG_ENABLE([headless-tests],
