diff -Naur inform-6.41-r2/Makefile inform-6.41-r2.patched/Makefile
--- inform-6.41-r2/Makefile	2022-11-21 19:31:13.000000000 -0500
+++ inform-6.41-r2.patched/Makefile	2023-01-18 15:22:43.134613101 -0500
@@ -178,7 +178,7 @@
 
 install-compiler: $(BINNAME)
 ifeq ($(wildcard $(DESTDIR)$(BINDIR)/$(BINNAME)),)
-	install -d -m $(DIRPERM) $(DESTDIR)$(BINDIR)
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(BINDIR)
 	install -c -m $(BINPERM) $(BINNAME) $(DESTDIR)$(BINDIR)
 	cd $(DESTDIR)$(BINDIR) ; ln -sf $(BINNAME) $(NAME)
 	@echo "** Done installing compiler."
@@ -218,24 +218,24 @@
 install-stdlib-all: install-stdlib install-stdinc install-stddemos install-stdtutor
 
 install-stdlib:
-	install -d -m $(DIRPERM) $(STDLIB)
-	install -c -m $(FILEPERM) $(wildcard ${STDSRC}/*) $(STDLIB)
-	@./fixlinks.sh $(STDLIB) $(STDLIB_LINKS)
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(STDLIB)
+	install -c -m $(FILEPERM) $(wildcard ${STDSRC}/*) $(DESTDIR)$(STDLIB)
+	@./fixlinks.sh $(DESTDIR)$(STDLIB) $(STDLIB_LINKS)
 	@echo "** Done installing Standard Library."
 
 install-stdinc:
-	install -d -m $(DIRPERM) $(STDINC)
-	install -c -m $(FILEPERM) $(wildcard ${STDINC_SRC}/*) $(STDINC)
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(STDINC)
+	install -c -m $(FILEPERM) $(wildcard ${STDINC_SRC}/*) $(DESTDIR)$(STDINC)
 	@echo "** Done installing Standard Library includes."
 
 install-stddemos: stddemos
-	install -d -m $(DIRPERM) $(STDDEMO)
-	install -c -m $(FILEPERM) $(wildcard ${STDDEMOS_SRC}/*) $(STDDEMO)
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(STDDEMO)
+	install -c -m $(FILEPERM) $(wildcard ${STDDEMOS_SRC}/*) $(DESTDIR)$(STDDEMO)
 	@echo "** Done installing Standard Library demos."
 
 install-stdtutor: stdtutor
-	install -d -m $(DIRPERM) $(STDTUTOR)
-	install -c -m $(FILEPERM) $(wildcard ${STDTUTOR_SRC}/*) $(STDTUTOR)
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(STDTUTOR)
+	install -c -m $(FILEPERM) $(wildcard ${STDTUTOR_SRC}/*) $(DESTDIR)$(STDTUTOR)
 	@echo "** Done installing Standard Library tutorials."
 
 
@@ -252,39 +252,41 @@
 
 install-punywrapper:
 	sed 's,^LIBPATH.*,LIBPATH=$(REAL_PREFIX)/share/$(NAME),g' $(PUNYINFORMSH) > $(PUNYINFORMSH)-scratch
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(BINDIR)
 	install -c -m $(BINPERM) $(PUNYINFORMSH)-scratch $(DESTDIR)$(BINDIR)/$(PUNYINFORMSH)
 	cd $(DESTDIR)$(BINDIR) ; ln -sf $(PUNYINFORMSH) $(PUNYINFORM)
 	cd $(DESTDIR)$(BINDIR) ; ln -sf $(PUNYINFORMSH) $(PUNYINFORM)-$(COMPVERSION)
 	rm -f $(PUNYINFORMSH)-scratch
 
 install-punylib:
-	install -d -m $(DIRPERM) $(PUNYLIB)
-	install -c -m $(FILEPERM) $(wildcard ${PUNYLIB_SRC}/*) $(PUNYLIB)
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(PUNYLIB)
+	install -c -m $(FILEPERM) $(wildcard ${PUNYLIB_SRC}/*) $(DESTDIR)$(PUNYLIB)
 	@echo "** Done installing PunyInform Library."
 
 install-punyhowto:
-	install -d -m $(DIRPERM) $(PUNYHOWTO)
-	install -c -m $(FILEPERM) $(wildcard ${PUNYHOWTO_SRC}/*) $(PUNYHOWTO)
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(PUNYHOWTO)
+	install -c -m $(FILEPERM) $(wildcard ${PUNYHOWTO_SRC}/*) $(DESTDIR)$(PUNYHOWTO)
 
 install-punytests:
-	install -d -m $(DIRPERM) $(PUNYTESTS)
-	install -c -m $(FILEPERM) $(wildcard ${PUNYTESTS_SRC}/*) $(PUNYTESTS)
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(PUNYTESTS)
+	install -c -m $(FILEPERM) $(wildcard ${PUNYTESTS_SRC}/*) $(DESTDIR)$(PUNYTESTS)
 
 install-punydocs:
-	install -d -m $(DIRPERM) $(PUNYDOCS)
-	install -d -m $(DIRPERM) $(PUNYDOCS1)
-	install -d -m $(DIRPERM) $(PUNYDOCS2)
-	install -d -m $(DIRPERM) $(PUNYDOCS3)
-	install -c -m $(FILEPERM) $(wildcard ${PUNYDOCS_SRC}/*pdf) $(PUNYDOCS)
-	install -c -m $(FILEPERM) $(wildcard ${PUNYDOCS_DIR1}/*) $(PUNYDOCS1)
-	install -c -m $(FILEPERM) $(wildcard ${PUNYDOCS_DIR2}/*) $(PUNYDOCS2)
-	install -c -m $(FILEPERM) $(wildcard ${PUNYDOCS_DIR3}/*) $(PUNYDOCS3)
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(PUNYDOCS)
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(PUNYDOCS1)
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(PUNYDOCS2)
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(PUNYDOCS3)
+	install -c -m $(FILEPERM) $(wildcard ${PUNYDOCS_SRC}/*pdf) $(DESTDIR)$(PUNYDOCS)
+	install -c -m $(FILEPERM) $(wildcard ${PUNYDOCS_DIR1}/*) $(DESTDIR)$(PUNYDOCS1)
+	install -c -m $(FILEPERM) $(wildcard ${PUNYDOCS_DIR2}/*) $(DESTDIR)$(PUNYDOCS2)
+	install -c -m $(FILEPERM) $(wildcard ${PUNYDOCS_DIR3}/*) $(DESTDIR)$(PUNYDOCS3)
 
 install-punymisc:
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(PUNY)
 	for file in $(PUNYMISC_FILES); do \
-		install -c -m $(FILEPERM) $(PUNYSRC)/$$file $(PUNY); \
+		install -c -m $(FILEPERM) $(PUNYSRC)/$$file $(DESTDIR)$(PUNY); \
 	done
-	install -c -m $(FILEPERM) $(wildcard ${PUNYSRC}/*.inf) $(PUNY)
+	install -c -m $(FILEPERM) $(wildcard ${PUNYSRC}/*.inf) $(DESTDIR)$(PUNY)
 
 install-puny:	install-punylib install-punywrapper install-punymisc install-punyhowto install-punytests install-punydocs
 
@@ -292,16 +294,17 @@
 #############################
 # Installing everything else
 install-manual:
-	install -d -m $(DIRPERM) $(MANDIR)
-	install -c -m $(FILEPERM) $(MANPAGE) $(MANDIR)
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(MANDIR)
+	install -c -m $(FILEPERM) $(MANPAGE) $(DESTDIR)$(MANDIR)
 
 install-contrib:
+	mkdir -p -m $(DIRPERM) $(DESTDIR)$(BINDIR)
 	install -c -m $(BINPERM) contrib/pblorb.pl $(DESTDIR)$(BINDIR)
 	install -c -m $(BINPERM) contrib/scanblorb.pl $(DESTDIR)$(BINDIR)
 	cd $(DESTDIR)$(BINDIR) ; ln -sf pblorb.pl pblorb
 	cd $(DESTDIR)$(BINDIR) ; ln -sf scanblorb.pl scanblorb
 
-install-strip: strip install
+install-strip: all strip install
 
 uninstall:
 	rm -f $(DESTDIR)$(BINDIR)/$(NAME)*
