diff -Naur inform-6.34-6.12.4-1/Makefile inform-6.34-6.12.4-1.patched/Makefile
--- inform-6.34-6.12.4-1/Makefile	2020-08-02 20:37:10.000000000 -0400
+++ inform-6.34-6.12.4-1.patched/Makefile	2020-11-05 00:32:19.772575474 -0500
@@ -67,7 +67,7 @@
 	$(CC) -o $(BINNAME) $(OBJECTS)
 	@echo "** Done building compiler."
 
-%.z5: %.inf
+%.z5: %.inf $(BINNAME)
 	$(PWD)/$(BINNAME) +lib $< $@
 
 demos:	$(BINNAME) $(DEMO_Z5)
@@ -115,6 +115,7 @@
 	install -c -m 644 $(wildcard ${TUTOR}/*) $(DESTDIR)$(TUTORDIR)
 
 install-contrib:
+	install -d -m 755 $(DESTDIR)$(BINDIR)
 	install -c -m 755 contrib/pblorb.pl $(DESTDIR)$(BINDIR)
 	install -c -m 755 contrib/scanblorb.pl $(DESTDIR)$(BINDIR)
 	cd $(DESTDIR)$(BINDIR) ; ln -s pblorb.pl pblorb
