From: Damián Viano <debian@damianv.com.ar>
Subject: Fixes segfault when $DISPLAY is not defined/can't be opened

diff -Nura yeahconsole-0.3.4/yeahconsole.c yeahconsole-0.3.4.r.bk/yeahconsole.c
--- yeahconsole-0.3.4/yeahconsole.c	2006-03-11 15:26:00.000000000 -0300
+++ yeahconsole-0.3.4.r.bk/yeahconsole.c	2006-07-01 17:49:14.000000000 -0300
@@ -107,7 +107,8 @@
     }
 
     if (!(dpy = XOpenDisplay(NULL))) {
-	fprintf(stderr, " can not open dpy %s", XDisplayName(NULL));
+	fprintf(stderr, "Can not open display %s\n", XDisplayName(NULL));
+	exit(1);
     }
     screen = DefaultScreen(dpy);
     root = RootWindow(dpy, screen);
