Description: Recognize Shift key, Meta, and Super key names
Author: Decklin Foster <decklin@red-bean.com>
Forwarded: mailto:knorke@phrat.de
Bug-Debian: https://bugs.debian.org/506062

--- a/yeahconsole.c
+++ b/yeahconsole.c
@@ -294,9 +294,11 @@
 	
     if (strstr(opt, "Control"))
 	modmask = modmask | ControlMask;
-    if (strstr(opt, "Alt"))
+    if (strstr(opt, "Shift"))
+	modmask = modmask | ShiftMask;
+    if (strstr(opt, "Meta") || strstr(opt, "Alt"))
 	modmask = modmask | Mod1Mask;
-    if (strstr(opt, "Win"))
+    if (strstr(opt, "Super") || strstr(opt, "Win"))
 	modmask = modmask | Mod4Mask;
     if (strstr(opt, "None"))
 	modmask = 0;
