--- luckybackup.pro.orig	2009-12-07 22:36:10.814594306 +0100
+++ luckybackup.pro	2009-12-08 00:18:03.065894763 +0100
@@ -18,16 +18,16 @@
 pixmap.path = /usr/share/pixmaps
 pixmap.files = menu/luckybackup.xpm menu/luckybackup.png
 
-documentation.path = /usr/share/doc/luckybackup
+documentation.path = /usr/doc/luckybackup-PROGVERSION
 documentation.files = manual
 
-manpage.path = /usr/share/man/man8
+manpage.path = /usr/man/man8
 manpage.files = manpage/luckybackup.8.gz
 
 translations.path = /usr/share/luckybackup
 translations.files = translations
 
-license.path = /usr/share/doc/luckybackup
+license.path = /usr/doc/luckybackup-PROGVERSION
 license.files = license
 
 INSTALLS += target menu debianmenu pixmap documentation manpage translations license
