--- HercStudio/HerculesExecutor.cpp	2011-07-13 03:13:06.000000000 -0300
+++ HercStudio_patched/HerculesExecutor.cpp	2012-08-09 15:34:30.000000000 -0300
@@ -28,6 +28,7 @@
 #include "NamedPipe.h"
 
 #include <cstdio>
+#include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
--- HercStudio/MainWindow.cpp	2011-07-13 03:13:06.000000000 -0300
+++ HercStudio_patched/MainWindow.cpp	2012-08-09 15:46:50.000000000 -0300
@@ -57,6 +57,7 @@
 #include <QTimer>
 
 #include <iostream>
+#include <unistd.h>
 #include "cerrno"
 #ifndef hFramework
 #include <err.h>
