From 42aeba19dbe79c6ebd3e860bff977595577ad90f Mon Sep 17 00:00:00 2001
From: orbea <orbea@riseup.net>
Date: Wed, 8 Jan 2020 12:12:34 -0800
Subject: [PATCH] gn: Fix build with newer gn versions.

---
 src/cpp/BUILD.gn | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/cpp/BUILD.gn b/src/cpp/BUILD.gn
index 58da775..b76d236 100644
--- a/src/cpp/BUILD.gn
+++ b/src/cpp/BUILD.gn
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 static_library("procyon-cpp") {
-  sources = [
+  public = [
     "include/pn/arg",
     "include/pn/array",
     "include/pn/data",
@@ -23,6 +23,9 @@ static_library("procyon-cpp") {
     "include/pn/output",
     "include/pn/string",
     "include/pn/value",
+  ]
+
+  sources = [
     "src/array.cpp",
     "src/common.hpp",
     "src/data.cpp",
