diff -upr compiz-plugins-main-0.8.8-orig/include/compiz-animation.h compiz-plugins-main-0.8.8/include/compiz-animation.h
--- compiz-plugins-main-0.8.8-orig/include/compiz-animation.h	2011-04-21 12:20:09.000000000 +0200
+++ compiz-plugins-main-0.8.8/include/compiz-animation.h	2015-05-13 11:31:11.970670459 +0200
@@ -215,7 +215,7 @@ typedef struct _AnimBaseFunctions {
 
 #define OPTION_GETTERS(extensionBaseFunctions,				\
 		       extensionPluginInfo, firstEffectOption)		\
-static inline CompOptionValue *						\
+extern inline CompOptionValue *						\
 animGetOptVal (CompWindow *w,						\
 	       int optionId)						\
 {									\
@@ -223,35 +223,35 @@ animGetOptVal (CompWindow *w,						\
     	(w, (extensionPluginInfo), optionId - (firstEffectOption));	\
 }						\
 						\
-inline Bool					\
+extern inline Bool				\
 animGetB (CompWindow *w,			\
 	  int optionId)				\
 {						\
     return animGetOptVal (w, optionId)->b;	\
 }						\
 						\
-inline int					\
+extern inline int				\
 animGetI (CompWindow *w,			\
 	  int optionId)				\
 {						\
     return animGetOptVal (w, optionId)->i;	\
 }						\
 						\
-inline float					\
+extern inline float				\
 animGetF (CompWindow *w,			\
 	  int optionId)				\
 {						\
     return animGetOptVal (w, optionId)->f;	\
 }						\
 						\
-inline char *					\
+extern inline char *				\
 animGetS (CompWindow *w,			\
 	  int optionId)				\
 {						\
     return animGetOptVal (w, optionId)->s;	\
 }						\
 						\
-inline unsigned short *				\
+extern inline unsigned short *			\
 animGetC (CompWindow *w,			\
 	  int optionId)				\
 {						\
@@ -260,23 +260,23 @@ animGetC (CompWindow *w,			\
 
 #define OPTION_GETTERS_HDR			\
 						\
-inline Bool					\
+extern inline Bool				\
 animGetB (CompWindow *w,			\
 	  int optionId);			\
 						\
-inline int					\
+extern inline int				\
 animGetI (CompWindow *w,			\
 	  int optionId);			\
 						\
-inline float					\
+extern inline float				\
 animGetF (CompWindow *w,			\
 	  int optionId);			\
 						\
-inline char *					\
+extern inline char *				\
 animGetS (CompWindow *w,			\
 	  int optionId);			\
 						\
-inline unsigned short *				\
+extern inline unsigned short *			\
 animGetC (CompWindow *w,			\
 	  int optionId);
 
diff -upr compiz-plugins-main-0.8.8-orig/src/animation/animation.c compiz-plugins-main-0.8.8/src/animation/animation.c
--- compiz-plugins-main-0.8.8-orig/src/animation/animation.c	2011-04-21 12:20:09.000000000 +0200
+++ compiz-plugins-main-0.8.8/src/animation/animation.c	2015-05-13 11:34:57.181935016 +0200
@@ -742,7 +742,7 @@ defaultUpdateWindowTransform (CompWindow
 }
 
 // Apply transform to wTransform
-inline void
+extern inline void
 applyTransform (CompTransform *wTransform,
 		CompTransform *transform)
 {
diff -upr compiz-plugins-main-0.8.8-orig/src/animation/animation-internal.h compiz-plugins-main-0.8.8/src/animation/animation-nternal.h
--- compiz-plugins-main-0.8.8-orig/src/animation/animation-internal.h	2011-04-21 12:20:09.000000000 +0200
+++ compiz-plugins-main-0.8.8/src/animation/animation-internal.h	2015-05-13 11:34:06.853428950 +0200
@@ -429,7 +429,7 @@ applyPerspectiveSkew (CompOutput *output
 		      CompTransform *transform,
 		      Point *center);
 
-inline void
+extern inline void
 applyTransform (CompTransform *wTransform,
 		CompTransform *transform);
 
@@ -616,7 +616,7 @@ fxZoomInit (CompWindow * w);
 void
 applyZoomTransform (CompWindow * w);
 
-void
+extern inline void
 getZoomCenterScale (CompWindow *w,
 		    Point *pCurCenter, Point *pCurScale);
 
diff -upr compiz-plugins-main-0.8.8-orig/src/animation/zoomside.c compiz-plugins-main-0.8.8/src/animation/zoomside.c
--- compiz-plugins-main-0.8.8-orig/src/animation/zoomside.c	2011-04-21 12:20:09.000000000 +0200
+++ compiz-plugins-main-0.8.8/src/animation/zoomside.c	2015-05-13 11:36:15.021717714 +0200
@@ -303,7 +303,7 @@ getZoomCenterScaleFull (CompWindow *w,
 	*pRotateProgress = rotateProgress;
 }
 
-inline void
+extern inline void
 getZoomCenterScale (CompWindow *w,
 		    Point *pCurCenter, Point *pCurScale)
 {

