diff --git a/src/mag/mag.c b/src/mag/mag.c
index f3c8fc3..6aac3a7 100644
--- a/src/mag/mag.c
+++ b/src/mag/mag.c
@@ -20,6 +20,7 @@
  *
  */
 
+#include <stdlib.h>
 #include <math.h>
 #include <string.h>
 
diff --git a/src/mousepoll/mousepoll.c b/src/mousepoll/mousepoll.c
index d009ed2..eb513e1 100644
--- a/src/mousepoll/mousepoll.c
+++ b/src/mousepoll/mousepoll.c
@@ -20,6 +20,7 @@
  *
  */
 
+#include <stdlib.h>
 #include <compiz-core.h>
 
 #include "compiz-mousepoll.h"
diff --git a/src/opacify/opacify.c b/src/opacify/opacify.c
index a4fb86c..3b96208 100644
--- a/src/opacify/opacify.c
+++ b/src/opacify/opacify.c
@@ -21,6 +21,7 @@
  *
  */
 
+#include <stdlib.h>
 #include <compiz-core.h>
 #include "opacify_options.h"
 
diff --git a/src/scaleaddon/scaleaddon.c b/src/scaleaddon/scaleaddon.c
index 8029fa4..d97fe11 100644
--- a/src/scaleaddon/scaleaddon.c
+++ b/src/scaleaddon/scaleaddon.c
@@ -24,6 +24,7 @@
  *
  */
 
+#include <stdlib.h>
 #include <math.h>
 #include <string.h>
 #include <X11/Xatom.h>
diff --git a/src/session/session.c b/src/session/session.c
index 7d1d6e9..bac2c0a 100644
--- a/src/session/session.c
+++ b/src/session/session.c
@@ -29,6 +29,7 @@
 
 #include <compiz-core.h>
 
+#include <stdlib.h>
 #include <string.h>
 #include <errno.h>
 #include <pwd.h>
diff --git a/src/vpswitch/vpswitch.c b/src/vpswitch/vpswitch.c
index 92faf48..3411671 100644
--- a/src/vpswitch/vpswitch.c
+++ b/src/vpswitch/vpswitch.c
@@ -27,6 +27,7 @@
  */
 
 #include <compiz-core.h>
+#include <stdlib.h>
 #include <string.h>
 #include <X11/keysymdef.h>
 #include "vpswitch_options.h"
diff --git a/src/winrules/winrules.c b/src/winrules/winrules.c
index d64623e..8a47b0a 100644
--- a/src/winrules/winrules.c
+++ b/src/winrules/winrules.c
@@ -21,6 +21,7 @@
 
 #include <compiz-core.h>
 
+#include <stdlib.h>
 #include <X11/Xatom.h>
 
 #define WINRULES_SCREEN_OPTION_SKIPTASKBAR_MATCH  0
diff --git a/src/workarounds/workarounds.c b/src/workarounds/workarounds.c
index 125c072..af92b26 100644
--- a/src/workarounds/workarounds.c
+++ b/src/workarounds/workarounds.c
@@ -20,6 +20,7 @@
  * This plug-in for Marco-like workarounds.
  */
 
+#include <stdlib.h>
 #include <string.h>
 #include <limits.h>
 
