diff -crB src/Satellite.cpp src.new//Satellite.cpp
*** src/Satellite.cpp	2010-06-24 08:32:19.923302138 +0300
--- src.new//Satellite.cpp	2010-06-24 08:31:30.857301301 +0300
***************
*** 1,4 ****
--- 1,5 ----
  #include <clocale>
+ #include <cstdio>
  #include <cstring>
  #include <iostream>
  using namespace std;
diff -crB src/Separation.cpp src.new//Separation.cpp
*** src/Separation.cpp	2010-06-24 08:32:19.911301998 +0300
--- src.new//Separation.cpp	2010-06-24 08:31:34.542300533 +0300
***************
*** 1,4 ****
--- 1,5 ----
  #include <cmath>
+ #include <cstdio>
  #include <iostream>
  
  #include "Separation.h"
