diff -Naur pianobar-2013.09.15-orig/Makefile pianobar-2013.09.15/Makefile
--- pianobar-2013.09.15-orig/Makefile	2013-09-15 09:54:04.000000000 -0400
+++ pianobar-2013.09.15/Makefile	2013-09-27 23:54:19.406220780 -0400
@@ -7,20 +7,7 @@
 MANDIR:=${PREFIX}/share/man
 DYNLINK:=0
 
-# Respect environment variables set by user; does not work with :=
-ifeq (${CFLAGS},)
-	CFLAGS=-O2 -DNDEBUG
-endif
-ifeq (${CC},cc)
-	OS := $(shell uname)
-	ifeq (${OS},Darwin)
-		CC=gcc -std=c99
-	else ifeq (${OS},FreeBSD)
-		CC=cc -std=c99
-	else
-		CC=c99
-	endif
-endif
+CFLAGS+=-std=c99 -O2 -DNDEBUG
 
 PIANOBAR_DIR:=src
 PIANOBAR_SRC:=\
