--- CMakeLists.txt.orig	2014-08-08 07:24:01.864876514 -0700
+++ CMakeLists.txt	2014-08-08 07:24:51.980873088 -0700
@@ -221,7 +221,10 @@
 ####### install
 # slim
 install(TARGETS slim RUNTIME DESTINATION bin)
-install(TARGETS slimlock RUNTIME DESTINATION bin)
+## we have to comment this out or else
+## build will fail on slackware as pam
+## is required for slimlock
+#install(TARGETS slimlock RUNTIME DESTINATION bin)
 
 if (BUILD_SHARED_LIBS)
 	set_target_properties(libslim PROPERTIES
