--- slim.conf.orig	2008-12-05 13:47:44.512494201 -0600
+++ slim.conf	2008-12-05 13:51:51.449743538 -0600
@@ -8,7 +8,7 @@
 halt_cmd            /sbin/shutdown -h now
 reboot_cmd          /sbin/shutdown -r now
 console_cmd         /usr/bin/xterm -C -fg white -bg black +sb -T "Console login" -e /bin/sh -c "/bin/cat /etc/issue; exec /bin/login"
-#suspend_cmd        /usr/sbin/suspend
+suspend_cmd        /usr/sbin/pm-suspend
 
 # Full path to the xauth binary
 xauth_path         /usr/bin/xauth 
