--- slim.conf.orig	2014-08-07 21:04:23.591900466 -0700
+++ slim.conf	2014-08-07 21:05:44.455894937 -0700
@@ -9,6 +9,8 @@
 reboot_cmd          /sbin/shutdown -r now
 console_cmd         /usr/bin/xterm -C -fg white -bg black +sb -T "Console login" -e /bin/sh -c "/bin/cat /etc/issue; exec /bin/login"
 #suspend_cmd        /usr/sbin/suspend
+## slackware suspend command
+suspend_cmd        /usr/sbin/pm-suspend
 
 # Full path to the xauth binary
 xauth_path         /usr/bin/xauth 
