--- Makefile.orig	2008-03-05 13:51:10.000000000 +0200
+++ Makefile	2008-03-05 13:58:47.000000000 +0200
@@ -6,26 +6,14 @@
 	@echo "** done **"
 
 esvn:
-	qmake esvn.pro
 	make -f esvn.mak
 	
 esvn-diff-wrapper:
-	qmake esvn-diff-wrapper.pro
 	make -f esvn-diff-wrapper.mak
 
 install:
 	make -f esvn.mak install
 	make -f esvn-diff-wrapper.mak install
-	mkdir -p /usr/share/doc/esvn
-	cp -f -r html-docs /usr/share/doc/esvn
-
-#cp -f esvn.pdf /usr/share/doc/esvn
-
-install-doc:	
-	mkdir -p /usr/share/doc/esvn
-	cp -f -r html-docs /usr/share/doc/esvn
-
-#cp -f esvn.pdf /usr/share/doc/esvn
 
 clean:
 	make -f esvn.mak clean
--- esvn.pro.orig	2008-03-05 13:51:22.000000000 +0200
+++ esvn.pro	2008-03-05 13:59:15.000000000 +0200
@@ -253,7 +253,6 @@
   VER=$$system(cat VERSION)
 
   target.path  = /usr/bin
-  target.files = esvn
   INSTALLS += target
 
   icons.path = /usr/share/pixmaps
--- esvn-diff-wrapper.pro.orig	2008-03-05 13:51:32.000000000 +0200
+++ esvn-diff-wrapper.pro	2008-03-05 14:01:38.000000000 +0200
@@ -13,7 +13,6 @@
 # Target is an application:  
 
 MAKEFILE = esvn-diff-wrapper.mak
-target.files = esvn-diff-wrapper
 target.path = /usr/bin 
 INSTALLS += target 
 unix{
