--- gl2ps-1.3.5-source/CMakeLists.txt	2009-10-16 16:14:07.000000000 -0300
+++ gl2ps-1.3.5-source_patched/CMakeLists.txt	2011-08-10 17:24:43.000000000 -0300
@@ -113,7 +113,7 @@
     set_target_properties(shared PROPERTIES COMPILE_FLAGS "-DGL2PSDLL -DGL2PSDLL_EXPORTS")
   endif(MSVC)
 
-  install(TARGETS lib shared DESTINATION lib)
+  install(TARGETS lib shared DESTINATION lib${LIB_SUFFIX})
 endif(OPENGL_FOUND)
 
 if(WIN32)
