diff --git a/glib/glue/list.c b/glib/glue/list.c
index dd3f0de..377df76 100644
--- a/glib/glue/list.c
+++ b/glib/glue/list.c
@@ -20,7 +20,7 @@
  */
 
 
-#include <glib/glist.h>
+#include <glib.h>
 
 /* Forward declarations */
 gpointer gtksharp_list_get_data (GList *l);
diff --git a/glib/glue/slist.c b/glib/glue/slist.c
index 80f5f17..ddd447c 100644
--- a/glib/glue/slist.c
+++ b/glib/glue/slist.c
@@ -20,7 +20,7 @@
  */
 
 
-#include <glib/gslist.h>
+#include <glib.h>
 
 /* Forward declarations */
 gpointer gtksharp_slist_get_data (GSList *l);
diff --git a/glib/glue/thread.c b/glib/glue/thread.c
index 4fcf8c6..3655cfc 100644
--- a/glib/glue/thread.c
+++ b/glib/glue/thread.c
@@ -20,7 +20,7 @@
  */
 
 
-#include <glib/gthread.h>
+#include <glib.h>
 
 gboolean glibsharp_g_thread_supported (void);
 
