--- configure.in~	2007-06-30 08:03:35.000000000 -0500
+++ configure.in	2007-06-30 08:21:46.000000000 -0500
@@ -8,15 +8,16 @@
 ##
 
 dnl --------------------------------------------------------------------
-dnl Made the prerequisite autoconf 2.13, lower it to each version we
+dnl Made the prerequisite autoconf 2.60, lower it to each version we
 dnl find to work correctly with this script
 dnl --------------------------------------------------------------------
-AC_PREREQ(2.13)
+AC_PREREQ(2.60)
 
 dnl --------------------------------------------------------------------
 dnl Process this file with autoconf to produce a configure script
 dnl --------------------------------------------------------------------
-AC_INIT(consts.h)
+AC_INIT
+AC_CONFIG_SRCDIR([consts.h])
 AM_INIT_AUTOMAKE(tuxnes,0.75)
 #AM_INIT_AUTOMAKE(tuxnes,devel)
 
@@ -28,16 +29,10 @@
 AC_CANONICAL_HOST
 
 dnl --------------------------------------------------------------------
-dnl Don't assume we want high level optimisations, it is known to
-dnl produce broken code on certain architectures using certain compilers
-dnl --------------------------------------------------------------------
-CFLAGS="-O"
-AC_SUBST(CFLAGS)
-
-dnl --------------------------------------------------------------------
 dnl Check for programs which we might need
 dnl --------------------------------------------------------------------
 AC_PROG_CC
+AM_PROG_AS
 AC_PROG_INSTALL
 
 dnl --------------------------------------------------------------------
@@ -175,7 +170,7 @@
   *-freebsd*)
   OLDFLAGS="$LDFLAGS"
   LDFLAGS="-L/usr/local/lib"
-  AC_CHECK_LIB(gnugetopt, getopt, [AC_DEFINE(HAVE_LIBGNUGETOPT) GNUGETOPT_LIB="-lgnugetopt"])
+  AC_CHECK_LIB(gnugetopt, getopt, [AC_DEFINE([HAVE_LIBGNUGETOPT], 1, [Define to 1 if you have the `gnugetopt' library (-lgnugetopt).]) GNUGETOPT_LIB="-lgnugetopt"])
   if test -n "$GNUGETOPT_LIB"; then
   	LDFLAGS="$OLDFLAGS -L/usr/local/lib -lgnugetopt"
 	AC_SUBST(LDFLAGS)
@@ -209,4 +204,5 @@
 AC_TYPE_SIGNAL
 AC_CHECK_FUNCS(gettimeofday strtod strtoul)
 
-AC_OUTPUT(Makefile)
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
