--- Makefile.old	2019-12-21 13:44:12.414553374 -0600
+++ Makefile	2019-12-21 14:12:56.166581253 -0600
@@ -2,8 +2,8 @@
 
 SHELL   = /bin/sh
 PROJECT = lpsk31
-BINDIR  = /usr/local/bin
-DOCDIR  = /usr/local/share/doc/
+BINDIR  = /usr/bin
+DOCDIR  = /usr/share/doc/
 CC      = gcc -Wall -march=native -O2 -D_FORTIFY_CODE=2
 
 objects = cat.o main.o phase.o pskcodec.o screen.o sound.o
@@ -20,10 +20,7 @@
 sound.o : sound.h
 
 install : $(PROJECT)
-	install -m 755 --strip $(PROJECT) $(BINDIR)
-	install -m 644 doc/$(PROJECT).html $(DOCDIR)
-	if test ! -d $(HOME)/$(PROJECT); then mkdir $(HOME)/$(PROJECT); fi
-	install -m 644 default/$(PROJECT)rc $(HOME)/$(PROJECT)
+	install -m 755 --strip $(PROJECT) $(DESTDIR)$(BINDIR)
 
 uninstall :
 	if test -d $(HOME)/$(PROJECT); then rm -r $(HOME)/$(PROJECT); fi
