--- Makefile.Linux.orig	2019-07-19 16:58:43.228052408 +0200
+++ Makefile.Linux	2019-07-19 17:00:32.528212655 +0200
@@ -42,16 +42,16 @@
 DEFS += -DUSE_CDROM
 
 # optional: avifile
-DEFS += -DUSE_AVIFILE
-INCS += `avifile-config --cflags`
-LIBS += `avifile-config --libs`
-TARGET += simple_aviplay$(EXESUFFIX)
-EXT_OBJS += AVIWrapper$(OBJSUFFIX)
+#DEFS += -DUSE_AVIFILE
+#INCS += `avifile-config --cflags`
+#LIBS += `avifile-config --libs`
+#TARGET += simple_aviplay$(EXESUFFIX)
+#EXT_OBJS += AVIWrapper$(OBJSUFFIX)
 
 # optional: lua
 DEFS += -DUSE_LUA
-INCS += -I/usr/include/lua5.1
-LIBS += -llua5.1
+INCS += -I/usr/include
+LIBS += -llua
 EXT_OBJS += LUAHandler$(OBJSUFFIX)
 
 # optional: force screen width for PDA
