diff -Naur twitter-1.03.orig/Twitter.pm twitter-1.03/Twitter.pm
--- twitter-1.03.orig/Twitter.pm	2008-06-16 23:39:19.000000000 +0200
+++ twitter-1.03/Twitter.pm	2009-04-06 16:30:59.000000000 +0200
@@ -10,8 +10,8 @@
 # This package implements a twitter object that is
 # used to connect to twitter, and "do stuff"
 
-my $twitter_post = q{http://twitter.com/statuses/update.json};
-my $twitter_friends_timeline = q{http://twitter.com/statuses/friends_timeline.rss};
+my $twitter_post = q{https://twitter.com/statuses/update.json};
+my $twitter_friends_timeline = q{https://twitter.com/statuses/friends_timeline.rss};
 
 # Constructor.  
 # new ( <user>, <pass>, [cookie jar] )
@@ -39,8 +39,8 @@
 	$self->{ua}->timeout(10);
 
 	# Basic auth is for the win.
-	$self->{ua}->credentials( "www.twitter.com:80", "Twitter API", $u, $p );
-	$self->{ua}->credentials( "twitter.com:80", "Twitter API", $u, $p );
+	$self->{ua}->credentials( "www.twitter.com:443", "Twitter API", $u, $p );
+	$self->{ua}->credentials( "twitter.com:443", "Twitter API", $u, $p );
 		
 	# It's fun to store our cookies.
 	$self->{ua}->cookie_jar( HTTP::Cookies->new( 
diff -Naur twitter-1.03.orig/twitter.pl twitter-1.03/twitter.pl
--- twitter-1.03.orig/twitter.pl	2008-07-09 20:38:39.000000000 +0200
+++ twitter-1.03/twitter.pl	2009-04-06 16:30:25.000000000 +0200
@@ -12,7 +12,6 @@
 # Licensed under the GNU GPL v2.
 
 use strict;
-use lib '/home/gabriel/lib';
 use Data::Dumper;
 use Getopt::Std;
 use Twitter;
