From 433c41f4a23130a079ed6e681e7016c0b187fb61 Mon Sep 17 00:00:00 2001
From: mancha <mancha1@hush.com>
Date: Thu, 26 Sep 2013
Subject: Use appropriate path on x86_64 systems

 tsocks |   14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

--- a/tsocks	2013-09-26
+++ b/tsocks	2013-09-26
@@ -46,14 +46,14 @@ case "$1" in
 	on)
 		if [ -z "$LD_PRELOAD" ]
 			then
-				export LD_PRELOAD="/usr/lib/libtsocks.so"
+				export LD_PRELOAD="/usr/lib64/libtsocks.so"
 			else
-				echo $LD_PRELOAD | grep -q "/usr/lib/libtsocks\.so" || \
-				export LD_PRELOAD="/usr/lib/libtsocks.so $LD_PRELOAD"
+				echo $LD_PRELOAD | grep -q "/usr/lib64/libtsocks\.so" || \
+				export LD_PRELOAD="/usr/lib64/libtsocks.so $LD_PRELOAD"
 		fi
 	;;
 	off)
-		export LD_PRELOAD=`echo -n $LD_PRELOAD | sed 's/\/usr\/lib\/libtsocks.so \?//'`
+		export LD_PRELOAD=`echo -n $LD_PRELOAD | sed 's/\/usr\/lib64\/libtsocks.so \?//'`
 		if [ -z "$LD_PRELOAD" ]
 			then
 				unset LD_PRELOAD
@@ -68,10 +68,10 @@ case "$1" in
 	*)
 		if [ -z "$LD_PRELOAD" ]
 		then
-			export LD_PRELOAD="/usr/lib/libtsocks.so"
+			export LD_PRELOAD="/usr/lib64/libtsocks.so"
 		else
-			echo $LD_PRELOAD | grep -q "/usr/lib/libtsocks\.so" || \
-			export LD_PRELOAD="/usr/lib/libtsocks.so $LD_PRELOAD"
+			echo $LD_PRELOAD | grep -q "/usr/lib64/libtsocks\.so" || \
+			export LD_PRELOAD="/usr/lib64/libtsocks.so $LD_PRELOAD"
 		fi
 
 		if [ $# = 0 ]
