--- radare2-0.10.5/shlr/Makefile.orig	2016-09-02 18:49:28.154542799 +0700
+++ radare2-0.10.5/shlr/Makefile	2016-09-02 18:53:09.710545379 +0700
@@ -219,8 +219,9 @@
 	rm -rf capstone
 	mv capstone-$(CS_VER) capstone
 ifeq ($(CS_PATCHES),1)
+        echo "no capstone installed"
 	#cd capstone ; for PATCH in ../capstone-patches/* ; do patch -tp1 < $$PATCH ; done
-	cd capstone ; for PATCH in ../capstone-patches/* ; do patch -p1 < $$PATCH ; done
+	#cd capstone ; for PATCH in ../capstone-patches/* ; do patch -p1 < $$PATCH ; done
 endif
 	mkdir -p capstone/include/capstone
 	cp -rf capstone/include/*.h capstone/include/capstone
@@ -234,9 +235,10 @@
 capstone-sync:
 	"$(SHELL)" capstone.sh "${CS_URL}" "${CS_BRA}" "${CS_TIP}" "${CS_REV}"
 ifeq ($(CS_PATCHES),1)
-	-cd capstone ; $(GIT) reset --hard
+	#-cd capstone ; $(GIT) reset --hard
 	#cd capstone ; for PATCH in ../capstone-patches/* ; do patch -tp1 < $$PATCH ; done
-	cd capstone ; for PATCH in ../capstone-patches/* ; do patch -p1 < $$PATCH ; done
+	#cd capstone ; for PATCH in ../capstone-patches/* ; do patch -p1 < $$PATCH ; done
+	echo "no capstone installed"
 endif
 
 .PHONY: capstone
