diff -Naur tap-reverbed-r0/src/file_io.c tap-reverbed-r0.patched/src/file_io.c
--- tap-reverbed-r0/src/file_io.c	2004-06-18 07:19:01.000000000 -0400
+++ tap-reverbed-r0.patched/src/file_io.c	2015-11-04 16:11:02.000000000 -0500
@@ -62,6 +62,8 @@
 }
 
 
+/* default config, used as fallback if ~/.reverbed not found */
+#define SYS_REVERBED "/etc/reverbed/reverbed.conf"
 
 REVTYPE *
 parse_input_file(void) {
@@ -85,8 +87,11 @@
 		home = ".";
 	sprintf(path, "%s/%s", home, NAME_REVERBED);
 	if ((infile = fopen(path,"rt")) == NULL) {
-		fprintf(stderr, "Unable to open %s\n", path);
-		return(NULL);
+		fprintf(stderr, "Unable to open %s, trying %s\n", path, SYS_REVERBED);
+		if ((infile = fopen(SYS_REVERBED ,"rt")) == NULL) {
+			fprintf(stderr, "Unable to open %s", SYS_REVERBED);
+			return(NULL);
+		}
 	}
 
 	if ((root = malloc(sizeof(REVTYPE))) == NULL) {
