--- ember-0.7.2/configure.old	2016-02-06 13:43:10.960997615 +0700
+++ ember-0.7.2/configure	2016-02-06 13:43:40.090962492 +0700
@@ -18020,7 +18020,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $boost_cv_lib_version" >&5
 $as_echo "$boost_cv_lib_version" >&6; }
     # e.g. "134" for 1_34_1 or "135" for 1_35
-    boost_major_version=`echo "$boost_cv_lib_version" | sed 's/_//;s/_.*//'`
+    boost_major_version=159
     case $boost_major_version in #(
       '' | *[!0-9]*)
         as_fn_error $? "invalid value: boost_major_version=$boost_major_version" "$LINENO" 5
