--- promise-1.2.0.tm.orig	2026-07-13 19:54:17.563172461 +1000
+++ promise-1.2.0.tm	2026-07-13 20:48:58.102345363 +1000
@@ -1282,16 +1282,7 @@
             }
         }
         install {
-            # Install in first native file system that exists on search path
-            foreach path [tcl::tm::path list] {
-                if {[lindex [file system $path] 0] eq "native"} {
-                    set dir $path
-                    if {[file isdirectory $path]} {
-                        break
-                    }
-                    # Else keep looking
-                }
-            }
+            set dir [string cat [lindex $::argv 1] [file join [tcl::pkgconfig get libdir,runtime] tcl8 8.6]]
             if {![file exists $dir]} {
                 file mkdir $dir
             }
