--- promise-1.1.0.tm.orig	2020-07-27 12:00:27.771781329 -0500
+++ promise-1.1.0.tm	2020-07-27 12:03:12.543787263 -0500
@@ -1281,7 +1281,8 @@
             }
         }
         install {
-            set dir [file join [tcl::pkgconfig get libdir,runtime] tcl8 8.6]
+            set dir [string cat [lindex $::argv 1] [file join [tcl::pkgconfig get libdir,runtime] tcl8 8.6]]
+            file mkdir $dir
             if {[file extension $filename] eq ".tm"} {
                 # We already are a .tm with version number
                 set target $filename
