diff -Nur wmii-3.6.orig/config.mk wmii-3.6/config.mk
--- wmii-3.6.orig/config.mk	2007-11-16 07:59:15.000000000 -0600
+++ wmii-3.6/config.mk	2009-08-03 01:14:13.179886673 -0500
@@ -10,11 +10,12 @@
 
 # Includes and libs
 INCPATH = .:${ROOT}/include:${INCLUDE}:/usr/include
-LIBS = -L/usr/lib -lc -L${ROOT}/lib
+LIBS = -L${LIBDIR} -lc -L${ROOT}/lib
 
 # Flags
 include ${ROOT}/mk/gcc.mk
-CFLAGS += -g -O0 -DIXPlint
+OPT ?= -O0
+CFLAGS += -g ${OPT} -DIXPlint
 LDFLAGS += -g ${LIBS}
 STATIC = -static
 MKDEP = cpp -M
@@ -29,8 +30,8 @@
 AWKPATH = $$(which awk)
 P9PATHS = ${PLAN9}:"'$${HOME}/plan9'":/usr/local/plan9:/usr/local/9:/opt/plan9:/opt/9:/usr/plan9:/usr/9
 
-INCX11 = -I/usr/X11R6/include
-LIBX11 = -L/usr/X11R6/lib -lX11
+INCX11 = -I${PREFIX}include
+LIBX11 = -L${LIBDIR}/X11 -lX11
 LIBICONV = # Leave blank if your libc includes iconv (glibc does)
 LIBIXP = ${ROOT}/libixp/libixp.a
 LIBIXP = ${LIBDIR}/libixp.a
