diff -Naur wmii+ixp-3.9.2.orig/mk/hdr.mk wmii+ixp-3.9.2/mk/hdr.mk
--- wmii+ixp-3.9.2.orig/mk/hdr.mk	2010-06-10 09:24:03.000000000 +0200
+++ wmii+ixp-3.9.2/mk/hdr.mk	2011-11-15 10:59:19.110546818 +0100
@@ -37,14 +37,8 @@
 
 PACKAGES = 2>/dev/null
 
-# and this:
-# Try to find a sane shell. /bin/sh is a last resort, because it's
-# usually bash on Linux, which means it's painfully slow.
-BINSH := $(shell \
-	   if [ -x /bin/dash ]; then echo /bin/dash; \
-	   elif [ -x /bin/ksh ]; then echo /bin/ksh; \
-	   else echo /bin/sh; fi)
-BINSH != echo /bin/sh
+# This is Slackware
+BINSH=/bin/sh
 
 include $(ROOT)/config.mk
 
