diff -Naur wmii+ixp-3.9.2.orig/config.mk wmii+ixp-3.9.2/config.mk
--- wmii+ixp-3.9.2.orig/config.mk	2010-06-10 09:24:04.000000000 +0200
+++ wmii+ixp-3.9.2/config.mk	2011-11-15 10:57:47.850346147 +0100
@@ -6,12 +6,12 @@
   MAN = $(PREFIX)/share/man
   DOC = $(PREFIX)/share/doc/wmii
   ETC = $(PREFIX)/etc
-  LIBDIR = $(PREFIX)/lib
+  LIBDIR = $(PREFIX)/lib$(LIBDIRSUFFIX)
   INCLUDE = $(PREFIX)/include
 
 # Includes and libs
 INCLUDES = -I. -I$(ROOT)/include -I$(INCLUDE) -I/usr/include
-LIBS = -L$(ROOT)/lib -L/usr/lib
+LIBS = -L$(ROOT)/lib$(LIBDIRSUFFIX) -L/usr/lib$(LIBDIRSUFFIX)
 
 TERMINAL = xterm
 
