--- httping-1.4.1/str.c	2010-01-10 15:35:22.000000000 -0500
+++ str.c	2010-07-08 00:20:04.633607710 -0400
@@ -19,13 +19,13 @@
 #include "utils.h"
 
 #ifndef _GNU_SOURCE
-char *strndup(char *in, int size)
-{
-	char *out = mymalloc(size + 1, "strndup");
+//char *strndup(char *in, int size)
+//{
+//	char *out = mymalloc(size + 1, "strndup");
 
-	memcpy(out, in, size);
-	out[size] = 0x00;
+//	memcpy(out, in, size);
+//	out[size] = 0x00;
 
-	return out;
-}
+//	return out;
+//}
 #endif
