diff -Nur snack2.2.10/generic/jkFormatMP3.c snack2.2.10.new/generic/jkFormatMP3.c
--- snack2.2.10/generic/jkFormatMP3.c	2005-12-14 11:29:38.000000000 +0000
+++ snack2.2.10.new/generic/jkFormatMP3.c	2011-01-01 01:24:04.723341697 +0000
@@ -29,6 +29,7 @@
 #include "snack.h"
 #include "jkFormatMP3.h"
 #include <string.h>
+#include <math.h>
 #define FRAS2(is,a) ((is) > 0 ? t_43[(is)]*(a):-t_43[-(is)]*(a))
 #define MAXFRAMESIZE 2106  /* frame size starting at header */
 #define roundf(x) (floor((x)+(float )0.5f))
