diff --git a/trunk/src/ladspa/ladspa_guitarix.cpp b/trunk/src/ladspa/ladspa_guitarix.cpp
index 1fccffb3..85c3a766 100644
--- a/trunk/src/ladspa/ladspa_guitarix.cpp
+++ b/trunk/src/ladspa/ladspa_guitarix.cpp
@@ -16,8 +16,6 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-#define always_inline inline __attribute__((always_inline))
-
 #include <stdlib.h>
 #include <string.h>
 #include <ladspa.h>
@@ -34,6 +32,8 @@
 #include "gx_faust_plugins.h"
 #include "../plugins/pluginlib.h"
 
+#define always_inline inline __attribute__((always_inline))
+
 #include "faust/gx_ampout_ladspa.cc"
 #include "faust/gx_outputlevel_ladspa.cc"
 
