--- src/fontwin.c
+++ src/fontwin.c	2006/01/17 09:08:26
@@ -126,7 +126,7 @@
 	fonttool_frame = XmCreateDialogShell(app_shell, "Font tool", NULL, 0);
 	handle_close(fonttool_frame);
         fonttool_panel = XtCreateWidget("fonttool_panel", xmFormWidgetClass, 
-                                        fonttool_frame, NULL, 0);
+                                        fonttool_frame, NULL, NULL);
 
         font_select_item = CreateFontChoice(fonttool_panel, "Font:");
         XtVaSetValues(font_select_item->menu,
