--- src/xmgrace.c	2007/04/26 09:39:20	1.1
+++ src/xmgrace.c	2007/04/26 09:40:26
@@ -864,10 +864,10 @@
 
     menupane = CreateMenu(menubar, "Help", 'H', TRUE);
 
-    CreateMenuButton(menupane, "User's Guide", 'G', HelpCB, "doc/UsersGuide.html");
-    CreateMenuButton(menupane, "Tutorial", 'T', HelpCB, "doc/Tutorial.html");
-    CreateMenuButton(menupane, "FAQ", 'Q', HelpCB, "doc/FAQ.html");
-    CreateMenuButton(menupane, "Changes", 'C', HelpCB, "doc/CHANGES.html");
+    CreateMenuButton(menupane, "User's Guide", 'G', HelpCB, "/usr/share/doc/packages/xmgrace/doc/UsersGuide.html");
+    CreateMenuButton(menupane, "Tutorial", 'T', HelpCB, "/usr/share/doc/packages/xmgrace/doc/Tutorial.html");
+    CreateMenuButton(menupane, "FAQ", 'Q', HelpCB, "/usr/share/doc/packages/xmgrace/doc/FAQ.html");
+    CreateMenuButton(menupane, "Changes", 'C', HelpCB, "/usr/share/doc/packages/xmgrace/doc/CHANGES.html");
 
     CreateMenuSeparator(menupane);
  
