diff -rupN dolphin-emu-2879cbd2b564-orig/CMakeLists.txt dolphin-emu-2879cbd2b564/CMakeLists.txt
--- dolphin-emu-2879cbd2b564-orig/CMakeLists.txt	2014-03-07 20:10:55.185158971 -0500
+++ dolphin-emu-2879cbd2b564/CMakeLists.txt	2014-03-07 20:16:00.388691555 -0500
@@ -707,13 +707,11 @@ if(NOT DISABLE_WX AND NOT ANDROID)
 			OUTPUT_VARIABLE DIST_NAME
 			ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
 		if(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.${CMAKE_PATCH_VERSION}
-				VERSION_EQUAL 2.8.2 OR "${DIST_NAME}" STREQUAL "natty")
-			check_lib(GTK2 gtk+-2.0 gtk.h REQUIRED)
+				VERSION_EQUAL 2.8.12 OR "${DIST_NAME}" STREQUAL "natty")
+			check_lib(GTK3 gtk+-3.0 gtk.h REQUIRED)
 		else()
-			include(FindGTK2)
-			if(GTK2_FOUND)
-				include_directories(${GTK2_INCLUDE_DIRS})
-			endif()
+			include(FindGTK)
+			include_directories(${GTK3_INCLUDE_DIRS})
 		endif()
 	endif()
 
diff -rupN dolphin-emu-2879cbd2b564-orig/Source/Core/DolphinWX/CMakeLists.txt dolphin-emu-2879cbd2b564/Source/Core/DolphinWX/CMakeLists.txt
--- dolphin-emu-2879cbd2b564-orig/Source/Core/DolphinWX/CMakeLists.txt	2014-03-07 20:10:55.186158963 -0500
+++ dolphin-emu-2879cbd2b564/Source/Core/DolphinWX/CMakeLists.txt	2014-03-07 20:18:55.977272101 -0500
@@ -11,7 +11,7 @@ set(LIBS	core
 			audiocommon
 			z
 			sfml-network
-			${GTK2_LIBRARIES})
+			${GTK3_LIBRARIES})
 
 if(NOT ANDROID)
 	if(USE_X11)
