--- anymeal-1.17/configure.ac.orig	2022-11-01 20:41:56.815000000 -0000
+++ anymeal-1.17/configure.ac	2022-11-01 20:41:52.771000000 -0000
@@ -15,17 +15,24 @@
   AC_MSG_ERROR([Could not find flex])
 fi
 
+AC_ARG_WITH(windres,
+  AS_HELP_STRING([--with-windres],
+    [Enable support for windres (default: auto)]),
+  [WINDRES=$with_windres], [WINDRES=auto])
+
 dnl Check for windres.
-AC_CHECK_PROG(WINDRES,windres,windres,no)
+if test "x$WINDRES" = "xauto"; then
+  AC_CHECK_PROG(WINDRES,windres,windres,no)
+fi
 AM_CONDITIONAL([HAVE_WINDRES], [test "x$WINDRES" = "xwindres"])
 
 dnl Check for Googletest framework.
-if test "x$GTEST" = "x"; then
-  GTEST="/usr/src/googletest/googletest"
-fi
-AS_IF([test -e "$GTEST/src/gtest-all.cc"],
-      AC_SUBST(GTEST),
-      AC_MSG_ERROR([Could not find $GTEST/src/gtest-all.cc]))
+dnl if test "x$GTEST" = "x"; then
+dnl   GTEST="/usr/src/googletest/googletest"
+dnl fi
+dnl AS_IF([test -e "$GTEST/src/gtest-all.cc"],
+dnl       AC_SUBST(GTEST),
+dnl       AC_MSG_ERROR([Could not find $GTEST/src/gtest-all.cc]))
 
 dnl Check for recode-library.
 AC_MSG_CHECKING([for recode-library])
