diff -Naur liquidwar6-0.0.6beta.orig/configure liquidwar6-0.0.6beta/configure
--- liquidwar6-0.0.6beta.orig/configure	2009-01-10 09:43:13.000000000 -0500
+++ liquidwar6-0.0.6beta/configure	2009-06-05 15:26:10.000000000 -0400
@@ -29368,7 +29368,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lsqlite3  $LIBS"
+LIBS="-lsqlite3 -ldl $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
