Backport of https://github.com/mypaint/mypaint/pull/1183

diff --git a/setup.cfg b/setup.cfg
index 4779b0fdd..49965f1f4 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -7,11 +7,11 @@
 
 [install]
 verbose=1
-install-scripts=$base/bin
-install-platlib=$base/lib/mypaint
-install-purelib=$base/lib/mypaint
-install-data=$base/share
-install-headers=$base/include
+install_scripts=$base/bin
+install_platlib=$base/lib/mypaint
+install_purelib=$base/lib/mypaint
+install_data=$base/share
+install_headers=$base/include
 
 [nosetests]
 with-doctest=1
index f12c28669..204236765 100644
--- a/setup.py
+++ b/setup.py
@@ -14,9 +14,6 @@
 import tempfile
 import shutil
 
-from distutils.command.build import build
-from distutils.command.clean import clean
-
 from setuptools import setup
 from setuptools import Extension
 from setuptools import Command
@@ -24,6 +21,11 @@
 from setuptools.command.install import install
 from setuptools.command.install_scripts import install_scripts
 
+# setuptools must be imported first since they ensure
+# their distutils implementation will be used.
+from distutils.command.build import build
+from distutils.command.clean import clean
+
 # Constants
 
 
