--- goffice/math/go-ryu.h	2023-05-20 19:48:32.000000000 -0300
+++ goffice/math/go-ryu.h.patched	2023-12-27 19:28:35.623167312 -0400
@@ -9,7 +9,9 @@
 /* ------------------------------------------------------------------------- */
 
 int go_ryu_d2s_buffered_n (double d, char *dst);
+#ifdef GOFFICE_WITH_LONG_DOUBLE
 int go_ryu_ld2s_buffered_n (long double d, char *dst);
+#endif
 
 /* ------------------------------------------------------------------------- */
 
