diff -Nur sash-3.7.orig/Makefile sash-3.7/Makefile
--- sash-3.7.orig/Makefile	2002-07-21 18:24:47.000000000 -0500
+++ sash-3.7/Makefile	2007-08-22 15:56:15.225837097 -0500
@@ -14,7 +14,7 @@
 MOUNT_TYPE		= '"ext3"'
 
 
-CFLAGS = -O3 -Wall -Wmissing-prototypes \
+CFLAGS = -O2 -march=i486 -mtune=i686 -Wall -Wmissing-prototypes \
 	-DHAVE_GZIP=$(HAVE_GZIP) \
 	-DHAVE_LINUX_ATTR=$(HAVE_LINUX_ATTR) \
 	-DHAVE_LINUX_MOUNT=$(HAVE_LINUX_MOUNT) \
