diff -Nur xtrs-4.9d.orig/Makefile.local xtrs-4.9d/Makefile.local
--- xtrs-4.9d.orig/Makefile.local	2009-06-15 18:32:57.000000000 -0500
+++ xtrs-4.9d/Makefile.local	2009-09-26 16:29:19.371490363 -0500
@@ -36,9 +36,9 @@
 # time, use these lines (with the appropriate file names).  The default file
 # name is ignored and can be omitted if the ROM for that model is built in.
 
-DEFAULT_ROM = -DDEFAULT_ROM='"/usr/local/lib/xtrs/level2rom.hex"' \
-              -DDEFAULT_ROM3='"/usr/local/lib/xtrs/romimage.m3"' \
-              -DDEFAULT_ROM4P='"/usr/local/lib/xtrs/romimage.m4p"'
+DEFAULT_ROM = -DDEFAULT_ROM='"/usr/share/xtrs/level2rom.hex"' \
+              -DDEFAULT_ROM3='"/usr/share/xtrs/romimage.m3"' \
+              -DDEFAULT_ROM4P='"/usr/share/xtrs/romimage.m4p"'
 
 # If you would like to change where xtrs looks for disk?-? files, edit
 # this line.  "." of course means the current working directory.
