--- toolchain-avr-7.3.0-atmel3.6.1-arduino7/build.conf.orig	2020-05-26 15:42:08.000000000 +0200
+++ toolchain-avr-7.3.0-atmel3.6.1-arduino7/build.conf	2022-01-06 15:07:46.251140445 +0100
@@ -17,7 +17,11 @@
 # MPC_VERSION=0.9
 
 GCC_VERSION=7.3.0
+AUTOCONF_VERSION=2.64
+AUTOMAKE_VERSION=1.11.1
+GMP_VERSION=5.0.2
 MPFR_VERSION=3.1.0
+MPC_VERSION=0.9
 
 ATMEL_ATMEGA_PACK_VERSION=1.4.351
 ATMEL_ATMEGA_PACK_FILENAME=Atmel.ATmega_DFP.${ATMEL_ATMEGA_PACK_VERSION}
@@ -43,7 +43,6 @@
 GREP=grep
 
 #fallback for grep without pearl regexp
-SOURCES_README="$(wget -O - ${AVR_SOURCES}/SOURCES.README 2>/dev/null)"
 [ -z "$GCC_VERSION"       ] && GCC_VERSION="$(echo "$SOURCES_README" | $GREP "GCC"  | egrep -o "([0-9]{1,}\.)+[0-9]{1,}")"
 [ -z "$AUTOMAKE_VERSION"  ] && AUTOMAKE_VERSION="$(echo "$SOURCES_README" | $GREP "automake"  | egrep -o "([0-9]{1,}\.)+[0-9]{1,}")"
 [ -z "$AUTOCONF_VERSION"  ] && AUTOCONF_VERSION="$(echo "$SOURCES_README" | $GREP "autoconf"  | egrep -o "([0-9]{1,}\.)+[0-9]{1,}")"
--- toolchain-avr-7.3.0-atmel3.6.1-arduino7/gcc.build.bash.orig	2020-05-26 15:42:08.000000000 +0200
+++ toolchain-avr-7.3.0-atmel3.6.1-arduino7/gcc.build.bash	2022-01-06 16:00:19.640966313 +0100
@@ -60,6 +60,7 @@
 
 # Apply the right patchset
 cd gcc && patch -p1 < ../avr-gcc-patches/atmel-patches-gcc.7.3.0-arduino2.patch && cd ..
+cd gcc && patch -p1 < ../avr-gcc-patches/gcc-add-missing-isl-includes.patch && cd ..
 
 #pushd gcc
 #pushd gcc/config/avr/
--- toolchain-avr-7.3.0-atmel3.6.1-arduino7/atpack.Dx.build.bash.orig	2020-05-26 15:42:08.000000000 +0200
+++ toolchain-avr-7.3.0-atmel3.6.1-arduino7/atpack.Dx.build.bash	2022-01-07 02:53:31.621802110 +0100
@@ -17,7 +17,7 @@
 
 source build.conf
 
-wget ${ATMEL_DX_PACK_URL}
+#wget ${ATMEL_DX_PACK_URL}
 
 mkdir -p atpack
 cd atpack
--- toolchain-avr-7.3.0-atmel3.6.1-arduino7/atpack.build.bash.orig	2020-05-26 15:42:08.000000000 +0200
+++ toolchain-avr-7.3.0-atmel3.6.1-arduino7/atpack.build.bash	2022-01-07 02:53:35.029801922 +0100
@@ -17,7 +17,7 @@
 
 source build.conf
 
-wget ${ATMEL_ATMEGA_PACK_URL}
+#wget ${ATMEL_ATMEGA_PACK_URL}
 
 mkdir -p atpack
 cd atpack
--- toolchain-avr-7.3.0-atmel3.6.1-arduino7/atpack.tiny.build.bash.orig	2020-05-26 15:42:08.000000000 +0200
+++ toolchain-avr-7.3.0-atmel3.6.1-arduino7/atpack.tiny.build.bash	2022-01-07 02:53:38.116801752 +0100
@@ -17,7 +17,7 @@
 
 source build.conf
 
-wget ${ATMEL_ATTINY_PACK_URL}
+#wget ${ATMEL_ATTINY_PACK_URL}
 
 mkdir -p atpack
 cd atpack
