diff -Naur x2goclient-4.1.2.2.orig/src/onmainwindow_privat.h x2goclient-4.1.2.2/src/onmainwindow_privat.h
--- x2goclient-4.1.2.2.orig/src/onmainwindow_privat.h	2020-02-13 11:36:07.000000000 +0100
+++ x2goclient-4.1.2.2/src/onmainwindow_privat.h	2022-08-23 20:49:00.556603000 +0200
@@ -84,7 +84,6 @@
 #include "contest.h"
 
 #if !defined Q_OS_WIN
-#include <sys/mount.h>
 #ifdef Q_OS_LINUX
 #include <X11/Xatom.h>
 #include <X11/xpm.h>
