--- data/pigui/libs/equipment-outfitter.lua.old	2025-02-08 00:53:12.018320147 +0900
+++ data/pigui/libs/equipment-outfitter.lua	2025-02-08 00:53:18.344388274 +0900
@@ -266,9 +266,11 @@
 ---@param data UI.EquipCard.Data
 function Outfitter:modifyEquipmentStats(data)
 	local stock = self:getStock(data.equip)
+	local tech_level = data.equip.tech_level
+	tech_level = tech_level == "MILITARY" and 11 or tech_level
 
 	table.insert(data.stats, 1, { l.AVAILABLE_STOCK, icons.cargo_crate, stock, fmt_number })
-	table.insert(data.stats, 2, { l.TECH_LEVEL, icons.station_orbital_large, data.equip.tech_level, fmt_number })
+	table.insert(data.stats, 2, { l.TECH_LEVEL, icons.station_orbital_large, tech_level, fmt_number })
 end
 
 function Outfitter:buildEquipmentList()
