--- data/modules/Equipment/Types.lua.old	2025-02-06 12:54:42.346204471 +0900
+++ data/modules/Equipment/Types.lua	2025-02-06 13:13:07.126097752 +0900
@@ -267,7 +267,7 @@
 		local amount = ship['nextJumpFuelUse']
 		ship:unsetprop('nextJumpFuelUse')
 
-		self.storedFuel = math.max(0, self.storedFuel - amount)
+		self:SetFuel(ship, math.max(0, self.storedFuel - amount))
 
 		if self.byproduct then
 			local cargoMgr = ship:GetComponent('CargoManager')
