--- data/default.in~	2007-05-27 13:16:01.000000000 +0200
+++ data/default.in	2007-05-27 13:22:56.000000000 +0200
@@ -134,13 +134,10 @@
     type = launchbar
     config {
         button {
-            id = pcmanfm.desktop
+            id = mozilla-firefox.desktop
         }
         button {
-            id = gnome-terminal.desktop
-        }
-        button {
-            id = firefox.desktop
+            id = mozilla-thunderbird.desktop
         }
     }
 }
