@@ -, +, @@ 
 src/plugins/batt/batt.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)
--- a/src/plugins/batt/batt.c	
+++ a/src/plugins/batt/batt.c	
@@ -95,7 +95,7 @@ typedef struct {
 typedef struct {
     char *command;
     sem_t *lock;
-} alarm;
+} Alarm;
 
 static void destructor(Plugin *p);
 static void update_display(lx_battery *lx_b, gboolean repaint);
@@ -103,7 +103,7 @@ static void update_display(lx_battery *lx_b, gboolean repaint);
 /* alarmProcess takes the address of a dynamically allocated alarm struct (which
    it must free). It ensures that alarm commands do not run concurrently. */
 static void * alarmProcess(void *arg) {
-    alarm *a = (alarm *) arg;
+    Alarm *a = (Alarm *) arg;
 
     sem_wait(a->lock);
     system(a->command);
@@ -157,7 +157,7 @@ void update_display(lx_battery *lx_b, gboolean repaint) {
 	/* Run the alarm command if it isn't already running */
 	if (alarmCanRun) {
 	    
-	    alarm *a = (alarm *) malloc(sizeof(alarm));
+	    Alarm *a = (Alarm *) malloc(sizeof(Alarm));
 	    a->command = lx_b->alarmCommand;
 	    a->lock = &(lx_b->alarmProcessLock);
 	    
