--- data/default/panels/panel.in~	2008-06-25 16:35:25.000000000 +0200
+++ data/default/panels/panel.in	2008-06-25 16:37:47.000000000 +0200
@@ -50,14 +50,15 @@
     type = launchbar
     Config {
         Button {
-            id=pcmanfm.desktop
+            id=mozilla-firefox.desktop
         }
         Button {
-            id=gnome-terminal.desktop
+            id=mozilla-thunderbird.desktop
         }
         Button {
-            id=firefox.desktop
+            id=Terminal.desktop
         }
+
     }
 }
 
