--- usr/share/lxpanel/profile/default/panels/panel.orig	2010-08-19 00:16:51.000000000 +0200
+++ usr/share/lxpanel/profile/default/panels/panel	2010-08-18 23:58:42.239083001 +0200
@@ -3,10 +3,10 @@
 
 Global {
     edge=bottom
-    allign=left
+    allign=center
     margin=0
     widthtype=percent
-    width=100
+    width=95
     height=26
     transparent=0
     tintcolor=#000000
@@ -53,10 +53,13 @@
             id=pcmanfm.desktop
         }
         Button {
-            id=gnome-terminal.desktop
+            id=mozilla-firefox.desktop
         }
         Button {
-            id=firefox.desktop
+            id=mozilla-thunderbird.desktop
+        }
+        Button {
+            id=lxterminal.desktop
         }
     }
 }
@@ -113,11 +116,44 @@
 }
 
 Plugin {
+    type = volumealsa
+}
+
+Plugin {
+    type = space
+    Config {
+        Size=4
+    }
+}
+
+Plugin {
+    type = netstatus
+    Config {
+        iface=eth0
+        configtool=network-admin --configure %i
+    }
+}
+
+Plugin {
+    type = space
+    Config {
+        Size=4
+    }
+}
+
+Plugin {
     type = cpu
 }
 
 Plugin {
-    type = tray
+    type = space
+    Config {
+        Size=4
+    }
+}
+
+Plugin {
+    type = cpufreq
 }
 
 Plugin {
@@ -126,6 +162,23 @@
         ClockFmt=%R
         TooltipFmt=%A %x
         BoldFont=0
+        IconOnly=0
+    }
+}
+
+Plugin {
+    type = space
+    Config {
+        Size=4
+    }
+}
+
+Plugin {
+    type = launchbar
+    Config {
+        Button {
+            id=lxde-logout.desktop
+        }
     }
 }
 
