diff -up ./CMakeLists.txt.orig ./CMakeLists.txt
--- ./CMakeLists.txt.orig	2016-01-19 20:49:33.000000000 +0100
+++ ./CMakeLists.txt	2016-02-05 13:30:27.456442363 +0100
@@ -81,5 +81,5 @@ ENDIF(NOT WIN32)
 configure_file (platform-config.cmake.in
                 platform-config.cmake @ONLY)
 install(FILES ${CMAKE_BINARY_DIR}/platform-config.cmake
-        DESTINATION ${CMAKE_INSTALL_LIBDIR_NOARCH}/platform)
+        DESTINATION ${CMAKE_INSTALL_LIBDIR_NOARCH}/cmake/${CMAKE_PROJECT_NAME})
 
