diff -ur fcitx-configtool-0.3.1/gtk/config_widget.h fcitx-configtool-0.3.1_fixed/gtk/config_widget.h
--- fcitx-configtool-0.3.1/gtk/config_widget.h	2011-09-09 03:55:49.000000000 +0000
+++ fcitx-configtool-0.3.1_fixed/gtk/config_widget.h	2012-09-14 14:40:24.016046036 +0000
@@ -23,7 +23,7 @@
 #define _FCITX_CONFIG_WIDGET
 
 #include <gtk/gtkwidget.h>
-#include <glib/gstring.h>
+#include <glib.h>
 #include <fcitx-config/fcitx-config.h>
 #include <gtk/gtkvbox.h>
 #include "sub_config_parser.h"
diff -ur fcitx-configtool-0.3.1/gtk/sub_config_parser.c fcitx-configtool-0.3.1_fixed/gtk/sub_config_parser.c
--- fcitx-configtool-0.3.1/gtk/sub_config_parser.c	2011-09-09 03:55:49.000000000 +0000
+++ fcitx-configtool-0.3.1_fixed/gtk/sub_config_parser.c	2012-09-14 14:32:49.512156233 +0000
@@ -22,8 +22,7 @@
 #include <dirent.h>
 #include <string.h>
 #include <sys/stat.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gpattern.h>
+#include <glib.h>
 
 #include <fcitx-config/xdg.h>
 #include <fcitx-utils/log.h>
diff -ur fcitx-configtool-0.3.1/gtk/sub_config_parser.h fcitx-configtool-0.3.1_fixed/gtk/sub_config_parser.h
--- fcitx-configtool-0.3.1/gtk/sub_config_parser.h	2011-09-09 03:55:49.000000000 +0000
+++ fcitx-configtool-0.3.1_fixed/gtk/sub_config_parser.h	2012-09-14 14:33:05.738265415 +0000
@@ -19,7 +19,7 @@
 
 #ifndef _SUB_CONFIG_PARSER_H
 #define _SUB_CONFIG_PARSER_H
-#include <glib/ghash.h>
+#include <glib.h>
 
 typedef enum
 {
