diff --git a/acinclude.m4 b/acinclude.m4
index 3b67d9b5cae2d766..408b9845f8f1f328 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -23,7 +23,7 @@ AC_DEFUN([AC_CHECK_ALIGNOF],
 			typedef struct {char s1;} CORBA_struct;
 			typedef void *CORBA_pointer;
 			struct test {char s1; $1 s2;};
-			main()
+			int main(void)
 			{
 			FILE *f=fopen("conftestval", "w");
 			if (!f) exit(1);
diff --git a/configure b/configure
index 1fed47a1d26c3f36..63c32a20c6574672 100755
--- a/configure
+++ b/configure
@@ -12346,7 +12346,7 @@ else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
  #include <stdio.h>
-             main ()
+             int main (void)
              {
                return 0;
              }
@@ -12387,7 +12387,7 @@ else
 			typedef struct {char s1;} CORBA_struct;
 			typedef void *CORBA_pointer;
 			struct test {char s1; CORBA_octet s2;};
-			main()
+			int main(void)
 			{
 			FILE *f=fopen("conftestval", "w");
 			if (!f) exit(1);
@@ -12434,7 +12434,7 @@ else
 			typedef struct {char s1;} CORBA_struct;
 			typedef void *CORBA_pointer;
 			struct test {char s1; CORBA_boolean s2;};
-			main()
+			int main(void)
 			{
 			FILE *f=fopen("conftestval", "w");
 			if (!f) exit(1);
@@ -12481,7 +12481,7 @@ else
 			typedef struct {char s1;} CORBA_struct;
 			typedef void *CORBA_pointer;
 			struct test {char s1; CORBA_char s2;};
-			main()
+			int main(void)
 			{
 			FILE *f=fopen("conftestval", "w");
 			if (!f) exit(1);
@@ -12528,7 +12528,7 @@ else
 			typedef struct {char s1;} CORBA_struct;
 			typedef void *CORBA_pointer;
 			struct test {char s1; CORBA_wchar s2;};
-			main()
+			int main(void)
 			{
 			FILE *f=fopen("conftestval", "w");
 			if (!f) exit(1);
@@ -12575,7 +12575,7 @@ else
 			typedef struct {char s1;} CORBA_struct;
 			typedef void *CORBA_pointer;
 			struct test {char s1; CORBA_short s2;};
-			main()
+			int main(void)
 			{
 			FILE *f=fopen("conftestval", "w");
 			if (!f) exit(1);
@@ -12622,7 +12622,7 @@ else
 			typedef struct {char s1;} CORBA_struct;
 			typedef void *CORBA_pointer;
 			struct test {char s1; CORBA_long s2;};
-			main()
+			int main(void)
 			{
 			FILE *f=fopen("conftestval", "w");
 			if (!f) exit(1);
@@ -12669,7 +12669,7 @@ else
 			typedef struct {char s1;} CORBA_struct;
 			typedef void *CORBA_pointer;
 			struct test {char s1; CORBA_long_long s2;};
-			main()
+			int main(void)
 			{
 			FILE *f=fopen("conftestval", "w");
 			if (!f) exit(1);
@@ -12716,7 +12716,7 @@ else
 			typedef struct {char s1;} CORBA_struct;
 			typedef void *CORBA_pointer;
 			struct test {char s1; CORBA_float s2;};
-			main()
+			int main(void)
 			{
 			FILE *f=fopen("conftestval", "w");
 			if (!f) exit(1);
@@ -12763,7 +12763,7 @@ else
 			typedef struct {char s1;} CORBA_struct;
 			typedef void *CORBA_pointer;
 			struct test {char s1; CORBA_double s2;};
-			main()
+			int main(void)
 			{
 			FILE *f=fopen("conftestval", "w");
 			if (!f) exit(1);
@@ -12810,7 +12810,7 @@ else
 			typedef struct {char s1;} CORBA_struct;
 			typedef void *CORBA_pointer;
 			struct test {char s1; CORBA_long_double s2;};
-			main()
+			int main(void)
 			{
 			FILE *f=fopen("conftestval", "w");
 			if (!f) exit(1);
@@ -12857,7 +12857,7 @@ else
 			typedef struct {char s1;} CORBA_struct;
 			typedef void *CORBA_pointer;
 			struct test {char s1; CORBA_struct s2;};
-			main()
+			int main(void)
 			{
 			FILE *f=fopen("conftestval", "w");
 			if (!f) exit(1);
@@ -12904,7 +12904,7 @@ else
 			typedef struct {char s1;} CORBA_struct;
 			typedef void *CORBA_pointer;
 			struct test {char s1; CORBA_pointer s2;};
-			main()
+			int main(void)
 			{
 			FILE *f=fopen("conftestval", "w");
 			if (!f) exit(1);
diff --git a/configure.in b/configure.in
index 04d4b2a397d53196..4a25f4e6ec84bdac 100644
--- a/configure.in
+++ b/configure.in
@@ -220,7 +220,7 @@ dnl
 dofus_save_libs="$LIBS"
 LIBS="$ORBIT_LIBS $LIBS"
 AC_TRY_RUN([ #include <stdio.h>
-             main ()
+             int main (void)
              {
                return 0;
              } ],,
