--- ./src/orb/GIOP/giop-recv-buffer.c~	2010-02-09 06:05:35.000000000 -0600
+++ ./src/orb/GIOP/giop-recv-buffer.c	2024-01-30 09:31:58.383855651 -0600
@@ -559,7 +559,7 @@
 	ent->src_thread = giop_thread_self ();
 	ent->async_cb = NULL;
 
-	ent->cnx = giop_connection_ref (cnx);
+	ent->cnx = (GIOPConnection *)giop_connection_ref (cnx);
 	ent->msg_type = msg_type;
 	ent->request_id = request_id;
 	ent->buffer = NULL;
