diff -Naur wgetpaste-2.13.orig/wgetpaste wgetpaste-2.13/wgetpaste
--- wgetpaste-2.13.orig/wgetpaste	2009-05-27 09:46:48.000000000 -0400
+++ wgetpaste-2.13/wgetpaste	2009-06-12 07:12:45.000000000 -0400
@@ -99,7 +99,7 @@
 noxclip() {
 	cat <<EOF >&2
 Could not find xclip on your system. In order to use --x$1 you must either
-emerge x11-misc/xclip or define x_$1() globally in /etc/wgetpaste.conf or
+install xclip or define x_$1() globally in /etc/wgetpaste.conf or
 per user in ~/.wgetpaste.conf to use another program (such as e.g. xcut or
 klipper) to $2 your clipboard.
 
@@ -603,8 +603,8 @@
 [[ -z $SOURCE ]] && SOURCE="stdin" && FILES[${#FILES[*]}]="/dev/stdin"
 CVT_TABS=No
 
-INFO_COMMAND=${INFO_COMMAND:-"emerge --info"}
-INFO_ARGS=${INFO_ARGS:-"--ignore-default-opts"}
+INFO_COMMAND=${INFO_COMMAND:-"wgetpaste_info"}
+INFO_ARGS=${INFO_ARGS:-""}
 
 ### everything below this should be independent of which service is being used...
 
