diff -Naur wgetpaste-2.20/wgetpaste wgetpaste-2.20.patched/wgetpaste
--- wgetpaste-2.20/wgetpaste	2012-05-05 15:15:32.000000000 -0400
+++ wgetpaste-2.20.patched/wgetpaste	2012-12-20 17:14:06.000000000 -0500
@@ -134,7 +134,7 @@
 noxclip() {
 	cat <<EOF >&2
 Could not find xclip on your system. In order to use --x$1 you must either
-emerge x11-misc/xclip or define x_$1() globally in /etc/wgetpaste.conf or
+install xclip or define x_$1() globally in /etc/wgetpaste.conf or
 per user in ~/.wgetpaste.conf to use another program (such as e.g. xcut or
 klipper) to $2 your clipboard.
 
@@ -312,9 +312,9 @@
     -c, --command COMMAND         paste COMMAND and the output of COMMAND
     -i, --info                    append the output of \`$INFO_COMMAND\`
     -I, --info-only               paste the output of \`$INFO_COMMAND\` only
-    -x, --xcut                    read input from clipboard (requires x11-misc/xclip)
-    -X, --xpaste                  write resulting url to the X primary selection buffer (requires x11-misc/xclip)
-    -C, --xclippaste              write resulting url to the X clipboard selection buffer (requires x11-misc/xclip)
+    -x, --xcut                    read input from clipboard (requires xclip)
+    -X, --xpaste                  write resulting url to the X primary selection buffer (requires xclip)
+    -C, --xclippaste              write resulting url to the X clipboard selection buffer (requires xclip)
 
     -r, --raw                     show url for the raw paste (no syntax highlighting or html)
     -t, --tee                     use tee to show what is being pasted
@@ -647,8 +647,8 @@
 [[ -z $SOURCE ]] && SOURCE="stdin"
 CVT_TABS=No
 
-INFO_COMMAND=${INFO_COMMAND:-"emerge --info"}
-INFO_ARGS=${INFO_ARGS:-"--ignore-default-opts"}
+INFO_COMMAND=${INFO_COMMAND:-"wgetpaste_info"}
+INFO_ARGS=${INFO_ARGS:-""}
 
 ### everything below this should be independent of which service is being used...
 
