
diff -Nur pcsc-lite-1.4.99.orig/DRIVERS pcsc-lite-1.4.99/DRIVERS
--- DRIVERS.bak	2007-10-13 16:21:11.000000000 -0500
+++ DRIVERS	2008-03-06 00:21:53.119195118 -0600
@@ -2,8 +2,8 @@
 
 To install USB drivers, download the driver from http://www.linuxnet.com/
 and copy it to a temporary location.  After you have unarchived the files
-copy the directory to /usr/local/pcsc/drivers/.  If the directory
-/usr/local/pcsc/drivers/ does not exist create it before copying the driver
+copy the directory to /usr/lib/pcsc/drivers/.  If the directory
+/usr/lib/pcsc/drivers/ does not exist create it before copying the driver
 directory to it.
 
 Example Mac OS X, Linux 2.4:
@@ -28,3 +28,7 @@
 Refer to the driver README for more information.
 
 $Id: DRIVERS 867 2004-05-13 13:25:06Z rousseau $
+
+# Edited by Robby Workman <rworkman@slackbuilds.org> to reflect correct
+# information when installed from SlackBuilds.org  --  20080305
+
