--- DRIVERS	2008-05-17 17:20:11.636054139 +0200
+++ DRIVERS.new	2008-05-17 17:21:51.637812109 +0200
@@ -2,18 +2,18 @@
 
 To install USB drivers, download the driver from http://www.linuxnet.com/
 and copy it to a temporary location.  After you have unarchived the files
-copy the directory to /usr/local/pcsc/drivers/.  If the directory
-/usr/local/pcsc/drivers/ does not exist create it before copying the driver
+copy the directory to <libdir>/pcsc/drivers/.  If the directory
+<libdir>/pcsc/drivers/ does not exist, create it before copying the driver
 directory to it.
 
 Example Mac OS X, Linux 2.4:
 
 tar -xzvf driver.bundle-0.1.0.tar.gz
-cp -r driver.bundle /usr/local/pcsc/drivers/
-cd /usr/local/pcsc/drivers/driver.bundle
+cp -r driver.bundle <libdir>/pcsc/drivers/
+cd <libdir>/pcsc/drivers/driver.bundle
 build or use Project Builder to build
 
-The /usr/local/pcsc/drivers/ directory should contain bundle directories only.
+The <libdir>/pcsc/drivers/ directory should contain bundle directories only.
 You must restart pcscd to use the new driver.
 
 
@@ -28,3 +28,7 @@
 Refer to the driver README for more information.
 
 $Id: DRIVERS 867 2004-05-13 13:25:06Z rousseau $
+
+# Edited by Robby Workman <rworkman@slackbuilds.org> to reflect correct
+# information when installed from SlackBuilds.org  --  20080305
+
