--- cpsokoban-1.0/configure.original	2014-04-26 10:50:11.334099417 +0300
+++ cpsokoban-1.0/configure	2014-04-26 11:12:26.817403578 +0300
@@ -308,7 +308,7 @@
 # include <unistd.h>
 #endif"
 
-ac_subst_vars='SHELL PATH_SEPARATOR PACKAGE_NAME PACKAGE_TARNAME PACKAGE_VERSION PACKAGE_STRING PACKAGE_BUGREPORT exec_prefix prefix program_transform_name bindir sbindir libexecdir datadir sysconfdir sharedstatedir localstatedir libdir includedir oldincludedir infodir mandir build_alias host_alias target_alias DEFS ECHO_C ECHO_N ECHO_T LIBS CC CFLAGS LDFLAGS CPPFLAGS ac_ct_CC EXEEXT OBJEXT CPP EGREP INCLUDES HAS_XLIB HAS_TIX SOKODIR TARDIR HELPCMD PIXMAPDIR SUBDIRS SET_MAKE LIBOBJS LTLIBOBJS'
+ac_subst_vars='SHELL PATH_SEPARATOR PACKAGE_NAME PACKAGE_TARNAME PACKAGE_VERSION PACKAGE_STRING PACKAGE_BUGREPORT exec_prefix prefix program_transform_name bindir sbindir libexecdir datadir sysconfdir sharedstatedir localstatedir libdir includedir oldincludedir infodir mandir docdir build_alias host_alias target_alias DEFS ECHO_C ECHO_N ECHO_T LIBS CC CFLAGS LDFLAGS CPPFLAGS ac_ct_CC EXEEXT OBJEXT CPP EGREP INCLUDES HAS_XLIB HAS_TIX SOKODIR TARDIR HELPCMD PIXMAPDIR SUBDIRS SET_MAKE LIBOBJS LTLIBOBJS'
 ac_subst_files=''
 
 # Initialize some variables set by options.
@@ -348,6 +348,7 @@
 oldincludedir='/usr/include'
 infodir='${prefix}/info'
 mandir='${prefix}/man'
+docdir='${prefix}/doc'
 
 ac_prev=
 for ac_option
@@ -450,6 +451,11 @@
   -infodir=* | --infodir=* | --infodi=* | --infod=* | --info=* | --inf=*)
     infodir=$ac_optarg ;;
 
+  -docdir | --docdir | --docdi | --docd | --doc) 
+    ac_prev=docdir ;;
+  -docdir=* | --docdir=* | --docdi=* | docd=* | --doc=*)
+    docdir=$ac_optarg ;;
+
   -libdir | --libdir | --libdi | --libd)
     ac_prev=libdir ;;
   -libdir=* | --libdir=* | --libdi=* | --libd=*)
@@ -667,7 +673,7 @@
 
 # Be sure to have absolute paths.
 for ac_var in bindir sbindir libexecdir datadir sysconfdir sharedstatedir \
-	      localstatedir libdir includedir oldincludedir infodir mandir
+	      localstatedir libdir includedir oldincludedir infodir mandir docdir
 do
   eval ac_val=$`echo $ac_var`
   case $ac_val in
@@ -825,6 +831,7 @@
   --includedir=DIR       C header files [PREFIX/include]
   --oldincludedir=DIR    C header files for non-gcc [/usr/include]
   --infodir=DIR          info documentation [PREFIX/info]
+  --docdir=DIR		 doc documentation [PREFIX/doc]
   --mandir=DIR           man documentation [PREFIX/man]
 _ACEOF
 
@@ -4208,6 +4215,7 @@
 s,@includedir@,$includedir,;t t
 s,@oldincludedir@,$oldincludedir,;t t
 s,@infodir@,$infodir,;t t
+s,@docdir@,$docdir,;t t
 s,@mandir@,$mandir,;t t
 s,@build_alias@,$build_alias,;t t
 s,@host_alias@,$host_alias,;t t
