diff -Naur LWP-UserAgent-Cached-0.06/Makefile.PL LWP-UserAgent-Cached-0.06.patched/Makefile.PL
--- LWP-UserAgent-Cached-0.06/Makefile.PL	2014-06-27 13:32:55.000000000 -0400
+++ LWP-UserAgent-Cached-0.06.patched/Makefile.PL	2015-05-12 18:40:40.000000000 -0400
@@ -4,33 +4,7 @@
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
 # the contents of the Makefile that is written.
 
-my %test_requires = (
-	'Test::Mock::LWP::Dispatch' => 0.02,
-	'File::Temp' => 0
-);
-
-my $not_installed_test_modules = '';
-while (my ($module, $ver) = each %test_requires) {
-	
-	eval "use $module $ver";
-	if ($@) {
-		$not_installed_test_modules .= "\t$module $ver\n";
-	}
-}
-
-if ($not_installed_test_modules) {
-	my $choose = prompt(
-		"We need few optional additional modules to test this package on your system:\n".
-		$not_installed_test_modules.
-		"If you'll choose `n' this tests will be skipped.\n".
-		"Append this modules to installation queue?",
-		"y"
-	);
-	
-	if ($choose =~ /n/) {
-		%test_requires = ();
-	}
-}
+my %test_requires = ();
 
 $test_requires{'Test::More'} = 0.88;
 
