--- msn-proxy-0.7_orig/Makefile	2009-01-20 13:34:19.000000000 -0200
+++ msn-proxy-0.7/Makefile	2009-12-26 23:17:08.000000000 -0200
@@ -1,18 +1,18 @@
 
 RM=/bin/rm
 INSTALL=/usr/bin/install
-SYSCONFDIR=$(DESTDIR)/usr/local/etc/msn-proxy
-BINDIR=$(DESTDIR)/usr/local/bin
+SYSCONFDIR=$(DESTDIR)/etc/msn-proxy
+BINDIR=$(DESTDIR)/usr/bin
 
-# FreeBSD
-MYSQLINC=/usr/local/include
-MYSQLLIB=/usr/local/lib/mysql
+# Slackware
+MYSQLINC=/usr/include/mysql
+MYSQLLIB=/usr/lib@LIBDIRSUFFIX@/mysql
 
 # Linux (most)
 #MYSQLINC=/usr/lib/mysql
 #MYSQLLIB=/usr/local/mysql/lib/mysql
 
-CCFLAGS=-g -Wall -O2 -pipe -I/usr/local/include -I$(MYSQLINC) -fno-builtin-log
+CCFLAGS=-g -Wall @SLKCFLAGS@ -pipe -I/usr/local/include -I$(MYSQLINC) -fno-builtin-log
 
 LDFLAGS=-L/usr/local/lib -levent -L$(MYSQLLIB) -lmysqlclient
 
