From e31fe347051e8dd051514398c56a816a3a71f8a4 Mon Sep 17 00:00:00 2001
From: Josue Ortega <josueortega@debian.org.gt>
Date: Sun, 15 Feb 2015 15:27:54 -0600
Subject: [PATCH] Removed unused variable

---
 curve/ed25519/additions/curve_sigs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/curve/ed25519/additions/curve_sigs.c b/curve/ed25519/additions/curve_sigs.c
index 51f2052..fa78eaf 100644
--- a/curve/ed25519/additions/curve_sigs.c
+++ b/curve/ed25519/additions/curve_sigs.c
@@ -7,7 +7,7 @@ void curve25519_keygen(unsigned char* curve25519_pubkey_out,
                        const unsigned char* curve25519_privkey_in)
 {
   ge_p3 ed; /* Ed25519 pubkey point */
-  fe ed_y, ed_y_plus_one, one_minus_ed_y, inv_one_minus_ed_y;
+  fe ed_y_plus_one, one_minus_ed_y, inv_one_minus_ed_y;
   fe mont_x;
 
   /* Perform a fixed-base multiplication of the Edwards base point,
