diff -Naur a/gui/crxdisplay.cpp b/gui/crxdisplay.cpp
--- a/gui/crxdisplay.cpp	2017-02-07 15:10:47.000000000 +0000
+++ b/gui/crxdisplay.cpp	2022-02-13 12:27:01.631479349 +0000
@@ -78,7 +78,7 @@
         Sound = new WaveInput ( -1 );
     else
       Sound = new CSound ( settings.serial );
-    if ( Sound <= NULL )
+    if ( Sound <= (void *) NULL )
       return false;
 
     connect ( Sound, SIGNAL ( samplesAvailable() ), this, SLOT ( process_rxdata() ) );
diff -Naur a/gui/linpsk.cpp b/gui/linpsk.cpp
--- a/gui/linpsk.cpp	2017-02-07 15:10:47.000000000 +0000
+++ b/gui/linpsk.cpp	2022-02-13 12:28:37.937992591 +0000
@@ -410,7 +410,7 @@
       Modulator = new BpskModulator ( 11025, Frequency, TxBuffer );
       break; */
   }
-  if ( Sound <= NULL ) // Only create Sound Device once for output
+  if ( Sound <= (void *) NULL ) // Only create Sound Device once for output
   {
     if ( settings.DemoMode )
     {
@@ -421,7 +421,7 @@
       Sound = new CSound ( settings.serial );
     connect ( Sound, SIGNAL ( samplesAvailable() ), this, SLOT ( process_txdata() ) );
   }
-  if ( Sound <= NULL )
+  if ( Sound <= (void *) NULL )
   {
     QMessageBox::critical ( 0, " Programm Error! LinPsk", "Could not create Sound Device for Output" );
     TxDisplay->TxFunctions->setStatus ( ON );
diff -Naur a/gui/modemenu.cpp b/gui/modemenu.cpp
--- a/gui/modemenu.cpp	2017-02-07 15:10:47.000000000 +0000
+++ b/gui/modemenu.cpp	2022-02-13 12:45:09.832995118 +0000
@@ -18,7 +18,7 @@
  *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
  ***************************************************************************/
 
-
+#include <QButtonGroup>
 #include "modemenu.h"
 #include "parameter.h"
 #include "crxchannel.h"
