diff -Nur helm-0.9.0/Makefile helm-0.9.0.new/Makefile
--- helm-0.9.0/Makefile	2017-07-08 23:40:42.000000000 +0200
+++ helm-0.9.0.new/Makefile	2018-12-13 00:48:34.637977851 +0100
@@ -40,7 +40,6 @@
 IMAGES  = $(SYSDATA)/icons
 PATCHES = $(SYSDATA)/patches
 MAN     = $(DESTDIR)/usr/share/man/man1/
-CHANGES = $(DESTDIR)/usr/share/doc/$(PROGRAM)/
 DESKTOP = $(DESTDIR)/usr/share/applications/
 
 ICONS   = $(DESTDIR)/usr/share/icons/hicolor/
@@ -96,14 +95,13 @@
 	cp -rf patches/* $(PATCHES)
 
 install_standalone: standalone install_patches install_icons
-	install -d $(BIN) $(IMAGES) $(MAN) $(CHANGES) $(DESKTOP)
+	install -d $(BIN) $(IMAGES) $(MAN) $(DESKTOP)
 	install standalone/builds/linux/build/$(PROGRAM) $(BIN)
 	install -m644 standalone/helm.desktop $(DESKTOP)/helm.desktop
 	install -m644 images/* $(IMAGES)
 	cp changelog changes
 	gzip -n -9 changelog
 	mv changes changelog
-	mv changelog.gz $(CHANGES)/changelog.gz
 	cp docs/helm.1.gz $(MAN)
 
 install_lv2: lv2 install_patches
