diff -Naur gecko-mediaplayer-1.0.6-orig/src/plugin.cpp gecko-mediaplayer-1.0.6/src/plugin.cpp
--- gecko-mediaplayer-1.0.6-orig/src/plugin.cpp	2012-03-21 12:23:46.000000000 -0400
+++ gecko-mediaplayer-1.0.6/src/plugin.cpp	2012-08-11 22:54:14.000000000 -0400
@@ -1391,7 +1391,8 @@
             }
         }
         // Disable playing of partially cached apple.com files, mplayer seems to crash on these files
-        /*
+	// Bring this Back in
+
            if (!item->opened) {
            if ((item->localsize >= (plugin->cache_size * 1024)) && (percent >= 0.2)) {
            //printf("Setting to play because %i > %i\n", item->localsize,
@@ -1423,10 +1424,6 @@
            }
 
            }
-         */
-
-        // try downloading entire file
-        ok_to_play = FALSE;
 
         // if not opened, over cache level and not an href target then try and open it
         if ((!item->opened) && ok_to_play == TRUE) {
