--- orig/opt/wicd/daemon.py	2008-01-03 11:48:40.000000000 -0600
+++ new/opt/wicd/daemon.py	2008-03-02 01:50:59.406343101 -0600
@@ -624,7 +624,7 @@
         print 'canceling connection attempt'
         if not self.wifi.ConnectingThread == None:
             self.wifi.ConnectingThread.ShouldDie = True
-        misc.Run("killall dhclient dhclient3 wpa_supplicant")
+        misc.Run("killall dhcpcd dhclient3 wpa_supplicant")
     #end function CancelConnect
 
     ########## WIRED FUNCTIONS
