diff -Naur glog-0.3.3/src/demangle_unittest.cc glog-0.3.3-patched/src/demangle_unittest.cc
--- glog-0.3.3/src/demangle_unittest.cc	2012-01-11 04:14:17.000000000 -0500
+++ glog-0.3.3-patched/src/demangle_unittest.cc	2014-04-24 12:13:53.967735046 -0400
@@ -41,6 +41,11 @@
 #include "googletest.h"
 #include "config.h"
 
+#ifdef HAVE_LIB_GFLAGS
+#include <gflags/gflags.h>
+using namespace gflags;
+#endif
+
 GLOG_DEFINE_bool(demangle_filter, false,
                  "Run demangle_unittest in filter mode");
 
diff -Naur glog-0.3.3/src/logging_unittest.cc glog-0.3.3-patched/src/logging_unittest.cc
--- glog-0.3.3/src/logging_unittest.cc	2013-01-30 23:39:40.000000000 -0500
+++ glog-0.3.3-patched/src/logging_unittest.cc	2014-04-24 12:12:11.977883984 -0400
@@ -61,6 +61,7 @@
 
 #ifdef HAVE_LIB_GFLAGS
 #include <gflags/gflags.h>
+using namespace gflags;
 #endif
 
 #ifdef HAVE_LIB_GMOCK
diff -Naur glog-0.3.3/src/signalhandler_unittest.cc glog-0.3.3-patched/src/signalhandler_unittest.cc
--- glog-0.3.3/src/signalhandler_unittest.cc	2012-01-11 04:14:17.000000000 -0500
+++ glog-0.3.3-patched/src/signalhandler_unittest.cc	2014-04-24 12:15:14.976822615 -0400
@@ -41,6 +41,11 @@
 #include <string>
 #include "glog/logging.h"
 
+#ifdef HAVE_LIB_GFLAGS
+#include <gflags/gflags.h>
+using namespace gflags;
+#endif
+
 using namespace GOOGLE_NAMESPACE;
 
 void* DieInThread(void*) {
diff -Naur glog-0.3.3/src/symbolize_unittest.cc glog-0.3.3-patched/src/symbolize_unittest.cc
--- glog-0.3.3/src/symbolize_unittest.cc	2012-01-11 04:14:17.000000000 -0500
+++ glog-0.3.3-patched/src/symbolize_unittest.cc	2014-04-24 12:14:18.077463477 -0400
@@ -41,6 +41,11 @@
 #include "googletest.h"
 #include "config.h"
 
+#ifdef HAVE_LIB_GFLAGS
+#include <gflags/gflags.h>
+using namespace gflags;
+#endif
+
 using namespace std;
 using namespace GOOGLE_NAMESPACE;
 
diff -Naur glog-0.3.3/src/utilities_unittest.cc glog-0.3.3-patched/src/utilities_unittest.cc
--- glog-0.3.3/src/utilities_unittest.cc	2012-01-11 04:14:17.000000000 -0500
+++ glog-0.3.3-patched/src/utilities_unittest.cc	2014-04-24 12:14:54.963048001 -0400
@@ -33,6 +33,11 @@
 #include "googletest.h"
 #include "glog/logging.h"
 
+#ifdef HAVE_LIB_GFLAGS
+#include <gflags/gflags.h>
+using namespace gflags;
+#endif
+
 using namespace GOOGLE_NAMESPACE;
 
 TEST(utilities, sync_val_compare_and_swap) {
