--- Makefile.back	2011-10-14 00:08:29.754834385 +0400
+++ Makefile	2011-10-14 00:47:36.227996214 +0400
@@ -14,7 +14,7 @@
 EMACS=emacs
 
 # Where local software is found
-prefix=/usr/local
+prefix=${DESTDIR}
 
 # Where local lisp files go.
 lispdir   = $(prefix)/share/emacs/site-lisp
@@ -246,7 +246,7 @@
 
 doc/org.html: doc/org.texi
 	(cd doc && $(TEXI2HTML) --no-split -o org.html org.texi)
-	UTILITIES/manfull.pl doc/org.html
+#	UTILITIES/manfull.pl doc/org.html
 
 doc/orgcard.pdf: doc/orgcard.tex
 	(cd doc && pdftex orgcard.tex)
@@ -277,7 +277,7 @@
 	$(TEXI2HTML) -o doc/guide doc/orgguide.texi
 	UTILITIES/guidesplit.pl doc/guide/*.html
 
-info:	doc/org
+info:	doc/org-7.7
 
 pdf:	doc/org.pdf doc/orgguide.pdf
 
