diff -Naur rumor-1.0.5.orig/src/notator.cc rumor-1.0.5.new/src/notator.cc
--- rumor-1.0.5.orig/src/notator.cc	2010-11-15 18:40:22.000000000 +0100
+++ rumor-1.0.5.new/src/notator.cc	2017-05-31 15:51:42.014855383 +0200
@@ -334,7 +334,7 @@
       }
     } else {
       OctaveTic=(RefPitch_wt>AbsPitch_wt?',':'\'');
-      NumTics=(abs(RefPitch_wt-AbsPitch_wt)+3)/7;
+      NumTics=(RefPitch_wt-AbsPitch_wt+3)/7;
       RefPitch_wt=AbsPitch_wt;
     }
     for (int i=0; i<NumTics; i++){
