diff -ur a/kurso4/main.cpp b/kurso4/main.cpp
--- a/kurso4/main.cpp	2012-06-08 22:08:38.000000000 -0400
+++ b/kurso4/main.cpp	2015-06-13 19:56:29.789088113 -0400
@@ -46,9 +46,12 @@
     QApplication a(argc, argv);
     a.setApplicationName("Kurso4");
 
+    /*
     Loko = QApplication::applicationDirPath();
     if (Loko.right(1) != "/")
         Loko.append("/");
+    */
+    Loko = "/usr/share/kurso/";
 
     QFontDatabase::addApplicationFont(":tiparoj/DejaVuSans.ttf");
     QFontDatabase::addApplicationFont(":tiparoj/DejaVuSans-Bold.ttf");
