diff -Naur mapper-0.8.2.orig/CMakeLists.txt mapper-0.8.2/CMakeLists.txt
--- mapper-0.8.2.orig/CMakeLists.txt	2018-08-12 11:01:50.129865377 +0300
+++ mapper-0.8.2/CMakeLists.txt	2018-08-12 11:03:52.807861119 +0300
@@ -152,7 +152,7 @@
 	set(MAPPER_RUNTIME_DESTINATION "bin")
 	set(MAPPER_LIBRARY_DESTINATION "lib/${Mapper_PACKAGE_NAME}")
 	set(MAPPER_DATA_DESTINATION    "share/${Mapper_PACKAGE_NAME}")
-	set(MAPPER_ABOUT_DESTINATION   "share/doc/${Mapper_PACKAGE_NAME}")
+	set(MAPPER_ABOUT_DESTINATION   "doc/${Mapper_PACKAGE_NAME}-${Mapper_VERSION}")
 endif()
 
 if(CMAKE_CROSSCOMPILING)
