diff -Naur connman-1.20.orig/src/connman-dbus.conf connman-1.20/src/connman-dbus.conf
--- connman-1.20.orig/src/connman-dbus.conf	2011-04-18 09:03:56.000000000 +0000
+++ connman-1.20/src/connman-dbus.conf	2014-01-01 07:52:53.758797399 +0000
@@ -8,6 +8,11 @@
         <allow send_interface="net.connman.Counter"/>
         <allow send_interface="net.connman.Notification"/>
     </policy>
+  <policy group="netdev">
+       <allow send_destination="net.connman"/>
+       <allow send_interface="net.connman.Agent"/>
+       <allow send_interface="net.connman.Counter"/>
+   </policy>
     <policy at_console="true">
         <allow send_destination="net.connman"/>
     </policy>
