diff -urpN a/configure b/configure
--- a/configure	2008-08-20 20:02:59.000000000 +0200
+++ b/configure	2008-12-11 13:48:35.000000000 +0100
@@ -29178,7 +29178,7 @@ done
 			if test "$ost_cv_nat_detected" = "no" ; then
 
 
-for ac_header in linux/netfilter_ipv4.h linux/netfilter_ipv6.h
+for ac_header in linux_netfilter_ipv4.h linux_netfilter_ipv6.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 { echo "$as_me:$LINENO: checking for $ac_header" >&5
diff -urpN a/linux_netfilter_ipv4.h b/linux_netfilter_ipv4.h
--- a/linux_netfilter_ipv4.h	1970-01-01 01:00:00.000000000 +0100
+++ b/linux_netfilter_ipv4.h	2008-12-11 13:44:47.000000000 +0100
@@ -0,0 +1,4 @@
+#include <sys/socket.h>
+#include <linux/in.h>
+#include <linux/in6.h>
+#include <linux/netfilter_ipv4.h>
diff -urpN a/linux_netfilter_ipv6.h b/linux_netfilter_ipv6.h
--- a/linux_netfilter_ipv6.h	1970-01-01 01:00:00.000000000 +0100
+++ b/linux_netfilter_ipv6.h	2008-12-11 13:44:58.000000000 +0100
@@ -0,0 +1,4 @@
+#include <sys/socket.h>
+#include <linux/in.h>
+#include <linux/in6.h>
+#include <linux/netfilter_ipv6.h>
diff -urpN a/m4/ost_socket.m4 b/m4/ost_socket.m4
--- a/m4/ost_socket.m4	2008-06-23 15:02:07.000000000 +0200
+++ b/m4/ost_socket.m4	2008-12-11 13:49:29.000000000 +0100
@@ -92,7 +92,7 @@ AC_DEFUN([OST_SYS_SOCKET],[
 				#endif])
 
 			if test "$ost_cv_nat_detected" = "no" ; then
-				AC_CHECK_HEADERS([linux/netfilter_ipv4.h linux/netfilter_ipv6.h],,,
+				AC_CHECK_HEADERS([linux_netfilter_ipv4.h linux_netfilter_ipv6.h],,,
 					[#ifdef HAVE_LIMITS_H
 					#include <limits.h>
 					#endif])
