--- sonic-visualiser-1.8/sonic-visualiser/sonic-visualiser.pro.implicit-dso	2011-04-08 17:56:20.000000000 +0200
+++ sonic-visualiser-1.8/sonic-visualiser/sonic-visualiser.pro	2011-05-31 23:11:17.157498460 +0200
@@ -23,7 +23,7 @@
 
 contains(DEFINES, BUILD_STATIC):LIBS -= -ljack
 
-LIBS = -L../svapp -L../svgui -L../svcore -lsvapp -lsvgui -lsvcore $$LIBS
+LIBS = -L../svapp -L../svgui -L../svcore -lsvapp -lsvgui -lsvcore -lX11 $$LIBS
 
 PRE_TARGETDEPS += ../svapp/libsvapp.a \
                   ../svgui/libsvgui.a \
