diff -Naur build-5.2-orig/src/CMakeLists.txt build-5.2/src/CMakeLists.txt
--- build-5.2-orig/src/CMakeLists.txt	2009-12-15 06:27:57.000000000 +0000
+++ build-5.2/src/CMakeLists.txt	2010-08-07 01:05:09.775042811 +0100
@@ -12,6 +12,8 @@
 	main.cpp
 	mainwindow.cpp
 	navigationpanel.cpp
+    toolbareditor.cpp
+    toolbarmanager.cpp
 	recentfiles.cpp
 	settings.cpp
 	tab_bookmarks.cpp
@@ -30,6 +32,7 @@
 	dialog_about.ui
 	dialog_setup.ui
 	dialog_topicselector.ui
+    toolbareditor.ui
 	mainwindow.ui
 	navigatorpanel.ui
 	tab_bookmarks.ui
