diff -Naur kchmviewer-5.3/CMakeLists.txt kchmviewer-5.3_patched/CMakeLists.txt
--- kchmviewer-5.3/CMakeLists.txt	2009-12-15 06:02:57.000000000 +0000
+++ kchmviewer-5.3_patched/CMakeLists.txt	2011-03-31 02:02:00.936125068 +0000
@@ -18,7 +18,8 @@
 
 # Run in those subdirectories
 ADD_SUBDIRECTORY( lib/libchmfile )
-ADD_SUBDIRECTORY( lib/kio-msits )
+# Disable use of kio-msits so it uses KDE's own.
+#ADD_SUBDIRECTORY( lib/kio-msits )          
 ADD_SUBDIRECTORY( src )
 ADD_SUBDIRECTORY( po )
 ADD_SUBDIRECTORY( packages )
