--- ext/hpricot_scan/hpricot_scan.c.orig	2026-08-31 18:34:00 UTC
+++ ext/hpricot_scan/hpricot_scan.c
@@ -6691,7 +6691,7 @@ static VALUE hpricot_struct_ref7(VALUE obj) {return H_
 static VALUE hpricot_struct_ref8(VALUE obj) {return H_ELE_GET(obj, 8);}
 static VALUE hpricot_struct_ref9(VALUE obj) {return H_ELE_GET(obj, 9);}
 
-static VALUE (*ref_func[10])() = {
+static VALUE (*ref_func[10])(VALUE) = {
   hpricot_struct_ref0,
   hpricot_struct_ref1,
   hpricot_struct_ref2,
@@ -6715,7 +6715,7 @@ static VALUE hpricot_struct_set7(VALUE obj, VALUE val)
 static VALUE hpricot_struct_set8(VALUE obj, VALUE val) {return H_ELE_SET(obj, 8, val);}
 static VALUE hpricot_struct_set9(VALUE obj, VALUE val) {return H_ELE_SET(obj, 9, val);}
 
-static VALUE (*set_func[10])() = {
+static VALUE (*set_func[10])(VALUE, VALUE) = {
   hpricot_struct_set0,
   hpricot_struct_set1,
   hpricot_struct_set2,
