diff -Naur wmmon-1.4/wmmon.c wmmon-1.4.patched/wmmon.c
--- wmmon-1.4/wmmon.c	2018-04-10 18:22:59.000000000 -0400
+++ wmmon-1.4.patched/wmmon.c	2023-02-14 00:38:41.868867398 -0500
@@ -800,7 +800,7 @@
 		 *  limited buffer. Or just divide by 2 (inefficient). Shouldn't
 		 * matter for graphing (we care about proportions, not numbers).  */
 		while ((getline(&line, &line_size, fp_diskstats)) > 0) {
-			if (strstr(line, "sd") || strstr(line, "sr")) {
+			if (strstr(line, "sd") || strstr(line, "sr") || strstr(line, "nvme")) {
 				p = strtok(line, tokens);
 				/* skip 3 tokens, then use fields from
 				`* linux/Documentation/iostats.txt	     */
