diff -u1 -r a/doc/srm.1.in b/doc/srm.1.in
--- a/doc/srm.1.in	2015-11-04 12:15:38.438987109 +0100
+++ b/doc/srm.1.in	2015-11-04 12:24:27.481981163 +0100
@@ -45,3 +45,3 @@
 \fB\-f\fR, \fB\-\-force\fR
-ignore nonexistent files, never prompt
+ignore non\-existent files, never prompt
 .TP 
diff -u1 -r a/src/main.c b/src/main.c
--- a/src/main.c	2015-11-04 12:52:07.568002966 +0100
+++ b/src/main.c	2015-11-04 12:54:09.159001600 +0100
@@ -90,7 +90,7 @@
 	   "Usage: %s [OPTION]... [FILE]...\n"
-	   "Overwrite and remove (unlink) the files. By default use the 35-pass Gutmann\n"
+	   "Overwrite and remove (unlink) the files. By default it uses the 35-pass Gutmann\n"
 	   "method to overwrite files.\n"
 	   "\n"
-	   "  -d, --directory       ignored (for compatability with rm(1))\n"
-	   "  -f, --force           ignore nonexistant files, never prompt\n"
+	   "  -d, --directory       ignored (for compatibility with rm(1))\n"
+	   "  -f, --force           ignore non-existent files, never prompt\n"
 	   "  -i, --interactive     prompt before any removal\n"
