diff -Naur xtetris-2.6/Xtetris.ad.all xtetris-2.6.patched/Xtetris.ad.all
--- xtetris-2.6/Xtetris.ad.all	1995-02-11 01:38:29.000000000 -0500
+++ xtetris-2.6.patched/Xtetris.ad.all	2024-11-29 01:08:38.882159942 -0500
@@ -4,7 +4,7 @@
 *BorderColor:			white
 *Font:				-*-helvetica-bold-r-*-*-*-120-*-*-*-*-*-*
 *Frame*sensitive:		true
-*Frame.Buttons*width:		80
+*Frame.Buttons*width:		100
 *NewGame.fromVert:		Start
 *NewGame.baseTranslations:	#replace <Btn1Down>,<Btn1Up>:NewGame() notify()
 *Pause.mapWhenManaged:		False
@@ -23,7 +23,7 @@
 *Buttons.fromVert:		Status
 *Buttons.hSpace:		0
 *Buttons.vertDistance:		10
-*Buttons.width:			100
+*Buttons.width:			120
 *Canvas.fromHoriz:		TitleBar
 *Canvas.fromVert:		TitleBar
 *Canvas.height:			480
@@ -65,7 +65,7 @@
 *NextObjectLabel.borderWidth:	0
 *NextObjectLabel.fromVert:	TitleBar
 *NextObjectLabel.vertDistance:	20
-*NextObjectLabel.width:		100
+*NextObjectLabel.width:		120
 *Shadow.fromHoriz:		TitleBar
 *Shadow.fromVert:		Canvas
 *Shadow.height:			16
@@ -76,7 +76,7 @@
 *Status*displayCaret:		False
 *Status*resize:			True
 *Status.Game.string:		"          "
-*Status*width:			80
+*Status*width:			100
 *Status.Level.string:		Level:	0
 *Status.Rows.string:		Rows:	0
 *Status.Score.string:		Score:	0
@@ -85,12 +85,12 @@
 *Status.left:			ChainLeft
 *Status.right:			ChainRight
 *Status.vertDistance:		175
-*Status.width:			100
+*Status.width:			120
 *TitleBar.Font:			-*-new century schoolbook-*-i-*-*-*-120-*-*-*-*-*-*
 *TitleBar.Label:		XTETRIS 2.6
 *TitleBar.foreground:		white
 *TitleBar.height:		20
-*TitleBar.width:		100
+*TitleBar.width:		120
 *Object.Background:		black
 ?.ScoreFrame*font:		-*-clean-bold-r-*-*-16-*-*-*-*-*-*-*
 ?.ScoreFrame*baseTranslations:	#override\
@@ -114,8 +114,8 @@
 	<KeyPress>:Done()
 *AboutText.displayCaret:	False
 *AboutText*Scroll:		whenNeeded
-*AboutText.width:		450
-*AboutText.height:		300
+*AboutText.width:		600
+*AboutText.height:		400
 #ifdef COLOR
 *customization:	    	    	.c
 *Foreground:	wheat
