diff -Naur xtetris-2.6/xtetris.man xtetris-2.6.patched/xtetris.man
--- xtetris-2.6/xtetris.man	1995-02-10 23:53:39.000000000 -0500
+++ xtetris-2.6.patched/xtetris.man	2024-11-28 13:47:55.529395013 -0500
@@ -40,7 +40,7 @@
 .I Xtetris
 requires the installation of an application defaults file, probably in the
 file 
-.I /usr/lib/X11/app-defaults/Xtetris.
+.I /etc/X11/app-defaults/Xtetris.
 If that file does not exist, 
 .I xtetris
 cannot run.  (It will probably print out something like "Width zero.
@@ -91,11 +91,12 @@
 Puts \fIxtetris\fP into Dan Quayle Mode.  "If we do not succeed, then 
 we face the risk of failure."
 .SH FILES
-.IP /usr/lib/X11/tetris_scores
-.IP /usr/lib/X11/app-defaults/Xtetris
-.IP /usr/lib/X11/app-defaults/Xtetris.c
-.IP /usr/lib/X11/app-defaults/Xtetris.bw
+.IP /var/games/xtetris/xtetris-scores
 Top 10 high scores.
+.IP /etc/X11/app-defaults/Xtetris
+.IP /etc/X11/app-defaults/Xtetris.c
+.IP /etc/X11/app-defaults/Xtetris.bw
+Default X resources.
 .SH "ENVIRONMENT VARIABLES"
 .IP XTETRIS
 If set, your desired name for the high score table.
