--- ntop-4.0.1_orig/Makefile.am	2010-04-25 09:52:28.000000000 -0300
+++ ntop-4.0.1/Makefile.am	2010-08-23 18:00:49.000000000 -0300
@@ -403,44 +403,9 @@
 dnetter:
 	@echo ""
 	@echo ""
-	@echo "Preparing "
+	@echo "Skipping download "
 	@echo ""
 
-	@if test -f $(ETTER_PASSIVE).old; then \
-	  echo "...Deleting prior file, $(ETTER_PASSIVE).old..."; \
-	  rm -rf $(ETTER_PASSIVE).old; \
-	fi;
-	@if test -f $(ETTER_PASSIVE); then \
-	  echo "...Moving current $(ETTER_PASSIVE) to $(ETTER_PASSIVE).old"; \
-	  mv $(ETTER_PASSIVE) $(ETTER_PASSIVE).old; \
-	fi;
-	@for file in $(ETTER_PASSIVE) $(ETTER_PASSIVE).gz \
-		$(ETTER_PASSIVE_FILE) $(ETTER_PASSIVE_FILE).gz; do \
-          if test -f $$file; then \
-	     echo "...Deleting prior file, $$file..."; \
-             rm -rf $$file; \
-          fi; \
-        done
-	@echo ""
-	@echo "...Downloading new file..."
-	@wget -O $(ETTER_PASSIVE_FILE) \
-		$(ETTER_PASSIVE_DOWNLOAD_FROM)/$(ETTER_PASSIVE_FILE)?$(ETTER_PASSIVE_DOWNLOAD_PARMS)
-	@echo ""
-	@echo "gziping downloaded file..."
-	@gzip $(ETTER_PASSIVE_FILE)
-	@echo ""
-	@echo ""
-	@if test -f $(ETTER_PASSIVE).old; then \
-	  echo -n "Old file lines were: "; \
-	  gunzip -c $(ETTER_PASSIVE).old | wc -l; \
-	fi;
-	@echo -n "New file lines are:  "
-	@gunzip -c $(ETTER_PASSIVE) | wc -l
-	@echo ""
-	@echo ""
-	@echo "New file is:"
-	@ls -l $(ETTER_PASSIVE)
-
 # ntop census
 census-fail:
 	@echo "This option is no longer available... thanks for trying"
