diff -up OpenCV-2.3.0/3rdparty/CMakeLists.txt.00 OpenCV-2.3.0/3rdparty/CMakeLists.txt
diff -up OpenCV-2.3.0/CMakeLists.txt.00 OpenCV-2.3.0/CMakeLists.txt
--- OpenCV-2.3.0/CMakeLists.txt.00	2011-07-13 23:38:23.168999995 -0500
+++ OpenCV-2.3.0/CMakeLists.txt	2011-07-14 11:44:13.217000018 -0500
@@ -621,9 +621,9 @@ string(REGEX MATCH "[0-9].[0-9]" PYTHON_
 if(UNIX)
     set(PYTHON_PLUGIN_INSTALL_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages/opencv)
     if(APPLE)
-        set(PYTHON_PACKAGES_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages CACHE PATH "Where to install the python packages.")
+        set(PYTHON_PACKAGES_PATH lib${LIB_SUFFIX}/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages CACHE PATH "Where to install the python packages.")
     else() #debian based assumed, install to the dist-packages.
-       set(PYTHON_PACKAGES_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/dist-packages CACHE PATH "Where to install the python packages.")
+       set(PYTHON_PACKAGES_PATH lib${LIB_SUFFIX}/python${PYTHON_VERSION_MAJOR_MINOR}/dist-packages CACHE PATH "Where to install the python packages.")
     endif()
 endif()
 if(WIN32)
@@ -1191,7 +1191,7 @@ endif("${CMAKE_CONFIGURE_LDFLAGS}")
 #                       PROCESS SUBDIRECTORIES:
 # ----------------------------------------------------------------------------
 # Save libs and executables in the same place
-set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/lib CACHE PATH "Output directory for libraries" )
+set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/lib${LIB_SUFFIX} CACHE PATH "Output directory for libraries" )
 set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin CACHE PATH "Output directory for applications" )
 
 if(WIN32)
@@ -1322,19 +1322,19 @@ endif(ANDROID)
 #  Part 2/2: ${BIN_DIR}/unix-install/opencv.pc -> For use *with* "make install"
 # prefix=/usr
 # exec_prefix=${prefix}
-# libdir=${exec_prefix}/lib
+# libdir=${exec_prefix}/lib${LIB_SUFFIX}
 # includedir=${prefix}/include/opencv
 # -------------------------------------------------------------------------------------------
 set(prefix      ${CMAKE_INSTALL_PREFIX})
 set(exec_prefix     "\${prefix}")
-set(libdir      "\${exec_prefix}/lib")
+set(libdir      "\${exec_prefix}/lib${LIB_SUFFIX}")
 set(includedir      "\${prefix}/include")
 set(VERSION             ${OPENCV_VERSION})
 
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/opencv.pc.cmake.in" "${CMAKE_BINARY_DIR}/unix-install/opencv.pc" @ONLY IMMEDIATE)
 
 if(UNIX)
-    install(FILES ${CMAKE_BINARY_DIR}/unix-install/opencv.pc DESTINATION lib/pkgconfig)
+    install(FILES ${CMAKE_BINARY_DIR}/unix-install/opencv.pc DESTINATION lib${LIB_SUFFIX}/pkgconfig)
 endif()
 
 
diff -up OpenCV-2.3.0/OpenCVModule.cmake.00 OpenCV-2.3.0/OpenCVModule.cmake
--- OpenCV-2.3.0/OpenCVModule.cmake.00	2011-07-14 11:27:32.563000017 -0500
+++ OpenCV-2.3.0/OpenCVModule.cmake	2011-07-14 11:27:57.963000018 -0500
@@ -97,8 +97,8 @@ macro(define_opencv_module name)
 
     install(TARGETS ${the_target}
         RUNTIME DESTINATION bin COMPONENT main
-        LIBRARY DESTINATION lib COMPONENT main
-        ARCHIVE DESTINATION lib COMPONENT main)
+        LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+        ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
     install(FILES ${lib_hdrs}
         DESTINATION include/opencv2/${name}
diff -up OpenCV-2.3.0/modules/gpu/CMakeLists.txt.00 OpenCV-2.3.0/modules/gpu/CMakeLists.txt
--- OpenCV-2.3.0/modules/gpu/CMakeLists.txt.00	2011-07-14 11:17:29.914000015 -0500
+++ OpenCV-2.3.0/modules/gpu/CMakeLists.txt	2011-07-14 11:18:25.774000018 -0500
@@ -149,8 +149,8 @@ add_dependencies(${the_target} ${DEPS})
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 install(FILES ${lib_hdrs}
 	DESTINATION include/opencv2/${name}
diff -up OpenCV-2.3.0/modules/highgui/CMakeLists.txt.00 OpenCV-2.3.0/modules/highgui/CMakeLists.txt
--- OpenCV-2.3.0/modules/highgui/CMakeLists.txt.00	2011-07-14 11:25:22.909000017 -0500
+++ OpenCV-2.3.0/modules/highgui/CMakeLists.txt	2011-07-14 11:25:53.793000017 -0500
@@ -362,8 +362,8 @@ endif()
 
 install(TARGETS ${the_target}
     RUNTIME DESTINATION bin COMPONENT main
-    LIBRARY DESTINATION lib COMPONENT main
-    ARCHIVE DESTINATION lib COMPONENT main)
+    LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+    ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 install(FILES ${highgui_ext_hdrs}
         DESTINATION include/opencv2/highgui
