diff -Naur OpenCV-2.1.0.orig/3rdparty/flann/CMakeLists.txt OpenCV-2.1.0/3rdparty/flann/CMakeLists.txt
--- OpenCV-2.1.0.orig/3rdparty/flann/CMakeLists.txt	2010-04-11 10:51:39.694283320 +0200
+++ OpenCV-2.1.0/3rdparty/flann/CMakeLists.txt	2010-04-11 10:56:06.674279640 +0200
@@ -89,8 +89,8 @@
 INSTALL (
     TARGETS flann  # flann_s
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
 )
 
 # INSTALL (
diff -Naur OpenCV-2.1.0.orig/CMakeLists.txt OpenCV-2.1.0/CMakeLists.txt
--- OpenCV-2.1.0.orig/CMakeLists.txt	2010-04-11 10:51:39.564279640 +0200
+++ OpenCV-2.1.0/CMakeLists.txt	2010-04-11 10:55:04.824279480 +0200
@@ -467,8 +467,8 @@
 
     string(REGEX MATCH "[0-9].[0-9]" PYTHON_VERSION_MAJOR_MINOR "${PYTHON_VERSION_FULL}")
     if(UNIX)
-        set(PYTHON_PLUGIN_INSTALL_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages/opencv)
-        set(PYTHON_PACKAGES_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages)
+        set(PYTHON_PLUGIN_INSTALL_PATH lib${LIB_SUFFIX}/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages/opencv)
+        set(PYTHON_PACKAGES_PATH lib${LIB_SUFFIX}/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages)
     endif()
     if(WIN32)
         get_filename_component(PYTHON_PATH "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\${PYTHON_VERSION_MAJOR_MINOR}\\InstallPath]" ABSOLUTE CACHE)
@@ -895,7 +895,7 @@
 #                       PROCESS SUBDIRECTORIES:
 # ----------------------------------------------------------------------------
 # Save libs and executables in the same place
-set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/lib CACHE PATH "Output directory for libraries" )
+set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/lib${LIB_SUFFIX} CACHE PATH "Output directory for libraries" )
 set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin CACHE PATH "Output directory for applications" )
 
 if(WIN32)
@@ -965,19 +965,19 @@
 #  Part 2/2: ${BIN_DIR}/unix-install/opencv.pc -> For use *with* "make install"
 # prefix=/usr
 # exec_prefix=${prefix}
-# libdir=${exec_prefix}/lib
+# libdir=${exec_prefix}/lib${LIB_SUFFIX}
 # includedir=${prefix}/include/opencv
 # -------------------------------------------------------------------------------------------
 set(prefix      ${CMAKE_INSTALL_PREFIX})
 set(exec_prefix     "\${prefix}")
-set(libdir      "\${exec_prefix}/lib")
+set(libdir      "\${exec_prefix}/lib${LIB_SUFFIX}")
 set(includedir      "\${prefix}/include")
 set(VERSION             ${OPENCV_VERSION})
 
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/opencv.pc.cmake.in" "${CMAKE_BINARY_DIR}/unix-install/opencv.pc" @ONLY IMMEDIATE)
 
 if(UNIX)
-    install(FILES ${CMAKE_BINARY_DIR}/unix-install/opencv.pc DESTINATION lib/pkgconfig)
+    install(FILES ${CMAKE_BINARY_DIR}/unix-install/opencv.pc DESTINATION lib${LIB_SUFFIX}/pkgconfig)
 endif()
 
 
diff -Naur OpenCV-2.1.0.orig/interfaces/ffopencv/CMakeLists.txt OpenCV-2.1.0/interfaces/ffopencv/CMakeLists.txt
--- OpenCV-2.1.0.orig/interfaces/ffopencv/CMakeLists.txt	2010-04-11 10:51:39.634281600 +0200
+++ OpenCV-2.1.0/interfaces/ffopencv/CMakeLists.txt	2010-04-11 10:56:58.114279520 +0200
@@ -40,6 +40,6 @@
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
     )
diff -Naur OpenCV-2.1.0.orig/src/cv/CMakeLists.txt OpenCV-2.1.0/src/cv/CMakeLists.txt
--- OpenCV-2.1.0.orig/src/cv/CMakeLists.txt	2010-04-11 10:51:39.644283200 +0200
+++ OpenCV-2.1.0/src/cv/CMakeLists.txt	2010-04-11 10:58:19.664279600 +0200
@@ -71,8 +71,8 @@
 
 install(TARGETS ${the_target}
     RUNTIME DESTINATION bin COMPONENT main
-    LIBRARY DESTINATION lib COMPONENT main
-    ARCHIVE DESTINATION lib COMPONENT main)
+    LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+    ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 install(FILES ${lib_hdrs}
     DESTINATION include/opencv
diff -Naur OpenCV-2.1.0.orig/src/cvaux/CMakeLists.txt OpenCV-2.1.0/src/cvaux/CMakeLists.txt
--- OpenCV-2.1.0.orig/src/cvaux/CMakeLists.txt	2010-04-11 10:51:39.634281600 +0200
+++ OpenCV-2.1.0/src/cvaux/CMakeLists.txt	2010-04-11 10:58:43.064279480 +0200
@@ -76,8 +76,8 @@
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 install(FILES ${aux_hdrs_ext}
         DESTINATION include/opencv
diff -Naur OpenCV-2.1.0.orig/src/cxcore/CMakeLists.txt OpenCV-2.1.0/src/cxcore/CMakeLists.txt
--- OpenCV-2.1.0.orig/src/cxcore/CMakeLists.txt	2010-04-11 10:51:39.654284200 +0200
+++ OpenCV-2.1.0/src/cxcore/CMakeLists.txt	2010-04-11 10:59:19.454279800 +0200
@@ -71,8 +71,8 @@
 
 install(TARGETS ${the_target}
     RUNTIME DESTINATION bin COMPONENT main
-    LIBRARY DESTINATION lib COMPONENT main
-    ARCHIVE DESTINATION lib COMPONENT main)
+    LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+    ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 # install each module headers separately
 install(FILES ${lib_hdrs}
diff -Naur OpenCV-2.1.0.orig/src/highgui/CMakeLists.txt OpenCV-2.1.0/src/highgui/CMakeLists.txt
--- OpenCV-2.1.0.orig/src/highgui/CMakeLists.txt	2010-04-11 10:51:39.634281600 +0200
+++ OpenCV-2.1.0/src/highgui/CMakeLists.txt	2010-04-11 11:00:25.174279600 +0200
@@ -282,8 +282,8 @@
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 install(FILES ${highgui_ext_hdrs}
         DESTINATION include/opencv
diff -Naur OpenCV-2.1.0.orig/src/ml/CMakeLists.txt OpenCV-2.1.0/src/ml/CMakeLists.txt
--- OpenCV-2.1.0.orig/src/ml/CMakeLists.txt	2010-04-11 10:51:39.654284200 +0200
+++ OpenCV-2.1.0/src/ml/CMakeLists.txt	2010-04-11 11:00:43.714279320 +0200
@@ -61,8 +61,8 @@
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 install(FILES ${lib_hdrs}
         DESTINATION include/opencv
diff -Naur OpenCV-2.1.0.orig/tests/cxts/CMakeLists.txt OpenCV-2.1.0/tests/cxts/CMakeLists.txt
--- OpenCV-2.1.0.orig/tests/cxts/CMakeLists.txt	2010-04-11 10:51:39.724279600 +0200
+++ OpenCV-2.1.0/tests/cxts/CMakeLists.txt	2010-04-11 11:01:36.324279440 +0200
@@ -63,6 +63,6 @@
 if(WIN32)
 install(TARGETS ${the_target}
     RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 endif()
