diff -up OpenCV-2.3.1/CMakeLists.txt.00 OpenCV-2.3.1/CMakeLists.txt
--- OpenCV-2.3.1/CMakeLists.txt.00	2011-12-28 14:46:28.825000098 -0600
+++ OpenCV-2.3.1/CMakeLists.txt	2011-12-28 14:48:31.277000108 -0600
@@ -18,7 +18,7 @@ if(NOT CMAKE_TOOLCHAIN_FILE)
     # Add these standard paths to the search paths for FIND_LIBRARY
     # to find libraries from these locations first
     if(UNIX)
-        set(CMAKE_LIBRARY_PATH ${CMAKE_LIBRARY_PATH} /lib /usr/lib)
+        set(CMAKE_LIBRARY_PATH ${CMAKE_LIBRARY_PATH} /lib${LIB_SUFFIX} /usr/lib${LIB_SUFFIX})
     endif()
 
     # it _must_ go before PROJECT(OpenCV) in order to work
@@ -117,7 +117,7 @@ endif()
 
 # SUBDIRECTORIES:
 # Save libs and executables in the same place
-set(LIBRARY_OUTPUT_PATH "${CMAKE_BINARY_DIR}/lib" CACHE PATH "Output directory for libraries" )
+set(LIBRARY_OUTPUT_PATH "${CMAKE_BINARY_DIR}/lib${LIB_SUFFIX}" CACHE PATH "Output directory for libraries" )
 set(EXECUTABLE_OUTPUT_PATH "${CMAKE_BINARY_DIR}/bin" CACHE PATH "Output directory for applications" )
 
 if(ANDROID OR WIN32)
@@ -665,9 +665,9 @@ if(CMAKE_HOST_UNIX)
                     OUTPUT_VARIABLE PYTHON_STD_PACKAGES_PATH
                     OUTPUT_STRIP_TRAILING_WHITESPACE)
     if("${PYTHON_STD_PACKAGES_PATH}" MATCHES "site-packages")
-        set(PYTHON_PACKAGES_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages CACHE PATH "Where to install the python packages.")
+        set(PYTHON_PACKAGES_PATH lib${LIB_SUFFIX}/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages CACHE PATH "Where to install the python packages.")
     else() #debian based assumed, install to the dist-packages.
-        set(PYTHON_PACKAGES_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/dist-packages CACHE PATH "Where to install the python packages.")
+        set(PYTHON_PACKAGES_PATH lib${LIB_SUFFIX}/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages CACHE PATH "Where to install the python packages.")
     endif()
 endif()
 if(CMAKE_HOST_WIN32)
