diff -baur opencv-2.4.7.orig/cmake/OpenCVGenPkgconfig.cmake opencv-2.4.7/cmake/OpenCVGenPkgconfig.cmake
--- opencv-2.4.7.orig/cmake/OpenCVGenPkgconfig.cmake	2013-11-12 20:31:27.253763202 +0000
+++ opencv-2.4.7/cmake/OpenCVGenPkgconfig.cmake	2013-11-12 20:41:48.936262732 +0000
@@ -10,7 +10,7 @@
 # -------------------------------------------------------------------------------------------
 set(prefix      "${CMAKE_INSTALL_PREFIX}")
 set(exec_prefix "\${prefix}")
-set(libdir      "\${exec_prefix}/${OPENCV_LIB_INSTALL_PATH}")
+set(libdir      "\${prefix}/${OPENCV_LIB_INSTALL_PATH}")
 set(includedir  "\${prefix}/${OPENCV_INCLUDE_INSTALL_PATH}")
 
 if(CMAKE_BUILD_TYPE MATCHES "Release")
