diff -up OpenCV-2.3.1/modules/highgui/src/cap_ffmpeg_impl.hpp.00 OpenCV-2.3.1/modules/highgui/src/cap_ffmpeg_impl.hpp
--- OpenCV-2.3.1/modules/highgui/src/cap_ffmpeg_impl.hpp.00	2011-12-28 14:50:07.038000125 -0600
+++ OpenCV-2.3.1/modules/highgui/src/cap_ffmpeg_impl.hpp	2011-12-28 14:51:11.687000116 -0600
@@ -104,6 +104,7 @@ extern "C" {
 #endif
 #if defined(HAVE_LIBAVCODEC_AVCODEC_H)
   #include <libavcodec/avcodec.h>
+  #include <libavutil/mathematics.h>
 #endif
 #if defined(HAVE_LIBSWSCALE_SWSCALE_H)
   #include <libswscale/swscale.h>
