diff -Naurp opencv-2.4.9-orig/CMakeLists.txt opencv-2.4.9/CMakeLists.txt
--- opencv-2.4.9-orig/CMakeLists.txt	2014-04-11 12:15:26.000000000 +0200
+++ opencv-2.4.9/CMakeLists.txt	2014-06-26 15:06:18.705309828 +0200
@@ -258,9 +258,9 @@ endif()
 if(ANDROID OR WIN32)
   set(OPENCV_DOC_INSTALL_PATH doc)
 elseif(INSTALL_TO_MANGLED_PATHS)
-  set(OPENCV_DOC_INSTALL_PATH share/OpenCV-${OPENCV_VERSION}/doc)
+  set(OPENCV_DOC_INSTALL_PATH share/doc/opencv-${OPENCV_VERSION})
 else()
-  set(OPENCV_DOC_INSTALL_PATH share/OpenCV/doc)
+  set(OPENCV_DOC_INSTALL_PATH share/doc/opencv)
 endif()
 
 if(WIN32)
@@ -313,8 +313,8 @@ else()
     set(OPENCV_SAMPLES_SRC_INSTALL_PATH    samples/native)
   else()
     set(OPENCV_LIB_INSTALL_PATH     lib${LIB_SUFFIX})
-    set(OPENCV_3P_LIB_INSTALL_PATH  share/OpenCV/3rdparty/${OPENCV_LIB_INSTALL_PATH})
-    set(OPENCV_SAMPLES_SRC_INSTALL_PATH    share/OpenCV/samples)
+    set(OPENCV_3P_LIB_INSTALL_PATH  share/opencv/3rdparty/${OPENCV_LIB_INSTALL_PATH})
+    set(OPENCV_SAMPLES_SRC_INSTALL_PATH    share/opencv/samples)
   endif()
   set(OPENCV_INCLUDE_INSTALL_PATH "include")
 
@@ -322,7 +322,7 @@ else()
   if(LIB_SUFFIX AND NOT SIZEOF_VOID_P_BITS EQUAL LIB_SUFFIX)
     set(OPENCV_CONFIG_INSTALL_PATH lib${LIB_SUFFIX}/cmake/opencv)
   else()
-    set(OPENCV_CONFIG_INSTALL_PATH share/OpenCV)
+    set(OPENCV_CONFIG_INSTALL_PATH share/opencv)
   endif()
 endif()
 
diff -Naurp opencv-2.4.9-orig/cmake/templates/opencv_run_all_tests_unix.sh.in opencv-2.4.9/cmake/templates/opencv_run_all_tests_unix.sh.in
--- opencv-2.4.9-orig/cmake/templates/opencv_run_all_tests_unix.sh.in	2014-04-11 12:15:26.000000000 +0200
+++ opencv-2.4.9/cmake/templates/opencv_run_all_tests_unix.sh.in	2014-06-26 15:09:21.456025129 +0200
@@ -1,7 +1,7 @@
 # Test binaries and data paths
 
 OPENCV_TEST_PATH=@CMAKE_INSTALL_PREFIX@/@OPENCV_TEST_INSTALL_PATH@
 OPENCV_PYTHON_TESTS=@OPENCV_PYTHON_TESTS_LIST@
-export OPENCV_TEST_DATA_PATH=@CMAKE_INSTALL_PREFIX@/share/OpenCV/testdata
+export OPENCV_TEST_DATA_PATH=@CMAKE_INSTALL_PREFIX@/share/opencv/testdata
 
 # Run tests
 
 SUMMARY_STATUS=0
 for t in "$OPENCV_TEST_PATH/"opencv_test_* "$OPENCV_TEST_PATH/"opencv_perf_*;
@@ -22,4 +22,4 @@ else
   echo "OpenCV tests finished with status $SUMMARY_STATUS"
 fi
 
-return $SUMMARY_STATUS
\ Manca newline alla fine del file
+return $SUMMARY_STATUS
diff -Naurp opencv-2.4.9-orig/data/CMakeLists.txt opencv-2.4.9/data/CMakeLists.txt
--- opencv-2.4.9-orig/data/CMakeLists.txt	2014-04-11 12:15:26.000000000 +0200
+++ opencv-2.4.9/data/CMakeLists.txt	2014-06-26 15:07:54.202115953 +0200
@@ -5,8 +5,8 @@ if(ANDROID)
   install(FILES ${HAAR_CASCADES} DESTINATION sdk/etc/haarcascades COMPONENT libs)
   install(FILES ${LBP_CASCADES}  DESTINATION sdk/etc/lbpcascades  COMPONENT libs)
 else()
-  install(FILES ${HAAR_CASCADES} DESTINATION share/OpenCV/haarcascades COMPONENT libs)
-  install(FILES ${LBP_CASCADES}  DESTINATION share/OpenCV/lbpcascades  COMPONENT libs)
+  install(FILES ${HAAR_CASCADES} DESTINATION share/opencv/haarcascades COMPONENT libs)
+  install(FILES ${LBP_CASCADES}  DESTINATION share/opencv/lbpcascades  COMPONENT libs)
 endif()
 
 if(INSTALL_TESTS AND OPENCV_TEST_DATA_PATH)
@@ -17,6 +17,6 @@ if(INSTALL_TESTS AND OPENCV_TEST_DATA_PA
     install(DIRECTORY ${OPENCV_TEST_DATA_PATH}
             DIRECTORY_PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE
               GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-            DESTINATION share/OpenCV/testdata COMPONENT tests)
+            DESTINATION share/opencv/testdata COMPONENT tests)
   endif()
-endif()
\ Manca newline alla fine del file
+endif()
diff -Naurp opencv-2.4.9-orig/modules/java/CMakeLists.txt opencv-2.4.9/modules/java/CMakeLists.txt
--- opencv-2.4.9-orig/modules/java/CMakeLists.txt	2014-04-11 12:15:26.000000000 +0200
+++ opencv-2.4.9/modules/java/CMakeLists.txt	2014-06-26 15:08:28.534686737 +0200
@@ -280,7 +280,7 @@ else(ANDROID)
   if(WIN32)
     set(JAR_INSTALL_DIR java)
   else(WIN32)
-    set(JAR_INSTALL_DIR share/OpenCV/java)
+    set(JAR_INSTALL_DIR share/opencv/java)
   endif(WIN32)
   install(FILES ${JAR_FILE} DESTINATION ${JAR_INSTALL_DIR} COMPONENT java)
 endif(ANDROID)

