--- opencv_contrib-4.0.0/modules/cvv/src/qtutil/filter/sobelfilterwidget.hpp.orig	2018-11-18 11:36:23.000000000 +1000
+++ opencv_contrib-4.0.0/modules/cvv/src/qtutil/filter/sobelfilterwidget.hpp	2018-11-28 15:39:14.367000000 +1000
@@ -11,6 +11,10 @@
 #include "grayfilterwidget.hpp"
 #include "channelreorderfilter.hpp"
 
+enum SpecialFilter {
+    FILTER_SCHARR = -1
+};
+
 namespace cvv
 {
 namespace qtutil
