--- CMakeLists.txt.orig	2017-12-23 05:46:21.000000000 +1000
+++ CMakeLists.txt	2018-02-06 11:04:07.434035665 +1000
@@ -351,7 +351,7 @@
 if(ANDROID OR WIN32)
   ocv_update(OPENCV_DOC_INSTALL_PATH doc)
 else()
-  ocv_update(OPENCV_DOC_INSTALL_PATH share/OpenCV/doc)
+  ocv_update(OPENCV_DOC_INSTALL_PATH share/opencv/doc)
 endif()
 
 if(WIN32 AND CMAKE_HOST_SYSTEM_NAME MATCHES Windows)
@@ -392,7 +392,7 @@
 elseif(WIN32)
   ocv_update(OPENCV_TEST_DATA_INSTALL_PATH "testdata")
 else()
-  ocv_update(OPENCV_TEST_DATA_INSTALL_PATH "share/OpenCV/testdata")
+  ocv_update(OPENCV_TEST_DATA_INSTALL_PATH "share/opencv/testdata")
 endif()
 
 if(ANDROID)
@@ -421,18 +421,18 @@
     ocv_update(OPENCV_CONFIG_INSTALL_PATH  ".")
   else()
     include(GNUInstallDirs)
-    ocv_update(OPENCV_LIB_INSTALL_PATH     ${CMAKE_INSTALL_LIBDIR}${LIB_SUFFIX})
-    ocv_update(OPENCV_3P_LIB_INSTALL_PATH  share/OpenCV/3rdparty/${OPENCV_LIB_INSTALL_PATH})
-    ocv_update(OPENCV_SAMPLES_SRC_INSTALL_PATH    share/OpenCV/samples)
-    ocv_update(OPENCV_JAR_INSTALL_PATH share/OpenCV/java)
-    ocv_update(OPENCV_OTHER_INSTALL_PATH   share/OpenCV)
+    ocv_update(OPENCV_LIB_INSTALL_PATH     ${CMAKE_INSTALL_LIBDIR})
+    ocv_update(OPENCV_3P_LIB_INSTALL_PATH  share/opencv/3rdparty/${OPENCV_LIB_INSTALL_PATH})
+    ocv_update(OPENCV_SAMPLES_SRC_INSTALL_PATH    share/opencv/samples)
+    ocv_update(OPENCV_JAR_INSTALL_PATH share/opencv/java)
+    ocv_update(OPENCV_OTHER_INSTALL_PATH   share/opencv)
 
     if(NOT DEFINED OPENCV_CONFIG_INSTALL_PATH)
       math(EXPR SIZEOF_VOID_P_BITS "8 * ${CMAKE_SIZEOF_VOID_P}")
       if(LIB_SUFFIX AND NOT SIZEOF_VOID_P_BITS EQUAL LIB_SUFFIX)
-        ocv_update(OPENCV_CONFIG_INSTALL_PATH ${CMAKE_INSTALL_LIBDIR}${LIB_SUFFIX}/cmake/opencv)
+        ocv_update(OPENCV_CONFIG_INSTALL_PATH ${CMAKE_INSTALL_LIBDIR}/cmake/opencv)
       else()
-        ocv_update(OPENCV_CONFIG_INSTALL_PATH share/OpenCV)
+        ocv_update(OPENCV_CONFIG_INSTALL_PATH share/opencv)
       endif()
     endif()
   endif()
56c56
< export OPENCV_TEST_DATA_PATH=@CMAKE_INSTALL_PREFIX@/share/OpenCV/testdata
---
> export OPENCV_TEST_DATA_PATH=@CMAKE_INSTALL_PREFIX@/share/opencv/testdata
