Index: libmusicbrainz-3.0.2/libmusicbrainz3.pc.cmake
===================================================================
--- libmusicbrainz-3.0.2.orig/libmusicbrainz3.pc.cmake	2008-12-22 21:53:56.000000000 +0000
+++ libmusicbrainz-3.0.2/libmusicbrainz3.pc.cmake	2008-12-22 21:54:00.000000000 +0000
@@ -7,7 +7,7 @@
 Description: The Musicbrainz Client Library.
 URL: http://musicbrainz.org/doc/libmusicbrainz
 Version: ${PROJECT_VERSION}
-Requires: neon >= 0.25, libdiscid
+Requires.private: neon >= 0.25, libdiscid
 Libs: -L${LIB_INSTALL_DIR} -lmusicbrainz3
 Cflags: -I${INCLUDE_INSTALL_DIR}
 
