diff -Nur nss-mdns-0.10.orig/src/Makefile.am nss-mdns-0.10/src/Makefile.am
--- nss-mdns-0.10.orig/src/Makefile.am	2007-02-08 10:42:23.000000000 -0600
+++ nss-mdns-0.10/src/Makefile.am	2019-08-28 23:24:49.690739951 -0500
@@ -22,7 +22,7 @@
 AM_CFLAGS = \
 	-DMDNS_ALLOW_FILE=\"$(sysconfdir)/mdns.allow\" \
 	-DRESOLV_CONF_FILE=\"$(sysconfdir)/resolv.conf\" \
-	-DAVAHI_SOCKET=\"$(localstatedir)/run/avahi-daemon/socket\"
+	-DAVAHI_SOCKET=\"/run/avahi-daemon/socket\"
 
 #AM_CFLAGS += -DNDEBUG=1 -Os
 
diff -Nur nss-mdns-0.10.orig/src/Makefile.in nss-mdns-0.10/src/Makefile.in
--- nss-mdns-0.10.orig/src/Makefile.in	2007-05-12 09:55:34.000000000 -0500
+++ nss-mdns-0.10/src/Makefile.in	2019-08-28 23:24:39.273681707 -0500
@@ -409,7 +409,7 @@
 # This cool debug trap works on i386/gcc only
 AM_CFLAGS = -DMDNS_ALLOW_FILE=\"$(sysconfdir)/mdns.allow\" \
 	-DRESOLV_CONF_FILE=\"$(sysconfdir)/resolv.conf\" \
-	-DAVAHI_SOCKET=\"$(localstatedir)/run/avahi-daemon/socket\" \
+	-DAVAHI_SOCKET=\"/run/avahi-daemon/socket\" \
 	'-DDEBUG_TRAP=__asm__("int $$3")'
 AM_LDFLAGS = -avoid-version -module -export-dynamic
 lib_LTLIBRARIES = $(am__append_1) $(am__append_2)
