diff --git a/src/sigc++/signal_base.h b/src/sigc++/signal_base.h
index 582a8f4..0d77ac5 100644
--- a/src/sigc++/signal_base.h
+++ b/src/sigc++/signal_base.h
@@ -21,6 +21,7 @@
 #ifndef _SIGC_SIGNAL_BASE_H_
 #define _SIGC_SIGNAL_BASE_H_
 
+#include <cstddef>
 #include <list>
 #include <sigc++config.h>
 #include <sigc++/type_traits.h>
diff --git a/stardict-plugins/stardict-wordnet-plugin/scene.hpp b/stardict-plugins/stardict-wordnet-plugin/scene.hpp
index 54003ca..270b2e2 100644
--- a/stardict-plugins/stardict-wordnet-plugin/scene.hpp
+++ b/stardict-plugins/stardict-wordnet-plugin/scene.hpp
@@ -9,6 +9,7 @@
 #include "partic.hpp"
 #include "spring.hpp"
 
+#include <cstdlib>
 #include <vector>
 using namespace std;
 
