--- session.cpp.orig	2011-09-08 00:59:26.732260001 +0200
+++ iTestServer/session.cpp	2011-09-08 01:00:36.932260001 +0200
@@ -111,7 +111,7 @@
 
 void Session::addLogEntry(int bgr, int bgg, int bgb, int fgr, int fgg, int fgb, const QString & text)
 {
-	s_log << LogEntry::LogEntry(bgr, bgg, bgb, fgr, fgg, fgb, text);
+	s_log << LogEntry(bgr, bgg, bgb, fgr, fgg, fgb, text);
 }
 
 int Session::numLogEntries() { return s_log.count(); }