diff -urN speedcrunch-0.10.orig/src/gui/aboutbox.cpp speedcrunch-0.10/src/gui/aboutbox.cpp
--- speedcrunch-0.10.orig/src/gui/aboutbox.cpp	2008-03-31 00:31:18.000000000 -0700
+++ speedcrunch-0.10/src/gui/aboutbox.cpp	2008-05-31 13:21:39.000000000 -0700
@@ -209,7 +209,7 @@
   infoLabel->setOpenExternalLinks( true );
 
   QLabel * iconLabel = new QLabel( this );
-  iconLabel->setPixmap( QPixmap( ":/crunch.png" ) );
+  iconLabel->setPixmap( QPixmap( ":/speedcrunch.png" ) );
   iconLabel->setAlignment( Qt::AlignVCenter | Qt::AlignRight );
 
   MarqueeText * marqueeText = new MarqueeText( this );
