diff -Naur desmume-0.9.11.orig/src/gtk/osmesa_3Demu.cpp desmume-0.9.11/src/gtk/osmesa_3Demu.cpp
--- desmume-0.9.11.orig/src/gtk/osmesa_3Demu.cpp	2015-02-14 15:05:26.000000000 +0000
+++ desmume-0.9.11/src/gtk/osmesa_3Demu.cpp	2015-05-09 02:00:11.279367208 +0000
@@ -20,6 +20,7 @@
 
 #ifdef HAVE_LIBOSMESA
 #include <stdlib.h>
+#include <stdio.h>
 #include <GL/osmesa.h>
 #include "../OGLRender.h"
 #include "osmesa_3Demu.h"
