--- Makefile.old	2006-03-08 14:59:53.000000000 -0500
+++ Makefile	2006-03-08 15:00:23.000000000 -0500
@@ -7,7 +7,7 @@
 OPT_MOD=-D__KERNEL__ -DMODULE -fomit-frame-pointer -fno-strict-aliasing -pipe -mpreferred-stack-boundary=2
 #LD_MOD=-r
 
-all: sdel-lib.o srm sfill sswap smem sdel-mod.o
+all: sdel-lib.o srm sfill sswap smem 
 	@echo
 	@echo "A Puritan is someone who is deathly afraid that someone, somewhere, is"
 	@echo "having fun."
@@ -15,9 +15,6 @@
 	@echo "I hope YOU have fun!"
 	@echo
 
-sdel-mod.o: sdel-mod.c
-	$(CC) $(OPT) $(OPT_MOD) $(LD_MOD) -I/lib/modules/`uname -r`/build/include -c sdel-mod.c
-
 sdel-lib.o: sdel-lib.c
 	$(CC) ${OPT} -c sdel-lib.c
 
@@ -35,7 +32,7 @@
 	-strip smem
 
 clean:
-	rm -f sfill srm sswap smem sdel sdel-lib.o sdel-mod.o core *~
+	rm -f sfill srm sswap smem sdel sdel-lib.o 
 
 install: all
 	mkdir -p -m 755 ${INSTALL_DIR} 2> /dev/null
@@ -47,8 +44,8 @@
 	chmod 644 ${MAN_DIR}/man1/srm.1 ${MAN_DIR}/man1/sfill.1 ${MAN_DIR}/man1/sswap.1 ${MAN_DIR}/man1/smem.1
 	mkdir -p -m 755 ${DOC_DIR} 2> /dev/null
 	cp -f CHANGES FILES README secure_delete.doc usenix6-gutmann.doc ${DOC_DIR}
-	-test -e sdel-mod.o && cp -f sdel-mod.o /lib/modules/`uname -r`/kernel/drivers/char
-#	@-test '!' -e sdel-mod.o -a `uname -s` = 'Linux' && echo "type \"make sdel-mod install\" to compile and install the Linux loadable kernel module for secure delete"
+	-test -e 
+#	@-test '!' -e 
 	@echo
 	@echo "If men could get pregnant, abortion would be a sacrament."
 	@echo
