--- Python-3.6.0/Include/pyport.h.orig	2017-01-01 07:58:08.575202311 +0700
+++ Python-3.6.0/Include/pyport.h	2017-01-01 07:58:20.332234160 +0700
@@ -37,7 +37,7 @@
  * integral synonyms.  Only define the ones we actually need.
  */
 
-// long long is required. Ensure HAVE_LONG_LONG is defined for compatibility.
+/* long long is required. Ensure HAVE_LONG_LONG is defined for compatibility. */
 #ifndef HAVE_LONG_LONG
 #define HAVE_LONG_LONG
 #endif
