--- ./Modules/readline.c.orig	2009-10-26 21:32:51.000000000 +0200
+++ ./Modules/readline.c	2010-05-30 14:07:10.000000000 +0300
@@ -12,6 +12,9 @@
 #include <errno.h>
 #include <sys/time.h>
 
+/* This seems to be needed for set_pre_input_hook to work */
+#define HAVE_RL_PRE_INPUT_HOOK 1
+
 #if defined(HAVE_SETLOCALE)
 /* GNU readline() mistakenly sets the LC_CTYPE locale.
  * This is evil.  Only the user or the app's main() should do this!
