--- exact-image-0.9.2/Makefile.orig	2016-12-12 09:23:11.937243882 +0000
+++ exact-image-0.9.2/Makefile	2016-12-12 09:27:03.967245907 +0000
@@ -48,7 +48,7 @@
 # we have some unimplemented colorspaces in the Image::iterator :-(
 CFLAGS += $(call cc-option,-Wno-switch -Wno-switch-enum,)
 
-CXXFLAGS := $(CFLAGS) -Wno-sign-compare
+CXXFLAGS := $(CFLAGS) -Wno-sign-compare -std=gnu++98
 
 ifeq "$(STATIC)" "1"
 X_EXEFLAGS += -static

