diff -Nur exact-image-0.8.1.orig//api/perl/Makefile exact-image-0.8.1/api/perl/Makefile
--- exact-image-0.8.1.orig//api/perl/Makefile	2008-12-09 03:42:19.000000000 -0600
+++ exact-image-0.8.1/api/perl/Makefile	2010-06-23 08:57:13.015010261 -0500
@@ -20,5 +20,6 @@
 all:: $($(X_MODULE)_BINARY)
 install:: $($(X_MODULE)_BINARY)
 	$(Q)echo "INSTALL PERL module $^"
-	$(Q)mkdir -p $(DESTDIR)$(libdir)/perl5/site_perl/
-	$(Q)install $^ $(dir $^)/ExactImage.pm $(DESTDIR)$(libdir)/perl5/site_perl/
+	eval `perl -V:vendorlib`
+	$(Q)mkdir -p $(DESTDIR)@vendorlib@
+	$(Q)install $^ $(dir $^)/ExactImage.pm $(DESTDIR)@vendorlib@
diff -Nur exact-image-0.8.1.orig//api/python/Makefile exact-image-0.8.1/api/python/Makefile
--- exact-image-0.8.1.orig//api/python/Makefile	2010-04-05 10:28:40.000000000 -0500
+++ exact-image-0.8.1/api/python/Makefile	2010-06-23 08:58:08.206169868 -0500
@@ -20,5 +20,5 @@
 all:: $($(X_MODULE)_BINARY)
 install:: $($(X_MODULE)_BINARY)
 	$(Q)echo "INSTALL PYTHON module $^"
-	$(Q)mkdir -p $(DESTDIR)$(libdir)/python2.5/site-packages/
+	$(Q)mkdir -p $(DESTDIR)$(shell python -c 'from distutils.sysconfig import get_python_lib; print get_python_lib()')
 	$(Q)install $(dir $^)/ExactImage.py $^ $(DESTDIR)$(shell python -c 'from distutils.sysconfig import get_python_lib; print get_python_lib()')/
