diff -Nur exact-image-0.8.3.orig//api/perl/Makefile exact-image-0.8.3/api/perl/Makefile
--- exact-image-0.8.3.orig//api/perl/Makefile
+++ exact-image-0.8.3/api/perl/Makefile
@@ -20,5 +20,6 @@
 all:: $($(X_MODULE)_BINARY)
 install:: $($(X_MODULE)_BINARY)
 	$(Q)echo "INSTALL PERL module $^"
-	$(Q)mkdir -p $(DESTDIR)$(libdir)/perl5/site_perl/
-	$(Q)install $^ $(dir $^)/ExactImage.pm $(DESTDIR)$(libdir)/perl5/site_perl/
+	eval `perl -V:vendorlib`
+	$(Q)mkdir -p $(DESTDIR)@vendorlib@
+	$(Q)install $^ $(dir $^)/ExactImage.pm $(DESTDIR)@vendorlib@
