diff -Nur exact-image-0.8.2.orig//api/perl/Makefile exact-image-0.8.2/api/perl/Makefile
--- exact-image-0.8.2.orig//api/perl/Makefile	2008-12-09 03:42:19.000000000 -0600
+++ exact-image-0.8.2/api/perl/Makefile	2010-06-23 08:57:13.015010261 -0500
@@ -20,5 +20,6 @@
 all:: $($(X_MODULE)_BINARY)
 install:: $($(X_MODULE)_BINARY)
 	$(Q)echo "INSTALL PERL module $^"
-	$(Q)mkdir -p $(DESTDIR)$(libdir)/perl5/site_perl/
-	$(Q)install $^ $(dir $^)/ExactImage.pm $(DESTDIR)$(libdir)/perl5/site_perl/
+	eval `perl -V:vendorlib`
+	$(Q)mkdir -p $(DESTDIR)@vendorlib@
+	$(Q)install $^ $(dir $^)/ExactImage.pm $(DESTDIR)@vendorlib@
