--- exact-image-0.9.2/codecs/raw.cc.orig	2017-07-05 09:07:52.988538841 +0700
+++ exact-image-0.9.2/codecs/raw.cc	2017-07-05 09:08:21.813539168 +0700
@@ -71,7 +71,7 @@
   if (!image.getRawData())
     return false;
 
-  return stream->write ((char*)image.getRawData(), image.stride()*image.h)
+  return (bool) stream->write ((char*)image.getRawData(), image.stride()*image.h)
     /* ==
        (size_t) image.stride()*image.h*/;
 }
