--- src/lib/net/SecureUtils.cpp	2021-11-01 21:46:43.000000000 +0100
+++ src/lib/net/SecureUtils.cpp	2024-02-06 22:12:48.553673626 +0100
@@ -190,7 +190,7 @@
 
     X509_sign(cert, private_key, EVP_sha256());
 
-    auto fp = fopen_utf8_path(path.c_str(), "r");
+    auto fp = fopen_utf8_path(path.c_str(), "w");
     if (!fp) {
         throw std::runtime_error("Could not open certificate output path");
     }
