--- gplates_2.3.0_src/src/gui/TopologyTools.cc	2021-07-05 07:28:32.000000000 +0200
+++ gplates_2.3.0_src-fixed/src/gui/TopologyTools.cc	2022-04-10 16:00:01.000000000 +0200
@@ -39,4 +39,6 @@
 #include <boost/none.hpp>
 #include <boost/numeric/conversion/cast.hpp>
+#include <boost/bind/placeholders.hpp>
+using namespace boost::placeholders;
 
 #include <QtDebug>
--- gplates_2.3.0_src/src/qt-widgets/ViewportWindow.cc 2021-08-05 07:40:35.000000000 +0200
+++ gplates_2.3.0_src-fixed/src/qt-widgets/ViewportWindow.cc 2022-04-10 16:10:01.000000000 +0200
@@ -37,4 +37,6 @@
 #include <boost/foreach.hpp>
 #include <boost/bind/bind.hpp>
+#include <boost/bind/placeholders.hpp>
+using namespace boost::placeholders;
 
 #include <QActionGroup>
--- gplates_2.3.0_src/src/presentation/ReconstructionGeometryRenderer.cc 2021-08-05 07:40:35.000000000 +0200
+++ gplates_2.3.0_src-fixed/src/presentation/ReconstructionGeometryRenderer.cc 2022-04-10 16:20:01.000000000 +0200
@@ -30,4 +30,6 @@
 #include <boost/bind/bind.hpp>
 #include <boost/foreach.hpp>
+#include <boost/bind/placeholders.hpp>
+using namespace boost::placeholders;
 
 #include "ReconstructionGeometryRenderer.h"
--- gplates_2.3.0_src/src/presentation/VisualLayerRegistry.cc 2021-08-05 07:40:35.000000000 +0200
+++ gplates_2.3.0_src-fixed/src/presentation/VisualLayerRegistry.cc 2022-04-10 16:30:01.000000000 +0200
@@ -27,4 +27,6 @@
 #include <boost/bind/bind.hpp>
 #include <boost/foreach.hpp>
+#include <boost/bind/placeholders.hpp>
+using namespace boost::placeholders;
 #include <QPixmap>
