--- Makefile.orig	2024-12-20 00:27:21.000000000 +1100
+++ Makefile	2025-01-06 09:23:36.095804676 +1100
@@ -74,6 +74,8 @@
 	install -D -m 0644 dkms.bash-completion $(DESTDIR)/usr/share/bash-completion/completions/dkms
 	install -D -m 0644 dkms.zsh-completion $(DESTDIR)/usr/share/zsh/site-functions/_dkms
 	install -D -m 0644 dkms.8 $(DESTDIR)/usr/share/man/man8/dkms.8
+	install -D -m 0755 dkms_autoinstaller $(DESTDIR)$(LIBDIR)/dkms_autoinstaller
+	install -D -m 0755 dkms_common.postinst $(DESTDIR)$(LIBDIR)/common.postinst
 
 install-redhat: install
 ifneq (,$(DESTDIR))
