--- Makefile.orig	2018-04-26 01:56:48.000000000 +1000
+++ Makefile	2018-07-09 23:29:27.236000000 +1000
@@ -97,7 +97,7 @@
 
 $(TARBALL):
 	mkdir -p $(deb_destdir)
-	tmp_dir=`mktemp -d --tmpdir dkms.XXXXXXXX` ; \
+	tmp_dir=`mktemp -d -t dkms.XXXXXXXX` ; \
 	cp -a ../$(RELEASE_NAME) $${tmp_dir}/$(RELEASE_STRING) ; \
 	sed -e "s/\[INSERT_VERSION_HERE\]/$(RELEASE_VERSION)/" dkms > $${tmp_dir}/$(RELEASE_STRING)/dkms ; \
 	sed -e "s/\[INSERT_VERSION_HERE\]/$(RELEASE_VERSION)/" dkms.spec > $${tmp_dir}/$(RELEASE_STRING)/dkms.spec ; \
@@ -114,7 +114,7 @@
 
 
 rpm: $(TARBALL) dkms.spec
-	tmp_dir=`mktemp -d --tmpdir dkms.XXXXXXXX` ; \
+	tmp_dir=`mktemp -d -t dkms.XXXXXXXX` ; \
 	echo $(tmp_dir); \
 	mkdir -p $${tmp_dir}/{BUILD,RPMS,SRPMS,SPECS,SOURCES} ; \
 	cp $(TARBALL) $${tmp_dir}/SOURCES ; \
@@ -139,7 +139,7 @@
 	cd -
 
 debs:
-	tmp_dir=`mktemp -d --tmpdir dkms.XXXXXXXX` ; \
+	tmp_dir=`mktemp -d -t dkms.XXXXXXXX` ; \
 	make debmagic DEB_TMP_BUILDDIR=$${tmp_dir} DIST=$(DIST); \
 	rm -rf $${tmp_dir}
 
