diff -Nur xapian-omega-1.2.4.orig//Makefile.in xapian-omega-1.2.4/Makefile.in
--- xapian-omega-1.2.4.orig//Makefile.in	2010-12-19 06:54:47.000000000 -0600
+++ xapian-omega-1.2.4/Makefile.in	2011-01-16 14:19:55.516062731 -0600
@@ -411,7 +411,7 @@
 	xapian-omega.spec
 
 AM_CPPFLAGS = \
-	-DCONFIGFILE_SYSTEM=\"$(sysconfdir)/omega.conf\" \
+	-DCONFIGFILE_SYSTEM=\"$(sysconfdir)/xapian-omega.conf\" \
 	-DPKGLIBBINDIR=\"$(pkglibbindir)\"
 
 pkglibbindir = $(pkglibdir)/bin
