diff -Nur xapian-omega-1.2.8.orig/Makefile.in xapian-omega-1.2.8/Makefile.in
--- xapian-omega-1.2.8.orig/Makefile.in	2011-12-14 11:04:48.000000000 +0530
+++ xapian-omega-1.2.8/Makefile.in	2012-01-18 19:30:25.686917983 +0530
@@ -420,7 +420,7 @@
 	xapian-omega.spec
 
 AM_CPPFLAGS = \
-	-DCONFIGFILE_SYSTEM=\"$(sysconfdir)/omega.conf\" \
+	-DCONFIGFILE_SYSTEM=\"$(sysconfdir)/xapian-omega.conf\" \
 	-DPKGLIBBINDIR=\"$(pkglibbindir)\"
 
 pkglibbindir = $(pkglibdir)/bin
