--- nagios-4.4.6/startup/default-init.in.orig	2020-04-28 22:48:29.000000000 +0200
+++ nagios-4.4.6/startup/default-init.in	2021-07-31 23:32:21.895857400 +0200
@@ -180,6 +180,10 @@
 
 		check_config
 
+		NagiosRunDir=`dirname $NagiosRunFile`
+		mkdir -p $NagiosRunDir
+		chown $NagiosUser:$NagiosGroup $NagiosRunDir
+
 		if test -f $NagiosRunFile; then
 			NagiosPID=`head -n 1 $NagiosRunFile`
 			if status_nagios; then
