--- nagios/html/Makefile.in	2011-07-24 14:53:52.000000000 -0300
+++ nagios_patched/html/Makefile.in	2011-07-26 20:56:28.000000000 -0300
@@ -75,7 +75,7 @@
 	do $(INSTALL) -m 664 $(INSTALL_OPTS) $$file $(DESTDIR)$(HTMLDIR)/images/logos; done
 	for file in includes/*.*; \
 	do $(INSTALL) -m 664 $(INSTALL_OPTS) $$file $(DESTDIR)$(HTMLDIR)/includes; done
-	for file in includes/rss/*; \
+	for file in $$(find includes/rss -type f -maxdepth 1); \
 	do $(INSTALL) -m 664 $(INSTALL_OPTS) $$file $(DESTDIR)$(HTMLDIR)/includes/rss; done
 	for file in includes/rss/extlib/*; \
 	do $(INSTALL) -m 664 $(INSTALL_OPTS) $$file $(DESTDIR)$(HTMLDIR)/includes/rss/extlib; done
