diff -ur cdparanoia-III-10.2.orig/main.c cdparanoia-III-10.2/main.c
--- cdparanoia-III-10.2.orig/main.c	2008-09-11 17:11:02.000000000 -0400
+++ cdparanoia-III-10.2/main.c	2010-09-20 23:24:18.753710419 -0400
@@ -1402,6 +1402,7 @@
 	    int i;
 	    /* read a sector and output the partial offset.  Save the
                rest for the next batch iteration */
+	    paranoia_seek(p, cursor, SEEK_SET);
 	    readbuf=paranoia_read_limited(p,callback,max_retries);
 	    err=cdda_errors(d);mes=cdda_messages(d);
 
