From 01f07c5ad90563dc258eba4134a2544d192d0a49 Mon Sep 17 00:00:00 2001
From: Thomas Vander Stichele <thomas (at) apestaart (dot) org>
Date: Sun, 5 Oct 2014 16:40:59 -0400
Subject: [PATCH] make sure rip cd rip -U works

---
 morituri/rip/cd.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/morituri/rip/cd.py b/morituri/rip/cd.py
index eaf5da4..c787d3f 100644
--- a/morituri/rip/cd.py
+++ b/morituri/rip/cd.py
@@ -123,7 +123,8 @@ def do(self, args):
             "full table's AR URL %s differs from toc AR URL %s" % (
             self.itable.getAccurateRipURL(), self.ittoc.getAccurateRipURL())
 
-        self.program.metadata.discid = self.ittoc.getMusicBrainzDiscId()
+        if self.program.metadata:
+            self.program.metadata.discid = self.ittoc.getMusicBrainzDiscId()
 
         # result
 
