--- hstr-1.22/configure.ac	2017-02-02 12:44:38.000000000 -0800
+++ hstr-1.22.new/configure.ac	2017-02-09 14:26:56.524105894 -0800
@@ -49,7 +49,7 @@
 
 # Checks for libraries.
 AC_CHECK_LIB(m, cos, [], [AC_MSG_ERROR([Could not find m library])])
-AC_CHECK_LIB(readline, using_history, [], [AC_MSG_ERROR([Could not find readline library])])
+AC_CHECK_LIB(readline, using_history, [], [AC_MSG_ERROR([Could not find readline library])],[-lncurses])
 
 # Checks for header files.
 AC_CHECK_HEADER(assert.h)
