--- roadfighter-1.0.1269//build/linux/Makefile.orig	2015-03-28 01:59:05.689956599 +0200
+++ roadfighter-1.0.1269/build/linux/Makefile	2015-03-28 02:00:02.974240447 +0200
@@ -3,11 +3,11 @@
 PREFIX = /usr
 EXTRADIR = ./build/linux
 
-GAMEDIR = $(PREFIX)/games
+GAMEDIR = $(PREFIX)/usr/bin
 STARTUP = $(GAMEDIR)/$(EXE)
-BINDIR = $(PREFIX)/share/games/$(EXE)
-ICNDIR = $(PREFIX)/share/pixmaps
-APPDIR = $(PREFIX)/share/applications
+BINDIR = $(PREFIX)/usr/share/games/$(EXE)
+ICNDIR = $(PREFIX)/usr/share/pixmaps
+APPDIR = $(PREFIX)/usr/share/applications
 
 DATA = fonts graphics maps sound readme.txt
 ICON = roadfighter.png
