diff -Naur zoom-1.1.5/configure zoom-1.1.5.patched/configure
--- zoom-1.1.5/configure	2011-11-05 10:29:03.000000000 -0400
+++ zoom-1.1.5.patched/configure	2012-11-28 02:34:45.000000000 -0500
@@ -7349,21 +7349,8 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 
-    # If Xft is installed, then there will be a xft-config file on the current path
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for xft-config" >&5
-$as_echo_n "checking for xft-config... " >&6; }
-    XFTCONFIG=`which xft-config`
-    XFT_CFLAGS=""
-    XFT_LIBS=""
-    if test -x ${XFTCONFIG}; then
-    	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $XFTCONFIG" >&5
-$as_echo "$XFTCONFIG" >&6; }
-    	XFT_CFLAGS=`${XFTCONFIG} --cflags`
-    	XFT_LIBS=`${XFTCONFIG} --libs`
-    else
-    	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-    fi
+    	XFT_CFLAGS=`pkg-config xft --cflags`
+    	XFT_LIBS=`pkg-config xft --libs`
 
     # Had to move this, Xft 2 again
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for XRenderQueryExtension in -lXrender" >&5
