diff -Naur zoom-1.1.5/src/interp.c zoom-1.1.5.patched/src/interp.c
--- zoom-1.1.5/src/interp.c	2011-11-05 07:40:53.000000000 -0400
+++ zoom-1.1.5.patched/src/interp.c	2017-06-21 03:26:42.543594700 -0400
@@ -101,7 +101,7 @@
 	} \
      }
 
-static inline void push(ZStack* stack, const ZWord word)
+static void push(ZStack* stack, const ZWord word)
 {
   *(stack->stack_top++) = word;
   stack->stack_size--;
@@ -131,7 +131,7 @@
 #endif
 }
 
-inline ZWord pop(ZStack* stack)
+ZWord pop(ZStack* stack)
 {
   stack->stack_size++;
 
@@ -159,7 +159,7 @@
   return *(--stack->stack_top);
 }
 
-inline ZWord top(ZStack* stack)
+ZWord top(ZStack* stack)
 {
     if (stack->current_frame)
     {
@@ -232,7 +232,7 @@
   return newframe;
 }
 
-inline void store(ZStack* stack, int var, ZWord value)
+void store(ZStack* stack, int var, ZWord value)
 {
 #ifdef DEBUG
   printf_debug("Storing %i in Variable #%x\n", value, var);
@@ -253,7 +253,7 @@
     }
 }
 
-inline void store_nopush(ZStack* stack, int var, ZWord value)
+void store_nopush(ZStack* stack, int var, ZWord value)
 {
 #ifdef DEBUG
     printf_debug("Storing %i in Variable #%x\n", value, var);
@@ -313,7 +313,7 @@
   int    isdefault;
 };
 
-static inline struct prop* get_object_prop_3(ZUWord object, ZWord property)
+static struct prop* get_object_prop_3(ZUWord object, ZWord property)
 {
   ZByte* obj;
   ZByte* prop;
@@ -364,7 +364,7 @@
   int header;
 };
 
-static inline struct propinfo* get_object_propinfo_4(ZByte* prop)
+static struct propinfo* get_object_propinfo_4(ZByte* prop)
 {
   static struct propinfo pinfo;
   
@@ -387,7 +387,7 @@
   return &pinfo;
 }
 
-static inline struct prop* get_object_prop_4(ZUWord object, ZWord property)
+static struct prop* get_object_prop_4(ZUWord object, ZWord property)
 {
   ZByte* obj;
   ZByte* prop;
@@ -539,7 +539,7 @@
 }
 #endif
 
-inline static int true_colour(int col)
+static int true_colour(int col)
 {
   switch (col)
     {
@@ -575,7 +575,7 @@
     }
 }
 
-inline static int convert_colour(int col)
+static int convert_colour(int col)
 {
   switch (col)
     {
@@ -1473,7 +1473,7 @@
   return -1;
 }
 
-static inline void zcode_setup_window(int window)
+static void zcode_setup_window(int window)
 {
   v6_set_window(window);
   v6_define_window(window,
@@ -1497,7 +1497,7 @@
 #endif
 }
 
-static inline int zcode_v6_push_stack(ZStack* stack,
+static int zcode_v6_push_stack(ZStack* stack,
 				      ZUWord  stk,
 				      ZUWord  value)
 {
@@ -1528,7 +1528,7 @@
   return 1;
 }
 
-static inline int v6_window(int win)
+static int v6_window(int win)
 {
   if (win > 7)
     zmachine_fatal("No such window: %i", win);
