diff -Naur fdupes-1.51.orig/Makefile fdupes-1.51/Makefile
--- fdupes-1.51.orig/Makefile	2013-04-20 12:51:58.000000000 -0500
+++ fdupes-1.51/Makefile	2013-07-16 09:22:30.169197997 -0500
@@ -46,6 +46,8 @@
 #
 PROGRAM_NAME=fdupes
 
+DESTDIR =
+
 #
 # BIN_DIR indicates directory where program is to be installed. 
 # Suggested value is "$(PREFIX)/bin"
@@ -103,12 +105,12 @@
 	$(CC) $(CFLAGS) -o fdupes $(OBJECT_FILES)
 
 installdirs:
-	test -d $(BIN_DIR) || $(MKDIR) $(BIN_DIR)
-	test -d $(MAN_DIR) || $(MKDIR) $(MAN_DIR)
+	test -d $(DESTDIR)$(BIN_DIR) || $(MKDIR) $(DESTDIR)$(BIN_DIR)
+	test -d $(DESTDIR)$(MAN_DIR) || $(MKDIR) $(DESTDIR)$(MAN_DIR)
 
 install: fdupes installdirs
-	$(INSTALL_PROGRAM)	fdupes   $(BIN_DIR)/$(PROGRAM_NAME)
-	$(INSTALL_DATA)		fdupes.1 $(MAN_DIR)/$(PROGRAM_NAME).$(MAN_EXT)
+	$(INSTALL_PROGRAM)      fdupes   $(DESTDIR)$(BIN_DIR)/$(PROGRAM_NAME)
+	$(INSTALL_DATA)         fdupes.1 $(DESTDIR)$(MAN_DIR)/$(PROGRAM_NAME).$(MAN_EXT)
 
 clean:
 	$(RM) $(OBJECT_FILES)
