Index: Makefile
===================================================================
--- Makefile.orig	2016-08-21 06:54:46.000000000 +0200
+++ Makefile	2016-11-04 13:46:21.037423459 +0100
@@ -11,7 +11,7 @@
 # determination of the actual installation directories.
 # Suggested values are "/usr/local", "/usr", "/pkgs/fdupes-$(VERSION)"
 #
-PREFIX = /usr/local
+PREFIX = /usr
 
 #
 # When compiling for 32-bit systems, FILEOFFSET_64BIT must be enabled
--- Makefile.orig	2019-01-31 06:02:58.000000000 +0200
+++ Makefile	2019-01-31 06:04:24.000000000 +0200
@@ -49,7 +49,7 @@
 # MAN_DIR indicates directory where the fdupes man page is to be 
 # installed. Suggested value is "$(PREFIX)/man/man1"
 #
-MAN_BASE_DIR = $(PREFIX)/share/man
+MAN_BASE_DIR = $(PREFIX)/man
 MAN_DIR = $(MAN_BASE_DIR)/man1
 MAN_EXT = 1
 
