--- openlierox-0.59/CMakeOlxCommon.original.cmake	2014-04-06 13:10:05.423413368 +0300
+++ openlierox-0.59/CMakeOlxCommon.cmake	2014-04-07 04:15:14.367646916 +0300
@@ -320,7 +320,7 @@
 		SET(LIBS ${LIBS} ${Boost_signals_LIBRARIES})
 	ELSE()
 		MESSAGE("boost-signals not found, trying just boost_signals-mt")
-		SET(LIBS ${LIBS} boost_signals-mt)
+		SET(LIBS ${LIBS} "/usr/lib/libboost_signals.so.1.54.0" "/usr/lib/libboost_system.so.1.54.0")
 	ENDIF()
 ENDIF(BOOST_LINK_STATIC)
 
