--- openlierox-0.58_rc3/include/HTTP.original.h	2014-04-19 19:58:09.213990582 +0300
+++ openlierox-0.58_rc3/include/HTTP.h	2014-04-21 13:45:11.485876525 +0300
@@ -21,7 +21,6 @@
 
 #include <string>
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 #include "Networking.h"
--- openlierox-0.58_rc3/src/common/HTTP.original.cpp	2014-04-19 20:48:13.255434896 +0300
+++ openlierox-0.58_rc3/src/common/HTTP.cpp	2014-04-21 13:48:08.493663623 +0300
@@ -25,7 +25,6 @@
 	#include <stdlib.h>
 #endif
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 #include "LieroX.h"
