diff -Naur hsetroot-1.0.2/configure.ac hsetroot-1.0.2.patched/configure.ac
--- hsetroot-1.0.2/configure.ac	2003-09-05 17:41:38.000000000 -0400
+++ hsetroot-1.0.2.patched/configure.ac	2023-01-26 12:45:06.365351344 -0500
@@ -39,15 +39,8 @@
 AC_C_CONST
 
 # Check for imlib2
-AC_CHECK_PROGS(imlib2config_cmd, imlib2-config)
-if test x$imlib2config_cmd = "x"; then
-  AC_MSG_ERROR([error. Imlib2 is required to compile.])
-fi
-
-IMLIB2_CFLAGS=`$imlib2config_cmd --cflags`
+PKG_CHECK_MODULES([IMLIB2], [imlib2])
 AC_SUBST(IMLIB2_CFLAGS)
-
-IMLIB2_LIBS=`$imlib2config_cmd --libs`
 AC_SUBST(IMLIB2_LIBS)
 
 # Some extra definitions for config.h
