Only in highlight-4.1/src/gui-qt: .highlight.pro.un~
diff -cr highlight-4.1.DIST/src/gui-qt/highlight.pro highlight-4.1/src/gui-qt/highlight.pro
*** highlight-4.1.DIST/src/gui-qt/highlight.pro	2021-05-10 18:47:34.000000000 +0100
--- highlight-4.1/src/gui-qt/highlight.pro	2022-07-27 14:39:31.295908472 +0100
***************
*** 36,44 ****
  unix {
      DESTDIR = ../
      LIBS += -L.. -lhighlight
!     LIBS += -llua
      CONFIG += link_pkgconfig
!     PKGCONFIG += lua
  
      # to make it run within Qt Creator
      !contains(DEFINES, DATA_DIR.*) {
--- 36,44 ----
  unix {
      DESTDIR = ../
      LIBS += -L.. -lhighlight
!     LIBS += -llua5.3
      CONFIG += link_pkgconfig
!     PKGCONFIG += lua53
  
      # to make it run within Qt Creator
      !contains(DEFINES, DATA_DIR.*) {
