
Fixes the clamd socket file name

--- clamd-1.0.2.orig/src/clamd/__init__.py	2015-11-12 20:33:36.923228615 -0700
+++ clamd-1.0.2/src/clamd/__init__.py	2015-11-12 20:34:37.925224478 -0700
@@ -274,7 +274,7 @@
     """
     Class for using clamd with an unix socket
     """
-    def __init__(self, path="/var/run/clamav/clamd.ctl", timeout=None):
+    def __init__(self, path="/var/run/clamav/clamd.socket", timeout=None):
         """
         class initialisation
 
