Index: tesseract-2.03/java/makefile
===================================================================
--- tesseract-2.03.orig/java/makefile	2008-04-24 21:52:12.000000000 +0200
+++ tesseract-2.03/java/makefile	2008-04-24 21:53:03.000000000 +0200
@@ -39,8 +39,11 @@
 clean :
 	rm -f ScrollView.jar *.class
 
+distclean : clean
+	rm -f Makefile
+
 # all-am does nothing, to make the java part optional.
-all all-am :
+all all-am install :
 
 # dist runs the autoconf makefile to archive the files correctly.
 dist distdir :
