diff -Nur a/viewer/scrollview.cpp b/viewer/scrollview.cpp
--- a/viewer/scrollview.cpp	2008-04-21 19:06:24.000000000 -0500
+++ b/viewer/scrollview.cpp	2009-06-22 21:33:53.747817922 -0500
@@ -36,6 +36,9 @@
 #include <vector>
 #include <string>
 
+#include <string.h>
+#include <limits.h>
+
 #include "svutil.h"
 
 #ifdef HAVE_LIBLEPT
diff -Nur a/viewer/svmnode.cpp b/viewer/svmnode.cpp
--- a/viewer/svmnode.cpp	2008-04-18 23:44:06.000000000 -0500
+++ b/viewer/svmnode.cpp	2009-06-22 21:34:12.375586651 -0500
@@ -28,6 +28,8 @@
 
 #include <iostream>
 
+#include <string.h>
+
 #include "scrollview.h"
 
 // Create the empty root menu node. with just a caption. All other nodes should
diff -Nur a/viewer/svutil.cpp b/viewer/svutil.cpp
--- a/viewer/svutil.cpp	2008-04-21 19:07:25.000000000 -0500
+++ b/viewer/svutil.cpp	2009-06-22 21:34:31.419367272 -0500
@@ -38,6 +38,8 @@
 
 #include <iostream>
 #include <string>
+#include <string.h>
+#include <stdlib.h>
 
 const int kBufferSize = 65536;
 const int kMaxMsgSize = 4096;
