--- bin/gazpacho.orig	2009-08-17 00:15:26.000000000 +0300
+++ bin/gazpacho	2009-08-17 00:15:44.000000000 +0300
@@ -40,30 +40,6 @@
                                    appname,
                                    string.split(sys.version)[0]))
 
-# Figure out the directy which is the prefix
-# path-of-current-file/..
-currentdir = os.path.dirname(os.path.abspath(sys.argv[0]))
-basedir = os.path.abspath(os.path.join(currentdir, '..'))
-
-# Add the base directory where the application is installed in to sys.path
-if os.path.exists(os.path.join(basedir, 'lib')):
-    if sys.platform == 'win32':
-        pythondir = os.path.join(basedir, 'Lib', 'site-packages')
-    else:
-        pythondir = os.path.join(basedir, 'lib',
-                                 'python%d.%d' % sys.version_info[:2],
-                                 'site-packages')
-    if not os.path.exists(pythondir):
-        raise SystemExit("ERROR: Could not find required directory: %s" %
-                         pythondir)
-elif not os.path.exists(os.path.join(basedir, dirname)):
-    raise SystemExit("ERROR: Could not find required directory: %s" %
-                     os.path.join(basedir, dirname))
-else:
-    pythondir = basedir
-
-sys.path.insert(0, pythondir)
-
 from gazpacho import main
 
 try:
