diff -Naur obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/cmake/legacy.cmake obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/cmake/legacy.cmake
--- obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/cmake/legacy.cmake	2023-06-29 02:45:21.000000000 +0200
+++ obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/cmake/legacy.cmake	2025-10-15 16:28:43.293050000 +0200
@@ -448,7 +448,6 @@
 
 elseif(OS_POSIX)
   target_sources(obs PRIVATE platform-x11.cpp)
-  target_link_libraries(obs PRIVATE Qt::GuiPrivate)
 
   target_compile_definitions(obs PRIVATE OBS_INSTALL_PREFIX="${OBS_INSTALL_PREFIX}"
                                          "$<$<BOOL:${LINUX_PORTABLE}>:LINUX_PORTABLE>")
diff -Naur obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/cmake/os-freebsd.cmake obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/cmake/os-freebsd.cmake
--- obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/cmake/os-freebsd.cmake	2023-06-29 02:39:47.000000000 +0200
+++ obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/cmake/os-freebsd.cmake	2025-10-15 16:28:11.585050000 +0200
@@ -1,6 +1,6 @@
 target_sources(obs-studio PRIVATE platform-x11.cpp)
 target_compile_definitions(obs-studio PRIVATE OBS_INSTALL_PREFIX="${OBS_INSTALL_PREFIX}")
-target_link_libraries(obs-studio PRIVATE Qt::GuiPrivate procstat)
+target_link_libraries(obs-studio PRIVATE procstat)
 
 if(TARGET OBS::python)
   find_package(Python REQUIRED COMPONENTS Interpreter Development)
diff -Naur obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/cmake/os-linux.cmake obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/cmake/os-linux.cmake
--- obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/cmake/os-linux.cmake	2023-06-29 02:39:47.000000000 +0200
+++ obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/cmake/os-linux.cmake	2025-10-15 16:27:50.879050000 +0200
@@ -1,6 +1,5 @@
 target_sources(obs-studio PRIVATE platform-x11.cpp)
 target_compile_definitions(obs-studio PRIVATE OBS_INSTALL_PREFIX="${OBS_INSTALL_PREFIX}")
-target_link_libraries(obs-studio PRIVATE Qt::GuiPrivate)
 
 if(TARGET OBS::python)
   find_package(Python REQUIRED COMPONENTS Interpreter Development)
diff -Naur obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/frontend-plugins/aja-output-ui/CMakeLists.txt obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/frontend-plugins/aja-output-ui/CMakeLists.txt
--- obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/frontend-plugins/aja-output-ui/CMakeLists.txt	2023-06-29 02:45:21.000000000 +0200
+++ obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/frontend-plugins/aja-output-ui/CMakeLists.txt	2025-10-15 16:26:11.193050000 +0200
@@ -76,7 +76,7 @@
   target_compile_options(aja-output-ui PRIVATE -Wno-error=deprecated-declarations)
 elseif(OS_LINUX OR OS_FREEBSD)
   find_package(X11 REQUIRED)
-  target_link_libraries(aja-output-ui PRIVATE X11::X11 Qt::GuiPrivate)
+  target_link_libraries(aja-output-ui PRIVATE X11::X11)
 endif()
 
 set_target_properties_obs(
diff -Naur obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/frontend-plugins/aja-output-ui/cmake/legacy.cmake obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/frontend-plugins/aja-output-ui/cmake/legacy.cmake
--- obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/frontend-plugins/aja-output-ui/cmake/legacy.cmake	2023-06-29 02:39:47.000000000 +0200
+++ obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/frontend-plugins/aja-output-ui/cmake/legacy.cmake	2025-10-15 16:26:31.178050000 +0200
@@ -80,7 +80,7 @@
   target_link_options(aja-output-ui PRIVATE "LINKER:/IGNORE:4099")
 else()
   find_package(X11 REQUIRED)
-  target_link_libraries(aja-output-ui PRIVATE X11::X11 Qt::GuiPrivate)
+  target_link_libraries(aja-output-ui PRIVATE X11::X11)
 endif()
 
 if(NOT MSVC)
diff -Naur obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/frontend-plugins/decklink-output-ui/CMakeLists.txt obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/frontend-plugins/decklink-output-ui/CMakeLists.txt
--- obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/frontend-plugins/decklink-output-ui/CMakeLists.txt	2023-06-29 02:45:21.000000000 +0200
+++ obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/frontend-plugins/decklink-output-ui/CMakeLists.txt	2025-10-15 16:27:03.438050000 +0200
@@ -53,7 +53,7 @@
   target_link_libraries(decklink-output-ui PRIVATE ${COCOA})
 elseif(OS_LINUX OR OS_FREEBSD)
   find_package(X11 REQUIRED)
-  target_link_libraries(decklink-output-ui PRIVATE X11::X11 Qt::GuiPrivate)
+  target_link_libraries(decklink-output-ui PRIVATE X11::X11)
 endif()
 
 set_target_properties_obs(
diff -Naur obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/frontend-plugins/decklink-output-ui/cmake/legacy.cmake obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/frontend-plugins/decklink-output-ui/cmake/legacy.cmake
--- obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/frontend-plugins/decklink-output-ui/cmake/legacy.cmake	2023-06-29 02:39:47.000000000 +0200
+++ obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/frontend-plugins/decklink-output-ui/cmake/legacy.cmake	2025-10-15 16:27:22.911050000 +0200
@@ -64,7 +64,7 @@
 
 elseif(OS_POSIX)
   find_package(X11 REQUIRED)
-  target_link_libraries(decklink-output-ui PRIVATE X11::X11 Qt::GuiPrivate)
+  target_link_libraries(decklink-output-ui PRIVATE X11::X11)
 endif()
 
 get_target_property(_SOURCES decklink-output-ui SOURCES)
diff -Naur obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/frontend-plugins/frontend-tools/CMakeLists.txt obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/frontend-plugins/frontend-tools/CMakeLists.txt
--- obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/frontend-plugins/frontend-tools/CMakeLists.txt	2023-06-29 02:45:21.000000000 +0200
+++ obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/frontend-plugins/frontend-tools/CMakeLists.txt	2025-10-15 16:24:29.662050000 +0200
@@ -36,8 +36,7 @@
 target_sources(frontend-tools PRIVATE forms/auto-scene-switcher.ui forms/captions.ui forms/output-timer.ui
                                       forms/scripts.ui)
 
-target_link_libraries(frontend-tools PRIVATE OBS::frontend-api OBS::libobs Qt::Widgets
-                                             $<$<PLATFORM_ID:Linux,FreeBSD>:Qt::GuiPrivate>)
+target_link_libraries(frontend-tools PRIVATE OBS::frontend-api OBS::libobs Qt::Widgets)
 
 add_subdirectory("${CMAKE_SOURCE_DIR}/deps/obs-scripting" "${CMAKE_BINARY_DIR}/deps/obs-scripting")
 
diff -Naur obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/frontend-plugins/frontend-tools/cmake/legacy.cmake obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/frontend-plugins/frontend-tools/cmake/legacy.cmake
--- obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c.orig/UI/frontend-plugins/frontend-tools/cmake/legacy.cmake	2023-06-29 02:39:47.000000000 +0200
+++ obs-studio-c58e511813c33e93da7637d50aa431ae0cddda0c/UI/frontend-plugins/frontend-tools/cmake/legacy.cmake	2025-10-15 16:25:11.017050000 +0200
@@ -49,10 +49,6 @@
 
 target_link_libraries(frontend-tools PRIVATE OBS::frontend-api OBS::libobs Qt::Widgets)
 
-if(OS_POSIX AND NOT OS_MACOS)
-  target_link_libraries(frontend-tools PRIVATE Qt::GuiPrivate)
-endif()
-
 if(ENABLE_SCRIPTING AND TARGET OBS::scripting)
   target_compile_definitions(frontend-tools PRIVATE ENABLE_SCRIPTING)
 
