diff -Naur newsboat-2.40.orig/include/utils.h newsboat-2.40/include/utils.h
--- newsboat-2.40.orig/include/utils.h	2025-06-21 21:46:53.000000000 +0200
+++ newsboat-2.40/include/utils.h	2025-10-12 09:30:34.994373349 +0200
@@ -126,7 +126,7 @@
 
 void set_common_curl_options(CurlHandle& handle, ConfigContainer& cfg);
 
-curl_proxytype get_proxy_type(const std::string& type);
+long get_proxy_type(const std::string& type);
 
 std::string get_content(xmlNode* node);
 std::string get_basename(const std::string& url);
diff -Naur newsboat-2.40.orig/rss/parser.cpp newsboat-2.40/rss/parser.cpp
--- newsboat-2.40.orig/rss/parser.cpp	2025-06-21 21:46:53.000000000 +0200
+++ newsboat-2.40/rss/parser.cpp	2025-10-12 09:30:34.995373349 +0200
@@ -29,7 +29,7 @@
 	const std::string& user_agent,
 	const std::string& proxy,
 	const std::string& proxy_auth,
-	curl_proxytype proxy_type,
+	long proxy_type,
 	const bool ssl_verify)
 	: to(timeout)
 	, ua(user_agent)
diff -Naur newsboat-2.40.orig/rss/parser.h newsboat-2.40/rss/parser.h
--- newsboat-2.40.orig/rss/parser.h	2025-06-21 21:46:53.000000000 +0200
+++ newsboat-2.40/rss/parser.h	2025-10-12 09:30:34.995373349 +0200
@@ -23,7 +23,7 @@
 		const std::string& user_agent = "",
 		const std::string& proxy = "",
 		const std::string& proxy_auth = "",
-		curl_proxytype proxy_type = CURLPROXY_HTTP,
+		long proxy_type = CURLPROXY_HTTP,
 		const bool ssl_verify = true);
 	~Parser();
 	Feed parse_url(const std::string& url,
@@ -53,7 +53,7 @@
 	const std::string ua;
 	const std::string prx;
 	const std::string prxauth;
-	curl_proxytype prxtype;
+	long prxtype;
 	const bool verify_ssl;
 	xmlDocPtr doc;
 	time_t lm;
diff -Naur newsboat-2.40.orig/src/utils.cpp newsboat-2.40/src/utils.cpp
--- newsboat-2.40.orig/src/utils.cpp	2025-06-21 21:46:53.000000000 +0200
+++ newsboat-2.40/src/utils.cpp	2025-10-12 09:30:34.995373349 +0200
@@ -644,7 +644,7 @@
 	return std::string(utils::bridged::get_basename(url));
 }
 
-curl_proxytype utils::get_proxy_type(const std::string& type)
+long utils::get_proxy_type(const std::string& type)
 {
 	if (type == "http") {
 		return CURLPROXY_HTTP;
