diff -Naur pokerth-20250404_1a8a157.orig/connectivity.pro pokerth-20250404_1a8a157/connectivity.pro
--- pokerth-20250404_1a8a157.orig/connectivity.pro	2025-04-04 06:18:26.000000000 +0200
+++ pokerth-20250404_1a8a157/connectivity.pro	2025-09-02 21:41:31.699345000 +0200
@@ -43,7 +43,6 @@
 	INCLUDEPATH += $${PREFIX}/include
 	LIB_DIRS = $${PREFIX}/lib $${PREFIX}/lib64 $$system($$QMAKE_QMAKE -query QT_INSTALL_LIBS)
 	BOOST_PROGRAM_OPTIONS = boost_program_options boost_program_options-mt
-	BOOST_SYS = boost_system boost_system-mt
 
 
 	#
@@ -61,18 +60,10 @@
 				message("Found $$lib")
 				BOOST_PROGRAM_OPTIONS = -l$$lib
 			}
-			for(lib, BOOST_SYS):exists($${dir}/lib$${lib}.so*) {
-				message("Found $$lib")
-				BOOST_SYS = -l$$lib
-			}
-			for(lib, BOOST_SYS):exists($${dir}/lib$${lib}.a) {
-				message("Found $$lib")
-				BOOST_SYS = -l$$lib
-			}
 		}
 	}
-	BOOST_LIBS = $$BOOST_PROGRAM_OPTIONS $$BOOST_SYS
-	!count(BOOST_LIBS, 2){
+	BOOST_LIBS = $$BOOST_PROGRAM_OPTIONS
+	!count(BOOST_LIBS, 1){
 		error("Unable to find boost libraries in PREFIX=$${PREFIX}")
 	}
 
diff -Naur pokerth-20250404_1a8a157.orig/load.pro pokerth-20250404_1a8a157/load.pro
--- pokerth-20250404_1a8a157.orig/load.pro	2025-04-04 06:18:26.000000000 +0200
+++ pokerth-20250404_1a8a157/load.pro	2025-09-02 21:41:31.699345000 +0200
@@ -45,7 +45,6 @@
 	LIB_DIRS = $${PREFIX}/lib $${PREFIX}/lib64 $$system($$QMAKE_QMAKE -query QT_INSTALL_LIBS)
 	BOOST_THREAD = boost_thread boost_thread-mt
 	BOOST_PROGRAM_OPTIONS = boost_program_options boost_program_options-mt
-	BOOST_SYS = boost_system boost_system-mt
 
 
 	#
@@ -71,18 +70,10 @@
 				message("Found $$lib")
 				BOOST_PROGRAM_OPTIONS = -l$$lib
 			}
-			for(lib, BOOST_SYS):exists($${dir}/lib$${lib}.so*) {
-				message("Found $$lib")
-				BOOST_SYS = -l$$lib
-			}
-			for(lib, BOOST_SYS):exists($${dir}/lib$${lib}.a) {
-				message("Found $$lib")
-				BOOST_SYS = -l$$lib
-			}
 		}
 	}
-	BOOST_LIBS = $$BOOST_PROGRAM_OPTIONS $$BOOST_SYS $$BOOST_THREAD
-	!count(BOOST_LIBS, 3){
+	BOOST_LIBS = $$BOOST_PROGRAM_OPTIONS $$BOOST_THREAD
+	!count(BOOST_LIBS, 2){
 		error("Unable to find boost libraries in PREFIX=$${PREFIX}")
 	}
 
diff -Naur pokerth-20250404_1a8a157.orig/pokerth_game.pro pokerth-20250404_1a8a157/pokerth_game.pro
--- pokerth-20250404_1a8a157.orig/pokerth_game.pro	2025-04-04 06:18:26.000000000 +0200
+++ pokerth-20250404_1a8a157/pokerth_game.pro	2025-09-02 21:42:35.863345000 +0200
@@ -426,8 +426,6 @@
 		boost_iostreams-mt
 	BOOST_REGEX = boost_regex \
 		boost_regex-mt
-	BOOST_SYS = boost_system \
-		boost_system-mt
 	BOOST_RANDOM = boost_random \
 		boost_random-mt
 	BOOST_CHRONO = boost_chrono \
@@ -479,14 +477,6 @@
 					BOOST_RANDOM = -l$$lib
 				}
 			}
-			for(lib, BOOST_SYS):exists($${dir}/lib$${lib}.so*) {
-				message("Found $$lib")
-				BOOST_SYS = -l$$lib
-			}
-			for(lib, BOOST_SYS):exists($${dir}/lib$${lib}.a) {
-				message("Found $$lib")
-				BOOST_SYS = -l$$lib
-			}
 			!c++11 { 
 				for(lib, BOOST_CHRONO):exists($${dir}/lib$${lib}.so*) {
 					message("Found $$lib")
@@ -504,9 +494,8 @@
 			$$BOOST_FS \
 			$$BOOST_IOSTREAMS \
 			$$BOOST_REGEX \
-			$$BOOST_RANDOM \
-			$$BOOST_SYS
-		!count(BOOST_LIBS, 6):error("Unable to find boost libraries in PREFIX=$${PREFIX}")
+			$$BOOST_RANDOM
+		!count(BOOST_LIBS, 5):error("Unable to find boost libraries in PREFIX=$${PREFIX}")
 		}
 		!c++11 {
 			BOOST_LIBS = $$BOOST_THREAD \
@@ -514,9 +503,8 @@
 			$$BOOST_IOSTREAMS \
 			$$BOOST_REGEX \
 			$$BOOST_RANDOM \
-			$$$$BOOST_SYS \
 			$$BOOST_CHRONO
-			!count(BOOST_LIBS, 7):error("Unable to find boost libraries in PREFIX=$${PREFIX}")
+			!count(BOOST_LIBS, 6):error("Unable to find boost libraries in PREFIX=$${PREFIX}")
 		}
 		if($$system(sdl-config --version)):error("sdl-config not found in PATH - libSDL_mixer, libSDL are required!")
 		UNAME = $$system(uname -s)
@@ -536,9 +524,8 @@
 		BOOST_LIBS = $$BOOST_THREAD \
 			$$BOOST_FS \
 			$$BOOST_IOSTREAMS \
-			$$BOOST_REGEX \
-			$$BOOST_SYS
-		!count(BOOST_LIBS, 5):error("Unable to find boost libraries in PREFIX=$${PREFIX}/armv5")
+			$$BOOST_REGEX
+		!count(BOOST_LIBS, 4):error("Unable to find boost libraries in PREFIX=$${PREFIX}/armv5")
 		LIBS += -ltinyxml
 		LIBS += $$BOOST_LIBS
 		LIBS += -lgsasl -lidn
diff -Naur pokerth-20250404_1a8a157.orig/pokerth_server.pro pokerth-20250404_1a8a157/pokerth_server.pro
--- pokerth-20250404_1a8a157.orig/pokerth_server.pro	2025-04-04 06:18:26.000000000 +0200
+++ pokerth-20250404_1a8a157/pokerth_server.pro	2025-09-02 21:41:31.700345000 +0200
@@ -185,7 +185,6 @@
 	BOOST_THREAD = boost_thread boost_thread-mt
 	BOOST_PROGRAM_OPTIONS = boost_program_options boost_program_options-mt
 	BOOST_IOSTREAMS = boost_iostreams boost_iostreams-mt
-	BOOST_SYS = boost_system boost_system-mt
 	BOOST_REGEX = boost_regex boost_regex-mt
 	BOOST_RANDOM = boost_random boost_random-mt
 	BOOST_CHRONO = boost_chrono boost_chrono-mt
@@ -245,14 +244,6 @@
 				message("Found $$lib")
 				BOOST_RANDOM = -l$$lib
 			}
-			for(lib, BOOST_SYS):exists($${dir}/lib$${lib}.so*) {
-				message("Found $$lib")
-				BOOST_SYS = -l$$lib
-			}
-			for(lib, BOOST_SYS):exists($${dir}/lib$${lib}.a) {
-				message("Found $$lib")
-				BOOST_SYS = -l$$lib
-			}
 			!c++11 { 
 				for(lib, BOOST_CHRONO):exists($${dir}/lib$${lib}.so*) {
 					message("Found $$lib")
@@ -266,14 +257,14 @@
 		}
 	}
 	c++11 { 
-		BOOST_LIBS = $$BOOST_THREAD $$BOOST_FS $$BOOST_PROGRAM_OPTIONS $$BOOST_IOSTREAMS $$BOOST_REGEX $$BOOST_RANDOM $$BOOST_SYS
-		!count(BOOST_LIBS, 7){
+		BOOST_LIBS = $$BOOST_THREAD $$BOOST_FS $$BOOST_PROGRAM_OPTIONS $$BOOST_IOSTREAMS $$BOOST_REGEX $$BOOST_RANDOM
+		!count(BOOST_LIBS, 6){
 			error("Unable to find boost libraries in PREFIX=$${PREFIX}")
 		}
 	}
 	!c++11 { 
-		BOOST_LIBS = $$BOOST_THREAD $$BOOST_FS $$BOOST_PROGRAM_OPTIONS $$BOOST_IOSTREAMS $$BOOST_REGEX $$BOOST_RANDOM $$BOOST_SYS $$BOOST_CHRONO
-		!count(BOOST_LIBS, 8){
+		BOOST_LIBS = $$BOOST_THREAD $$BOOST_FS $$BOOST_PROGRAM_OPTIONS $$BOOST_IOSTREAMS $$BOOST_REGEX $$BOOST_RANDOM $$BOOST_CHRONO
+		!count(BOOST_LIBS, 7){
 			error("Unable to find boost libraries in PREFIX=$${PREFIX}")
 		}
 	}
