diff -Naur pokerth-20230930_ecf031e.orig/chatcleaner.pro pokerth-20230930_ecf031e/chatcleaner.pro
--- pokerth-20230930_ecf031e.orig/chatcleaner.pro	2023-09-30 09:15:23.000000000 +0200
+++ pokerth-20230930_ecf031e/chatcleaner.pro	2024-12-03 10:34:39.004430000 +0100
@@ -38,6 +38,8 @@
 LIBS += -lpokerth_lib \
 	-lpokerth_protocol \
 	-lprotobuf \
+	-labsl_log_internal_check_op \
+	-labsl_log_internal_message \
 	-ltinyxml
 
 win32 {
diff -Naur pokerth-20230930_ecf031e.orig/pokerth_game.pro pokerth-20230930_ecf031e/pokerth_game.pro
--- pokerth-20230930_ecf031e.orig/pokerth_game.pro	2023-09-30 09:15:23.000000000 +0200
+++ pokerth-20230930_ecf031e/pokerth_game.pro	2024-12-03 11:01:41.902430000 +0100
@@ -524,7 +524,10 @@
 		kFreeBSD = $$find(UNAME, "kFreeBSD")
 		LIBS += -lsqlite3 \
 				-ltinyxml \
-				-lprotobuf
+				-lprotobuf \
+				-labsl_log_internal_check_op \
+				-labsl_log_internal_nullguard \
+				-labsl_log_internal_message
 		LIBS += $$BOOST_LIBS
 		LIBS += -lSDL \
 				-lSDL_mixer \
diff -Naur pokerth-20230930_ecf031e.orig/pokerth_server.pro pokerth-20230930_ecf031e/pokerth_server.pro
--- pokerth-20230930_ecf031e.orig/pokerth_server.pro	2023-09-30 09:15:23.000000000 +0200
+++ pokerth-20230930_ecf031e/pokerth_server.pro	2024-12-03 10:35:49.439430000 +0100
@@ -285,7 +285,10 @@
 	LIBS += $$BOOST_LIBS
 	LIBS += -lsqlite3 \
 			-ltinyxml \
-			-lprotobuf
+			-lprotobuf \
+			-labsl_log_internal_check_op \
+			-labsl_log_internal_nullguard \
+			-labsl_log_internal_message
 	LIBS += -lgsasl
 	!isEmpty( BSD ): isEmpty( kFreeBSD ){
 		LIBS += -lcrypto -liconv
