diff -aurp ./src/sdlport/sound.cpp ../abuse-0.8//src/sdlport/sound.cpp
--- ./src/sdlport/sound.cpp	2011-05-09 03:58:30.000000000 -0400
+++ ../abuse-0.8//src/sdlport/sound.cpp	2011-05-21 20:46:44.487000027 -0400
@@ -65,7 +65,7 @@ int sound_init( int argc, char **argv )
     }
     free( sfxdir );
 
-    if (Mix_OpenAudio(44100, AUDIO_S16SYS, 2, 128) < 0)
+    if (Mix_OpenAudio(44100, AUDIO_S16SYS, 2, 1024) < 0)
     {
         printf( "Sound: Unable to open audio - %s\nSound: Disabled (error)\n", SDL_GetError() );
         return 0;
