--- src/emoteshortcut.cpp.old	2010-05-24 17:17:18.000000000 +0200
+++ src/emoteshortcut.cpp	2010-05-24 17:18:40.000000000 +0200
@@ -26,7 +26,7 @@
 
 #include "utils/stringutils.h"
 
-EmoteShortcut::EmoteShortcut *emoteShortcut;
+EmoteShortcut *emoteShortcut;
 
 EmoteShortcut::EmoteShortcut():
     mEmoteSelected(0)
--- src/itemshortcut.cpp.old	2010-05-24 17:19:37.000000000 +0200
+++ src/itemshortcut.cpp	2010-05-24 17:19:49.000000000 +0200
@@ -30,7 +30,7 @@
 
 #include "utils/stringutils.h"
 
-ItemShortcut::ItemShortcut *itemShortcut;
+ItemShortcut *itemShortcut;
 
 ItemShortcut::ItemShortcut():
     mItemSelected(-1)
