diff -Nur ekiga-2.0.11.orig/lib/gmconf/gmconf-glib.c ekiga-2.0.11/lib/gmconf/gmconf-glib.c
--- ekiga-2.0.11.orig/lib/gmconf/gmconf-glib.c	2007-09-18 04:00:21.000000000 -0500
+++ ekiga-2.0.11/lib/gmconf/gmconf-glib.c	2007-12-09 01:50:26.380594315 -0600
@@ -1473,7 +1473,7 @@
 }
 
 gboolean 
-gm_conf_is_key_writable (gchar *key)
+gm_conf_is_key_writable (const gchar *key)
 {
   g_return_val_if_fail (key != NULL, FALSE);
 
