--- Makefile.orig	2008-10-26 12:14:07.558133556 -0700
+++ Makefile	2008-10-26 12:15:55.282116793 -0700
@@ -23,7 +23,7 @@
 
 	install -d $(DESTDIR)/$(PREFIX)/share/icons/hicolor/scalable/apps
 	install $(APPNAME).svg $(DESTDIR)/$(PREFIX)/share/icons/hicolor/scalable/apps
-	ln -sf $(DESTDIR)/$(PREFIX)/share/icons/hicolor/scalable/apps/$(APPNAME).svg $(DESTDIR)/$(PREFIX)/share/$(APPNAME)/$(APPNAME).svg
+	ln -sf $(PREFIX)/share/icons/hicolor/scalable/apps/$(APPNAME).svg $(DESTDIR)/$(PREFIX)/share/$(APPNAME)/$(APPNAME).svg
 
 	install -d $(DESTDIR)/$(PREFIX)/share/doc/$(APPNAME)
 	install AUTHORS $(DESTDIR)/$(PREFIX)/share/doc/$(APPNAME)
@@ -36,7 +36,7 @@
 	install --mode=644 $(APPNAME).desktop $(DESTDIR)/$(PREFIX)/share/applications
 
 	cp -rf locale $(DESTDIR)/$(PREFIX)/share
-	ln -sf $(DESTDIR)/$(PREFIX)/share/locale $(DESTDIR)/$(PREFIX)/share/$(APPNAME)/locale
+	ln -sf $(PREFIX)/share/locale $(DESTDIR)/$(PREFIX)/share/$(APPNAME)/locale
 
 uninstall:
 	rm -f $(DESTDIR)/$(PREFIX)/share/applications/$(APPNAME).desktop
