--- Makefile.orig	2008-03-18 12:50:03.638894634 -0700
+++ Makefile	2008-03-18 12:50:55.641857937 -0700
@@ -23,7 +23,7 @@
 
 	install -d $(DESTDIR)/$(PREFIX)/share/pixmaps
 	install $(APPNAME).svg $(DESTDIR)/$(PREFIX)/share/pixmaps
-	ln -sf $(DESTDIR)/$(PREFIX)/share/pixmaps/$(APPNAME).svg $(DESTDIR)/$(PREFIX)/share/$(APPNAME)/$(APPNAME).svg
+	ln -sf $(PREFIX)/share/pixmaps/$(APPNAME).svg $(DESTDIR)/$(PREFIX)/share/$(APPNAME)/$(APPNAME).svg
 
 	install -d $(DESTDIR)/$(PREFIX)/share/doc
 	install AUTHORS $(DESTDIR)/$(PREFIX)/share/doc
@@ -36,7 +36,7 @@
 	install --mode=644 $(APPNAME).desktop $(DESTDIR)/$(PREFIX)/share/applications
 
 	cp -rf locale $(DESTDIR)/$(PREFIX)/share
-	ln -sf $(DESTDIR)/$(PREFIX)/share/locale $(DESTDIR)/$(PREFIX)/share/$(APPNAME)/locale
+	ln -sf $(PREFIX)/share/locale $(DESTDIR)/$(PREFIX)/share/$(APPNAME)/locale
 
 uninstall:
 	rm -f $(DESTDIR)/$(PREFIX)/share/applications/$(APPNAME).desktop
