diff -Naur Unhide-20220611/unhideGui.py Unhide-20220611.patched/unhideGui.py
--- Unhide-20220611/unhideGui.py	2022-06-11 05:30:24.000000000 -0400
+++ Unhide-20220611.patched/unhideGui.py	2023-08-03 17:03:50.945488351 -0400
@@ -1,4 +1,4 @@
-#!/bin/python3
+#!/usr/bin/env python3
 
 """
 Copyright © 2020-2022 Patrick Gouin
@@ -19,13 +19,15 @@
 """
 __author__ = "Patrick Gouin"
 __copyright__ = "Copyright 2020-2022, Patrick Gouin"
-__credits__ = [daichifukui]
+__credits__ = "daichifukui"
 __license__ = "GPL V3"
 __version__ = "1.1"
 __maintainer__ = "Patrick Gouin"
 __email__ = "patrickg.github@free.fr"
 __status__ = "Production"
 
+import sys
+sys.path.append("/usr/share/unhide")
 
 from tkinter import *
 from tkinter.ttk import *
@@ -187,7 +189,7 @@
     
 
 def GenCmd() :
-    Cmd = './unhide-linux '
+    Cmd = '/usr/sbin/unhide-linux '
     idx = 0
     for opt in OptionBut :
         if opt[VARB].get() == '1' :
@@ -209,7 +211,7 @@
     CmdText.config(width = len(Cmd))
         
 def GenTcpCmd() :
-    Cmd = './unhide-tcp '
+    Cmd = '/usr/sbin/unhide-tcp '
     idx = 0
     for opt in TcpOptionBut :
         if opt[VARB].get() == '1' :
