diff -Naur Unhide-20240510/unhideGui.py Unhide-20240510.patched/unhideGui.py
--- Unhide-20240510/unhideGui.py	2024-12-11 09:33:11.000000000 -0500
+++ Unhide-20240510.patched/unhideGui.py	2025-06-14 17:58:49.115052346 -0400
@@ -1,5 +1,4 @@
-#!/bin/python3
-# #!/usr/bin/python3
+#!/usr/bin/python3
 
 """
 Copyright © 2020-2024 Patrick Gouin
@@ -27,6 +26,8 @@
 __email__ = "patrickg.github@free.fr"
 __status__ = "Production"
 
+import sys
+sys.path.append("/usr/share/unhide")
 
 from tkinter import *
 from tkinter.ttk import *
@@ -191,7 +192,7 @@
     
 
 def GenCmd() :
-    Cmd = unhPath
+    Cmd = '/usr/sbin/unhide-linux '
     idx = 0
     for opt in OptionBut :
         if opt[VARB].get() == '1' :
@@ -213,7 +214,7 @@
     CmdText.config(width = len(Cmd))
         
 def GenTcpCmd() :
-    Cmd = unhtcpPath
+    Cmd = '/usr/sbin/unhide-tcp '
     idx = 0
     for opt in TcpOptionBut :
         if opt[VARB].get() == '1' :
@@ -460,23 +461,6 @@
 screen_height = root.winfo_screenheight()
 root.geometry('+%d+%d' % (screen_width/3, screen_height/3))
 
-# look for unhide path
-# we prefer the local version
-if os.path.exists("./unhide-linux") :
-    unhPath = "./unhide-linux "
-elif os.path.exists("./sbin/unhide-linux") :
-    unhPath = "./sbin/unhide-linux "
-else :
-    unhPath = "./usr/sbin/unhide-linux "
-
-if os.path.exists("./unhide-tcp") :
-    unhtcpPath = "./unhide-tcp "
-elif os.path.exists("./sbin/unhide-tcp") :
-    unhtcpPath = "./sbin/unhide-tcp "
-else :
-    unhtcpPath = "./usr/sbin/unhide-tcp "
-
-
 root.update()
 
 root.mainloop()
