--- udftools-1.0.0b3.orig/cdrwtool/cdrwtool.c	2002-12-28 15:48:51.000000000 +1100
+++ udftools-1.0.0b3/cdrwtool/cdrwtool.c	2013-03-09 16:28:32.685490490 +1100
@@ -27,6 +27,7 @@
 #include <unistd.h>
 #include <getopt.h>
 #include <signal.h>
+#include <limits.h>
 
 #include <sys/ioctl.h>
 #include <asm/param.h>
--- udftools-1.0.0b3.orig/cdrwtool/main.c	2004-02-23 14:33:11.000000000 +1100
+++ udftools-1.0.0b3/cdrwtool/main.c	2013-03-09 16:27:35.630406411 +1100
@@ -30,6 +30,7 @@
 #include <time.h>
 #include <sys/time.h>
 #include <errno.h>
+#include <string.h>
 
 #include "cdrwtool.h"
 #include "defaults.h"
--- udftools-1.0.0b3.orig/cdrwtool/options.c	2003-10-24 11:16:04.000000000 +1000
+++ udftools-1.0.0b3/cdrwtool/options.c	2013-03-09 16:27:49.846422176 +1100
@@ -23,6 +23,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <malloc.h>
+#include <string.h>
 
 #include "cdrwtool.h"
 #include "libudffs.h"
--- udftools-1.0.0b3.orig/include/udf_endian.h	2002-11-26 18:18:51.000000000 +1100
+++ udftools-1.0.0b3/include/udf_endian.h	2013-03-09 16:24:42.000920607 +1100
@@ -23,6 +23,7 @@
 #ifndef __UDF_ENDIAN_H
 #define __UDF_ENDIAN_H
 
+#include <string.h>
 #include "bswap.h"
 
 static inline lb_addr lelb_to_cpu(lb_addr in)
--- udftools-1.0.0b3.orig/libudffs/file.c	2002-11-26 18:18:51.000000000 +1100
+++ udftools-1.0.0b3/libudffs/file.c	2013-03-09 16:25:21.175013609 +1100
@@ -21,6 +21,7 @@
  */
 
 #include <malloc.h>
+#include <string.h>
 
 #include "libudffs.h"
 #include "defaults.h"
--- udftools-1.0.0b3.orig/libudffs/unicode.c	2002-11-26 18:18:51.000000000 +1100
+++ udftools-1.0.0b3/libudffs/unicode.c	2013-03-09 16:25:59.433223040 +1100
@@ -20,6 +20,8 @@
  *
  */
 
+#include <string.h>
+
 #include "libudffs.h"
 #include "defaults.h"
 #include "config.h"
--- udftools-1.0.0b3.orig/mkudffs/main.c	2004-02-23 14:33:11.000000000 +1100
+++ udftools-1.0.0b3/mkudffs/main.c	2013-03-09 16:26:27.967171881 +1100
@@ -31,6 +31,7 @@
 #include <sys/time.h>
 #include <errno.h>
 #include <limits.h>
+#include <string.h>
 
 #include "mkudffs.h"
 #include "defaults.h"
--- udftools-1.0.0b3.orig/mkudffs/mkudffs.c	2004-02-23 14:35:33.000000000 +1100
+++ udftools-1.0.0b3/mkudffs/mkudffs.c	2013-03-09 16:26:55.098244722 +1100
@@ -30,6 +30,7 @@
 #include <time.h>
 #include <sys/time.h>
 #include <errno.h>
+#include <string.h>
 
 #include "mkudffs.h"
 #include "defaults.h"
--- udftools-1.0.0b3.orig/mkudffs/options.c	2002-11-26 18:18:51.000000000 +1100
+++ udftools-1.0.0b3/mkudffs/options.c	2013-03-09 16:27:17.188691854 +1100
@@ -24,6 +24,7 @@
 #include <stdlib.h>
 #include <getopt.h>
 #include <malloc.h>
+#include <string.h>
 
 #include "mkudffs.h"
 #include "defaults.h"
--- udftools-1.0.0b3.orig/pktsetup/pktsetup.c	2002-11-26 18:18:51.000000000 +1100
+++ udftools-1.0.0b3/pktsetup/pktsetup.c	2013-03-09 16:30:53.851742260 +1100
@@ -23,6 +23,7 @@
 #include <getopt.h>
 #include <bits/types.h>
 #include <sys/types.h>
+#include <limits.h>
 
 #include <linux/cdrom.h>
 
--- udftools-1.0.0b3.orig/wrudf/ide-pc.c	2002-11-26 18:18:51.000000000 +1100
+++ udftools-1.0.0b3/wrudf/ide-pc.c	2013-03-09 16:29:57.634001857 +1100
@@ -31,6 +31,7 @@
 #include <sys/types.h>		/* for u_char etc. */
 #include <linux/cdrom.h>
 #include <unistd.h>		/* sleep() */
+#include <stdlib.h>
 
 #include "bswap.h"
 #include "ide-pc.h"
--- udftools-1.0.0b3.orig/wrudf/wrudf-cdr.c	2002-11-26 18:18:51.000000000 +1100
+++ udftools-1.0.0b3/wrudf/wrudf-cdr.c	2013-03-09 16:31:43.156091936 +1100
@@ -6,6 +6,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <string.h>
+
 #include "wrudf.h"
 #include "ide-pc.h"
 #include "bswap.h"
--- udftools-1.0.0b3.orig/wrudf/wrudf-cdrw.c	2002-11-26 18:18:51.000000000 +1100
+++ udftools-1.0.0b3/wrudf/wrudf-cdrw.c	2013-03-09 16:29:05.652037738 +1100
@@ -31,6 +31,7 @@
 #include <sys/stat.h>
 #include <sys/ioctl.h>
 #include <linux/cdrom.h>		/* for CDROM_DRIVE_STATUS  */
+#include <string.h>
 
 #include "wrudf.h"
 #include "ide-pc.h"
--- udftools-1.0.0b3.orig/wrudf/wrudf.c	2002-11-26 18:18:51.000000000 +1100
+++ udftools-1.0.0b3/wrudf/wrudf.c	2013-03-09 16:31:10.001538000 +1100
@@ -245,7 +245,7 @@
 	    } else if( strncmp( spm->partIdent.ident, UDF_ID_VIRTUAL, strlen(UDF_ID_VIRTUAL)) == 0 )
 		virtualPartitionNum = i;
 	}
-	(char*)spm += spm->partitionMapLength;
+	spm += spm->partitionMapLength;
     }
 
     if( medium == CDR ) {
