diff -Naur simgear-2020.1.2/CMakeLists.txt simgear-2020.1.2.new/CMakeLists.txt
--- simgear-2020.1.2/CMakeLists.txt	2020-05-24 11:13:35.000000000 -0400
+++ simgear-2020.1.2.new/CMakeLists.txt	2020-06-01 09:36:21.608814301 -0400
@@ -43,13 +43,13 @@
 set(CMAKE_CXX_STANDARD_REQUIRED YES)
 
 # read 'version' file into a variable (stripping any newlines or spaces)
-file(READ version versionFile)
+file(READ simgear-version versionFile)
 string(STRIP ${versionFile} SIMGEAR_VERSION)
 
 project(SimGear VERSION ${SIMGEAR_VERSION} LANGUAGES C CXX)
 
 # add a dependency on the version file
-set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS version)
+set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS simgear-version)
 
 set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS TRUE)
 
diff -Naur simgear-2020.1.2/simgear-version simgear-2020.1.2.new/simgear-version
--- simgear-2020.1.2/simgear-version	1969-12-31 19:00:00.000000000 -0500
+++ simgear-2020.1.2.new/simgear-version	2020-05-24 11:13:36.000000000 -0400
@@ -0,0 +1 @@
+2020.1.2
diff -Naur simgear-2020.1.2/version simgear-2020.1.2.new/version
--- simgear-2020.1.2/version	2020-05-24 11:13:36.000000000 -0400
+++ simgear-2020.1.2.new/version	1969-12-31 19:00:00.000000000 -0500
@@ -1 +0,0 @@
-2020.1.2
