diff --git a/device/lib/Makefile.in b/device/lib/Makefile.in
--- a/device/lib/Makefile.in
+++ b/device/lib/Makefile.in
@@ -522,11 +522,11 @@
 
 	for src in $(MODELS) ds390 ds400 z80 z180 r2k r3ka tlcs90 gbz80 hc08 s08 stm8 pic14 pic16; do \
 	  find $(sdcc_libdir)/src/$$src -depth \
-	    \( -name '*.rel' -o -name '*.dump*' -o -name '*.sym' -o -name '*.o' -o -name '*.lib' -o \
-	    -name '*.lst' -o -name '*.asm' -o -name '.svn' -o -name 'autom4te.cache' -o \
+	    \( -name '*.rel' -o -name '*.dump*' -o -name '*.sym' -o -name '*.o' -o -name '*.a' -o -name '*.lib' -o \
+	    -name '*.lst' -o -name '*.asm' -o -name '*.cod' -o -name '*.hex' -o -name '.svn' -o -name 'autom4te.cache' -o \
 	    -name 'build' -o -name 'bin' -o -name 'Makefile*' -o -name 'config*' -o \
-	    -name 'depcomp' -o -name 'missing' -o -name 'install-sh' -o \
-	    -name 'bootstrap.sh' \) \
+	    -name 'depcomp' -o -name 'missing' -o -name 'install-sh' -o -name 'stamp-h1' -o \
+	    -name 'bootstrap.sh' -o -name '.deps' -o -name '.checkdevices' -o -name '.checksupport' -o -name 'pics.supported' \) \
 	    -exec rm -rf {} + ; \
 	done
 	find $(sdcc_libdir)/src/mcs51 -depth \
diff --git a/device/non-free/lib/Makefile.in b/device/non-free/lib/Makefile.in
--- a/device/non-free/lib/Makefile.in
+++ b/device/non-free/lib/Makefile.in
@@ -357,11 +357,11 @@
 
 	for src in $(ENABLED_PORTS); do \
 	  find $(sdcc_libdir)/src/$$src -depth \
-	    \( -name '*.rel' -o -name '*.dump*' -o -name '*.sym' -o -name '*.o' -o -name '*.lib' -o \
-	    -name '*.lst' -o -name '*.asm' -o -name '.svn' -o -name 'autom4te.cache' -o \
+	    \( -name '*.rel' -o -name '*.dump*' -o -name '*.sym' -o -name '*.o' -o -name '*.a' -o -name '*.lib' -o \
+	    -name '*.lst' -o -name '*.asm' -o -name '*.cod' -o -name '*.hex' -o -name '.svn' -o -name 'autom4te.cache' -o \
 	    -name 'build' -o -name 'bin' -o -name 'Makefile*' -o -name 'config*' -o \
-	    -name 'depcomp' -o -name 'missing' -o -name 'install-sh' -o \
-	    -name 'bootstrap.sh' \) \
+	    -name 'depcomp' -o -name 'missing' -o -name 'install-sh' -o -name 'stamp-h1' -o \
+	    -name 'bootstrap.sh' -o -name '.deps' -o -name '.checkdevices' -o -name '.checksupport' -o -name 'pics.supported' \) \
 	    -exec rm -rf {} + ; \
 	done
 endif
