diff -Naur libvmime-0.9.2.orig/Makefile.in libvmime-0.9.2/Makefile.in
--- libvmime-0.9.2.orig/Makefile.in	2012-01-10 10:33:02.000000000 +0100
+++ libvmime-0.9.2/Makefile.in	2012-09-26 05:00:51.899012736 +0200
@@ -42,7 +42,7 @@
 DIST_COMMON = INSTALL NEWS README AUTHORS ChangeLog \
 	$(srcdir)/Makefile.in $(srcdir)/Makefile.am \
 	$(top_srcdir)/configure $(am__configure_deps) \
-	$(srcdir)/config.h.in $(srcdir)/vmime.pc.in
+	$(srcdir)/config.h.in $(srcdir)/vmime_zarafa.pc.in
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/m4/acx_pthread.m4 \
 	$(top_srcdir)/m4/iconv.m4 $(top_srcdir)/m4/lib-ld.m4 \
@@ -56,7 +56,7 @@
  configure.lineno config.status.lineno
 mkinstalldirs = $(install_sh) -d
 CONFIG_HEADER = config.h
-CONFIG_CLEAN_FILES = vmime.pc
+CONFIG_CLEAN_FILES = vmime_zarafa.pc
 CONFIG_CLEAN_VPATH_FILES =
 SOURCES =
 RECURSIVE_TARGETS = all-recursive check-recursive dvi-recursive \
@@ -314,7 +314,7 @@
 
 distclean-hdr:
 	-rm -f config.h stamp-h1
-vmime.pc: $(top_builddir)/config.status $(srcdir)/vmime.pc.in
+vmime_zarafa.pc: $(top_builddir)/config.status $(srcdir)/vmime_zarafa.pc.in
 	cd $(top_builddir) && $(SHELL) ./config.status $@
 
 mostlyclean-libtool:
diff -Naur libvmime-0.9.2.orig/configure libvmime-0.9.2/configure
--- libvmime-0.9.2.orig/configure	2012-01-10 10:33:01.000000000 +0100
+++ libvmime-0.9.2/configure	2012-09-26 04:55:30.835971484 +0200
@@ -699,10 +699,10 @@
 MAKEFLAGS=
 
 # Identity of this package.
-PACKAGE_NAME='VMime Library'
-PACKAGE_TARNAME='vmime'
+PACKAGE_NAME='VMime Library for Zarafa'
+PACKAGE_TARNAME='vmime_zarafa'
 PACKAGE_VERSION='0.9.2'
-PACKAGE_STRING='VMime Library 0.9.2'
+PACKAGE_STRING='VMime Library 0.9.2 for Zarafa'
 PACKAGE_BUGREPORT='vincent@vincent-richard.net'
 PACKAGE_URL=''
 
@@ -2939,13 +2939,13 @@
 
 
 # Library name
-GENERIC_LIBRARY_NAME="vmime"
+GENERIC_LIBRARY_NAME="vmime_zarafa"
 
 
-GENERIC_VERSIONED_LIBRARY_NAME="vmime"
+GENERIC_VERSIONED_LIBRARY_NAME="vmime_zarafa"
 
 
-LIBRARY_NAME="libvmime"
+LIBRARY_NAME="libvmime_zarafa"
 
 
 # Library version
@@ -3508,7 +3508,7 @@
 
 
 # Define the identity of the package.
- PACKAGE=vmime
+ PACKAGE=vmime_zarafa
  VERSION=0.9.2
 
 
@@ -21556,7 +21556,7 @@
 
 LIBS=`echo $LIBS | sed -e 's|^ ||g' | sed -e 's|  | |g'`
 
-ac_config_files="$ac_config_files vmime.pc Makefile src/Makefile vmime/Makefile"
+ac_config_files="$ac_config_files vmime_zarafa.pc Makefile src/Makefile vmime/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -22754,7 +22754,7 @@
     "config.h") CONFIG_HEADERS="$CONFIG_HEADERS config.h" ;;
     "depfiles") CONFIG_COMMANDS="$CONFIG_COMMANDS depfiles" ;;
     "libtool") CONFIG_COMMANDS="$CONFIG_COMMANDS libtool" ;;
-    "vmime.pc") CONFIG_FILES="$CONFIG_FILES vmime.pc" ;;
+    "vmime_zarafa.pc") CONFIG_FILES="$CONFIG_FILES vmime_zarafa.pc" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
     "src/Makefile") CONFIG_FILES="$CONFIG_FILES src/Makefile" ;;
     "vmime/Makefile") CONFIG_FILES="$CONFIG_FILES vmime/Makefile" ;;
@@ -24301,7 +24301,7 @@
 
 
 // Name of package
-#define VMIME_PACKAGE \"libvmime\"
+#define VMIME_PACKAGE \"libvmime_zarafa\"
 
 // Version number of package
 #define VMIME_VERSION \"0.9.2\"
diff -Naur libvmime-0.9.2.orig/src/Makefile.in libvmime-0.9.2/src/Makefile.in
--- libvmime-0.9.2.orig/src/Makefile.in	2012-01-10 10:33:02.000000000 +0100
+++ libvmime-0.9.2/src/Makefile.in	2012-09-26 04:41:07.134000010 +0200
@@ -149,8 +149,8 @@
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(libdir)"
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libvmime_la_LIBADD =
-am__libvmime_la_SOURCES_DIST = address.cpp addressList.cpp \
+libvmime_zarafa_la_LIBADD =
+am__libvmime_zarafa_la_SOURCES_DIST = address.cpp addressList.cpp \
 	attachmentHelper.cpp base.cpp body.cpp bodyPart.cpp \
 	bodyPartAttachment.cpp charset.cpp charsetConverter.cpp \
 	component.cpp constants.cpp contentDisposition.cpp \
@@ -286,7 +286,7 @@
 @VMIME_BUILTIN_PLATFORM_POSIX_TRUE@	platforms_posix_posixFile.lo \
 @VMIME_BUILTIN_PLATFORM_POSIX_TRUE@	platforms_posix_posixHandler.lo \
 @VMIME_BUILTIN_PLATFORM_POSIX_TRUE@	platforms_posix_posixSocket.lo
-am_libvmime_la_OBJECTS = address.lo addressList.lo attachmentHelper.lo \
+am_libvmime_zarafa_la_OBJECTS = address.lo addressList.lo attachmentHelper.lo \
 	base.lo body.lo bodyPart.lo bodyPartAttachment.lo charset.lo \
 	charsetConverter.lo component.lo constants.lo \
 	contentDisposition.lo contentDispositionField.lo \
@@ -322,10 +322,10 @@
 	$(am__objects_2) $(am__objects_3) $(am__objects_4) \
 	$(am__objects_5) $(am__objects_6) $(am__objects_7) \
 	$(am__objects_8) $(am__objects_9) $(am__objects_10)
-libvmime_la_OBJECTS = $(am_libvmime_la_OBJECTS)
-libvmime_la_LINK = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) \
+libvmime_zarafa_la_OBJECTS = $(am_libvmime_zarafa_la_OBJECTS)
+libvmime_zarafa_la_LINK = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CXXLD) $(AM_CXXFLAGS) \
-	$(CXXFLAGS) $(libvmime_la_LDFLAGS) $(LDFLAGS) -o $@
+	$(CXXFLAGS) $(libvmime_zarafa_la_LDFLAGS) $(LDFLAGS) -o $@
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
 depcomp =
 am__depfiles_maybe =
@@ -338,7 +338,7 @@
 CXXLINK = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=link $(CXXLD) $(AM_CXXFLAGS) $(CXXFLAGS) $(AM_LDFLAGS) \
 	$(LDFLAGS) -o $@
-SOURCES = $(libvmime_la_SOURCES)
+SOURCES = $(libvmime_zarafa_la_SOURCES)
 DIST_SOURCES = $(am__libvmime_la_SOURCES_DIST)
 HEADERS = $(noinst_HEADERS)
 ETAGS = etags
@@ -499,9 +499,9 @@
 AUTOMAKE_OPTIONS = no-dependencies foreign
 INTERNALS = 
 INCLUDES = -I$(prefix)/include -I$(top_srcdir) @PKGCONFIG_CFLAGS@ @EXTRA_CFLAGS@
-lib_LTLIBRARIES = libvmime.la
-libvmime_la_LDFLAGS = -export-dynamic -version-info @LIBRARY_VERSION@ @PKGCONFIG_LIBS@ @EXTRA_LIBS@
-libvmime_la_SOURCES = address.cpp addressList.cpp attachmentHelper.cpp \
+lib_LTLIBRARIES = libvmime_zarafa.la
+libvmime_zarafa_la_LDFLAGS = -export-dynamic -version-info @LIBRARY_VERSION@ @PKGCONFIG_LIBS@ @EXTRA_LIBS@
+libvmime_zarafa_la_SOURCES = address.cpp addressList.cpp attachmentHelper.cpp \
 	base.cpp body.cpp bodyPart.cpp bodyPartAttachment.cpp \
 	charset.cpp charsetConverter.cpp component.cpp constants.cpp \
 	contentDisposition.cpp contentDispositionField.cpp \
@@ -607,8 +607,8 @@
 	  echo "rm -f \"$${dir}/so_locations\""; \
 	  rm -f "$${dir}/so_locations"; \
 	done
-libvmime.la: $(libvmime_la_OBJECTS) $(libvmime_la_DEPENDENCIES) 
-	$(libvmime_la_LINK) -rpath $(libdir) $(libvmime_la_OBJECTS) $(libvmime_la_LIBADD) $(LIBS)
+libvmime_zarafa.la: $(libvmime_zarafa_la_OBJECTS) $(libvmime_zarafa_la_DEPENDENCIES) 
+	$(libvmime_zarafa_la_LINK) -rpath $(libdir) $(libvmime_zarafa_la_OBJECTS) $(libvmime_zarafa_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
diff -Naur libvmime-0.9.2.orig/vmime.pc.in libvmime-0.9.2/vmime.pc.in
--- libvmime-0.9.2.orig/vmime.pc.in	2012-01-10 10:32:57.000000000 +0100
+++ libvmime-0.9.2/vmime.pc.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,14 +0,0 @@
-# File automatically generated by SConstruct ('scons autotools')
-# DO NOT EDIT!
-
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
-
-Name: @GENERIC_LIBRARY_NAME@
-Description: VMime C++ Mail Library (http://www.vmime.org)
-Version: @VERSION@
-Requires: @GSASL_REQUIRED@
-Libs: -L${libdir} -l@GENERIC_VERSIONED_LIBRARY_NAME@ @GSASL_LIBS@ @LIBGNUTLS_LIBS@ @LIBICONV@ @PTHREAD_LIBS@ @LIBICONV@ @PTHREAD_LIBS@ @VMIME_ADDITIONAL_PC_LIBS@
-Cflags: -I${includedir}/ @LIBGNUTLS_CFLAGS@
diff -Naur libvmime-0.9.2.orig/vmime_zarafa.pc.in libvmime-0.9.2/vmime_zarafa.pc.in
--- libvmime-0.9.2.orig/vmime_zarafa.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ libvmime-0.9.2/vmime_zarafa.pc.in	2012-01-10 10:32:57.000000000 +0100
@@ -0,0 +1,14 @@
+# File automatically generated by SConstruct ('scons autotools')
+# DO NOT EDIT!
+
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: @GENERIC_LIBRARY_NAME@
+Description: VMime C++ Mail Library (http://www.vmime.org)
+Version: @VERSION@
+Requires: @GSASL_REQUIRED@
+Libs: -L${libdir} -l@GENERIC_VERSIONED_LIBRARY_NAME@ @GSASL_LIBS@ @LIBGNUTLS_LIBS@ @LIBICONV@ @PTHREAD_LIBS@ @LIBICONV@ @PTHREAD_LIBS@ @VMIME_ADDITIONAL_PC_LIBS@
+Cflags: -I${includedir}/ @LIBGNUTLS_CFLAGS@
