--- libvmime-0.7.1_orig/configure	2005-06-21 19:11:26.000000000 -0300
+++ libvmime-0.7.1/configure	2009-12-26 15:30:12.000000000 -0200
@@ -1557,7 +1557,7 @@
 GENERIC_VERSIONED_LIBRARY_NAME="vmime"
 
 
-LIBRARY_NAME="libvmime"
+LIBRARY_NAME="libvmime-zarafa"
 
 
 # Library version
@@ -25237,7 +25237,7 @@
 
 
 // Name of package
-#define VMIME_PACKAGE \"libvmime\"
+#define VMIME_PACKAGE \"libvmime-zarafa\"
 
 // Version number of package
 #define VMIME_VERSION \"0.7.1\"
