--- pypanel	2005-06-27 02:24:43.000000000 +0300
+++ pypanel.new	2007-06-23 21:47:56.000000000 +0300
@@ -95,6 +95,7 @@
         self.root.change_attributes(event_mask=(X.PropertyChangeMask)) 
         self.window.map()
         self.display.flush()
+        self.updatePanel(self.root, self.window, self.panel)
         self.loop(self.display, self.root, self.window, self.panel)
                 
     #------------------------------------
@@ -944,6 +945,9 @@
     main   = 2.4
     config = globals().get("VERSION", None)
     
+    # Get the startup delay
+    delay  = globals().get("STARTUP_DELAY", None)
+    
     # Set locale to user's default
     locale.setlocale(locale.LC_ALL, "")
     
@@ -953,5 +957,9 @@
         sys.stderr.write("\nA current pypanelrc example can be found here -\n")
         sys.stderr.write("%s/pypanel/pypanelrc\n\n" % sysconfig.get_python_lib())
     del main, config
-
+    
+    # If delay is set, pause, and let windowmanager load
+    if delay:
+	    time.sleep(delay)
+	    
     PyPanel(display.Display())
--- pypanelrc	2005-06-27 02:27:37.000000000 +0300
+++ pypanelrc.new	2007-06-23 21:46:01.000000000 +0300
@@ -9,6 +9,12 @@
 VERSION         = 2.4           # Config file version
 
 #------------------------------------------------------------------------------
+# Startup delay: The time to wait before painting the pypanel window. This lets
+# the background load up so that pypanel will be displayed correctly. 
+#------------------------------------------------------------------------------
+STARTUP_DELAY   = 2
+
+#------------------------------------------------------------------------------
 # Colors: Format is hex triplet - 0xrrggbb
 #------------------------------------------------------------------------------
 BG_COLOR        = "0xd6d6d6"    # Panel background and tinting color
