diff -Naur hatari-2.2.1/python-ui/hatariui hatari-2.2.1.patched/python-ui/hatariui
--- hatari-2.2.1/python-ui/hatariui	2019-02-08 03:30:10.000000000 -0500
+++ hatari-2.2.1.patched/python-ui/hatariui	2020-04-27 00:47:57.786768381 -0400
@@ -3,6 +3,13 @@
 # Don't modify the 'path' or 'conf' variable names or initial values,
 # those will be replaced by Makefile when this script is installed.
 
+# 20200426 bkw: create default config file if missing
+[ -e ~/.config/hatari/hatari.cfg ] || (
+	mkdir -p ~/.config/hatari/
+	cd ~/.config/hatari/
+	hatari --saveconfig
+)
+
 path=${0%/*}
 name=${0##*/}
 
