diff -ru GMapCatcher-0.7.7.2/setup.py GMapCatcher-0.7.7.2.new/setup.py
--- GMapCatcher-0.7.7.2/setup.py	2012-04-28 10:35:10.000000000 -0300
+++ GMapCatcher-0.7.7.2.new/setup.py	2012-07-09 11:06:37.054422933 -0300
@@ -24,7 +24,7 @@
         scripts = ['mapcatcher', 'mapdownloader'],
         packages = ['gmapcatcher', 'gmapcatcher.mapServers',
                     'gmapcatcher.pyGPSD', 'gmapcatcher.pyGPSD.nmea',
-                    'gmapcatcher.pyGPSD.nmea.serial']
+                    'gmapcatcher.pyGPSD.nmea.serial', 'gmapcatcher.tilesRepo']
     )
 else:
     import py2exe
