--- pcem-965ea55331e7c1c5a3ca169e006150615cb65776.orig/src/plugin-api/meson.build	2022-03-19 19:19:47.223202352 +0900
+++ pcem-965ea55331e7c1c5a3ca169e006150615cb65776/src/plugin-api/meson.build	2022-03-19 19:20:10.085117202 +0900
@@ -26,7 +26,9 @@
         )

 if get_option('use-plugin-engine')
-        pcem_plugin_api = shared_library('pcem', sources: pcem_pluginapi_src, dependencies: pcem_dependencies, include_directories: pcem_include_dirs, install: true)
+        libdl = compiler.find_library('dl', required: false)
+        pcem_dependencies += libdl
+        pcem_plugin_api = shared_library('pcem-plugin-api', sources: pcem_pluginapi_src, dependencies: pcem_dependencies, include_directories: pcem_include_dirs, install: true)
         pcem_project_libraries += pcem_plugin_api
 else
         pcem_src += pcem_pluginapi_src
