diff -Naur gnome-session-42.0/gnome-session/gnome-session.in gnome-session-42.0-patched/gnome-session/gnome-session.in
--- gnome-session-42.0/gnome-session/gnome-session.in	2022-03-21 15:13:31.651419000 -0500
+++ gnome-session-42.0-patched/gnome-session/gnome-session.in	2022-08-09 01:15:35.293264706 -0500
@@ -13,6 +13,16 @@
   fi
 fi
 
+# Force blocaled to update LANG according to /etc/profile.d/lang.sh:
+if [ -f /etc/profile.d/lang.sh ]; then
+  (
+  source /etc/profile.d/lang.sh
+  busctl call org.freedesktop.locale1 /org/freedesktop/locale1 org.freedesktop.locale1 SetLocale "asb" 1 "LANG=$LANG" 0
+  )
+  # If we are GDM then source /etc/profile.d/lang.sh for real:
+  [ "x$XDG_SESSION_CLASS" = "xgreeter" ] && source /etc/profile.d/lang.sh
+fi
+
 SETTING=$(G_MESSAGES_DEBUG='' gsettings get org.gnome.system.locale region)
 REGION=${SETTING#\'}
 REGION=${REGION%\'}
