--- antlr-2.7.7/lib/cpp/antlr/CharScanner.hpp	2006-11-01 22:37:17.000000000 +0100
+++ antlr-2.7.7-patch/lib/cpp/antlr/CharScanner.hpp	2017-02-06 15:04:59.554926371 +0100
@@ -18,10 +18,13 @@
 #include <cctype>
 #endif
 
-#if ( _MSC_VER == 1200 )
-// VC6 seems to need this
+#if ( _MSC_VER == 1200 ) || ( __GNUC__ )
+// VC6 and GCC seem to need this
 // note that this is not a standard C++ include file.
 # include <stdio.h>
+#if ( __GNUC__ )
+#include <strings.h>
+#endif
 #endif
 
 #include <antlr/TokenStream.hpp>
