*** audio/drivers/pipewire.c.orig	2026-08-06 20:52:50.321742970 -0000
--- audio/drivers/pipewire.c	2026-08-06 20:55:27.681066242 -0000
***************
*** 511,518 ****
  
     /* calculate the total no of bytes to read data from buffer */
     n_bytes = buf->datas[0].maxsize;
!    if (b->requested)
!       n_bytes = MIN(b->requested * audio->frame_size, n_bytes);
  
     avail = spa_ringbuffer_get_read_index(&audio->ring, &idx);
  
--- 511,520 ----
  
     /* calculate the total no of bytes to read data from buffer */
     n_bytes = buf->datas[0].maxsize;
!   #if PW_CHECK_VERSION(0, 3, 49)
! 	 if (b->requested)
! 		n_bytes = MIN(b->requested * audio->frame_size, n_bytes);
!   #endif
  
     avail = spa_ringbuffer_get_read_index(&audio->ring, &idx);
  
