diff -Naur a/Makefile b/Makefile
--- a/Makefile	2012-05-18 22:32:47.000000000 -0400
+++ b/Makefile	2014-02-09 05:51:36.000000000 -0500
@@ -1,5 +1,5 @@
-CFLAGS = -O3 -Wall -Wextra --std=c99 `pkg-config --cflags argtable2 sndfile lilv-0`
-LDLIBS = `pkg-config --libs argtable2 sndfile lilv-0` -lm
+CFLAGS = -O3 -Wall -Wextra --std=c99 `pkg-config --cflags sndfile lilv-0`
+LDLIBS = `pkg-config --libs sndfile lilv-0` -largtable2 -lm
 BINDIR = $(DESTDIR)/usr/bin
 INSTALL_PROGRAM = install
 
