diff -Naur calibre.orig/setup/install.py calibre/setup/install.py
--- calibre.orig/setup/install.py	2010-03-05 21:13:33.000000000 +0000
+++ calibre/setup/install.py	2010-03-07 00:07:41.000000000 +0000
@@ -142,7 +142,9 @@
 
     def install_env_module(self):
         import distutils.sysconfig as s
-        libdir = s.get_python_lib(prefix=self.opts.staging_root)
+        libdir = s.get_python_lib(prefix=(self.opts.staging_root + sys.prefix))
+        if not os.path.exists(libdir):
+            os.makedirs(libdir)
         if os.path.exists(libdir):
             path = os.path.join(libdir, 'init_calibre.py')
             self.info('Installing calibre environment module: '+path)
