--- calibre/setup/extensions.py.orig	2012-12-28 04:59:30.000000000 +0100
+++ calibre/setup/extensions.py	2012-12-28 23:05:09.572200438 +0100
@@ -183,13 +183,6 @@
                 sip_files = ['calibre/gui2/progress_indicator/QProgressIndicator.sip']
                 ),
 
-    Extension('qt_hack',
-                ['calibre/ebooks/pdf/render/qt_hack.cpp'],
-                inc_dirs = qt_private_inc + ['calibre/ebooks/pdf/render', 'qt-harfbuzz/src'],
-                headers = ['calibre/ebooks/pdf/render/qt_hack.h'],
-                sip_files = ['calibre/ebooks/pdf/render/qt_hack.sip']
-                ),
-
     Extension('unrar',
               ['unrar/%s.cpp'%(x.partition('.')[0]) for x in '''
                rar.o strlist.o strfn.o pathfn.o savepos.o smallfn.o global.o file.o
--- calibre/src/calibre/constants.py.orig	2012-12-28 04:59:30.000000000 +0100
+++ calibre/src/calibre/constants.py	2012-12-28 23:07:10.690575371 +0100
@@ -100,7 +100,6 @@
                 'freetype',
                 'woff',
                 'unrar',
-                'qt_hack',
             ]
         if iswindows:
             plugins.extend(['winutil', 'wpd', 'winfonts'])
