diff -Naur calibre-0.5.12.orig/src/calibre/linux.py calibre-0.5.12/src/calibre/linux.py
--- calibre-0.5.12.orig/src/calibre/linux.py	2009-05-15 21:05:52.000000000 +0000
+++ calibre-0.5.12/src/calibre/linux.py	2009-05-16 04:09:44.000000000 +0000
@@ -417,7 +417,7 @@
 def install_man_pages(fatal_errors, use_destdir=False):
     from calibre.utils.help2man import create_man_page
     prefix = os.environ.get('DESTDIR', '/') if use_destdir else '/'
-    manpath = os.path.join(prefix, 'usr/share/man/man1')
+    manpath = os.path.join(prefix, 'usr/man/man1')
     if not os.path.exists(manpath):
         os.makedirs(manpath)
     print 'Installing MAN pages...'
@@ -431,7 +431,7 @@
             continue
         parser = parser()
         raw = create_man_page(prog, parser)
-        manfile = os.path.join(manpath, prog+'.1'+__appname__+'.bz2')
+        manfile = os.path.join(manpath, prog+'.1'+'.bz2')
         print '\tInstalling MAN page for', prog
         open(manfile, 'wb').write(raw)
         manifest.append(manfile)
