diff -Naur calibre.orig/src/calibre/linux.py calibre/src/calibre/linux.py
--- calibre.orig/src/calibre/linux.py	2010-03-05 21:13:33.000000000 +0000
+++ calibre/src/calibre/linux.py	2010-03-07 00:02:26.000000000 +0000
@@ -325,7 +325,7 @@
             if isfreebsd:
                 manpath = os.path.join(self.opts.staging_root, 'man/man1')
             else:
-                manpath = os.path.join(self.opts.staging_sharedir, 'man/man1')
+                manpath = os.path.join(self.opts.staging_root, 'usr/man/man1')
             if not os.path.exists(manpath):
                 os.makedirs(manpath)
             self.info('Installing MAN pages...')
@@ -341,7 +341,7 @@
                 if isfreebsd:
                     manfile = os.path.join(manpath, prog+'.1')
                 else:
-                    manfile = os.path.join(manpath, prog+'.1'+__appname__+'.bz2')
+                    manfile = os.path.join(manpath, prog+'.1'+'.bz2')
                 self.info('\tInstalling MAN page for', prog)
                 open(manfile, 'wb').write(raw)
                 self.manifest.append(manfile)
