diff -Naur calibre.orig/src/calibre/ebooks/markdown/extensions/def_list.py calibre/src/calibre/ebooks/markdown/extensions/def_list.py
--- calibre.orig/src/calibre/ebooks/markdown/extensions/def_list.py	2012-06-29 04:53:39.000000000 +0000
+++ calibre/src/calibre/ebooks/markdown/extensions/def_list.py	2012-07-03 03:31:12.127833772 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env Python
+#!/usr/bin/env python
 """
 Definition List Extension for Python-Markdown
 =============================================
diff -Naur calibre.orig/src/calibre/ebooks/markdown/extensions/meta.py calibre/src/calibre/ebooks/markdown/extensions/meta.py
--- calibre.orig/src/calibre/ebooks/markdown/extensions/meta.py	2012-06-29 04:53:39.000000000 +0000
+++ calibre/src/calibre/ebooks/markdown/extensions/meta.py	2012-07-03 03:31:34.910833575 +0000
@@ -1,4 +1,4 @@
-#!usr/bin/python
+#!/usr/bin/python
 
 """
 Meta Data Extension for Python-Markdown
diff -Naur calibre.orig/src/calibre/ebooks/markdown/extensions/tables.py calibre/src/calibre/ebooks/markdown/extensions/tables.py
--- calibre.orig/src/calibre/ebooks/markdown/extensions/tables.py	2012-06-29 04:53:39.000000000 +0000
+++ calibre/src/calibre/ebooks/markdown/extensions/tables.py	2012-07-03 03:31:45.957833583 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env Python
+#!/usr/bin/env python
 """
 Tables Extension for Python-Markdown
 ====================================
