diff -Naur calibre-3.48.0.orig/src/calibre/utils/icu.c calibre-3.48.0/src/calibre/utils/icu.c
--- calibre-3.48.0.orig/src/calibre/utils/icu.c	2019-09-13 04:46:46.000000000 +0200
+++ calibre-3.48.0/src/calibre/utils/icu.c	2020-12-05 09:20:59.063932000 +0100
@@ -243,14 +243,14 @@
 
     a = python_to_icu(a_, &asz);
     if (a == NULL) goto end;
-    if (asz == 0) { found = TRUE; goto end; }
+    if (asz == 0) { found = true; goto end; }
     b = python_to_icu(b_, &bsz);
     if (b == NULL) goto end;
 
     search = usearch_openFromCollator(a, asz, b, bsz, self->collator, NULL, &status);
     if (U_SUCCESS(status)) {
         pos = usearch_first(search, &status);
-        if (pos != USEARCH_DONE) found = TRUE;
+        if (pos != USEARCH_DONE) found = true;
     }
 end:
     if (search != NULL) usearch_close(search);
diff -Naur calibre-3.48.0.orig/src/calibre/utils/matcher.c calibre-3.48.0/src/calibre/utils/matcher.c
--- calibre-3.48.0.orig/src/calibre/utils/matcher.c	2019-09-13 04:46:46.000000000 +0200
+++ calibre-3.48.0/src/calibre/utils/matcher.c	2020-12-05 09:26:57.800932000 +0100
@@ -15,7 +15,6 @@
 #define inline
 #endif
 
-typedef unsigned char bool;
 #define TRUE 1
 #define FALSE 0
 #define MAX(x, y) ((x > y) ? x : y)
