diff -Naur calibre-0.5.14.orig/setup.py calibre-0.5.14/setup.py
--- calibre-0.5.14.orig/setup.py	2009-05-29 17:25:45.000000000 +0000
+++ calibre-0.5.14/setup.py	2009-06-27 03:26:58.000000000 +0000
@@ -55,8 +55,6 @@
                         upload_to_pypi, stage3, stage2, stage1, upload, \
                         upload_rss
 
-    entry_points['console_scripts'].append(
-                            'calibre_postinstall = calibre.linux:post_install')
     optional = []
 
 
@@ -198,6 +196,4 @@
 
     if 'develop' in ' '.join(sys.argv) and islinux:
         subprocess.check_call('calibre_postinstall --do-not-reload-udev-hal', shell=True)
-    if 'install' in sys.argv and islinux:
-        subprocess.check_call('calibre_postinstall', shell=True)
 
diff -Naur calibre-0.5.14.orig/src/calibre/linux.py calibre-0.5.14/src/calibre/linux.py
--- calibre-0.5.14.orig/src/calibre/linux.py	2009-05-29 17:25:45.000000000 +0000
+++ calibre-0.5.14/src/calibre/linux.py	2009-06-27 03:26:23.000000000 +0000
@@ -444,7 +444,6 @@
     global use_destdir
     use_destdir = opts.destdir
     manifest = []
-    setup_desktop_integration(opts.fatal_errors)
     if opts.no_root or os.geteuid() == 0:
         manifest += install_man_pages(opts.fatal_errors, use_destdir)
         manifest += setup_udev_rules(opts.group_file, not opts.dont_reload, opts.fatal_errors)
@@ -462,13 +461,6 @@
     if opts.save_manifest_to:
         open(opts.save_manifest_to, 'wb').write('\n'.join(manifest)+'\n')
 
-    from calibre.utils.config import config_dir
-    if os.path.exists(config_dir):
-        os.chdir(config_dir)
-        for f in os.listdir('.'):
-            if os.stat(f).st_uid == 0:
-                os.unlink(f)
-
 def binary_install():
     manifest = os.path.join(getattr(sys, 'frozen_path'), 'manifest')
     exes = [x.strip() for x in open(manifest).readlines()]
