diff -Naur calibre.orig/src/calibre/linux.py calibre/src/calibre/linux.py
--- calibre.orig/src/calibre/linux.py	2010-06-11 22:28:24.000000000 +0000
+++ calibre/src/calibre/linux.py	2010-06-13 23:07:49.000000000 +0000
@@ -133,9 +133,6 @@
         if islinux:
             self.setup_completion()
         self.install_man_pages()
-        if islinux:
-            self.setup_desktop_integration()
-        self.create_uninstaller()
 
         from calibre.utils.config import config_dir
         if os.path.exists(config_dir):
