diff -Naur calibre-3.48.0.orig/src/calibre/linux.py calibre-3.48.0/src/calibre/linux.py
--- calibre-3.48.0.orig/src/calibre/linux.py	2019-09-13 04:46:46.000000000 +0200
+++ calibre-3.48.0/src/calibre/linux.py	2020-03-04 15:02:49.789055000 +0100
@@ -725,10 +725,6 @@
         self.appdata_resources = []
         if islinux or isbsd:
             self.setup_completion()
-        if islinux or isbsd:
-            self.setup_desktop_integration()
-        if not getattr(self.opts, 'staged_install', False):
-            self.create_uninstaller()
 
         from calibre.utils.config import config_dir
         if os.path.exists(config_dir):
