diff -Naur calibre-0.5.14.orig/src/tagging/forms.py calibre-0.5.14/src/tagging/forms.py
--- calibre-0.5.14.orig/src/tagging/forms.py	2009-05-29 17:25:45.000000000 +0000
+++ calibre-0.5.14/src/tagging/forms.py	2009-07-01 22:58:17.000000000 +0000
@@ -16,7 +16,7 @@
         value = self.cleaned_data['name']
         tag_names = parse_tag_input(value)
         if len(tag_names) > 1:
-            raise ValidationError(_('Multiple tags were given.'))
+            raise forms.ValidationError(_('Multiple tags were given.'))
         elif len(tag_names[0]) > settings.MAX_TAG_LENGTH:
             raise forms.ValidationError(
                 _('A tag may be no more than %s characters long.') %
