diff -Naur atom-0.94.0/build/Gruntfile.coffee atom-0.94.0.a/build/Gruntfile.coffee
--- atom-0.94.0/build/Gruntfile.coffee	2014-05-06 10:56:03.000000000 -0600
+++ atom-0.94.0.a/build/Gruntfile.coffee	2014-05-09 18:58:26.226372541 -0600
@@ -58,7 +58,7 @@
     installDir = path.join('/Applications', appName)
   else
     appName = 'Atom'
-    tmpDir = '/tmp'
+    tmpDir = process.env.BUILD_DIR ? '/tmp'
     buildDir = grunt.option('build-dir') ? path.join(tmpDir, 'atom-build')
     symbolsDir = path.join(buildDir, 'Atom.breakpad.syms')
     shellAppDir = path.join(buildDir, appName)
