diff -Naur atom-0.99.0/build/Gruntfile.coffee atom-0.99.0.a/build/Gruntfile.coffee
--- atom-0.99.0/build/Gruntfile.coffee	2014-05-23 15:00:33.000000000 -0600
+++ atom-0.99.0.a/build/Gruntfile.coffee	2014-05-26 14:12:39.212928087 -0600
@@ -36,7 +36,7 @@
     grunt.log.write = (args...) -> grunt.log
 
   [major, minor, patch] = packageJson.version.split('.')
-  tmpDir = os.tmpdir()
+  tmpDir = process.env.BUILD_DIR ? os.tmpdir()
   appName = if process.platform is 'darwin' then 'Atom.app' else 'Atom'
   buildDir = grunt.option('build-dir') ? path.join(tmpDir, 'atom-build')
   atomShellDownloadDir = path.join(os.tmpdir(), 'atom-cached-atom-shells')
