From 7ebdb30fdf91d1f63b4fb07e54b089136de5507b Mon Sep 17 00:00:00 2001
From: Lukas Fleischer <lfleischer@lfos.de>
Date: Wed, 5 Apr 2017 06:38:27 +0200
Subject: [PATCH 07/15] Remove unused variable from sort_section()

Signed-off-by: Lukas Fleischer <lfleischer@lfos.de>
---
 ui-repolist.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/ui-repolist.c b/ui-repolist.c
index 20a4f56..7272e87 100644
--- a/ui-repolist.c
+++ b/ui-repolist.c
@@ -225,7 +225,6 @@ static int sort_section(const void *a, const void *b)
 	const struct cgit_repo *r1 = a;
 	const struct cgit_repo *r2 = b;
 	int result;
-	time_t t;
 
 	result = cmp(r1->section, r2->section);
 	if (!result) {
-- 
2.14.1

