--- FeLib/Include/fearray.h.old	2006-05-21 16:02:19.000000000 +0200
+++ FeLib/Include/fearray.h	2006-05-21 16:03:16.000000000 +0200
@@ -34,7 +34,7 @@
 };
 
 template <class type>
-inline fearray<type>::fearray<type>(const fearray<type>& A)
+inline fearray<type>::fearray(const fearray<type>& A)
 : Data(A.Data), Size(A.Size)
 {
   if(Data)
@@ -42,7 +42,7 @@
 }
 
 template <class type>
-inline fearray<type>::fearray<type>(const type* Array, sizetype Size)
+inline fearray<type>::fearray(const type* Array, sizetype Size)
 : Size(Size)
 {
   char* Ptr = new char[Size * sizeof(type) + sizeof(ulong)];
--- Main/Source/script.cpp.old	2006-05-21 16:14:43.000000000 +0200
+++ Main/Source/script.cpp	2006-05-21 16:16:12.000000000 +0200
@@ -471,7 +471,7 @@
   INIT_ENTRY(Flags);
 }
 
-contentscript<character>::contentscript<character>()
+contentscript<character>::contentscript()
 : INIT(Team, DEFAULT_TEAM),
   INIT(Flags, 0)
 { }
@@ -498,7 +498,7 @@
   return Instance;
 }
 
-contentscript<item>::contentscript<item>()
+contentscript<item>::contentscript()
 : INIT(Category, ANY_CATEGORY),
   INIT(MinPrice, 0),
   INIT(MaxPrice, MAX_PRICE),
@@ -592,7 +592,7 @@
   INIT_ENTRY(IsInside);
 }
 
-contentscript<olterrain>::contentscript<olterrain>()
+contentscript<olterrain>::contentscript()
 : INIT(VisualEffects, 0),
   INIT(AttachedArea, DEFAULT_ATTACHED_AREA),
   INIT(AttachedEntry, DEFAULT_ATTACHED_ENTRY)
@@ -679,7 +679,7 @@
   }
 }
 
-template <class type, class contenttype> contentmap<type, contenttype>::contentmap<type, contenttype>() : ContentMap(0) { }
+template <class type, class contenttype> contentmap<type, contenttype>::contentmap() : ContentMap(0) { }
 
 template <class type, class contenttype> contentmap<type, contenttype>::~contentmap<type, contenttype>()
 {
