diff -uprNEBZ --suppress-blank-empty a/compton-conf.desktop.in b/compton-conf.desktop.in
--- a/compton-conf.desktop.in	2015-11-27 11:29:17.187956000 -0600
+++ b/compton-conf.desktop.in	2015-11-27 11:32:12.851906393 -0600
@@ -6,6 +6,6 @@ Comment=Configure Compton window effects
 TryExec=compton-conf
 Exec=compton-conf
 Icon=preferences-system-windows
-Categories=Settings;DesktopSettings;Qt;LXQt;
+Categories=Settings;DesktopSettings;Qt;X-LXQt;

 #TRANSLATIONS_DIR=translations
