--- a/config.layout	2007-02-26 20:55:04.000000000 +0100
+++ b/config.layout	2007-02-26 20:58:49.000000000 +0100
@@ -322,3 +322,27 @@
     installbuilddir: ${prefix}/etc/apache2/build
     errordir:      ${datadir}/error
 </Layout>
+
+# FHS layout
+<Layout Slackware-FHS>
+    prefix:          /usr
+    exec_prefix:     ${prefix}
+    bindir:          ${prefix}/bin
+    sbindir:         ${prefix}/sbin
+    libdir:          ${prefix}/lib/apache2
+    libexecdir:      ${prefix}/libexec/apache2
+    installbuilddir: ${prefix}/libexec/apache2/build
+    mandir:          ${prefix}/man
+    sysconfdir:      /etc/apache2
+    datadir:         /var/www
+    iconsdir:        ${datadir}/icons
+    htdocsdir:       ${datadir}/htdocs
+    manualdir:       ${htdocsdir}/manual
+    cgidir:          ${datadir}/cgi-bin
+    errordir:        ${datadir}/error
+    includedir:      ${prefix}/include/apache2
+    localstatedir:   /var
+    runtimedir:      ${localstatedir}/run/apache2
+    logfiledir:      ${localstatedir}/log/apache2
+    proxycachedir:   ${localstatedir}/cache/apache2
+</Layout>
