--- a/httpd.conf	2007-02-26 21:44:23.000000000 +0100
+++ b/httpd.conf	2007-02-26 22:24:12.000000000 +0100
@@ -88,7 +88,7 @@
 LoadModule proxy_http_module libexec/apache2/mod_proxy_http.so
 LoadModule proxy_ajp_module libexec/apache2/mod_proxy_ajp.so
 LoadModule proxy_balancer_module libexec/apache2/mod_proxy_balancer.so
-LoadModule ssl_module libexec/apache2/mod_ssl.so
+#LoadModule ssl_module libexec/apache2/mod_ssl.so
 LoadModule mime_module libexec/apache2/mod_mime.so
 LoadModule dav_module libexec/apache2/mod_dav.so
 LoadModule status_module libexec/apache2/mod_status.so
@@ -116,8 +116,8 @@
 # It is usually good practice to create a dedicated user and group for
 # running httpd, as with most system services.
 #
-User daemon
-Group daemon
+User apache2
+Group apache2
 </IfModule>
 
 # 'Main' server configuration
@@ -419,34 +419,34 @@
 # necessary.
 
 # Server-pool management (MPM specific)
-#Include /etc/apache2/extra/httpd-mpm.conf
+Include /etc/apache2/extra/httpd-mpm.conf
 
 # Multi-language error messages
 #Include /etc/apache2/extra/httpd-multilang-errordoc.conf
 
 # Fancy directory listings
-#Include /etc/apache2/extra/httpd-autoindex.conf
+Include /etc/apache2/extra/httpd-autoindex.conf
 
 # Language settings
 #Include /etc/apache2/extra/httpd-languages.conf
 
 # User home directories
-#Include /etc/apache2/extra/httpd-userdir.conf
+Include /etc/apache2/extra/httpd-userdir.conf
 
 # Real-time info on requests and configuration
-#Include /etc/apache2/extra/httpd-info.conf
+Include /etc/apache2/extra/httpd-info.conf
 
 # Virtual hosts
 #Include /etc/apache2/extra/httpd-vhosts.conf
 
 # Local access to the Apache HTTP Server Manual
-#Include /etc/apache2/extra/httpd-manual.conf
+Include /etc/apache2/extra/httpd-manual.conf
 
 # Distributed authoring and versioning (WebDAV)
 #Include /etc/apache2/extra/httpd-dav.conf
 
 # Various default settings
-#Include /etc/apache2/extra/httpd-default.conf
+Include /etc/apache2/extra/httpd-default.conf
 
 # Secure (SSL/TLS) connections
 #Include /etc/apache2/extra/httpd-ssl.conf
