diff -aur a/extra/httpd-default.conf b/extra/httpd-default.conf
--- a/extra/httpd-default.conf	2007-01-21 21:19:30.000000000 +0100
+++ b/extra/httpd-default.conf	2007-01-21 21:23:20.000000000 +0100
@@ -52,7 +52,7 @@
 # Set to one of:  Full | OS | Minor | Minimal | Major | Prod
 # where Full conveys the most information, and Prod the least.
 #
-ServerTokens Full
+ServerTokens Prod
 
 #
 # Optionally add a line containing the server version and virtual host
@@ -62,7 +62,7 @@
 # Set to "EMail" to also include a mailto: link to the ServerAdmin.
 # Set to one of:  On | Off | EMail
 #
-ServerSignature On
+ServerSignature Off
 
 #
 # HostnameLookups: Log the names of clients or just their IP addresses
diff -aur a/extra/httpd-info.conf b/extra/httpd-info.conf
--- a/extra/httpd-info.conf	2007-01-21 21:19:30.000000000 +0100
+++ b/extra/httpd-info.conf	2007-01-21 21:24:32.000000000 +0100
@@ -14,7 +14,7 @@
     SetHandler server-status
     Order deny,allow
     Deny from all
-    Allow from .example.com
+    Allow from localhost
 </Location>
 
 #
@@ -33,5 +33,5 @@
     SetHandler server-info
     Order deny,allow
     Deny from all
-    Allow from .example.com
+    Allow from localhost
 </Location>
