diff -Nur comgt.0.32/Makefile comgt.0.32.new/Makefile
--- comgt.0.32/Makefile	2006-10-20 14:49:46.000000000 +0000
+++ comgt.0.32.new/Makefile	2010-12-16 00:03:14.416680046 +0000
@@ -23,12 +23,12 @@
 #
 #
 
-LIB     = -L/usr/local/lib
-INC     = -I/usr/local/include
-EXE	= /usr/local/bin
-MAN     = /usr/share/man/man1
+#LIB     = -L/lib
+#INC     = -I/usr/include
+EXE	= $(DESTDIR)/usr/bin
+MAN     = $(DESTDIR)/usr/man/man1
 CPROG	= comgt
-SCRIPTPATH = /etc/comgt/
+SCRIPTPATH = $(DESTDIR)/etc/comgt/
 SCRIPTSRC = ./scripts/
 BIN     = $(CPROG) 
 MANP	= comgt.1 sigmon.1
