diff -Naur madwifi-0.9.4.orig/ath/if_athvar.h madwifi-0.9.4/ath/if_athvar.h
--- madwifi-0.9.4.orig/ath/if_athvar.h	2008-02-12 06:05:28.000000000 +0000
+++ madwifi-0.9.4/ath/if_athvar.h	2009-09-25 15:46:40.000000000 +0000
@@ -94,11 +94,13 @@
 /*
  * Guess how the interrupt handler should work.
  */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,29)
 #if !defined(IRQ_NONE)
 typedef void irqreturn_t;
 #define	IRQ_NONE
 #define	IRQ_HANDLED
 #endif /* !defined(IRQ_NONE) */
+#endif /* Linux < 2.6.29 */
 
 #ifndef SET_MODULE_OWNER
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,23)
