From 94bd4ae76f0fb3233eda58794e0105f6e3be92ee Mon Sep 17 00:00:00 2001
From: arvidn <arvid@libtorrent.org>
Date: Fri, 13 Dec 2019 21:58:09 +0100
Subject: [PATCH] support building with new boost version

---
 ChangeLog                          | 1 +
 include/libtorrent/socket_type.hpp | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/ChangeLog b/ChangeLog
index 6c74a8067a..e060ab190e 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,6 @@
 1.1.14 release
 
+	* support building with new boost version
 	* add handling on invalid piece index to torrent_handle::have_piece()
 	* fix error in treatment of move_storage() flags
 	* fix division by zero in anti-leech choker
diff --git a/include/libtorrent/socket_type.hpp b/include/libtorrent/socket_type.hpp
index 578b4534a9..69eabd2569 100644
--- a/include/libtorrent/socket_type.hpp
+++ b/include/libtorrent/socket_type.hpp
@@ -190,6 +190,10 @@ namespace libtorrent
 		typedef tcp::socket::protocol_type protocol_type;
 
 		typedef tcp::socket::receive_buffer_size receive_buffer_size;
+
+#if BOOST_VERSION >= 106600
+		using executor_type = tcp::socket::executor_type;
+#endif
 		typedef tcp::socket::send_buffer_size send_buffer_size;
 
 		explicit socket_type(io_service& ios): m_io_service(ios), m_type(0) {}
