--- include/ptypes.h.orig	2016-08-20 00:40:10.316942041 -0400
+++ include/ptypes.h	2016-08-20 00:40:17.052941703 -0400
@@ -42,7 +42,7 @@
     { return (T*)pexchange((void**)target, (void*)value); }
 
 
-#if ((__GNUC__ == 3) && (__GNUC_MINOR__ >= 3)) || (__GNUC__ == 4) || defined(__hpux)
+#if ((__GNUC__ == 3) && (__GNUC_MINOR__ >= 3)) || (__GNUC__ >= 4) || defined(__hpux)
 #  define VARIANT_TYPECAST_HACK
 #endif
 
