--- Makefile.orig	2014-01-06 15:05:19.565762653 +0100
+++ Makefile	2014-01-06 15:05:28.701192221 +0100
@@ -29,7 +29,7 @@ CFLAGS	 	+= -DUSE_WCHAR
 # for example on Mac OS X.
 STATIC		 = -static
 # Linux requires -pthread to statically link with libdb.
-#STATIC		+= -pthread
+STATIC		+= -pthread
 
 CFLAGS		+= -g -DHAVE_CONFIG_H
 CFLAGS     	+= -W -Wall -Wstrict-prototypes -Wno-unused-parameter -Wwrite-strings
@@ -54,7 +54,7 @@ INSTALL_MAN	 = $(INSTALL_DATA)
 # However, if you don't have -ldb at all (or it's not native), then
 # comment out apropos and mandocdb. 
 #
-#DBLIB		 = -ldb
+DBLIB		 = -ldb
 DBBIN		 = apropos mandocdb man.cgi catman whatis
 DBLN		 = llib-lapropos.ln llib-lmandocdb.ln llib-lman.cgi.ln llib-lcatman.ln
 
