diff -Naur bstone-1.2.8/src/3d_main.cpp bstone-1.2.8.patched/src/3d_main.cpp
--- bstone-1.2.8/src/3d_main.cpp	2020-07-10 09:58:35.000000000 -0400
+++ bstone-1.2.8.patched/src/3d_main.cpp	2020-08-04 03:07:36.266707676 -0400
@@ -10722,20 +10722,7 @@
 
 const std::string& get_default_data_dir()
 {
-	static std::string result;
-	static auto is_initialized = false;
-
-	if (!is_initialized)
-	{
-		is_initialized = true;
-
-		result = bstone::file_system::get_working_dir();
-
-#ifdef __vita__
-		result = "ux0:/data/bstone/";
-#endif
-	}
-
+	static std::string result = "/usr/share/games/bstone/";
 	return result;
 }
 
