diff -Naur bstone-1.1.14/src/3d_main.cpp bstone-1.1.14.patched/src/3d_main.cpp
--- bstone-1.1.14/src/3d_main.cpp	2019-11-09 05:08:15.000000000 -0500
+++ bstone-1.1.14.patched/src/3d_main.cpp	2019-11-27 14:08:41.481084182 -0500
@@ -10523,24 +10523,7 @@
 
 const std::string& get_default_data_dir()
 {
-	static std::string result;
-	static auto is_initialized = false;
-
-	if (!is_initialized)
-	{
-		is_initialized = true;
-
-		auto sdl_dir = ::SDL_GetBasePath();
-
-		if (sdl_dir)
-		{
-			result = sdl_dir;
-			::SDL_free(sdl_dir);
-		}
-	}
-#ifdef __vita__
-	result = "ux0:/data/bstone/";
-#endif
+	static std::string result = "/usr/share/games/bstone/";
 	return result;
 }
 
