diff -Naur bstone-1.2.11/src/3d_main.cpp bstone-1.2.11.patched/src/3d_main.cpp
--- bstone-1.2.11/src/3d_main.cpp	2021-03-21 15:24:36.000000000 -0400
+++ bstone-1.2.11.patched/src/3d_main.cpp	2021-10-04 19:12:08.818938051 -0400
@@ -10711,20 +10711,7 @@
 
 const std::string& get_default_data_dir()
 {
-	static std::string result;
-	static auto is_initialized = false;
-
-	if (!is_initialized)
-	{
-		is_initialized = true;
-
-		result = bstone::file_system::get_working_dir();
-
-#ifdef __vita__
-		result = "ux0:/data/bstone/";
-#endif
-	}
-
+	static std::string result = "/usr/share/games/bstone/";
 	return result;
 }
 
diff -Naur bstone-1.2.11/src/jm_free.cpp bstone-1.2.11.patched/src/jm_free.cpp
--- bstone-1.2.11/src/jm_free.cpp	2021-03-21 15:24:36.000000000 -0400
+++ bstone-1.2.11.patched/src/jm_free.cpp	2021-10-04 19:12:39.570935256 -0400
@@ -718,6 +718,7 @@
 				bstone::file_system::get_working_dir());
 
 			add_search_path("working dir", working_full_dir, search_paths);
+			add_search_path("system dir", "/usr/share/games/bstone", search_paths);
 		}
 
 		{
