diff -Naur bstone-1.2.13/src/bstone/src/3d_main.cpp bstone-1.2.13.patched/src/bstone/src/3d_main.cpp
--- bstone-1.2.13/src/bstone/src/3d_main.cpp	2024-12-14 10:29:49.000000000 -0500
+++ bstone-1.2.13.patched/src/bstone/src/3d_main.cpp	2025-01-29 03:30:03.334126326 -0500
@@ -10837,20 +10837,7 @@
 
 const std::string& get_default_data_dir()
 {
-	static std::string result;
-	static auto is_initialized = false;
-
-	if (!is_initialized)
-	{
-		is_initialized = true;
-
-		result = bstone::fs_utils::get_working_dir();
-
-#ifdef __vita__
-		result = "ux0:/data/bstone/";
-#endif
-	}
-
+	static std::string result = "/usr/share/games/bstone/";
 	return result;
 }
 
diff -Naur bstone-1.2.13/src/bstone/src/jm_free.cpp bstone-1.2.13.patched/src/bstone/src/jm_free.cpp
--- bstone-1.2.13/src/bstone/src/jm_free.cpp	2024-12-14 10:29:49.000000000 -0500
+++ bstone-1.2.13.patched/src/bstone/src/jm_free.cpp	2025-01-29 03:30:37.581123103 -0500
@@ -677,6 +677,7 @@
 			bstone::fs_utils::get_working_dir());
 
 		add_search_path("working dir", working_full_dir, search_paths);
+		add_search_path("system dir", "/usr/share/games/bstone", search_paths);
 	}
 
 	{
