diff --git a/avidemux/ADM_userInterfaces/ADM_QT4/ui_support.cpp b/avidemux/ADM_userInterfaces/ADM_QT4/ui_support.cpp
index 4cce636..b875528 100644
--- a/avidemux/ADM_userInterfaces/ADM_QT4/ui_support.cpp
+++ b/avidemux/ADM_userInterfaces/ADM_QT4/ui_support.cpp
@@ -73,7 +73,7 @@ void loadTranslator(void)
 #ifdef __APPLE__
 	QString appdir = QCoreApplication::applicationDirPath() + "/../Resources/locale/";
 #else
-	QString appdir = QCoreApplication::applicationDirPath() + "/i18n/";
+	QString appdir = "/usr/share/avidemux/i18n/";
 #endif
 
 	loadTranslation(&qtTranslator, appdir + "qt_" + QLocale::system().name());
@@ -164,4 +164,4 @@ const char* getNativeRendererDesc(void)
 	}
 
 	return "Qt";
-}
\ No newline at end of file
+}
diff --git a/cmake/Ts.cmake b/cmake/Ts.cmake
index 15eb89b..e40f078 100644
--- a/cmake/Ts.cmake
+++ b/cmake/Ts.cmake
@@ -64,7 +64,7 @@ MACRO(COMPILE_AVIDEMUX_TS_FILES ts_subdir _sources)
                 
             SET(qm_files ${qm_files} ${_outXml} ${_out})
 
-			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/${BIN_DIR}/i18n")
+			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/share/avidemux/i18n")
         ENDFOREACH(ts_input ${ts_files})
 
         SET(${_sources} ${${_sources}} ${qm_files})
@@ -95,7 +95,7 @@ MACRO(COMPILE_QT_TS_FILES ts_subdir _sources)
                 
             SET(qm_files ${qm_files} ${_out})
 
-			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/${BIN_DIR}/i18n")
+			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/share/avidemux/i18n")
         ENDFOREACH(ts_input ${ts_files})
 
         SET(${_sources} ${${_sources}} ${qm_files})
