diff -ur avidemux_2.4.orig/ConfigureChecks.cmake avidemux_2.4/ConfigureChecks.cmake
--- avidemux_2.4.orig/ConfigureChecks.cmake	2007-12-04 14:52:16.000000000 +0100
+++ avidemux_2.4/ConfigureChecks.cmake	2007-12-23 04:10:19.000000000 +0100
@@ -441,7 +441,7 @@
 IF (USE_LATE_BINDING)
 	CHECK_INCLUDE_FILES(dts.h USE_LIBDCA)
 ELSE (USE_LATE_BINDING)
-	ADM_CHECK_HL(libdca dts.h dts dts_init USE_LIBDCA)
+	ADM_CHECK_HL(libdca dts.h dca dca_init USE_LIBDCA)
 ENDIF (USE_LATE_BINDING)
 
 SET(CMAKE_REQUIRED_LIBRARIES)
