diff -Naur avidemux_2.8.1/avidemux_plugins/ADM_videoEncoder/x265/ADM_x265Setup.cpp avidemux_2.8.1.patched/avidemux_plugins/ADM_videoEncoder/x265/ADM_x265Setup.cpp
--- avidemux_2.8.1/avidemux_plugins/ADM_videoEncoder/x265/ADM_x265Setup.cpp	2022-09-17 09:59:17.000000000 -0400
+++ avidemux_2.8.1.patched/avidemux_plugins/ADM_videoEncoder/x265/ADM_x265Setup.cpp	2024-12-05 04:27:42.629540408 -0500
@@ -207,13 +207,13 @@
                         {
                              param.rc.bStatWrite=1;
                              param.rc.bStatRead=0;
-                             param.rc.statFileName=strdup(logFile);
+                             strncpy(param.rc.statFileName, logFile, 256);
  
                         }else
                         {
                              param.rc.bStatWrite=0;
                              param.rc.bStatRead=1;
-                             param.rc.statFileName=strdup(logFile);
+                             strncpy(param.rc.statFileName, logFile, 256);
                              if(!ADM_fileExist(logFile))
                              {
                                    ADM_error("Logfile %s does not exist \n",logFile);
