diff --git a/configure b/configure
index 78ddf3e..a730c3d 100755
--- a/configure
+++ b/configure
@@ -5553,7 +5553,7 @@ $as_echo_n "checking for the kernel version... " >&6; }
         kernel=`uname -r`
 
         case "${kernel}" in
-             2.6.*|3.*)
+             2.6.*|3.*|4.*)
         	    { $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.6.x/3.x family (${kernel})" >&5
 $as_echo "2.6.x/3.x family (${kernel})" >&6; }
 
@@ -6832,7 +6832,7 @@ $as_echo "#define GLIBC_2_19 1" >>confdefs.h
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
-     2.20)
+     2.2?)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.20 family" >&5
 $as_echo "2.20 family" >&6; }
 
