--- carbon-0.9.13-pre1/conf/carbon.conf.example.orig	2014-12-31 17:50:50.000000000 +0100
+++ carbon-0.9.13-pre1/conf/carbon.conf.example	2015-02-04 00:36:17.680898110 +0100
@@ -14,30 +14,21 @@
 # To change other directory paths, add settings to this file. The following
 # configuration variables are available with these default values:
 #
-#   STORAGE_DIR    = $GRAPHITE_STORAGE_DIR
-#   LOCAL_DATA_DIR = STORAGE_DIR/whisper/
-#   WHITELISTS_DIR = STORAGE_DIR/lists/
-#   CONF_DIR       = STORAGE_DIR/conf/
-#   LOG_DIR        = STORAGE_DIR/log/
-#   PID_DIR        = STORAGE_DIR/
-#
-# For FHS style directory structures, use:
-#
-#   STORAGE_DIR    = /var/lib/carbon/
-#   CONF_DIR       = /etc/carbon/
-#   LOG_DIR        = /var/log/carbon/
-#   PID_DIR        = /var/run/
-#
-#LOCAL_DATA_DIR = /opt/graphite/storage/whisper/
+STORAGE_DIR    = /var/lib/graphite/
+CONF_DIR       = /etc/carbon/
+LOG_DIR        = /var/log/carbon/
+PID_DIR        = /var/run/
+
+LOCAL_DATA_DIR = /var/lib/graphite/whisper/
 
 # Enable daily log rotation. If disabled, carbon will automatically re-open
 # the file if it's rotated out of place (e.g. by logrotate daemon)
-ENABLE_LOGROTATION = True
+ENABLE_LOGROTATION = False
 
 # Specify the user to drop privileges to
 # If this is blank carbon runs as the user that invokes it
 # This user must have write access to the local data directory
-USER =
+USER = graphite
 #
 # NOTE: The above settings must be set under [relay] and [aggregator]
 #       to take effect for those daemons as well
