--- carbon-0.9.13-pre1/bin/validate-storage-schemas.py.orig	2014-12-31 17:50:50.000000000 +0100
+++ carbon-0.9.13-pre1/bin/validate-storage-schemas.py	2015-02-04 00:40:16.244602801 +0100
@@ -22,7 +22,7 @@
   SCHEMAS_FILE = sys.argv[1]
   print "Loading storage-schemas configuration from: '%s'" % SCHEMAS_FILE
 else:
-  SCHEMAS_FILE = realpath(join(dirname(__file__), '..', 'conf', 'storage-schemas.conf'))
+  SCHEMAS_FILE = '/etc/carbon/storage-schemas.conf'
   print "Loading storage-schemas configuration from default location at: '%s'" % SCHEMAS_FILE
 
 config_parser = ConfigParser()
