diff --git a/src/3rdparty/autotrace/autotrace.h b/src/3rdparty/autotrace/autotrace.h
index 2ce6f272a997cee8c7839412a520106d5584ce4c..e56be9a74c4308cacc4043ef0855edae9cdef0a6 100644
--- a/src/3rdparty/autotrace/autotrace.h
+++ b/src/3rdparty/autotrace/autotrace.h
@@ -23,6 +23,9 @@
 
 #include <stdio.h>
 
+#include "types.h"
+#include "color.h"
+
 #ifdef __cplusplus
 extern "C" {
 #endif                          /* __cplusplus */
@@ -35,9 +38,6 @@ extern "C" {
  * Typedefs
  * ===================================================================== */
 
-#include "types.h"
-#include "color.h"
-
 /* Third degree is the highest we deal with.  */
   enum _at_polynomial_degree {
     AT_LINEARTYPE = 1,
diff --git a/src/3rdparty/autotrace/color.h b/src/3rdparty/autotrace/color.h
index e50ab30ae04d1861c4d9aa0f393650d2f2c102da..88651db9f73571cdc5f14cdf4645e8fa69e03912 100644
--- a/src/3rdparty/autotrace/color.h
+++ b/src/3rdparty/autotrace/color.h
@@ -24,6 +24,10 @@
 #include <glib.h>
 #include <glib-object.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif                          /* __cplusplus */
+
 typedef struct _at_color at_color;
 struct _at_color {
   guint8 r;
@@ -43,4 +47,7 @@ void at_color_free(at_color * color);
 GType at_color_get_type(void);
 #define AT_TYPE_COLOR (at_color_get_type ())
 
+#ifdef __cplusplus
+}
+#endif                          /* __cplusplus */
 #endif /* not AT_COLOR_H */
