diff -ru vile-9.8s.orig/filters/sh-filt.l vile-9.8s/filters/sh-filt.l
--- vile-9.8s.orig/filters/sh-filt.l	2016-12-17 18:24:00.000000000 -0500
+++ vile-9.8s/filters/sh-filt.l	2018-10-13 13:40:34.077559351 -0400
@@ -198,7 +198,7 @@
 			  }
 			}
 <NORMAL>{KSH_TEST_UP}	{ if (!pop_backtic1(yytext, yyleng, LEN_KSH_TEST_DOWN)) {
-			    FLEX_PRINTF((stderr, "cannot pop '[[' level %d\n", stk_level));
+			    FLEX_PRINTF((stderr, "cannot pop '\[\[' level %d\n", stk_level));
 			    handle_parens(yytext, 1, 1, -1, NORMAL);
 			    unput(R_BLOCK);
 			  }
