diff -Naur grub-2.00.orig/util/grub.d/10_linux.in grub-2.00/util/grub.d/10_linux.in
--- grub-2.00.orig/util/grub.d/10_linux.in	2012-04-18 23:24:38.000000000 +0200
+++ grub-2.00/util/grub.d/10_linux.in	2012-06-30 07:53:03.765625589 +0200
@@ -198,7 +198,8 @@
 	   "initramfs-genkernel-${version}" \
 	   "initramfs-genkernel-${alt_version}" \
 	   "initramfs-genkernel-${GENKERNEL_ARCH}-${version}" \
-	   "initramfs-genkernel-${GENKERNEL_ARCH}-${alt_version}"; do
+	   "initramfs-genkernel-${GENKERNEL_ARCH}-${alt_version}" \
+	   "initrd.gz"; do
     if test -e "${dirname}/${i}" ; then
       initrd="$i"
       break
