diff -Naur svox-1.0+git20130326/pico/bin/pico2wave.c svox-1.0+git20130326.patched/pico/bin/pico2wave.c
--- svox-1.0+git20130326/pico/bin/pico2wave.c	2018-06-17 02:50:56.203338391 -0400
+++ svox-1.0+git20130326.patched/pico/bin/pico2wave.c	2018-06-17 03:00:01.026324549 -0400
@@ -371,8 +371,6 @@
         pico_terminate(&picoSystem);
         picoSystem = NULL;
     }
-    if (text)
-      free(text);
     exit(ret);
 }
 
