--- build_files/cmake/buildinfo.cmake.orig	2016-10-25 19:59:23.000000000 +1000
+++ build_files/cmake/buildinfo.cmake	2016-11-13 17:31:44.921023097 +1000
@@ -8,9 +8,10 @@
 
 # Extract working copy information for SOURCE_DIR into MY_XXX variables
 # with a default in case anything fails, for example when using git-svn
-set(MY_WC_HASH "unknown")
-set(MY_WC_BRANCH "unknown")
-set(MY_WC_COMMIT_TIMESTAMP 0)
+# Timestamp from $(date --date='TZ="UTC" 15:33 24-Feb-17' +%s) - time & date of commit
+set(MY_WC_HASH "e92f23528307")
+set(MY_WC_BRANCH "master")
+set(MY_WC_COMMIT_TIMESTAMP 1487950380)
 
 # Guess if this is a git working copy and then look up the revision
 if(EXISTS ${SOURCE_DIR}/.git)
