--- build_files/cmake/buildinfo.cmake.orig	2015-07-28 11:59:14.020100604 +0000
+++ build_files/cmake/buildinfo.cmake	2015-07-28 12:18:20.021045004 +0000
@@ -138,11 +138,12 @@
 	execute_process(COMMAND cmd /c time /t OUTPUT_VARIABLE BUILD_TIME OUTPUT_STRIP_TRAILING_WHITESPACE)
 endif()
 
+# Timestamp from $(date --date='TZ="UTC" 16:56 07-Jul-15' +%s) - time & date of commit
 # Write a file with the BUILD_HASH define
 file(WRITE buildinfo.h.txt
-	"#define BUILD_HASH \"${MY_WC_HASH}\"\n"
-	"#define BUILD_COMMIT_TIMESTAMP ${MY_WC_COMMIT_TIMESTAMP}\n"
-	"#define BUILD_BRANCH \"${MY_WC_BRANCH}\"\n"
+	"#define BUILD_HASH \"rB000dfc03190d\"\n"
+	"#define BUILD_COMMIT_TIMESTAMP 1427816340\n"
+	"#define BUILD_BRANCH \"master\"\n"
 	"#define BUILD_DATE \"${BUILD_DATE}\"\n"
 	"#define BUILD_TIME \"${BUILD_TIME}\"\n"
 )
