--- build_files/cmake/buildinfo.cmake.orig	2017-09-11 14:34:59.000000000 +1000
+++ build_files/cmake/buildinfo.cmake	2017-09-14 11:10:52.213995610 +1000
@@ -8,9 +8,10 @@
 
 # Extract working copy information for SOURCE_DIR into MY_XXX variables
 # with a default in case anything fails, for example when using git-svn
-set(MY_WC_HASH "unknown")
-set(MY_WC_BRANCH "unknown")
-set(MY_WC_COMMIT_TIMESTAMP 0)
+# Timestamp from $(date --date='TZ="UTC" 01:24 24-Mar-18' +%s) - time & date of commit
+set(MY_WC_HASH "f4dc9f9d68bd")
+set(MY_WC_BRANCH "master")
+set(MY_WC_COMMIT_TIMESTAMP 1521854640)
 
 # Guess if this is a git working copy and then look up the revision
 if(EXISTS ${SOURCE_DIR}/.git)
