--- build_files/cmake/buildinfo.cmake.orig	2019-07-24 17:41:38.000000000 +1000
+++ build_files/cmake/buildinfo.cmake	2019-07-31 17:36:21.614279355 +1000
@@ -8,9 +8,10 @@
 
 # Extract working copy information for SOURCE_DIR into MY_XXX variables
 # with a default in case anything fails, for example when using git-svn
-set(MY_WC_HASH "unknown")
-set(MY_WC_BRANCH "unknown")
-set(MY_WC_COMMIT_TIMESTAMP 0)
+# Timestamp from $(date --date='TZ="UTC" 16:47 29-Jul-19' +%s) - time & date of commit
+set(MY_WC_HASH "f6cb5f54494e")
+set(MY_WC_BRANCH "blender-v2.80-release")
+set(MY_WC_COMMIT_TIMESTAMP 1564418820)
 
 # Guess if this is a git working copy and then look up the revision
 if(EXISTS ${SOURCE_DIR}/.git)
