diff -Naur blender-3.3.10.orig/CMakeLists.txt blender-3.3.10/CMakeLists.txt
--- blender-3.3.10.orig/CMakeLists.txt	2023-05-15 15:48:41.000000000 +0200
+++ blender-3.3.10/CMakeLists.txt	2024-02-01 12:13:12.018443284 +0100
@@ -170,7 +170,8 @@
 mark_as_advanced(BUILDINFO_OVERRIDE_DATE)
 mark_as_advanced(BUILDINFO_OVERRIDE_TIME)
 
-if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.16")
+# CMAKE 3.28.2 has issues with the combination of PCH and unity builds, disable for now.
+if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.16" AND NOT ${CMAKE_VERSION} VERSION_EQUAL "3.28.2")
   option(WITH_UNITY_BUILD   "Enable unity build for modules that support it to improve compile times" ON)
   mark_as_advanced(WITH_UNITY_BUILD)
 else()
