diff -Nru blender-2.74.orig/build_files/cmake/Modules/FindOpenJPEG.cmake blender-2.74/build_files/cmake/Modules/FindOpenJPEG.cmake
--- blender-2.74.orig/build_files/cmake/Modules/FindOpenJPEG.cmake	2015-03-25 05:56:13.000000000 -0500
+++ blender-2.74/build_files/cmake/Modules/FindOpenJPEG.cmake	2016-01-09 20:17:20.521735656 -0600
@@ -43,11 +43,12 @@
   PATH_SUFFIXES
     include
     include/openjpeg-1.5
+    include/openmj2-2.1
 )
 
 FIND_LIBRARY(OPENJPEG_LIBRARY
   NAMES
-    openjpeg
+    openmj2
   HINTS
     ${_openjpeg_SEARCH_DIRS}
   PATH_SUFFIXES
