--- avogadrolibs-1.98.1/avogadro/qtgui/CMakeLists.txt
+++ avogadrolibs-1.98.1/avogadro/qtgui/CMakeLists.txt
@@ -19,7 +19,7 @@
 endif()
 
 # Find python for input generator scripts:
-find_package(PythonInterp 3)
+find_package(Python3)
 
 file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/avogadropython.h.in"
 "namespace Avogadro {

--- avogadrolibs-1.98.1/python/CMakeLists.txt
+++ avogadrolibs-1.98.1/python/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.3 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.5...3.27 FATAL_ERROR)
 
 if (NOT TARGET Avogadro::Core)
   find_package(AvogadroLibs REQUIRED)
