diff -Naur librra-0.13/python/pyrra.pyx librra-0.13_patched/python/pyrra.pyx
--- librra-0.13/python/pyrra.pyx	2008-06-29 11:49:42.000000000 +0000
+++ librra-0.13_patched/python/pyrra.pyx	2011-04-01 15:22:24.152898428 +0000
@@ -162,7 +162,7 @@
 	cdef bool connected
 	cdef uint32_t ntypes
 	
-	def __new__(self):
+	def __cinit__(self):
 		self.instance = <RRA_SyncMgr *>rra_syncmgr_new()
 		if not self.instance:
 			raise RRAError(-1)
