diff -Naur xbmc-17.1-Krypton.orig/xbmc/filesystem/SMBDirectory.cpp xbmc-17.1-Krypton/xbmc/filesystem/SMBDirectory.cpp
--- xbmc-17.1-Krypton.orig/xbmc/filesystem/SMBDirectory.cpp	2017-03-20 16:17:49.000000000 +0000
+++ xbmc-17.1-Krypton/xbmc/filesystem/SMBDirectory.cpp	2017-03-20 23:01:52.444160749 +0000
@@ -45,7 +45,7 @@
 #include "linux/XTimeUtils.h"
 #endif
 
-#include <libsmbclient.h>
+#include <samba-4.0/libsmbclient.h>
 
 struct CachedDirEntry
 {
diff -Naur xbmc-17.1-Krypton.orig/xbmc/filesystem/SMBFile.cpp xbmc-17.1-Krypton/xbmc/filesystem/SMBFile.cpp
--- xbmc-17.1-Krypton.orig/xbmc/filesystem/SMBFile.cpp	2017-03-20 16:17:49.000000000 +0000
+++ xbmc-17.1-Krypton/xbmc/filesystem/SMBFile.cpp	2017-03-20 23:00:17.355154034 +0000
@@ -26,7 +26,7 @@
 #include "SMBFile.h"
 #include "PasswordManager.h"
 #include "SMBDirectory.h"
-#include <libsmbclient.h>
+#include <samba-4.0/libsmbclient.h>
 #include "settings/AdvancedSettings.h"
 #include "settings/Settings.h"
 #include "threads/SingleLock.h"
