diff -Naur xbmc-14.0-Helix.orig/xbmc/filesystem/SMBDirectory.cpp xbmc-14.0-Helix/xbmc/filesystem/SMBDirectory.cpp
--- xbmc-14.0-Helix.orig/xbmc/filesystem/SMBDirectory.cpp	2014-12-22 23:11:35.000000000 +0000
+++ xbmc-14.0-Helix/xbmc/filesystem/SMBDirectory.cpp	2014-12-24 03:32:09.227643266 +0000
@@ -42,7 +42,7 @@
 #include "threads/SingleLock.h"
 #include "PasswordManager.h"
 
-#include <libsmbclient.h>
+#include <samba-4.0/libsmbclient.h>
 
 #if defined(TARGET_DARWIN)
 #define XBMC_SMB_MOUNT_PATH "Library/Application Support/Kodi/Mounts/"
diff -Naur xbmc-14.0-Helix.orig/xbmc/filesystem/SMBFile.cpp xbmc-14.0-Helix/xbmc/filesystem/SMBFile.cpp
--- xbmc-14.0-Helix.orig/xbmc/filesystem/SMBFile.cpp	2014-12-22 23:11:35.000000000 +0000
+++ xbmc-14.0-Helix/xbmc/filesystem/SMBFile.cpp	2014-12-24 03:32:52.697913705 +0000
@@ -26,7 +26,7 @@
 #include "SMBFile.h"
 #include "PasswordManager.h"
 #include "SMBDirectory.h"
-#include <libsmbclient.h>
+#include <samba-4.0/libsmbclient.h>
 #include "settings/AdvancedSettings.h"
 #include "settings/Settings.h"
 #include "threads/SingleLock.h"
