diff -Naur xbmc-Gotham_alpha9.orig/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamRTMP.cpp xbmc-Gotham_alpha9/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamRTMP.cpp
--- xbmc-Gotham_alpha9.orig/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamRTMP.cpp	2013-10-31 16:48:45.000000000 +0000
+++ xbmc-Gotham_alpha9/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamRTMP.cpp	2013-11-28 05:47:36.418763917 +0000
@@ -21,9 +21,7 @@
 #if (defined HAVE_CONFIG_H) && (!defined TARGET_WINDOWS)
   #include "config.h"
 #endif
-#ifdef TARGET_WINDOWS
 #include "system.h" // just for HAS_LIBRTMP
-#endif
 
 #ifdef HAS_LIBRTMP
 #include "settings/AdvancedSettings.h"
