--- tightvncserver.conf.orig	2021-03-27 00:33:41.773056870 -0400
+++ tightvncserver.conf	2021-03-27 00:38:34.195076768 -0400
@@ -23,11 +23,11 @@
 # $vncUserDir = "$ENV{HOME}/.vnc";
 # $fontPath = "unix/:7100";
 # $authType = "-rfbauth $vncUserDir/passwd";
-# $colorPath = "/usr/lib/X11/rgb";
+$colorPath = "/usr/share/X11/rgb";
 
 ## Here is another example of setting the font path:
 #
-# $fontPath = "/usr/lib/X11/fonts/misc/"
+$fontPath = "/usr/share/fonts/misc/"
 # $fontPath = "$fontPath,/usr/lib/X11/fonts/75dpi/";
 
 ## You might wish to make your vnc directory under /tmp, to make sure
