diff -Nur audit-2.0.4.orig//init.d/auditd.init audit-2.0.4/init.d/auditd.init
--- audit-2.0.4.orig//init.d/auditd.init	2009-12-07 15:16:41.000000000 -0600
+++ audit-2.0.4/init.d/auditd.init	2010-06-13 02:07:13.368552889 -0500
@@ -9,7 +9,7 @@
 #              will be sent to syslog.
 #
 # processname: /sbin/auditd
-# config: /etc/sysconfig/auditd
+# config: /etc/rc.d/rc.auditd.conf
 # config: /etc/audit/auditd.conf
 # pidfile: /var/run/auditd.pid
 #
@@ -42,7 +42,7 @@
 test $EUID = 0  ||  exit 4
 
 # Check config
-test -f /etc/sysconfig/auditd && . /etc/sysconfig/auditd
+test -f /etc/rc.d/rc.auditd.conf && . /etc/rc.d/rc.auditd.conf
 
 RETVAL=0
 
