From 8176cc23549f60ddd326badf61f7e268ea1af897 Mon Sep 17 00:00:00 2001
From: Pierre Carrier <pierre@gcarrier.fr>
Date: Wed, 13 Feb 2013 03:43:50 +0100
Subject: [PATCH] src/afuse.c: avoid early exit in afuse_releasedir

Should solve #1.
---
 src/afuse.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/afuse.c b/src/afuse.c
index 37825ac..9c93fc6 100644
--- a/src/afuse.c
+++ b/src/afuse.c
@@ -931,6 +931,7 @@ static int afuse_releasedir(const char *path, struct fuse_file_info *fi)
 		if (dp)
 			closedir(dp);
 		retval = 0;
+		break;
 
 	default:
 		DEFAULT_CASE_INVALID_ENUM;
-- 
1.8.1.3

