--- gnuradio-3.7.10.1.orig/volk/CMakeLists.txt	2017-02-03 09:52:01.876198540 -0800
+++ gnuradio-3.7.10.1/volk/CMakeLists.txt	2017-02-03 09:48:17.651151087 -0800
@@ -27,7 +27,6 @@
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
-add_definitions(-D_GLIBCXX_USE_CXX11_ABI=1)
 
 set(CMAKE_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}) #allows this to be a sub-project
 set(CMAKE_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}) #allows this to be a sub-project
