https://bbs.archlinux.org/viewtopic.php?id=290483

diff --git a/src/addhelper/addhelper.c b/src/addhelper/addhelper.c
index 3bc8a6b..78242e9 100644
--- a/src/addhelper/addhelper.c
+++ b/src/addhelper/addhelper.c
@@ -24,6 +24,9 @@
  */
 
 #include <compiz-core.h>
+
+#include <stdlib.h>
+
 #include "addhelper_options.h"
 
 #define GET_ADD_DISPLAY(d)                            \
diff --git a/src/bench/bench.c b/src/bench/bench.c
index 9cdef8c..4610021 100644
--- a/src/bench/bench.c
+++ b/src/bench/bench.c
@@ -22,6 +22,8 @@
 
 #include <compiz-core.h>
 
+#include <stdlib.h>
+
 #include "bench_tex.h"
 #include "bench_options.h"
 
diff --git a/src/firepaint/firepaint.c b/src/firepaint/firepaint.c
index b147b03..cb8a508 100644
--- a/src/firepaint/firepaint.c
+++ b/src/firepaint/firepaint.c
@@ -21,6 +21,8 @@
 
 #include <compiz-core.h>
 
+#include <stdlib.h>
+
 #include "firepaint_options.h"
 #include "firepaint_tex.h"
 
diff --git a/src/grid/grid.c b/src/grid/grid.c
index a8d5152..cb827e6 100644
--- a/src/grid/grid.c
+++ b/src/grid/grid.c
@@ -23,7 +23,10 @@
  */
 
 #include <compiz-core.h>
+
+#include <stdlib.h>
 #include <string.h>
+
 #include "grid_options.h"
 
 #define GRID_DEBUG 0
diff --git a/src/highlightcontent/highlightcontent.c b/src/highlightcontent/highlightcontent.c
index 33bf932..5d66978 100644
--- a/src/highlightcontent/highlightcontent.c
+++ b/src/highlightcontent/highlightcontent.c
@@ -21,6 +21,7 @@
  */
 
 #include <math.h>
+#include <stdlib.h>
 #include <string.h>
 
 #include <compiz-core.h>
diff --git a/src/loginout/loginout.c b/src/loginout/loginout.c
index e2889d2..466f7df 100644
--- a/src/loginout/loginout.c
+++ b/src/loginout/loginout.c
@@ -21,6 +21,8 @@
 
 #include <compiz-core.h>
 
+#include <stdlib.h>
+
 #include "loginout_options.h"
 
 static int displayPrivateIndex = 0;
diff --git a/src/mblur/mblur.c b/src/mblur/mblur.c
index 103cd32..4e15c6d 100644
--- a/src/mblur/mblur.c
+++ b/src/mblur/mblur.c
@@ -20,6 +20,7 @@
  */
 
 #include <math.h>
+#include <stdlib.h>
 
 #include <compiz-core.h>
 
diff --git a/src/reflex/reflex.c b/src/reflex/reflex.c
index 8c6ebc3..648c9c9 100644
--- a/src/reflex/reflex.c
+++ b/src/reflex/reflex.c
@@ -20,6 +20,8 @@
  */
 #include <compiz-core.h>
 
+#include <stdlib.h>
+
 #include "reflex_options.h"
 
 
diff --git a/src/scalefilter/scalefilter.c b/src/scalefilter/scalefilter.c
index 55e2a96..c10c5d1 100644
--- a/src/scalefilter/scalefilter.c
+++ b/src/scalefilter/scalefilter.c
@@ -25,6 +25,7 @@
 
 #define _GNU_SOURCE
 #include <math.h>
+#include <stdlib.h>
 #include <string.h>
 #include <wchar.h>
 #include <locale.h>
diff --git a/src/shelf/shelf.c b/src/shelf/shelf.c
index cd17839..e4fd09c 100644
--- a/src/shelf/shelf.c
+++ b/src/shelf/shelf.c
@@ -33,6 +33,7 @@
 #include <X11/extensions/shape.h>
 #include <X11/cursorfont.h>
 #include <math.h>
+#include <stdlib.h>
 #include <string.h>
 #include "shelf_options.h"
 
diff --git a/src/showmouse/showmouse.c b/src/showmouse/showmouse.c
index fe972a2..32c3cb9 100644
--- a/src/showmouse/showmouse.c
+++ b/src/showmouse/showmouse.c
@@ -25,6 +25,7 @@
  */
 
 #include <math.h>
+#include <stdlib.h>
 #include <string.h>
 
 #include <compiz-core.h>
diff --git a/src/splash/splash.c b/src/splash/splash.c
index c72b9b0..986a3ec 100644
--- a/src/splash/splash.c
+++ b/src/splash/splash.c
@@ -20,6 +20,7 @@
  */
 
 #include <math.h>
+#include <stdlib.h>
 
 #include <compiz-core.h>
 #include <X11/Xatom.h>
diff --git a/src/trailfocus/trailfocus.c b/src/trailfocus/trailfocus.c
index ecfa84d..05c50f4 100644
--- a/src/trailfocus/trailfocus.c
+++ b/src/trailfocus/trailfocus.c
@@ -25,6 +25,7 @@
  *
  */
 
+#include <stdlib.h>
 #include <string.h>
 #include <compiz-core.h>
 #include "trailfocus_options.h"
diff --git a/src/vidcap/vidcap.c b/src/vidcap/vidcap.c
index 10157b2..a08126d 100644
--- a/src/vidcap/vidcap.c
+++ b/src/vidcap/vidcap.c
@@ -27,6 +27,7 @@
 
 #define _GNU_SOURCE /* For asprintf */
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
diff --git a/src/wallpaper/wallpaper.c b/src/wallpaper/wallpaper.c
index fd23b69..2c47fa7 100644
--- a/src/wallpaper/wallpaper.c
+++ b/src/wallpaper/wallpaper.c
@@ -23,6 +23,7 @@
  */
 
 #include <stdarg.h>
+#include <stdlib.h>
 #include <string.h>
 #include <math.h>
 #include <dirent.h>
diff --git a/src/widget/widget.c b/src/widget/widget.c
index ce7d4f0..6d146fa 100644
--- a/src/widget/widget.c
+++ b/src/widget/widget.c
@@ -26,6 +26,7 @@
  *
  */
 
+#include <stdlib.h>
 #include <string.h>
 #include <compiz-core.h>
 #include <X11/Xatom.h>
