Index: compiz-plugins-extra-0.8.8/src/notification/notification.c
===================================================================
--- compiz-plugins-extra-0.8.8.orig/src/notification/notification.c
+++ compiz-plugins-extra-0.8.8/src/notification/notification.c
@@ -104,9 +104,6 @@ notifyLogMessage (const char   *componen
     n = notify_notification_new (logLevel,
                                  message,
                                  iconUri
-#if !defined (HAVE_LIBNOTIFY_0_6_1)
-                                 , NULL
-#endif
                                  );
 
     notify_notification_set_timeout (n, nd->timeout);
