diff -uNr incron-0.5.9/Makefile incron-0.5.9_patched/Makefile
--- incron-0.5.9/Makefile	2009-06-21 23:27:24.000000000 +0300
+++ incron-0.5.9_patched/Makefile	2011-04-06 18:09:10.661890554 +0300
@@ -44,33 +44,35 @@
 distclean: clean
 
 install:	all install-man
-	[ -d $(PREFIX) ]
-	$(INSTALL) -m 04755 -o $(USER) incrontab $(PREFIX)/bin/
-	$(INSTALL) -m 0755 incrond $(PREFIX)/sbin/
-	$(INSTALL) -m 0755 -o $(USER) -d $(USERDATADIR)
-	$(INSTALL) -m 0755 -o $(USER) -d $(SYSDATADIR)
-	$(INSTALL) -m 0644 -o $(USER) incron.conf.example $(CFGDIR)
+	[ -d $(DESTDIR)$(PREFIX) ]
+	$(INSTALL) -m 0755 -d $(DESTDIR)$(PREFIX)/bin
+	$(INSTALL) -m 0755 -d $(DESTDIR)$(PREFIX)/sbin
+	$(INSTALL) -m 04755 -o $(USER) incrontab $(DESTDIR)$(PREFIX)/bin/
+	$(INSTALL) -m 0755 incrond $(DESTDIR)$(PREFIX)/sbin/
+	$(INSTALL) -m 0755 -o $(USER) -d $(DESTDIR)$(USERDATADIR)
+	$(INSTALL) -m 0755 -o $(USER) -d $(DESTDIR)$(SYSDATADIR)
+	$(INSTALL) -m 0644 -o $(USER) incron.conf.example $(DESTDIR)$(CFGDIR)
 
 install-man:	incrontab.1 incrontab.5 incrond.8 incron.conf.5
-	$(INSTALL) -m 0755 -d $(MANPATH)/man1
-	$(INSTALL) -m 0755 -d $(MANPATH)/man5
-	$(INSTALL) -m 0755 -d $(MANPATH)/man8
-	$(INSTALL) -m 0644 incrontab.1 $(MANPATH)/man1
-	$(INSTALL) -m 0644 incrontab.5 $(MANPATH)/man5
-	$(INSTALL) -m 0644 incrond.8 $(MANPATH)/man8
-	$(INSTALL) -m 0644 incron.conf.5 $(MANPATH)/man5
+	$(INSTALL) -m 0755 -d $(DESTDIR)$(MANPATH)/man1
+	$(INSTALL) -m 0755 -d $(DESTDIR)$(MANPATH)/man5
+	$(INSTALL) -m 0755 -d $(DESTDIR)$(MANPATH)/man8
+	$(INSTALL) -m 0644 incrontab.1 $(DESTDIR)$(MANPATH)/man1
+	$(INSTALL) -m 0644 incrontab.5 $(DESTDIR)$(MANPATH)/man5
+	$(INSTALL) -m 0644 incrond.8 $(DESTDIR)$(MANPATH)/man8
+	$(INSTALL) -m 0644 incron.conf.5 $(DESTDIR)$(MANPATH)/man5
 
 uninstall:	uninstall-man
-	[ -d $(PREFIX) ]
-	rm -f $(PREFIX)/bin/incrontab
-	rm -f $(PREFIX)/sbin/incrond
-	rm -f $(CFGDIR)/incron.conf.example
+	[ -d $(DESTDIR)$(PREFIX) ]
+	rm -f $(DESTDIR)$(PREFIX)/bin/incrontab
+	rm -f $(DESTDIR)$(PREFIX)/sbin/incrond
+	rm -f $(DESTDIR)$(CFGDIR)/incron.conf.example
 
 uninstall-man:
-	rm -f $(MANPATH)/man1/incrontab.1
-	rm -f $(MANPATH)/man5/incrontab.5
-	rm -f $(MANPATH)/man8/incrond.8
-	rm -f $(MANPATH)/man5/incron.conf.5
+	rm -f $(DESTDIR)$(MANPATH)/man1/incrontab.1
+	rm -f $(DESTDIR)$(MANPATH)/man5/incrontab.5
+	rm -f $(DESTDIR)$(MANPATH)/man8/incrond.8
+	rm -f $(DESTDIR)$(MANPATH)/man5/incron.conf.5
 
 update:		uninstall install
 
