diff -uNr incron-0.5.9/appargs.cpp incron-0.5.9_patched/appargs.cpp
--- incron-0.5.9/appargs.cpp	2009-06-21 23:27:24.000000000 +0300
+++ incron-0.5.9_patched/appargs.cpp	2011-04-06 15:35:32.565888675 +0300
@@ -22,6 +22,8 @@
 
 #include <cstring>
 
+#include <stdio.h>
+
 #include "strtok.h"
 
 #include "appargs.h"
diff -uNr incron-0.5.9/icd-main.cpp incron-0.5.9_patched/icd-main.cpp
--- incron-0.5.9/icd-main.cpp	2009-06-21 23:27:24.000000000 +0300
+++ incron-0.5.9_patched/icd-main.cpp	2011-04-06 15:32:20.007891134 +0300
@@ -23,6 +23,7 @@
 #include <errno.h>
 #include <sys/poll.h>
 #include <sys/stat.h>
+#include <stdio.h>
 #include <cstring>
 
 #include "inotify-cxx.h"
diff -uNr incron-0.5.9/incroncfg.cpp incron-0.5.9_patched/incroncfg.cpp
--- incron-0.5.9/incroncfg.cpp	2009-06-21 23:27:24.000000000 +0300
+++ incron-0.5.9_patched/incroncfg.cpp	2011-04-06 15:35:04.422890150 +0300
@@ -181,7 +181,7 @@
 
 bool IncronCfg::IsComment(const char* s)
 {
-  char* sx = strchr(s, '#');
+  const char* sx = strchr(s, '#');
   if (sx == NULL)
     return false;
     
diff -uNr incron-0.5.9/inotify-cxx.cpp incron-0.5.9_patched/inotify-cxx.cpp
--- incron-0.5.9/inotify-cxx.cpp	2009-06-21 23:27:24.000000000 +0300
+++ incron-0.5.9_patched/inotify-cxx.cpp	2011-04-06 15:32:39.878890586 +0300
@@ -22,6 +22,7 @@
 
 #include <errno.h>
 #include <unistd.h>
+#include <stdio.h>
 #include <fcntl.h>
 
 #include "inotify-cxx.h"
diff -uNr incron-0.5.9/usertable.cpp incron-0.5.9_patched/usertable.cpp
--- incron-0.5.9/usertable.cpp	2009-06-21 23:27:24.000000000 +0300
+++ incron-0.5.9_patched/usertable.cpp	2011-04-06 15:33:09.499888673 +0300
@@ -25,6 +25,7 @@
 #include <unistd.h>
 #include <grp.h>
 #include <stdlib.h>
+#include <stdio.h>
 #include <sys/stat.h>
 #include <cstring>
 
