diff -Naur odamex-src-12.3.0.orig/client/sdl/i_music.cpp odamex-src-12.3.0/client/sdl/i_music.cpp
--- odamex-src-12.3.0.orig/client/sdl/i_music.cpp	2026-08-05 23:55:14.000000000 -0400
+++ odamex-src-12.3.0/client/sdl/i_music.cpp	2026-09-29 01:49:32.130010270 -0400
@@ -206,13 +206,10 @@
 			break;
 		#endif	// PORTMIDI
 
-		case MS_LIBADLMIDI:
-			musicsystem = new AdlMidiMusicSystem();
-			break;
-
 		case MS_SDLMIXER:	// fall through
+		case MS_LIBADLMIDI:
 		default:
-			musicsystem = new SdlMixerMusicSystem();
+			musicsystem = new AdlMidiMusicSystem();
 			break;
 	}
 
diff -Naur odamex-src-12.3.0.orig/client/src/m_options.cpp odamex-src-12.3.0/client/src/m_options.cpp
--- odamex-src-12.3.0.orig/client/src/m_options.cpp	2026-08-05 23:55:14.000000000 -0400
+++ odamex-src-12.3.0/client/src/m_options.cpp	2026-09-29 01:49:58.729010169 -0400
@@ -570,7 +570,7 @@
 static value_t MusSys[] = {
 	{ MS_AUTO,		"Auto"},
 	#ifndef _WIN32
-	{ MS_SDLMIXER,	"SDL Mixer"},
+	{ MS_SDLMIXER,	"libADLMIDI (OPL3 FM)"},
 	#endif
 	{ MS_LIBADLMIDI,"libADLMIDI (OPL3 FM)"},
 	#ifdef OSX
