diff -Naur odamex-src-0.5.6/client/src/d_main.cpp odamex-src-0.5.6.patched//client/src/d_main.cpp
--- odamex-src-0.5.6/client/src/d_main.cpp	2011-11-05 00:00:05.000000000 -0400
+++ odamex-src-0.5.6.patched//client/src/d_main.cpp	2011-12-23 07:23:12.000000000 -0500
@@ -845,6 +845,7 @@
 	D_AddSearchDir(dirs, getenv("DOOMWADPATH"), separator);
 	D_AddSearchDir(dirs, getenv("HOME"), separator);
 	D_AddSearchDir(dirs, waddirs.cstring(), separator);
+	D_AddSearchDir(dirs, "/usr/share/games/doom", separator);
 
     dirs.erase(std::unique(dirs.begin(), dirs.end()), dirs.end());
 
diff -Naur odamex-src-0.5.6/server/src/d_main.cpp odamex-src-0.5.6.patched//server/src/d_main.cpp
--- odamex-src-0.5.6/server/src/d_main.cpp	2011-11-04 23:59:48.000000000 -0400
+++ odamex-src-0.5.6.patched//server/src/d_main.cpp	2011-12-23 07:22:56.000000000 -0500
@@ -629,6 +629,7 @@
 	D_AddSearchDir(dirs, getenv("DOOMWADPATH"), separator);
     D_AddSearchDir(dirs, getenv("HOME"), separator);
     D_AddSearchDir(dirs, waddirs.cstring(), separator);
+	D_AddSearchDir(dirs, "/usr/share/games/doom", separator);
 
 
 	dirs.erase(std::unique(dirs.begin(), dirs.end()), dirs.end());
