Index: cairo-dock-2.3.0~2/src/gldit/cairo-dock-keybinder.h
===================================================================
--- cairo-dock-2.3.0~2.orig/src/gldit/cairo-dock-keybinder.h
+++ cairo-dock-2.3.0~2/src/gldit/cairo-dock-keybinder.h
@@ -31,7 +31,7 @@
 #ifndef __CD_KEY_BINDER_H__
 #define __CD_KEY_BINDER_H__
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "cairo-dock-struct.h"
 #include "cairo-dock-manager.h"
 
