--- Makefile.in	2002-03-06 17:59:37.000000000 +0100
+++ Makefile.in.patched	2015-08-10 16:10:50.439007299 +0200
@@ -41,10 +41,12 @@
 	rm -f *.o
 
 install: $(PROGS)
-	$(INSTALL) -m 0755 aggregate $(prefix)/bin/
-	$(INSTALL) -m 0644 aggregate.1 $(prefix)/man/man1/
-	$(INSTALL) -m 0755 aggregate-ios $(prefix)/bin/
-	$(INSTALL) -m 0644 aggregate-ios.1 $(prefix)/man/man1/
+	$(INSTALL) -m 0755 -d $(DESTDIR)$(prefix)/bin/
+	$(INSTALL) -m 0755 -d $(DESTDIR)$(prefix)/man/man1/
+	$(INSTALL) -m 0755 aggregate $(DESTDIR)$(prefix)/bin/
+	$(INSTALL) -m 0644 aggregate.1 $(DESTDIR)$(prefix)/man/man1/
+	$(INSTALL) -m 0755 aggregate-ios $(DESTDIR)$(prefix)/bin/
+	$(INSTALL) -m 0644 aggregate-ios.1 $(DESTDIR)$(prefix)/man/man1/
 
 .c.o:
 	$(CC) -c $(CPPFLAGS) $(DEFS) $(CFLAGS) $<
