diff --git a/README b/README
index f89157f..ad8c529 100644
--- a/README
+++ b/README
@@ -112,9 +112,9 @@ Binary Calculator   Control-/
        like the best thing to do here.  Or else I'd be highlighting all
        the rest of the file.
      - Undo: for undo, if the modified byte is on the screen, it is reverted
-       to it's previous state.  If it is not on the screen, you are brought
+       to its previous state.  If it is not on the screen, you are brought
        to the byte's location so it is viewable on the screen.  The next
-       Undo command will revert the byte back to it's previous state.
+       Undo command will revert the byte back to its previous state.
      - Environment variable HEXEDIT can be used to save your command line
        options.  Example, export HEXEDIT=-8r would put hexedit into
        readonly mode with 8-bit printing on by default.
@@ -124,7 +124,7 @@ BUGS - DRAWBACKS
       - You currently can only edit files and disks up to 4.2 gigs.
       - On SunOS if you use control-v for page down you have to do it
         twice.  This has to do with the terminal driver interpreting
-        Control-v for it's purposes.  If there is a portable way to
+        Control-v for its purposes.  If there is a portable way to
         changes this let me know.
         *** I've been informed that this is a normal terminal driver thingy.
         Still, how do I get around it?
diff --git a/docs/hexedit.1 b/docs/hexedit.1
index 3f11291..dd66a7a 100644
--- a/docs/hexedit.1
+++ b/docs/hexedit.1
@@ -21,7 +21,7 @@ hexedit \- Full screen curses Hex editor
 .I Hexedit
 is an editor to edit binary (or any) files or disks.  The Display consists
 of the current offset from 0 (the first byte), the next sixteen bytes
-(aligned by bytes or 32-bit words), and it's ASCII or EBCDIC text
+(aligned by bytes or 32-bit words), and its ASCII or EBCDIC text
 representation.
 .SH OPTIONS
 .TP
diff --git a/docs/hexedit.texi b/docs/hexedit.texi
index 62eb729..8cb1d5a 100644
--- a/docs/hexedit.texinfo
+++ b/docs/hexedit.texinfo
@@ -65,7 +65,7 @@ by the Free Software Foundation.
 
 @strong{hexedit} is a Curses based Hex editor.
 Unlike a text editor, which is used for editing text documents in the desired language,
-@strong{hexedit} lets you edit any file as it's byte@footnote{I assume a byte is an 8-bit
+@strong{hexedit} lets you edit any file as its byte@footnote{I assume a byte is an 8-bit
 octet.  If this is not the case, I really recommend NOT using this program.}
 for byte representation.  It can even let you view and edit your fixed disks on your Linux
 system.  This is not ideal for writing a letter or writing c code, but
@@ -508,11 +508,11 @@ While editing a file, it is quite likely that you may make a mistake@footnote{Or
 To facilitate this, an extensive Undo command has been implemented on your behalf.
 @emph{Control-U} Undoes the previous change to the file.  The editor keeps track
 of all modifications, insertions, and deletions.  What this means is that at any time
-you can, if you wish, revert the file back to it's original state by a series of
+you can, if you wish, revert the file back to its original state by a series of
 Undo's.
 
 If you do @emph{Control-U} and the last changed byte is currently viewable
-in the Main Window, then the cursor will relocate to that byte and revert it to it's
+in the Main Window, then the cursor will relocate to that byte and revert it to its
 previous state.  If the last changed byte is not currently viewable, the currently
 viewable area will shift so that the changed byte is viewable.  In other words, you
 will be relocated to that part of the file, with the cursor on that byte at that 
