Index: build/common.gypi
===================================================================
--- build/common.gypi
+++ build/common.gypi
@@ -124,7 +124,7 @@
       'use_third_party_translations%': 0,
 
       # Remoting compilation is enabled by default. Set to 0 to disable.
-      'remoting%': 1,
+      'remoting%': 0,
 
       # If this is set, the clang plugins used on the buildbot will be used.
       # Run tools/clang/scripts/update.sh to make sure they are compiled.
