--- chromium-21.0.1168.0/chrome/browser/memory_details_linux.cc.orig        2012-09-25 10:48:48.208739739 +0700
+++ chromium-21.0.1168.0/chrome/browser/memory_details_linux.cc     2012-09-25 10:49:19.195357731 +0700
@@ -14,6 +14,7 @@
 #include "chrome/common/chrome_constants.h"
 #include "content/public/browser/browser_thread.h"
 #include "content/public/common/process_type.h"
+#include <unistd.h>
 
 using base::ProcessEntry;
 using content::BrowserThread;