diff -Naur dis6502-0.12.orig/initopts.c dis6502-0.12/initopts.c
--- dis6502-0.12.orig/initopts.c	2004-01-17 18:25:29.000000000 -0500
+++ dis6502-0.12/initopts.c	2006-11-24 18:56:32.000000000 -0500
@@ -30,7 +30,7 @@
                    "  options:  -a             assembly output\n"
                    "            -p <file>      predefs\n"
 	           "            -v <address>   alternate vector address\n"
-	   	   "            -7             mask character data to 7-bit",
+	   	   "            -7             mask character data to 7-bit\n",
 	   progname);
   exit (1);
 }
