diff -Naur a/src/bwish/CMakeLists.txt b/src/bwish/CMakeLists.txt
--- a/src/bwish/CMakeLists.txt	2013-06-14 01:55:27.000000000 -0400
+++ b/src/bwish/CMakeLists.txt	2013-11-21 20:57:03.386263490 -0500
@@ -14,6 +14,8 @@
 include_directories(
   ${CMAKE_SOURCE_DIR}/src/other/tcl/generic
   ${CMAKE_SOURCE_DIR}/src/other/tcl/unix
+  ${CMAKE_SOURCE_DIR}/src/other/tk/generic
+  ${CMAKE_SOURCE_DIR}/src/other/tk/unix
   ${CMAKE_SOURCE_DIR}/src/other/incrTcl/itcl/generic
   ${CMAKE_SOURCE_DIR}/src/other/incrTcl/itk/generic
   )
diff -Naur a/src/libtclcad/CMakeLists.txt b/src/libtclcad/CMakeLists.txt
--- a/src/libtclcad/CMakeLists.txt	2013-06-14 01:55:25.000000000 -0400
+++ b/src/libtclcad/CMakeLists.txt	2013-11-21 20:43:02.865252378 -0500
@@ -10,6 +10,8 @@
 set(TCLCAD_LOCAL_INCLUDE_DIRS
   ${CMAKE_SOURCE_DIR}/src/other/tcl/generic
   ${CMAKE_SOURCE_DIR}/src/other/tcl/unix
+  ${CMAKE_SOURCE_DIR}/src/other/tk/generic
+  ${CMAKE_SOURCE_DIR}/src/other/tk/unix
   ${CMAKE_SOURCE_DIR}/src/other/incrTcl/itcl/generic
   ${CMAKE_SOURCE_DIR}/src/other/incrTcl/itk/generic
   ${PNG_INCLUDE_DIR}
diff -Naur a/src/mged/CMakeLists.txt b/src/mged/CMakeLists.txt
--- a/src/mged/CMakeLists.txt	2013-06-14 01:55:49.000000000 -0400
+++ b/src/mged/CMakeLists.txt	2013-11-21 21:07:51.444272058 -0500
@@ -15,6 +15,8 @@
 include_directories(
   ${CMAKE_SOURCE_DIR}/src/other/tcl/generic
   ${CMAKE_SOURCE_DIR}/src/other/tcl/unix
+  ${CMAKE_SOURCE_DIR}/src/other/tk/generic
+  ${CMAKE_SOURCE_DIR}/src/other/tk/unix
   ${CMAKE_SOURCE_DIR}/src/other/incrTcl/itcl/generic
   ${CMAKE_SOURCE_DIR}/src/other/incrTcl/itk/generic
   )
