From 0891528c3ebbf59b0634d01b57499183bac1974b Mon Sep 17 00:00:00 2001
From: orbea <orbea@riseup.net>
Date: Mon, 23 Sep 2019 12:09:37 -0700
Subject: [PATCH] Fix build with gcc and an external fmt.

---
 src/libixion/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libixion/Makefile.am b/src/libixion/Makefile.am
index 1121e5f0..bb310b3f 100644
--- a/src/libixion/Makefile.am
+++ b/src/libixion/Makefile.am
@@ -3,6 +3,7 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/src/include \
 	-DIXION_BUILD \
+	-DFMT_HEADER_ONLY \
 	$(MDDS_CFLAGS) \
 	$(SPDLOG_CFLAGS) \
 	$(BOOST_CPPFLAGS)
-- 
2.22.0

