diff -up gwenhywfar-4.13.1/src/sio/syncio_tls.c.foo gwenhywfar-4.13.1/src/sio/syncio_tls.c
--- gwenhywfar-4.13.1/src/sio/syncio_tls.c.foo	2015-12-11 22:35:48.474181431 -0500
+++ gwenhywfar-4.13.1/src/sio/syncio_tls.c	2015-12-11 22:41:27.414733356 -0500
@@ -361,12 +361,12 @@ int GWEN_SyncIo_Tls_Prepare(GWEN_SYNCIO
 
   /* possibly force protocol priority */
   if (lflags & GWEN_SYNCIO_TLS_FLAGS_FORCE_SSL_V3) {
-    const int proto_prio[2] = { GNUTLS_SSL3, 0 };
+    char *force_sslv3_str = "NORMAL:-VERS-TLS-ALL:+VERS-SSL3.0";
 
     DBG_INFO(GWEN_LOGDOMAIN, "Forcing SSL v3");
-    rv=gnutls_protocol_set_priority(xio->session, proto_prio);
+    rv=gnutls_priority_set_direct(xio->session, force_sslv3_str, NULL);
     if (rv) {
-      DBG_ERROR(GWEN_LOGDOMAIN, "gnutls_protocol_set_priority: %d (%s)", rv, gnutls_strerror(rv));
+      DBG_ERROR(GWEN_LOGDOMAIN, "gnutls_priority_set_direct: %d (%s)", rv, gnutls_strerror(rv));
       gnutls_deinit(xio->session);
       return GWEN_ERROR_GENERIC;
     }
