diff -Nur vuze.orig/azureus vuze/azureus
--- vuze.orig/azureus	2011-05-02 20:04:44.738300650 -0400
+++ vuze/azureus	2011-05-02 20:07:23.191928931 -0400
@@ -3,11 +3,11 @@
 ######## CONFIGURATION OPTIONS ########
 SCRIPT_NOT_CHANGED=0	# change this to 1 if you don't want your script overwritten!
 JAVA_PROGRAM_DIR=""	# use full path to java bin dir, ex. "/usr/java/j2sdk1.4.2/bin/"
-#PROGRAM_DIR="/home/username/apps/azureus"	# use full path to Azureus bin dir
+PROGRAM_DIR="@LIBDIR@/vuze"    # use full path to Azureus bin dir
 JAVA_ARGS="-Xmx128m"
 
-#export MOZILLA_FIVE_HOME="/path/to/gre"	# Full path to GRE/Mozilla. When commenting out this line, also comment out the next line
-#if [ "$LD_LIBRARY_PATH x" = " x" ]; then export LD_LIBRARY_PATH=$MOZILLA_FIVE_HOME; else export LD_LIBRARY_PATH=$MOZILLA_FIVE_HOME:$LD_LIBRARY_PATH; fi
+export MOZILLA_FIVE_HOME="@LIBDIR@/seamonkey"  # Full path to GRE/Mozilla. When commenting out this line, also comment out the next line
+if [ "$LD_LIBRARY_PATH x" = " x" ]; then export LD_LIBRARY_PATH=$MOZILLA_FIVE_HOME; else export LD_LIBRARY_PATH=$MOZILLA_FIVE_HOME:$LD_LIBRARY_PATH; fi
 #######################################
 
 
diff -Nur vuze.orig/vuze vuze/vuze
--- vuze.orig/vuze	2011-05-02 20:04:44.739302770 -0400
+++ vuze/vuze	2011-05-02 20:08:29.022931571 -0400
@@ -3,11 +3,11 @@
 ######## CONFIGURATION OPTIONS ########
 SCRIPT_NOT_CHANGED=0	# change this to 1 if you don't want your script overwritten!
 JAVA_PROGRAM_DIR=""	# use full path to java bin dir, ex. "/usr/java/j2sdk1.4.2/bin/"
-#PROGRAM_DIR="/home/username/apps/azureus"	# use full path to Azureus bin dir
+PROGRAM_DIR="@LIBDIR@/vuze"    # use full path to Azureus bin dir
 JAVA_ARGS="-Xmx128m"
 
-#export MOZILLA_FIVE_HOME="/path/to/gre"	# Full path to GRE/Mozilla. When commenting out this line, also comment out the next line
-#if [ "$LD_LIBRARY_PATH x" = " x" ]; then export LD_LIBRARY_PATH=$MOZILLA_FIVE_HOME; else export LD_LIBRARY_PATH=$MOZILLA_FIVE_HOME:$LD_LIBRARY_PATH; fi
+export MOZILLA_FIVE_HOME="@LIBDIR@/seamonkey"  # Full path to GRE/Mozilla. When commenting out this line, also comment out the next line
+if [ "$LD_LIBRARY_PATH x" = " x" ]; then export LD_LIBRARY_PATH=$MOZILLA_FIVE_HOME; else export LD_LIBRARY_PATH=$MOZILLA_FIVE_HOME:$LD_LIBRARY_PATH; fi
 #######################################
 
 
