diff -Naur sunpinyin-2.0.3.orig/src/ime-core/ic_history.cpp sunpinyin-2.0.3/src/ime-core/ic_history.cpp
--- sunpinyin-2.0.3.orig/src/ime-core/ic_history.cpp	2011-02-18 04:39:52.000000000 +0100
+++ sunpinyin-2.0.3/src/ime-core/ic_history.cpp	2012-09-18 05:32:14.687001314 +0200
@@ -41,6 +41,7 @@
 
 #include <fcntl.h>
 #include <stdio.h>
+#include <unistd.h>
 #include <stdint.h>
 #include <cassert>
 #include <sys/stat.h>
diff -Naur sunpinyin-2.0.3.orig/src/lexicon/genPYT.cpp sunpinyin-2.0.3/src/lexicon/genPYT.cpp
--- sunpinyin-2.0.3.orig/src/lexicon/genPYT.cpp	2011-02-18 04:39:52.000000000 +0100
+++ sunpinyin-2.0.3/src/lexicon/genPYT.cpp	2012-09-18 05:29:24.570735820 +0200
@@ -37,6 +37,7 @@
 
 #include <locale.h>
 #include <stdlib.h>
+#include <unistd.h>
 
 #include "pytrie.h"
 #include "pytrie_gen.h"
