--- driftnet-0.1.6_p20090401/gif.c.old	2016-01-04 23:08:50.734142148 +0700
+++ driftnet-0.1.6_p20090401/gif.c	2016-01-04 23:11:58.759931300 +0700
@@ -20,7 +20,7 @@
  */
 int gif_load_hdr(img I) {
     GifFileType *g;
-    g = I->us = DGifOpenFileHandle(fileno(I->fp));
+    g = I->us = DGifOpenFileHandle(fileno(I->fp), NULL);
     if (!I->us) {
         I->err = IE_HDRFORMAT;
         return 0;
@@ -36,7 +36,7 @@
  * Abort loading a GIF file after the header is done.
  */
 int gif_abort_load(img I) {
-    DGifCloseFile((GifFileType*)I->us);
+    DGifCloseFile((GifFileType*)I->us, NULL);
     return 1;
 }
 
@@ -114,7 +114,7 @@
     ret = 1;
 fail:
 
-    DGifCloseFile(g);
+    DGifCloseFile(g, NULL);
     
     return ret;
 }
