Description: Fix FTBFS: package build loops on tmpfs
Forwarded: no
Author: Bart Martens <bartm@debian.org>
Last-Update: 2009-10-09
Index: gxmessage-2.20.0/po/Makefile.in.in
===================================================================
--- gxmessage-2.20.0.orig/po/Makefile.in.in	2012-02-25 02:13:49.000000000 +0000
+++ gxmessage-2.20.0/po/Makefile.in.in	2012-05-09 18:39:35.000000000 +0000
@@ -211,6 +211,7 @@
 	cd $(top_builddir) \
 	  && CONFIG_FILES=$(subdir)/Makefile.in CONFIG_HEADERS= CONFIG_LINKS= \
 	       $(SHELL) ./config.status
+	touch stamp-it
 
 # Tell versions [3.59,3.63) of GNU make not to export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
