From: Santiago Vila <sanvila@debian.org>
Subject: Modify doc/Makefile.in to avoid /usr/share/info/dir.gz.
Bug-Debian: https://bugs.debian.org/799715
--- a/doc/Makefile.in
+++ b/doc/Makefile.in
@@ -862,7 +862,7 @@
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(infodir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(infodir)" || exit $$?; done
 	@$(POST_INSTALL)
-	@if $(am__can_run_installinfo); then \
+	@if false; then \
 	  list='$(INFO_DEPS)'; test -n "$(infodir)" || list=; \
 	  for file in $$list; do \
 	    relfile=`echo "$$file" | sed 's|^.*/||'`; \
