diff -Nru nginx-0.7.65.orig//auto/lib/perl/conf nginx-0.7.65/auto/lib/perl/conf
--- nginx-0.7.65.orig//auto/lib/perl/conf	2006-12-24 16:23:11.000000000 -0200
+++ nginx-0.7.65/auto/lib/perl/conf	2010-06-07 00:42:23.398070172 -0300
@@ -47,7 +47,7 @@
     if test -n "$NGX_PERL_MODULES"; then
         have=NGX_PERL_MODULES value="(u_char *) \"$NGX_PERL_MODULES\""
         . auto/define
-        NGX_PERL_MODULES_MAN=$NGX_PERL_MODULES/man3
+        NGX_PERL_MODULES_MAN=/usr/man/man3
     fi
 
 else
