diff -Nur nginx-0.8.53.orig//auto/lib/perl/conf nginx-0.8.53//auto/lib/perl/conf
--- nginx-0.8.53.orig//auto/lib/perl/conf	2006-12-24 12:23:11.000000000 -0600
+++ nginx-0.8.53//auto/lib/perl/conf	2010-12-11 17:01:03.152535901 -0600
@@ -47,7 +47,7 @@
     if test -n "$NGX_PERL_MODULES"; then
         have=NGX_PERL_MODULES value="(u_char *) \"$NGX_PERL_MODULES\""
         . auto/define
-        NGX_PERL_MODULES_MAN=$NGX_PERL_MODULES/man3
+        NGX_PERL_MODULES_MAN=/usr/man/man3
     fi
 
 else
