--- Makefile.old	2019-07-29 07:34:04.356456054 +0200
+++ Makefile	2019-07-29 07:34:20.286630637 +0200
@@ -46,7 +46,7 @@
 #	-DPACKAGE_DATA_DIR=\"$(PACKAGE_DATA_DIR)\" $(FONT_USE_PIXMAP) $(MUSIC) $(DESIGNER)
 #LDFLAGS=-pg
 LINK=$(CC)
-LIBS=`$(SDL_CONFIG) --libs` -lSDL_image -lSDL_mixer $(FONT_USE_SDL_TTF)
+LIBS=`$(SDL_CONFIG) --libs` -lSDL_image -lSDL_mixer $(FONT_USE_SDL_TTF) -lm
 
 # You won't need to alter anything below
 all: $(SOURCES) $(TARGET)
