diff -Naur wolf4sdl-2.0+20210408_f31f41a/wl_main.c wolf4sdl-2.0+20210408_f31f41a.patched/wl_main.c
--- wolf4sdl-2.0+20210408_f31f41a/wl_main.c	2023-01-17 21:39:16.439986452 -0500
+++ wolf4sdl-2.0+20210408_f31f41a.patched/wl_main.c	2023-01-17 22:45:03.503411046 -0500
@@ -1608,6 +1608,10 @@
     bool hasError = false, showHelp = false;
     bool sampleRateGiven = false, audioBufferGiven = false;
     int i,defaultSampleRate = param_samplerate;
+	 char *exename = argv[0], *p;
+
+	 for (p = argv[0]; *p; p++)
+		 if (*p == '/') exename = p + 1;
 
     for(i = 1; i < argc; i++)
     {
@@ -1814,7 +1818,7 @@
             "Wolf4SDL v2.0\n"
             "Ported by Chaos-Software, additions by the community\n"
             "Original Wolfenstein 3D by id Software\n\n"
-            "Usage: Wolf4SDL [options]\n"
+				"Usage: %s [options]\n"
             "Options:\n"
             " --help                 This help page\n"
             " --tedlevel <level>     Starts the game in the given level\n"
@@ -1853,7 +1857,7 @@
             "                        (default: 0 -> .sod, 1-3 -> .sd*)\n"
             " --goodtimes            Disable copy protection quiz\n"
 #endif
-            , defaultSampleRate
+            , exename, defaultSampleRate
         );
         exit(1);
     }
