diff -Naur audtty-0.1.12.orig/Makefile.in audtty-0.1.12/Makefile.in
--- audtty-0.1.12.orig/Makefile.in	2010-02-28 19:10:48.000000000 +0100
+++ audtty-0.1.12/Makefile.in	2012-09-15 09:06:34.570080268 +0200
@@ -2,15 +2,14 @@
 #
 # A fork of xmms-curses
 
-DESTDIR=
 # autoconf is dumb.
-prefix=${DESTDIR}
+prefix=/usr
 exec_prefix=@prefix@
 bindir=@bindir@
-mandir=@prefix@/share/man/man1
+mandir=@prefix@/man/man1
 sysconfdir=@sysconfdir@
 
-AUDACIOUS_CFLAGS=@AUDACIOUS_CFLAGS@ -I/usr/include/dbus-1.0
+AUDACIOUS_CFLAGS=@AUDACIOUS_CFLAGS@ -I/usr/include/dbus-1.0 -I/usr/lib@LIBDIRSUFFIX@/glib-2.0/include -I/usr/include/glib-2.0
 
 SOURCES=main.c curses_printf.c playlist.c playlist_jump.c playlist_addurl.c settings.c connect.c browse.c playlist_create.c
 BINS=$(SOURCES:.c=.o)
@@ -20,18 +19,18 @@
 all: audtty
 
 audtty: ${BINS}
-	cc -g -O2 -g2 -Wall -Werror -lncursesw -laudclient ${LDFLAGS} -o audtty $(BINS)
+	$(CC) @SLKCFLAGS@ -Wall ${LDFLAGS} -lncursesw -laudclient -lglib-2.0 -ldbus-glib-1 -lgobject-2.0 -o audtty $(BINS)
 
 .c.o:
-	cc -g -Wall ${AUDACIOUS_CFLAGS} ${CFLAGS} -o $@ -c $<
+	$(CC) -g -Wall ${AUDACIOUS_CFLAGS} ${CFLAGS} -o $@ -c $<
 
 install: audtty
-	mkdir -p ${bindir}
-	install -m 0755 audtty ${bindir}/audtty
-	mkdir -p ${mandir}
-	install -m 0644 audtty.1 ${mandir}/audtty.1
-	mkdir -p ${sysconfdir}
-	install -m 0644 audtty.conf ${sysconfdir}/audtty.conf
+	mkdir -p $(DESTDIR)${bindir}
+	install -m 0755 audtty $(DESTDIR)${bindir}/audtty
+	mkdir -p $(DESTDIR)${mandir}
+	install -m 0644 audtty.1 $(DESTDIR)${mandir}/audtty.1
+	mkdir -p $(DESTDIR)${sysconfdir}
+	install -m 0644 audtty.conf $(DESTDIR)${sysconfdir}/audtty.conf
 
 uninstall:
 	rm ${bindir}/audtty || false
