diff -Nur chirp-0.4.0.orig/share/chirp.desktop chirp-0.4.0/share/chirp.desktop
--- chirp-0.4.0.orig/share/chirp.desktop	2012-02-15 13:46:54.000000000 -0600
+++ chirp-0.4.0/share/chirp.desktop	2014-05-18 22:39:37.171092107 -0500
@@ -1,13 +1,11 @@
 [Desktop Entry]
-Version=0.1
-Encoding=UTF-8
+Version=1.0
 Type=Application
 Exec=chirpw
-Icon=chirp.png
+Icon=chirp
 StartupNotify=true
 Terminal=false
-Categories=Application;HamRadio
-MimeType=inode/directory
+Categories=System;HamRadio;
+MimeType=inode/directory;
 Name=CHIRP
 Comment=CHIRP Radio Programming Tool
-GenericName=CHIRP Radio Programming Tool
