--- Makefile	2003-08-27 22:44:07.000000000 -0300
+++ Makefile.new	2009-02-14 23:00:24.000000000 -0300
@@ -21,7 +21,7 @@ PREFIX = /usr
 # with that, you don't even need to uncomment the next line.
 # CDROM = /dev/cdrom
 # Or, if you use devfs, uncomment this line:
-CDROM = /dev/cdroms/cdrom0
+# CDROM = /dev/cdroms/cdrom0
 
 
 
@@ -51,8 +51,6 @@ NETWORK = 1
 
 ifdef DEBUG
 EXTRA_CFLAGS = -DDEBUG=${DEBUG} -Wall -pedantic -g
-else
-EXTRA_CFLAGS = -O2 # -march=k6 # the arch optimization is for dave!
 endif
 
 ifdef CDROM
