diff -Naur gst-plugins-bad-1.16.2.orig/ext/vulkan/vkerror.c gst-plugins-bad-1.16.2/ext/vulkan/vkerror.c
--- gst-plugins-bad-1.16.2.orig/ext/vulkan/vkerror.c	2019-04-19 11:16:36.000000000 +0200
+++ gst-plugins-bad-1.16.2/ext/vulkan/vkerror.c	2020-06-21 18:46:08.911292000 +0200
@@ -63,8 +63,6 @@
 
   if (result >= 0)
     return NULL;
-  if (result < VK_RESULT_BEGIN_RANGE)
-    return "Unknown Error";
 
   for (i = 0; i < G_N_ELEMENTS (vk_result_string_map); i++) {
     if (result == vk_result_string_map[i].result)
