Description: do not use MAKEFLAGS
Author: Thorsten Alteholz <debian@alteholz.de>
Last-Update: 2011-07-03
Index: vcftools-0.1.7/Makefile
===================================================================
--- vcftools_0.1.7.org/Makefile	2011-07-04 15:22:33.000000000 +0200
+++ vcftools_0.1.7/Makefile	2011-07-04 15:22:56.000000000 +0200
@@ -22,7 +22,7 @@
 DIRS = cpp perl
 install:
 	    @mkdir -p $(BINDIR); mkdir -p $(MODDIR); \
-        for dir in $(DIRS); do cd $$dir && $(MAKE) $(MAKEFLAGS) && cd ..; done
+        for dir in $(DIRS); do cd $$dir && $(MAKE) && cd ..; done
 
 clean:
 		@for dir in $(DIRS); do cd $$dir && $(MAKE) clean && cd ..; done
