Description: do not use MAKEFLAGS and not MANDIR
Author: Thorsten Alteholz <debian@alteholz.de>
Last-Update: 2014-04-03
Index: vcftools_0.1.12/Makefile
===================================================================
--- vcftools_0.1.12.orig/Makefile	2014-04-04 10:37:59.000000000 +0200
+++ vcftools_0.1.12/Makefile	2014-04-04 10:38:34.000000000 +0200
@@ -25,9 +25,8 @@
 
 DIRS = cpp perl
 install:
-	    @mkdir -p $(BINDIR); mkdir -p $(MODDIR); mkdir -p $(MANDIR); \
-	    cp ${PREFIX}/cpp/vcftools.1.gz $(MANDIR); \
-        for dir in $(DIRS); do cd $$dir && $(MAKE) $(MAKEFLAGS) && cd ..; done
+	    @mkdir -p $(BINDIR); mkdir -p $(MODDIR); \
+        for dir in $(DIRS); do cd $$dir && $(MAKE) && cd ..; done
 
 clean:
 		@for dir in $(DIRS); do cd $$dir && $(MAKE) clean && cd ..; done
