Author: Andreas Tille <tille@debian.org>
Date: Sat, 12 May 2012 09:31:58 +0200
Description: Enable propagation of hardening flags

Index: vcftools_0.1.11/cpp/Makefile
===================================================================
--- vcftools_0.1.11.orig/cpp/Makefile	2013-06-13 16:40:54.000000000 +0200
+++ vcftools_0.1.11/cpp/Makefile	2013-06-27 19:54:45.000000000 +0200
@@ -12,9 +12,9 @@
 	VCFTOOLS_PCA = 0
 endif
 # Compiler flags
-CFLAGS = -O2 -m64
+CFLAGS += -O2
 #CFLAGS = -Wall -O2 -pg -m64
-CPPFLAGS = -O2 -D_FILE_OFFSET_BITS=64 
+CPPFLAGS += -O2 -D_FILE_OFFSET_BITS=64
 #CPPFLAGS = -O2 -Wall -pg -D_FILE_OFFSET_BITS=64
 # Included libraries (zlib)
 LIB = -lz 
@@ -38,7 +38,7 @@
 endif
 
 vcftools: $(OBJS)
-	$(CPP) $(CPPFLAGS) $(OBJS) -o vcftools $(LIB)
+	$(CPP) $(CPPFLAGS) $(OBJS) $(LDFLAGS) -o vcftools $(LIB)
 ifdef BINDIR
 	cp $(CURDIR)/$@ $(BINDIR)/$@
 endif
@@ -50,8 +50,8 @@
 -include $(OBJS:.o=.d)
 
 %.o: %.cpp
-	$(CPP) -c $(CPPFLAGS) $*.cpp -o $*.o
-	$(CPP) -MM $(CPPFLAGS) $*.cpp > $*.d
+	$(CPP) -c $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) $*.cpp -o $*.o
+	$(CPP) -MM $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) $*.cpp > $*.d
 
 # remove compilation products
 clean:
