diff -Naur ircii-20240918/source/edit.c ircii-20240918.patched/source/edit.c
--- ircii-20240918/source/edit.c	2024-09-20 01:30:10.000000000 -0400
+++ ircii-20240918.patched/source/edit.c	2024-11-26 00:12:58.858074498 -0500
@@ -1330,7 +1330,7 @@
 				}
 				else
 				{
-					say("WHO -HOSTS: missing arguement");
+					say("WHO -HOSTS: missing argument");
 					new_free(&cmd);
 					return;
 				}
@@ -1349,7 +1349,7 @@
 				}
 				else
 				{
-					say("WHO -SERVERS: missing arguement");
+					say("WHO -SERVERS: missing argument");
 					new_free(&cmd);
 					return;
 				}
@@ -1364,7 +1364,7 @@
 				}
 				else
 				{
-					say("WHO -NAME: missing arguement");
+					say("WHO -NAME: missing argument");
 					new_free(&cmd);
 					return;
 				}
@@ -1379,7 +1379,7 @@
 				}
 				else
 				{
-					say("WHO -REALNAME: missing arguement");
+					say("WHO -REALNAME: missing argument");
 					new_free(&cmd);
 					return;
 				}
@@ -1394,7 +1394,7 @@
 				}
 				else
 				{
-					say("WHO -NICK: missing arguement");
+					say("WHO -NICK: missing argument");
 					new_free(&cmd);
 					return;
 				}
@@ -1409,7 +1409,7 @@
 				}
 				else
 				{
-					say("WHO -FILE: missing arguement");
+					say("WHO -FILE: missing argument");
 					new_free(&cmd);
 					return;
 				}
@@ -1553,7 +1553,7 @@
 		{
 			if (!(nick = recv_nick))
 			{
-				say("You have not recieved a message from \
+				say("You have not received a message from \
 						anyone yet");
 				goto out;
 			}
@@ -3593,7 +3593,7 @@
 
 /*
  * timercmd: the bit that handles the TIMER command.  If there are no
- * arguements, then just list the currently pending timers, if we are
+ * arguments, then just list the currently pending timers, if we are
  * give a -DELETE flag, attempt to delete the timer from the list.  Else
  * consider it to be a timer to add, and add it.
  */
@@ -3842,7 +3842,7 @@
 }
 
 /*
- * inputcmd:  the INPUT command.   Takes a couple of arguements...
+ * inputcmd:  the INPUT command.   Takes a couple of arguments...
  * the first surrounded in double quotes, and the rest makes up
  * a normal ircII command.  The command is evalutated, with $*
  * being the line that you input.  Used add_wait_prompt() to prompt
diff -Naur ircii-20240918/source/keys.c ircii-20240918.patched/source/keys.c
--- ircii-20240918/source/keys.c	2024-09-20 01:30:10.000000000 -0400
+++ ircii-20240918.patched/source/keys.c	2024-11-26 00:13:22.274072294 -0500
@@ -1682,7 +1682,7 @@
 			key_names[i].func(0, args);
 			break;
 		default:
-			say("Ambigious function %s", arg);
+			say("Ambiguous function %s", arg);
 			break;
 		}
 	}
@@ -1803,7 +1803,7 @@
 			break;
 
 		default:
-			say("Ambigious function %s", arg);
+			say("Ambiguous function %s", arg);
 			return;
 		}
 
