commit 4a037b06e349a3ec0c6ad5e1dcf0814c4b44e6de
Author: Zhu Qun-Ying <zhu.qunying@gmail.com>
Date:   Thu Dec 5 21:09:56 2024 -0800

    fix compile error

diff --git a/core/src/gnatcoll-json.adb b/core/src/gnatcoll-json.adb
index 64b4217c..e4753312 100644
--- a/core/src/gnatcoll-json.adb
+++ b/core/src/gnatcoll-json.adb
@@ -666,7 +666,7 @@ package body GNATCOLL.JSON is
 
       Deallocate (Current_Key);
       Free (Read_States);
-      return (Success => True, Value => Result);
+      return R_Result : constant Read_Result := (Success => True, Value => Result);
    exception
       when E : Invalid_JSON_Stream =>
          Free (Read_States);
@@ -675,7 +675,7 @@ package body GNATCOLL.JSON is
          begin
             Data.Current_Text_Position (Line, Column);
 
-            return (Success => False,
+            return R_Result : constant Read_Result:= (Success => False,
                     Error => (Line, Column,
                     To_Unbounded_String
                        (Ada.Exceptions.Exception_Message (E))));
