diff -Naur nikto-2.1.2.orig/nikto.pl nikto-2.1.2/nikto.pl
--- nikto-2.1.2.orig/nikto.pl	2010-07-12 02:27:03.000000000 +0200
+++ nikto-2.1.2/nikto.pl	2010-07-12 16:54:23.081240167 +0200
@@ -6,7 +6,7 @@
 use Getopt::Long;
 Getopt::Long::Configure('no_ignore_case');
 
-# use LW2;                   ### Change this line to use a different installed version
+use LW2;                   ### Change this line to use a different installed version
 
 ###############################################################################
 #                               Nikto                                         #
@@ -89,7 +89,7 @@
 require "$NIKTOCONFIG{'PLUGINDIR'}/nikto_core.plugin";
 nprint("T:$starttime: Starting", "d");
 require "$NIKTOCONFIG{'PLUGINDIR'}/nikto_single.plugin";
-require "$NIKTOCONFIG{'PLUGINDIR'}/LW2.pm";
+#require "$NIKTOCONFIG{'PLUGINDIR'}/LW2.pm";
 
 my ($a, $b) = split(/\./, $LW2::VERSION);
 die("- You must use LW2 2.4 or later\n") if ($a != 2 || $b < 4);
