diff -Naur mediaextract-v1.1.1/source/src/riff.c mediaextract-v1.1.1.patched/source/src/riff.c
--- mediaextract-v1.1.1/source/src/riff.c	2021-06-25 10:54:31.000000000 -0400
+++ mediaextract-v1.1.1.patched/source/src/riff.c	2022-03-19 13:19:13.506696806 -0400
@@ -217,28 +217,7 @@
 	CHUNK('V','P','8','X',    0)
 );
 
-#define MAX_CHUNK_SPEC_COUNT \
-	MAX(CHUNK_SPEC_COUNT(riff_empty_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_wav_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_avi_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_ani_fram_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_ani_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_pal_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_aud_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_dmbd_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_dmpr_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_dmcn_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_dsbc_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_dsfx_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_dmsc_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_sgt_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_sty_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_dmtl_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_dmtg_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_dmtk_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_dmbt_body), \
-	MAX(CHUNK_SPEC_COUNT(riff_dmpt_body), \
-	    CHUNK_SPEC_COUNT(riff_webp_body)))))))))))))))))))))
+#define MAX_CHUNK_SPEC_COUNT 9
 
 static const struct riff_file_spec riff_file_specs[] = {
 	{ RIFF('W','A','V','E', riff_wav_body ), "wav"  },
