diff -Naur tt-15.1.orig/tt.c tt-15.1/tt.c
--- tt-15.1.orig/tt.c	2006-06-01 09:36:53.000000000 +0200
+++ tt-15.1/tt.c	2013-01-19 16:05:27.358173158 +0100
@@ -43,8 +43,8 @@
 
 /*-------------------------------------------------------------------------*/
 
-char *LOCK_FILE = "/home/mike/lib/ttlock/ttlock";
-char *SCORE_FILE = "/home/mike/lib/ttscores";
+char *LOCK_FILE = "ttlock";
+char *SCORE_FILE = "ttscores";
 
 /* These four variables determine which game we're playing */
 struct piece *pieces = pieces4;
