diff -up aubio-0.4.9/src/io/source_avcodec.c.omv~ aubio-0.4.9/src/io/source_avcodec.c
--- aubio-0.4.9/src/io/source_avcodec.c.omv~	2022-01-16 03:12:59.164776223 +0100
+++ aubio-0.4.9/src/io/source_avcodec.c	2022-01-16 03:13:44.000066377 +0100
@@ -68,6 +68,10 @@
 #define AUBIO_AVCODEC_MAX_BUFFER_SIZE AV_INPUT_BUFFER_MIN_SIZE
 #endif
 
+#if LIBAVCODEC_VERSION_MAJOR >= 59
+#define FF_API_LAVF_AVCTX 1
+#endif
+
 struct _aubio_source_avcodec_t {
   uint_t hop_size;
   uint_t samplerate;
