--- nFlood-0.5.0/Makefile.original	2014-04-26 06:36:09.348650457 +0300
+++ nFlood-0.5.0/Makefile	2014-04-26 06:54:41.880741892 +0300
@@ -46,11 +46,11 @@
 
 EXEC_PREFIX = $(PREFIX)
 DATAROOTDIR = $(PREFIX)/share
-MANROOT     = $(DATAROOTDIR)/man
+MANROOT     = $(PREFIX)/man
 
 MANNUMBER   = 6
 
-BINDIR      = $(EXEC_PREFIX)/games
+BINDIR      = $(EXEC_PREFIX)/bin
 MANDIR      = $(MANROOT)/man$(MANNUMBER)
 
 MANFILE     = $(PACKAGE).$(MANNUMBER)
@@ -104,10 +104,10 @@
 	# Installing...
 	$(MUTE)install -d -m 755 $(BINDIR)
 	$(MUTE)install -m 755 bin/$(EXE) $(BINDIR)
-	-$(MUTE)cat $(MANPAGE) | sed -e "s|DATE|$(DATE)|g" -e "s|VERSION|$(VERSION)|g" >$(MANFILE)
-	$(MUTE)install -d $(MANDIR)
-	$(MUTE)install $(MANFILE) $(MANDIR)
-	$(MUTE)rm -f $(MANFILE)
+	# -$(MUTE)cat $(MANPAGE) | sed -e "s|DATE|$(DATE)|g" -e "s|VERSION|$(VERSION)|g" >$(MANFILE)
+	# $(MUTE)install -d $(MANDIR)
+	# $(MUTE)install $(MANFILE) $(MANDIR)
+	# $(MUTE)rm -f $(MANFILE)
 	# $(PACKAGE) successfuly installed!
 
 uninstall:
