--- /home/glitch/builds/munin-node/munin-2.0.25/Makefile	2014-11-24 16:46:24.000000000 -0500
+++ Makefile	2015-07-16 14:32:54.688903959 -0400
@@ -30,6 +30,7 @@
 MAN8		 := master/_bin/munin-update master/_bin/munin-limits master/_bin/munin-html master/_bin/munin-graph
 PODMAN8          := build/master/doc/munin-cron master/doc/munin master/doc/munin-check
 PODMAN5          := build/master/doc/munin.conf node/doc/munin-node.conf
+MAN3EXT		 := $(shell $(PERL) -e 'use Config; print $$Config{man3ext};')
 
 .PHONY: install install-pre install-master-prime install-node-prime install-node-pre install-common-prime install-doc install-man \
         build build-common-prime build-common-pre build-doc \
@@ -284,7 +285,7 @@
              -e 's|\@\@GOODSH\@\@|$(GOODSH)|g;'                      \
              -e 's|\@\@BASH\@\@|$(BASH)|g;'                          \
              -e 's|\@\@HASSETR\@\@|$(HASSETR)|g;'                    \
-             ./master/blib/libdoc/Munin::Master::HTMLOld.3pm    \
+             ./master/blib/libdoc/Munin::Master::HTMLOld.$(MAN3EXT)  \
              ./master/blib/lib/Munin/Master/HTMLOld.pm          \
              ./node/blib/sbin/munin-node-configure              \
              ./node/blib/sbin/munin-node                        \
