diff -Naur imwheel-1.0.0pre12-orig/imwheel.c imwheel-1.0.0pre12/imwheel.c
--- imwheel-1.0.0pre12-orig/imwheel.c	2004-08-29 11:22:03.000000000 +0100
+++ imwheel-1.0.0pre12/imwheel.c	2010-02-26 14:36:08.000000000 +0000
@@ -586,6 +586,8 @@
 				}
 				history[HISTORY_LENGTH-1].motion=button;
 				history[HISTORY_LENGTH-1].t     =time(NULL);
+
+				// patching this out
 				// Configure if in root and toggling wheel
 				if(!pointer_window)
 				{
@@ -595,6 +597,7 @@
 						if(strcmp(wap->id,"\\(root\\)"))
 							continue; //no root action defined!
 					}
+/*
 					else
 					{
 						for(j=1,i=0;j&&i<HISTORY_LENGTH;i++)
@@ -608,6 +611,7 @@
 							Printf("No config...\n");
 						continue; // No wheel actions needed in root window!
 					}
+*/
 				}
 				else
 					wap=findWA(d,button,wname,xch.res_name,xch.res_class,xmk,km);
