--- qtwitter-app/x11/qtwitter.desktop.orig	2012-08-07 15:34:04.781987614 +0300
+++ qtwitter-app/x11/qtwitter.desktop	2012-08-07 15:34:21.032000002 +0300
@@ -3,7 +3,7 @@
 Name=qTwitter
 Icon=qtwitter
 Type=Application
-Categories=Qt;Network;Instant Messaging;
+Categories=Qt;Network;InstantMessaging;
 Comment=A microblogging client
 Comment[ca]=El client de Twitter i Identi.ca
 Comment[de]=Ein Twitter und Identi.ca Kunde
