--- qtwitter-app/qtwitter-app.pro.orig	2012-08-07 15:32:14.975000266 +0300
+++ qtwitter-app/qtwitter-app.pro	2012-08-07 15:32:43.860999601 +0300
@@ -82,7 +82,8 @@
 }
 else:unix {
     LIBS += -L$${DESTDIR} \
-        $$TWITTERAPI_LIB
+        $$TWITTERAPI_LIB \
+        -lX11
     target.path = $${INSTALL_PREFIX}/bin
     PLUGINS_DIR = $${INSTALL_PREFIX}/lib$${LIB_SUFFIX}/$${TARGET}/plugins
     DEFINES += PLUGINS_DIR='\\\"$${PLUGINS_DIR}\\\"'
