--- glob2-0.9.4.4/libgag/src/TextStream.original.cpp	2014-04-06 22:37:28.204008827 +0300
+++ glob2-0.9.4.4/libgag/src/TextStream.cpp	2014-04-06 22:37:59.509617453 +0300
@@ -17,11 +17,11 @@
   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 */
 
-#include <TextStream.h>
 #include <assert.h>
 #include <fstream>
 #include <iostream>
 #include <stack>
+#include <TextStream.h>
 #ifdef WIN32
 #define snprintf _snprintf
 #define vsnprintf _vsnprintf
--- glob2-0.9.4.4/SConstruct.original	2014-04-09 06:25:12.710804438 +0300
+++ glob2-0.9.4.4/SConstruct	2014-04-09 06:47:08.010360930 +0300
@@ -109,6 +109,14 @@
         else:
             print "Could not find libz or zlib1.dll"
             missing.append("zlib")
+    
+    boost_system = ''
+    if conf.CheckLib("boost_system"):
+        boost_system="boost_system"
+    else:
+        print "Could not find libboost_system"
+        missing.append("libboost_system")
+    env.Append(LIBS=[boost_system])
 
     boost_thread = ''
     if conf.CheckLib("boost_thread") and conf.CheckCXXHeader("boost/thread/thread.hpp"):
Index: glob2-0.9.4.4/src/Game.h
===================================================================
--- glob2-0.9.4.4.orig/src/Game.h	2014-07-14 10:36:36.705074927 +0800
+++ glob2-0.9.4.4/src/Game.h	2014-07-14 10:38:28.733077827 +0800
@@ -148,7 +148,8 @@
 		TOP_TO_BOTTOM,
 		BOTTOM_TO_TOP
 	};
-	
+
+public:
 	struct BuildProject
 	{
 		int posX;

