diff -u -Nur texlive-20091107-source/libs/gd/gd-2.0.35/gd_png.c texlive-20091107-source.fixed/libs/gd/gd-2.0.35/gd_png.c
--- texlive-20091107-source/libs/gd/gd-2.0.35/gd_png.c	2009-03-20 10:58:16.000000000 +0300
+++ texlive-20091107-source.fixed/libs/gd/gd-2.0.35/gd_png.c	2010-03-05 10:17:14.350872597 +0300
@@ -149,7 +149,7 @@
 	return NULL;
   }
 
-	if (!png_check_sig (sig, 8)) { /* bad signature */
+	if (!(png_sig_cmp(sig, 0, 8) == 0)) { /* bad signature */
     	return NULL;		/* bad signature */
 	}
 
diff -u -Nur texlive-20091107-source/texk/dvipdfmx/src/pngimage.c texlive-20091107-source.fixed/texk/dvipdfmx/src/pngimage.c
--- texlive-20091107-source/texk/dvipdfmx/src/pngimage.c	2009-05-26 03:12:24.000000000 +0400
+++ texlive-20091107-source.fixed/texk/dvipdfmx/src/pngimage.c	2010-03-05 10:12:39.976897269 +0300
@@ -148,7 +148,7 @@
   rewind (png_file);
   if (fread (sigbytes, 1, sizeof(sigbytes), png_file) !=
       sizeof(sigbytes) ||
-      (!png_check_sig (sigbytes, sizeof(sigbytes))))
+      (!(png_sig_cmp(sigbytes, 0, sizeof(sigbytes)) == 0)))
     return 0;
   else
     return 1;
diff -u -Nur texlive-20091107-source/texk/web2c/xetexdir/pngimage.c texlive-20091107-source.fixed/texk/web2c/xetexdir/pngimage.c
--- texlive-20091107-source/texk/web2c/xetexdir/pngimage.c	2006-07-26 20:19:45.000000000 +0400
+++ texlive-20091107-source.fixed/texk/web2c/xetexdir/pngimage.c	2010-03-05 10:15:45.010872612 +0300
@@ -101,7 +101,7 @@
   rewind (png_file);
   if (fread (sigbytes, 1, sizeof(sigbytes), png_file) !=
       sizeof(sigbytes) ||
-      (!png_check_sig (sigbytes, sizeof(sigbytes))))
+      (!(png_sig_cmp(sigbytes, 0, sizeof(sigbytes)) == 0)))
     return 0;
   else
     return 1;
diff -u -Nur texlive-20091107-source/texk/xdvipdfmx/src/pngimage.c texlive-20091107-source.fixed/texk/xdvipdfmx/src/pngimage.c
--- texlive-20091107-source/texk/xdvipdfmx/src/pngimage.c	2007-11-22 19:48:06.000000000 +0300
+++ texlive-20091107-source.fixed/texk/xdvipdfmx/src/pngimage.c	2010-03-05 10:37:12.920343277 +0300
@@ -146,7 +146,7 @@
   rewind (png_file);
   if (fread (sigbytes, 1, sizeof(sigbytes), png_file) !=
       sizeof(sigbytes) ||
-      (!png_check_sig (sigbytes, sizeof(sigbytes))))
+      (!(png_sig_cmp(sigbytes, 0, sizeof(sigbytes)) == 0)))
     return 0;
   else
     return 1;
diff -u -Nur texlive-20091107-source/texk/xdvipdfmx/src/xbb.c texlive-20091107-source.fixed/texk/xdvipdfmx/src/xbb.c
--- texlive-20091107-source/texk/xdvipdfmx/src/xbb.c	2008-05-25 19:00:35.000000000 +0400
+++ texlive-20091107-source.fixed/texk/xdvipdfmx/src/xbb.c	2010-03-05 10:15:23.713902927 +0300
@@ -246,7 +246,7 @@
   unsigned char sigbytes[4];
   rewind (png_file);
   if (fread(sigbytes, 1, sizeof(sigbytes), png_file) != sizeof(sigbytes) ||
-      (!png_check_sig (sigbytes, sizeof(sigbytes)))) return 0;
+      (!(png_sig_cmp(sigbytes, 0, sizeof(sigbytes)) == 0))) return 0;
   else return 1;
 }             
 
