diff -Naur csv2sql-0.1/csv2sql csv2sql-0.1.patched/csv2sql
--- csv2sql-0.1/csv2sql	2015-03-07 13:20:09.000000000 -0500
+++ csv2sql-0.1.patched/csv2sql	2022-07-08 15:50:54.975335386 -0400
@@ -24,13 +24,9 @@
     else
       error "the CSV file that you specified (#{file}) does not exist. Exiting."
     end
-  # Otherwise, verify that standard input exists
+  # Otherwise, read from stdin
   else
-    if $stdin.fcntl(Fcntl::F_GETFL, 0) == 0
-      verbose "Will read data from standard input."
-    else
-      error "you did not specify a CSV input file, or provide standard input. Exiting."
-    end
+    verbose "Will read data from standard input."
   end
 end
 
