Index: src/oap.c
===================================================================
--- src/oap.c
+++ src/oap.c
@@ -633,7 +633,7 @@
   if (((buffer_size_t) TO_CLIENT_BUFFER_SIZE) - to_client_end
       < strlen (msg))
     {
-      tracef ("   send_to_client out of space (%i < %i)\n",
+      tracef ("   send_to_client out of space (%i < %zu)\n",
               ((buffer_size_t) TO_CLIENT_BUFFER_SIZE) - to_client_end,
               strlen (msg));
       return TRUE;
Index: src/oxpd.c
===================================================================
--- src/oxpd.c
+++ src/oxpd.c
@@ -247,7 +247,8 @@
                 tracef ("<= client  Input may contain password, suppressed.\n");
               else
                 tracef ("<= client  \"%.*s\"\n",
-                        count,
+			/* Cast is safe because count is bounded by from_client size. */
+                        (int) count,
                         from_client + from_client_end);
 #else
               tracef ("<= client  %i bytes\n", count);
