diff -Naur straw-viewer-0.0.1/bin/gtk-straw-viewer straw-viewer-0.0.1.patched/bin/gtk-straw-viewer
--- straw-viewer-0.0.1/bin/gtk-straw-viewer	2020-02-26 05:17:22.000000000 -0500
+++ straw-viewer-0.0.1.patched/bin/gtk-straw-viewer	2020-02-28 17:04:39.723002906 -0500
@@ -643,11 +643,11 @@
             undef $history{lc($line)};
         }
 
-        require List::Util;
+        require List::MoreUtils;
 
         # Keep only the most recent non-duplicated entries
-        @history        = reverse(List::Util::uniq(reverse(@history)));
-        @search_history = List::Util::uniq(@search_history);
+        @history        = reverse(List::MoreUtils::uniq(reverse(@history)));
+        @search_history = List::MoreUtils::uniq(@search_history);
 
         # Set entry completion
         $completion = Gtk3::EntryCompletion->new;
