--- MP/Makefile.orig	2016-12-20 16:59:48.565402393 +0100
+++ MP/Makefile	2016-12-20 17:00:04.089401418 +0100
@@ -2692,9 +2692,9 @@
 	$(DO_DED_CC)
 
 # Extra dependencies to ensure the git version is incorporated
-  $(B)/client/cl_console.o : ../.git/index
-  $(B)/client/common.o : ../.git/index
-  $(B)/ded/common.o : ../.git/index
+#  $(B)/client/cl_console.o : ../.git/index
+#  $(B)/client/common.o : ../.git/index
+#  $(B)/ded/common.o : ../.git/index
 
 
 #############################################################################
