diff -Naur unifdef-2.12/tests/outeperm.sh unifdef-2.12.patched/tests/outeperm.sh
--- unifdef-2.12/tests/outeperm.sh	2020-02-14 11:58:52.000000000 -0500
+++ unifdef-2.12.patched/tests/outeperm.sh	2026-09-27 19:15:14.377465595 -0400
@@ -1,6 +1,10 @@
 mkdir -p outdir
 chmod 555 outdir
-unifdef -DFOO=1 -DFOOB=42 -UBAR -ooutdir/outfile.c if1.c
+if [ "$( id -u )" = "0" ]; then
+  su nobody -s /bin/sh -c "../unifdef -DFOO=1 -DFOOB=42 -UBAR -ooutdir/outfile.c if1.c"
+else
+  unifdef -DFOO=1 -DFOOB=42 -UBAR -ooutdir/outfile.c if1.c
+fi
 e=$?
 rmdir outdir
 exit $e
