--- conf.py.orig	2016-08-17 20:36:40.160264073 +0200
+++ conf.py	2016-08-17 20:36:12.631648017 +0200
@@ -43,8 +43,8 @@
 master_doc = 'index'
 
 # General information about the project.
-project = 'Borg - Deduplicating Archiver'
-copyright = '2010-2014 Jonas Borgström, 2015-2016 The Borg Collective (see AUTHORS file)'
+project = u'Borg - Deduplicating Archiver'
+copyright = u'2010-2014 Jonas Borgström, 2015-2016 The Borg Collective (see AUTHORS file)'
 
 # The version info for the project you're documenting, acts as replacement for
 # |version| and |release|, also used in various other places throughout the
