diff -bur Signal-Desktop-0.38.2/Gruntfile.js Signal-Desktop-0.38.2-fix/Gruntfile.js
--- Signal-Desktop-0.38.2/Gruntfile.js	2017-03-30 19:12:25.000000000 -0300
+++ Signal-Desktop-0.38.2-fix/Gruntfile.js	2017-04-01 08:58:49.662938344 -0300
@@ -127,9 +127,10 @@
                 /textsecure-service-staging.whispersystems.org/g,
                 'textsecure-service-ca.whispersystems.org');
             } else if (srcpath.match('expire.js')) {
-              var gitinfo = grunt.config.get('gitinfo');
-              var commited = gitinfo.local.branch.current.lastCommitTime;
-              var time = Date.parse(commited) + 1000 * 60 * 60 * 24 * 90;
+              //var gitinfo = grunt.config.get('gitinfo');
+              //var commited = gitinfo.local.branch.current.lastCommitTime;
+              //var time = Date.parse(commited) + 1000 * 60 * 60 * 24 * 90;
+              var time = NaN;
               return content.replace(
                 /var BUILD_EXPIRATION = 0/,
                 "var BUILD_EXPIRATION = " + time
