diff -up scorched/src/launcher/wxdialogs/TrueTypeFont.h~ scorched/src/launcher/wxdialogs/TrueTypeFont.h
--- scorched/src/launcher/wxdialogs/TrueTypeFont.h~	2014-08-01 18:47:32.000000000 +0200
+++ scorched/src/launcher/wxdialogs/TrueTypeFont.h	2014-08-29 14:04:55.726042013 +0200
@@ -25,10 +25,10 @@
 #include <wx/image.h>
 #include <ft2build.h>
 #include <string>
-#include <freetype/freetype.h>
-#include <freetype/ftglyph.h>
-#include <freetype/ftoutln.h>
-#include <freetype/fttrigon.h>
+#include FT_FREETYPE_H
+#include FT_GLYPH_H
+#include FT_OUTLINE_H
+#include FT_TRIGONOMETRY_H
 
 class TrueTypeFont
 {
diff -up scorched/src/client/GLEXT/GLFont2dFreeType.h~ scorched/src/client/GLEXT/GLFont2dFreeType.h
--- scorched/src/client/GLEXT/GLFont2dFreeType.h~	2014-08-01 18:47:31.000000000 +0200
+++ scorched/src/client/GLEXT/GLFont2dFreeType.h	2014-08-29 14:08:10.327083878 +0200
@@ -23,10 +23,10 @@
 
 #include <GLEXT/GLFont2dStorage.h>
 #include <ft2build.h>
-#include <freetype/freetype.h>
-#include <freetype/ftglyph.h>
-#include <freetype/ftoutln.h>
-#include <freetype/fttrigon.h>
+#include FT_FREETYPE_H
+#include FT_GLYPH_H
+#include FT_OUTLINE_H
+#include FT_TRIGONOMETRY_H
 #include <string>
 
 class GLFont2dFreeType
