--- trunk/vhba-module/vhba.c	2010/08/15 20:11:18	691
+++ trunk/vhba-module/vhba.c	2011/02/27 15:56:27	730
@@ -363,7 +363,7 @@
         spin_unlock_irqrestore(&vhost->cmd_lock, flags);
 }
 
-static int vhba_queuecommand(struct scsi_cmnd *cmd, void (*done)(struct scsi_cmnd *))
+static int vhba_queuecommand_lck(struct scsi_cmnd *cmd, void (*done)(struct scsi_cmnd *))
 {
         struct vhba_device *vdev;
         int retval;
@@ -388,6 +388,12 @@
         return retval;
 }
 
+#ifdef DEF_SCSI_QCMD
+DEF_SCSI_QCMD(vhba_queuecommand)
+#else
+#define vhba_queuecommand vhba_queuecommand_lck
+#endif
+
 static int vhba_abort(struct scsi_cmnd *cmd)
 {
         struct vhba_device *vdev;
@@ -796,7 +802,7 @@
         .poll = vhba_ctl_poll,
         .unlocked_ioctl = vhba_ctl_ioctl,
 #ifdef CONFIG_COMPAT
-	.compat_ioctl = vhba_ctl_compat_ioctl,
+        .compat_ioctl = vhba_ctl_compat_ioctl,
 #endif
 };
