--- vhba-module-1.1.0.orig/kat/scatterlist_has_page_link.c	2008-09-13 18:44:36.000000000 +0200
+++ vhba-module-1.1.0/kat/scatterlist_has_page_link.c	2008-09-13 19:01:33.000000000 +0200
@@ -0,0 +1,9 @@
+#include <linux/autoconf.h>
+#include <linux/scatterlist.h>
+
+/* Note: scatterlist.page_link is used in kernel (2.6.24 <= version < X) */
+
+struct scatterlist s = { .page_link = 0 };
+
+//T#define KAT_SCATTERLIST_HAS_PAGE_LINK
+//F#include <linux/scatterlist.h>
