diff -Naur Miro-2.0.4.orig/platform/gtk-x11/plat/renderers/gstreamerrenderer.py Miro-2.0.4/platform/gtk-x11/plat/renderers/gstreamerrenderer.py
--- Miro-2.0.4.orig/platform/gtk-x11/plat/renderers/gstreamerrenderer.py	2009-03-30 20:04:53.000000000 +0000
+++ Miro-2.0.4/platform/gtk-x11/plat/renderers/gstreamerrenderer.py	2009-04-13 08:12:45.000000000 +0000
@@ -281,4 +281,5 @@
                 position + (rate * gst.SECOND))
  
     def movie_data_program_info(self, movie_path, thumbnail_path):
-        return ((sys.executable, 'plat/renderers/gst_extractor.py', movie_path, thumbnail_path), None)
+        extractor_path = os.path.join(os.path.split(__file__)[0], "gst_extractor.py")
+        return ((sys.executable, extractor_path, movie_path, thumbnail_path), None)
