diff -Naur miro-4.0.4.orig/linux/miro miro-4.0.4/linux/miro
--- miro-4.0.4.orig/linux/miro	2011-12-22 14:45:47.000000000 +0000
+++ miro-4.0.4/linux/miro	2011-12-28 21:26:20.298219802 +0000
@@ -42,7 +42,7 @@
 then
     MIROREAL=./miro.real
 else
-    MIROREAL=`which gdb`
+    MIROREAL=`which miro.real`
 fi
 
 echo "using ${MIROREAL}"
@@ -60,7 +60,7 @@
     fi
 
     $GDB -ex 'set breakpoint pending on' \
-        -ex 'run' --args $PYTHON ./miro.real --sync "$@"
+        -ex 'run' --args $PYTHON $MIROREAL --sync "$@"
 else
     miro.real "$@"
 fi
