diff -Naur Miro-2.0.3.orig/platform/gtk-x11/setup.py Miro-2.0.3/platform/gtk-x11/setup.py
--- Miro-2.0.3.orig/platform/gtk-x11/setup.py	2009-03-12 20:24:46.000000000 +0000
+++ Miro-2.0.3/platform/gtk-x11/setup.py	2009-03-15 12:46:33.000000000 +0000
@@ -466,8 +466,12 @@
             xpcom_lib = 'firefox-xpcom'
             gtkmozembed_lib = 'firefox-gtkmozembed'
 
+        elif re.search("^seamonkey-xpcom", packages, re.MULTILINE):
+            xpcom_lib = 'seamonkey-xpcom'
+            gtkmozembed_lib = 'seamonkey-gtkmozembed'
+
         else:
-            sys.exit("Can't find libxul, xulrunner-xpcom, mozilla-xpcom or firefox-xpcom")
+            sys.exit("Can't find libxul, xulrunner-xpcom, mozilla-xpcom, firefox-xpcom or seamonkey-xpcom")
 
     print "using xpcom_lib: ", repr(xpcom_lib)
     print "using gtkmozembed_lib: ", repr(gtkmozembed_lib)
