diff -Nru gtkman-0.6.orig/install.sh gtkman-0.6/install.sh
--- gtkman-0.6.orig/install.sh	2013-06-14 11:07:18.000000000 +0200
+++ gtkman-0.6/install.sh	2013-06-15 10:29:15.522760342 +0200
@@ -6,7 +6,7 @@
 
 # Install manpage
 if [ -f man/gtkman.man ]; then
-	install -D -m 644 man/gtkman.man $DESTDIR/usr/share/man/man1/gtkman.1
+	install -D -m 644 man/gtkman.man $DESTDIR/usr/man/man1/gtkman.1
 fi
 
 # Install icons
