diff --git a/CMakeLists.txt b/CMakeLists.txt
index d131cb48..8b3d420a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,4 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 3.1.0 FATAL_ERROR)
-cmake_policy(SET CMP0043 OLD) # cmake --help-policy CMP0043
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5...4.0)
 
 PROJECT(NitrokeyApp LANGUAGES CXX)
 SET(PROJECT_VERSION "1.5.0")
diff -Naur libnitrokey-3.6.orig/CMakeLists.txt libnitrokey-3.6/CMakeLists.txt
--- a/libnitrokey/CMakeLists.txt	2020-09-19 17:11:29.000000000 +0200
+++ b/libnitrokey/CMakeLists.txt	2026-06-12 18:02:59.906427507 +0200
@@ -3,7 +3,7 @@
     SET(CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_NO_WARNINGS ON)
 ENDIF()
 
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5...4.0)
 IF (UNIX)
     OPTION(ADD_ASAN "Use ASAN to show memory issues" FALSE)
     OPTION(ADD_TSAN "Use TSAN to show thread issues" FALSE)
