diff -Nur vlock-2.1.orig/Makefile vlock-2.1/Makefile
--- vlock-2.1.orig/Makefile	2007-09-08 13:04:25.000000000 -0500
+++ vlock-2.1/Makefile	2007-12-31 23:06:31.231101644 -0600
@@ -63,10 +63,10 @@
 .PHONY: install-programs
 install-programs: $(PROGRAMS)
 	$(INSTALL) -D -m 755 -o root -g $(ROOT_GROUP) vlock $(DESTDIR)$(PREFIX)/bin/vlock
-	$(INSTALL) -D -m 4711 -o root -g $(ROOT_GROUP) vlock-current $(DESTDIR)$(PREFIX)/sbin/vlock-current
+	$(INSTALL) -D -m 2751 -o root -g shadow vlock-current $(DESTDIR)$(PREFIX)/sbin/vlock-current
 	$(INSTALL) -D -m 755 -o root -g $(ROOT_GROUP) vlock-all $(DESTDIR)$(PREFIX)/sbin/vlock-all
 	$(INSTALL) -D -m $(VLOCK_MODE) -o root -g $(VLOCK_GROUP) vlock-nosysrq $(DESTDIR)$(PREFIX)/sbin/vlock-nosysrq
-	$(INSTALL) -D -m $(VLOCK_MODE) -o root -g $(VLOCK_GROUP) vlock-new $(DESTDIR)$(PREFIX)/sbin/vlock-new
+	$(INSTALL) -D -m 2751 -o root -g tty vlock-new $(DESTDIR)$(PREFIX)/sbin/vlock-new
 
 .PHONY: install-man
 install-man:
