diff -Naur argtable2-13/doc/argtable2.3.in argtable2-13.patched/doc/argtable2.3.in
--- argtable2-13/doc/argtable2.3.in	2011-01-10 05:44:01.000000000 -0500
+++ argtable2-13.patched/doc/argtable2.3.in	2014-02-10 08:52:25.000000000 -0500
@@ -59,7 +59,7 @@
 Argtable is an ANSI C library for parsing GNU style command line arguments with a minimum of fuss. It enables the programmer to define their program's argument syntax directly in the source code as an array of structs. The command line is then parsed according to that specification and the resulting values stored directly into user\-defined program variables where they are accessible to the main program.
 .PP
 This man page is only for reference.
-Introductory documentation and example source code is typically installed under \fB@prefix@/share/doc/argtable2/\fP
+Introductory documentation and example source code is typically installed under \fB@docdir@/\fP
 and is also available from the argtable homepage at http://argtable.sourceforge.net.
 
 .SS Constructing an arg_<xxx> data structure
@@ -352,15 +352,13 @@
 .SH "FILES"
 @prefix@/include/argtable2.h
 .br
-@prefix@/lib/libargtable2.a
+@libdir@/libargtable2.so
 .br
-@prefix@/lib/libargtable2.so
+@mandir@/man3/argtable2.3.gz
 .br
-@prefix@/man3/argtable2.3
+@docdir@/
 .br
-@prefix@/share/doc/argtable2/
-.br
-@prefix@/share/doc/argtable2/example/
+@docdir@/example/
 
 .SH "AUTHOR"
 Stewart Heitmann <sheitmann@users.sourceforge.net>
diff -Naur argtable2-13/doc/argtable2.html.in argtable2-13.patched/doc/argtable2.html.in
--- argtable2-13/doc/argtable2.html.in	2011-01-10 06:29:53.000000000 -0500
+++ argtable2-13.patched/doc/argtable2.html.in	2014-02-10 08:52:20.000000000 -0500
@@ -115,7 +115,7 @@
 <p style="margin-left:11%; margin-top: 1em">This man page
 is only for reference. Introductory documentation and
 example source code is typically installed under
-<b>@prefix@/share/doc/argtable2/</b> and is also available
+<b>@docdir@/</b> and is also available
 from the argtable homepage at
 http://argtable.sourceforge.net.</p>
 
@@ -577,11 +577,10 @@
 
 <p style="margin-left:11%; margin-top: 1em">@prefix@/include/argtable2.h
 <br>
-@prefix@/lib/libargtable2.a <br>
-@prefix@/lib/libargtable2.so <br>
-@prefix@/man3/argtable2.3 <br>
-@prefix@/share/doc/argtable2/ <br>
-@prefix@/share/doc/argtable2/example/</p>
+@libdir@/libargtable2.so <br>
+@mandir@/man3/argtable2.3.gz <br>
+@docdir@/ <br>
+@docdir@/example/</p>
 
 <h2>AUTHOR
 <a name="AUTHOR"></a>
