diff --color -rupN mozc.orig/src/third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc mozc/src/third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc
--- mozc.orig/src/third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc	2021-03-15 22:29:19.136624465 +0800
+++ mozc/src/third_party/abseil-cpp/absl/synchronization/internal/graphcycles.cc	2021-03-15 22:29:50.682448271 +0800
@@ -37,6 +37,7 @@
 
 #include <algorithm>
 #include <array>
+#include <limits>
 #include "absl/base/internal/hide_ptr.h"
 #include "absl/base/internal/raw_logging.h"
 #include "absl/base/internal/spinlock.h"
