--- third_party/abseil-cpp/absl/strings/internal/str_format/float_conversion.cc	2023-09-05 17:36:23.973250174 +0900
+++ third_party/abseil-cpp/absl/strings/internal/str_format/float_conversion.cc	2023-09-05 17:37:06.585477262 +0900
@@ -5,6 +5,7 @@
 #include <algorithm>
 #include <cassert>
 #include <cmath>
+#include <cstdint>
 #include <limits>
 #include <string>
 
--- third_party/abseil-cpp/absl/strings/internal/str_format/extension.h	2023-09-05 17:45:04.104106229 +0900
+++ third_party/abseil-cpp/absl/strings/internal/str_format/extension.h	2023-09-05 17:45:24.484098612 +0900
@@ -20,6 +20,7 @@
 
 #include <cstddef>
 #include <cstring>
+#include <cstdint>
 #include <ostream>
 
 #include "absl/base/config.h"
