diff -Nur pdftk-1.41.orig/pdftk/Makefile.Generic pdftk-1.41/pdftk/Makefile.Generic
--- pdftk-1.41.orig/pdftk/Makefile.Generic	2006-09-21 12:37:12.000000000 -0500
+++ pdftk-1.41/pdftk/Makefile.Generic	2009-11-26 00:03:40.823641779 -0600
@@ -23,12 +23,13 @@
 # if you want pdftk to ask before overwriting a file, set
 # ASK_ABOUT_WARNINGS to true; otherwise: false; override this default
 # with the dont_ask or do_ask command-line options
-CPPFLAGS= -O2 -DPATH_DELIM=0x2f -DASK_ABOUT_WARNINGS=false -fdollars-in-identifiers
-CXXFLAGS= -lgcj
+OPT= -O2
+CPPFLAGS= $(OPT) -DPATH_DELIM=0x2f -DASK_ABOUT_WARNINGS=false -fdollars-in-identifiers
+CXXFLAGS= $(OPT) -lgcj
 
 # itext compiler flags
 # -O3 might cause pdftk to segfault on cat operation (gcc 3.4.4)
-export GCJFLAGS= -O2
+export GCJFLAGS= $(OPT) -w
 
 #
 export ARFLAGS= rs
