--- Makefile~	2008-06-18 22:11:05.000000000 -0500
+++ Makefile	2008-06-18 22:11:31.000000000 -0500
@@ -1,8 +1,9 @@
 BINDIR=/usr/bin
 LOCALESDIR=/usr/share/locale
-MANDIR=/usr/share/man/man1
+MANDIR=/usr/man/man1
 WARNFLAGS=-Wall  -W -Wshadow
-CFLAGS?=-O1 -g ${WARNFLAGS}
+CFLAGS?=-O1 -g
+CFLAGS+=${WARNFLAGS}
 CC?=gcc
 
 
@@ -24,7 +25,7 @@
 	@(cd po/ && $(MAKE))
 
 powertop.1.gz: powertop.1
-	gzip -c $< > $@
+	gzip -9 -c $< > $@
 
 install: powertop powertop.1.gz
 	mkdir -p ${DESTDIR}${BINDIR}
