--- core/libs/dimg/loaders/pgfloader.cpp.orig   2019-03-17 16:52:40.102888509 +0700
+++ core/libs/dimg/loaders/pgfloader.cpp        2019-03-17 16:52:51.404020146 +0700
@@ -442,14 +442,6 @@
             }
         }
 
-#ifdef PGFCodecVersionID
-#   if PGFCodecVersionID < 0x061142
-        header.background.rgbtBlue  = 0;
-        header.background.rgbtGreen = 0;
-        header.background.rgbtRed   = 0;
-#   endif
-#endif
-
         pgf.SetHeader(header);
 
         // NOTE: see bug #273765 : Loading PGF thumbs with OpenMP support through a separated thread do not work properlly with libppgf 6.11.24
--- core/libs/threadimageio/pgfutils.cpp.orig   2019-03-17 17:07:35.586244318 +0700
+++ core/libs/threadimageio/pgfutils.cpp        2019-03-17 17:07:47.110376670 +0700
@@ -262,13 +262,6 @@
         header.mode                 = ImageModeRGBA;
         header.usedBitsPerChannel   = 0;             // Auto
 
-#ifdef PGFCodecVersionID
-#   if PGFCodecVersionID < 0x061142
-        header.background.rgbtBlue  = 0;
-        header.background.rgbtGreen = 0;
-        header.background.rgbtRed   = 0;
-#   endif
-#endif
         pgfImg.SetHeader(header);
 
         // NOTE: see bug #273765 : Loading PGF thumbs with OpenMP support through a separated thread do not work properlly with libppgf 6.11.24
