diff -Nur gnucash-4.9.orig/CMakeLists.txt gnucash-4.9/CMakeLists.txt
--- gnucash-4.9.orig/CMakeLists.txt	2021-12-17 16:17:34.000000000 -0500
+++ gnucash-4.9/CMakeLists.txt	2022-02-01 18:20:28.027299079 -0500
@@ -211,7 +211,7 @@
     pkg_check_modules (WEBKIT REQUIRED IMPORTED_TARGET webkitgtk-3.0)
     set(WEBKIT1 1 CACHE INTERNAL "WebKitGtk")
   else()
-    pkg_check_modules (WEBKIT REQUIRED IMPORTED_TARGET webkit2gtk-4.0>=2.14.0)
+    pkg_check_modules (WEBKIT REQUIRED IMPORTED_TARGET webkit2gtk-4.1>=2.14.0)
     set(WEBKIT2 1 CACHE INTERNAL "WebKit2Gtk4")
   endif()
 
