--- museeq/CMakeLists.txt
+++ museeq/CMakeLists.txt
@@ -305,7 +305,12 @@
 
 add_executable(museeq ${MUSEEQ_SOURCES} ${MUSEEQ_MOC_HEADERS})
 
-target_link_libraries(museeq Mucipher ${QT_LIBRARIES} ${QSA_LIBRARY})
+set(LINK_LIBRARIES  Mucipher ${QT_LIBRARIES}) 
+if (QSA) 
+  set(LINK_LIBRARIES ${LINK_LIBRARIES} ${QSA_LIBRARY}) 
+endif (QSA) 
+ 
+target_link_libraries(museeq ${LINK_LIBRARIES}) 
 
 install(
   TARGETS museeq
