--- qwtconfig.pri.orig	2016-06-13 17:14:23.031601095 +1000
+++ qwtconfig.pri	2018-09-14 14:50:18.247000000 +1000
@@ -16,11 +16,8 @@
 # Install paths
 ######################################################################
 
-QWT_INSTALL_PREFIX = $$[QT_INSTALL_PREFIX]
-
 unix {
-    QWT_INSTALL_PREFIX    = /usr/local/qwt-$$QWT_VERSION
-    # QWT_INSTALL_PREFIX = /usr/local/qwt-$$QWT_VERSION-qt-$$QT_VERSION
+    QWT_INSTALL_PREFIX    = /usr
 }
 
 win32 {
@@ -29,8 +26,10 @@
 }
 
 QWT_INSTALL_DOCS      = $${QWT_INSTALL_PREFIX}/doc
-QWT_INSTALL_HEADERS   = $${QWT_INSTALL_PREFIX}/include
-QWT_INSTALL_LIBS      = $${QWT_INSTALL_PREFIX}/lib
+QWT_INSTALL_HEADERS   = $${QWT_INSTALL_PREFIX}/include/qwt
+QWT_INSTALL_LIBS      = $${QWT_INSTALL_PREFIX}/lib$${LIBDIRSUFFIX}
+
+
 
 ######################################################################
 # Designer plugin
@@ -42,14 +41,12 @@
 # runtime environment of designer/creator.
 ######################################################################
 
-QWT_INSTALL_PLUGINS   = $${QWT_INSTALL_PREFIX}/plugins/designer
-
 # linux distributors often organize the Qt installation
 # their way and QT_INSTALL_PREFIX doesn't offer a good
 # path. Also QT_INSTALL_PREFIX is only one of the default
 # search paths of the designer - not the Qt creator
 
-#QWT_INSTALL_PLUGINS   = $$[QT_INSTALL_PREFIX]/plugins/designer
+QWT_INSTALL_PLUGINS   = $$[QT_INSTALL_PLUGINS]/designer
 
 ######################################################################
 # Features
@@ -63,8 +60,7 @@
 # with every Qt upgrade. 
 ######################################################################
 
-QWT_INSTALL_FEATURES  = $${QWT_INSTALL_PREFIX}/features
-# QWT_INSTALL_FEATURES  = $$[QT_INSTALL_PREFIX]/features
+QWT_INSTALL_FEATURES  = $$[QMAKE_MKSPECS]/features
 
 ######################################################################
 # Build the static/shared libraries.
@@ -141,7 +137,7 @@
 # Otherwise you have to build them from the examples directory.
 ######################################################################
 
-#QWT_CONFIG     += QwtExamples
+QWT_CONFIG     += QwtExamples
 
 ######################################################################
 # The playground is primarily intended for the Qwt development 
@@ -171,5 +167,5 @@
 
 unix {
 
-    #QWT_CONFIG     += QwtPkgConfig
+    QWT_CONFIG     += QwtPkgConfig
 }
--- src/src.pro.orig	2018-09-14 14:52:33.531000000 +1000
+++ src/src.pro	2018-09-14 14:53:59.816000000 +1000
@@ -17,7 +17,12 @@
 QWT_OUT_ROOT = $${OUT_PWD}/..
 
 TEMPLATE          = lib
-TARGET            = $$qwtLibraryTarget(qwt)
+greaterThan(QT_MAJOR_VERSION, 4) {
+    TARGET            = $$qwtLibraryTarget(qwt-qt5)
+}
+else {
+    TARGET            = $$qwtLibraryTarget(qwt)
+}
 
 DESTDIR           = $${QWT_OUT_ROOT}/lib
 
--- designer/designer.pro.orig	2016-06-13 17:14:23.019600853 +1000
+++ designer/designer.pro	2018-09-14 17:20:37.038000000 +1000
@@ -83,8 +83,16 @@
         # compile the path for finding the Qwt library
         # into the plugin. Not supported on Windows !
 
-        QMAKE_RPATHDIR *= $${QWT_INSTALL_LIBS}
-        qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt)
+        #QMAKE_RPATHDIR *= $${QWT_INSTALL_LIBS}
+        #qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt)
+	contains(QWT_CONFIG, QwtDll) {
+	    greaterThan(QT_MAJOR_VERSION, 4) {
+ 		qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt-qt5)
+	    }
+	    else {
+ 		qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt)
+	    }
+	}
 
         contains(QWT_CONFIG, QwtDll) {
 
--- examples/examples.pri.orig	2016-06-13 17:14:23.023600934 +1000
+++ examples/examples.pri	2018-09-14 17:44:03.727000000 +1000
@@ -34,8 +34,13 @@
     }
 }
 
-QMAKE_RPATHDIR *= $${QWT_OUT_ROOT}/lib
-qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt)
+#QMAKE_RPATHDIR *= $${QWT_OUT_ROOT}/lib
+greaterThan(QT_MAJOR_VERSION, 4) {
+	qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt-qt5)
+}
+else {
+	qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt)
+}
 
 greaterThan(QT_MAJOR_VERSION, 4) {
 
