--- qwtconfig.pri.orig	2014-12-11 14:13:13.513186492 +0000
+++ qwtconfig.pri	2015-09-27 08:33:01.948490369 +0000
@@ -16,11 +16,8 @@
 # Install paths
 ######################################################################
 
-QWT_INSTALL_PREFIX = $$[QT_INSTALL_PREFIX]
-
 unix {
-    QWT_INSTALL_PREFIX    = /usr/local/qwt-$$QWT_VERSION
-    # QWT_INSTALL_PREFIX = /usr/local/qwt-$$QWT_VERSION-qt-$$QT_VERSION
+    QWT_INSTALL_PREFIX    = /usr
 }
 
 win32 {
@@ -29,8 +26,10 @@
 }
 
 QWT_INSTALL_DOCS      = $${QWT_INSTALL_PREFIX}/doc
-QWT_INSTALL_HEADERS   = $${QWT_INSTALL_PREFIX}/include
-QWT_INSTALL_LIBS      = $${QWT_INSTALL_PREFIX}/lib
+QWT_INSTALL_HEADERS   = $${QWT_INSTALL_PREFIX}/include/qwt
+QWT_INSTALL_LIBS      = $${QWT_INSTALL_PREFIX}/lib$${LIBDIRSUFFIX}
+
+
 
 ######################################################################
 # Designer plugin
@@ -42,14 +41,12 @@
 # runtime environment of designer/creator.
 ######################################################################
 
-QWT_INSTALL_PLUGINS   = $${QWT_INSTALL_PREFIX}/plugins/designer
-
 # linux distributors often organize the Qt installation
 # their way and QT_INSTALL_PREFIX doesn't offer a good
 # path. Also QT_INSTALL_PREFIX is only one of the default
 # search paths of the designer - not the Qt creator
 
-#QWT_INSTALL_PLUGINS   = $$[QT_INSTALL_PREFIX]/plugins/designer
+QWT_INSTALL_PLUGINS   = $$[QT_INSTALL_PLUGINS]/designer
 
 ######################################################################
 # Features
@@ -63,8 +60,7 @@
 # with every Qt upgrade. 
 ######################################################################
 
-QWT_INSTALL_FEATURES  = $${QWT_INSTALL_PREFIX}/features
-# QWT_INSTALL_FEATURES  = $$[QT_INSTALL_PREFIX]/features
+QWT_INSTALL_FEATURES  = $$[QMAKE_MKSPECS]/features
 
 ######################################################################
 # Build the static/shared libraries.
@@ -141,7 +137,7 @@
 # Otherwise you have to build them from the examples directory.
 ######################################################################
 
-#QWT_CONFIG     += QwtExamples
+QWT_CONFIG     += QwtExamples
 
 ######################################################################
 # The playground is primarily intended for the Qwt development 
