--- src/src.pro.orig	2014-12-12 00:13:13.461186493 +1000
+++ src/src.pro	2015-10-05 08:55:43.504194164 +1000
@@ -21,6 +21,19 @@
 
 DESTDIR           = $${QWT_OUT_ROOT}/lib
 
+# Add pkgconfig support
+unix {
+	CONFIG  += create_pc create_prl no_install_prl
+	QMAKE_PKGCONFIG_NAME            = qwt
+	QMAKE_PKGCONFIG_DESCRIPTION     = GUI components
+	QMAKE_PKGCONFIG_PREFIX          = $${QWT_INSTALL_PREFIX}
+	QMAKE_PKGCONFIG_LIBDIR          = $${QWT_INSTALL_LIBS}
+	QMAKE_PKGCONFIG_INCDIR          = $${QWT_INSTALL_HEADERS}
+	QMAKE_PKGCONFIG_REQUIRES        = QtCore
+	QMAKE_PKGCONFIG_DESTDIR         = pkgconfig
+}
+
+
 contains(QWT_CONFIG, QwtDll) {
 
     CONFIG += dll
