diff -Nur qwt-5.1.1.orig/qwtconfig.pri qwt-5.1.1/qwtconfig.pri
--- qwt-5.1.1.orig/qwtconfig.pri	2008-05-24 11:47:36.000000000 -0500
+++ qwt-5.1.1/qwtconfig.pri	2009-06-25 02:06:19.375460237 -0500
@@ -8,14 +8,14 @@
 VERSION      = $$VER_MAJ.$$VER_MIN.$$VER_PAT
 
 unix {
-    INSTALLBASE    = /usr/local/qwt-5.1.1
+    INSTALLBASE    = /usr
 }
 
 win32 {
     INSTALLBASE    = C:/Qwt-5.1.1
 }
 
-target.path    = $$INSTALLBASE/lib
+target.path    = $$INSTALLBASE/@BASELIBDIR@
 headers.path   = $$INSTALLBASE/include
 doc.path       = $$INSTALLBASE/doc
 
