diff -Naur qwt-6.0.2.orig/qwtconfig.pri qwt-6.0.2/qwtconfig.pri
--- qwt-6.0.2.orig/qwtconfig.pri	2012-11-30 14:22:47.397127932 +0100
+++ qwt-6.0.2/qwtconfig.pri	2013-01-07 13:16:08.747317951 +0100
@@ -19,7 +19,7 @@
 QWT_INSTALL_PREFIX = $$[QT_INSTALL_PREFIX]
 
 unix {
-    QWT_INSTALL_PREFIX    = /usr/local/qwt-$$QWT_VERSION
+    QWT_INSTALL_PREFIX    = /usr
 }
 
 win32 {
@@ -28,13 +28,13 @@
 
 QWT_INSTALL_DOCS      = $${QWT_INSTALL_PREFIX}/doc
 QWT_INSTALL_HEADERS   = $${QWT_INSTALL_PREFIX}/include
-QWT_INSTALL_LIBS      = $${QWT_INSTALL_PREFIX}/lib
+QWT_INSTALL_LIBS      = $${QWT_INSTALL_PREFIX}/lib@LIBDIRSUFFIX@
 
 ######################################################################
 # Designer plugin
 ######################################################################
 
-QWT_INSTALL_PLUGINS   = $${QWT_INSTALL_PREFIX}/plugins/designer
+QWT_INSTALL_PLUGINS   = @QT4DIR@/plugins/designer
 # QWT_INSTALL_PLUGINS   = $${QT_INSTALL_PREFIX}/plugins/designer
 
 ######################################################################
@@ -49,7 +49,7 @@
 # with every Qt upgrade. 
 ######################################################################
 
-QWT_INSTALL_FEATURES  = $${QWT_INSTALL_PREFIX}/features
+QWT_INSTALL_FEATURES  = @QT4DIR@/features
 # QWT_INSTALL_FEATURES  = $${QT_INSTALL_PREFIX}/features
 
 ######################################################################
