--- dvdauthor/src/subgen-image.c	2010-03-13 05:14:07.000000000 -0300
+++ dvdauthor_patched/src/subgen-image.c	2010-06-11 08:10:04.000000000 -0300
@@ -226,7 +226,7 @@
     png_struct *ps;
     png_info *pi;
     png_byte **rowp;
-    unsigned long width,height;
+    png_uint_32 width,height;
     int bit_depth,color_type,channels,x,y;
 
     fp=fopen(s->fname,"rb");
@@ -271,7 +271,7 @@
     assert(bit_depth==8); // 8bpp, not 1, 2, 4, or 16
     assert(!(color_type&PNG_COLOR_MASK_PALETTE)); // not a palette
     if( width>MAXX || height>MAXY ) {
-        fprintf(stderr,"ERR: PNG %s is too big: %lux%lu\n",s->fname,width,height);
+        fprintf(stderr,"ERR: PNG %s is too big: %lux%lu\n",s->fname,(long unsigned int)width,(long unsigned int)height);
         png_destroy_read_struct(&ps,&pi,NULL);
         return -1;
     }
