diff -Naupr ipvsadm-1.25.orig/Makefile ipvsadm-1.25/Makefile
--- ipvsadm-1.25.orig/Makefile	2008-09-23 18:14:46.000000000 +0200
+++ ipvsadm-1.25/Makefile	2008-12-22 20:03:53.793372130 +0100
@@ -62,14 +62,7 @@ RPMBUILD = $(shell				\
 	fi )
 
 ifeq (,$(FORCE_GETOPT))
-LIB_SEARCH = /lib /usr/lib /usr/local/lib
-POPT_LIB = $(shell for i in $(LIB_SEARCH); do \
-  if [ -f $$i/libpopt.a ]; then \
-    if nm $$i/libpopt.a | fgrep -q poptGetContext; then \
-	echo "-lpopt"; \
-    fi; \
-  fi; \
-done)
+POPT_LIB = "-lpopt"
 endif
 
 ifneq (,$(POPT_LIB))
