Last-Update: 2016-06-16
Forwarded: not-needed
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=827421
Author: Dmitry Smirnov <onlyjob@debian.org>
Description: allow build with QT-5.5+

--- a/src/phantomjs.pro
+++ b/src/phantomjs.pro
@@ -1,8 +1,4 @@

-if(!equals(QT_MAJOR_VERSION, 5)|!equals(QT_MINOR_VERSION, 5)) {
-    error("This program can only be compiled with Qt 5.5.x.")
-}
-
 TEMPLATE = app
 TARGET = phantomjs
 QT += network webkitwidgets
