diff -Naur QtCurve-KDE4-1.5.0.orig/config/CMakeLists.txt QtCurve-KDE4-1.5.0/config/CMakeLists.txt
--- QtCurve-KDE4-1.5.0.orig/config/CMakeLists.txt	2010-06-27 21:55:45.000000000 +0100
+++ QtCurve-KDE4-1.5.0/config/CMakeLists.txt	2010-06-28 18:40:19.000000000 +0100
@@ -1,5 +1,5 @@
 if (QTC_STYLE_SUPPORT)
-    set(kstyle_qtcurve_config_PART_SRCS qtcurveconfig.cpp exportthemedialog.cpp qtcurvekwinconfig.cpp ../kwin/qtcurveconfig.cpp ../kwin/qtcurveshadowconfiguration.cpp)
+    set(kstyle_qtcurve_config_PART_SRCS qtcurveconfig.cpp exportthemedialog.cpp ../kwinconfig/qtcurvekwinconfig.cpp ../kwin/qtcurveconfig.cpp ../kwin/qtcurveshadowconfiguration.cpp)
 else (QTC_STYLE_SUPPORT)
     set(kstyle_qtcurve_config_PART_SRCS qtcurveconfig.cpp ../kwinconfig/qtcurvekwinconfig.cpp ../kwin/qtcurveconfig.cpp ../kwin/qtcurveshadowconfiguration.cpp)
 endif (QTC_STYLE_SUPPORT)
