--- Makefile.orig	2011-09-10 06:59:30.000000000 +0000
+++ Makefile	2011-11-23 14:49:17.000000000 +0000
@@ -58,15 +58,15 @@
 install: zyGrib
 	mkdir -p $(INSTALLDIR)
 	mkdir -p $(INSTALLDIR)/bin
+	mkdir -p $(INSTALLDIR)/../../usr/bin
 	cp -a data $(INSTALLDIR)
 	cp -a grib $(INSTALLDIR)
 	rm -f ./zyGrib
 	rm -f $(INSTALLDIR)/bin/zyGrib
 	cp  -a src/zyGrib $(INSTALLDIR)/bin
-	echo "#!/bin/bash" >> ./zyGrib
-	echo "cd $(INSTALLDIR)" >> ./zyGrib
-	echo 'bin/zyGrib $$*' >> ./zyGrib
-	chmod 755 ./zyGrib
-	cp -af zyGrib $(INSTALLDIR)
+	echo "#!/bin/bash" >> $(INSTALLDIR)/../../usr/bin/zyGrib
+	echo "cd /opt/zyGrib" >> $(INSTALLDIR)/../../usr/bin/zyGrib
+	echo 'bin/zyGrib $$*' >> $(INSTALLDIR)/../../usr/bin/zyGrib
+	chmod 755 $(INSTALLDIR)/../../usr/bin/zyGrib
 
 
