diff -Naur gelemental-1.2.0.orig/src/dialogs.cc gelemental-1.2.0/src/dialogs.cc
--- gelemental-1.2.0.orig/src/dialogs.cc	2007-09-25 06:49:33.000000000 +0200
+++ gelemental-1.2.0/src/dialogs.cc	2017-07-27 21:45:34.356776770 +0200
@@ -247,7 +247,7 @@
 	const Gtk::TreePath& tpath, bool)
 {
 	Gtk::TreeIter iter = store->get_iter (tpath);
-	return iter ? iter->get_value (cols.property) : false;
+	return iter ? static_cast<bool>(iter->get_value (cols.property)) : false;
 }
 
 
