Index: gelemental-1.2.0/libelemental/misc/extras.cc
===================================================================
--- gelemental-1.2.0.orig/libelemental/misc/extras.cc
+++ gelemental-1.2.0/libelemental/misc/extras.cc
@@ -21,7 +21,7 @@
 
 #include <algorithm>
 #include <stdexcept>
-#include <glib/gmem.h>
+#include <glib.h>
 #include <glibmm/utility.h>
 
 namespace misc {
Index: gelemental-1.2.0/libelemental/misc/widgets.cc
===================================================================
--- gelemental-1.2.0.orig/libelemental/misc/widgets.cc
+++ gelemental-1.2.0/libelemental/misc/widgets.cc
@@ -20,7 +20,7 @@
 #include "macros.hh"
 #include "widgets.hh"
 
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <gtkmm/alignment.h>
 #include <gtkmm/image.h>
 #include <gtkmm/label.h>
Index: gelemental-1.2.0/src/main.cc
===================================================================
--- gelemental-1.2.0.orig/src/main.cc
+++ gelemental-1.2.0/src/main.cc
@@ -26,8 +26,7 @@
 #include <clocale>
 #include <iostream>
 #include <locale>
-#include <glib/goption.h>
-#include <glib/gutils.h>
+#include <glib.h>
 #include <glibmm/optionentry.h>
 #include <gtk/gtkaboutdialog.h>
 #include <gtk/gtkversion.h>
