--- trunk/source/shared_lib/sources/util/util.cpp	2012/04/14 21:21:09	3244
+++ trunk/source/shared_lib/sources/util/util.cpp	2012/05/01 14:55:06	3279
@@ -22,6 +22,8 @@
 
 #ifdef WIN32
   #include <io.h> // for open()
+#else
+  #include <unistd.h>
 #endif
 
 #include <sys/stat.h> // for open()
--- trunk/source/shared_lib/include/graphics/math_util.h	2012/04/16 20:15:57	3252
+++ trunk/source/shared_lib/include/graphics/math_util.h	2012/05/01 14:55:06	3279
@@ -184,10 +184,10 @@
 
 	Rect2<T> computeBoundingRect() const{
 		return Rect2i(
-			min(p[0].x, p[1].x), 
-			min(p[0].y, p[2].y), 
-			max(p[2].x, p[3].x), 
-			max(p[1].y, p[3].y));
+			std::min(p[0].x, p[1].x),
+			std::min(p[0].y, p[2].y),
+			std::max(p[2].x, p[3].x),
+			std::max(p[1].y, p[3].y));
 	}
 
 	bool isInside(const Vec2<T> &pt) const{
