diff -Naur megaglest-3.9.0.orig/source/g3d_viewer/CMakeLists.txt megaglest-3.9.0/source/g3d_viewer/CMakeLists.txt
--- megaglest-3.9.0.orig/source/g3d_viewer/CMakeLists.txt	2013-11-18 23:10:38.000000000 +0000
+++ megaglest-3.9.0/source/g3d_viewer/CMakeLists.txt	2013-11-19 07:19:28.601093314 +0000
@@ -167,7 +167,7 @@
 			MESSAGE("***-- Found xvfb-run: ${XVFB_EXEC} about to build manpage with it.")
 
 		  	add_custom_target(${TARGET_NAME_MANPAGE} ALL
-		    		COMMAND ${XVFB_EXEC} --auto-servernum --server-num=770 ${HELP2MAN} --name='${MG_MANPAGE_DESCRIPTION}' --section=6 -N -o ${HELP2MAN_OUT_PATH}${TARGET_NAME_MANPAGE} ${HELP2MAN_OUT_PATH}${TARGET_NAME}
+		    		COMMAND ${XVFB_EXEC} --auto-servernum ${HELP2MAN} --name='${MG_MANPAGE_DESCRIPTION}' --section=6 -N -o ${HELP2MAN_OUT_PATH}${TARGET_NAME_MANPAGE} ${HELP2MAN_OUT_PATH}${TARGET_NAME}
 		    	DEPENDS ${TARGET_NAME})
 		ELSE()
 		  	add_custom_target(${TARGET_NAME_MANPAGE} ALL
