diff -Naur grip-4.2.3/src/grip.c grip-4.2.3.patched/src/grip.c
--- grip-4.2.3/src/grip.c	2021-07-07 13:53:25.000000000 -0400
+++ grip-4.2.3.patched/src/grip.c	2021-12-13 14:10:40.912094097 -0500
@@ -370,6 +370,10 @@
   screen = gtk_widget_get_screen (widget);
 
   gtk_show_uri(screen, "ghelp:grip", gtk_get_current_event_time(), &error);
+  if(error != NULL) {
+    error = NULL;
+    gtk_show_uri(screen, "file:///usr/doc/grip-" VERSION "/html/grip.html", gtk_get_current_event_time(), &error);
+  }
 
   if (error != NULL) {
     GtkWidget *d;
