diff -Naur sdlmess0129.orig/makefile.sdl sdlmess0129/makefile.sdl
--- sdlmess0129.orig/makefile.sdl	2009-01-06 23:41:45.000000000 -0500
+++ sdlmess0129/makefile.sdl	2009-03-07 18:43:52.000000000 -0500
@@ -242,8 +242,11 @@
 # compile-time definitions
 #-------------------------------------------------
 
+# 20090304 bkw: Look for mess.ini in ~/.mess, then /etc
+DEFS = -DINI_PATH="\"\$$HOME/.mess;/etc\""
+
 # CR/LF setup: use both on win32/os2, CR only on everything else
-DEFS = -DCRLF=2
+DEFS += -DCRLF=2
 
 ifeq ($(TARGETOS),win32)
 DEFS = -DCRLF=3
