diff -ur avogadro-1.0.3/CMakeLists.txt avogadro-1.0.3-mkspecs-dir/CMakeLists.txt
--- avogadro-1.0.3/CMakeLists.txt	2011-04-25 07:29:58.000000000 +0200
+++ avogadro-1.0.3-mkspecs-dir/CMakeLists.txt	2011-04-27 02:42:26.000000000 +0200
@@ -491,7 +491,7 @@
 if(QT_MKSPECS_DIR AND INSTALL_QMAKE_PRF)
     install(FILES
       "${CMAKE_CURRENT_BINARY_DIR}/avogadro.prf"
-      DESTINATION "${CMAKE_INSTALL_PREFIX}/${QT_MKSPECS_RELATIVE}/features"
+      DESTINATION "${QT_MKSPECS_DIR}/features"
     )
 else(QT_MKSPECS_DIR AND INSTALL_QMAKE_PRF)
   message(WARNING "Qt mkspecs directory not found. File 'avogadro.prf' will be installed to ${SHARE_INSTALL_PREFIX}/avogadro/qmake.")
