diff -Naur avogadro-1.0.3.orig/libavogadro/src/extensions/pythonterminal.h avogadro-1.0.3/libavogadro/src/extensions/pythonterminal.h
--- avogadro-1.0.3.orig/libavogadro/src/extensions/pythonterminal.h	2011-04-25 05:22:18.000000000 +0000
+++ avogadro-1.0.3/libavogadro/src/extensions/pythonterminal.h	2012-07-12 02:35:19.324975289 +0000
@@ -23,6 +23,10 @@
 #ifndef PYTHONTERMINAL_H
 #define PYTHONTERMINAL_H
 
+#ifdef Q_MOC_RUN
+#define BOOST_TT_HAS_OPERATOR_HPP_INCLUDED
+#endif
+
 #include <avogadro/extension.h>
 #include <avogadro/primitive.h>
 #include <avogadro/glwidget.h>
diff -Naur avogadro-1.0.3.orig/libavogadro/src/pythonengine_p.h avogadro-1.0.3/libavogadro/src/pythonengine_p.h
--- avogadro-1.0.3.orig/libavogadro/src/pythonengine_p.h	2011-04-25 05:22:18.000000000 +0000
+++ avogadro-1.0.3/libavogadro/src/pythonengine_p.h	2012-07-12 02:32:39.429980751 +0000
@@ -25,6 +25,10 @@
 #ifndef PYTHONENGINE_H
 #define PYTHONENGINE_H
 
+#ifdef Q_MOC_RUN
+#define BOOST_TT_HAS_OPERATOR_HPP_INCLUDED
+#endif
+
 #include <avogadro/global.h>
 #include <avogadro/engine.h>
 #include <boost/python.hpp>
diff -Naur avogadro-1.0.3.orig/libavogadro/src/pythonextension_p.h avogadro-1.0.3/libavogadro/src/pythonextension_p.h
--- avogadro-1.0.3.orig/libavogadro/src/pythonextension_p.h	2011-04-25 05:22:18.000000000 +0000
+++ avogadro-1.0.3/libavogadro/src/pythonextension_p.h	2012-07-12 02:33:46.838978444 +0000
@@ -26,6 +26,10 @@
 #ifndef PYTHONEXTENSION_H
 #define PYTHONEXTENSION_H
 
+#ifdef Q_MOC_RUN
+#define BOOST_TT_HAS_OPERATOR_HPP_INCLUDED
+#endif
+
 #include <avogadro/extension.h>
 #include <avogadro/primitive.h>
 #include <avogadro/glwidget.h>
diff -Naur avogadro-1.0.3.orig/libavogadro/src/pythontool_p.h avogadro-1.0.3/libavogadro/src/pythontool_p.h
--- avogadro-1.0.3.orig/libavogadro/src/pythontool_p.h	2011-04-25 05:22:18.000000000 +0000
+++ avogadro-1.0.3/libavogadro/src/pythontool_p.h	2012-07-12 02:34:14.068977512 +0000
@@ -25,6 +25,10 @@
 #ifndef PYTHONTOOL_H
 #define PYTHONTOOL_H
 
+#ifdef Q_MOC_RUN
+#define BOOST_TT_HAS_OPERATOR_HPP_INCLUDED
+#endif
+
 #include <avogadro/global.h>
 #include <avogadro/tool.h>
 #include <boost/python.hpp>
