diff -Naur avogadro-1.0.3.orig/avogadro/src/mainwindow.h avogadro-1.0.3/avogadro/src/mainwindow.h
--- avogadro-1.0.3.orig/avogadro/src/mainwindow.h	2011-04-25 05:22:18.000000000 +0000
+++ avogadro-1.0.3/avogadro/src/mainwindow.h	2012-07-11 13:57:36.624006453 +0000
@@ -24,6 +24,7 @@
 #ifndef MAINWINDOW_H
 #define MAINWINDOW_H
 
+#include <cstddef>
 #include <Eigen/Geometry>
 #include "ui_mainwindow.h"
 #include "flattabwidget.h"
diff -Naur avogadro-1.0.3.orig/libavogadro/src/tools/eyecandy.h avogadro-1.0.3/libavogadro/src/tools/eyecandy.h
--- avogadro-1.0.3.orig/libavogadro/src/tools/eyecandy.h	2011-04-25 05:22:18.000000000 +0000
+++ avogadro-1.0.3/libavogadro/src/tools/eyecandy.h	2012-07-11 13:43:22.892050693 +0000
@@ -25,6 +25,7 @@
 #ifndef EYECANDY_H
 #define EYECANDY_H
 
+#include <cstddef>
 #include <Eigen/Core>
 
 namespace Avogadro {
