diff -ruN w3m-0.4.1.orig/scripts/w3mhelp.cgi.in w3m-0.4.1/scripts/w3mhelp.cgi.in
--- w3m-0.4.1.orig/scripts/w3mhelp.cgi.in	2003-06-12 21:03:09.000000000 +0900
+++ w3m-0.4.1/scripts/w3mhelp.cgi.in	2003-06-12 21:25:44.000000000 +0900
@@ -51,6 +51,12 @@
     }
 }
 
+if ($lang eq 'ja') {
+  $help_charset='EUC-JP';
+} else {
+  $help_charset='UTF-8';
+}
+
 %f = %keyfunc;
 %keyfunc = ();
 while (($key, $fname) = each %f) {
@@ -88,6 +94,7 @@
 
 <HTML>
 <HEAD>
+<META http-equiv="Content-Type" content="text/html; charset=$help_charset">
 <TITLE>w3m help page</TITLE>
 </HEAD>
 <BODY>
