diff -ruN w3m-0.5.1.orig/acinclude.m4 w3m-0.5.1/acinclude.m4
--- w3m-0.5.1.orig/acinclude.m4	2006-03-06 18:24:38.000000000 +0900
+++ w3m-0.5.1/acinclude.m4	2006-03-06 18:27:55.000000000 +0900
@@ -394,7 +394,7 @@
 [lib=$1
  AC_MSG_CHECKING(for -l$lib)
  extlib="not found"
- for dir in /lib /usr/lib /usr/local/lib /usr/ucblib /usr/ccslib /usr/ccs/lib
+ for dir in /lib64 /lib /usr/lib64 /usr/lib /usr/local/lib64 /usr/local/lib /usr/ucblib /usr/ccslib /usr/ccs/lib
  do
    if test -f $dir/lib$lib.a -o -f $dir/lib$lib.so ; then 
     LIBS="$LIBS -l$lib"
