diff --git a/CMakeLists.txt b/CMakeLists.txt
index d15214545..fe277fcd6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -138,7 +138,7 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
     # all compilers until they all accept the C++17 standard
     if(CMAKE_COMPILER_IS_GNUCC)
         set(COMMON_COMPILE_FLAGS "${COMMON_COMPILE_FLAGS} -Wno-stringop-truncation -Wno-stringop-overflow")
-        if(CMAKE_CXX_COMPILER_VERSION GREATER_EQUAL 7.1)
+        if(CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 7.1 OR CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7.1)
             set(COMMON_COMPILE_FLAGS "${COMMON_COMPILE_FLAGS} -Wimplicit-fallthrough=0")
         endif()
     endif()
