Index: mp3check-0.8.7/tappconfig.cc
===================================================================
--- mp3check-0.8.7.orig/tappconfig.cc
+++ mp3check-0.8.7/tappconfig.cc
@@ -970,7 +970,7 @@ void TAppConfigItem::setValue(const tstr
 			      const tstring& rcfile, const tstring& line_context) {
    if(only_app)       
      userError("internal parameter name '%s' is private to the application may not be set by the user!\n", name.c_str());
-   if(only_cl && (!setin==COMMAND_LINE))       
+   if(only_cl && (setin!=COMMAND_LINE))       
      userError("parameter name '%s' is only available on the command line!\n", name.c_str());
 
    // remember verbatim value for non string parameters in string_value
