See https://code.google.com/p/ocropus/issues/detail?id=365

diff -Naur ocropus-20120920.orig/ocropy/setup.py ocropus-20120920/ocropy/setup.py
--- ocropus-20120920.orig/ocropy/setup.py	2012-09-20 06:48:34.000000000 +0200
+++ ocropus-20120920/ocropy/setup.py	2012-09-20 11:16:24.784307573 +0200
@@ -4,6 +4,9 @@
 from distutils.core import setup, Extension, Command
 from distutils.command.install_data import install_data
 
+import matplotlib
+matplotlib.use('Agg')
+
 from ocrolib import default
 modeldir = "models/"
 modelfiles = default.installable
