--- BeastieWorker-0.4/src/Makefile.original	2014-04-21 12:16:55.317087746 +0300
+++ BeastieWorker-0.4/src/Makefile	2014-04-21 13:35:23.362229082 +0300
@@ -3,8 +3,8 @@
 DEL=rm
 
 LFLAGS=      -O2 -pthread
-INCS=        -I/usr/local/include/SDL11 -I/usr/X11R6/include/
-LIBS=        -L/usr/local/lib -lSDL-1.1 -L/usr/X11R6/lib -lGL -lGLU
+INCS=        -I/usr/include/SDL -I/usr/X11R6/include/
+LIBS=        -L/usr/lib -lSDL -L/usr/X11R6/lib -lGL -lGLU
 OBJS=        cursor.o font.o main.o matrixmath.o menu.o model3d.o player.o scene.o sound.o strings.o texture.o sky.o
 MAIN=        BeastieWorker
 
