--- fvwm3-1.0.4.orig/fvwm/move_resize.c	2021-07-17 18:03:08.000000000 +0300
+++ fvwm3-1.0.4/fvwm/move_resize.c	2021-07-21 01:20:34.428964491 +0300
@@ -3269,13 +3269,14 @@
 
 static void set_geom_win_visible_val(char *token, bool val)
 {
-	if (token == NULL)
-		return;
-
 	Scr.gs.do_hide_position_window = !val;
 	Scr.gs.do_hide_resize_window = !val;
 
-	if (StrEquals(token, "never"))
+	if (token == NULL)
+	{
+		return;
+	}
+	else if (StrEquals(token, "never"))
 	{
 		Scr.gs.do_hide_position_window = val;
 		Scr.gs.do_hide_resize_window = val;
