diff -Nur fvwm3-1.0.2.orig/fvwm/screen.h fvwm3-1.0.2/fvwm/screen.h
--- fvwm3-1.0.2.orig/fvwm/screen.h	2020-12-12 03:37:08.000000000 +0300
+++ fvwm3-1.0.2/fvwm/screen.h	2020-12-22 03:21:41.103120674 +0300
@@ -485,8 +485,7 @@
 		get_unshaded_geometry((fw), &g);			   \
 		mnew = FindScreenOfXY((fw)->g.frame.x, (fw)->g.frame.y);   \
 		/* Avoid unnecessary updates. */			   \
-		if (mnew == (fw)->m &&					   \
-		    (fw)->Desk == mnew->virtual_scr.CurrentDesk)	   \
+		if (mnew == (fw)->m)					   \
 			break;						   \
 		(fw)->m_prev = (fw)->m;					   \
 		(fw)->m = mnew;						   \
