diff -Naur kmediafactory-r691.orig/lib/kmftools.cpp kmediafactory-r691/lib/kmftools.cpp
--- kmediafactory-r691.orig/lib/kmftools.cpp	2011-06-10 19:05:46.000000000 +0200
+++ kmediafactory-r691/lib/kmftools.cpp	2012-09-19 07:54:15.485298848 +0200
@@ -51,6 +51,7 @@
 #include <fontconfig/fontconfig.h>
 #include <sys/stat.h>
 #include <stdlib.h>
+#include <unistd.h>
 
 #include "Q4puGenericSignalSpy.h"
 
