--- lib/videofile.cpp.ffmpeg	2012-07-30 04:42:30.000000000 +0000
+++ lib/videofile.cpp	2012-07-30 04:43:24.000000000 +0000
@@ -58,7 +58,6 @@
     static bool init=false;
     if(!init) {
         av_register_all();
-        avcodec_init();
         avcodec_register_all();
         av_log_set_level(0);
         init=true;
@@ -66,7 +65,7 @@
 
     close();
 
-    if ( av_open_input_file(&avFromatContext, QFile::encodeName(filename).constData(), 0L, 0, 0L) != 0 ||
+    if ( avformat_open_input(&avFromatContext, QFile::encodeName(filename).constData(), 0L, 0L) != 0 ||
          av_find_stream_info(avFromatContext) < 0) {
         close();
     }
