Index: libdmapsharing-2.1.9/libdmapsharing/dpap-share.c
===================================================================
--- libdmapsharing-2.1.9.orig/libdmapsharing/dpap-share.c
+++ libdmapsharing-2.1.9/libdmapsharing/dpap-share.c
@@ -29,8 +29,7 @@
 #include <stdlib.h>
 
 #include <glib/gi18n.h>
-#include <glib/gslist.h>
-#include <glib/gmappedfile.h>
+#include <glib.h>
 
 #include <unistd.h>
 #include <sys/types.h>
