diff -Naur hidapi-hidapi-0.8.0-rc1.orig/configure.ac hidapi-hidapi-0.8.0-rc1/configure.ac
--- hidapi-hidapi-0.8.0-rc1.orig/configure.ac	2013-10-07 00:43:37.000000000 +0200
+++ hidapi-hidapi-0.8.0-rc1/configure.ac	2020-12-13 19:26:32.602903000 +0100
@@ -18,8 +18,7 @@
 lt_age="0"
 LTLDFLAGS="-version-info ${lt_current}:${lt_revision}:${lt_age}"
 
-AC_CONFIG_MACRO_DIR([m4])
-AM_INIT_AUTOMAKE([foreign -Wall -Werror])
+AM_INIT_AUTOMAKE([foreign -Wall])
 AC_CONFIG_MACRO_DIR([m4])
 
 m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
