diff -ur fribid-0.1.2.orig/client/Makefile fribid-0.1.2/client/Makefile
--- fribid-0.1.2.orig/client/Makefile	2010-03-24 00:32:27.000000000 +0100
+++ fribid-0.1.2/client/Makefile	2010-04-07 21:24:16.000000000 +0200
@@ -22,10 +22,10 @@
 
 CFLAGS ?= -O2 -g
 COMMONCFLAGS=$(CFLAGS) -Wall -Wextra -std=c99 -pedantic -Wno-unused-parameter
-CCFLAGS=$(COMMONCFLAGS) -I../npapi/include `pkg-config --cflags 'gtk+-2.0 >= 2.12' glib-2.0 'nss >= 3.10' nspr` -DGTK_DISABLE_DEPRECATED=1 -DGDK_DISABLE_DEPRECATED=1 -DG_DISABLE_DEPRECATED=1
+CCFLAGS=$(COMMONCFLAGS) -I../npapi/include `pkg-config --cflags 'gtk+-2.0 >= 2.12' glib-2.0 'nss >= 2.0' nspr` -DGTK_DISABLE_DEPRECATED=1 -DGDK_DISABLE_DEPRECATED=1 -DG_DISABLE_DEPRECATED=1
 # You may have to add -lpthread after $(LDFLAGS) on OpenBSD
 LINKFLAGS=$(CFLAGS) $(LDFLAGS) -Wl,--as-needed
-LIBS=`pkg-config --libs 'gtk+-2.0 >= 2.12' glib-2.0 'nss >= 3.10' nspr`
+LIBS=`pkg-config --libs 'gtk+-2.0 >= 2.12' glib-2.0 'nss >= 2.0' nspr`
 
 # Files to be installed
 LIB_PATH=`../configure --internal--get-define=LIB_PATH`
Endast i fribid-0.1.2/common: config.h
diff -ur fribid-0.1.2.orig/configure fribid-0.1.2/configure
--- fribid-0.1.2.orig/configure	2010-03-24 00:32:27.000000000 +0100
+++ fribid-0.1.2/configure	2010-04-07 21:25:17.000000000 +0200
@@ -171,7 +171,7 @@
 
 oldifs=IFS
 IFS=";"
-pkgconfigDeps="gtk+-2.0 >= 2.12;glib-2.0;nss >= 3.10;nspr;x11"
+pkgconfigDeps="gtk+-2.0 >= 2.12;glib-2.0;nss >= 2.0;nspr;x11"
 
 if ! pkg-config --exists $pkgconfigDeps; then
     echo "Error: Unsatisfied dependencies: " >&2
Endast i fribid-0.1.2/: configure~
