diff -Naur python-spidermonkey-0.0.5.orig/setup.py python-spidermonkey-0.0.5/setup.py
--- python-spidermonkey-0.0.5.orig/setup.py	2009-03-13 20:04:25.000000000 +0000
+++ python-spidermonkey-0.0.5/setup.py	2009-03-18 05:47:12.000000000 +0000
@@ -16,8 +16,7 @@
 import os
 import subprocess as sp
 import sys
-import ez_setup
-ez_setup.use_setuptools()
+import setuptools
 from setuptools import setup, Extension
 
 def find_sources(extensions=[".c", ".cpp"]):
