
Set the EXECDIR for nikto to /usr/lib64/nikto

--- nikto-2.1.5.orig/nikto.conf	2015-11-07 19:48:17.854292534 -0700
+++ nikto-2.1.5/nikto.conf	2015-11-07 19:50:49.311286043 -0700
@@ -61,7 +61,7 @@
 CHECKMETHODS=HEAD GET
 
 # If you want to specify the location of any of the files, specify them here
-# EXECDIR=/opt/nikto				# Location of Nikto
+EXECDIR=/usr/lib64/nikto				# Location of Nikto
 # PLUGINDIR=/opt/nikto/plugins			# Location of plugin dir
 # DBDIR=/opt/nikto/databases			# Location of database dir
 # TEMPLATEDIR=/opt/nikto/templates		# Location of template dir
