
Fix nikto.pl paths to use libwhisker2 from SBo.

--- nikto-2.1.5.orig/nikto.pl	2012-09-10 02:22:36.000000000 -0600
+++ nikto-2.1.5/nikto.pl	2015-11-07 22:14:55.380915446 -0700
@@ -56,14 +56,14 @@
 setup_dirs();
 require "$CONFIGFILE{'PLUGINDIR'}/nikto_core.plugin";
 nprint("T:" . localtime($COUNTERS{'scan_start'}) . ": Starting", "d");
-require "$CONFIGFILE{'PLUGINDIR'}/LW2.pm";
+#require "$CONFIGFILE{'PLUGINDIR'}/LW2.pm";
 require "$CONFIGFILE{'PLUGINDIR'}/JSON-PP.pm";
 $VARIABLES{'GMTOFFSET'} = gmt_offset();
 
-# use LW2;                   ### Change this line to use a different installed version
+use LW2;                   ### Change this line to use a different installed version
 
 #set SSL Engine
-LW2::init_ssl_engine($CONFIGFILE{'LW_SSL_ENGINE'});
+#LW2::init_ssl_engine($CONFIGFILE{'LW_SSL_ENGINE'});
 
 my ($a, $b) = split(/\./, $LW2::VERSION);
 die("- You must use LW2 2.4 or later\n") if ($a != 2 || $b < 4);
