--- clamav-unofficial-sigs.conf.orig	2013-09-16 18:58:03.738768005 -0500
+++ clamav-unofficial-sigs.conf	2013-09-16 18:59:41.609098853 -0500
@@ -31,6 +31,10 @@
 PATH="/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin"
 export PATH
 
+# Slackware Package Management Variables
+pkg_mgr="pkgtool"
+pkg_rm="removepkg clamav-unofficial-sigs"
+
 # Set the appropriate ClamD user and group accounts for your system.
 # If you do not want the script to set user and group permissions on
 # files and directories, comment the next two variables.
