--- clamav-unofficial-sigs.conf.orig	2009-05-11 08:07:00.739452000 -0500
+++ clamav-unofficial-sigs.conf	2009-05-11 08:08:44.213661407 -0500
@@ -34,6 +34,10 @@
 PATH="/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin"
 export PATH
 
+# Slackware package management info
+pkg_mgr="pkgtool"
+pkg_rm="removepkg clamav-unofficial-sigs"
+
 # Set the appropriate ClamD user and group accounts for your system.
 clam_user="clamav"
 clam_group="clamav"
