--- CMakeLists.txt.orig	2023-08-29 11:13:57.755000000 +1000
+++ CMakeLists.txt	2023-08-29 11:19:42.759000000 +1000
@@ -605,8 +605,14 @@
         if (NOT DEFINED OSD_CUDA_NVCC_FLAGS)
             if (CUDA_VERSION_MAJOR LESS 6)
                 set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_11 )
-            elseif (CUDA_VERSION_MAJOR LESS 8)
+            elseif (CUDA_VERSION_MAJOR LESS 9)
                 set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_20 )
+            elseif (CUDA_VERSION_MAJOR LESS 11)
+                set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_30 )
+            elseif (CUDA_VERSION_MAJOR LESS 12)
+                set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_35 )
+            else ()
+                set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_50 )
             endif()
         endif()
     endif()
