--- bumblebee-status-2.1.1/bumblebee_status/core/theme.py	2020-12-07 10:00:20.000000000 +0100
+++ bumblebee-status-2.1.1/bumblebee_status/core/theme.py.patched	2020-12-20 21:12:27.216695638 +0100
@@ -17,6 +17,7 @@
     os.path.join(THEME_BASE_DIR, "../../themes"),
     os.path.expanduser("~/.config/bumblebee-status/themes"),
     os.path.expanduser("~/.local/share/bumblebee-status/themes"),  # PIP
+    "/usr/share/bumblebee-status/themes",
 ]
 
 
