--- configure.orig	2025-07-31 14:11:26.463704405 +0300
+++ configure	2025-12-27 03:00:11.682420502 +0200
@@ -23935,7 +23935,7 @@
             as_fn_error $? "Illegal value -$with_smb_idmap_interface_version- for option --with-smb-idmap-interface-version" "$LINENO" 5
         fi
     else
-        sambalibdir="`$PKG_CONFIG --variable=libdir smbclient`"/samba
+        sambalibdir="`$PKG_CONFIG --variable=libdir smbclient`"
         { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking Samba's idmap library" >&5
 printf %s "checking Samba's idmap library... " >&6; }
         if test -f "${sambalibdir}/libidmap-private-samba.so"; then
