diff -Nur libtunepimp-0.5.3.orig/include/tunepimp-0.5/metadata.h libtunepimp-0.5.3/include/tunepimp-0.5/metadata.h
--- libtunepimp-0.5.3.orig/include/tunepimp-0.5/metadata.h	2006-11-18 04:52:08.000000000 -0600
+++ libtunepimp-0.5.3/include/tunepimp-0.5/metadata.h	2009-07-05 22:25:09.566603724 -0500
@@ -27,6 +27,7 @@
 #ifndef __METADATA_H__
 #define __METADATA_H__
 
+#include <string.h>
 #include <string>
 #include <cstdio>
 
diff -Nur libtunepimp-0.5.3.orig/lib/c_wrapper.cpp libtunepimp-0.5.3/lib/c_wrapper.cpp
--- libtunepimp-0.5.3.orig/lib/c_wrapper.cpp	2006-11-18 04:52:33.000000000 -0600
+++ libtunepimp-0.5.3/lib/c_wrapper.cpp	2009-07-05 22:25:09.558605363 -0500
@@ -24,6 +24,7 @@
    $Id: c_wrapper.cpp 8360 2006-08-07 21:05:12Z luks $
 
 ----------------------------------------------------------------------------*/
+#include <stdlib.h>
 #include "tunepimp.h"
 #include "mutex.h"
 #include "tp_c.h"
diff -Nur libtunepimp-0.5.3.orig/lib/fileio.cpp libtunepimp-0.5.3/lib/fileio.cpp
--- libtunepimp-0.5.3.orig/lib/fileio.cpp	2006-11-18 04:52:33.000000000 -0600
+++ libtunepimp-0.5.3/lib/fileio.cpp	2009-07-05 22:25:09.562578982 -0500
@@ -27,6 +27,8 @@
 
 #include <assert.h>
 #include <errno.h>
+#include <stdlib.h>
+#include <string.h>
 #include <string>
 #ifndef WIN32
 #include <unistd.h>
diff -Nur libtunepimp-0.5.3.orig/lib/metadata.cpp libtunepimp-0.5.3/lib/metadata.cpp
--- libtunepimp-0.5.3.orig/lib/metadata.cpp	2006-11-18 04:52:33.000000000 -0600
+++ libtunepimp-0.5.3/lib/metadata.cpp	2009-07-05 22:25:09.558605363 -0500
@@ -26,6 +26,7 @@
 ----------------------------------------------------------------------------*/
 #include <math.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include "metadata.h"
 #include "../config.h"
 using namespace std;
diff -Nur libtunepimp-0.5.3.orig/lib/protocol.cpp libtunepimp-0.5.3/lib/protocol.cpp
--- libtunepimp-0.5.3.orig/lib/protocol.cpp	2006-11-28 14:25:04.000000000 -0600
+++ libtunepimp-0.5.3/lib/protocol.cpp	2009-07-05 22:25:09.558605363 -0500
@@ -8,6 +8,7 @@
 -------------------------------------------------------------------*/
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <string>
 #include <map>
 #include <expat.h>
diff -Nur libtunepimp-0.5.3.orig/lib/readmeta.cpp libtunepimp-0.5.3/lib/readmeta.cpp
--- libtunepimp-0.5.3.orig/lib/readmeta.cpp	2006-11-18 04:52:33.000000000 -0600
+++ libtunepimp-0.5.3/lib/readmeta.cpp	2009-07-05 22:25:09.562578982 -0500
@@ -31,6 +31,7 @@
 #endif
 
 #include <stdio.h>
+#include <stdlib.h>
 #ifndef WIN32
 #include <unistd.h>
 #endif
diff -Nur libtunepimp-0.5.3.orig/lib/tunepimp.cpp libtunepimp-0.5.3/lib/tunepimp.cpp
--- libtunepimp-0.5.3.orig/lib/tunepimp.cpp	2006-11-18 04:52:33.000000000 -0600
+++ libtunepimp-0.5.3/lib/tunepimp.cpp	2009-07-05 22:25:09.558605363 -0500
@@ -30,7 +30,9 @@
 #endif
 #endif
 
+#include <string.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <map>
 using namespace std;
 
diff -Nur libtunepimp-0.5.3.orig/lib/utf8/utf8util.cpp libtunepimp-0.5.3/lib/utf8/utf8util.cpp
--- libtunepimp-0.5.3.orig/lib/utf8/utf8util.cpp	2006-11-18 04:52:26.000000000 -0600
+++ libtunepimp-0.5.3/lib/utf8/utf8util.cpp	2009-07-05 22:25:09.558605363 -0500
@@ -26,6 +26,7 @@
 ----------------------------------------------------------------------------*/
 
 #include <stdio.h>
+#include <stdlib.h>
 #include "utf8util.h"
 #include "utf8.h"
 #ifdef WIN32
diff -Nur libtunepimp-0.5.3.orig/plugins/mp3/id3_2_3_meta.cpp libtunepimp-0.5.3/plugins/mp3/id3_2_3_meta.cpp
--- libtunepimp-0.5.3.orig/plugins/mp3/id3_2_3_meta.cpp	2006-11-18 04:51:07.000000000 -0600
+++ libtunepimp-0.5.3/plugins/mp3/id3_2_3_meta.cpp	2009-07-05 22:25:09.562578982 -0500
@@ -26,6 +26,7 @@
 ----------------------------------------------------------------------------*/
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <assert.h>
 #include <ctype.h>
 #include <musicbrainz/mb_c.h>
diff -Nur libtunepimp-0.5.3.orig/plugins/mp3/id3_meta.cpp libtunepimp-0.5.3/plugins/mp3/id3_meta.cpp
--- libtunepimp-0.5.3.orig/plugins/mp3/id3_meta.cpp	2006-11-18 04:51:07.000000000 -0600
+++ libtunepimp-0.5.3/plugins/mp3/id3_meta.cpp	2009-07-05 22:25:09.562578982 -0500
@@ -26,6 +26,7 @@
 ----------------------------------------------------------------------------*/
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <time.h>
 #include <ctype.h>
 #include <musicbrainz/mb_c.h>
diff -Nur libtunepimp-0.5.3.orig/plugins/mp4/mp4.cpp libtunepimp-0.5.3/plugins/mp4/mp4.cpp
--- libtunepimp-0.5.3.orig/plugins/mp4/mp4.cpp	2006-11-18 04:51:08.000000000 -0600
+++ libtunepimp-0.5.3/plugins/mp4/mp4.cpp	2009-07-05 22:25:09.562578982 -0500
@@ -27,6 +27,7 @@
 
 #include <string.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <mp4.h>
 #include "metadata.h"
 #include "plugin.h"
diff -Nur libtunepimp-0.5.3.orig/plugins/wma/wma.cpp libtunepimp-0.5.3/plugins/wma/wma.cpp
--- libtunepimp-0.5.3.orig/plugins/wma/wma.cpp	2006-11-18 04:51:28.000000000 -0600
+++ libtunepimp-0.5.3/plugins/wma/wma.cpp	2009-07-05 22:25:09.562578982 -0500
@@ -27,6 +27,7 @@
 
 #include <string.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <wmafile.h>
 #include "metadata.h"
 #include "plugin.h"
diff -Nur libtunepimp-0.5.3.orig/plugins/wma/wmafile.cpp libtunepimp-0.5.3/plugins/wma/wmafile.cpp
--- libtunepimp-0.5.3.orig/plugins/wma/wmafile.cpp	2006-11-18 04:51:28.000000000 -0600
+++ libtunepimp-0.5.3/plugins/wma/wmafile.cpp	2009-07-05 22:25:09.566603724 -0500
@@ -19,6 +19,7 @@
  *   USA                                                                   *
  ***************************************************************************/
 
+#include <string.h>
 #include <tstring.h>
 #include <wmafile.h>
 #include <wmatag.h>
