--- a/lxqt-admin-user/usermanager.cpp
+++ b/lxqt-admin-user/usermanager.cpp
@@ -436,7 +436,12 @@
                 QByteArray line = file.readLine().trimmed();
                 if(line.isEmpty() || line.startsWith('#'))
                     continue;
-                mAvailableShells.append(QString::fromLocal8Bit(line));
+                QString shell = QString::fromLocal8Bit(line);
+                 if (shell.endsWith(QLatin1String("/bash")) ) {
+                    mAvailableShells.prepend(shell);
+                } else {
+                    mAvailableShells.append(shell);
+                }
             }
             file.close();
         }
