--- briquolo-0.5.7/desktop/briquolo.desktop	2006-03-07 21:31:26.000000000 +0100
+++ briquolo.desktop	2012-05-21 23:00:50.000000000 +0200
@@ -1,22 +1,12 @@
 [Desktop Entry]
-Version=1.0
 Encoding=UTF-8
 Name=Briquolo
-Name[fr]=Briquolo
-
-Comment=An OpenGL breakout
-Comment[fr]=Un casse brique OpenGL
-
-Type=Application
-Categories=ArcadeGame;Game;
-
-TryExec=briquolo
-Exec=briquolo
-FilePattern=briquolo
-
-Icon=briquolo.svg
+Icon=/usr/share/pixmaps/briquolo.svg
+GenericName=An OpenGL breakout clone
+GenericName[de_DE]=Ein OpenGL Breakout-Klon
+Exec=/usr/bin/briquolo
 Terminal=false
-
-GenericName=BreakOut
-GenericName[fr]=Casse Briques
-Categories=Application;Game;
+StartupNotify=false
+Type=Application
+TryExec=/usr/bin/supertuxkart
+Categories=Game;ArcadeGame;
\ Kein Zeilenumbruch am Dateiende.
