--- cal3d-0.11.0/src/cal3d/hardwaremodel.original.cpp	2014-04-03 06:06:40.944313033 +0300
+++ cal3d-0.11.0/src/cal3d/hardwaremodel.cpp	2014-04-03 06:07:02.795039862 +0300
@@ -10,6 +10,7 @@
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
+#include "string.h"
 #endif
 
 
--- cal3d-0.11.0/src/cal3d/platform.original.cpp	2014-04-03 06:09:37.926100456 +0300
+++ cal3d-0.11.0/src/cal3d/platform.cpp	2014-04-03 06:10:55.935125209 +0300
@@ -10,6 +10,7 @@
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
+#include "string.h"
 #endif
 
 //****************************************************************************//
--- cal3d-0.11.0/src/cal3d/renderer.original.cpp	2014-04-03 06:12:09.135210082 +0300
+++ cal3d-0.11.0/src/cal3d/renderer.cpp	2014-04-03 06:13:02.877538211 +0300
@@ -10,6 +10,7 @@
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
+#include "string.h"
 #endif
 
 //****************************************************************************//
--- cal3d-0.11.0/src/cal3d/submesh.original.cpp	2014-04-03 06:14:34.770389392 +0300
+++ cal3d-0.11.0/src/cal3d/submesh.cpp	2014-04-03 06:14:50.839188504 +0300
@@ -10,6 +10,7 @@
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
+#include "string.h"
 #endif
 
 #include "cal3d/submesh.h"
--- cal3d-0.11.0/src/cal3d_converter.original.cpp	2014-04-03 06:16:15.541129584 +0300
+++ cal3d-0.11.0/src/cal3d_converter.cpp	2014-04-03 06:16:38.281845286 +0300
@@ -10,6 +10,7 @@
 //****************************************************************************//
 
 #include "cal3d/cal3d.h"
+#include "string.h"
 
 #define SKELETON 0
 #define MESH 1
