diff -ur ucspi-tcp-0.88.old/rblsmtpd.c ucspi-tcp-0.88/rblsmtpd.c
--- ucspi-tcp-0.88.old/rblsmtpd.c	Sat Mar 18 09:18:42 2000
+++ ucspi-tcp-0.88/rblsmtpd.c	Mon Apr  1 09:26:07 2002
@@ -22,7 +22,7 @@
 }
 void usage(void)
 {
-  strerr_die1x(100,"rblsmtpd: usage: rblsmtpd [ -b ] [ -R ] [ -t timeout ] [ -r base ] [ -a base ] smtpd [ arg ... ]");
+  strerr_die1x(100,"rblsmtpd: usage: rblsmtpd -r base [ -b ] [ -R ] [ -t timeout ] [ -r base ] [ -a base ] smtpd [ arg ... ]");
 }
 
 char *ip_env;
@@ -190,7 +190,7 @@
   argv += optind;
   if (!*argv) usage();
 
-  if (flagwantdefaultrbl) rbl("rbl.maps.vix.com");
+  if (flagwantdefaultrbl) usage();
   if (decision >= 2) rblsmtpd();
 
   pathexec_run(*argv,argv,envp);
