--- Unvanquished-0.50.0/daemon/CMakeLists.txt.orig	2016-06-08 14:53:22.525135294 +0300
+++ Unvanquished-0.50.0/daemon/CMakeLists.txt	2016-06-08 17:10:11.316629219 +0300
@@ -343,32 +343,32 @@
     if (NOT EXISTS ${DEPS_DIR})
         get_filename_component(BASENAME ${DEPS_DIR} NAME)
         set(REMOTE "http://dl.unvanquished.net/deps/${BASENAME}${DEPS_EXT}")
-        message(STATUS "Downloading dependencies from '${REMOTE}'")
-        file(DOWNLOAD ${REMOTE} ${OBJ_DIR}/${BASENAME}${DEPS_EXT}
-            SHOW_PROGRESS
-            STATUS DOWNLOAD_RESULT
-            LOG DOWNLOAD_LOG
-        )
-        list(GET DOWNLOAD_RESULT 0 DOWNLOAD_STATUS)
-        list(GET DOWNLOAD_RESULT 1 DOWNLOAD_STRING)
-        if (NOT DOWNLOAD_STATUS EQUAL 0)
-            message(FATAL_ERROR "Error downloading '${REMOTE}':
-                Status code: ${DOWNLOAD_STATUS}
-                Error string: ${DOWNLOAD_STRING}
-                Download log: ${DOWNLOAD_LOG}"
-            )
-        endif()
-        message(STATUS "Download completed successfully")
+        #message(STATUS "Downloading dependencies from '${REMOTE}'")
+        #file(DOWNLOAD ${REMOTE} ${OBJ_DIR}/${BASENAME}${DEPS_EXT}
+        #    SHOW_PROGRESS
+        #    STATUS DOWNLOAD_RESULT
+        #    LOG DOWNLOAD_LOG
+        #)
+        #list(GET DOWNLOAD_RESULT 0 DOWNLOAD_STATUS)
+        #list(GET DOWNLOAD_RESULT 1 DOWNLOAD_STRING)
+        #if (NOT DOWNLOAD_STATUS EQUAL 0)
+        #    message(FATAL_ERROR "Error downloading '${REMOTE}':
+        #        Status code: ${DOWNLOAD_STATUS}
+        #        Error string: ${DOWNLOAD_STRING}
+        #        Download log: ${DOWNLOAD_LOG}"
+        #    )
+        #endif()
+        #message(STATUS "Download completed successfully")
 
         # Extract the downloaded archive
-        execute_process(
-            COMMAND ${CMAKE_COMMAND} -E tar xzf ${OBJ_DIR}/${BASENAME}${DEPS_EXT}
-            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/external_deps
-            RESULT_VARIABLE EXTRACT_RESULT
-        )
-        if (NOT EXTRACT_RESULT EQUAL 0)
-            message(FATAL_ERROR "Could not extract ${BASENAME}${DEPS_EXT}")
-        endif()
+        #execute_process(
+        #    COMMAND ${CMAKE_COMMAND} -E tar xzf ${OBJ_DIR}/${BASENAME}${DEPS_EXT}
+        #    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/external_deps
+        #    RESULT_VARIABLE EXTRACT_RESULT
+        #)
+        #if (NOT EXTRACT_RESULT EQUAL 0)
+        #    message(FATAL_ERROR "Could not extract ${BASENAME}${DEPS_EXT}")
+        #endif()
     endif()
 
     # Add to paths
