--- ./clang/lib/Driver/ToolChains/Gnu.cpp.orig	2024-03-06 13:59:35.328900971 -0600
+++ ./clang/lib/Driver/ToolChains/Gnu.cpp	2024-03-06 14:02:01.218892901 -0600
@@ -2490,6 +2490,7 @@
   static const char *const X86Triples[] = {
       "i586-linux-gnu",      "i686-linux-gnu",        "i686-pc-linux-gnu",
       "i386-redhat-linux6E", "i686-redhat-linux",     "i386-redhat-linux",
+      "i586-slackware-linux",  "i686-slackware-linux",
       "i586-suse-linux",     "i686-montavista-linux",
   };
 
