--- ./libcxx/test/tools/clang_tidy_checks/CMakeLists.txt.orig	2025-03-18 18:15:15.000000000 -0500
+++ ./libcxx/test/tools/clang_tidy_checks/CMakeLists.txt	2025-03-22 15:21:45.943189808 -0500
@@ -3,6 +3,7 @@
 # the system's libc++. Restoring these setting after testing fixes this issue.
 set(LLVM_DIR_SAVE ${LLVM_DIR})
 set(Clang_DIR_SAVE ${Clang_DIR})
+return()
 
 # Since the Clang C++ ABI is not stable the Clang libraries and clang-tidy
 # versions must match. Otherwise there likely will be ODR-violations. This had
