--- coolreader-cr3.2.49/cr3qt/src/settings.ui	2020-09-11 07:17:23.000000000 -0400
+++ settings.ui	2020-10-19 12:30:04.557739030 -0400
@@ -1566,7 +1566,7 @@
          </item>
          <item row="12" column="1">
           <widget class="QComboBox" name="cbFontShaping">
-           <property name="currentText">
+           <property name="currentText" stdset="0">
             <string>Simple (FreeType only, fastest)</string>
            </property>
            <item>
@@ -1733,8 +1733,8 @@
              <rect>
               <x>0</x>
               <y>0</y>
-              <width>393</width>
-              <height>620</height>
+              <width>387</width>
+              <height>625</height>
              </rect>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_9">
