diff -Naur brainparty.orig/main.cpp brainparty/main.cpp
--- brainparty.orig/main.cpp	2010-03-11 00:15:34.000000000 +0000
+++ brainparty/main.cpp	2010-04-03 22:35:45.000000000 +0000
@@ -26,6 +26,8 @@
 BPGame* Game;
 
 int main(int argc, char *argv[]) {
+	chdir("/usr/share/brainparty");
+
 	if ( SDL_Init(SDL_INIT_VIDEO | SDL_INIT_AUDIO) < 0 ) {
 		fprintf(stderr, "Unable to init SDL: %s\n", SDL_GetError());
 		exit(1);
