diff -rupN DevIL.orig/DevIL/src-IL/CMakeLists.txt DevIL/DevIL/src-IL/CMakeLists.txt
--- DevIL.orig/DevIL/src-IL/CMakeLists.txt	2017-01-01 16:03:56.000000000 -0800
+++ DevIL/DevIL/src-IL/CMakeLists.txt	2018-08-08 20:18:42.144194766 -0700
@@ -147,7 +147,7 @@ else(LIBSQUISH_FOUND)
 endif(LIBSQUISH_FOUND)
 
 # Sets the output folders
-set(LIBDIR "../lib/")
+set(LIBDIR "../lib64/")
 set(LIBDIR "${LIBDIR}${EX_PLATFORM_NAME}")
 if(UNICODE)
 	set(LIBDIR "${LIBDIR}/unicode")
@@ -248,14 +248,14 @@ configure_file( pkgconfig/IL.pc.cmake.in
 
 # Installation
 install (TARGETS IL
-    ARCHIVE DESTINATION lib
-    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib64
+    LIBRARY DESTINATION lib64
     RUNTIME DESTINATION bin
 )
 install (FILES ../include/IL/il.h DESTINATION include/IL)
 
 install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/IL.pc
-	DESTINATION lib/pkgconfig
+	DESTINATION lib64/pkgconfig
 )
 
diff -rupN DevIL.orig/DevIL/src-ILU/CMakeLists.txt DevIL/DevIL/src-ILU/CMakeLists.txt
--- DevIL.orig/DevIL/src-ILU/CMakeLists.txt	2017-01-01 16:03:56.000000000 -0800
+++ DevIL/DevIL/src-ILU/CMakeLists.txt	2018-08-08 20:18:20.040193100 -0700
@@ -58,7 +58,7 @@ configure_file( pkgconfig/ILU.pc.cmake.i
     ${CMAKE_CURRENT_BINARY_DIR}/ILU.pc @ONLY)
 
 # Sets the output folders
-set(LIBDIR "../lib/")
+set(LIBDIR "../lib64/")
 set(LIBDIR "${LIBDIR}${EX_PLATFORM_NAME}")
 if(UNICODE)
 	set(LIBDIR "${LIBDIR}/unicode")
@@ -74,14 +74,14 @@ endif(UNICODE)
 
 # Installation
 install (TARGETS ILU
-    ARCHIVE DESTINATION lib
-    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib64
+    LIBRARY DESTINATION lib64
     RUNTIME DESTINATION bin
 )
 install (FILES ../include/IL/ilu.h DESTINATION include/IL)
 
 install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/ILU.pc
-	DESTINATION lib/pkgconfig
+	DESTINATION lib64/pkgconfig
 )
 
diff -rupN DevIL.orig/DevIL/src-ILUT/CMakeLists.txt DevIL/DevIL/src-ILUT/CMakeLists.txt
--- DevIL.orig/DevIL/src-ILUT/CMakeLists.txt	2017-01-01 16:03:56.000000000 -0800
+++ DevIL/DevIL/src-ILUT/CMakeLists.txt	2018-08-08 20:18:32.104194009 -0700
@@ -56,7 +56,7 @@ configure_file( pkgconfig/ILUT.pc.cmake.
     ${CMAKE_CURRENT_BINARY_DIR}/ILUT.pc @ONLY)
 
 # Sets the output folders
-set(LIBDIR "../lib/")
+set(LIBDIR "../lib64/")
 set(LIBDIR "${LIBDIR}${EX_PLATFORM_NAME}")
 if(UNICODE)
 	set(LIBDIR "${LIBDIR}/unicode")
@@ -72,14 +72,14 @@ endif(UNICODE)
 
 # Installation
 install (TARGETS ILUT
-    ARCHIVE DESTINATION lib
-    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib64
+    LIBRARY DESTINATION lib64
     RUNTIME DESTINATION bin
 )
 install (FILES ../include/IL/ilut.h DESTINATION include/IL)
 
 install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/ILUT.pc
-	DESTINATION lib/pkgconfig
+	DESTINATION lib64/pkgconfig
 )
 
