--- share/applications/org.qt-project.qtcreator.desktop	2020-03-27 04:34:53.000000000 +0100
+++ share/applications/org.qt-project.qtcreator.desktop	2020-04-10 18:27:44.756614495 +0200
@@ -2,11 +2,10 @@
 Type=Application
 Exec=qtcreator %F
 Name=Qt Creator
-GenericName=C++ IDE for developing Qt applications
+GenericName=C/C++ and Qt IDE
 X-KDE-StartupNotify=true
 Icon=QtProject-qtcreator
 StartupWMClass=qtcreator
 Terminal=false
 Categories=Development;IDE;Qt;
-MimeType= text/x-c++src;text/x-c++hdr;text/x-xsrc;application/x-designer;application/vnd.qt.qmakeprofile;application/vnd.qt.xml.resource;
-
+MimeType=text/x-csrc;text/x-c++src;text/x-c++hdr;text/x-xsrc;text/x-python;application/x-designer;application/vnd.qt.qmakeprofile;application/vnd.qt.xml.resource;
