--- a/src/plugins/clangformat/clangformatutils.cpp	2019-09-23 22:09:30.394472064 +1000
+++ a/src/plugins/clangformat/clangformatutils.cpp	2019-09-23 22:11:54.426645315 +1000
@@ -60,7 +60,7 @@ 
     style.AllowShortBlocksOnASingleLine = false;
     style.AllowShortCaseLabelsOnASingleLine = false;
     style.AllowShortFunctionsOnASingleLine = FormatStyle::SFS_Inline;
-    style.AllowShortIfStatementsOnASingleLine = false;
+    style.AllowShortIfStatementsOnASingleLine = FormatStyle::SIS_Never;
     style.AllowShortLoopsOnASingleLine = false;
     style.AlwaysBreakAfterReturnType = FormatStyle::RTBS_None;
     style.AlwaysBreakBeforeMultilineStrings = false;
