--- src/shared/clang/clang_installation.pri.orig	2018-09-18 10:30:26.000000000 +0200
+++ src/shared/clang/clang_installation.pri	2018-09-25 20:25:05.873974281 +0200
@@ -180,6 +180,7 @@
     LLVM_CXXFLAGS ~= s,-fPIC,
     LLVM_CXXFLAGS ~= s,-pedantic,
     LLVM_CXXFLAGS ~= s,-Wstring-conversion,
+    LLVM_CXXFLAGS ~= s,-Werror=unguarded-availability-new,
     # split-dwarf needs objcopy which does not work via icecc out-of-the-box
     LLVM_CXXFLAGS ~= s,-gsplit-dwarf,
 
