diff -ur cpprestsdk-2.9.0.old/Release/src/CMakeLists.txt cpprestsdk-2.9.0/Release/src/CMakeLists.txt
--- cpprestsdk-2.9.0.old/Release/src/CMakeLists.txt	2016-10-18 17:30:28.000000000 -0200
+++ cpprestsdk-2.9.0/Release/src/CMakeLists.txt	2016-12-20 22:36:53.079208022 -0200
@@ -147,7 +147,7 @@
 
   install(
     TARGETS cpprest
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
     )
 endif()
