--- tests/Makefile.am~	2010-08-29 06:56:00.000000000 -0500
+++ tests/Makefile.am	2013-02-07 13:47:03.187506469 -0600
@@ -55,7 +55,7 @@
 	MODELS_DIR="$(srcdir)/models"
 
 TESTS = test-moniker$(EXEEXT) test-event-source$(EXEEXT) test-object$(EXEEXT) \
-	test-stream-mem$(EXEEXT) $(srcdir)/test-properties.sh test-storage-mem$(EXEEXT) \
+	test-stream-mem$(EXEEXT) tests/test-properties.sh test-storage-mem$(EXEEXT) \
 	test-main-loop$(EXEEXT)
 
 # TESTS += test-generic-factory.sh
