--- pdal/Geometry.cpp.orig	2023-04-13 12:07:49.000000000 -0600
+++ pdal/Geometry.cpp	2023-06-10 08:58:52.245292599 -0600
@@ -153,7 +153,7 @@
 
 bool Geometry::srsValid() const
 {
-    OGRSpatialReference *srs = m_geom->getSpatialReference();
+    const OGRSpatialReference *srs = m_geom->getSpatialReference();
     return srs && srs->GetRoot();
 }
 
@@ -172,7 +172,7 @@
         return StatusWithReason(-2,
             "Geometry::transform() failed.  NULL target SRS.");
 
-    OGRSpatialReference *inSrs = m_geom->getSpatialReference();
+    const OGRSpatialReference *inSrs = m_geom->getSpatialReference();
     SrsTransform transform(*inSrs, OGRSpatialReference(out.getWKT().data()));
     if (m_geom->transform(transform.get()) != OGRERR_NONE)
         return StatusWithReason(-1, "Geometry::transform() failed.");
