--- Makefile.orig	2006-08-21 11:13:12.000000000 -0500
+++ Makefile	2006-08-21 14:58:59.000000000 -0500
@@ -5,13 +5,12 @@
 
 install: all
 	mkdir -p $(DESTDIR)/include
-	install -m 644 -o root -g root inklevel.h $(DESTDIR)/include/inklevel.h
+	install -m 644 inklevel.h $(DESTDIR)/include/inklevel.h
 	mkdir -p $(DESTDIR)/lib
-	install -m 755 -o root -g root libinklevel.so.2.0.6.5 \
+	install -m 755 libinklevel.so.2.0.6.5 \
 	$(DESTDIR)/lib/libinklevel.so.2.0.6.5
 	cd $(DESTDIR)/lib && rm -fr libinklevel.so && \
 	ln -s libinklevel.so.2.0.6.5 libinklevel.so
-	ldconfig $(DESTDIR)/lib
 
 clean:
 	rm -fr *.o *.so.2.0.6.5 *~
