diff -ur proxygen-eb6235bf1f9c36756934b6619bf82a4070828288.old/proxygen/lib/test/Makefile.am proxygen-eb6235bf1f9c36756934b6619bf82a4070828288/proxygen/lib/test/Makefile.am
--- proxygen-eb6235bf1f9c36756934b6619bf82a4070828288.old/proxygen/lib/test/Makefile.am	2016-12-30 17:26:15.000000000 -0200
+++ proxygen-eb6235bf1f9c36756934b6619bf82a4070828288/proxygen/lib/test/Makefile.am	2017-01-04 23:48:53.053074604 -0200
@@ -2,11 +2,6 @@
 
 BUILT_SOURCES = googletest-release-1.8.0/googletest/src/gtest-all.cc
 
-googletest-release-1.8.0/googletest/src/gtest-all.cc:
-	wget https://github.com/google/googletest/archive/release-1.8.0.zip
-	unzip release-1.8.0.zip
-
-
 check_LTLIBRARIES = libtesttransport.la
 libtesttransport_la_SOURCES = TestAsyncTransport.cpp
 
