diff -Naur alsaequal-orig/Makefile alsaequal/Makefile
--- alsaequal-orig/Makefile	2010-03-05 17:30:39.000000000 +0000
+++ alsaequal/Makefile	2011-01-17 22:13:10.156344120 +0000
@@ -45,11 +45,11 @@
 
 install: all
 	@echo Installing...
-	$(Q)install -m 644 $(SND_PCM_BIN) ${DESTDIR}/usr/lib/alsa-lib/
-	$(Q)install -m 644 $(SND_CTL_BIN) ${DESTDIR}/usr/lib/alsa-lib/
+	$(Q)install -m 644 $(SND_PCM_BIN) ${DESTDIR}/usr/lib@LIBDIRSUFFIX@/alsa-lib/
+	$(Q)install -m 644 $(SND_CTL_BIN) ${DESTDIR}/usr/lib@LIBDIRSUFFIX@/alsa-lib/
 
 uninstall:
 	@echo Un-installing...
-	$(Q)rm ${DESTDIR}/usr/lib/alsa-lib/$(SND_PCM_BIN)
-	$(Q)rm ${DESTDIR}/usr/lib/alsa-lib/$(SND_CTL_BIN)
+	$(Q)rm ${DESTDIR}/usr/lib@LIBDIRSUFFIX@/alsa-lib/$(SND_PCM_BIN)
+	$(Q)rm ${DESTDIR}/usr/lib@LIBDIRSUFFIX@/alsa-lib/$(SND_CTL_BIN)
 	
diff -Naur alsaequal-orig/ctl_equal.c alsaequal/ctl_equal.c
--- alsaequal-orig/ctl_equal.c	2009-01-31 22:06:47.000000000 +0000
+++ alsaequal/ctl_equal.c	2011-01-17 23:08:54.273018904 +0000
@@ -166,7 +166,7 @@
 	snd_config_iterator_t it, next;
 	snd_ctl_equal_t *equal;
 	const char *controls = ".alsaequal.bin";
-	const char *library = "/usr/lib/ladspa/caps.so";
+	const char *library = "/usr/lib@LIBDIRSUFFIX@/ladspa/caps.so";
 	const char *module = "Eq";
 	long channels = 2;
 	const char *sufix = " Playback Volume";
diff -Naur alsaequal-orig/pcm_equal.c alsaequal/pcm_equal.c
--- alsaequal-orig/pcm_equal.c	2010-02-01 22:55:00.000000000 +0000
+++ alsaequal/pcm_equal.c	2011-01-17 23:07:56.424897961 +0000
@@ -150,7 +150,7 @@
 	snd_pcm_equal_t *equal;
 	snd_config_t *sconf = NULL;
 	const char *controls = ".alsaequal.bin";
-	const char *library = "/usr/lib/ladspa/caps.so";
+	const char *library = "/usr/lib@LIBDIRSUFFIX@/ladspa/caps.so";
 	const char *module = "Eq";
 	long channels = 2;
 	int err;
