--- INSTALL.vsm	1998/06/15 10:53:16	1.1
+++ INSTALL.vsm	2017/02/20 08:47:19
@@ -43,7 +43,7 @@
 3. Using sendmail's delivery agent
 
 sendmail uses binmail to deliver to /var/spool/mail. binmail is shipped
-with the operating system as /bin/mail or /usr/libexec/mail.local.
+with the operating system as /bin/mail or /usr/sbin/mail.local.
 
 There is some variation in binmail syntax among systems. The most common
 interfaces are shown in /var/qmail/boot/binm1, /var/qmail/boot/binm2,
--- REMOVE.binmail	1998/06/15 10:53:16	1.1
+++ REMOVE.binmail	2017/02/20 08:47:19
@@ -2,11 +2,11 @@
 configured qmail to use binmail for local delivery.
 
 
-1. Find the binmail binary on your system: /usr/libexec/mail.local if
+1. Find the binmail binary on your system: /usr/sbin/mail.local if
    that exists, otherwise /bin/mail.
 
 2. Remove permissions from the binmail binary:
-      # chmod 0 /usr/libexec/mail.local
+      # chmod 0 /usr/sbin/mail.local
 
 3. If the binmail binary was /bin/mail, make sure that ``mail'' still
    invokes a usable mailer. Under SVR4 you may want to link mail to
--- binm1+df.sh	1998/06/15 10:53:16	1.1
+++ binm1+df.sh	2017/02/20 08:47:19
@@ -3,9 +3,9 @@
 # Using splogger to send the log through syslog.
 # Using dot-forward to support sendmail-style ~/.forward files.
 # Using binmail to deliver messages to /var/spool/mail/$USER by default.
-# Using BSD 4.4 binmail interface: /usr/libexec/mail.local -r
+# Using BSD 4.4 binmail interface: /usr/sbin/mail.local -r
 
 exec env - PATH="QMAIL/bin:$PATH" \
 qmail-start '|dot-forward .forward
-|preline -f /usr/libexec/mail.local -r "${SENDER:-MAILER-DAEMON}" -d "$USER"' \
+|preline -f /usr/sbin/mail.local -r "${SENDER:-MAILER-DAEMON}" -d "$USER"' \
 splogger qmail
--- binm1.sh	1998/06/15 10:53:16	1.1
+++ binm1.sh	2017/02/20 08:47:19
@@ -2,9 +2,9 @@
 
 # Using splogger to send the log through syslog.
 # Using binmail to deliver messages to /var/spool/mail/$USER by default.
-# Using BSD 4.4 binmail interface: /usr/libexec/mail.local -r
+# Using BSD 4.4 binmail interface: /usr/sbin/mail.local -r
 
 exec env - PATH="QMAIL/bin:$PATH" \
 qmail-start \
-'|preline -f /usr/libexec/mail.local -r "${SENDER:-MAILER-DAEMON}" -d "$USER"' \
+'|preline -f /usr/sbin/mail.local -r "${SENDER:-MAILER-DAEMON}" -d "$USER"' \
 splogger qmail
