--- QtDMM-0.9.7/src/src.pro	2016-10-30 04:37:47.000000000 -0400
+++ src.pro	2025-03-16 16:16:07.060280105 -0400
@@ -17,7 +17,7 @@
 QMAKE_EXTRA_COMPILERS += lrelease
 lrelease.input         = TRANSLATIONS
 lrelease.output        = ${QMAKE_FILE_BASE}.qm
-lrelease.commands      = $$[QT_INSTALL_BINS]/lrelease-qt5 ${QMAKE_FILE_IN} -qm translations/${QMAKE_FILE_BASE}.qm
+lrelease.commands      = $$[QT_INSTALL_BINS]/lrelease ${QMAKE_FILE_IN} -qm translations/${QMAKE_FILE_BASE}.qm
 lrelease.CONFIG       += no_link target_predeps
 
 FORMS =      forms/uiconfigdlg.ui \
