diff -Naur kernel.org/conftest.sh kernel/conftest.sh
--- kernel.org/conftest.sh	2024-05-02 22:25:33.000000000 +0800
+++ kernel/conftest.sh	2024-07-26 21:24:43.434506976 +0800
@@ -101,7 +101,8 @@
 build_cflags() {
     BASE_CFLAGS="-O2 -D__KERNEL__ \
 -DKBUILD_BASENAME=\"#conftest$$\" -DKBUILD_MODNAME=\"#conftest$$\" \
--nostdinc -isystem $ISYSTEM"
+-nostdinc -isystem $ISYSTEM -fshort-wchar \
+-Wno-implicit-function-declaration -Wno-strict-prototypes"
 
     if [ "$OUTPUT" != "$SOURCES" ]; then
         OUTPUT_CFLAGS="-I$OUTPUT/include2 -I$OUTPUT/include"
