diff -Naur libofa-0.9.3.orig/examples/uselame.cpp libofa-0.9.3/examples/uselame.cpp
--- libofa-0.9.3.orig/examples/uselame.cpp	2006-05-10 20:05:51.000000000 +0200
+++ libofa-0.9.3/examples/uselame.cpp	2012-03-30 06:19:44.072989020 +0200
@@ -6,6 +6,8 @@
    No rights reserved.
 
 -------------------------------------------------------------------*/
+#include <string.h>
+#include <unistd.h>
 #include "protocol.h"
 #ifdef WIN32
 #include "windows.h"
diff -Naur libofa-0.9.3.orig/examples/wavefile.cpp libofa-0.9.3/examples/wavefile.cpp
--- libofa-0.9.3.orig/examples/wavefile.cpp	2006-05-10 20:05:56.000000000 +0200
+++ libofa-0.9.3/examples/wavefile.cpp	2012-03-30 06:20:11.808008983 +0200
@@ -6,6 +6,7 @@
    No rights reserved.
 
 -------------------------------------------------------------------*/
+#include <unistd.h>
 #include "protocol.h"
 #ifdef WIN32
 #include "io.h"
