--- conf/ifplugd.action	2022-10-21 04:32:02.560985145 +1100
+++ conf/ifplugd.action	2022-10-21 04:37:32.173998517 +1100
@@ -24,7 +24,13 @@
 	exit 1
 fi
 
-[ "$2" = "up" ] && exec /sbin/ifup $1
-[ "$2" = "down" ] && exec /sbin/ifdown $1
+# Place custom actions/commands/scripts below. The interface is passed
+# through "$1", and the state of interface is passed via "$2".
+# i.e.  if [ "$1" = "eth0" ] && [ "$2" = "down" ]; then
+
+# This script originally used ifup and ifdown which Slackware does
+# not include by default.  i.e. -
+#[ "$2" = "up" ] && exec /sbin/ifup $1
+#[ "$2" = "down" ] && exec /sbin/ifdown $1
 
 exit 1
