diff -Naur penguin-command-1.6.11/penguin-command.6 penguin-command-1.6.11.patched/penguin-command.6
--- penguin-command-1.6.11/penguin-command.6	2007-03-18 11:21:13.000000000 -0400
+++ penguin-command-1.6.11.patched/penguin-command.6	2014-04-17 19:15:48.000000000 -0400
@@ -16,10 +16,10 @@
 .SH OPTIONS
 .TP
 .B \-\-fullscreen, \-f
-Run in fullscreen mode (default).
+Run in fullscreen mode.
 .TP
 .B \-\-windowed, \-w
-Run in windowed mode.
+Run in windowed mode (default).
 .TP
 .B \-\-nosound, \-s
 Run without sound.
diff -Naur penguin-command-1.6.11/src/main.c penguin-command-1.6.11.patched/src/main.c
--- penguin-command-1.6.11/src/main.c	2007-03-18 11:21:13.000000000 -0400
+++ penguin-command-1.6.11.patched/src/main.c	2014-04-17 19:15:05.000000000 -0400
@@ -32,7 +32,7 @@
 #endif
 
 
-    int fullscreen=1, CityDestroyed[6]={0,0,0,0,0,0},
+    int fullscreen=0, CityDestroyed[6]={0,0,0,0,0,0},
 	CannonDestroyed[3]={0,0,0}, Score=0, HWCursor=0, NoAlpha=0, ArcadeMode=0, MusicVol=10, SoundVol=10;
 //images
     SDL_Surface *ShotPic,*CityPic,*BackPic,*CityHitPic,*Cannon[21],*Font,
