--- Makefile.orig	2012-11-03 21:13:55.065215509 +0100
+++ Makefile	2012-11-03 21:14:13.244222002 +0100
@@ -90,7 +90,7 @@
 
 $(TARBALL):
 	mkdir -p $(deb_destdir)
-	tmp_dir=`mktemp -d --tmpdir dkms.XXXXXXXX` ; \
+	tmp_dir=`mktemp -d -t dkms.XXXXXXXX` ; \
 	cp -a ../$(RELEASE_NAME) $${tmp_dir}/$(RELEASE_STRING) ; \
 	sed -e "s/\[INSERT_VERSION_HERE\]/$(RELEASE_VERSION)/" dkms > $${tmp_dir}/$(RELEASE_STRING)/dkms ; \
 	sed -e "s/\[INSERT_VERSION_HERE\]/$(RELEASE_VERSION)/" dkms.spec > $${tmp_dir}/$(RELEASE_STRING)/dkms.spec ; \
@@ -107,7 +107,7 @@
 
 
 rpm: $(TARBALL) dkms.spec
-	tmp_dir=`mktemp -d --tmpdir dkms.XXXXXXXX` ; \
+	tmp_dir=`mktemp -d -t dkms.XXXXXXXX` ; \
 	mkdir -p $${tmp_dir}/{BUILD,RPMS,SRPMS,SPECS,SOURCES} ; \
 	cp $(TARBALL) $${tmp_dir}/SOURCES ; \
 	sed "s/\[INSERT_VERSION_HERE\]/$(RELEASE_VERSION)/" dkms.spec > $${tmp_dir}/SPECS/dkms.spec ; \
@@ -131,7 +131,7 @@
 	cd -
 
 debs:
-	tmp_dir=`mktemp -d --tmpdir dkms.XXXXXXXX` ; \
+	tmp_dir=`mktemp -d -t dkms.XXXXXXXX` ; \
 	make debmagic DEB_TMP_BUILDDIR=$${tmp_dir} DIST=$(DIST); \
 	rm -rf $${tmp_dir}
 
