Index: ufraw-0.22/dcraw.cc
===================================================================
--- ufraw-0.22.orig/dcraw.cc
+++ ufraw-0.22/dcraw.cc
@@ -9240,9 +9240,9 @@ canon_a5:
       filters = 0x16161616;
     }
     if (make[0] == 'O') {
-      i = find_green (12, 32, 1188864, 3576832);
-      c = find_green (12, 32, 2383920, 2387016);
-      if (abs(i) < abs(c)) {
+      i = abs(find_green (12, 32, 1188864, 3576832));
+      c = abs(find_green (12, 32, 2383920, 2387016));
+      if (i < c) {
 	SWAP(i,c);
 	load_flags = 24;
       }
