diff -Naur kbgoffice-2.1.orig/src/translator_manager.cpp kbgoffice-2.1/src/translator_manager.cpp
--- kbgoffice-2.1.orig/src/translator_manager.cpp	2010-04-30 14:37:42.000000000 +0200
+++ kbgoffice-2.1/src/translator_manager.cpp	2017-06-01 18:10:20.308920738 +0200
@@ -141,11 +141,11 @@
                 isDataOk = isDataOk && tr->init(tr->BG_EN, string(string(dataDir) + prop->getString("data")).c_str());
         } else {
                 cerr << "TranslatorManager::getTestDictionaryObject - Wrong description file: " << testDictionaries[index].file << "\n";
-                return false;
+                return NULL;
         }
         if (!isDataOk) {
                 cerr << "TranslatorManager::getTestDictionaryObject - Problem initialazing dictionary: " << testDictionaries[index].file << "\n";
-                return false;
+                return NULL;
         }
         char c[16];
         sprintf(c, "%d", level);
