diff -Naur hexe-20120305/hexe.c hexe-20120305.patched/hexe.c
--- hexe-20120305/hexe.c	2012-03-05 19:08:11.000000000 -0500
+++ hexe-20120305.patched/hexe.c	2020-11-16 15:55:21.419107371 -0500
@@ -63,7 +63,7 @@
                         break;
                     }
                 }
-                else if(!strcmp(argv[i], "-c"))
+                else if(!strcmp(argv[i], "-b"))
                 {
                     sscanf(argv[++i], "%i", &bytegroup);
                     goto bytegroup_check;
