--- statifier-1.6.15/configs/config.x86_64	2005-02-16 09:06:35.000000000 -0200
+++ statifier-1.6.15_patched/configs/config.x86_64	2009-12-25 23:42:05.000000000 -0200
@@ -9,7 +9,7 @@
 
 # ELF32
    # Have elf32 ?
-   ELF32             := yes
+   ELF32             := no
    # Special flags to compile elf32
    FLAGS_32          := -m32
    # Flags to compile C to asm elf32 (mostly to be sure calling convention)
