diff -Nur abcde-2.3.3.orig/Makefile abcde-2.3.3/Makefile
--- abcde-2.3.3.orig/Makefile	2005-08-25 17:43:27.000000000 -0500
+++ abcde-2.3.3/Makefile	2007-02-01 10:21:55.000000000 -0600
@@ -4,7 +4,7 @@
 # Installation directories
 prefix = ${DESTDIR}/usr
 exec_prefix = ${prefix}
-mandir = ${prefix}/share/man/man1
+mandir = ${prefix}/man/man1
 bindir = ${exec_prefix}/bin
 etcdir = ${DESTDIR}/etc
 
@@ -14,13 +14,13 @@
 
 install:
 	$(INSTALL) -d -m 755 $(bindir)
-	$(INSTALL) -m 755 -o 0 abcde $(bindir)
-	$(INSTALL) -m 755 -o 0 cddb-tool $(bindir)
+	$(INSTALL) -m 755 abcde $(bindir)
+	$(INSTALL) -m 755 cddb-tool $(bindir)
 	$(INSTALL) -d -m 755 $(mandir)
-	$(INSTALL) -m 644 -o 0 abcde.1 $(mandir)
-	$(INSTALL) -m 644 -o 0 cddb-tool.1 $(mandir)
+	$(INSTALL) -m 644 abcde.1 $(mandir)
+	$(INSTALL) -m 644 cddb-tool.1 $(mandir)
 	$(INSTALL) -d -m 755 $(etcdir)
-	$(INSTALL) -m 644 -o 0 abcde.conf $(etcdir)
+	$(INSTALL) -m 644 abcde.conf $(etcdir)
 
 tarball:
 	@cd .. && tar czvf abcde_$(VERSION).orig.tar.gz \
