--- Makefile.org	2012-04-24 22:51:05.485128589 -0700
+++ Makefile	2012-04-24 22:53:29.216123458 -0700
@@ -91,15 +91,15 @@
 install_lib:
 	mkdir -p $(PREFIX)/include/ahven
 	mkdir -p $(LIBDIR)/ahven
-	mkdir -p $(PREFIX)/lib/gnat
+	mkdir -p $(LIBDIR)/gnat
 	$(INSTALL) -m 644 $(SOURCES) $(PREFIX)/include/ahven
 	$(INSTALL) -m 444 $(ALI_FILES) $(LIBDIR)/ahven
 	$(INSTALL) -m 644 lib/$(STATIC_LIBRARY) $(LIBDIR)/ahven
-	$(INSTALL) -m 644 $(GPR_FILE) $(PREFIX)/lib/gnat
+	$(INSTALL) -m 644 $(GPR_FILE) $(LIBDIR)/gnat
 
 install_docs:
-	mkdir -p $(PREFIX)/share/doc/ahven
-	cp -r doc/manual/en/build/html $(PREFIX)/share/doc/ahven
+	mkdir -p $(PREFIX)/doc/ahven
+	cp -r doc/manual/en/build/html $(PREFIX)/doc/ahven
 
 check: build_tests
 	./tester -c
