diff -Nur ming-0.4.2.orig/php_ext/Makefile.in ming-0.4.2/php_ext/Makefile.in
--- ming-0.4.2.orig/php_ext/Makefile.in	2008-09-17 19:59:59.000000000 -0500
+++ ming-0.4.2/php_ext/Makefile.in	2009-01-05 09:24:44.012689279 -0600
@@ -359,7 +359,7 @@
 
 install: all
 	$(INSTALL) -d "$(DESTDIR)"/`php-config --extension-dir`
-	$(INSTALL) -m 755 tmp/modules/ming.so" "$(DESTDIR)"/`php-config --extension-dir`
+	$(INSTALL) -m 755 tmp/modules/ming.so "$(DESTDIR)"/`php-config --extension-dir`
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
