Only in restbed-4.0: .CMakeLists.txt.un~
diff -ur restbed-4.0.old/CMakeLists.txt restbed-4.0/CMakeLists.txt
--- restbed-4.0.old/CMakeLists.txt	2016-04-14 18:38:26.000000000 -0300
+++ restbed-4.0/CMakeLists.txt	2017-01-03 18:15:29.883784733 -0200
@@ -71,5 +71,5 @@
 install( FILES "${INCLUDE_DIR}/${PROJECT_NAME}" DESTINATION "include" )
 install( FILES ${BUILD_ARTIFACTS} DESTINATION "include/corvusoft/${PROJECT_NAME}" )
 install( TARGETS ${PROJECT_NAME}
-         LIBRARY DESTINATION "library"
-         ARCHIVE DESTINATION "library" COMPONENT library )
+	LIBRARY DESTINATION "lib${LIB_SUFFIX}"
+	ARCHIVE DESTINATION "lib${LIB_SUFFIX}" COMPONENT library )
