--- amavisd-signer.orig	2014-11-08 21:02:36.183683393 -0600
+++ amavisd-signer	2014-11-08 21:04:50.079221328 -0600
@@ -86,11 +86,11 @@
 # Please adjust the following settings as necessary:
 #
 
-$daemon_user  = 'vscan';
-$daemon_group = 'vscan';
+$daemon_user  = 'amavis';
+$daemon_group = 'amavis';
 # $daemon_chroot_dir = '/var/amavis';   # chroot directory or undef
 
-# $daemonize = 1;
+$daemonize = 1;
 
 $log_level = 2;  # 0..5
 $syslog_facility = 'mail';
