--- amavisd-snmp-subagent.orig	2014-11-08 20:58:13.439786270 -0600
+++ amavisd-snmp-subagent	2014-11-08 21:01:24.141242836 -0600
@@ -69,7 +69,7 @@
 $syslog_facility = LOG_MAIL;
 
 my($db_home) =  # DB databases directory
-  defined $ENV{'AMAVISD_DB_HOME'} ? $ENV{'AMAVISD_DB_HOME'} : '/var/amavis/db';
+  defined $ENV{'AMAVISD_DB_HOME'} ? $ENV{'AMAVISD_DB_HOME'} : '/var/lib/amavis/db';
 
 my($mta_queue_dir);
 
