--- amavisd-submit.orig	2014-11-08 20:54:37.766290064 -0600
+++ amavisd-submit	2014-11-08 20:55:19.205752430 -0600
@@ -72,9 +72,9 @@
 ### USER CONFIGURABLE:
 
   $log_level = 0;
-  $tempbase = '/var/amavis/tmp';  # where to create a temp directory with a msg
+  $tempbase = '/var/lib/amavis/tmp';  # where to create a temp directory with a msg
 
-  $socketname = '/var/amavis/amavisd.sock';
+  $socketname = '/var/run/amavis/amavisd.sock';
 # $socketname = '127.0.0.1:9998';
 # $socketname = '[::1]:9998';
 
