--- amavisd-agent.orig	2014-11-08 20:48:15.953806621 -0600
+++ amavisd-agent	2014-11-08 20:48:21.655008087 -0600
@@ -53,7 +53,7 @@
 
 my($dbfile) = 'snmp.db';
 my($db_home) =  # DB databases directory
-  defined $ENV{'AMAVISD_DB_HOME'} ? $ENV{'AMAVISD_DB_HOME'} : '/var/amavis/db';
+  defined $ENV{'AMAVISD_DB_HOME'} ? $ENV{'AMAVISD_DB_HOME'} : '/var/lib/amavis/db';
 
 my($wakeuptime) = 10;  # -w, sleep time in seconds, may be fractional
 my($repeatcount);      # -c, repeat count (when defined)
