--- amavisd-signer.orig	2016-04-26 23:00:03.028973260 -0500
+++ amavisd-signer	2016-04-26 23:05:54.207764264 -0500
@@ -88,11 +88,11 @@
 # Please adjust the following settings as necessary:
 #
 
-$daemon_user   = 'vscan';
-@daemon_groups = 'vscan';
+$daemon_user   = 'amavis';
+@daemon_groups = 'amavis';
 # $daemon_chroot_dir = '/var/amavis';   # chroot directory or undef
 
-# $daemonize = 1;
+$daemonize = 1;
 
 $log_level = 2;  # 0..5
 $syslog_facility = 'mail';
