diff -ur suphp-0.7.1.orig/src/Application.cpp suphp-0.7.1/src/Application.cpp
--- suphp-0.7.1.orig/src/Application.cpp	2009-03-14 18:55:25.000000000 +0100
+++ suphp-0.7.1/src/Application.cpp	2009-04-05 13:03:10.000000000 +0200
@@ -174,12 +174,13 @@
         throw SecurityException("Calling user is not webserver user!",
                                 __FILE__, __LINE__);
     }
-
+#if 0
     if (!api.getEffectiveProcessUser().isSuperUser()) {
         throw SecurityException(
             "Do not have root privileges. Executable not set-uid root?",
             __FILE__, __LINE__);
     }
+#endif
 }
 
 
