diff -Naur elinks-git20131231/doc/man/man1/elinks.1.in elinks-git20131231.patched/doc/man/man1/elinks.1.in
--- elinks-git20131231/doc/man/man1/elinks.1.in	2013-12-18 10:38:57.000000000 -0500
+++ elinks-git20131231.patched/doc/man/man1/elinks.1.in	2020-06-25 03:02:54.523412259 -0400
@@ -141,12 +141,13 @@
 Look up specified host and print all DNS resolved IP addresses\&.
 .RE
 .PP
-\fB\-no\-connect\fR \fI[0|1]\fR (default: 0)
+\fB\-no\-connect\fR \fI[0|1]\fR (default: 1)
 .RS 4
 Run
 ELinks
-as a separate instance instead of connecting to an existing instance\&. Note that normally no runtime state files (bookmarks, history, etc\&.) are written to the disk when this option is used\&. See also
-\-touch\-files\&.
+as a separate instance instead of connecting to an existing instance\&. Enabled
+by default in this build; use \fB\-no\-connect 0\fR to connect to an existing
+instance\&.
 .RE
 .PP
 \fB\-no\-home\fR \fI[0|1]\fR (default: 0)
@@ -238,13 +239,13 @@
 Print given URLs in source form to stdout\&.
 .RE
 .PP
-\fB\-touch\-files\fR \fI[0|1]\fR (default: 0)
+\fB\-touch\-files\fR \fI[0|1]\fR (default: 1)
 .RS 4
 When enabled, runtime state files (bookmarks, history, etc\&.) are written to disk, even when
 \-no\-connect
 or
 \-session\-ring
-is used\&. The option has no effect if not used in conjunction with any of these options\&.
+is used\&. The option has no effect if not used in conjunction with any of these options\&. This option is enabled by default; use \fB\-touch\-files 0\fR to disable\&.
 .RE
 .PP
 \fB\-verbose\fR \fI<num>\fR (default: 1)
diff -Naur elinks-git20131231/src/config/cmdline.c elinks-git20131231.patched/src/config/cmdline.c
--- elinks-git20131231/src/config/cmdline.c	2013-12-18 10:38:57.000000000 -0500
+++ elinks-git20131231.patched/src/config/cmdline.c	2020-06-25 02:58:31.595442844 -0400
@@ -848,11 +848,9 @@
 		"addresses.")),
 
 	INIT_OPT_BOOL("", N_("Run as separate instance"),
-		"no-connect", 0, 0,
+		"no-connect", 0, 1,
 		N_("Run ELinks as a separate instance instead of connecting "
-		"to an existing instance. Note that normally no runtime state "
-		"files (bookmarks, history, etc.) are written to the disk "
-		"when this option is used. See also -touch-files.")),
+		"to an existing instance. Enabled by default in this build.")),
 
 	INIT_OPT_BOOL("", N_("Disable use of files in ~/.elinks"),
 		"no-home", 0, 0,
@@ -921,7 +919,7 @@
 	INIT_OPT_COMMAND("", NULL, "stdin", OPT_HIDDEN, redir_cmd, NULL),
 
 	INIT_OPT_BOOL("", N_("Touch files in ~/.elinks when running with -no-connect/-session-ring"),
-		"touch-files", 0, 0,
+		"touch-files", 0, 1,
 		N_("When enabled, runtime state files (bookmarks, history, "
 		"etc.) are written to disk, even when -no-connect or "
 		"-session-ring is used. The option has no effect if not used "
