--- CMakeLists.txt.old	2025-07-06 20:24:13.735435385 +0900
+++ CMakeLists.txt	2025-07-06 20:25:00.055892785 +0900
@@ -2,12 +2,13 @@
 # Originally created for "roxlu build system" to compile libyuv on windows
 # Run with -DUNIT_TEST=ON to build unit tests
 
-include(CheckCSourceCompiles)
+cmake_minimum_required( VERSION 3.16 )
 
 project ( YUV C CXX )	# "C" is required even for C++ projects
-cmake_minimum_required( VERSION 2.8.12 )
 option( UNIT_TEST "Built unit tests" OFF )
 
+include(CheckCSourceCompiles)
+
 set ( ly_base_dir	${PROJECT_SOURCE_DIR} )
 set ( ly_src_dir	${ly_base_dir}/source )
 set ( ly_inc_dir	${ly_base_dir}/include )
