diff -Naur mysql-workbench-gpl-5.2.43-src.orig/frontend/linux/workbench/mysql-workbench.in mysql-workbench-gpl-5.2.43-src/frontend/linux/workbench/mysql-workbench.in
--- mysql-workbench-gpl-5.2.43-src.orig/frontend/linux/workbench/mysql-workbench.in	2012-09-11 17:47:07.000000000 +0200
+++ mysql-workbench-gpl-5.2.43-src/frontend/linux/workbench/mysql-workbench.in	2013-02-28 18:24:14.217249244 +0100
@@ -4,6 +4,9 @@
 # This will cause passwords to be stored only temporarily for the session.
 #WB_NO_GNOME_KEYRING=1
 
+# create config dir (safe if it already exists)                                                                                                                                                                              
+mkdir -p $HOME/.mysql/workbench
+
 # force disable the Mac style single menu hack in Ubuntu Unity
 UBUNTU_MENUPROXY=0
 
