--- mysql-workbench-community-6.3.8-src/plugins/migration/copytable/copytable.cpp.orig	2016-10-23 15:39:11.356781884 +0700
+++ mysql-workbench-community-6.3.8-src/plugins/migration/copytable/copytable.cpp	2016-10-23 15:39:36.864471189 +0700
@@ -2847,14 +2847,7 @@
   // This is needed because the escaping depends on the character set in use by the server
   unsigned long ret_length = 0;
   
-  #if MYSQL_CHECK_VERSION(5, 7, 6)
-    if (_target->is_mysql_version_at_least(5, 7, 6))
-      ret_length += mysql_real_escape_string_quote(_mysql, buffer + length, data, (unsigned long)dlength, '`');
-    else
-      ret_length += mysql_real_escape_string(_mysql, buffer + length, data, (unsigned long)dlength);
-  #else
-    ret_length += mysql_real_escape_string(_mysql, buffer + length, data, (unsigned long)dlength);
-  #endif
+  ret_length += mysql_real_escape_string(_mysql, buffer + length, data, (unsigned long)dlength);
 
   if( ret_length != (unsigned long) -1)
     length += ret_length;
