--- mysql-workbench-community-8.0.34-src/library/cdbc/src/driver_manager.h.orig	2023-07-19 21:15:37.648964114 +0700
+++ mysql-workbench-community-8.0.34-src/library/cdbc/src/driver_manager.h	2023-07-19 21:15:45.990061252 +0700
@@ -32,6 +32,7 @@
 
 #include "grts/structs.db.mgmt.h"
 #include <cppconn/connection.h>
+#include <cstdint> // for std::uint64_t
 
 namespace wb {
   class SSHTunnel;
--- mysql-workbench-community-8.0.34-src/library/forms/mforms/treeview.h.orig	2023-07-19 21:35:04.750432184 +0700
+++ mysql-workbench-community-8.0.34-src/library/forms/mforms/treeview.h	2023-07-19 21:35:20.883616910 +0700
@@ -24,6 +24,7 @@
 #pragma once
 
 #include <mforms/view.h>
+#include <cstdint> // for std::uint64_t
 
 /**
  * Implementation of a control class for a treeview control based on node objects.
 
