--- mysql-workbench-community-6.2.2-src/backend/wbpublic/grt/spatial_handler.h.orig	2014-09-08 13:59:56.557339109 +0700
+++ mysql-workbench-community-6.2.2-src/backend/wbpublic/grt/spatial_handler.h	2014-09-08 14:00:11.515354241 +0700
@@ -20,12 +20,12 @@
 #ifndef SPATIAL_HANDLER_H_
 #define SPATIAL_HANDLER_H_
 
-#include <gdal/ogrsf_frmts.h>
-#include <gdal/ogr_api.h>
-#include <gdal/gdal_pam.h>
-#include <gdal/memdataset.h>
-#include <gdal/gdal_alg.h>
-#include <gdal/gdal.h>
+#include <ogrsf_frmts.h>
+#include <ogr_api.h>
+#include <gdal_pam.h>
+#include <memdataset.h>
+#include <gdal_alg.h>
+#include <gdal.h>
 #include <deque>
 #include "base/geometry.h"
 #include "wbpublic_public_interface.h"
--- mysql-workbench-community-6.2.2-src/backend/wbpublic/grtui/geom_draw_box.h.orig     2014-09-08 14:32:05.677224760 +0700
+++ mysql-workbench-community-6.2.2-src/backend/wbpublic/grtui/geom_draw_box.h  2014-09-08 14:32:13.122224590 +0700
@@ -22,7 +22,7 @@
 #define _GEOM_DRAW_BOX_H_
 
 #include <mforms/drawbox.h>
-#include <gdal/ogr_geometry.h>
+#include <ogr_geometry.h>
 #include "wbpublic_public_interface.h"
 
 class WBPUBLICBACKEND_PUBLIC_FUNC GeomDrawBox : public mforms::DrawBox
