--- mysql-workbench-community-6.2.1-src/backend/wbprivate/sqlide/spatial_handler.h.orig 2014-08-30 11:27:11.434520066 +0700
+++ mysql-workbench-community-6.2.1-src/backend/wbprivate/sqlide/spatial_handler.h      2014-08-30 11:27:28.737541213 +0700
@@ -20,12 +20,12 @@
 #ifndef SPATIAL_HANDLER_H_
 #define SPATIAL_HANDLER_H_
 
-#include <gdal/ogrsf_frmts.h>
-#include <gdal/ogr_api.h>
-#include <gdal/gdal_pam.h>
-#include <gdal/memdataset.h>
-#include <gdal/gdal_alg.h>
-#include <gdal/gdal.h>
+#include <ogrsf_frmts.h>
+#include <ogr_api.h>
+#include <gdal_pam.h>
+#include <memdataset.h>
+#include <gdal_alg.h>
+#include <gdal.h>
 #include <deque>
 #include "base/geometry.h"
 

