diff -Naur pwsafe-1.01BETA.orig/src/os/unix/dir.cpp pwsafe-1.01BETA/src/os/unix/dir.cpp
--- pwsafe-1.01BETA.orig/src/os/unix/dir.cpp	2016-12-21 21:10:25.000000000 +0100
+++ pwsafe-1.01BETA/src/os/unix/dir.cpp	2017-01-07 04:05:12.837349604 +0100
@@ -173,10 +173,10 @@
 
 stringT pws_os::getxmldir(void)
 {
-  return _S("/usr/share/pwsafe/xml/");
+  return _S("/usr/share/passwordsafe/xml/");
 }
 
 stringT pws_os::gethelpdir(void)
 {
-  return _S("/usr/share/doc/passwordsafe/help/");
+  return _S("/usr/share/passwordsafe/help/");
 }
