--- CMakeLists.txt	2023-12-03 14:22:05.000000000 +0000
+++ CMakeLists.txt.new	2023-12-04 22:05:56.210467471 +0000
@@ -4,7 +4,7 @@
 message(STATUS "~~~ ${PROJECT_NAME} v${PROJECT_VERSION} ~~~")
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
-option(PLUG_UNITTEST "Build Unit Tests" ON)
+option(PLUG_UNITTEST "Build Unit Tests" OFF)
 message(STATUS "Unit Tests : ${PLUG_UNITTEST}")
 
 option(PLUG_COVERAGE "Enable Coverage" OFF)
