diff -Naur unpaper-7.0.0.orig/meson.build unpaper-7.0.0/meson.build
--- unpaper-7.0.0.orig/meson.build	2023-08-12 12:37:30.389999033 -0400
+++ unpaper-7.0.0/meson.build	2023-08-12 12:37:57.307998930 -0400
@@ -36,17 +36,6 @@
     install : true,
 )
 
-sphinx = find_program('sphinx-build', required: true, version: '>= 3.4')
-
-custom_target(
-    'man',
-    command: [sphinx, '-b', 'man', join_paths(meson.source_root(), 'doc'), '@OUTDIR@'],
-    input: ['doc/conf.py', 'doc/index.rst', 'doc/unpaper.1.rst'],
-    output: 'unpaper.1',
-    install: true,
-    install_dir: join_paths(get_option('prefix'), get_option('mandir'), 'man1'),
-)
-
 pymod = import('python')
 python = pymod.find_installation(required: false, modules: ['pytest', 'PIL'], disabler: true)
 
