--- CEGUITinyXMLParser.cpp.orig	2009-10-16 11:56:28.000000000 +0100
+++ CEGUITinyXMLParser.cpp	2009-10-16 11:56:35.000000000 +0100
@@ -131,10 +131,10 @@
         {
             switch(childNode->Type())
             {
-            case CEGUI_TINYXML_NAMESPACE::TiXmlNode::ELEMENT:
+            case CEGUI_TINYXML_NAMESPACE::TiXmlNode::TINYXML_ELEMENT:
                 processElement(childNode->ToElement());
                 break;
-            case CEGUI_TINYXML_NAMESPACE::TiXmlNode::TEXT:
+            case CEGUI_TINYXML_NAMESPACE::TiXmlNode::TINYXML_TEXT:
                 if (childNode->ToText()->Value() != '\0')
                     d_handler->text((utf8*)childNode->ToText()->Value());
                 break;
