diff -Naur airoscript.orig/Makefile airoscript.new/Makefile
--- airoscript.orig/Makefile	2009-02-05 12:41:57.000000000 +0000
+++ airoscript.new/Makefile	2009-03-13 03:43:04.000000000 +0000
@@ -7,14 +7,14 @@
 
 
 # Dirs
-PREF="/usr/local"
+PREF="/usr"
 BINMODE="755"
 SBINDIR=$(DESTDIR)$(PREF)"/sbin"
 ETCDIR=$(DESTDIR)"/etc"
 SHAREDIR=$(DESTDIR)$(PREF)"/share/airoscript"
-LOCALEDIR=$(DESTDIR)$(PREF)"/share/locale/"
-MANDIR=$(DESTDIR)$(PREF)"/share/man/man1"
-DOCDIR=$(DESTDIR)$(PREF)"/share/doc/airoscript"
+LOCALEDIR=$(DESTDIR)$(PREF)"/share/locale"
+MANDIR=$(DESTDIR)$(PREF)"/man/man1"
+DOCDIR=$(DESTDIR)$(PREF)"/doc/airoscript-2.1.1"
 
 install: airopdate
 	@echo -en "Installing files into:$(BINDIR) $(ETCDIR) $(SHAREDIR) $(DOCDIR) $(SBINDIR) "
@@ -23,7 +23,7 @@
 	@cp -r $(CURDIR)/src/themes/*					                   $(SHAREDIR)/themes/
 	@chown $(UID):$(GID) $(SHAREDIR)/themes/*
 	@chmod $(BINMODE) $(SHAREDIR)/themes/*.theme
-	@install -D -o $(UID) -g $(GID) -m $(BINMODE) $(CURDIR)/src/airoscript.conf        $(ETCDIR)/airoscript.conf
+	@install -D -o $(UID) -g $(GID) -m 644        $(CURDIR)/src/airoscript.conf        $(ETCDIR)/airoscript.conf
 	@install    -o $(UID) -g $(GID) -m $(BINMODE) $(CURDIR)/src/airopdate.sh           $(SHAREDIR)/airopdate
 	@install    -o $(UID) -g $(GID) -m $(BINMODE) $(CURDIR)/src/airoscfunc.sh          $(SHAREDIR)/airoscfunc.sh
 	@install    -o $(UID) -g $(GID) -m $(BINMODE) $(CURDIR)/src/airoscfunc_screen.sh   $(SHAREDIR)/airoscfunc_screen.sh
