diff -Naur rott-1.1.1.orig/rott/develop.h rott-1.1.1/rott/develop.h
--- rott-1.1.1.orig/rott/develop.h	2008-05-24 02:29:04.000000000 -0400
+++ rott-1.1.1/rott/develop.h	2010-03-16 23:52:12.000000000 -0400
@@ -41,8 +41,8 @@
 
 #define BNACRASHPREVENT  1 //bna added 
 // Make sure only one of the following are on at one time
-#define SHAREWARE   1
-#define SUPERROTT   0
+#define SHAREWARE   0
+#define SUPERROTT   1
 #define SITELICENSE 0
 
 // cute little dopefish thing, only works with special patch?
