diff -Naur bmon-2.1.0/configure.ac bmon-2.1.0.patched/configure.ac
--- bmon-2.1.0/configure.ac	2005-04-05 11:01:33.000000000 -0400
+++ bmon-2.1.0.patched/configure.ac	2009-09-30 23:43:25.000000000 -0400
@@ -188,26 +188,6 @@
 
 #####################################################################
 ##
-## libnl check
-##
-#####################################################################
-NL="No "
-AC_CHECK_LIB(nl, nl_connect, [
-	AC_DEFINE_UNQUOTED(HAVE_NL, "1", [have libnl])
-	LIBNL="-lnl"
-	NL="Yes"
-],[
-	case ${target_os} in
-	    *linux*)
-		echo
-		echo "*** Warning: Building bmon on Linuxx w/o libnl ***"
-		echo
-	    ;;
-	esac
-])
-
-#####################################################################
-##
 ## libm check
 ##
 #####################################################################
