diff -Naur bmon-2.1.0.old/src/out_audio.c bmon-2.1.0.patched/src/out_audio.c
--- bmon-2.1.0.old/src/out_audio.c	2005-04-05 11:01:33.000000000 -0400
+++ bmon-2.1.0.patched/src/out_audio.c	2009-09-30 23:48:28.000000000 -0400
@@ -141,7 +141,7 @@
 	.om_draw = audio_draw,
 	.om_set_opts = audio_set_opts,
 	.om_probe = audio_probe,
-	.om_shutdown audio_shutdown,
+	.om_shutdown = audio_shutdown,
 };
 
 static void __init audio_init(void)
diff -Naur bmon-2.1.0.old/src/out_xml_event.c bmon-2.1.0.patched/src/out_xml_event.c
--- bmon-2.1.0.old/src/out_xml_event.c	2005-04-05 11:01:33.000000000 -0400
+++ bmon-2.1.0.patched/src/out_xml_event.c	2009-09-30 23:46:22.000000000 -0400
@@ -127,7 +127,7 @@
 	.om_draw = xml_event_draw,
 	.om_set_opts = xml_event_set_opts,
 	.om_probe = xml_event_probe,
-	.om_shutdown xml_event_shutdown,
+	.om_shutdown = xml_event_shutdown,
 };
 
 static void __init xml_event_init(void)
