From: "FeRD (Frank Dana)" <ferdnyc@gmail.com>
Date: Thu, 24 Feb 2022 07:55:03 -0500
Subject: [PATCH 2/3] FFmpegWriter: Don't force deprecated stream member

---
 src/FFmpegWriter.cpp | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src/FFmpegWriter.cpp b/src/FFmpegWriter.cpp
index c43dcff..7a45ab4 100644
--- a/src/FFmpegWriter.cpp
+++ b/src/FFmpegWriter.cpp
@@ -1266,13 +1266,6 @@ AVStream *FFmpegWriter::add_video_stream() {
 	st->avg_frame_rate = av_inv_q(c->time_base);
 	st->time_base.num = info.video_timebase.num;
 	st->time_base.den = info.video_timebase.den;
-#if (LIBAVFORMAT_VERSION_MAJOR >= 58)
-	#pragma GCC diagnostic push
-	#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-	st->codec->time_base.num = info.video_timebase.num;
-	st->codec->time_base.den = info.video_timebase.den;
-	#pragma GCC diagnostic pop
-#endif
 
 	c->gop_size = 12; /* TODO: add this to "info"... emit one intra frame every twelve frames at most */
 	c->max_b_frames = 10;
