Index: /trunk/boost/filesystem/path.hpp
===================================================================
--- /trunk/boost/filesystem/path.hpp (revision 47923)
+++ /trunk/boost/filesystem/path.hpp (revision 48374)
@@ -230,4 +230,6 @@
       string_type  leaf() const { return filename(); }
       basic_path   branch_path() const { return parent_path(); }
+      bool has_leaf() const            { return !m_path.empty(); }
+      bool has_branch_path() const     { return !parent_path().empty(); }
 # endif
 
@@ -238,5 +240,5 @@
       bool has_root_directory() const;
       bool has_relative_path() const   { return !relative_path().empty(); }
-      bool has_filename() const            { return !m_path.empty(); }
+      bool has_filename() const        { return !m_path.empty(); }
       bool has_parent_path() const     { return !parent_path().empty(); }
 
