Index: /trunk/boost/filesystem/convenience.hpp
===================================================================
--- /trunk/boost/filesystem/convenience.hpp (revision 47181)
+++ /trunk/boost/filesystem/convenience.hpp (revision 48462)
@@ -95,4 +95,5 @@
       { return create_directories<wpath>(ph); }
 
+# ifndef BOOST_FILESYSTEM_NO_DEPRECATED
     inline std::string extension(const path& ph)
       { return extension<path>(ph); }
@@ -109,4 +110,5 @@
     inline wpath change_extension( const wpath & ph, const std::wstring& new_ex )
       { return change_extension<wpath>( ph, new_ex ); }
+# endif
 
 # endif
