--- ../orig/einstein-2.0.dfsg.2/main.cpp	2007-08-31 08:42:16.000000000 +0200
+++ main.cpp	2007-08-31 10:39:20.000000000 +0200
@@ -70,7 +70,7 @@
 #ifdef __APPLE__
     dirs.push_back(getResourcesPath(selfPath));
 #else
-    dirs.push_back(PREFIX L"/share/einstein/res");
+    dirs.push_back("/usr" L"/share/einstein/res");
     dirs.push_back(fromMbcs(getenv("HOME")) + L"/.einstein/res");
 #endif
 #endif
