# Obtained from Springdale Linux (SDL) RPM
diff -up LuaBitOp-1.0.2/Makefile.destdir LuaBitOp-1.0.2/Makefile
--- Makefile.destdir	2012-05-08 13:15:00.000000000 -0600
+++ Makefile	2014-06-26 11:20:14.380263499 -0600
@@ -39,7 +39,9 @@
	$(SOCC) $(SOLDFLAGS) -o $@ $<

 install: $(MODSO)
-	$(INSTALL) $< `$(INSTALLPATH) $(MODNAME)`
+	file=$(DESTDIR)`$(INSTALLPATH) $(MODNAME)`; \
+	mkdir -p `dirname $$file`; \
+	$(INSTALL) $< $$file

 test: $(MODSO)
	@$(LUA) bittest.lua && echo "basic test OK"
