--- Qt/ApplicationComponents/pqParaViewMenuBuilders.cxx.orig	2016-07-27 12:54:37.000000000 +1000
+++ Qt/ApplicationComponents/pqParaViewMenuBuilders.cxx	2016-07-28 15:10:34.755501222 +1000
@@ -323,8 +323,8 @@
 #else
   QString appdir = QCoreApplication::applicationDirPath();
   QString documentationPath = QFileInfo(appdir).fileName() == "bin" ?
-    /* w/o shared forwarding */ appdir + "/../share/paraview-" PARAVIEW_VERSION "/doc"  :
-    /* w/ shared forwarding  */ appdir + "/../../share/paraview-" PARAVIEW_VERSION "/doc";
+    /* w/o shared forwarding */ appdir + "/../doc/paraview-" PARAVIEW_VERSION_FULL  :
+    /* w/ shared forwarding  */ appdir + "/../../doc/paraview-" PARAVIEW_VERSION_FULL ;
 #endif
 
   QString paraViewGuideFile = documentationPath + "/Guide.pdf";
--- Qt/ApplicationComponents/pqWelcomeDialog.cxx.orig	2016-07-27 12:54:37.000000000 +1000
+++ Qt/ApplicationComponents/pqWelcomeDialog.cxx	2016-07-28 15:14:43.187484954 +1000
@@ -44,8 +44,8 @@
 #else
   QString appdir = QCoreApplication::applicationDirPath();
   QString documentationPath = QFileInfo(appdir).fileName() == "bin" ?
-    /* w/o shared forwarding */ appdir + "/../share/paraview-" PARAVIEW_VERSION "/doc"  :
-    /* w/ shared forwarding  */ appdir + "/../../share/paraview-" PARAVIEW_VERSION "/doc";
+    /* w/o shared forwarding */ appdir + "/../doc/paraview-" PARAVIEW_VERSION_FULL :
+    /* w/ shared forwarding  */ appdir + "/../../doc/paraview-" PARAVIEW_VERSION_FULL ;
 #endif
   QString paraViewGettingStartedFile = documentationPath + "/GettingStarted.pdf";
   QUrl gettingStartedURL = QUrl::fromLocalFile(paraViewGettingStartedFile);
