--- ParaViewCore/ClientServerCore/Default/vtkPVFileInformation.cxx.orig	2018-11-07 05:03:24.000000000 +1000
+++ ParaViewCore/ClientServerCore/Default/vtkPVFileInformation.cxx	2018-11-23 18:17:38.214000000 +1000
@@ -1227,7 +1227,8 @@
 //-----------------------------------------------------------------------------
 std::string vtkPVFileInformation::GetParaViewDocDirectory()
 {
-  return vtkPVFileInformation::GetParaViewSharedResourcesDirectory() + "/doc";
+  /* return vtkPVFileInformation::GetParaViewSharedResourcesDirectory() + "/doc"; */
+  return "/usr/doc/paraview-" PARAVIEW_VERSION_FULL ;
 }
 
 //-----------------------------------------------------------------------------
