--- ParaViewCore/ClientServerCore/Default/vtkPVFileInformation.cxx.orig	2018-04-07 06:03:33.000000000 +1000
+++ ParaViewCore/ClientServerCore/Default/vtkPVFileInformation.cxx	2018-04-09 22:02:03.621998659 +1000
@@ -1238,7 +1238,8 @@
 //-----------------------------------------------------------------------------
 std::string vtkPVFileInformation::GetParaViewDocDirectory()
 {
-  return vtkPVFileInformation::GetParaViewSharedResourcesDirectory() + "/doc";
+  /* return vtkPVFileInformation::GetParaViewSharedResourcesDirectory() + "/doc"; */
+  return "/usr/doc/paraview-" PARAVIEW_VERSION_FULL ;
 }
 
 //-----------------------------------------------------------------------------
--- Qt/ApplicationComponents/pqParaViewMenuBuilders.cxx.orig	2018-04-07 06:03:33.000000000 +1000
+++ Qt/ApplicationComponents/pqParaViewMenuBuilders.cxx	2018-04-10 10:03:16.737203216 +1000
@@ -464,6 +464,7 @@
     (menu.addAction(QIcon(":/pqWidgets/Icons/pdf.png"), "Getting Started with ParaView")
                                   << pqSetName("actionGettingStarted")));
 
+/* Original
   // ParaView Guide
   QAction* guide = menu.addAction(QIcon(":/pqWidgets/Icons/pdf.png"), "ParaView Guide");
   guide->setObjectName("actionGuide");
@@ -472,6 +473,12 @@
                                      "download.php?submit=Download&version=v5.5&type=binary&os="
                                      "Sources&downloadFile=ParaViewGuide-5.5.0.pdf"),
     guide);
+*/
+/* Now use local Guide */
+  QString paraViewGuideFile = documentationPath + "/Guide.pdf";
+  new pqDesktopServicesReaction(QUrl::fromLocalFile(paraViewGuideFile),
+    (menu.addAction(QIcon(":/pqWidgets/Icons/pdf.png"), "ParaView Guide")
+                                  << pqSetName("actionGuide")));
 
 #ifdef PARAVIEW_USE_QTHELP
   // Help
@@ -482,12 +489,19 @@
   // -----------------
   menu.addSeparator();
 
+/* Originally
   // ParaView Tutorial notes
   new pqDesktopServicesReaction(QUrl("https://www.paraview.org/paraview-downloads/"
                                      "download.php?submit=Download&version=v5.5&type=binary&os="
                                      "Sources&downloadFile=ParaViewTutorial.pdf"),
     (menu.addAction(QIcon(":/pqWidgets/Icons/pdf.png"), "ParaView Tutorial")
                                   << pqSetName("actionTutorialNotes")));
+*/
+/* Now use local Tutorial */
+  QString paraTutorialFile = documentationPath + "/Tutorial.pdf";
+  new pqDesktopServicesReaction(QUrl::fromLocalFile(paraTutorialFile),
+    (menu.addAction(QIcon(":/pqWidgets/Icons/pdf.png"), "ParaView Tutorial")
+                                  << pqSetName("actionTutorial")));
 
   // Sandia National Labs Tutorials
   new pqDesktopServicesReaction(QUrl("http://www.paraview.org/Wiki/SNL_ParaView_4_Tutorials"),
