--- Remoting/Core/vtkPVFileInformation.cxx.orig	2021-05-14 02:47:34.000000000 +1000
+++ Remoting/Core/vtkPVFileInformation.cxx	2021-05-23 20:53:31.322000000 +1000
@@ -1227,13 +1227,15 @@
 //-----------------------------------------------------------------------------
 std::string vtkPVFileInformation::GetParaViewExampleFilesDirectory()
 {
-  return vtkPVFileInformation::GetParaViewSharedResourcesDirectory() + "/examples";
+  /* return vtkPVFileInformation::GetParaViewSharedResourcesDirectory() + "/examples"; */
+  return "/usr/doc/paraview-"PARAVIEW_VERSION_FULL ;
 }
 
 //-----------------------------------------------------------------------------
 std::string vtkPVFileInformation::GetParaViewDocDirectory()
 {
-  return vtkPVFileInformation::GetParaViewSharedResourcesDirectory() + "/doc";
+  /* return vtkPVFileInformation::GetParaViewSharedResourcesDirectory() + "/doc"; */
+  return "/usr/doc/paraview-"PARAVIEW_VERSION_FULL ;
 }
 
 //-----------------------------------------------------------------------------
