--- VTK/Utilities/Doxygen/CMakeLists.txt.orig	2017-03-11 05:22:48.000000000 +1000
+++ VTK/Utilities/Doxygen/CMakeLists.txt	2017-03-15 21:49:40.289474063 +1000
@@ -80,7 +80,4 @@
   install(FILES doc_readme.txt
     DESTINATION ${VTK_INSTALL_DOXYGEN_DIR}
     COMPONENT Development)
-  install(DIRECTORY ${VTK_BINARY_DIR}/Utilities/Doxygen/doc/html
-      DESTINATION ${VTK_INSTALL_DOXYGEN_DIR}
-      COMPONENT Development)
 endif()
