--- Plugins/GeodesicMeasurementPlugin/FmmMesh/CMakeLists.txt.orig	2016-07-27 12:54:37.000000000 +1000
+++ Plugins/GeodesicMeasurementPlugin/FmmMesh/CMakeLists.txt	2016-07-28 10:09:59.293353301 +1000
@@ -59,6 +59,6 @@
 
 install(TARGETS ${PROJECT_NAME}
   RUNTIME DESTINATION bin
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib${LIB_SUFFIX}
+  ARCHIVE DESTINATION lib${LIB_SUFFIX}
 )
