--- qTox-1.12.1/src/video/camerasource.cpp.orig	2017-10-11 23:49:18.000000000 +0200
+++ qTox-1.12.1/src/video/camerasource.cpp	2017-10-25 00:10:50.632026656 +0200
@@ -332,6 +332,8 @@
         emit openFailed();
         return;
     }
+
+    cctx->refcounted_frames = 1;
 #else
     // Create a context for our codec, using the existing parameters
     cctx = avcodec_alloc_context3(codec);
@@ -342,8 +344,6 @@
     }
 #endif
 
-    cctx->refcounted_frames = 1;
-
     // Open codec
     if (avcodec_open2(cctx, codec, nullptr) < 0) {
         qWarning() << "Can't open codec";
