diff -Naur -r meshlab.orig/src/meshlabplugins/edit_arc3D/edit_arc3D.pro meshlab/src/meshlabplugins/edit_arc3D/edit_arc3D.pro
--- meshlab.orig/src/meshlabplugins/edit_arc3D/edit_arc3D.pro	2016-10-31 21:31:43.301625217 +0100
+++ meshlab/src/meshlabplugins/edit_arc3D/edit_arc3D.pro	2016-10-31 21:36:46.694965117 +0100
@@ -35,9 +35,9 @@
 win32-msvc2013:LIBS	   += ../../external/lib/win32-msvc2013/bz2.lib
 win32-msvc2015:LIBS	   += ../../external/lib/win32-msvc2015/bz2.lib
 win32-g++:LIBS	+= ../../external/lib/win32-gcc/libbz2.a
-linux-g++:LIBS	+= ../../external/lib/linux-g++/libbz2.a
-linux-g++-32:LIBS	+= ../../external/lib/linux-g++-32/libbz2.a
-linux-g++-64:LIBS	+= ../../external/lib/linux-g++-64/libbz2.a
+linux-g++:LIBS	+= -lbz2
+linux-g++-32:LIBS	+= -lbz2
+linux-g++-64:LIBS	+= -lbz2
 mac:LIBS   += -lbz2
 
 win32-msvc2005:INCLUDEPATH += ../../external/bzip2-1.0.5
@@ -48,9 +48,6 @@
 win32-msvc2015:INCLUDEPATH += ../../external/bzip2-1.0.5
 
 win32-g++:INCLUDEPATH  += ../../external/bzip2-1.0.5
-linux-g++:INCLUDEPATH	+= ../../external/bzip2-1.0.5
-linux-g++-32:INCLUDEPATH	+= ../../external/bzip2-1.0.5
-linux-g++-64:INCLUDEPATH	+= ../../external/bzip2-1.0.5
 
 
 CONFIG(release, debug|release) {
