diff --git a/src/meshlabplugins/filter_csg/filter_csg.pro b/src/meshlabplugins/filter_csg/filter_csg.pro
index 52147b1..f41385d 100644
--- meshlab.orig/src/meshlabplugins/filter_csg/filter_csg.pro
+++ meshlab/src/meshlabplugins/filter_csg/filter_csg.pro
@@ -20,7 +20,7 @@ win32-msvc2008:LIBS		 += ../../external/lib/win32-msvc2008/mpirxx.lib ../../exte
 win32-msvc2010:LIBS		 += ../../external/lib/win32-msvc2008/mpirxx.lib ../../external/lib/win32-msvc2008/mpir.lib
 win32-msvc2012:LIBS		 += ../../external/lib/win32-msvc2008/mpirxx.lib ../../external/lib/win32-msvc2008/mpir.lib
 win32-msvc2015:LIBS		 += ../../external/lib/win32-msvc2015/mpirxx.lib ../../external/lib/win32-msvc2015/mpir.lib
-linux-g++:INCLUDEPATH += ../../external/inc/linux-g++/mpir-2.4.0
-linux-g++:LIBS += ../../external/lib/linux-g++/libmpirxx.a ../../external/lib/linux-g++/libmpir.a
-linux-g++-64:INCLUDEPATH += ../../external/inc/linux-g++/mpir-2.4.0
-linux-g++-64:LIBS += ../../external/lib/linux-g++-64/libmpirxx.a ../../external/lib/linux-g++-64/libmpir.a
+unix:LIBS += -lmpirxx -lmpir
+linux-g++-:LIBS += -lmpirxx -lmpir
+linux-g++-32:LIBS += -lmpirxx -lmpir
+linux-g++-64:LIBS += -lmpirxx -lmpir
