diff -Naur /tmp/SBo/wireshark-1.6.4-orig//dumpcap.c wireshark-1.6.4//dumpcap.c
--- /tmp/SBo/wireshark-1.6.4-orig//dumpcap.c	2011-11-18 19:14:44.000000000 +0000
+++ wireshark-1.6.4//dumpcap.c	2011-12-06 19:36:17.965368736 +0000
@@ -2491,6 +2491,9 @@
 #ifndef USE_THREADS
     int       sel_ret;
 #endif
+#ifdef MUST_DO_SELECT
+    int       sel_ret;
+#endif
 
     packet_count_before = ld->packet_count;
     if (ld->from_cap_pipe) {
