diff -Naur linuxsampler-1.0.0.orig/src/common/Pool.h linuxsampler-1.0.0/src/common/Pool.h
--- linuxsampler-1.0.0.orig/src/common/Pool.h	2008-12-07 01:18:04.000000000 +0100
+++ linuxsampler-1.0.0/src/common/Pool.h	2012-09-17 15:27:17.580041864 +0200
@@ -393,7 +393,7 @@
         inline Iterator allocAppend() {
             if (pPool->poolIsEmpty()) return RTListBase<T>::begin();
             Iterator element = pPool->alloc();
-            append(element);
+            this->append(element);
             #if CONFIG_DEVMODE
             element.list = this;
             #endif // CONFIG_DEVMODE
