Index: sqliteodbc-0.9996/Makefile.in
===================================================================
--- sqliteodbc-0.9996.orig/Makefile.in
+++ sqliteodbc-0.9996/Makefile.in
@@ -5,7 +5,7 @@ exec_prefix =	@exec_prefix@
 libdir =	@libdir@
 drvdir =	$(libdir)@DRVDIR@
 CC =		@CC@
-INSTALL =	@INSTALL@
+INSTALL =	./install-sh
 LIBTOOL =	./libtool
 CFLAGS=		@CFLAGS@ -DDRIVER_VER_INFO=\"@VER_INFO@\" @SQLITE3_A10N_FLAGS@
 SQLITE_INC =	@SQLITE_INC@
@@ -138,7 +138,7 @@ xpath.lo:	xpath.c
 
 install-2:	libsqliteodbc.la
 		$(LIBTOOL) --mode=install $(INSTALL) \
-		    libsqliteodbc.la $(DESTDIR)$(drvdir)
+		    libsqliteodbc.la $(DESTDIR)$(drvdir)/libsqliteodbc.la
 
 drvinst-2:	install-2
 		sh drvdsninst.sh SQLITE "SQLite Datasource" \
@@ -154,22 +154,22 @@ uninstall-2:
 install-3:	libsqlite3odbc.la @EXT_BLOBTOXY@ @EXT_IMPEXP@ \
 		@EXT_CSVTABLE@ @EXT_ZIPFILE@ @EXT_XPATH@
 		$(LIBTOOL) --mode=install $(INSTALL) \
-		    libsqlite3odbc.la $(DESTDIR)$(drvdir)
+		    libsqlite3odbc.la $(DESTDIR)$(drvdir)/libsqlite3odbc.la
 		test -z "@EXT_BLOBTOXY@" || \
 		    $(LIBTOOL) --mode=install $(INSTALL) \
-		    @EXT_BLOBTOXY@ $(DESTDIR)$(drvdir)
+		    @EXT_BLOBTOXY@ $(DESTDIR)$(drvdir)/@EXT_BLOBTOXY@
 		test -z "@EXT_IMPEXP@" || \
 		    $(LIBTOOL) --mode=install $(INSTALL) \
-		    @EXT_IMPEXP@ $(DESTDIR)$(drvdir)
+		    @EXT_IMPEXP@ $(DESTDIR)$(drvdir)/@EXT_IMPEXP@
 		test -z "@EXT_CSVTABLE@" || \
 		    $(LIBTOOL) --mode=install $(INSTALL) \
-		    @EXT_CSVTABLE@ $(DESTDIR)$(drvdir)
+		    @EXT_CSVTABLE@ $(DESTDIR)$(drvdir)/@EXT_CSVTABLE@
 		test -z "@EXT_ZIPFILE@" || \
 		    $(LIBTOOL) --mode=install $(INSTALL) \
-		    @EXT_ZIPFILE@ $(DESTDIR)$(drvdir)
+		    @EXT_ZIPFILE@ $(DESTDIR)$(drvdir)/@EXT_ZIPFILE@
 		test -z "@EXT_XPATH@" || \
 		    $(LIBTOOL) --mode=install $(INSTALL) \
-		    @EXT_XPATH@ $(DESTDIR)$(drvdir)
+		    @EXT_XPATH@ $(DESTDIR)$(drvdir)/@EXT_XPATH@
 
 uninstall-3:
 		$(LIBTOOL) --mode=uninstall \
@@ -200,7 +200,7 @@ drvuninst-3:	uninstall-3
 
 install-4:	libsqlite4odbc.la
 		$(LIBTOOL) --mode=install $(INSTALL) \
-		    libsqlite4odbc.la $(DESTDIR)$(drvdir)
+		    libsqlite4odbc.la $(DESTDIR)$(drvdir)/libsqlite4odbc.la
 
 uninstall-4:
 		$(LIBTOOL) --mode=uninstall \
