--- schroot-1.4.0/sbuild/sbuild-auth-pam-conv-tty.h.orig	2010-02-20 10:30:25.426447989 -0500
+++ schroot-1.4.0/sbuild/sbuild-auth-pam-conv-tty.h	2010-02-20 10:30:48.196230114 -0500
@@ -24,7 +24,6 @@
 #include <sbuild/sbuild-custom-error.h>
 
 #include <security/pam_appl.h>
-#include <security/pam_misc.h>
 
 namespace sbuild
 {
--- schroot-1.4.0/sbuild/sbuild-auth-pam.cc.orig	2010-02-20 10:35:54.444312134 -0500
+++ schroot-1.4.0/sbuild/sbuild-auth-pam.cc	2010-02-20 10:37:25.747231482 -0500
@@ -42,8 +42,11 @@
 #define PAM_TEXT_DOMAIN "Linux-PAM"
 #elif defined(__sun__)
 #define PAM_TEXT_DOMAIN "SUNW_OST_SYSOSPAM"
+#elif defined(OPENPAM)
+#define PAM_TEXT_DOMAIN "OpenPAM"
 #endif
 
+
 namespace
 {
 

