--- gui-ufw-12.04.1/bin/gufw	2012-01-12 11:43:02.000000000 -0700
+++ gui-ufw-12.04.1-fix/bin/gufw	2016-01-12 18:50:01.738311270 -0700
@@ -1,7 +1,7 @@
 #!/bin/bash
 LOCATIONS=(
-"/usr/lib/python2.7/site-packages/gufw/gufw.py"
-"/usr/lib/python2.6/site-packages/gufw/gufw.py"
+"/usr/lib64/python2.7/site-packages/gufw/gufw.py"
+"/usr/lib64/python2.6/site-packages/gufw/gufw.py"
 "/usr/lib/python2.7/dist-packages/gufw/gufw.py"
 "/usr/lib/python2.6/dist-packages/gufw/gufw.py"
 )
