diff -urp libflashsupport-MODIFICATO/configure.ac libflashsupport-MODIFICATO-NOPULSE/configure.ac
--- libflashsupport-MODIFICATO/configure.ac	2008-07-02 11:43:50.000000000 +0200
+++ libflashsupport-MODIFICATO-NOPULSE/configure.ac	2008-07-02 11:48:52.000000000 +0200
@@ -79,12 +79,12 @@ elif test -d ../pulseaudio ; then
    PULSE_CFLAGS='-I$(top_srcdir)/../pulseaudio/src'
    PULSE_LIBS='-L$(top_srcdir)/../pulseaudio/src/.libs -lpulse'
    echo "*** Found pulseaudio in ../pulseaudio, using that version ***"
-else
-   PKG_CHECK_MODULES(PULSE, [ libpulse >= 0.9.5 ])
+#else
+#   PKG_CHECK_MODULES(PULSE, [ libpulse >= 0.9.5 ])
 fi

-AC_SUBST(PULSE_LIBS)
-AC_SUBST(PULSE_CFLAGS)
+#AC_SUBST(PULSE_LIBS)
+#AC_SUBST(PULSE_CFLAGS)

 PKG_CHECK_MODULES(LIBSSL, [ libssl ])

diff -urp libflashsupport-MODIFICATO/flashsupport.c libflashsupport-MODIFICATO-NOPULSE/flashsupport.c
--- libflashsupport-MODIFICATO/flashsupport.c	2008-07-02 11:46:33.000000000 +0200
+++ libflashsupport-MODIFICATO-NOPULSE/flashsupport.c	2008-07-02 11:48:21.000000000 +0200
@@ -54,7 +54,7 @@
 //#define ALSA
 #define OSS
 //#define V4L1
-#define PULSEAUDIO
+//#define PULSEAUDIO

 ////////////////////////////////////////////////////////////////////////////////////////////////////
 //
