diff -Naur solfege-3.22.0-orig/Makefile.in solfege-3.22.0/Makefile.in
--- solfege-3.22.0-orig/Makefile.in	2013-05-25 18:07:56.000000000 -0400
+++ solfege-3.22.0/Makefile.in	2013-09-07 10:12:15.833153860 -0400
@@ -106,13 +106,6 @@
 -include debian-dir/Makefile
 -include debian/Makefile
 
-%.1: %.1.txt
-	txt2man -r "GNU Solfege @VERSION@" -t Solfege -s 1 $< > $@
-
-# We need solfege/_version.py because solfege imports it at runtime.
-solfege.1: solfege.1.txt solfege/_version.py
-	LANGUAGE=C $(PYTHON) -c "import tools.buildutil; tools.buildutil.create_manpage()" | txt2man -t Solfege -s 1 > $@
-
 rsync-manual: $(foreach lang,$(USER_MANUAL_LANGUAGES),help/$(lang)/one-big-page.html) help/C/one-big-page.html
 	rm -rf tmp-build
 	make prefix=`pwd`/tmp-build
