diff -ru solfege-3.10.4-orig/run-solfege.py.in solfege-3.10.4/run-solfege.py.in
--- solfege-3.10.4-orig/run-solfege.py.in	2008-04-29 12:31:56.000000000 -0500
+++ solfege-3.10.4/run-solfege.py.in	2008-10-31 13:55:37.000000000 -0500
@@ -31,8 +31,8 @@
     # variables. Example dir: Gehör
     prefix = prefix.decode(sys.getfilesystemencoding())
 os.chdir(os.path.join(prefix, "share", "solfege"))
-sys.path.append(os.path.join(prefix, "lib", "solfege"))
-sys.path.append(os.path.join(prefix, "share", "solfege"))
+sys.path.insert(0, os.path.join(prefix, "lib", "solfege"))
+sys.path.insert(0, os.path.join(prefix, "share", "solfege"))
 
 import src
 import src.cfg
