--- clutter-gtk-0.10.2/clutter-gtk/gtk-clutter-embed.c	2009-06-27 11:21:06.000000000 -0300
+++ clutter-gtk-0.10.2_patched/clutter-gtk/gtk-clutter-embed.c	2011-03-31 20:49:50.000000000 -0300
@@ -131,7 +131,7 @@
 {
   GtkClutterEmbedPrivate *priv = GTK_CLUTTER_EMBED (widget)->priv;
 
-  if (GTK_WIDGET_REALIZED (widget))
+  if (gtk_widget_get_realized (widget))
     clutter_actor_show (priv->stage);
 
   GTK_WIDGET_CLASS (gtk_clutter_embed_parent_class)->show (widget);
@@ -178,7 +178,7 @@
   }
 #endif
 
-  GTK_WIDGET_SET_FLAGS (widget, GTK_REALIZED);
+  gtk_widget_set_realized (widget,TRUE);
 
   attributes.window_type = GDK_WINDOW_CHILD;
   attributes.x = widget->allocation.x;
@@ -222,7 +222,7 @@
 
   clutter_actor_realize (priv->stage);
 
-  if (GTK_WIDGET_VISIBLE (widget))
+  if (gtk_widget_get_visible (widget))
     clutter_actor_show (priv->stage);
 
   gtk_clutter_embed_send_configure (GTK_CLUTTER_EMBED (widget));
@@ -246,7 +246,7 @@
 
   widget->allocation = *allocation;
 
-  if (GTK_WIDGET_REALIZED (widget))
+  if (gtk_widget_get_realized (widget))
     {
       gdk_window_move_resize (widget->window,
                               allocation->x, allocation->y,
@@ -586,8 +586,8 @@
 
   embed->priv = priv = GTK_CLUTTER_EMBED_GET_PRIVATE (embed);
 
-  GTK_WIDGET_SET_FLAGS (embed, GTK_CAN_FOCUS);
-  GTK_WIDGET_UNSET_FLAGS (embed, GTK_NO_WINDOW);
+  gtk_widget_set_can_focus(GTK_WIDGET(embed),TRUE);
+  gtk_widget_set_has_window(GTK_WIDGET(embed),TRUE);
 
   /* disable double-buffering: it's automatically provided
    * by OpenGL
--- clutter-gtk-0.10.2/clutter-gtk/gtk-clutter-util.c	2009-07-28 14:20:35.000000000 -0300
+++ clutter-gtk-0.10.2_patched/clutter-gtk/gtk-clutter-util.c	2011-03-31 20:57:19.000000000 -0300
@@ -622,7 +622,7 @@
     return CLUTTER_INIT_ERROR_GTK;
 
 #if defined(HAVE_CLUTTER_GTK_X11)
-  clutter_x11_set_display (GDK_DISPLAY());
+  clutter_x11_set_display (GDK_DISPLAY_XDISPLAY(gdk_display_get_default()));
   clutter_x11_disable_event_retrieval ();
 #elif defined(HAVE_CLUTTER_GTK_WIN32)
   clutter_win32_disable_event_retrieval ();
@@ -673,7 +673,7 @@
     return CLUTTER_INIT_ERROR_GTK;
 
 #if defined(GDK_WINDOWING_X11)
-  clutter_x11_set_display (GDK_DISPLAY());
+  clutter_x11_set_display (GDK_DISPLAY_XDISPLAY(gdk_display_get_default()));
   clutter_x11_disable_event_retrieval ();
 #elif defined(GDK_WINDOWING_WIN32)
   clutter_win32_disable_event_retrieval ();
