diff -rup pcsx-df-1.9/configure.ac pcsx-df-1.9.new.clean/configure.ac
--- pcsx-df-1.9/configure.ac	2008-01-17 13:11:07.000000000 +0100
+++ pcsx-df-1.9.new.clean/configure.ac	2008-07-25 15:55:38.857104605 +0200
@@ -23,6 +23,11 @@ AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE], ["
 PKG_CHECK_MODULES(GLIB2, glib-2.0, [], AC_MSG_ERROR([*** glib2 not found!]))
 PKG_CHECK_MODULES(GTK2, gtk+-2.0, [], AC_MSG_ERROR([*** libgtk2 not found!]))
 PKG_CHECK_MODULES(GLADE2, libglade-2.0, [], AC_MSG_ERROR([*** libglade2 not found!]))
+
+PKG_CHECK_MODULES(GTHREAD, gthread-2.0, [], AC_MSG_ERROR([*** gthread-2.0 not found!]))
+AC_SUBST(GTREAD_CFLAGS)
+AC_SUBST(GTREAD_LIBS)
+
 AC_ARG_ENABLE(nautilusburn, [  --enable-nautilusburn    build dfiso with libnautilusburn (default yes)],
 [ NAUTILUS_BURN="$enableval" ],[ NAUTILUS_BURN="yes" ])
 AM_CONDITIONAL(ENABLE_NAUTILUSBURN,test "x$NAUTILUS_BURN" = xyes)
diff -rup pcsx-df-1.9/gui/Makefile.am pcsx-df-1.9.new.clean/gui/Makefile.am
--- pcsx-df-1.9/gui/Makefile.am	2008-01-12 16:13:30.000000000 +0100
+++ pcsx-df-1.9.new.clean/gui/Makefile.am	2008-07-25 15:55:38.917113978 +0200
@@ -5,6 +5,8 @@ INCLUDES = -DPACKAGE_DATA_DIR=\"${datadi
 	$(GTK2_CFLAGS) $(GLADE2_CFLAGS) \
 	-I$(top_srcdir)/libpcsxcore -I$(top_srcdir)/include
 
+AM_CFLAGS= $(GTRHEAD_CFLAGS) $(GLIB2_CFLAGS) $(GTK2_CFLAGS) $(GLADE2_CFLAGS)
+
 bin_PROGRAMS = pcsx
 
 pcsx_SOURCES = \
@@ -14,6 +16,6 @@ pcsx_SOURCES = \
 	Config.c	\
 	Gtk2Gui.c	\
 	hdebug.c
-	
-pcsx_LDADD = \
-	$(GTK2_LIBS) $(GLADE2_LIBS) ../libpcsxcore/libpcsxcore.la
+
+pcsx_LDADD = $(GTHREAD_LIBS) $(GLIB2_LIBS) $(GTK2_LIBS) \
+             $(GLADE2_LIBS) ../libpcsxcore/libpcsxcore.la
\ Pas de fin de ligne à la fin du fichier.
