--- ndiswrapper/driver/loader.c	2010/09/13 08:35:26	2725
+++ ndiswrapper/driver/loader.c	2010/09/13 08:43:15	2726
@@ -750,8 +750,13 @@
 }
 
 /* called with loader_mutex is down */
+#ifdef HAVE_UNLOCKED_IOCTL
+static long wrapper_ioctl(struct file *file,
+			 unsigned int cmd, unsigned long arg)
+#else
 static int wrapper_ioctl(struct inode *inode, struct file *file,
 			 unsigned int cmd, unsigned long arg)
+#endif
 {
 	struct load_driver *load_driver;
 	struct load_device load_device;
@@ -831,7 +836,12 @@
 
 static struct file_operations wrapper_fops = {
 	.owner          = THIS_MODULE,
+
+#ifdef HAVE_UNLOCKED_IOCTL
+	.unlocked_ioctl	= wrapper_ioctl,
+#else
 	.ioctl		= wrapper_ioctl,
+#endif
 	.release	= wrapper_ioctl_release,
 };
 
