--- MEGAsync-5.2.1.0_Linux/src/MEGAShellExtDolphin/CMakeLists.txt	2024-04-17 20:25:54.000000000 +0900
+++ MEGAsync-5.2.1.0_Linux.New/src/MEGAShellExtDolphin/CMakeLists.txt	2024-05-10 19:38:46.571062863 +0900
@@ -58,4 +58,4 @@
 target_link_libraries(${MEGASYNCPLUGIN} KF5::CoreAddons KF5::KIOCore KF5::KIOWidgets)
 
 install(FILES megasync-plugin.desktop DESTINATION ${KDE_INSTALL_KSERVICES5DIR})
-install(TARGETS ${MEGASYNCPLUGIN} DESTINATION ${CMAKE_INSTALL_LIBDIR}/qt5/plugins)
+install(TARGETS ${MEGASYNCPLUGIN} DESTINATION ${PLUGIN_INSTALL_DIR})
