From 0137f46c58e26352b2ba88cb581974f3f263ba3f Mon Sep 17 00:00:00 2001
From: Chris Pickel <sfiera@twotaled.com>
Date: Fri, 5 Jun 2020 20:09:00 +0900
Subject: [PATCH] Update to python3

---
 scripts/antares_launcher.py     |  4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/scripts/antares_launcher.py b/scripts/antares_launcher.py
index d20f343e..fc5e4d32 100755
--- a/scripts/antares_launcher.py
+++ b/scripts/antares_launcher.py
@@ -1,10 +1,8 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (C) 2017 The Antares Authors
 # This file is part of Antares, a tactical space combat game.
 # Antares is free software, distributed under the LGPL+. See COPYING.
 
-from __future__ import division, print_function, unicode_literals
-
 import gi
 import json
 import os
