diff -Naur teapot-2.3.0/CMakeLists.txt teapot-2.3.0.patched/CMakeLists.txt
--- teapot-2.3.0/CMakeLists.txt	2020-11-19 00:18:02.217657446 -0500
+++ teapot-2.3.0.patched/CMakeLists.txt	2020-11-19 00:22:05.201635357 -0500
@@ -40,7 +40,7 @@
 	install(TARGETS teapot DESTINATION bin)
 endif ()
 
-find_package(FLTK NO_MODULE)
+find_package(FLTK)
 if (FLTK_FOUND)
 	include("${FLTK_USE_FILE}")
 
@@ -52,13 +52,13 @@
 		if (ENABLE_STATIC)
 			target_link_libraries(fteapot teapotlib fltk fltk_images ${LIB_PORTABLEXDR})
 		else ()
-			target_link_libraries(fteapot teapotlib fltk_SHARED fltk_images_SHARED ${LIB_PORTABLEXDR})
+			target_link_libraries(fteapot teapotlib fltk fltk_images ${LIB_PORTABLEXDR})
 		endif ()
 	else ()
 		if (ENABLE_STATIC)
 			target_link_libraries(fteapot teapotlib fltk fltk_images ${LIB_PORTABLEXDR})
 		else ()
-			target_link_libraries(fteapot teapotlib fltk_SHARED ${LIB_PORTABLEXDR})
+			target_link_libraries(fteapot teapotlib fltk fltk_images ${LIB_PORTABLEXDR})
 		endif ()
 	endif ()
 	install(TARGETS fteapot DESTINATION bin)
diff -Naur teapot-2.3.0/fteapot.fl teapot-2.3.0.patched/fteapot.fl
--- teapot-2.3.0/fteapot.fl	2020-11-19 00:15:47.506669691 -0500
+++ teapot-2.3.0.patched/fteapot.fl	2020-11-19 00:16:05.569668049 -0500
@@ -11,6 +11,9 @@
 decl {\#include <fcntl.h>} {private global
 }
 
+decl {\#include <unistd.h>} {private global
+}
+
 decl {\#define shadow _shadow} {private global
 }
 
