--- easystroke-thjaeger/Makefile	2026-09-26 19:58:27.585660069 +1200
+++ easystroke.sbopkg/Makefile	2026-09-27 16:22:44.123771043 +1300
@@ -43,7 +43,7 @@
 GENFILES = gui.c desktop.c po/POTFILES.in easystroke.desktop
 GZFILES  = $(wildcard *.gz)
 
-VERSION  = $(shell test -e debian/changelog && grep '(.*)' debian/changelog | sed 's/.*(//' | sed 's/).*//' | head -n1 || (test -e version && cat version || git describe))
+VERSION  = $(shell test -e debian/changelog && grep '(.*)' debian/changelog | sed 's/.*(//' | sed 's/).*//' | head -n1 || (test -e version && cat version || git describe --always 2>/dev/null || echo "0.6.0"))
 GIT      = $(wildcard .git/index version)
 DIST     = easystroke-$(VERSION)
 
@@ -116,7 +116,7 @@
 man:	$(MANPAGE)
 
 $(MANPAGE):	$(BINARY)
-	help2man -N -n "X11 gesture recognition application" ./$(BINARY) > $@
+	help2man -N -n "X11 gesture recognition application" --version-string="$(VERSION)" ./$(BINARY) > $@
 
 debug:
 	@echo "The version is: '$(VERSION)'"
