diff -up LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan1_2p.cpp.unique LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan1_2p.cpp
--- LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan1_2p.cpp.unique	2019-06-06 13:55:41.522433921 -0400
+++ LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan1_2p.cpp	2019-06-06 13:56:06.386992097 -0400
@@ -38,7 +38,7 @@ Foundation, Inc., 51 Franklin Street, Fi
 
 namespace{
 //list of entity types supported by current action
-auto enTypeList={RS2::EntityLine, RS2::EntityArc, RS2::EntityCircle};
+auto enTypeList1_2={RS2::EntityLine, RS2::EntityArc, RS2::EntityCircle};
 }
 
 struct RS_ActionDrawCircleTan1_2P::Points {
@@ -181,7 +181,7 @@ void RS_ActionDrawCircleTan1_2P::mouseMo
     }
     case SetCenter: {
 
-        //        RS_Entity*  en = catchEntity(e, enTypeList, RS2::ResolveAll);
+        //        RS_Entity*  en = catchEntity(e, enTypeList1_2, RS2::ResolveAll);
 		pPoints->coord= graphicView->toGraph(e->x(), e->y());
         //        circles[getStatus()]=static_cast<RS_Line*>(en);
         if(preparePreview()) {
@@ -272,7 +272,7 @@ bool RS_ActionDrawCircleTan1_2P::prepare
 
 RS_Entity* RS_ActionDrawCircleTan1_2P::catchCircle(QMouseEvent* e) {
 	RS_Entity* ret=nullptr;
-	RS_Entity* en = catchEntity(e,enTypeList, RS2::ResolveAll);
+	RS_Entity* en = catchEntity(e,enTypeList1_2, RS2::ResolveAll);
 	if (!en) return ret;
 	if (!en->isVisible()) return ret;
 	if (en->getParent()) {
diff -up LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan2_1p.cpp.unique LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan2_1p.cpp
--- LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan2_1p.cpp.unique	2019-06-06 13:56:17.533766673 -0400
+++ LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan2_1p.cpp	2019-06-06 13:56:42.592259905 -0400
@@ -36,7 +36,7 @@ Foundation, Inc., 51 Franklin Street, Fi
 #include "rs_debug.h"
 
 namespace {
-auto enTypeList={RS2::EntityLine, RS2::EntityArc, RS2::EntityCircle};
+auto enTypeList2_1={RS2::EntityLine, RS2::EntityArc, RS2::EntityCircle};
 }
 
 struct RS_ActionDrawCircleTan2_1P::Points {
@@ -184,7 +184,7 @@ bool RS_ActionDrawCircleTan2_1P::prepare
 
 RS_Entity* RS_ActionDrawCircleTan2_1P::catchCircle(QMouseEvent* e) {
 	RS_Entity* ret=nullptr;
-    RS_Entity*  en = catchEntity(e,enTypeList, RS2::ResolveAll);
+    RS_Entity*  en = catchEntity(e,enTypeList2_1, RS2::ResolveAll);
 	if (!en) return ret;
 	if (!en->isVisible()) return ret;
 	for(auto p: pPoints->circles){
diff -up LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan2.cpp.unique LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan2.cpp
--- LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan2.cpp.unique	2019-06-06 13:56:52.743054623 -0400
+++ LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan2.cpp	2019-06-06 13:57:14.371617229 -0400
@@ -34,7 +34,7 @@ Foundation, Inc., 51 Franklin Street, Fi
 #include "rs_debug.h"
 
 namespace {
-auto enTypeList={RS2::EntityLine, RS2::EntityArc, RS2::EntityCircle};
+auto enTypeList2={RS2::EntityLine, RS2::EntityArc, RS2::EntityCircle};
 }
 
 struct RS_ActionDrawCircleTan2::Points {
@@ -121,7 +121,7 @@ void RS_ActionDrawCircleTan2::mouseMoveE
 
     switch(getStatus() ){
     case SetCenter: {
-        //        RS_Entity*  en = catchEntity(e, enTypeList, RS2::ResolveAll);
+        //        RS_Entity*  en = catchEntity(e, enTypeList2, RS2::ResolveAll);
 		pPoints->coord= graphicView->toGraph(e->x(), e->y());
         //        circles[getStatus()]=static_cast<RS_Line*>(en);
         if(preparePreview()) {
@@ -166,7 +166,7 @@ bool RS_ActionDrawCircleTan2::preparePre
 }
 
 RS_Entity* RS_ActionDrawCircleTan2::catchCircle(QMouseEvent* e) {
-    RS_Entity*  en = catchEntity(e,enTypeList, RS2::ResolveAll);
+    RS_Entity*  en = catchEntity(e,enTypeList2, RS2::ResolveAll);
 	if (!en) return nullptr;
 	if (!en->isVisible()) return nullptr;
 	for (int i=0;i<getStatus();i++) {
diff -up LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan3.cpp.unique LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan3.cpp
--- LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan3.cpp.unique	2019-06-06 13:57:22.978443168 -0400
+++ LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawcircletan3.cpp	2019-06-06 13:57:40.469089451 -0400
@@ -36,7 +36,7 @@ Foundation, Inc., 51 Franklin Street, Fi
 #include "rs_debug.h"
 
 namespace {
-auto enTypeList={RS2::EntityArc, RS2::EntityCircle, RS2::EntityLine, RS2::EntityPoint};
+auto enTypeList3={RS2::EntityArc, RS2::EntityCircle, RS2::EntityLine, RS2::EntityPoint};
 }
 
 struct RS_ActionDrawCircleTan3::Points {
@@ -125,7 +125,7 @@ void RS_ActionDrawCircleTan3::mouseMoveE
 
 	switch(getStatus() ){
 	case SetCenter: {
-		//        RS_Entity*  en = catchEntity(e, enTypeList, RS2::ResolveAll);
+		//        RS_Entity*  en = catchEntity(e, enTypeList3, RS2::ResolveAll);
 		pPoints->coord= graphicView->toGraph(e->x(), e->y());
 		//        circles[getStatus()]=static_cast<RS_Line*>(en);
 		deletePreview();
@@ -326,7 +326,7 @@ bool RS_ActionDrawCircleTan3::preparePre
 
 RS_Entity* RS_ActionDrawCircleTan3::catchCircle(QMouseEvent* e) {
     RS_Entity* ret=nullptr;
-	RS_Entity*  en = catchEntity(e,enTypeList, RS2::ResolveAll);
+	RS_Entity*  en = catchEntity(e,enTypeList3, RS2::ResolveAll);
 	if (!en) return ret;
 	if (!en->isVisible()) return ret;
 	for(int i=0;i<getStatus();++i) {
diff -up LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawlinetangent1.cpp.unique LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawlinetangent1.cpp
--- LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawlinetangent1.cpp.unique	2019-06-06 13:58:15.207386934 -0400
+++ LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawlinetangent1.cpp	2019-06-06 13:58:33.095026437 -0400
@@ -37,7 +37,7 @@
 #include "rs_debug.h"
 
 namespace{
-auto circleType={RS2::EntityArc, RS2::EntityCircle,
+auto circleType1={RS2::EntityArc, RS2::EntityCircle,
 				 RS2::EntityEllipse, RS2::EntitySplinePoints
 				};
 }
@@ -106,7 +106,7 @@ void RS_ActionDrawLineTangent1::mouseMov
 		break;
 
 	case SetCircle: {
-		RS_Entity* en = catchEntity(e, circleType, RS2::ResolveAll);
+		RS_Entity* en = catchEntity(e, circleType1, RS2::ResolveAll);
 		if (en && (en->isArc() ||
 				   en->rtti()==RS2::EntitySplinePoints)) {
 			if(circle){
diff -up LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawlinetangent2.cpp.unique LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawlinetangent2.cpp
--- LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawlinetangent2.cpp.unique	2019-06-06 13:58:41.362860565 -0400
+++ LibreCAD-2.2.0-rc1/librecad/src/actions/rs_actiondrawlinetangent2.cpp	2019-06-06 13:58:58.335520052 -0400
@@ -36,7 +36,7 @@
 #include "rs_debug.h"
 
 namespace{
-auto circleType={RS2::EntityArc, RS2::EntityCircle, RS2::EntityEllipse};
+auto circleType2={RS2::EntityArc, RS2::EntityCircle, RS2::EntityEllipse};
 }
 
 RS_ActionDrawLineTangent2::RS_ActionDrawLineTangent2(
@@ -100,7 +100,7 @@ void RS_ActionDrawLineTangent2::mouseMov
 //    RS_DEBUG->print("RS_ActionDrawLineTangent2::mouseMoveEvent begin");
 	e->accept();
     if(getStatus() != SetCircle2) return;
-	RS_Entity* en= catchEntity(e, circleType, RS2::ResolveAll);
+	RS_Entity* en= catchEntity(e, circleType2, RS2::ResolveAll);
 	if(!en || en==circle1) return;
 	if(circle2){
 		circle2->setHighlighted(false);
@@ -140,7 +140,7 @@ void RS_ActionDrawLineTangent2::mouseRel
     switch (getStatus()) {
     case SetCircle1:
     {
-        circle1 = catchEntity(e, circleType, RS2::ResolveAll);
+        circle1 = catchEntity(e, circleType2, RS2::ResolveAll);
 		if(!circle1) return;
         circle1->setHighlighted(true);
 		graphicView->drawEntity(circle1);
