diff -Naur dockapps-c2fa6a3/example-wmsupermonrc dockapps-c2fa6a3.patched/example-wmsupermonrc
--- dockapps-c2fa6a3/example-wmsupermonrc	2016-08-03 20:18:50.000000000 -0400
+++ dockapps-c2fa6a3.patched/example-wmsupermonrc	2020-06-10 02:10:51.872534759 -0400
@@ -30,11 +30,11 @@
 Source = /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq
 Scale = 1000
 
-# temperature from the second sensor (cpu temperature on my motherboard)
+# temperature from the first sensor (cpu temperature on my motherboard)
 [Tem]
-Source = /sys/bus/i2c/drivers/w83627hf/9191-0290/temp2_input
+Source = /sys/class/hwmon/hwmon0/temp1_input
 Scale  = 1000
-Range  = 37 .. 55
+Range  = 37 .. 75
 
 # second fan speed (cpu fan on my motherboard)
 [fan]
