--- lockdev-1.0.3.orig/src/lockdev.c
+++ lockdev-1.0.3/src/lockdev.c
@@ -125,14 +125,23 @@
 #include <sys/file.h>
 #if defined (__GNU_LIBRARY__)
 # include <sys/sysmacros.h>
+#ifdef __GNU__
+# define MAJOR(dev) major (dev)
+# define MINOR(dev) minor (dev)
+#else
 # define MAJOR(dev) gnu_dev_major (dev)
 # define MINOR(dev) gnu_dev_minor (dev)
+#endif
 #else
 #  error "put here a define for MAJOR and MINOR"
 #endif
 
 #include "lockdev.h"
 
+#ifndef MAXPATHLEN
+#define MAXPATHLEN 4096
+#endif
+
 /*
  *	PROTOTYPES for internal functions
  *	all internal functions names start with _dl_
