--- Amaya11.3.1/Amaya/thotlib/base/platform.c	2009-12-02 09:23:07.000000000 -0200
+++ Amaya11.3.1_patched/Amaya/thotlib/base/platform.c	2011-03-25 00:43:18.000000000 -0300
@@ -329,9 +329,9 @@
     {
       tmp = (char *)TtaGetMemory (strlen(targetFileName)+10);
       sprintf (tmp, "%s.tmp", targetFileName);
-      wxFile::wxFile (targetFile, wxFile::write);
+      wxFile (targetFile, wxFile::write);
       tmpFile = TtaConvMessageToWX(tmp);
-      wxFile::wxFile (tmpFile, wxFile::write);
+      wxFile (tmpFile, wxFile::write);
       result = wxFile::Exists(targetFile);
       if (result)
         result = wxConcatFiles (tmpFile, sourceFile, targetFile);
--- Amaya11.3.1/Amaya/amaya/HTMLedit.c	2009-12-09 08:11:22.000000000 -0200
+++ Amaya11.3.1_patched/Amaya/amaya/HTMLedit.c	2011-03-25 01:05:24.000000000 -0300
@@ -1689,7 +1689,7 @@
                 {
 #ifdef _WX
                   wxString tmpFile = TtaConvMessageToWX(tempURL);
-                  wxFile::wxFile (tmpFile, wxFile::write);
+                  wxFile (tmpFile, wxFile::write);
 #endif /* _WX */
                 }
             }
