diff -Naur a/linux-user/signal.c b/linux-user/signal.c
--- a/linux-user/signal.c	2009-12-02 20:27:02.000000000 +0000
+++ b/linux-user/signal.c	2013-11-18 00:48:22.345001621 +0000
@@ -3059,6 +3059,10 @@
     uint32_t tramp[2];
 };
 
+// fix for 'incomplete type struct siginfo info' error
+typedef struct siginfo {
+} siginfo;
+
 struct rt_signal_frame {
     struct siginfo info;
     struct ucontext uc;
@@ -3276,6 +3280,10 @@
         uint8_t retcode[8];       /* Trampoline code. */
 };
 
+// fix for 'incomplete type struct siginfo info' error
+typedef struct siginfo {
+} siginfo;
+
 struct rt_signal_frame {
         struct siginfo *pinfo;
         void *puc;
