diff -NarU5 a/include/headers.h b/include/headers.h
--- a/include/headers.h	2015-04-02 16:51:48.000000000 -0400
+++ b/include/headers.h	2016-03-16 14:52:14.386853750 -0400
@@ -80,11 +80,10 @@
 #include <assert.h>
 #include <ctype.h>
 #include <errno.h>
 #include <string.h>
 #include <time.h>
-#include <math.h>
 #include <float.h>
 #include <sys/types.h>
 
 #ifdef WIN32
 
diff -NarU5 a/src/Client.cpp b/src/Client.cpp
--- a/src/Client.cpp	2015-04-02 16:51:48.000000000 -0400
+++ b/src/Client.cpp	2016-03-16 14:52:14.387853701 -0400
@@ -50,10 +50,11 @@
  * A client thread initiates a connect to the server and handles
  * sending and receiving data, then closes the socket.
  * ------------------------------------------------------------------- */
 
 #include <time.h>
+#include <cmath>
 #include "headers.h"
 #include "Client.hpp"
 #include "Thread.h"
 #include "SocketAddr.h"
 #include "PerfSocket.hpp"
diff -NarU5 a/src/ReportDefault.c b/src/ReportDefault.c
--- a/src/ReportDefault.c	2015-04-02 16:51:49.000000000 -0400
+++ b/src/ReportDefault.c	2016-03-16 14:52:14.387853701 -0400
@@ -47,10 +47,11 @@
  * ReportDefault.c
  * by Kevin Gibbs <kgibbs@nlanr.net>
  *
  * ________________________________________________________________ */
 
+#include <math.h>
 #include "headers.h"
 #include "Settings.hpp"
 #include "util.h"
 #include "Reporter.h"
 #include "report_default.h"
diff -NarU5 a/src/Reporter.c b/src/Reporter.c
--- a/src/Reporter.c	2015-04-02 16:51:49.000000000 -0400
+++ b/src/Reporter.c	2016-03-16 14:52:14.388853652 -0400
@@ -47,10 +47,11 @@
  * Reporter.c
  * by Kevin Gibbs <kgibbs@nlanr.net>
  *
  * ________________________________________________________________ */
 
+#include <math.h>
 #include "headers.h"
 #include "Settings.hpp"
 #include "util.h"
 #include "Reporter.h"
 #include "Thread.h"
diff -NarU5 a/src/Server.cpp b/src/Server.cpp
--- a/src/Server.cpp	2015-04-02 16:51:49.000000000 -0400
+++ b/src/Server.cpp	2016-03-16 14:52:14.388853652 -0400
@@ -52,10 +52,11 @@
  * Handles sending and receiving data, and then closes socket.
  * Changes to this version : The server can be run as a daemon
  * ------------------------------------------------------------------- */
 
 #define HEADERS()
+#include <cmath>
 #include "headers.h"
 #include "Server.hpp"
 #include "List.h"
 #include "Extractor.h"
 #include "Reporter.h"
