--- a/config.def.h	2022-05-12 00:58:25.427926600 -0300
+++ b/config.def.h	2022-05-14 00:30:50.588985094 -0300
@@ -63,7 +63,7 @@
 #define SHCMD(cmd) { .v = (const char*[]){ "/bin/sh", "-c", cmd, NULL } }
 
 /* commands */
-static const char *termcmd[] = { "alacritty", NULL };
+static const char *termcmd[] = { "foot", NULL };
 static const char *menucmd[] = { "bemenu-run", NULL };
 
 static const Key keys[] = {
