diff -Nur irda-utils-0.9.18.orig/findchip/nsc.c irda-utils-0.9.18/findchip/nsc.c
--- irda-utils-0.9.18.orig/findchip/nsc.c	2006-07-11 01:16:06.000000000 -0500
+++ irda-utils-0.9.18/findchip/nsc.c	2009-10-14 00:22:39.372013759 -0500
@@ -33,7 +33,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
-#include <asm/io.h>
+#include <sys/io.h>
 
 #include "findchip.h"
 
diff -Nur irda-utils-0.9.18.orig/findchip/smc.c irda-utils-0.9.18/findchip/smc.c
--- irda-utils-0.9.18.orig/findchip/smc.c	2006-07-11 01:16:06.000000000 -0500
+++ irda-utils-0.9.18/findchip/smc.c	2009-10-14 00:22:42.770014109 -0500
@@ -32,7 +32,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
-#include <asm/io.h>
+#include <sys/io.h>
 
 #include "findchip.h"
 
diff -Nur irda-utils-0.9.18.orig/findchip/winbond.c irda-utils-0.9.18/findchip/winbond.c
--- irda-utils-0.9.18.orig/findchip/winbond.c	2006-07-11 01:16:06.000000000 -0500
+++ irda-utils-0.9.18/findchip/winbond.c	2009-10-14 00:22:45.938995740 -0500
@@ -32,7 +32,7 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <fcntl.h>
-#include <asm/io.h>
+#include <sys/io.h>
 
 #include "findchip.h"
 
