--- smcinit/Makefile~	2007-06-11 21:06:06.000000000 +0200
+++ smcinit/Makefile	2007-06-11 21:06:39.000000000 +0200
@@ -26,17 +26,17 @@
 
 smcinit: smcinit.o
 	$(prn_cc_o)
-	$(ECMD)$(CC) -o $@ $^ $(LIBS) -lpci
+	$(ECMD)$(CC) -o $@ $^ $(LIBS) -lpci -lz
 
 
 tosh1800-smcinit: tosh1800-smcinit.o
 	$(prn_cc_o)
-	$(ECMD)$(CC) $(XCFLAGS) -o $@ $^ $(LIBS) -lpci
+	$(ECMD)$(CC) $(XCFLAGS) -o $@ $^ $(LIBS) -lpci -lz
 
 
 tosh2450-smcinit: tosh2450-smcinit.o
 	$(prn_cc_o)
-	$(ECMD)$(CC) $(XCFLAGS) -o $@ $^ $(LIBS) -lpci
+	$(ECMD)$(CC) $(XCFLAGS) -o $@ $^ $(LIBS) -lpci -lz
 
 
 %o: %c
