diff --git a/usr/bin/microsoft-edge-stable b/usr/bin/microsoft-edge-stable
deleted file mode 120000
index 275612f..0000000
--- a/usr/bin/microsoft-edge-stable
+++ /dev/null
@@ -1 +0,0 @@
-/opt/microsoft/msedge/microsoft-edge
\ No newline at end of file
diff --git a/usr/bin/microsoft-edge-stable b/usr/bin/microsoft-edge-stable
new file mode 100755
index 0000000..b8c06ea
--- /dev/null
+++ b/usr/bin/microsoft-edge-stable
@@ -0,0 +1,15 @@
+#!/usr/bin/env sh
+# Launches MS Edge with flags specified in $XDG_CONFIG_HOME/microsoft-edge-beta-flags.conf
+
+# Make script fail if `cat` fails for some reason
+set -e
+
+# Set default value if variable is unset/null
+XDG_CONFIG_HOME="${XDG_CONFIG_HOME:-${HOME}/.config}"
+
+# Attempt to read a config file if it exists
+if [ -r "${XDG_CONFIG_HOME}/microsoft-edge-stable-flags.conf" ]; then
+  EDGE_USER_FLAGS="$(cat "$XDG_CONFIG_HOME/microsoft-edge-stable-flags.conf")"
+fi
+
+exec /usr/lib64/msedge/microsoft-edge $EDGE_USER_FLAGS "$@"
diff --git a/usr/lib64/msedge/default-app-block b/usr/lib64/msedge/default-app-block
index 7ac7165..7ad030e 100644
--- a/usr/lib64/msedge/default-app-block
+++ b/usr/lib64/msedge/default-app-block
@@ -1,10 +1,10 @@
     <web-browser>
       <name>Microsoft Edge</name>
-      <executable>/opt/microsoft/msedge/microsoft-edge</executable>
-      <command>/opt/microsoft/msedge/microsoft-edge %s</command>
+      <executable>/usr/lib64/msedge/microsoft-edge</executable>
+      <command>/usr/lib64/msedge/microsoft-edge %s</command>
       <icon-name>microsoft-edge</icon-name>
       <run-in-terminal>false</run-in-terminal>
       <netscape-remote>true</netscape-remote>
-      <tab-command>/opt/microsoft/msedge/microsoft-edge %s</tab-command>
-      <win-command>/opt/microsoft/msedge/microsoft-edge --new-window %s</win-command>
+      <tab-command>/usr/lib64/msedge/microsoft-edge %s</tab-command>
+      <win-command>/usr/lib64/msedge/microsoft-edge --new-window %s</win-command>
     </web-browser>
diff --git a/usr/share/gnome-control-center/default-apps/microsoft-edge.xml b/usr/share/gnome-control-center/default-apps/microsoft-edge.xml
index 80dec09..fbdf731 100644
--- a/usr/share/gnome-control-center/default-apps/microsoft-edge.xml
+++ b/usr/share/gnome-control-center/default-apps/microsoft-edge.xml
@@ -8,13 +8,13 @@
   <web-browsers>
     <web-browser>
       <name>Microsoft Edge</name>
-      <executable>/opt/microsoft/msedge/microsoft-edge</executable>
-      <command>/opt/microsoft/msedge/microsoft-edge %s</command>
+      <executable>/usr/lib64/msedge/microsoft-edge</executable>
+      <command>/usr/lib64/msedge/microsoft-edge %s</command>
       <icon-name>microsoft-edge</icon-name>
       <run-in-terminal>false</run-in-terminal>
       <netscape-remote>true</netscape-remote>
-      <tab-command>/opt/microsoft/msedge/microsoft-edge %s</tab-command>
-      <win-command>/opt/microsoft/msedge/microsoft-edge --new-window %s</win-command>
+      <tab-command>/usr/lib64/msedge/microsoft-edge %s</tab-command>
+      <win-command>/usr/lib64/msedge/microsoft-edge --new-window %s</win-command>
     </web-browser>
   </web-browsers>
 </default-apps>
diff --git a/usr/share/menu/microsoft-edge.menu b/usr/share/menu/microsoft-edge.menu
new file mode 100644
index 0000000..7871356
--- /dev/null
+++ b/usr/share/menu/microsoft-edge.menu
@@ -0,0 +1,6 @@
+?package(microsoft-edge-stable):needs="x11" \
+  section="Applications/Network/Web Browsing" \
+  hints="Web browsers" \
+  title="Microsoft Edge" \
+  icon="/usr/lib64/msedge/product_logo_32.xpm" \
+  command="/usr/lib64/msedge/microsoft-edge"
