--- oneTBB-2021.6.0.orig/test/conformance/conformance_flowgraph.h	2023-12-17 14:34:12.234366082 +0900
+++ oneTBB-2021.6.0/test/conformance/conformance_flowgraph.h	2023-12-17 15:27:02.512416476 +0900
@@ -79,8 +79,9 @@
 template<typename V>
 typename std::enable_if<!std::is_default_constructible<V>::value, std::vector<V>>::type get_values( test_push_receiver<V>& rr ) {
     std::vector<V> messages;
-    int val = 0;
-    for(V tmp(0); rr.try_get(tmp); ++val) {
+    V tmp(0);
+
+    while (rr.try_get(tmp)) {
         messages.push_back(tmp);
     }
     return messages;
@@ -89,8 +90,9 @@
 template<typename V>
 typename std::enable_if<std::is_default_constructible<V>::value, std::vector<V>>::type get_values( test_push_receiver<V>& rr ) {
     std::vector<V> messages;
-    int val = 0;
-    for(V tmp; rr.try_get(tmp); ++val) {
+    V tmp;
+
+    while (rr.try_get(tmp)) {
         messages.push_back(tmp);
     }
     return messages;
--- oneTBB-2021.6.0.orig/test/conformance/conformance_join_node.cpp	2023-12-17 14:34:12.235366077 +0900
+++ oneTBB-2021.6.0/test/conformance/conformance_join_node.cpp	2023-12-17 15:45:20.864114506 +0900
@@ -28,8 +28,8 @@
 
 std::vector<my_input_tuple> get_values( conformance::test_push_receiver<my_input_tuple>& rr ) {
     std::vector<my_input_tuple> messages;
-    int val = 0;
-    for(my_input_tuple tmp(0, 0.f, input_msg(0)); rr.try_get(tmp); ++val) {
+    my_input_tuple tmp(0, 0.f, input_msg(0));
+    while(rr.try_get(tmp)) {
         messages.push_back(tmp);
     }
     return messages;
