diff -Naur vwm.orig/modules/vwmterm3/Makefile vwm/modules/vwmterm3/Makefile
--- vwm.orig/modules/vwmterm3/Makefile	2009-10-20 16:02:05.000000000 -0400
+++ vwm/modules/vwmterm3/Makefile	2017-08-25 19:55:30.380418408 -0400
@@ -20,7 +20,7 @@
 
 vwmterm3:
 	gcc $(CFLAGS) $(DEFS) $(PKG_CFG) $(includedir) -c -fpic *.c 
-	gcc $(CFLAGS) -shared -o vwmterm3.so *.o
+	gcc $(CFLAGS) -shared -o vwmterm3.so *.o $(VTERM_A)
 
 vwmterm3_wide:
 	gcc $(CFLAGS) $(DEFS) $(PKG_CFG) $(includedir) -D_VIPER_WIDE -c -fpic *.c 
diff -Naur vwm.orig/modules/vwmterm3/init.c vwm/modules/vwmterm3/init.c
--- vwm.orig/modules/vwmterm3/init.c	2009-10-30 17:31:44.000000000 -0400
+++ vwm/modules/vwmterm3/init.c	2017-08-25 19:56:03.036416671 -0400
@@ -36,10 +36,6 @@
 	if(g_module_open("libutil.so",G_MODULE_BIND_LAZY)==NULL)
  		return "\ncould not preload libutil.so";
 
-   // preload libvterm for use with this module
-   if(g_module_open("libvterm.so",G_MODULE_BIND_LAZY)==NULL)
-      return "\ncould not preload libvterm.so";
-
 	/*	register with vwm 	*/
 	libfilename=(gchar*)g_module_name(module);
 	vwm_module_add("Tools","VTerm",vwmterm_init,NULL,libfilename);
