--- vwm/modules/vwmterm2/Makefile	2009-04-21 23:55:19.000000000 +0100
+++ vwm/modules/vwmterm2/Makefile.new	2010-01-19 19:57:52.754821551 +0000
@@ -2,10 +2,10 @@
 PKG_CFG = `pkg-config --cflags glib-2.0 gmodule-2.0`
 DEFS = -D_REENTRANT -D_GNU_SOURCE
 prefix = /usr/local
-libdir = ${prefix}/lib
+libdir = ${prefix}/lib${arch}
 includedir = -I${prefix}/include -I../..
 bindir = ${prefix}/bin
-moddir = /usr/lib/vwm/modules
+moddir = ${libdir}/vwm/modules
 
 makefile: all
 
