--- Makefile.orig	2009-05-22 14:47:31.000000000 +0100
+++ Makefile	2009-05-22 15:54:30.000000000 +0100
@@ -8,7 +8,7 @@
 includedir = ${prefix}/include -I./
 bindir = ${prefix}/bin
 headerdir = ${prefix}/include
-moddir = /usr/lib/vwm/modules
+moddir = ${libdir}/vwm/modules
 
 makefile: all
 
@@ -32,7 +32,7 @@
 	rm -f vwm
 
 install:
-	mkdir -p $(moddir)
+	mkdir -p $(moddir) $(bindir) $(headerdir) $(libdir)
 	chmod 644 vwm.h
 	cp -f vwm.h $(headerdir)
 	chmod 755 vwm
