--- a/pm/sleep.d/99video
+++ b/pm/sleep.d/99video
@@ -144,7 +144,7 @@ suspend_video()
 	local acpi_flag=0
 	quirk "${QUIRK_S3_BIOS}" && 		acpi_flag=$(($acpi_flag + 1))
 	quirk "${QUIRK_S3_MODE}" && 		acpi_flag=$(($acpi_flag + 2))
-	[ 0 -ne $acpi_flag ] && sysctl -w kernel.acpi_video_flags=$acpi_flag
+	sysctl -w kernel.acpi_video_flags=$acpi_flag
 	
 	quirk "${QUIRK_NOFB}" && 		die_if_framebuffer
 	quirk "${QUIRK_VBESTATE_RESTORE}" && 	vbe_savestate
@@ -166,8 +166,6 @@ resume_video()
 	quirk "${QUIRK_RADEON_OFF}" && 		radeon_on
 	quirk "${QUIRK_DPMS_ON}" && 		vbe dpms on
 	quirk "${QUIRK_RESET_BRIGHTNESS}" && 	reset_brightness
-	# always reset acpi_video_flag -- makes quirk testing more reliable.
-	sysctl -w kernel.acpi_video_flags=0
 	return 0  # avoid spurious hook exit failure message.
 }
 
