diff -pur adns-1.6.1.orig/common.make.in adns-1.6.1/common.make.in
--- adns-1.6.1.orig/common.make.in	2024-05-05 23:28:40.000000000 +0200
+++ adns-1.6.1/common.make.in	2024-06-19 12:27:28.903419587 +0200
@@ -47,17 +47,6 @@ bindir		:= @bindir@
 libdir		:= @libdir@
 includedir	:= @includedir@
 
-ifneq ($(DESTDIR),)
-# autoconf likes to define some of these in terms of each other
-# so if we want to add a prefix to each one we have to do it this way:
-destdir_prefix=$(DESTDIR)/
-prefix		:= ${destdir_prefix}${prefix}
-exec_prefix	:= ${destdir_prefix}${exec_prefix}
-bindir		:= ${destdir_prefix}${bindir}
-libdir		:= ${destdir_prefix}${libdir}
-includedir	:= ${destdir_prefix}${includedir}
-endif
-
 AC_INSTALL=	@INSTALL@
 ifeq ($(AC_INSTALL),./install-sh -c)
 INSTALL=	$(srcdir)/../$(AC_INSTALL)
@@ -67,4 +56,4 @@ endif
 
 INSTALL_PROGRAM=	$(INSTALL) -m 755 $(INSTALL_PROGRAM_FLAGS)
 INSTALL_DATA=		$(INSTALL) -m 644
-INSTALL_DIR=		$(INSTALL) -d 755
+INSTALL_DIR=		$(INSTALL) -m 755 -d
