diff -Naur hedgewars-src-1.0.2.orig/CMakeLists.txt hedgewars-src-1.0.2/CMakeLists.txt
--- hedgewars-src-1.0.2.orig/CMakeLists.txt	2022-09-13 00:33:59.000000000 +0200
+++ hedgewars-src-1.0.2/CMakeLists.txt	2025-07-23 14:32:06.333368259 +0200
@@ -11,7 +11,7 @@
 
 foreach(hwpolicy CMP0026 CMP0068)
     if(POLICY ${hwpolicy})
-        cmake_policy(SET ${hwpolicy} OLD)
+        cmake_policy(SET ${hwpolicy} NEW)
     endif()
 endforeach()
 
diff -Naur hedgewars-src-1.0.2.orig/misc/libphyslayer/CMakeLists.txt hedgewars-src-1.0.2/misc/libphyslayer/CMakeLists.txt
--- hedgewars-src-1.0.2.orig/misc/libphyslayer/CMakeLists.txt	2022-09-12 16:27:31.000000000 +0200
+++ hedgewars-src-1.0.2/misc/libphyslayer/CMakeLists.txt	2025-07-23 14:34:08.847368259 +0200
@@ -21,12 +21,10 @@
 install(TARGETS physlayer RUNTIME DESTINATION ${target_binary_install_dir}
                           LIBRARY DESTINATION ${target_library_install_dir}
                           ARCHIVE DESTINATION ${target_library_install_dir})
-get_target_property(physlayer_fullpath physlayer LOCATION)
 
 
 ## added standard variables (FORCE or cmake won't pick 'em)
 set(PHYSLAYER_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR} CACHE STRING "Physlayer include dir" FORCE)
-set(PHYSLAYER_LIBRARY ${physlayer_fullpath} CACHE STRING "Physlayer library" FORCE)
 
 if(BUILD_ENGINE_JS)
     set_target_properties(physlayer PROPERTIES SUFFIX ".bc")
