diff -Naur hedgewars-src-1.0.2.orig/CMakeLists.txt hedgewars-src-1.0.2/CMakeLists.txt
--- hedgewars-src-1.0.2.orig/CMakeLists.txt	2022-09-13 00:33:59.000000000 +0200
+++ hedgewars-src-1.0.2/CMakeLists.txt	2025-07-23 14:32:06.333368259 +0200
@@ -11,7 +11,7 @@
 
 foreach(hwpolicy CMP0026 CMP0068)
     if(POLICY ${hwpolicy})
-        cmake_policy(SET ${hwpolicy} OLD)
+        cmake_policy(SET ${hwpolicy} NEW)
     endif()
 endforeach()
 
