Description: Fix/Workaround build failure with newer fpc 3.2.0
Origin: http://hg.hedgewars.org/hedgewars/rev/6832dab555ae
Bug-Debian: https://bugs.debian.org/968125
Last-Update: 2020-08-10

--- a/hedgewars/uWorld.pas	Sun Aug 09 14:43:02 2020 +0200
+++ b/hedgewars/uWorld.pas	Mon Aug 10 12:56:53 2020 +0300
@@ -1129,8 +1129,8 @@
 procedure ShiftWorld(Dir: LongInt); inline;
 begin
     preShiftWorldDx:= WorldDx;
-    WorldDx:= WorldDx + LongInt(Dir * LongInt(playWidth));
-
+    Dir := Dir * LongInt(playWidth);
+    WorldDx:= WorldDx + Dir;
 end;
 
 procedure UnshiftWorld(); inline;


