--- cmake_modules/paths.cmake.orig	2015-01-03 23:11:50.455837007 -0600
+++ cmake_modules/paths.cmake	2015-01-03 23:14:19.209742864 -0600
@@ -15,7 +15,7 @@
 #resource paths
 if(UNIX AND NOT APPLE)
     set(target_binary_install_dir "bin" CACHE PATH "install dest for binaries")
-    set(target_library_install_dir "lib" CACHE PATH "install dest for libs")
+    set(target_library_install_dir "lib${LIB_SUFFIX}" CACHE PATH "install dest for libs")
 
     string(SUBSTRING "${DATA_INSTALL_DIR}" 0 1 sharepath_start)
     if(${sharepath_start} MATCHES "/")
