From 0eb95ff5c1c29887f14ecb2bb5f9c3cdd8b3ff29 Mon Sep 17 00:00:00 2001
From: felixonmars <felixonmars@archlinux.org>
Date: Thu, 4 Jun 2020 12:33:24 +0200
Subject: [PATCH] Fix build with Qt 5.15+

---
 QTfrontend/ui/page/pagegamestats.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/QTfrontend/ui/page/pagegamestats.cpp b/QTfrontend/ui/page/pagegamestats.cpp
index c0ad9a974f..655b70d37b 100644
--- a/QTfrontend/ui/page/pagegamestats.cpp
+++ b/QTfrontend/ui/page/pagegamestats.cpp
@@ -22,6 +22,7 @@
 #include <QGraphicsScene>
 #include <QGroupBox>
 #include <QSizePolicy>
+#include <QPainterPath>
 
 #include "pagegamestats.h"
 #include "team.h"
