diff -Naur man-pages-ja-20140715/script/configure.perl man-pages-ja-20140715.patched/script/configure.perl
--- man-pages-ja-20140715/script/configure.perl	2010-11-09 13:14:23.000000000 -0500
+++ man-pages-ja-20140715.patched/script/configure.perl	2014-08-09 15:55:43.000000000 -0400
@@ -14,49 +14,10 @@
 my @pack_method = ("none", "gzip", "bzip2", "compress");
 my $podsec = "1";
 
-#
-# インストール情報
-#
-print "[INSTALLATION INFORMATION]\n";
-print "(just Return if you accept default)\n";
-
-do{
-    $MANROOT = "/usr/share/man/$LANG";
-    print "   Install directory   [$MANROOT] ?: ";
-    $ans = <STDIN>; chomp $ans;
-    if ($ans ne "") {$MANROOT = $ans;}
-
-    print "   compress manual with..\n";
-    for my $i (0 .. $#pack_method){
-	print "      $i: $pack_method[$i]\n";
-    }
-    print "   select [0..$#pack_method] : ";
-    $ans = <STDIN>; chomp $ans;
-    if  ($ans eq "") {$ans = 0;}
-    if ($ans < 0 || $ans > $#pack_method) {$PACK = $pack_method[0]}
-    else {$PACK = $pack_method[$ans]}
-
-    $OWNER = "root";
-    print "   uname of page owner [$OWNER] ?: ";
-    $ans = <STDIN>; chomp $ans;
-    if ($ans ne "") {$OWNER = $ans;}
-
-    $GROUP = "root";
-    print "   group of page owner [$GROUP] ?: ";
-    $ans = <STDIN>; chomp $ans;
-    if ($ans ne "") {$GROUP = $ans;}
-
-    print "\n";
-    print "   Directory:    $MANROOT\n";
-    print "   Compression:  $PACK\n";
-    print "   Page uid/gid: $OWNER/$GROUP\n";
-    print "\n";
-    do {
-	print "All OK? (Yes, [C]ontinue / No, [R]eselect) : ";
-	$ans = <STDIN>; chomp $ans;
-    } until ($ans =~ /^[yYnNcCrR]/);
-
-} until ($ans =~ /^[yYcC]/);
+$MANROOT = $ENV{MANROOT};
+$PACK = "gzip";
+$OWNER = "root";
+$GROUP = "root";
 
 #
 # パッケージデータの読み込み (ファイルの先にあるほうが
@@ -80,30 +41,7 @@
 }
 close(PL);
 
-#
-# インストールするパッケージの選択
-# インストールしない奴は優先度 $pw{$name} を -1 にする。
-#
-print "\n\n";
-print "[INSTALL PACKAGE SELECTION]\n";
-print "(just Return if you accept default choice)\n";
-print "(you can change the default by editing script/pkgs.list)\n";
-
-do{
-    for $i (0 .. $#pl){
-	my $qstr = ($main::pw{$pl[$i]} > -1) ? "[Y/n]" : "[y/N]";
-	printf "   [%2d/%2d] %-15s %s ?: ", $i, $#pl, $pl[$i], $qstr;
-	$ans = <STDIN>; chomp $ans;
-	if ($ans =~ /[Yy].*/) { $main::pw{$pl[$i]} = $i;}
-	if ($ans =~ /[Nn].*/) { $main::pw{$pl[$i]} = -1;}
-    }
-
-    do {
-	print "All OK? (Yes, [C]ontinue / No, [R]eselect) : ";
-	$ans = <STDIN>; chomp $ans;
-    } until ($ans =~ /[yYnNcCrR].*/);
-
-} until ($ans =~ /^[yYcC]/);
+for $i (0 .. $#pl) { $main::pw{$pl[$i]} = $i; }
 
 #
 # translation_list からインストールすべきファイルを取得.
@@ -147,7 +85,7 @@
 print "[RESOLVE CONFLICTS]\n";
 print "(just Return if you accept item 0)\n";
 
-do{
+#do{
     my $i = 0;
     foreach my $key (sort keys %ppage) {
 	if (@{$ppage{$key}} < 2) { next; }
@@ -165,7 +103,8 @@
 	}
 	print "   Which to install? (0..$cf) : ";
 
-	$ans = <STDIN>; chomp $ans;
+#	$ans = <STDIN>; chomp $ans;
+	$ans = 0;
 	if ($ans eq "") {$ans = 0};
 	if ($ans < 0 || $ans > $cf) { $ans = 0 };
 
@@ -178,12 +117,12 @@
 	$i++;
     }
 
-    do {
-	print "All OK? (Yes, [C]ontinue / No, [R]eselect) : ";
-	$ans = <STDIN>; chomp $ans;
-    } until ($ans =~ /^[yYnNcCrR]/);
+#    do {
+#	print "All OK? (Yes, [C]ontinue / No, [R]eselect) : ";
+#	$ans = <STDIN>; chomp $ans;
+#    } until ($ans =~ /^[yYnNcCrR]/);
 
-} until ($ans =~ /^[yYcC]/);
+#} until ($ans =~ /^[yYcC]/);
 
 #
 # インストールスクリプトを作る。
