--- disc/ifomodule.c~	2004-07-17 10:46:52.000000000 +0200
+++ disc/ifomodule.c	2006-04-11 07:59:38.030697690 +0200
@@ -9,6 +9,7 @@
 #include <unistd.h>
 #include <assert.h>
 #include <inttypes.h>
+#include <stdint.h>
 
 #include <dvdread/dvd_reader.h>
 #include <dvdread/ifo_types.h>
--- disc/cdrommodule.c~	2003-06-23 21:26:16.000000000 +0200
+++ disc/cdrommodule.c	2006-04-11 07:58:02.463087444 +0200
@@ -36,6 +36,7 @@
 #include "Python.h"
 #include <fcntl.h>
 #include <sys/ioctl.h>
+#include <stdint.h>
 
 #ifdef __linux__
 #include <linux/cdrom.h>
