diff -Naur gri-2.12.27/configure.ac gri-2.12.27b/configure.ac
--- gri-2.12.27/configure.ac	2024-04-29 21:37:38.000000000 +0100
+++ gri-2.12.27b/configure.ac	2026-10-05 15:27:54.494714282 +0100
@@ -331,17 +331,17 @@
 dnl AC_SUBST(EXTRA_LIBS)
 
 
-dnl	Check for Imgemagick's convert, needed to make PNG files in HTML manual
-AC_CHECK_PROG(HAVE_CONVERT, convert, "yes", "no")
+dnl	Check for Imgemagick's magick (was convert), needed to make PNG files in HTML manual
+AC_CHECK_PROG(HAVE_CONVERT, magick, "yes", "no")
 if test $HAVE_CONVERT = "no" ; then
 	echo " **WARNING:  Cannot make proper HTML manual without Imagemagick."
 	echo " **WARNING:  It is needed to make PNG images from PostScript files."
 	echo " **WARNING:  Expect broken image links in the manual!"
 fi
 
-dnl	AC_MSG_CHECKING(for ImageMagick's convert)
+dnl	AC_MSG_CHECKING(for ImageMagick's magick)
 dnl	define(CHECK_FOR_CONVERT, [
-dnl		if test -f /usr/bin/convert ; then
+dnl		if test -f /usr/bin/magick ; then
 dnl			HAVE_CONVERT="yes"
 dnl			echo "yes"
 dnl		else
diff -Naur gri-2.12.27/doc/examples/Makefile.am gri-2.12.27b/doc/examples/Makefile.am
--- gri-2.12.27/doc/examples/Makefile.am	2024-04-29 21:37:38.000000000 +0100
+++ gri-2.12.27b/doc/examples/Makefile.am	2026-10-05 23:44:03.644909649 +0100
@@ -63,28 +63,36 @@
  example12.ps example13.ps \
  logo.ps FEM.ps
 
-DISTCLEANFILES = $(png_files) $(eps_files) $(html_files) $(txt_files) $(svg_files)
-
+#DISTCLEANFILES = $(png_files) $(eps_files) $(html_files) $(txt_files) $(svg_files)
+DISTCLEANFILES = $(png_files) $(html_files) $(txt_files) $(svg_files)
 %.svg : %.gri
-	gri -output $@ $< 	
+	gri -output $@ $<
 %.txt : %.gri
 	echo "$<" | cat >$@
 	chmod +w $@
-%.eps : %.ps
-	cp $< $@
-	chmod +w $@
-%.png : %.ps
-	-convert -strip -background white $< $@
+
+# Reduce image load of documentation
+#%.eps : %.ps
+#	cp $< $@
+#	chmod +w $@
+
+# Conversion from ps is blocked by /etc/ImageMagick-*/policy.xml
+%.png :
+	$<
+#%.png : %.ps
+#	-magick $< -strip -background white $@
 %-tiny.png : %.png
-	-convert -strip -background white -geometry 90x999 $< $@
-%.pdf : %.ps
-	convert $< $@ 
-#	ps2pdf $< $@
+	-magick $< -strip -background white -geometry 90x999 $@
+#%.pdf : %.ps
+#	magick $< $@
+##	ps2pdf $< $@
+
 %.html : %.gri
 	perl $(srcdir)/../gri2html $< $@
-all: html png eps txt
+#all: html png eps txt
+all: html png txt
 html: $(html_files)
-eps: $(eps_files)
+#eps: $(eps_files)
 png: $(png_files)
 txt: $(txt_files)
 
diff -Naur gri-2.12.27/doc/screenshots/Makefile.am gri-2.12.27b/doc/screenshots/Makefile.am
--- gri-2.12.27/doc/screenshots/Makefile.am	2024-04-29 21:37:38.000000000 +0100
+++ gri-2.12.27b/doc/screenshots/Makefile.am	2026-10-05 14:43:11.249369615 +0100
@@ -12,31 +12,33 @@
 
 DISTCLEANFILES = $(the_eps_files)
 
-all: eps png
+#all: eps png
+all: png
 png: $(the_png_files)
-eps: $(the_eps_files)
+#eps: $(the_eps_files)
 
-pdf: $(the_pdf_files)
+#pdf: $(the_pdf_files)
 
 
-%.pgm : %.png
-	-convert  -geometry 500x999 $< $@
+#%.pgm : %.png
+#	-magick $< -geometry 500x999 $@
 
-%.eps : %.pgm
-	-convert $< $@
+## ImageMagick policy blocks conversions
+#%.eps : %.pgm
+#	-magick $< $@
 
-%.pdf : %.eps
-	convert $< $@
-#	ps2pdf $< $@
+#%.pdf : %.eps
+#	magick $< $@
+##	ps2pdf $< $@
 
-# This was good for color; 
+# This was good for color;
 # now let's put them in grayscale for the PostScript manual
 #%.eps : %.png
-#	-convert  -geometry 500x999 $< eps2:$@
+#	-magick  -geometry 500x999 $< eps2:$@
 
 %-tiny.png : %.png
-	-convert -geometry 999x120 $< $@
+	-magick $< -geometry 999x120 $@
 
 # OLD STUFF, PROBABLY SHOULD BE REMOVED.
 old_clean:
-	rm -f *-tiny.png *.eps *.pgm *.pdf *~ 
+	rm -f *-tiny.png *.eps *.pgm *.pdf *~
