From 90f648bf6182d558ff6489caefb52240fb2eab4e Mon Sep 17 00:00:00 2001
From: Alexander Polakov <polachok@gmail.com>
Date: Sun, 29 May 2011 15:34:55 +0400
Subject: [PATCH 4/8] ksh/vi: make Ctrl-f in command mode behave the same as
 in insert.

---
 vi.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git vi.c vi.c
index d0de478..6eb5d8c 100644
--- vi.c
+++ vi.c
@@ -1132,7 +1132,7 @@ vi_cmd(int argcnt, const char *cmd)
 				return -1;
 		case '\\':			/* at&t ksh */
 		case Ctrl('f'):			/* Nonstandard vi/ksh */
-			complete_word(1, argcnt, 0);
+			complete_word(1, argcnt, XCF_FORCE_COMMAND);
 			break;
 
 
-- 
1.7.5

