diff -Naur webkitgtk-2.20.5.orig/Source/WebCore/dom/Document.cpp webkitgtk-2.20.5/Source/WebCore/dom/Document.cpp
--- webkitgtk-2.20.5.orig/Source/WebCore/dom/Document.cpp	2018-06-11 11:32:58.000000000 +0200
+++ webkitgtk-2.20.5/Source/WebCore/dom/Document.cpp	2019-10-07 17:35:29.096058000 +0200
@@ -4688,12 +4688,12 @@
     unsigned i = 0;
 
     UChar32 c;
-    U16_NEXT(characters, i, length, c)
+    U16_NEXT(characters, i, length, c);
     if (!isValidNameStart(c))
         return false;
 
     while (i < length) {
-        U16_NEXT(characters, i, length, c)
+        U16_NEXT(characters, i, length, c);
         if (!isValidNamePart(c))
             return false;
     }
@@ -4753,7 +4753,7 @@
 
     for (unsigned i = 0; i < length; ) {
         UChar32 c;
-        U16_NEXT(qualifiedName, i, length, c)
+        U16_NEXT(qualifiedName, i, length, c);
         if (c == ':') {
             if (sawColon)
                 return Exception { InvalidCharacterError };
