--- tripod/src/ipodexportdialog.cpp~	2007-02-01 10:36:04.000000000 +0100
+++ tripod/src/ipodexportdialog.cpp	2007-05-29 15:27:19.000000000 +0200
@@ -494,7 +494,7 @@
 
         // get last album. we don't use lastItem() because it returns children. FIXME?? O(n)
         QListViewItem *last = m_ipodAlbumList->firstChild();
-        while( last->nextSibling() )
+        while( last && last->nextSibling() )
             last = last->nextSibling();
 
         Itdb_PhotoAlbum *photoAlbum = itdb_photodb_photoalbum_create( m_itdb, QFile::encodeName( newAlbum ), -1/*end*/ );
