diff --git a/src/gtkhex.c b/src/gtkhex.c
index c34b57f..b155838 100644
--- a/src/gtkhex.c
+++ b/src/gtkhex.c
@@ -2951,6 +2951,17 @@ hex_widget_init (HexWidget *self)
 	gtk_widget_add_controller (widget,
 			GTK_EVENT_CONTROLLER(controller));
 
+	/* Event controller - focus */
+
+	controller = gtk_event_controller_focus_new ();
+
+	g_signal_connect_swapped (controller, "enter",
+			G_CALLBACK(gtk_widget_queue_draw), widget);
+	g_signal_connect_swapped (controller, "leave",
+			G_CALLBACK(gtk_widget_queue_draw), widget);
+
+	gtk_widget_add_controller (widget, GTK_EVENT_CONTROLLER(controller));
+
 	/* Connect signal for adjustment */
 
 	g_signal_connect (self->adj, "value-changed",
