diff -Naur bozohttpd-20170201/auth-bozo.c bozohttpd-20170201.patched/auth-bozo.c
--- bozohttpd-20170201/auth-bozo.c	2017-01-31 23:21:23.000000000 -0500
+++ bozohttpd-20170201.patched/auth-bozo.c	2018-06-29 15:42:59.076820050 -0400
@@ -34,9 +34,12 @@
 
 #ifdef DO_HTPASSWD
 
+#define _XOPEN_SOURCE
+
 #include <sys/param.h>
 
 #include <string.h>
+#include <strings.h>
 #include <stdlib.h>
 #include <unistd.h>
 
diff -Naur bozohttpd-20170201/bozohttpd.c bozohttpd-20170201.patched/bozohttpd.c
--- bozohttpd-20170201/bozohttpd.c	2017-01-31 23:21:23.000000000 -0500
+++ bozohttpd-20170201.patched/bozohttpd.c	2018-06-29 15:08:08.066873176 -0400
@@ -132,6 +132,10 @@
  * And so it begins ..
  */
 
+/* need this for vasprintf(): */
+#define _GNU_SOURCE
+#include <stdio.h>
+
 #include <sys/param.h>
 #include <sys/socket.h>
 #include <sys/time.h>
diff -Naur bozohttpd-20170201/lua-bozo.c bozohttpd-20170201.patched/lua-bozo.c
--- bozohttpd-20170201/lua-bozo.c	2017-01-31 23:21:23.000000000 -0500
+++ bozohttpd-20170201.patched/lua-bozo.c	2018-06-29 15:43:38.424819051 -0400
@@ -32,6 +32,9 @@
 
 #ifndef NO_LUA_SUPPORT
 
+#define _GNU_SOURCE
+#include <stdio.h>
+
 #include <sys/param.h>
 
 #include <lua.h>
diff -Naur bozohttpd-20170201/ssl-bozo.c bozohttpd-20170201.patched/ssl-bozo.c
--- bozohttpd-20170201/ssl-bozo.c	2017-01-31 23:21:23.000000000 -0500
+++ bozohttpd-20170201.patched/ssl-bozo.c	2018-06-29 15:07:50.813873614 -0400
@@ -32,6 +32,9 @@
 
 /* this code implements SSL and backend IO for bozohttpd */
 
+/* need this for vasprintf(): */
+#define _GNU_SOURCE
+
 #include <stdarg.h>
 #include <stdio.h>
 #include <syslog.h>
