diff -up PyXML-0.8.4/xml/xpath/ParsedAbbreviatedAbsoluteLocationPath.py.patch PyXML-0.8.4/xml/xpath/ParsedAbbreviatedAbsoluteLocationPath.py
--- PyXML-0.8.4/xml/xpath/ParsedAbbreviatedAbsoluteLocationPath.py.patch	2008-12-24 06:35:27.000000000 +0100
+++ PyXML-0.8.4/xml/xpath/ParsedAbbreviatedAbsoluteLocationPath.py	2008-12-24 06:35:39.000000000 +0100
@@ -24,8 +24,8 @@ class ParsedAbbreviatedAbsoluteLocationP
         self._rel = rel
         nt = ParsedNodeTest.ParsedNodeTest('node', '')
         ppl = ParsedPredicateList.ParsedPredicateList([])
-        as = ParsedAxisSpecifier.ParsedAxisSpecifier('descendant-or-self')
-        self._step = ParsedStep.ParsedStep(as, nt, ppl)
+        as_ = ParsedAxisSpecifier.ParsedAxisSpecifier('descendant-or-self')
+        self._step = ParsedStep.ParsedStep(as_, nt, ppl)
         return
 
     def evaluate(self, context):
diff -up PyXML-0.8.4/xml/xpath/ParsedAbbreviatedRelativeLocationPath.py.patch PyXML-0.8.4/xml/xpath/ParsedAbbreviatedRelativeLocationPath.py
--- PyXML-0.8.4/xml/xpath/ParsedAbbreviatedRelativeLocationPath.py.patch	2008-12-24 06:37:27.000000000 +0100
+++ PyXML-0.8.4/xml/xpath/ParsedAbbreviatedRelativeLocationPath.py	2008-12-24 06:37:48.000000000 +0100
@@ -28,8 +28,8 @@ class ParsedAbbreviatedRelativeLocationP
         self._right = right
         nt = ParsedNodeTest.ParsedNodeTest('node','')
         ppl = ParsedPredicateList.ParsedPredicateList([])
-        as = ParsedAxisSpecifier.ParsedAxisSpecifier('descendant-or-self')
-        self._middle = ParsedStep.ParsedStep(as, nt, ppl)
+        as_ = ParsedAxisSpecifier.ParsedAxisSpecifier('descendant-or-self')
+        self._middle = ParsedStep.ParsedStep(as_, nt, ppl)
 
     def evaluate(self, context):
         res = []
diff -ru PyXML-0.8.4.orig/xml/xslt/Stylesheet.py PyXML-0.8.4/xml/xslt/Stylesheet.py
--- PyXML-0.8.4/xml/xslt/Stylesheet.py.patch	2009-03-03 08:20:01.000000000 +0000
+++ PyXML-0.8.4/xml/xslt/Stylesheet.py	2009-03-03 08:21:18.000000000 +0000
@@ -373,8 +373,8 @@
 
         #Attribute sets
         attribute_sets = filter(lambda x: x.nodeType == Node.ELEMENT_NODE and (x.namespaceURI, x.localName) == (XSL_NAMESPACE, 'attribute-set'), self.childNodes)
-        for as in attribute_sets:
-            as.instantiate(context, processor)
+        for as_ in attribute_sets:
+            as_.instantiate(context, processor)
         overridden_params = {}
         for k in topLevelParams.keys():
             if type(k) != types.TupleType:
