diff -u -r libsigrokdecode-0.3.1-orig/configure.ac libsigrokdecode-0.3.1/configure.ac
--- libsigrokdecode-0.3.1-orig/configure.ac	2015-07-12 21:35:14.000000000 +0100
+++ libsigrokdecode-0.3.1/configure.ac	2015-12-21 19:23:07.809995006 +0000
@@ -99,6 +99,11 @@
 	CPPFLAGS_PYTHON="$CPPFLAGS_PYTHON $python3_CFLAGS";
 	LDFLAGS_PYTHON="$LDFLAGS_PYTHON $python3_LIBS";
 	MODNAME_PYTHON="python3"],
+[PKG_CHECK_MODULES([python35], [python-3.5 >= 3.5],
+	[pyver=`$PKG_CONFIG --modversion python-3.5`;
+	CPPFLAGS_PYTHON="$CPPFLAGS_PYTHON $python35_CFLAGS";
+	LDFLAGS_PYTHON="$LDFLAGS_PYTHON $python35_LIBS";
+	MODNAME_PYTHON="python-3.5"],
 [PKG_CHECK_MODULES([python34], [python-3.4 >= 3.4],
 	[pyver=`$PKG_CONFIG --modversion python-3.4`;
 	CPPFLAGS_PYTHON="$CPPFLAGS_PYTHON $python34_CFLAGS";
@@ -114,7 +119,7 @@
 	CPPFLAGS_PYTHON="$CPPFLAGS_PYTHON $python32_CFLAGS";
 	LDFLAGS_PYTHON="$LDFLAGS_PYTHON $python32_LIBS";
 	MODNAME_PYTHON="python-3.2"],
-)])])])
+)])])])])
 AC_SUBST(CPPFLAGS_PYTHON)
 AC_SUBST(LDFLAGS_PYTHON)
 AC_SUBST(MODNAME_PYTHON)
