--- iptraf-ng-1.0.2.orig/src/Makefile.am	2010-03-24 15:19:47.000000000 +0200
+++ iptraf-ng-1.0.2/src/Makefile.am	2011-01-26 21:36:14.433000336 +0200
@@ -1,6 +1,6 @@
-bin_PROGRAMS = iptraf rvnamed rawtime
+bin_PROGRAMS = iptraf-ng rvnamed-ng rawtime-ng
 
-iptraf_SOURCES = iptraf.c									\
+iptraf_ng_SOURCES = iptraf.c									\
 					itrafmon.c								\
 					packet.c		packet.h				\
 					tcptable.c		tcptable.h				\
@@ -36,28 +36,28 @@
 					addproto.h links.h tcptimeout.h			\
 					arphdr.h descrec.h  externs.h ospf.h attrs.h fltdefs.h logvars.h
 
-iptraf_CFLAGS =	$(CFLAGS)		\
+iptraf_ng_CFLAGS =	$(CFLAGS)		\
 					-Wall		\
 					-std=gnu99	\
 					-pedantic	\
 					-I$(srcdir)/../support
 
-iptraf_LDFLAGS = -l$(ncurses_lib) -l$(panel_lib)
-iptraf_LDADD   = ../support/libtextbox.la
+iptraf_ng_LDFLAGS = -l$(ncurses_lib) -l$(panel_lib)
+iptraf_ng_LDADD   = ../support/libtextbox.la
 
-man_MANS = iptraf.8 rvnamed.8
+man_MANS = iptraf-ng.8 rvnamed-ng.8
 
-rvnamed_SOURCES = rvnamed.c rvnamed.h getpath.c dirs.h getpath.h
+rvnamed_ng_SOURCES = rvnamed.c rvnamed.h getpath.c dirs.h getpath.h
 
-rvnamed_CFLAGS = $(CFLAGS)		\
+rvnamed_ng_CFLAGS = $(CFLAGS)		\
 					-Wall		\
 					-std=gnu99	\
 					-pedantic	\
 					-I$(srcdir)/../support
 
-rawtime_SOURCES = rawtime.c
+rawtime_ng_SOURCES = rawtime.c
 
-rawtime_CFLAGS = $(CFLAGS)		\
+rawtime_ng_CFLAGS = $(CFLAGS)		\
 					-Wall		\
 					-std=gnu99	\
 					-pedantic	\
