diff -Naur gst-plugins-bad-0.10.23.orig/ext/modplug/gstmodplug.cc gst-plugins-bad-0.10.23/ext/modplug/gstmodplug.cc
--- gst-plugins-bad-0.10.23.orig/ext/modplug/gstmodplug.cc	2012-01-18 12:32:55.000000000 +0000
+++ gst-plugins-bad-0.10.23/ext/modplug/gstmodplug.cc	2015-02-03 16:31:44.428094306 +0000
@@ -50,7 +50,7 @@
 #define WORDS_BIGENDIAN 0
 #endif
 
-#include <stdafx.h>
+#include <libmodplug/stdafx.h>
 #include <libmodplug/sndfile.h>
 
 #include "gstmodplug.h"
