diff -Nur kvm-kmod-3.2.orig/scripts/65-kvm.rules kvm-kmod-3.2/scripts/65-kvm.rules
--- kvm-kmod-3.2.orig/scripts/65-kvm.rules	2012-01-08 07:02:08.000000000 -0600
+++ kvm-kmod-3.2/scripts/65-kvm.rules	2012-02-12 11:59:06.317519510 -0600
@@ -1,2 +1,2 @@
-KERNEL=="kvm", MODE="0660", GROUP="kvm"
+KERNEL=="kvm", MODE="0660", GROUP="@KVMGROUP@"
 ACTION=="add|change", SUBSYSTEM=="dmi", KERNEL=="id", RUN+="/bin/sh -c 'grep -q vmx /proc/cpuinfo && /sbin/modprobe kvm-intel; grep -q svm /proc/cpuinfo && /sbin/modprobe kvm-amd'"
