--- setup.py.orig	2022-03-04 16:47:38.000000000 +1000
+++ setup.py	2024-08-16 19:58:40.508399337 +1000
@@ -23,13 +23,6 @@
     build_java = "ON" if get_build_env_var_by_name("java") else "OFF"
 
     install_requires = [
-        'numpy>=1.13.3; python_version<"3.7"',
-        'numpy>=1.14.5; python_version>="3.7"',
-        'numpy>=1.17.3; python_version>="3.8"',
-        'numpy>=1.19.3; python_version>="3.9"',
-        'numpy>=1.21.2; python_version>="3.10"',
-        'numpy>=1.19.3; python_version>="3.6" and platform_system=="Linux" and platform_machine=="aarch64"',
-        'numpy>=1.21.2; python_version>="3.6" and platform_system=="Darwin" and platform_machine=="arm64"',
     ]
 
     python_version = cmaker.CMaker.get_python_version()
@@ -451,7 +444,7 @@
             data_files,
             # To get around a check that prepends source dir to paths and breaks package detection code.
             cmake_source_dir="",
-            cmake_install_dir=cmake_install_reldir,
+            _cmake_install_dir=cmake_install_reldir,
         )
 
 
