diff -ru guitarix-0.47.0.orig/wscript guitarix-0.47.0/wscript
--- guitarix-0.47.0.orig/wscript	2025-10-17 19:30:30.505748521 +0200
+++ guitarix-0.47.0/wscript	2025-10-17 19:32:27.524851009 +0200
@@ -182,14 +182,6 @@
 # configuration helper functions
 #
 def check_boost(conf):
-    code="""
-    #include <boost/system/error_code.hpp>
-    int main() { boost::system::error_code c; }
-    """
-    msg = "Checking for boost-system "
-    conf.check_cxx(msg = msg, fragment=code, lib="boost_system", uselib_store="BOOST_SYSTEM", mandatory=1,includes='/opt/local/include', libpath='/opt/local/lib')
-    # some boost (>1.49) versions depend on boost-system so we will check for it first
-    # and later link against boost_system were boost headers are included.
     boost_atleast_version = 104200
     boost_atleast_vermsg = "1.42"
     code="""
@@ -201,12 +193,10 @@
     #endif
     int main(){ return 0; }
     """ % boost_atleast_version
-    msg = "Checking for boost >= %s" % boost_atleast_vermsg
-    conf.check_cxx(msg = msg, fragment=code, lib="boost_system", mandatory=1)
     #conf.check_cxx(msg = msg, fragment=code, lib="boost_system-mt", mandatory=1,includes='/opt/local/include', libpath='/opt/local/lib')
 
     msg = "Checking for boost_iostreams >= %s" % boost_atleast_vermsg
-    conf.check_cxx(msg = msg, fragment=code, lib=["boost_iostreams","boost_system"], uselib_store="BOOST_IOSTREAMS", mandatory=1)
+    conf.check_cxx(msg = msg, fragment=code, lib=["boost_iostreams"], uselib_store="BOOST_IOSTREAMS", mandatory=1)
     #conf.check_cxx(msg = msg, fragment=code, lib=["boost_iostreams","boost_system-mt"], uselib_store="BOOST_IOSTREAMS", mandatory=1, includes='/opt/local/include', libpath='/opt/local/lib')
  
     #try:
