diff -rupN Vertex-Maia.orig/gtk-2.0/gtkrc Vertex-Maia/gtk-2.0/gtkrc
--- Vertex-Maia.orig/gtk-2.0/gtkrc	2016-04-16 02:17:41.000000000 +0200
+++ Vertex-Maia/gtk-2.0/gtkrc	2016-04-17 16:08:29.546981860 +0200
@@ -2381,6 +2383,18 @@ style "xfwm-tabwin-button" {
   bg[SELECTED] = @selected_bg_color
 }
 
+style "darkback" {
+   bg[NORMAL] = "#454947"
+   bg[ACTIVE] = "#1A1A1A"
+   fg[NORMAL] = "#f9f9f9"
+}
+
+style "darktree" {
+   base[PRELIGHT] = "#436E65"
+   text[NORMAL] = "#1A1A1A"
+   text[ACTIVE] = "#ffffff"
+}
+
 # Chromium
 style "chrome-gtk-frame" {
 
@@ -2503,3 +2517,5 @@ widget "*libreoffice-toplevel*GtkSpinBut
 widget_class "*XfdesktopIconView*"                                  style "xfdesktop-icon-view"
 widget "xfwm4-tabwin*"                                              style "xfwm-tabwin"
 widget "xfwm4-tabwin*GtkButton*"                                    style "xfwm-tabwin-button"
+widget "whiskermenu-window*"                                        style "darkback"
+widget "whiskermenu-window*TreeView*"                               style "darktree"
