--- cngpij/cngpij/bjcups.c.orig	2013-07-17 11:33:45.000000000 +1000
+++ cngpij/cngpij/bjcups.c	2019-03-05 21:47:19.748034963 +1000
@@ -21,6 +21,7 @@
 #include	<config.h>
 #endif	// HAVE_CONFIG_H
 
+#define _IPP_PRIVATE_STRUCTURES 1
 #include <cups/cups.h>
 #include <cups/language.h>
 #include <cups/ppd.h>
--- cngpijmnt/src/main.c.orig	2013-07-17 11:33:45.000000000 +1000
+++ cngpijmnt/src/main.c	2019-03-05 21:53:32.803734632 +1000
@@ -21,6 +21,7 @@
 #include	<config.h>
 #endif	// HAVE_CONFIG_H
 
+#define _IPP_PRIVATE_STRUCTURES 1
 #include <cups/cups.h>
 #include <cups/language.h>
 #include <cups/ppd.h>
