diff -Naur gzdoom-g2.4.0/game-music-emu/CMakeLists.txt gzdoom-g2.4.0.patched/game-music-emu/CMakeLists.txt
--- gzdoom-g2.4.0/game-music-emu/CMakeLists.txt	2017-03-19 16:37:06.000000000 -0400
+++ gzdoom-g2.4.0.patched/game-music-emu/CMakeLists.txt	2021-10-09 14:02:12.973117056 -0400
@@ -89,7 +89,7 @@
    if (__LIBGME_TEST_VISIBILITY)
       # get the gcc version
       exec_program(${CMAKE_CXX_COMPILER} ARGS --version OUTPUT_VARIABLE _gcc_version_info)
-      string (REGEX MATCH "[3-9]\\.[0-9]\\.[0-9]" _gcc_version "${_gcc_version_info}")
+      string (REGEX MATCH "[1-9][0-9]*\\.[0-9]\\.[0-9]" _gcc_version "${_gcc_version_info}")
 
       # gcc <4.1 had poor support for symbol visibility
       if ((${_gcc_version} VERSION_GREATER "4.1") OR (${_gcc_version} VERSION_EQUAL "4.1"))
diff -Naur gzdoom-g2.4.0/src/sound/oalsound.cpp gzdoom-g2.4.0.patched/src/sound/oalsound.cpp
--- gzdoom-g2.4.0/src/sound/oalsound.cpp	2017-03-19 16:37:06.000000000 -0400
+++ gzdoom-g2.4.0.patched/src/sound/oalsound.cpp	2021-10-09 13:59:39.997130962 -0400
@@ -40,6 +40,7 @@
 #include <dlfcn.h>
 #endif
 
+#include <functional>
 #include <memory>
 #include <chrono>
 
