diff -Naur gzdoom-g2.1.1.orig/src/gitinfo.cpp gzdoom-g2.1.1/src/gitinfo.cpp
--- gzdoom-g2.1.1.orig/src/gitinfo.cpp	2016-02-23 04:20:43.000000000 -0500
+++ gzdoom-g2.1.1/src/gitinfo.cpp	2016-08-03 15:18:12.164389643 -0400
@@ -44,22 +44,15 @@
 
 const char *GetGitHash()
 {
-	return GIT_HASH;
+	return "release";
 }
 
 const char *GetGitTime()
 {
-	return GIT_TIME;
+	return __DATE__;
 }
 
 const char *GetVersionString()
 {
-	if (GetGitDescription()[0] == '\0')
-	{
-		return VERSIONSTR;
-	}
-	else
-	{
-		return GIT_DESCRIPTION;
-	}
+	return VERSIONSTR;
 }
