diff -uar haruna-0.12.3.orig/src/mpv/mpvitem.cpp haruna-0.12.3/src/mpv/mpvitem.cpp
--- haruna-0.12.3.orig/src/mpv/mpvitem.cpp  2023-11-15 20:44:52.000000000 +0200
+++ haruna-0.12.3/src/mpv/mpvitem.cpp   2024-06-02 13:43:21.945207485 +0300
@@ -116,6 +116,8 @@
     //    setProperty(QStringLiteral("terminal"), InformationSettings::mpvLogging());
     //    setProperty(QStringLiteral("msg-level"), QStringLiteral("all=v"));
 
+    setProperty(MpvProperties::self()->VO, QStringLiteral("libmpv"));
+
     QString hwdec = PlaybackSettings::useHWDecoding() ? PlaybackSettings::hWDecoding() : QStringLiteral("no");
     setProperty(MpvProperties::self()->HardwareDecoding, hwdec);
     setProperty(MpvProperties::self()->VolumeMax, QStringLiteral("100"));
diff -uar haruna-0.12.3.orig/src/mpv/mpvpreview.cpp haruna-0.12.3/src/mpv/mpvpreview.cpp
--- haruna-0.12.3.orig/src/mpv/mpvpreview.cpp   2023-11-15 20:44:52.000000000 +0200
+++ haruna-0.12.3/src/mpv/mpvpreview.cpp    2024-06-02 13:40:21.902051345 +0300
@@ -11,6 +11,7 @@
 
 MpvPreview::MpvPreview()
 {
+    setProperty(MpvProperties::self()->VO, QStringLiteral("libmpv"));
     mpv_observe_property(m_mpv, 0, "time-pos", MPV_FORMAT_DOUBLE);
 
     setProperty(MpvProperties::self()->Mute, true);
diff -uar haruna-0.12.3.orig/src/mpv/mpvproperties.h haruna-0.12.3/src/mpv/mpvproperties.h
--- haruna-0.12.3.orig/src/mpv/mpvproperties.h  2023-11-15 20:44:52.000000000 +0200
+++ haruna-0.12.3/src/mpv/mpvproperties.h   2024-06-02 13:43:04.678619289 +0300
@@ -20,6 +20,9 @@
         return &p;
     }
 
+    Q_PROPERTY(QString VO MEMBER Pause CONSTANT)
+    const QString VO{QStringLiteral("vo")};
+
     Q_PROPERTY(QString Pause MEMBER Pause CONSTANT)
     const QString Pause{QStringLiteral("pause")};
 
