diff -Naur open-iscsi-2.0.875.orig/iscsiuio/src/unix/libs/bnx2.c open-iscsi-2.0.875/iscsiuio/src/unix/libs/bnx2.c
--- open-iscsi-2.0.875.orig/iscsiuio/src/unix/libs/bnx2.c	2017-09-01 20:05:23.000000000 +0200
+++ open-iscsi-2.0.875/iscsiuio/src/unix/libs/bnx2.c	2019-09-06 14:24:47.943039000 +0200
@@ -47,6 +47,7 @@
 #include <sys/user.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <sys/sysmacros.h>
 
 #include "config.h"
 
diff -Naur open-iscsi-2.0.875.orig/iscsiuio/src/unix/libs/bnx2x.c open-iscsi-2.0.875/iscsiuio/src/unix/libs/bnx2x.c
--- open-iscsi-2.0.875.orig/iscsiuio/src/unix/libs/bnx2x.c	2017-09-01 20:05:23.000000000 +0200
+++ open-iscsi-2.0.875/iscsiuio/src/unix/libs/bnx2x.c	2019-09-06 14:24:23.527039000 +0200
@@ -51,6 +51,7 @@
 #include <sys/user.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <sys/sysmacros.h>
 
 #include "config.h"
 
diff -Naur open-iscsi-2.0.875.orig/iscsiuio/src/unix/libs/qedi.c open-iscsi-2.0.875/iscsiuio/src/unix/libs/qedi.c
--- open-iscsi-2.0.875.orig/iscsiuio/src/unix/libs/qedi.c	2017-09-01 20:05:23.000000000 +0200
+++ open-iscsi-2.0.875/iscsiuio/src/unix/libs/qedi.c	2019-09-06 14:29:00.780039000 +0200
@@ -58,6 +58,7 @@
 #include <sys/user.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <sys/sysmacros.h>
 
 #include "config.h"
 
