--- corectrl-v1.3.0/src/CMakeLists.txt	2022-09-18 12:56:52.000000000 -0500
+++ corectrl-v1.3.0/src/CMakeLists.txt.new	2022-10-05 11:51:27.147208476 -0500
@@ -417,7 +417,7 @@
 )
 
 install(TARGETS corectrl corectrl_lib
-        RUNTIME DESTINATION bin
-        LIBRARY DESTINATION lib)
+        RUNTIME DESTINATION ${CMAKE_INSTALL_FULL_BINDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR})
 
 add_subdirectory(helper)
