diff -Naur osrf-sdformat-e5ac5aaa49f2.orig/CMakeLists.txt osrf-sdformat-e5ac5aaa49f2/CMakeLists.txt
--- osrf-sdformat-e5ac5aaa49f2.orig/CMakeLists.txt	2016-10-10 23:51:37.000000000 +0200
+++ osrf-sdformat-e5ac5aaa49f2/CMakeLists.txt	2017-07-29 19:46:16.021921859 +0200
@@ -186,11 +186,13 @@
 # OS Specific initialization
 if (UNIX)
   sdf_setup_unix()
-else (WIN32)
+endif ()
+if (WIN32)
   sdf_setup_windows()
-else (APPLE)
+endif ()
+if (APPLE)
   sdf_setup_apple()
-endif()
+endif ()
 
 #################################################
 # Print warnings and errors
