--- /tmp/SBo/tinytldr-1.2.2/Makefile	2026-07-29 14:04:03.000000000 +0100
+++ 0/patched/Makefile.new	2026-07-29 22:20:40.983897402 +0100
@@ -6,7 +6,7 @@
 LIBS += $(LIBNOTIFY_LIBS)
 INCS += $(LIBNOTIFY_INCS)
 
-CFLAGS += -std=c23
+CFLAGS += -std=c2x
 CFLAGS += -pedantic
 CFLAGS += -Wall
 CFLAGS += -Wextra
@@ -16,7 +16,8 @@
 CFLAGS += -Wno-format-truncation
 
 CFLAGS += -D_XOPEN_SOURCE=500
-CFLAGS += -DGIT_DESC=\"$(shell git describe --tags --always --dirty)\"
+#CFLAGS += -DGIT_DESC=\"$(shell git describe --tags --always --dirty)\"
+CFLAGS += -DGIT_DESC=\"Slackware\"
 
 CFLAGS += $(INCS)
 LDFLAGS += $(LIBS)
