diff -Naur rakarrack-git-a6208406d94a1da978f435605072ee5caefe1491/src/rakarrack.cxx rakarrack-git-a6208406d94a1da978f435605072ee5caefe1491.patched/src/rakarrack.cxx
--- rakarrack-git-a6208406d94a1da978f435605072ee5caefe1491/src/rakarrack.cxx	2015-08-14 00:38:44.000000000 -0400
+++ rakarrack-git-a6208406d94a1da978f435605072ee5caefe1491.patched/src/rakarrack.cxx	2024-12-06 18:51:32.829129235 -0500
@@ -23583,7 +23583,7 @@
     Fl_Widget *m = fl_message_icon();
     m->parent()->copy_label(rkr->jackcliname);
     sprintf(temp2,"Overwrite \"%s\"?",w->label());
-    ok=fl_choice(temp2, "No","Yes", NULL);
+    ok=fl_choice("%s",temp2, "No","Yes", NULL);
    if (!ok)
    { 
    o->value(0);
@@ -24850,7 +24850,7 @@
    w->parent()->copy_label(rkr->jackcliname);
    
   
-   ok=fl_choice("Bank was modified, but not saved", "Discard","Save",NULL);
+   ok=fl_choice("%s","Bank was modified, but not saved", "Discard","Save",NULL);
   
   
   
@@ -28008,7 +28008,7 @@
   } 
   
   sprintf(temp2,"Delete? \"%s\"",s->text());
-  ok=fl_choice("%d",temp2,"No","Yes",NULL);
+  ok=fl_choice("%s",temp2,"No","Yes",NULL);
   if (!ok) return;
   memset(Rname,0,sizeof(Rname));
   sprintf(Rname,"%s",s->text());
