--- scangearmp2-source-4.90-1/scangearmp2/include/libcnnet2_type.h	2024-12-18 20:35:47.000000000 +1300
+++ scangearmp2-source-4.90-1.new/scangearmp2/include/libcnnet2_type.h	2025-11-02 12:54:36.401900773 +1300
@@ -47,9 +47,11 @@
   CNNET2_SETTING_FLAG_DISCOVER_PACKET_WAIT_MILLIS,        
 } CNNET2_SETTING_FLAGS;
 
-#ifndef __cplusplus
-typedef char bool;
-#endif
+#if defined(__GNUC__) && __GNUC__ < 15
+  #ifndef __cplusplus
+  typedef char bool;
+  #endif
+#endif
 
 typedef struct {
   int nicIndex_;
