diff -ru libreadline-java-0.8.0.orig/src/native/Makefile libreadline-java-0.8.0/src/native/Makefile
--- libreadline-java-0.8.0.orig/src/native/Makefile	2003-01-07 10:14:35.000000000 +0000
+++ libreadline-java-0.8.0/src/native/Makefile	2014-01-05 22:00:43.569763351 +0000
@@ -51,7 +51,7 @@
 ifeq (cygwin,$(WIN32))
 JavaGetline_LIBS = -lcygwin
 endif
-CC = gcc
+#CC = gcc
 OBJ_EXT := o
 LIB_PRE := lib
 LIB_EXT := so
diff -ru libreadline-java-0.8.0.orig/src/native/org_gnu_readline_Readline.c libreadline-java-0.8.0/src/native/org_gnu_readline_Readline.c
--- libreadline-java-0.8.0.orig/src/native/org_gnu_readline_Readline.c	2003-01-07 10:14:35.000000000 +0000
+++ libreadline-java-0.8.0/src/native/org_gnu_readline_Readline.c	2014-01-05 22:10:54.306920519 +0000
@@ -430,7 +430,7 @@
   jtext = (*jniEnv)->NewStringUTF(jniEnv,text);
 
   if (jniMethodId == 0) {
-    return;
+    return 0;
   }
 
   completion = (*jniEnv)->CallObjectMethod(jniEnv, jniObject,
