diff -rupN podofo-0.9.4/test/unit/EncryptTest.h podofo-0.9.4-new/test/unit/EncryptTest.h
--- podofo-0.9.4/test/unit/EncryptTest.h	2012-12-12 23:55:46.000000000 +0100
+++ podofo-0.9.4-new/test/unit/EncryptTest.h	2016-09-18 10:41:30.476635230 +0200
@@ -83,7 +83,7 @@ class EncryptTest : public CppUnit::Test
 
  private:
   char* m_pEncBuffer;
-  long  m_lLen;
+  PoDoFo::pdf_long  m_lLen;
   int   m_protection;
   
 };
