diff -up podofo-0.9.1/test/unit/TestUtils.cpp.unistd podofo-0.9.1/test/unit/TestUtils.cpp
--- podofo-0.9.1/test/unit/TestUtils.cpp.unistd	2012-01-16 18:12:52.000000000 +0100
+++ podofo-0.9.1/test/unit/TestUtils.cpp	2012-01-16 18:13:00.000000000 +0100
@@ -23,6 +23,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 
 #if defined(_WIN32) || defined(_WIN64)
 #include <windows.h>
