--- a/jack_capture.c	2017-05-19 05:56:00.000000000 -0500
+++ b/jack_capture.c	2025-04-05 23:16:46.219938571 -0500
@@ -909,6 +909,7 @@
 #define vsnprintf _vsnprintf
 #endif
 #ifndef __APPLE__
+#ifndef HAVE_ASPRINTF
 int asprintf(char **buffer, char *fmt, ...) {
     /* Guess we need no more than 200 chars of space. */
     int size = 200;
@@ -947,6 +948,7 @@
 }
 #endif
 #endif
+#endif
 
 #define ARGS_ADD_ARGV(FMT,ARG) \
   argv=(char**) realloc((void*)argv, (argc+2)*sizeof(char*)); \
--- a/gen_das_config_h.sh	2017-05-19 05:56:00.000000000 -0500
+++ b/gen_das_config_h.sh	2025-04-05 23:47:40.843411772 -0500
@@ -54,6 +54,18 @@
     echo "#define NEW_JACK_LATENCY_API 0"
 fi
 
+echo "#include <stdio.h>" >temp$$.c
+echo "int main() {" >>temp$$.c
+echo "    char *buf;" >>temp$$.c
+echo "    asprintf(&buf, "test %d", 42);" >>temp$$.c
+echo "    return 0;" >>temp$$.c
+echo "}" >>temp$$.c
+
+if gcc temp$$.c 2>/dev/null; then
+    echo "#define HAVE_ASPRINTF 1"
+else
+    echo "#define HAVE_ASPRINTF 0"
+fi
 
 rm temp$$.c
 
