--- Makefile	2022-06-02 21:37:14.244285088 +0100
+++ Makefile.new	2022-06-02 22:21:33.238469204 +0100
@@ -477,7 +477,7 @@
 	echo "Generated file is : ZEsarUX_src-$(EMULATOR_VERSION).tar.gz"
 
 bintargz: all
-	./generate_install_sh.sh
+	./generate_install_sh.sh $(DESTDIR)
 	rm -fR bintargztemp
 	mkdir -p bintargztemp/ZEsarUX-$(EMULATOR_VERSION)
 	cp -a $(COMMONFILES) $(ONLYUNIXBINARIES) bintargztemp/ZEsarUX-$(EMULATOR_VERSION)/
@@ -557,7 +557,7 @@
 
 
 install: all
-	./generate_install_sh.sh
+	./generate_install_sh.sh $(DESTDIR)
 	./install.sh
 
 #utilities: $(UTILITIES)
