diff -u -r mongo-cxx-driver-r3.1.1-orig/examples/mongocxx/CMakeLists.txt mongo-cxx-driver-r3.1.1-patched/examples/mongocxx/CMakeLists.txt
--- mongo-cxx-driver-r3.1.1-orig/examples/mongocxx/CMakeLists.txt	2016-12-20 20:23:11.000000000 +0000
+++ mongo-cxx-driver-r3.1.1-patched/examples/mongocxx/CMakeLists.txt	2017-03-14 12:06:28.955217844 +0000
@@ -18,7 +18,7 @@
 )
 
 link_directories(
-    ${CMAKE_INSTALL_PREFIX}/lib
+    ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}
 )
 
 set(MONGOCXX_EXAMPLES
diff -u -r mongo-cxx-driver-r3.1.1-orig/src/bsoncxx/CMakeLists.txt mongo-cxx-driver-r3.1.1-patched/src/bsoncxx/CMakeLists.txt
--- mongo-cxx-driver-r3.1.1-orig/src/bsoncxx/CMakeLists.txt	2016-12-20 20:23:11.000000000 +0000
+++ mongo-cxx-driver-r3.1.1-patched/src/bsoncxx/CMakeLists.txt	2017-03-14 12:04:02.763279762 +0000
@@ -114,7 +114,7 @@
 # library names
 if (WIN32)
     set_target_properties(bsoncxx_static PROPERTIES
-        PREFIX lib
+        PREFIX lib${LIB_SUFFIX}
     )
 endif()
 
@@ -172,25 +172,25 @@
     install(TARGETS
         bsoncxx
         RUNTIME DESTINATION bin COMPONENT runtime
-        LIBRARY DESTINATION lib COMPONENT runtime
-        ARCHIVE DESTINATION lib COMPONENT dev
+        LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT runtime
+        ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT dev
     )
 endif()
 
 install(TARGETS
     bsoncxx_static
-    ARCHIVE DESTINATION lib COMPONENT dev
+    ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT dev
 )
 
 set(PACKAGE_INCLUDE_INSTALL_DIRS ${BSONCXX_HEADER_INSTALL_DIR})
-set(PACKAGE_LIBRARY_INSTALL_DIRS lib)
+set(PACKAGE_LIBRARY_INSTALL_DIRS lib${LIB_SUFFIX})
 set(PACKAGE_LIBRARIES bsoncxx)
 
 include(CMakePackageConfigHelpers)
 
 configure_package_config_file(
   cmake/libbsoncxx-config.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/libbsoncxx-config.cmake
-  INSTALL_DESTINATION lib/cmake/libbsoncxx-${BSONCXX_VERSION}
+  INSTALL_DESTINATION lib${LIB_SUFFIX}/cmake/libbsoncxx-${BSONCXX_VERSION}
   PATH_VARS PACKAGE_INCLUDE_INSTALL_DIRS PACKAGE_LIBRARY_INSTALL_DIRS
 )
 
@@ -202,7 +202,7 @@
 
 install(
   FILES ${CMAKE_CURRENT_BINARY_DIR}/libbsoncxx-config.cmake ${CMAKE_CURRENT_BINARY_DIR}/libbsoncxx-config-version.cmake
-  DESTINATION lib/cmake/libbsoncxx-${BSONCXX_VERSION}
+  DESTINATION lib${LIB_SUFFIX}/cmake/libbsoncxx-${BSONCXX_VERSION}
 )
 
 add_subdirectory(test)
diff -u -r mongo-cxx-driver-r3.1.1-orig/src/bsoncxx/config/CMakeLists.txt mongo-cxx-driver-r3.1.1-patched/src/bsoncxx/config/CMakeLists.txt
--- mongo-cxx-driver-r3.1.1-orig/src/bsoncxx/config/CMakeLists.txt	2016-12-20 20:23:11.000000000 +0000
+++ mongo-cxx-driver-r3.1.1-patched/src/bsoncxx/config/CMakeLists.txt	2017-03-14 12:05:42.959495804 +0000
@@ -42,6 +42,6 @@
 
 install(FILES
     "${CMAKE_CURRENT_BINARY_DIR}/libbsoncxx.pc"
-    DESTINATION lib/pkgconfig
+    DESTINATION lib${LIB_SUFFIX}/pkgconfig
     COMPONENT dev
 )
diff -u -r mongo-cxx-driver-r3.1.1-orig/src/bsoncxx/config/libbsoncxx.pc.in mongo-cxx-driver-r3.1.1-patched/src/bsoncxx/config/libbsoncxx.pc.in
--- mongo-cxx-driver-r3.1.1-orig/src/bsoncxx/config/libbsoncxx.pc.in	2016-12-20 20:23:11.000000000 +0000
+++ mongo-cxx-driver-r3.1.1-patched/src/bsoncxx/config/libbsoncxx.pc.in	2017-03-14 12:38:29.143868341 +0000
@@ -14,7 +14,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 includedir=${prefix}/include
-libdir=${prefix}/lib
+libdir=${prefix}/lib@LIB_SUFFIX@
 
 Name: libbsoncxx
 Description: The MongoDB C++11 BSON Library
diff -u -r mongo-cxx-driver-r3.1.1-orig/src/mongocxx/CMakeLists.txt mongo-cxx-driver-r3.1.1-patched/src/mongocxx/CMakeLists.txt
--- mongo-cxx-driver-r3.1.1-orig/src/mongocxx/CMakeLists.txt	2016-12-20 20:23:11.000000000 +0000
+++ mongo-cxx-driver-r3.1.1-patched/src/mongocxx/CMakeLists.txt	2017-03-14 12:01:22.963720065 +0000
@@ -120,7 +120,7 @@
 # library names
 if (WIN32)
     set_target_properties(mongocxx_static PROPERTIES
-        PREFIX lib
+        PREFIX lib${LIB_SUFFIX}
     )
 endif()
 
@@ -183,25 +183,25 @@
     install(TARGETS
         mongocxx
         RUNTIME DESTINATION bin COMPONENT runtime
-        LIBRARY DESTINATION lib COMPONENT runtime
-        ARCHIVE DESTINATION lib COMPONENT dev
+        LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT runtime
+        ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT dev
     )
 endif()
 
 install(TARGETS
     mongocxx_static
-    ARCHIVE DESTINATION lib COMPONENT dev
+    ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT dev
 )
 
 set(PACKAGE_INCLUDE_INSTALL_DIRS ${MONGOCXX_HEADER_INSTALL_DIR})
-set(PACKAGE_LIBRARY_INSTALL_DIRS lib)
+set(PACKAGE_LIBRARY_INSTALL_DIRS lib${LIB_SUFFIX})
 set(PACKAGE_LIBRARIES mongocxx)
 
 include(CMakePackageConfigHelpers)
 
 configure_package_config_file(
   cmake/libmongocxx-config.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/libmongocxx-config.cmake
-  INSTALL_DESTINATION lib/cmake/libmongocxx-${MONGOCXX_VERSION}
+  INSTALL_DESTINATION lib${LIB_SUFFIX}/cmake/libmongocxx-${MONGOCXX_VERSION}
   PATH_VARS PACKAGE_INCLUDE_INSTALL_DIRS PACKAGE_LIBRARY_INSTALL_DIRS
 )
 
@@ -213,7 +213,7 @@
 
 install(
   FILES ${CMAKE_CURRENT_BINARY_DIR}/libmongocxx-config.cmake ${CMAKE_CURRENT_BINARY_DIR}/libmongocxx-config-version.cmake
-  DESTINATION lib/cmake/libmongocxx-${MONGOCXX_VERSION}
+  DESTINATION lib${LIB_SUFFIX}/cmake/libmongocxx-${MONGOCXX_VERSION}
 )
 
 add_subdirectory(test)
diff -u -r mongo-cxx-driver-r3.1.1-orig/src/mongocxx/config/CMakeLists.txt mongo-cxx-driver-r3.1.1-patched/src/mongocxx/config/CMakeLists.txt
--- mongo-cxx-driver-r3.1.1-orig/src/mongocxx/config/CMakeLists.txt	2016-12-20 20:23:11.000000000 +0000
+++ mongo-cxx-driver-r3.1.1-patched/src/mongocxx/config/CMakeLists.txt	2017-03-14 12:02:27.633923054 +0000
@@ -42,6 +42,6 @@
 
 install(FILES
     "${CMAKE_CURRENT_BINARY_DIR}/libmongocxx.pc"
-    DESTINATION lib/pkgconfig
+    DESTINATION lib${LIB_SUFFIX}/pkgconfig
     COMPONENT dev
 )
diff -u -r mongo-cxx-driver-r3.1.1-orig/src/mongocxx/config/libmongocxx.pc.in mongo-cxx-driver-r3.1.1-patched/src/mongocxx/config/libmongocxx.pc.in
--- mongo-cxx-driver-r3.1.1-orig/src/mongocxx/config/libmongocxx.pc.in	2016-12-20 20:23:11.000000000 +0000
+++ mongo-cxx-driver-r3.1.1-patched/src/mongocxx/config/libmongocxx.pc.in	2017-03-14 12:38:14.955262043 +0000
@@ -14,7 +14,7 @@
 
 prefix=@CMAKE_INSTALL_PREFIX@
 includedir=${prefix}/include
-libdir=${prefix}/lib
+libdir=${prefix}/lib@LIB_SUFFIX@
 
 Name: libmongocxx
 Description: The MongoDB C++11 Driver Library
