diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2164,7 +2164,7 @@
   [], [], [SuiteSparse_time], [],
   [don't link directly with suitesparseconfig library],
   [],
-  SuiteSparse_config)
+  suitesparseconfig)
 LIBS="$save_LIBS"
 
 if test -n "$SUITESPARSECONFIG_LIBS" && test -n "$LIB_CLOCK_GETTIME"; then
@@ -2183,7 +2183,7 @@
   [amd_postorder],
   [], [don't use AMD library, disable some sparse matrix functionality],
   [],
-  AMD)
+  amd)
 
 ### Check for CAMD library.
 
@@ -2193,7 +2193,7 @@
   [camd_postorder],
   [], [don't use CAMD library, disable some sparse matrix functionality],
   [],
-  CAMD)
+  camd)
 
 ### Check for COLAMD library.
 
@@ -2203,7 +2203,7 @@
   [colamd],
   [], [don't use COLAMD library, disable some sparse matrix functionality],
   [],
-  COLAMD)
+  colamd)
 
 ### Check for CCOLAMD library.
 
@@ -2213,7 +2213,7 @@
   [ccolamd],
   [], [don't use CCOLAMD library, disable some sparse matrix functionality],
   [],
-  CCOLAMD)
+  ccolamd)
 
 ### Check for CHOLMOD library.
 dnl If your cholmod library requires cblas, then you will need to configure
@@ -2227,7 +2227,7 @@
   [cholmod${CHOLMOD_TAG}start],
   [], [don't use CHOLMOD library, disable some sparse matrix functionality],
   [],
-  CHOLMOD)
+  cholmod)
 LIBS="$save_LIBS"
 
 ### Check for SPQR library.
@@ -2240,7 +2240,7 @@
   [SuiteSparseQR_C],
   [C++], [don't use SPQR library, disable some sparse matrix functionality],
   [],
-  SPQR)
+  spqr)
 LIBS="$save_LIBS"
 
 ### Check for CXSparse library.
@@ -2251,7 +2251,7 @@
   [cs${CXSPARSE_TAG}sqr],
   [C++], [don't use CXSparse library, disable some sparse matrix functionality],
   [],
-  CXSparse)
+  cxsparse)
 
 if test -n "$CXSPARSE_LIBS"; then
   ## Additional check for CXSparse version >= 2.2
@@ -2275,7 +2275,7 @@
   [umfpack${UMFPACK_TAG}get_determinant],
   [], [don't use UMFPACK, disable some sparse matrix functionality],
   [],
-  UMFPACK)
+  umfpack)
 CPPFLAGS="$save_CPPFLAGS"
 LIBS="$save_LIBS"
 
@@ -2291,7 +2291,7 @@
     [umfpack${UMFPACK_TAG}get_determinant],
     [], [don't use UMFPACK, disable some sparse matrix functionality],
     [],
-    UMFPACK)
+    umfpack)
   if test -n "$UMFPACK_LIBS"; then
     UMFPACK_LIBS="$UMFPACK_LIBS -lcblas"
   fi
@@ -2305,7 +2305,7 @@
   [klu.h suitesparse/klu.h klu/klu.h], [klu_solve],
   [], [don't use KLU library, disable some sparse matrix functionality],
   [],
-  KLU)
+  klu)
 
 ## Restore LIBS after all suitesparse configuration tests
 LIBS="$suitesparse_save_LIBS"
