--- kicad-bzr_20100608/common/edaappl.cpp	2010-09-25 08:49:59.000000000 -0300
+++ kicad-bzr_20100608_patched/common/edaappl.cpp	2010-09-25 13:46:19.000000000 -0300
@@ -891,7 +891,6 @@
     // from a distribution files are in /usr/share/doc/kicad/help and binaries
     // in /usr/bin or /usr/local/bin
     subdirs.Add( wxT( "share" ) );
-    subdirs.Add( _T( "doc" ) );
     subdirs.Add( wxT( "kicad" ) );
     subdirs.Add( _T( "help" ) );
 
