--- kicad-bzr_20101223/common/edaappl.cpp	2010-12-31 17:48:16.000000000 -0200
+++ kicad-bzr_20101223_patched/common/edaappl.cpp	2010-12-31 17:49:46.000000000 -0200
@@ -903,7 +903,6 @@
     // from a distribution files are in /usr/share/doc/kicad/help and binaries
     // in /usr/bin or /usr/local/bin
     subdirs.Add( wxT( "share" ) );
-    subdirs.Add( _T( "doc" ) );
     subdirs.Add( wxT( "kicad" ) );
     subdirs.Add( _T( "help" ) );
 
