--- kicad-bzr_20120521/include/boost/polygon/polygon_90_set_data.hpp	2012-09-14 12:19:08.000000000 -0300
+++ kicad-bzr_20120521_patched/include/boost/polygon/polygon_90_set_data.hpp	2012-09-14 12:55:52.000000000 -0300
@@ -16,6 +16,7 @@
 #include "detail/iterator_points_to_compact.hpp"
 #include "detail/iterator_compact_to_points.hpp"
 #include "polygon_traits.hpp"
+#include "detail/polygon_sort_adaptor.hpp"
 
 //manhattan boolean algorithms
 #include "detail/boolean_op.hpp"
