--- kicad-bzr_20101223_orig/common/edaappl.cpp	2011-01-01 14:30:14.000000000 -0200
+++ kicad-bzr_20101223/common/edaappl.cpp	2011-01-01 14:29:33.000000000 -0200
@@ -680,7 +680,7 @@
         delete m_Locale;
     m_Locale = new wxLocale;
 
-    if( !m_Locale->Init( m_LanguageId ) )
+    if( !m_Locale->Init( m_LanguageId, wxLOCALE_CONV_ENCODING ) )
     {
         wxLogDebug( wxT("This language is not supported by the system.") );
 
