Description: fix FTBFS on kFreeBSD
Author: Petr Salinger <Petr.Salinger@seznam.cz>
Bug-Debian: http://bugs.debian.org/537883
Forwarded: no
Last-Update: 2013-10-30

--- a/utils/wvcrash.cc
+++ b/utils/wvcrash.cc
@@ -26,7 +26,7 @@
 #endif
 
 // FIXME: this file mostly only works in Linux
-#ifdef __linux
+#if 1
 
 # include <execinfo.h>
 #include <unistd.h>
--- a/utils/wvcrashbase.cc
+++ b/utils/wvcrashbase.cc
@@ -66,7 +66,7 @@
 
 
 // FIXME: leaving of a will and catching asserts mostly only works in Linux
-#ifdef __linux
+#if 1
 
 #ifdef __USE_GNU
 static const char *argv0 = program_invocation_short_name;
