Description: fix return type checking of X509_REQ_verify
Author: Moritz Mühlenhoff <jmm@debian.org>
Bug: https://code.google.com/p/wvstreams/issues/detail?id=40
Bug-Debian: http://bugs.debian.org/513539
Forwarded: yes
Last-Update: 2013-10-30

--- a/crypto/wvx509.cc
+++ b/crypto/wvx509.cc
@@ -325,7 +325,7 @@
     }
 
     int verify_result = X509_REQ_verify(certreq, pk);
-    if (verify_result == 0)
+    if (verify_result == 0 || verify_result == -1)
     {
 	debug(WvLog::Warning, "Self signed request failed");
 	X509_REQ_free(certreq);
