Description: make it possible to disable readline
Author: Jakub Wilk <jwilk@debian.org>
Bug: https://code.google.com/p/wvstreams/issues/detail?id=38
Forwarded: no
Last-Update: 2013-10-30

--- a/configure.ac
+++ b/configure.ac
@@ -59,6 +59,7 @@
 
 AC_ARG_WITH(dbus, AC_HELP_STRING([--with-dbus], [DBUS]))
 AC_ARG_WITH(openssl, AC_HELP_STRING([--with-openssl], [OpenSSL >= 0.9.7 (required)]))
+AC_ARG_WITH(readline, AC_HELP_STRING([--with-readline], [readline]))
 AC_ARG_WITH(pam, AC_HELP_STRING([--with-pam], [PAM]))
 AC_ARG_WITH(tcl, AC_HELP_STRING([--with-tcl], [Tcl]))
 AC_ARG_WITH(qt, AC_HELP_STRING([--with-qt], [Qt]))
