Description: fix typos
Author: Jakub Wilk <jwilk@debian.org>
Bug: https://code.google.com/p/wvstreams/issues/detail?id=39
Forwared: not-needed
Last-Update: 2013-10-30

--- a/ChangeLog
+++ b/ChangeLog
@@ -745,7 +745,7 @@
     "wvtestmain ''" actually still runs all the tests, so skip the warning.
     
     wvtest.cc prints a message "WARNING: WvTest: only ran tests starting
-    with specifed prefix(es)" if you give it any parameters restricting the
+    with specified prefix(es)" if you give it any parameters restricting the
     tests to be run.  But 'make runtests' actually sometimes provides an
     existing but empty parameter, which doesn't restrict anything because *any*
     test starts with the empty string.  So suppress the warning in that case;
--- a/utils/wvtest.cc
+++ b/utils/wvtest.cc
@@ -252,7 +252,7 @@
     
     if (prefixes && *prefixes && **prefixes)
 	printf("WvTest: WARNING: only ran tests starting with "
-	       "specifed prefix(es).\n");
+	       "specified prefix(es).\n");
     else
 	printf("WvTest: ran all tests.\n");
     printf("WvTest: %d test%s, %d failure%s.\n",
--- a/uniconf/tests/uni.8.subst
+++ b/uniconf/tests/uni.8.subst
@@ -78,7 +78,7 @@
 .SH COMMANDS
 .TP
 get
-Retreive the
+Retrieve the
 .I VALUE
 associated with the provided
 .I KEY
