--- amavisd-signer.orig	2012-05-04 22:27:39.397598579 -0500
+++ amavisd-signer	2012-05-04 22:28:55.609603047 -0500
@@ -83,11 +83,11 @@
 # Please adjust the following settings as necessary:
 #
 
-$daemon_user  = 'vscan';
-$daemon_group = 'vscan';
+$daemon_user  = 'amavis';
+$daemon_group = 'amavis';
 # $daemon_chroot_dir = '/var/amavis';   # chroot directory or undef
 
-# $daemonize = 1;
+$daemonize = 1;
 
 $log_level = 2;  # 0..5
 $syslog_facility = 'mail';
