--- config.mk.orig	2010-12-03 20:58:19.000000000 +1000
+++ config.mk	2011-01-29 22:12:00.937180408 +1000
@@ -5,7 +5,7 @@
 #MANCOUNTRIES=en_GB
 
 CFLAGS=-I. -I.. -ggdb -Wall -O2 -I../lib
-LDFLAGS=-lsqlite3
+LDFLAGS=-lsqlite3 -ldl
 # Add -lwrap to LDFLAGS if compiling with tcp wrappers support.
 
 CC=gcc
@@ -15,6 +15,6 @@
 MSGFMT=msgfmt
 DOCBOOK2MAN=docbook2man.pl
 
-prefix=/usr/local
-mandir=${prefix}/share/man
+prefix=/usr
+mandir=${prefix}/man
 localedir=${prefix}/share/locale
