--- src/CMakeLists.txt.orig	2015-02-18 11:44:09.000000000 +1000
+++ src/CMakeLists.txt	2015-02-21 14:24:10.423277309 +1000
@@ -113,7 +113,7 @@
 	if (APPLE)
 		set_target_properties(mosquitto PROPERTIES LINK_FLAGS "-Wl,-exported_symbols_list -Wl,${mosquitto_SOURCE_DIR}/src/linker-macosx.syms")
 	else (APPLE)
-		set_target_properties(mosquitto PROPERTIES LINK_FLAGS "-Wl,-exported_symbols_list -Wl,${mosquitto_SOURCE_DIR}/src/linker.syms")
+		set_target_properties(mosquitto PROPERTIES LINK_FLAGS "-Wl,-dynamic-list -Wl,${mosquitto_SOURCE_DIR}/src/linker.syms")
 	endif (APPLE)
 endif (UNIX)
 
