--- a/gwyddion-2.26/plugins/process/Makefile.am	2007-10-18 20:06:34.000000000 +0700
+++ b/gwyddion-2.26/plugins/process/Makefile.am	2012-02-11 12:11:13.000000000 +0700
@@ -84,7 +84,7 @@
 	$(cpp_plugins_sources) \
 	$(all_rgis)
 
-PASCAL_FLAGS = -Mdelphi -l- -Sh -WC -O2 -Xc -XX
+PASCAL_FLAGS = -Mdelphi -l- -Sh -O2 -Xc -XX
 
 clean-local:
 	rm -f core.* *~ *.ppu
--- a/gwyddion-2.26/plugins/process/Makefile.in	2011-12-17 16:58:01.000000000 +0700
+++ b/gwyddion-2.26/plugins/process/Makefile.in	2012-02-11 12:10:59.000000000 +0700
@@ -401,7 +401,7 @@
 	$(cpp_plugins_sources) \
 	$(all_rgis)
 
-PASCAL_FLAGS = -Mdelphi -l- -Sh -WC -O2 -Xc -XX
+PASCAL_FLAGS = -Mdelphi -l- -Sh -O2 -Xc -XX
 all: all-am
 
 .SUFFIXES:
