--- gwyddion-2.46/libdraw/gwyselection.c
+++ gwyddion/libdraw/gwyselection.c
@@ -638,7 +638,8 @@
 
     if (nselected) {
         g_return_if_fail(data);
-        memcpy((gdouble*)selection->objects->data, data, nselected*object_size);
+        gwy_assign((gdouble*)selection->objects->data, data,
+                   nselected*object_size);
     }
     selection->n = nselected;
     g_signal_emit(selection, selection_signals[CHANGED], 0, -1);
