Index: modules/tools/stats.c
===================================================================
--- modules/tools/stats.c	(revision 16602)
+++ modules/tools/stats.c	(working copy)
@@ -1096,7 +1096,7 @@
     area = report_data->same_units ? fmt_val(area) : g_strdup(_("N.A."));
     projarea = fmt_val(projarea);
 
-    vf = gwy_si_unit_get_format_with_digits(siunitarea,
+    vf = gwy_si_unit_get_format_with_digits(siunitvar,
                                             GWY_SI_UNIT_FORMAT_PLAIN,
                                             report_data->results.var, 3, vf);
     var = fmt_val(var);
