Description: tint2conf also uses libm

Bug-Debian: https://bugs.debian.org/1082699
Forwarded: no
Last-Update: 2024-12-18

--- tint2-17.0.1.orig/src/tint2conf/CMakeLists.txt
+++ tint2-17.0.1/src/tint2conf/CMakeLists.txt
@@ -65,6 +65,7 @@ target_link_libraries( tint2conf ${X11_T
                                  ${GTHREAD2_LIBRARIES}
                                  ${GTK3_LIBRARIES}
                                  ${RSVG_LIBRARIES} )
+target_link_libraries( tint2conf m )
 
 add_definitions( -DINSTALL_PREFIX=\"${CMAKE_INSTALL_PREFIX}\" )
 add_definitions( -DLOCALEDIR=\"${CMAKE_INSTALL_FULL_LOCALEDIR}\" )
