diff -ur audtty.orig/Makefile.in audtty/Makefile.in
--- audtty.orig/Makefile.in	2008-05-08 09:50:26.000000000 +0300
+++ audtty/Makefile.in	2008-07-05 09:36:31.000000000 +0300
@@ -20,18 +20,18 @@
 all: audtty
 
 audtty: ${BINS}
-	cc -Wall -lncurses -laudclient ${LDFLAGS} -o audtty $(BINS)
+	$(CC) -Wall ${LDFLAGS} -o audtty $(BINS) -lncurses -laudclient
 
 .c.o:
-	cc -Wall ${AUDACIOUS_CFLAGS} ${CFLAGS} -o $@ -c $<
+	$(CC) -Wall ${AUDACIOUS_CFLAGS} ${CFLAGS} -o $@ -c $<
 
 install: audtty
-	mkdir -p ${bindir}
-	install -m 0755 audtty ${bindir}/audtty
-	mkdir -p ${mandir}
-	install -m 0644 audtty.1 ${mandir}/audtty.1
-	mkdir -p ${sysconfdir}
-	install -m 0644 audtty.conf ${sysconfdir}/audtty.conf
+	mkdir -p $(DESTDIR)${bindir}
+	install -m 0755 audtty $(DESTDIR)${bindir}/audtty
+	mkdir -p $(DESTDIR)${mandir}
+	install -m 0644 audtty.1 $(DESTDIR)${mandir}/audtty.1
+	mkdir -p $(DESTDIR)${sysconfdir}
+	install -m 0644 audtty.conf $(DESTDIR)${sysconfdir}/audtty.conf
 
 uninstall:
 	rm ${bindir}/audtty || false
