--- cd-discid-1.4.orig/Makefile	2013-05-14 05:02:00.000000000 -0400
+++ cd-discid-1.4/Makefile	2014-06-22 13:30:36.873539224 -0400
@@ -6,16 +6,16 @@
 STRIP = strip
 
 CFLAGS ?= -g -O2
-CFLAGS := -DVERSION=\"$(VERSION)\" $(CFLAGS)
+# CFLAGS := -DVERSION=\"$(VERSION)\" $(CFLAGS)
 CPPFLAGS ?=
 LDFLAGS ?=
 
 SRCS = cd-discid.c
 OBJS = $(SRCS:.c=.o)
 
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 BINDIR = ${PREFIX}/bin
-MANDIR = ${PREFIX}/share/man/man1
+MANDIR = ${PREFIX}/man/man1
 
 .SUFFIXES: .c .o
 
