diff -ur ucspi-ssl-0.70.old/package/command-ln ucspi-ssl-0.70/package/command-ln
--- ucspi-ssl-0.70.old/package/command-ln	2005-07-17 18:59:02.000000000 -0500
+++ ucspi-ssl-0.70/package/command-ln	2009-05-16 20:28:13.000000000 -0500
@@ -1,5 +1,3 @@
-/package/host/superscript.com/command
-/usr/local/bin
 
 Directories to soft link commands into, one per line.
 The first empty line terminates the list.
Only in ucspi-ssl-0.70/package: command-ln~
diff -ur ucspi-ssl-0.70.old/package/upgrade ucspi-ssl-0.70/package/upgrade
--- ucspi-ssl-0.70.old/package/upgrade	2009-05-16 20:20:29.000000000 -0500
+++ ucspi-ssl-0.70/package/upgrade	2009-05-16 20:19:25.000000000 -0500
@@ -62,14 +62,14 @@
 
 [ "X$targets" = "X" ] && barf "no targets"
 
-version="`head -1 package/version`"
-here="`pwd`"
-command="`echo $here | sed -e 's/-'$version'$//'`/command"
-package="`basename $here | sed -e 's/-'$version'$//'`"
-echo "symlink $package -> $package-$version"
-safe rm -f $package
-safe ln -s $package-$version $package
-safe mv -f $package ..
+#version="`head -1 package/version`"
+#here="`pwd`"
+#command="`echo $here | sed -e 's/-'$version'$//'`/command"
+#package="`basename $here | sed -e 's/-'$version'$//'`"
+#echo "symlink $package -> $package-$version"
+#safe rm -f $package
+#safe ln -s $package-$version $package
+#safe mv -f $package ..
 
 commands=
 for i in $targets
diff -ur ucspi-ssl-0.70.old/src/conf-cadir ucspi-ssl-0.70/src/conf-cadir
--- ucspi-ssl-0.70.old/src/conf-cadir	2005-07-17 18:59:02.000000000 -0500
+++ ucspi-ssl-0.70/src/conf-cadir	2009-05-16 20:13:51.000000000 -0500
@@ -1,4 +1,4 @@
-/usr/local/ssl/certs
+/etc/ssl/certs
 
 This is the ucspi-ssl CA directory.
 An empty name means no certificate directory is compiled in.
diff -ur ucspi-ssl-0.70.old/src/conf-dhfile ucspi-ssl-0.70/src/conf-dhfile
--- ucspi-ssl-0.70.old/src/conf-dhfile	2005-07-17 18:59:02.000000000 -0500
+++ ucspi-ssl-0.70/src/conf-dhfile	2009-05-16 20:23:04.000000000 -0500
@@ -1,3 +1,3 @@
-/usr/local/ssl/pem/dh1024.pem
+/etc/ssl/pem/dh1024.pem
 
 This is the sslserver DH parameter file.
diff -ur ucspi-ssl-0.70.old/src/conf-tcpbin ucspi-ssl-0.70/src/conf-tcpbin
--- ucspi-ssl-0.70.old/src/conf-tcpbin	2005-07-17 18:59:02.000000000 -0500
+++ ucspi-ssl-0.70/src/conf-tcpbin	2009-05-16 20:17:25.000000000 -0500
@@ -1,4 +1,4 @@
-/usr/local/bin
+/usr/bin
 
 This is the ucspi-tcp bin directory.  The ucspi-tcp executables
 should appear in it.
