Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(Revision 1117624)
+++ CMakeLists.txt	(Arbeitskopie)
@@ -1,5 +1,20 @@
 PROJECT(libkexiv2)
 
+FIND_PACKAGE(KDE4 REQUIRED)
+INCLUDE(KDE4Defaults)
+INCLUDE(MacroLibrary)
+INCLUDE(MacroOptionalAddSubdirectory)
+INCLUDE(FindPackageHandleStandardArgs)
+
+ADD_DEFINITIONS (${QT_DEFINITIONS} ${QT_QTDBUS_DEFINITIONS} ${KDE4_DEFINITIONS})
+INCLUDE_DIRECTORIES (${QDBUS_INCLUDE_DIRS} ${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR} ${KDE4_INCLUDES})
+
+# Preview acess was added in Exiv2 0.18
+# IMPORTANT : Several fixes done in 0.19. Set as default version when KDE 4.4 will be released
+set(EXIV2_MIN_VERSION "0.18")
+macro_optional_find_package(Exiv2)
+macro_log_feature(EXIV2_FOUND "Exiv2" "A library to access image metadata" "http://www.exiv2.org" FALSE ${EXIV2_MIN_VERSION} "Required to build libkexiv2 and gwenview.")
+
 # =======================================================
 # Information to update before to release this library. 
