diff -Naur wifi-radar-2.0.s08-orig/man/man5/wifi-radar.conf.5 wifi-radar-2.0.s08/man/man5/wifi-radar.conf.5
--- wifi-radar-2.0.s08-orig/man/man5/wifi-radar.conf.5	2010-04-04 19:00:21.000000000 +0100
+++ wifi-radar-2.0.s08/man/man5/wifi-radar.conf.5	2010-06-01 18:33:19.000000000 +0100
@@ -175,7 +175,7 @@
 The parameters to pass to the DHCP client when acquiring a lease
 (i.e. setting up a connection).
 .IP
-.B (default: -D -o -i dhcp_client -t %(timeout)s)
+.B (default: -t %(timeout)s)
 .
 .TP 4
 .B command
diff -Naur wifi-radar-2.0.s08-orig/wifi-radar wifi-radar-2.0.s08/wifi-radar
--- wifi-radar-2.0.s08-orig/wifi-radar	2010-04-04 19:03:02.000000000 +0100
+++ wifi-radar-2.0.s08/wifi-radar	2010-06-01 18:33:35.000000000 +0100
@@ -2919,7 +2919,7 @@
 				# How long to wait for an IP addr from DHCP server
 				'timeout': '30',
 				# Arguments to use with DHCP client on connect
-				'args': '-D -o -i dhcp_client -t %(timeout)s',
+				'args': '-t %(timeout)s',
 				# Argument to use with DHCP client on disconnect
 				'kill_args': '-k',
 				# The file where DHCP client PID is written
