diff -Naur wifi-radar-2.0.s08-orig/wifi-radar.desktop wifi-radar-2.0.s08/wifi-radar.desktop
--- wifi-radar-2.0.s08-orig/wifi-radar.desktop	2010-04-04 19:00:21.000000000 +0100
+++ wifi-radar-2.0.s08/wifi-radar.desktop	2010-06-03 18:09:07.000000000 +0100
@@ -8,5 +8,5 @@
 Icon=wifi-radar.svg
 FilePattern=wifi-radar
 TryExec=wifi-radar
-Exec=gksudo -S wifi-radar
+Exec=kdesu wifi-radar
 Categories=Application;Network;
