diff --color -ur soundconverter-4.1.0/soundconverter/interface/theme.py soundconverter-4.1.0-patched/soundconverter/interface/theme.py
--- soundconverter-4.1.0/soundconverter/interface/theme.py	2025-06-23 19:49:18.000000000 +0100
+++ soundconverter-4.1.0-patched/soundconverter/interface/theme.py	2026-08-04 15:52:03.800653949 +0100
@@ -44,7 +44,7 @@
         return
 
     # Determine dark preference from 'color-scheme' GSetting
-    color_scheme = gnome_interface_settings.get_string("color-scheme")
+    color_scheme = "default"
     prefers_dark = color_scheme == "prefer-dark"
 
     current_theme_set = gtk_settings.get_property("gtk-theme-name")
