diff -pruN old/src/main_window.c new/src/main_window.c
--- old/src/main_window.c	2004-07-23 02:10:24.000000000 +0300
+++ new/src/main_window.c	2023-04-27 11:02:28.366906925 +0300
@@ -21,7 +21,6 @@
 #include "packages.h"
 #include "details_window.h"
 #include "progress_window.h"
-#include "build_window.h"
 #include "pkglist.h"
 #include "gpt.h"
 
@@ -37,8 +36,6 @@ static gboolean on_menu_close_cb (GtkWid
                                   GptMainWindow *mw);
 static gboolean on_menu_about_cb (GtkWidget     *widget,
                                   GptMainWindow *mw);
-static gboolean on_menu_build_cb (GtkWidget     *widget,
-                                  GptMainWindow *mw);
 static gboolean on_menu_install_cb (GtkWidget           *widget,
                                     GptMainWindow       *mw);
 static gboolean on_menu_refresh_cb (GtkWidget           *widget,
@@ -128,13 +125,11 @@ gpt_main_window_set_property (GObject
         if (g_value_get_boolean (val))
           {
             gtk_widget_set_sensitive (mw->priv->menu_install, FALSE);
-            gtk_widget_set_sensitive (mw->priv->menu_build, FALSE);
             gtk_widget_set_sensitive (mw->priv->btn_uninstall, FALSE);
           }
         else
           {
             gtk_widget_set_sensitive (mw->priv->menu_install, TRUE);
-            gtk_widget_set_sensitive (mw->priv->menu_build, TRUE);
             gtk_widget_set_sensitive (mw->priv->btn_uninstall, TRUE);
           }
         break;
@@ -180,11 +175,6 @@ gpt_main_window_init (GptMainWindow *mw)
         (gpointer) on_menu_install_cb, (gpointer) mw, NULL,
         GNOME_APP_PIXMAP_STOCK, "gtk-add",
         0, (GdkModifierType) 0, NULL },
-      { GNOME_APP_UI_ITEM, "_Build From Source...",
-	"Build package from source tarball",
-	(gpointer) on_menu_build_cb, (gpointer) mw, NULL,
-	GNOME_APP_PIXMAP_STOCK, "gtk-convert",
-	0, (GdkModifierType) 0, NULL },
       GNOMEUIINFO_SEPARATOR,
       GNOMEUIINFO_MENU_CLOSE_ITEM (on_menu_close_cb, NULL),
       GNOMEUIINFO_END
@@ -233,7 +223,6 @@ gpt_main_window_init (GptMainWindow *mw)
 	
   /* remember pointers to some menu items */
   mw->priv->menu_install = menu_file_uiinfo[0].widget;
-  mw->priv->menu_build = menu_file_uiinfo[1].widget;
 
   /* paned */
   align = gtk_alignment_new (0.5, 0.7, 1, 1);
@@ -398,16 +387,6 @@ gpt_main_window_init (GptMainWindow *mw)
 }
 
 static gboolean
-on_menu_build_cb (GtkWidget *widget, GptMainWindow *mw)
-{
-  GtkWidget *bw = gpt_build_window_new ();
-  gtk_window_set_transient_for(GTK_WINDOW (bw), GTK_WINDOW (mw));
-  gtk_window_set_destroy_with_parent (GTK_WINDOW (bw), TRUE);
-  gtk_widget_show(bw);
-  return TRUE;
-}
-
-static gboolean
 on_menu_about_cb (GtkWidget *widget, GptMainWindow *mw)
 {
   GtkWidget *dlg;
