--- abcde-2.6/cddb-tool	2014-10-26 14:21:14.000000000 -0400
+++ abcde-2.6/cddb-tool.patched	2014-12-14 18:08:31.324695765 -0500
@@ -220,7 +220,7 @@
 		if [ "$UTF_LOCALE"x != ""x ] ; then
 			export LC_ALL=$UTF_LOCALE
 		fi
-		grep -v "^#CATEGORY=" "$FILE" | iconv -t utf-8 | mail -s "cddb $CDDBGENRE $DISCID" "$ADDRESS"
+		grep -v "^#CATEGORY=" "$FILE" | iconv -t utf-8 | mailx -s "cddb $CDDBGENRE $DISCID" "$ADDRESS"
 	fi
 	;;
 
