diff -Naur o2em118src/src/debug.c o2em118src.patched//src/debug.c
--- o2em118src/src/debug.c	2006-11-24 05:41:34.000000000 -0500
+++ o2em118src.patched//src/debug.c	2011-01-12 16:02:24.000000000 -0500
@@ -16,6 +16,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <errno.h>
 #include "cpu.h"
 #include "keyboard.h"
 #include "vmachine.h"
@@ -1042,4 +1043,4 @@
 
 	printf("\n");
 	return 1;
-}
\ No newline at end of file
+}
diff -Naur o2em118src/src/makefile.linux o2em118src.patched//src/makefile.linux
--- o2em118src/src/makefile.linux	2006-05-05 16:45:14.000000000 -0400
+++ o2em118src.patched//src/makefile.linux	2011-01-12 16:07:46.000000000 -0500
@@ -1,8 +1,8 @@
 CC = gcc
 
-CFLAGS = -O3 -Wall -fomit-frame-pointer -I/usr/include
+CFLAGS = -Wall $(SLKCFLAGS) `allegro-config --cflags`
 LFLAGS = -s
-LIBALLEG = /usr/lib/liballeg.so.4.2 -lalleg_unsharable
+LIBALLEG = `allegro-config --libs`
 
 
 all: o2em dis48
diff -Naur o2em118src/src/score.c o2em118src.patched//src/score.c
--- o2em118src/src/score.c	2006-05-01 01:58:08.000000000 -0400
+++ o2em118src.patched//src/score.c	2011-01-12 16:02:35.000000000 -0500
@@ -19,6 +19,7 @@
 #include <string.h>
 #include <ctype.h>
 #include <time.h>
+#include <errno.h>
 #include "vmachine.h"
 #include "types.h"
 #include "score.h"
diff -Naur o2em118src/src/vmachine.c o2em118src.patched//src/vmachine.c
--- o2em118src/src/vmachine.c	2006-11-24 11:59:26.000000000 -0500
+++ o2em118src.patched//src/vmachine.c	2011-01-12 16:02:43.000000000 -0500
@@ -18,6 +18,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <time.h>
+#include <errno.h>
 #include "audio.h"
 #include "types.h"
 #include "cpu.h"
