diff -Nur lxi-tools-1.18.orig/src/gui/lxi-gui/Makefile.am lxi-tools-1.18/src/gui/lxi-gui/Makefile.am > src_gui_lxi-gui_Makefile.am.diff
--- lxi-tools-1.18.orig/src/gui/lxi-gui/Makefile.am	2018-01-29 20:02:39.000000000 -0600
+++ lxi-tools-1.18/src/gui/lxi-gui/Makefile.am	2018-02-10 08:09:20.049138969 -0600
@@ -11,12 +11,13 @@
              workerthread.h

 Makefile.qmake: lxi-gui.pro
-	$(QTCHOOSER) -run-tool=qmake -qt=5 -makefile -o Makefile.qmake lxi-gui.pro \
+	$(QMAKE) -makefile -o Makefile.qmake lxi-gui.pro \
 		INCDIR="$(SNAPCRAFT_STAGE)/$(includedir)" \
 		LIBDIR="$(SNAPCRAFT_STAGE)/$(libdir)" \
 		SNAPCRAFT="$(SNAPCRAFT)" \
 		SNAPCRAFT_STAGE="$(SNAPCRAFT_STAGE)" \
-		$(QMAKE_ARGUMENTS)
+		$(QMAKE_ARGUMENTS) \
+		LIBS="-lncurses"

 all-local: Makefile.qmake
 	$(MAKE) -f Makefile.qmake $(AM_MAKEFLAGS) all
