--- Makefile.in	2013-10-08 07:00:17.000000000 -0500
+++ Makefile.in.fixed	2014-07-13 01:12:54.405819192 -0500
@@ -821,13 +821,6 @@
 	  done \
 	  fi
 
-	if test -f $(HOME)/.$(PACKAGE)rc; then \
-	  echo Config file exists: $(HOME)/.$(PACKAGE)rc - not copying; \
-	  else \
-	  echo Copying $(PACKAGE) config file to $(HOME)/.$(PACKAGE); \
-	  cp $(srcdir)/default/.$(PACKAGE)rc $(HOME); \
-	  fi
-
 dist-hook:
 	if test -d pixmaps; then \
 	  mkdir $(distdir)/pixmaps; \
