From 2afe4a11d43e8767cbeb90372177d4b580384e9a Mon Sep 17 00:00:00 2001
From: BearOso <bearoso@gmail.com>
Date: Mon, 1 May 2023 11:38:54 -0500
Subject: [PATCH] CI acting funky. See if this fixes it.

---
 vulkan/vulkan_context.hpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/vulkan/vulkan_context.hpp b/vulkan/vulkan_context.hpp
index cfc29eda2..33ec20c5f 100644
--- a/vulkan/vulkan_context.hpp
+++ b/vulkan/vulkan_context.hpp
@@ -7,6 +7,8 @@
 #undef WINVER
 #define WINVER 0x599
 #endif
+#include <cstdio>
+#include <cstdint>
 #include "vk_mem_alloc.hpp"
 #include "vulkan/vulkan.hpp"
 #include "vulkan_swapchain.hpp"
@@ -64,4 +66,4 @@ class Context
     vk::CommandBuffer one_time_use_cmd;
 };
 
-} // namespace Vulkan
\ No newline at end of file
+} // namespace Vulkan
