From b383b94c8c94e3a28b09e7dd3d72cac2a783ba55 Mon Sep 17 00:00:00 2001
From: BearOso <bearoso@gmail.com>
Date: Sun, 30 Apr 2023 17:18:39 -0500
Subject: [PATCH] slang: Fix CI build.

---
 vulkan/slang_shader.hpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/vulkan/slang_shader.hpp b/vulkan/slang_shader.hpp
index 8af9f48d9..6e24bcd8d 100644
--- a/vulkan/slang_shader.hpp
+++ b/vulkan/slang_shader.hpp
@@ -1,6 +1,7 @@
 #pragma once
 #include <string>
 #include <vector>
+#include <cstdint>
 
 struct SlangShader
 {
@@ -101,4 +102,4 @@ struct SlangShader
     int ubo_binding;
     std::vector<Uniform> uniforms;
     std::vector<Sampler> samplers;
-};
\ No newline at end of file
+};
