From 03fad27e1f0f18ec83c9a4bc5f03f28948cd44fb Mon Sep 17 00:00:00 2001
From: Jan Holthuis <jan.holthuis@ruhr-uni-bochum.de>
Date: Fri, 23 Aug 2019 15:26:21 +0200
Subject: [PATCH] scons: Fix UIC invocation via map() in Python 3

---
 build/depends.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/build/depends.py b/build/depends.py
index bf81d20cb89..152a6de5c11 100644
--- a/build/depends.py
+++ b/build/depends.py
@@ -1297,7 +1297,11 @@ def sources(self, build):
             'src/preferences/dialog/dlgprefvinyldlg.ui',
             'src/preferences/dialog/dlgprefwaveformdlg.ui',
         ]
-        map(Qt.uic(build), ui_files)
+
+        # In Python 3.x, map() returns a "map object" (instead of a list),
+        # which is evaluated on-demand rather than at once. To invoke uic
+        # for all *.ui files at once, we need to cast it to a list here.
+        list(map(Qt.uic(build), ui_files))
 
         if build.platform_is_windows:
             # Add Windows resource file with icons and such
