diff -Naur a/src/SDL_gfxPrimitives.c b/src/SDL_gfxPrimitives.c
--- a/src/SDL_gfxPrimitives.c	2007-03-07 10:35:26.000000000 -0500
+++ b/src/SDL_gfxPrimitives.c	2016-08-14 15:36:20.470686223 -0400
@@ -10,6 +10,7 @@
 #include <stdlib.h>
 #include <math.h>
 #include <string.h>
+#include <limits.h>
 
 #include "SDL_gfxPrimitives.h"
 #include "SDL_gfxPrimitives_font.h"
diff -Naur a/src/global.cpp b/src/global.cpp
--- a/src/global.cpp	2007-03-07 10:34:56.000000000 -0500
+++ b/src/global.cpp	2016-08-14 15:39:23.252692634 -0400
@@ -9,7 +9,7 @@
 
 #include "Global.h"
 #include "unixutils.h"
-
+#include <limits.h>
 
 
 SDL_Surface *screen;
diff -Naur a/src/levels.cpp b/src/levels.cpp
--- a/src/levels.cpp	2007-03-07 10:35:04.000000000 -0500
+++ b/src/levels.cpp	2016-08-14 15:38:08.660690018 -0400
@@ -6,6 +6,7 @@
 *********************************************************************/ 
 
 
+#include <limits.h>
 #include "Global.h"
 #include "unixutils.h"
 
diff -Naur a/src/main.cpp b/src/main.cpp
--- a/src/main.cpp	2007-03-07 10:35:04.000000000 -0500
+++ b/src/main.cpp	2016-08-14 15:38:42.660691210 -0400
@@ -10,6 +10,7 @@
 #include "iniparser.h"
 #include "unixutils.h"
 #include <time.h>
+#include <limits.h>
 
 
 bool running;
diff -Naur a/src/menu.cpp b/src/menu.cpp
--- a/src/menu.cpp	2007-03-07 10:35:10.000000000 -0500
+++ b/src/menu.cpp	2016-08-14 15:40:10.108694278 -0400
@@ -7,6 +7,7 @@
 
 #include "Global.h"
 #include "unixutils.h"
+#include <limits.h>
 
 bool hideMenu;
 
