diff -Naur a/src/functions.cpp b/src/functions.cpp
--- a/src/functions.cpp	2007-03-07 10:34:56.000000000 -0500
+++ b/src/functions.cpp	2016-08-14 16:26:04.208790873 -0400
@@ -447,7 +447,7 @@
 	}
 	else
 	{
-		sprintf(tmp, "mp3music/%s",filename);
+		sprintf(tmp, DATADIR "mp3music/%s",filename);
 		tmp[strlen(tmp)-3]='m';
 		tmp[strlen(tmp)-2]='p';
 		tmp[strlen(tmp)-1]='3';
diff -Naur a/src/main.cpp b/src/main.cpp
--- a/src/main.cpp	2007-03-07 10:35:04.000000000 -0500
+++ b/src/main.cpp	2016-08-14 16:25:41.230790067 -0400
@@ -413,7 +413,7 @@
 
 
 fstream fin;
-fin.open("mp3music/music_available.dat",ios::in);
+fin.open(DATADIR "mp3music/music_available.dat",ios::in);
 if( fin.is_open() )
 {
 	HighQualityMusicFound=1;
