From 928c81d326ec2b9529e183cc1816484252703357 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?George=20Florea=20B=C4=83nu=C8=99?= <georgefb899@gmail.com>
Date: Fri, 6 Jan 2023 13:43:53 +0200
Subject: [PATCH] recentfilesmodel: don't try to add files if option was
 disabled (MaxRecentFiles = 0)

BUG: 463737
---
 src/models/recentfilesmodel.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/models/recentfilesmodel.cpp b/src/models/recentfilesmodel.cpp
index 92a94ea..be51e0e 100644
--- a/src/models/recentfilesmodel.cpp
+++ b/src/models/recentfilesmodel.cpp
@@ -84,6 +84,10 @@ void RecentFilesModel::populate()
 
 void RecentFilesModel::addUrl(const QString &path, const QString &name)
 {
+    if (maxRecentFiles() == 0) {
+        return;
+    }
+
     auto config = KSharedConfig::openConfig(Global::instance()->appConfigFilePath());
     QUrl url(path);
     if (!url.isLocalFile() && url.scheme().isEmpty()) {
-- 
GitLab


