diff -Naur a/include/cgmtek.h b/include/cgmtek.h
--- a/include/cgmtek.h	1993-08-10 04:10:54.000000000 -0400
+++ b/include/cgmtek.h	2017-12-20 00:25:38.734568061 -0500
@@ -187,7 +187,7 @@
 };
 
 /* Tektronix colour table variables*/
-static n_coltab = DEF_N_COLTAB;
+static int n_coltab = DEF_N_COLTAB;
 static RGBcolour TEKcoltab[MAX_N_COLTAB];   /* The TEK colour table */
 static Enum      TEKcoluse[MAX_N_COLTAB];   /* The TEK colour flags */
 static Int colset_n;        /* Number of colour table entries set */
diff -Naur a/src/cgmlib.c b/src/cgmlib.c
--- a/src/cgmlib.c	1993-08-10 04:10:34.000000000 -0400
+++ b/src/cgmlib.c	2017-12-20 00:33:41.535568880 -0500
@@ -38,6 +38,12 @@
 #define CGMLIB_C
 #include "cgmlib.h"
 
+extern void CGMOhpgl(FILE *, Code, Long*, Float*, char* );
+
+#ifdef TEK4200
+   extern void CGMOtek( FILE*, Code, long *, float *, char * );
+#endif
+
 /* Driver Routines */
 
 #ifdef PROTO
diff -Naur a/src/cgmmain.c b/src/cgmmain.c
--- a/src/cgmmain.c	1995-06-13 11:32:22.000000000 -0400
+++ b/src/cgmmain.c	2017-12-20 00:23:32.606567847 -0500
@@ -280,6 +280,7 @@
 #define OUTMESS      (void) fprintf( (FILE *) cgmerr,
 
 /******************************************************** main *********/
+int
 #ifdef PROTO
 main ( int argc, char **argv)
 #else
diff -Naur a/src/cgmoxw.c b/src/cgmoxw.c
--- a/src/cgmoxw.c	1994-04-15 09:30:07.000000000 -0400
+++ b/src/cgmoxw.c	2017-12-20 00:24:37.448567957 -0500
@@ -111,6 +111,8 @@
 
  */
 
+#include <unistd.h>
+
 #define CGMOXW_C
 #include "cgmout.h"
 
diff -Naur a/src/cgmtrap.c b/src/cgmtrap.c
--- a/src/cgmtrap.c	1995-06-13 11:33:02.000000000 -0400
+++ b/src/cgmtrap.c	2017-12-20 00:27:03.439568205 -0500
@@ -45,6 +45,7 @@
 #include "cgmelem.h"
 #include "cgmpar.h"
 #include "cgmpts.h"
+#include "cgmgraph.h"
 
 /* Fixed Limit to the number trapezoids (protect CPU from bug) */
 #define MAXTRAPS 100000
diff -Naur a/src/genbez.c b/src/genbez.c
--- a/src/genbez.c	1995-06-16 03:26:44.000000000 -0400
+++ b/src/genbez.c	2017-12-20 00:22:46.548567769 -0500
@@ -37,7 +37,7 @@
 BezDir  Dir;
 BezFont Bfont;
 
-main()
+int main(int argc, char **argv)
 {
     FILE *generator, *directory;
     char *dirname, *genname;
diff -Naur a/src/genher.c b/src/genher.c
--- a/src/genher.c	1995-06-16 03:26:48.000000000 -0400
+++ b/src/genher.c	2017-12-20 00:22:56.659567786 -0500
@@ -51,7 +51,7 @@
 static char *func = "genher";
 Logical cgmerrcount=FALSE, cgmquiet=FALSE;
 
-main()
+int main(int argc, char **argv)
 
 {
     int last, code, i;
