--- oyranos-0.4.0/configure_tests.sh  2012-03-10 19:28:12.016514261 +0100
+++ oyranos-0.4.0/configure_tests.sh.new      2012-03-10 19:30:04.230678948 +0100
@@ -3,7 +3,7 @@
 set > testset.txt
 ERROR=0
 WARNING=0
-STRIPOPT='s/-O.// ; s/-isysroot [[:graph:]]*// ; s/-arch ppc// ; s/-arch i386//'
+STRIPOPT='s/-Wl[^ ]*//g ; s/-O.// ; s/-isysroot [[:graph:]]*// ; s/-arch ppc// ; s/-arch i386//'
 
 mkdir -p tests
