--- oyranos-0.4.0/examples/libraw/oyranos_file.cpp    2012-03-10 20:39:13.754934907 +0100
+++ oyranos-0.4.0/examples/libraw/oyranos_file.cpp.new        2012-03-10 20:39:18.317895815 +0100
@@ -15,7 +15,7 @@
  */
 /* c++ -Wall -g oyranos_file.cpp `pkg-config --cflags --libs oyranos` -o oyranos-file */
 
-#include <alpha/oyranos_alpha.h>
+#include <oyranos_alpha.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
