--- oyranos-0.4.0/examples/xcmsevents/qcmsevents.cpp  2012-05-02 21:58:50.309004915 +0200
+++ oyranos-0.4.0/examples/xcmsevents/qcmsevents.cpp.new      2012-05-02 21:58:32.533739653 +0200
@@ -118,8 +118,8 @@
   else if (XCME_MSG_DISPLAY_STATUS)
   {
     int i;
-    if(strstr(text, "atom: \""XCM_COLOR_DESKTOP"\":"))
-      sscanf( text, "atom: \""XCM_COLOR_DESKTOP"\": %d %d", &i, &pid );
+    if(strstr(text, "atom: \"" XCM_COLOR_DESKTOP "\":"))
+      sscanf( text, "atom: \"" XCM_COLOR_DESKTOP "\": %d %d", &i, &pid );
     if(pid == 0)
     {
       colour.setHsvF( 0.6, 0.4, 0.9 );

