diff -Naur a/gui/src/MainForm.cpp b/gui/src/MainForm.cpp
--- a/gui/src/MainForm.cpp	2012-02-19 21:54:05.000000000 +0000
+++ b/gui/src/MainForm.cpp	2014-11-28 09:13:01.930327087 +0000
@@ -20,6 +20,7 @@
  *
  */
 #include "version.h"
+#include "unistd.h"
 
 
 #include <hydrogen/hydrogen.h>
diff -Naur a/libs/hydrogen/src/IO/portmidi_driver.cpp b/libs/hydrogen/src/IO/portmidi_driver.cpp
--- a/libs/hydrogen/src/IO/portmidi_driver.cpp	2012-01-16 23:20:51.000000000 +0000
+++ b/libs/hydrogen/src/IO/portmidi_driver.cpp	2014-11-28 09:13:01.931327087 +0000
@@ -20,6 +20,7 @@
  *
  */
 
+#include "unistd.h"
 #include "PortMidiDriver.h"
 
 #include <hydrogen/Preferences.h>
