diff -Naur hydrogen-0.9.5.orig/gui/src/MainForm.cpp hydrogen-0.9.5/gui/src/MainForm.cpp
--- hydrogen-0.9.5.orig/gui/src/MainForm.cpp	2011-02-08 10:15:04.000000000 +0100
+++ hydrogen-0.9.5/gui/src/MainForm.cpp	2012-09-12 17:40:17.305970299 +0200
@@ -20,6 +20,7 @@
  *
  */
 #include "version.h"
+#include "unistd.h"
 
 
 #include <hydrogen/hydrogen.h>
diff -Naur hydrogen-0.9.5.orig/gui/src/SongEditor/SongEditor.cpp hydrogen-0.9.5/gui/src/SongEditor/SongEditor.cpp
--- hydrogen-0.9.5.orig/gui/src/SongEditor/SongEditor.cpp	2010-03-28 12:06:05.000000000 +0200
+++ hydrogen-0.9.5/gui/src/SongEditor/SongEditor.cpp	2012-09-12 17:37:17.764043771 +0200
@@ -23,6 +23,7 @@
 #include <assert.h>
 #include <algorithm>
 #include <memory>
+#include <unistd.h>
 
 #include <hydrogen/Song.h>
 #include <hydrogen/hydrogen.h>
diff -Naur hydrogen-0.9.5.orig/libs/hydrogen/src/IO/portmidi_driver.cpp hydrogen-0.9.5/libs/hydrogen/src/IO/portmidi_driver.cpp
--- hydrogen-0.9.5.orig/libs/hydrogen/src/IO/portmidi_driver.cpp	2010-02-01 22:06:17.000000000 +0100
+++ hydrogen-0.9.5/libs/hydrogen/src/IO/portmidi_driver.cpp	2012-09-12 17:45:38.426002346 +0200
@@ -20,6 +20,7 @@
  *
  */
 
+#include "unistd.h"
 #include "PortMidiDriver.h"
 
 #include <hydrogen/Preferences.h>
