diff -Naur a/Sconstruct b/Sconstruct
--- a/Sconstruct	2011-03-15 12:22:35.000000000 +0000
+++ b/Sconstruct	2011-10-28 03:17:21.000000000 +0100
@@ -214,12 +214,7 @@
 	for file in files:
 		if file.endswith(".png"):
 			
-			if env['prefix'].endswith("/"):
-				dname = dir[2:]  
-			else:
-				dname = dir[1:]  
-
-			env.Alias(target="install", source=env.Install(dir= env['DESTDIR'] + env['prefix'] + '/share/hydrogen/' + dname, source= dir + "/" + file))
+			env.Alias(target="install", source=env.Install(dir= env['DESTDIR'] + env['prefix'] + '/share/hydrogen/img/gray'  + file))
 
 def get_hydrogen_gui( lib_hydrogen , opts ):
 	includes, cppflags, ldflags = get_platform_flags( opts )
@@ -301,7 +296,7 @@
 	env.Alias(target="install", source=env.Install(dir= env['DESTDIR'] + env['prefix'] + '/share/hydrogen/data', source="./data/img"))
 
 	#add every img in ./data/img to the install list. 
-	os.path.walk("./data/img/",install_images,env) 
+	os.path.walk("./data/img/gray",install_images,env) 
 
 
 	env.Alias(target="install", source=env.Install(dir= env['DESTDIR'] + env['prefix'] + '/share/hydrogen/data', source="./data/drumkits"))
