diff -Naur hydrogen-0.9.5.orig/Sconstruct hydrogen-0.9.5/Sconstruct
--- hydrogen-0.9.5.orig/Sconstruct	2011-03-15 13:22:35.000000000 +0100
+++ hydrogen-0.9.5/Sconstruct	2012-09-12 17:32:56.865019039 +0200
@@ -81,6 +81,7 @@
 	if str(env['libarchive']) == "1": cppflags.append('-DLIBARCHIVE_SUPPORT')
 
 	includes.append( './' )
+	includes.append( '/usr/include/raptor2' )
 	includes.append( 'gui/src/' )
 	includes.append( '3rdparty/install/include' )
 
