diff --git a/scripts/check_rootdir b/scripts/check_rootdir
index a34eb2a..5f55159 100755
--- a/scripts/check_rootdir
+++ b/scripts/check_rootdir
@@ -90,7 +90,7 @@ haveallvars TESTLINK HOSTNAME || exit 1
 # Checking to see if the root directory is on the correct INODE number
 inode=`$LS -di / | $AWK '{print $1}'`
 fstype=`$GETFS | $GREP " / " | $AWK -F " " '{ print $5;}'`
-[ $inode != 2 ] && [ "$fstype" = "ext2" -o "$fstype" = "ext3" ] &&
+[ $inode != 2 ] && [ "$fstype" = "ext2" -o "$fstype" = "ext3" -o "$fstype" = "ext4" ] &&
    message FAIL rootdir001f "" "Root directory on incorrect inode"
 
 
