diff -urpN tiger-3.2.3.orig/scripts/check_signatures tiger-3.2.3/scripts/check_signatures
--- tiger-3.2.3.orig/scripts/check_signatures	2011-08-14 05:44:29.474729020 +0000
+++ tiger-3.2.3/scripts/check_signatures	2011-08-14 05:46:36.840361481 +0000
@@ -91,7 +91,7 @@ do
 	  *[!0-9a-f]*) {
 	    std_signature="$p1"
 	    comment="$p2 $p3 $p4 $p5 $p6 $p7 $p8 $comment"
-	    [ -n "$MD5" ] && loc_signature="`$MD5 < $file`"
+	    [ -n "$MD5" ] && loc_signature="`$MD5 < $file | $AWK '{print $1}'`"
 	  }
 	  ;;
 	  *) {
