diff -urpN tiger-3.2.3.orig/util/mkfilelst tiger-3.2.3/util/mkfilelst
--- tiger-3.2.3.orig/util/mkfilelst	2011-08-07 21:16:16.361620719 +0000
+++ tiger-3.2.3/util/mkfilelst	2011-08-07 21:18:30.710039906 +0000
@@ -138,7 +138,7 @@ do
 	if [ $glob -lt 2 ] ; then
 	   [ -f "$file" -o -d "$file" ] && \
 	   getpermit $file |
-           while read _file owner group ur uw ux gr gw gx or ow ox stk suid sgid stk
+           while read _file owner group ur uw ux gr gw gx or ow ox suid sgid stk
 	  do
 	    echo "$file $owner $group $suid $sgid $ur $uw $ux $gr $gw $gx $or $ow $ox $lvl $ownmid $grpmid $prmmid"
 	  done
