From c0815d8bd760a3b5b62d6caa452b1ae1430040f9 Mon Sep 17 00:00:00 2001
From: pyllyukko <pyllyukko@maimed.org>
Date: Sat, 6 Oct 2012 23:18:03 +0300
Subject: Fixed the aide_run script to support Aide 0.15.1 reports

---
 scripts/aide_run |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/aide_run b/scripts/aide_run
index 363ef10..b776589 100755
--- a/scripts/aide_run
+++ b/scripts/aide_run
@@ -152,11 +152,11 @@ $AIDE --check --config=${AIDE_CFG} --report=stdout ${AIDE_DB} \
 # i.e. check $? and determine what happened
 
 # Check temporary report for the changes signal:
-$GREP ${AIDE_RPT} -qe "AIDE found differences"
+$GREP ${AIDE_RPT} -qe "AIDE \([0-9.]\+ \)\?found differences"
 
 case "$?" in
 0)	# Summary
-	AIDE_SUMMARY=`$GREP ${AIDE_RPT} -e "^Total number"`
+	AIDE_SUMMARY=`$GREP ${AIDE_RPT} -e "^ *Total number"`
 	message INFO aide003i "" "Summary: ${AIDE_SUMMARY}"
 	# Differentiate between added, changed and removed files
 	for status in changed removed detected; do
-- 
1.7.4.4

