From 160ba685d08651d1cc755d2bb0167dc7170f8e2b Mon Sep 17 00:00:00 2001
From: pyllyukko <pyllyukko@maimed.org>
Date: Sat, 6 Oct 2012 23:43:07 +0300
Subject: Added facl.{tmpl|list} to Makefile.in.

---
 Makefile.in |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 9d6eab8..86ab714 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -167,6 +167,8 @@ MISCFILES=./initdefs \
 	./util/mksig \
 	./util/setsh \
 	./util/sgrep \
+	./util/facl.lst \
+	./util/facl.tmpl \
         ./doc/explain.idx
 
 MISCDIRS=./bin \
-- 
1.7.4.4

