--- libdbus-c++-0.9.0/src/message.cpp     2016-02-20 12:15:52.069704535 +0100
+++ libdbus-c++-0.9.0-patch/src/message.cpp    2016-02-20 12:31:34.159854472 +0100
@@ -347,6 +347,7 @@
       (
         (DBusMessageIter *) & (to._iter),
         from.type(),
-        from.type() == DBUS_TYPE_VARIANT ? NULL : sig,
+       from.type() == DBUS_TYPE_DICT_ENTRY ||
+       from.type() == DBUS_TYPE_STRUCT ? NULL : sig,
         (DBusMessageIter *) & (to_container._iter)
       );
