--- brlcad-7.16.6/src/other/tkimg/pngtcl/pngtclDecls.h	2010-01-19 00:27:35.000000000 -0200
+++ brlcad-7.16.6_patched/src/other/tkimg/pngtcl/pngtclDecls.h	2010-04-23 17:58:07.000000000 -0300
@@ -46,11 +46,11 @@
 				png_voidp error_ptr, png_error_ptr error_fn,
 				png_error_ptr warn_fn));
 /* 6 */
-EXTERN png_uint_32	png_get_compression_buffer_size _ANSI_ARGS_((
+EXTERN png_size_t	png_get_compression_buffer_size _ANSI_ARGS_((
 				png_structp png_ptr));
 /* 7 */
 EXTERN void		png_set_compression_buffer_size _ANSI_ARGS_((
-				png_structp png_ptr, png_uint_32 size));
+				png_structp png_ptr, png_size_t size));
 /* 8 */
 EXTERN int		png_reset_zstream _ANSI_ARGS_((png_structp png_ptr));
 /* 9 */
@@ -337,7 +337,7 @@
 				png_bytep new_row));
 /* 94 */
 EXTERN png_voidp	png_malloc _ANSI_ARGS_((png_structp png_ptr,
-				png_uint_32 size));
+				png_size_t size));
 /* 95 */
 EXTERN void		png_free _ANSI_ARGS_((png_structp png_ptr,
 				png_voidp ptr));
@@ -351,7 +351,7 @@
 				png_uint_32 mask));
 /* 98 */
 EXTERN png_voidp	png_malloc_default _ANSI_ARGS_((png_structp png_ptr,
-				png_uint_32 size));
+				png_size_t size));
 /* 99 */
 EXTERN void		png_free_default _ANSI_ARGS_((png_structp png_ptr,
 				png_voidp ptr));
@@ -380,7 +380,7 @@
 EXTERN png_uint_32	png_get_valid _ANSI_ARGS_((png_structp png_ptr,
 				png_infop info_ptr, png_uint_32 flag));
 /* 108 */
-EXTERN png_uint_32	png_get_rowbytes _ANSI_ARGS_((png_structp png_ptr,
+EXTERN png_size_t	png_get_rowbytes _ANSI_ARGS_((png_structp png_ptr,
 				png_infop info_ptr));
 /* 109 */
 EXTERN png_bytepp	png_get_rows _ANSI_ARGS_((png_structp png_ptr,
--- brlcad-7.16.6/src/fb/png-fb.c	2010-01-19 00:29:40.000000000 -0200
+++ brlcad-7.16.6_patched/src/fb/png-fb.c	2010-04-23 18:28:14.000000000 -0300
@@ -189,7 +189,7 @@
 	bu_exit(EXIT_FAILURE,  "ERROR: Failed while reading file header!!!\n" );
     }
 
-    if (!png_check_sig((png_bytep)header, 8)) {
+    if (png_sig_cmp((png_bytep)header, 0, 8)) {
 	bu_exit(EXIT_FAILURE,  "This is not a PNG file!!!\n" );
     }
 
--- brlcad-7.16.6/src/util/png-bw.c	2010-01-19 00:33:04.000000000 -0200
+++ brlcad-7.16.6_patched/src/util/png-bw.c	2010-04-23 19:08:46.000000000 -0300
@@ -139,7 +139,7 @@
     if ( fread( header, 8, 1, fp_in ) != 1 )
 	bu_exit( EXIT_FAILURE, "ERROR: Failed while reading file header!!!\n" );
 
-    if ( !png_check_sig( (png_bytep)header, 8 ) )
+    if ( png_sig_cmp((png_bytep)header, 0, 8))
 	bu_exit( EXIT_FAILURE, "This is not a PNG file!!!\n" );
 
     png_p = png_create_read_struct( PNG_LIBPNG_VER_STRING, NULL, NULL, NULL );
--- brlcad-7.16.6/src/util/png-pix.c	2010-01-19 00:33:02.000000000 -0200
+++ brlcad-7.16.6_patched/src/util/png-pix.c	2010-04-23 19:10:30.000000000 -0300
@@ -95,7 +95,7 @@
     if ( fread( header, 8, 1, fp_in ) != 1 )
 	bu_exit( EXIT_FAILURE, "ERROR: Failed while reading file header!!!\n" );
 
-    if ( !png_check_sig( (png_bytep)header, 8 ) )
+    if (png_sig_cmp( (png_bytep)header, 0, 8))
 	bu_exit( EXIT_FAILURE, "This is not a PNG file!!!\n" );
 
     png_p = png_create_read_struct( PNG_LIBPNG_VER_STRING, NULL, NULL, NULL );
--- brlcad-7.16.6/src/util/png_info.c	2010-01-19 00:33:05.000000000 -0200
+++ brlcad-7.16.6_patched/src/util/png_info.c	2010-04-23 19:12:00.000000000 -0300
@@ -76,7 +76,7 @@
     if ( fread( header, 8, 1, fp_in ) != 1 )
 	bu_exit( EXIT_FAILURE, "ERROR: Failed while reading file header!!!\n" );
 
-    if ( !png_check_sig( (png_bytep)header, 8 ) )
+    if (png_sig_cmp((png_bytep)header, 0, 8))
 	bu_exit( EXIT_FAILURE, "This is not a PNG file!!!\n" );
 
     png_p = png_create_read_struct( PNG_LIBPNG_VER_STRING, NULL, NULL, NULL );
--- brlcad-7.16.6/src/canon/png-ipu.c	2010-01-19 00:30:24.000000000 -0200
+++ brlcad-7.16.6_patched/src/canon/png-ipu.c	2010-04-23 19:13:10.000000000 -0300
@@ -125,7 +125,7 @@
     if ( fread( header, 8, 1, stdin ) != 1 )
 	bu_exit(1, "png-ipu: ERROR: Failed while reading file header!");
 
-    if ( !png_check_sig( (png_bytep)header, 8 ) )
+    if (png_sig_cmp((png_bytep)header, 0, 8))
 	bu_exit(2, "png-ipu: This is not a valid PNG file\n");
 
     png_p = png_create_read_struct( PNG_LIBPNG_VER_STRING, NULL, NULL, NULL );
