diff --git a/Makefile.in b/Makefile.in
index 10d7b32..25e79b0 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -133,7 +133,7 @@ config.status: configure
 
 # two steps to prevent the creation of a bogus siglist.out
 siglist.out: config.h sh.h siglist.in siglist.sh
-	$(srcdir)/siglist.sh "$(CPP) $(CPPFLAGS) $(DEFS) -I. -I$(srcdir)" < $(srcdir)/siglist.in > tmpsiglist.out
+	$(srcdir)/siglist.sh "$(CPP) $(CPPFLAGS) $(DEFS) -I. -I$(srcdir) -P" < $(srcdir)/siglist.in > tmpsiglist.out
 	mv tmpsiglist.out siglist.out
 
 # two steps to prevent the creation of a bogus emacs.out

diff --git a/Makefile.in b/Makefile.in
index 25e79b0..5779948 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -80,8 +80,8 @@ all: $(SHELL_PROG)$(exe_suffix) $(SHELL_PROG).1
 	$(CC) -c $(CPPFLAGS) $(DEFS) -I. -I$(srcdir) $(CFLAGS) $<
 
 install: installdirs all
-	$(INSTALL_PROGRAM) $(SHELL_PROG)$(exe_suffix) $(bindir)/`echo $(SHELL_PROG)|sed '$(transform)'`$(exe_suffix)
-	-$(INSTALL_DATA) $(SHELL_PROG).1 $(mandir)/`echo $(SHELL_PROG)|sed '$(transform)'`.$(manext)
+	$(INSTALL_PROGRAM) $(SHELL_PROG)$(exe_suffix) -D $(DESTDIR)$(bindir)/`echo $(SHELL_PROG)|sed '$(transform)'`$(exe_suffix)
+	-$(INSTALL_DATA) $(SHELL_PROG).1 -D $(DESTDIR)$(mandir)/`echo $(SHELL_PROG)|sed '$(transform)'`.$(manext)
 	-@prog=$(bindir)/`echo $(SHELL_PROG)|sed '$(transform)'`$(exe_suffix);\
 	    test -f /etc/shells \
 	    && (grep "^$$prog\$$" /etc/shells > /dev/null \
