diff -Naur veracrypt-1.26.7.orig/src/Common/PCSCException.cpp veracrypt-1.26.7/src/Common/PCSCException.cpp
--- veracrypt-1.26.7.orig/src/Common/PCSCException.cpp	2023-10-01 18:52:55.000000000 +0200
+++ veracrypt-1.26.7/src/Common/PCSCException.cpp	2024-06-04 10:00:30.424227000 +0200
@@ -140,14 +140,17 @@
 	{
 		Exception::Deserialize(stream);
 		Serializer sr(stream);
-		sr.Deserialize("ErrorCode", ErrorCode);
+		int64 v;
+		sr.Deserialize("ErrorCode", v);
+		ErrorCode = (LONG_PCSC)v;
 	}
 
 	void PCSCException::Serialize(shared_ptr <Stream> stream) const
 	{
 		Exception::Serialize(stream);
 		Serializer sr(stream);
-		sr.Serialize("ErrorCode", ErrorCode);
+		int64 v = (int64)ErrorCode;
+		sr.Serialize("ErrorCode", v);
 	}
 
 #	define TC_EXCEPTION(TYPE) TC_SERIALIZER_FACTORY_ADD(TYPE)
