diff -Naur wmgtemp-1.1/src/wmgtemp.c wmgtemp-1.1.patched/src/wmgtemp.c
--- wmgtemp-1.1/src/wmgtemp.c	2010-06-21 06:54:37.000000000 -0400
+++ wmgtemp-1.1.patched/src/wmgtemp.c	2016-08-16 08:38:01.208866564 -0400
@@ -90,8 +90,8 @@
 int SUBFEAT_NUM_CPU = 0;
 int SUBFEAT_NUM_SYS = 0;
 
-double cpu_history[59];
-double sys_history[59];
+double cpu_history[60];
+double sys_history[60];
 
 double display_min = 20;
 double display_max = 35;
