diff -u -r flow-0.29.0/hack/heap/hh_shared.c flow-0.29.0-patched/hack/heap/hh_shared.c
--- flow-0.29.0/hack/heap/hh_shared.c	2016-07-14 18:29:48.000000000 +0100
+++ flow-0.29.0-patched/hack/heap/hh_shared.c	2016-07-23 17:55:27.041786018 +0100
@@ -130,19 +130,14 @@
  * appeared in Linux 3.17.
  ****************************************************************************/
 #if !defined __APPLE__ && !defined _WIN32
+
   // Linux version for the architecture must support syscall memfd_create
-  #if defined(__x86_64__)
-    #define SYS_memfd_create 319
-  #elif defined(__powerpc64__)
-    #define SYS_memfd_create 360
-  #elif defined(__aarch64__)
-    #define SYS_memfd_create 385
-  #else
-    #error "hh_shared.c requires a architecture that supports memfd_create"
+  #include <sys/syscall.h>
+  #if !defined(SYS_memfd_create)
+    #error "hh_shared.c requires an architecture that supports memfd_create"
   #endif
 
   #define MEMFD_CREATE 1
-  #include <asm/unistd.h>
 
   /* Originally this function would call uname(), parse the linux
    * kernel release version and make a decision based on whether
