diff -Nur 4th-3.62.5-unix/sources/Makefile 4th-3.62.5-unix.new/sources/Makefile
--- 4th-3.62.5-unix/sources/Makefile	2015-04-23 07:59:18.000000000 +0200
+++ 4th-3.62.5-unix.new/sources/Makefile	2018-01-27 01:39:36.000000000 +0100
@@ -13,13 +13,13 @@
 
 # The following variables may need to be changed
 
-LIBRARIES=/usr/lib
-INCLUDES=/usr/include
-BINARIES=/usr/local/bin
-MANDIR=/usr/share/man
-DOCDIR=/usr/share/doc
+LIBRARIES=$(DESTDIR)/usr/lib$(LIBSUFFIX)
+INCLUDES=$(DESTDIR)/usr/include
+BINARIES=$(DESTDIR)/usr/bin
+MANDIR=$(DESTDIR)/usr/man
+DOCDIR=$(DESTDIR)/usr/doc
 SOURCES=../4th
-CFLAGS= -DUNIX -fsigned-char -Wall -O3 -s
+CFLAGS=$(CXXFLAGS) -DUNIX -fsigned-char -Wall -s
 
 # Cross compilation variables
 LD=$(CROSS)ld
@@ -125,7 +125,8 @@
 
 install: mostlyinstall
 	install -Dm644 ../documentation/4th.1 $(MANDIR)/man1/4th.1
-	install -Dm644 ../documentation/4tHmanual.txt $(DOCDIR)/4th/
+	install -Dm644 ../documentation/4tHmanual.txt $(DOCDIR)/4th-3.62.5/
+	install -Dm644 ../sources/4th.h $(INCLUDES)/4th.h
 
 uninstall:
 	-rm -f $(LIBRARIES)/lib4th.{a,so*}
