diff -Nur 4th-3.61.2-unix/sources/4th.h 4th-3.61.2-unix.new/sources/4th.h
--- 4th-3.61.2-unix/sources/4th.h	2009-11-20 23:52:02.000000000 +0000
+++ 4th-3.61.2-unix.new/sources/4th.h	2011-08-15 19:55:51.674547949 +0000
@@ -60,11 +60,11 @@
 #define M4DUPNAM 26
 #define M4CABORT 27
 
-#define CELL_MIN LONG_MIN
-#define CELL_MAX LONG_MAX
-#define strtocell(a,b,c) strtol((a),(b),(c))
+#define CELL_MIN INT_MIN
+#define CELL_MAX INT_MAX
+#define strtocell(a,b,c) (int)strtol((a),(b),(c))
 
-typedef long cell;
+typedef int cell;
 typedef unsigned char unit;
 
 typedef struct {
