diff -Nur 4th-3.61.2-unix/sources/Makefile 4th-3.61.2-unix.new/sources/Makefile
--- 4th-3.61.2-unix/sources/Makefile	2011-08-18 20:34:47.000000000 +0000
+++ 4th-3.61.2-unix.new/sources/Makefile	2011-08-20 01:10:29.653081410 +0000
@@ -16,13 +16,13 @@
 
 # The following variables may need to be changed
 
-LIBRARIES=/usr/lib
-INCLUDES=/usr/include
-BINARIES=/usr/local/bin
-MANDIR=/usr/share/man
-DOCDIR=/usr/share/doc
+LIBRARIES=$(DESTDIR)/usr/lib$(LIBSUFFIX)
+INCLUDES=$(DESTDIR)/usr/include
+BINARIES=$(DESTDIR)/usr/bin
+MANDIR=$(DESTDIR)/usr/man
+DOCDIR=$(DESTDIR)/usr/doc
 SOURCES=../4th
-CFLAGS= -DUNIX -fsigned-char -Wall -O3 -s
+CFLAGS=$(CXXFLAGS) -DUNIX -fsigned-char -Wall -s
 
 # Cross compilation variables
 LD=$(CROSS)ld
@@ -125,7 +125,7 @@
 
 install: mostlyinstall
 	install -Dm644 ../documentation/4th.1 $(MANDIR)/man1/4th.1
-	install -Dm644 ../documentation/4tHmanual.txt $(DOCDIR)/4th/
+	install -Dm644 ../documentation/4tHmanual.txt $(DOCDIR)/4th-3.61.2/
 
 uninstall:
 	-rm -f $(LIBRARIES)/lib4th.{a,so*}
