diff -Nur 4th-3.62.5-unix/sources/4th.h 4th-3.62.5-unix.new/sources/4th.h
--- 4th-3.62.5-unix/sources/4th.h	2012-04-09 18:15:03.000000000 +0200
+++ 4th-3.62.5-unix.new/sources/4th.h	2014-04-13 16:23:07.122971512 +0200
@@ -62,8 +62,8 @@
 #define M4BADNAM 28
 #define M4CABORT 29
 
-#define CELL_MIN LONG_MIN
-#define CELL_MAX LONG_MAX
+#define CELL_MIN INT_MIN
+#define CELL_MAX INT_MAX
 
 typedef long cell;
 typedef unsigned char unit;
