diff -Nur 4th-3.64.1/sources/4th.h 4th-3.64.1.new/sources/4th.h
--- 4th-3.64.1/sources/4th.h	2019-03-17 17:32:08.000000000 +0100
+++ 4th-3.64.1.new/sources/4th.h	2022-01-20 19:38:11.022969349 +0100
@@ -62,12 +62,12 @@
 #define M4BADNAM 28
 #define M4CABORT 29
 
-#define CELL_MIN LONG_MIN
-#define CELL_MAX LONG_MAX
+#define CELL_MIN INT_MIN
+#define CELL_MAX INT_MAX
 
 #define SYMLEN       24                /* length of word name (incl. \0) */
 
-typedef long cell;
+typedef int cell;
 typedef unsigned char unit;
 
 typedef struct {
