diff -Nur cmatrix-1.2a.orig/configure cmatrix-1.2a/configure
--- cmatrix-1.2a.orig/configure	2002-03-31 19:17:59.000000000 -0600
+++ cmatrix-1.2a/configure	2009-09-06 23:43:05.089753804 -0500
@@ -1876,7 +1876,7 @@
 if test x$CONSOLECHARS != x -o x$CONSOLECHARS != x'"$CONSOLECHARS"'; then
      if test x$SETFONT != x -o x$SETFONT != x'"$SETFONT"'; then
 
-		for ac_file in /usr/lib/kbd/consolefonts /usr/share/consolefonts
+		for ac_file in /usr/lib/kbd/consolefonts /usr/share/consolefonts /usr/share/kbd/consolefonts
 do
 
 ac_safe=`echo "$ac_file" | sed 'y%./+-%__p_%'`
@@ -1911,15 +1911,17 @@
 
 	if test "x$ac_cv_file__usr_lib_kbd_consolefonts" = "xno"; then
 	    if test "x$ac_cv_file__usr_share_consolefonts" = "xno"; then
-		echo "configure: warning:  
+	        if test "x$ac_cv_file__usr_share_kbd_consolefonts" = "xno"; then
+		    echo "configure: warning:  
 
 *** You do not appear to have a consolefonts directory in a standard location
-*** (/usr/lib/kbd or /usr/share), even though you appear to have the
+*** (/usr/lib/kbd or /usr/share/kbd or /usr/share), even though you appear to have the
 *** consolechars and/or setfont command.  The matrix font for the console
 *** will not be installed.  This means you will not be able to use the
 *** matrix console font (and the -l command line switch) unless the font
 *** is located in your current directory when you run CMatrix.
 " 1>&2
+	        fi
 	    fi
 	fi
      fi
@@ -1962,7 +1964,7 @@
   echo "$ac_t""no" 1>&6
 fi
 
-for ac_file in /usr/lib/X11/fonts/misc /usr/X11R6/lib/X11/fonts/misc
+for ac_file in /usr/lib/X11/fonts/misc /usr/X11R6/lib/X11/fonts/misc /usr/share/fonts/misc
 do
 
 ac_safe=`echo "$ac_file" | sed 'y%./+-%__p_%'`
@@ -1997,17 +1999,19 @@
 
 if test "x$ac_cv_file__usr_lib_X11_fonts_misc" = "xno"; then
     if test "x$ac_cv_file__usr_X11R6_lib_X11_fonts_misc" = "xno"; then
-	echo "configure: warning:  
+        if test "x$ac_cv_file__usr_share_fonts_misc" = "xno"; then
+	    echo "configure: warning:  
 
 *** You do not appear to have an X window fonts directory in the standard
-*** locations (/usr/lib/X11/fonts/misc or /usr/X11R6/lib/X11/fonts/misc). The
-*** mtx.pcf font will not be installed.  This means you will probably not
-*** be able to use the mtx fonts in your x terminals, and hence be unable
-*** to use the -x command line switch.  Sorry about that...
+*** locations (/usr/lib/X11/fonts/misc or /usr/X11R6/lib/X11/fonts/misc or
+*** /usr/share/fonts/misc). The mtx.pcf font will not be installed.  This 
+*** means you will probably not be able to use the mtx fonts in your x 
+*** terminals, and hence be unable to use the -x command line switch.  
+*** Sorry about that...
 " 1>&2
 	fi
     fi
-
+fi
 
 LIBS="$LIBS $CURSES_LIB"
 # Check whether --enable-debug or --disable-debug was given.
