diff -Naur mysql-workbench-gpl-5.2.43-src.orig/frontend/linux/workbench/mysql-workbench.in mysql-workbench-gpl-5.2.43-src/frontend/linux/workbench/mysql-workbench.in
--- mysql-workbench-gpl-5.2.43-src.orig/frontend/linux/workbench/mysql-workbench.in	2012-09-11 17:47:07.000000000 +0200
+++ mysql-workbench-gpl-5.2.43-src/frontend/linux/workbench/mysql-workbench.in	2013-02-28 18:24:14.217249244 +0100
@@ -4,6 +4,9 @@
 # This will cause passwords to be stored only temporarily for the session.
 #WB_NO_GNOME_KEYRING=1
 
+# create config dir (safe if it already exists)                                                                                                                                                                              
+mkdir -p $HOME/.mysql/workbench
+
 # force disable the Mac style single menu hack in Ubuntu Unity
 UBUNTU_MENUPROXY=0
 
diff -Naur mysql-workbench-gpl-5.2.43-src.orig/plugins/wb.admin/backend/wb_server_control.py mysql-workbench-gpl-5.2.43-src/plugins/wb.admin/backend/wb_server_control.py
--- mysql-workbench-gpl-5.2.43-src.orig/plugins/wb.admin/backend/wb_server_control.py	2012-09-11 17:47:17.000000000 +0200
+++ mysql-workbench-gpl-5.2.43-src/plugins/wb.admin/backend/wb_server_control.py	2013-02-28 18:25:09.285245528 +0100
@@ -32,7 +32,7 @@
 
 #-------------------------------------------------------------------------------
 def get_local_ip_list():
-    cmd    = "/bin/sh -c ifconfig"
+    cmd    = "/bin/sh -c /sbin/ifconfig"
     regexp = "inet addr:([0-9a-f:\.]+)"
 
     if hasattr(sys, 'getwindowsversion'):
