--- mysql-workbench-gpl-5.2.29-src/frontend/linux/linux_utilities/toolbar_manager.cpp	2010-10-11 04:28:19.000000000 -0300
+++ mysql-workbench-gpl-5.2.29-src_patched/frontend/linux/linux_utilities/toolbar_manager.cpp	2011-05-02 08:51:38.000000000 -0300
@@ -31,7 +31,7 @@
   }
 
   Glib::RefPtr<Gtk::ListStore> model= Gtk::ListStore::create(*color_combo_columns);
-  Gtk::ComboBox *combo= new Gtk::ComboBox(model);
+  Gtk::ComboBox *combo= new Gtk::ComboBox((Glib::RefPtr<Gtk::TreeModel> &)model);
 
   combo->pack_start(color_combo_columns->image);
 
