diff -Nur mysql-workbench-gpl-5.2.43-src/res/mysql.profiles/Makefile.am.orig mysql-workbench-gpl-5.2.43-src/res/mysql.profiles/Makefile.am
--- mysql-workbench-gpl-5.2.43-src/res/mysql.profiles/Makefile.am.orig 2012-09-13 21:49:26.288841700 +0700
+++ mysql-workbench-gpl-5.2.43-src/res/mysql.profiles/Makefile.am      2012-09-13 21:50:35.127099727 +0700
@@ -2,30 +2,30 @@
 workbenchdir=$(pkgdatadir)/mysql.profiles
 
 workbench_DATA=\
-Fedora_Linux_(MySQL_Package).xml\
-Fedora_Linux_(Vendor_Package).xml\
-Fedora_Linux_16_(Vendor_Package).xml\
-Fedora_Linux_15_(Vendor_Package).xml\
-FreeBSD_(MySQL_Package).xml\
-Generic_Linux_(MySQL_tar_package).xml\
-Linux_(Custom).xml\
-MacOS_X_(MySQL_Package).xml\
-OpenSolaris_(MySQL_Package).xml\
-RHEL_(MySQL_Package).xml\
-SLES_(MySQL_Package).xml\
-Ubuntu_Linux_(MySQL_Package).xml\
-Ubuntu_Linux_(Vendor_Package).xml\
-Ubuntu_Linux_Upstart_(MySQL_Package).xml\
-Ubuntu_Linux_Upstart_(Vendor_Package).xml\
-Windows_(MySQL_5.0_x64_Installer_Package).xml\
-Windows_(MySQL_5.0_x86_Installer_Package).xml\
-Windows_x64_(MySQL_5.0_x86_Installer_Package).xml\
-Windows_(MySQL_5.1_x64_Installer_Package).xml\
-Windows_(MySQL_5.1_x86_Installer_Package).xml\
-Windows_(MySQL_5.5_x64_Installer_Package).xml\
-Windows_(MySQL_5.5_x86_Installer_Package).xml\
-Windows_x64_(MySQL_5.1_x86_Installer_Package).xml\
-Windows_x64_(MySQL_5.5_x86_Installer_Package).xml\
-Windows_(MySQL_x64_zip_Package).xml\
-Windows_(MySQL_x86_zip_Package).xml\
-Windows_x64_(MySQL_x86_zip_Package).xml
+Fedora_Linux_\(MySQL_Package\).xml\
+Fedora_Linux_\(Vendor_Package\).xml\
+Fedora_Linux_16_\(Vendor_Package\).xml\
+Fedora_Linux_15_\(Vendor_Package\).xml\
+FreeBSD_\(MySQL_Package\).xml\
+Generic_Linux_\(MySQL_tar_package\).xml\
+Linux_\(Custom\).xml\
+MacOS_X_\(MySQL_Package\).xml\
+OpenSolaris_\(MySQL_Package\).xml\
+RHEL_\(MySQL_Package\).xml\
+SLES_\(MySQL_Package\).xml\
+Ubuntu_Linux_\(MySQL_Package\).xml\
+Ubuntu_Linux_\(Vendor_Package\).xml\
+Ubuntu_Linux_Upstart_\(MySQL_Package\).xml\
+Ubuntu_Linux_Upstart_\(Vendor_Package\).xml\
+Windows_\(MySQL_5.0_x64_Installer_Package\).xml\
+Windows_\(MySQL_5.0_x86_Installer_Package\).xml\
+Windows_x64_\(MySQL_5.0_x86_Installer_Package\).xml\
+Windows_\(MySQL_5.1_x64_Installer_Package\).xml\
+Windows_\(MySQL_5.1_x86_Installer_Package\).xml\
+Windows_\(MySQL_5.5_x64_Installer_Package\).xml\
+Windows_\(MySQL_5.5_x86_Installer_Package\).xml\
+Windows_x64_\(MySQL_5.1_x86_Installer_Package\).xml\
+Windows_x64_\(MySQL_5.5_x86_Installer_Package\).xml\
+Windows_\(MySQL_x64_zip_Package\).xml\
+Windows_\(MySQL_x86_zip_Package\).xml\
+Windows_x64_\(MySQL_x86_zip_Package\).xml
diff -Nur mysql-workbench-gpl-5.2.43-src/res/mysql.profiles/Makefile.in.orig mysql-workbench-gpl-5.2.43-src/res/mysql.profiles/Makefile.in
--- mysql-workbench-gpl-5.2.43-src/res/mysql.profiles/Makefile.in.orig 2012-09-13 21:50:59.495836763 +0700
+++ mysql-workbench-gpl-5.2.43-src/res/mysql.profiles/Makefile.in      2012-09-13 21:51:53.093257821 +0700
@@ -221,33 +221,33 @@
 unixodbc = @unixodbc@
 workbenchdir = $(pkgdatadir)/mysql.profiles
 workbench_DATA = \
-Fedora_Linux_(MySQL_Package).xml\
-Fedora_Linux_(Vendor_Package).xml\
-Fedora_Linux_16_(Vendor_Package).xml\
-Fedora_Linux_15_(Vendor_Package).xml\
-FreeBSD_(MySQL_Package).xml\
-Generic_Linux_(MySQL_tar_package).xml\
-Linux_(Custom).xml\
-MacOS_X_(MySQL_Package).xml\
-OpenSolaris_(MySQL_Package).xml\
-RHEL_(MySQL_Package).xml\
-SLES_(MySQL_Package).xml\
-Ubuntu_Linux_(MySQL_Package).xml\
-Ubuntu_Linux_(Vendor_Package).xml\
-Ubuntu_Linux_Upstart_(MySQL_Package).xml\
-Ubuntu_Linux_Upstart_(Vendor_Package).xml\
-Windows_(MySQL_5.0_x64_Installer_Package).xml\
-Windows_(MySQL_5.0_x86_Installer_Package).xml\
-Windows_x64_(MySQL_5.0_x86_Installer_Package).xml\
-Windows_(MySQL_5.1_x64_Installer_Package).xml\
-Windows_(MySQL_5.1_x86_Installer_Package).xml\
-Windows_(MySQL_5.5_x64_Installer_Package).xml\
-Windows_(MySQL_5.5_x86_Installer_Package).xml\
-Windows_x64_(MySQL_5.1_x86_Installer_Package).xml\
-Windows_x64_(MySQL_5.5_x86_Installer_Package).xml\
-Windows_(MySQL_x64_zip_Package).xml\
-Windows_(MySQL_x86_zip_Package).xml\
-Windows_x64_(MySQL_x86_zip_Package).xml
+Fedora_Linux_\(MySQL_Package\).xml\
+Fedora_Linux_\(Vendor_Package\).xml\
+Fedora_Linux_16_\(Vendor_Package).xml\
+Fedora_Linux_15_\(Vendor_Package\).xml\
+FreeBSD_\(MySQL_Package\).xml\
+Generic_Linux_\(MySQL_tar_package\).xml\
+Linux_\(Custom\).xml\
+MacOS_X_\(MySQL_Package\).xml\
+OpenSolaris_\(MySQL_Package\).xml\
+RHEL_\(MySQL_Package\).xml\
+SLES_\(MySQL_Package\).xml\
+Ubuntu_Linux_\(MySQL_Package\).xml\
+Ubuntu_Linux_\(Vendor_Package\).xml\
+Ubuntu_Linux_Upstart_\(MySQL_Package\).xml\
+Ubuntu_Linux_Upstart_\(Vendor_Package\).xml\
+Windows_\(MySQL_5.0_x64_Installer_Package\).xml\
+Windows_\(MySQL_5.0_x86_Installer_Package\).xml\
+Windows_x64_\(MySQL_5.0_x86_Installer_Package\).xml\
+Windows_\(MySQL_5.1_x64_Installer_Package\).xml\
+Windows_\(MySQL_5.1_x86_Installer_Package\).xml\
+Windows_\(MySQL_5.5_x64_Installer_Package\).xml\
+Windows_\(MySQL_5.5_x86_Installer_Package\).xml\
+Windows_x64_\(MySQL_5.1_x86_Installer_Package\).xml\
+Windows_x64_\(MySQL_5.5_x86_Installer_Package\).xml\
+Windows_\(MySQL_x64_zip_Package\).xml\
+Windows_\(MySQL_x86_zip_Package\).xml\
+Windows_x64_\(MySQL_x86_zip_Package\).xml
 
 all: all-am
