diff -Naur swh-lv2-1.0.15/Makefile swh-lv2-1.0.15.patched/Makefile
--- swh-lv2-1.0.15/Makefile	2008-01-09 16:02:08.000000000 -0500
+++ swh-lv2-1.0.15.patched/Makefile	2010-01-02 19:30:31.000000000 -0500
@@ -72,7 +72,7 @@
 
 %.so: NAME = $(shell echo -n $@ | sed 's/plugins\/\(.*\)-swh.*/\1/')
 %.so: %.xml %.o %.ttl
-	$(CC) $*.o $(PLUGIN_LDFLAGS) $($(NAME)_LDFLAGS) -o $@
+	$(CC) $*.o $(PLUGIN_LDFLAGS) $($(NAME)_LDFLAGS) -o $@ -lm
 	cp $@ $*-$(OS).$(EXT)
 	sed 's/@OS@/$(OS)/g' < `dirname $@`/manifest.ttl.in > `dirname $@`/manifest.ttl
 
