commit 994f03cf736229044a168835ae7387696041658f
Author: rezso <rezso@example.com>
Date:   Mon May 19 09:20:40 2014 +0100

    fix installing lucene++ headers in 3.0.6

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b0f3f20..6ae96b7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -138,6 +138,14 @@ if(NOT WIN32)
     DESTINATION "${LIB_DESTINATION}/pkgconfig")
 endif()
 
+#################################
+# install Config.h
+#################################
+install(
+  FILES
+  "${CMAKE_CURRENT_BINARY_DIR}/include/Config.h"
+  DESTINATION include/lucene++)
+
 ####################################
 # custom targets
 ####################################
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 83eb578..7d6153d 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -20,7 +20,7 @@ file(GLOB_RECURSE lucene_internal_headers
 )
 
 file(GLOB_RECURSE lucene_headers
-  include/*.h
+  "${lucene++_SOURCE_DIR}/include/*.h"
 )
 
 add_definitions(-DLPP_BUILDING_LIB)
