diff -Nur libgphoto2-2.4.3.orig/packaging/generic/check-ptp-camera libgphoto2-2.4.3/packaging/generic/check-ptp-camera
--- libgphoto2-2.4.3.orig/packaging/generic/check-ptp-camera	2007-07-27 00:52:50.000000000 -0500
+++ libgphoto2-2.4.3/packaging/generic/check-ptp-camera	2008-10-21 12:43:06.121742696 -0500
@@ -3,8 +3,8 @@
 # check if any interface of this device is a PTP camera interface (06)
 INTERFACE="${1:-06/01/01}"
 
-BASENAME=${PHYSDEVPATH##*/}
-for d in /sys/${PHYSDEVPATH}/${BASENAME}:*; do
+BASENAME=${DEVPATH##*/}
+for d in /sys/${DEVPATH}/${BASENAME}:*; do
 	[[ -d ${d} ]] || continue
 	INTERFACEID="$(< ${d}/bInterfaceClass)"
 	INTERFACEID="${INTERFACEID}/$(< ${d}/bInterfaceSubClass)"
