diff -Nur dwm-5.7.2.orig/config.mk dwm-5.7.2/config.mk
--- dwm-5.7.2.orig/config.mk	2009-09-27 14:20:23.000000000 -0500
+++ dwm-5.7.2/config.mk	2009-10-14 22:09:46.313425339 -0500
@@ -21,7 +21,7 @@
 # flags
 CPPFLAGS = -DVERSION=\"${VERSION}\" ${XINERAMAFLAGS}
 #CFLAGS = -g -std=c99 -pedantic -Wall -O0 ${INCS} ${CPPFLAGS}
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
+CFLAGS = -std=c99 -pedantic -Wall ${OPTS} ${INCS} ${CPPFLAGS}
 #LDFLAGS = -g ${LIBS}
 LDFLAGS = -s ${LIBS}
 
