diff --git a/data/desktop/tilem2.desktop b/data/desktop/tilem2.desktop
index 8118c97..2432f5f 100644
--- a/data/desktop/tilem2.desktop
+++ b/data/desktop/tilem2.desktop
@@ -7,4 +7,4 @@ MimeType=application/x-tigroup;application/x-ti73-variables;application/x-ti73-p
 StartupNotify=true
 Terminal=false
 Type=Application
-Categories=Education;Science;Math;Emulator;
+Categories=Education;Science;Math;System;Emulator;
