diff -ur tilem-2.0/gui/macro.c tilem-2.0-64bit/gui/macro.c
--- tilem-2.0/gui/macro.c	2011-11-26 21:47:07.000000000 +0100
+++ tilem-2.0-64bit/gui/macro.c	2012-07-23 22:48:24.000000000 +0200
@@ -134,7 +134,7 @@
 					char * lengthchar = g_new0(char, 4);
 					int length = strlen(emu->macro->actions[i]->value);
 					fwrite("file=", 1, 5, fp);
-					sprintf(lengthchar, "%04d", strlen(emu->macro->actions[i]->value));
+					sprintf(lengthchar, "%04zd", strlen(emu->macro->actions[i]->value));
 					fwrite(lengthchar, 1, sizeof(int), fp);
 					fwrite("-", 1, 1, fp);
 					fwrite(emu->macro->actions[i]->value, 1, length, fp);
