--- allegro-4.2.2/makefile.in	2007-07-22 02:55:54.000000000 -0300
+++ allegro-4.2.2_patched/makefile.in	2011-04-03 20:38:48.000000000 -0300
@@ -398,37 +398,32 @@
 
 # -------- rules for deleting the generated files --------
 
-clean:
-
    define RM_OBJ_CLEAN_FILES
-      $(foreach file, $(OBJ_CLEAN_FILES), rm -f $(file)
-      )
+      $(foreach file, $(OBJ_CLEAN_FILES), rm -f $(file))
    endef
 
    define RM_OTHER_CLEAN_FILES
-      $(foreach file, $(OTHER_CLEAN_FILES), rm -f $(file)
-      )
+      $(foreach file, $(OTHER_CLEAN_FILES), rm -f $(file))
    endef
 
-	$(RM_OBJ_CLEAN_FILES)
-	$(RM_OTHER_CLEAN_FILES)
-
-distclean: clean
-
    define RM_DISTCLEAN_FILES
       $(foreach file, $(DISTCLEAN_FILES) $(ALLEGRO_LIB_X_EXES), rm -f $(file)
       )
    endef
 
-	$(RM_DISTCLEAN_FILES)
-
-veryclean: distclean
-
    define RM_VERYCLEAN_FILES
       $(foreach file, $(VERYCLEAN_FILES), rm -f $(file)
       )
    endef
 
+clean:
+	$(RM_OBJ_CLEAN_FILES)
+	$(RM_OTHER_CLEAN_FILES)
+
+distclean: clean
+	$(RM_DISTCLEAN_FILES)
+
+veryclean: distclean
 	$(RM_VERYCLEAN_FILES)
 	rm -f makefile
 
