diff -Naur libaosd-xinerama-8a0d0d702e3b61c84439ce2f3f8de43bb4dcff07/examples/aosd_cat/aosd_cat.1.in libaosd-xinerama-8a0d0d702e3b61c84439ce2f3f8de43bb4dcff07.patched/examples/aosd_cat/aosd_cat.1.in
--- libaosd-xinerama-8a0d0d702e3b61c84439ce2f3f8de43bb4dcff07/examples/aosd_cat/aosd_cat.1.in	2014-11-27 08:01:03.000000000 -0500
+++ libaosd-xinerama-8a0d0d702e3b61c84439ce2f3f8de43bb4dcff07.patched/examples/aosd_cat/aosd_cat.1.in	2026-08-23 19:48:04.047416427 -0400
@@ -1,4 +1,4 @@
-.TH AOSD "1" "November 2007" "aosd_cat" "User Commands"
+.TH AOSD "1" "August 2026" "aosd_cat" "User Commands"
 .SH NAME
 aosd_cat - displays text on the screen
 .SH SYNOPSIS
@@ -15,6 +15,10 @@
 .TP
 Geometry Options:
 .TP
+\fB\-O\fR, \fB\-\-output\fR
+Sets the Xinerama output (head). Range is 0 to (number of heads - 1).
+Default value is \fB-1\fR.
+.TP
 \fB\-p\fR, \fB\-\-position\fR
 Sets the OSD window position. Or to be more precise, selects the point of the
 screen which is told to be the beginning of the coordinates (the 0,0 point)
@@ -47,6 +51,10 @@
 wrapped on screen width or will not be wrapped at all if other
 parameters make it impossible to layout correctly. Default value is \fB0\fR.
 .TP
+\fB\-A\fR, \fB\-\-alignment\fR
+Sets the text alignment (0 = left, 1 = center, 2 = right). Default value
+is \fB0\fR.
+.TP
 Coloring Options:
 .TP
 \fB\-B\fR, \fB\-\-back\-color\fR
@@ -90,8 +98,20 @@
 .TP
 Application Options:
 .TP
+\fB\-W,\fR \fB\-\-wait\fR
+Wait until the display is clear.
+.TP
+\fB\-k,\fR \fB\-\-keep-reading\fR
+Reopen input on EOF (doesn't have any effect when reading from stdin).
+.TP
+\fB\-m,\fR \fB\-\-markup\fR
+Enable Pango markup language.
+.TP
 \fB\-i,\fR \fB\-\-input\fR
 Sets the input text source. Default value is \fB-\fR.
+.TP
+\fB\-T,\fR \fB\-\-input-timeout\fR
+Sets the input timeout in seconds. Default value is \fB0\fR.
 .SH BUGS
 This manpage may be out of date. Use \-\-help to get the latest options.
 
