diff -Nur libpri-1.4.10.1.orig/Makefile libpri-1.4.10.1/Makefile
--- libpri-1.4.10.1.orig/Makefile	2008-08-06 13:20:51.000000000 -0500
+++ libpri-1.4.10.1/Makefile	2009-07-05 15:11:09.417405564 -0500
@@ -43,7 +43,7 @@
 DYNAMIC_LIBRARY:=libpri.so.$(SONAME)
 STATIC_OBJS=copy_string.o pri.o q921.o prisched.o q931.o pri_facility.o version.o
 DYNAMIC_OBJS=copy_string.lo pri.lo q921.lo prisched.lo q931.lo pri_facility.lo version.lo
-CFLAGS=-Wall -Werror -Wstrict-prototypes -Wmissing-prototypes -g -fPIC $(ALERTING) $(LIBPRI_COUNTERS)
+CFLAGS=-Wall -Werror -Wstrict-prototypes -Wmissing-prototypes -g -fPIC $(EXTRA_CFLAGS) $(ALERTING) $(LIBPRI_COUNTERS)
 INSTALL_PREFIX=$(DESTDIR)
 INSTALL_BASE=/usr
 libdir?=$(INSTALL_BASE)/lib
