diff -Nur NetworkManager-0.8.2.orig//initscript/Slackware/rc.networkmanager NetworkManager-0.8.2/initscript/Slackware/rc.networkmanager
--- NetworkManager-0.8.2.orig//initscript/Slackware/rc.networkmanager	2010-11-03 15:11:32.000000000 -0500
+++ NetworkManager-0.8.2/initscript/Slackware/rc.networkmanager	2010-12-23 00:39:39.858508485 -0600
@@ -27,11 +27,6 @@
     return
 	fi
   
-	if [ "`pgrep hald`" = "" ]; then
-	 echo "HAL must be running to start NetworkManager"
-	 return
-	fi
- 
   # Just in case the pidfile is still there, we may need to nuke it.
   if [ -e "$PIDFILE" ]; then
     rm -f $PIDFILE
@@ -67,6 +62,7 @@
 nm_restart()
 {
   nm_stop
+  sleep 2
   nm_start
 }
 
diff -Nur NetworkManager-0.8.2.orig//initscript/Slackware/rc.networkmanager.in NetworkManager-0.8.2/initscript/Slackware/rc.networkmanager.in
--- NetworkManager-0.8.2.orig//initscript/Slackware/rc.networkmanager.in	2009-12-23 12:30:29.000000000 -0600
+++ NetworkManager-0.8.2/initscript/Slackware/rc.networkmanager.in	2010-12-23 00:39:52.207382807 -0600
@@ -27,11 +27,6 @@
     return
 	fi
   
-	if [ "`pgrep hald`" = "" ]; then
-	 echo "HAL must be running to start NetworkManager"
-	 return
-	fi
- 
   # Just in case the pidfile is still there, we may need to nuke it.
   if [ -e "$PIDFILE" ]; then
     rm -f $PIDFILE
@@ -67,6 +62,7 @@
 nm_restart()
 {
   nm_stop
+  sleep 2
   nm_start
 }
 
