From 71fcccbacd51e939f168d44301393301e9dcb5fe Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ji=C5=99=C3=AD=20Klime=C5=A1?= <jklimes@redhat.com>
Date: Mon, 13 Dec 2010 20:55:02 +0100
Subject: [PATCH] libnm-util: fix a memory leaks in nm_setting_to_string()

---
 libnm-util/nm-setting.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/libnm-util/nm-setting.c b/libnm-util/nm-setting.c
index 06994c7..190199c 100644
--- a/libnm-util/nm-setting.c
+++ b/libnm-util/nm-setting.c
@@ -620,6 +620,8 @@ nm_setting_to_string (NMSetting *setting)
 		is_serializable = prop_spec->flags & NM_SETTING_PARAM_SERIALIZE;
 		is_default = g_param_value_defaults (prop_spec, &value);
 
+		g_value_unset (&value);
+
 		if (is_serializable || is_default) {
 			g_string_append (string, " (");
 
-- 
1.7.3.4

