diff -Nur NetworkManager-0.8.2.orig//initscript/Slackware/rc.networkmanager NetworkManager-0.8.2/initscript/Slackware/rc.networkmanager
--- NetworkManager-0.8.2.orig//initscript/Slackware/rc.networkmanager	2010-11-03 15:11:32.000000000 -0500
+++ NetworkManager-0.8.2/initscript/Slackware/rc.networkmanager	2010-12-18 15:41:06.884120091 -0600
@@ -67,6 +67,7 @@
 nm_restart()
 {
   nm_stop
+  sleep 2
   nm_start
 }
 
