diff --git a/configure b/configure
index fa7e189..9beaf07 100755
--- a/configure
+++ b/configure
@@ -4622,11 +4622,7 @@ LIBS=$ac_check_lib_save_LIBS
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_readline_rl_save_prompt" >&5
 $as_echo "$ac_cv_lib_readline_rl_save_prompt" >&6; }
-if test "x$ac_cv_lib_readline_rl_save_prompt" = xyes; then :
-   EXTRA_LIBS="${EXTRA_LIBS} -lreadline" ;
-else
-  as_fn_error $? "no libreadline found" "$LINENO" 5
-fi
+EXTRA_LIBS="${EXTRA_LIBS} -lreadline" ;
 
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libconfig" >&5
