Index: regexxer-0.9/src/globalstrings.h
===================================================================
--- regexxer-0.9.orig/src/globalstrings.h
+++ regexxer-0.9/src/globalstrings.h
@@ -21,7 +21,7 @@
 #ifndef REGEXXER_GLOBALSTRINGS_H_INCLUDED
 #define REGEXXER_GLOBALSTRINGS_H_INCLUDED
 
-#include <glib/gutils.h>  /* for G_DIR_SEPARATOR_S */
+#include <glib.h> /* for G_DIR_SEPARATOR_S */
 
 namespace Regexxer
 {
Index: regexxer-0.9/src/translation.h
===================================================================
--- regexxer-0.9.orig/src/translation.h
+++ regexxer-0.9/src/translation.h
@@ -21,7 +21,7 @@
 #ifndef REGEXXER_TRANSLATION_H_INCLUDED
 #define REGEXXER_TRANSLATION_H_INCLUDED
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glibmm/ustring.h>
 
 #ifndef gettext_noop
