diff -ur flatbuffers-1.5.0.old/CMakeLists.txt flatbuffers-1.5.0/CMakeLists.txt
--- flatbuffers-1.5.0.old/CMakeLists.txt	2017-01-03 20:56:44.000000000 -0200
+++ flatbuffers-1.5.0/CMakeLists.txt	2017-01-05 18:41:09.742061141 -0200
@@ -224,13 +224,13 @@
 if(FLATBUFFERS_INSTALL)
   install(DIRECTORY include/flatbuffers DESTINATION include)
   if(FLATBUFFERS_BUILD_FLATLIB)
-    install(TARGETS flatbuffers DESTINATION lib)
+    install(TARGETS flatbuffers DESTINATION lib${LIB_SUFFIX})
   endif()
   if(FLATBUFFERS_BUILD_FLATC)
     install(TARGETS flatc DESTINATION bin)
   endif()
   if(FLATBUFFERS_BUILD_SHAREDLIB)
-    install(TARGETS flatbuffers_shared DESTINATION lib)
+    install(TARGETS flatbuffers_shared DESTINATION lib${LIB_SUFFIX})
   endif()
 endif()
 
