diff -up oesenc_pi-2.9.1601/CMakeLists.txt.orig1 oesenc_pi-2.9.1601/CMakeLists.txt
--- oesenc_pi-2.9.1601/CMakeLists.txt.orig1	2019-04-03 20:56:51.982839060 -0500
+++ oesenc_pi-2.9.1601/CMakeLists.txt	2019-04-03 20:58:02.487835992 -0500
@@ -99,6 +99,10 @@ IF (NOT WIN32)
       ENDIF (CMAKE_SYSTEM_PROCESSOR MATCHES "arm*")
     ENDIF (EXISTS /etc/debian_version)
     
+   IF (EXISTS /etc/slackware-version)
+     SET (LIB_INSTALL_DIR "lib${LIB_SUFFIX}")
+   ENDIF (EXISTS /etc/slackware-version)
+    
     IF (EXISTS /etc/redhat-release)
       SET (PACKAGE_FORMAT "RPM")
       SET (PACKAGE_DEPS  "opencpn")
