diff -up oesenc_pi-3.2.0/CMakeLists.txt.orig1 oesenc_pi-3.2.0/CMakeLists.txt
--- oesenc_pi-3.2.0/CMakeLists.txt.orig1	2019-09-01 23:51:42.965365609 -0500
+++ oesenc_pi-3.2.0/CMakeLists.txt	2019-09-02 00:10:07.037326117 -0500
@@ -116,18 +116,17 @@ IF (NOT WIN32)
     ENDIF(NOT DEFINED PACKAGE_FORMAT)
  
     IF(NOT DEFINED PACKAGE_FORMAT)
-     IF (EXISTS /etc/os-release OR EXISTS /etc/sysconfig/SuSEfirewall2.d OR EXISTS /etc/suse-release OR EXISTS /etc/SuSE-release)
-      MESSAGE (STATUS "*** OpenSUSE detected  ***")
-      SET (PACKAGE_FORMAT "RPM")
-      SET (PACKAGE_DEPS  "opencpn")
+     IF (EXISTS /etc/slackware-version)
+      MESSAGE (STATUS "*** Slackware detected  ***")
+      SET (PACKAGE_FORMAT "TXZ")
       IF (CMAKE_SIZEOF_VOID_P MATCHES "8")
         SET (ARCH "x86_64")
-        SET (LIB_INSTALL_DIR "lib")
+        SET (LIB_INSTALL_DIR "lib64")
       ELSE (CMAKE_SIZEOF_VOID_P MATCHES "8")
         SET (ARCH "i386")
         SET (LIB_INSTALL_DIR "lib")
       ENDIF (CMAKE_SIZEOF_VOID_P MATCHES "8")
-     ENDIF (EXISTS /etc/os-release OR EXISTS /etc/sysconfig/SuSEfirewall2.d OR EXISTS /etc/suse-release OR EXISTS /etc/SuSE-release)
+     ENDIF (EXISTS /etc/slackware-version)
     ENDIF(NOT DEFINED PACKAGE_FORMAT)
 
   ENDIF (UNIX AND NOT APPLE)
