diff -Naur recipe-0.7.1.1-1~ppa2/docs/surl.1 recipe-0.7.1.1-1~ppa2.patched/docs/surl.1
--- recipe-0.7.1.1-1~ppa2/docs/surl.1	2013-09-01 08:27:48.000000000 -0400
+++ recipe-0.7.1.1-1~ppa2.patched/docs/surl.1	2020-11-14 00:21:20.723851357 -0500
@@ -124,16 +124,16 @@
 .INDENT 0.0
 .IP \(bu 2
 .
-\fCecho \(aqhttp://google.co.uk\(aq | surl \-a myapikey123 \-u example \-s bit.ly\fP
+\fCecho \(aqhttp://google.co.uk\(aq | surl \-a myapikey123 \-u example \-s turl.ca\fP
 .IP \(bu 2
 .
-\fCecho \(aqhttp://www.google.com\(aq | surl \-s tr.im\fP
+\fCecho \(aqhttp://www.google.com\(aq | surl \-s clck.ru\fP
 .IP \(bu 2
 .
-\fCsurl \-c http://www.google.com \-s tr.im\fP
+\fCsurl \-c http://www.google.com \-s is.gd\fP
 .IP \(bu 2
 .
-\fCsurl \-f example.txt \-i \-s tr.im\fP
+\fCsurl \-f example.txt \-i \-s is.gd\fP
 .UNINDENT
 .SH BUGS
 .sp
diff -Naur recipe-0.7.1.1-1~ppa2/docs/surl.rst recipe-0.7.1.1-1~ppa2.patched/docs/surl.rst
--- recipe-0.7.1.1-1~ppa2/docs/surl.rst	2013-09-01 08:27:48.000000000 -0400
+++ recipe-0.7.1.1-1~ppa2.patched/docs/surl.rst	2020-11-14 00:19:23.019862057 -0500
@@ -64,10 +64,10 @@
 
 EXAMPLES
 ========
-* ``echo 'http://google.co.uk' | surl -a myapikey123 -u example -s bit.ly``
-* ``echo 'http://www.google.com' | surl -s tr.im``
-* ``surl -c http://www.google.com -s tr.im``
-* ``surl -f example.txt -i -s tr.im``
+* ``echo 'http://google.co.uk' | surl -a myapikey123 -u example -s turl.ca``
+* ``echo 'http://www.google.com' | surl -s clck.ru``
+* ``surl -c http://www.google.com -s is.gd``
+* ``surl -f example.txt -i -s is.gd``
 
 BUGS
 ====
diff -Naur recipe-0.7.1.1-1~ppa2/surl/services.py recipe-0.7.1.1-1~ppa2.patched/surl/services.py
--- recipe-0.7.1.1-1~ppa2/surl/services.py	2013-09-01 08:27:48.000000000 -0400
+++ recipe-0.7.1.1-1~ppa2.patched/surl/services.py	2020-11-14 00:15:04.403885566 -0500
@@ -29,30 +29,11 @@
 def supportedServices():
     """ Returns a dictionary with the supported surl services """
     dict_service = {
-        'bit.ly'       : bitly_com('bit.ly'),
-        'bitly.com'    : bitly_com('bit.ly'),
-        'j.mp'         : bitly_com('j.mp'),
         'chilp.it'     : chilp_it(),
         'clck.ru'      : clck_ru(),
-        'crum.bs'      : crum_bs(),
-        'decenturl.com': decenturl_com(),
-        'goo.gl'       : goo_gl(),
         'is.gd'        : is_gd(),
-        'liip.to'      : liip_to(),
-        'metamark.net' : metamark_net(),
-        'redir.ec'     : redir_ec(),
-        'ri.ms'        : tinyarro_ws('ri.ms'),
-        'ta.gd'        : tinyarro_ws('ta.gd'),
-        'tinyarro.ws'  : tinyarro_ws('tinyarro.ws'),
-        'sn.im'        : sn_im('sn.im'),
-        'snipr.com'    : sn_im('snipr.com'),
-        'snipurl.com'  : sn_im('snipurl.com'),
-        'snurl.com'    : sn_im('snurl.com'),
-        'tiny.cc'      : tiny_cc(),
         'tinyurl.com'  : tinyurl_com(),
         'turl.ca'      : turl_ca(),
-        'twurl.nl'     : twurl_nl(),
-        'urlx.ru'      : urlx_ru(),
     }
     return dict_service
 
diff -Naur recipe-0.7.1.1-1~ppa2/surl/surl.py recipe-0.7.1.1-1~ppa2.patched/surl/surl.py
--- recipe-0.7.1.1-1~ppa2/surl/surl.py	2013-09-01 08:27:48.000000000 -0400
+++ recipe-0.7.1.1-1~ppa2.patched/surl/surl.py	2020-11-14 00:17:42.283871214 -0500
@@ -98,10 +98,10 @@
         print("  -r, --script <path>\t\tThe surlscript to execute")
         print("\n\nSupported URL shortening services:")
         print(" %s\n" % ', '.join(service_list))
-        print("Example:\n  echo 'http://google.co.uk' | surl -a myapikey123 -u example -s bit.ly")
-        print("  echo 'http://www.google.com' | surl -s tr.im")
-        print("  surl -c http://www.google.com -s tr.im")
-        print("  surl -f example.txt -i -s tr.im")
+        print("Example:\n  echo 'http://google.co.uk' | surl -a myapikey123 -u example -s turl.ca")
+        print("  echo 'http://www.google.com' | surl -s clck.ru")
+        print("  surl -c http://www.google.com -s is.gd")
+        print("  surl -f example.txt -i -s is.gd")
         ps.call("Help()")
         ps.unload()
         sys.exit(error)
