diff -r 7324ff1929d5 kallithea/lib/markup_renderer.py
--- a/kallithea/lib/markup_renderer.py	Thu May 27 21:28:32 2021 +0200
+++ b/kallithea/lib/markup_renderer.py	Thu Mar 14 12:25:06 2024 +0100
@@ -142,7 +142,7 @@
                   'strong', 'sub', 'sup', 'table', 'tbody', 'td', 'th',
                   'thead', 'tr', 'ul'],
             attributes=['class', 'id', 'style', 'label', 'title', 'alt', 'href', 'src'],
-            styles=['color'],
+            css_sanitizer=['color'],
             protocols=['http', 'https', 'mailto'],
             )
 
