diff -c a/setup.py b/setup.py
*** a/setup.py	2019-03-31 21:31:17.000000000 +0200
--- b/setup.py	2019-06-12 15:39:28.093303343 +0200
***************
*** 37,67 ****
  requirements = [
!     "alembic >= 0.8.0, < 1.1",
      "gearbox < 1",
!     "waitress >= 0.8.8, < 1.2",
!     "WebOb >= 1.7, < 1.8", # turbogears2 2.3.12 requires WebOb<1.8.0
      "backlash >= 0.1.2, < 1",
!     "TurboGears2 >= 2.3.10, < 2.4",
      "tgext.routes >= 0.2.0, < 1",
      "Beaker >= 1.7.0, < 2",
      "WebHelpers >= 1.3, < 1.4",
      "FormEncode >= 1.2.4, < 1.4",
!     "SQLAlchemy >= 1.1, < 1.3",
!     "Mako >= 0.9.0, < 1.1",
!     "Pygments >= 2.0, < 2.3",
      "Whoosh >= 2.5.0, < 2.8",
!     "celery >= 3.1, < 4.0", # celery 4 doesn't work
!     "Babel >= 1.3, < 2.7",
!     "python-dateutil >= 1.5.0, < 2.8",
!     "Markdown >= 2.2.1, < 2.7",
      "docutils >= 0.11, < 0.15",
      "URLObject >= 2.3.4, < 2.5",
!     "Routes >= 1.13, < 2",
      "dulwich >= 0.14.1, < 0.20",
!     "mercurial >= 4.1.1, < 4.10",
!     "decorator >= 3.3.2, < 4.4",
!     "Paste >= 2.0.3, < 3.1",
!     "bleach >= 3.0, < 3.1",
!     "Click >= 7.0, < 8",
  ]
  
  if sys.version_info < (2, 7):
--- 37,67 ----
  requirements = [
!     "alembic >= 0.8.0",
      "gearbox < 1",
!     "waitress >= 0.8.8",
!     "WebOb >= 1.7",
      "backlash >= 0.1.2, < 1",
!     "TurboGears2 >= 2.3.10",
      "tgext.routes >= 0.2.0, < 1",
      "Beaker >= 1.7.0, < 2",
      "WebHelpers >= 1.3, < 1.4",
      "FormEncode >= 1.2.4, < 1.4",
!     "SQLAlchemy >= 1.1",
!     "Mako >= 0.9.0",
!     "Pygments >= 2.0",
      "Whoosh >= 2.5.0, < 2.8",
!     "celery >= 3.1, < 4.0",  # celery 4 doesn't work
!     "Babel >= 1.3",
!     "python-dateutil >= 1.5.0",
!     "Markdown >= 2.2.1",
      "docutils >= 0.11, < 0.15",
      "URLObject >= 2.3.4, < 2.5",
!     "Routes >= 1.13",
      "dulwich >= 0.14.1, < 0.20",
!     "mercurial >= 4.1.1",
!     "decorator >= 3.3.2",
!     "Paste >= 2.0.3",
!     "bleach >= 3.0",
!     "Click >= 7.0",
  ]
  
  if sys.version_info < (2, 7):
