--- graphite-web-0.9.15/setup.py.orig	2015-11-27 19:37:16.000000000 +0100
+++ graphite-web-0.9.15/setup.py	2015-12-04 02:50:48.311066851 +0100
@@ -85,9 +85,8 @@
         'graphite.thirdparty.pytz',
       ],
       package_data={'graphite' :
-        ['templates/*', 'local_settings.py.example']},
-      scripts=glob('bin/*'),
-      data_files=webapp_content.items() + storage_dirs + conf_files + examples,
+        ['templates/*']},
+      data_files=webapp_content.items() + conf_files,
       **setup_kwargs
     )
 finally:
