--- graphite-web-0.9.15/setup.cfg.orig	2015-11-27 19:37:16.000000000 +0100
+++ graphite-web-0.9.15/setup.cfg	2015-12-04 00:24:12.045817187 +0100
@@ -1,16 +0,0 @@
-[bdist_rpm]
-requires = Django => 1.1.4
-           django-tagging
-           carbon
-           whisper
-           mod_wsgi
-           pycairo
-           pycairo-devel
-           python-simplejson
-           python-sqlite2
-           python-hashlib
-
-post-install = distro/redhat/misc/postinstall
-
-provides = graphite
-obsoletes = graphite <= 0.9.9
