--- graphite-web-0.9.13-pre1/setup.py.orig	2014-12-31 17:51:36.000000000 +0100
+++ graphite-web-0.9.13-pre1/setup.py	2015-02-19 18:45:25.173599991 +0100
@@ -58,8 +58,7 @@
     'graphite.thirdparty.pytz',
   ],
   package_data={'graphite' :
-    ['templates/*', 'local_settings.py.example']},
-  scripts=glob('bin/*'),
-  data_files=webapp_content.items() + storage_dirs + conf_files + examples,
+    ['templates/*']},
+  data_files=webapp_content.items() + conf_files,
   **setup_kwargs
 )
