--- Makefile	2006-09-09 21:25:16.000000000 -0300
+++ Makefile.new	2009-04-20 16:06:57.905013404 -0300
@@ -7,14 +7,17 @@
 #altere a linha abaixo se o seu tetex não puder se detetado
 #pelo comando kpsewich, na dúvida faça um "make test"
 #TEXDIR= /tmp/test
-export TEXDIR= ${shell kpsewhich --expand-path='$$TEXMFMAIN'}
+export TEXDIR= ${DESTDIR}/${shell kpsewhich --expand-path='$$TEXMFMAIN'}
 
-export USRLOCAL=/usr/local/bin
+export USRLOCAL=${DESTDIR}/usr/bin
 
 #Do "man lyx":
 #The system directory is determined by  searching  for  the
 #       file  "chkconfig.ltx".
-export LYXDIR=${dir ${shell  find /usr/share/lyx -name chkconfig.ltx}}
+export LYXDIR=${dir ${shell  find /usr/share/lyx -name chkconfig.ltx}}
+ifneq (${LYXDIR},)
+export LYXDIR:=${DESTDIR}/${LYXDIR}
+endif
 
 #diretório corrente onde se expandiu o cvs do abntex
 export ABNTEXDIR=$(notdir $(shell pwd))
@@ -119,14 +122,17 @@
 
 #install e uninstall específico para tetex localizado em ${TEXDIR}
 install-tex:
+	mkdir -p ${TEXDIR}
 	cp ${OPVR} --preserve texmf/* ${TEXDIR}
 
 install-lyx:
 ifneq (${LYXDIR},)
+	mkdir -p ${LYXDIR}
 	cp ${OPVR} --preserve lyx/* ${LYXDIR}
 endif
 
 install-bin:
+	mkdir -p ${USRLOCAL}
 	cp ${OPVR} --preserve bin/* ${USRLOCAL}
 
 post-install:
