diff -Naur fsviewer-app-0.2.6/src/FSFileView.c fsviewer-app-0.2.6.patched/src/FSFileView.c
--- fsviewer-app-0.2.6/src/FSFileView.c	2007-10-14 15:06:02.000000000 -0400
+++ fsviewer-app-0.2.6.patched/src/FSFileView.c	2013-01-21 23:55:02.000000000 -0500
@@ -430,7 +430,7 @@
 				   WMResizableWindowMask);
 	attributes.window_level = WMNormalWindowLevel;
 	attributes.extra_flags = GSFullKeyboardEventsFlag;
-	attributes.flags = (GSWindowStyleAttr | GSWindowLevelAttr |
+	attributes.flags = (/* GSWindowStyleAttr | */ GSWindowLevelAttr |
 			    GSExtraFlagsAttr);
 	WMSetWindowAttributes(fView->dpy, WMWidgetXID(fView->fileView),
 			      &attributes);
