diff -Naur fsviewer-app-0.2.6/defs/chdef.c fsviewer-app-0.2.6.patched/defs/chdef.c
--- fsviewer-app-0.2.6/defs/chdef.c	2007-10-14 12:18:25.000000000 -0400
+++ fsviewer-app-0.2.6.patched/defs/chdef.c	2013-01-21 20:12:00.000000000 -0500
@@ -374,7 +374,7 @@
 	SetIntegerForKey(0, "DisplayMCListPixmap");
 	
 	WMWritePropListToFile(filesDB,
-                wdefaultspathfordomain("FSViewer"), True);
+                wdefaultspathfordomain("FSViewer"));
 	result = 0;
     }	    
     else
diff -Naur fsviewer-app-0.2.6/src/extnInspector.c fsviewer-app-0.2.6.patched/src/extnInspector.c
--- fsviewer-app-0.2.6/src/extnInspector.c	2006-07-23 03:24:15.000000000 -0400
+++ fsviewer-app-0.2.6.patched/src/extnInspector.c	2013-01-21 20:09:39.000000000 -0500
@@ -229,7 +229,7 @@
 
     if(numRows > 0)
 	WMWritePropListToFile(filesDB,
-	  wdefaultspathfordomain("FSViewer"), True);
+	  wdefaultspathfordomain("FSViewer"));
 
    if(extn)
 	free(extn);
