diff -rupN ardour-2.8.12.old/libs/pbd/path.cc ardour-2.8.12/libs/pbd/path.cc
--- ardour-2.8.12.old/libs/pbd/path.cc	2009-02-24 07:38:16.000000000 -0500
+++ ardour-2.8.12/libs/pbd/path.cc	2012-05-02 23:39:41.446682468 -0400
@@ -17,6 +17,7 @@
 */
 
 #include <cerrno>
+#include <unistd.h>
 
 #include <glib.h>
 #include <glib/gstdio.h>
diff -rupN ardour-2.8.12.old/libs/rubberband/src/main.cpp ardour-2.8.12/libs/rubberband/src/main.cpp
--- ardour-2.8.12.old/libs/rubberband/src/main.cpp	2009-02-24 07:38:17.000000000 -0500
+++ ardour-2.8.12/libs/rubberband/src/main.cpp	2012-05-03 00:00:47.390285238 -0400
@@ -21,6 +21,7 @@
 #include <time.h>
 #include <cstdlib>
 #include <cstring>
+#include <unistd.h>
 #include "sysutils.h"
 
 #ifdef __MSVC__
