--- Makefile	2017-01-16 00:29:52.284871037 +0700
+++ Makefile	2017-01-16 01:19:42.578737717 +0700
@@ -2,7 +2,7 @@
 LINKFLAGS	= -lpthread
 
 DESTDIR		=
-PREFIX		= /usr/local
+PREFIX		= /usr
 SBINDIR		= $(PREFIX)/sbin
 
 OSD		= osdep
@@ -23,7 +23,7 @@
 
 install: mdk3
 	$(MAKE) -C $(OSD) install
-	install -D -m 0755 $^ $(DESTDIR)/$(SBINDIR)/$^
+	install -D -m 0755 $^ $(DESTDIR)$(SBINDIR)/$^
 
 clean:
 	rm -f mdk3
