--- aria2-1.19.0/src/a2gmp.h.orig	2015-05-26 23:16:33.145925238 +0700
+++ aria2-1.19.0/src/a2gmp.h	2015-05-26 23:17:01.022556391 +0700
@@ -35,6 +35,7 @@
 #ifndef D_A2GMP_H
 #define D_A2GMP_H
 
+#include <stddef.h>
 #include <gmp.h>
 
 namespace aria2 {
--- aria2-1.19.0/src/LibgmpDHKeyExchange.h.orig	2015-05-26 23:34:32.190146786 +0700
+++ aria2-1.19.0/src/LibgmpDHKeyExchange.h	2015-05-26 23:34:46.847939101 +0700
@@ -36,7 +36,7 @@
 #define D_LIBGMP_DH_KEY_EXCHANGE_H
 
 #include "common.h"
-
+#include <stddef.h>
 #include <gmp.h>
 
 namespace aria2 {
