diff -Nur kover-4.orig/src/directory.h kover-4/src/directory.h
--- kover-4.orig/src/directory.h	2008-11-15 15:59:26.000000000 -0600
+++ kover-4/src/directory.h	2009-11-28 22:55:39.994692905 -0600
@@ -10,6 +10,7 @@
 #include <config.h>
 #include <string>
 #include <list>
+#include <stdio.h>
 using namespace std;
 
 class directory
diff -Nur kover-4.orig/src/inexact_dialog.h kover-4/src/inexact_dialog.h
--- kover-4.orig/src/inexact_dialog.h	2008-11-15 15:59:26.000000000 -0600
+++ kover-4/src/inexact_dialog.h	2009-11-28 22:56:06.061713788 -0600
@@ -26,6 +26,7 @@
 #include <list>
 #include <string>
 #include <QListWidget>
+#include <stdio.h>
 
 using namespace std;
 
