--- asr-manpages-1.3.orig/debian/changelog
+++ asr-manpages-1.3/debian/changelog
@@ -0,0 +1,110 @@
+asr-manpages (1.3-6) unstable; urgency=low
+
+  * Rebuilt to force current standards (closes: #322807)
+  * While at that updated standards-version
+
+ -- Pawel Wiecek <coven@debian.org>  Thu, 22 Sep 2005 21:42:46 +0200
+
+asr-manpages (1.3-5) unstable; urgency=low
+
+  * re-applied some fixes that got lost during repackaging
+  * fixed one more type in luser.1fun
+
+ -- Pawel Wiecek <coven@debian.org>  Tue, 12 Jun 2001 15:52:50 +0200
+
+asr-manpages (1.3-4) unstable; urgency=low
+
+  * added symlinks for {axe,chainsaw,cutter}.8fun.gz (closes: #99541)
+  * now compliant with 3.5.5 policy
+
+ -- Pawel Wiecek <coven@debian.org>  Fri,  8 Jun 2001 09:38:33 +0200
+
+asr-manpages (1.3-3) unstable; urgency=low
+
+  * Added Build-Depends line to debian/control (was missing)
+
+ -- Pawel Wiecek <coven@debian.org>  Tue, 22 May 2001 12:25:58 +0200
+
+asr-manpages (1.3-2) unstable; urgency=low
+
+  * Re-applied fix to knife.8fun, that somehow got lost during repackaging
+    (closes: #96925)
+
+ -- Pawel Wiecek <coven@debian.org>  Thu, 10 May 2001 09:33:15 +0200
+
+asr-manpages (1.3-1) unstable; urgency=low
+
+  * Rebuilt archive, moving all manpages to fun subsection to avoid possible
+    name clashes
+  * General cleanup (new standards version, etc.)
+
+ -- Pawel Wiecek <coven@debian.org>  Mon,  7 May 2001 11:28:26 +0200
+
+asr-manpages (1.2-4) unstable; urgency=low
+
+  * Correction to knife.8 (closes: #58422)
+  * Updated standards version
+
+ -- Pawel Wiecek <coven@debian.org>  Fri,  1 Dec 2000 15:53:24 +0100
+
+asr-manpages (1.2-3) unstable; urgency=low
+
+  * Fixed a couple of typos (fixes bugs #42461, #34173)
+  * Symlinked n.1 to undocumented.7 (fixes #36617)
+  * Updated to standards version 3.0.1
+
+ -- Pawel Wiecek <coven@debian.org>  Tue, 26 Oct 1999 21:00:37 +0200
+
+asr-manpages (1.2-2) unstable; urgency=low
+
+  * Updated Standards-Version
+
+ -- Pawel Wiecek <coven@debian.org>  Sat, 31 Oct 1998 22:53:44 +0100
+
+asr-manpages (1.2-1) unstable; urgency=low
+
+  * Version number changed 'cause I messed .orig somehow
+
+ -- Pawel Wiecek <coven@pwr.wroc.pl>  Thu, 15 Jan 1998 00:00:10 +0100
+
+asr-manpages (1.1-5) unstable; urgency=low
+
+  * Uubp(1) moved to funny-manpages since it's not present in canonical
+    asr-manpages distribution.
+
+ -- Pawel Wiecek <coven@pwr.wroc.pl>  Sun, 11 Jan 1998 19:39:51 +0100
+
+asr-manpages (1.1-4) unstable; urgency=low
+
+  * Added Suggests: funny-manpages to control file to encourage people to
+    install non-a.s.r pages too
+
+ -- Pawel Wiecek <coven@pwr.wroc.pl>  Wed, 10 Dec 1997 01:37:38 +0100
+
+asr-manpages (1.1-3) unstable; urgency=low
+
+  * Fixed a typo in the description
+
+ -- Pawel Wiecek <coven@pwr.wroc.pl>  Sat, 16 Aug 1997 01:28:33 +0200
+
+asr-manpages (1.1-2) unstable; urgency=low
+
+  * Corrected copyright file (copyright info for c(1) page was missing)
+
+ -- Pawel Wiecek <coven@pwr.wroc.pl>  Mon, 28 Jul 1997 23:53:48 +0200
+
+asr-manpages (1.1-1) unstable; urgency=low
+
+  * New man page for c(1) added
+  * Added man page for uubp(1) which was missing from distribution
+    (required some reverse engeneering)
+
+ -- Pawel Wiecek <coven@pwr.wroc.pl>  Mon, 28 Jul 1997 00:56:58 +0200
+
+asr-manpages (1.0-1) unstable; urgency=low
+
+  * Initial Release.
+  * Added symlinks to axe, cutter and chainsaw
+
+ -- Pawel Wiecek <coven@pwr.wroc.pl>  Sun, 1 Jun 1997 23:13:56 +0200
+
--- asr-manpages-1.3.orig/debian/control
+++ asr-manpages-1.3/debian/control
@@ -0,0 +1,19 @@
+Source: asr-manpages
+Section: doc
+Priority: optional
+Maintainer: Pawel Wiecek <coven@debian.org>
+Build-Depends: debhelper (>> 3.0.0)
+Standards-Version: 3.6.2
+
+Package: asr-manpages
+Architecture: all
+Suggests: funny-manpages (>=1.3-2)
+Description: alt.sysadmin.recovery manual pages
+ A set of humorous manual pages developed on
+ alt.sysadmin.recovery (don't treat them seriously!).
+ They document a set of really useful tools that for some
+ strange reason are not included in any implementation of
+ Unix. This includes such famous commands as lart, sysadmin,
+ luser, bosskill and others.
+ The authors recommend these man pages should be installed
+ on every system.
--- asr-manpages-1.3.orig/debian/copyright
+++ asr-manpages-1.3/debian/copyright
@@ -0,0 +1,27 @@
+This package was debianized by Pawel Wiecek coven@pwr.wroc.pl on
+Sun, 1 Jun 1997 23:13:56 +0200.
+
+It was downloaded from ftp://ftp.winternet.com/users/eric/asr.pages.tar
+
+Uubp page is missing from distribution - it was reverse-engeneered from
+HTML version found on http://www.bofh.org.pl/man/.
+
+The authors of the pages are:
+ - c - Nick Williams <nickw@iname.com>
+ - lart - Tim Bandy <timn8r@risk.cs.umn.edu>
+ - slave - Ingvar Mattsson
+ - sysadmin - Abby Franquemont-Guillory <abbyfg@tezcat.com>
+ - think - John Guthrie <guthrie@math.upenn.edu>
+ - whack - Eric L. Pederson <eric@winternet.com>
+ - people - Ingvar Mattsson
+ - chastise - Ingvar Mattsson
+ - normality - Ingvar Mattsson
+ - bosskill - Eric L. Pederson <eric@winternet.com>
+ - ctluser - Simon Burr <simes@tcp.co.uk>
+ - guru - unknown author <user@foo.com>
+ - knife - Kurt Hockenbury <kurt@something.com>
+ - luser - Eric L. Pederson <eric@winternet.com>
+ - nuke - Eric S. Raymond <esr@thyrus.com>
+ - pmsd - Eric L. Pederson <eric@winternet.com>
+
+To the best of my knowledge they are all completely free.
--- asr-manpages-1.3.orig/debian/dirs
+++ asr-manpages-1.3/debian/dirs
@@ -0,0 +1 @@
+/usr/share/man/man8
--- asr-manpages-1.3.orig/debian/rules
+++ asr-manpages-1.3/debian/rules
@@ -0,0 +1,88 @@
+#!/usr/bin/make -f
+# Sample debian/rules that uses debhelper.
+# GNU copyright 1997 to 1999 by Joey Hess.
+
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+
+# This is the debhelper compatability version to use.
+export DH_COMPAT=3
+
+configure: configure-stamp
+configure-stamp:
+	dh_testdir
+	# Add here commands to configure the package.
+	
+
+	touch configure-stamp
+
+build: configure-stamp build-stamp
+build-stamp:
+	dh_testdir
+
+	# Add here commands to compile the package.
+#	$(MAKE)
+	#/usr/bin/docbook-to-man debian/asr-manpages.sgml > asr-manpages.1
+
+	touch build-stamp
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp configure-stamp
+
+	# Add here commands to clean up after the build process.
+#	-$(MAKE) clean
+
+	dh_clean
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_clean -k
+	dh_installdirs
+
+	# Add here commands to install the package into debian/asr-manpages.
+#	$(MAKE) install DESTDIR=$(CURDIR)/debian/asr-manpages
+	ln -s knife.8fun.gz debian/asr-manpages/usr/share/man/man8/axe.8fun.gz
+	ln -s knife.8fun.gz debian/asr-manpages/usr/share/man/man8/chainsaw.8fun.gz
+	ln -s knife.8fun.gz debian/asr-manpages/usr/share/man/man8/cutter.8fun.gz
+
+
+# Build architecture-independent files here.
+binary-indep: build install
+# We have nothing to do by default.
+
+# Build architecture-dependent files here.
+binary-arch: build install
+	dh_testdir
+	dh_testroot
+#	dh_installdebconf	
+	dh_installdocs
+#	dh_installexamples
+#	dh_installmenu
+#	dh_installlogrotate
+#	dh_installemacsen
+#	dh_installpam
+#	dh_installmime
+#	dh_installinit
+#	dh_installcron
+#	dh_installman		# dh_installman *sucks*
+	dh_installmanpages
+#	dh_installinfo
+#	dh_undocumented
+	dh_installchangelogs 
+#	dh_link
+	dh_strip
+	dh_compress
+	dh_fixperms
+#	dh_makeshlibs
+	dh_installdeb
+#	dh_perl
+	dh_shlibdeps
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary install configure
--- asr-manpages-1.3.orig/luser.8fun
+++ asr-manpages-1.3/luser.8fun
@@ -2,7 +2,7 @@
 .\" entertainment of the denizens of alt.sysadmin.recovery
 .TH LUSER 8 "25 September 1995"
 .SH NAME
-luser \- process to control the clueless induhviduals
+luser \- process to control the clueless individuals
 who (mis)use computer systems, peripheral devices
 and system administrators. Word play on "loser" and "user".
 .SH SYNOPSIS
--- asr-manpages-1.3.orig/people.2fun
+++ asr-manpages-1.3/people.2fun
@@ -27,7 +27,7 @@
   int is_amoron;
   int is_aluser;
   int has_aclue;
-  pid_t prgp_leader;
+  pid_t pgrp_leader;
 };
 .fi
 .SH RETURN VALUES
@@ -57,7 +57,7 @@
 is system dependent, but is usually about 3.
 .SH EXAMPLE
 .nf
-#include <sys/pople.h>
+#include <sys/people.h>
 #include <asr.h>
 #include <signal.h>
 
@@ -71,7 +71,7 @@
   if (rv!=-1) {
     for (c=0;c<rv;c++)
       if ((ttys[c].is_amoron)||(ttys[c].is_aluser)) {
-	kill(-(ttys[c].pgrp_leader),SIGKILL);
+        kill(-(ttys[c].pgrp_leader),SIGKILL);
       }
   } else {
     ; /* Handle errors in a graceful way... */
--- asr-manpages-1.3.orig/slave.1fun
+++ asr-manpages-1.3/slave.1fun
@@ -18,7 +18,7 @@
 .SH THE OPTION
 .TP
 \-\fIf\fP
-Send a rapid succession of \fISIHUP\fPs to \fIslave\fP, mainly used
+Send a rapid succession of \fISIGHUP\fPs to \fIslave\fP, mainly used
 when one needs to blow off a little steam.
 .SH THE COMMANDS
 Almost any phrase in pidgin English is acceptable input to the
--- asr-manpages-1.3.orig/sysadmin.1fun
+++ asr-manpages-1.3/sysadmin.1fun
@@ -1,5 +1,5 @@
-\." Written by Abby Franquement-Guillory (abbyfg@tezcat.com).
-\." Convered to troff by Eric Pederson (eric@winternet.com).
+.\" Written by Abby Franquement-Guillory (abbyfg@tezcat.com).
+.\" Convered to troff by Eric Pederson (eric@winternet.com).
 .TH SYSADMIN 1 "25 September 1995"
 .SH NAME
 sysadmin \- responsible for everything imaginable that may
--- asr-manpages-1.3.orig/knife.8fun
+++ asr-manpages-1.3/knife.8fun
@@ -1,5 +1,5 @@
 .TH KNIFE 8 
-.SH KNIFE
+.SH NAME
 knife, axe, cutter, chainsaw \- tools to improve network performance via SNIP
 .SH SYNOPSIS
 .B knife 
