diff -r 26b5095aa992 -r cfacb1bd4433 rc.h
--- a/rc.h	Sat Oct 18 21:54:02 2025 -0400
+++ b/rc.h	Sat Oct 18 21:57:59 2025 -0400
@@ -7,6 +7,9 @@
 /* for struct stat */
 #include <sys/stat.h>
 
+/* for bool, true & false */
+#include <stdbool.h>
+
 #define RC "rc: "
 
 /* datatypes */
@@ -47,9 +50,14 @@
 	eError, eBreak, eReturn, eVarstack, eArena, eFifo, eFd, eContinue
 } ecodes;
 
+#if __bool_true_false_are_defined
+#define TRUE true
+#define FALSE false
+#else
 typedef enum bool {
 	FALSE, TRUE
 } bool;
+#endif
 
 typedef enum redirtype {
 	rFrom, rCreate, rAppend, rHeredoc, rHerestring
