diff -ru MilkyTracker-1.0.0/cmake/FindSDL2.cmake MilkyTracker-sdlmain/cmake/FindSDL2.cmake
--- MilkyTracker-1.0.0/cmake/FindSDL2.cmake	2017-03-12 21:30:25.000000000 +0200
+++ MilkyTracker-sdlmain/cmake/FindSDL2.cmake	2017-03-19 20:23:15.773398294 +0200
@@ -93,7 +93,7 @@
 endif()
 
 set(SDL2_INCLUDE_DIRS ${SDL2_INCLUDE_DIR})
-set(SDL2_LIBRARIES ${SDL2MAIN_LIBRARY} ${SDL2_LIBRARY})
+set(SDL2_LIBRARIES ${SDL2_LIBRARY})
 
 include(FindPackageHandleStandardArgs)
 
