--- OpenBoard-1.6.0a/plugins/cffadaptor/UBCFFAdaptor.pro.orig	2020-04-07 08:26:29.240677360 +0700
+++ OpenBoard-1.6.0a/plugins/cffadaptor/UBCFFAdaptor.pro	2020-04-07 08:27:03.695993158 +0700
@@ -39,7 +39,7 @@
     QMAKE_POST_LINK += bash postScript_mac.sh "$$DESTDIR" "$$SANKORE_DIR/$$BUILD_DIR/product/Open-Sankore.app/Contents/MacOS"
 }
 
-LIBS     += "-L$$THIRD_PARTY_PATH/quazip/lib/$$SUB_DIR" "-lquazip"
+LIBS     += "-lquazip5"
 
 QT       += xml xmlpatterns core
 QT       += gui
@@ -51,10 +51,7 @@
 
 INCLUDEPATH += src
 
-DEPENDPATH  += $$THIRD_PARTY_PATH/quazip/
-INCLUDEPATH += $$THIRD_PARTY_PATH/quazip/
-include($$THIRD_PARTY_PATH/quazip/quazip.pri)
-
+INCLUDEPATH += /usr/include/quazip5/
 
 SOURCES += \
     src/UBCFFAdaptor.cpp
@@ -66,4 +63,4 @@
     src/UBCFFConstants.h
 
 RESOURCES += \
-    resources/resources.qrc
\ No newline at end of file
+    resources/resources.qrc
--- OpenBoard-1.6.0a3/OpenBoard.pro.orig	2020-07-20 08:23:09.859918480 +0700
+++ OpenBoard-1.6.0a3/OpenBoard.pro	2020-07-20 08:23:17.762002419 +0700
@@ -434,7 +434,7 @@
     #LIBS += -lprofiler
     LIBS += -lX11
     LIBS += -lquazip5
-    INCLUDEPATH += "/usr/include/quazip"
+    INCLUDEPATH += "/usr/include/quazip5"
 
     LIBS += -lpoppler
     INCLUDEPATH += "/usr/include/poppler"
