--- pkginfo-1.4.0/setup.orig.py	2016-11-05 05:43:07.149052800 +0200
+++ pkginfo-1.4.0/setup.py	2016-11-05 05:43:34.139715370 +0200
@@ -16,14 +16,14 @@
 
 here = os.path.abspath(os.path.dirname(__file__))
 README = open(os.path.join(here, 'README.txt')).read()
-CHANGES = open(os.path.join(here, 'CHANGES.txt')).read()
+# CHANGES = open(os.path.join(here, 'CHANGES.txt')).read()
 
 setup(
     name='pkginfo',
     version='1.4.0',
     description='Query metadatdata from sdists / bdists / installed packages.',
     platforms=['Unix', 'Windows'],
-    long_description='\n\n'.join([README, CHANGES]),
+    long_description='\n\n'.join([README]),
     keywords='distribution sdist installed metadata',
     url='https://code.launchpad.net/~tseaver/pkginfo/trunk',
     author='Tres Seaver, Agendaless Consulting',
