diff -Naur muse-3.0.2/muse/mplugins/mitplugin.cpp muse-3.0.2.patched/muse/mplugins/mitplugin.cpp
--- muse-3.0.2/muse/mplugins/mitplugin.cpp	2017-12-04 22:01:18.000000000 +0100
+++ muse-3.0.2.patched/muse/mplugins/mitplugin.cpp	2020-02-26 23:27:20.013277911 +0100
@@ -21,6 +21,7 @@
 //
 //=========================================================
 
+#include <QtWidgets>
 #include "mitplugin.h"
 #include "app.h"
 #include "event.h"
diff -Naur muse-3.0.2/muse/widgets/genset.cpp muse-3.0.2.patched/muse/widgets/genset.cpp
--- muse-3.0.2/muse/widgets/genset.cpp	2018-01-22 17:43:28.000000000 +0100
+++ muse-3.0.2.patched/muse/widgets/genset.cpp	2020-02-26 21:47:19.446707723 +0100
@@ -24,6 +24,7 @@
 
 #include <stdio.h>
 
+#include <QButtonGroup>
 #include <QFileDialog>
 #include <QFileInfo>
 #include <QRect>
