diff -Naur gqradio-1.9.2/src/ui_fileops.c gqradio-1.9.2-new/src/ui_fileops.c
--- gqradio-1.9.2/src/ui_fileops.c	2005-02-22 20:40:49.000000000 -0300
+++ gqradio-1.9.2-new/src/ui_fileops.c	2014-05-05 21:52:18.121233702 -0300
@@ -44,7 +44,7 @@
 	gchar *text_l;
 
 	text_l = g_locale_from_utf8(text_utf8, -1, NULL, NULL, NULL);
-	printf((text_l) ? text_l : text_utf8);
+	printf("%s", (text_l) ? text_l : text_utf8);
 	g_free(text_l);
 }
 
