diff -Naur gqradio-1.9.2/configure gqradio-1.9.2-new/configure
--- gqradio-1.9.2/configure	2005-02-22 21:58:29.000000000 -0300
+++ gqradio-1.9.2-new/configure	2010-02-25 07:12:37.000000000 -0300
@@ -4327,6 +4327,7 @@
             echo "$as_me:$LINENO: checking GQRADIO_LIBS" >&5
 echo $ECHO_N "checking GQRADIO_LIBS... $ECHO_C" >&6
             GQRADIO_LIBS=`$PKG_CONFIG --libs "gtk+-2.0 >= 2.4.0"`
+            GQRADIO_LIBS="$GQRADIO_LIBS -lX11 -lm"
             echo "$as_me:$LINENO: result: $GQRADIO_LIBS" >&5
 echo "${ECHO_T}$GQRADIO_LIBS" >&6
         else
