diff -Naur gqradio-1.9.2/configure gqradio-1.9.2-new/configure
--- gqradio-1.9.2/configure	2005-02-22 21:58:29.000000000 -0300
+++ gqradio-1.9.2-new/configure	2012-10-25 20:07:45.745138407 -0200
@@ -4430,8 +4430,8 @@
 
 
 if test "${ac_cv_header_linux_videodev_h+set}" = set; then
-  echo "$as_me:$LINENO: checking for linux/videodev.h" >&5
-echo $ECHO_N "checking for linux/videodev.h... $ECHO_C" >&6
+  echo "$as_me:$LINENO: checking for linux/videodev2.h" >&5
+echo $ECHO_N "checking for linux/videodev2.h... $ECHO_C" >&6
 if test "${ac_cv_header_linux_videodev_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 fi
@@ -4439,8 +4439,8 @@
 echo "${ECHO_T}$ac_cv_header_linux_videodev_h" >&6
 else
   # Is the header compilable?
-echo "$as_me:$LINENO: checking linux/videodev.h usability" >&5
-echo $ECHO_N "checking linux/videodev.h usability... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking linux/videodev2.h usability" >&5
+echo $ECHO_N "checking linux/videodev2.h usability... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -4448,7 +4448,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <linux/videodev.h>
+#include <linux/videodev2.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
@@ -4484,15 +4484,15 @@
 echo "${ECHO_T}$ac_header_compiler" >&6
 
 # Is the header present?
-echo "$as_me:$LINENO: checking linux/videodev.h presence" >&5
-echo $ECHO_N "checking linux/videodev.h presence... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking linux/videodev2.h presence" >&5
+echo $ECHO_N "checking linux/videodev2.h presence... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <linux/videodev.h>
+#include <linux/videodev2.h>
 _ACEOF
 if { (eval echo "$as_me:$LINENO: \"$ac_cpp conftest.$ac_ext\"") >&5
   (eval $ac_cpp conftest.$ac_ext) 2>conftest.er1
@@ -4526,25 +4526,25 @@
 # So?  What about this header?
 case $ac_header_compiler:$ac_header_preproc:$ac_c_preproc_warn_flag in
   yes:no: )
-    { echo "$as_me:$LINENO: WARNING: linux/videodev.h: accepted by the compiler, rejected by the preprocessor!" >&5
-echo "$as_me: WARNING: linux/videodev.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
-    { echo "$as_me:$LINENO: WARNING: linux/videodev.h: proceeding with the compiler's result" >&5
-echo "$as_me: WARNING: linux/videodev.h: proceeding with the compiler's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: linux/videodev2.h: accepted by the compiler, rejected by the preprocessor!" >&5
+echo "$as_me: WARNING: linux/videodev2.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
+    { echo "$as_me:$LINENO: WARNING: linux/videodev2.h: proceeding with the compiler's result" >&5
+echo "$as_me: WARNING: linux/videodev2.h: proceeding with the compiler's result" >&2;}
     ac_header_preproc=yes
     ;;
   no:yes:* )
-    { echo "$as_me:$LINENO: WARNING: linux/videodev.h: present but cannot be compiled" >&5
-echo "$as_me: WARNING: linux/videodev.h: present but cannot be compiled" >&2;}
-    { echo "$as_me:$LINENO: WARNING: linux/videodev.h:     check for missing prerequisite headers?" >&5
-echo "$as_me: WARNING: linux/videodev.h:     check for missing prerequisite headers?" >&2;}
-    { echo "$as_me:$LINENO: WARNING: linux/videodev.h: see the Autoconf documentation" >&5
-echo "$as_me: WARNING: linux/videodev.h: see the Autoconf documentation" >&2;}
-    { echo "$as_me:$LINENO: WARNING: linux/videodev.h:     section \"Present But Cannot Be Compiled\"" >&5
-echo "$as_me: WARNING: linux/videodev.h:     section \"Present But Cannot Be Compiled\"" >&2;}
-    { echo "$as_me:$LINENO: WARNING: linux/videodev.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: linux/videodev.h: proceeding with the preprocessor's result" >&2;}
-    { echo "$as_me:$LINENO: WARNING: linux/videodev.h: in the future, the compiler will take precedence" >&5
-echo "$as_me: WARNING: linux/videodev.h: in the future, the compiler will take precedence" >&2;}
+    { echo "$as_me:$LINENO: WARNING: linux/videodev2.h: present but cannot be compiled" >&5
+echo "$as_me: WARNING: linux/videodev2.h: present but cannot be compiled" >&2;}
+    { echo "$as_me:$LINENO: WARNING: linux/videodev2.h:     check for missing prerequisite headers?" >&5
+echo "$as_me: WARNING: linux/videodev2.h:     check for missing prerequisite headers?" >&2;}
+    { echo "$as_me:$LINENO: WARNING: linux/videodev2.h: see the Autoconf documentation" >&5
+echo "$as_me: WARNING: linux/videodev2.h: see the Autoconf documentation" >&2;}
+    { echo "$as_me:$LINENO: WARNING: linux/videodev2.h:     section \"Present But Cannot Be Compiled\"" >&5
+echo "$as_me: WARNING: linux/videodev2.h:     section \"Present But Cannot Be Compiled\"" >&2;}
+    { echo "$as_me:$LINENO: WARNING: linux/videodev2.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: linux/videodev2.h: proceeding with the preprocessor's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: linux/videodev2.h: in the future, the compiler will take precedence" >&5
+echo "$as_me: WARNING: linux/videodev2.h: in the future, the compiler will take precedence" >&2;}
     (
       cat <<\_ASBOX
 ## ------------------------------------------ ##
@@ -4555,8 +4555,8 @@
       sed "s/^/$as_me: WARNING:     /" >&2
     ;;
 esac
-echo "$as_me:$LINENO: checking for linux/videodev.h" >&5
-echo $ECHO_N "checking for linux/videodev.h... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking for linux/videodev2.h" >&5
+echo $ECHO_N "checking for linux/videodev2.h... $ECHO_C" >&6
 if test "${ac_cv_header_linux_videodev_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
diff -Naur gqradio-1.9.2/configure.in gqradio-1.9.2-new/configure.in
--- gqradio-1.9.2/configure.in	2005-02-22 21:58:06.000000000 -0300
+++ gqradio-1.9.2-new/configure.in	2012-10-25 20:06:54.853152918 -0200
@@ -22,7 +22,7 @@
 AC_SUBST([GQRADIO_LIBS])
 
 dnl Video4Linux
-AC_CHECK_HEADER(linux/videodev.h, AC_DEFINE(HAVE_VIDEO4LINUX,,[Video4Linux support]))
+AC_CHECK_HEADER(linux/videodev2.h, AC_DEFINE(HAVE_VIDEO4LINUX,,[Video4Linux support]))
 
 dnl bktr
 AC_CHECK_HEADER(machine/ioctl_bt848.h, AC_DEFINE(HAVE_IOCTL_BT848,,[FreeBSD bktr support]))
