diff -Nur gnumeric-1.10.12.orig//configure gnumeric-1.10.12//configure
--- gnumeric-1.10.12.orig//configure	2010-11-20 11:27:24.000000000 -0600
+++ gnumeric-1.10.12//configure	2010-11-28 04:22:21.807932393 -0600
@@ -12807,7 +12807,7 @@
 
 libspreadsheet_reqs="
 	libgoffice-${GOFFICE_API_VER}	>= 0.8.10
-	libgsf-1		>= 1.14.18
+	libgsf-1		>= 1.14.16
 	libxml-2.0		>= 2.4.12
 "
 gnumeric_reqs="$libspreadsheet_reqs
