--- xen-4.5.0/tools/qemu-xen/libcacard/Makefile.orig	2014-12-02 11:41:02.000000000 +0100
+++ xen-4.5.0/tools/qemu-xen/libcacard/Makefile	2015-03-14 21:05:54.361325689 +0100
@@ -26,8 +26,8 @@
 libcacard.la: LDFLAGS += -rpath $(libdir) -no-undefined \
 	-export-syms $(SRC_PATH)/libcacard/libcacard.syms
 libcacard.la: LIBS = $(libcacard_libs)
-libcacard.la: $(libcacard-lobj-y)
-	$(call LINK,$^)
+libcacard.la: $(libcacard-obj-y)
+	$(call LINK,$(libcacard-lobj-y))
 
 libcacard.pc: $(SRC_PATH)/libcacard/libcacard.pc.in
 	$(call quiet-command,sed -e 's|@LIBDIR@|$(libdir)|' \
