Subject: [PATCH] Fix gcc7 warn

---
 tools/firmware/etherboot/patches/series | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tools/firmware/etherboot/patches/series b/tools/firmware/etherboot/patches/series
index 86cb300..780c6c6 100644
--- a/tools/firmware/etherboot/patches/series
+++ b/tools/firmware/etherboot/patches/series
@@ -1 +1,4 @@
 boot_prompt_option.patch
+patch-inbuild-ipxe-gcc7-implicit-fallthrough.patch
+patch-inbuild-ipxe-gcc7-implicit-fallthrough-ath5k.patch
+patch-inbuild-ipxe-gcc7-implicit-fallthrough-curses.patch
-- 
2.13.0

