--- xen-4.2.1/tools/hotplug/Linux/xend.rules.ORIG	2012-12-17 16:00:35.000000000 +0100
+++ xen-4.2.1/tools/hotplug/Linux/xend.rules	2013-03-01 04:26:34.247176457 +0100
@@ -1,4 +1,4 @@
-SUBSYSTEM=="pci", RUN+="socket:/org/xen/xend/udev_event"
-SUBSYSTEM=="scsi", RUN+="socket:/org/xen/xend/udev_event"
-SUBSYSTEM=="usb", RUN+="socket:/org/xen/xend/udev_event"
+#SUBSYSTEM=="pci", RUN+="socket:/org/xen/xend/udev_event"
+#SUBSYSTEM=="scsi", RUN+="socket:/org/xen/xend/udev_event"
+#SUBSYSTEM=="usb", RUN+="socket:/org/xen/xend/udev_event"
 #SUBSYSTEM=="net", KERNEL!="vif[0-9]*.[0-9]*|tap[0-9]*.[0-9]*", RUN+="socket:/org/xen/xend/udev_event"
