--- xen-4.2.0/tools/xenstore/Makefile.ORIG	2012-09-17 12:21:19.000000000 +0200
+++ xen-4.2.0/tools/xenstore/Makefile	2012-10-21 22:59:54.585759242 +0200
@@ -64,7 +64,7 @@
 	$(AR) cr $@ $^
 
 $(CLIENTS): xenstore
-	ln -f xenstore $@
+	ln -sf xenstore $@
 
 xenstore: xenstore_client.o $(LIBXENSTORE)
 	$(CC) $(LDFLAGS) $< $(LDLIBS_libxenstore) $(SOCKET_LIBS) -o $@ $(APPEND_LDFLAGS)
@@ -116,7 +116,7 @@
 	$(INSTALL_PROG) xenstore-control $(DESTDIR)$(BINDIR)
 	$(INSTALL_PROG) xenstore $(DESTDIR)$(BINDIR)
 	set -e ; for c in $(CLIENTS) ; do \
-		ln -f $(DESTDIR)$(BINDIR)/xenstore $(DESTDIR)$(BINDIR)/$${c} ; \
+		ln -sf xenstore $(DESTDIR)$(BINDIR)/$${c} ; \
 	done
 	$(INSTALL_DIR) $(DESTDIR)$(LIBDIR)
 	$(INSTALL_PROG) libxenstore.so.$(MAJOR).$(MINOR) $(DESTDIR)$(LIBDIR)
