--- xen-4.3.1/tools/firmware/Makefile.ORIG	2013-10-30 09:18:28.000000000 +0100
+++ xen-4.3.1/tools/firmware/Makefile	2013-11-04 12:28:10.056796654 +0100
@@ -18,7 +18,7 @@
 	cp ovmf-makefile ovmf/Makefile;
 
 seabios-dir:
-	GIT=$(GIT) $(XEN_ROOT)/scripts/git-checkout.sh $(SEABIOS_UPSTREAM_URL) $(SEABIOS_UPSTREAM_TAG) seabios-dir
+#	GIT=$(GIT) $(XEN_ROOT)/scripts/git-checkout.sh $(SEABIOS_UPSTREAM_URL) $(SEABIOS_UPSTREAM_TAG) seabios-dir
 	cp seabios-config seabios-dir/.config;
 
 .PHONY: all
