--- xen-4.3.0/tools/firmware/etherboot/Makefile.ORIG	2013-07-09 12:46:56.000000000 +0200
+++ xen-4.3.0/tools/firmware/etherboot/Makefile	2013-07-28 13:34:32.994197893 +0200
@@ -28,12 +28,12 @@
 	$(MAKE) -C $D/src bin/$(*F).rom
 
 $T:
-	if ! $(FETCHER) _$T $(IPXE_TARBALL_URL); then \
-		$(GIT) clone $(IPXE_GIT_URL) $D.git; \
-		(cd $D.git && $(GIT) archive --format=tar --prefix=$D/ \
-		$(IPXE_GIT_TAG) | gzip >../_$T); \
-		rm -rf $D.git; \
-	fi
+#	if ! $(FETCHER) _$T $(IPXE_TARBALL_URL); then \
+#		$(GIT) clone $(IPXE_GIT_URL) $D.git; \
+#		(cd $D.git && $(GIT) archive --format=tar --prefix=$D/ \
+#		$(IPXE_GIT_TAG) | gzip >../_$T); \
+#		rm -rf $D.git; \
+#	fi
 	mv _$T $T
 
 $D/src/arch/i386/Makefile: $T Config
