diff -Nur xen-4.1.2.orig/tools/firmware/etherboot/Makefile xen-4.1.2/tools/firmware/etherboot/Makefile
--- xen-4.1.2.orig/tools/firmware/etherboot/Makefile	2011-10-20 12:05:41.000000000 -0500
+++ xen-4.1.2/tools/firmware/etherboot/Makefile	2011-12-10 00:16:40.217911034 -0600
@@ -35,12 +35,6 @@
 	mv -f $@.new $@
 
 $T:
-	if ! wget -O _$T $(IPXE_TARBALL_URL); then \
-		$(GIT) clone $(IPXE_GIT_URL) $D.git; \
-		(cd $D.git && $(GIT) archive --format=tar --prefix=$D/ \
-		$(IPXE_GIT_TAG) | gzip >../_$T); \
-		rm -rf $D.git; \
-	fi
 	mv _$T $T
 
 $D/src/arch/i386/Makefile: $T Config
