diff -ur portmidi-orig/pm_common/CMakeLists.txt portmidi/pm_common/CMakeLists.txt
--- portmidi-orig/pm_common/CMakeLists.txt	2009-10-19 19:41:21.000000000 +0100
+++ portmidi/pm_common/CMakeLists.txt	2009-12-06 22:52:04.000000000 +0000
@@ -69,7 +69,7 @@
         CACHE STRING "where to find Java SDK include directory")
     set(JAVA_INCLUDE_PATHS ${JAVA_INCLUDE_PATH} ${JAVA_INCLUDE_PATH}/linux)
     # libjvm.so is found relative to JAVA_INCLUDE_PATH:
-    set(JAVAVM_LIB ${JAVA_INCLUDE_PATH}/../jre/lib/i386/client/libjvm.so)
+    set(JAVAVM_LIB @LIBDIR@/libjvm.so)
 
     set(LINUXSRC pmlinuxalsa pmlinux finddefault)
     prepend_path(LIBSRC ../pm_linux/ ${LINUXSRC})
