diff -U 3 -H -d -r -N -- portmidi-old/pm_common/CMakeLists.txt portmidi/pm_common/CMakeLists.txt
--- portmidi-old/pm_common/CMakeLists.txt	2009-11-20 07:18:06.000000000 +0100
+++ portmidi/pm_common/CMakeLists.txt	2010-10-17 20:02:53.278000042 +0200
@@ -125,9 +125,9 @@
 # install the libraries (Linux and Mac OS X command line)
 if(UNIX)
   INSTALL(TARGETS portmidi-static pmjni
-    LIBRARY DESTINATION /usr/local/lib
-    ARCHIVE DESTINATION /usr/local/lib)
+    LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
 # .h files installed by pm_dylib/CMakeLists.txt, so don't need them here
 #  INSTALL(FILES portmidi.h ../porttime/porttime.h
-#    DESTINATION /usr/local/include)
+#    DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
 endif(UNIX)
diff -U 3 -H -d -r -N -- portmidi-old/pm_dylib/CMakeLists.txt portmidi/pm_dylib/CMakeLists.txt
--- portmidi-old/pm_dylib/CMakeLists.txt	2009-11-20 06:41:09.000000000 +0100
+++ portmidi/pm_dylib/CMakeLists.txt	2010-10-17 20:02:53.279000042 +0200
@@ -59,7 +59,7 @@
 
     set(JAVAVM_LIB "${FRAMEWORK_PATH}/JavaVM.framework")
     set(JAVA_INCLUDE_PATHS ${JAVAVM_LIB}/Headers)
-    set(INSTALL_NAME_DIR "/usr/local/lib")
+    set(INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
     message(STATUS "SYSROOT: " ${CMAKE_OSX_SYSROOT})
   else(APPLE)
     # LINUX settings...
@@ -120,8 +120,8 @@
 # install the libraries (Linux and Mac OS X command line)
 if(UNIX)
   INSTALL(TARGETS portmidi-dynamic
-    LIBRARY DESTINATION /usr/local/lib
-    ARCHIVE DESTINATION /usr/local/lib)
+    LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
   INSTALL(FILES ../pm_common/portmidi.h ../porttime/porttime.h
-    DESTINATION /usr/local/include)
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
 endif(UNIX)
diff -U 3 -H -d -r -N -- portmidi-old/pm_java/CMakeLists.txt portmidi/pm_java/CMakeLists.txt
--- portmidi-old/pm_java/CMakeLists.txt	2009-11-04 16:20:44.000000000 +0100
+++ portmidi/pm_java/CMakeLists.txt	2010-10-17 20:10:46.411000043 +0200
@@ -39,9 +39,9 @@
 
     # install the libraries (Linux only)
     INSTALL(FILES ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/pmdefaults.jar
-      DESTINATION /usr/share/java)
+      DESTINATION ${CMAKE_INSTALL_PREFIX}/share/java)
     INSTALL(PROGRAMS ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/pmdefaults
-      DESTINATION /usr/local/bin)
+      DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
   endif(APPLE)
 endif(UNIX)
 # In windows, use pm_java/make.bat
