diff -Naur portmidi-orig/pm_common/CMakeLists.txt portmidi/pm_common/CMakeLists.txt
--- portmidi-orig/pm_common/CMakeLists.txt	2009-11-20 06:18:06.000000000 +0000
+++ portmidi/pm_common/CMakeLists.txt	2010-02-27 00:43:32.000000000 +0000
@@ -125,9 +125,9 @@
 # install the libraries (Linux and Mac OS X command line)
 if(UNIX)
   INSTALL(TARGETS portmidi-static pmjni
-    LIBRARY DESTINATION /usr/local/lib
-    ARCHIVE DESTINATION /usr/local/lib)
+    LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
 # .h files installed by pm_dylib/CMakeLists.txt, so don't need them here
 #  INSTALL(FILES portmidi.h ../porttime/porttime.h
-#    DESTINATION /usr/local/include)
+#    DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
 endif(UNIX)
diff -Naur portmidi-orig/pm_dylib/CMakeLists.txt portmidi/pm_dylib/CMakeLists.txt
--- portmidi-orig/pm_dylib/CMakeLists.txt	2009-11-20 05:41:09.000000000 +0000
+++ portmidi/pm_dylib/CMakeLists.txt	2010-02-27 00:43:32.000000000 +0000
@@ -59,7 +59,7 @@
 
     set(JAVAVM_LIB "${FRAMEWORK_PATH}/JavaVM.framework")
     set(JAVA_INCLUDE_PATHS ${JAVAVM_LIB}/Headers)
-    set(INSTALL_NAME_DIR "/usr/local/lib")
+    set(INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
     message(STATUS "SYSROOT: " ${CMAKE_OSX_SYSROOT})
   else(APPLE)
     # LINUX settings...
@@ -120,8 +120,8 @@
 # install the libraries (Linux and Mac OS X command line)
 if(UNIX)
   INSTALL(TARGETS portmidi-dynamic
-    LIBRARY DESTINATION /usr/local/lib
-    ARCHIVE DESTINATION /usr/local/lib)
+    LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
   INSTALL(FILES ../pm_common/portmidi.h ../porttime/porttime.h
-    DESTINATION /usr/local/include)
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
 endif(UNIX)
diff -Naur portmidi-orig/pm_java/CMakeLists.txt portmidi/pm_java/CMakeLists.txt
--- portmidi-orig/pm_java/CMakeLists.txt	2009-11-04 15:20:44.000000000 +0000
+++ portmidi/pm_java/CMakeLists.txt	2010-02-27 00:43:32.000000000 +0000
@@ -41,7 +41,7 @@
     INSTALL(FILES ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/pmdefaults.jar
       DESTINATION /usr/share/java)
     INSTALL(PROGRAMS ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/pmdefaults
-      DESTINATION /usr/local/bin)
+      DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
   endif(APPLE)
 endif(UNIX)
 # In windows, use pm_java/make.bat
diff -Naur portmidi-orig/pm_linux/Makefile portmidi/pm_linux/Makefile
--- portmidi-orig/pm_linux/Makefile	2009-01-20 04:01:54.000000000 +0000
+++ portmidi/pm_linux/Makefile	2010-02-27 00:42:47.000000000 +0000
@@ -147,10 +147,10 @@
 	cd pm_java; chmod +x pmdefaults/pmdefaults
 
 install:
-	cp pm_java/pmjni/libpmjni.so /usr/local/lib
+	cp pm_java/pmjni/libpmjni.so ${CMAKE_INSTALL_PREFIX}/lib
 	cp pm_java/pmdefaults.jar /usr/share/java
 	# do not chmod +x here since we are probably root
-	cp pm_java/pmdefaults/pmdefaults /usr/local/bin
+	cp pm_java/pmdefaults/pmdefaults ${CMAKE_INSTALL_PREFIX}/bin
 
 clean:
 	rm -f *.o *~ core* */*.o */*.so */*~ */core* pm_test/*/pm_dll.dll 
