diff -Naur efiboots-1.0.orig/efiboots.py efiboots-1.0/efiboots.py
--- efiboots-1.0.orig/efiboots.py	2022-04-12 01:36:59.888996574 -0400
+++ efiboots-1.0/efiboots.py	2022-04-12 01:37:32.991997960 -0400
@@ -272,7 +272,7 @@
 			boot = run_efibootmgr()
 		except subprocess.CalledProcessError as e:
 			logging.exception("Error running efibootmgr. Please check that it is correctly installed.")
-			error_dialog(parent=self.window, title="efibootmgr utility not installed!", message="Please check that the efibootmgr utility is correctly installed, as this program requires its output.\n" + e)
+			error_dialog(parent=self.window, title="efibootmgr utility not installed!", message="Please check that the efibootmgr utility is correctly installed, as this program requires its output.\n" + e.output)
 			sys.exit(-1)
 		except UnicodeDecodeError as e:
 			logging.exception("Error decoding efibootmgr -v output.")
