diff -U 3 -dHrN -- nethack-3.4.3-x11/include/config.h nethack-3.4.3-qt/include/config.h
--- nethack-3.4.3-x11/include/config.h	2007-01-02 23:10:07.000000000 -0600
+++ nethack-3.4.3-qt/include/config.h	2007-01-03 11:58:11.000000000 -0600
@@ -44,7 +44,7 @@
  */
 #define TTY_GRAPHICS	/* good old tty based graphics */
 #define X11_GRAPHICS 	/* X11 interface */
-/* #define QT_GRAPHICS */	/* Qt interface */
+#define QT_GRAPHICS 	/* Qt interface */
 /* #define GNOME_GRAPHICS */	/* Gnome interface */
 /* #define MSWIN_GRAPHICS */	/* Windows NT, CE, Graphics */
 
diff -U 3 -dHrN -- nethack-3.4.3-x11/sys/unix/Makefile.src nethack-3.4.3-qt/sys/unix/Makefile.src
--- nethack-3.4.3-x11/sys/unix/Makefile.src	2007-01-02 23:20:01.000000000 -0600
+++ nethack-3.4.3-qt/sys/unix/Makefile.src	2007-01-03 12:12:55.000000000 -0600
@@ -186,7 +186,8 @@
 # Files for a Qt port
 #
 WINQTSRC = ../win/Qt/qt_win.cpp ../win/Qt/qt_clust.cpp ../win/Qt/qttableview.cpp
-WINQTOBJ = qt_win.o qt_clust.o qttableview.o tile.o
+# WINQTOBJ = qt_win.o qt_clust.o qttableview.o tile.o
+WINQTOBJ = qt_win.o qt_clust.o qttableview.o
 #
 # Files for a Gnome port
 #
@@ -214,8 +215,8 @@
 
 #
 #
-WINSRC = $(WINTTYSRC) $(WINX11SRC)
-WINOBJ = $(WINTTYOBJ) $(WINX11OBJ)
+WINSRC = $(WINTTYSRC) $(WINX11SRC) $(WINQTSRC)
+WINOBJ = $(WINTTYOBJ) $(WINX11OBJ) $(WINQTOBJ)
 
 # on some systems the termcap library is in -ltermcap or -lcurses
 # on 386 Xenix, the -ltermlib tputs() seems not to work; use -lcurses instead
@@ -241,7 +242,7 @@
 # WINX11LIB = -lXaw -lXmu -lXpm -lXext -lXt -lX11 -lSM -lICE -lm # BSD/OS 2.0
 #
 # libraries for Qt
-WINQTLIB = -L$(QTDIR)/lib -lqt
+WINQTLIB = -L$(QTDIR)/lib -lqt-mt
 #
 # libraries for KDE (with Qt)
 WINKDELIB = -lkdecore -lkdeui -lXext
@@ -255,7 +256,7 @@
 # libraries for BeOS 
 WINBELIB = -lbe
 
-WINLIB = $(WINTTYLIB) $(WINX11LIB)
+WINLIB = $(WINTTYLIB) $(WINX11LIB) $(WINQTLIB)
 
 # any other strange libraries your system needs (for Sysunix only -- the more
 # specialized targets should already be right)
diff -U 3 -dHrN -- nethack-3.4.3-x11/sys/unix/Makefile.top nethack-3.4.3-qt/sys/unix/Makefile.top
--- nethack-3.4.3-x11/sys/unix/Makefile.top	2007-01-02 23:12:32.000000000 -0600
+++ nethack-3.4.3-qt/sys/unix/Makefile.top	2007-01-03 12:01:32.000000000 -0600
@@ -42,7 +42,7 @@
 # per discussion in Install.X11 and Install.Qt
 # VARDATND = 
 # VARDATND = x11tiles NetHack.ad pet_mark.xbm
-VARDATND = x11tiles NetHack.ad pet_mark.xbm rip.xpm
+VARDATND = x11tiles NetHack.ad pet_mark.xbm rip.xpm nhsplash.xpm
 # for Atari/Gem
 # VARDATND = nh16.img title.img GEM_RSC.RSC rip.img
 # for BeOS
