diff -U 3 -dHrN -- nethack-3.4.3/include/config.h nethack-3.4.3-enviro/include/config.h
--- nethack-3.4.3/include/config.h	2003-12-07 17:39:13.000000000 -0600
+++ nethack-3.4.3-enviro/include/config.h	2007-01-02 22:38:51.000000000 -0600
@@ -169,11 +169,11 @@
 
 #ifdef UNIX
 /* path and file name extension for compression program */
-#define COMPRESS "/usr/bin/compress"	/* Lempel-Ziv compression */
-#define COMPRESS_EXTENSION ".Z"		/* compress's extension */
+/* #define COMPRESS "/usr/bin/compress"	*/ /* Lempel-Ziv compression */
+/* #define COMPRESS_EXTENSION ".Z"		*/ /* compress's extension */
 /* An example of one alternative you might want to use: */
-/* #define COMPRESS "/usr/local/bin/gzip" */	/* FSF gzip compression */
-/* #define COMPRESS_EXTENSION ".gz" */		/* normal gzip extension */
+#define COMPRESS "/usr/bin/gzip" 	/* FSF gzip compression */
+#define COMPRESS_EXTENSION ".gz" 		/* normal gzip extension */
 #endif
 
 #ifndef COMPRESS
@@ -185,7 +185,7 @@
  *	a tar-like file, thus making a neater installation.  See *conf.h
  *	for detailed configuration.
  */
-/* #define DLB */	/* not supported on all platforms */
+#define DLB 	/* not supported on all platforms */
 
 /*
  *	Defining INSURANCE slows down level changes, but allows games that
@@ -204,7 +204,7 @@
  * otherwise it will be the current directory.
  */
 # ifndef HACKDIR
-#  define HACKDIR "/usr/games/lib/nethackdir"
+#  define HACKDIR "/usr/share/games/nethack"
 # endif
 
 /*
@@ -300,7 +300,7 @@
  * functions that have been macroized.
  */
 
-/* #define VISION_TABLES */ /* use vision tables generated at compile time */
+#define VISION_TABLES  /* use vision tables generated at compile time */
 #ifndef VISION_TABLES
 # ifndef NO_MACRO_CPATH
 #  define MACRO_CPATH	/* use clear_path macros instead of functions */
@@ -348,8 +348,8 @@
  * bugs left here.
  */
 
-/*#define GOLDOBJ */	/* Gold is kept on obj chains - Helge Hafting */
-/*#define AUTOPICKUP_EXCEPTIONS */ /* exceptions to autopickup */
+#define GOLDOBJ 	/* Gold is kept on obj chains - Helge Hafting */
+#define AUTOPICKUP_EXCEPTIONS  /* exceptions to autopickup */
 
 /* End of Section 5 */
 
diff -U 3 -dHrN -- nethack-3.4.3/include/unixconf.h nethack-3.4.3-enviro/include/unixconf.h
--- nethack-3.4.3/include/unixconf.h	2003-12-07 17:39:13.000000000 -0600
+++ nethack-3.4.3-enviro/include/unixconf.h	2007-01-02 22:39:38.000000000 -0600
@@ -32,12 +32,12 @@
 
 
 /* define any of the following that are appropriate */
-#define SVR4		/* use in addition to SYSV for System V Release 4 */
+/*n#define SVR4	*/	/* use in addition to SYSV for System V Release 4 */
 			/* including Solaris 2+ */
 #define NETWORK		/* if running on a networked system */
 			/* e.g. Suns sharing a playground through NFS */
 /* #define SUNOS4 */	/* SunOS 4.x */
-/* #define LINUX */	/* Another Unix clone */
+#define LINUX 	/* Another Unix clone */
 /* #define CYGWIN32 */	/* Unix on Win32 -- use with case sensitive defines */
 /* #define GENIX */	/* Yet Another Unix Clone */
 /* #define HISX */	/* Bull Unix for XPS Machines */
@@ -102,7 +102,7 @@
  * If you want the static parts of your playground on a read-only file
  * system, define VAR_PLAYGROUND to be where the variable parts are kept.
  */
-/* #define VAR_PLAYGROUND "/var/lib/games/nethack" */
+#define VAR_PLAYGROUND "/var/games/nethack" 
 
 
 /*
diff -U 3 -dHrN -- nethack-3.4.3/sys/unix/Makefile.src nethack-3.4.3-enviro/sys/unix/Makefile.src
--- nethack-3.4.3/sys/unix/Makefile.src	2003-12-07 17:39:13.000000000 -0600
+++ nethack-3.4.3-enviro/sys/unix/Makefile.src	2007-01-02 22:40:09.000000000 -0600
@@ -230,8 +230,8 @@
 # WINTTYLIB = -ltermcap
 # WINTTYLIB = -lcurses
 # WINTTYLIB = -lcurses16
-# WINTTYLIB = -lncurses
-WINTTYLIB = -ltermlib
+WINTTYLIB = -lncurses
+# WINTTYLIB = -ltermlib
 #
 # libraries for X11
 # If USE_XPM is defined in config.h, you will also need -lXpm here.
diff -U 3 -dHrN -- nethack-3.4.3/sys/unix/Makefile.top nethack-3.4.3-enviro/sys/unix/Makefile.top
--- nethack-3.4.3/sys/unix/Makefile.top	2003-12-07 17:39:13.000000000 -0600
+++ nethack-3.4.3-enviro/sys/unix/Makefile.top	2007-01-02 22:36:23.000000000 -0600
@@ -14,11 +14,11 @@
 # MAKE = make
 
 # make NetHack
-PREFIX	 = /usr
+PREFIX	 = $(DESTDIR)/usr
 GAME     = nethack
 # GAME     = nethack.prg
-GAMEUID  = games
-GAMEGRP  = bin
+GAMEUID  = root
+GAMEGRP  = root
 
 # Permissions - some places use setgid instead of setuid, for instance
 # See also the option "SECURE" in include/config.h
@@ -35,8 +35,8 @@
 # therefore there should not be anything in GAMEDIR that you want to keep
 # (if there is, you'll have to do the installation by hand or modify the
 # instructions)
-GAMEDIR  = $(PREFIX)/games/lib/$(GAME)dir
-VARDIR  = $(GAMEDIR)
+GAMEDIR  = $(PREFIX)/share/games/nethack
+VARDIR  = $(DESTDIR)/var/games/nethack
 SHELLDIR = $(PREFIX)/games
 
 # per discussion in Install.X11 and Install.Qt
@@ -188,6 +188,7 @@
 	-rm -f $(SHELLDIR)/$(GAME)
 	sed -e 's;/usr/games/lib/nethackdir;$(GAMEDIR);' \
 		-e 's;HACKDIR/nethack;HACKDIR/$(GAME);' \
+		-e 's;$(DESTDIR);;' \
 		< sys/unix/nethack.sh \
 		> $(SHELLDIR)/$(GAME)
 # set up their permissions
diff -U 3 -dHrN -- nethack-3.4.3/sys/unix/Makefile.utl nethack-3.4.3-enviro/sys/unix/Makefile.utl
--- nethack-3.4.3/sys/unix/Makefile.utl	2003-12-07 17:39:13.000000000 -0600
+++ nethack-3.4.3-enviro/sys/unix/Makefile.utl	2007-01-02 22:36:46.000000000 -0600
@@ -101,11 +101,11 @@
 
 # yacc/lex programs to use to generate *_comp.h, *_lex.c, and *_yacc.c.
 # if, instead of yacc/lex you have bison/flex, comment/uncomment the following.
-YACC     = yacc
-LEX      = lex
-# YACC     = bison -y
-# YACC     = byacc
-# LEX      = flex
+# YACC     = yacc
+# LEX      = lex
+YACC     = bison -y
+YACC     = byacc
+LEX      = flex
  
 # these are the names of the output files from YACC/LEX. Under MS-DOS
 # and similar systems, they may differ
