diff -Nur mpqc-2.3.1.orig/configure mpqc-2.3.1/configure
--- mpqc-2.3.1.orig/configure	2006-03-22 13:13:52.000000000 -0600
+++ mpqc-2.3.1/configure	2009-09-27 20:57:25.129509290 -0500
@@ -2354,14 +2354,14 @@
 if test $exec_prefix = "NONE"; then
   if test $prefix = "NONE"; then
     scbindir=$ac_default_prefix/bin;
-    sclibdir=$ac_default_prefix/lib;
+    sclibdir=$ac_default_prefix/lib64;
   else
     scbindir=$prefix/bin;
-    sclibdir=$prefix/lib;
+    sclibdir=$prefix/lib64;
   fi
 else
   scbindir=$exec_prefix/bin
-  sclibdir=$exec_prefix/lib
+  sclibdir=$exec_prefix/lib64
 fi
 
 
