diff -up tagainijisho-1.0.3/src/sqlite/sqlite3ext.cc.fts3_tokenizer tagainijisho-1.0.3/src/sqlite/sqlite3ext.cc
--- tagainijisho-1.0.3/src/sqlite/sqlite3ext.cc.fts3_tokenizer	2018-07-22 14:22:31.597931031 +0200
+++ tagainijisho-1.0.3/src/sqlite/sqlite3ext.cc	2018-07-22 14:22:35.858945041 +0200
@@ -118,6 +118,10 @@ int register_tokenizer(sqlite3 *db, cons
 	sqlite3_stmt *pStmt;
 	const char *zSql = "SELECT fts3_tokenizer(?, ?)";
 
+#ifdef SQLITE_DBCONFIG_ENABLE_FTS3_TOKENIZER
+	sqlite3_db_config(db, SQLITE_DBCONFIG_ENABLE_FTS3_TOKENIZER, 1, NULL);
+#endif
+
 	rc = sqlite3_prepare_v2(db, zSql, -1, &pStmt, 0);
 	if( rc!=SQLITE_OK ){
 		return rc;
