diff -Naur j4-dmenu-desktop-master/src/Applications.hh j4-dmenu-desktop-master.patched/src/Applications.hh
--- j4-dmenu-desktop-master/src/Applications.hh	2014-05-27 07:02:41.000000000 -0400
+++ j4-dmenu-desktop-master.patched/src/Applications.hh	2014-07-24 15:23:39.000000000 -0400
@@ -44,10 +44,11 @@
 		if((shell = getenv("SHELL")) == 0)
 		    shell = "/bin/sh";
 		
-		fprintf(stderr, "%s -i -c '%s'\n", shell, choice.c_str());
+		fprintf(stderr, "%s -c '%s'\n", shell, choice.c_str());
 		
 		// -i -c was tested with both bash and zsh.
-		exit(execl(shell, shell, "-i", "-c",  choice.c_str(), 0));
+		// 20140724 bkw: remove -i, it causes problems.
+		exit(execl(shell, shell, "-c",  choice.c_str(), 0));
 	    }
 
             // +1 b/c there must be whitespace we add back later...
diff -Naur j4-dmenu-desktop-master/src/Main.hh j4-dmenu-desktop-master.patched/src/Main.hh
--- j4-dmenu-desktop-master/src/Main.hh	2014-05-27 07:02:41.000000000 -0400
+++ j4-dmenu-desktop-master.patched/src/Main.hh	2014-07-24 15:28:38.000000000 -0400
@@ -47,9 +47,10 @@
             if((shell = getenv("SHELL")) == 0)
                 shell = "/bin/sh";
 
-	    fprintf(stderr, "%s -i -c '%s'\n", shell, command.c_str());
+	    fprintf(stderr, "%s -c '%s'\n", shell, command.c_str());
 
-            return execl(shell, shell, "-i", "-c", command.c_str(), 0);
+		 // 20140724 bkw: get rid of -i, it causes problems.
+            return execl(shell, shell, "-c", command.c_str(), 0);
         }
         return 0;
     }
