Description: Fix FTBFS with gcc14
Origin: vendor
Forwarded: not-needed
Bug-Debian: http://bugs.debian.org/1075604
Author: Kartik Mistry <kartik@debian.org>
--- a/enttbl.c
+++ b/enttbl.c
@@ -21,6 +21,7 @@
 #include "config.h"
 #include <stdlib.h>
 #include "unicode.h"
+#include <string.h>
 
 struct ent {
   char *name;
--- a/uni2ascii.c
+++ b/uni2ascii.c
@@ -2689,7 +2689,7 @@ int SubsAvailable = 0;
  * search may be advisable. For the time being we just do a linear search.
  */
 
-SubstituteChar(UTF32 c) {
+int SubstituteChar(UTF32 c) {
   int i;
   for(i = 0; i < SubCnt; i++) {
     if(c == SubList[i].u) {
@@ -2703,7 +2703,7 @@ SubstituteChar(UTF32 c) {
   return 0;
 }
 
-AddCustomSubstitution(char *str){
+int AddCustomSubstitution(char *str){
   char *Left;
   char *Right;
   char *Delim;
@@ -3186,6 +3186,7 @@ int main (int ac, char *av[])
       }
       break;
     default:
+      extern void putu8 (unsigned long);
       if(PassThroughP) putu8(c);
       else if(!PureP && (c <= 0x7F)) putchar((int)c);
       else {
