From 60be4725ec2ee8a9557a02f7eb7b6aa748961f48 Mon Sep 17 00:00:00 2001
From: Contarino Rosario <contarino.rosario@gmail.com>
Date: Thu, 28 Apr 2011 23:59:06 +0200
Subject: [PATCH] panel-plugin/main.c: Fix start bug

---
 panel-plugin/main.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/panel-plugin/main.c b/panel-plugin/main.c
index f59886e..8d51ae6 100644
--- a/panel-plugin/main.c
+++ b/panel-plugin/main.c
@@ -836,6 +836,8 @@ static void genmon_construct (XfcePanelPlugin *plugin)
 
     SetMonitorFont (genmon);
 
+    SetTimer (genmon);
+
     g_signal_connect (plugin, "free-data", G_CALLBACK (genmon_free), genmon);
 
     g_signal_connect (plugin, "save", G_CALLBACK (genmon_write_config),
