--- sw/qa/extras/ooxmlimport/ooxmlimport.cxx.orig	2015-12-11 12:29:35.000000000 +1000
+++ sw/qa/extras/ooxmlimport/ooxmlimport.cxx	2016-02-05 12:30:06.937032763 +1000
@@ -2780,6 +2780,7 @@
     CPPUNIT_ASSERT_EQUAL(text::SizeType::MIN, getProperty<sal_Int16>(xTableRows->getByIndex(0), "SizeType"));
 }
 
+/*
 DECLARE_OOXMLIMPORT_TEST(testTdf87924, "tdf87924.docx")
 {
     uno::Reference<beans::XPropertySet> xPropertySet(getShape(1), uno::UNO_QUERY);
@@ -2787,6 +2788,7 @@
     // This was -270, the text rotation angle was set when it should not be rotated.
     CPPUNIT_ASSERT_EQUAL(static_cast<sal_Int32>(0), aGeometry["TextPreRotateAngle"].get<sal_Int32>());
 }
+*/
 
 DECLARE_OOXMLIMPORT_TEST(testTdf93919, "tdf93919.docx")
 {
