--- sal/osl/unx/file.cxx.orig	2024-02-26 01:15:17.000000000 +1000
+++ sal/osl/unx/file.cxx	2024-03-08 21:40:36.345000000 +1000
@@ -64,6 +64,14 @@
 #include <vector>
 #endif
 
+#ifdef LINUX
+#include <sys/vfs.h>
+// As documented by the kernel
+#define SMB_SUPER_MAGIC  static_cast<__fsword_t>(0x517B)
+#define CIFS_SUPER_MAGIC static_cast<__fsword_t>(0xFF534D42)
+#define SMB2_SUPER_MAGIC static_cast<__fsword_t>(0xFE534D42)
+#endif
+
 namespace {
 
 enum class State
