--- framework/source/services/substitutepathvars.cxx.orig	2016-09-22 01:12:13.000000000 +1000
+++ framework/source/services/substitutepathvars.cxx	2016-09-30 10:54:26.208653704 +1000
@@ -1095,7 +1095,8 @@
             if ( nPos >= 0 )
             {
                 bool bMatch = true;
-                if ( i->eVariable == PREDEFVAR_LANGID ||
+                if ( i->eVariable == PREDEFVAR_USERNAME ||
+                     i->eVariable == PREDEFVAR_LANGID ||
                      i->eVariable == PREDEFVAR_VLANG )
                 {
                     // Special path variables as they can occur in the middle of a path. Only match if they
