--- orrs/bin/Makefile	2011-04-03 12:27:46.000000000 -0400
+++ orrs/bin/Makefile.orrs	2016-09-12 20:36:36.000000000 -0400
@@ -1,25 +1,12 @@
 
 SRC = ../src
-PLTOBJ = /home/codes/Xplot/libPlt.a
 
 #========================================
 #  Default compiler flags
-FC = f77
-FLG = -O
+FC = gfortran
+FLG = -O2 -fdefault-real-8
 PLTLIB = -lX11
 FTNLIB =
-
-#========================================
-#  Uncomment for Intel Fortran Compiler
-FC = ifort
-FLG = -O -fpe0 -CB
-PLTLIB = -L/usr/X11R6/lib -lX11
-FTNLIB = 
-
-#  Additional IFC stuff needed on MIT's Athena system
-#FC = ifc
-#FTNLIB = -Vaxlib /usr/lib/C-ctype.o /usr/lib/C_name.o /usr/lib/ctype-info.o
-
 #========================================
 
 
