From e5d7c0b90cfe3218d67b56e122140456206f6a65 Mon Sep 17 00:00:00 2001
From: Hoai-Thu Vuong <thuvh87@gmail.com>
Date: Tue, 12 Jan 2016 17:30:45 +0700
Subject: [PATCH] using echo and tag php in code

---
 webfrontend/www/design.inc.php | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/webfrontend/www/design.inc.php b/webfrontend/www/design.inc.php
index 174e04a..9fcd7b0 100644
--- a/webfrontend/www/design.inc.php
+++ b/webfrontend/www/design.inc.php
@@ -29,12 +29,12 @@ function doHeader($title="", $options=FALSE) {
       <meta name="description" content="tvprobe webfrontend" />
       <meta name="publisher"   content="ComX Networks A/S" />
 
-      <link rel="stylesheet" type="text/css" href="<?=$incdir?>css/motorola.css">
+      <link rel="stylesheet" type="text/css" href="<?php echo $incdir; ?>css/motorola.css">
 <?php
        if ($include_javascript == TRUE) {
 ?>
       <script
-	 src="<?=$incdir?>functions.js" type="text/javascript" language='javascript'>
+	 src="<?php echo $incdir; ?>functions.js" type="text/javascript" language='javascript'>
       </script>
 <?php
        }
@@ -44,10 +44,10 @@ function doHeader($title="", $options=FALSE) {
 ?>
       <!--Epoch's styles-->
       <link rel="stylesheet" type="text/css"
-	    href="<?=$incdir?>js/epoch_v202_en/epoch_styles.css" />
+	    href="<?php echo $incdir; ?>js/epoch_v202_en/epoch_styles.css" />
 
       <!--Epoch's Code-->
-      <script type="text/javascript" src="<?=$incdir?>js/epoch_v202_en/epoch_classes.js">
+      <script type="text/javascript" src="<?php echo $incdir; ?>js/epoch_v202_en/epoch_classes.js">
       </script>
 
       <script type="text/javascript">
