diff -Naur a/data/scripts/Xsession b/data/scripts/Xsession
--- a/data/scripts/Xsession	2017-11-03 11:49:19.374409491 +0000
+++ b/data/scripts/Xsession	2017-11-03 11:50:17.807893273 +0000
@@ -67,7 +67,6 @@
 # by the scripts to work
 xsessionddir="/etc/X11/Xsession.d"
 OPTIONFILE=/etc/X11/Xsession.options
-USERXSESSION=$HOME/.xsession
 USERXSESSIONRC=$HOME/.xsessionrc
 ALTUSERXSESSION=$HOME/.Xsession
 
@@ -90,10 +89,6 @@
 fi
 [ -f $HOME/.Xresources ] && xrdb -merge $HOME/.Xresources
 
-if [ -f "$USERXSESSION" ]; then
-  . "$USERXSESSION"
-fi
-
 if [ -z "$@" ]; then
     exec xmessage -center -buttons OK:0 -default OK "Sorry, $DESKTOP_SESSION is no valid session."
 else
