diff -Naur a/data/scripts/Xsession b/data/scripts/Xsession
--- a/data/scripts/Xsession	2018-07-18 11:31:40.000000000 +0100
+++ b/data/scripts/Xsession	2018-08-07 20:06:16.937412206 +0100
@@ -68,6 +68,7 @@
 # by the scripts to work
 xsessionddir="/etc/X11/Xsession.d"
 OPTIONFILE=/etc/X11/Xsession.options
+# This seems to break with some WMs
 USERXSESSION=$HOME/.xsession
 USERXSESSIONRC=$HOME/.xsessionrc
 ALTUSERXSESSION=$HOME/.Xsession
@@ -91,9 +92,10 @@
 fi
 [ -f $HOME/.Xresources ] && xrdb -merge $HOME/.Xresources
 
-if [ -f "$USERXSESSION" ]; then
-  . "$USERXSESSION"
-fi
+# This seems to break with some WMs
+#if [ -f "$USERXSESSION" ]; then
+#  . "$USERXSESSION"
+#fi
 
 if [ -z "$*" ]; then
     exec xmessage -center -buttons OK:0 -default OK "Sorry, $DESKTOP_SESSION is no valid session."
