diff -rupN ./CMakeLists.txt ./CMakeLists.txt
--- ./CMakeLists.txt	2015-09-26 02:34:29.547077049 -0700
+++ ./CMakeLists.txt	2015-09-25 22:58:00.000000000 -0700
@@ -65,4 +65,4 @@ add_executable(ggmud WIN32 alias.c ansi.
                            ${ADDITIONAL_SOURCES}
               )
 
-target_link_libraries(ggmud tt lua sq3 httpfetch zlib ${GTK2_LIBRARIES} ${ADDITIONAL_LIBS} ${GTHREAD_LIBRARIES})
+target_link_libraries(ggmud tt lua sq3 httpfetch zlib dl m ${GTK2_LIBRARIES} ${ADDITIONAL_LIBS} ${GTHREAD_LIBRARIES})
