--- include/repair/Makefile.am~	2009-01-08 15:34:27.000000000 -0600
+++ include/repair/Makefile.am	2011-02-23 22:23:25.654384304 -0600
@@ -3,4 +3,4 @@
 			  master.h format.h journal.h node.h place.h item.h \
 			  filter.h disk_scan.h twig_scan.h add_missing.h \
 			  semantic.h lost_found.h cleanup.h tree.h alloc.h \
-			  object.h status.h backup.h oid.h pset.h
+			  status.h backup.h oid.h pset.h
