diff -ur gmrun-0.9.2/src/ci_string.h gmrun-0.9.2.new/src/ci_string.h
--- gmrun-0.9.2/src/ci_string.h	2001-05-16 17:39:31.000000000 +0300
+++ gmrun-0.9.2.new/src/ci_string.h	2008-01-15 09:10:39.000000000 +0200
@@ -7,6 +7,7 @@
 #define __CI_STRING_H__
 
 #include <string>
+#include <string.h>
 #include <ctype.h>
 
 struct ci_char_traits : public std::char_traits<char>
diff -ur gmrun-0.9.2/src/gtkcompletionline.cc gmrun-0.9.2.new/src/gtkcompletionline.cc
--- gmrun-0.9.2/src/gtkcompletionline.cc	2003-11-16 12:55:07.000000000 +0200
+++ gmrun-0.9.2.new/src/gtkcompletionline.cc	2008-01-15 09:10:39.000000000 +0200
@@ -24,6 +24,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
+#include <string.h>
 
 #include <iostream>
 #include <set>
diff -ur gmrun-0.9.2/src/main.cc gmrun-0.9.2.new/src/main.cc
--- gmrun-0.9.2/src/main.cc	2003-11-16 12:55:07.000000000 +0200
+++ gmrun-0.9.2.new/src/main.cc	2008-01-15 09:10:39.000000000 +0200
@@ -20,6 +20,7 @@
 #include <algorithm>
 #include <iterator>
 #include <popt.h>
+#include <string.h>
 
 using namespace std;
 
diff -ur gmrun-0.9.2/src/prefs.cc gmrun-0.9.2.new/src/prefs.cc
--- gmrun-0.9.2/src/prefs.cc	2002-08-16 13:48:22.000000000 +0300
+++ gmrun-0.9.2.new/src/prefs.cc	2008-01-16 19:48:39.000000000 +0200
@@ -13,6 +13,7 @@
 #include <fstream>
 #include <iostream>
 #include <stdio.h>
+#include <stdlib.h>
 
 #include <list>
 
