--- gmrun-0.9.2/config/gmrunrc.old	2003-11-16 12:43:41.000000000 +0200
+++ gmrun-0.9.2/config/gmrunrc	2007-01-18 16:05:47.000000000 +0200
@@ -3,7 +3,7 @@
 # GPL v2.0 applies
 
 # Set terminal
-Terminal = gnome-terminal --start-factory-server --use-factory
+Terminal = xterm
 TermExec = ${Terminal} -e
 AlwaysInTerm = ssh telnet ftp lynx mc vi vim pine centericq perldoc man
 
@@ -32,18 +32,14 @@
 #   - %u gets replaced with the whole URL ("http://www.google.com")
 #   - %s gets replaced with "//www.google.com".  This is useful for URL-s
 #     like "man:printf" --> %s will get replaced with "printf"
-URL_http = mozilla -remote "openURL(%u, new-window)"
-URL_mailto = mozilla -remote "mailto(%s)"
+URL_http = xdg-open %u
+URL_mailto = xdg-email %u
+URL_file = xdg-open %s
 URL_man = ${TermExec} 'man %s'
 URL_info = ${TermExec} 'info %s'
-URL_pd = ${TermExec} 'perldoc %s'
-URL_file = nautilus %s
-URL_readme = ${TermExec} 'less /usr/doc/%s/README'
-URL_info = ${TermExec} 'info %s'
-URL_sh = sh -c '%s'
+URL_search = xdg-open 'http://www.google.com/search?q=%s'
 
 # extension handlers
-EXT:doc,rtf = AbiWord %s
-EXT:txt,cc,cpp,h,java,html,htm,epl,tex,latex,js,css,xml,xsl,am = emacs %s
-EXT:ps = gv %s
-EXT:pdf = xpdf %s
+# Customize your own extension handler.
+EXT:doc,rtf,txt,cc,cpp,h,java,html,htm,epl,tex,latex,js,css,xml,xsl,am,ps,pdf = xdg-open %s
+
