--- gmrun-0.9.2/src/gtkcompletionline.cc.orig	2009-06-19 22:38:14.000000000 +0400
+++ gmrun-0.9.2/src/gtkcompletionline.cc	2009-06-19 22:37:14.000000000 +0400
@@ -376,7 +377,7 @@
   return 0;
 }
 
-int my_alphasort(const void* va, const void* vb) {
+int my_alphasort(const dirent** va, const dirent** vb) {
   const struct dirent** a = (const struct dirent**)va;
   const struct dirent** b = (const struct dirent**)vb;
 
