--- Makefile.orig	2009-05-06 21:10:37.578993290 -0500
+++ Makefile	2009-05-06 21:11:37.266485486 -0500
@@ -38,7 +38,7 @@
 	docbook2man offlineimap.sgml
 	docbook2html -u offlineimap.sgml
 	mv offlineimap.html manual.html
-	man -t -l offlineimap.1 > manual.ps
+	man -t ./offlineimap.1 > manual.ps
 	ps2pdf manual.ps
 	groff -Tascii -man offlineimap.1 | sed $$'s/.\b//g' > manual.txt
 	-rm manpage.links manpage.refs manual.ps
