--- 2ManDVD/mainfrm.cpp.orig	2010-07-04 15:28:28.000000030 +0200
+++ 2ManDVD/mainfrm.cpp	2010-07-04 15:29:26.670000516 +0200
@@ -1045,8 +1045,8 @@
 	outext << "growisofs" << endl;
 	outext << "mpeg2enc" << endl;
         outext << "jpegtopnm /dev/null" << endl;
-        outext << "cat " + qApp->applicationDirPath() + "/fake.pl" << endl;
-        outext << "rm -f " + homeDir.path() + "/.mdvdep.sh" << endl;
+        outext << "cat " << qApp->applicationDirPath() << "/fake.pl" << endl;
+        outext << "rm -f " << homeDir.path() << "/.mdvdep.sh" << endl;
 	outext << "exit" << endl;
 	wmyficext.close();
 	QString programf = "bash";
