--- a/vtwm-20120316/add_window.c
+++ b/vtwm-20120316/add_window.c
@@ -37,6 +37,7 @@
  *
  */
 #include <stdio.h>
+#include <time.h>
 #include <string.h>
 #include "twm.h"
 #include <X11/Xatom.h>
@@ -408,6 +409,6 @@
       !(tmp_win->hints.flags & USPosition) && (tmp_win->group != tmp_win->w))
   {
-    static lastwingroup = 0;
-    static lastwin = 0;
-    static lasttime = 0;
-    int curtime = time(NULL);
+    static Window lastwingroup = 0;
+    static Window lastwin = 0;
+    static time_t lasttime = 0;
+    time_t curtime = time(NULL);
     Window junk;
--- a/vtwm-20120316/menus.c
+++ b/vtwm-20120316/menus.c
@@ -45,6 +45,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <signal.h>
+#include <sys/wait.h>
 #include <errno.h>
 #include <unistd.h>
