--- CMakeLists.txt.orig	2017-01-14 07:55:01.000000000 +1000
+++ CMakeLists.txt	2017-01-23 11:18:46.243967086 +1000
@@ -92,10 +92,10 @@
 	set (UTPP_INSTALL_DESTINATION "include/UnitTestPP")
 endif()
 
-set(config_install_dir_ lib/cmake/${PROJECT_NAME})
+set(config_install_dir_ lib${LIB_SUFFIX}/cmake/${PROJECT_NAME})
 set(targets_export_name_ "${PROJECT_NAME}Targets")
 
-install(TARGETS UnitTest++ EXPORT "${targets_export_name_}" DESTINATION lib)
+install(TARGETS UnitTest++ EXPORT "${targets_export_name_}" DESTINATION lib${LIB_SUFFIX})
 install(FILES ${headers_} DESTINATION ${UTPP_INSTALL_DESTINATION})
 install(FILES ${platformHeaders_} DESTINATION ${UTPP_INSTALL_DESTINATION}/${platformDir_})
 install(FILES cmake/UnitTest++Config.cmake DESTINATION "${config_install_dir_}")
@@ -103,8 +103,8 @@
 
 set(prefix      ${CMAKE_INSTALL_PREFIX})
 set(exec_prefix ${CMAKE_INSTALL_PREFIX}/bin)
-set(libdir      ${CMAKE_INSTALL_PREFIX}/lib)
+set(libdir      ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
 set(includedir  ${CMAKE_INSTALL_PREFIX}/include/UnitTest++)
 configure_file("UnitTest++.pc.in" "UnitTest++.pc" @ONLY)
 install(FILES       "${CMAKE_CURRENT_BINARY_DIR}/UnitTest++.pc"
-        DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
+        DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/pkgconfig")
