diff -Naur gst-plugins-bad-1.18.5/ext/x265/gstx265enc.c gst-plugins-bad-1.18.5.patched/ext/x265/gstx265enc.c
--- gst-plugins-bad-1.18.5/ext/x265/gstx265enc.c	2021-09-08 15:03:40.879254300 -0400
+++ gst-plugins-bad-1.18.5.patched/ext/x265/gstx265enc.c	2025-08-05 03:05:25.472399516 -0400
@@ -942,8 +942,9 @@
        * HEVC uses gbr order
        * See spec D.3.28 display_primaries_x and display_primaries_y
        */
-      encoder->x265param.masteringDisplayColorVolume =
-          g_strdup_printf ("G(%hu,%hu)B(%hu,%hu)R(%hu,%hu)WP(%hu,%hu)L(%u,%u)",
+      snprintf(encoder->x265param.masteringDisplayColorVolume,
+          X265_MAX_STRING_SIZE,
+          "G(%hu,%hu)B(%hu,%hu)R(%hu,%hu)WP(%hu,%hu)L(%u,%u)",
           minfo.display_primaries[1].x, minfo.display_primaries[1].y,
           minfo.display_primaries[2].x, minfo.display_primaries[2].y,
           minfo.display_primaries[0].x, minfo.display_primaries[0].y,
