--- ncspot-0.11.0/src/main.rs	2022-08-21 07:07:08.000000000 +0900
+++ ncspot-0.11.0/src/main.rs.new	2022-10-16 14:36:37.635552107 +0900
@@ -166,6 +166,7 @@
         credentials = credentials_prompt(Some(error_msg))?;
     }

+    println!("Connecting to Spotify..");
     let mut cursive = cursive::default().into_runner();
     cursive.set_window_title("ncspot");

@@ -174,7 +175,6 @@

     let event_manager = EventManager::new(cursive.cb_sink().clone());

-    println!("Connecting to Spotify..");
     let spotify = spotify::Spotify::new(event_manager.clone(), credentials, cfg.clone());

     let library = Arc::new(Library::new(&event_manager, spotify.clone(), cfg.clone()));
