diff -urN cowsay-3.03.old/cows/bob.cow cowsay-3.03.new/cows/bob.cow
--- cowsay-3.03.old/cows/bob.cow	1969-12-31 18:00:00.000000000 -0600
+++ cowsay-3.03.new/cows/bob.cow	2009-12-23 13:57:40.000000000 -0600
@@ -0,0 +1,10 @@
+$the_cow = <<"EOC";
+    $thoughts     ___
+     $thoughts   /_ _\\
+      $thoughts |/ ~ \\|
+        |'-.-`|
+        (|e|e|?
+        `._^_,'
+         \\\\=//
+         U/_/
+EOC
diff -urN cowsay-3.03.old/cowsay cowsay-3.03.new/cowsay
--- cowsay-3.03.old/cowsay	2000-05-19 20:29:49.000000000 -0500
+++ cowsay-3.03.new/cowsay	2009-12-23 11:32:29.000000000 -0600
@@ -16,7 +16,7 @@
 $progname = basename($0);
 $eyes = "oo";
 $tongue = "  ";
-$cowpath = $ENV{'COWPATH'} || '%PREFIX%/share/cows';
+$cowpath = $ENV{'COWPATH'} || '/usr/share/games/cows';
 @message = ();
 $thoughts = "";
 
diff -urN cowsay-3.03.old/cowsay.1 cowsay-3.03.new/cowsay.1
--- cowsay-3.03.old/cowsay.1	1999-11-04 13:50:40.000000000 -0600
+++ cowsay-3.03.new/cowsay.1	2009-12-23 13:38:08.000000000 -0600
@@ -149,12 +149,12 @@
 .B PATH or
 .BR MANPATH .
 It should always contain the
-.B /usr/local/share/cows
+.B /usr/share/games/cows
 directory, or at least a directory with a file called 
 .B default.cow
 in it.
 .SH FILES
-.B %PREFIX%/share/cows
+.B /usr/share/games/cows
 holds a sample set of cowfiles.  If your
 .B COWPATH
 is not explicitly set, it automatically contains this directory.
diff -urN cowsay-3.03.old/install.sh cowsay-3.03.new/install.sh
--- cowsay-3.03.old/install.sh	1999-11-01 14:19:21.000000000 -0600
+++ cowsay-3.03.new/install.sh	2009-12-23 11:31:34.000000000 -0600
@@ -67,26 +67,26 @@
 echo Okay, time to install this puppy.
 
 echo s,%BANGPERL%,!$usethisperl,\; > install.pl
-echo s,%PREFIX%,$PREFIX,\; >> install.pl
+#echo s,%PREFIX%,$PREFIX,\; >> install.pl
 set -x
-mkdir -p $PREFIX/bin || (mkdir $PREFIX; mkdir $PREFIX/bin)
-$usethisperl -p install.pl cowsay > $PREFIX/bin/cowsay
-chmod a+x $PREFIX/bin/cowsay
-ln -s cowsay $PREFIX/bin/cowthink
-mkdir -p $PREFIX/man/man1 || ($mkdir $PREFIX; mkdir $PREFIX/man; mkdir $PREFIX/man/man1)
-$usethisperl -p install.pl cowsay.1 > $PREFIX/man/man1/cowsay.1
-chmod a+r $PREFIX/man/man1/cowsay.1
-ln -s cowsay.1 $PREFIX/man/man1/cowthink.1
-mkdir -p $PREFIX/share/cows || (mkdir $PREFIX; mkdir $PREFIX/share; mkdir $PREFIX/share/cows)
-tar -cf - $filelist | (cd $PREFIX/share && tar -xvf -)
+mkdir -p $DESTDIR/$PREFIX/games || (mkdir $DESTDIR/$PREFIX; mkdir $DESTDIR/$PREFIX/games)
+$usethisperl -p install.pl cowsay > $DESTDIR/$PREFIX/games/cowsay
+chmod a+x $DESTDIR/$PREFIX/games/cowsay
+ln -s cowsay $DESTDIR/$PREFIX/games/cowthink
+mkdir -p $DESTDIR/$PREFIX/man/man1 || ($mkdir $DESTDIR/$PREFIX; mkdir $DESTDIR/$PREFIX/man; mkdir $DESTDIR/$PREFIX/man/man1)
+$usethisperl -p install.pl cowsay.1 > $DESTDIR/$PREFIX/man/man1/cowsay.1
+chmod a+r $DESTDIR/$PREFIX/man/man1/cowsay.1
+ln -s cowsay.1 $DESTDIR/$PREFIX/man/man1/cowthink.1
+mkdir -p $DESTDIR/$PREFIX/share/games/cows || (mkdir $DESTDIR/$PREFIX; mkdir $DESTDIR/$PREFIX/share; mkdir $DESTDIR/$PREFIX/share/games; mkdir $DESTDIR/$PREFIX/share/games/cows)
+tar -cf - $filelist | (cd $DESTDIR/$PREFIX/share/games && tar -xvf -)
 set +x
 
-echo Okay, let us see if the install actually worked.
+#echo Okay, let us see if the install actually worked.
 
-if [ ! -f $PREFIX/share/cows/default.cow ]; then
-    echo The default cow file did not make it across!
-    echo Ooops, it failed...sorry!
-    exit 1
-fi
+#if [ ! -f $PREFIX/share/cows/default.cow ]; then
+#    echo The default cow file did not make it across!
+#    echo Ooops, it failed...sorry!
+#    exit 1
+#fi
 
 echo Installation complete!  Enjoy the cows!
