diff -urpN fvwm95-2.0.43f.orig/modules/FvwmScroll/GrabWindow.c fvwm95-2.0.43f/modules/FvwmScroll/GrabWindow.c
--- fvwm95-2.0.43f.orig/modules/FvwmScroll/GrabWindow.c	2022-06-16 22:45:46.614676791 +0200
+++ fvwm95-2.0.43f/modules/FvwmScroll/GrabWindow.c	2022-06-17 23:58:40.056000000 +0200
@@ -38,10 +38,10 @@
 #include <X11/Intrinsic.h>
 
 #include "FvwmScroll.h"
-char *MyName;
+extern char *MyName;
 
-Display *dpy;			/* which display are we talking to */
-int x_fd,fd_width;
+extern Display *dpy;			/* which display are we talking to */
+extern int x_fd,fd_width;
 int Width = 300, Height = 300;
 int target_width, target_height;
 int target_x_offset = 0, target_y_offset = 0;
@@ -55,9 +55,9 @@ int Reduction_V = 2;
 #define PAD_WIDTH2 3
 #define PAD_WIDTH3 5
 
-Window Root;
-int screen;
-int d_depth;
+extern Window Root;
+extern int screen;
+extern int d_depth;
 
 Window main_win,holder_win;
 Pixel back_pix, fore_pix, hilite_pix,shadow_pix;


diff -urpN fvwm95-2.0.43f.orig/modules/FvwmTaskBar/Goodies.c fvwm95-2.0.43f/modules/FvwmTaskBar/Goodies.c
---  fvwm95-2.0.43f.orig/modules/FvwmTaskBar/Goodies.c	2022-06-16 22:45:46.615676777 +0200
+++ fvwm95-2.0.43f/modules/FvwmTaskBar/Goodies.c	2022-06-17 23:59:05.760000000 +0200
@@ -46,7 +46,8 @@ XFontSet StatusFontset;
 #endif
 int stwin_width = 100, old_stwin_width = 100, goodies_width = 0;
 int anymail, unreadmail, newmail, mailcleared = 0;
-int fontheight, clock_width;
+extern int fontheight;
+int clock_width;
 int BellVolume = DEFAULT_BELL_VOLUME;
 Pixmap mailpix, wmailpix, pmask, pclip, speakerpix, speakeroffpix,s_mask;
 XpmAttributes s_attr;

