diff -Naur caps-0.9.24/AutoFilter.cc caps-0.9.24.patched/AutoFilter.cc
--- caps-0.9.24/AutoFilter.cc	2014-11-10 03:12:26.000000000 -0500
+++ caps-0.9.24.patched/AutoFilter.cc	2017-06-22 16:26:26.421820579 -0400
@@ -69,7 +69,7 @@
 void
 AutoFilter::cycle (uint frames)
 {
-	div_t qr = div (frames, blocksize);
+	div_t qr = div ((int)frames, (int)blocksize);
 	int blocks = qr.quot;
 	if (qr.rem) ++blocks;
 	double over_blocks = 1./blocks;
