--- src/zm_event.cpp    2019-01-21 23:17:19.881961455 -0500
+++ src/zm_event.cpp.new        2019-01-21 23:17:38.452315254 -0500
@@ -196,7 +196,7 @@

   if ( monitor->GetOptVideoWriter() != 0 ) {
     snprintf(video_name, sizeof(video_name), "%" PRIu64 "-%s", id, "video.mp4");
-    snprintf(video_file, sizeof(video_file), staticConfig.video_file_format, path, video_name);
+    snprintf(video_file, sizeof(video_file), "%s/%s", path, video_name);
     Debug(1,"Writing video file to %s", video_file );

     /* X264 MP4 video writer */
