diff -Naur vocoder/src/output_jack.cpp vocoder.patched/src/output_jack.cpp
--- vocoder/src/output_jack.cpp	2009-05-25 17:56:46.000000000 -0400
+++ vocoder.patched/src/output_jack.cpp	2012-11-22 15:10:01.000000000 -0500
@@ -5,6 +5,7 @@
  */
 
 // Includes /*fold00*/
+#include <unistd.h>
 #include <iostream>
 #include "output_jack.h"
 
diff -Naur vocoder/util/play_loop.cc vocoder.patched/util/play_loop.cc
--- vocoder/util/play_loop.cc	2009-03-07 05:42:17.000000000 -0500
+++ vocoder.patched/util/play_loop.cc	2012-11-22 15:09:10.000000000 -0500
@@ -5,6 +5,7 @@
  */ 
 
 // Includes /*fold00*/
+#include <unistd.h>
 #include <string.h>
 #include <getopt.h>
 #include <sndfile.h>
