From: John H. Robinson, IV <jaqque@debian.org>
Subject: Modified toplevel Makefile to support $(DESTDIR)
--- nullidentd-1.0.orig/Makefile
+++ nullidentd-1.0/Makefile
@@ -1,5 +1,9 @@
 
-INSTALL=/usr/local/sbin
+# Edited for Debian GNU/Linux.
+DESTDIR=
+
+INSTALL=$(DESTDIR)/usr/sbin
+
 
 nullidentd: nullidentd.c version.h
 	gcc -O2 -o nullidentd nullidentd.c
@@ -21,9 +25,10 @@ clean:
 	rm -f .version version.h nullidentd
 
 install: nullidentd
-	rm -f $(INSTALL)/nullidentd
+	#rm -f $(INSTALL)/nullidentd
 	cp nullidentd $(INSTALL)/nullidentd
 	chown root.root $(INSTALL)/nullidentd
-	chmod a-rw $(INSTALL)/nullidentd
-	chmod a+x $(INSTALL)/nullidentd
+	#chmod a-rw $(INSTALL)/nullidentd
+	#chmod a+x $(INSTALL)/nullidentd
+	chmod 0755 $(INSTALL)/nullidentd
 
