From 85422b744ae2294d60fdae9074ad8eae53323e66 Mon Sep 17 00:00:00 2001
From: Ebben Aries <exa@dscp.org>
Date: Thu, 16 Jun 2022 08:18:06 -0700
Subject: [PATCH] Fix CMakeLists.txt to honor CMAKE_INSTALL_LIBDIR

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 21d24ba6e..8f83f22c4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,11 +52,11 @@ project(${PACKAGE_NAME} CXX C ASM)
 
 set(INCLUDE_INSTALL_DIR include CACHE STRING
     "The subdirectory where header files should be installed")
-set(LIB_INSTALL_DIR lib CACHE STRING
+set(LIB_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR} CACHE STRING
     "The subdirectory where libraries should be installed")
 set(BIN_INSTALL_DIR bin CACHE STRING
     "The subdirectory where binaries should be installed")
-set(CMAKE_INSTALL_DIR lib/cmake/folly CACHE STRING
+set(CMAKE_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/folly CACHE STRING
     "The subdirectory where CMake package config files should be installed")
 
 option(BUILD_SHARED_LIBS
-- 
2.36.1

