diff -Naur mupdf-1.5-source/source/fitz/load-jpx.c mupdf-1.5-source.patched/source/fitz/load-jpx.c
--- mupdf-1.5-source/source/fitz/load-jpx.c	2014-06-10 11:09:28.000000000 -0400
+++ mupdf-1.5-source.patched/source/fitz/load-jpx.c	2014-08-21 23:49:21.000000000 -0400
@@ -116,7 +116,8 @@
 	opj_stream_set_read_function(stream, fz_opj_stream_read);
 	opj_stream_set_skip_function(stream, fz_opj_stream_skip);
 	opj_stream_set_seek_function(stream, fz_opj_stream_seek);
-	opj_stream_set_user_data(stream, &sb);
+	opj_stream_set_user_data(stream, &sb, NULL);
+
 	/* Set the length to avoid an assert */
 	opj_stream_set_user_data_length(stream, size);
 
