diff -uar Python-3.1.2.orig//Lib/distutils/command/install.py Python-3.1.2/Lib/distutils/command/install.py
--- Python-3.1.2.orig//Lib/distutils/command/install.py	2010-05-30 21:01:34.000000000 +0300
+++ Python-3.1.2/Lib/distutils/command/install.py	2010-05-30 21:03:07.000000000 +0300
@@ -46,14 +46,14 @@
 
 INSTALL_SCHEMES = {
     'unix_prefix': {
-        'purelib': '$base/lib/python$py_version_short/site-packages',
+        'purelib': '$base/lib64/python$py_version_short/site-packages',
         'platlib': '$platbase/lib64/python$py_version_short/site-packages',
         'headers': '$base/include/python$py_version_short/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
         },
     'unix_home': {
-        'purelib': '$base/lib/python',
+        'purelib': '$base/lib64/python',
         'platlib': '$base/lib64/python',
         'headers': '$base/include/python/$dist_name',
         'scripts': '$base/bin',
diff -uar Python-3.1.2.orig//Lib/distutils/sysconfig.py Python-3.1.2/Lib/distutils/sysconfig.py
--- Python-3.1.2.orig//Lib/distutils/sysconfig.py	2010-05-30 21:01:34.000000000 +0300
+++ Python-3.1.2/Lib/distutils/sysconfig.py	2010-05-30 21:03:41.000000000 +0300
@@ -120,7 +120,7 @@
         if plat_specific or standard_lib:
             lib = "lib64"
         else:
-            lib = "lib"
+            lib = "lib64"
         libpython = os.path.join(prefix,
                                  lib, "python" + get_python_version())
         if standard_lib:
