
diff -r 8715c044a7ee -r 58fc2f01c432 src/managesieve-login/managesieve-proxy.c
--- a/src/managesieve-login/managesieve-proxy.c	Sat Jan 07 16:29:23 2012 +0100
+++ b/src/managesieve-login/managesieve-proxy.c	Fri Feb 03 23:51:53 2012 +0100
@@ -325,13 +325,13 @@
 
 			(void)client_skip_line(msieve_client);
 			client_proxy_finish_destroy_client(client);
-			
+
 			return 1;
-		} 
-		
+		}
+
 		/* Login failed */
 
-		if ( client->set->verbose_auth ) {
+		if ( client->set->auth_verbose ) {
 			const char *log_line = line;
 
 			if (strncasecmp(log_line, "NO ", 3) == 0)




