diff -Naur verlihub-0.9.8e-r2.orig/src/script_api.cpp verlihub-0.9.8e-r2/src/script_api.cpp
--- verlihub-0.9.8e-r2.orig/src/script_api.cpp	2009-05-16 16:42:07.000000000 +0200
+++ verlihub-0.9.8e-r2/src/script_api.cpp	2017-06-01 20:33:58.923440440 +0200
@@ -218,12 +218,12 @@
 	cServerDC *server = GetCurrentVerlihub();
 	if(!server) {
 		cerr << "Server verlihub is unfortunately not running or not found." << endl;
-		return false;
+		return NULL;
 	}
 	cUser *usr = GetUser((char *) server->mC.hub_security.c_str());
 	printf("%p\n", usr);
 	printf("%p", usr->mxConn);
-	if ((!usr) || (usr && !usr->mxConn)) return false;
+	if ((!usr) || (usr && !usr->mxConn)) return NULL;
 	cout << "here" << endl;
 	if (!server->mP.ParseForCommands(command_line, usr->mxConn)) {
 		// unknown command	
