--- nikto-2.1.6.orig/program/replay.pl	2015-07-09 08:06:04.000000000 -0600
+++ nikto-2.1.6/program/replay.pl	2017-04-21 17:57:45.292892040 -0600
@@ -22,7 +22,7 @@
 ###############################################################################
 use Getopt::Long;
 use JSON::PP;
-require 'plugins/LW2.pm';
+use LW2;
 my ($infile, $proxy, %request, $header, %result, $s_request);
 LW2::http_init_request(\%request);
 