diff -ruN SOGo-5.4.0/SoObjects/common.make SOGo-5.4.0-b/SoObjects/common.make
--- SOGo-5.4.0/SoObjects/common.make	2021-12-16 11:03:37.000000000 -0600
+++ SOGo-5.4.0-b/SoObjects/common.make	2022-01-16 03:04:05.560313315 -0600
@@ -18,7 +18,7 @@
         -I../../SOPE
 
 ADDITIONAL_LIB_DIRS += \
-        -L../SOGo/SOGo.framework/Versions/Current/sogo/ -lSOGo \
+        -L../SOGo/SOGo.framework/Versions/Current/sogo/ \
 	-L../../SOGo/$(GNUSTEP_OBJ_DIR)/ \
 	-L../../SOPE/NGCards/$(GNUSTEP_OBJ_DIR)/ -lNGCards \
         -L../../SOPE/GDLContentStore/$(GNUSTEP_OBJ_DIR)/ -lGDLContentStore \
