--- support/Startup/argus.orig	2010-07-06 11:23:38.032000082 -0400
+++ support/Startup/argus	2010-07-06 11:27:15.466000084 -0400
@@ -43,17 +43,17 @@
 fi
 
 # Source networking configuration.
-. /etc/sysconfig/network
+#. /etc/sysconfig/network
 
 # Check that networking is up.
-[ ${NETWORKING} = "no" ] && exit 1
+# [ ${NETWORKING} = "no" ] && exit 1
 
 
 # Set argus path by defining $ARGUSHOME for this script.
 # If argus was installed in another way, modify PATH to
 # include the directory where the argus binary was installed.
 
-ARGUSDIR=/usr/local/sbin
+ARGUSDIR=/usr/sbin
 ARGUSHOME=$ARGUSDIR
 export PATH=$ARGUSHOME:$PATH
 
@@ -71,7 +71,7 @@
                 then
                         mkdir /var/log/argus
                 fi
-                argus -de `hostname` -w /var/log/argus/argus.out \
+                argus -p -de `hostname` -w /var/log/argus/argus.out \
                         > /dev/null 2>&1
                 RETVAL=$?
         else
