diff -Naur opencomal/src/Makefile opencomal.patched/src/Makefile
--- opencomal/src/Makefile	2025-01-13 23:01:02.000000000 -0500
+++ opencomal.patched/src/Makefile	2025-01-14 16:51:00.468867796 -0500
@@ -37,20 +37,20 @@
 	pdcseg.o pdcval.o pdccloop.o pdcmod.o pdccomp.o pdccexp.o
 
 # Choose OS below
-ASAN=-fsanitize=address -fno-omit-frame-pointer
+#ASAN=-fsanitize=address -fno-omit-frame-pointer
 
 # MacOS
-OS=pdcmac.o pdcposix.o
-CC=clang
-COPTS=-c -Wall -ggdb -DOS_MAC $(PROFILE) -I/opt/homebrew/opt/readline/include -pg $(ASAN)
-OSLDOPTS=-lncurses  -L/opt/homebrew/opt/readline/lib -lreadline $(ASAN)
+# OS=pdcmac.o pdcposix.o
+# CC=clang
+# COPTS=-c -Wall -ggdb -DOS_MAC $(PROFILE) -I/opt/homebrew/opt/readline/include -pg $(ASAN)
+# OSLDOPTS=-lncurses  -L/opt/homebrew/opt/readline/lib -lreadline $(ASAN)
 
 # Linux
-# OS=pdclinux.o pdcposix.o
-# CC=gcc
-# COPTS=-c -ggdb -pedantic -Wall -DOS_LINUX $(PROFILE) $(ASAN)
+OS=pdclinux.o pdcposix.o
+CC=gcc
+COPTS=-c -pedantic -Wall -DOS_LINUX $(PROFILE) $(ASAN)
 # COPTS=-c -pedantic -Wall -DOS_LINUX $(PROFILE)
-# OSLDOPTS=$(ASAN) -lncurses -lreadline
+OSLDOPTS=$(ASAN) -lncurses -lreadline
 
 # Webasm
 # OS=pdcwasm.c
@@ -97,7 +97,7 @@
 
 
 buildinfo:		
-		../tools/genversion
+		sh ../tools/genversion
 
 pdcpars.tab.c:	pdcpars.y
 		bison -vd pdcpars.y
