diff -Naurp gtkdialog-0.7.20-orig/src/glade_support.c gtkdialog-0.7.20/src/glade_support.c
--- gtkdialog-0.7.20-orig/src/glade_support.c	2007-03-16 10:25:49.000000000 -0400
+++ gtkdialog-0.7.20/src/glade_support.c	2009-09-06 00:10:21.405853809 -0400
@@ -37,10 +37,10 @@
  *                                                                       *
  *                                                                       *
  *************************************************************************/
-typedef struct signal {
+typedef struct gtkd_signal {
 	gchar     *name;
 	GCallback callback;
-} signal;
+} gtkd_signal;
 
 /*
 ** Signal handler callbascks.
@@ -206,7 +206,7 @@ on_any_widget_almost_any_gdk_event(
 static gboolean
 find_and_connect_handler(
 		GtkWidget *widget,
-		signal *signals,
+		gtkd_signal *signals,
 		const gchar *signal_name,
 		const gchar *handler_name)
 {
@@ -288,7 +288,7 @@ gtk_entry_signal_handler_connector(
 		gpointer user_data)
 {
 	gint n;
-	signal entry_signals[] = {
+	gtkd_signal entry_signals[] = {
 		{ "activate",           (GCallback)on_any_entry_almost_any },
 		{ "backspace",          (GCallback)on_any_entry_almost_any },
 		{ "copy-clipboard",     (GCallback)on_any_entry_almost_any },
@@ -319,7 +319,7 @@ gtk_widget_signal_handler_connector(
 		gpointer user_data)
 {
 	variable *var;
-	signal widget_signals[] = {
+	gtkd_signal widget_signals[] = {
 		{ "accel-closures-changed",    (GCallback)on_any_widget_almost_any },
 		{ "composited-changed",        (GCallback)on_any_widget_almost_any },
 		{ "grab-focus",                (GCallback)on_any_widget_almost_any },
