--- Makefile	2007-04-22 14:18:21.000000000 -0400
+++ Makefile.new	2007-04-22 14:31:49.000000000 -0400
@@ -12,16 +12,16 @@
 	gcc -Wall -g -o set_rlimits set_rlimits.c
 
 install:	set_rlimits
-	test -d $(PREFIX)/bin || mkdir -p $(PREFIX)/bin
-	test -d $(PREFIX)/man/man8 || mkdir -p $(PREFIX)/man/man8
-	test -d $(SYSCONFDIR) || mkdir -p $(SYSCONFDIR)
-	cp -p set_rlimits $(PREFIX)/bin/
-	chown root.root $(PREFIX)/bin/set_rlimits
-	chmod u+s $(PREFIX)/bin/set_rlimits
-	test -e $(SYSCONFDIR)/set_rlimits.conf || cp set_rlimits.conf $(SYSCONFDIR)/
-	chown root.root $(SYSCONFDIR)/set_rlimits.conf
-	cp -p set_rlimits.8 $(PREFIX)/man/man8/
-	chown root.root $(PREFIX)/man/man8/set_rlimits.8
+	test -d $(DESTDIR)$(PREFIX)/bin || mkdir -p $(DESTDIR)$(PREFIX)/bin
+	test -d $(DESTDIR)$(PREFIX)/man/man8 || mkdir -p $(DESTDIR)$(PREFIX)/man/man8
+	test -d $(DESTDIR)$(SYSCONFDIR) || mkdir -p $(DESTDIR)$(SYSCONFDIR)
+	cp -p set_rlimits $(DESTDIR)$(PREFIX)/bin/
+	chown root.root $(DESTDIR)$(PREFIX)/bin/set_rlimits
+	chmod u+s $(DESTDIR)$(PREFIX)/bin/set_rlimits
+	test -e $(DESTDIR)$(SYSCONFDIR)/set_rlimits.conf || cp set_rlimits.conf $(DESTDIR)$(SYSCONFDIR)/
+	chown root.root $(DESTDIR)$(SYSCONFDIR)/set_rlimits.conf
+	cp -p set_rlimits.8 $(DESTDIR)$(PREFIX)/man/man8/
+	chown root.root $(DESTDIR)$(PREFIX)/man/man8/set_rlimits.8
 
 .PHONY:	clean tidy
 
