diff --git a/autoconf/Makefile.am b/autoconf/Makefile.am
index e943a79..b983bc7 100644
--- a/autoconf/Makefile.am
+++ b/autoconf/Makefile.am
@@ -25,7 +25,6 @@ EXTRA_DIST = \
   gtk.m4 \
   gtk-2.0.m4 \
   libFLAC.m4 \
-  libOggFLAC.m4 \
   ogg.m4 \
   utils.m4 \
-  vorbis.m4
\ No newline at end of file
+  vorbis.m4
diff --git a/autoconf/alsa.m4 b/autoconf/alsa.m4
index 3429d08..2b65bc6 100644
--- a/autoconf/alsa.m4
+++ b/autoconf/alsa.m4
@@ -14,7 +14,7 @@ dnl
 dnl For backwards compatibility, if ACTION_IF_NOT_FOUND is not specified,
 dnl and the alsa libraries are not found, a fatal AC_MSG_ERROR() will result.
 dnl
-AC_DEFUN(AM_PATH_ALSA,
+AC_DEFUN([AM_PATH_ALSA],
 [dnl Save the original CFLAGS, LDFLAGS, and LIBS
 alsa_save_CFLAGS="$CFLAGS"
 alsa_save_LDFLAGS="$LDFLAGS"
@@ -151,4 +151,4 @@ fi
 dnl That should be it.  Now just export out symbols:
 AC_SUBST(ALSA_CFLAGS)
 AC_SUBST(ALSA_LIBS)
-])
\ No newline at end of file
+])
diff --git a/autoconf/ao.m4 b/autoconf/ao.m4
index a2bfa67..a852ad8 100644
--- a/autoconf/ao.m4
+++ b/autoconf/ao.m4
@@ -6,7 +6,7 @@
 dnl XIPH_PATH_AO([ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]])
 dnl Test for libao, and define AO_CFLAGS and AO_LIBS
 dnl
-AC_DEFUN(XIPH_PATH_AO,
+AC_DEFUN([XIPH_PATH_AO],
 [dnl 
 dnl Get the cflags and libraries
 dnl
diff --git a/autoconf/arts.m4 b/autoconf/arts.m4
index 7ac4daa..f6b1d94 100644
--- a/autoconf/arts.m4
+++ b/autoconf/arts.m4
@@ -9,7 +9,7 @@ dnl Snarfed from the ESD code below - but the faults are all mine!
 dnl AM_PATH_ARTS([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]]])
 dnl Test for aRts, and define ARTS_CFLAGS and ARTS_LIBS
 dnl
-AC_DEFUN(AM_PATH_ARTS,
+AC_DEFUN([AM_PATH_ARTS],
 	[dnl 
 dnl Get the cflags and libraries from the arts-config script
 dnl
@@ -156,4 +156,4 @@ int main ()
   AC_SUBST(ARTS_LIBS)
   rm -f conf.artstest
 ])
-dnl End of aRts
\ No newline at end of file
+dnl End of aRts
diff --git a/autoconf/esd.m4 b/autoconf/esd.m4
index 289a6c0..24666c9 100644
--- a/autoconf/esd.m4
+++ b/autoconf/esd.m4
@@ -7,7 +7,7 @@
 dnl AM_PATH_ESD([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]]])
 dnl Test for ESD, and define ESD_CFLAGS and ESD_LIBS
 dnl
-AC_DEFUN(AM_PATH_ESD,
+AC_DEFUN([AM_PATH_ESD],
 [dnl 
 dnl Get the cflags and libraries from the esd-config script
 dnl
@@ -172,7 +172,7 @@ int main ()
 dnl AM_ESD_SUPPORTS_MULTIPLE_RECORD([ACTION-IF-SUPPORTS [, ACTION-IF-NOT-SUPPORTS]])
 dnl Test, whether esd supports multiple recording clients (version >=0.2.21)
 dnl
-AC_DEFUN(AM_ESD_SUPPORTS_MULTIPLE_RECORD,
+AC_DEFUN([AM_ESD_SUPPORTS_MULTIPLE_RECORD],
 [dnl
   AC_MSG_NOTICE([whether installed esd version supports multiple recording clients])
   ac_save_ESD_CFLAGS="$ESD_CFLAGS"
diff --git a/autoconf/gtk-2.0.m4 b/autoconf/gtk-2.0.m4
index a0c53ac..36a1f9f 100644
--- a/autoconf/gtk-2.0.m4
+++ b/autoconf/gtk-2.0.m4
@@ -5,7 +5,7 @@ dnl AM_PATH_GTK_2_0([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND [,
 dnl Test for GTK+, and define GTK_CFLAGS and GTK_LIBS, if gthread is specified in MODULES, 
 dnl pass to pkg-config
 dnl
-AC_DEFUN(AM_PATH_GTK_2_0,
+AC_DEFUN([AM_PATH_GTK_2_0],
 [dnl 
 dnl Get the cflags and libraries from pkg-config
 dnl
diff --git a/autoconf/gtk.m4 b/autoconf/gtk.m4
index b8cf402..f2dd472 100644
--- a/autoconf/gtk.m4
+++ b/autoconf/gtk.m4
@@ -4,7 +4,7 @@
 dnl AM_PATH_GTK([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND [, MODULES]]]])
 dnl Test for GTK, and define GTK_CFLAGS and GTK_LIBS
 dnl
-AC_DEFUN(AM_PATH_GTK,
+AC_DEFUN([AM_PATH_GTK],
 [dnl 
 dnl Get the cflags and libraries from the gtk-config script
 dnl
diff --git a/autoconf/libFLAC.m4 b/autoconf/libFLAC.m4
index 88d8e84..b6950e7 100644
--- a/autoconf/libFLAC.m4
+++ b/autoconf/libFLAC.m4
@@ -4,7 +4,7 @@
 dnl AM_PATH_LIBFLAC([ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]])
 dnl Test for libFLAC, and define LIBFLAC_CFLAGS and LIBFLAC_LIBS
 dnl
-AC_DEFUN(AM_PATH_LIBFLAC,
+AC_DEFUN([AM_PATH_LIBFLAC],
 [dnl 
 dnl Get the cflags and libraries
 dnl
diff --git a/autoconf/libOggFLAC.m4 b/autoconf/libOggFLAC.m4
index 5ce8bda..9b5f6d2 100644
--- a/autoconf/libOggFLAC.m4
+++ b/autoconf/libOggFLAC.m4
@@ -4,7 +4,7 @@
 dnl AM_PATH_LIBOGGFLAC([ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]])
 dnl Test for libOggFLAC, and define LIBOGGFLAC_CFLAGS and LIBOGGFLAC_LIBS
 dnl
-AC_DEFUN(AM_PATH_LIBOGGFLAC,
+AC_DEFUN([AM_PATH_LIBOGGFLAC],
 [dnl 
 dnl Get the cflags and libraries
 dnl
diff --git a/autoconf/ogg.m4 b/autoconf/ogg.m4
index 89558ab..5c13745 100644
--- a/autoconf/ogg.m4
+++ b/autoconf/ogg.m4
@@ -5,7 +5,7 @@
 dnl AM_PATH_OGG([ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]])
 dnl Test for libogg, and define OGG_CFLAGS and OGG_LIBS
 dnl
-AC_DEFUN(AM_PATH_OGG,
+AC_DEFUN([AM_PATH_OGG],
 [dnl 
 dnl Get the cflags and libraries
 dnl
diff --git a/autoconf/utils.m4 b/autoconf/utils.m4
index ecf03c8..97a2e85 100644
--- a/autoconf/utils.m4
+++ b/autoconf/utils.m4
@@ -17,7 +17,7 @@ dnl along with this program; if not, write to the Free Software
 dnl Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 
 dnl MY_DEFINE(VARIABLE)
-AC_DEFUN(MY_DEFINE,
+AC_DEFUN([MY_DEFINE],
 [cat >> confdefs.h <<EOF
 [#define] $1 1
 EOF
@@ -29,7 +29,7 @@ dnl                  action-if-yes-or-dynamic,
 dnl		     $5
 dnl		     action-if-yes,action-if-dynamic,action-if-no)
 dnl		     $6            $7                $8
-AC_DEFUN(CONFIG_INTERFACE,
+AC_DEFUN([CONFIG_INTERFACE],
 [AC_ARG_ENABLE($1,[$4],
 [case "x$enable_$1" in xyes|xdynamic) $5 ;; esac])
 case "x$enable_$1" in
@@ -53,7 +53,7 @@ dnl CHECK_DLSYM_UNDERSCORE([ACTION-IF-NEED [, ACTION IF-NOT-NEED]])
 dnl variable input:
 dnl   CC CFLAGS CPPFLAGS LDFLAGS LIBS SHCFLAGS SHLD SHLDFLAGS
 dnl   ac_cv_header_dlfcn_h lib_dl_opt so
-AC_DEFUN(CHECK_DLSYM_UNDERSCORE,
+AC_DEFUN([CHECK_DLSYM_UNDERSCORE],
 [dnl Check if dlsym need a leading underscore
 AC_MSG_CHECKING(whether your dlsym() needs a leading underscore)
 AC_CACHE_VAL(timidity_cv_func_dlsym_underscore,
@@ -144,7 +144,7 @@ esac
 
 dnl contains program from perl5
 dnl CONTAINS_INIT()
-AC_DEFUN(CONTAINS_INIT,
+AC_DEFUN([CONTAINS_INIT],
 [dnl Some greps do not return status, grrr.
 AC_MSG_CHECKING(whether grep returns status)
 echo "grimblepritz" >grimble
@@ -171,7 +171,7 @@ esac
 ])
 
 dnl CONTAINS(word,filename,action-if-found,action-if-not-found)
-AC_DEFUN(CONTAINS,
+AC_DEFUN([CONTAINS],
 [if $contains "^[$1]"'[$]' $2 >/dev/null 2>&1; then
   [$3]
 else
@@ -180,7 +180,7 @@ fi
 ])
 
 dnl SET_UNIQ_WORDS(shell-variable,words...)
-AC_DEFUN(SET_UNIQ_WORDS,
+AC_DEFUN([SET_UNIQ_WORDS],
 [rm -f wordtmp >/dev/null 2>&1
 val=''
 for f in $2; do
@@ -193,7 +193,7 @@ rm -f wordtmp >/dev/null 2>&1
 
 dnl WAPI_CHECK_FUNC(FUNCTION, INCLUDES, TEST-BODY,
 		    [ACTION-FI-FOUND [, ACTION-IF-NOT-FOUND]])
-AC_DEFUN(WAPI_CHECK_FUNC,
+AC_DEFUN([WAPI_CHECK_FUNC],
 [AC_MSG_CHECKING(for $1)
 AC_CACHE_VAL(wapi_cv_func_$1,
 [AC_TRY_LINK([#include <windows.h>
@@ -214,7 +214,7 @@ dnl WAPI_CHECK_LIB(LIBRARY, FUNCTION,
 dnl		INCLUDES, TEST-BODY
 dnl		[, ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND
 dnl		[, OTHER-LIBRARIES]]])
-AC_DEFUN(WAPI_CHECK_LIB,
+AC_DEFUN([WAPI_CHECK_LIB],
 [AC_MSG_CHECKING([for $2 in -l$1])
 ac_lib_var=`echo $1['_']$2 | sed 'y%./+-%__p_%'`
 AC_CACHE_VAL(wapi_cv_lib_$ac_lib_var,
@@ -238,18 +238,18 @@ fi
 ])
 
 dnl EXTRACT_CPPFLAGS(CPPFLAGS-to-append,others-to-append,FLAGS)
-AC_DEFUN(EXTRACT_CPPFLAGS,
+AC_DEFUN([EXTRACT_CPPFLAGS],
 [for f in $3; do
     case ".$f" in
 	.-I?*|.-D?*)	$1="[$]$1 $f" ;;
-	*)		$2="[$]$1 $f" ;;
+	*)		$2="[$]$2 $f" ;;
     esac
 done
 ])
 
 
 dnl CHECK_COMPILER_OPTION(OPTIONS [, ACTION-IF-SUCCEED [, ACTION-IF-FAILED]])
-AC_DEFUN(CHECK_COMPILER_OPTION,
+AC_DEFUN([CHECK_COMPILER_OPTION],
 [AC_MSG_CHECKING([whether -$1 option is recognized])
 ac_ccoption=`echo $1 | sed 'y%./+-%__p_%'`
 AC_CACHE_VAL(timidity_cv_ccoption_$ac_ccoption,
@@ -282,7 +282,7 @@ dnl MY_SEARCH_LIBS(FUNCTION, LIBRARIES [, ACTION-IF-FOUND
 dnl            [, ACTION-IF-NOT-FOUND [, OTHER-LIBRARIES]]])
 dnl Search for a library defining FUNC, if it's not already available.
 
-AC_DEFUN(MY_SEARCH_LIBS,
+AC_DEFUN([MY_SEARCH_LIBS],
 [AC_CACHE_CHECK([for library containing $1], [timidity_cv_search_$1],
 [ac_func_search_save_LIBS="$LIBS"
 timidity_cv_search_$1="no"
@@ -295,4 +295,4 @@ if test "$timidity_cv_search_$1" != "no"; then
   $3
 else :
   $4
-fi])
\ No newline at end of file
+fi])
diff --git a/autoconf/vorbis.m4 b/autoconf/vorbis.m4
index 9910dad..0e266bd 100644
--- a/autoconf/vorbis.m4
+++ b/autoconf/vorbis.m4
@@ -5,7 +5,7 @@
 dnl AM_PATH_VORBIS([ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]])
 dnl Test for libvorbis, and define VORBIS_CFLAGS and VORBIS_LIBS
 dnl
-AC_DEFUN(AM_PATH_VORBIS,
+AC_DEFUN([AM_PATH_VORBIS],
 [dnl 
 dnl Get the cflags and libraries
 dnl
diff --git a/configure.in b/configure.in
index 9f2835b..04897e6 100644
--- a/configure.in
+++ b/configure.in
@@ -388,7 +388,7 @@ dnl End of X configure
 
 # Checks for header files.
 AC_HEADER_STDC
-AC_HEADER_STDBOOL
+#AC_HEADER_STDBOOL
 AC_HEADER_SYS_WAIT
 if test "x$WATCOM_C" != xyes ; then
 AC_HEADER_TIME
@@ -689,7 +689,7 @@ dnl arts(R):	aRts
 dnl esd(e):     EsounD
 dnl portaudio(p) PortAudio
 dnl vorbis(v):  Ogg Vorbis
-dnl flac(F):    FLAC / OggFLAC
+dnl flac(F):    FLAC
 dnl speex(S):   Ogg Speex
 dnl gogo(g):    MP3 GOGO
 dnl jack(j):    JACK
@@ -723,7 +723,7 @@ AC_ARG_ENABLE(audio,
                               jack:      JACK
                               ao:        Libao
                               vorbis:    Ogg Vorbis
-                              flac:      FLAC / OggFLAC
+                              flac:      FLAC
                               speex:     Ogg Speex
                               gogo:      MP3 GOGO (Only Windows is supported)],
   [ enable_audio=$enableval
@@ -1198,11 +1198,6 @@ if test "x$CYGNUS" = xyes || test "x$MSYS" = xyes ; then
       EXTRACT_CPPFLAGS(CPPFLAGS,CFLAGS,$LIBFLAC_CFLAGS)
       LIBS="$LIBS $LIBFLAC_LIBS"
     ])
-    AM_PATH_LIBOGGFLAC([
-      EXTRADEFS="$EXTRADEFS -DAU_OGGFLAC -DAU_OGGFLAC_DLL"
-      EXTRACT_CPPFLAGS(CPPFLAGS,CFLAGS,$LIBOGGFLAC_CFLAGS)
-      LIBS="$LIBS $LIBOGGFLAC_LIBS"
-    ])
   else
     AC_MSG_RESULT(no)
   fi
@@ -1216,11 +1211,6 @@ else
       EXTRACT_CPPFLAGS(CPPFLAGS,CFLAGS,$LIBFLAC_CFLAGS)
       LIBS="$LIBS $LIBFLAC_LIBS"
     ])
-    AM_PATH_LIBOGGFLAC([
-      EXTRADEFS="$EXTRADEFS -DAU_OGGFLAC"
-      EXTRACT_CPPFLAGS(CPPFLAGS,CFLAGS,$LIBOGGFLAC_CFLAGS)
-      LIBS="$LIBS $LIBOGGFLAC_LIBS"
-    ])
   else
     AC_MSG_RESULT(no)
   fi
diff --git a/interface/Makefile.am b/interface/Makefile.am
index b02d085..1f50572 100644
--- a/interface/Makefile.am
+++ b/interface/Makefile.am
@@ -247,7 +247,7 @@ clean:
 
 tclIndex: $(TCLSRCS)
 	rm -f tclIndex
-	echo 'auto_mkindex . *.tcl; exit' | $(WISH)
+	echo "auto_mkindex . *.tcl" | tclsh
 
 if ENABLE_TCLTK
 install_tk = install.tk
diff --git a/timidity/Makefile.am b/timidity/Makefile.am
index a411d87..d5141ba 100644
--- a/timidity/Makefile.am
+++ b/timidity/Makefile.am
@@ -155,11 +155,7 @@ EXTRA_timidity_SOURCES = \
 	w32_libFLAC_dll.c \
 	w32_libFLAC_dll.h \
 	w32_libFLAC_dll_g.h \
-	w32_libFLAC_dll_i.h \
-	w32_libOggFLAC_dll.c \
-	w32_libOggFLAC_dll.h \
-	w32_libOggFLAC_dll_g.h \
-	w32_libOggFLAC_dll_i.h
+	w32_libFLAC_dll_i.h
 
 calcnewt_SOURCES = calcnewt.c
 
@@ -308,7 +304,7 @@ endif
 version.$(OBJEXT): version.c ../configure
 	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(CFLAGS) -c $(srcdir)/version.c
 
-resample.c: newton_table.c
+$(srcdir)/resample.c: newton_table.c
 
 if VCPP
 newton_table.c: calcnewt$(EXEEXT)
