diff --git a/interface/timidity.el b/interface/timidity.el
index f03ba25..c2019bf 100644
--- a/interface/timidity.el
+++ b/interface/timidity.el
@@ -28,14 +28,14 @@
 ;;
 ;; You write follows in ~/.emacs
 ;; (autoload 'timidity "timidity" "TiMidity Interface" t)
-;; (setq timidity-prog-path "/usr/local/bin/timidity")
+;; (setq timidity-prog-path "/usr/bin/timidity")
 ;;
 ;; Then you type:
 ;; M-x timidity
 
 ;; Configuration parameters.
 ; Absolute path of timidity.
-(defvar timidity-prog-path "/usr/local/bin/timidity")
+(defvar timidity-prog-path "/usr/bin/timidity")
 
 ; String list for timidity program options.
 (defvar timidity-default-options nil)
diff --git a/interface/tkmidity.ptcl b/interface/tkmidity.ptcl
index caf5e0a..7168bc4 100755
--- a/interface/tkmidity.ptcl
+++ b/interface/tkmidity.ptcl
@@ -1,4 +1,4 @@
-#!/usr/local/bin/wishx -f
+#!/usr/bin/wish -f
 #
 # TiMidity++ -- MIDI to WAVE converter and player
 # Copyright (C) 1999-2002 Masanao Izumo <mo@goice.co.jp>
diff --git a/interface/tkpanel.tcl b/interface/tkpanel.tcl
index 0a392b1..e826c54 100755
--- a/interface/tkpanel.tcl
+++ b/interface/tkpanel.tcl
@@ -1,4 +1,4 @@
-#!/usr/local/bin/wishx -f
+#!/usr/bin/wish -f
 #
 # TiMidity++ -- MIDI to WAVE converter and player
 # Copyright (C) 1999-2002 Masanao Izumo <mo@goice.co.jp>
