diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -14785,7 +14785,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 
-  #include <speex.h>
+  #include <speex/speex.h>
   #include <ogg/ogg.h>
 
 int
diff --git a/configure.in b/configure.in
--- a/configure.in
+++ b/configure.in
@@ -1230,7 +1230,7 @@
 if test "x$au_enable_speex" = xyes; then
   AC_CACHE_VAL(have_speex,
   [AC_TRY_LINK([
-  #include <speex.h>
+  #include <speex/speex.h>
   #include <ogg/ogg.h>
   ],
   [
diff --git a/timidity/speex_a.c b/timidity/speex_a.c
--- a/timidity/speex_a.c
+++ b/timidity/speex_a.c
@@ -33,8 +33,8 @@
 #include <fcntl.h>
 #endif
 
-#include <speex.h>
-#include <speex_header.h>
+#include <speex/speex.h>
+#include <speex/speex_header.h>
 #include <ogg/ogg.h>
 
 #include "timidity.h"
