--- src/protocols/livejournal/ljio/editfriendsjob.cpp	2009-10-23 17:15:11.000000000 +0200
+++ src/protocols/livejournal/ljio/editfriendsjob.cpp.old	2010-07-02 09:44:18.000000000 +0200
@@ -28,10 +28,10 @@
 {
 public:
 	AddUser( void ):
-	user( 0 ),
+	user(),
 	fg( QColor() ),
 	bg( QColor() ),
-	groupmask( 0 ) {}
+	groupmask() {}
 	
 	AddUser( const QString & user,
 	         const QColor & fg,
--- src/core/config/accountwizard.cpp	2009-10-23 17:16:07.000000000 +0200
+++ src/core/config/accountwizard.cpp.old	2010-07-02 09:10:56.000000000 +0200
@@ -46,8 +46,8 @@
 
 BlokkalAccountWizard::BlokkalAccountWizard( QWidget *parent ) :
 KAssistantDialog( parent ),
-m_editAccountPage( 0 ),
-m_accountName( 0 ),
+m_editAccountPage(),
+m_accountName(),
 m_fetcher( new RsdDataFetcher( this ) )
 {
 	setWindowTitle( i18n( "Add Account" ) );
