diff -Naur eduke32_20160704-5811.orig/build/src/util/generateicon.c eduke32_20160704-5811/build/src/util/generateicon.c
--- eduke32_20160704-5811.orig/build/src/util/generateicon.c	2013-10-06 03:50:16.000000000 -0400
+++ eduke32_20160704-5811/build/src/util/generateicon.c	2016-08-07 23:58:35.091530858 -0400
@@ -64,7 +64,7 @@
 
     memset(&icon, 0, sizeof(icon));
 
-    kpzload(argv[1], (intptr_t*)&icon.pixels, (int32_t*)&bpl, (int32_t*)&icon.width, (int32_t*)&icon.height);
+    kpzload(argv[1], (intptr_t*)&icon.pixels, (int32_t*)&icon.width, (int32_t*)&icon.height);
     if (!icon.pixels) {
         Bfprintf(stderr, "Failure loading %s\n", argv[1]);
         return 1;
