diff -Naur eduke32_20211102-9751-8970754aa/source/duke3d/src/common.cpp eduke32_20211102-9751-8970754aa.patched/source/duke3d/src/common.cpp
--- eduke32_20211102-9751-8970754aa/source/duke3d/src/common.cpp	2021-11-03 00:31:19.000000000 -0400
+++ eduke32_20211102-9751-8970754aa.patched/source/duke3d/src/common.cpp	2021-11-09 12:06:08.273092333 -0500
@@ -449,6 +449,7 @@
 
     if (autoload)
     {
+        G_LoadGroupsInDir("/usr/share/games/eduke32/autoload");
         G_LoadGroupsInDir("autoload");
 
         if (i != -1)
