diff -Naur eduke32_20211102-9751-8970754aa/GNUmakefile eduke32_20211102-9751-8970754aa.patched/GNUmakefile
--- eduke32_20211102-9751-8970754aa/GNUmakefile	2021-11-03 00:31:19.000000000 -0400
+++ eduke32_20211102-9751-8970754aa.patched/GNUmakefile	2021-11-07 15:41:44.198560499 -0500
@@ -439,13 +439,12 @@
 
 tools_cflags := $(engine_cflags)
 
-tools_deps := engine_tools
+tools_deps := engine_tools mimalloc
 
 tools_targets := \
     arttool \
     bsuite \
     cacheinfo \
-    generateicon \
     givedepth \
     ivfrate \
     kextract \
