diff -Naur eduke32_20211102-9751-8970754aa/source/sw/src/grpscan.cpp eduke32_20211102-9751-8970754aa.patched/source/sw/src/grpscan.cpp
--- eduke32_20211102-9751-8970754aa/source/sw/src/grpscan.cpp	2021-11-03 00:31:20.000000000 -0400
+++ eduke32_20211102-9751-8970754aa.patched/source/sw/src/grpscan.cpp	2021-11-07 15:00:27.613722670 -0500
@@ -50,7 +50,7 @@
     { "Shadow Warrior Shareware 1.1", 0xF21A6B35u, 25833456, GAMEFLAG_SHAREWARE, 0, 0, nullptr },
     { "Shadow Warrior Shareware 1.2", 0x08A7FA1Fu, 26056769, GAMEFLAG_SHAREWARE, 0, 0, nullptr },
     { "Shadow Warrior Mac Demo",      0x4227F535u, 26056769, GAMEFLAG_SHAREWARE, 0, 0, nullptr },
-    { "Wanton Destruction",           SWWD_CRC, 48698128, GAMEFLAG_SWWD, GRP_HAS_DEPENDENCY, SWREG12_CRC, nullptr },
+    { "Wanton Destruction",           SWWD_CRC, 48698128, GAMEFLAG_SWWD, 0, 0, nullptr },
     { "Twin Dragon",                  SWTD_CRC, 12499012, GAMEFLAG_SWTD, GRP_HAS_DEPENDENCY, SWREG12_CRC, process_tdragongrp },
     { "Twin Dragon",                  0xACC8DCDEu, 6235578, GAMEFLAG_SWTD, GRP_HAS_DEPENDENCY, SWREG12_CRC, nullptr },
     { "Twin Dragon",                  0xB5B71277u, 6236287, GAMEFLAG_SWTD, GRP_HAS_DEPENDENCY, SWREG12_CRC, nullptr }, // commonly available repack
