Index: megatunix-0.9.23/msloader/mtxloader.c
===================================================================
--- megatunix-0.9.23.orig/msloader/mtxloader.c
+++ megatunix-0.9.23/msloader/mtxloader.c
@@ -5,7 +5,7 @@
 #include <getfiles.h>
 #include <gtk/gtk.h>
 #include <glib/gstdio.h>
-#include <glib/gstring.h>
+#include <glib.h>
 #include <loader_common.h>
 #include <ms1_loader.h>
 #include <ms2_loader.h>
