--- ibam-0.5.2/ibam.inl	2009-03-09 19:53:20.000000000 +0300
+++ ibam-0.5.2/ibam.inl	2010-11-26 04:52:29.590364097 +0300
@@ -942,10 +942,11 @@ 
 {
    string pmu_path = "/proc/pmu";    // These strings are already in ibam.hpp,
    string acpi_path = "/proc/acpi";  // maybe a static function should check this
+   string acpi_ver_path = "/sys/module/acpi/parameters/acpica_version";
    string sysfs_path = "/sys/class/power_supply"; // ...
    ifstream pmu,acpi,sysfs;
    pmu.open((pmu_path+"/info").c_str());
-   acpi.open((acpi_path+"/info").c_str());
+   acpi.open(acpi_ver_path.c_str());
    if (pmu.is_open()) {
 #ifdef IBAM_DEBUG
       cout << "using pmu" << endl;
