diff -uNr ibam-0.5.2.ORIG//Makefile ibam-0.5.2/Makefile
--- ibam-0.5.2.ORIG//Makefile	2010-10-19 21:49:39.000000000 +0100
+++ ibam-0.5.2/Makefile	2010-10-19 21:53:27.000000000 +0100
@@ -30,12 +30,12 @@
 all:    ibam
 
 ibam:	ibam.cpp ibam.inl ibam.hpp
-	$(CC) $(CFLAGS) ibam.cpp -o ibam
+	$(CC) $(LDFLAGS) $(CFLAGS) ibam.cpp -o ibam
 
 krell:	ibam-krell.so
 
 ibam-krell.so: ibam-krell.o
-	$(CC) $(CFLAGS) -fPIC -shared -Wl -o ibam-krell.so ibam-krell.o
+	$(CC) $(LDFLAGS) $(CFLAGS) -fPIC -shared -Wl -o ibam-krell.so ibam-krell.o
 
 ibam-krell.o: ibam-krell.cpp ibam.hpp ibam.inl
 	$(CC) $(CFLAGS) -fPIC `pkg-config gtk+-2.0 --cflags` `gdk-pixbuf-config --cflags` -c ibam-krell.cpp
@@ -43,7 +43,7 @@
 krell1:	ibam-krell1.so
 
 ibam-krell1.so: ibam-krell1.o
-	$(CC) $(CFLAGS) -shared -Wl -o ibam-krell1.so ibam-krell1.o
+	$(CC) $(LDFLAGS) $(CFLAGS) -shared -Wl -o ibam-krell1.so ibam-krell1.o
 
 ibam-krell1.o: ibam-krell1.cpp ibam.hpp ibam.inl
 	$(CC) $(CFLAGS) -fPIC `gtk-config --cflags` `imlib-config --cflags-gdk` -c ibam-krell1.cpp
