diff -Nur libopensync-0.22.orig/configure libopensync-0.22/configure
--- libopensync-0.22.orig/configure	2007-03-27 07:21:55.000000000 -0500
+++ libopensync-0.22/configure	2009-08-15 01:17:55.905706460 -0500
@@ -21813,7 +21813,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   am_cv_python_pythondir=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(1,0,prefix='$PYTHON_PREFIX')" 2>/dev/null ||
-     echo "$PYTHON_PREFIX/lib/python$PYTHON_VERSION/site-packages"`
+     echo "$PYTHON_PREFIX/@baselibdir@/python$PYTHON_VERSION/site-packages"`
 fi
 { echo "$as_me:$LINENO: result: $am_cv_python_pythondir" >&5
 echo "${ECHO_T}$am_cv_python_pythondir" >&6; }
@@ -21830,7 +21830,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   am_cv_python_pyexecdir=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(1,0,prefix='$PYTHON_EXEC_PREFIX')" 2>/dev/null ||
-     echo "${PYTHON_EXEC_PREFIX}/lib/python${PYTHON_VERSION}/site-packages"`
+     echo "${PYTHON_EXEC_PREFIX}/@baselibdir@/python${PYTHON_VERSION}/site-packages"`
 fi
 { echo "$as_me:$LINENO: result: $am_cv_python_pyexecdir" >&5
 echo "${ECHO_T}$am_cv_python_pyexecdir" >&6; }
@@ -22040,7 +22040,7 @@
 	{ echo "$as_me:$LINENO: checking for Python library path" >&5
 echo $ECHO_N "checking for Python library path... $ECHO_C" >&6; }
 	python_path=`echo $PYTHON | sed "s,/bin.*$,,"`
-	for i in "$python_path/lib/python$PYTHON_VERSION/config/" "$python_path/lib/python$PYTHON_VERSION/" "$python_path/lib/python/config/" "$python_path/lib/python/" "$python_path/" ; do
+	for i in "$python_path/@baselibdir@/python$PYTHON_VERSION/config/" "$python_path/@baselibdir@/python$PYTHON_VERSION/" "$python_path/@baselibdir@/python/config/" "$python_path/@baselibdir@/python/" "$python_path/" ; do
 		python_path=`find $i -type f -name libpython$PYTHON_VERSION.* -print | sed "1q"`
 		if test -n "$python_path" ; then
 			break
