--- CMakeLists.txt.orig	2017-12-16 19:36:37.881668295 -0800
+++ CMakeLists.txt	2017-12-16 19:36:51.612529617 -0800
@@ -424,7 +424,7 @@
 install( FILES source/unix/icons/svg/nespad.svg    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/scalable/apps )
 
 # documentation
-install( FILES AUTHORS ChangeLog README.md README.unix DESTINATION ${CMAKE_INSTALL_DOCDIR} )
+install( FILES AUTHORS ChangeLog README.md DESTINATION ${CMAKE_INSTALL_DOCDIR} )
 install( FILES readme.html DESTINATION ${CMAKE_INSTALL_DOCDIR}/html )
 
 option( ENABLE_DOC "Install full documentation suite" )
