--- Makefile.old	2013-01-02 12:58:22.586959169 -0800
+++ Makefile	2013-01-02 14:56:29.603157717 -0800
@@ -26,7 +26,7 @@
 LDFLAGS += -Wl,--as-needed
 
 EXE  = nestopia
-LIBS = -lstdc++ -lm -lz -lasound $(shell sdl-config --libs) $(shell pkg-config --libs gtk+-3.0)
+LIBS = -lX11 -lstdc++ -lm -lz -lasound $(shell sdl-config --libs) $(shell pkg-config --libs gtk+-3.0)
 
 PREFIX = /usr/local
 BINDIR = $(PREFIX)/bin
@@ -41,12 +41,12 @@
 CXXFLAGS += -DDATADIR=\"$(DATADIR)\"
 
 # Linux objs
-OBJS = objs/linux/main.o objs/linux/oss.o objs/linux/interface.o objs/linux/settings.o 
+OBJS = objs/linux/main.o objs/linux/oss.o objs/linux/interface.o objs/linux/settings.o
 OBJS += objs/linux/auxio.o objs/linux/input.o objs/linux/kentry.o objs/linux/controlconfig.o objs/linux/cheats.o
 OBJS += objs/linux/seffect.o objs/linux/uihelp.o
 
 # 7-zip decoder (from LZMA SDK 4.58 beta)
-OBJS += objs/linux/7zip/7zAlloc.o objs/linux/7zip/7zBuf.o objs/linux/7zip/7zCrc.o objs/linux/7zip/7zDecode.o objs/linux/7zip/7zExtract.o 
+OBJS += objs/linux/7zip/7zAlloc.o objs/linux/7zip/7zBuf.o objs/linux/7zip/7zCrc.o objs/linux/7zip/7zDecode.o objs/linux/7zip/7zExtract.o
 OBJS += objs/linux/7zip/7zHeader.o objs/linux/7zip/7zIn.o objs/linux/7zip/7zItem.o objs/linux/7zip/LzmaDec.o
 OBJS += objs/linux/7zip/Alloc.o objs/linux/7zip/Bcj2.o objs/linux/7zip/Bra.o objs/linux/7zip/Bra86.o objs/linux/7zip/BraIA64.o
 
@@ -157,7 +157,7 @@
 OBJS += objs/core/vssystem/NstVsRbiBaseball.o  objs/core/vssystem/NstVsSuperXevious.o  objs/core/vssystem/NstVsSystem.o  objs/core/vssystem/NstVsTkoBoxing.o
 
 # object dirs
-OBJDIRS = objs objs/core objs/core/api objs/core/board objs/core/input objs/core/vssystem objs/nes_ntsc 
+OBJDIRS = objs objs/core objs/core/api objs/core/board objs/core/input objs/core/vssystem objs/nes_ntsc
 OBJDIRS += objs/linux objs/linux/7zip objs/linux/unzip
 
 # build rules
@@ -184,13 +184,13 @@
 
 install:
 	mkdir -p $(DATADIR)/icons
-	install -m 0755 $(EXE) $(BINDIR)
-	install -m 0644 source/linux/icons/nestopia.desktop $(DATADIR)
-	install -m 0644 NstDatabase.xml $(DATADIR)
+	install -Dm 0755 $(EXE) $(BINDIR)/$(EXE)
+	install -Dm 0644 source/linux/icons/nestopia.desktop $(DATADIR)/nestopia.desktop
+	install -Dm 0644 NstDatabase.xml $(DATADIR)/NstDatabase.xml
 	install -m 0644 source/linux/icons/*.png $(DATADIR)/icons
 	install -m 0644 source/linux/icons/*.svg $(DATADIR)/icons
-	install -m 0644 source/linux/icons/nestopia.svg /usr/share/pixmaps
-	xdg-desktop-menu install --novendor $(DATADIR)/nestopia.desktop
+	install -Dm 0644 source/linux/icons/nestopia.svg $(PREFIX)/share/pixmaps/nestopia.svg
+	install -Dm 0644 $(DATADIR)/nestopia.desktop $(PREFIX)/share/applications/nestopia.desktop
 
 uninstall:
 	xdg-desktop-menu uninstall $(DATADIR)/nestopia.desktop
