--- updater.ini	2014-01-08 16:38:25.957198862 +0200
+++ updater.ini	2014-01-08 16:44:42.207217074 +0200
@@ -25,17 +25,17 @@
 SaveReportsToFile=true
 AppendToPreviousFile=true
 SizeLogFileValue=1048576
-ReportFileName=report.txt
+ReportFileName=/var/log/kasp_updater.log
 DeleteIfSize=true
 DeleteIfNumDay=false
 NoChangeLogFile=false
 NumDayLifeLOgFileValue=7
 
 [DirectoriesSettings]
-MoveToCurrentFolder=true
-MoveToCustomFolder=false
-UpdatesFolder=
-TempFolder=
+MoveToCurrentFolder=false
+MoveToCustomFolder=true
+UpdatesFolder=/var/kav
+TempFolder=/var/kav
 ClearTempFolder=true
 
 [UpdatesSourceSettings]
