
Corrects the path to the latex binary on Slackware.

--- halberd-0.2.4.orig/doc/GNUmakefile	2015-11-17 19:57:03.993466727 -0700
+++ halberd-0.2.4/doc/GNUmakefile	2015-11-17 19:58:43.780459960 -0700
@@ -15,7 +15,7 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 
 
-LATEX := /usr/bin/latex
+LATEX := /usr/share/texmf/bin/latex
 DVIPDF := /usr/bin/dvipdf
 LATEX2HTML := /usr/bin/latex2html
 
