diff -rupN a/clib/unique.c b/clib/unique.c
--- a/clib/unique.c	2024-01-30 20:31:08.000000000 +0100
+++ b/clib/unique.c	2024-01-31 08:22:48.919908308 +0100
@@ -78,11 +78,7 @@ luaH_unique_new(lua_State *L)
 
     GError *error = NULL;
     if (!globalconf.application) {
-#if GTK_CHECK_VERSION(2,74,0)
-        globalconf.application = gtk_application_new(name, G_APPLICATION_DEFAULT_FLAGS);
-#else
         globalconf.application = gtk_application_new(name, G_APPLICATION_FLAGS_NONE);
-#endif
     }
 
     g_application_register(G_APPLICATION(globalconf.application), NULL, &error);
