diff -Nur xfmedia-0.9.2.orig/doc/C/Makefile.in xfmedia-0.9.2/doc/C/Makefile.in
--- xfmedia-0.9.2.orig/doc/C/Makefile.in	2006-11-27 14:20:51.000000000 -0600
+++ xfmedia-0.9.2/doc/C/Makefile.in	2009-08-15 17:43:00.612828573 -0500
@@ -59,7 +59,7 @@
     *) f=$$p;; \
   esac;
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
-am__installdirs = "$(DESTDIR)$(docdir)" "$(DESTDIR)$(imagesdir)"
+am__installdirs = "$(DESTDIR)$(htmldir)" "$(DESTDIR)$(imagesdir)"
 docDATA_INSTALL = $(INSTALL_DATA)
 imagesDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(doc_DATA) $(images_DATA)
@@ -265,7 +265,8 @@
 build_vendor = @build_vendor@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(datadir)/xfmedia/doc/C
+docdir = @docdir@
+htmldir = @htmldir@
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
@@ -307,7 +308,7 @@
 	html/ch04.html \
 	html/ch04s02.html
 
-imagesdir = $(datadir)/xfmedia/doc/C/images
+imagesdir = $(htmldir)/images
 images_DATA = \
 	xml/images/mainwin.png \
 	xml/images/prefs-options.png \
@@ -361,20 +362,20 @@
 uninstall-info-am:
 install-docDATA: $(doc_DATA)
 	@$(NORMAL_INSTALL)
-	test -z "$(docdir)" || $(mkdir_p) "$(DESTDIR)$(docdir)"
+	test -z "$(htmldir)" || $(mkdir_p) "$(DESTDIR)$(htmldir)"
 	@list='$(doc_DATA)'; for p in $$list; do \
 	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
 	  f=$(am__strip_dir) \
-	  echo " $(docDATA_INSTALL) '$$d$$p' '$(DESTDIR)$(docdir)/$$f'"; \
-	  $(docDATA_INSTALL) "$$d$$p" "$(DESTDIR)$(docdir)/$$f"; \
+	  echo " $(docDATA_INSTALL) '$$d$$p' '$(DESTDIR)$(htmldir)/$$f'"; \
+	  $(docDATA_INSTALL) "$$d$$p" "$(DESTDIR)$(htmldir)/$$f"; \
 	done
 
 uninstall-docDATA:
 	@$(NORMAL_UNINSTALL)
 	@list='$(doc_DATA)'; for p in $$list; do \
 	  f=$(am__strip_dir) \
-	  echo " rm -f '$(DESTDIR)$(docdir)/$$f'"; \
-	  rm -f "$(DESTDIR)$(docdir)/$$f"; \
+	  echo " rm -f '$(DESTDIR)$(htmldir)/$$f'"; \
+	  rm -f "$(DESTDIR)$(htmldir)/$$f"; \
 	done
 install-imagesDATA: $(images_DATA)
 	@$(NORMAL_INSTALL)
@@ -574,7 +575,7 @@
 all-am: Makefile $(DATA)
 installdirs: installdirs-recursive
 installdirs-am:
-	for dir in "$(DESTDIR)$(docdir)" "$(DESTDIR)$(imagesdir)"; do \
+	for dir in "$(DESTDIR)$(htmldir)" "$(DESTDIR)$(imagesdir)"; do \
 	  test -z "$$dir" || $(mkdir_p) "$$dir"; \
 	done
 install: install-recursive
diff -Nur xfmedia-0.9.2.orig/doc/Makefile.in xfmedia-0.9.2/doc/Makefile.in
--- xfmedia-0.9.2.orig/doc/Makefile.in	2006-11-27 14:20:52.000000000 -0600
+++ xfmedia-0.9.2/doc/Makefile.in	2009-08-15 17:41:11.197985898 -0500
@@ -295,7 +295,7 @@
 SUBDIRS = \
 	C
 
-cssdir = $(datadir)/xfmedia/doc
+cssdir = $(htmldir)
 css_DATA = xfce.css
 noinst_DATA = \
 	catalog.xml.in \
