--- a/modules/access/dvdnav.c	2020-04-07 23:35:05.000000000 +1000
+++ b/modules/access/dvdnav.c	2020-04-16 22:32:34.175000000 +1000
@@ -59,6 +59,10 @@
 
 
 #include <dvdnav/dvdnav.h>
+/* libdvdnav-5.0.3 doesn't define DVDNAV_VERSION */
+#ifndef DVDNAV_VERSION
+#define DVDNAV_VERSION (50003)
+#endif
 /* Expose without patching headers */
 dvdnav_status_t dvdnav_jump_to_sector_by_time(dvdnav_t *, uint64_t, int32_t);
 
