--- a/modules/access/dvdread.c	2020-03-28 23:39:33.000000000 +1000
+++ b/modules/access/dvdread.c	2020-04-16 22:10:43.065000000 +1000
@@ -55,6 +55,12 @@
 #include <unistd.h>
 
 #include <dvdread/dvd_reader.h>
+#ifndef DVDREAD_VERSION_CODE
+#define DVDREAD_VERSION_CODE(major, minor, micro) \
+    (((major) * 10000) +                          \
+     ((minor) *   100) +                          \
+     ((micro) *     1))
+#endif
 #include <dvdread/ifo_types.h>
 #include <dvdread/ifo_read.h>
 #include <dvdread/nav_read.h>
