--- modules/access/sftp.c.orig	2018-06-26 05:59:03.000000000 +1000
+++ modules/access/sftp.c	2019-03-23 16:20:21.647000000 +1000
@@ -306,7 +306,7 @@
         case LIBSSH2_HOSTKEY_TYPE_DSS:
             knownhost_fingerprint_algo = LIBSSH2_KNOWNHOST_KEY_SSHDSS;
             break;
-#if LIBSSH2_VERSION_NUM >= 0x010801
+#if LIBSSH2_VERSION_NUM >= 0x010900
         case LIBSSH2_HOSTKEY_TYPE_ECDSA_256:
             knownhost_fingerprint_algo = LIBSSH2_KNOWNHOST_KEY_ECDSA_256;
             break;
