--- configure.ac.orig	2015-02-27 06:25:58.000000000 +1000
+++ configure.ac	2015-06-02 23:10:53.544894508 +1000
@@ -2243,8 +2243,8 @@
 [  --enable-avcodec        libavcodec codec (default enabled)])
 AS_IF([test "${enable_avcodec}" != "no"], [
   PKG_CHECK_MODULES(AVCODEC,[libavcodec >= 53.34.0 libavutil >= 51.22.0], [
-    PKG_CHECK_EXISTS([libavcodec < 56],, [
-      AC_MSG_ERROR([libavcodec versions 56 and later are not supported yet.])
+    PKG_CHECK_EXISTS([libavcodec < 57],, [
+      AC_MSG_ERROR([libavcodec versions 57 and later are not supported yet.])
     ])
     VLC_SAVE_FLAGS
     CPPFLAGS="${CPPFLAGS} ${AVCODEC_CFLAGS}"
