--- bin/Makefile.am.orig	2015-06-03 10:49:51.266282736 +1000
+++ bin/Makefile.am	2015-06-03 10:50:20.469282310 +1000
@@ -72,6 +72,10 @@
 vlc_cache_gen_DEPENDENCIES = vlc_win32_rc.$(OBJEXT)
 endif
 
+if HAVE_GOBJECT
+vlc_cache_gen_LDADD += $(GOBJECT_LIBS)
+endif
+
 #
 # Plug-ins cache
 #
--- configure.ac.orig	2015-06-03 10:53:22.299279657 +1000
+++ configure.ac	2015-06-03 09:44:33.338339889 +1000
@@ -762,6 +762,12 @@
     ])
 ])
 
+dnl
+dnl Check for gobject
+dnl
+have_gobject="no"
+PKG_CHECK_MODULES([GOBJECT], [gobject-2.0], [ have_gobject="yes" ])
+AM_CONDITIONAL(HAVE_GOBJECT, [ test "${have_gobject}" = "yes" ])
 
 dnl
 dnl Check for zlib.h and -lz along with system -lminizip if available
