--- configure.ac.prev	2019-03-10 13:15:05.033000000 +1000
+++ configure.ac	2019-03-10 13:19:03.840000000 +1000
@@ -1848,7 +1848,7 @@
 dnl
 dnl  OpenCV wrapper and example filters
 dnl
-PKG_ENABLE_MODULES_VLC([OPENCV], [opencv_example opencv_wrapper], [opencv > 2.0], (OpenCV (computer vision) filter), [auto])
+PKG_ENABLE_MODULES_VLC([OPENCV], [opencv_wrapper], [opencv4 > 4.0], (OpenCV (computer vision) filter), [auto])
 
 
 dnl
--- modules/video_filter/Makefile.am.orig	2017-11-25 01:29:18.000000000 +1000
+++ modules/video_filter/Makefile.am	2019-03-12 14:21:25.417000000 +1000
@@ -147,8 +147,8 @@
 libdeinterlace_plugin_la_LIBADD = libdeinterlace_common.la
 video_filter_LTLIBRARIES += libdeinterlace_plugin.la
 
-libopencv_wrapper_plugin_la_SOURCES = video_filter/opencv_wrapper.c
-libopencv_wrapper_plugin_la_CPPFLAGS = $(AM_CPPFLAGS) $(OPENCV_CFLAGS)
+libopencv_wrapper_plugin_la_SOURCES = video_filter/opencv_wrapper.cpp
+libopencv_wrapper_plugin_la_CPPFLAGS = $(AM_CPPFLAGS) $(OPENCV_CFLAGS) -fpermissive
 libopencv_wrapper_plugin_la_LIBADD = $(OPENCV_LIBS)
 libopencv_wrapper_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(video_filterdir)'
 video_filter_LTLIBRARIES += $(LTLIBopencv_wrapper)
