--- modules/visualization/projectm.cpp.orig	2013-09-13 01:18:34.000000000 +1000
+++ modules/visualization/projectm.cpp	2014-09-04 14:53:21.510242036 +1000
@@ -79,8 +79,8 @@
 # define FONT_PATH_MENU "C:\\WINDOWS\\Fonts\\arial.ttf"
 # define PRESET_PATH    NULL
 #else
-# define FONT_PATH      "/usr/share/fonts/truetype/ttf-dejavu/DejaVuSans.ttf"
-# define FONT_PATH_MENU "/usr/share/fonts/truetype/ttf-dejavu/DejaVuSansMono.ttf"
+# define FONT_PATH      "/usr/share/fonts/TTF/DejaVuSans.ttf"
+# define FONT_PATH_MENU "/usr/share/fonts/TTF/DejaVuSansMono.ttf"
 # define PRESET_PATH    "/usr/share/projectM/presets"
 #endif
 
