diff -up glew-1.9.0/include/GL/glxew.h.orig glew-1.9.0/include/GL/glxew.h
--- glew-1.9.0/include/GL/glxew.h.orig	2012-08-06 17:59:08.000000000 +0200
+++ glew-1.9.0/include/GL/glxew.h	2013-05-23 17:29:09.712473195 +0200
@@ -1393,7 +1393,7 @@ typedef int ( * PFNGLXVIDEORESIZESUNPROC
 /* ------------------------------------------------------------------------- */
 
 #ifdef GLEW_MX
-#define GLXEW_FUN_EXPORT
+#define GLXEW_FUN_EXPORT GLEW_FUN_EXPORT
 #define GLXEW_VAR_EXPORT
 #else
 #define GLXEW_FUN_EXPORT GLEW_FUN_EXPORT

