diff -Naur ranger-master.orig/Makefile ranger-master/Makefile
--- ranger-master.orig/Makefile	2010-04-25 23:41:15.000000000 +0100
+++ ranger-master/Makefile	2010-04-26 03:42:48.000000000 +0100
@@ -53,10 +53,12 @@
 		false; \
 	fi
 	@echo "Installing..."
-	cp ranger.py $(PREFIX)/bin/ranger
-	cp -ruT ranger $(DEST)
-	chmod 755 $(PREFIX)/bin/ranger
-	chmod -R +rX $(DEST)
+	mkdir -p $(DESTDIR)$(PREFIX)/bin
+	mkdir -p $(DESTDIR)$(DEST)
+	cp ranger.py $(DESTDIR)$(PREFIX)/bin/ranger
+	cp -ruT ranger $(DESTDIR)$(DEST)
+	chmod 755 $(DESTDIR)$(PREFIX)/bin/ranger
+	chmod -R +rX $(DESTDIR)$(DEST)
 	@echo '--------------------------------------'
 	@echo 'Finished.'
 	@echo 'If you use BASH or ZSH, you can activate an extra feature now:'
