--- /var/lib/sbopkg/r0t/14.2/games/nethack/nethack.SlackBuild	2019-06-08 22:37:23.000000000 +0000
+++ nethack.SlackBuild	2020-03-26 02:19:34.386984395 +0000
@@ -25,7 +25,8 @@
 # ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 PRGNAM=nethack
-VERSION=${VERSION:-3.6.2}
+VERSION=${VERSION:-3.6.6}
+SRCDIR=NetHack-NetHack-${VERSION}_Released
 SRCVERSION=$(echo $VERSION | sed "s,\.,,g")
 BUILD=${BUILD:-1}
 TAG=${TAG:-_SBo}
@@ -63,9 +64,9 @@
 rm -rf $PKG
 mkdir -p $TMP $PKG $OUTPUT
 cd $TMP
-rm -rf $PRGNAM-$VERSION
+rm -rf $SRCDIR
 tar xvf $CWD/$PRGNAM-$SRCVERSION-src.tgz
-cd $PRGNAM-$VERSION
+cd $SRCDIR
 chown -R root:root .
 find -L . \
  \( -perm 777 -o -perm 775 -o -perm 750 -o -perm 711 -o -perm 555 -o -perm 511 \) \
@@ -97,7 +98,7 @@
 cd sys/unix
 sh setup.sh hints/linux
 cd ../..
-make -j1 install PREFIX=$PKG
+make -j1 install PREFIX=$PKG WINTTYLIB=-lncurses
 mkdir -p $PKG/usr/man/man6
 make PREFIX=$PKG -j1 install manpages # Multi-threaded builds fail.
 
