diff -Naur -x '.*.swp' frotz-2.52/src/sdl/sf_util.c frotz-2.52.patched/src/sdl/sf_util.c
--- frotz-2.52/src/sdl/sf_util.c	2020-05-10 19:51:48.000000000 -0400
+++ frotz-2.52.patched/src/sdl/sf_util.c	2020-12-16 18:33:26.144268039 -0500
@@ -380,7 +380,6 @@
 static char *new_dirname(const char *path)
 {
 	char *p = strdup(path), *p2 = strdup(dirname(p));
-	free(p);
 	return p2;
 }
 
@@ -392,7 +391,6 @@
 static char *new_basename(const char *path)
 {
 	char *p = strdup(path), *p2 = strdup(basename(p));
-	free(p);
 	return p2;
 }
 
