--- CEGUI-0.7.9/cegui/include/RendererModules/Ogre/CEGUIOgreRenderer.h.orig	2016-02-06 11:05:31.177684112 +0700
+++ CEGUI-0.7.9/cegui/include/RendererModules/Ogre/CEGUIOgreRenderer.h	2016-02-06 11:06:00.183625951 +0700
@@ -54,7 +54,9 @@
 class Root;
 class RenderSystem;
 class RenderTarget;
-class TexturePtr;
+class Texture;
+template<typename T> class SharedPtr;
+typedef SharedPtr<Texture> TexturePtr;
 class Matrix4;
 }
 
