--- goonies-1.4.1528/build/linux/Makefile.orig	2015-03-26 04:24:44.994794899 +0200
+++ goonies-1.4.1528/build/linux/Makefile	2015-03-26 05:08:09.457234638 +0200
@@ -3,11 +3,11 @@
 PREFIX = /usr
 EXTRADIR = ./build/linux
 
-GAMEDIR = $(PREFIX)/games
+GAMEDIR = $(PREFIX)/usr/bin
 STARTUP = $(GAMEDIR)/$(EXE)
-BINDIR = $(PREFIX)/share/games/$(EXE)
-ICNDIR = $(PREFIX)/share/pixmaps
-APPDIR = $(PREFIX)/share/applications
+BINDIR = $(PREFIX)/usr/share/games/$(EXE)
+ICNDIR = $(PREFIX)/usr/share/pixmaps
+APPDIR = $(PREFIX)/usr/share/applications
 
 DATA = graphics maps sound
 ICON = goonies.png
@@ -58,7 +58,7 @@
 
 CC = gcc
 CFLAGS = -g3 -O3 -Wno-write-strings `sdl-config --cflags` -I/usr/X11R6/include
-LDFLAGS = `sdl-config --libs` -L/usr/X11R6/lib/ -lSDL_image -lSDL_mixer -lGL -lGLU
+LDFLAGS = `sdl-config --libs` -L/usr/X11R6/lib/ -lSDL_image -lSDL_mixer -lGL -lGLU -lm -lstdc++
 RM = rm -f
 CP = cp -r
 MD = mkdir -p
