From 9149e5b11bb25fbbdb5c1d7ad58c9a06dbf9f26f Mon Sep 17 00:00:00 2001
From: Fabio Valentini <decathorpe@gmail.com>
Date: Thu, 4 Feb 2016 12:24:43 +0100
Subject: [PATCH] src/develop/masks/path.c: fix int -> float (compile error w/
 GCC6)

---
 src/develop/masks/path.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/develop/masks/path.c b/src/develop/masks/path.c
index 62c0095..f08dd4c 100644
--- a/src/develop/masks/path.c
+++ b/src/develop/masks/path.c
@@ -2555,8 +2555,8 @@ static int dt_path_get_mask_roi(dt_iop_module_t *module, dt_dev_pixelpipe_iop_t
   // now check if feather is at least partially within roi
   for(int i = nb_corner * 3; i < border_count; i++)
   {
-    int xx = border[i * 2];
-    int yy = border[i * 2 + 1];
+    float xx = border[i * 2];
+    float yy = border[i * 2 + 1];
     if(isnan(xx))
     {
       if(isnan(yy)) break; // that means we have to skip the end of the border path
