diff -u -r darktable-0.9.2//cmake/modules/FindGTK2.cmake darktable-0.9.2-patched//cmake/modules/FindGTK2.cmake
--- darktable-0.9.2//cmake/modules/FindGTK2.cmake	2011-08-24 10:20:29.000000000 +0100
+++ darktable-0.9.2-patched//cmake/modules/FindGTK2.cmake	2011-08-29 21:10:41.758237394 +0100
@@ -170,8 +170,6 @@
  /usr/lib/i386-linux-gnu/
  /usr/lib/x86_64-linux-gnu/
  # end fix for Ubuntu >= 11.04 (Natty Narwhal)
-        # look in debian/ubuntu multi-arch path (needs cmake >= 2.85~rc2)
-        /usr/lib/${CMAKE_LIBRARY_ARCHITECTURE}
             /usr/local/lib64
             /usr/local/lib
             /usr/lib64
diff -u -r darktable-0.9.2//cmake/modules/FindGlib.cmake darktable-0.9.2-patched//cmake/modules/FindGlib.cmake
--- darktable-0.9.2//cmake/modules/FindGlib.cmake	2011-08-29 21:55:53.743097178 +0100
+++ darktable-0.9.2-patched//cmake/modules/FindGlib.cmake	2011-08-29 21:55:09.927007977 +0100
@@ -21,7 +21,7 @@
 find_path(GlibConfig_INCLUDE_DIR
   NAMES glibconfig.h
   PATHS ${Glib_PKGCONF_INCLUDE_DIRS} /usr
-  PATH_SUFFIXES lib/glib-2.0/include ../lib/glib-2.0/include
+  PATH_SUFFIXES lib${LIB_SUFFIX}/glib-2.0/include ../lib${LIB_SUFFIX}/glib-2.0/include
 )
 
 # Finally the library itself
