diff -Naur darktable-0.7.orig/src/common/exif.cc
darktable-0.7/src/common/exif.cc
--- darktable-0.7.orig/src/common/exif.cc 2010-11-20 12:35:27.000000000 +0100
+++ darktable-0.7/src/common/exif.cc 2011-01-03 19:05:34.242131001 +0100
@@ -29,11 +29,7 @@
 #include "common/imageio.h"
 }
 // #include <libexif/exif-data.h>
-#include <exiv2/xmp.hpp>
-#include <exiv2/error.hpp>
-#include <exiv2/image.hpp>
-#include <exiv2/exif.hpp>
-#include <exiv2/canonmn.hpp>
+#include <exiv2/exiv2.hpp>
 #include <sqlite3.h>
 #include <iostream>
 #include <fstream>
