diff -Nur darktable-2.0.0.orig/cmake/modules/FindOpenJPEG.cmake darktable-2.0.0/cmake/modules/FindOpenJPEG.cmake
--- darktable-2.0.0.orig/cmake/modules/FindOpenJPEG.cmake	2015-12-22 07:56:37.000000000 -0600
+++ darktable-2.0.0/cmake/modules/FindOpenJPEG.cmake	2016-01-09 17:19:38.605300279 -0600
@@ -29,7 +29,7 @@
 
 include(LibFindMacros)
 
-libfind_pkg_check_modules(PC_OPENJPEG libopenjpeg1)
+libfind_pkg_check_modules(PC_OPENJPEG libopenmj2)
 
 SET(_openjpeg_SEARCH_DIRS
   ${OPENJPEG_ROOT_DIR}
@@ -52,7 +52,7 @@
 
 FIND_LIBRARY(OPENJPEG_LIBRARY
   NAMES
-    openjpeg
+    openmj2
   HINTS
     ${PC_OPENJPEG_LIBDIR}
     ${PC_OPENJPEG_LIBRARY_DIRS}
