From 22cb2d442fecf4dd8122347309dbf27ad79e5559 Mon Sep 17 00:00:00 2001
From: Petri Hintukainen <phintuka@gmail.com>
Date: Fri, 11 Mar 2016 12:04:34 +0200
Subject: [PATCH 2/5] Improve error reporting when cache directory creation
 fails

---
 src/libdvdcss.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libdvdcss.c b/src/libdvdcss.c
index 7d86a08..408ec2f 100644
--- a/src/libdvdcss.c
+++ b/src/libdvdcss.c
@@ -305,7 +305,7 @@ static int init_cache_dir( dvdcss_t dvdcss )
     i_ret = mkdir( dvdcss->psz_cachefile, 0755 );
     if( i_ret < 0 && errno != EEXIST )
     {
-        print_error( dvdcss, "failed creating cache directory" );
+        print_error( dvdcss, "failed creating cache directory '%s'", dvdcss->psz_cachefile );
         dvdcss->psz_cachefile[0] = '\0';
         return -1;
     }
-- 
2.10.0

