From 5bec036cfee3950d002f3c46c17bbd4e375d5bd6 Mon Sep 17 00:00:00 2001
From: Hannes Domani <ssbssa@yahoo.de>
Date: Thu, 25 Feb 2016 21:03:29 +0100
Subject: [PATCH 5/5] Don't close any handle if callback functions are used

Signed-off-by: Jean-Baptiste Kempf <jb@videolan.org>
---
 src/device.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/device.c b/src/device.c
index af735e0..baa4bc5 100644
--- a/src/device.c
+++ b/src/device.c
@@ -406,6 +406,11 @@ int dvdcss_open_device ( dvdcss_t dvdcss )
 
 int dvdcss_close_device ( dvdcss_t dvdcss )
 {
+    if( dvdcss->p_stream )
+    {
+        return 0;
+    }
+
 #if defined( _WIN32 )
     /* Free readv temporary buffer */
     free( dvdcss->p_readv_buffer );
-- 
2.10.0

