diff -Naur a/po/bs/CMakeLists.txt b/po/bs/CMakeLists.txt
--- a/po/bs/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/bs/CMakeLists.txt	2015-11-04 14:08:11.439085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(bs ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/ca/CMakeLists.txt b/po/ca/CMakeLists.txt
--- a/po/ca/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/ca/CMakeLists.txt	2015-11-04 14:08:11.501085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(ca ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/ca@valencia/CMakeLists.txt b/po/ca@valencia/CMakeLists.txt
--- a/po/ca@valencia/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/ca@valencia/CMakeLists.txt	2015-11-04 14:08:11.494085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(ca@valencia ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/cs/CMakeLists.txt b/po/cs/CMakeLists.txt
--- a/po/cs/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/cs/CMakeLists.txt	2015-11-04 14:08:11.334085810 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(cs ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/da/CMakeLists.txt b/po/da/CMakeLists.txt
--- a/po/da/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/da/CMakeLists.txt	2015-11-04 14:08:11.543085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(da ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/de/CMakeLists.txt b/po/de/CMakeLists.txt
--- a/po/de/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/de/CMakeLists.txt	2015-11-04 14:08:11.384085810 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(de ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/el/CMakeLists.txt b/po/el/CMakeLists.txt
--- a/po/el/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/el/CMakeLists.txt	2015-11-04 14:08:11.377085810 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(el ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/es/CMakeLists.txt b/po/es/CMakeLists.txt
--- a/po/es/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/es/CMakeLists.txt	2015-11-04 14:08:11.445085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(es ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/et/CMakeLists.txt b/po/et/CMakeLists.txt
--- a/po/et/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/et/CMakeLists.txt	2015-11-04 14:08:11.353085810 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(et ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/fi/CMakeLists.txt b/po/fi/CMakeLists.txt
--- a/po/fi/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/fi/CMakeLists.txt	2015-11-04 14:08:11.457085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(fi ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/ga/CMakeLists.txt b/po/ga/CMakeLists.txt
--- a/po/ga/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/ga/CMakeLists.txt	2015-11-04 14:08:11.507085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(ga ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/gl/CMakeLists.txt b/po/gl/CMakeLists.txt
--- a/po/gl/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/gl/CMakeLists.txt	2015-11-04 14:08:11.477085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(gl ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/he/CMakeLists.txt b/po/he/CMakeLists.txt
--- a/po/he/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/he/CMakeLists.txt	2015-11-04 14:08:11.408085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(he ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/hr/CMakeLists.txt b/po/hr/CMakeLists.txt
--- a/po/hr/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/hr/CMakeLists.txt	2015-11-04 14:08:11.346085810 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(hr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/hu/CMakeLists.txt b/po/hu/CMakeLists.txt
--- a/po/hu/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/hu/CMakeLists.txt	2015-11-04 14:08:11.488085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(hu ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/it/CMakeLists.txt b/po/it/CMakeLists.txt
--- a/po/it/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/it/CMakeLists.txt	2015-11-04 14:08:11.390085810 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(it ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/ja/CMakeLists.txt b/po/ja/CMakeLists.txt
--- a/po/ja/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/ja/CMakeLists.txt	2015-11-04 14:08:11.402085810 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(ja ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/lt/CMakeLists.txt b/po/lt/CMakeLists.txt
--- a/po/lt/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/lt/CMakeLists.txt	2015-11-04 14:08:11.365085810 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(lt ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/mr/CMakeLists.txt b/po/mr/CMakeLists.txt
--- a/po/mr/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/mr/CMakeLists.txt	2015-11-04 14:08:11.414085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(mr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/nb/CMakeLists.txt b/po/nb/CMakeLists.txt
--- a/po/nb/CMakeLists.txt	2014-05-16 14:46:42.000000000 +0400
+++ b/po/nb/CMakeLists.txt	2015-11-04 14:08:11.371085810 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(nb ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/nds/CMakeLists.txt b/po/nds/CMakeLists.txt
--- a/po/nds/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/nds/CMakeLists.txt	2015-11-04 14:08:11.420085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(nds ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/nl/CMakeLists.txt b/po/nl/CMakeLists.txt
--- a/po/nl/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/nl/CMakeLists.txt	2015-11-04 14:08:11.470085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(nl ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/pl/CMakeLists.txt b/po/pl/CMakeLists.txt
--- a/po/pl/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/pl/CMakeLists.txt	2015-11-04 14:08:11.427085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(pl ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/pt/CMakeLists.txt b/po/pt/CMakeLists.txt
--- a/po/pt/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/pt/CMakeLists.txt	2015-11-04 14:08:11.463085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(pt ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/pt_BR/CMakeLists.txt b/po/pt_BR/CMakeLists.txt
--- a/po/pt_BR/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/pt_BR/CMakeLists.txt	2015-11-04 14:08:11.531085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(pt_BR ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/ro/CMakeLists.txt b/po/ro/CMakeLists.txt
--- a/po/ro/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/ro/CMakeLists.txt	2015-11-04 14:08:11.525085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(ro ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/ru/CMakeLists.txt b/po/ru/CMakeLists.txt
--- a/po/ru/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/ru/CMakeLists.txt	2015-11-04 14:08:11.451085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(ru ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/sk/CMakeLists.txt b/po/sk/CMakeLists.txt
--- a/po/sk/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/sk/CMakeLists.txt	2015-11-04 14:08:11.483085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(sk ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/sl/CMakeLists.txt b/po/sl/CMakeLists.txt
--- a/po/sl/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/sl/CMakeLists.txt	2015-11-04 14:08:11.359085810 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(sl ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/sv/CMakeLists.txt b/po/sv/CMakeLists.txt
--- a/po/sv/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/sv/CMakeLists.txt	2015-11-04 14:08:11.537085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(sv ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/tr/CMakeLists.txt b/po/tr/CMakeLists.txt
--- a/po/tr/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/tr/CMakeLists.txt	2015-11-04 14:08:11.340085810 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(tr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/ug/CMakeLists.txt b/po/ug/CMakeLists.txt
--- a/po/ug/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/ug/CMakeLists.txt	2015-11-04 14:08:11.513085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(ug ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/uk/CMakeLists.txt b/po/uk/CMakeLists.txt
--- a/po/uk/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/uk/CMakeLists.txt	2015-11-04 14:08:11.433085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(uk ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/zh_CN/CMakeLists.txt b/po/zh_CN/CMakeLists.txt
--- a/po/zh_CN/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/zh_CN/CMakeLists.txt	2015-11-04 14:08:11.328085810 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(zh_CN ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
diff -Naur a/po/zh_TW/CMakeLists.txt b/po/zh_TW/CMakeLists.txt
--- a/po/zh_TW/CMakeLists.txt	2014-05-16 14:46:43.000000000 +0400
+++ b/po/zh_TW/CMakeLists.txt	2015-11-04 14:08:11.519085811 +0400
@@ -1,2 +1,2 @@
 file(GLOB _po_files *.po)
-GETTEXT_PROCESS_PO_FILES(fr ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
+GETTEXT_PROCESS_PO_FILES(zh_TW ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )
