--- src/xoper.f	2013-12-08 13:21:22.000000000 -0500
+++ src.xoper.f.new	2016-09-15 20:50:03.000000000 -0400
@@ -3064,32 +3064,32 @@
         ione = int( (fnum-float(10*iten)) / 0.99999 )
         idec = int( (fnum-float(10*iten)-float(ione)) / 0.09999 )
 
-        fname = char(iten+izero) 
-     &       // char(ione+izero) 
-     &       // char(idec+izero) // '.bl'
-        lu = 44
-        open(lu,file=fname,status='unknown')
-        rewind(lu)
-        write(lu,'(a,a)') 
-     &'#       s         ue          H          P         K ',
-     &'        x    -m du/dx'
-c       1234567890 1234567890 1234567890 1234567890 1234567890 1234567890
-        do ibl = 2, iblte(is)
-          iblm = max( ibl-1 , 2 )
-          iblp = min( ibl+1 , iblte(is) )
-          i  = ipan(ibl ,is)
-          hk = dstr(ibl,is) / thet(ibl,is)
-          ddef = dstr(ibl,is)*uedg(ibl,is)
-          pdef = thet(ibl,is)*uedg(ibl,is)**2
-          edef = tstr(ibl,is)*uedg(ibl,is)**3 * 0.5
-          duds = (uedg(iblp,is)-uedg(iblm,is))
-     &         / (xssi(iblp,is)-xssi(iblm,is))
-          dpds = -ddef*duds
-          write(lu,9977) 
-     &       xssi(ibl,is), uedg(ibl,is), hk, pdef, edef, x(i), dpds
- 9977     format(1x, 3f11.4, 2f11.6, f11.3, e14.6 )
-        enddo
-        close(lu)
+c        fname = char(iten+izero) 
+c     &       // char(ione+izero) 
+c     &       // char(idec+izero) // '.bl'
+c        lu = 44
+c        open(lu,file=fname,status='unknown')
+c        rewind(lu)
+c        write(lu,'(a,a)') 
+c     &'#       s         ue          H          P         K ',
+c     &'        x    -m du/dx'
+cc       1234567890 1234567890 1234567890 1234567890 1234567890 1234567890
+c        do ibl = 2, iblte(is)
+c          iblm = max( ibl-1 , 2 )
+c          iblp = min( ibl+1 , iblte(is) )
+c          i  = ipan(ibl ,is)
+c          hk = dstr(ibl,is) / thet(ibl,is)
+c          ddef = dstr(ibl,is)*uedg(ibl,is)
+c          pdef = thet(ibl,is)*uedg(ibl,is)**2
+c          edef = tstr(ibl,is)*uedg(ibl,is)**3 * 0.5
+c          duds = (uedg(iblp,is)-uedg(iblm,is))
+c     &         / (xssi(iblp,is)-xssi(iblm,is))
+c          dpds = -ddef*duds
+c          write(lu,9977) 
+c     &       xssi(ibl,is), uedg(ibl,is), hk, pdef, edef, x(i), dpds
+c 9977     format(1x, 3f11.4, 2f11.6, f11.3, e14.6 )
+c        enddo
+c        close(lu)
 
 
 
