--- crystal-2.2.1/client/crystalclient.cpp.orig	2016-02-15 07:32:55.748512951 +0700
+++ crystal-2.2.1//client/crystalclient.cpp	2016-02-15 07:33:14.018499876 +0700
@@ -118,7 +118,7 @@
 	if (isPreview()) {
 		char c[512];
 		QLabel *label;
-		sprintf(c,"<center><b>Crystal %s Preview</b><br>Built: %s</center>",VERSION, __DATE__);
+		sprintf(c,"<center><b>Crystal %s Preview</b><br>Built: %s</center>",2.2, __DATE__);
 		mainlayout->addItem(new QSpacerItem(1, 1,QSizePolicy::Expanding,QSizePolicy::Fixed), 0, 1);
 		mainlayout->addItem(new QSpacerItem(1, ::factory->borderwidth,QSizePolicy::Expanding,QSizePolicy::Expanding), 3, 1);
 		label = new QLabel(i18n(c), widget());
