commit 87e11a7e57ec734059924e192656fc70d1ce3f50
Author: orbea <orbea@riseup.net>
Date:   Mon Jun 8 10:29:26 2020 -0700

    Fix build with newer glslang.

diff --git a/src/glsl/glslang.cc b/src/glsl/glslang.cc
index 5546120..07cf8f1 100644
--- a/src/glsl/glslang.cc
+++ b/src/glsl/glslang.cc
@@ -210,6 +210,7 @@ const TBuiltInResource DefaultTBuiltInResource = {
     /* .maxTaskWorkGroupSizeY_NV = */ 1,
     /* .maxTaskWorkGroupSizeZ_NV = */ 1,
     /* .maxMeshViewCountNV = */ 4,
+    /* .maxDualSourceDrawBuffersEXT = */ 1,
 #endif
 
     /* .limits = */ {
