diff -ur libplacebo-v1.18.0.orig/src/meson.build libplacebo-v1.18.0/src/meson.build
--- libplacebo-v1.18.0.orig/src/meson.build	2019-02-14 10:11:33.000000000 -0800
+++ libplacebo-v1.18.0/src/meson.build	2019-08-09 07:23:04.838656296 -0700
@@ -63,6 +63,8 @@
     cxx.find_library('OGLCompiler', required: glslang_req),
     cxx.find_library('OSDependent', required: glslang_req),
     cxx.find_library('SPIRV',       required: glslang_req),
+    cxx.find_library('SPIRV-Tools', required: glslang_req),
+    cxx.find_library('SPIRV-Tools-opt', required: glslang_req),
     cxx.find_library('SPVRemapper', required: glslang_req),
   ]
 
