diff -Naur xrdp-v0.6.0-orig/sesman/sesman.ini xrdp-v0.6.0/sesman/sesman.ini
--- xrdp-v0.6.0-orig/sesman/sesman.ini	2012-02-21 14:00:55.000000000 -0500
+++ xrdp-v0.6.0/sesman/sesman.ini	2013-09-07 18:48:37.377849867 -0400
@@ -2,11 +2,11 @@
 ListenAddress=127.0.0.1
 ListenPort=3350
 EnableUserWindowManager=1
-UserWindowManager=startwm.sh
-DefaultWindowManager=startwm.sh
+UserWindowManager=.xrdp-xinitrc
+DefaultWindowManager=xrdp-xinitrc
 
 [Security]
-AllowRootLogin=1
+AllowRootLogin=0
 MaxLoginRetry=4
 TerminalServerUsers=tsusers
 TerminalServerAdmins=tsadmins
@@ -20,7 +20,7 @@
 
 [Logging]
 LogFile=/var/log/xrdp-sesman.log
-LogLevel=DEBUG
+LogLevel=INFO
 EnableSyslog=0
 SyslogLevel=DEBUG
 
