diff -Nur xfce4-screenshooter-1.7.9.orig//Makefile.in xfce4-screenshooter-1.7.9/Makefile.in
--- xfce4-screenshooter-1.7.9.orig//Makefile.in	2010-02-07 07:45:15.000000000 -0600
+++ xfce4-screenshooter-1.7.9/Makefile.in	2010-05-23 01:04:12.713542580 -0500
@@ -91,8 +91,7 @@
 	lib/lib_libscreenshooter_la-screenshooter-dialogs.lo \
 	lib/lib_libscreenshooter_la-screenshooter-job.lo \
 	lib/lib_libscreenshooter_la-screenshooter-simple-job.lo \
-	lib/lib_libscreenshooter_la-screenshooter-utils.lo \
-	lib/lib_libscreenshooter_la-screenshooter-zimagez.lo
+	lib/lib_libscreenshooter_la-screenshooter-utils.lo
 lib_libscreenshooter_la_OBJECTS =  \
 	$(am_lib_libscreenshooter_la_OBJECTS)
 lib_libscreenshooter_la_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
@@ -412,8 +411,7 @@
 	lib/screenshooter-global.h \
 	lib/screenshooter-job.c lib/screenshooter-job.h \
 	lib/screenshooter-simple-job.c lib/screenshooter-simple-job.h \
-	lib/screenshooter-utils.c lib/screenshooter-utils.h \
-	lib/screenshooter-zimagez.c lib/screenshooter-zimagez.h
+	lib/screenshooter-utils.c lib/screenshooter-utils.h
 
 lib_libscreenshooter_la_CFLAGS = \
 	-I$(top_srcdir)	\
@@ -528,9 +526,7 @@
 astimagesdir = $(datadir)/xfce4/doc/ast/images
 astimages_DATA = \
 	docs/manual/ast/images/xfce4-screenshooter-dialog1.png \
-	docs/manual/ast/images/xfce4-screenshooter-dialog2.png \
-	docs/manual/ast/images/xfce4-screenshooter-zimagez-login-dialog.png \
-	docs/manual/ast/images/xfce4-screenshooter-zimagez-result-dialog.png
+	docs/manual/ast/images/xfce4-screenshooter-dialog2.png
 
 
 # C documentation
@@ -539,9 +535,7 @@
 cimagesdir = $(datadir)/xfce4/doc/C/images
 cimages_DATA = \
 	docs/manual/C/images/xfce4-screenshooter-dialog1.png \
-	docs/manual/C/images/xfce4-screenshooter-dialog2.png \
-	docs/manual/C/images/xfce4-screenshooter-zimagez-login-dialog.png \
-	docs/manual/C/images/xfce4-screenshooter-zimagez-result-dialog.png
+	docs/manual/C/images/xfce4-screenshooter-dialog2.png
 
 
 # Catalan documentation
@@ -550,9 +544,7 @@
 caimagesdir = $(datadir)/xfce4/doc/ca/images
 caimages_DATA = \
 	docs/manual/ca/images/xfce4-screenshooter-dialog1.png \
-	docs/manual/ca/images/xfce4-screenshooter-dialog2.png \
-	docs/manual/ca/images/xfce4-screenshooter-zimagez-login-dialog.png \
-	docs/manual/ca/images/xfce4-screenshooter-zimagez-result-dialog.png
+	docs/manual/ca/images/xfce4-screenshooter-dialog2.png
 
 
 # Danish documentation
@@ -561,9 +553,7 @@
 daimagesdir = $(datadir)/xfce4/doc/da/images
 daimages_DATA = \
 	docs/manual/da/images/xfce4-screenshooter-dialog1.png \
-	docs/manual/da/images/xfce4-screenshooter-dialog2.png \
-	docs/manual/da/images/xfce4-screenshooter-zimagez-login-dialog.png \
-	docs/manual/da/images/xfce4-screenshooter-zimagez-result-dialog.png
+	docs/manual/da/images/xfce4-screenshooter-dialog2.png
 
 
 # Portuguese documentation
@@ -572,9 +562,7 @@
 ptimagesdir = $(datadir)/xfce4/doc/pt/images
 ptimages_DATA = \
 	docs/manual/pt/images/xfce4-screenshooter-dialog1.png \
-	docs/manual/pt/images/xfce4-screenshooter-dialog2.png \
-	docs/manual/pt/images/xfce4-screenshooter-zimagez-login-dialog.png \
-	docs/manual/pt/images/xfce4-screenshooter-zimagez-result-dialog.png
+	docs/manual/pt/images/xfce4-screenshooter-dialog2.png
 
 
 # Spanish documentation
@@ -583,9 +571,7 @@
 esimagesdir = $(datadir)/xfce4/doc/es/images
 esimages_DATA = \
 	docs/manual/es/images/xfce4-screenshooter-dialog1.png \
-	docs/manual/es/images/xfce4-screenshooter-dialog2.png \
-	docs/manual/es/images/xfce4-screenshooter-zimagez-login-dialog.png \
-	docs/manual/es/images/xfce4-screenshooter-zimagez-result-dialog.png
+	docs/manual/es/images/xfce4-screenshooter-dialog2.png
 
 
 # French documentation
@@ -594,9 +580,7 @@
 frimagesdir = $(datadir)/xfce4/doc/fr/images
 frimages_DATA = \
 	docs/manual/fr/images/xfce4-screenshooter-dialog1.png \
-	docs/manual/fr/images/xfce4-screenshooter-dialog2.png \
-	docs/manual/fr/images/xfce4-screenshooter-zimagez-login-dialog.png \
-	docs/manual/fr/images/xfce4-screenshooter-zimagez-result-dialog.png
+	docs/manual/fr/images/xfce4-screenshooter-dialog2.png
 
 
 # Galician documentation
@@ -605,9 +589,7 @@
 glimagesdir = $(datadir)/xfce4/doc/gl/images
 glimages_DATA = \
 	docs/manual/gl/images/xfce4-screenshooter-dialog1.png \
-	docs/manual/gl/images/xfce4-screenshooter-dialog2.png \
-	docs/manual/gl/images/xfce4-screenshooter-zimagez-login-dialog.png \
-	docs/manual/gl/images/xfce4-screenshooter-zimagez-result-dialog.png
+	docs/manual/gl/images/xfce4-screenshooter-dialog2.png
 
 
 # Indonesian documentation
@@ -616,9 +598,7 @@
 idimagesdir = $(datadir)/xfce4/doc/id/images
 idimages_DATA = \
 	docs/manual/id/images/xfce4-screenshooter-dialog1.png \
-	docs/manual/id/images/xfce4-screenshooter-dialog2.png \
-	docs/manual/id/images/xfce4-screenshooter-zimagez-login-dialog.png \
-	docs/manual/id/images/xfce4-screenshooter-zimagez-result-dialog.png
+	docs/manual/id/images/xfce4-screenshooter-dialog2.png
 
 
 # Italian documentation
@@ -627,9 +607,7 @@
 itimagesdir = $(datadir)/xfce4/doc/it/images
 itimages_DATA = \
 	docs/manual/it/images/xfce4-screenshooter-dialog1.png \
-	docs/manual/it/images/xfce4-screenshooter-dialog2.png \
-	docs/manual/it/images/xfce4-screenshooter-zimagez-login-dialog.png \
-	docs/manual/it/images/xfce4-screenshooter-zimagez-result-dialog.png
+	docs/manual/it/images/xfce4-screenshooter-dialog2.png
 
 
 # Japanese documentation
@@ -638,9 +616,7 @@
 jaimagesdir = $(datadir)/xfce4/doc/ja/images
 jaimages_DATA = \
 	docs/manual/ja/images/xfce4-screenshooter-dialog1.png \
-	docs/manual/ja/images/xfce4-screenshooter-dialog2.png \
-	docs/manual/ja/images/xfce4-screenshooter-zimagez-login-dialog.png \
-	docs/manual/ja/images/xfce4-screenshooter-zimagez-result-dialog.png
+	docs/manual/ja/images/xfce4-screenshooter-dialog2.png
 
 
 # Turkish documentation
@@ -649,9 +625,7 @@
 trimagesdir = $(datadir)/xfce4/doc/tr/images
 trimages_DATA = \
 	docs/manual/tr/images/xfce4-screenshooter-dialog1.png \
-	docs/manual/tr/images/xfce4-screenshooter-dialog2.png \
-	docs/manual/tr/images/xfce4-screenshooter-zimagez-login-dialog.png \
-	docs/manual/tr/images/xfce4-screenshooter-zimagez-result-dialog.png
+	docs/manual/tr/images/xfce4-screenshooter-dialog2.png
 
 
 # Siplified Chinese documentation
@@ -660,9 +634,7 @@
 zhcnimagesdir = $(datadir)/xfce4/doc/zh_CN/images
 zhcnimages_DATA = \
 	docs/manual/zh_CN/images/xfce4-screenshooter-dialog1.png \
-	docs/manual/zh_CN/images/xfce4-screenshooter-dialog2.png \
-	docs/manual/zh_CN/images/xfce4-screenshooter-zimagez-login-dialog.png \
-	docs/manual/zh_CN/images/xfce4-screenshooter-zimagez-result-dialog.png
+	docs/manual/zh_CN/images/xfce4-screenshooter-dialog2.png
 
 
 # Extra dist and distclean rules
@@ -823,8 +795,6 @@
 	lib/$(am__dirstamp) lib/$(DEPDIR)/$(am__dirstamp)
 lib/lib_libscreenshooter_la-screenshooter-utils.lo:  \
 	lib/$(am__dirstamp) lib/$(DEPDIR)/$(am__dirstamp)
-lib/lib_libscreenshooter_la-screenshooter-zimagez.lo:  \
-	lib/$(am__dirstamp) lib/$(DEPDIR)/$(am__dirstamp)
 lib/libscreenshooter.la: $(lib_libscreenshooter_la_OBJECTS) $(lib_libscreenshooter_la_DEPENDENCIES) lib/$(am__dirstamp)
 	$(lib_libscreenshooter_la_LINK)  $(lib_libscreenshooter_la_OBJECTS) $(lib_libscreenshooter_la_LIBADD) $(LIBS)
 install-binPROGRAMS: $(bin_PROGRAMS)
@@ -929,8 +899,6 @@
 	-rm -f lib/lib_libscreenshooter_la-screenshooter-simple-job.lo
 	-rm -f lib/lib_libscreenshooter_la-screenshooter-utils.$(OBJEXT)
 	-rm -f lib/lib_libscreenshooter_la-screenshooter-utils.lo
-	-rm -f lib/lib_libscreenshooter_la-screenshooter-zimagez.$(OBJEXT)
-	-rm -f lib/lib_libscreenshooter_la-screenshooter-zimagez.lo
 	-rm -f lib/lib_libscreenshooter_la-sexy-url-label.$(OBJEXT)
 	-rm -f lib/lib_libscreenshooter_la-sexy-url-label.lo
 	-rm -f panel-plugin/panel_plugin_xfce4_screenshooter_plugin-screenshooter-plugin.$(OBJEXT)
@@ -949,7 +917,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@lib/$(DEPDIR)/lib_libscreenshooter_la-screenshooter-marshal.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@lib/$(DEPDIR)/lib_libscreenshooter_la-screenshooter-simple-job.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@lib/$(DEPDIR)/lib_libscreenshooter_la-screenshooter-utils.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@lib/$(DEPDIR)/lib_libscreenshooter_la-screenshooter-zimagez.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@lib/$(DEPDIR)/lib_libscreenshooter_la-sexy-url-label.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@panel-plugin/$(DEPDIR)/panel_plugin_xfce4_screenshooter_plugin-screenshooter-plugin.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@src/$(DEPDIR)/src_xfce4_screenshooter-main.Po@am__quote@
@@ -1055,13 +1022,6 @@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(lib_libscreenshooter_la_CFLAGS) $(CFLAGS) -c -o lib/lib_libscreenshooter_la-screenshooter-utils.lo `test -f 'lib/screenshooter-utils.c' || echo '$(srcdir)/'`lib/screenshooter-utils.c
 
-lib/lib_libscreenshooter_la-screenshooter-zimagez.lo: lib/screenshooter-zimagez.c
-@am__fastdepCC_TRUE@	$(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(lib_libscreenshooter_la_CFLAGS) $(CFLAGS) -MT lib/lib_libscreenshooter_la-screenshooter-zimagez.lo -MD -MP -MF lib/$(DEPDIR)/lib_libscreenshooter_la-screenshooter-zimagez.Tpo -c -o lib/lib_libscreenshooter_la-screenshooter-zimagez.lo `test -f 'lib/screenshooter-zimagez.c' || echo '$(srcdir)/'`lib/screenshooter-zimagez.c
-@am__fastdepCC_TRUE@	mv -f lib/$(DEPDIR)/lib_libscreenshooter_la-screenshooter-zimagez.Tpo lib/$(DEPDIR)/lib_libscreenshooter_la-screenshooter-zimagez.Plo
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='lib/screenshooter-zimagez.c' object='lib/lib_libscreenshooter_la-screenshooter-zimagez.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(lib_libscreenshooter_la_CFLAGS) $(CFLAGS) -c -o lib/lib_libscreenshooter_la-screenshooter-zimagez.lo `test -f 'lib/screenshooter-zimagez.c' || echo '$(srcdir)/'`lib/screenshooter-zimagez.c
-
 panel-plugin/panel_plugin_xfce4_screenshooter_plugin-screenshooter-plugin.o: panel-plugin/screenshooter-plugin.c
 @am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(panel_plugin_xfce4_screenshooter_plugin_CFLAGS) $(CFLAGS) -MT panel-plugin/panel_plugin_xfce4_screenshooter_plugin-screenshooter-plugin.o -MD -MP -MF panel-plugin/$(DEPDIR)/panel_plugin_xfce4_screenshooter_plugin-screenshooter-plugin.Tpo -c -o panel-plugin/panel_plugin_xfce4_screenshooter_plugin-screenshooter-plugin.o `test -f 'panel-plugin/screenshooter-plugin.c' || echo '$(srcdir)/'`panel-plugin/screenshooter-plugin.c
 @am__fastdepCC_TRUE@	mv -f panel-plugin/$(DEPDIR)/panel_plugin_xfce4_screenshooter_plugin-screenshooter-plugin.Tpo panel-plugin/$(DEPDIR)/panel_plugin_xfce4_screenshooter_plugin-screenshooter-plugin.Po
diff -Nur xfce4-screenshooter-1.7.9.orig//configure xfce4-screenshooter-1.7.9/configure
--- xfce4-screenshooter-1.7.9.orig//configure	2010-02-07 07:45:16.000000000 -0600
+++ xfce4-screenshooter-1.7.9/configure	2010-05-23 01:00:44.506665929 -0500
@@ -12876,62 +12876,6 @@
   fi
 
 
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libsoup-2.4 >= 2.26.0" >&5
-$as_echo_n "checking for libsoup-2.4 >= 2.26.0... " >&6; }
-  if $PKG_CONFIG "--atleast-version=2.26.0" "libsoup-2.4" >/dev/null 2>&1; then
-    SOUP_VERSION=`$PKG_CONFIG --modversion "libsoup-2.4"`
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: $SOUP_VERSION" >&5
-$as_echo "$SOUP_VERSION" >&6; }
-
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking SOUP_CFLAGS" >&5
-$as_echo_n "checking SOUP_CFLAGS... " >&6; }
-    SOUP_CFLAGS=`$PKG_CONFIG --cflags "libsoup-2.4"`
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: $SOUP_CFLAGS" >&5
-$as_echo "$SOUP_CFLAGS" >&6; }
-
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking SOUP_LIBS" >&5
-$as_echo_n "checking SOUP_LIBS... " >&6; }
-    SOUP_LIBS=`$PKG_CONFIG --libs "libsoup-2.4"`
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: $SOUP_LIBS" >&5
-$as_echo "$SOUP_LIBS" >&6; }
-
-    SOUP_REQUIRED_VERSION=2.26.0
-
-
-
-
-
-
-
-  elif $PKG_CONFIG --exists "libsoup-2.4" >/dev/null 2>&1; then
-    xdt_cv_version=`$PKG_CONFIG --modversion "libsoup-2.4"`
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: found, but $xdt_cv_version" >&5
-$as_echo "found, but $xdt_cv_version" >&6; }
-
-
-      echo "*** The required package libsoup-2.4 was found on your system,"
-      echo "*** but the installed version ($xdt_cv_version) is too old."
-      echo "*** Please upgrade libsoup-2.4 to atleast version 2.26.0, or adjust"
-      echo "*** the PKG_CONFIG_PATH environment variable if you installed"
-      echo "*** the new version of the package in a nonstandard prefix so"
-      echo "*** pkg-config is able to find it."
-      exit 1
-
-  else
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: not found" >&5
-$as_echo "not found" >&6; }
-
-
-      echo "*** The required package libsoup-2.4 was not found on your system."
-      echo "*** Please install libsoup-2.4 (atleast version 2.26.0) or adjust"
-      echo "*** the PKG_CONFIG_PATH environment variable if you"
-      echo "*** installed the package in a nonstandard prefix so that"
-      echo "*** pkg-config is able to find it."
-      exit 1
-
-  fi
-
-
 # Check whether --enable-xsltproc was given.
 if test "${enable_xsltproc+set}" = set; then :
   enableval=$enable_xsltproc;
diff -Nur xfce4-screenshooter-1.7.9.orig//lib/screenshooter-actions.c xfce4-screenshooter-1.7.9/lib/screenshooter-actions.c
--- xfce4-screenshooter-1.7.9.orig//lib/screenshooter-actions.c	2010-01-02 05:41:32.000000000 -0600
+++ xfce4-screenshooter-1.7.9/lib/screenshooter-actions.c	2010-05-23 01:00:44.506665929 -0500
@@ -123,20 +123,7 @@
           if (sd->action == OPEN)
             screenshooter_open_screenshot (screenshot_path, sd->app);
           else
-            {
-              gchar *new_last_user = NULL;
-
-              screenshooter_upload_to_zimagez (screenshot_path,
-                                               sd->last_user,
-                                               sd->title,
-                                               &new_last_user);
-
-              if (new_last_user)
-                {
-                  g_free (sd->last_user);
-                  sd->last_user = new_last_user;
-                }
-            }
+            return;
         }
 
       g_object_unref (temp_dir);
diff -Nur xfce4-screenshooter-1.7.9.orig//lib/screenshooter-actions.h xfce4-screenshooter-1.7.9/lib/screenshooter-actions.h
--- xfce4-screenshooter-1.7.9.orig//lib/screenshooter-actions.h	2009-11-07 17:19:33.000000000 -0600
+++ xfce4-screenshooter-1.7.9/lib/screenshooter-actions.h	2010-05-23 01:00:44.507665998 -0500
@@ -24,7 +24,6 @@
 #include "screenshooter-capture.h"
 #include "screenshooter-global.h"
 #include "screenshooter-dialogs.h"
-#include "screenshooter-zimagez.h"
 
 gboolean screenshooter_take_screenshot_idle (ScreenshotData *sd);
 gboolean screenshooter_action_idle          (ScreenshotData *sd);
diff -Nur xfce4-screenshooter-1.7.9.orig//lib/screenshooter-dialogs.c xfce4-screenshooter-1.7.9/lib/screenshooter-dialogs.c
--- xfce4-screenshooter-1.7.9.orig//lib/screenshooter-dialogs.c	2010-01-02 05:41:48.000000000 -0600
+++ xfce4-screenshooter-1.7.9/lib/screenshooter-dialogs.c	2010-05-23 01:00:44.507665998 -0500
@@ -53,9 +53,6 @@
 cb_clipboard_toggled               (GtkToggleButton    *tb,
                                     ScreenshotData     *sd);
 static void
-cb_zimagez_toggled                 (GtkToggleButton    *tb,
-                                    ScreenshotData     *sd);
-static void
 cb_delay_spinner_changed           (GtkWidget          *spinner,
                                     ScreenshotData     *sd);
 static gchar
@@ -184,14 +181,6 @@
 
 
 
-static void cb_zimagez_toggled (GtkToggleButton *tb, ScreenshotData *sd)
-{
-  if (gtk_toggle_button_get_active (tb))
-    sd->action = UPLOAD;
-}
-
-
-
 /* Set the delay according to the spinner */
 static void cb_delay_spinner_changed (GtkWidget *spinner, ScreenshotData *sd)
 {
@@ -868,7 +857,6 @@
   GtkWidget *actions_label, *actions_alignment, *actions_box;
   GtkWidget *save_radio_button;
   GtkWidget *clipboard_radio_button, *open_with_radio_button;
-  GtkWidget *zimagez_radio_button;
 
   GtkListStore *liststore;
   GtkWidget *combobox, *open_box;
@@ -997,19 +985,6 @@
   /* Run the callback functions to grey/ungrey the correct widgets */
   cb_toggle_set_sensi (GTK_TOGGLE_BUTTON (open_with_radio_button), combobox);
 
-  /* Upload to zimagez radio button */
-  zimagez_radio_button =
-    gtk_radio_button_new_with_label_from_widget (GTK_RADIO_BUTTON (save_radio_button),
-                                                 _("Host on ZimageZ"));
-  gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (zimagez_radio_button),
-                                (sd->action == UPLOAD));
-  gtk_widget_set_tooltip_text (zimagez_radio_button,
-                               _("Host the screenshot on ZimageZ, a free online "
-                                 "image hosting service"));
-  g_signal_connect (G_OBJECT (zimagez_radio_button), "toggled",
-                    G_CALLBACK (cb_zimagez_toggled), sd);
-  gtk_box_pack_start (GTK_BOX (actions_box), zimagez_radio_button, FALSE, FALSE, 0);
-
   /* Preview box */
   preview_box = gtk_vbox_new (FALSE, 6);
   gtk_container_set_border_width (GTK_CONTAINER (preview_box), 0);
