--- srcML-src.orig/src/libsrcml/srcml_reader_handler.hpp	2015-05-20 03:36:48.000000000 +0200
+++ srcML-src/src/libsrcml/srcml_reader_handler.hpp	2017-04-07 17:10:40.137339217 +0200
@@ -456,7 +456,7 @@
 
             if(uri == SRCML_CPP_NS_URI) {
 
-                if(archive->language != 0) {
+                if(archive->language) {
 
                     if(*archive->language == "C++" || *archive->language == "C" || *archive->language == "Objective-C")
                         archive->options |= SRCML_OPTION_CPP | SRCML_OPTION_CPP_NOMACRO;
