diff -Naur Image-BMP-1.26/lib/Image/BMP.pm Image-BMP-1.26.patched/lib/Image/BMP.pm
--- Image-BMP-1.26/lib/Image/BMP.pm	2024-02-06 19:37:00.000000000 -0500
+++ Image-BMP-1.26.patched/lib/Image/BMP.pm	2024-08-19 15:45:02.250718304 -0400
@@ -382,7 +382,10 @@
 	$bmp->{YpixelsPerM} = read_bmp($bmp,4);
 	$bmp->{ColorsUsed} = read_bmp($bmp,4);
 	$bmp->{ColorsImportant} = read_bmp($bmp,4);
-	$bmp->{ColorsUsed} = 1<<$bmp->{BitCount} if $bmp->{ColorsUsed} == 0;
+	{
+		use bigint;
+		$bmp->{ColorsUsed} = 1<<$bmp->{BitCount} if $bmp->{ColorsUsed} == 0;
+	}
 
 	$bmp->_debug(1,"Image: $bmp->{BitCount}/$bmp->{ColorsUsed} colors. Geometry: $bmp->{Width}x$bmp->{Height} $bmp->{ImageSize} [comp: $compStr ($bmp->{Compression})]\n");
 
