diff -Naur freeorion-0.4.8/parse/ConditionParser.h freeorion-0.4.8.patched/parse/ConditionParser.h
--- freeorion-0.4.8/parse/ConditionParser.h	2018-08-23 05:41:14.000000000 -0400
+++ freeorion-0.4.8.patched/parse/ConditionParser.h	2021-02-17 04:34:17.063751926 -0500
@@ -1,6 +1,7 @@
 #ifndef _ConditionParser_h_
 #define _ConditionParser_h_
 
+#include "../universe/ValueRef.h"
 #include "Lexer.h"
 #include "ParseImpl.h"
 #include "ValueRefParser.h"
diff -Naur freeorion-0.4.8/server/ServerApp.cpp freeorion-0.4.8.patched/server/ServerApp.cpp
--- freeorion-0.4.8/server/ServerApp.cpp	2018-08-23 05:41:14.000000000 -0400
+++ freeorion-0.4.8.patched/server/ServerApp.cpp	2021-02-17 16:33:47.472827568 -0500
@@ -1715,7 +1715,7 @@
             // make a link to new connection
             m_player_empire_ids[player_connection->PlayerID()] = empire.first;
 
-            const OrderSet dummy;
+            OrderSet dummy;
             const OrderSet& orders = orders_it->second ? *orders_it->second : dummy;
 
             auto player_info_map = GetPlayerInfoMap();
diff -Naur freeorion-0.4.8/universe/Universe.h freeorion-0.4.8.patched/universe/Universe.h
--- freeorion-0.4.8/universe/Universe.h	2018-08-23 05:41:14.000000000 -0400
+++ freeorion-0.4.8.patched/universe/Universe.h	2021-02-17 04:03:51.908917840 -0500
@@ -3,7 +3,8 @@
 
 
 #include "EnumsFwd.h"
-#include "ValueRefFwd.h"
+#include "ValueRef.h"
+#include "Condition.h"
 #include "ObjectMap.h"
 #include "UniverseObject.h"
 #include "../util/Pending.h"
