index b257c83..8189a72 100644
--- a/src/nsgenbind-lexer.l
+++ b/src/nsgenbind-lexer.l
@@ -19,6 +19,16 @@
     yylloc->first_column = yylloc->last_column + 1; \
     yylloc->last_column += yyleng;
 
+
+/* Ensure compatability with bison 2.6 and later */
+#if ! defined YYSTYPE && ! defined YYSTYPE_IS_DECLARED && defined NSGENBIND_STYPE_IS_DECLARED
+#define YYSTYPE NSGENBIND_STYPE
+#endif
+
+#if ! defined YYLTYPE && ! defined YYLTYPE_IS_DECLARED && defined NSGENBIND_LTYPE_IS_DECLARED
+#define YYLTYPE NSGENBIND_LTYPE
+#endif
+
 %}
