diff -rupN a/src/netsurf-3.0/gtk/res/tabcontents.gtk2.ui b/src/netsurf-3.0/gtk/res/tabcontents.gtk2.ui
--- a/src/netsurf-3.0/gtk/res/tabcontents.gtk2.ui	2013-04-19 21:27:45.000000000 +0100
+++ b/src/netsurf-3.0/gtk/res/tabcontents.gtk2.ui	2014-01-17 11:47:53.000000000 +0000
@@ -79,13 +79,13 @@
       </object>
   <object class="GtkAdjustment" id="layouthadjustment">
     <property name="upper">100</property>
-    <property name="step_increment">1</property>
+    <property name="step_increment">30</property>
     <property name="page_increment">10</property>
     <property name="page_size">10</property>
   </object>
   <object class="GtkAdjustment" id="layoutvadjustment">
     <property name="upper">100</property>
-    <property name="step_increment">1</property>
+    <property name="step_increment">30</property>
     <property name="page_increment">10</property>
     <property name="page_size">10</property>
   </object>
