index 5369d99..74b9bb8 100644
--- a/src/webidl-lexer.l
+++ b/src/webidl-lexer.l
@@ -27,6 +27,15 @@
     yylloc->first_column = yylloc->last_column + 1; \
     yylloc->last_column += yyleng;
 
+/* Ensure compatability with bison 2.6 and later */
+#if ! defined YYSTYPE && ! defined YYSTYPE_IS_DECLARED && defined WEBIDL_STYPE_IS_DECLARED
+#define YYSTYPE WEBIDL_STYPE
+#endif
+
+#if ! defined YYLTYPE && ! defined YYLTYPE_IS_DECLARED && defined WEBIDL_LTYPE_IS_DECLARED
+#define YYLTYPE WEBIDL_LTYPE
+#endif
+
 %}
