From e5b34644f148c66211cc4a66a484c7fd706411fa Mon Sep 17 00:00:00 2001
From: Andreas Baumann <mail@andreasbaumann.cc>
Date: Thu, 2 Jul 2026 09:56:15 +0200
Subject: [PATCH] fixed macro guard for DGifCloseFile to work for 6.x.x and
 beyond

---
 src/gif.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gif.c b/src/gif.c
index 9d492c2..fc8ea16 100644
--- a/src/gif.c
+++ b/src/gif.c
@@ -96,11 +96,11 @@ static int gif_read(struct image *img){
 
 void gif_close(struct image *img){
 	struct gif_t *g = (struct gif_t *)img;
-#if defined(GIFLIB_MAJOR) && defined(GIFLIB_MINOR) && (GIFLIB_MAJOR == 5 && GIFLIB_MINOR >= 1)
+#if defined(GIFLIB_MAJOR) && defined(GIFLIB_MINOR) && (GIFLIB_MAJOR > 5 || (GIFLIB_MAJOR == 5 && GIFLIB_MINOR >= 1))
 	int ret;
 #endif
 
-#if defined(GIFLIB_MAJOR) && defined(GIFLIB_MINOR) && (GIFLIB_MAJOR == 5 && GIFLIB_MINOR >= 1)
+#if defined(GIFLIB_MAJOR) && defined(GIFLIB_MINOR) && (GIFLIB_MAJOR > 5 || (GIFLIB_MAJOR == 5 && GIFLIB_MINOR >= 1))
 	DGifCloseFile(g->gif,&ret);
 	if(ret != GIF_OK) {
 #if defined(GIFLIB_MAJOR) && GIFLIB_MAJOR >= 5
