--- meh-0.3+20260702_e5b3464.orig/src/gif.c	2026-07-02 07:56:15.000000000 +0000
+++ meh-0.3+20260702_e5b3464/src/gif.c	2026-08-28 18:52:36.181465263 +0000
@@ -101,7 +101,7 @@
 #endif
 
 #if defined(GIFLIB_MAJOR) && defined(GIFLIB_MINOR) && (GIFLIB_MAJOR > 5 || (GIFLIB_MAJOR == 5 && GIFLIB_MINOR >= 1))
-	DGifCloseFile(g->gif,&ret);
+	ret = DGifCloseFile(g->gif,&ret);
 	if(ret != GIF_OK) {
 #if defined(GIFLIB_MAJOR) && GIFLIB_MAJOR >= 5
 		fprintf(stderr, "GIFLIB: %s\n", GifErrorString(ret));
