diff -Nur cgmadness/Makefile cgmadness.new/Makefile
--- cgmadness/Makefile	2011-09-09 19:55:46.000000000 +0200
+++ cgmadness.new/Makefile	2012-10-30 23:36:52.000000000 +0100
@@ -18,7 +18,7 @@
 CPREFIX :=
 
 CC := $(CPREFIX)gcc
-CFLAGS := -Wall -Werror
+CFLAGS := -Wall
 
 CXX := $(CPREFIX)g++
 CXXFLAGS = $(CFLAGS) -pedantic -I.
@@ -53,7 +53,7 @@
 	LIBS_NETWORK += -lpthreadGCE2 -lwsock32 -lws2_32
 	EXECSUFFIX := .exe
 endif
-ifneq ($(findstring linux-gnu,$(MACHINE)),)
+ifneq ($(findstring slackware-linux,$(MACHINE)),)
 	CXXFLAGS += -ansi
 	LIBS_OPENGL += -lglut -lGLEW
 	LIBS_NETWORK += -lpthread
@@ -115,7 +115,7 @@
 
 %$(EXECSUFFIX): $(BUILD)/%.o
 	@echo "  LINK $@"
-	@$(LD) $(LDFLAGS) $^ $(foreach find, $(FIND_LIBS), $(call $(find),$^)) -o $@
+	@$(LD) $(LDFLAGS) $^ $(foreach find, $(FIND_LIBS), $(call $(find),$^)) -o $@ -lGL -lGLU
 
 $(BUILD)/%.o: %.c | $$(@D)/.
 	@echo "  CC $@"
