diff -Naur thttpd-2.25b.orig/cgi-src/Makefile.in thttpd-2.25b/cgi-src/Makefile.in
--- thttpd-2.25b.orig/cgi-src/Makefile.in	2010-10-28 23:33:28.000000000 +0200
+++ thttpd-2.25b/cgi-src/Makefile.in	2010-10-29 00:05:43.000000000 +0200
@@ -31,10 +31,9 @@
 MANDIR =	@mandir@
 
 CC =		@CC@
-CCOPT =		@V_CCOPT@
 DEFS =		@DEFS@
 INCLS =		-I..
-CFLAGS =	$(CCOPT) $(DEFS) $(INCLS)
+CFLAGS +=	$(DEFS) $(INCLS)
 LDFLAGS =	@LDFLAGS@ @V_STATICFLAG@
 LIBS =		@LIBS@
 NETLIBS =	@V_NETLIBS@
diff -Naur thttpd-2.25b.orig/extras/Makefile.in thttpd-2.25b/extras/Makefile.in
--- thttpd-2.25b.orig/extras/Makefile.in	2010-10-28 23:33:28.000000000 +0200
+++ thttpd-2.25b/extras/Makefile.in	2010-10-29 00:06:36.000000000 +0200
@@ -32,11 +32,9 @@
 MANDIR =	@mandir@
 
 CC =		@CC@
-CCOPT =		@V_CCOPT@
 DEFS =		@DEFS@
 INCLS =		-I..
-CFLAGS =	$(CCOPT) $(DEFS) $(INCLS)
-STATICFLAG =	@V_STATICFLAG@
+CFLAGS +=	$(DEFS) $(INCLS)
 LDFLAGS =	@LDFLAGS@
 LIBS =		@LIBS@
 NETLIBS =	@V_NETLIBS@
@@ -59,7 +57,7 @@
 	$(CC) $(CFLAGS) -DWEBDIR=\"$(WEBDIR)\" -c makeweb.c
 
 htpasswd:	htpasswd.o
-	$(CC) $(LDFLAGS) $(STATICFLAG) htpasswd.o -o htpasswd $(LIBS)
+	$(CC) $(LDFLAGS) htpasswd.o -o htpasswd $(LIBS)
 
 htpasswd.o:	htpasswd.c ../config.h
 	$(CC) $(CFLAGS) -DWEBDIR=\"$(WEBDIR)\" -c htpasswd.c
diff -Naur thttpd-2.25b.orig/Makefile.in thttpd-2.25b/Makefile.in
--- thttpd-2.25b.orig/Makefile.in	2010-10-28 23:33:28.000000000 +0200
+++ thttpd-2.25b/Makefile.in	2010-10-29 00:05:43.000000000 +0200
@@ -47,10 +47,9 @@
 # You shouldn't need to edit anything below here.
 
 CC =		@CC@
-CCOPT =		@V_CCOPT@
 DEFS =		@DEFS@
 INCLS =		-I.
-CFLAGS =	$(CCOPT) $(DEFS) $(INCLS)
+CFLAGS +=	$(DEFS) $(INCLS)
 LDFLAGS =	@LDFLAGS@
 LIBS =		@LIBS@
 NETLIBS =	@V_NETLIBS@
