--- yetris-2.0.1/Makefile.original	2014-04-23 15:48:09.448984387 +0300
+++ yetris-2.0.1/Makefile	2014-04-23 15:54:12.061451107 +0300
@@ -49,11 +49,11 @@
 
 EXEC_PREFIX = $(PREFIX)
 DATAROOTDIR = $(PREFIX)/share
-MANROOT     = $(DATAROOTDIR)/man
+MANROOT     = $(PREFIX)/man
 
 MANNUMBER   = 6
 
-BINDIR      = $(EXEC_PREFIX)/games
+BINDIR      = $(EXEC_PREFIX)/bin
 MANDIR      = $(MANROOT)/man$(MANNUMBER)
 
 MANFILE     = $(PACKAGE).$(MANNUMBER)
