--- doc/CMakeLists.txt.orig	2019-02-16 20:30:36.724781344 +0100
+++ doc/CMakeLists.txt	2019-02-16 20:31:01.294782000 +0100
@@ -107,17 +107,17 @@
         # we need to exclude it when installing man3. Same for
         # editorconfig-format.3
         install(DIRECTORY ${EC_MANPAGE3_DIR}
-            DESTINATION share/man
+            DESTINATION man
             PATTERN editorconfig.3 EXCLUDE
             PATTERN editorconfig-format.3 EXCLUDE
             REGEX ._include_. EXCLUDE)
 
         install(FILES
             ${EC_MANPAGE1_DIR}/editorconfig.1
-            DESTINATION share/man/man1)
+            DESTINATION man/man1)
         install(FILES
             ${EC_MANPAGE5_DIR}/editorconfig-format.5
-            DESTINATION share/man/man5)
+            DESTINATION man/man5)
 
         # "make clean" should also clean generated docs
         set_directory_properties(PROPERTIES
