Qualify placeholders
Matteo Bernardini <ponce@slackbuilds.org>

diff -Naur libLAS-1.8.1.orig/src/c_api.cpp libLAS-1.8.1/src/c_api.cpp
--- libLAS-1.8.1.orig/src/c_api.cpp	2016-08-22 15:06:34.000000000 +0200
+++ libLAS-1.8.1/src/c_api.cpp	2020-08-04 14:54:22.311996000 +0200
@@ -498,7 +498,7 @@
         
         transforms.erase( std::remove_if( transforms.begin(), 
                                   transforms.end(),
-                                  boost::bind( &IsReprojectionTransform, _1 ) ),
+                                  boost::bind( &IsReprojectionTransform, boost::placeholders::_1 ) ),
                   transforms.end());
         
         liblas::TransformPtr srs_transform = liblas::TransformPtr(new liblas::ReprojectionTransform(in_ref, *out_ref, &h));
@@ -1840,7 +1840,7 @@
         
         transforms.erase( std::remove_if( transforms.begin(), 
                                   transforms.end(),
-                                  boost::bind( &IsReprojectionTransform, _1 ) ),
+                                  boost::bind( &IsReprojectionTransform, boost::placeholders::_1 ) ),
                   transforms.end());
         
         liblas::TransformPtr srs_transform = liblas::TransformPtr(new liblas::ReprojectionTransform(in_ref, *out_ref, &h));
diff -Naur libLAS-1.8.1.orig/src/header.cpp libLAS-1.8.1/src/header.cpp
--- libLAS-1.8.1.orig/src/header.cpp	2016-08-22 15:06:34.000000000 +0200
+++ libLAS-1.8.1/src/header.cpp	2020-08-04 14:53:36.599996000 +0200
@@ -610,7 +610,7 @@
 
     m_vlrs.erase( std::remove_if( m_vlrs.begin(),
                                   m_vlrs.end(),
-                                  boost::bind( &SameVLRs, name, id, _1 ) ),
+                                  boost::bind( &SameVLRs, name, id, boost::placeholders::_1 ) ),
                   m_vlrs.end());
 
     m_recordsCount = static_cast<uint32_t>(m_vlrs.size());
