Matteo Bernardini <ponce@slackbuilds.org>
---
Don't hardcode python versions

diff -Naur gattlib-0.20150805.orig/setup.py gattlib-0.20150805/setup.py
--- gattlib-0.20150805.orig/setup.py	2015-08-05 15:16:56.000000000 +0200
+++ gattlib-0.20150805/setup.py	2018-05-02 09:45:04.625893000 +0200
@@ -19,9 +19,9 @@
     glib_libs = [x.strip() for x in glib_libs if x]
 
     if sys.version_info.major == 3:
-        boost_libs = ["boost_python-py34"]
+        boost_libs = ["boost_python3"+str(sys.version_info.minor)]
     else:
-        boost_libs = ["boost_python"]
+        boost_libs = ["boost_python2"+str(sys.version_info.minor)]
     extension_modules = [
         Extension(
             'gattlib',
