--- vala-0.34.2/vapi/gtk+-3.0.vapi.orig	2016-09-27 22:44:37.000000000 +1000
+++ vala-0.34.2/vapi/gtk+-3.0.vapi	2016-11-14 16:24:16.635334244 +1000
@@ -840,6 +840,7 @@
 		[CCode (has_construct_function = false, type = "GtkWidget*")]
 		public Button.with_mnemonic (string label);
 		public bool always_show_image { get; set construct; }
+                public bool focus_on_click { get; set; }
 		public Gtk.Widget image { get; set; }
 		public Gtk.PositionType image_position { get; set; }
 		public string label { get; set construct; }
@@ -1573,6 +1574,7 @@
 		public Gtk.CellArea cell_area { owned get; construct; }
 		public int column_span_column { get; set; }
 		public int entry_text_column { get; set; }
+                public bool focus_on_click { get; set; }
 		public bool has_entry { get; construct; }
 		[NoAccessorMethod]
 		public bool has_frame { get; set; }
@@ -2080,6 +2082,7 @@
 		[CCode (has_construct_function = false, type = "GtkWidget*")]
 		public FileChooserButton.with_dialog (Gtk.Dialog dialog);
 		public Gtk.FileChooser dialog { construct; }
+                public bool focus_on_click { get; set; }
 		public string title { get; set; }
 		public int width_chars { get; set; }
 		public virtual signal void file_set ();
@@ -6692,7 +6695,6 @@
 		public unowned Gdk.Display get_display ();
 		public bool get_double_buffered ();
 		public int get_events ();
-		public bool get_focus_on_click ();
 		public unowned Pango.FontMap? get_font_map ();
 		public unowned Cairo.FontOptions? get_font_options ();
 		public unowned Gdk.FrameClock? get_frame_clock ();
@@ -6859,7 +6861,6 @@
 		public void set_direction (Gtk.TextDirection dir);
 		public void set_double_buffered (bool double_buffered);
 		public void set_events (int events);
-		public void set_focus_on_click (bool focus_on_click);
 		public void set_font_map (Pango.FontMap? font_map);
 		public void set_font_options (Cairo.FontOptions? options);
 		public void set_halign (Gtk.Align align);
@@ -6927,7 +6928,6 @@
 		public Gdk.EventMask events { get; set; }
 		[NoAccessorMethod]
 		public bool expand { get; set; }
-		public bool focus_on_click { get; set; }
 		public Gtk.Align halign { get; set; }
 		[NoAccessorMethod]
 		public bool has_default { get; set; }
