diff --git a/opendmarc/opendmarc.c b/opendmarc/opendmarc.c
index 0b3457a..1c0e27a 100644
--- a/opendmarc/opendmarc.c
+++ b/opendmarc/opendmarc.c
@@ -1863,7 +1863,13 @@ mlfi_envfrom(SMFICTX *ctx, char **envfrom)
 
 	if (conf->conf_ignoreauthclients &&
 	    dmarcf_getsymval(ctx, "{auth_authen}") != NULL)
+	{
+		if (curconf->conf_dolog)
+		{
+			syslog(LOG_INFO, "ignoring authenticated client, mailfrom=%s", envfrom[0]);
+		}
 		return SMFIS_ACCEPT;
+	}
 
 	dfc = (DMARCF_MSGCTX) malloc(sizeof(struct dmarcf_msgctx));
 	if (dfc == NULL)
