diff --git a/reports/opendmarc-reports.in b/reports/opendmarc-reports.in
index 5dfa5ee..6826ff7 100755
--- a/reports/opendmarc-reports.in
+++ b/reports/opendmarc-reports.in
@@ -878,7 +878,7 @@ foreach (@$domainset)
 			$mailout .= "Content-Type: text/plain;\n";
 			$mailout .= "\n";
 			$mailout .= "This is a DMARC aggregate report for $domain\n";
-			$mailout .= "generated at " . localtime() . "\n";
+			$mailout .= "generated at " . strftime("%a, %b %e %Y %H:%M:%S %z (%Z)", localtime()) . "\n";
 			$mailout .= "\n";
 			$mailout .= "--$boundary\n";
 			$mailout .= "Content-Type: application/zip\n";
