diff --git a/opendmarc/opendmarc.c b/opendmarc/opendmarc.c
index 2a60b92..07e21a4 100644
--- a/opendmarc/opendmarc.c
+++ b/opendmarc/opendmarc.c
@@ -2859,9 +2859,9 @@ mlfi_eom(SMFICTX *ctx)
 			dmarcf_dstring_printf(dfc->mctx_afrf,
 			                      "This is an authentication "
 			                      "failure report for an email "
-			                      "message received from IP\n"
-			                      "%s on %s.\n\n",
-			                      cc->cctx_ipstr, timebuf);
+			                      "message received from\n"
+			                      "%s [%s] on %s.\n\n",
+			                      cc->cctx_host, cc->cctx_ipstr, timebuf);
 
 			dmarcf_dstring_printf(dfc->mctx_afrf,
 			                      "--%s:%s\n"
