Index: opendmarc/reports/opendmarc-import.in
===================================================================
--- opendmarc.orig/reports/opendmarc-import.in	2018-12-17 01:41:32.570336411 -0500
+++ opendmarc/reports/opendmarc-import.in	2018-12-17 01:41:32.566336411 -0500
@@ -529,7 +529,7 @@
 				}
 
 	  case "from"		{
-					$fdomain = $value;
+					$fdomain = lc($value);
 				}
 
 	  case "job"		{
@@ -567,7 +567,7 @@
 				}
 
 	  case "mfrom"		{
-					$envdomain = $value;
+					$envdomain = lc($value);
 				}
 
 	  case "p"		{
@@ -579,7 +579,7 @@
 				}
 
 	  case "pdomain"	{
-					$pdomain = $value;
+					$pdomain = lc($value);
 				}
 
 	  case "policy"		{
