diff --git a/opendmarc/opendmarc.c b/opendmarc/opendmarc.c
index bdad10d..d29036f 100644
--- a/opendmarc/opendmarc.c
+++ b/opendmarc/opendmarc.c
@@ -3005,19 +3005,9 @@ mlfi_eom(SMFICTX *ctx)
 			dmarcf_dstring_printf(dfc->mctx_afrf, "Date: %s\n",
 			                      timebuf);
 
-			h = dmarcf_findheader(dfc, "subject", 0);
-			if (h == NULL)
-			{
-				dmarcf_dstring_printf(dfc->mctx_afrf,
-				                      "Subject: DMARC failure report for job %s\n",
-				                      dfc->mctx_jobid);
-			}
-			else
-			{
-				dmarcf_dstring_printf(dfc->mctx_afrf,
-				                      "Subject: FW: %s\n",
-				                      h->hdr_value);
-			}
+			dmarcf_dstring_printf(dfc->mctx_afrf,
+					      "Subject: DMARC failure report for %s received from %s\n",
+					      dfc->mctx_fromdomain, cc->cctx_host);
 
 			dmarcf_dstring_cat(dfc->mctx_afrf,
 			                   "MIME-Version: 1.0\n");
