diff --git a/opendmarc/opendmarc.c b/opendmarc/opendmarc.c
index 0645bcf..2a60b92 100644
--- a/opendmarc/opendmarc.c
+++ b/opendmarc/opendmarc.c
@@ -2852,6 +2852,7 @@ mlfi_eom(SMFICTX *ctx)
 
 			dmarcf_dstring_printf(dfc->mctx_afrf,
 			                      "--%s:%s\n"
+			                      "Content-Description: Notification\n"
 			                      "Content-Type: text/plain\n\n",
 			                      hostname, dfc->mctx_jobid);
 
@@ -2864,6 +2865,7 @@ mlfi_eom(SMFICTX *ctx)
 
 			dmarcf_dstring_printf(dfc->mctx_afrf,
 			                      "--%s:%s\n"
+			                      "Content-Description: DMARC failure report\n"
 			                      "Content-Type: message/feedback-report\n\n",
 			                      hostname, dfc->mctx_jobid);
 
@@ -2902,6 +2904,7 @@ mlfi_eom(SMFICTX *ctx)
 
 			dmarcf_dstring_printf(dfc->mctx_afrf,
 			                      "--%s:%s\n"
+					      "Content-Description: Failed message headers\n"
 			                      "Content-Type: text/rfc822-headers\n\n",
 			                      hostname, dfc->mctx_jobid);
 
