--- ring-daemon-20181026_114c66440/contrib/src/libressl/rules.mak.orig	2018-10-26 17:20:13.000000000 +0200
+++ ring-daemon-20181026_114c66440/contrib/src/libressl/rules.mak	2018-10-27 22:36:00.608283629 +0200
@@ -25,9 +25,9 @@
 OPENBSD_URL := https://github.com/libressl-portable/openbsd/archive/$(OPENBSD_VERSION).tar.gz
 
 # Check if openssl or libressl is already present on the system
-ifeq ($(call need_pkg,"openssl >= 1.0.0" || call need_pkg,"libressl >= 1.0.0"),)
-PKGS_FOUND += libressl
-endif
+#ifeq ($(call need_pkg,"openssl >= 1.0.0" || call need_pkg,"libressl >= 1.0.0"),)
+#PKGS_FOUND += libressl
+#endif
 
 # Pure dependency of restbed: do not add to PKGS.
 
