--- ring-daemon-4.0.0/src/ringdht/namedirectory.cpp.orig	2017-07-19 23:27:21.000000000 +0200
+++ ring-daemon-4.0.0/src/ringdht/namedirectory.cpp	2017-08-01 15:50:13.311715765 +0200
@@ -90,9 +90,9 @@
 size_t getContentLength(restbed::Response& reply)
 {
     size_t length = 0;
-#ifndef RESTBED_OLD_API
-    length =
-#endif
+//#ifndef RESTBED_OLD_API
+//    length =
+//#endif
     reply.get_header("Content-Length", length);
     return length;
 }
