--- ring-daemon-4.0.0/configure.ac.orig	2017-07-19 23:27:21.000000000 +0200
+++ ring-daemon-4.0.0/configure.ac	2017-08-01 16:34:48.826571353 +0200
@@ -368,8 +368,8 @@
 AS_IF([test "x$with_dbus" = "xyes"], [
        PKG_CHECK_MODULES(DBUSCPP, dbus-c++-1,, AC_MSG_WARN([Missing dbus development files]))
 
-       AS_AC_EXPAND(LIBDIR, $libdir)
-       AC_SUBST(LIBDIR)
+       AS_AC_EXPAND(LIBEXECDIR, $libexecdir)
+       AC_SUBST(LIBEXECDIR)
 
        AC_CONFIG_FILES([bin/dbus/Makefile
                         bin/dbus/cx.ring.Ring.service])
--- ring-daemon-4.0.0/globals.mk.orig	2017-07-19 23:27:21.000000000 +0200
+++ ring-daemon-4.0.0/globals.mk	2017-08-01 16:36:01.520567429 +0200
@@ -1,7 +1,7 @@
 # Global variables
 
 src=$(abs_top_srcdir)
-ringlibdir=$(DESTDIR)$(libdir)/ring
+ringlibdir=$(DESTDIR)$(libexecdir)/ring
 
 ASTYLERC="$(top_srcdir)/../astylerc"
 indent="/usr/bin/astyle"
--- ring-daemon-4.0.0/bin/dbus/cx.ring.Ring.service.in.orig	2017-07-19 23:27:21.000000000 +0200
+++ ring-daemon-4.0.0/bin/dbus/cx.ring.Ring.service.in	2017-08-01 16:39:22.606556575 +0200
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=cx.ring.Ring
-Exec=@LIBDIR@/ring/dring
+Exec=@LIBEXECDIR@/ring/dring
--- ring-daemon-4.0.0/bin/Makefile.am.orig	2017-07-19 23:27:21.000000000 +0200
+++ ring-daemon-4.0.0/bin/Makefile.am	2017-08-01 16:51:06.985518556 +0200
@@ -1,4 +1,4 @@
-ringlibdir = $(libdir)/ring
+ringlibdir = $(libexecdir)/ring
 
 if HAVE_WIN32
 ringlib_PROGRAMS = ringcli
