diff -Nuar SOAP-Lite-0.710.08.orig/Makefile.PL SOAP-Lite-0.710.08/Makefile.PL
--- SOAP-Lite-0.710.08.orig/Makefile.PL	2008-06-28 11:37:20.000000000 -0700
+++ SOAP-Lite-0.710.08/Makefile.PL	2008-12-07 23:42:03.341939336 -0800
@@ -43,7 +43,7 @@
     ["DIME messages","SOAP::Packager::DIME",{"IO::Scalar" => "2.105", "DIME::Tools" => 0.03, "Data::UUID" => "0.11"},0],
     ["SSL Support for TCP Transport","SOAP::Transport::TCP",{"IO::Socket::SSL" => 0},0],
     ["Compression support for HTTP","SOAP::Transport::HTTP",{"Compress::Zlib" => 0},0],
-    ["MIME interoperability w/ Axis","SOAP::Lite",{"MIME::Parser" => "6.106"},0],
+    ["MIME interoperability w/ Axis","SOAP::Lite",{"MIME::Parser" => "5.413"},0],
 );
 
 use Getopt::Long;
