diff -Naur gerbmerge-1.8old/setup.py gerbmerge-1.8new/setup.py
--- gerbmerge-1.8old/setup.py	2011-07-11 19:15:46.000000000 +0200
+++ gerbmerge-1.8new/setup.py	2011-07-11 19:13:20.000000000 +0200
@@ -37,7 +37,7 @@
   BinFiles = ['misc/gerbmerge.bat']
   BinDir = '.'
 else:
-  DestLib = distutils.sysconfig.get_config_var('LIBPYTHON')
+  DestLib = "/usr/lib/python2.6"
   DestDir = os.path.join(DestLib, 'gerbmerge')
   BinFiles = ['misc/gerbmerge']
   BinDir = distutils.sysconfig.get_config_var('BINDIR')  
@@ -74,10 +74,7 @@
        url = "http://ruggedcircuits.com/gerbmerge",
        packages = ['gerbmerge'],
        platforms = ['all'],
-       data_files = [ (DestDir, AuxFiles), 
-                      (os.path.join(DestDir,'testdata'), SampleFiles),
-                      (os.path.join(DestDir,'doc'), DocFiles),
-                      (BinDir, BinFiles) ]
+       data_files = [ (BinDir, BinFiles) ]
 )
 
 do_fix_perms = 0
