diff -Nur ./gqview-2.1.5.orig/src/bar_exif.c ./gqview-2.1.5.new/src/bar_exif.c
--- ./gqview-2.1.5.orig/src/bar_exif.c	2006-11-26 23:38:29.000000000 +0000
+++ ./gqview-2.1.5.new/src/bar_exif.c	2023-12-03 21:08:00.619450308 +0000
@@ -26,29 +26,6 @@
 
 #define BAR_EXIF_DATA_COLUMN_WIDTH 250
 
-static const gchar *bar_exif_key_list_real[] = {
-	"fCamera",
-	"fDateTime",
-	"fShutterSpeed",
-	"fAperture",
-	"ExposureProgram",
-	"fExposureBias",
-	"fISOSpeedRating",
-	"fFocalLength",
-	"fSubjectDistance",
-	"MeteringMode",
-	"fFlash",
-	"LightSource",
-	"fResolution",
-	"Orientation",
-	"ImageDescription",
-	"Copyright"
-};
-
-const gchar **bar_exif_key_list = bar_exif_key_list_real;
-const gint bar_exif_key_count = (sizeof(bar_exif_key_list_real) / sizeof(gchar *));
-
-
 /*
  *-------------------------------------------------------------------
  * table util
diff -Nur ./gqview-2.1.5.orig/src/bar_exif.h ./gqview-2.1.5.new/src/bar_exif.h
--- ./gqview-2.1.5.orig/src/bar_exif.h	2006-11-03 15:12:39.000000000 +0000
+++ ./gqview-2.1.5.new/src/bar_exif.h	2023-12-03 21:08:14.331450821 +0000
@@ -24,8 +24,27 @@
 
 /* these are exposed for when duplication of the exif bar's text is needed */
 
-const gchar **bar_exif_key_list;
-const gint bar_exif_key_count;
+static const gchar *bar_exif_key_list_real[] = {
+	"fCamera",
+	"fDateTime",
+	"fShutterSpeed",
+	"fAperture",
+	"ExposureProgram",
+	"fExposureBias",
+	"fISOSpeedRating",
+	"fFocalLength",
+	"fSubjectDistance",
+	"MeteringMode",
+	"fFlash",
+	"LightSource",
+	"fResolution",
+	"Orientation",
+	"ImageDescription",
+	"Copyright"
+};
+
+static const gchar **bar_exif_key_list = bar_exif_key_list_real;
+static const gint bar_exif_key_count = (sizeof(bar_exif_key_list_real) / sizeof(gchar *));
 
 gchar *bar_exif_validate_text(gchar *text);
 
diff -Nur ./gqview-2.1.5.orig/src/editors.c ./gqview-2.1.5.new/src/editors.c
--- ./gqview-2.1.5.orig/src/editors.c	2006-12-03 01:15:04.000000000 +0000
+++ ./gqview-2.1.5.new/src/editors.c	2023-12-04 18:56:19.956040817 +0000
@@ -47,7 +47,7 @@
 
 
 static gchar *editor_slot_defaults[] = {
-	N_("The Gimp"), "gimp-remote -n %f",
+	N_("The Gimp"), "gimp -n %f",
 	N_("XV"), "xv %f",
 	N_("Xpaint"), "xpaint %f",
 	NULL, NULL,
