diff -Naur smc-20140328.orig/configure.ac smc-20140328/configure.ac
--- smc-20140328.orig/configure.ac	2014-03-29 00:06:03.000000000 +0100
+++ smc-20140328/configure.ac	2025-10-02 21:03:41.393373349 +0200
@@ -26,10 +26,6 @@
 AC_CHECK_LIB([boost_thread], [main], ,
 	[AC_MSG_ERROR([Unable to find Boost Thread library])])
 
-# Check for the Boost System library
-AC_CHECK_LIB([boost_system], [main], ,
-	[AC_MSG_ERROR([Unable to find Boost System library])])
-
 # Check for the OpenGL and GLU library
 case "${host}" in
 *darwin*|*macosx*)
diff -Naur smc-20140328.orig/src/core/filesystem/filesystem.cpp smc-20140328/src/core/filesystem/filesystem.cpp
--- smc-20140328.orig/src/core/filesystem/filesystem.cpp	2014-03-29 00:06:04.000000000 +0100
+++ smc-20140328/src/core/filesystem/filesystem.cpp	2025-10-02 21:14:57.185373349 +0200
@@ -15,8 +15,12 @@
 
 #include "../../core/filesystem/filesystem.h"
 #include "../../core/game_core.h"
+#include "../../core/global_basic.h"
+#include <boost/filesystem.hpp>
+
+using namespace std;
+
 // boost filesystem
-#include "boost/filesystem/convenience.hpp"
 namespace fs = boost::filesystem;
 // needed for the stat function and to get the user directory on unix
 #include <sys/stat.h>
