--- trunk/cdemu-daemon/src/cdemud-daemon.c	2008/07/25 20:39:19	406
+++ trunk/cdemu-daemon/src/cdemud-daemon.c	2008/07/25 20:39:47	407
@@ -30,19 +30,19 @@
 #define BUF_SIZE (512 * (MAX_SECTORS + OTHER_SECTORS))
 
 struct vhba_request {
-    __u32 tag;
-    __u32 lun;
+    guint32 tag;
+    guint32 lun;
 #define MAX_COMMAND_SIZE       16
 
-    __u8 cdb[MAX_COMMAND_SIZE];
-    __u8 cdb_len;
-    __u32 data_len;
+    guint8 cdb[MAX_COMMAND_SIZE];
+    guint8 cdb_len;
+    guint32 data_len;
 };
 
 struct vhba_response {
-    __u32 tag;
-    __u32 status;
-    __u32 data_len;
+    guint32 tag;
+    guint32 status;
+    guint32 data_len;
 };
 
 
