diff -up ./cmake/VersionString.cmake.orig ./cmake/VersionString.cmake
--- ./cmake/VersionString.cmake.orig	2016-03-22 19:18:55.656290322 +0200
+++ ./cmake/VersionString.cmake	2016-03-22 19:19:08.239813098 +0200
@@ -33,7 +33,7 @@ function(version_file SRC DST VERSION_SO
 	
 	foreach(arg IN LISTS VERSION_SOURCES)
 		
-		if(mode STREQUAL "variable")
+		if("${mode}" STREQUAL "variable")
 			set(mode "file")
 		else()
 			get_filename_component(arg "${arg}" ABSOLUTE)
