--- phply-0.9.1.orig/setup.py	2015-11-16 13:07:31.666674204 -0700
+++ phply-0.9.1/setup.py	2015-11-16 13:15:02.571666129 -0700
@@ -19,6 +19,15 @@
       license='BSD',
       url='http://www.github.com/ramen/phply',
 
+      scripts=[
+        'tools/php2jinja.py',
+        'tools/php2json.py',
+        'tools/php2python.py',
+        'tools/phpshell.py',
+        'tools/tokenize.php',
+        'tools/unparse.py',
+        ],
+
       classifiers=[
         'Development Status :: 2 - Pre-Alpha',
         'Environment :: Console',
