From 82c2287bfc61012ac2fbba0a7c8d359048f09b92 Mon Sep 17 00:00:00 2001
From: Steffen Macke <sdteffen@sdteffen.de>
Date: Sun, 18 Aug 2013 13:22:35 +0200
Subject: [PATCH 20/24] Fixed black light shape reference.

---
 sheets/Lights.sheet.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sheets/Lights.sheet.in b/sheets/Lights.sheet.in
index a246043..4613fa1 100644
--- a/sheets/Lights.sheet.in
+++ b/sheets/Lights.sheet.in
@@ -28,7 +28,7 @@
       <_description>Blinders</_description>
     </object>
     <br/>
-    <object name="Lights - Black light">
+    <object name="Lights - Blacklight">
       <_description>Black light</_description>
     </object>
     <object name="Lights - Stroboscope">
-- 
1.8.4.4

