diff -u a/configure.in b/configure.in
--- a/configure.in	2015-10-10 20:23:14.257728818 +0200
+++ b/configure.in	2015-10-10 20:23:44.512728478 +0200
@@ -532,10 +532,10 @@
  AM_CONDITIONAL(WITH_JW, test "xno" != "xno")
 fi
 hardbook_ok=no
-xsltproc_ok=no
+xsltproc_ok=yes
 if test "x$with_hardbooks" = "xno"; then
   AM_CONDITIONAL(WITH_JW, test "xno" != "xno")
-  AM_CONDITIONAL(WITH_HTMLDOC, test "xno" != "xno")
+  AM_CONDITIONAL(WITH_HTMLDOC, test "x$xsltproc" != "xno")
   AM_CONDITIONAL(WITH_PDFDOC, test "xno" != "xno")
   AM_CONDITIONAL(WITH_PSDOC, test "xno" != "xno")
 else
@@ -549,9 +549,6 @@
       hardbook_ok=yes
     fi
   fi
-  if test "x$xsltproc" != "xno" ; then
-    xsltproc_ok=yes
-  fi
 fi
 
 
