diff -Naur old/app/diagram_tree_menu.c new/app/diagram_tree_menu.c
--- old/app/diagram_tree_menu.c	2014-08-24 11:46:01.000000000 -0400
+++ new/app/diagram_tree_menu.c	2025-12-26 15:10:31.863151896 -0500
@@ -45,35 +45,35 @@
 static GtkItemFactoryEntry common_items_[] = {
   { "/sep0", NULL, NULL, 0, "<Separator>" },
   { N_("/_Sort objects"), NULL, NULL, 0, "<Branch>" },
-  { N_("/Sort objects/by _name"), NULL, on_sort_objects_activate,
+  { N_("/Sort objects/by _name"), NULL, (void (*)(void)) on_sort_objects_activate,
     DIA_TREE_SORT_NAME, "" },
-  { N_("/Sort objects/by _type"), NULL, on_sort_objects_activate,
+  { N_("/Sort objects/by _type"), NULL, (void (*)(void)) on_sort_objects_activate,
     DIA_TREE_SORT_TYPE, "" },
-  { N_("/Sort objects/as _inserted"), NULL, on_sort_objects_activate,
+  { N_("/Sort objects/as _inserted"), NULL, (void (*)(void)) on_sort_objects_activate,
     DIA_TREE_SORT_INSERT, "" },
   { "/Sort objects/sep0", NULL, NULL, 0, "<Separator>" },
-  { N_("/Sort objects/All by name"), NULL, on_sort_all_objects_activate,
+  { N_("/Sort objects/All by name"), NULL, (void (*)(void)) on_sort_all_objects_activate,
     DIA_TREE_SORT_NAME, "" },
-  { N_("/Sort objects/All by type"), NULL, on_sort_all_objects_activate,
+  { N_("/Sort objects/All by type"), NULL, (void (*)(void)) on_sort_all_objects_activate,
     DIA_TREE_SORT_TYPE, "" },
-  { N_("/Sort objects/All as inserted"), NULL, on_sort_all_objects_activate,
+  { N_("/Sort objects/All as inserted"), NULL, (void (*)(void)) on_sort_all_objects_activate,
     DIA_TREE_SORT_INSERT, "" },
   { N_("/Sort objects/_Default"), NULL, NULL, 0, "<Branch>"},
-  { N_("/Sort objects/Default/by _name"), NULL, on_sort_def_activate,
+  { N_("/Sort objects/Default/by _name"), NULL, (void (*)(void)) on_sort_def_activate,
     DIA_TREE_SORT_NAME, "<RadioItem>" },
-  { N_("/Sort objects/Default/by _type"), NULL, on_sort_def_activate,
+  { N_("/Sort objects/Default/by _type"), NULL, (void (*)(void)) on_sort_def_activate,
     DIA_TREE_SORT_TYPE, "/Sort objects/Default/by name" },
-  { N_("/Sort objects/Default/as _inserted"), NULL, on_sort_def_activate,
+  { N_("/Sort objects/Default/as _inserted"), NULL, (void (*)(void)) on_sort_def_activate,
     DIA_TREE_SORT_INSERT, "/Sort objects/Default/by name" },
   { N_("/Sort _diagrams"), NULL, NULL, 0, "<Branch>" },
-  { N_("/Sort _diagrams/by _name"), NULL, on_sort_diagrams_activate,
+  { N_("/Sort _diagrams/by _name"), NULL, (void (*)(void)) on_sort_diagrams_activate,
     DIA_TREE_SORT_NAME, "" },
-  { N_("/Sort _diagrams/as _inserted"), NULL, on_sort_diagrams_activate,
+  { N_("/Sort _diagrams/as _inserted"), NULL, (void (*)(void)) on_sort_diagrams_activate,
     DIA_TREE_SORT_INSERT, "" },
   { N_("/Sort diagrams/_Default"), NULL, NULL, 0, "<Branch>"},
-  { N_("/Sort diagrams/Default/by _name"), NULL, on_sort_dia_def_activate,
+  { N_("/Sort diagrams/Default/by _name"), NULL, (void (*)(void)) on_sort_dia_def_activate,
     DIA_TREE_SORT_NAME, "<RadioItem>" },
-  { N_("/Sort diagrams/Default/as _inserted"), NULL, on_sort_dia_def_activate,
+  { N_("/Sort diagrams/Default/as _inserted"), NULL, (void (*)(void)) on_sort_dia_def_activate,
     DIA_TREE_SORT_INSERT, "/Sort diagrams/Default/by name" },
 };
 
@@ -81,9 +81,9 @@
 sizeof (common_items_) / sizeof (common_items_[0]);
 
 static GtkItemFactoryEntry object_items_[] = {
-  { N_("/_Locate"), NULL, on_locate_object_activate, 0, "" },
-  { N_("/_Properties"), NULL, on_properties_activate, 0, "" },
-  { N_("/_Hide this type"), NULL, on_hide_object_activate, 0, "" },
+  { N_("/_Locate"), NULL, (void (*)(void)) on_locate_object_activate, 0, "" },
+  { N_("/_Properties"), NULL, (void (*)(void)) on_properties_activate, 0, "" },
+  { N_("/_Hide this type"), NULL, (void (*)(void)) on_hide_object_activate, 0, "" },
   { N_(SHOW_TYPE_PATH), NULL, NULL, 0, "<Branch>" },
   /*  { "/sep1", NULL, NULL, 0, "<Separator>" },
       { N_("/_Delete"), NULL, on_delete_object_activate, 0, "" }, */
@@ -92,7 +92,7 @@
 #define OBJ_ITEMS_SIZE_  (sizeof (object_items_) / sizeof (object_items_[0]))
 
 static GtkItemFactoryEntry dia_items_[] = {
-  { N_("/_Locate"), NULL, on_locate_dia_activate, 0, "" },
+  { N_("/_Locate"), NULL, (void (*)(void)) on_locate_dia_activate, 0, "" },
   { N_(SHOW_TYPE_PATH), NULL, NULL, 0, "<Branch>" },
 };
 
