diff -ru dwdiff-2.1.4.orig/config.pkg dwdiff-2.1.4/config.pkg
--- dwdiff-2.1.4.orig/config.pkg	2020-12-13 06:38:01.000000000 -0500
+++ dwdiff-2.1.4/config.pkg	2021-10-26 09:04:45.389583300 -0400
@@ -11,6 +11,8 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+INSTALLDIRS="mandir"
+
 EXTENSIONS="c verbose_compile gettext"
 
 SWITCHES="+unicode +dwfilter"
