diff -Naurp dwdiff-2.0.1.orig/src/doDiff.c dwdiff-2.0.1/src/doDiff.c
--- dwdiff-2.0.1.orig/src/doDiff.c	2011-10-26 14:32:22.000000000 -0400
+++ dwdiff-2.0.1/src/doDiff.c	2011-10-27 00:34:59.785336664 -0400
@@ -190,7 +190,9 @@ static void handleNextWhitespace(InputFi
 	whitespace.
 */
 static void handleSynchronizedNextWhitespace(bool printNew) {
+#ifdef USE_UNICODE
 	static_assert(CRLF_GRAPHEME_CLUSTER_BREAK == 0);
+#endif
 
 	bool BValid = true;
 	unsigned int *lineNumberA, *lineNumberB;
