diff -ru dwdiff-2.1.4.orig/src/option.c dwdiff-2.1.4/src/option.c
--- dwdiff-2.1.4.orig/src/option.c	2020-12-13 06:38:01.000000000 -0500
+++ dwdiff-2.1.4/src/option.c	2021-10-26 08:48:13.189583300 -0400
@@ -470,8 +470,10 @@
 	option.output = stdout;
 
 	initOptions();
+#ifdef USE_UNICODE
   UErrorCode error = U_ZERO_ERROR;
 	ONLY_UNICODE(option.decomposition = unorm2_getNFDInstance(&error);)
+#endif
 
 	option.needStartStop = true;
 
