--- Makefile.orig	2009-07-01 13:06:33.857658500 -0500
+++ Makefile	2009-07-01 16:09:55.195727429 -0500
@@ -7,13 +7,13 @@
 #PREFIX += /usr/local
 #BINDIR = ${PREFIX}/bin
 #MANDIR = ${PREFIX}/man
-PREFIX += /usr/local
+PREFIX += /usr
 BINDIR = ${PREFIX}/bin
 MANDIR = ${PREFIX}/man
-DOCDIR = ${PREFIX}/share/doc/vobcopy
+DOCDIR = ${PREFIX}/doc/vobcopy-@VERSION@
 LFS    = -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
-CFLAGS += -I/usr/local//include
-LDFLAGS += -ldvdread -L/usr/local//lib
+CFLAGS += -I/usr/include
+LDFLAGS += -ldvdread -L/usr/lib@LIBDIRSUFFIX@
 
 #This specifies the conversion from .c to .o 
 .c.o:
