diff -Naur xmms-mp4_20050213/mp4ff/mp4ffint.h xmms-mp4_20050213.patched//mp4ff/mp4ffint.h
--- xmms-mp4_20050213/mp4ff/mp4ffint.h	2005-02-13 05:41:31.000000000 -0500
+++ xmms-mp4_20050213.patched//mp4ff/mp4ffint.h	2011-10-23 15:22:30.000000000 -0400
@@ -347,7 +347,7 @@
 mp4ff_t *mp4ff_open_edit(mp4ff_callback_t *f);
 #endif
 void mp4ff_close(mp4ff_t *ff);
-void mp4ff_track_add(mp4ff_t *f);
+static void mp4ff_track_add(mp4ff_t *f);
 int32_t parse_sub_atoms(mp4ff_t *f, const uint64_t total_size,int meta_only);
 int32_t parse_atoms(mp4ff_t *f,int meta_only);
 
