diff -up alienarena-7.66/source/ref_gl/r_shadowmaps.c.no-qglBlitFramebufferEXT alienarena-7.66/source/ref_gl/r_shadowmaps.c
--- alienarena-7.66/source/ref_gl/r_shadowmaps.c.no-qglBlitFramebufferEXT	2013-11-05 14:46:37.611487279 -0500
+++ alienarena-7.66/source/ref_gl/r_shadowmaps.c	2013-11-05 15:00:47.510933729 -0500
@@ -95,12 +95,12 @@ void R_CheckFBOExtensions (void)
 	
 	//must check for ability to blit(Many old ATI drivers do not support)
 	//TODO: redundant with previous check?
-	if(gl_state.hasFBOblit) {
-		if(!qglBlitFramebufferEXT) {
-			Com_Printf("glBlitFramebufferEXT not found...\n");
-			gl_state.hasFBOblit = false;
-		}
-	}
+	// if(gl_state.hasFBOblit) {
+	//	if(!qglBlitFramebufferEXT) {
+	//		Com_Printf("glBlitFramebufferEXT not found...\n");
+	//		gl_state.hasFBOblit = false;
+	//	}
+	//}
 }
     
 
