diff -Naur alienarena-7.66-svn4307/source/ref_gl/r_shadowmaps.c alienarena-7.66-svn4307.patched/source/ref_gl/r_shadowmaps.c
--- alienarena-7.66-svn4307/source/ref_gl/r_shadowmaps.c	2017-08-02 05:34:56.000000000 -0400
+++ alienarena-7.66-svn4307.patched/source/ref_gl/r_shadowmaps.c	2022-02-22 23:27:40.462295225 -0500
@@ -92,15 +92,6 @@
 	{
 		Com_Printf("...GL_EXT_framebuffer_blit not found\n");
 	}
-	
-	//must check for ability to blit(Many old ATI drivers do not support)
-	//TODO: redundant with previous check?
-	if(gl_state.hasFBOblit) {
-		if(!qglBlitFramebufferEXT) {
-			Com_Printf("glBlitFramebufferEXT not found...\n");
-			gl_state.hasFBOblit = false;
-		}
-	}
 }
     
 
