diff -r 2debcb532cbd Sources/ptConstants.h
--- a/Sources/ptConstants.h	Sun May 25 15:40:27 2014 +0700
+++ b/Sources/ptConstants.h	Sun Nov 23 17:39:52 2014 +0100
@@ -222,7 +222,9 @@
 const short ptLfunDistModel_None      = LF_DIST_MODEL_NONE;
 const short ptLfunDistModel_Poly3     = LF_DIST_MODEL_POLY3;
 const short ptLfunDistModel_Poly5     = LF_DIST_MODEL_POLY5;
+#ifdef LF_DIST_MODEL_FOV1
 const short ptLfunDistModel_Fov1      = LF_DIST_MODEL_FOV1;
+#endif
 const short ptLfunDistModel_PTLens    = LF_DIST_MODEL_PTLENS;
 
 
diff -r 2debcb532cbd Sources/ptGuiOptions.cpp
--- a/Sources/ptGuiOptions.cpp	Sun May 25 15:40:27 2014 +0700
+++ b/Sources/ptGuiOptions.cpp	Sun Nov 23 17:39:52 2014 +0100
@@ -177,7 +177,9 @@
   {ptLfunDistModel_None,     QObject::tr("None") },
   {ptLfunDistModel_Poly3,    QObject::tr("3rd order polynomial") },
   {ptLfunDistModel_Poly5,    QObject::tr("5th order polynomial") },
+#ifdef LF_DIST_MODEL_FOV1
   {ptLfunDistModel_Fov1,     QObject::tr("1st order field of view") },
+#endif
   {ptLfunDistModel_PTLens,   QObject::tr("Panotools lens model") },
   {-1,NULL}};
 
diff -r 2debcb532cbd Sources/ptMainWindow.cpp
--- a/Sources/ptMainWindow.cpp	Sun May 25 15:40:27 2014 +0700
+++ b/Sources/ptMainWindow.cpp	Sun Nov 23 17:39:52 2014 +0100
@@ -3067,7 +3067,9 @@
   short DistModel = Settings->GetInt("LfunDistModel");
   LfunDistPoly3Container->setVisible(DistModel == ptLfunDistModel_Poly3);
   LfunDistPoly5Container->setVisible(DistModel == ptLfunDistModel_Poly5);
+#ifdef LF_DIST_MODEL_FOV1
   LfunDistFov1Container->setVisible(DistModel == ptLfunDistModel_Fov1);
+#endif
   LfunDistPTLensContainer->setVisible(DistModel == ptLfunDistModel_PTLens);
 }
 
diff -r 2debcb532cbd Sources/ptProcessor.cpp
--- a/Sources/ptProcessor.cpp	Sun May 25 15:40:27 2014 +0700
+++ b/Sources/ptProcessor.cpp	Sun Nov 23 17:39:52 2014 +0100
@@ -2238,12 +2238,14 @@
         DistortionData.Terms[1] = Settings->GetDouble("LfunDistPoly5K2");
         DistortionData.Terms[2] = 0.0;
         break;
+#ifdef LF_DIST_MODEL_FOV1
       case LF_DIST_MODEL_FOV1:
         modflags |= LF_MODIFY_DISTORTION;
         DistortionData.Terms[0] = Settings->GetDouble("LfunDistFov1Omega");
         DistortionData.Terms[1] = 0.0;
         DistortionData.Terms[2] = 0.0;
         break;
+#endif	
       case LF_DIST_MODEL_PTLENS:
         modflags |= LF_MODIFY_DISTORTION;
         DistortionData.Terms[0] = Settings->GetDouble("LfunDistPTLensA");
