diff -Naur wiimms-wbfs-tool-0.25a.orig/Makefile wiimms-wbfs-tool-0.25a/Makefile
--- wiimms-wbfs-tool-0.25a.orig/Makefile	2010-01-04 14:47:11.265517000 +0100
+++ wiimms-wbfs-tool-0.25a/Makefile	2010-01-04 15:36:37.033724127 +0100
@@ -105,7 +105,7 @@
 #LDFLAGS	+= -Xlinker --strip-all
 LDFLAGS		:= $(strip $(LDFLAGS))
 
-INSTALL_PATH	= /usr/local
+INSTALL_PATH	= /usr
 INSTALL_SCRIPTS	= install.sh load-titles.sh
 RM_FILES	+= $(INSTALL_SCRIPTS)
 SCRIPTS		= ./scripts
