--- XMonad/Util/Paste.hs.orig	2012-05-17 16:47:40.345000003 +0300
+++ XMonad/Util/Paste.hs	2012-05-17 16:47:55.724000002 +0300
@@ -89,9 +89,3 @@
                   sendEvent d w True keyPressMask ev
                   setEventType ev keyRelease
                   sendEvent d w True keyReleaseMask ev
-
--- | A null 'KeyMask'. Used when you don't want a character or string shifted, control'd, or what.
---
---   TODO: This really should be a function in the X11 binding. When noModMask shows up there, remove.
-noModMask :: KeyMask
-noModMask = 0
