r24405 | peter | 2011-10-26 10:00:04 +0000 (Wed, 26 Oct 2011) | 1 line

branch2011: distribute README.branch2011

Index: Build/source/Makefile.in
===================================================================
--- Build/source/Makefile.in	(revision 24404)
+++ Build/source/Makefile.in	(revision 24405)
@@ -342,6 +342,7 @@
 distcleancheck_listfiles = find . -type f -o -type l -print
 EXTRA_DIST = \
 	Build \
+	README.branch2011 \
 	README.coding \
 	README.config \
 	README.cross \
Index: Build/source/Makefile.am
===================================================================
--- Build/source/Makefile.am	(revision 24404)
+++ Build/source/Makefile.am	(revision 24405)
@@ -1,6 +1,6 @@
 ## Makefile.am for the TeX Live top-level
 ##
-## Copyright (C) 2009 - 2011 Peter Breitenlohner <tex-live@tug.org>
+## Copyright (C) 2009-2011 Peter Breitenlohner <tex-live@tug.org>
 ## You may freely use, modify and/or distribute this file.
 ##
 ## Use 'aclocal --force' (for "computed m4_sinclude")
@@ -15,6 +15,7 @@
 
 EXTRA_DIST = \
 	Build \
+	README.branch2011 \
 	README.coding \
 	README.config \
 	README.cross \
