r24407 | peter | 2011-10-26 10:05:08 +0000 (Wed, 26 Oct 2011) | 1 line

branch2011: update generated files

Index: Build/source/libs/zziplib/configure
===================================================================
--- Build/source/libs/zziplib/configure	(revision 24406)
+++ Build/source/libs/zziplib/configure	(revision 24407)
@@ -727,8 +727,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/libs/teckit/configure
===================================================================
--- Build/source/libs/teckit/configure	(revision 24406)
+++ Build/source/libs/teckit/configure	(revision 24407)
@@ -738,13 +738,8 @@
 CCC
 CC
 CFLAGS
-LDFLAGS
-LIBS
-CPPFLAGS
 CXXCPP
-CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/afm2pl/configure
===================================================================
--- Build/source/texk/afm2pl/configure	(revision 24406)
+++ Build/source/texk/afm2pl/configure	(revision 24407)
@@ -758,8 +758,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/dtl/configure
===================================================================
--- Build/source/texk/dtl/configure	(revision 24406)
+++ Build/source/texk/dtl/configure	(revision 24407)
@@ -758,8 +758,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/xdvik/configure
===================================================================
--- Build/source/texk/xdvik/configure	(revision 24406)
+++ Build/source/texk/xdvik/configure	(revision 24407)
@@ -800,7 +800,6 @@
 LIBS
 CPPFLAGS
 CPP
-CPPFLAGS
 XMKMF'
 ac_subdirs_all='squeeze'
 
Index: Build/source/texk/lacheck/configure
===================================================================
--- Build/source/texk/lacheck/configure	(revision 24406)
+++ Build/source/texk/lacheck/configure	(revision 24407)
@@ -755,8 +755,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/dvipos/configure
===================================================================
--- Build/source/texk/dvipos/configure	(revision 24406)
+++ Build/source/texk/dvipos/configure	(revision 24407)
@@ -758,8 +758,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/devnag/configure
===================================================================
--- Build/source/texk/devnag/configure	(revision 24406)
+++ Build/source/texk/devnag/configure	(revision 24407)
@@ -669,8 +669,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/dvi2tty/configure
===================================================================
--- Build/source/texk/dvi2tty/configure	(revision 24406)
+++ Build/source/texk/dvi2tty/configure	(revision 24407)
@@ -759,8 +759,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/bibtex8/configure
===================================================================
--- Build/source/texk/bibtex8/configure	(revision 24406)
+++ Build/source/texk/bibtex8/configure	(revision 24407)
@@ -758,8 +758,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/seetexk/configure
===================================================================
--- Build/source/texk/seetexk/configure	(revision 24406)
+++ Build/source/texk/seetexk/configure	(revision 24407)
@@ -758,8 +758,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/xdvipdfmx/configure
===================================================================
--- Build/source/texk/xdvipdfmx/configure	(revision 24406)
+++ Build/source/texk/xdvipdfmx/configure	(revision 24407)
@@ -787,8 +787,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/detex/configure
===================================================================
--- Build/source/texk/detex/configure	(revision 24406)
+++ Build/source/texk/detex/configure	(revision 24407)
@@ -762,8 +762,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/ttf2pk/configure
===================================================================
--- Build/source/texk/ttf2pk/configure	(revision 24406)
+++ Build/source/texk/ttf2pk/configure	(revision 24407)
@@ -765,8 +765,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/gsftopk/configure
===================================================================
--- Build/source/texk/gsftopk/configure	(revision 24406)
+++ Build/source/texk/gsftopk/configure	(revision 24407)
@@ -758,8 +758,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/ttfdump/configure
===================================================================
--- Build/source/texk/ttfdump/configure	(revision 24406)
+++ Build/source/texk/ttfdump/configure	(revision 24407)
@@ -751,8 +751,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/makejvf/configure
===================================================================
--- Build/source/texk/makejvf/configure	(revision 24406)
+++ Build/source/texk/makejvf/configure	(revision 24407)
@@ -762,8 +762,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/dvidvi/configure
===================================================================
--- Build/source/texk/dvidvi/configure	(revision 24406)
+++ Build/source/texk/dvidvi/configure	(revision 24407)
@@ -758,8 +758,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/dvipng/configure
===================================================================
--- Build/source/texk/dvipng/configure	(revision 24406)
+++ Build/source/texk/dvipng/configure	(revision 24407)
@@ -813,8 +813,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/cjkutils/configure
===================================================================
--- Build/source/texk/cjkutils/configure	(revision 24406)
+++ Build/source/texk/cjkutils/configure	(revision 24407)
@@ -768,8 +768,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/tex4htk/configure
===================================================================
--- Build/source/texk/tex4htk/configure	(revision 24406)
+++ Build/source/texk/tex4htk/configure	(revision 24407)
@@ -762,8 +762,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/ptexenc/configure
===================================================================
--- Build/source/texk/ptexenc/configure	(revision 24406)
+++ Build/source/texk/ptexenc/configure	(revision 24407)
@@ -762,8 +762,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/ps2pkm/configure
===================================================================
--- Build/source/texk/ps2pkm/configure	(revision 24406)
+++ Build/source/texk/ps2pkm/configure	(revision 24407)
@@ -758,8 +758,7 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP
-CPPFLAGS'
+CPP'
 
 
 # Initialize some variables set by options.
Index: Build/source/texk/xdv2pdf/configure
===================================================================
--- Build/source/texk/xdv2pdf/configure	(revision 24406)
+++ Build/source/texk/xdv2pdf/configure	(revision 24407)
@@ -763,7 +763,6 @@
 enable_libtool_lock
 enable_largefile
 enable_compiler_warnings
-with_gnu_ld
 with_system_kpathsea
 with_kpathsea_includes
 with_kpathsea_libdir
@@ -777,19 +776,12 @@
 LIBS
 CPPFLAGS
 CPP
-CPPFLAGS
 CXX
 CXXFLAGS
-LDFLAGS
-LIBS
-CPPFLAGS
 CCC
 CXXCPP
 OBJC
-OBJCFLAGS
-LDFLAGS
-LIBS
-OBJC'
+OBJCFLAGS'
 
 
 # Initialize some variables set by options.
