--- rejoystick-0.8.1/src/main.c.orig	2014-12-20 06:55:37.951753670 +0700
+++ rejoystick-0.8.1/src/main.c	2014-12-20 06:56:23.666092712 +0700
@@ -137,13 +137,9 @@
 
 	pthread_t	sdl_thread;
 
-	char*		home;
-	char*		keyfile;
-
-	home = (char*)getenv("HOME");
-	keyfile = KEYFILE;
-	filename = strcat(home, keyfile);
-
+        const char * home = getenv("HOME");
+        filename = malloc(strlen(home) + strlen(KEYFILE) + 1);
+        sprintf(filename, "%s%s", home, KEYFILE);
 	global_argv = argv;
 
 
