--- MEGAsync-4.7.1.0_Win/src/MEGASync/mega/src/gfx/freeimage.cpp	2022-09-27 20:24:16.000000000 +0900
+++ MEGAsync-4.7.1.0_Win.New/src/MEGASync/mega/src/gfx/freeimage.cpp	2022-10-12 10:30:22.849460973 +0900
@@ -313,7 +313,7 @@
 
     // Find decoder for video stream
     AVCodecID codecId = codecParm->codec_id;
-    AVCodec* decoder = avcodec_find_decoder(codecId);
+    auto decoder = avcodec_find_decoder(codecId);
     if (!decoder)
     {
         LOG_warn << "Codec not found: " << codecId;
@@ -330,7 +330,6 @@
 
     // Force seeking to key frames
     formatContext->seek2any = false;
-    videoStream->skip_to_keyframe = true;
     if (decoder->capabilities & CAP_TRUNCATED)
     {
         codecContext->flags |= CAP_TRUNCATED;
