From ef8b0c26df78f5bca48803fa350cf772eef351c8 Mon Sep 17 00:00:00 2001
From: Klas Lindfors <klas@yubico.com>
Date: Wed, 15 May 2019 13:57:11 +0200
Subject: [PATCH 01/25] bump versions after release

Signed-off-by: Gustavo B. Schenkel <gustavo.schenkel@gmail.com>
---
 NEWS         | 2 ++
 configure.ac | 4 ++--
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/NEWS b/NEWS
index a85d667..fc933b8 100644
--- a/NEWS
+++ b/NEWS
@@ -1,5 +1,7 @@
 libu2f-host NEWS -- History of user visible changes.
 
+* Version 1.1.11 (unreleased)
+
 * Version 1.1.10 (released 2019-05-15)
 
 ** Add new devices to udev rules.
diff --git a/configure.ac b/configure.ac
index d5251d3..0031f76 100644
--- a/configure.ac
+++ b/configure.ac
@@ -13,7 +13,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-AC_INIT([libu2f-host], [1.1.10], [yubico-devel@googlegroups.com])
+AC_INIT([libu2f-host], [1.1.11], [yubico-devel@googlegroups.com])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_AUX_DIR([build-aux])
@@ -21,7 +21,7 @@ AC_CONFIG_AUX_DIR([build-aux])
 # http://www.gnu.org/s/libtool/manual/html_node/Updating-version-info.html
 AC_SUBST(LT_CURRENT, 1)  # Interfaces removed:    CURRENT++, AGE=0, REVISION=0
 AC_SUBST(LT_AGE, 1)      # Interfaces added:      CURRENT++, AGE++, REVISION=0
-AC_SUBST(LT_REVISION, 10) # No interfaces changed:                   REVISION++
+AC_SUBST(LT_REVISION, 11) # No interfaces changed:                   REVISION++
 
 AM_INIT_AUTOMAKE([gnits dist-xz no-dist-gzip std-options -Wall])
 AM_SILENT_RULES([yes])
-- 
2.32.0

