From 24c6b57d25ff574ade39bdcbeb5eed2169f4b532 Mon Sep 17 00:00:00 2001
From: Ed Maste <emaste@freebsd.org>
Date: Sat, 8 Jun 2019 01:36:29 -0400
Subject: [PATCH 02/25] README: add FreeBSD build information

Signed-off-by: Gustavo B. Schenkel <gustavo.schenkel@gmail.com>
---
 README | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/README b/README
index e17d0e5..fefc96b 100644
--- a/README
+++ b/README
@@ -177,6 +177,16 @@ Both of these require that a release tarball of the project exists in the
 current directory. The value of the VERSION variable must match the version
 on that tarball.
 
+Building from Git works on FreeBSD (excluding documentation), using gmake.
+Dependencies may be installed via:
+
+  # pkg install gengetopt help2man hidapi json-c u2f-devd
+
+And then configure and build using:
+
+  $ ./configure --disable-gtk-doc
+  $ gmake
+
 Namespaces
 ----------
 
-- 
2.32.0

