From 08e32414f36b902127d679f55d439816e33dc0dc Mon Sep 17 00:00:00 2001
From: Klas Lindfors <klas@yubico.com>
Date: Mon, 10 Jun 2019 12:08:33 +0200
Subject: [PATCH 03/25] add note about maintenance mode and libfido2

Signed-off-by: Gustavo B. Schenkel <gustavo.schenkel@gmail.com>
---
 README | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/README b/README
index e17d0e5..bc0c2fb 100644
--- a/README
+++ b/README
@@ -10,6 +10,10 @@ device and perform the U2F Register and U2F Authenticate operations.
 For the server-side aspect, see our
 https://developers.yubico.com/libu2f-server/[libu2f-server project].
 
+NOTE: This project is in maintenance mode,
+https://developers.yubico.com/libfido2/[libfido2] is a new project with
+support for U2F and FIDO2.
+
 License
 -------
 
-- 
2.32.0

