From 3b8cf649c04e484d9162ab9325a4aa064df1f2ee Mon Sep 17 00:00:00 2001
From: Nicolas Braud-Santoni <nicolas@braud-santoni.eu>
Date: Fri, 30 Aug 2019 12:21:13 +0200
Subject: [PATCH 07/25] devs/init_device: Fix misleading comment (logical
 inversion)

Signed-off-by: Gustavo B. Schenkel <gustavo.schenkel@gmail.com>
---
 u2f-host/devs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/u2f-host/devs.c b/u2f-host/devs.c
index f10048f..aa76c9f 100644
--- a/u2f-host/devs.c
+++ b/u2f-host/devs.c
@@ -303,7 +303,7 @@ init_device (u2fh_devs * devs, struct u2fdevice *dev)
        &resplen) == U2FH_OK)
     {
       int offs = sizeof (nonce);
-      /* the response has to be at least 17 bytes, if it's more we discard that */
+      /* the response has to be at least 17 bytes, if it's less we discard it */
       if (resplen < 17)
 	{
 	  return U2FH_SIZE_ERROR;
-- 
2.32.0

