From bf3a625d846e22ff0ac3abac386781e0c09ffcab Mon Sep 17 00:00:00 2001
From: Jason Cawood <jasoncawood@gmail.com>
Date: Thu, 24 Oct 2019 16:14:27 -0700
Subject: [PATCH 08/25] Update 70-u2f.rules to include product id 0121

Signed-off-by: Gustavo B. Schenkel <gustavo.schenkel@gmail.com>
---
 70-u2f.rules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/70-u2f.rules b/70-u2f.rules
index 40506d8..c2babc4 100644
--- a/70-u2f.rules
+++ b/70-u2f.rules
@@ -17,7 +17,7 @@
 ACTION!="add|change", GOTO="u2f_end"
 
 # Yubico YubiKey
-KERNEL=="hidraw*", SUBSYSTEM=="hidraw", ATTRS{idVendor}=="1050", ATTRS{idProduct}=="0113|0114|0115|0116|0120|0200|0402|0403|0406|0407|0410", TAG+="uaccess", GROUP="plugdev", MODE="0660"
+KERNEL=="hidraw*", SUBSYSTEM=="hidraw", ATTRS{idVendor}=="1050", ATTRS{idProduct}=="0113|0114|0115|0116|0120|0121|0200|0402|0403|0406|0407|0410", TAG+="uaccess", GROUP="plugdev", MODE="0660"
 
 # Happlink (formerly Plug-Up) Security KEY
 KERNEL=="hidraw*", SUBSYSTEM=="hidraw", ATTRS{idVendor}=="2581", ATTRS{idProduct}=="f1d0", TAG+="uaccess", GROUP="plugdev", MODE="0660"
-- 
2.32.0

