From 420a18112a0aab09f19915cd7465e471e29717cd Mon Sep 17 00:00:00 2001
From: Gabriel Kihlman <g.kihlman@yubico.com>
Date: Wed, 14 Oct 2020 14:16:07 +0200
Subject: [PATCH 25/25] Add deprecation notice

Signed-off-by: Gustavo B. Schenkel <gustavo.schenkel@gmail.com>
---
 README | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/README b/README
index bc920d6..4b5bc8c 100644
--- a/README
+++ b/README
@@ -10,7 +10,7 @@ device and perform the U2F Register and U2F Authenticate operations.
 For the server-side aspect, see our
 https://developers.yubico.com/libu2f-server/[libu2f-server project].
 
-NOTE: This project is in maintenance mode,
+NOTE: This project is deprecated and is no longer being maintained.
 https://developers.yubico.com/libfido2/[libfido2] is a new project with
 support for U2F and FIDO2.
 
-- 
2.32.0

