Only in sysdig-0.11.0.fix: build
diff -ru sysdig-0.11.0/userspace/libsinsp/json_query.h sysdig-0.11.0.fix/userspace/libsinsp/json_query.h
--- sysdig-0.11.0/userspace/libsinsp/json_query.h	2016-07-14 10:11:40.000000000 +0200
+++ sysdig-0.11.0.fix/userspace/libsinsp/json_query.h	2016-07-21 21:50:23.582406170 +0200
@@ -24,17 +24,9 @@
 // jq is not C++-friendly
 extern "C"
 {
-	#include "compile.h"
 	#include "jv.h"
 	#include "jq.h"
 	//+++ jq C++ compile fixes
-	#ifndef NDEBUG
-	#	define NDEBUG
-	#	include "jv_alloc.h"
-	#	undef NDEBUG
-	#else
-	#	include "jv_alloc.h"
-	#endif // NDEBUG
 	#ifndef HAVE_MKSTEMP
 	#	include <stdlib.h>
 	#	include <fcntl.h>
@@ -58,8 +50,6 @@
 		}
 	#	define HAVE_MKSTEMP
 	#endif // HAVE_MKSTEMP
-	//--- jq C++ compile fix
-	#include "util.h"
 }
 
 #include <string>
