--- collada-dom-2.5.0.orig/dom/src/dae/daeMetaElement.cpp	2026-08-14 02:53:20.449450825 +0000
+++ collada-dom-2.5.0/dom/src/dae/daeMetaElement.cpp	2026-08-14 02:53:22.802340188 +0000
@@ -175,7 +175,7 @@
     else
         _metaAttributes.append(attr);
 
-    if ((attr->getName() != NULL) &&
+    if ((const char*)(attr->getName()) &&
         (strcmp(attr->getName(),"id") == 0)) {
         _metaID = attr;
         _isTrackableForQueries = true;
