--- fgrun-1.6.2/src/wizard_funcs.cxx.orig	2012-09-11 14:57:53.176639212 +0200
+++ fgrun-1.6.2/src/wizard_funcs.cxx	2012-09-11 15:11:35.995000085 +0200
@@ -296,7 +296,7 @@
     {
         fg_scenery = buf;
     }
-    else if (reloadPath && systemPrefs.get( "fg_scenery", buf, "", buflen-1))
+    else if (systemPrefs.get( "fg_scenery", buf, "", buflen-1))
     {
         fg_scenery = buf;
         prefs.set("fg_scenery", buf);
@@ -338,7 +338,7 @@
     {
         ts_dir = iVal;
     }
-    else if (reloadPath && systemPrefs.get("ts_dir", iVal, 0))
+    else if (systemPrefs.get("ts_dir", iVal, 0))
     {
         prefs.set("ts_dir", ts_dir);
         systemPrefs.get("ts_dir_init", iVal, 0);
@@ -353,7 +353,7 @@
     {
         ts_exe_->value(buf);
     }
-    else if (reloadPath && systemPrefs.get( "ts_exe", buf, "", buflen-1))
+    else if (systemPrefs.get( "ts_exe", buf, "", buflen-1))
     {
         ts_exe_->value( buf );
         prefs.set("ts_exe", buf);
