diff -Naur menulibre-2.2.1.org/menulibre/MenulibreApplication.py menulibre-2.2.1/menulibre/MenulibreApplication.py
--- menulibre-2.2.1.org/menulibre/MenulibreApplication.py	2019-05-31 01:55:18.000000000 +0100
+++ menulibre-2.2.1/menulibre/MenulibreApplication.py	2021-06-03 20:43:35.507256870 +0100
@@ -26,6 +26,8 @@
 
 from locale import gettext as _
 
+import gi
+gi.require_version("Gtk", "3.0")
 from gi.repository import Gio, GLib, GObject, Gtk, Gdk, GdkPixbuf
 
 from . import MenulibreStackSwitcher, MenulibreIconSelection
diff -Naur menulibre-2.2.1.org/menulibre_lib/helpers.py menulibre-2.2.1/menulibre_lib/helpers.py
--- menulibre-2.2.1.org/menulibre_lib/helpers.py	2019-05-31 01:55:18.000000000 +0100
+++ menulibre-2.2.1/menulibre_lib/helpers.py	2021-06-03 20:44:04.194017263 +0100
@@ -18,6 +18,8 @@
 import logging
 import os
 
+import gi
+gi.require_version("Gtk", "3.0")
 from gi.repository import Gtk
 
 from . menulibreconfig import get_data_file
