Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 728)
+++ CMakeLists.txt	(revision 729)
@@ -30,6 +30,9 @@
   ENDIF(BUILD_SHARED_LIBS)
 ENDIF(WIN32)
 ADD_LIBRARY(${OPENJPEG_LIBRARY_NAME} ${OPENJPEG_SRCS})
+IF(UNIX)
+  TARGET_LINK_LIBRARIES(${OPENJPEG_LIBRARY_NAME} m)
+ENDIF(UNIX)
 SET_TARGET_PROPERTIES(${OPENJPEG_LIBRARY_NAME} PROPERTIES ${OPENJPEG_LIBRARY_PROPERTIES})
 
 # Install library
