--- docs/jinjaext.py	2014-01-10 12:14:43.000000000 +0200
+++ docs/jinjaext.py.patched	2014-01-24 23:31:50.511110780 +0200
@@ -23,7 +23,7 @@
 from pygments.token import Keyword, Name, Comment, String, Error, \
      Number, Operator, Generic
 from jinja2 import Environment, FileSystemLoader
-from jinja2.utils import next
+from jinja2._compat import next
 
 
 def parse_rst(state, content_offset, doc):
