--- easyspice-0.6.8.orig/src/util.c
+++ easyspice-0.6.8/src/util.c
@@ -655,12 +655,12 @@
 void 
 commands_set_default(ExternalProgramms *prog)
 {
-  prog->editor=update_str(prog->editor, "xemacs");
+  prog->editor=update_str(prog->editor, "sensible-editor");
   prog->plot_command=update_str(prog->plot_command, "ngspice");
   prog->spice_command=update_str(prog->spice_command, "ngspice");
-  prog->browser=update_str(prog->browser, "firefox");
+  prog->browser=update_str(prog->browser, "sensible-browser");
   prog->html_files=update_str(prog->html_files,
-			      "file:///usr/local/doc/spice/index.html");
+			      "http://ngspice.sourceforge.net/docs.html");
   prog->spicehelp_command=update_str(prog->spicehelp_command, "nghelp");
 }
 
