Description: Added missing math lib to link, fixing FTBFS.
Author: Daniel T Chen <crimsun@ubuntu.com>
Bug-Debian: https://bugs.debian.org/747052
Last-Update: 2013-04-11

Index: easyspice-0.6.8/src/Makefile.am
===================================================================
--- easyspice-0.6.8.orig/src/Makefile.am	2013-04-11 09:44:23.680002124 -0400
+++ easyspice-0.6.8/src/Makefile.am	2013-04-11 09:44:34.571928687 -0400
@@ -18,7 +18,7 @@
 	spice.c \
 	util.c 
 
-easy_spice_LDADD = @PACKAGE_LIBS@ $(INTLLIBS)
+easy_spice_LDADD = @PACKAGE_LIBS@ $(INTLLIBS) -lm
 
 localedir = @datadir@/locale
 DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
Index: easyspice-0.6.8/src/Makefile.in
===================================================================
--- easyspice-0.6.8.orig/src/Makefile.in	2013-04-11 09:44:17.416036093 -0400
+++ easyspice-0.6.8/src/Makefile.in	2013-04-11 09:44:43.115859927 -0400
@@ -181,7 +181,7 @@
 	spice.c \
 	util.c 
 
-easy_spice_LDADD = @PACKAGE_LIBS@ $(INTLLIBS)
+easy_spice_LDADD = @PACKAGE_LIBS@ $(INTLLIBS) -lm
 AM_CFLAGS = -Wall
 all: all-am
 
