diff -u -r libkate-0.4.1-orig/Makefile.in libkate-0.4.1/Makefile.in
--- libkate-0.4.1-orig/Makefile.in	2011-08-21 23:46:58.000000000 +0100
+++ libkate-0.4.1/Makefile.in	2015-05-28 23:52:21.389946971 +0100
@@ -260,7 +260,6 @@
 LD = @LD@
 LDFLAGS = @LDFLAGS@
 LEX = @LEX@
-LEXLIB = @LEXLIB@
 LEX_OUTPUT_ROOT = @LEX_OUTPUT_ROOT@
 LIBKATE_SHARED_VERSION = @LIBKATE_SHARED_VERSION@
 LIBOBJS = @LIBOBJS@
diff -u -r libkate-0.4.1-orig/configure.ac libkate-0.4.1/configure.ac
--- libkate-0.4.1-orig/configure.ac	2011-08-21 23:44:32.000000000 +0100
+++ libkate-0.4.1/configure.ac	2015-05-30 06:28:33.306682481 +0100
@@ -7,7 +7,7 @@
 
 AC_CANONICAL_TARGET
 
-AM_INIT_AUTOMAKE([-Wall -Werror foreign subdir-objects])
+AM_INIT_AUTOMAKE([-Wall foreign subdir-objects])
 AC_CONFIG_HEADERS([include/config.h])
 
 ifdef([AM_SILENT_RULES],[AM_SILENT_RULES])
diff -u -r libkate-0.4.1-orig/doc/Makefile.in libkate-0.4.1/doc/Makefile.in
--- libkate-0.4.1-orig/doc/Makefile.in	2011-08-21 23:47:10.000000000 +0100
+++ libkate-0.4.1/doc/Makefile.in	2015-05-28 23:52:49.742054572 +0100
@@ -125,7 +125,6 @@
 LD = @LD@
 LDFLAGS = @LDFLAGS@
 LEX = @LEX@
-LEXLIB = @LEXLIB@
 LEX_OUTPUT_ROOT = @LEX_OUTPUT_ROOT@
 LIBKATE_SHARED_VERSION = @LIBKATE_SHARED_VERSION@
 LIBOBJS = @LIBOBJS@
diff -u -r libkate-0.4.1-orig/tests/Makefile.in libkate-0.4.1/tests/Makefile.in
--- libkate-0.4.1-orig/tests/Makefile.in	2011-08-21 23:46:58.000000000 +0100
+++ libkate-0.4.1/tests/Makefile.in	2015-05-28 23:52:36.542004463 +0100
@@ -156,7 +156,6 @@
 LD = @LD@
 LDFLAGS = @LDFLAGS@
 LEX = @LEX@
-LEXLIB = @LEXLIB@
 LEX_OUTPUT_ROOT = @LEX_OUTPUT_ROOT@
 LIBKATE_SHARED_VERSION = @LIBKATE_SHARED_VERSION@
 LIBOBJS = @LIBOBJS@
diff -u -r libkate-0.4.1-orig/tools/KateDJ/Makefile.in libkate-0.4.1/tools/KateDJ/Makefile.in
--- libkate-0.4.1-orig/tools/KateDJ/Makefile.in	2011-08-21 23:46:58.000000000 +0100
+++ libkate-0.4.1/tools/KateDJ/Makefile.in	2015-05-28 23:53:11.981139017 +0100
@@ -99,7 +99,6 @@
 LD = @LD@
 LDFLAGS = @LDFLAGS@
 LEX = @LEX@
-LEXLIB = @LEXLIB@
 LEX_OUTPUT_ROOT = @LEX_OUTPUT_ROOT@
 LIBKATE_SHARED_VERSION = @LIBKATE_SHARED_VERSION@
 LIBOBJS = @LIBOBJS@
diff -u -r libkate-0.4.1-orig/tools/Makefile.am libkate-0.4.1/tools/Makefile.am
--- libkate-0.4.1-orig/tools/Makefile.am	2011-08-21 13:42:10.000000000 +0100
+++ libkate-0.4.1/tools/Makefile.am	2015-05-28 23:54:03.293333914 +0100
@@ -20,7 +20,7 @@
 endif
 katedec_SOURCES=katedec.c kkate.c ksrt.c klrc.c kutil.c kfuzz.c kstream.c kread.c kstrings.c
 katalyzer_SOURCES=katalyzer.c kutil.c kstream.c kread.c kstrings.c kstats.c
-kateenc_LDADD=../lib/liboggkate.la ../lib/libkate.la @OGG_LIBS@ @PNG_LIBS@ @LEXLIB@
+kateenc_LDADD=../lib/liboggkate.la ../lib/libkate.la @OGG_LIBS@ @PNG_LIBS@
 katedec_LDADD=../lib/liboggkate.la ../lib/libkate.la @OGG_LIBS@
 katalyzer_LDADD=../lib/liboggkate.la ../lib/libkate.la @OGG_LIBS@
 kateenc_CFLAGS=@CWARNFLAGS_LIGHT@ @CFLAGS_FORTIFY_SOURCE@ @CFLAGS_DEBUG@ @OGG_CFLAGS@ @PNG_CFLAGS@
diff -u -r libkate-0.4.1-orig/tools/Makefile.in libkate-0.4.1/tools/Makefile.in
--- libkate-0.4.1-orig/tools/Makefile.in	2011-08-21 23:46:58.000000000 +0100
+++ libkate-0.4.1/tools/Makefile.in	2015-05-28 23:54:21.805404270 +0100
@@ -223,7 +223,6 @@
 LD = @LD@
 LDFLAGS = @LDFLAGS@
 LEX = @LEX@
-LEXLIB = @LEXLIB@
 LEX_OUTPUT_ROOT = @LEX_OUTPUT_ROOT@
 LIBKATE_SHARED_VERSION = @LIBKATE_SHARED_VERSION@
 LIBOBJS = @LIBOBJS@
@@ -351,7 +350,7 @@
 @HAVE_OGG_TRUE@	$(am__append_1)
 @HAVE_OGG_TRUE@katedec_SOURCES = katedec.c kkate.c ksrt.c klrc.c kutil.c kfuzz.c kstream.c kread.c kstrings.c
 @HAVE_OGG_TRUE@katalyzer_SOURCES = katalyzer.c kutil.c kstream.c kread.c kstrings.c kstats.c
-@HAVE_OGG_TRUE@kateenc_LDADD = ../lib/liboggkate.la ../lib/libkate.la @OGG_LIBS@ @PNG_LIBS@ @LEXLIB@
+@HAVE_OGG_TRUE@kateenc_LDADD = ../lib/liboggkate.la ../lib/libkate.la @OGG_LIBS@ @PNG_LIBS@
 @HAVE_OGG_TRUE@katedec_LDADD = ../lib/liboggkate.la ../lib/libkate.la @OGG_LIBS@
 @HAVE_OGG_TRUE@katalyzer_LDADD = ../lib/liboggkate.la ../lib/libkate.la @OGG_LIBS@
 @HAVE_OGG_TRUE@kateenc_CFLAGS = @CWARNFLAGS_LIGHT@ @CFLAGS_FORTIFY_SOURCE@ @CFLAGS_DEBUG@ @OGG_CFLAGS@ @PNG_CFLAGS@
