diff -r -U 2 a/ucview-0.33/src/callbacks.c b/ucview-0.33/src/callbacks.c
--- a/ucview-0.33/src/callbacks.c	2010-03-18 10:28:46.000000000 +0100
+++ b/ucview-0.33/src/callbacks.c	2015-02-10 19:29:27.329625095 +0100
@@ -229,4 +229,7 @@
       
       path = gconf_client_get_string( window->client, UCVIEW_GCONF_DIR "/video_file_path", NULL );
+      if( !path ){
+         path = g_get_home_dir();
+      }
       if( !g_file_test( path, ( G_FILE_TEST_IS_DIR ) ) )
       {
