diff -r -U 2 a/ucview-0.33/src/ucview-window.c b/ucview-0.33/src/ucview-window.c
--- a/ucview-0.33/src/ucview-window.c	2009-12-08 18:24:08.000000000 +0100
+++ b/ucview-0.33/src/ucview-window.c	2015-02-10 19:34:41.104729578 +0100
@@ -714,4 +714,7 @@
    if( !filename ){
       path = gconf_client_get_string( window->client, UCVIEW_GCONF_DIR "/still_image_file_path", NULL );
+      if( !path ){
+         path = g_get_home_dir();
+      }
       if( !g_file_test( path, ( G_FILE_TEST_IS_DIR ) ) ){
 	 g_set_error( error, UCVIEW_ERROR, 0, _("Path '%s' does not exist"), path );
@@ -755,5 +758,5 @@
    sidebar_add_image( SIDEBAR( window->sidebar ), buffer, filename );
    
-   g_free( path );
+/* g_free( path );  */
    g_free( filetype );
    g_free( filename );
