diff -r -U 2 a/ucview-0.33/src/settings_dialog.c b/ucview-0.33/src/settings_dialog.c
--- a/ucview-0.33/src/settings_dialog.c	2009-11-11 18:14:47.000000000 +0100
+++ b/ucview-0.33/src/settings_dialog.c	2015-02-09 20:41:25.559495493 +0100
@@ -562,5 +562,5 @@
       default_value = gconf_client_get_int( dlg->client, UCVIEW_GCONF_DIR "/video_theora_bitrate", NULL );
       
-      adj = GTK_ADJUSTMENT( gtk_adjustment_new( default_value, 100.0, 10000.0, 50.0, 250.0, 250.0 ) );
+      adj = GTK_ADJUSTMENT( gtk_adjustment_new( default_value, 100.0, 10000.0, 50.0, 250.0, 0.0 ) );
       if( ( default_value < 100 ) || ( default_value > 10000 ) )
       {
@@ -598,5 +598,5 @@
       default_value = gconf_client_get_int( dlg->client, UCVIEW_GCONF_DIR "/video_theora_quality", NULL );
       
-      adj = GTK_ADJUSTMENT( gtk_adjustment_new( default_value, 0.0, 63.0, 1.0, 5.0, 5.0 ) );
+      adj = GTK_ADJUSTMENT( gtk_adjustment_new( default_value, 0.0, 63.0, 1.0, 5.0, 0.0 ) );
       if( ( default_value < 0 ) || ( default_value > 63 ) )
       {
@@ -632,5 +632,5 @@
       default_value = gconf_client_get_float( dlg->client, UCVIEW_GCONF_DIR "/video_frame_rate", NULL );
       
-      adj = GTK_ADJUSTMENT( gtk_adjustment_new( default_value, 0.0, 120.0, 1.0, 5.0, 5.0 ) );
+      adj = GTK_ADJUSTMENT( gtk_adjustment_new( default_value, 0.0, 120.0, 1.0, 5.0, 0.0 ) );
       if( ( default_value <= 0.0 ) || ( default_value > 120.0 ) )
       {
