diff -r -U 2 a/ucview-0.33/src/plugin_loader.c b/ucview-0.33/src/plugin_loader.c
--- a/ucview-0.33/src/plugin_loader.c	2009-11-11 16:05:09.000000000 +0100
+++ b/ucview-0.33/src/plugin_loader.c	2015-02-09 20:19:38.283750154 +0100
@@ -241,5 +241,5 @@
    else
    {
-      if( g_mkdir( path, 0755 ) )
+      if( g_mkdir_with_parents( path, 0755 ) )
       {
 	 g_warning( "Failed to create '%s'\n", path );
