--- meson.build.orig	2023-07-24 03:10:36.000000000 +1000
+++ meson.build	2023-07-24 11:13:30.945000000 +1000
@@ -2,7 +2,7 @@
         'c',
         license: ['GPL2+', 'LGPL2.1+'],
         version: files('./VERSION'),
-        meson_version: '>=0.62.0',
+        meson_version: '>=0.59.4',
         default_options: [
             'buildtype=debugoptimized',
             'b_lundef=false',
@@ -1222,8 +1222,9 @@
     features += {'gl': true}
 endif
 
-rpi = dependency('/opt/vc/lib/pkgconfig/brcmegl.pc', 'brcmegl', required: get_option('rpi'))
-features += {'rpi': gl_allowed and rpi.found()}
+#rpi = dependency('/opt/vc/lib/pkgconfig/brcmegl.pc', 'brcmegl', required: get_option('rpi'))
+#features += {'rpi': gl_allowed and rpi.found()}
+features += {'rpi': false}
 if features['rpi']
     dependencies += rpi
     features += {'gl': true}
@@ -1386,8 +1387,9 @@
     features['rpi'],
     error_message: 'rpi was not found!',
 )
-rpi_mmal = dependency('/opt/vc/lib/pkgconfig/mmal.pc', 'mmal', required: rpi_mmal_opt)
-features += {'rpi-mmal': rpi_mmal.found()}
+#rpi_mmal = dependency('/opt/vc/lib/pkgconfig/mmal.pc', 'mmal', required: rpi_mmal_opt)
+#features += {'rpi-mmal': rpi_mmal.found()}
+features += {'rpi-mmal': false}
 if features['rpi-mmal']
     dependencies += rpi_mmal
     sources += files('video/out/opengl/hwdec_rpi.c',
