From 96378f7975ffae9d8b789daf14daf4bdc937195a Mon Sep 17 00:00:00 2001
From: "William J. Bowman" <wjb@williamjbowman.com>
Date: Thu, 26 Jul 2012 12:37:55 -0400
Subject: [PATCH] Edit to fix declaration mismatch.

---
 heaplayers/wrappers/gnuwrapper.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/heaplayers/wrappers/gnuwrapper.cpp b/heaplayers/wrappers/gnuwrapper.cpp
index f1c4aee..1fff480 100644
--- a/heaplayers/wrappers/gnuwrapper.cpp
+++ b/heaplayers/wrappers/gnuwrapper.cpp
@@ -65,7 +65,7 @@ extern "C" {
   static void * (*old_realloc_hook)(void *ptr, size_t size, const void *caller);
   static void * (*old_memalign_hook)(size_t alignment, size_t size, const void *caller);
 
-  void (* __malloc_initialize_hook) (void) = my_init_hook;
+  void (* volatile  __malloc_initialize_hook) (void) = my_init_hook;
 
   static void my_init_hook (void) {
     // Store the old hooks.
-- 
1.7.11.3

