--- Makefile.orig	2018-03-29 09:57:17.598456271 -0700
+++ Makefile	2018-03-29 09:58:53.390396527 -0700
@@ -1,18 +1,20 @@
 
 CC = gcc
 
-CFLAGS = -Wall			 	\
+DEF_FLAGS = -Wall			 	\
 	-DG_DISABLE_DEPRECATED 	 	\
 	-DGDK_DISABLE_DEPRECATED 	\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED \
 	-DGTK_DISABLE_DEPRECATED
 
+CFLAGS = -O2
+
 GKT =  `pkg-config gtk+-2.0 --cflags --libs`
 
 all: myGtkMenu
 
 myGtkMenu: main.c
-	$(CC) main.c -o myGtkMenu $(CFLAGS) $(GKT)
+	$(CC) main.c -o myGtkMenu $(CFLAGS) $(DEF_FLAGS) $(GKT)
 
 clean:
 	rm -f *.o myGtkMenu
