diff -Naur mygui-MyGUI3.4.1.orig/CMakeLists.txt mygui-MyGUI3.4.1/CMakeLists.txt
--- mygui-MyGUI3.4.1.orig/CMakeLists.txt	2021-03-01 00:41:43.000000000 +0100
+++ mygui-MyGUI3.4.1/CMakeLists.txt	2025-07-14 09:20:55.636062000 +0200
@@ -8,7 +8,7 @@
 cmake_minimum_required(VERSION 2.8.12)
 cmake_policy(SET CMP0003 NEW)
 cmake_policy(SET CMP0011 NEW)
-cmake_policy(SET CMP0048 OLD)
+cmake_policy(SET CMP0048 NEW)
 
 # for link time optimization, remove if cmake version is >= 3.9
 if(POLICY CMP0069) # LTO
