diff -Naur src/libtiled/libtiled.pro src/libtiled/libtiled.pro
--- src/libtiled/libtiled.pro	2019-11-12 15:18:10.000000000 +0100
+++ src/libtiled/libtiled.pro	2019-11-12 21:20:24.366384856 +0100
@@ -8,7 +8,7 @@
     DESTDIR = ../../bin/Tiled.app/Contents/Frameworks
     QMAKE_LFLAGS_SONAME = -Wl,-install_name,@executable_path/../Frameworks/
 } else {
-    DESTDIR = ../../lib
+    DESTDIR = ../../lib64
 }
 DLLDESTDIR = ../..
 
diff -Naur src/libtiled/pluginmanager.cpp src/libtiled/pluginmanager.cpp
--- src/libtiled/pluginmanager.cpp	2019-11-12 15:18:10.000000000 +0100
+++ src/libtiled/pluginmanager.cpp	2019-11-12 21:20:24.366384856 +0100
@@ -196,7 +196,7 @@
 #elif defined(TILED_PLUGIN_DIR)
     QString pluginPath = QLatin1String(TILED_PLUGIN_DIR);
 #else
-    pluginPath += QLatin1String("/../lib/tiled/plugins");
+    pluginPath += QLatin1String("/../lib64/tiled/plugins");
 #endif
 
     // Load dynamic plugins
diff -Naur src/plugins/plugin.pri src/plugins/plugin.pri
--- src/plugins/plugin.pri	2019-11-12 15:18:10.000000000 +0100
+++ src/plugins/plugin.pri	2019-11-12 21:20:24.366384856 +0100
@@ -10,7 +10,7 @@
 } else:macx {
     DESTDIR = $$OUT_PWD/../../../bin/Tiled.app/Contents/PlugIns
 } else {
-    DESTDIR = $$OUT_PWD/../../../lib/tiled/plugins
+    DESTDIR = $$OUT_PWD/../../../lib64/tiled/plugins
 }
 
 include(../../tiled.pri)
@@ -21,9 +21,9 @@
 macx {
     QMAKE_LIBDIR += $$OUT_PWD/../../../bin/Tiled.app/Contents/Frameworks
 } else:win32 {
-    LIBS += -L$$OUT_PWD/../../../lib
+    LIBS += -L$$OUT_PWD/../../../lib64
 } else {
-    QMAKE_LIBDIR += $$OUT_PWD/../../../lib
+    QMAKE_LIBDIR += $$OUT_PWD/../../../lib64
 }
 
 # Set rpath so that the plugin will resolve libtiled correctly
diff -Naur src/tiled/tiled.pro src/tiled/tiled.pro
--- src/tiled/tiled.pro	2019-11-12 15:18:10.000000000 +0100
+++ src/tiled/tiled.pro	2019-11-12 21:20:24.366384856 +0100
@@ -28,12 +28,12 @@
 } else:win32 {
     LIBS += -L$$OUT_PWD/../../lib
 } else {
-    QMAKE_LIBDIR = $$OUT_PWD/../../lib $$QMAKE_LIBDIR
+    QMAKE_LIBDIR = $$OUT_PWD/../../lib64 $$QMAKE_LIBDIR
 }
 
 # Make sure the Tiled executable can find libtiled
 !win32:!macx:!cygwin:contains(RPATH, yes) {
-    QMAKE_RPATHDIR += \$\$ORIGIN/../lib
+    QMAKE_RPATHDIR += \$\$ORIGIN/../lib64
 
     # It is not possible to use ORIGIN in QMAKE_RPATHDIR, so a bit manually
     QMAKE_LFLAGS += -Wl,-z,origin \'-Wl,-rpath,$$join(QMAKE_RPATHDIR, ":")\'
diff -Naur src/tmxrasterizer/tmxrasterizer.pro src/tmxrasterizer/tmxrasterizer.pro
--- src/tmxrasterizer/tmxrasterizer.pro	2019-11-12 15:18:10.000000000 +0100
+++ src/tmxrasterizer/tmxrasterizer.pro	2019-11-12 21:20:24.366384856 +0100
@@ -19,12 +19,12 @@
 } else:win32 {
     LIBS += -L$$OUT_PWD/../../lib
 } else {
-    QMAKE_LIBDIR = $$OUT_PWD/../../lib $$QMAKE_LIBDIR
+    QMAKE_LIBDIR = $$OUT_PWD/../../lib64 $$QMAKE_LIBDIR
 }
 
 # Make sure the executable can find libtiled
 !win32:!macx:!cygwin:contains(RPATH, yes) {
-    QMAKE_RPATHDIR += \$\$ORIGIN/../lib
+    QMAKE_RPATHDIR += \$\$ORIGIN/../lib64
 
     # It is not possible to use ORIGIN in QMAKE_RPATHDIR, so a bit manually
     QMAKE_LFLAGS += -Wl,-z,origin \'-Wl,-rpath,$$join(QMAKE_RPATHDIR, ":")\'
diff -Naur src/tmxviewer/tmxviewer.pro src/tmxviewer/tmxviewer.pro
--- src/tmxviewer/tmxviewer.pro	2019-11-12 15:18:10.000000000 +0100
+++ src/tmxviewer/tmxviewer.pro	2019-11-12 21:20:24.366384856 +0100
@@ -19,12 +19,12 @@
 } else:win32 {
     LIBS += -L$$OUT_PWD/../../lib
 } else {
-    QMAKE_LIBDIR = $$OUT_PWD/../../lib $$QMAKE_LIBDIR
+    QMAKE_LIBDIR = $$OUT_PWD/../../lib64 $$QMAKE_LIBDIR
 }
 
 # Make sure the executable can find libtiled
 !win32:!macx:!cygwin:contains(RPATH, yes) {
-    QMAKE_RPATHDIR += \$\$ORIGIN/../lib
+    QMAKE_RPATHDIR += \$\$ORIGIN/../lib64
 
     # It is not possible to use ORIGIN in QMAKE_RPATHDIR, so a bit manually
     QMAKE_LFLAGS += -Wl,-z,origin \'-Wl,-rpath,$$join(QMAKE_RPATHDIR, ":")\'
diff -Naur src/terraingenerator/terraingenerator.pro src/terraingenerator/terraingenerator.pro
--- src/terraingenerator/terraingenerator.pro	2019-11-12 15:18:10.000000000 +0100
+++ src/terraingenerator/terraingenerator.pro	2019-11-12 21:20:24.367384856 +0100
@@ -19,12 +19,12 @@
 } else:win32 {
     LIBS += -L$$OUT_PWD/../../lib
 } else {
-    QMAKE_LIBDIR_FLAGS += -L$$OUT_PWD/../../lib
+    QMAKE_LIBDIR_FLAGS += -L$$OUT_PWD/../../lib64
 }
 
 # Make sure the executable can find libtiled
 !win32:!macx:!cygwin:contains(RPATH, yes) {
-    QMAKE_RPATHDIR += \$\$ORIGIN/../lib
+    QMAKE_RPATHDIR += \$\$ORIGIN/../lib64
 
     # It is not possible to use ORIGIN in QMAKE_RPATHDIR, so a bit manually
     QMAKE_LFLAGS += -Wl,-z,origin \'-Wl,-rpath,$$join(QMAKE_RPATHDIR, ":")\'
diff -Naur src/tiledquickplugin/tiledquickplugin.pro src/tiledquickplugin/tiledquickplugin.pro
--- src/tiledquickplugin/tiledquickplugin.pro	2019-11-12 15:18:10.000000000 +0100
+++ src/tiledquickplugin/tiledquickplugin.pro	2019-11-12 21:50:05.469424699 +0100
@@ -17,12 +17,12 @@
 } else:win32 {
     LIBS += -L$$OUT_PWD/../../lib
 } else {
-    QMAKE_LIBDIR = $$OUT_PWD/../../lib $$QMAKE_LIBDIR
+    QMAKE_LIBDIR = $$OUT_PWD/../../lib64 $$QMAKE_LIBDIR
 }
 
 # Make sure the Tiled executable can find libtiled
 !win32:!macx:contains(RPATH, yes) {
-    QMAKE_RPATHDIR += \$\$ORIGIN/../../../../lib
+    QMAKE_RPATHDIR += \$\$ORIGIN/../../../../lib64
 
     # It is not possible to use ORIGIN in QMAKE_RPATHDIR, so a bit manually
     QMAKE_LFLAGS += -Wl,-z,origin \'-Wl,-rpath,$$join(QMAKE_RPATHDIR, ":")\'
