diff -Naur src/libtiled/libtiled.pro src/libtiled/libtiled.pro
--- src/libtiled/libtiled.pro	2017-09-21 10:28:51.737708697 +0200
+++ src/libtiled/libtiled.pro	2017-09-20 16:53:45.976355498 +0200
@@ -8,7 +8,7 @@
     DESTDIR = ../../bin/Tiled.app/Contents/Frameworks
     QMAKE_LFLAGS_SONAME = -Wl,-install_name,@executable_path/../Frameworks/
 } else {
-    DESTDIR = ../../lib
+    DESTDIR = ../../lib64
 }
 DLLDESTDIR = ../..
 
diff -Naur src/libtiled/pluginmanager.cpp src/libtiled/pluginmanager.cpp
--- src/libtiled/pluginmanager.cpp	2017-09-21 10:28:51.739708697 +0200
+++ src/libtiled/pluginmanager.cpp	2017-09-21 09:25:05.377930367 +0200
@@ -201,7 +201,7 @@
 #elif defined(TILED_PLUGIN_DIR)
     QString pluginPath = QLatin1String(TILED_PLUGIN_DIR);
 #else
-    pluginPath += QLatin1String("/../lib/tiled/plugins");
+    pluginPath += QLatin1String("/../lib64/tiled/plugins");
 #endif
 
     // Load dynamic plugins
diff -Naur src/plugins/plugin.pri src/plugins/plugin.pri
--- src/plugins/plugin.pri	2017-09-21 10:28:51.740708697 +0200
+++ src/plugins/plugin.pri	2017-09-20 16:11:02.307504018 +0200
@@ -10,7 +10,7 @@
 } else:macx {
     DESTDIR = $$OUT_PWD/../../../bin/Tiled.app/Contents/PlugIns
 } else {
-    DESTDIR = $$OUT_PWD/../../../lib/tiled/plugins
+    DESTDIR = $$OUT_PWD/../../../lib64/tiled/plugins
 }
 
 include(../../tiled.pri)
@@ -21,9 +21,9 @@
 macx {
     QMAKE_LIBDIR += $$OUT_PWD/../../../bin/Tiled.app/Contents/Frameworks
 } else:win32 {
-    LIBS += -L$$OUT_PWD/../../../lib
+    LIBS += -L$$OUT_PWD/../../../lib64
 } else {
-    QMAKE_LIBDIR += $$OUT_PWD/../../../lib
+    QMAKE_LIBDIR += $$OUT_PWD/../../../lib64
 }
 
 # Set rpath so that the plugin will resolve libtiled correctly
diff -Naur src/tiled/tiled.pro src/tiled/tiled.pro
--- src/tiled/tiled.pro	2017-09-21 10:28:51.750708696 +0200
+++ src/tiled/tiled.pro	2017-09-20 15:52:06.112569840 +0200
@@ -58,12 +58,12 @@
 } else:win32 {
     LIBS += -L$$OUT_PWD/../../lib
 } else {
-    QMAKE_LIBDIR = $$OUT_PWD/../../lib $$QMAKE_LIBDIR
+    QMAKE_LIBDIR = $$OUT_PWD/../../lib64 $$QMAKE_LIBDIR
 }
 
 # Make sure the Tiled executable can find libtiled
 !win32:!macx:!cygwin:contains(RPATH, yes) {
-    QMAKE_RPATHDIR += \$\$ORIGIN/../lib
+    QMAKE_RPATHDIR += \$\$ORIGIN/../lib64
 
     # It is not possible to use ORIGIN in QMAKE_RPATHDIR, so a bit manually
     QMAKE_LFLAGS += -Wl,-z,origin \'-Wl,-rpath,$$join(QMAKE_RPATHDIR, ":")\'
diff -Naur src/tmxrasterizer/tmxrasterizer.pro src/tmxrasterizer/tmxrasterizer.pro
--- src/tmxrasterizer/tmxrasterizer.pro	2017-09-21 11:18:41.212535509 +0200
+++ src/tmxrasterizer/tmxrasterizer.pro	2017-09-21 11:18:29.837536168 +0200
@@ -19,12 +19,12 @@
 } else:win32 {
     LIBS += -L$$OUT_PWD/../../lib
 } else {
-    QMAKE_LIBDIR = $$OUT_PWD/../../lib $$QMAKE_LIBDIR
+    QMAKE_LIBDIR = $$OUT_PWD/../../lib64 $$QMAKE_LIBDIR
 }
 
 # Make sure the executable can find libtiled
 !win32:!macx:!cygwin:contains(RPATH, yes) {
-    QMAKE_RPATHDIR += \$\$ORIGIN/../lib
+    QMAKE_RPATHDIR += \$\$ORIGIN/../lib64
 
     # It is not possible to use ORIGIN in QMAKE_RPATHDIR, so a bit manually
     QMAKE_LFLAGS += -Wl,-z,origin \'-Wl,-rpath,$$join(QMAKE_RPATHDIR, ":")\'
diff -Naur src/tmxviewer/tmxviewer.pro src/tmxviewer/tmxviewer.pro
--- src/tmxviewer/tmxviewer.pro	2017-09-21 11:18:12.497537173 +0200
+++ src/tmxviewer/tmxviewer.pro	2017-09-21 11:17:51.468538391 +0200
@@ -19,12 +19,12 @@
 } else:win32 {
     LIBS += -L$$OUT_PWD/../../lib
 } else {
-    QMAKE_LIBDIR = $$OUT_PWD/../../lib $$QMAKE_LIBDIR
+    QMAKE_LIBDIR = $$OUT_PWD/../../lib64 $$QMAKE_LIBDIR
 }
 
 # Make sure the executable can find libtiled
 !win32:!macx:!cygwin:contains(RPATH, yes) {
-    QMAKE_RPATHDIR += \$\$ORIGIN/../lib
+    QMAKE_RPATHDIR += \$\$ORIGIN/../lib64
 
     # It is not possible to use ORIGIN in QMAKE_RPATHDIR, so a bit manually
     QMAKE_LFLAGS += -Wl,-z,origin \'-Wl,-rpath,$$join(QMAKE_RPATHDIR, ":")\'
diff -Naur src/terraingenerator/terraingenerator.pro src/terraingenerator/terraingenerator.pro
--- src/terraingenerator/terraingenerator.pro	2017-08-29 09:06:51.000000000 +0200
+++ src/terraingenerator/terraingenerator.pro	2017-09-21 11:19:06.508534044 +0200
@@ -19,12 +19,12 @@
 } else:win32 {
     LIBS += -L$$OUT_PWD/../../lib
 } else {
-    QMAKE_LIBDIR_FLAGS += -L$$OUT_PWD/../../lib
+    QMAKE_LIBDIR_FLAGS += -L$$OUT_PWD/../../lib64
 }
 
 # Make sure the executable can find libtiled
 !win32:!macx:!cygwin:contains(RPATH, yes) {
-    QMAKE_RPATHDIR += \$\$ORIGIN/../lib
+    QMAKE_RPATHDIR += \$\$ORIGIN/../lib64
 
     # It is not possible to use ORIGIN in QMAKE_RPATHDIR, so a bit manually
     QMAKE_LFLAGS += -Wl,-z,origin \'-Wl,-rpath,$$join(QMAKE_RPATHDIR, ":")\'
diff -Naur src/automappingconverter/automappingconverter.pro src/automappingconverter/automappingconverter.pro
--- src/automappingconverter/automappingconverter.pro	2017-09-21 12:00:09.574391353 +0200
+++ src/automappingconverter/automappingconverter.pro	2017-09-21 11:59:56.557392107 +0200
@@ -12,12 +12,12 @@
 } else:win32 {
     LIBS += -L$$OUT_PWD/../../lib
 } else {
-    QMAKE_LIBDIR = $$OUT_PWD/../../lib $$QMAKE_LIBDIR
+    QMAKE_LIBDIR = $$OUT_PWD/../../lib64 $$QMAKE_LIBDIR
 }
 
 # Make sure the executable can find libtiled
 !win32:!macx:!cygwin:contains(RPATH, yes) {
-    QMAKE_RPATHDIR += \$\$ORIGIN/../lib
+    QMAKE_RPATHDIR += \$\$ORIGIN/../lib64
 
     # It is not possible to use ORIGIN in QMAKE_RPATHDIR, so a bit manually
     QMAKE_LFLAGS += -Wl,-z,origin \'-Wl,-rpath,$$join(QMAKE_RPATHDIR, ":")\'
