diff -up grantlee-5.1.0/CMakeLists.txt.orig grantlee-5.1.0/CMakeLists.txt
--- grantlee-5.1.0/CMakeLists.txt.orig	2016-04-19 01:33:17.000000000 -0500
+++ grantlee-5.1.0/CMakeLists.txt	2016-04-19 06:44:06.371122149 -0500
@@ -100,7 +100,7 @@ set (LIB_SUFFIX "" CACHE STRING "Define
 set( LIB_INSTALL_DIR lib${LIB_SUFFIX} )
 set( PLUGIN_INSTALL_DIR ${LIB_INSTALL_DIR}/grantlee/${Grantlee5_MAJOR_MINOR_VERSION_STRING} )
 set( BIN_INSTALL_DIR bin )
-set( INCLUDE_INSTALL_DIR include )
+set( INCLUDE_INSTALL_DIR include/Grantlee5 )
 set( DATA_INSTALL_DIR share/apps )
 
 # set up RPATH/install_name_dir
diff -up grantlee-5.1.0/templates/lib/CMakeLists.txt.orig grantlee-5.1.0/templates/lib/CMakeLists.txt
--- grantlee-5.1.0/templates/lib/CMakeLists.txt.orig	2016-04-19 06:44:06.371122149 -0500
+++ grantlee-5.1.0/templates/lib/CMakeLists.txt	2016-04-19 06:44:55.763336264 -0500
@@ -107,6 +107,10 @@ if (Qt5Script_FOUND)
 
   target_sources(Grantlee_Templates PRIVATE ${scriptabletags_SRCS})
   target_include_directories(Grantlee_Templates PRIVATE ../scriptabletags)
+  target_include_directories(Grantlee_Templates
+    INTERFACE "$<INSTALL_INTERFACE:${INCLUDE_INSTALL_DIR}>"
+  )
+
   target_link_libraries(Grantlee_Templates
     LINK_PRIVATE Qt5::Script
   )
diff -up grantlee-5.1.0/textdocument/lib/CMakeLists.txt.orig grantlee-5.1.0/textdocument/lib/CMakeLists.txt
--- grantlee-5.1.0/textdocument/lib/CMakeLists.txt.orig	2016-04-19 06:44:06.371122149 -0500
+++ grantlee-5.1.0/textdocument/lib/CMakeLists.txt	2016-04-19 06:45:56.268598553 -0500
@@ -43,6 +43,10 @@ if (CMAKE_GENERATOR MATCHES "Visual Stud
   endforeach()
 endif()
 
+target_include_directories(Grantlee_TextDocument
+    INTERFACE "$<INSTALL_INTERFACE:${INCLUDE_INSTALL_DIR}>"
+)
+
 target_link_libraries(Grantlee_TextDocument
   LINK_PUBLIC Qt5::Gui
 )
