--- usr/bin/teams.old	2022-03-21 18:53:40.958533543 -0300
+++ usr/bin/teams	2022-03-21 18:54:01.813813567 -0300
@@ -8,5 +8,5 @@
 
 mkdir -p "$TEAMS_LOGS"
 
-nohup "$TEAMS_PATH" "$@" --disable-namespace-sandbox --disable-setuid-sandbox > "$TEAMS_LOGS/teams-startup.log" 2>&1 &
+nohup "$TEAMS_PATH" "$@" --disable-namespace-sandbox --disable-setuid-sandbox --disable-seccomp-filter-sandbox > "$TEAMS_LOGS/teams-startup.log" 2>&1 &
 
