--- mysqlcc-0.9.8-src/src/main.cpp	2006-08-11 17:29:12.000000000 -0300
+++ mysqlcc-0.9.8-src_patched/src/main.cpp	2010-06-17 08:49:29.000000000 -0300
@@ -90,28 +90,28 @@
 
   {"version", 'V', "Print version information and exit.", 0, 0, 0, GET_NO_ARG, NO_ARG, 0, 0, 0, 0, 0, 0},
 
-  {"connect_timeout", OPT_CONNECT_TIMEOUT, "", (gptr*) &opt_connect_timeout,
-   (gptr*) &opt_connect_timeout, 0, GET_ULONG, REQUIRED_ARG, 0, 0, 3600*12, 0,
-   0, 1},
+  {"connect_timeout", OPT_CONNECT_TIMEOUT, "", (uchar**) &opt_connect_timeout,
+   (uchar**) &opt_connect_timeout, 0, GET_ULONG, REQUIRED_ARG, 0, 0, 3600*12, 0,
+   0, (void*)1},
 
-  {"select_limit", OPT_SELECT_LIMIT, "", (gptr*) &select_limit,
-   (gptr*) &select_limit, 0, GET_ULONG, REQUIRED_ARG, 1000L, 1, ~0L, 0, 1, 0},
+  {"select_limit", OPT_SELECT_LIMIT, "", (uchar**) &select_limit,
+   (uchar**) &select_limit, 0, GET_ULONG, REQUIRED_ARG, 1000L, 1, ~0L, 0, 1, 0},
   
   {"net_buffer_length", OPT_NET_BUFFER_LENGTH, "",
-   (gptr*) &my_net_buffer_length, (gptr*) &my_net_buffer_length, 0, GET_ULONG,
+   (uchar**) &my_net_buffer_length, (uchar**) &my_net_buffer_length, 0, GET_ULONG,
    REQUIRED_ARG, 16384, 1024, 512*1024*1024L, 0, 1024, 0},
 
   {"max_allowed_packet", OPT_MAX_ALLOWED_PACKET, "",
-   (gptr*) &my_max_allowed_packet, (gptr*) &my_max_allowed_packet, 0, GET_ULONG,
+   (uchar**) &my_max_allowed_packet, (uchar**) &my_max_allowed_packet, 0, GET_ULONG,
    REQUIRED_ARG, 16 *1024L*1024L, 4096, 512*1024L*1024L, 0,
    1024, 0},
 
   {"local-infile", OPT_LOCAL_INFILE, "Enable/disable LOAD DATA LOCAL INFILE.",
-   (gptr*) &opt_local_infile,
-   (gptr*) &opt_local_infile, 0, GET_BOOL, OPT_ARG, 0, 0, 0, 0, 0, 0},
+   (uchar**) &opt_local_infile,
+   (uchar**) &opt_local_infile, 0, GET_BOOL, OPT_ARG, 0, 0, 0, 0, 0, 0},
 
-  {"max_join_size", OPT_MAX_JOIN_SIZE, "", (gptr*) &max_join_size,
-   (gptr*) &max_join_size, 0, GET_ULONG, REQUIRED_ARG, 1000000L, 1, ~0L, 0, 1,
+  {"max_join_size", OPT_MAX_JOIN_SIZE, "", (uchar**) &max_join_size,
+   (uchar**) &max_join_size, 0, GET_ULONG, REQUIRED_ARG, 1000000L, 1, ~0L, 0, 1,
    0},
 
   { 0, 0, 0, 0, 0, 0, GET_NO_ARG, NO_ARG, 0, 0, 0, 0, 0, 0}
@@ -155,7 +155,7 @@
           break;
       }
       if (opt->var_type == GET_NO_ARG && opt->arg_type == NO_ARG)  //Boolean options
-        add_argument(id, "");
+        add_argument(id, (char*)"");
       else
         add_argument(id, argument);
     }
