diff -Naur svkbd-0.2.1/drw.c svkbd-0.2.1.patched/drw.c
--- svkbd-0.2.1/drw.c	2020-09-18 07:02:35.000000000 -0400
+++ svkbd-0.2.1.patched/drw.c	2020-11-02 12:45:36.377138299 -0500
@@ -133,6 +133,7 @@
 		die("no font specified.");
 	}
 
+#ifdef FC_COLOR
 	/* Do not allow using color fonts. This is a workaround for a BadLength
 	 * error from Xft with color glyphs. Modelled on the Xterm workaround. See
 	 * https://bugzilla.redhat.com/show_bug.cgi?id=1498269
@@ -145,6 +146,7 @@
 		XftFontClose(drw->dpy, xfont);
 		return NULL;
 	}
+#endif
 
 	font = ecalloc(1, sizeof(Fnt));
 	font->xfont = xfont;
@@ -351,7 +353,9 @@
 			fcpattern = FcPatternDuplicate(drw->fonts->pattern);
 			FcPatternAddCharSet(fcpattern, FC_CHARSET, fccharset);
 			FcPatternAddBool(fcpattern, FC_SCALABLE, FcTrue);
+#ifdef FC_COLOR
 			FcPatternAddBool(fcpattern, FC_COLOR, FcFalse);
+#endif
 
 			FcConfigSubstitute(NULL, fcpattern, FcMatchPattern);
 			FcDefaultSubstitute(fcpattern);
