diff -Naur spectranet-tnfs-fuse-fd46ffca3281db4a0c60c04dc83c9cf2cb80bc9e/tnfs-fuse.py spectranet-tnfs-fuse-fd46ffca3281db4a0c60c04dc83c9cf2cb80bc9e.patched/tnfs-fuse.py
--- spectranet-tnfs-fuse-fd46ffca3281db4a0c60c04dc83c9cf2cb80bc9e/tnfs-fuse.py	2013-01-27 13:02:20.000000000 -0500
+++ spectranet-tnfs-fuse-fd46ffca3281db4a0c60c04dc83c9cf2cb80bc9e.patched/tnfs-fuse.py	2022-07-18 14:47:20.902204356 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # The MIT License
 #
@@ -22,6 +22,9 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
+import sys
+sys.path.insert(0, '@LIBDIR@')
+
 import fuse
 fuse.fuse_python_api = (0, 2)
 
