diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -9810,7 +9810,7 @@
 
     if test -z "$_ver"; then
         ## If empty, try date [fontforge]
-        eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep '[0-9]\{6,8\}' \
+        eval _ver=\"\`("$exe" --version || "$exe" -V) 2>&1 | grep 'libfontforge [0-9]\{6,8\}' \
 	    | head -n 1 \
 	    | sed -e 's/^[^.0-9]*//' -e 's/[^.0-9]*$//'\`\"
     fi
