--- src/src.pro~	2009-08-08 11:37:32.000000000 +0300
+++ src/src.pro	2010-02-13 00:52:54.000000000 +0200
@@ -62,7 +62,12 @@
     system( ./pcfile.sh $${INSTALL_PREFIX} $${VERSION} )
     pkgconfig.files = qoauth.pc
 
-    target.path = $${INSTALL_PREFIX}/lib
+    contains(QMAKE_HOST.arch, x86_64) {
+      target.path = $${INSTALL_PREFIX}/lib64
+    } else {
+      target.path = $${INSTALL_PREFIX}/lib
+    }
+
     headers.path = $${INSTALL_PREFIX}/include/QtOAuth
     docs.path = $${INSTALL_PREFIX}/share/doc/$${TARGET}-$${VERSION}
     pkgconfig.path = $${target.path}/pkgconfig
