diff -Naur krename-r247.orig/src/batchrenamer.cpp krename-r247/src/batchrenamer.cpp
--- krename-r247.orig/src/batchrenamer.cpp	2013-11-24 22:46:05.000000000 +0100
+++ krename-r247/src/batchrenamer.cpp	2017-06-20 14:42:57.330437870 +0200
@@ -402,7 +402,7 @@
                 break;
             case eRenameMode_Link:
                 // In case of link delete created file
-                job = KIO::file_delete( dstUrl, false );
+                job = KIO::file_delete( dstUrl, NULL );
                 break;
             case eRenameMode_Copy: // no undo possible
                 // TODO: Maybe we should delete the created files
