diff -Naur vera-1.21a/vera.b vera-1.21a.patched/vera.b
--- vera-1.21a/vera.b	2014-01-04 19:20:08.000000000 -0500
+++ vera-1.21a.patched/vera.b	2014-07-29 14:43:13.000000000 -0400
@@ -801,7 +801,8 @@
 @item BP
 Base Pointer [register] (CPU, Intel, assembler)
 
-BP;DR        Behind Paywall; Didn't Read (slang, Usenet, IRC)
+@item BP;DR
+Behind Paywall; Didn't Read (slang, Usenet, IRC)
 
 @item BPB
 BIOS Parameter Block (BIOS, DOS, HDD, FDD)
@@ -812,6 +813,9 @@
 @item BPD
 BankParameterDaten (DDBAC)
 
+@item BPDR
+Behind Paywall; Didn't Read (slang, Usenet, IRC)
+
 @item BPDU
 Bridge Protocol Data Unit (PDU)
 
diff -Naur vera-1.21a/vera.t vera-1.21a.patched/vera.t
--- vera-1.21a/vera.t	2014-01-04 19:20:08.000000000 -0500
+++ vera-1.21a.patched/vera.t	2014-07-29 14:44:06.000000000 -0400
@@ -789,7 +789,8 @@
 @item TL
 Transform and Lighting (3D), "T&L"
 
-TL;DR        Too Long; Didn't Read (slang, Usenet, IRC)
+@item TL;DR
+Too Long; Didn't Read (slang, Usenet, IRC)
 
 @item TLA
 Three Letter Acronym (slang)
@@ -806,6 +807,9 @@
 @item TLDP
 The Linux Documentation Project (Linux), http://www.tldp.org
 
+@item TLDR
+Too Long; Didn't Read (slang, Usenet, IRC)
+
 @item TLER
 Time Limited Error Recovery (WD, RAID)
 
