--- t-engine4-src-1.1.5/build/physfs.make.original	2014-03-17 11:20:03.904158009 +0200
+++ t-engine4-src-1.1.5/build/physfs.make	2014-03-17 11:20:43.826658909 +0200
@@ -32,7 +32,7 @@
   TARGETDIR  = ../bin/Debug
   TARGET     = $(TARGETDIR)/libphysfs.a
   DEFINES   += -DGLEW_STATIC -DPHYSFS_SUPPORTS_ZIP
-  INCLUDES  += -I../src -I../src/luasocket -I../src/fov -I../src/expat -I../src/lxp -I../src/libtcod_import -I../src/physfs -I../src/zlib -I../src/bzip2 -I/opt/SDL-2.0/include/SDL2 -I/usr/include/GL -I../src/luajit2/src -I../src/luajit2/dynasm
+  INCLUDES  += -I../src -I../src/luasocket -I../src/fov -I../src/expat -I../src/lxp -I../src/libtcod_import -I../src/physfs -I../src/zlib -I../src/bzip2 -I/usr/include/SDL2 -I/usr/include/GL -I../src/luajit2/src -I../src/luajit2/dynasm
   CPPFLAGS  += -MMD -MP $(DEFINES) $(INCLUDES)
   CFLAGS    += $(CPPFLAGS) $(ARCH) -g -ggdb -O3
   CXXFLAGS  += $(CFLAGS) 
@@ -54,7 +54,7 @@
   TARGETDIR  = ../bin/Release
   TARGET     = $(TARGETDIR)/libphysfs.a
   DEFINES   += -DGLEW_STATIC -DNDEBUG=1 -DPHYSFS_SUPPORTS_ZIP
-  INCLUDES  += -I../src -I../src/luasocket -I../src/fov -I../src/expat -I../src/lxp -I../src/libtcod_import -I../src/physfs -I../src/zlib -I../src/bzip2 -I/opt/SDL-2.0/include/SDL2 -I/usr/include/GL -I../src/luajit2/src -I../src/luajit2/dynasm
+  INCLUDES  += -I../src -I../src/luasocket -I../src/fov -I../src/expat -I../src/lxp -I../src/libtcod_import -I../src/physfs -I../src/zlib -I../src/bzip2 -I/usr/include/SDL2 -I/usr/include/GL -I../src/luajit2/src -I../src/luajit2/dynasm
   CPPFLAGS  += -MMD -MP $(DEFINES) $(INCLUDES)
   CFLAGS    += $(CPPFLAGS) $(ARCH) -O2 -fomit-frame-pointer -O2
   CXXFLAGS  += $(CFLAGS) 
