--- t-engine4-src-1.1.5/build/TEngine.make.original	2014-03-17 11:45:51.098815414 +0200
+++ t-engine4-src-1.1.5/build/TEngine.make	2014-03-17 11:46:28.742344806 +0200
@@ -32,7 +32,7 @@
   TARGETDIR  = ../bin/Debug
   TARGET     = $(TARGETDIR)/t-engine
   DEFINES   += -DGLEW_STATIC -D_DEFAULT_VIDEOMODE_FLAGS_='SDL_HWSURFACE|SDL_DOUBLEBUF' -DTENGINE_HOME_PATH='".t-engine"' -DTE4CORE_VERSION=17 -DSELFEXE_LINUX
-  INCLUDES  += -I../src -I../src/luasocket -I../src/fov -I../src/expat -I../src/lxp -I../src/libtcod_import -I../src/physfs -I../src/zlib -I../src/bzip2 -I/opt/SDL-2.0/include/SDL2 -I/usr/include/GL -I../src/luajit2/src -I../src/luajit2/dynasm
+  INCLUDES  += -I../src -I../src/luasocket -I../src/fov -I../src/expat -I../src/lxp -I../src/libtcod_import -I../src/physfs -I../src/zlib -I../src/bzip2 -I/usr/include/SDL2 -I/usr/include/GL -I../src/luajit2/src -I../src/luajit2/dynasm
   CPPFLAGS  += -MMD -MP $(DEFINES) $(INCLUDES)
   CFLAGS    += $(CPPFLAGS) $(ARCH) -g -ggdb -O3
   CXXFLAGS  += $(CFLAGS) 
@@ -56,7 +56,7 @@
   TARGETDIR  = ../bin/Release
   TARGET     = $(TARGETDIR)/t-engine
   DEFINES   += -DGLEW_STATIC -DNDEBUG=1 -D_DEFAULT_VIDEOMODE_FLAGS_='SDL_HWSURFACE|SDL_DOUBLEBUF' -DTENGINE_HOME_PATH='".t-engine"' -DTE4CORE_VERSION=17 -DSELFEXE_LINUX
-  INCLUDES  += -I../src -I../src/luasocket -I../src/fov -I../src/expat -I../src/lxp -I../src/libtcod_import -I../src/physfs -I../src/zlib -I../src/bzip2 -I/opt/SDL-2.0/include/SDL2 -I/usr/include/GL -I../src/luajit2/src -I../src/luajit2/dynasm
+  INCLUDES  += -I../src -I../src/luasocket -I../src/fov -I../src/expat -I../src/lxp -I../src/libtcod_import -I../src/physfs -I../src/zlib -I../src/bzip2 -I/usr/include/SDL2 -I/usr/include/GL -I../src/luajit2/src -I../src/luajit2/dynasm
   CPPFLAGS  += -MMD -MP $(DEFINES) $(INCLUDES)
   CFLAGS    += $(CPPFLAGS) $(ARCH) -O2 -fomit-frame-pointer -O2
   CXXFLAGS  += $(CFLAGS) 
