diff -Nur r2e-2.6.6.orig/rss2email.py r2e-2.6.6/rss2email.py
--- r2e-2.6.6.orig/rss2email.py	2009-01-05 13:15:15.000000000 -0600
+++ r2e-2.6.6/rss2email.py	2009-03-27 20:35:32.561832322 -0500
@@ -200,7 +200,7 @@
 				print >>warn, ('Fatal error: sendmail exited with code %s' % status)
 				sys.exit(1)
 		except:
-			print '''Error attempting to send email via sendmail. Possibly you need to configure your config.py to use a SMTP server? Please refer to the rss2email documentation or website (http://rss2email.infogami.com) for complete documentation of config.py. The options below may suffice for configuring email:
+			print '''Error attempting to send email via sendmail. Possibly you need to configure your /etc/r2e/config.py to use a SMTP server? Please refer to the rss2email documentation or website (http://rss2email.infogami.com) for complete documentation of /etc/r2e/config.py. The options below may suffice for configuring email:
 # 1: Use SMTP_SERVER to send mail.
 # 0: Call /usr/sbin/sendmail to send mail.
 SMTP_SEND = 0
