From: David Watson <dwatson@debian.org>
Date: Mon, 18 Apr 2011 10:41:04 +0200
Subject: remove-newlines

---
 html2text.py |    1 +
 1 file changed, 1 insertion(+)

diff --git a/html2text.py b/html2text.py
index 0ed4cec..1cd5194 100755
--- a/html2text.py
+++ b/html2text.py
@@ -305,6 +305,7 @@ class _html2text(HTMLParser.HTMLParser):
             if has_key(attrs, 'src'):
                 attrs['href'] = attrs['src']
                 alt = attrs.get('alt', '')
+                alt = re.sub('\n', ' ', alt)
                 i = self.previousIndex(attrs)
                 if i is not None:
                     attrs = self.a[i]
