--- winff-1/winff/unit3.pas	2022-09-03 11:44:10.000000000 +0530
+++ winff-2/winff/unit3.pas	2022-09-24 16:04:02.473969501 +0530
@@ -2,7 +2,7 @@
 
 // WInFF 1.0 Copyright 2006-2013 Matthew Weatherford
 // WinFF 1.3.2 Copyright 2011 Alexey Osipov <lion-simba@pridelands.ru>
-// http://winff.org
+// https://github.com/WinFF/winff
 // Licensed under the GPL v3 or any later version
 //
 // This program is free software: you can redistribute it and/or modify
@@ -62,7 +62,6 @@
     procedure lblBellardClick(Sender: TObject);
     procedure lblGeversClick(Sender: TObject);
     procedure lblStoffbergClick(Sender: TObject);
-    procedure lblGmaqClick(Sender: TObject);
     procedure lblURLZeranoClick(Sender: TObject);
     procedure lblWeatherfordClick(Sender: TObject);
     procedure lblURLWinffOrgClick(Sender: TObject);
@@ -126,7 +125,7 @@
     lbThanks.items.add(rsLaunchpadContrib);
 
     frmAbout.lblWinff.Caption       := StringReplace(frmAbout.lblWinff.Caption, '%s', '1.5.5',[]);
-    frmAbout.lblCopyright.Caption   := StringReplace(frmAbout.lblCopyright.Caption, '%s', '2006-2016',[]);
+    frmAbout.lblCopyright.Caption   := StringReplace(frmAbout.lblCopyright.Caption, '%s', '2006-2022',[]);
 end;
 
 procedure TfrmAbout.lblBellardClick(Sender: TObject);
@@ -136,22 +135,17 @@
 
 procedure TfrmAbout.lblGeversClick(Sender: TObject);
 begin
-  frmMain.LaunchBrowser('https://launchpad.net/~paul-climbing/+archive/ppa/');
+  frmMain.LaunchBrowser('https://tracker.debian.org/pkg/winff');
 end;
 
 procedure TfrmAbout.lblStoffbergClick(Sender: TObject);
 begin
-frmMain.LaunchBrowser('http://istoff.blogspot.com');
-end;
-
-procedure TfrmAbout.lblGmaqClick(Sender: TObject);
-begin
-  frmMain.LaunchBrowser('http://www.bandshed.net/');
+frmMain.LaunchBrowser('https://istoff.blogspot.com');
 end;
 
 procedure TfrmAbout.lblURLZeranoClick(Sender: TObject);
 begin
-  frmMain.LaunchBrowser('http://ffmpeg.zeranoe.com');
+  frmMain.LaunchBrowser('https://www.gyan.dev/ffmpeg/');
 end;
 
 procedure TfrmAbout.btnOkClick(Sender: TObject);
@@ -161,22 +155,22 @@
 
 procedure TfrmAbout.lblWeatherfordClick(Sender: TObject);
 begin
-   frmMain.LaunchBrowser('http://www.biggmatt.com');
+   frmMain.LaunchBrowser('https://www.biggmatt.com/p/winff.html');
 end;
 
 procedure TfrmAbout.lblURLWinffOrgClick(Sender: TObject);
 begin
-  frmMain.LaunchBrowser('http://www.winff.org');
+  frmMain.LaunchBrowser('https://github.com/WinFF/winff');
 end;
 
 procedure TfrmAbout.lblURLffmpegClick(Sender: TObject);
 begin
-  frmMain.LaunchBrowser('http://ffmpeg.org/');
+  frmMain.LaunchBrowser('https://ffmpeg.org/');
 end;
 
 procedure TfrmAbout.lblwinbuildsClick(Sender: TObject);
 begin
-  frmMain.LaunchBrowser('http://ffmpeg.zeranoe.com/builds/');
+  frmMain.LaunchBrowser('https://www.gyan.dev/ffmpeg/builds/');
 end;
 
 procedure TfrmAbout.lblWindowsBuildsClick(Sender: TObject);
