--- unit6.pas	2022-09-03 11:44:10.000000000 +0530
+++ unit6.pas	2022-09-18 21:24:18.544977572 +0530
@@ -2,7 +2,7 @@
 
 // WInFF 1.0 Copyright 2006-2009 Matthew Weatherford
 // WinFF 1.3.2 Copyright 2011 Alexey Osipov <lion-simba@pridelands.ru>
-// http://winff.org
+// https://github.com/WinFF/winff
 // Licensed under the GPL v3 or any later version
 //
 // This program is free software: you can redistribute it and/or modify
