--- unit1.pas	2022-09-03 11:44:10.000000000 +0530
+++ unit1.pas	2022-09-18 21:46:23.128973768 +0530
@@ -2,7 +2,7 @@
 
 // WInFF 1.0 Copyright 2006-2012 Matthew Weatherford
 // WinFF 1.3.2 Copyright 2011 Alexey Osipov <lion-simba@pridelands.ru>
-// http://winff.org
+// https://github.com/WinFF/winff
 // Licensed under the GPL v3 or any later version
 //
 // This program is free software: you can redistribute it and/or modify
@@ -1390,10 +1390,12 @@
 begin
   {$ifdef linux}
   s:='';
+  if fileexists('/usr/bin/okular') then s:='/usr/bin/okular';
   if fileexists('/usr/bin/evince') then s:='/usr/bin/evince';
   if fileexists('/usr/bin/kpdf') then s:='/usr/bin/kpdf';
   if fileexists('/usr/bin/xpdf') then s:='/usr/bin/xpdf';
   if fileexists('/usr/bin/acroread') then s:='/usr/bin/acroread';
+  if fileexists('/usr/bin/xdg-open') then s:='/usr/bin/xdg-open';
   if s='' then
      begin
      Showmessage('More information can be found at ' + pdffile);
@@ -1984,16 +1986,16 @@
   language:=leftstr(lang,2);
   {$ifdef linux}
   s :='';
-  if fileexists('/usr/share/doc/winff/WinFF.' + language + '.pdf.gz') then s:='/usr/share/doc/winff/WinFF.' + language + '.pdf.gz';
-  if fileexists('/usr/share/doc/winff/WinFF.' + language + '.pdf') then s:='/usr/share/doc/winff/WinFF.' + language + '.pdf';
+  if fileexists('/usr/doc/winff-20220903_8e4491b/WinFF.' + language + '.pdf.gz') then s:='/usr/doc/winff-20220903_8e4491b/WinFF.' + language + '.pdf.gz';
+  if fileexists('/usr/doc/winff-20220903_8e4491b/WinFF.' + language + '.pdf') then s:='/usr/doc/winff-20220903_8e4491b/WinFF.' + language + '.pdf';
   if fileexists('/usr/share/winff/WinFF.' + language + '.pdf') then s:='/usr/share/winff/WinFF.' + language + '.pdf';
   if fileexists('/usr/share/winff/WinFF.' + language + '.pdf.gz') then s:='/usr/share/winff/WinFF.' + language + '.pdf.gz';
   if fileexists('/usr/share/doc/packages/winff/WinFF.' + language + '.pdf.gz') then s:='/usr/share/doc/packages/winff/WinFF.' + language + '.pdf.gz';
   if fileexists('/usr/share/doc/packages/winff/WinFF.' + language + '.pdf') then s:='/usr/share/doc/packages/winff/WinFF.' + language + '.pdf';
   if s='' then
      begin
-       s := '/usr/share/doc/winff/WinFF.en.pdf.gz';
-       if fileexists('/usr/share/doc/winff/WinFF.en.pdf') then s:='/usr/share/doc/winff/WinFF.en.pdf';
+       s := '/usr/doc/winff-20220903_8e4491b/WinFF.en.pdf.gz';
+       if fileexists('/usr/doc/winff-20220903_8e4491b/WinFF.en.pdf') then s:='/usr/doc/winff-20220903_8e4491b/WinFF.en.pdf';
        if fileexists('/usr/share/winff/WinFF.en.pdf') then s:='/usr/share/winff/WinFF.en.pdf';
        if fileexists('/usr/share/winff/WinFF.en.pdf.gz') then s:='/usr/share/winff/WinFF.en.pdf.gz';
        if fileexists('/usr/share/doc/packages/winff/WinFF.en.pdf.gz') then s:='/usr/share/doc/packages/winff/WinFF.en.pdf.gz';
@@ -2011,7 +2013,7 @@
 procedure TfrmMain.mitForumsClick(Sender: TObject);
 
 begin
-  launchbrowser('http://www.winff.org/forums/');
+  launchbrowser('http://github.com/WinFF/winff/issues');
 end;
 
 procedure TfrmMain.mitRestoreDefaultsClick(Sender: TObject);
@@ -2110,7 +2112,7 @@
 procedure TfrmMain.mitWinffClick(Sender: TObject);
 
 begin
-  launchbrowser('http://www.winff.org/');
+  launchbrowser('https://github.com/WinFF/winff');
 end;
 
 // menu: about
