--- unit3.lfm	2023-10-03 19:20:47.000000000 +0530
+++ unit3n.lfm	2024-01-11 20:09:24.951994073 +0530
@@ -341,7 +341,9 @@
         BorderSpacing.Top = 2
         BorderSpacing.Bottom = 2
         Caption = 'Trademark Fabrice Bellard'
+        Font.Color = clBlue
         ParentColor = False
+        ParentFont = False
         OnClick = lblBellardClick
       end
       object lblURLffmpeg: TLabel
@@ -354,18 +356,21 @@
         BorderSpacing.Left = 5
         BorderSpacing.Top = 2
         BorderSpacing.Bottom = 5
-        Caption = 'http://ffmpeg.org'
+        Caption = 'https://ffmpeg.org'
         Font.Color = clBlue
         ParentColor = False
         ParentFont = False
         OnClick = lblURLffmpegClick
       end
       object lblWindowsBuilds: TLabel
-        Left = 0
-        Height = 20
-        Top = 69
+        Left = 5
+        Height = 25
+        Top = 75
         Width = 376
         Align = alTop
+        BorderSpacing.Left = 5
+        BorderSpacing.Top = 2
+        BorderSpacing.Bottom = 5
         Caption = 'Windows Builds by'
         ParentColor = False
         ShowAccelChar = False
@@ -373,12 +378,15 @@
       end
       object lblwinbuilds: TLabel
         Cursor = crHandPoint
-        Left = 0
-        Height = 16
-        Top = 89
+        Left = 5
+        Height = 30
+        Top = 95
         Width = 376
         Align = alTop
-        Caption = 'Kyle Schwarz'
+        BorderSpacing.Left = 5
+        BorderSpacing.Top = 2
+        BorderSpacing.Bottom = 7
+        Caption = 'Gyan Doshi'
         Font.CharSet = ANSI_CHARSET
         Font.Color = clBlue
         Font.Height = -13
