--- winff-1/winff/unit3.lfm	2022-09-03 11:44:10.000000000 +0530
+++ winff-2/winff/unit3.lfm	2022-09-24 15:47:28.210972357 +0530
@@ -1,12 +1,12 @@
 object frmAbout: TfrmAbout
   Left = 323
-  Height = 300
+  Height = 325
   Top = 212
   Width = 384
   BorderIcons = [biSystemMenu]
   BorderStyle = bsDialog
   Caption = 'About WinFF'
-  ClientHeight = 300
+  ClientHeight = 350
   ClientWidth = 384
   Color = clForm
   Constraints.MinHeight = 300
@@ -144,8 +144,8 @@
         object lblGevers: TLabel
           Cursor = crHandPoint
           Left = 4
-          Height = 20
-          Top = 136
+          Height = 25
+          Top = 140
           Width = 372
           Align = alTop
           BorderSpacing.Left = 4
@@ -188,11 +188,11 @@
             AnchorSideTop.Side = asrBottom
             Cursor = crHandPoint
             Left = 57
-            Height = 20
-            Top = 29
+            Height = 25
+            Top = 33
             Width = 137
             BorderSpacing.Around = 5
-            Caption = 'http://www.winff.org'
+            Caption = 'https://github.com/WinFF/winff'
             Font.Color = clBlue
             ParentColor = False
             ParentFont = False
@@ -303,8 +303,8 @@
         object lblStoffberg: TLabel
           Cursor = crHandPoint
           Left = 4
-          Height = 20
-          Top = 116
+          Height = 25
+          Top = 120
           Width = 372
           Align = alTop
           BorderSpacing.Left = 4
@@ -322,7 +322,7 @@
       ClientWidth = 376
       object lblFFmeg: TLabel
         Left = 5
-        Height = 20
+        Height = 25
         Top = 0
         Width = 371
         Align = alTop
@@ -347,25 +347,28 @@
       object lblURLffmpeg: TLabel
         Cursor = crHandPoint
         Left = 5
-        Height = 20
-        Top = 44
+        Height = 25
+        Top = 45
         Width = 371
         Align = alTop
         BorderSpacing.Left = 5
         BorderSpacing.Top = 2
         BorderSpacing.Bottom = 5
-        Caption = 'http://ffmpeg.org'
+        Caption = 'https://ffmpeg.org'
         Font.Color = clBlue
         ParentColor = False
         ParentFont = False
         OnClick = lblURLffmpegClick
       end
       object lblWindowsBuilds: TLabel
-        Left = 0
-        Height = 20
-        Top = 69
+        Left = 5
+        Height = 25
+        Top = 75
         Width = 376
         Align = alTop
+        BorderSpacing.Left = 5
+        BorderSpacing.Top = 2
+        BorderSpacing.Bottom = 5
         Caption = 'Windows Builds by'
         ParentColor = False
         ShowAccelChar = False
@@ -373,18 +376,16 @@
       end
       object lblwinbuilds: TLabel
         Cursor = crHandPoint
-        Left = 0
-        Height = 16
-        Top = 89
+        Left = 5
+        Height = 30
+        Top = 95
         Width = 376
         Align = alTop
-        Caption = 'Kyle Schwarz'
-        Font.CharSet = ANSI_CHARSET
+        BorderSpacing.Left = 5
+        BorderSpacing.Top = 2
+        BorderSpacing.Bottom = 7
+        Caption = 'Gyan Doshi'
         Font.Color = clBlue
-        Font.Height = -13
-        Font.Name = 'Arial'
-        Font.Pitch = fpVariable
-        Font.Quality = fqDraft
         ParentColor = False
         ParentFont = False
         OnClick = lblwinbuildsClick
