--- unit3.pas	2024-04-17 01:41:03.000000000 +0530
+++ unit3n.pas	2024-04-25 17:30:56.674161644 +0530
@@ -2,9 +2,9 @@
 
 // About form
 
-// WInFF 1.0 Copyright 2006-2013 Matthew Weatherford
+// WInFF 1.0 Copyright 2006-2024 Matthew Weatherford
 // WinFF 1.3.2 Copyright 2011 Alexey Osipov <lion-simba@pridelands.ru>
-// http://winff.org
+// https://github.com/WinFF/winff
 // Licensed under the GPL v3 or any later version
 
 // This program is free software: you can redistribute it and/or modify
@@ -60,7 +60,6 @@
         procedure lblBellardClick(Sender: TObject);
         procedure lblGeversClick(Sender: TObject);
         procedure lblStoffbergClick(Sender: TObject);
-        procedure lblGmaqClick(Sender: TObject);
         procedure lblURLZeranoClick(Sender: TObject);
         procedure lblWeatherfordClick(Sender: TObject);
         procedure lblURLWinffOrgClick(Sender: TObject);
@@ -128,27 +127,22 @@
 
 procedure TfrmAbout.lblBellardClick(Sender: TObject);
 begin
-
+    frmMain.LaunchBrowser('https://bellard.org/');
 end;
 
 procedure TfrmAbout.lblGeversClick(Sender: TObject);
 begin
-    frmMain.LaunchBrowser('https://launchpad.net/~paul-climbing/+archive/ppa/');
+    frmMain.LaunchBrowser('https://tracker.debian.org/pkg/winff');
 end;
 
 procedure TfrmAbout.lblStoffbergClick(Sender: TObject);
 begin
-    frmMain.LaunchBrowser('http://istoff.blogspot.com');
-end;
-
-procedure TfrmAbout.lblGmaqClick(Sender: TObject);
-begin
-    frmMain.LaunchBrowser('http://www.bandshed.net/');
+    frmMain.LaunchBrowser('https://istoff.blogspot.com');
 end;
 
 procedure TfrmAbout.lblURLZeranoClick(Sender: TObject);
 begin
-    frmMain.LaunchBrowser('http://ffmpeg.zeranoe.com');
+    frmMain.LaunchBrowser('https://www.gyan.dev/ffmpeg/builds');
 end;
 
 procedure TfrmAbout.btnOkClick(Sender: TObject);
@@ -158,7 +152,7 @@
 
 procedure TfrmAbout.lblWeatherfordClick(Sender: TObject);
 begin
-    frmMain.LaunchBrowser('http://www.biggmatt.com');
+    frmMain.LaunchBrowser('https://github.com/WinFF/winff');
 end;
 
 procedure TfrmAbout.lblURLWinffOrgClick(Sender: TObject);
@@ -168,12 +162,12 @@
 
 procedure TfrmAbout.lblURLffmpegClick(Sender: TObject);
 begin
-    frmMain.LaunchBrowser('http://ffmpeg.org/');
+    frmMain.LaunchBrowser('https://ffmpeg.org/');
 end;
 
 procedure TfrmAbout.lblwinbuildsClick(Sender: TObject);
 begin
-    frmMain.LaunchBrowser('http://ffmpeg.zeranoe.com/builds/');
+    frmMain.LaunchBrowser('https://www.gyan.dev/ffmpeg/builds');
 end;
 
 procedure TfrmAbout.lblWindowsBuildsClick(Sender: TObject);
