--- unit1.pas	2023-10-03 19:20:47.000000000 +0530
+++ unit1n.pas	2024-01-10 22:26:54.221464261 +0530
@@ -1,8 +1,8 @@
 unit Unit1;
 
-// WInFF 1.0 Copyright 2006-2012 Matthew Weatherford
+// WInFF 1.0 Copyright 2006-2024 Matthew Weatherford
 // WinFF 1.3.2 Copyright 2011 Alexey Osipov <lion-simba@pridelands.ru>
-// http://winff.org
+// https://github.com/WinFF/winff
 // Licensed under the GPL v3 or any later version
 //
 // This program is free software: you can redistribute it and/or modify
@@ -1988,16 +1988,16 @@
   language:=leftstr(lang,2);
   {$ifdef unix}
   s :='';
-  if fileexists('/usr/share/doc/winff/WinFF.' + language + '.pdf.gz') then s:='/usr/share/doc/winff/WinFF.' + language + '.pdf.gz';
-  if fileexists('/usr/share/doc/winff/WinFF.' + language + '.pdf') then s:='/usr/share/doc/winff/WinFF.' + language + '.pdf';
+  if fileexists('/usr/share/doc/winff/WinFF.' + language + '.pdf.gz') then s:='/usr/doc/winff-20231003_31b79e3/WinFF.' + language + '.pdf.gz';
+  if fileexists('/usr/share/doc/winff/WinFF.' + language + '.pdf') then s:='/usr/doc/winff-20231003_31b79e3/WinFF.' + language + '.pdf';
   if fileexists('/usr/share/winff/WinFF.' + language + '.pdf') then s:='/usr/share/winff/WinFF.' + language + '.pdf';
   if fileexists('/usr/share/winff/WinFF.' + language + '.pdf.gz') then s:='/usr/share/winff/WinFF.' + language + '.pdf.gz';
   if fileexists('/usr/share/doc/packages/winff/WinFF.' + language + '.pdf.gz') then s:='/usr/share/doc/packages/winff/WinFF.' + language + '.pdf.gz';
   if fileexists('/usr/share/doc/packages/winff/WinFF.' + language + '.pdf') then s:='/usr/share/doc/packages/winff/WinFF.' + language + '.pdf';
   if s='' then
      begin
-       s := '/usr/share/doc/winff/WinFF.en.pdf.gz';
-       if fileexists('/usr/share/doc/winff/WinFF.en.pdf') then s:='/usr/share/doc/winff/WinFF.en.pdf';
+       s := '/usr/doc/winff-20231003_31b79e3/WinFF.en.pdf.gz';
+       if fileexists('/usr/doc/winff-20231003_31b79e3/WinFF.en.pdf') then s:='/usr/doc/winff-20231003_31b79e3/WinFF.en.pdf';
        if fileexists('/usr/share/winff/WinFF.en.pdf') then s:='/usr/share/winff/WinFF.en.pdf';
        if fileexists('/usr/share/winff/WinFF.en.pdf.gz') then s:='/usr/share/winff/WinFF.en.pdf.gz';
        if fileexists('/usr/share/doc/packages/winff/WinFF.en.pdf.gz') then s:='/usr/share/doc/packages/winff/WinFF.en.pdf.gz';
@@ -2015,7 +2015,7 @@
 procedure TfrmMain.mitForumsClick(Sender: TObject);
 
 begin
-  launchbrowser('https://groups.google.com/g/winff');
+  launchbrowser('https://github.com/WinFF/winff/issues');
 end;
 
 procedure TfrmMain.mitRestoreDefaultsClick(Sender: TObject);
