--- winff-1/winff/winff.lpi	2022-09-03 11:44:10.000000000 +0530
+++ winff-2/winff/winff.lpi	2022-09-23 15:25:23.822970637 +0530
@@ -20,7 +20,7 @@
       <MajorVersionNr Value="1"/>
       <MinorVersionNr Value="5"/>
       <RevisionNr Value="5"/>
-      <StringTable CompanyName="WinFF.org" FileDescription="WinFF Video Converter" InternalName="winff" LegalCopyright="Copyright 2006-2016 Matthew Weatherford" OriginalFilename="winff.exe" ProductVersion=""/>
+      <StringTable CompanyName="WinFF.org" FileDescription="WinFF Video Converter" InternalName="winff" LegalCopyright="Copyright 2006-2022 Matthew Weatherford" OriginalFilename="winff.exe" ProductVersion=""/>
     </VersionInfo>
     <BuildModes Count="1">
       <Item1 Name="default" Default="True"/>
@@ -117,6 +117,11 @@
         <UseAnsiStrings Value="False"/>
       </SyntaxOptions>
     </Parsing>
+    <CodeGeneration>
+        <Optimizations>
+            <OptimizationLevel Value="2"/>
+        </Optimizations>
+    </CodeGeneration>
     <Linking>
       <Debugging>
         <DebugInfoType Value="dsStabs"/>
@@ -124,7 +129,7 @@
       <LinkSmart Value="True"/>
       <Options>
         <PassLinkerOptions Value="True"/>
-        <LinkerOptions Value="--as-needed"/>
+        <LinkerOptions Value="--as-needed -zrelro -znow"/>
         <Win32>
           <GraphicApplication Value="True"/>
         </Win32>
