--- winff.lpr	2023-10-03 19:20:47.000000000 +0530
+++ winffn.lpr	2024-01-10 22:48:22.389222629 +0530
@@ -1,7 +1,7 @@
 program winff;
 
-// WInFF 1.0 Copyright 2006-2009 Matthew Weatherford
-// http://winff.org
+// WInFF 1.0 Copyright 2006-2024 Matthew Weatherford
+// https://github.com/WinFF/winff
 // Licensed under the GPL v3 or any later version
 //
 // This program is free software: you can redistribute it and/or modify
