--- winff.lpi	2024-04-17 01:41:03.000000000 +0530
+++ winffn.lpi	2024-04-25 17:33:23.998161221 +0530
@@ -20,7 +20,7 @@
       <MajorVersionNr Value="1"/>
       <MinorVersionNr Value="6"/>
       <RevisionNr Value="0"/>
-      <StringTable CompanyName="WinFF.org" FileDescription="WinFF Video Converter" InternalName="winff" LegalCopyright="Copyright 2006-2023 Matthew Weatherford" OriginalFilename="winff.exe" ProductVersion=""/>
+      <StringTable CompanyName="WinFF.org" FileDescription="WinFF Video Converter" InternalName="winff" LegalCopyright="Copyright 2006-2024 Matthew Weatherford" OriginalFilename="winff.exe" ProductVersion=""/>
     </VersionInfo>
     <BuildModes Count="1">
       <Item1 Name="default" Default="True"/>
@@ -129,7 +129,7 @@
       </Debugging>
       <Options>
         <PassLinkerOptions Value="True"/>
-        <LinkerOptions Value="--as-needed -zrelro -znow"/>
+        <LinkerOptions Value="--as-needed -zrelro -znow -zdefs -pie"/>
         <Win32>
           <GraphicApplication Value="True"/>
         </Win32>
