--- winff.lpi	2023-10-03 19:20:47.000000000 +0530
+++ winffn.lpi	2024-01-10 22:47:30.406244263 +0530
@@ -20,7 +20,7 @@
       <MajorVersionNr Value="1"/>
       <MinorVersionNr Value="6"/>
       <RevisionNr Value="0"/>
-      <StringTable CompanyName="WinFF.org" FileDescription="WinFF Video Converter" InternalName="winff" LegalCopyright="Copyright 2006-2023 Matthew Weatherford" OriginalFilename="winff.exe" ProductVersion=""/>
+      <StringTable CompanyName="WinFF.org" FileDescription="WinFF Video Converter" InternalName="winff" LegalCopyright="Copyright 2006-2024 Matthew Weatherford" OriginalFilename="winff.exe" ProductVersion=""/>
     </VersionInfo>
     <BuildModes Count="1">
       <Item1 Name="default" Default="True"/>
@@ -117,6 +117,11 @@
         <UseAnsiStrings Value="False"/>
       </SyntaxOptions>
     </Parsing>
+    <CodeGeneration>
+        <Optimizations>
+            <OptimizationLevel Value="2"/>
+        </Optimizations>
+    </CodeGeneration>
     <Linking>
       <Debugging>
         <DebugInfoType Value="dsStabs"/>
@@ -124,7 +129,7 @@
       <LinkSmart Value="True"/>
       <Options>
         <PassLinkerOptions Value="True"/>
-        <LinkerOptions Value="--as-needed"/>
+        <LinkerOptions Value="--as-needed -zrelro -znow"/>
         <Win32>
           <GraphicApplication Value="True"/>
         </Win32>
