diff -Nur moria-5.6.orig/source/config.h moria-5.6/source/config.h
--- moria-5.6.orig/source/config.h	2008-10-13 18:44:36.000000000 -0500
+++ moria-5.6/source/config.h	2012-08-10 17:20:22.987707262 -0500
@@ -221,41 +221,23 @@
 
 #else
 
-#if 0
-/* Debian standards for file location */
 /* This must be unix; change file names as appropriate.  */
 #define MORIA_SAV	".moria-save"
 #define MORIA_HOU	"/etc/moria-hours"
-#define MORIA_MOR	"/usr/lib/games/moria/news"
-#define MORIA_GPL	"/usr/lib/games/moria/COPYING"
+#define MORIA_MOR	"/usr/share/games/moria/news"
+#define MORIA_GPL	"/usr/share/games/moria/COPYING"
 #define MORIA_TOP	"/var/games/moria/scores"
-#define MORIA_HELP	"/usr/lib/games/moria/roglcmds.hlp"
-#define MORIA_ORIG_HELP	"/usr/lib/games/moria/origcmds.hlp"
-#define MORIA_WIZ_HELP	"/usr/lib/games/moria/rwizcmds.hlp"
-#define MORIA_OWIZ_HELP	"/usr/lib/games/moria/owizcmds.hlp"
-#define MORIA_WELCOME	"/usr/lib/games/moria/welcome.hlp"
-#define MORIA_VER	"/usr/lib/games/moria/version.hlp"
+#define MORIA_HELP	"/usr/share/games/moria/roglcmds.hlp"
+#define MORIA_ORIG_HELP	"/usr/share/games/moria/origcmds.hlp"
+#define MORIA_WIZ_HELP	"/usr/share/games/moria/rwizcmds.hlp"
+#define MORIA_OWIZ_HELP	"/usr/share/games/moria/owizcmds.hlp"
+#define MORIA_WELCOME	"/usr/share/games/moria/welcome.hlp"
+#define MORIA_VER	"/usr/share/games/moria/version.hlp"
 
-#else
-
-/* Generic UNIX */
-/* This must be unix; change file names as appropriate.  */
-#define MORIA_SAV	"moria-save"
-#define MORIA_HOU	"/home/dgrabiner/moria-5.6/files/hours"
-#define MORIA_MOR	"/home/dgrabiner/moria-5.6/files/news"
-#define MORIA_GPL	"/home/dgrabiner/moria-5.6/files/COPYING"
-#define MORIA_TOP	"/home/dgrabiner/moria-5.6/files/scores"
-#define MORIA_HELP	"/home/dgrabiner/moria-5.6/files/roglcmds.hlp"
-#define MORIA_ORIG_HELP	"/home/dgrabiner/moria-5.6/files/origcmds.hlp"
-#define MORIA_WIZ_HELP	"/home/dgrabiner/moria-5.6/files/rwizcmds.hlp"
-#define MORIA_OWIZ_HELP	"/home/dgrabiner/moria-5.6/files/owizcmds.hlp"
-#define MORIA_WELCOME	"/home/dgrabiner/moria-5.6/files/welcome.hlp"
-#define MORIA_VER	"/home/dgrabiner/moria-5.6/files/version.hlp"
 
 #endif
 #endif
 #endif
-#endif
 #endif
 #endif
 #endif
