diff -urN DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/Makefile DPO_RT5572_LinuxSTA_2.6.1.3_20121022/Makefile
--- DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/Makefile	2012-10-22 05:52:59.000000000 +0200
+++ DPO_RT5572_LinuxSTA_2.6.1.3_20121022/Makefile	2013-09-29 08:42:46.842805675 +0200
@@ -355,28 +355,28 @@
 endif
 
 ifeq ($(RT28xx_MODE),AP)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)ap.o /tftpboot
-ifeq ($(OSABL),YES)
-	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)ap.o /tftpboot
-	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)ap.o /tftpboot
-endif
+#	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)ap.o /tftpboot
+#ifeq ($(OSABL),YES)
+#	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)ap.o /tftpboot
+#	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)ap.o /tftpboot
+#endif
 ifeq ($(PLATFORM),INF_AMAZON_SE)
 	cp -f /tftpboot/rt2870ap.o /backup/ifx/build/root_filesystem/lib/modules/2.4.31-Amazon_SE-3.6.2.2-R0416_Ralink/kernel/drivers/net
 endif
-else	
-ifeq ($(RT28xx_MODE),APSTA)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)apsta.o /tftpboot
-ifeq ($(OSABL),YES)
-	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)apsta.o /tftpboot
-	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)apsta.o /tftpboot
-endif
-else
-	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)sta.o /tftpboot
-ifeq ($(OSABL),YES)
-	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)sta.o /tftpboot
-	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)sta.o /tftpboot
-endif
-endif	
+#else	
+#ifeq ($(RT28xx_MODE),APSTA)
+#	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)apsta.o /tftpboot
+#ifeq ($(OSABL),YES)
+#	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)apsta.o /tftpboot
+#	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)apsta.o /tftpboot
+#endif
+#else
+#	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)sta.o /tftpboot
+#ifeq ($(OSABL),YES)
+#	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)sta.o /tftpboot
+#	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)sta.o /tftpboot
+#endif
+#endif	
 endif	
 else
 
@@ -402,27 +402,27 @@
 endif
 
 ifeq ($(RT28xx_MODE),AP)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)ap.ko /tftpboot
-ifeq ($(OSABL),YES)
-	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)ap.ko /tftpboot
-	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)ap.ko /tftpboot
-endif
+#	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)ap.ko /tftpboot
+#ifeq ($(OSABL),YES)
+#	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)ap.ko /tftpboot
+#	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)ap.ko /tftpboot
+#endif
 	rm -f os/linux/rt$(MODULE)ap.ko.lzma
 	/root/bin/lzma e os/linux/rt$(MODULE)ap.ko os/linux/rt$(MODULE)ap.ko.lzma
-else	
-ifeq ($(RT28xx_MODE),APSTA)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)apsta.ko /tftpboot
-ifeq ($(OSABL),YES)
-	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)apsta.ko /tftpboot
-	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)apsta.ko /tftpboot
-endif
-else
-	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)sta.ko /tftpboot
-ifeq ($(OSABL),YES)
-	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)sta.ko /tftpboot
-	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)sta.ko /tftpboot
-endif
-endif
+#else	
+#ifeq ($(RT28xx_MODE),APSTA)
+#	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)apsta.ko /tftpboot
+#ifeq ($(OSABL),YES)
+#	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)apsta.ko /tftpboot
+#	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)apsta.ko /tftpboot
+#endif
+#else
+#	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)sta.ko /tftpboot
+#ifeq ($(OSABL),YES)
+#	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)sta.ko /tftpboot
+#	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)sta.ko /tftpboot
+#endif
+#endif
 endif
 endif
 
diff -urN DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/os/linux/Makefile.4 DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/Makefile.4
--- DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/os/linux/Makefile.4	2012-10-22 05:52:59.000000000 +0200
+++ DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/Makefile.4	2013-09-29 08:37:25.444820107 +0200
@@ -549,13 +549,13 @@
 endif
 
 install:
-	rm -rf $(DAT_PATH)
-	$(shell [ ! -f /etc/Wireless ] && mkdir /etc/Wireless)
-	mkdir $(DAT_PATH)
-	cp $(RT28xx_DIR)/$(DAT_FILE_NAME) $(DAT_PATH)/.
-	install -d $(LINUX_SRC_MODULE)
-	install -m 644 -c $(addsuffix .o,$(MOD_NAME)) $(LINUX_SRC_MODULE)
-	/sbin/depmod -a ${shell uname -r}
+	rm -rf $(DESTDIR)$(DAT_PATH)
+	$(shell [ ! -f $(DESTDIR)/etc/Wireless ] && mkdir -p $(DESTDIR)/etc/Wireless)
+	mkdir -p $(DESTDIR)$(DAT_PATH)
+	cp $(RT28xx_DIR)/$(DAT_FILE_NAME) $(DESTDIR)$(DAT_PATH)/.
+	install -d $(DESTDIR)$(LINUX_SRC_MODULE)
+	install -m 644 -c $(addsuffix .o,$(MOD_NAME)) $(DESTDIR)$(LINUX_SRC_MODULE)
+#	/sbin/depmod -a ${shell uname -r}
 
 uninstall:
 #	rm -rf $(DAT_PATH)
diff -urN DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/os/linux/Makefile.4.netif DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/Makefile.4.netif
--- DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/os/linux/Makefile.4.netif	2012-10-22 05:52:59.000000000 +0200
+++ DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/Makefile.4.netif	2013-09-29 08:28:23.784844428 +0200
@@ -148,9 +148,9 @@
 endif
 
 install:
-	install -d $(LINUX_SRC_MODULE)
-	install -m 644 -c $(addsuffix .o,$(MOD_NAME)) $(LINUX_SRC_MODULE)
-	/sbin/depmod -a ${shell uname -r}
+	install -d $(DESTDIR)$(LINUX_SRC_MODULE)
+	install -m 644 -c $(addsuffix .o,$(MOD_NAME)) $(DESTDIR)$(LINUX_SRC_MODULE)
+#	/sbin/depmod -a ${shell uname -r}
 
 uninstall:
 	rm -rf $(addprefix $(LINUX_SRC_MODULE),$(addsuffix .o,$(MOD_NAME)))
diff -urN DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/os/linux/Makefile.4.util DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/Makefile.4.util
--- DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/os/linux/Makefile.4.util	2012-10-22 05:52:59.000000000 +0200
+++ DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/Makefile.4.util	2013-09-29 08:27:06.948847878 +0200
@@ -63,9 +63,9 @@
 endif
 
 install:
-	install -d $(LINUX_SRC_MODULE)
-	install -m 644 -c $(addsuffix .o,$(MOD_NAME)) $(LINUX_SRC_MODULE)
-	/sbin/depmod -a ${shell uname -r}
+	install -d $(DESTDIR)$(LINUX_SRC_MODULE)
+	install -m 644 -c $(addsuffix .o,$(MOD_NAME)) $(DESTDIR)$(LINUX_SRC_MODULE)
+#	/sbin/depmod -a ${shell uname -r}
 
 uninstall:
 	rm -rf $(addprefix $(LINUX_SRC_MODULE),$(addsuffix .o,$(MOD_NAME)))
diff -urN DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/os/linux/Makefile.6 DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/Makefile.6
--- DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/os/linux/Makefile.6	2012-10-22 05:52:59.000000000 +0200
+++ DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/Makefile.6	2013-09-29 08:37:37.333819573 +0200
@@ -529,13 +529,13 @@
 endif
 
 install:
-	rm -rf $(DAT_PATH)
-	$(shell [ ! -f /etc/Wireless ] && mkdir /etc/Wireless)
-	mkdir $(DAT_PATH)
-	cp $(RT28xx_DIR)/$(DAT_FILE_NAME) $(DAT_PATH)/.
-	install -d $(LINUX_SRC_MODULE)
-	install -m 644 -c $(addsuffix .ko,$(MOD_NAME)) $(LINUX_SRC_MODULE)
-	/sbin/depmod -a ${shell uname -r}
+	rm -rf $(DESTDIR)$(DAT_PATH)
+	$(shell [ ! -f $(DESTDIR)/etc/Wireless ] && mkdir -p $(DESTDIR)/etc/Wireless)
+	mkdir -p $(DESTDIR)$(DAT_PATH)
+	cp $(RT28xx_DIR)/$(DAT_FILE_NAME) $(DESTDIR)$(DAT_PATH)/.
+	install -d $(DESTDIR)$(LINUX_SRC_MODULE)
+	install -m 644 -c $(addsuffix .ko,$(MOD_NAME)) $(DESTDIR)$(LINUX_SRC_MODULE)
+#	/sbin/depmod -a ${shell uname -r}
 
 uninstall:
 #	rm -rf $(DAT_PATH)
diff -urN DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/os/linux/Makefile.6.netif DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/Makefile.6.netif
--- DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/os/linux/Makefile.6.netif	2012-10-22 05:52:59.000000000 +0200
+++ DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/Makefile.6.netif	2013-09-29 08:27:56.146845669 +0200
@@ -126,9 +126,9 @@
 endif
 
 install:
-	install -d $(LINUX_SRC_MODULE)
-	install -m 644 -c $(addsuffix .ko,$(MOD_NAME)) $(LINUX_SRC_MODULE)
-	/sbin/depmod -a ${shell uname -r}
+	install -d $(DESTDIR)$(LINUX_SRC_MODULE)
+	install -m 644 -c $(addsuffix .ko,$(MOD_NAME)) $(DESTDIR)$(LINUX_SRC_MODULE)
+#	/sbin/depmod -a ${shell uname -r}
 
 uninstall:
 	rm -rf $(addprefix $(LINUX_SRC_MODULE),$(addsuffix .ko,$(MOD_NAME)))
diff -urN DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/os/linux/Makefile.6.util DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/Makefile.6.util
--- DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/os/linux/Makefile.6.util	2012-10-22 05:52:59.000000000 +0200
+++ DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/Makefile.6.util	2013-09-29 08:27:31.339846782 +0200
@@ -48,9 +48,9 @@
 endif
 
 install:
-	install -d $(LINUX_SRC_MODULE)
-	install -m 644 -c $(addsuffix .ko,$(MOD_NAME)) $(LINUX_SRC_MODULE)
-	/sbin/depmod -a ${shell uname -r}
+	install -d $(DESTDIR)$(LINUX_SRC_MODULE)
+	install -m 644 -c $(addsuffix .ko,$(MOD_NAME)) $(DESTDIR)$(LINUX_SRC_MODULE)
+#	/sbin/depmod -a ${shell uname -r}
 
 uninstall:
 	rm -rf $(addprefix $(LINUX_SRC_MODULE),$(addsuffix .ko,$(MOD_NAME)))
diff -urN DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/os/linux/config.mk DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/config.mk
--- DPO_RT5572_LinuxSTA_2.6.1.3_20121022.orig/os/linux/config.mk	2012-10-22 05:52:59.000000000 +0200
+++ DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/config.mk	2013-09-29 11:30:32.537931142 +0200
@@ -23,7 +23,7 @@
 
 # Support Wpa_Supplicant
 # i.e. wpa_supplicant -Dralink
-HAS_WPA_SUPPLICANT=n
+HAS_WPA_SUPPLICANT=y
 
 
 # Support Native WpaSupplicant for Network Maganger
@@ -32,7 +32,7 @@
 # what if user want to use wpa_supplicant to serve P2P function/feature, 
 # in case, it must use Ralink Propriectary wpa_supplicant to do.
 # and this compile flag will report P2P Related Event to Ralink wpa_supplicant.
-HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=n
+HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=y
 
 #Support Net interface block while Tx-Sw queue full
 HAS_BLOCK_NET_IF=n
