--- src/editor/ui_menus/editor_main_menu_random_map.cc.old	2010-05-18 17:17:15.000000000 +0200
+++ src/editor/ui_menus/editor_main_menu_random_map.cc	2010-05-18 17:18:29.000000000 +0200
@@ -251,7 +251,7 @@
 		 posx, posy, width, height,
 		 g_gr->get_picture(PicMod_UI, "pics/but1.png"),
 		 &Main_Menu_New_Random_Map::button_clicked, *this, 8,
-		 Widelands::World::World(m_worlds[m_currentworld].c_str()).get_name());
+		 Widelands::World(m_worlds[m_currentworld].c_str()).get_name());
 
 	posy += height + spacing + spacing + spacing;
 
@@ -322,7 +322,7 @@
 		if (m_currentworld == m_worlds.size())
 			m_currentworld = 0;
 		m_world->set_title
-			(Widelands::World::World(m_worlds[m_currentworld].c_str()).get_name());
+			(Widelands::World(m_worlds[m_currentworld].c_str()).get_name());
 		break;
 	case 9:
 		break;
@@ -476,7 +476,7 @@
 			(strcmp(mapInfo.worldName.c_str(), m_worlds[m_currentworld].c_str()))
 			++m_currentworld;
 		m_world->set_title
-			(Widelands::World::World(m_worlds[m_currentworld].c_str()).get_name());
+			(Widelands::World(m_worlds[m_currentworld].c_str()).get_name());
 
 		button_clicked(-1);  // Update other values in UI as well
 
--- src/editor/ui_menus/editor_main_menu_new_map.cc.old	2010-05-18 17:38:29.000000000 +0200
+++ src/editor/ui_menus/editor_main_menu_new_map.cc	2010-05-18 17:38:55.000000000 +0200
@@ -113,7 +113,7 @@
 		 posx, posy, width, height,
 		 g_gr->get_picture(PicMod_UI, "pics/but1.png"),
 		 &Main_Menu_New_Map::button_clicked, *this, 4,
-		 Widelands::World::World(m_worlds[m_currentworld].c_str()).get_name());
+		 Widelands::World(m_worlds[m_currentworld].c_str()).get_name());
 
 	posy += height + spacing + spacing + spacing;
 
@@ -142,7 +142,7 @@
 		if (m_currentworld == m_worlds.size())
 			m_currentworld = 0;
 		m_world->set_title
-			(Widelands::World::World(m_worlds[m_currentworld].c_str()).get_name
+			(Widelands::World(m_worlds[m_currentworld].c_str()).get_name
 			 	());
 		break;
 	}
