diff -Naur update.orig/source/src/Makefile update/source/src/Makefile
--- update.orig/source/src/Makefile	2009-02-14 16:02:36.000000000 +0000
+++ update/source/src/Makefile	2009-06-29 16:57:39.000000000 +0000
@@ -1,12 +1,12 @@
-CXXOPTFLAGS=-O3 -fomit-frame-pointer
-INCLUDES1=-I/usr/X11R6/include -I../enet/include -I../src
+CXXOPTFLAGS?=-O3 -fomit-frame-pointer
+INCLUDES1=-I../enet/include -I/usr/X11R6/include -I../src
 INCLUDES2=`sdl-config --cflags`
 CXXCUSTOMFLAGS=-fsigned-char -Wall -Wextra -Wno-missing-field-initializers -Wno-unused -Wpointer-arith -Wwrite-strings -Wconversion -rdynamic
 CXXFLAGS=$(CXXOPTFLAGS) $(INCLUDES1) $(INCLUDES2) $(CUSTOMFLAGS)
 CXXFLAGSSERVER=$(CXXOPTFLAGS) $(INCLUDES1) $(CUSTOMFLAGS)
 
 PLATFORM_PREFIX=native
-CLIENT_LIBS=-L/usr/X11R6/lib -L../enet -lenet `sdl-config --libs` -lSDL_image -lz -lGL -lopenal -lvorbisfile
+CLIENT_LIBS=-L../enet -lenet -L/usr/X11R6/lib `sdl-config --libs` -lSDL_image -lz -lGL -lopenal -lvorbisfile
 CLIENT_OBJS1= \
 	client.o \
 	clientgame.o \
@@ -69,7 +69,7 @@
 all: libenet client server
 
 ../enet/Makefile:
-	cd ../enet; ./configure
+	cd ../enet; ./configure --enable-crc32
 
 libenet: ../enet/Makefile
 	-$(MAKE) -C ../enet all
