diff -Naur 1.1.0.0.orig/source/src/Makefile 1.1.0.0/source/src/Makefile
--- 1.1.0.0.orig/source/src/Makefile	2010-08-03 17:33:18.000000000 +0000
+++ 1.1.0.0/source/src/Makefile	2010-08-05 23:42:36.631639906 +0000
@@ -1,5 +1,5 @@
 //CXX=g++-gcc-4.2.3
-CXXFLAGS= -O3 -fomit-frame-pointer
+CXXFLAGS= $(EXTRA_CXXFLAGS) -fomit-frame-pointer
 override CXXFLAGS+= -Wall -fsigned-char
 
 PLATFORM= $(shell uname -s)
