diff -Naur 1.1.0.4.orig/source/src/Makefile 1.1.0.4/source/src/Makefile
--- 1.1.0.4.orig/source/src/Makefile	2010-11-14 18:46:24.000000000 +0000
+++ 1.1.0.4/source/src/Makefile	2012-07-10 03:43:41.110712746 +0000
@@ -26,7 +26,7 @@
 override CXXFLAGS+= -rdynamic
 CLIENT_INCLUDES= $(INCLUDES) -I/usr/include `sdl-config --cflags` -idirafter ../include
 # -lSDL_ttf 
-CLIENT_LIBS= -L../enet/.libs -lenet -L$(USRLIB) `sdl-config --libs` -lSDL_image -lz -lGL -lopenal -lvorbisfile
+CLIENT_LIBS= -L../enet/.libs -lenet -L$(USRLIB) -lX11 `sdl-config --libs` -lSDL_image -lz -lGL -lopenal -lvorbisfile
 endif
 
 CLIENT_OBJS= \
