diff -r 9c4fdd8369ca src/pkg/runtime/crash_cgo_test.go
--- a/src/pkg/runtime/crash_cgo_test.go	Mon May 05 10:43:37 2014 -0700
+++ b/src/pkg/runtime/crash_cgo_test.go	Tue May 06 17:21:45 2014 -0400
@@ -8,6 +8,7 @@
 
 import (
 	"runtime"
+	"strings"
 	"testing"
 )
 
@@ -21,7 +22,7 @@
 	}
 	got := executeTest(t, cgoSignalDeadlockSource, nil)
 	want := "OK\n"
-	if got != want {
+	if !strings.Contains(got, want) {
 		t.Fatalf("expected %q, but got %q", want, got)
 	}
 }
@@ -29,7 +30,7 @@
 func TestCgoTraceback(t *testing.T) {
 	got := executeTest(t, cgoTracebackSource, nil)
 	want := "OK\n"
-	if got != want {
+	if !strings.Contains(got, want) {
 		t.Fatalf("expected %q, but got %q", want, got)
 	}
 }
