diff -Naur GTorrentViewer-0.2b/src/mainwindow.c GTorrentViewer-0.2b.patched/src/mainwindow.c
--- GTorrentViewer-0.2b/src/mainwindow.c	2023-11-07 21:01:19.243506719 -0500
+++ GTorrentViewer-0.2b.patched/src/mainwindow.c	2023-11-07 21:02:04.947502418 -0500
@@ -423,7 +423,7 @@
 
   node = benc_node_find_key(torrent, "announce");
   gtk_list_store_append(liststore, &iter);
-  gtk_list_store_set(liststore, &iter, 0, benc_node_data(node), -1);
+  gtk_list_store_set(liststore, &iter, 0, node!=NULL?benc_node_data(node):"", -1);
 
   node = benc_node_find_key(torrent, "announce-list");
   if(node != NULL) /* multi-tracker support */
