--- GTorrentViewer-0.2b/data/gtorrentviewer.desktop.in.orig	2004-10-21 05:40:16.000000000 +0100
+++ GTorrentViewer-0.2b/data/gtorrentviewer.desktop.in	2007-08-12 12:14:52.000000000 +0100
@@ -2,7 +2,7 @@
 _Name=Torrent Metainfo Viewer
 _GenericName=Torrent Viewer
 _Comment=Viewer and Editor for .torrent files
-Exec=gtorrentviewer %U
+Exec=gtorrentviewer %f
 Icon=gtorrentviewer.png
 StartupNotify=false
 Terminal=false
@@ -10,4 +10,4 @@
 Encoding=UTF-8
 Categories=Application;Network;
 MimeType=application/x-bittorrent;
-Version=0.2
+Version=1.0
