Index: gq-1.2.3/src/gq-hash.h
===================================================================
--- gq-1.2.3.orig/src/gq-hash.h
+++ gq-1.2.3/src/gq-hash.h
@@ -24,7 +24,7 @@
 #ifndef GQ_ENCRYPTION_H
 #define GQ_ENCRYPTION_H
 
-#include <glib/garray.h>
+#include <glib.h>
 #include "util.h"
 
 G_BEGIN_DECLS
Index: gq-1.2.3/src/gq-server-list.h
===================================================================
--- gq-1.2.3.orig/src/gq-server-list.h
+++ gq-1.2.3/src/gq-server-list.h
@@ -26,7 +26,7 @@
 #ifndef GQ_SERVER_LIST_H
 #define GQ_SERVER_LIST_H
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include "gq-server.h" // for GqServer
 
 G_BEGIN_DECLS
Index: gq-1.2.3/src/state.c
===================================================================
--- gq-1.2.3.orig/src/state.c
+++ gq-1.2.3/src/state.c
@@ -36,7 +36,7 @@
 #include <errno.h>
 
 #include <glib/gi18n.h>
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <gtk/gtk.h>
 
 #include "xmlparse.h"
