diff -r e0fc0db3d486 configure
--- a/configure	Wed Mar 07 22:18:10 2018 +0900
+++ b/configure	Wed Mar 07 23:58:10 2018 +0900
@@ -19871,7 +19871,7 @@
 	fi
 	mkdir -p ${top_srcdir-$srcdir}/fb
 	for file in ChangeLog Makefile.in common configure etc gtk inputmethod java \
-		main man vtemu script scrollbar uitoolkit doc ; do
+		main man vtemu script scrollbar uitoolkit doc libvterm; do
 		cp -R ${top_srcdir-$srcdir}/$file ${top_srcdir-$srcdir}/fb/
 	done
 fi
diff -r e0fc0db3d486 configure.in
--- a/configure.in	Wed Mar 07 22:18:10 2018 +0900
+++ b/configure.in	Wed Mar 07 23:58:10 2018 +0900
@@ -82,7 +82,7 @@
 	fi
 	mkdir -p ${top_srcdir-$srcdir}/fb
 	for file in ChangeLog Makefile.in common configure etc gtk inputmethod java \
-		main man vtemu script scrollbar uitoolkit doc ; do
+		main man vtemu script scrollbar uitoolkit doc libvterm; do
 		cp -R ${top_srcdir-$srcdir}/$file ${top_srcdir-$srcdir}/fb/
 	done
 fi
