--- multipathd/Makefile.orig	2021-09-08 11:49:01.000000000 +0200
+++ multipathd/Makefile	2021-12-05 23:47:00.654767573 +0100
@@ -15,7 +15,7 @@
 LDFLAGS += $(BIN_LDFLAGS)
 LIBDEPS += -L$(multipathdir) -lmultipath -L$(mpathpersistdir) -lmpathpersist \
 	   -L$(mpathcmddir) -lmpathcmd -ludev -ldl -lurcu -lpthread \
-	   -ldevmapper -lreadline
+	   -ldevmapper -lreadline -lncurses
 CFLAGS += $(shell $(PKGCONFIG) --modversion liburcu 2>/dev/null | \
 	awk -F. '{ printf("-DURCU_VERSION=0x%06x", 256 * ( 256 * $$1 + $$2) + $$3); }')
 
