--- multipathd/Makefile.orig	2017-11-10 17:52:36.070562376 +0200
+++ multipathd/Makefile	2017-11-10 18:10:47.854895039 +0200
@@ -11,7 +11,7 @@ CFLAGS += $(BIN_CFLAGS) -I$(multipathdir
 LDFLAGS += $(BIN_LDFLAGS)
 LIBDEPS += -L$(multipathdir) -lmultipath -L$(mpathpersistdir) -lmpathpersist \
 	   -L$(mpathcmddir) -lmpathcmd -ludev -ldl -lurcu -lpthread \
-	   -ldevmapper -lreadline
+	   -ldevmapper -lreadline -lncurses
 
 ifdef SYSTEMD
 	CFLAGS += -DUSE_SYSTEMD=$(SYSTEMD)
