Index: qpdfview-0.4.18/sources/djvumodel.cpp
===================================================================
--- qpdfview-0.4.18.orig/sources/djvumodel.cpp
+++ qpdfview-0.4.18/sources/djvumodel.cpp
@@ -29,6 +29,7 @@ along with qpdfview.  If not, see <http:
 #include <cstdio>
 
 #include <QFile>
+#include <QPainterPath>
 #include <qmath.h>
 
 #if defined(Q_OS_WIN) && defined(DJVU_STATIC)
Index: qpdfview-0.4.18/sources/model.h
===================================================================
--- qpdfview-0.4.18.orig/sources/model.h
+++ qpdfview-0.4.18/sources/model.h
@@ -27,6 +27,7 @@ along with qpdfview.  If not, see <http:
 #include <QtPlugin>
 #include <QWidget>
 #include <QVector>
+#include <QPainterPath>
 
 class QAbstractItemModel;
 class QColor;
