--- orig/perlprimer.pl	2016-11-22 21:06:25.135978000 +0200
+++ new/perlprimer.pl	2016-11-22 21:08:09.001971413 +0200
@@ -5226,7 +5226,7 @@ sub generate_report {
 	}
 	
 	my ($hlist_sel) = $$hlist_ref->selectionGet;
-	unless (defined(@$slist) && defined($hlist_sel)) {
+	unless (@$slist && defined($hlist_sel)) {
 		dialogue("The Generate Report function saves the statistics and alignment of a particular primer pair - please select a primer pair first");
 		return;
 	}
