diff -Nur plplot-5.15.0.orig/bindings/qt_gui/plqt.cpp plplot-5.15.0/bindings/qt_gui/plqt.cpp
--- plplot-5.15.0.orig/bindings/qt_gui/plqt.cpp	2019-06-01 18:24:33.000000000 -0500
+++ plplot-5.15.0/bindings/qt_gui/plqt.cpp	2022-02-26 18:05:49.916622395 -0600
@@ -23,6 +23,7 @@
 //
 
 #include "qt.h"
+#include <QPainterPath>
 
 // Global variables for Qt driver.
 PLDLLIMPEXP_QT_DATA( int ) vectorize = 0;
