diff -Nur rapidjson-1.1.0.orig/include/rapidjson/internal/regex.h rapidjson-1.1.0/include/rapidjson/internal/regex.h
--- rapidjson-1.1.0.orig/include/rapidjson/internal/regex.h	2016-08-25 02:59:05.000000000 -0500
+++ rapidjson-1.1.0/include/rapidjson/internal/regex.h	2017-12-27 01:10:17.765316098 -0600
@@ -29,6 +29,9 @@
 #ifdef __GNUC__
 RAPIDJSON_DIAG_PUSH
 RAPIDJSON_DIAG_OFF(effc++)
+#if __GNUC__ >= 7
+RAPIDJSON_DIAG_OFF(implicit-fallthrough)
+#endif
 #endif
 
 #ifdef _MSC_VER
