diff --git a/etc/systemd/system/windscribe-helper.service b/etc/systemd/system/windscribe-helper.service
index 602d584..d9baeab 100644
--- a/etc/systemd/system/windscribe-helper.service
+++ b/etc/systemd/system/windscribe-helper.service
@@ -5,7 +5,7 @@ Wants=network-pre.target
 
 [Service]
 Type=simple
-ExecStart=/usr/local/windscribe/helper
+ExecStart=/opt/windscribe/helper
 
 [Install]
 WantedBy=multi-user.target
\ No newline at end of file
diff --git a/usr/polkit-1/actions/com.windscribe.authhelper.policy b/usr/polkit-1/actions/com.windscribe.authhelper.policy
index 4d81580..301fb8d 100644
--- a/usr/polkit-1/actions/com.windscribe.authhelper.policy
+++ b/usr/polkit-1/actions/com.windscribe.authhelper.policy
@@ -11,7 +11,7 @@
              <allow_inactive>auth_admin_keep</allow_inactive>
              <allow_active>auth_admin_keep</allow_active>
          </defaults>        
-         <annotate key="org.freedesktop.policykit.exec.path">/usr/local/windscribe/windscribe-authhelper</annotate>         
+         <annotate key="org.freedesktop.policykit.exec.path">/opt/windscribe/windscribe-authhelper</annotate>         
          <annotate key="org.freedesktop.policykit.exec.allow_gui">true</annotate>
      </action>                                                                       
 </policyconfig>
diff --git a/usr/share/applications/windscribe.desktop b/usr/share/applications/windscribe.desktop
index 8118766..731074b 100644
--- a/usr/share/applications/windscribe.desktop
+++ b/usr/share/applications/windscribe.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Terminal=false
-Exec=/usr/local/windscribe/Windscribe %F
+Exec=/opt/windscribe/Windscribe %F
 Name=Windscribe
 Icon=windscribe
 Categories=Utility;Security
\ No newline at end of file
