diff -ur src.old/activity.c src/activity.c
--- src.old/activity.c	2024-10-13 11:42:59.998881132 +0900
+++ src/activity.c	2024-10-13 11:44:49.745270696 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/args.c src/args.c
--- src.old/args.c	2024-10-13 11:42:59.999881145 +0900
+++ src/args.c	2024-10-13 11:48:38.331145970 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/blog.c src/blog.c
--- src.old/blog.c	2024-10-13 11:43:00.000881158 +0900
+++ src/blog.c	2024-10-13 11:43:54.910577158 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/category.c src/category.c
--- src.old/category.c	2024-10-13 11:42:59.999881145 +0900
+++ src/category.c	2024-10-13 11:43:24.710194543 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/collection.c src/collection.c
--- src.old/collection.c	2024-10-13 11:42:59.999881145 +0900
+++ src/collection.c	2024-10-13 11:43:15.118072925 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/comments.c src/comments.c
--- src.old/comments.c	2024-10-13 11:42:59.999881145 +0900
+++ src/comments.c	2024-10-13 11:44:31.003033817 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/contacts.c src/contacts.c
--- src.old/contacts.c	2024-10-13 11:42:59.998881132 +0900
+++ src/contacts.c	2024-10-13 11:45:19.570647295 +0900
@@ -24,6 +24,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/context.c src/context.c
--- src.old/context.c	2024-10-13 11:43:00.000881158 +0900
+++ src/context.c	2024-10-13 11:49:15.717613873 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/exif.c src/exif.c
--- src.old/exif.c	2024-10-13 11:42:59.999881145 +0900
+++ src/exif.c	2024-10-13 11:49:04.788477160 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/gallery.c src/gallery.c
--- src.old/gallery.c	2024-10-13 11:42:59.998881132 +0900
+++ src/gallery.c	2024-10-13 11:48:54.363346699 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/group.c src/group.c
--- src.old/group.c	2024-10-13 11:42:59.998881132 +0900
+++ src/group.c	2024-10-13 11:49:30.564799510 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/legacy-auth.c src/legacy-auth.c
--- src.old/legacy-auth.c	2024-10-13 11:42:59.999881145 +0900
+++ src/legacy-auth.c	2024-10-13 11:56:39.293976791 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/members.c src/members.c
--- src.old/members.c	2024-10-13 11:42:59.999881145 +0900
+++ src/members.c	2024-10-13 11:54:35.846594448 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/method.c src/method.c
--- src.old/method.c	2024-10-13 11:42:59.999881145 +0900
+++ src/method.c	2024-10-13 11:54:39.492639524 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/note.c src/note.c
--- src.old/note.c	2024-10-13 11:42:59.999881145 +0900
+++ src/note.c	2024-10-13 11:54:32.676555252 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/oauth.c src/oauth.c
--- src.old/oauth.c	2024-10-13 11:43:00.000881158 +0900
+++ src/oauth.c	2024-10-13 11:56:34.588936076 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/perms.c src/perms.c
--- src.old/perms.c	2024-10-13 11:43:00.000881158 +0900
+++ src/perms.c	2024-10-13 11:55:31.035276137 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/photo.c src/photo.c
--- src.old/photo.c	2024-10-13 11:42:59.998881132 +0900
+++ src/photo.c	2024-10-13 11:54:25.302464059 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 #include <ctype.h>
 
 #ifdef HAVE_CONFIG_H
diff -ur src.old/photoset.c src/photoset.c
--- src.old/photoset.c	2024-10-13 11:42:59.999881145 +0900
+++ src/photoset.c	2024-10-13 11:54:28.555504292 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/sha1.c src/sha1.c
--- src.old/sha1.c	2024-10-13 11:42:59.999881145 +0900
+++ src/sha1.c	2024-10-13 11:56:51.346081132 +0900
@@ -92,6 +92,7 @@
 #include <string.h>
 #include <stdarg.h>
 #include <stdint.h>
+#include <stdlib.h>
 
 #include <flickcurl.h>
 #include <flickcurl_internal.h>
diff -ur src.old/size.c src/size.c
--- src.old/size.c	2024-10-13 11:42:59.998881132 +0900
+++ src/size.c	2024-10-13 11:54:16.726357972 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/stat.c src/stat.c
--- src.old/stat.c	2024-10-13 11:42:59.999881145 +0900
+++ src/stat.c	2024-10-13 11:54:09.052263016 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/tags.c src/tags.c
--- src.old/tags.c	2024-10-13 11:42:59.999881145 +0900
+++ src/tags.c	2024-10-13 11:53:55.022089347 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/ticket.c src/ticket.c
--- src.old/ticket.c	2024-10-13 11:42:59.999881145 +0900
+++ src/ticket.c	2024-10-13 11:54:13.112313256 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -ur src.old/user_upload_status.c src/user_upload_status.c
--- src.old/user_upload_status.c	2024-10-13 11:42:59.998881132 +0900
+++ src/user_upload_status.c	2024-10-13 11:54:04.058201207 +0900
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
+#include <stdlib.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
