Author: Reiner Herrmann <reiner@reiner-h.de>
Description: Fix FTBFS with GCC 10
Bug-Debian: https://bugs.debian.org/957872

--- a/tetris.c
+++ b/tetris.c
@@ -32,6 +32,7 @@
 signed char specials[MAX_SPECIALS] = {-1}; /* Special block inventory */
 int next_piece;		/* Next piece to fall */
 
+PieceData piecedata[7][4];
 static struct timeval timeout;	/* Time of next action */
 int current_piece;	/* Current piece number */
 int current_rotation;	/* Current rotation value */
--- a/tetris.h
+++ b/tetris.h
@@ -50,7 +50,7 @@
     char shape[4][4];	/* Shape data for the piece */
 } PieceData;
 
-PieceData piecedata[7][4];
+extern PieceData piecedata[7][4];
 
 extern int current_piece, current_rotation;
 
