diff -Naur tetrinet-0.11/tetris.c tetrinet-0.11.patched/tetris.c
--- tetrinet-0.11/tetris.c	2003-09-07 12:29:29.000000000 -0400
+++ tetrinet-0.11.patched/tetris.c	2020-11-21 00:32:56.473867963 -0500
@@ -902,6 +902,7 @@
     switch (c) {
       case K_UP:	/* Rotate clockwise */
       case 'x':
+      case 'k':
 	if (piece_waiting)
 	    break;
 	rnew = (current_rotation+1) % 4;
@@ -942,6 +943,7 @@
 	break;
 
       case K_LEFT:	/* Move left */
+		case 'h':
 	if (piece_waiting)
 	    break;
 	if (x + pd->left > 0) {
@@ -957,6 +959,7 @@
 	break;
 
       case K_RIGHT:	/* Move right */
+		case 'l':
 	if (piece_waiting)
 	    break;
 	if (x + pd->right < FIELD_WIDTH-1) {
@@ -972,6 +975,7 @@
 	break;
 
       case K_DOWN:	/* Down one space */
+		case 'j':
 	if (piece_waiting)
 	    break;
 	step_down();
