--- libaaio-0.3.1.orig/configure.in	2005-09-22 07:34:37.000000000 -0600
+++ libaaio-0.3.1/configure.ac	2016-03-30 14:17:30.214587130 -0600
@@ -4,8 +4,10 @@
 AC_PREREQ(2.13)
-dnl AC_INIT(libaaio, 0.2.0, macbishop@users.sf.net)
-AC_INIT(aaio.c)
+AC_INIT([libaaio], [0.3.1])
 
+AC_CONFIG_FILES([Makefile man/Makefile man/man3/Makefile])
 dnl AC_CONFIG_HEADER([config.h])
+AC_CONFIG_SRCDIR([aaio.c])
 dnl DON'T FORGET TO CHANGE LIBTOOL VERSION IN MAKEFILE.AM
-AM_INIT_AUTOMAKE(libaaio,0.3.1)
+AC_CONFIG_MACRO_DIRS([m4])
+AM_INIT_AUTOMAKE
 
@@ -14,3 +16,5 @@
 dnl AC_PROG_LIBTOOL
-AM_PROG_LIBTOOL
+dnl AM_PROG_LIBTOOL
+AM_PROG_AR
+LT_INIT
 
@@ -42,3 +46,2 @@
 
-AC_OUTPUT([Makefile man/Makefile man/man3/Makefile])
-
+AC_OUTPUT
--- libaaio-0.3.1.orig/Makefile.am	2005-09-22 07:33:44.000000000 -0600
+++ libaaio-0.3.1/Makefile.am	2016-03-30 14:16:57.622994581 -0600
@@ -10,3 +10,5 @@
 
-CFLAGS=-Wall -pedantic -std=c99
+AM_CFLAGS=-Wall -pedantic -std=c99
+
+ACLOCAL_AMFLAGS=-I m4
 
