diff -rupN POTFILES.in.orig POTFILES.in.diff > Fix_POTFILES.patch
--- POTFILES.in.orig//po/POTFILES.in	2010-08-03 20:48:09.603076845 -0300
+++ POTFILES.in.diff//po/POTFILES.in	2010-08-03 21:32:27.439507572 -0300
@@ -20,7 +20,7 @@ cinelerra/channeledit.C
 cinelerra/channelpicker.C
 cinelerra/clipedit.C
 # cinelerra/compresspopup.C
-cinelerra/confirmquit.C
+# cinelerra/confirmquit.C
 cinelerra/confirmsave.C
 cinelerra/cpanel.C
 # cinelerra/cropvideo.C
@@ -56,7 +56,7 @@ cinelerra/indexfile.C
 cinelerra/indexthread.C
 cinelerra/interfaceprefs.C
 cinelerra/keyframepopup.C
-cinelerra/labelnavigate.C
+# cinelerra/labelnavigate.C
 cinelerra/loadfile.C
 cinelerra/loadmode.C
 cinelerra/main.C
@@ -65,7 +65,7 @@ cinelerra/mainmenu.C
 cinelerra/mainprogress.C
 cinelerra/manualgoto.C
 cinelerra/menueffects.C
-cinelerra/messages.C
+# cinelerra/messages.C
 cinelerra/meterpanel.C
 cinelerra/mwindow.C
 cinelerra/mwindowedit.C
@@ -95,7 +95,7 @@ cinelerra/recordmonitor.C
 cinelerra/recordprefs.C
 cinelerra/recordtransport.C
 cinelerra/recordvideo.C
-cinelerra/recordwindow.C
+# cinelerra/recordwindow.C
 # cinelerra/reindex.C
 cinelerra/render.C
 cinelerra/renderfarm.C
@@ -109,7 +109,7 @@ cinelerra/setformat.C
 cinelerra/sharedlocation.C
 cinelerra/splashgui.C
 cinelerra/statusbar.C
-cinelerra/threadindexer.C
+# cinelerra/threadindexer.C
 cinelerra/tipwindow.C
 # cinelerra/toggleauto.C
 cinelerra/trackcanvas.C
