From: Boyuan Yang <byang@debian.org>
Date: Sun, 15 Aug 2021 13:18:32 -0400
Subject: Use proper library version for libproxychains

Forwarded: not-needed
---
 Makefile        | 4 ++--
 src/proxyresolv | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 7f3f4fa..0524bf8 100644
--- a/Makefile
+++ b/Makefile
@@ -37,11 +37,11 @@ AR      = $(CROSS_COMPILE)ar
 RANLIB  = $(CROSS_COMPILE)ranlib
 SOCKET_LIBS =

-LDSO_SUFFIX = so
+LDSO_SUFFIX = so.4
 LD_SET_SONAME = -Wl,-soname=
 INSTALL = ./tools/install.sh

-LDSO_PATHNAME = libproxychains4.$(LDSO_SUFFIX)
+LDSO_PATHNAME = libproxychains.$(LDSO_SUFFIX)

 SHARED_LIBS = $(LDSO_PATHNAME)
 ALL_LIBS = $(SHARED_LIBS)
diff --git a/src/proxyresolv b/src/proxyresolv
index f10a412..fafed7e 100755
--- a/src/proxyresolv
+++ b/src/proxyresolv
@@ -12,7 +12,7 @@ if [ $# = 0 ] ; then
 fi


-test -z $LD_PRELOAD && export LD_PRELOAD=libproxychains4.so
+test -z $LD_PRELOAD && export LD_PRELOAD=libproxychains.so.4.0.0

 if type dig 1>/dev/null 2>&1 ; then
 dig       $1 @$DNS_SERVER +tcp | awk '/A.?[0-9]+\.[0-9]+\.[0-9]/{print $5; }'
