diff -ruN higan_v093-source/ananke/Makefile higan_v093-source_MOD/ananke/Makefile
--- higan_v093-source/ananke/Makefile	2013-08-02 17:36:56.000000000 -0300
+++ higan_v093-source_MOD/ananke/Makefile	2013-09-24 09:42:36.399970000 -0300
@@ -25,9 +25,9 @@
 
 install: uninstall
 ifeq ($(platform),x)
-	if [ ! -d ~/.config/ananke ]; then mkdir ~/.config/ananke; fi
-	sudo cp libananke.so $(path)/libananke.so.1
-	sudo ln -s $(path)/libananke.so.1 $(path)/libananke.so
+	mkdir -p $(path)
+	cp libananke.so $(path)/libananke.so.1
+	ln -s $(path)/libananke.so.1 $(path)/libananke.so
 else ifeq ($(platform),osx)
 	if [ ! -d ~/Library/Application\ Support/ananke ]; then mkdir ~/Library/Application\ Support/ananke; fi
 	sudo cp libananke.dylib $(path)/libananke.1.dylib
diff -ruN higan_v093-source/target-ethos/Makefile higan_v093-source_MOD/target-ethos/Makefile
--- higan_v093-source/target-ethos/Makefile	2013-08-16 20:39:20.000000000 -0300
+++ higan_v093-source_MOD/target-ethos/Makefile	2013-09-24 09:53:39.367766000 -0300
@@ -16,7 +16,7 @@
 # platform
 ifeq ($(platform),x)
   ruby := video.glx video.xv video.sdl
-  ruby += audio.alsa audio.openal audio.oss audio.pulseaudio audio.pulseaudiosimple audio.ao
+  ruby += audio.alsa audio.openal audio.oss audio.ao
   ruby += input.sdl input.x
 else ifeq ($(platform),osx)
   ruby := video.cgl
@@ -86,14 +86,14 @@
 
 install:
 ifeq ($(platform),x)
-	sudo install -D -m 755 out/$(name) $(DESTDIR)$(prefix)/bin/$(name)
-	sudo install -D -m 644 data/$(name).png $(DESTDIR)$(prefix)/share/pixmaps/$(name).png
-	sudo install -D -m 644 data/$(name).desktop $(DESTDIR)$(prefix)/share/applications/$(name).desktop
-
-	sudo mkdir -p /usr/share/$(name)
-	sudo cp -R profile/* /usr/share/$(name)
-	sudo cp data/cheats.bml /usr/share/$(name)/cheats.bml
-	sudo chmod -R 777 /usr/share/$(name)
+	install -D -m 755 out/$(name) $(DESTDIR)$(prefix)/bin/$(name)
+	install -D -m 644 data/$(name).png $(DESTDIR)$(prefix)/share/pixmaps/$(name).png
+	install -D -m 644 data/$(name).desktop $(DESTDIR)$(prefix)/share/applications/$(name).desktop
+
+	mkdir -p $(DESTDIR)/usr/share/$(name)
+	cp -R profile/* $(DESTDIR)/usr/share/$(name)
+	cp data/cheats.bml $(DESTDIR)/usr/share/$(name)/cheats.bml
+	chmod -R 777 $(DESTDIR)/usr/share/$(name)
 else ifeq ($(platform),osx)
 	sudo mkdir -p /Library/Application\ Support/$(name)
 	sudo cp -R profile/* /Library/Application\ Support/$(name)
