--- mog-0.63.1548/build/linux/Makefile.orig	2015-03-28 01:25:41.748009321 +0200
+++ mog-0.63.1548/build/linux/Makefile	2015-03-28 01:26:40.989268703 +0200
@@ -3,11 +3,11 @@
 PREFIX = /usr
 EXTRADIR = ./build/linux
 
-GAMEDIR = $(PREFIX)/games
+GAMEDIR = $(PREFIX)/usr/bin
 STARTUP = $(GAMEDIR)/$(EXE)
-BINDIR = $(PREFIX)/share/games/$(EXE)
-ICNDIR = $(PREFIX)/share/pixmaps
-APPDIR = $(PREFIX)/share/applications
+BINDIR = $(PREFIX)/usr/share/games/$(EXE)
+ICNDIR = $(PREFIX)/usr/share/pixmaps
+APPDIR = $(PREFIX)/usr/share/applications
 
 DATA = graphics rooms sound leeme.txt liesmich.txt readme.txt MoG-FAQ.txt LICENSE
 ICON = mog.png
