diff -Naur freecad-0.13.1830.orig/src/Doc/CMakeLists.txt freecad-0.13.1830/src/Doc/CMakeLists.txt
--- freecad-0.13.1830.orig/src/Doc/CMakeLists.txt	2013-02-02 18:09:17.000000000 +0100
+++ freecad-0.13.1830/src/Doc/CMakeLists.txt	2013-11-19 21:23:26.655733113 +0100
@@ -53,13 +53,6 @@
     SET(DOXYGEN_IMAGE_PATH ${CMAKE_SOURCE_DIR}/src/Gui/Icons)
     SET(DOXYGEN_LAYOUT_FILE ${CMAKE_SOURCE_DIR}/src/Doc/FreecadDoxygenLayout.xml)
 
-    find_package(Coin3DDoc)
-    if( COIN3D_DOC_FOUND )
-        SET(DOXYGEN_TAGFILES 
-            ${COIN3D_DOC_TAGFILE}=${COIN3D_DOC_PATH}
-        )
-    endif( COIN3D_DOC_FOUND )
-
     configure_file(${CMAKE_CURRENT_SOURCE_DIR}/BuildDevDoc.cfg.in  
                    ${CMAKE_CURRENT_BINARY_DIR}/BuildDevDoc.cfg @ONLY)
 
