--- FreeCAD-0.11.3729/src/3rdParty/salomesmesh/src/SMESH/SMESH_MeshEditor.cpp	2010-10-17 05:59:46.000000000 -0200
+++ FreeCAD-0.11.3729_patched/src/3rdParty/salomesmesh/src/SMESH/SMESH_MeshEditor.cpp	2011-08-10 20:04:59.000000000 -0300
@@ -2205,8 +2205,8 @@
   if ( projector.IsDone() ) {
     double u, v, minVal = DBL_MAX;
     for ( int i = projector.NbExt(); i > 0; i-- )
-      if ( projector.Value( i ) < minVal ) {
-        minVal = projector.Value( i );
+      if ( projector.SquareDistance( i ) < minVal ) {
+        minVal = projector.SquareDistance( i );
         projector.Point( i ).Parameter( u, v );
       }
     result.SetCoord( u, v );
--- FreeCAD-0.11.3729/src/3rdParty/salomesmesh/src/SMESH/SMESH_Pattern.cpp	2010-10-17 05:59:46.000000000 -0200
+++ FreeCAD-0.11.3729_patched/src/3rdParty/salomesmesh/src/SMESH/SMESH_Pattern.cpp	2011-08-10 20:06:21.000000000 -0300
@@ -436,8 +436,8 @@
   }
   double u, v, minVal = DBL_MAX;
   for ( int i = theProjectorPS.NbExt(); i > 0; i-- )
-    if ( theProjectorPS.Value( i ) < minVal ) {
-      minVal = theProjectorPS.Value( i );
+    if ( theProjectorPS.SquareDistance( i ) < minVal ) {
+      minVal = theProjectorPS.SquareDistance( i );
       theProjectorPS.Point( i ).Parameter( u, v );
     }
   return gp_XY( u, v );
--- FreeCAD-0.11.3729/src/Mod/Part/App/PrimitiveFeature.cpp	2010-08-27 11:22:14.000000000 -0300
+++ FreeCAD-0.11.3729_patched/src/Mod/Part/App/PrimitiveFeature.cpp	2011-08-10 20:41:35.000000000 -0300
@@ -141,9 +141,6 @@
     case BRepBuilderAPI_ParametersOutOfRange:
         error = "parameters out of range";
         break;
-    case BRepBuilderAPI_SurfaceNotC2:
-        error = "surface not C2";
-        break;
     default:
         error = "unknown error";
         break;
--- FreeCAD-0.11.3729/src/Mod/Part/App/TopoShape.cpp	2010-11-14 06:53:08.000000000 -0200
+++ FreeCAD-0.11.3729_patched/src/Mod/Part/App/TopoShape.cpp	2011-08-10 20:43:03.000000000 -0300
@@ -139,8 +139,6 @@
         return "Curve projection failed";
     case BRepBuilderAPI_ParametersOutOfRange:
         return "Parameters out of range";
-    case BRepBuilderAPI_SurfaceNotC2:
-        return "Surface not C2-continous";
     default:
         return "Unknown creation error";
     }
