diff -Nur texlive-20140525-source.orig/texk/xdvik/events.c texlive-20140525-source/texk/xdvik/events.c
--- texlive-20140525-source.orig/texk/xdvik/events.c	2014-04-14 03:35:50.000000000 -0500
+++ texlive-20140525-source/texk/xdvik/events.c	2014-06-15 13:18:40.762860567 -0500
@@ -5306,7 +5306,7 @@
 		}
 
 		if (actp->proc == Act_wheel) {
-#  if XAW
+#  if !MOTIF
 		    if (globals.widgets.y_bar != NULL)
 			XtCallCallbacks(globals.widgets.y_bar, XtNscrollProc,
 			  cast_int_to_XtPointer(dist));
@@ -5317,7 +5317,7 @@
 #  endif /* MOTIF */
 		}
 		else {	/* Act_hwheel */
-#  if XAW
+#  if !MOTIF
 		    if (globals.widgets.x_bar != NULL)
 			XtCallCallbacks(globals.widgets.x_bar, XtNscrollProc,
 			  cast_int_to_XtPointer(dist));
