diff -Nur texlive-20110705-source.orig//texk/kpathsea/texmf.cnf texlive-20110705-source/texk/kpathsea/texmf.cnf
--- texlive-20110705-source.orig//texk/kpathsea/texmf.cnf	2011-06-18 17:10:46.000000000 -0500
+++ texlive-20110705-source/texk/kpathsea/texmf.cnf	2011-07-22 21:30:42.459806437 -0500
@@ -59,7 +59,7 @@
 TEXMFDIST = $TEXMFROOT/texmf-dist
 
 % Local additions to the distribution trees.
-TEXMFLOCAL = $TEXMFROOT/../texmf-local
+TEXMFLOCAL = $TEXMFROOT/texmf-local
 
 % TEXMFSYSVAR, where *-sys store cached runtime data.
 TEXMFSYSVAR = $TEXMFROOT/texmf-var
