--- share/texmf-dist/scripts/texcount/texcount.pl.orig	2013-11-11 18:16:18.388313783 +0000
+++ share/texmf-dist/scripts/texcount/texcount.pl	2013-11-11 18:16:32.407314117 +0000
@@ -2218,7 +2218,7 @@
   }
   $template=~s/\{VER\}/$versionnumber/gi;
   # TODO: Should base warnings and errors on TeXcode or Main object
-  $template=__process_template($template,"W|WARN|WARNING|WARNINGS",length(%warnings));
+  $template=__process_template($template,"W|WARN|WARNING|WARNINGS",scalar(keys %warnings));
   $template=__process_template($template,"E|ERR|ERROR|ERRORS",$errorcount);
   $template=__process_template($template,"S|SUM",get_sum_count($count));
   $template=__process_template($template,"T|TITLE",$count->{'title'}||"");
