--- texlive-20120701-texmf/texmf-dist/web2c/texmf.cnf.orig	2013-06-22 04:30:50.766408343 -0500
+++ texlive-20120701-texmf/texmf-dist/web2c/texmf.cnf	2013-06-22 04:31:13.717121421 -0500
@@ -51,7 +51,7 @@
 % Redistributors will probably want $SELFAUTODIR/share, i.e., /usr/share.
 % Kpathsea sets SELFAUTOLOC (the directory with the binary),
 % SELFAUTODIR (its parent), and SELFAUTOPARENT (its grandparent).
-TEXMFROOT = $SELFAUTOPARENT
+TEXMFROOT = $SELFAUTODIR/share
 
 % The tree containing runtime files related to the specific
 % distribution and version.
@@ -61,7 +61,7 @@
 TEXMFDIST = $TEXMFROOT/texmf-dist
 
 % Local additions to the distribution trees.
-TEXMFLOCAL = $TEXMFROOT/../texmf-local
+TEXMFLOCAL = $TEXMFROOT/texmf-local
 
 % TEXMFSYSVAR, where *-sys store cached runtime data.
 TEXMFSYSVAR = $TEXMFROOT/texmf-var
