diff -Naur sms_sdl-0.9.4a-r7.1/sdl/Makefile sms_sdl-0.9.4a-r7.1.patched/sdl/Makefile
--- sms_sdl-0.9.4a-r7.1/sdl/Makefile	2003-07-16 23:37:38.000000000 -0400
+++ sms_sdl-0.9.4a-r7.1.patched/sdl/Makefile	2009-02-05 17:39:28.000000000 -0500
@@ -12,7 +12,7 @@
 NAME	  = sms_sdl
 
 CC        = gcc
-CFLAGS    = `sdl-config --cflags` -O2
+CFLAGS    = `sdl-config --cflags` $(OPTFLAGS)
 DEFINES   = -DLSB_FIRST -DX86_ASM
 INCLUDES  = -I. -I.. -I../cpu -I../sound
 LIBS	  = `sdl-config --libs`
@@ -22,9 +22,9 @@
             ../cpu/z80.o ../sound/emu2413.o ../sound/sn76496.o
 
 # (un)comment to enable ZIP support
-#DEFINES  += -DUSE_ZLIB
-#LIBS     += -Lz
-#OBJECTS  += unzip.o
+DEFINES  += -DUSE_ZLIB
+LIBS     += -lz
+OBJECTS  += unzip.o
 
 
 all: $(NAME)
