--- sms_sdl-0.9.4a-r7.1/sdl/main.c	2003-07-21 12:40:56.000000000 -0300
+++ sms_sdl-0.9.4a-r7.1.patched/sdl/main.c	2009-05-12 17:42:29.362768285 -0300
@@ -45,21 +45,19 @@
   int i;
 
   /* default virtual console emulation settings */
-  sms.use_fm = 0;
-  sms.country = TYPE_OVERSEAS;
+  cfg.fm = 0;
+  cfg.japan = 0;
 
   cfg.filter = -1;
   strcpy(cfg.game_name, argv[1]);
 
   for(i = 2; i < argc; ++i) {
     if(strcasecmp(argv[i], "--fm") == 0)
-      sms.use_fm = 1;
+      cfg.fm = 1;
     else if(strcasecmp(argv[i], "--japan") == 0)
-      sms.country = TYPE_DOMESTIC;
-    else if(strcasecmp(argv[i], "--usesram") == 0) {
+      cfg.japan = 1;
+    else if(strcasecmp(argv[i], "--usesram") == 0)
       cfg.usesram = 1;
-      sms.save = 1;
-    }
     else if(strcasecmp(argv[i], "--fskip") == 0) {
       if(++i<argc) {
         cfg.frameskip = atoi(argv[i]);
