diff -Naur sms_sdl-0.9.4a-r7.1/sdl/Makefile sms_sdl-0.9.4a-r7.1.patched/sdl/Makefile
--- sms_sdl-0.9.4a-r7.1/sdl/Makefile	2009-09-30 18:39:29.000000000 -0400
+++ sms_sdl-0.9.4a-r7.1.patched/sdl/Makefile	2009-09-30 18:40:10.000000000 -0400
@@ -13,7 +13,7 @@
 
 CC        = gcc
 CFLAGS    = `sdl-config --cflags` $(OPTFLAGS)
-DEFINES   = -DLSB_FIRST -DX86_ASM
+DEFINES   = -DLSB_FIRST
 INCLUDES  = -I. -I.. -I../cpu -I../sound
 LIBS	  = `sdl-config --libs`
 
diff -Naur sms_sdl-0.9.4a-r7.1/types.h sms_sdl-0.9.4a-r7.1.patched/types.h
--- sms_sdl-0.9.4a-r7.1/types.h	2003-01-07 12:37:58.000000000 -0500
+++ sms_sdl-0.9.4a-r7.1.patched/types.h	2009-09-30 18:40:19.000000000 -0400
@@ -4,11 +4,11 @@
 
 typedef unsigned char uint8;
 typedef unsigned short int uint16;
-typedef unsigned long int uint32;
+typedef unsigned int uint32;
 
 typedef signed char int8;
 typedef signed short int int16;
-typedef signed long int int32;
+typedef signed int int32;
 
 #endif /* _TYPES_H_ */
 
