--- cpan2tgz-0.6/buildslackpkg.sh.orig	2006-08-23 15:51:09.546266236 -0500
+++ cpan2tgz-0.6/buildslackpkg.sh	2006-08-23 16:02:48.273371442 -0500
@@ -1,8 +1,11 @@
 #!/bin/sh
 
-DESTINATION=/tmp/$$
+PKG=/tmp/SBo/package-cpan2tgz
+OUTPUT=/tmp
+DESTINATION=$PKG
 VERSION=`grep VERSION cpan2tgz |awk '{print $4}'|cut -f2 -d\"`
 PKGNAME=cpan2tgz
+TAG=_SBo
 
 perl Makefile.PL
 make
@@ -32,11 +35,8 @@
 	if [ -d ./usr/share/man ]; then
 		mv ./usr/share/man ./usr
 	fi
-	if [ -d ./usr/bin ]; then
-		chown -R root.bin ./usr/bin
-		chmod 755 ./usr/bin/*
-	fi
 	chmod 644 ./usr/man/man?/*
+	gzip -9 ./usr/man/man?/*
 	rmdir ./usr/share
 	mkdir install
 
@@ -51,8 +51,6 @@
 	  rm $PERLLOCALPOD
   fi
 
-	echo "perl" > ./install/slack-required
-
 	cat >./install/slack-desc <<EOF
 # HOW TO EDIT THIS FILE:
 # The "handy ruler" below makes it easier to edit a package description.  Line
@@ -73,7 +71,7 @@
 ${PKGNAME}: http://software.jaos.org/
 ${PKGNAME}: 
 EOF
-	makepkg -l y -c n /usr/src/${PKGNAME}-${VERSION}-noarch-1.tgz
+	makepkg -l y -c n ${OUTPUT}/${PKGNAME}-${VERSION}-noarch-1$TAG.tgz
 )
 
 make distclean
