diff -Naur jezzball-1.1/src/main.cpp jezzball-1.1.patched/src/main.cpp
--- jezzball-1.1/src/main.cpp	2007-09-09 18:07:10.000000000 -0400
+++ jezzball-1.1.patched/src/main.cpp	2014-05-12 13:36:14.000000000 -0400
@@ -250,7 +250,7 @@
 	// sub-bitmap is deleted before parent bitmap which
 	// coincidentally works out well for us :)
 	Bitmap sub_work( work, offset_x, offset_y, 600, 460 );
-	bool quit = false;
+	bool quit = false, died = false;
 	speed_counter = 0;
 	while ( speed_counter < 15 );
 	speed_counter = 0;
@@ -263,7 +263,8 @@
 		}
 		// if ( key[ KEY_R ] ) quit = true;
 
-		if ( show_insults == -1 && rnd( 40 ) == rnd( 40 ) ){
+		if ( show_insults == -1 && died ) {
+			died = false;
 			show_insults = 60;
 			insult = rnd( num_insults );
 			// insult_x = rnd( 200 ) + 100;
@@ -284,6 +285,7 @@
 			for ( int q = 0; q < 5; q++ )
 				if ( game->runGame( &mouse, offset_x, offset_y ) ){
 					player->incLives( -1 );
+					died = true;
 					redraw_name = true;
 				}
 			speed_counter--;
