diff -urN qTox-3223707.orig/src/friend.cpp qTox-3223707/src/friend.cpp
--- qTox-3223707.orig/src/friend.cpp	2015-04-05 22:37:15.000000000 +0200
+++ qTox-3223707/src/friend.cpp	2015-04-16 05:55:03.531187202 +0200
@@ -40,6 +40,11 @@
         chatForm->loadHistory(QDateTime::currentDateTime().addDays(-7), true);
         widget->historyLoaded = true;
     }
+    if (Settings::getInstance().getEnableLogging())
+    {
+        chatForm->loadHistory(QDateTime::currentDateTime().addDays(-7), true);
+        widget->historyLoaded = true;
+    }
 }
 
 Friend::~Friend()
diff -urN qTox-3223707.orig/src/widget/form/genericchatform.cpp qTox-3223707/src/widget/form/genericchatform.cpp
--- qTox-3223707.orig/src/widget/form/genericchatform.cpp	2015-04-05 22:37:15.000000000 +0200
+++ qTox-3223707/src/widget/form/genericchatform.cpp	2015-04-16 05:57:01.339540592 +0200
@@ -374,6 +374,15 @@
                 return res;
         }
     }
+    // try load alias from settings
+    QString lastTry = Settings::getInstance().getFriendAlias(id);
+    if (!lastTry.isEmpty())
+        return lastTry;
+
+    // try load name from tox
+    lastTry = Core::getInstance()->getPeerName(id);
+    if (!lastTry.isEmpty())
+        return lastTry;
 
     return QString();
 }
