--- qTox-3223707/src/widget/form/settings/generalsettings.ui.orig	2015-04-05 22:37:15.000000000 +0200
+++ qTox-3223707/src/widget/form/settings/generalsettings.ui	2015-04-16 05:22:17.389288167 +0200
@@ -406,12 +406,6 @@
        </item>
        <item>
         <widget class="QGroupBox" name="themeGroup">
-         <property name="minimumSize">
-          <size>
-           <width>0</width>
-           <height>232</height>
-          </size>
-         </property>
          <property name="title">
           <string>Theme</string>
          </property>
