--- qTox-3223707/src/widget/systemtrayicon.cpp.orig	2015-04-05 22:37:15.000000000 +0200
+++ qTox-3223707/src/widget/systemtrayicon.cpp	2015-04-16 02:22:08.955184668 +0200
@@ -230,8 +233,8 @@
             g_signal_connect(item, "activate", G_CALLBACK(callback), a);
             gtk_widget_show(item);
         }
-        void (*callbackMenu)(StatusNotifier*, gint, gint, gpointer) =
-                [](StatusNotifier*, gint, gint, gpointer data)
+        void (*callbackMenu)(GtkMenu*, gint, gint, gpointer) =
+                [](GtkMenu*, gint, gint, gpointer data)
         {
             gtk_widget_show_all(((SystemTrayIcon*)data)->gtkMenu);
             gtk_menu_popup(GTK_MENU(((SystemTrayIcon*)data)->gtkMenu), 0, 0, 0, 0, 3, gtk_get_current_event_time());
