--- qTox-3223707/src/widget/systemtrayicon.cpp.orig	2015-04-05 22:37:15.000000000 +0200
+++ qTox-3223707/src/widget/systemtrayicon.cpp	2015-04-16 02:22:08.955184668 +0200
@@ -9,6 +9,9 @@
 SystemTrayIcon::SystemTrayIcon()
 {
     QString desktop = getenv("XDG_CURRENT_DESKTOP");
+    if (desktop.isEmpty()) {
+        desktop = getenv("DESKTOP_SESSION");
+    }
     desktop = desktop.toLower();
     if (false);
     #ifdef ENABLE_SYSTRAY_UNITY_BACKEND
