--- cmake/Toolset.cmake	2023-01-02 17:31:12.369073592 -0700
+++ cmake/Toolset.cmake	2023-01-02 17:32:15.127507305 -0700
@@ -56,10 +56,10 @@
 find_program(GOLD_PATH NAMES "ld.gold" "gold")
 
 if(NOT LINKER_NAME)
-    if(LLD_PATH)
-        set(LINKER_NAME "lld")
-    elseif(GOLD_PATH)
+    if(GOLD_PATH)
         set(LINKER_NAME "gold")
+    elseif(LLD_PATH)
+        set(LINKER_NAME "lld")
     else()
         message(STATUS "Use generic 'ld' as a linker")
     endif()
