--- cmake/Toolset.cmake	2020-09-23 18:31:57.275661294 -0600
+++ cmake/Toolset.cmake	2020-09-23 18:32:34.812659218 -0600
@@ -45,10 +45,10 @@
 find_program(GOLD_PATH NAMES "ld.gold" "gold")
 
 if(NOT LINKER_NAME)
-    if(LLD_PATH)
-        set(LINKER_NAME "lld")
-    elseif(GOLD_PATH)
+    if(GOLD_PATH)
         set(LINKER_NAME "gold")
+    elseif(LLD_PATH)
+        set(LINKER_NAME "lld")
     else()
         message(STATUS "Use generic 'ld' as a linker")
     endif()
