diff --color -Naur old/CodeLite/fileutils.cpp new/CodeLite/fileutils.cpp
--- old/CodeLite/fileutils.cpp	2022-04-18 01:09:00.000000000 -0400
+++ new/CodeLite/fileutils.cpp	2025-12-26 14:36:13.800162566 -0500
@@ -58,6 +58,7 @@
 #endif
 #include <functional>
 #include <memory>
+#include <cstdint>
 #include <wx/filename.h>
 
 using namespace std;
diff --color -Naur old/Plugin/dtl/Diff.hpp new/Plugin/dtl/Diff.hpp
--- old/Plugin/dtl/Diff.hpp	2022-04-18 01:09:00.000000000 -0400
+++ new/Plugin/dtl/Diff.hpp	2025-12-26 15:02:20.168154445 -0500
@@ -165,7 +165,7 @@
 
     bool trivialEnabled() const { return trivial; }
 
-    void enableTrivial() const { this->trivial = true; }
+    void enableTrivial() { this->trivial = true; }
 
     void disableTrivial() { this->trivial = false; }
 
diff --color -Naur old/wxcrafter/myxh_propgrid.cpp new/wxcrafter/myxh_propgrid.cpp
--- old/wxcrafter/myxh_propgrid.cpp	2022-04-18 01:09:00.000000000 -0400
+++ new/wxcrafter/myxh_propgrid.cpp	2025-12-26 16:13:56.736132168 -0500
@@ -48,7 +48,7 @@
 
         // add the splitter property after the children were added
         int splitterLeft = GetBool("splitterleft");
-        if(splitterLeft) { m_pgmgr->GetPage(0)->SetSplitterLeft(); }
+        if(splitterLeft) { m_pgmgr->GetPage(0)->GetGrid()->SetSplitterLeft(); }
 
         int splitterPos = GetLong("splitterpos", wxNOT_FOUND);
         if(splitterPos != wxNOT_FOUND) { m_pgmgr->GetPage(0)->SetSplitterPosition(splitterPos); }
