diff -Naur stressapptest-1.0.9/configure.ac stressapptest-1.0.9.patched/configure.ac
--- stressapptest-1.0.9/configure.ac	2018-08-09 20:01:24.000000000 -0400
+++ stressapptest-1.0.9.patched/configure.ac	2020-06-06 13:34:23.000966542 -0400
@@ -19,6 +19,10 @@
     AC_DEFINE([STRESSAPPTEST_CPU_X86_64],[],
               [Defined if the target CPU is x86_64])
     ], 
+  [*i586*], [
+    AC_DEFINE([STRESSAPPTEST_CPU_I686],[],
+              [Defined if the target CPU is i586])
+    ], 
   [*i686*], [
     AC_DEFINE([STRESSAPPTEST_CPU_I686],[],
               [Defined if the target CPU is i686])
@@ -35,7 +39,7 @@
     AC_DEFINE([STRESSAPPTEST_CPU_AARCH64],[],
               [Defined if the target CPU is aarch64])
     ],
-[AC_MSG_WARN([Unsupported CPU: $host_cpu! Try x86_64, i686, powerpc, armv7a, or aarch64])]
+[AC_MSG_WARN([Unsupported CPU: $host_cpu! Try x86_64, i586, i686, powerpc, armv7a, or aarch64])]
 )
 
 ## The following allows like systems to share settings. This is not meant to
