--- stressapptest-1.0.11/configure.ac.orig	2024-05-16 06:24:36.806255976 +0700
+++ stressapptest-1.0.11/configure.ac	2024-05-16 06:25:18.914725236 +0700
@@ -26,6 +26,10 @@
     AC_DEFINE([STRESSAPPTEST_CPU_X86_64],[],
               [Defined if the target CPU is x86_64])
     ],
+  [*i586*], [
+    AC_DEFINE([STRESSAPPTEST_CPU_I586],[],
+              [Defined if the target CPU is i586])
+    ],
   [*i686*], [
     AC_DEFINE([STRESSAPPTEST_CPU_I686],[],
               [Defined if the target CPU is i686])
@@ -50,7 +54,7 @@
     AC_DEFINE([STRESSAPPTEST_CPU_LOONGARCH],[],
               [Defined if the target CPU is LOONGARCH])
     ],
-[AC_MSG_WARN([Unsupported CPU: $host_cpu! Try x86_64, i686, mips, powerpc, armv7a, aarch64 or loongarch])]
+[AC_MSG_WARN([Unsupported CPU: $host_cpu! Try x86_64, i586, i686, mips, powerpc, armv7a, aarch64 or loongarch])]
 )
 
 ## The following allows like systems to share settings. This is not meant to
