--- a/helix-term/build.rs	2025-01-20 00:25:42.000000000 +0300
+++ b/helix-term/build.rs	2025-06-14 14:11:03.753775574 +0300
@@ -1,8 +1,7 @@
-use helix_loader::grammar::{build_grammars, fetch_grammars};
+use helix_loader::grammar::build_grammars;
 
 fn main() {
     if std::env::var("HELIX_DISABLE_AUTO_GRAMMAR_BUILD").is_err() {
-        fetch_grammars().expect("Failed to fetch tree-sitter grammars");
         build_grammars(Some(std::env::var("TARGET").unwrap()))
             .expect("Failed to compile tree-sitter grammars");
     }
