--- libavogadro/src/python/sip.cpp.old	2010-11-14 14:55:44.928487067 -0800
+++ libavogadro/src/python/sip.cpp	2010-11-14 15:37:31.589487066 -0800
@@ -198,13 +198,19 @@
 #endif
     
     // reinterpret to sipWrapper
+#if SIP_API_MAJOR_NR >= 8
+    sipSimpleWrapper *wrapper = reinterpret_cast<sipSimpleWrapper*>(obj_ptr);
+    return wrapper->data;
+#else
+
 #ifdef SIP_4_8
     sipSimpleWrapper *wrapper = reinterpret_cast<sipSimpleWrapper*>(obj_ptr);
 #else
     sipWrapper *wrapper = reinterpret_cast<sipWrapper*>(obj_ptr);
+     // return the C++ pointer
 #endif
-    // return the C++ pointer
     return wrapper->u.cppPtr;
+#endif
   }
     
   QClass_converters()
