diff -Naur avogadro-1.0.3.orig/avogadro/src/mainwindow.h avogadro-1.0.3/avogadro/src/mainwindow.h
--- avogadro-1.0.3.orig/avogadro/src/mainwindow.h	2011-04-25 05:22:18.000000000 +0000
+++ avogadro-1.0.3/avogadro/src/mainwindow.h	2012-07-11 13:57:36.624006453 +0000
@@ -24,6 +24,7 @@
 #ifndef MAINWINDOW_H
 #define MAINWINDOW_H
 
+#include <cstddef>
 #include <Eigen/Geometry>
 #include "ui_mainwindow.h"
 #include "flattabwidget.h"
diff -Naur avogadro-1.0.3.orig/libavogadro/src/tools/eyecandy.h avogadro-1.0.3/libavogadro/src/tools/eyecandy.h
--- avogadro-1.0.3.orig/libavogadro/src/tools/eyecandy.h	2011-04-25 05:22:18.000000000 +0000
+++ avogadro-1.0.3/libavogadro/src/tools/eyecandy.h	2012-07-11 13:43:22.892050693 +0000
@@ -25,6 +25,7 @@
 #ifndef EYECANDY_H
 #define EYECANDY_H
 
+#include <cstddef>
 #include <Eigen/Core>
 
 namespace Avogadro {
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/crystallography/avospglib.h avogadro-1.1.0/libavogadro/src/extensions/crystallography/avospglib.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/crystallography/avospglib.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/crystallography/avospglib.h	2012-12-02 23:46:44.297457487 +0000
@@ -18,6 +18,7 @@
 #ifndef AVOSPGLIB_H
 #define AVOSPGLIB_H
 
+#include <cstddef>
 #include <Eigen/Core>
 
 #include <QtCore/QList>
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/qtaim/qtaimcriticalpointlocator.cpp avogadro-1.1.0/libavogadro/src/extensions/qtaim/qtaimcriticalpointlocator.cpp
--- avogadro-1.1.0.orig/libavogadro/src/extensions/qtaim/qtaimcriticalpointlocator.cpp	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/qtaim/qtaimcriticalpointlocator.cpp	2012-12-03 00:46:44.820773931 +0000
@@ -28,6 +28,7 @@
 #include "qtaimlsodaintegrator.h"
 #include "qtaimmathutilities.h"
 
+#include <cstddef>
 #include <Eigen/Core>
 
 #include <QList>
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/qtaim/qtaimlsodaintegrator.h avogadro-1.1.0/libavogadro/src/extensions/qtaim/qtaimlsodaintegrator.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/qtaim/qtaimlsodaintegrator.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/qtaim/qtaimlsodaintegrator.h	2012-12-03 00:32:41.095760808 +0000
@@ -30,6 +30,7 @@
 #include <QVector3D>
 #include <QPair>
 
+#include <cstddef>
 #include <Eigen/Core>
 
 #include "qtaimwavefunction.h"
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/qtaim/qtaimmathutilities.cpp avogadro-1.1.0/libavogadro/src/extensions/qtaim/qtaimmathutilities.cpp
--- avogadro-1.1.0.orig/libavogadro/src/extensions/qtaim/qtaimmathutilities.cpp	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/qtaim/qtaimmathutilities.cpp	2012-12-03 00:46:24.188773609 +0000
@@ -23,6 +23,7 @@
 **********************************************************************/
 
 #include <cmath>
+#include <cstddef>
 #include <Eigen/QR>
 
 #include "qtaimmathutilities.h"
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/qtaim/qtaimmathutilities.h avogadro-1.1.0/libavogadro/src/extensions/qtaim/qtaimmathutilities.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/qtaim/qtaimmathutilities.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/qtaim/qtaimmathutilities.h	2012-12-03 00:32:01.134760196 +0000
@@ -26,6 +26,7 @@
 
 #include <QtGlobal>
 
+#include <cstddef>
 #include <Eigen/Core>
 
 using namespace Eigen;
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/qtaim/qtaimodeintegrator.h avogadro-1.1.0/libavogadro/src/extensions/qtaim/qtaimodeintegrator.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/qtaim/qtaimodeintegrator.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/qtaim/qtaimodeintegrator.h	2012-12-03 00:33:01.432761053 +0000
@@ -30,6 +30,7 @@
 #include <QVector3D>
 #include <QPair>
 
+#include <cstddef>
 #include <Eigen/Core>
 
 #include "qtaimwavefunction.h"
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/qtaim/qtaimwavefunctionevaluator.h avogadro-1.1.0/libavogadro/src/extensions/qtaim/qtaimwavefunctionevaluator.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/qtaim/qtaimwavefunctionevaluator.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/qtaim/qtaimwavefunctionevaluator.h	2012-12-03 00:33:28.571761559 +0000
@@ -10,6 +10,7 @@
 
 #include "qtaimwavefunction.h"
 
+#include <cstddef>
 #include <Eigen/Core>
 
 using namespace Eigen;
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/atom.h avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/atom.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/atom.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/atom.h	2012-12-02 23:57:35.872728048 +0000
@@ -19,6 +19,7 @@
 
 #include "molecule.h"
 
+#include <cstddef>
 #include <Eigen/Core>
 
 namespace OpenQube {
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/cube.h avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/cube.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/cube.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/cube.h	2012-12-03 00:16:50.638746161 +0000
@@ -20,6 +20,7 @@
 #include "openqubeabi.h"
 
 #include <vector>
+#include <cstddef>
 #include <Eigen/Core>
 #include <QtCore/QString>
 
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/gamessukout.h avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/gamessukout.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/gamessukout.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/gamessukout.h	2012-12-03 00:17:25.300746590 +0000
@@ -19,6 +19,7 @@
 
 #include "gaussianset.h"
 
+#include <cstddef>
 #include <Eigen/Core>
 
 #include <vector>
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/gamessus.h avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/gamessus.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/gamessus.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/gamessus.h	2012-12-03 00:17:40.584746828 +0000
@@ -18,6 +18,7 @@
 #define GAMESSUS_H
 
 #include <QtCore/QIODevice>
+#include <cstddef>
 #include <Eigen/Core>
 #include <vector>
 
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/gaussianfchk.h avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/gaussianfchk.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/gaussianfchk.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/gaussianfchk.h	2012-12-03 00:17:52.130747005 +0000
@@ -18,6 +18,7 @@
 #define GAUSSIANFCHK_H
 
 #include <QtCore/QIODevice>
+#include <cstddef>
 #include <Eigen/Core>
 #include <vector>
 
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/gaussianset.h avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/gaussianset.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/gaussianset.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/gaussianset.h	2012-12-03 00:18:04.072747212 +0000
@@ -22,6 +22,7 @@
 
 #include <QtCore/QFuture>
 
+#include <cstddef>
 #include <Eigen/Core>
 #include <vector>
 
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/molden.h avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/molden.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/molden.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/molden.h	2012-12-03 00:18:23.164747493 +0000
@@ -17,6 +17,7 @@
 #ifndef MOLDEN_H
 #define MOLDEN_H
 
+#include <cstddef>
 #include <Eigen/Core>
 #include <vector>
 
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/molecule.h avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/molecule.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/molecule.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/molecule.h	2012-12-03 00:08:10.094737989 +0000
@@ -22,6 +22,7 @@
 #include "atom.h"
 
 #include <vector>
+#include <cstddef>
 #include <Eigen/Core>
 
 namespace OpenQube {
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/mopacaux.h avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/mopacaux.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/mopacaux.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/mopacaux.h	2012-12-03 00:18:48.113747871 +0000
@@ -18,6 +18,7 @@
 #define MOPACAUX_H
 
 #include <QtCore/QTextStream>
+#include <cstddef>
 #include <Eigen/Core>
 #include <vector>
 
diff -Naur avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/slaterset.h avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/slaterset.h
--- avogadro-1.1.0.orig/libavogadro/src/extensions/surfaces/openqube/slaterset.h	2012-09-03 17:07:00.000000000 +0000
+++ avogadro-1.1.0/libavogadro/src/extensions/surfaces/openqube/slaterset.h	2012-12-03 00:19:11.592748237 +0000
@@ -21,6 +21,7 @@
 
 #include <QtCore/QFuture>
 
+#include <cstddef>
 #include <Eigen/Core>
 #include <vector>
 
