--- ./Pinta/Pinta.csproj	2010-07-07 02:20:09.000000000 -0700
+++ ./Pinta/Pinta.csproj.clean	2010-07-07 02:20:59.000000000 -0700
@@ -214,8 +214,6 @@
     <InstallPrefix>/usr/local</InstallPrefix>
   </PropertyGroup>
   <ItemGroup>
-    <BinFiles Include="$(OutputPath)/Pinta.exe" />
-    <BinFiles Include="$(OutputPath)/Pinta.*.dll" />
   </ItemGroup>
   <Target Name="Install">
     <Copy SourceFiles="@(BinFiles)" DestinationFolder="$(InstallPrefix)/lib/pinta" />
