diff -Naur joyce-2.2.15/bin/PcKeyboard.cxx joyce-2.2.15.patched/bin/PcKeyboard.cxx
--- joyce-2.2.15/bin/PcKeyboard.cxx	2017-07-07 14:12:52.000000000 -0400
+++ joyce-2.2.15.patched/bin/PcKeyboard.cxx	2021-08-25 15:05:05.208713380 -0400
@@ -199,7 +199,7 @@
 {
         bool b = false;
         int rv = 0;
-	int n, kc_entry;
+	unsigned int n, kc_entry;
         Uint16 *km;
         SDLKey keysym = SDLK_UNKNOWN;
 
