diff -Naur gnome-commander-20120801_2f83e80.orig/src/gnome-cmd-collection.h gnome-commander-20120801_2f83e80/src/gnome-cmd-collection.h
--- gnome-commander-20120801_2f83e80.orig/src/gnome-cmd-collection.h	2012-08-01 21:05:40.000000000 +0200
+++ gnome-commander-20120801_2f83e80/src/gnome-cmd-collection.h	2012-09-28 20:41:46.264480154 +0200
@@ -35,9 +35,9 @@
     template <typename T>
     struct Collection<T *>: std::set<T *>
     {
-        void add(T *t)            {  insert(t);                          }
-        void remove(T *t)         {  erase(t);                           }
-        bool contain(T *t) const  {  return find(t)!=Collection::end();  }
+        void add(T *t)            {  this->insert(t);                          }
+        void remove(T *t)         {  this->erase(t);                           }
+        bool contain(T *t) const  {  return this->find(t)!=Collection::end();  }
 
         GList *get_list();
     };
