--- Makefile-orig	2010-04-09 20:02:02.000000000 +0400
+++ Makefile	2010-04-09 20:02:39.000000000 +0400
@@ -53,8 +53,8 @@
 # but we use a local copy if we don't find it.
 #
 #KERNELSRC=/lib/modules/`uname -r`/build/
-KERNELSRC?=./linux
-OSFLAGS?= -DLINUX -I$(KERNELSRC)/include/
+KERNELSRC=/usr/src/linux
+OSFLAGS= -DLINUX -I$(KERNELSRC)/include/
 #
 # Uncomment the following to use the kernel interface under Linux
 # This requires the pppol2tp-linux-2.4.27.patch patch from contrib
@@ -62,7 +62,7 @@
 # are packages seperately (eg kernel-headers on Fedora)
 # Note: 2.6.23+ support still needs some changes in the xl2tpd source
 #
-#OSFLAGS+= -DUSE_KERNEL
+OSFLAGS+= -DUSE_KERNEL
 #
 #
 # Uncomment the next line for FreeBSD
@@ -97,10 +97,10 @@
 #LIBS= $(OSLIBS) # -lefence # efence for malloc checking
 EXEC=xl2tpd
 
-PREFIX?=/usr/local
+PREFIX?=/usr
 SBINDIR?=$(DESTDIR)${PREFIX}/sbin
 BINDIR?=$(DESTDIR)${PREFIX}/bin
-MANDIR?=$(DESTDIR)${PREFIX}/share/man
+MANDIR?=$(DESTDIR)${PREFIX}/man
 
 
 all: $(EXEC) pfc
@@ -113,7 +113,7 @@
 
 pfc:
 	$(CC) $(CFLAGS) -c contrib/pfc.c
-	$(CC) $(LDFLAGS) -lpcap $(LDLIBS) -o pfc pfc.o
+	$(CC)  pfc.o $(LDFLAGS) -lpcap $(LDLIBS) -o pfc
 
 romfs:
 	$(ROMFSINST) /bin/$(EXEC)
