--- qtile-0.11.1/setup.py.orig	2018-03-04 19:04:35.588264201 +0100
+++ qtile-0.11.1/setup.py	2018-03-04 19:05:20.151265263 +0100
@@ -121,7 +121,6 @@
     url="http://qtile.org",
     license="MIT",
     install_requires=dependencies,
-    setup_requires=dependencies,
     extras_require={
         'ipython': ["ipykernel", "jupyter_console"],
     },
@@ -155,7 +154,6 @@
     data_files=[
         ('share/man/man1', ['resources/qtile.1',
                             'resources/qshell.1'])],
-    cmdclass={'install': CheckCairoXcb},
     cffi_modules=[
         'libqtile/ffi_build.py:pango_ffi',
         'libqtile/ffi_build.py:xcursors_ffi',
