--- profile-sync-daemon/init/psd.openrc.orig	2014-11-19 14:35:06.049935645 +0100
+++ profile-sync-daemon/init/psd.openrc	2014-11-19 14:37:33.235934241 +0100
@@ -1,29 +1,21 @@
-#!/sbin/runscript
+#!/bin/sh
 # Copyright 1999-2014 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
+# Adapted by Marcel Saegebarth for slackbuilds.org for
+# http://slackbuilds.org/result/?search=profile-sync-daemon.
 
 description="Webbrowser profile syncing"
 extra_commands="resync"
 description_resync="Manually sync the disk profile with running tmpfs image"
 
-
-depend() {
-	need devfs localmount
-	after swapfiles # bug 398431#c8
-}
-
 start() {
-	ebegin "Starting Profile-Sync-Daemon"
-
+	echo "Starting Profile-Sync-Daemon"
 	/usr/bin/profile-sync-daemon sync
-	eend $?
 }
 
 stop() {
-	ebegin "Stopping Profile-Sync-Daemon"
-
+	echo "Stopping Profile-Sync-Daemon"
 	/usr/bin/profile-sync-daemon unsync
-	eend $?
 }
 
 status() {
@@ -31,7 +23,25 @@
 }
 
 resync() {
-	ebegin "Syncing browser profiles in tmpfs to physical disc"
+	echo "Syncing browser profiles in tmpfs to physical disc"
 	/usr/bin/profile-sync-daemon resync
-	eend $?
 }
+
+case "$1" in
+	start)
+		start
+		;;
+	stop)
+		stop
+		;;
+	resync)
+		resync
+		;;
+
+	status)
+                status
+		;;
+	*)
+		echo $"Usage: $0 {start|stop|resync|status}"
+		exit 1
+esac
