diff -Naur dwm-6.3-orig/config.def.h dwm-6.3/config.def.h
--- dwm-6.3-orig/config.def.h	2022-06-09 07:57:13.193221688 +0000
+++ dwm-6.3/config.def.h	2022-06-09 07:57:42.563068451 +0000
@@ -58,7 +58,7 @@
 /* commands */
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
 static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
-static const char *termcmd[]  = { "st", NULL };
+static const char *termcmd[]  = { "uxterm", NULL };
 
 static Key keys[] = {
 	/* modifier                     key        function        argument */
