diff -rupN Tk-804.028.org/TextList/reindex.pl Tk-804.028.new/TextList/reindex.pl
--- Tk-804.028.org/TextList/reindex.pl	2007-05-05 15:38:18.000000000 -0300
+++ Tk-804.028.new/TextList/reindex.pl	2009-11-11 18:46:31.000000000 -0200
@@ -1,4 +1,4 @@
-#!/bin/perl
+#!/usr/bin/perl
 
 use lib qw(/home1/gbartels/textlist);
 use Tk;
diff -rupN Tk-804.028.org/Tk/FBox.pm Tk-804.028.new/Tk/FBox.pm
--- Tk-804.028.org/Tk/FBox.pm	2007-05-05 15:37:56.000000000 -0300
+++ Tk-804.028.new/Tk/FBox.pm	2009-11-11 18:46:48.000000000 -0200
@@ -906,7 +906,7 @@ sub Done {
 	if ($w->cget('-multiple')) {
 	    $selectFilePath = [];
 	    for my $f (@{ $w->{'selectFile'} }) {
-		push @$selectFilePath, JoinFile($w->_get_select_Path, $f);
+		push @$selectFilePath, JoinFile($w->_get_select_path, $f);
 	    }
 	} else {
 	    $selectFilePath = JoinFile($w->_get_select_path,
diff -rupN Tk-804.028.org/pTk/Tcl-pTk Tk-804.028.new/pTk/Tcl-pTk
--- Tk-804.028.org/pTk/Tcl-pTk	2007-05-05 15:39:02.000000000 -0300
+++ Tk-804.028.new/pTk/Tcl-pTk	2009-11-11 18:48:58.000000000 -0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 use open IO => ':bytes'; # Avoid UTF-8 issues with some perl5.8.0 (RedHat)
 use Carp;
 my $verbose = 0;
diff -rupN Tk-804.028.org/pTk/mTk/generic/tk.h Tk-804.028.new/pTk/mTk/generic/tk.h
--- Tk-804.028.org/pTk/mTk/generic/tk.h	2007-05-05 15:41:02.000000000 -0300
+++ Tk-804.028.new/pTk/mTk/generic/tk.h	2009-11-11 18:49:56.000000000 -0200
@@ -677,17 +677,15 @@ typedef struct Tk_GeomMgr {
  *
  *---------------------------------------------------------------------------
  */
-#define VirtualEvent        (LASTEvent)
-#define ActivateNotify      (LASTEvent + 1)
-#define DeactivateNotify    (LASTEvent + 2)
-#define MouseWheelEvent     (LASTEvent + 3)
-#define TK_LASTEVENT        (LASTEvent + 4)
+#define VirtualEvent        (MappingNotify + 1)
+#define ActivateNotify      (MappingNotify + 2)
+#define DeactivateNotify    (MappingNotify + 3)
+#define MouseWheelEvent     (MappingNotify + 4)
+#define TK_LASTEVENT        (MappingNotify + 5)
 
 #define MouseWheelMask      (1L << 28)
-
 #define ActivateMask        (1L << 29)
 #define VirtualEventMask    (1L << 30)
-#define TK_LASTEVENT        (LASTEvent + 4)
 
 
 /*
diff -rupN Tk-804.028.org/pTk/mkVFunc Tk-804.028.new/pTk/mkVFunc
--- Tk-804.028.org/pTk/mkVFunc	2007-05-05 15:38:59.000000000 -0300
+++ Tk-804.028.new/pTk/mkVFunc	2009-11-11 18:49:14.000000000 -0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 use strict;
 
 
