diff -ru Tk-804.029.org//TextList/reindex.pl Tk-804.029/TextList/reindex.pl
--- Tk-804.029.org//TextList/reindex.pl	2007-02-10 05:54:45.000000000 -0300
+++ Tk-804.029/TextList/reindex.pl	2010-08-05 13:53:36.447371174 -0300
@@ -1,4 +1,4 @@
-#!/bin/perl
+#!/usr/bin/perl
 
 use lib qw(/home1/gbartels/textlist);
 use Tk;
diff -ru Tk-804.029.org//pTk/Tcl-pTk Tk-804.029/pTk/Tcl-pTk
--- Tk-804.029.org//pTk/Tcl-pTk	2007-02-10 05:55:23.000000000 -0300
+++ Tk-804.029/pTk/Tcl-pTk	2010-08-05 13:54:06.527369253 -0300
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 use open IO => ':bytes'; # Avoid UTF-8 issues with some perl5.8.0 (RedHat)
 use Carp;
 my $verbose = 0;
diff -ru Tk-804.029.org//pTk/mkVFunc Tk-804.029/pTk/mkVFunc
--- Tk-804.029.org//pTk/mkVFunc	2008-11-04 20:06:50.000000000 -0200
+++ Tk-804.029/pTk/mkVFunc	2010-08-05 13:54:20.074244152 -0300
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 use strict;
 
 
