--- lout-3.39-org/makefile	2010-04-21 15:52:47.000000000 -0700
+++ lout-3.39/makefile	2012-05-14 14:26:11.095459297 -0700
@@ -386,12 +386,6 @@
 	mkdir $(LOUTLIBDIR)/$(LIBLOCA)
 	chmod 755 $(LOUTLIBDIR)/$(LIBLOCA)
 	@echo ""
-	@echo "(c) Initializing run (should be silent, no errors expected)"
-	$(BINDIR)/lout -x -s $(LOUTLIBDIR)/$(LIBINCL)/init
-	@echo ""
-	@echo "(d) Changing mode of files just created by initializing run"
-	chmod 644 $(LOUTLIBDIR)/$(LIBDATA)/*
-	chmod 644 $(LOUTLIBDIR)/$(LIBHYPH)/*
 
 installman:
 	@echo ""
