--- lout-3.42/makefile	2020-12-27 16:45:43.000000000 -0800
+++ lout-3.42-slackware/makefile	2021-01-08 10:42:01.398114307 -0800
@@ -402,12 +402,6 @@
 	$(MKDIR) $(LOUTLIBDIR)/$(LIBLOCA)
 	chmod 755 $(LOUTLIBDIR)/$(LIBLOCA)
 	@echo ""
-	@echo "(c) Initializing run (should be silent, no errors expected)"
-	$(BINDIR)/lout -x -s $(LOUTLIBDIR)/$(LIBINCL)/init -I $(LOUTLIBDIR)/$(LIBINCL) -H $(LOUTLIBDIR)/$(LIBHYPH) -D $(LOUTLIBDIR)/$(LIBDATA)
-	@echo ""
-	@echo "(d) Changing mode of files just created by initializing run"
-	chmod 644 $(LOUTLIBDIR)/$(LIBDATA)/*
-	chmod 644 $(LOUTLIBDIR)/$(LIBHYPH)/*
 
 installman:
 	@echo ""
