--- qwtconfig.pri	2016-12-13 10:08:11.000000000 -0200
+++ qwtconfig.pri.new	2016-12-13 10:07:49.000000000 -0200
@@ -8,7 +8,7 @@
 VERSION      = $${VER_MAJ}.$${VER_MIN}.$${VER_PAT}
 
 unix {
-    INSTALLBASE    = /usr/local/qwt-$$VERSION
+    INSTALLBASE    = /usr
 }
 
 win32 {
@@ -16,7 +16,7 @@
 }
 
 target.path    = $$INSTALLBASE/lib
-headers.path   = $$INSTALLBASE/include
+headers.path   = $$INSTALLBASE/include/qwt5
 doc.path       = $$INSTALLBASE/doc
 
 ######################################################################
@@ -64,7 +64,7 @@
 ######################################################################
 
 DEBUG_SUFFIX        = 
-RELEASE_SUFFIX      = 
+RELEASE_SUFFIX      = 5 
 
 win32 {
     DEBUG_SUFFIX      = d
@@ -98,7 +98,7 @@
 # QwtSVGItem.
 ######################################################################
 
-#CONFIG     += QwtSVGItem
+CONFIG     += QwtSVGItem
 
 ######################################################################
 # You can use the MathML renderer of the Qt solutions package to 
