--- Makefile.orig	2009-10-31 19:59:54.000000000 +0300
+++ Makefile	2009-10-31 20:09:49.000000000 +0300
@@ -26,9 +26,9 @@
 # Look where your install program is
 #
 INSTALL = /usr/bin/install
-prefix = /usr
+prefix = ${DESTDIR}/usr
 bindir = $(prefix)/bin
-mandir = $(prefix)/share/$(MAN)
+mandir = $(prefix)/$(MAN)
 
 obj = db_dump.o dbview.o version.o
 
@@ -46,8 +46,8 @@
 install: all
 	$(INSTALL) -d -o root -g root -m 755 $(bindir)
 	$(INSTALL) -s -o root -g root -m 755 dbview $(bindir)
-	$(INSTALL) -d -o $(MAN) -g $(MAN) -m 755 $(mandir)/man1
-	$(INSTALL) -o $(MAN) -g $(MAN) -m 644 dbview.1 $(mandir)/man1
+	$(INSTALL) -d -o root -g root -m 755 $(mandir)/man1
+	$(INSTALL) -o root -g root -m 644 dbview.1 $(mandir)/man1
 
 clean:
 	rm -f dbview *.o *.log *~ *.orig Makefile.bak;
