--- abiword-2.8.1/asio.hpp	1969-12-31 16:00:00.000000000 -0800
+++ abiword-2.8.1/asio.hpp	2009-10-31 19:01:58.977141321 -0700
@@ -0,0 +1,10 @@
+#include <boost/asio.hpp>
+#include <boost/thread.hpp>
+
+namespace asio
+{
+   using namespace boost::asio;
+   using boost::system::error_code;
+   using boost::system::system_error;
+   using boost::thread;
+}
--- abiword-2.8.1/configure	2009-10-30 23:30:30.761312681 -0700
+++ abiword-2.8.1/configure	2009-10-31 19:18:16.581095503 -0700
@@ -22662,7 +22662,7 @@
 ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
 
-for ac_header in asio.hpp
+for ac_header in boost/asio.hpp
 do
 as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -22834,7 +22834,7 @@
 ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
 
-for ac_header in asio.hpp
+for ac_header in boost/asio.hpp
 do
 as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -23838,7 +23838,7 @@
 
 if test "$enable_collab_backend_tcp" == "yes" || \
    test "$enable_collab_backend_service" == "yes"; then
-	COLLAB_LIBS="$COLLAB_LIBS -lgcrypt -lpthread"
+	COLLAB_LIBS="$COLLAB_LIBS -lgcrypt -lpthread -lboost_system-mt -lboost_thread-mt"
 fi
 
 test "$enable_collab" == "auto" && PLUGINS="$PLUGINS collab"
