diff -Nur abiword-2.8.5.orig//asio.hpp abiword-2.8.5/asio.hpp
--- abiword-2.8.5.orig//asio.hpp	1969-12-31 18:00:00.000000000 -0600
+++ abiword-2.8.5/asio.hpp	2010-06-04 02:13:57.279465007 -0500
@@ -0,0 +1,10 @@
+#include <boost/asio.hpp>
+#include <boost/thread.hpp>
+
+namespace asio
+{
+   using namespace boost::asio;
+   using boost::system::error_code;
+   using boost::system::system_error;
+   using boost::thread;
+}
diff -Nur abiword-2.8.5.orig//configure abiword-2.8.5/configure
--- abiword-2.8.5.orig//configure	2010-05-30 14:34:39.000000000 -0500
+++ abiword-2.8.5/configure	2010-06-04 02:13:39.961560728 -0500
@@ -19815,7 +19815,7 @@
 ac_link='$CXX -o conftest$ac_exeext $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
-		for ac_header in asio.hpp
+		for ac_header in boost/asio.hpp
 do :
   ac_fn_cxx_check_header_mongrel "$LINENO" "asio.hpp" "ac_cv_header_asio_hpp" "$ac_includes_default"
 if test "x$ac_cv_header_asio_hpp" = x""yes; then :
@@ -19847,7 +19847,7 @@
 ac_link='$CXX -o conftest$ac_exeext $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
-	for ac_header in asio.hpp
+	for ac_header in boost/asio.hpp
 do :
   ac_fn_cxx_check_header_mongrel "$LINENO" "asio.hpp" "ac_cv_header_asio_hpp" "$ac_includes_default"
 if test "x$ac_cv_header_asio_hpp" = x""yes; then :
@@ -20353,7 +20353,7 @@
 
 if test "$enable_collab_backend_tcp" = "yes" || \
    test "$enable_collab_backend_service" = "yes"; then
-	COLLAB_LIBS="$COLLAB_LIBS -lgcrypt -lpthread"
+	COLLAB_LIBS="$COLLAB_LIBS -lgcrypt -lpthread -lboost_system -lboost_thread"
 fi
 
 test "$enable_collab" = "auto" && PLUGINS="$PLUGINS collab"
