--- abiword-2.8.1/src/af/util/xp/ut_go_file.cpp	2009-10-30 23:30:24.005096842 -0700
+++ abiword-2.8.1/src/af/util/xp/ut_go_file.cpp	2009-10-31 09:32:24.028095222 -0700
@@ -1665,12 +1665,17 @@
 #else
 	GError *err = NULL;
 #ifdef HAVE_GTK214
-	gtk_show_uri (NULL, url, GDK_CURRENT_TIME, &err);
-	return err;
-#elif defined(WITH_GNOMEVFS)
-	gnome_vfs_url_show (url);
-	return err;
-#else
+	if(gtk_show_uri (NULL, url, GDK_CURRENT_TIME, &err))
+	{
+		return err;
+	}
+#endif
+#if defined(WITH_GNOMEVFS)
+	if(gnome_vfs_url_show (url))
+	{
+		return err;
+	}
+#endif
 	gchar *browser = NULL;
 	gchar *clean_url = NULL;
 
@@ -1679,6 +1684,7 @@
 
 	if (browser == NULL) {
 		static char const * const browsers[] = {
+			"xdg-open",		/* freedesktop.org */
 			"sensible-browser",	/* debian */
 			"epiphany",		/* primary gnome */
 			"galeon",		/* secondary gnome */
@@ -1688,6 +1694,7 @@
 			"mozilla",
 			"netscape",
 			"konqueror",
+			"dillo",
 			"xterm -e w3m",
 			"xterm -e lynx",
 			"xterm -e links"
@@ -1735,7 +1742,6 @@
 	g_free (clean_url);
 	return err;
 #endif
-#endif
 }
 
 /**
