diff -Naur read-edid-3.0.2/get-edid/classic.c read-edid-3.0.2.patched/get-edid/classic.c
--- read-edid-3.0.2/get-edid/classic.c	2015-04-28 21:08:34.000000000 -0400
+++ read-edid-3.0.2.patched/get-edid/classic.c	2021-08-24 03:34:15.182335445 -0400
@@ -26,7 +26,7 @@
 #define dosmemput(buffer,length,offset) memcpy(offset,buffer,length)
 
 #define display(...) if (quiet == 0) { fprintf(stderr, __VA_ARGS__); }
-int quiet;
+extern int quiet;
 
 real_ptr far_ptr_to_real_ptr( uint32 farptr )
 {
diff -Naur read-edid-3.0.2/get-edid/i2c.c read-edid-3.0.2.patched/get-edid/i2c.c
--- read-edid-3.0.2/get-edid/i2c.c	2014-11-21 06:52:13.000000000 -0500
+++ read-edid-3.0.2.patched/get-edid/i2c.c	2021-08-24 03:34:21.390334880 -0400
@@ -15,7 +15,7 @@
 
 //Ideas (but not too much actual code) taken from i2c-tools. Thanks guys.
 
-int quiet;
+extern int quiet;
 
 #define display(...) if (quiet == 0) { fprintf(stderr, __VA_ARGS__); }
 
