diff -Naur dis-0.5/dis dis-0.5.patched/dis
--- dis-0.5/dis	2017-09-07 16:42:33.595299125 -0400
+++ dis-0.5.patched/dis	2017-09-07 16:42:51.714298161 -0400
@@ -539,7 +539,8 @@
 sub arg {
     my ($opts, $args, $file) = @_;
     return if grep $file eq $_, $opts->{arg};
-    open my $fh, $file or die "ERROR: Cannot open $file: $!\n";
+	 my $fh;
+    open $fh, $file or open $fh, "/usr/share/dis/$file" or die "ERROR: Cannot open $file: $!\n";
     my %args = map { /(\w+)/; my $a = $1; $a => [/=s/, /@/] } @$args;
     while (<$fh>) {
         s/;.*//;
