--- sources/pyside2-tools/CMakeLists.txt	2020-11-11 13:51:30.000000000 +0100
+++ ../SBo/pyside-setup-opensource-src-5.15.2/sources/pyside2-tools/CMakeLists.txt	2020-12-19 02:04:45.416344099 +0100
@@ -47,27 +47,27 @@
 endif()
 
 set(TOOLS_PATH "${_qt5Core_install_prefix}/bin")
-set(UIC_PATH "${TOOLS_PATH}/uic${EXE_EXT}")
-set(RCC_PATH "${TOOLS_PATH}/rcc${EXE_EXT}")
+# set(UIC_PATH "${TOOLS_PATH}/uic${EXE_EXT}")
+# set(RCC_PATH "${TOOLS_PATH}/rcc${EXE_EXT}")
 if (APPLE)
     set(DESIGNER_PATH "${TOOLS_PATH}/Designer.app")
 else()
     set(DESIGNER_PATH "${TOOLS_PATH}/designer${EXE_EXT}")
 endif()
 
-install(FILES "${UIC_PATH}"
-        DESTINATION bin
-        PERMISSIONS
-        OWNER_EXECUTE OWNER_WRITE OWNER_READ
-        GROUP_EXECUTE GROUP_READ
-        WORLD_EXECUTE WORLD_READ)
+# install(FILES "${UIC_PATH}"
+#       DESTINATION bin
+#       PERMISSIONS
+#       OWNER_EXECUTE OWNER_WRITE OWNER_READ
+#	GROUP_EXECUTE GROUP_READ
+#       WORLD_EXECUTE WORLD_READ)
 
-install(FILES "${RCC_PATH}"
-        DESTINATION bin
-        PERMISSIONS
-        OWNER_EXECUTE OWNER_WRITE OWNER_READ
-        GROUP_EXECUTE GROUP_READ
-        WORLD_EXECUTE WORLD_READ)
+#install(FILES "${RCC_PATH}"
+#        DESTINATION bin
+#        PERMISSIONS
+#        OWNER_EXECUTE OWNER_WRITE OWNER_READ
+#        GROUP_EXECUTE GROUP_READ
+#        WORLD_EXECUTE WORLD_READ)
 
 if (EXISTS ${DESIGNER_PATH})
     if (APPLE)
