diff -Naur chroma-1.15.orig/sdlshadowdisplay.c chroma-1.15/sdlshadowdisplay.c
--- chroma-1.15.orig/sdlshadowdisplay.c	2013-04-20 08:07:00.000000000 +0200
+++ chroma-1.15/sdlshadowdisplay.c	2017-08-02 12:07:56.327010753 +0200
@@ -88,7 +88,7 @@
 void displayshadowed_level(struct level* plevel);
 void displayshadowed_movers(struct level* plevel, int redraw);
 
-inline void displayshadowed_piece(struct level* plevel, int p, int x, int y, int d)
+void displayshadowed_piece(struct level* plevel, int p, int x, int y, int d)
 {
     SDL_Surface *pimage;
 
@@ -255,7 +255,7 @@
     }
 }
 
-inline void displayshadowed_pieceshadow(struct level* plevel, int p, int x, int y, int d)
+void displayshadowed_pieceshadow(struct level* plevel, int p, int x, int y, int d)
 {
     SDL_Surface *pimage;
 
@@ -310,7 +310,7 @@
     SDL_BlitSurface(pimage, &srect, screen_surface, &drect);
 }
 
-inline void displayshadowed_piecebase(struct level* plevel, int x, int y)
+void displayshadowed_piecebase(struct level* plevel, int x, int y)
 {
     int p;
     SDL_Surface *pimage;
