diff -Nur libnfc-1.8.0.orig/utils/CMakeLists.txt libnfc-1.8.0/utils/CMakeLists.txt
--- libnfc-1.8.0.orig/utils/CMakeLists.txt	2020-05-22 05:20:17.000000000 -0500
+++ libnfc-1.8.0/utils/CMakeLists.txt	2022-02-21 19:14:09.544499311 -0600
@@ -61,5 +61,5 @@
 IF(NOT WIN32)
   # Manuals for the examples
   FILE(GLOB manuals "${CMAKE_CURRENT_SOURCE_DIR}/*.1")
-  INSTALL(FILES ${manuals} DESTINATION ${SHARE_INSTALL_PREFIX}/man/man1 COMPONENT manuals)
+  INSTALL(FILES ${manuals} DESTINATION ${CMAKE_INSTALL_PREFIX}/man/man1 COMPONENT manuals)
 ENDIF(NOT WIN32)
