diff -Naur a/configure b/configure
--- a/configure	2018-01-02 20:58:13.000000000 +0000
+++ b/configure	2018-01-08 18:31:55.480995247 +0000
@@ -8072,7 +8072,7 @@
 done
 
     LIBEXT="dylib"
-    DLLFLAGS="$DLLFLAGS -fPIC"
+    DLLFLAGS="$DLLFLAGS "
     LDRPATH_INSTALL="-Wl,-rpath,@loader_path/\`\$(MAKEDEP) -R \${bindir} \${libdir}\`"
     LDRPATH_LOCAL="-Wl,-rpath,@loader_path/\$(top_builddir)/libs/wine"
         COREFOUNDATION_LIBS="-framework CoreFoundation"
@@ -8304,14 +8304,14 @@
     ;;
 
   linux-android*)
-    DLLFLAGS="$DLLFLAGS -fPIC"
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports -fPIC -Wl,--export-dynamic" >&5
-$as_echo_n "checking whether the compiler supports -fPIC -Wl,--export-dynamic... " >&6; }
+    DLLFLAGS="$DLLFLAGS "
+    { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports  -Wl,--export-dynamic" >&5
+$as_echo_n "checking whether the compiler supports  -Wl,--export-dynamic... " >&6; }
 if ${ac_cv_cflags__fPIC__Wl___export_dynamic+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_wine_try_cflags_saved=$CFLAGS
-CFLAGS="$CFLAGS -fPIC -Wl,--export-dynamic"
+CFLAGS="$CFLAGS  -Wl,--export-dynamic"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(int argc, char **argv) { return 0; }
@@ -8330,14 +8330,14 @@
 if test "x$ac_cv_cflags__fPIC__Wl___export_dynamic" = xyes; then :
   LDEXECFLAGS="-Wl,--export-dynamic"
 fi
-    as_ac_var=`$as_echo "ac_cv_cflags_-fPIC -Wl,--rpath,\\$ORIGIN/../lib" | $as_tr_sh`
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports -fPIC -Wl,--rpath,\$ORIGIN/../lib" >&5
-$as_echo_n "checking whether the compiler supports -fPIC -Wl,--rpath,\$ORIGIN/../lib... " >&6; }
+    as_ac_var=`$as_echo "ac_cv_cflags_ -Wl,--rpath,\\$ORIGIN/../lib" | $as_tr_sh`
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports  -Wl,--rpath,\$ORIGIN/../lib" >&5
+$as_echo_n "checking whether the compiler supports  -Wl,--rpath,\$ORIGIN/../lib... " >&6; }
 if eval \${$as_ac_var+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_wine_try_cflags_saved=$CFLAGS
-CFLAGS="$CFLAGS -fPIC -Wl,--rpath,\$ORIGIN/../lib"
+CFLAGS="$CFLAGS  -Wl,--rpath,\$ORIGIN/../lib"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(int argc, char **argv) { return 0; }
@@ -8358,14 +8358,14 @@
   LDRPATH_INSTALL="-Wl,--rpath,\\\$\$ORIGIN/\`\$(MAKEDEP) -R \${bindir} \${libdir}\`"
                      LDRPATH_LOCAL="-Wl,--rpath,\\\$\$ORIGIN/\$(top_builddir)/libs/wine"
 else
-  as_ac_var=`$as_echo "ac_cv_cflags_-fPIC -Wl,-R,\\$ORIGIN/../lib" | $as_tr_sh`
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports -fPIC -Wl,-R,\$ORIGIN/../lib" >&5
-$as_echo_n "checking whether the compiler supports -fPIC -Wl,-R,\$ORIGIN/../lib... " >&6; }
+  as_ac_var=`$as_echo "ac_cv_cflags_ -Wl,-R,\\$ORIGIN/../lib" | $as_tr_sh`
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports  -Wl,-R,\$ORIGIN/../lib" >&5
+$as_echo_n "checking whether the compiler supports  -Wl,-R,\$ORIGIN/../lib... " >&6; }
 if eval \${$as_ac_var+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_wine_try_cflags_saved=$CFLAGS
-CFLAGS="$CFLAGS -fPIC -Wl,-R,\$ORIGIN/../lib"
+CFLAGS="$CFLAGS  -Wl,-R,\$ORIGIN/../lib"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(int argc, char **argv) { return 0; }
@@ -8448,7 +8448,7 @@
     ;;
 
   *)
-    DLLFLAGS="$DLLFLAGS -fPIC"
+    DLLFLAGS="$DLLFLAGS "
     LDEXECFLAGS=""
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether we can build a GNU style ELF dll" >&5
 $as_echo_n "checking whether we can build a GNU style ELF dll... " >&6; }
@@ -8456,7 +8456,7 @@
   $as_echo_n "(cached) " >&6
 else
   ac_wine_try_cflags_saved=$CFLAGS
-CFLAGS="$CFLAGS -fPIC -shared -Wl,-Bsymbolic"
+CFLAGS="$CFLAGS  -shared -Wl,-Bsymbolic"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 void myfunc() {}
@@ -8475,13 +8475,13 @@
     if test "$ac_cv_c_dll_gnuelf" = "yes"
     then
       LIBWINE_LDFLAGS="-shared"
-      { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports -fPIC -shared -Wl,-soname,confest.so.1" >&5
-$as_echo_n "checking whether the compiler supports -fPIC -shared -Wl,-soname,confest.so.1... " >&6; }
+      { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports  -shared -Wl,-soname,confest.so.1" >&5
+$as_echo_n "checking whether the compiler supports  -shared -Wl,-soname,confest.so.1... " >&6; }
 if ${ac_cv_cflags__fPIC__shared__Wl__soname_confest_so_1+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_wine_try_cflags_saved=$CFLAGS
-CFLAGS="$CFLAGS -fPIC -shared -Wl,-soname,confest.so.1"
+CFLAGS="$CFLAGS  -shared -Wl,-soname,confest.so.1"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(int argc, char **argv) { return 0; }
@@ -8500,13 +8500,13 @@
 if test "x$ac_cv_cflags__fPIC__shared__Wl__soname_confest_so_1" = xyes; then :
   LIBWINE_LDFLAGS="-shared -Wl,-soname,libwine.so.$libwine_soversion"
 else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports -fPIC -shared -Wl,-h,confest.so.1" >&5
-$as_echo_n "checking whether the compiler supports -fPIC -shared -Wl,-h,confest.so.1... " >&6; }
+  { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports  -shared -Wl,-h,confest.so.1" >&5
+$as_echo_n "checking whether the compiler supports  -shared -Wl,-h,confest.so.1... " >&6; }
 if ${ac_cv_cflags__fPIC__shared__Wl__h_confest_so_1+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_wine_try_cflags_saved=$CFLAGS
-CFLAGS="$CFLAGS -fPIC -shared -Wl,-h,confest.so.1"
+CFLAGS="$CFLAGS  -shared -Wl,-h,confest.so.1"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(int argc, char **argv) { return 0; }
@@ -8528,13 +8528,13 @@
 fi
 
       echo '{ global: *; };' >conftest.map
-      { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports -fPIC -shared -Wl,--version-script=conftest.map" >&5
-$as_echo_n "checking whether the compiler supports -fPIC -shared -Wl,--version-script=conftest.map... " >&6; }
+      { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports  -shared -Wl,--version-script=conftest.map" >&5
+$as_echo_n "checking whether the compiler supports  -shared -Wl,--version-script=conftest.map... " >&6; }
 if ${ac_cv_cflags__fPIC__shared__Wl___version_script_conftest_map+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_wine_try_cflags_saved=$CFLAGS
-CFLAGS="$CFLAGS -fPIC -shared -Wl,--version-script=conftest.map"
+CFLAGS="$CFLAGS  -shared -Wl,--version-script=conftest.map"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(int argc, char **argv) { return 0; }
@@ -8555,13 +8555,13 @@
 fi
       rm -f conftest.map
 
-      { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports -fPIC -Wl,--export-dynamic" >&5
-$as_echo_n "checking whether the compiler supports -fPIC -Wl,--export-dynamic... " >&6; }
+      { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports  -Wl,--export-dynamic" >&5
+$as_echo_n "checking whether the compiler supports  -Wl,--export-dynamic... " >&6; }
 if ${ac_cv_cflags__fPIC__Wl___export_dynamic+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_wine_try_cflags_saved=$CFLAGS
-CFLAGS="$CFLAGS -fPIC -Wl,--export-dynamic"
+CFLAGS="$CFLAGS  -Wl,--export-dynamic"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(int argc, char **argv) { return 0; }
@@ -8581,14 +8581,14 @@
   LDEXECFLAGS="-Wl,--export-dynamic"
 fi
 
-      as_ac_var=`$as_echo "ac_cv_cflags_-fPIC -Wl,--rpath,\\$ORIGIN/../lib" | $as_tr_sh`
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports -fPIC -Wl,--rpath,\$ORIGIN/../lib" >&5
-$as_echo_n "checking whether the compiler supports -fPIC -Wl,--rpath,\$ORIGIN/../lib... " >&6; }
+      as_ac_var=`$as_echo "ac_cv_cflags_ -Wl,--rpath,\\$ORIGIN/../lib" | $as_tr_sh`
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports  -Wl,--rpath,\$ORIGIN/../lib" >&5
+$as_echo_n "checking whether the compiler supports  -Wl,--rpath,\$ORIGIN/../lib... " >&6; }
 if eval \${$as_ac_var+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_wine_try_cflags_saved=$CFLAGS
-CFLAGS="$CFLAGS -fPIC -Wl,--rpath,\$ORIGIN/../lib"
+CFLAGS="$CFLAGS  -Wl,--rpath,\$ORIGIN/../lib"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(int argc, char **argv) { return 0; }
@@ -8609,14 +8609,14 @@
   LDRPATH_INSTALL="-Wl,--rpath,\\\$\$ORIGIN/\`\$(MAKEDEP) -R \${bindir} \${libdir}\`"
                        LDRPATH_LOCAL="-Wl,--rpath,\\\$\$ORIGIN/\$(top_builddir)/libs/wine"
 else
-  as_ac_var=`$as_echo "ac_cv_cflags_-fPIC -Wl,-R,\\$ORIGIN/../lib" | $as_tr_sh`
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports -fPIC -Wl,-R,\$ORIGIN/../lib" >&5
-$as_echo_n "checking whether the compiler supports -fPIC -Wl,-R,\$ORIGIN/../lib... " >&6; }
+  as_ac_var=`$as_echo "ac_cv_cflags_ -Wl,-R,\\$ORIGIN/../lib" | $as_tr_sh`
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports  -Wl,-R,\$ORIGIN/../lib" >&5
+$as_echo_n "checking whether the compiler supports  -Wl,-R,\$ORIGIN/../lib... " >&6; }
 if eval \${$as_ac_var+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_wine_try_cflags_saved=$CFLAGS
-CFLAGS="$CFLAGS -fPIC -Wl,-R,\$ORIGIN/../lib"
+CFLAGS="$CFLAGS  -Wl,-R,\$ORIGIN/../lib"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(int argc, char **argv) { return 0; }
@@ -8804,7 +8804,7 @@
   $as_echo_n "(cached) " >&6
 else
   ac_wine_try_cflags_saved=$CFLAGS
-CFLAGS="$CFLAGS -fPIC -Wl,-G,-h,conftest.so.1.0,-B,symbolic"
+CFLAGS="$CFLAGS  -Wl,-G,-h,conftest.so.1.0,-B,symbolic"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 void myfunc() {}
