diff -Naur rbenv-1.0.0.orig/libexec/rbenv rbenv-1.0.0/libexec/rbenv
--- rbenv-1.0.0.orig/libexec/rbenv	2015-12-25 07:12:47.000000000 +1300
+++ rbenv-1.0.0/libexec/rbenv	2016-04-02 14:50:55.733994971 +1300
@@ -20,7 +20,7 @@
   exit 1
 }
 
-if enable -f "${BASH_SOURCE%/*}"/../libexec/rbenv-realpath.dylib realpath 2>/dev/null; then
+if enable -f /usr/lib/rbenv/libexec/rbenv-realpath.dylib realpath 2>/dev/null; then
   abs_dirname() {
     local path="$(realpath "$1")"
     echo "${path%/*}"
