Author: Andrea Veri <and@debian.org>
Description: Disable GPC at build time.

diff -urNad debian.source~/Makefile.am debian.source/Makefile.am
--- debian.source~/Makefile.am	2007-10-10 16:13:11.000000000 +0200
+++ debian.source/Makefile.am	2007-10-10 16:26:51.000000000 +0200
@@ -1,4 +1,7 @@
+SUBDIRS = src font_freetype font_win32_tt include examples
+if ENABLE_GPC
 SUBDIRS = gpc src font_freetype font_win32_tt include examples
+endif
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libagg.pc
