Author: Andrea Veri <and@debian.org>
Description: Patch to prevent a FTBFS on kfreebsd-*

Index: agg-2.5+dfsg1/src/Makefile
===================================================================
--- agg-2.5+dfsg1.orig/src/Makefile	2007-10-11 00:06:16.000000000 +0200
+++ agg-2.5+dfsg1/src/Makefile	2012-05-01 21:30:08.492401874 +0200
@@ -1,4 +1,9 @@
-include ../Makefile.in.$(shell uname)
+PLATFORM=$(shell uname)
+ifeq (GNU/kFreeBSD,$(PLATFORM))
+	PLATFORM=Linux
+endif
+
+include ../Makefile.in.$(PLATFORM)
 
 CXXFLAGS= $(AGGCXXFLAGS) -I../include -L./
 
