Author: Barry deFreese <bdefreese@debian.org>
Description: Patch to prevent a FTBFS on GNU/Hurd.

Index: agg-2.5+dfsg1/src/Makefile
===================================================================
--- agg-2.5+dfsg1.orig/src/Makefile	2012-06-04 23:05:40.000000000 +0000
+++ agg-2.5+dfsg1/src/Makefile	2012-06-04 23:08:30.000000000 +0000
@@ -3,6 +3,10 @@
 	PLATFORM=Linux
 endif
 
+ifeq (GNU,$(PLATFORM))
+	PLATFORM=Linux
+endif
+
 include ../Makefile.in.$(PLATFORM)
 
 CXXFLAGS= $(AGGCXXFLAGS) -I../include -L./
