--- Makefile	2012-03-26 19:27:20.000000000 -0300
+++ Makefile.new	2014-06-20 22:54:21.000000000 -0300
@@ -1,14 +1,14 @@
 CC	?= gcc
 CFLAGS	?= -g -pipe
 
-PREFIX	?= /usr/local
+PREFIX	?= /usr
 
 INSTALL_DIR	:= $(PREFIX)/bin
 PIXMAP_DIR	:= $(PREFIX)/share/pixmaps
-MANPAGE_DIR	:= $(PREFIX)/share/man/man1
+MANPAGE_DIR	:= $(PREFIX)/man/man1
 
-CFLAGS 	+= $(shell pkg-config audacious --cflags) $(shell pkg-config dbus-1 --cflags) -DPIXMAP_DIR="\"$(PIXMAP_DIR)\""
-LIBS 	:= $(shell pkg-config audacious --libs) $(shell pkg-config audclient --libs)  $(shell pkg-config dbus-1 --libs)
+CFLAGS 	+= $(shell pkg-config audclient --cflags) $(shell pkg-config dbus-glib-1 --cflags) -DPIXMAP_DIR="\"$(PIXMAP_DIR)\""
+LIBS 	:= $(shell pkg-config audclient --libs)  $(shell pkg-config dbus-glib-1 --libs) -lX11
 
 CFLAGS  += $(shell pkg-config gtk+-2.0 --cflags)
 LIBS    += $(shell pkg-config gtk+-2.0 --libs)
@@ -17,7 +17,7 @@
 HEADERS = dock-master.xpm
 
 wmauda:	$(OBJS) $(HEADERS)
-	$(CC) -o wmauda $(OBJS) $(CFLAGS) $(LIBS)
+	$(CC) $(LDFLAGS) -o wmauda $(OBJS) $(CFLAGS) $(LIBS)
 
 all:	wmauda
 
--- wmauda.c	2012-03-26 19:23:14.000000000 -0300
+++ wmauda.c.new	2014-06-20 23:03:15.000000000 -0300
@@ -161,8 +161,9 @@
 
 void action_eject(void)
 {
-	audacious_remote_playlist_clear(dbus_proxy);
-	audacious_remote_stop(dbus_proxy);
+/*	audacious_remote_playlist_clear(dbus_proxy); */
+/*	audacious_remote_stop(dbus_proxy); */
+       audacious_remote_eject(dbus_proxy);
 }
 
 void action_prev(void)
