From b9b83bef7abd0b234b9641d523be9012ea1cf718 Mon Sep 17 00:00:00 2001
From: Andrey Rijov <rdron@users.sourceforge.net>
Date: Sun, 21 Aug 2011 12:53:11 +0400
Subject: [PATCH 2/7] Fixed bug #3395228

---
 README |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/README b/README
index ad70462..f8c08e2 100644
--- a/README
+++ b/README
@@ -37,7 +37,7 @@ INSTALLATION (For Unix):
         Sample `cmake -DCMAKE_INSTALL_PREFIX=/usr/local/' this install AQEMU binary
         in /usr/local/bin and install data in /usr/local/share/aqemu
 
-    -DWITHOUT_EMBEDDED_DISPLAY = on | off (Default is: on)
+    -DWITHOUT_EMBEDDED_DISPLAY = on | off (Default is: off)
         For building without support Embedded VNC Display.
 
     -DDEBUG = on | off (Default is: off)
-- 
1.7.8.3

