From 9446cc47b96cbbe52976e8893fc50ad730fbaa9e Mon Sep 17 00:00:00 2001
From: Andrey Rijov <rdron@users.sourceforge.net>
Date: Sun, 13 Nov 2011 11:36:13 +0400
Subject: [PATCH 6/7] Fixed emulator monitor. (BUG ID: 3432871)

---
 Monitor_Window.cpp |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/Monitor_Window.cpp b/Monitor_Window.cpp
index ace256c..a4b30ea 100644
--- a/Monitor_Window.cpp
+++ b/Monitor_Window.cpp
@@ -36,6 +36,7 @@ Monitor_Window::Monitor_Window( QWidget *parent )
 
 void Monitor_Window::Add_QEMU_Out( const QString &new_text )
 {
+	ui.Edit_Monitor_Out->moveCursor( QTextCursor::End );
 	ui.Edit_Monitor_Out->insertPlainText( new_text );
 }
 
-- 
1.7.8.3

