diff -Naur zarafa-7.1.0.orig/inetmapi/outputStreamMAPIAdapter.h zarafa-7.1.0/inetmapi/outputStreamMAPIAdapter.h
--- zarafa-7.1.0.orig/inetmapi/outputStreamMAPIAdapter.h	2012-08-08 17:14:57.000000000 +0200
+++ zarafa-7.1.0/inetmapi/outputStreamMAPIAdapter.h	2012-09-13 19:37:26.072059371 +0200
@@ -51,6 +51,7 @@
 #define OUTPUT_STREAM_MAPI_ADAPTER_H
 
 #include "mapidefs.h"
+#include <unistd.h>
 #include <vmime/utility/stream.hpp>
 
 class outputStreamMAPIAdapter : public vmime::utility::outputStream {
