--- zarafa-6.30.5/ECtools/ECTestTools/ICalTests/ical2mapi.cpp	2009-11-12 20:55:01.000000000 -0200
+++ zarafa-6.30.5_patched/ECtools/ECTestTools/ICalTests/ical2mapi.cpp	2009-12-12 12:28:15.000000000 -0200
@@ -178,9 +178,9 @@
 	if (readfile(argv[4], &icalstring))
 		goto exit;
 
-	CreateICalToMapi(lpInbox, lpAdrBook, &lpICalToMapi);
+	CreateICalToMapi(lpInbox, lpAdrBook, 0, &lpICalToMapi);
 
-	hr = lpICalToMapi->ParseICal(icalstring, "UTF-8", "UTC", 0); // assume ical file is in utf-8
+	hr = lpICalToMapi->ParseICal(icalstring, "UTF-8", "UTC", NULL, 0); // assume ical file is in utf-8
 	if (FAILED(hr)) {
 		cerr << "unable to parse" << endl;
 		goto exit;
