--- zarafa-6.40.2/ECtools/ECTestTools/RecurrenceTest/rectest.cpp	2010-08-30 08:01:44.000000000 -0300
+++ zarafa-6.40.2_patched/ECtools/ECTestTools/RecurrenceTest/rectest.cpp	2010-09-04 13:27:42.000000000 -0300
@@ -79,7 +79,7 @@
     
     RecurrenceState r;
     
-    hr = r.ParseBlob((char *)strBin.c_str(), strBin.size());
+    hr = r.ParseBlob((char *)strBin.c_str(), strBin.size(), 0);
     
     if(hr == hrSuccess) {
         cerr << "Recurrence OK" << std::endl;
--- zarafa-6.40.2/ECtools/ECTestTools/ICalTests/mapi2ical.cpp	2010-08-30 08:01:44.000000000 -0300
+++ zarafa-6.40.2_patched/ECtools/ECTestTools/ICalTests/mapi2ical.cpp	2010-09-04 13:55:38.000000000 -0300
@@ -170,7 +170,7 @@
 			goto next;
 
 
-		hr = lpMapiToICal->AddMessage(lpMessage, 0);
+		hr = lpMapiToICal->AddMessage(lpMessage, std::string(), 0);
 		if (hr != hrSuccess) {
 			cerr << "-- broken message!" << stringify(hr, 1) << endl;
 			goto next;
--- zarafa-6.40.2/ECtools/ECTestTools/ICalTests/Makefile.in	2010-08-30 04:31:48.000000000 -0300
+++ zarafa-6.40.2_patched/ECtools/ECTestTools/ICalTests/Makefile.in	2010-09-04 15:09:39.000000000 -0300
@@ -272,6 +272,7 @@
 	${top_builddir}/common/libcommon_mapi.la \
 	${top_builddir}/common/libcommon_util.la \
 	${top_builddir}/libicalmapi/libicalmapi.la \
+	${top_builddir}/libfreebusy/libfreebusy.la \
 	$(PROG_LIBS) $(ICAL_LIBS)
 
 mapi2ical_SOURCES = mapi2ical.cpp
