diff -ruNp yadifa-3.0.2-11672/CMakeLists.txt yadifa-3.0.2-11672.new/CMakeLists.txt
--- yadifa-3.0.2-11672/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/CMakeLists.txt	2025-03-15 13:43:17.310828599 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 include(${CMAKE_SOURCE_DIR}/cmake/CheckCHeaders.cmake)
 include(${CMAKE_SOURCE_DIR}/cmake/CheckCompilerFlags.cmake)
diff -ruNp yadifa-3.0.2-11672/bin/yadifa/CMakeLists.txt yadifa-3.0.2-11672.new/bin/yadifa/CMakeLists.txt
--- yadifa-3.0.2-11672/bin/yadifa/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/bin/yadifa/CMakeLists.txt	2025-03-15 13:43:17.310828599 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 project(yadifa LANGUAGES C)
 
diff -ruNp yadifa-3.0.2-11672/doc/CMakeLists.txt yadifa-3.0.2-11672.new/doc/CMakeLists.txt
--- yadifa-3.0.2-11672/doc/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/doc/CMakeLists.txt	2025-03-15 13:43:17.310828599 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 ################################################################################
 #
diff -ruNp yadifa-3.0.2-11672/etc/CMakeLists.txt yadifa-3.0.2-11672.new/etc/CMakeLists.txt
--- yadifa-3.0.2-11672/etc/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/etc/CMakeLists.txt	2025-03-15 13:43:17.310828599 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 ################################################################################
 #
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/CMakeLists.txt	2025-03-15 13:43:17.310828599 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 include(${CMAKE_SOURCE_DIR}/cmake/EURidToolbox.cmake)
 
 project(dnscore LANGUAGES C)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/acl-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/acl-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/acl-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/acl-test/CMakeLists.txt	2025-03-15 13:43:17.310828599 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/alarm-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/alarm-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/alarm-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/alarm-test/CMakeLists.txt	2025-03-15 13:43:17.310828599 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/async-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/async-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/async-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/async-test/CMakeLists.txt	2025-03-15 13:43:17.310828599 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/basexx-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/basexx-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/basexx-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/basexx-test/CMakeLists.txt	2025-03-15 13:43:17.310828599 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/basic_priority_queue-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/basic_priority_queue-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/basic_priority_queue-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/basic_priority_queue-test/CMakeLists.txt	2025-03-15 13:43:17.311828650 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/buffer_input_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/buffer_input_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/buffer_input_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/buffer_input_stream-test/CMakeLists.txt	2025-03-15 13:43:17.311828650 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/buffer_output_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/buffer_output_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/buffer_output_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/buffer_output_stream-test/CMakeLists.txt	2025-03-15 13:43:17.311828650 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/buffered_file-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/buffered_file-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/buffered_file-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/buffered_file-test/CMakeLists.txt	2025-03-15 13:43:17.311828650 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/bytearray_input_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/bytearray_input_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/bytearray_input_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/bytearray_input_stream-test/CMakeLists.txt	2025-03-15 13:43:17.311828650 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/bytearray_output_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/bytearray_output_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/bytearray_output_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/bytearray_output_stream-test/CMakeLists.txt	2025-03-15 13:43:17.311828650 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/bytezarray_output_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/bytezarray_output_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/bytezarray_output_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/bytezarray_output_stream-test/CMakeLists.txt	2025-03-15 13:43:17.311828650 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/cache-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/cache-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/cache-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/cache-test/CMakeLists.txt	2025-03-15 13:43:17.311828650 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/checked_output_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/checked_output_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/checked_output_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/checked_output_stream-test/CMakeLists.txt	2025-03-15 13:43:17.311828650 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/chroot-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/chroot-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/chroot-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/chroot-test/CMakeLists.txt	2025-03-15 13:43:17.311828650 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/concat_input_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/concat_input_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/concat_input_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/concat_input_stream-test/CMakeLists.txt	2025-03-15 13:43:17.311828650 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/config-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/config-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/config-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/config-test/CMakeLists.txt	2025-03-15 13:43:17.311828650 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/config_key-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/config_key-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/config_key-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/config_key-test/CMakeLists.txt	2025-03-15 13:43:17.312828701 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/config_logger-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/config_logger-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/config_logger-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/config_logger-test/CMakeLists.txt	2025-03-15 13:43:17.312828701 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/counter_output_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/counter_output_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/counter_output_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/counter_output_stream-test/CMakeLists.txt	2025-03-15 13:43:17.312828701 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/crc32-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/crc32-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/crc32-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/crc32-test/CMakeLists.txt	2025-03-15 13:43:17.312828701 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/crypto-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/crypto-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/crypto-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/crypto-test/CMakeLists.txt	2025-03-15 13:43:17.312828701 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/digest-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/digest-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/digest-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/digest-test/CMakeLists.txt	2025-03-15 13:43:17.312828701 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/dns_message-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/dns_message-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/dns_message-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/dns_message-test/CMakeLists.txt	2025-03-15 13:43:17.312828701 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/dns_packet_reader-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/dns_packet_reader-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/dns_packet_reader-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/dns_packet_reader-test/CMakeLists.txt	2025-03-15 13:43:17.312828701 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/dns_packet_writer-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/dns_packet_writer-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/dns_packet_writer-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/dns_packet_writer-test/CMakeLists.txt	2025-03-15 13:43:17.312828701 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/dns_resource_record-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/dns_resource_record-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/dns_resource_record-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/dns_resource_record-test/CMakeLists.txt	2025-03-15 13:43:17.312828701 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/dnscore-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/dnscore-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/dnscore-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/dnscore-test/CMakeLists.txt	2025-03-15 13:43:17.312828701 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/dnscore_extension-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/dnscore_extension-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/dnscore_extension-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/dnscore_extension-test/CMakeLists.txt	2025-03-15 13:43:17.312828701 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/dnsformat-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/dnsformat-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/dnsformat-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/dnsformat-test/CMakeLists.txt	2025-03-15 13:43:17.312828701 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/dnskey-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/dnskey-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/dnskey-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/dnskey-test/CMakeLists.txt	2025-03-15 13:43:17.313828752 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c keys.c settings.h)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/dnsname-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/dnsname-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/dnsname-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/dnsname-test/CMakeLists.txt	2025-03-15 13:43:17.313828752 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/empty_input_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/empty_input_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/empty_input_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/empty_input_stream-test/CMakeLists.txt	2025-03-15 13:43:17.313828752 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/error_state-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/error_state-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/error_state-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/error_state-test/CMakeLists.txt	2025-03-15 13:43:17.313828752 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/fdtools-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/fdtools-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/fdtools-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/fdtools-test/CMakeLists.txt	2025-03-15 13:43:17.313828752 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/file_input_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/file_input_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/file_input_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/file_input_stream-test/CMakeLists.txt	2025-03-15 13:43:17.313828752 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/file_output_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/file_output_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/file_output_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/file_output_stream-test/CMakeLists.txt	2025-03-15 13:43:17.313828752 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/file_pool-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/file_pool-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/file_pool-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/file_pool-test/CMakeLists.txt	2025-03-15 13:43:17.313828752 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/filesystem_file-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/filesystem_file-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/filesystem_file-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/filesystem_file-test/CMakeLists.txt	2025-03-15 13:43:17.314828803 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/format-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/format-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/format-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/format-test/CMakeLists.txt	2025-03-15 13:43:17.314828803 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/hash-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/hash-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/hash-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/hash-test/CMakeLists.txt	2025-03-15 13:43:17.314828803 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/hmac_evp-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/hmac_evp-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/hmac_evp-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/hmac_evp-test/CMakeLists.txt	2025-03-15 13:43:17.314828803 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/host_address-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/host_address-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/host_address-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/host_address-test/CMakeLists.txt	2025-03-15 13:43:17.314828803 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/http_chunk_output_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/http_chunk_output_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/http_chunk_output_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/http_chunk_output_stream-test/CMakeLists.txt	2025-03-15 13:43:17.314828803 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/input_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/input_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/input_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/input_stream-test/CMakeLists.txt	2025-03-15 13:43:17.314828803 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/json-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/json-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/json-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/json-test/CMakeLists.txt	2025-03-15 13:43:17.314828803 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/limited_input_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/limited_input_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/limited_input_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/limited_input_stream-test/CMakeLists.txt	2025-03-15 13:43:17.314828803 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/list_dl-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/list_dl-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/list_dl-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/list_dl-test/CMakeLists.txt	2025-03-15 13:43:17.314828803 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/list_dl_nodata-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/list_dl_nodata-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/list_dl_nodata-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/list_dl_nodata-test/CMakeLists.txt	2025-03-15 13:43:17.314828803 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/list_sl-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/list_sl-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/list_sl-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/list_sl-test/CMakeLists.txt	2025-03-15 13:43:17.314828803 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/list_sl_debug-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/list_sl_debug-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/list_sl_debug-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/list_sl_debug-test/CMakeLists.txt	2025-03-15 13:43:17.314828803 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/logger-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/logger-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/logger-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/logger-test/CMakeLists.txt	2025-03-15 13:43:17.314828803 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/mapped_file-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/mapped_file-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/mapped_file-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/mapped_file-test/CMakeLists.txt	2025-03-15 13:43:17.315828853 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/message_dnsupdate-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/message_dnsupdate-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/message_dnsupdate-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/message_dnsupdate-test/CMakeLists.txt	2025-03-15 13:43:17.315828853 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/mt_output_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/mt_output_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/mt_output_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/mt_output_stream-test/CMakeLists.txt	2025-03-15 13:43:17.315828853 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/mutex-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/mutex-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/mutex-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/mutex-test/CMakeLists.txt	2025-03-15 13:43:17.315828853 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/network-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/network-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/network-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/network-test/CMakeLists.txt	2025-03-15 13:43:17.315828853 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/nsec3_hash-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/nsec3_hash-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/nsec3_hash-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/nsec3_hash-test/CMakeLists.txt	2025-03-15 13:43:17.315828853 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/output_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/output_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/output_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/output_stream-test/CMakeLists.txt	2025-03-15 13:43:17.315828853 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/pace-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/pace-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/pace-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/pace-test/CMakeLists.txt	2025-03-15 13:43:17.315828853 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/parser-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/parser-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/parser-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/parser-test/CMakeLists.txt	2025-03-15 13:43:17.315828853 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/parsing-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/parsing-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/parsing-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/parsing-test/CMakeLists.txt	2025-03-15 13:43:17.315828853 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/pid-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/pid-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/pid-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/pid-test/CMakeLists.txt	2025-03-15 13:43:17.315828853 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/pipe_steam-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/pipe_steam-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/pipe_steam-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/pipe_steam-test/CMakeLists.txt	2025-03-15 13:43:17.315828853 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/pool-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/pool-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/pool-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/pool-test/CMakeLists.txt	2025-03-15 13:43:17.316828904 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/popen_output_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/popen_output_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/popen_output_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/popen_output_stream-test/CMakeLists.txt	2025-03-15 13:43:17.316828904 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/process-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/process-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/process-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/process-test/CMakeLists.txt	2025-03-15 13:43:17.316828904 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/ptr_treemap-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/ptr_treemap-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/ptr_treemap-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/ptr_treemap-test/CMakeLists.txt	2025-03-15 13:43:17.316828904 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/ptr_treemap_debug-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/ptr_treemap_debug-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/ptr_treemap_debug-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/ptr_treemap_debug-test/CMakeLists.txt	2025-03-15 13:43:17.316828904 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/ptr_treeset-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/ptr_treeset-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/ptr_treeset-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/ptr_treeset-test/CMakeLists.txt	2025-03-15 13:43:17.316828904 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/ptr_treeset_debug-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/ptr_treeset_debug-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/ptr_treeset_debug-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/ptr_treeset_debug-test/CMakeLists.txt	2025-03-15 13:43:17.316828904 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/ptr_vector-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/ptr_vector-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/ptr_vector-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/ptr_vector-test/CMakeLists.txt	2025-03-15 13:43:17.316828904 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/pushback_input_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/pushback_input_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/pushback_input_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/pushback_input_stream-test/CMakeLists.txt	2025-03-15 13:43:17.316828904 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/random-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/random-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/random-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/random-test/CMakeLists.txt	2025-03-15 13:43:17.316828904 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/rest_server-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/rest_server-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/rest_server-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/rest_server-test/CMakeLists.txt	2025-03-15 13:43:17.316828904 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/rewind_input_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/rewind_input_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/rewind_input_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/rewind_input_stream-test/CMakeLists.txt	2025-03-15 13:43:17.317828955 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/rfc-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/rfc-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/rfc-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/rfc-test/CMakeLists.txt	2025-03-15 13:43:17.317828955 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/serial-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/serial-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/serial-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/serial-test/CMakeLists.txt	2025-03-15 13:43:17.317828955 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/service-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/service-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/service-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/service-test/CMakeLists.txt	2025-03-15 13:43:17.317828955 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/shared_circular_buffer-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/shared_circular_buffer-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/shared_circular_buffer-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/shared_circular_buffer-test/CMakeLists.txt	2025-03-15 13:43:17.317828955 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/shared_heap-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/shared_heap-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/shared_heap-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/shared_heap-test/CMakeLists.txt	2025-03-15 13:43:17.317828955 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/signal-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/signal-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/signal-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/signal-test/CMakeLists.txt	2025-03-15 13:43:17.317828955 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/skeleton-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/skeleton-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/skeleton-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/skeleton-test/CMakeLists.txt	2025-03-15 13:43:17.317828955 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/sys_get_cpu_count-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/sys_get_cpu_count-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/sys_get_cpu_count-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/sys_get_cpu_count-test/CMakeLists.txt	2025-03-15 13:43:17.317828955 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/thread-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/thread-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/thread-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/thread-test/CMakeLists.txt	2025-03-15 13:43:17.317828955 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/threaded_dll_cw-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/threaded_dll_cw-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/threaded_dll_cw-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/threaded_dll_cw-test/CMakeLists.txt	2025-03-15 13:43:17.317828955 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/threaded_qsl_cw-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/threaded_qsl_cw-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/threaded_qsl_cw-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/threaded_qsl_cw-test/CMakeLists.txt	2025-03-15 13:43:17.317828955 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/threaded_ringbuffer_cw-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/threaded_ringbuffer_cw-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/threaded_ringbuffer_cw-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/threaded_ringbuffer_cw-test/CMakeLists.txt	2025-03-15 13:43:17.317828955 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/threaded_slg-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/threaded_slg-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/threaded_slg-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/threaded_slg-test/CMakeLists.txt	2025-03-15 13:43:17.317828955 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/timeformat-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/timeformat-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/timeformat-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/timeformat-test/CMakeLists.txt	2025-03-15 13:43:17.318829006 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/timems-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/timems-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/timems-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/timems-test/CMakeLists.txt	2025-03-15 13:43:17.318829006 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/tools-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/tools-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/tools-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/tools-test/CMakeLists.txt	2025-03-15 13:43:17.318829006 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/tsig-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/tsig-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/tsig-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/tsig-test/CMakeLists.txt	2025-03-15 13:43:17.318829006 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/typebitmap-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/typebitmap-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/typebitmap-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/typebitmap-test/CMakeLists.txt	2025-03-15 13:43:17.318829006 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/u64_treemap-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/u64_treemap-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/u64_treemap-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/u64_treemap-test/CMakeLists.txt	2025-03-15 13:43:17.318829006 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/uri-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/uri-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/uri-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/uri-test/CMakeLists.txt	2025-03-15 13:43:17.318829006 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/utf8-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/utf8-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/utf8-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/utf8-test/CMakeLists.txt	2025-03-15 13:43:17.318829006 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/xfr_input_stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/xfr_input_stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/xfr_input_stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/xfr_input_stream-test/CMakeLists.txt	2025-03-15 13:43:17.318829006 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/zalloc-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/zalloc-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/zalloc-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/zalloc-test/CMakeLists.txt	2025-03-15 13:43:17.318829006 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/zone_reader_axfr-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/zone_reader_axfr-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/zone_reader_axfr-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/zone_reader_axfr-test/CMakeLists.txt	2025-03-15 13:43:17.318829006 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnscore/tests/zone_reader_text-test/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/tests/zone_reader_text-test/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/tests/zone_reader_text-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/tests/zone_reader_text-test/CMakeLists.txt	2025-03-15 13:43:17.318829006 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 cmake_path(GET CMAKE_CURRENT_SOURCE_DIR FILENAME NAME)
 set(SOURCE_FILES main.c)
diff -ruNp yadifa-3.0.2-11672/lib/dnsdb/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnsdb/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnsdb/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnsdb/CMakeLists.txt	2025-03-15 13:43:17.318829006 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 project(dnsdb LANGUAGES C)
 
 include(${CMAKE_SOURCE_DIR}/cmake/EURidToolbox.cmake)
diff -ruNp yadifa-3.0.2-11672/lib/dnslg/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnslg/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnslg/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnslg/CMakeLists.txt	2025-03-15 13:43:17.318829006 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 project(dnslg LANGUAGES C)
 set(CMAKE_C_STANDARD 11)
diff -ruNp yadifa-3.0.2-11672/lib/glibchooks/CMakeLists.txt yadifa-3.0.2-11672.new/lib/glibchooks/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/glibchooks/CMakeLists.txt	2025-03-11 13:40:04.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/glibchooks/CMakeLists.txt	2025-03-15 13:43:17.319829057 +0300
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 project(glibchooks LANGUAGES C)
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
diff -ruNp yadifa-3.0.2-11672/sbin/yadifad/CMakeLists.txt yadifa-3.0.2-11672.new/sbin/yadifad/CMakeLists.txt
--- yadifa-3.0.2-11672/sbin/yadifad/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/sbin/yadifad/CMakeLists.txt	2025-03-15 13:43:17.319829057 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 project(yadifad LANGUAGES C)
 
diff -ruNp yadifa-3.0.2-11672/sbin/yakeyrolld/CMakeLists.txt yadifa-3.0.2-11672.new/sbin/yakeyrolld/CMakeLists.txt
--- yadifa-3.0.2-11672/sbin/yakeyrolld/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/sbin/yakeyrolld/CMakeLists.txt	2025-03-15 13:43:17.319829057 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/circular-file-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/circular-file-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/circular-file-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/circular-file-test/CMakeLists.txt	2025-03-15 13:43:17.319829057 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/cjf-scan/CMakeLists.txt yadifa-3.0.2-11672.new/tests/cjf-scan/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/cjf-scan/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/cjf-scan/CMakeLists.txt	2025-03-15 13:43:17.319829057 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/cmdline-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/cmdline-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/cmdline-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/cmdline-test/CMakeLists.txt	2025-03-15 13:43:17.319829057 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/collection-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/collection-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/collection-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/collection-test/CMakeLists.txt	2025-03-15 13:43:17.319829057 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/daemonised-logger-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/daemonised-logger-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/daemonised-logger-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/daemonised-logger-test/CMakeLists.txt	2025-03-15 13:43:17.319829057 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/dns-message-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/dns-message-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/dns-message-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/dns-message-test/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/dnssec-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/dnssec-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/dnssec-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/dnssec-test/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/dsfromkey-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/dsfromkey-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/dsfromkey-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/dsfromkey-test/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/ethbenchprobe/CMakeLists.txt yadifa-3.0.2-11672.new/tests/ethbenchprobe/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/ethbenchprobe/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/ethbenchprobe/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/file-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/file-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/file-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/file-test/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/freebsd12-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/freebsd12-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/freebsd12-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/freebsd12-test/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/hmac-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/hmac-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/hmac-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/hmac-test/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/ipc-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/ipc-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/ipc-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/ipc-test/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/journal-from-ixfr/CMakeLists.txt yadifa-3.0.2-11672.new/tests/journal-from-ixfr/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/journal-from-ixfr/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/journal-from-ixfr/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/journal-fusion/CMakeLists.txt yadifa-3.0.2-11672.new/tests/journal-fusion/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/journal-fusion/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/journal-fusion/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/journal-speed-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/journal-speed-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/journal-speed-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/journal-speed-test/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/keygen-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/keygen-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/keygen-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/keygen-test/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/limiter-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/limiter-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/limiter-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/limiter-test/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/message-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/message-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/message-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/message-test/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/network-interfaces-forall-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/network-interfaces-forall-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/network-interfaces-forall-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/network-interfaces-forall-test/CMakeLists.txt	2025-03-15 13:43:17.320829108 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/notify-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/notify-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/notify-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/notify-test/CMakeLists.txt	2025-03-15 13:43:17.321829159 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/nsec3-answer/CMakeLists.txt yadifa-3.0.2-11672.new/tests/nsec3-answer/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/nsec3-answer/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/nsec3-answer/CMakeLists.txt	2025-03-15 13:43:17.321829159 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/nsec3-hash/CMakeLists.txt yadifa-3.0.2-11672.new/tests/nsec3-hash/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/nsec3-hash/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/nsec3-hash/CMakeLists.txt	2025-03-15 13:43:17.321829159 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/pid-file-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/pid-file-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/pid-file-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/pid-file-test/CMakeLists.txt	2025-03-15 13:43:17.321829159 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/protocol-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/protocol-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/protocol-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/protocol-test/CMakeLists.txt	2025-03-15 13:43:17.321829159 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/query-hammer-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/query-hammer-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/query-hammer-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/query-hammer-test/CMakeLists.txt	2025-03-15 13:43:17.321829159 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/rndc-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/rndc-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/rndc-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/rndc-test/CMakeLists.txt	2025-03-15 13:43:17.321829159 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/rrsig-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/rrsig-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/rrsig-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/rrsig-test/CMakeLists.txt	2025-03-15 13:43:17.321829159 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/shared-circular-buffer-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/shared-circular-buffer-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/shared-circular-buffer-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/shared-circular-buffer-test/CMakeLists.txt	2025-03-15 13:43:17.321829159 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/shared-heap-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/shared-heap-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/shared-heap-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/shared-heap-test/CMakeLists.txt	2025-03-15 13:43:17.321829159 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/simple-http-server-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/simple-http-server-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/simple-http-server-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/simple-http-server-test/CMakeLists.txt	2025-03-15 13:43:17.321829159 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/skeleton/CMakeLists.txt yadifa-3.0.2-11672.new/tests/skeleton/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/skeleton/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/skeleton/CMakeLists.txt	2025-03-15 13:43:17.322829210 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/stream-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/stream-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/stream-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/stream-test/CMakeLists.txt	2025-03-15 13:43:17.322829210 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/tag-gen/CMakeLists.txt yadifa-3.0.2-11672.new/tests/tag-gen/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/tag-gen/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/tag-gen/CMakeLists.txt	2025-03-15 13:43:17.322829210 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/threaded-queue/CMakeLists.txt yadifa-3.0.2-11672.new/tests/threaded-queue/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/threaded-queue/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/threaded-queue/CMakeLists.txt	2025-03-15 13:43:17.322829210 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/udp-echo-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/udp-echo-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/udp-echo-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/udp-echo-test/CMakeLists.txt	2025-03-15 13:43:17.322829210 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/update-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/update-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/update-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/update-test/CMakeLists.txt	2025-03-15 13:43:17.322829210 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/tests/zone-save-test/CMakeLists.txt yadifa-3.0.2-11672.new/tests/zone-save-test/CMakeLists.txt
--- yadifa-3.0.2-11672/tests/zone-save-test/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/tests/zone-save-test/CMakeLists.txt	2025-03-15 13:43:17.322829210 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 # include(${CMAKE_SOURCE_DIR}/cmake/FindYADIFA.cmake)
 
diff -ruNp yadifa-3.0.2-11672/var/CMakeLists.txt yadifa-3.0.2-11672.new/var/CMakeLists.txt
--- yadifa-3.0.2-11672/var/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/var/CMakeLists.txt	2025-03-15 13:43:17.322829210 +0300
@@ -30,7 +30,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.21)
 
 message("var will be installed in ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/var")
 
