diff -ruNp yadifa-3.0.2-11672/var/CMakeLists.txt yadifa-3.0.2-11672.new/var/CMakeLists.txt
--- yadifa-3.0.2-11672/var/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/var/CMakeLists.txt	2025-03-15 13:42:14.823645290 +0300
@@ -39,17 +39,17 @@ install(DIRECTORY DESTINATION ${CMAKE_IN
 install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/log/yadifa)
 install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/log/yadifad)
 install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/log/yakeyrolld)
-install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/plans)
+install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/lib/yadifa/plans)
 install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/run)
-install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/zones)
-install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/zones/keys)
-install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/zones/primaries)
-install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/zones/secondaries)
-install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/zones/xfr)
+install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/lib/yadifa/zones)
+install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/lib/yadifa/zones/keys)
+install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/lib/yadifa/zones/primaries)
+install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/lib/yadifa/zones/secondaries)
+install(DIRECTORY DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/lib/yadifa/zones/xfr)
 
 install(FILES   zones/primaries/0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.ip6.arpa.zone
                 zones/primaries/0.0.127.in-addr.arpa.zone
                 zones/primaries/localhost6.zone
                 zones/primaries/localhost.zone
                 zones/primaries/somedomain.eu.zone
-        DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/zones/primaries)
+        DESTINATION ${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/lib/yadifa/zones/primaries)
