diff -ruNp yadifa-3.0.2-11672/lib/dnscore/CMakeLists.txt yadifa-3.0.2-11672.new/lib/dnscore/CMakeLists.txt
--- yadifa-3.0.2-11672/lib/dnscore/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/CMakeLists.txt	2025-03-16 18:22:25.349492905 +0300
@@ -764,7 +764,7 @@ add_subdirectory(tests/dns_resource_reco
 add_subdirectory(tests/dnscore_extension-test)
 add_subdirectory(tests/dnsformat-test)
 add_subdirectory(tests/dnskey-test)
-add_subdirectory(tests/dnsname-test)
+#add_subdirectory(tests/dnsname-test)
 add_subdirectory(tests/error_state-test) # slow
 add_subdirectory(tests/file_pool-test)
 add_subdirectory(tests/format-test)
