diff -ruNp yadifa-3.0.2-11672/etc/CMakeLists.txt yadifa-3.0.2-11672.new/etc/CMakeLists.txt
--- yadifa-3.0.2-11672/etc/CMakeLists.txt	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/etc/CMakeLists.txt	2025-03-16 18:24:59.436989412 +0300
@@ -158,6 +158,6 @@ install(DIRECTORY DESTINATION ${CMAKE_IN
 install(CODE
         "
         #file(TOUCH \"${CMAKE_INSTALL_FULL_SYSCONFDIR}/yadifa.conf\")
-        file(TOUCH \"${CMAKE_INSTALL_FULL_SYSCONFDIR}/yadifad.conf\")
-        file(TOUCH \"${CMAKE_INSTALL_FULL_SYSCONFDIR}/yakeyrolld.conf\")
+        #file(TOUCH \"${CMAKE_INSTALL_FULL_SYSCONFDIR}/yadifad.conf\")
+        #file(TOUCH \"${CMAKE_INSTALL_FULL_SYSCONFDIR}/yakeyrolld.conf\")
         ")
