diff -ruNp yadifa-3.0.2-11672/lib/dnscore/include/dnscore/thread.h yadifa-3.0.2-11672.new/lib/dnscore/include/dnscore/thread.h
--- yadifa-3.0.2-11672/lib/dnscore/include/dnscore/thread.h	2025-03-11 14:26:10.000000000 +0300
+++ yadifa-3.0.2-11672.new/lib/dnscore/include/dnscore/thread.h	2025-03-16 18:28:44.449936509 +0300
@@ -56,7 +56,6 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <stdatomic.h>
-static inline pid_t gettid() { return getpid(); }
 #endif
 
 #ifdef __cplusplus
