--- a/main/dlist.cc	2006-04-06 01:14:54.000000000 +0600
+++ b/main/dlist.cc	2008-04-17 08:50:22.000000000 +0600
@@ -1023,11 +1023,12 @@
 				cpfrom=oldname;
 			}else
 				cpfrom=newname;
-				 
-			WL->log_printf(LOG_WARNING,_("Trying to copy %s to %s due 'Content-Disposition'"),
+			if(strcmp(cpfrom,tmp)){
+				WL->log_printf(LOG_WARNING,_("Trying to copy %s to %s due 'Content-Disposition'"),
 				       cpfrom,tmp);
-			if (file_copy(cpfrom,tmp,buf,1000))
-				WL->log(LOG_ERROR,_("Error during copying!"));
+				if (file_copy(cpfrom,tmp,buf,1000))
+					WL->log(LOG_ERROR,_("Error during copying!"));
+			}
 			delete[] buf;
 			delete[] tmp;
 			delete[] newname;
