--- src/xlib/gtk.c.orig	2018-03-26 14:52:33.000000000 +1000
+++ src/xlib/gtk.c	2018-04-02 09:59:02.201013926 +1000
@@ -24,8 +24,8 @@
 #include <stdlib.h>
 #include <string.h>
 
-#define LIBGTK_FILENAME "libgtk-3.so"
-#define LIBGTK_FILENAME_FALLBACK "libgtk-3.so.0"
+#define LIBGTK_FILENAME "libgtk-x11-2.0.so"
+#define LIBGTK_FILENAME_FALLBACK "libgtk-x11-2.0.so.0"
 
 #define GTK_FILE_CHOOSER_ACTION_OPEN 0
 #define GTK_FILE_CHOOSER_ACTION_SAVE 1
