--- uTox-0.5.0/Makefile	2015-12-20 00:57:28.000000000 -0700
+++ uTox-0.5.0-fix/Makefile	2015-12-24 10:57:21.787213435 -0700
@@ -127,8 +127,8 @@
 	install -m 644 utox.desktop $(DESTDIR)$(PREFIX)/share/applications/utox.desktop
 	if [ "$(UNITY)" -eq "1" ]; then echo "X-MessagingMenu-UsesChatSection=true" >> $(DESTDIR)$(PREFIX)/share/applications/utox.desktop; fi
 
-	mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1
-	install -m 644 utox.1 $(DESTDIR)$(PREFIX)/share/man/man1/utox.1
+	mkdir -p $(DESTDIR)$(PREFIX)/man/man1
+	install -m 644 utox.1 $(DESTDIR)$(PREFIX)/man/man1/utox.1
 
 $(OBJ): %.o: %.c $(HEADERS)
 	@echo "  CC    $@"
