--- uTox-0.4.5/Makefile	2015-12-09 16:56:06.000000000 -0700
+++ uTox-0.4.5-fix/Makefile	2015-12-13 06:28:47.453261588 -0700
@@ -104,8 +104,8 @@
 	install -m 644 utox.desktop $(DESTDIR)$(PREFIX)/share/applications/utox.desktop
 	if [ "$(UNITY)" -eq "1" ]; then echo "X-MessagingMenu-UsesChatSection=true" >> $(DESTDIR)$(PREFIX)/share/applications/utox.desktop; fi
 
-	mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1
-	install -m 644 utox.1 $(DESTDIR)$(PREFIX)/share/man/man1/utox.1
+	mkdir -p $(DESTDIR)$(PREFIX)/man/man1
+	install -m 644 utox.1 $(DESTDIR)$(PREFIX)/man/man1/utox.1
 
 main.o: xlib/main.c xlib/keysym2ucs.c
 
