diff -uarN freenx-server-0.7.3.orig/nxloadconfig freenx-server-0.7.3/nxloadconfig
--- freenx-server-0.7.3.orig/nxloadconfig	2008-08-22 02:44:43.000000000 +0200
+++ freenx-server-0.7.3/nxloadconfig	2008-11-16 01:28:14.000000000 +0100
@@ -70,7 +70,7 @@
 APPLICATION_LIBRARY_PRELOAD="" #Calculated
 
 # the name of the authorized keys file for ssh
-SSH_AUTHORIZED_KEYS="authorized_keys2"
+SSH_AUTHORIZED_KEYS="authorized_keys"
 
 # retest values like xauth, netcat and checks for nxdesktop/nxviewer availability
 # set to 0 if you are sure, you set the right values
