diff -uarN freenx-server-0.7.3.orig/node.conf.sample freenx-server-0.7.3/node.conf.sample
--- freenx-server-0.7.3.orig/node.conf.sample	2008-08-22 02:44:43.000000000 +0200
+++ freenx-server-0.7.3/node.conf.sample	2009-01-27 21:03:36.000000000 +0100
@@ -498,12 +498,12 @@
 # and .Xsession
 # Depending on distribution DEFAULT_X_SESSION might be /etc/X11/xdm/Xsession,
 # /etc/X11/Sessions/Xsession or /etc/X11/xinit/xinitrc
-#USER_X_STARTUP_SCRIPT=.Xclients
-#DEFAULT_X_SESSION=/etc/X11/xdm/Xsession
+#USER_X_STARTUP_SCRIPT=.xinitrc
+#DEFAULT_X_SESSION=/etc/X11/xinit/xinitrc
 
 # The key that contains the name of the script that starts a KDE session.
 # It's run when a 'unix-kde' session is requested by the client.
-#COMMAND_START_KDE=startkde
+#COMMAND_START_KDE=/etc/X11/xinit/xinitrc.kde
 
 # The key that contains the name of the script that starts a gnome session.
 # It's run when a 'unix-gnome' session is requested by the client.
diff -uarN freenx-server-0.7.3.orig/nxloadconfig freenx-server-0.7.3/nxloadconfig
--- freenx-server-0.7.3.orig/nxloadconfig	2008-08-22 02:44:43.000000000 +0200
+++ freenx-server-0.7.3/nxloadconfig	2009-01-27 21:01:57.000000000 +0100
@@ -56,8 +56,8 @@
 NX_LICENSE="OS (GPL, using backend: %BACKEND%)"
 
 # Where can different nx components be found
-NX_DIR=/usr
-PATH_BIN=$NX_DIR/bin # if you change that, be sure to also change the public keys
+NX_DIR=/usr/lib/NX
+PATH_BIN=/usr/bin # if you change that, be sure to also change the public keys
 PATH_LIB=$NX_DIR/lib
 NX_ETC_DIR=/etc/nxserver
 NX_SESS_DIR=/var/lib/nxserver/db
@@ -184,9 +184,9 @@
 
 DEFAULT_X_WM=""
 KILL_DEFAULT_X_WM="1"
-USER_X_STARTUP_SCRIPT=.Xclients
-DEFAULT_X_SESSION=/etc/X11/xdm/Xsession
-COMMAND_START_KDE=startkde
+USER_X_STARTUP_SCRIPT=.xinitrc
+DEFAULT_X_SESSION=/etc/X11/xinit/xinitrc
+COMMAND_START_KDE=/etc/X11/xinit/xinitrc.kde
 COMMAND_START_GNOME=gnome-session
 COMMAND_START_CDE=cdwm
 COMMAND_XTERM=xterm
@@ -317,7 +317,7 @@
 [ -z "$APPLICATION_LIBRARY_PATH" ] && APPLICATION_LIBRARY_PATH=$PATH_LIB
 [ -z "$APPLICATION_LIBRARY_PRELOAD" ] && APPLICATION_LIBRARY_PRELOAD="$APPLICATION_LIBRARY_PATH/libX11.so.6.2:$APPLICATION_LIBRARY_PATH/libXext.so.6.4:$APPLICATION_LIBRARY_PATH/libXcomp.so:$APPLICATION_LIBRARY_PATH/libXcompext.so:$APPLICATION_LIBRARY_PATH/libXrender.so.1.2"
 
-NX_BACKEND_VERSION=$(strings $PATH_BIN/nxagent 2>/dev/null | egrep 'NXAGENT - Version' | sed 's/.*Version //g')
+NX_BACKEND_VERSION=$(strings $NX_DIR/nxagent 2>/dev/null | egrep 'NXAGENT - Version' | sed 's/.*Version //g')
 
 [ "$ENABLE_1_5_0_BACKEND" = "1" ] && NX_BACKEND_VERSION="1.5.0" # forced the backend
 
@@ -606,7 +606,7 @@
 	[ -z $(echo "$ENABLE_ROOTLESS_MODE" | egrep "^[0|1]$") ] && \
 		ERROR="yes" && echo "Error: Invalid value \"ENABLE_ROOTLESS_MODE=$ENABLE_ROOTLESS_MODE\""
 
-	[ -z "$(strings $PATH_BIN/nxagent | egrep 'NXAGENT - Version 1.5.0|NXAGENT - Version 2.[01].0|NXAGENT - Version 3.[012].0')" ] && \
+	[ -z "$(strings $NX_DIR/nxagent | egrep 'NXAGENT - Version 1.5.0|NXAGENT - Version 2.[01].0|NXAGENT - Version 3.[012].0')" ] && \
 		WARNING="yes" && echo "Error: Could not find 1.5.0 or 2.[01].0 or 3.[01].0 version string in nxagent. NX 1.5.0 or 2.[01].0 or 3.[012].0 backend is needed for this version of FreeNX."
 		
 	[ -z $(echo "$ENABLE_USESSION" | egrep "^[0|1]$") ] && \
