diff -Naur platform-platform-2.1.0.orig/CMakeLists.txt platform-platform-2.1.0/CMakeLists.txt
--- platform-platform-2.1.0.orig/CMakeLists.txt	2016-10-24 09:56:25.000000000 +0000
+++ platform-platform-2.1.0/CMakeLists.txt	2017-03-01 16:41:56.443161968 +0000
@@ -81,5 +81,5 @@
 configure_file (p8-platform-config.cmake.in
                 p8-platform-config.cmake @ONLY)
 install(FILES ${CMAKE_BINARY_DIR}/p8-platform-config.cmake
-        DESTINATION ${CMAKE_INSTALL_LIBDIR}/p8-platform)
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/p8-platform)
 
