diff -Nur rodent-5.0.5.orig/Build/plugins/Makefile.am rodent-5.0.5/Build/plugins/Makefile.am
--- rodent-5.0.5.orig/Build/plugins/Makefile.am	2013-05-31 17:55:27.000000000 +0200
+++ rodent-5.0.5/Build/plugins/Makefile.am	2013-06-03 10:32:52.647215769 +0200
@@ -253,6 +253,6 @@
 	)
 
 install-exec-local:
-	$(mkinstalldirs) $(bindir)
-	$(top_srcdir)/install-sh $(dirplug)/filesystem_user_space/rfm-simple-agent $(bindir)
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	$(top_srcdir)/install-sh $(dirplug)/filesystem_user_space/rfm-simple-agent $(DESTDIR)$(bindir)
 
diff -Nur rodent-5.0.5.orig/Build/plugins/Makefile.in rodent-5.0.5/Build/plugins/Makefile.in
--- rodent-5.0.5.orig/Build/plugins/Makefile.in	2013-05-31 19:02:34.000000000 +0200
+++ rodent-5.0.5/Build/plugins/Makefile.in	2013-06-03 10:33:06.296045136 +0200
@@ -1386,8 +1386,8 @@
 	)
 
 install-exec-local:
-	$(mkinstalldirs) $(bindir)
-	$(top_srcdir)/install-sh $(dirplug)/filesystem_user_space/rfm-simple-agent $(bindir)
+	$(mkinstalldirs) $(DESTDIR)$(bindir)
+	$(top_srcdir)/install-sh $(dirplug)/filesystem_user_space/rfm-simple-agent $(DESTDIR)$(bindir)
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
