diff -ur rodent-4.8.0.orig//Build/plugins/Makefile.am rodent-4.8.0/Build/plugins/Makefile.am
--- rodent-4.8.0.orig//Build/plugins/Makefile.am	2012-06-25 15:49:48.000000000 +0200
+++ rodent-4.8.0/Build/plugins/Makefile.am	2012-06-28 11:32:27.775686892 +0200
@@ -239,5 +239,5 @@
 
 if WANT_EXPERIMENTAL
 install-exec-local:
-	$(top_srcdir)/install-sh $(dirplug)/filesystem_user_space/rfm-simple-agent $(bindir)
+	$(top_srcdir)/install-sh $(dirplug)/filesystem_user_space/rfm-simple-agent $(DESTDIR)$(bindir)
 endif
diff -ur rodent-4.8.0.orig//Build/plugins/Makefile.in rodent-4.8.0/Build/plugins/Makefile.in
--- rodent-4.8.0.orig//Build/plugins/Makefile.in	2012-06-27 21:39:52.000000000 +0200
+++ rodent-4.8.0/Build/plugins/Makefile.in	2012-06-28 11:46:03.518488703 +0200
@@ -1326,7 +1326,7 @@
 	)
 
 @WANT_EXPERIMENTAL_TRUE@install-exec-local:
-@WANT_EXPERIMENTAL_TRUE@	$(top_srcdir)/install-sh $(dirplug)/filesystem_user_space/rfm-simple-agent $(bindir)
+@WANT_EXPERIMENTAL_TRUE@	$(top_srcdir)/install-sh $(dirplug)/filesystem_user_space/rfm-simple-agent $(DESTDIR)$(bindir)
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
