diff -Naur html2ps-1.0b7/html2ps html2ps-1.0b7.patched/html2ps
--- html2ps-1.0b7/html2ps	2010-05-07 07:22:12.000000000 -0400
+++ html2ps-1.0b7.patched/html2ps	2014-09-12 13:37:55.000000000 -0400
@@ -3405,7 +3405,6 @@
   local($optlist)=@_;
   local(@args,$_,$opt,$opts,$rest,$olist,$plist,$found,@popts);
   local($errs)=0;
-  local($[)=0;
   @args=split( /\|/, $optlist );
   for $opt (@args) {
     if(substr($opt,-1,1) ne ':') {$olist.=$opt}
