--- cnijfilter2-source-6.90-1/lgmon3/configure.in	2024-12-18 20:35:00.000000000 +1300
+++ cnijfilter2-source-6.90-1.new/lgmon3/configure.in	2025-11-01 16:43:40.416873662 +1300
@@ -72,7 +72,7 @@
 
 CFLAGS="-O2"
 
-XML2_CFLAGS=`xml2-config --cflags`
+XML2_CFLAGS=`pkg-config libxml-2.0 --cflags`
 AC_SUBST(XML2_CFLAGS)
 
 AC_OUTPUT([
--- cnijfilter2-source-6.90-1/tocnpwg/configure.in	2024-12-18 20:35:00.000000000 +1300
+++ cnijfilter2-source-6.90-1.new/tocnpwg/configure.in	2025-11-01 16:44:13.320873000 +1300
@@ -41,7 +41,7 @@
 
 CFLAGS="-O2"
 
-XML_2_CFLAGS=`xml2-config --cflags`
+XML_2_CFLAGS=`pkg-config libxml-2.0 --cflags`
 AC_SUBST(XML_2_CFLAGS)
 
 AC_OUTPUT(Makefile
