--- cnijfilter2-source-6.20-1.orig/lgmon3/src/cnijlgmon3.c	2021-02-20 22:08:56.000000000 +1300
+++ cnijfilter2-source-6.20-1/lgmon3/src/cnijlgmon3.c	2021-11-05 23:35:18.626436603 +1300
@@ -55,7 +55,7 @@
 int (*GET_STATUS)(char *, int, int *, int * , char *);
 int (*GET_STATUS2)(char *, int, char *, int *, int * , char *, char *);
 int (*GET_STATUS2_MAINTENANCE)(char *, int, char *, int *, int * , char *, char *);
-int (*GET_PROTOCOL)(char *, size_t);
+static int (*GET_PROTOCOL)(char *, size_t);
 
 
  int main(int argc, char *argv[])
