--- atop-1.20/atop.init.orig	2008-01-08 20:44:34.327288088 -0600
+++ atop-1.20/atop.init	2008-01-08 20:45:25.786220566 -0600
@@ -7,7 +7,7 @@
 #
 
 # Check existance of binaries 
-[ -f /usr/bin/atop ] || exit 0
+[ -f /usr/bin/atop ] || exit 1
 
 PIDFILE=/var/run/atop.pid
 RETVAL=0
@@ -22,7 +22,7 @@
 		:
 	else
 		# Start atop
-		/etc/atop/atop.daily
+		/etc/cron.daily/atop
 	fi
 	;;
 
@@ -43,11 +43,11 @@
 	;;
 
   reload)
-	/etc/atop/atop.daily
+	/etc/cron.daily/atop
 	;;
 
   restart)
-	/etc/atop/atop.daily
+	/etc/cron.daily/atop
 	;;
 
   *)
