Description: Extend desktop files
Author: <eriks@debian.org>
Last-Update: 2009-06-04

diff -urNad antiword-0.37~/Unix-only/KDE3-only/Antiword.desktop.eu antiword-0.37/Unix-only/KDE3-only/Antiword.desktop.eu
--- antiword-0.37~/Unix-only/KDE3-only/Antiword.desktop.eu	2001-07-08 20:45:44.000000000 +0200
+++ antiword-0.37/Unix-only/KDE3-only/Antiword.desktop.eu	2009-06-04 10:03:59.508015784 +0200
@@ -1,8 +1,16 @@
 [Desktop Entry]
+Encoding=UTF-8
 BinaryPattern=kantiword;Kantiword
-MimeType=application/msword
-Name=Antiword
+Comment=*.doc Dateien per Drag&Drop auf das Icon anzeigen
+Comment[de]=*.doc Dateien per Drag&Drop auf das Icon anzeigen
+Comment[en]=displays *.doc files via drag'n'drop on this icon
 Exec=kantiword a4 "%f"
+GenericName=Anzeige für MS-Word-Dokument
+GenericName[de]=Anzeige für MS-Word-Dokument
+GenericName[en]=MS-Word document viewer
 Icon=antiword
+MimeType=application/msword
+Name=Kantiword
+StartupNotify=false
+Terminal=false
 Type=Application
-Terminal=0
diff -urNad antiword-0.37~/Unix-only/KDE3-only/Antiword.desktop.us antiword-0.37/Unix-only/KDE3-only/Antiword.desktop.us
--- antiword-0.37~/Unix-only/KDE3-only/Antiword.desktop.us	2001-07-08 20:45:54.000000000 +0200
+++ antiword-0.37/Unix-only/KDE3-only/Antiword.desktop.us	2009-06-04 10:03:59.508015784 +0200
@@ -1,8 +1,16 @@
 [Desktop Entry]
+Encoding=UTF-8
 BinaryPattern=kantiword;Kantiword
-MimeType=application/msword
-Name=Antiword
+Comment=*.doc Dateien per Drag&Drop auf das Icon anzeigen
+Comment[de]=*.doc Dateien per Drag&Drop auf das Icon anzeigen
+Comment[en]=displays *.doc files via drag'n'drop on this icon
 Exec=kantiword letter "%f"
+GenericName=Anzeige für MS-Word-Dokument
+GenericName[de]=Anzeige für MS-Word-Dokument
+GenericName[en]=MS-Word document viewer
 Icon=antiword
+MimeType=application/msword
+Name=Kantiword
+StartupNotify=false
+Terminal=false
 Type=Application
-Terminal=0
