diff -up fceu2.1.4a/SConstruct.00 fceu2.1.4a/SConstruct
--- fceu2.1.4a/SConstruct.00	2010-06-05 20:53:37.621706694 -0700
+++ fceu2.1.4a/SConstruct	2010-06-05 20:53:51.370717520 -0700
@@ -10,7 +10,7 @@ opts.AddVariables(
   BoolVariable('DEBUG',     'Build with debugging symbols', 1),
   BoolVariable('LUA',       'Enable Lua support', 1),
   BoolVariable('NEWPPU',    'Enable new PPU core', 1),
-  BoolVariable('CREATE_AVI', 'Enable avi creation support (SDL only)', 0),
+  BoolVariable('CREATE_AVI', 'Enable avi creation support (SDL only)', 1),
   BoolVariable('LOGO', 'Enable a logoscreen when creating avis (SDL only)', '1'),
   BoolVariable('GTK', 'Enable GTK2 GUI (SDL only)', 1),
   BoolVariable('GTK_LITE', 'Enable GTK2 for dialogs only', 0)
