diff -Naur fceur2668.orig/SConstruct fceur2668/SConstruct
--- fceur2668.orig/SConstruct	2012-09-11 11:58:00.000000000 +0200
+++ fceur2668/SConstruct	2012-09-17 11:11:09.082042995 +0200
@@ -19,7 +19,7 @@
   BoolVariable('LUA',       'Enable Lua support', 1),
   BoolVariable('NEWPPU',    'Enable new PPU core', 1),
   BoolVariable('CREATE_AVI', 'Enable avi creation support (SDL only)', 1),
-  BoolVariable('LOGO', 'Enable a logoscreen when creating avis (SDL only)', '1'),
+  BoolVariable('LOGO', 'Enable a logoscreen when creating avis (SDL only)', '0'),
   BoolVariable('GTK', 'Enable GTK2 GUI (SDL only)', 1),
   BoolVariable('GTK3', 'Enable GTK3 GUI (SDL only)', 0),
   BoolVariable('CLANG', 'Compile with llvm-clang instead of gcc', 0)
@@ -30,8 +30,8 @@
 env = Environment(options = opts)
 
 #### Uncomment this for a public release ###
-# env.Append(CPPDEFINES=["PUBLIC_RELEASE"])
-# env['DEBUG'] = 0
+env.Append(CPPDEFINES=["PUBLIC_RELEASE"])
+env['DEBUG'] = 0
 ############################################
 
 # LSB_FIRST must be off for PPC to compile
@@ -97,6 +97,7 @@
     env.ParseConfig('pkg-config --cflags --libs gtk+-3.0')
     env.Append(CPPDEFINES=["_GTK3"])
     env.Append(CCFLAGS = ["-D_GTK"])
+  env.Append(LIBS =["-lz","-ldl"])
 
   ### Lua platform defines
   ### Applies to all files even though only lua needs it, but should be ok
