diff -up fceu2.1.5/SConstruct.00 fceu2.1.5/SConstruct
--- fceu2.1.5/SConstruct.00	2011-06-11 08:32:32.807001855 -0700
+++ fceu2.1.5/SConstruct	2011-06-11 08:33:05.347001851 -0700
@@ -11,7 +11,7 @@ opts.AddVariables(
   BoolVariable('LUA',       'Enable Lua support', 1),
   BoolVariable('NEWPPU',    'Enable new PPU core', 1),
   BoolVariable('CREATE_AVI', 'Enable avi creation support (SDL only)', 1),
-  BoolVariable('LOGO', 'Enable a logoscreen when creating avis (SDL only)', '1'),
+  BoolVariable('LOGO', 'Enable a logoscreen when creating avis (SDL only)', '0'),
   BoolVariable('GTK', 'Enable GTK2 GUI (SDL only)', 1),
   BoolVariable('GTK3', 'Enable GTK3 GUI (BROKEN/EXPERIMENTAL)', 0),
 )
@@ -70,6 +70,9 @@ else:
     env.ParseConfig('pkg-config --cflags --libs gtk+-3.0')
     env.Append(CPPDEFINES=["_GTK3"])
     env.Append(CCFLAGS = ["-D_GTK"])
+    env.Append(LIBS =["-lz"])
+    env.Append(LIBS =["-ldl"])
+
   ### Lua platform defines
   ### Applies to all files even though only lua needs it, but should be ok
   if env['LUA']:
