diff -up fceux/SConstruct.00 fceux/SConstruct
--- fceux/SConstruct.00	2013-01-05 01:02:40.508959444 -0600
+++ fceux/SConstruct	2013-01-05 01:05:40.071959252 -0600
@@ -22,7 +22,7 @@ opts.AddVariables(
   BoolVariable('SYSTEM_LUA','Use system lua instead of static lua provided with fceux', 1),
   BoolVariable('NEWPPU',    'Enable new PPU core', 1),
   BoolVariable('CREATE_AVI', 'Enable avi creation support (SDL only)', 1),
-  BoolVariable('LOGO', 'Enable a logoscreen when creating avis (SDL only)', 1),
+  BoolVariable('LOGO', 'Enable a logoscreen when creating avis (SDL only)', 0),
   BoolVariable('GTK', 'Enable GTK2 GUI (SDL only)', 1),
   BoolVariable('GTK3', 'Enable GTK3 GUI (SDL only)', 0),
   BoolVariable('CLANG', 'Compile with llvm-clang instead of gcc', 0)
@@ -100,6 +100,9 @@ else:
     env.ParseConfig('pkg-config --cflags --libs gtk+-3.0')
     env.Append(CPPDEFINES=["_GTK3"])
     env.Append(CCFLAGS = ["-D_GTK"])
+  env.Append(LIBS =["-lz"])
+  env.Append(LIBS =["-ldl"])
+  env.Append(LIBS =["-lX11"])  
 
   ### Lua platform defines
   ### Applies to all files even though only lua needs it, but should be ok
