diff -Naur wmmon.app/wmgeneral/wmgeneral.c wmmon.app.patched/wmgeneral/wmgeneral.c
--- wmmon.app/wmgeneral/wmgeneral.c	2009-04-20 17:03:25.000000000 -0400
+++ wmmon.app.patched/wmgeneral/wmgeneral.c	2009-04-20 17:02:58.000000000 -0400
@@ -95,6 +95,8 @@
 	fp = fopen(filename, "r");
 	if (fp) {
 		while (getline(&line, &line_size, fp) >= 0) {
+			if(*line == '#')
+				continue;
 			key = 0;
 			while (key >= 0 && keys[key].label) {
 				if ((p = strstr(line, keys[key].label))) {
