From 64ca3fd1568e56f2d0e1fe992f3a0f5ba57ea694 Mon Sep 17 00:00:00 2001
From: Stephen Abraham <stephenabraham@tophatorbotics.com>
Date: Mon, 13 Sep 2021 16:22:46 -0400
Subject: [PATCH] Removed 2to3 command

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index c9de5c9..6a80b4e 100755
--- a/setup.py
+++ b/setup.py
@@ -127,7 +127,6 @@ def finalize_options(self):
       license='GPL',
       extras_require={'ble': ['gattlib']},
       package_dir=package_dir,
-      use_2to3=True,
       install_requires=install_requires,
       package_data=package_data,
       eager_resources=eager_resources,
