--- wmMoonClock-1.27/Src/wmMoonClock.c 1999-06-07 11:27:18.000000000 -0300
+++ wmMoonClock-1.??/Src/wmMoonClock.c 2004-12-15 18:59:50.714764064 -0200
@@ -244,6 +244,7 @@
		nMAX = 1000;
	        ImageNumber = (int)(c.MoonPhase * 60.0 + 0.5);
	        if (ImageNumber > 59) ImageNumber = 0;
+	        if (Glat < 0) ImageNumber = 59 - ImageNumber;
	        j = ImageNumber/10;
	        i = ImageNumber%10;
	        copyXPMArea(67+58*i, 2+58*j, 54, 54, 5, 5);
