--- Makefile	2022-01-26 00:21:45.000000000 +0200
+++ Makefile.new	2024-01-11 03:29:04.935018734 +0200
@@ -35,14 +35,11 @@
 # Mac OS X users might need to add -D__unix__ to CFLAGS
 # and use CC=cc or CC=gcc
 
-OPTIMISE=-O3
-
 CFLAGS+=-c -pedantic -Wall -Wshadow -Wformat -Wpointer-arith \
 	-Wstrict-prototypes -Wmissing-prototypes -Wwrite-strings \
 	-Wsign-compare -Wimplicit-function-declaration $(DEBUGINFO) \
 	-I/usr/local/lib/ansi-include -std=c99 \
-        $(CPPFLAGS) \
-        $(OPTIMISE)
+        $(CPPFLAGS)
 	 
 CC=gcc
 LIBS=-lm
