--- cdiomodule.h.orig	2015-11-09 19:48:07.220880278 -0600
+++ cdiomodule.h	2015-11-09 19:48:32.765284762 -0600
@@ -1,11 +1,6 @@
 #include <Python.h>
-#ifdef OLD_PARANOIA
-#include <cdio/cdda.h>
-#include <cdio/paranoia.h>
-#else
-#include <cdio/cdparanoia/cdda.h>
-#include <cdio/cdparanoia/paranoia.h>
-#endif
+#include <cdio/paranoia/cdda.h>
+#include <cdio/paranoia/paranoia.h>
 
 /********************************************************
  Audio Tools, a module and set of tools for manipulating audio data
