--- a/dist/arch/gnome/PKGBUILD
+++ b/dist/arch/gnome/PKGBUILD
@@ -13,7 +13,7 @@ depends=(
   'qt6-wayland'
   'gnome-shell-extension-appindicator'
   'thunderbird>=91'
-  'thunderbird<137'
+  'thunderbird<999'
 )
 makedepends=(
   'git'
--- a/dist/arch/kde/PKGBUILD
+++ b/dist/arch/kde/PKGBUILD
@@ -13,7 +13,7 @@ depends=(
   'qt6-wayland'
   'kstatusnotifieritem'
   'thunderbird>=91'
-  'thunderbird<137'
+  'thunderbird<999'
 )
 makedepends=(
   'git'
--- a/dist/arch/minimal/PKGBUILD
+++ b/dist/arch/minimal/PKGBUILD
@@ -12,7 +12,7 @@ depends=(
   'qt6-base'
   'qt6-wayland'
   'thunderbird>=91'
-  'thunderbird<137'
+  'thunderbird<999'
 )
 makedepends=(
   'git'
--- a/dist/deb/gnome/debian.control
+++ b/dist/deb/gnome/debian.control
@@ -6,7 +6,7 @@ Build-Depends: debhelper (>= 4.1.16), zip, g++, libx11-dev, qt5-qmake, qtbase5-d
 
 Package: systray-x-gnome
 Architecture: any
-Depends: ${shlibs:Depends}, gnome-shell-extension-appindicator, thunderbird (>= 1:91), thunderbird (< 1:137)
+Depends: ${shlibs:Depends}, gnome-shell-extension-appindicator, thunderbird (>= 1:91), thunderbird (< 1:999)
 Description: SysTray-X is a system tray extension for Thunderbird (GNOME).
  This version is optimized for the GNOME desktop.
  The add-on uses the WebExtension API's to control an external system
--- a/dist/deb/kde/debian.control
+++ b/dist/deb/kde/debian.control
@@ -6,7 +6,7 @@ Build-Depends: debhelper (>= 4.1.16), zip, g++, libx11-dev, qt5-qmake, qtbase5-d
 
 Package: systray-x
 Architecture: any
-Depends: ${shlibs:Depends}, thunderbird (>= 1:91), thunderbird (< 1:137)
+Depends: ${shlibs:Depends}, thunderbird (>= 1:91), thunderbird (< 1:999)
 Description: SysTray-X is a system tray extension for Thunderbird (KDE).
  This version is optimized for the KDE desktop.
  The add-on uses the WebExtension API's to control an external system
--- a/dist/deb/minimal/debian.control
+++ b/dist/deb/minimal/debian.control
@@ -6,7 +6,7 @@ Build-Depends: debhelper (>= 4.1.16), zip, g++, libx11-dev, qt5-qmake, qtbase5-d
 
 Package: systray-x-minimal
 Architecture: any
-Depends: ${shlibs:Depends}, thunderbird (>= 1:91), thunderbird (< 1:137)
+Depends: ${shlibs:Depends}, thunderbird (>= 1:91), thunderbird (< 1:999)
 Description: SysTray-X is a system tray extension for Thunderbird (non-GNOME,non-KDE).
  This version is for non-KDE and non-GNOME desktops.
  The add-on uses the WebExtension API's to control an external system
--- a/dist/rpm/gnome/systray-x-gnome.spec
+++ b/dist/rpm/gnome/systray-x-gnome.spec
@@ -43,10 +43,10 @@ BuildRequires:  libqt5-qtx11extras-devel
 BuildRequires:  pkgconfig(x11)
 %if 0%{?fedora_version}
 Requires:       thunderbird >= 91
-Requires:       thunderbird < 137
+Requires:       thunderbird < 999
 %else
 Requires:       MozillaThunderbird >= 91
-Requires:       MozillaThunderbird < 137
+Requires:       MozillaThunderbird < 999
 %endif
 %if 0%{?suse_version} > 1600 || 0%{?fedora_version} > 39
 %if 0%{?fedora_version}
--- a/dist/rpm/kde/systray-x.spec
+++ b/dist/rpm/kde/systray-x.spec
@@ -54,10 +54,10 @@ Requires:       libKF5Notifications5
 BuildRequires:  pkgconfig(x11)
 %if 0%{?fedora_version}
 Requires:       thunderbird >= 91
-Requires:       thunderbird < 137
+Requires:       thunderbird < 999
 %else
 Requires:       MozillaThunderbird >= 91
-Requires:       MozillaThunderbird < 137
+Requires:       MozillaThunderbird < 999
 %endif
 %if 0%{?suse_version} > 1600 || 0%{?fedora_version} > 39
 %if 0%{?fedora_version}
--- a/dist/rpm/minimal/systray-x-minimal.spec
+++ b/dist/rpm/minimal/systray-x-minimal.spec
@@ -43,10 +43,10 @@ BuildRequires:  libqt5-qtx11extras-devel
 BuildRequires:  pkgconfig(x11)
 %if 0%{?fedora_version} || 0%{?centos_version}
 Requires:       thunderbird >= 91
-Requires:       thunderbird < 137
+Requires:       thunderbird < 999
 %else
 Requires:       MozillaThunderbird >= 91
-Requires:       MozillaThunderbird < 137
+Requires:       MozillaThunderbird < 999
 %endif
 %if 0%{?suse_version} > 1600 || 0%{?fedora_version} > 39
 %if 0%{?fedora_version}
--- a/webext/manifest.json
+++ b/webext/manifest.json
@@ -10,7 +10,7 @@
     "gecko": {
       "id": "systray-x@Ximi1970",
       "strict_min_version": "91.0",
-      "strict_max_version": "136.*"
+      "strict_max_version": "998.*"
     }
   },
