From 58dcdd9338d965e54c8f03ce3d2757388d82b7a3 Mon Sep 17 00:00:00 2001
From: gustavo <gbuschle@hotmail.com>
Date: Thu, 11 Sep 2014 14:12:17 +0000
Subject: Add support for giflib 5.1

---
diff --git a/libbase/GnashImageGif.cpp b/libbase/GnashImageGif.cpp
index 84b09ce..2ae5ddb 100644
--- a/libbase/GnashImageGif.cpp
+++ b/libbase/GnashImageGif.cpp
@@ -120,7 +120,11 @@ GifInput::GifInput(std::shared_ptr<IOChannel> in)
 GifInput::~GifInput()
 {
     // Clean up allocated data.
-    DGifCloseFile(_gif);
+#if GIFLIB_MAJOR==5 && GIFLIB_MINOR==1
+	DGifCloseFile(_gif, 0);
+#else
+	DGifCloseFile(_gif);
+#endif
 }
 
 size_t
--
cgit v0.9.0.2
