--- gnash-0.8.10/libbase/accumulator.h.orig	2016-02-08 02:56:17.059694787 +0700
+++ gnash-0.8.10/libbase/accumulator.h	2016-02-08 02:57:58.375615249 +0700
@@ -61,6 +61,11 @@
     virtual unsigned min_tokens() const { return 0; }
     virtual unsigned max_tokens() const { return 0; }
 
+    #if BOOST_VERSION >= 105900
+    /// There are no tokens, so this is indifferent.
+    virtual bool adjacent_tokens_only() const { return false; }
+    #endif
+
     /// Accumulating from different sources is silly.
     virtual bool is_composing() const { return false; }
 
