diff -Naur SRB2-SRB2_release_2.2.15.orig/src/d_think.h SRB2-SRB2_release_2.2.15/src/d_think.h
--- SRB2-SRB2_release_2.2.15.orig/src/d_think.h	2025-01-25 23:35:40.000000000 +0100
+++ SRB2-SRB2_release_2.2.15/src/d_think.h	2026-03-22 10:30:16.645282573 +0100
@@ -30,9 +30,12 @@
 //
 typedef void (*actionf_v)();
 typedef void (*actionf_p1)(void *);
+struct mobj_s;
+typedef void (*actionf_m1)(struct mobj_s *);
 
 typedef union
 {
+	actionf_m1 acm1;
 	actionf_v acv;
 	actionf_p1 acp1;
 } actionf_t;
diff -Naur SRB2-SRB2_release_2.2.15.orig/src/doomtype.h SRB2-SRB2_release_2.2.15/src/doomtype.h
--- SRB2-SRB2_release_2.2.15.orig/src/doomtype.h	2025-01-25 23:35:40.000000000 +0100
+++ SRB2-SRB2_release_2.2.15/src/doomtype.h	2026-03-22 10:30:52.536666445 +0100
@@ -156,6 +156,7 @@
 #endif
 
 #ifndef __cplusplus
+#if ( __STDC_VERSION__ < 202000 )
 #ifndef _WIN32
 enum {false = 0, true = 1};
 #else
@@ -163,6 +164,7 @@
 #define true TRUE
 #endif
 #endif
+#endif
 
 /* 7.18.2.1  Limits of exact-width integer types */
 
