diff -Naur qiv/Makefile qiv.patched/Makefile
--- qiv/Makefile	2025-04-03 02:54:29.000000000 -0400
+++ qiv.patched/Makefile	2025-06-16 22:57:44.205086413 -0400
@@ -23,7 +23,7 @@

 # This sets the file extentions to filter on (other file types will be
 # skipped.) It should reflect whatever can be loaded via gdk-pixbuf-loader.
-EXTNS = GIF TIFF XPM PNG PPM PNM PGM PCX BMP EIM JPEG SVG WMF ICO WEBP HEIF
+EXTNS = GIF TIFF XPM PNG PPM PNM PGM PCX BMP EIM JPEG SVG WMF ICO WEBP HEIF XCF PSD ICNS TGA RAW JP2
 
 # Comment this line out if your system doesn't have getopt_long().
 GETOPT_LONG = -DHAVE_GETOPT_LONG
diff -Naur qiv/main.h qiv.patched/main.h
--- qiv/main.h	2025-04-03 02:54:29.000000000 -0400
+++ qiv.patched/main.h	2025-06-16 23:00:45.234069377 -0400
@@ -206,7 +206,22 @@
     ".webp",
 #endif
 #ifdef EXTN_HEIF
-    ".heic", ".heif",
+    ".heic", ".heif", ".avif",
+#endif
+#ifdef EXTN_XCF
+    ".xcf",
+#endif
+#ifdef EXTN_PSD
+    ".psd",
+#endif
+#ifdef EXTN_ICNS
+    ".icns",
+#endif
+#ifdef EXTN_JP2
+    ".jp2", ".jpc", ".jpx", ".j2k", ".jpf",
+#endif
+#ifdef EXTN_RAW
+   ".dng", ".cr2", ".crw", ".nef", ".orf", ".pef", ".arw", ".erf", ".mrw", ".raf",
 #endif
     NULL
 };
@@ -252,6 +267,28 @@
 #endif
 #ifdef EXTN_HEIF
   "ISO Media, HEIF Image",
+  "ISO Media, AVIF Image",
+#endif
+#ifdef EXTN_XCF
+  "GIMP XCF image data",
+#endif
+#ifdef EXTN_PSD
+  "Adobe Photoshop Image",
+#endif
+#ifdef EXTN_ICNS
+  "Mac OS X icon",
+#endif
+#ifdef EXTN_JP2
+  "JPEG 2000",
+#endif
+#ifdef EXTN_RAW
+  /* 20191231 bkw: some of these (dng, nef, erf, pef) show up as TIFF
+    with the file command, so we can't include them here. cr3 is "ISO Media",
+    which fails to work if added here. */
+  "Canon CIFF",
+  "Canon CR2",
+  "Olympus ORF",
+  "Minolta Dimage",
 #endif
  NULL
 };
