diff -Nur iscsitarget-0.4.17.orig/usr/Makefile iscsitarget-0.4.17/usr/Makefile
--- iscsitarget-0.4.17.orig/usr/Makefile	2008-06-18 18:10:19.000000000 -0500
+++ iscsitarget-0.4.17/usr/Makefile	2009-10-10 22:09:54.221991340 -0500
@@ -1,4 +1,4 @@
-CFLAGS += -O2 -fno-inline -Wall -Wstrict-prototypes -g -I../include
+CFLAGS += $(OPTS) -fno-inline -Wall -Wstrict-prototypes -g -I../include
 CFLAGS += -D_GNU_SOURCE # required for glibc >= 2.8
 PROGRAMS = ietd ietadm
 LIBS = -lcrypto
