--- iscsitarget-1.4.19-source/usr/Makefile	2008-06-19 00:10:19.000000000 +0100
+++ iscsitarget-1.4.19/usr/Makefile	2010-01-09 17:32:28.000000000 +0000
@@ -1,4 +1,4 @@
-CFLAGS += -O2 -fno-inline -Wall -Wstrict-prototypes -g -I../include
+CFLAGS += @ARCH@ -fno-inline -Wall -Wstrict-prototypes -g -I../include
 CFLAGS += -D_GNU_SOURCE # required for glibc >= 2.8
 PROGRAMS = ietd ietadm
 LIBS = -lcrypto
