diff -Nur NetworkManager-openconnect-0.8.1.orig//nm-openconnect-service.conf NetworkManager-openconnect-0.8.1/nm-openconnect-service.conf
--- NetworkManager-openconnect-0.8.1.orig//nm-openconnect-service.conf	2009-04-20 06:09:58.000000000 -0500
+++ NetworkManager-openconnect-0.8.1/nm-openconnect-service.conf	2010-09-12 11:57:41.100526093 -0500
@@ -6,10 +6,6 @@
 		<allow own="org.freedesktop.NetworkManager.openconnect"/>
 		<allow send_destination="org.freedesktop.NetworkManager.openconnect"/>
 	</policy>
-	<policy user="nm-openconnect">
-		<allow own="org.freedesktop.NetworkManager.openconnect"/>
-		<allow send_destination="org.freedesktop.NetworkManager.openconnect"/>
-	</policy>
 	<policy context="default">
 		<deny own="org.freedesktop.NetworkManager.openconnect"/>
 		<deny send_destination="org.freedesktop.NetworkManager.openconnect"/>
