diff -Nur cd-discid-0.9.orig/Makefile cd-discid-0.9/Makefile
--- cd-discid-0.9.orig/Makefile	2003-01-05 14:18:07.000000000 -0600
+++ cd-discid-0.9/Makefile	2007-02-01 09:35:54.000000000 -0600
@@ -9,7 +9,7 @@
 # Installation directories
 prefix = ${DESTDIR}/usr
 exec_prefix = ${prefix}
-mandir = ${prefix}/share/man/man1
+mandir = ${prefix}/man/man1
 bindir = ${exec_prefix}/bin
 etcdir = ${DESTDIR}/etc
 
@@ -32,9 +32,9 @@
 
 install: cd-discid
 	$(INSTALL) -d -m 755 $(bindir)
-	$(INSTALL) -s -m 755 -o 0 cd-discid $(bindir)
+	$(INSTALL) -s -m 755 cd-discid $(bindir)
 	$(INSTALL) -d -m 755 $(mandir)
-	$(INSTALL) -m 644 -o 0 cd-discid.1 $(mandir)
+	$(INSTALL) -m 644 cd-discid.1 $(mandir)
 
 tarball:
 	@cd .. && tar czvf cd-discid_$(VERSION).orig.tar.gz \
