--- wxVillaLib/ThumbnailFactory.cpp.org	2012-08-20 07:10:49.071537020 +0200
+++ wxVillaLib/ThumbnailFactory.cpp	2012-08-20 07:11:27.153489557 +0200
@@ -251,7 +251,7 @@
 		free(fnameEscaped);
 	wxString fname = filename;
 	if (info->mimeType == _T("concat"))
-		fname = filename.substr(7).BeforeFirst(wxT('|'));
+		fname = filename.Mid(7).BeforeFirst(wxT('|'));
 	info->mtime = wxFileName(fname).GetModificationTime();
 #endif
 	return info;
