diff -ru DVDStyler-3.2.1.orig/src/mediaenc_ffmpeg.cpp DVDStyler-3.2.1/src/mediaenc_ffmpeg.cpp
--- DVDStyler-3.2.1.orig/src/mediaenc_ffmpeg.cpp	2026-08-05 15:41:10.059018056 +0200
+++ DVDStyler-3.2.1/src/mediaenc_ffmpeg.cpp	2026-08-05 15:42:34.027342649 +0200
@@ -247,10 +247,12 @@
 }
 
 bool hasSampleFmt(AVCodec* codec, AVSampleFormat sample_fmt) {
-	if (codec != NULL && codec->sample_fmts != NULL) {
+	const AVSampleFormat *codec_sample_fmts = nullptr;
+	avcodec_get_supported_config(nullptr, codec, AV_CODEC_CONFIG_SAMPLE_FORMAT, 0, (const void**)&codec_sample_fmts, nullptr);
+	if (codec != NULL && codec_sample_fmts != NULL) {
 		int fIdx = 0;
-		while (codec->sample_fmts[fIdx] >= 0) {
-			if (codec->sample_fmts[fIdx] == sample_fmt)
+		while (codec_sample_fmts[fIdx] >= 0) {
+			if (codec_sample_fmts[fIdx] == sample_fmt)
 				return true;
 			fIdx++;
 		}
