From 67948b17aea120636e27c07ec8bb42dcdb0e6438 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Mon, 1 Jun 2026 20:57:17 +0200
Subject: [PATCH 4/4] added support of ffmpeg 8.0

---
 src/mediaenc_ffmpeg.cpp | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/mediaenc_ffmpeg.cpp b/src/mediaenc_ffmpeg.cpp
index 2c3e4ce..2274d7a 100644
--- a/src/mediaenc_ffmpeg.cpp
+++ b/src/mediaenc_ffmpeg.cpp
@@ -216,8 +216,13 @@ bool wxFfmpegMediaEncoder::addVideoStream(int codecId, VideoFormat videoFormat,
 
 	m_videoOutbuf = (uint8_t*) av_malloc(VIDEO_BUF_SIZE);
 
-	AVCPBProperties *props;
-	props = (AVCPBProperties*) av_stream_new_side_data(m_videoStm, AV_PKT_DATA_CPB_PROPERTIES, sizeof(*props));
+	AVPacketSideData* sd = av_packet_side_data_new(&m_videoStm->codecpar->coded_side_data,
+							&m_videoStm->codecpar->nb_coded_side_data,
+							AV_PKT_DATA_CPB_PROPERTIES,
+							sizeof(AVCPBProperties),
+							0);
+	if (!sd) abort();
+	AVCPBProperties *props = (AVCPBProperties*)sd->data;
 	props->buffer_size = VIDEO_BUF_SIZE;
 	props->max_bitrate = 0;
 	props->min_bitrate = 0;
-- 
2.54.0
