diff -Naur love-HEAD/src/modules/graphics/opengl/GLee.h love-HEAD.patched/src/modules/graphics/opengl/GLee.h
--- love-HEAD/src/modules/graphics/opengl/GLee.h	2011-05-04 12:00:10.000000000 -0400
+++ love-HEAD.patched/src/modules/graphics/opengl/GLee.h	2021-09-03 14:16:42.123563880 -0400
@@ -62,8 +62,9 @@
     #define GL_GLEXT_LEGACY
 	#include <OpenGL/gl.h>
 #else // GLX
-	#define __glext_h_  /* prevent glext.h from being included  */
-	#define __glxext_h_ /* prevent glxext.h from being included */
+    #define GL_GLEXT_LEGACY
+    #define GLX_GLEXT_LEGACY
+    #define GLX_GLXEXT_LEGACY
 	#define GLX_GLXEXT_PROTOTYPES
 	#include <GL/gl.h>
 	#include <GL/glx.h>
