--- ssvnc-1.0.30/Makefile.orig	2014-01-26 08:11:30.168836072 +0700
+++ ssvnc-1.0.30/Makefile	2014-01-26 08:12:03.465391347 +0700
@@ -24,7 +24,7 @@
 PREFIX  ?= /usr/local
 ROOT    ?=
 BIN      = bin
-LIB      = lib/ssvnc
+LIB     ?= lib/ssvnc
 MAN      = man
 MANDIR  ?= $(MAN)
 APPS    ?= share/applications
@@ -32,7 +32,7 @@
 
 
 VSRC = vnc_unixsrc
-JSRC = ultraftp
+JSRC ?= ultraftp
 PSRC = vncstorepw
 
 VIEWER  = $(VSRC)/vncviewer/vncviewer
