diff -Naur engauge.orig/src/pointmatch.cpp engauge/src/pointmatch.cpp
--- engauge.orig/src/pointmatch.cpp	2007-04-11 08:16:12.000000000 +0200
+++ engauge/src/pointmatch.cpp	2012-09-13 11:40:19.094978840 +0200
@@ -80,6 +80,7 @@
  *                                                                         *
  ***************************************************************************/
 
+#include <unistd.h>
 #include <qobject.h>
 #include <qdialog.h>
 #include <qlabel.h>
