--- fsvs-1.2.3/src/helper.c	2010-02-16 08:20:50.000000000 -0200
+++ fsvs-1.2.3_patched/src/helper.c	2011-04-05 10:56:59.000000000 -0300
@@ -1738,7 +1738,7 @@
 {
 	static int last=0;
 	/* Sadly GCC doesn't statically solve sizeof(rev)*log(10)/log(2) ... */
-	static char buffers[2][(int)(sizeof(rev)*3.32)+3];
+	static char buffers[2][(int)(sizeof(rev)*4)+3];
 
 	last++;
 	if (last>= sizeof(buffers)/sizeof(buffers[0])) last=0;
--- fsvs-1.2.3/src/options.c	2010-02-13 09:49:16.000000000 -0200
+++ fsvs-1.2.3_patched/src/options.c	2011-04-05 10:58:28.000000000 -0300
@@ -740,7 +740,7 @@
  * loop via \c getenv() over all options? */
 char *opt__variable_from_option(enum opt__settings_e which)
 {
-	static char buffer[ strlen(ENV_PREFIX) + 
+	static char buffer[ 5 + 
 		sizeof(opt__list[0].name) + 1] = ENV_PREFIX;
 	char * const target=buffer+strlen(ENV_PREFIX);
 	int i;
--- fsvs-1.2.3/src/status.c	2010-02-13 15:56:59.000000000 -0200
+++ fsvs-1.2.3_patched/src/status.c	2011-04-05 10:59:00.000000000 -0300
@@ -451,7 +451,7 @@
 	time_t now;
 	int print;
 	const int bar_chart_width=20;
-	static const char bar_chart[bar_chart_width+1]="###################>";
+	static const char bar_chart[20+1]="###################>";
 	float pct;
 
 
