diff -Naur lcal-2.1.0_orig/Makefile lcal-2.1.0/Makefile
--- lcal-2.1.0_orig/Makefile	2006-08-04 19:02:48.000000000 -0700
+++ lcal-2.1.0/Makefile	2011-01-01 18:03:23.492235946 -0800
@@ -103,7 +103,7 @@
 	MANDIR = $(DJDIR)/man/man1
 	CATDIR = $(DJDIR)/man/cat1
 else   # Unix
-	BINDIR = /usr/local/bin
+	BINDIR = /usr/bin
 	MANDIR = /usr/man/man1
 	CATDIR = /usr/man/cat1
 endif
@@ -192,12 +192,11 @@
 	groff -man -Tascii $(DOCDIR)/lcal.man >$(DOCDIR)/lcal-help.txt
 
 install:	$(EXECDIR)/$(LCAL) man
-	cp $(EXECDIR)/$(LCAL) $(BINDIR)
-	if [ -d $(MANDIR) ]; then \
-		cp $(DOCDIR)/lcal.man $(MANDIR)/lcal.1; \
-		$(PACK) $(MANDIR)/lcal.1; \
-	fi
-	if [ -d $(CATDIR) ]; then \
-		cp $(DOCDIR)/lcal.cat $(CATDIR)/lcal.1; \
-		$(PACK) $(CATDIR)/lcal.1; \
-	fi
+	mkdir -p $(DESTDIR)/$(BINDIR)
+	mkdir -p $(DESTDIR)/$(MANDIR)
+	mkdir -p $(DESTDIR)/$(CATDIR)
+	cp $(EXECDIR)/$(LCAL) $(DESTDIR)/$(BINDIR)
+	cp $(DOCDIR)/lcal.man $(DESTDIR)/$(MANDIR)/lcal.1
+	$(PACK) $(DESTDIR)/$(MANDIR)/lcal.1              
+	cp $(DOCDIR)/lcal.cat $(DESTDIR)/$(CATDIR)/lcal.1
+	$(PACK) $(DESTDIR)/$(CATDIR)/lcal.1              
