Description: Fix a few typos in the man & help docs.
Author: B. Watson <yahlcru@gmail.com>

--

diff -Naur lv-4.51.orig/lv.1 lv-4.51.orig.patched/lv.1
--- lv-4.51.orig/lv.1	2020-11-12 01:44:33.656105671 -0500
+++ lv-4.51.orig.patched/lv.1	2020-11-12 01:45:58.092097996 -0500
@@ -47,7 +47,7 @@
 but also as a coding-system translation filter
 like \fInkf\fP (1) and \fItcs\fP (1).
 .IP "\fIMultilingual regular expressions / Multilingual grep\fP"
-lv can recognize multi-bytes patterns as regular expressions,
+lv can recognize multi-byte patterns as regular expressions,
 and lv also provides multilingual \fIgrep\fP (1) functionality
 by giving it another name, \fIlgrep\fP.
 Pattern matching is conducted in the charset level,
@@ -203,7 +203,7 @@
 Allow physical lines of each logical line printed on the screen
 to be concatenated for cut and paste after screen refresh
 .IP "-s"
-Force old pages to be swept out from the screen Smoothly
+Force old pages to be swept out from the screen smoothly
 .IP "-u"
 Unify several character sets, eg. JIS X0208 and C6226.
 In addition, lv equates ISO 646 variants,
@@ -237,7 +237,7 @@
 Show this help
 .SH "CONFIGURATION"
 Options can be described in configuration file ``.lv'' (``_lv'' on MSDOS)
-located at you HOME directory.  If and only if you use MSDOS, you can locate
+located in your HOME directory.  If and only if you use MSDOS, you can locate
 ``_lv'' at current working directory.
 They can be also described in the environment variable LV.
 Every configuration will be overloaded in this order if there is. Command line
diff -Naur lv-4.51.orig/lv.hlp lv-4.51.orig.patched/lv.hlp
--- lv-4.51.orig/lv.hlp	2020-11-12 01:44:33.635105673 -0500
+++ lv-4.51.orig.patched/lv.hlp	2020-11-12 01:45:58.098097995 -0500
@@ -37,7 +37,7 @@
               _t_c_s (1).
 
        _M_u_l_t_i_l_i_n_g_u_a_l _r_e_g_u_l_a_r _e_x_p_r_e_s_s_i_o_n_s _/ _M_u_l_t_i_l_i_n_g_u_a_l _g_r_e_p
-              lv  can  recognize  multi-bytes patterns as regular expressions,
+              lv  can  recognize  multi-byte patterns as regular expressions,
               and lv also provides multilingual _g_r_e_p (1) functionality by giv-
               ing  it  another  name, _l_g_r_e_p.  Pattern matching is conducted in
               the charset level, so an EUC fragment, for example, can be found
@@ -183,7 +183,7 @@
        -l     Allow physical lines of each logical line printed on the  screen
               to be concatenated for cut and paste after screen refresh
 
-       -s     Force old pages to be swept out from the screen Smoothly
+       -s     Force old pages to be swept out from the screen smoothly
 
        -u     Unify several character sets, eg. JIS X0208 and C6226.  In addi-
               tion, lv equates ISO 646  variants,  eg.  JIS  X0201-Roman,  and
@@ -219,7 +219,7 @@
 
 CCOONNFFIIGGUURRAATTIIOONN
        Options  can  be  described  in  configuration file ``.lv'' (``_lv'' on
-       MSDOS) located at you HOME directory.  If and only if  you  use  MSDOS,
+       MSDOS) located in your HOME directory.  If and only if  you  use  MSDOS,
        you  can locate ``_lv'' at current working directory.  They can be also
        described in the environment variable LV.  Every configuration will  be
        overloaded  in  this order if there is. Command line options are always
