--- youtube-dl-2018.06.25/setup.py.orig	2018-06-30 08:56:35.007072725 +0200
+++ youtube-dl-2018.06.25/setup.py	2018-06-30 08:56:54.365073186 +0200
@@ -59,10 +59,10 @@
     params = py2exe_params
 else:
     files_spec = [
-        ('etc/bash_completion.d', ['youtube-dl.bash-completion']),
-        ('etc/fish/completions', ['youtube-dl.fish']),
-        ('share/doc/youtube_dl', ['README.txt']),
-        ('share/man/man1', ['youtube-dl.1'])
+        ('/etc/bash_completion.d', ['youtube-dl.bash-completion']),
+        ('/etc/fish/completions', ['youtube-dl.fish']),
+        ('doc/youtube_dl', ['README.txt']),
+        ('man/man1', ['youtube-dl.1'])
     ]
     root = os.path.dirname(os.path.abspath(__file__))
     data_files = []
