--- youtube-dl-2015.02.21/setup.orig.py	2015-02-23 13:12:26.419954401 +0200
+++ youtube-dl-2015.02.21/setup.py	2015-02-23 13:13:23.668238699 +0200
@@ -46,10 +46,10 @@
     params = py2exe_params
 else:
     files_spec = [
-        ('etc/bash_completion.d', ['youtube-dl.bash-completion']),
-        ('etc/fish/completions', ['youtube-dl.fish']),
-        ('share/doc/youtube_dl', ['README.txt']),
-        ('share/man/man1', ['youtube-dl.1'])
+        ('/etc/bash_completion.d', ['youtube-dl.bash-completion']),
+        ('/etc/fish/completions', ['youtube-dl.fish']),
+        ('/usr/doc/youtube_dl', ['README.txt']),
+        ('/usr/man/man1', ['youtube-dl.1'])
     ]
     root = os.path.dirname(os.path.abspath(__file__))
     data_files = []
