--- youtube-dl-2018.05.01/Makefile.orig	2018-05-06 18:08:39.581459715 +0200
+++ youtube-dl-2018.05.01/Makefile	2018-05-06 18:10:24.291462210 +0200
@@ -5,11 +5,11 @@
 	find . -name "*.pyc" -delete
 	find . -name "*.class" -delete

-PREFIX ?= /usr/local
+PREFIX ?= /usr
 BINDIR ?= $(PREFIX)/bin
 MANDIR ?= $(PREFIX)/man
 SHAREDIR ?= $(PREFIX)/share
-PYTHON ?= /usr/bin/env python
+PYTHON ?= /usr/bin/python

 # set SYSCONFDIR to /etc if PREFIX=/usr or PREFIX=/usr/local
 SYSCONFDIR = $(shell if [ $(PREFIX) = /usr -o $(PREFIX) = /usr/local ]; then echo /etc; else echo $(PREFIX)/etc; fi)
@@ -55,7 +55,7 @@

 .PHONY: all clean install test tar bash-completion pypi-files zsh-completion fish-completion ot offlinetest codetest supportedsites

-pypi-files: youtube-dl.bash-completion README.txt youtube-dl.1 youtube-dl.fish
+pypi-files: youtube-dl.bash-completion README.md youtube-dl.1 youtube-dl.fish

 youtube-dl: youtube_dl/*.py youtube_dl/*/*.py
 mkdir -p zip
@@ -85,11 +85,11 @@
 	$(PYTHON) devscripts/make_supportedsites.py docs/supportedsites.md

 README.txt: README.md
-	pandoc -f $(MARKDOWN) -t plain README.md -o README.txt
+	#pandoc -f $(MARKDOWN) -t plain README.md -o README.txt

 youtube-dl.1: README.md
 	$(PYTHON) devscripts/prepare_manpage.py youtube-dl.1.temp.md
-	pandoc -s -f $(MARKDOWN) -t man youtube-dl.1.temp.md -o youtube-dl.1
+	#pandoc -s -f $(MARKDOWN) -t man youtube-dl.1.temp.md -o youtube-dl.1
 	rm -f youtube-dl.1.temp.md

 youtube-dl.bash-completion: youtube_dl/*.py youtube_dl/*/*.py devscripts/bash-completion.in
--- youtube-dl-2018.05.01/setup.py.orig	2018-05-06 18:08:48.362459924 +0200
+++ youtube-dl-2018.05.01/setup.py	2018-05-06 18:11:02.935463131 +0200
@@ -59,10 +59,10 @@
     params = py2exe_params
 else:
     files_spec = [
-        ('etc/bash_completion.d', ['youtube-dl.bash-completion']),
-        ('etc/fish/completions', ['youtube-dl.fish']),
-        ('share/doc/youtube_dl', ['README.txt']),
-        ('share/man/man1', ['youtube-dl.1'])
+        ('/etc/bash_completion.d', ['youtube-dl.bash-completion']),
+        ('/etc/fish/completions', ['youtube-dl.fish']),
+        ('usr/doc/youtube_dl', ['README.txt']),
+        ('usr/man/man1', ['youtube-dl.1'])
     ]
     root = os.path.dirname(os.path.abspath(__file__))
     data_files = []
