From ae298e5c0cf64d8d9d01f3b2eddfaa783f85ecf9 Mon Sep 17 00:00:00 2001
From: Robby Workman <rworkman@slackware.com>
Date: Mon, 11 Nov 2013 17:24:57 -0600
Subject: [PATCH] virtinst/osdict.py: Add Slackware to OS choices

---
 virtinst/osdict.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/virtinst/osdict.py b/virtinst/osdict.py
index 5dd727a..82ffef1 100644
--- a/virtinst/osdict.py
+++ b/virtinst/osdict.py
@@ -276,6 +276,8 @@ _add_var("mageia1", "Mageia 1 (or later)", urldistro="mandriva", supported=True,
 
 _add_var("altlinux", "ALT Linux (or later)", urldistro="altlinux", supported=True, virtiodisk=True, virtionet=True, inputtype="tablet", inputbus="usb", parent="linux")
 
+_add_var("slackware14.1", "Slackware 14.1 (or later)", parent="linux", supported=True, virtiodisk=True, virtionet=True, inputtype="tablet", inputbus="usb")
+
 _add_var("debianetch", "Debian Etch", urldistro="debian", sortby="debian4", parent="linux")
 _add_var("debianlenny", "Debian Lenny", sortby="debian5", supported=True, virtiodisk=True, virtionet=True, parent="debianetch")
 _add_var("debiansqueeze", "Debian Squeeze", sortby="debian6", virtiodisk=True, virtionet=True, inputtype="tablet", inputbus="usb", parent="debianlenny")
-- 
1.8.4

