diff -Nur virt-manager-0.10.0.orig/virtinst/osdict.py virt-manager-0.10.0/virtinst/osdict.py
--- virt-manager-0.10.0.orig/virtinst/osdict.py	2013-06-13 14:10:00.000000000 -0500
+++ virt-manager-0.10.0/virtinst/osdict.py	2013-11-11 17:47:51.907305561 -0600
@@ -619,7 +619,15 @@
             VIDEO : VMVGA_VIDEO,
         },
     },
-
+    "slackware": {
+        "label": "Slackware",
+        "distro": "slackware",
+        "supported": True,
+        "devices" : {
+            DISK : VIRTIO_DISK,
+            NET  : VIRTIO_NET,
+        },
+    },
     "generic24": {
         "label": "Generic 2.4.x kernel"
    },
