--- a/misc/Makefile
+++ b/misc/Makefile
@@ -29,7 +29,8 @@
 	@printf "make clean: 	        Remove build directories.\n"
 
 install:
-	python3 setup.py install --root=$(DESTDIR)/ --prefix=$(PREFIX) --record=install_log.txt
+	python3 -m build --no-isolation
+	python3 -m installer -d $(DESTDIR) dist/*.whl
 	install -Dm644 misc/vimiv.desktop $(APPDIR)/vimiv.desktop
 	install -Dm644 misc/org.karlch.vimiv.qt.metainfo.xml $(METAINFODIR)/org.karlch.vimiv.qt.metainfo.xml
 	install -Dm644 LICENSE $(LICENSEDIR)/vimiv/LICENSE
