diff -Naur hping3-20051105.orig/configure hping3-20051105/configure
--- hping3-20051105.orig/configure	2004-06-04 08:39:10.000000000 +0200
+++ hping3-20051105/configure	2008-12-17 23:16:57.000000000 +0100
@@ -63,7 +63,7 @@
 #
 for TCLPATH_TRY in "/usr/bin/" "/usr/local/bin/" "/bin/"
 do
-	for TCLVER_TRY in "8.4" "8.3" "8.2" "8.1" "8.0"
+	for TCLVER_TRY in "8.5" "8.4" "8.3" "8.2" "8.1" "8.0"
 	do
 		if [ -z $TCLSH ]
 		then
diff -Naur hping3-20051105.orig/libpcap_stuff.c hping3-20051105/libpcap_stuff.c
--- hping3-20051105.orig/libpcap_stuff.c	2004-04-10 01:38:56.000000000 +0200
+++ hping3-20051105/libpcap_stuff.c	2008-12-17 23:20:52.000000000 +0100
@@ -17,7 +17,7 @@
 #include <stdlib.h>
 #include <sys/ioctl.h>
 #include <pcap.h>
-#include <net/bpf.h>
+#include <pcap-bpf.h>
 
 #include "globals.h"
 
diff -Naur hping3-20051105.orig/script.c hping3-20051105/script.c
--- hping3-20051105.orig/script.c	2004-05-29 08:48:13.000000000 +0200
+++ hping3-20051105/script.c	2008-12-17 23:21:15.000000000 +0100
@@ -24,7 +24,7 @@
 
 #include <sys/ioctl.h>
 #include <pcap.h>
-#include <net/bpf.h>
+#include <pcap-bpf.h>
 
 #include "release.h"
 #include "hping2.h"
