--- auitabart.h	2021-10-22 17:19:37.000000000 +0100
+++ auitabart.h.new	2023-02-19 11:05:12.071625681 +0000
@@ -6,7 +6,7 @@
 class AuiTabArt : public wxAuiGenericTabArt {
 public:
 	wxSize GetTabSize(wxDC& dc, wxWindow* wnd, const wxString& caption, const wxBitmap& bitmap,
-	                  bool active, int close_button_state, int* x_extent) override {
+	                  bool active, int close_button_state, int* x_extent) {
 		return wxAuiGenericTabArt::GetTabSize(dc, wnd, caption, bitmap, active, close_button_state,
 		                                      x_extent) +
 		       wxSize(0, 2);
