diff --git a/Demo/Makefile.am b/Demo/Makefile.am
new file mode 100644
--- /dev/null
+++ b/Demo/Makefile.am
@@ -0,0 +1,273 @@
+AM_CPPFLAGS = -I$(top_srcdir)/Include $(SUITESPARSECONFIG_CFLAGS)
+LDADD = $(top_builddir)/Source/libcxsparse.la
+
+check_PROGRAMS = \
+	cs_demo1 \
+	cs_demo2 \
+	cs_demo3 \
+	cs_di_demo1 \
+	cs_di_demo2 \
+	cs_di_demo3 \
+	cs_dl_demo1 \
+	cs_dl_demo2 \
+	cs_dl_demo3 \
+	cs_ci_demo1 \
+	cs_ci_demo2 \
+	cs_ci_demo3 \
+	cs_cl_demo1 \
+	cs_cl_demo2 \
+	cs_cl_demo3 \
+	cs_idemo \
+	cs_ldemo
+
+check_LTLIBRARIES = \
+	libcs_demo.la \
+	libcs_di_demo.la \
+	libcs_dl_demo.la \
+	libcs_ci_demo.la \
+	libcs_cl_demo.la
+
+cs_demo1_SOURCES = cs_demo1.c
+cs_demo2_SOURCES = cs_demo2.c
+cs_demo3_SOURCES = cs_demo3.c
+libcs_demo_la_SOURCES = cs_demo.c cs_demo.h
+cs_demo2_LDADD = libcs_demo.la $(LDADD)
+cs_demo3_LDADD = libcs_demo.la $(LDADD)
+
+cs_di_demo1_SOURCES = cs_di_demo1.c
+cs_di_demo2_SOURCES = cs_di_demo2.c
+cs_di_demo3_SOURCES = cs_di_demo3.c
+libcs_di_demo_la_SOURCES = cs_di_demo.c cs_di_demo.h
+cs_di_demo2_LDADD = libcs_di_demo.la $(LDADD)
+cs_di_demo3_LDADD = libcs_di_demo.la $(LDADD)
+
+cs_dl_demo1_SOURCES = cs_dl_demo1.c
+cs_dl_demo2_SOURCES = cs_dl_demo2.c
+cs_dl_demo3_SOURCES = cs_dl_demo3.c
+libcs_dl_demo_la_SOURCES = cs_dl_demo.c cs_dl_demo.h
+cs_dl_demo2_LDADD = libcs_dl_demo.la $(LDADD)
+cs_dl_demo3_LDADD = libcs_dl_demo.la $(LDADD)
+cs_dl_demo1_CPPFLAGS = $(AM_CPPFLAGS) -DCS_LONG
+cs_dl_demo2_CPPFLAGS = $(AM_CPPFLAGS) -DCS_LONG
+cs_dl_demo3_CPPFLAGS = $(AM_CPPFLAGS) -DCS_LONG
+libcs_dl_demo_la_CPPFLAGS = $(AM_CPPFLAGS) -DCS_LONG
+
+cs_ci_demo1_SOURCES = cs_ci_demo1.c
+cs_ci_demo2_SOURCES = cs_ci_demo2.c
+cs_ci_demo3_SOURCES = cs_ci_demo3.c
+libcs_ci_demo_la_SOURCES = cs_ci_demo.c cs_ci_demo.h
+libcs_ci_demo_la_LIBADD = -lm
+cs_ci_demo2_LDADD = libcs_ci_demo.la $(LDADD)
+cs_ci_demo3_LDADD = libcs_ci_demo.la $(LDADD)
+cs_ci_demo1_CPPFLAGS = $(AM_CPPFLAGS) -DCS_COMPLEX
+cs_ci_demo2_CPPFLAGS = $(AM_CPPFLAGS) -DCS_COMPLEX
+cs_ci_demo3_CPPFLAGS = $(AM_CPPFLAGS) -DCS_COMPLEX
+libcs_ci_demo_la_CPPFLAGS = $(AM_CPPFLAGS) -DCS_COMPLEX
+
+cs_cl_demo1_SOURCES = cs_cl_demo1.c
+cs_cl_demo2_SOURCES = cs_cl_demo2.c
+cs_cl_demo3_SOURCES = cs_cl_demo3.c
+libcs_cl_demo_la_SOURCES = cs_cl_demo.c cs_cl_demo.h
+libcs_cl_demo_la_LIBADD = -lm
+cs_cl_demo2_LDADD = libcs_cl_demo.la $(LDADD)
+cs_cl_demo3_LDADD = libcs_cl_demo.la $(LDADD)
+cs_cl_demo1_CPPFLAGS = $(AM_CPPFLAGS) -DCS_LONG -DCS_COMPLEX
+cs_cl_demo2_CPPFLAGS = $(AM_CPPFLAGS) -DCS_LONG -DCS_COMPLEX
+cs_cl_demo3_CPPFLAGS = $(AM_CPPFLAGS) -DCS_LONG -DCS_COMPLEX
+libcs_cl_demo_la_CPPFLAGS = $(AM_CPPFLAGS) -DCS_LONG -DCS_COMPLEX
+
+cs_idemo_SOURCES = cs_idemo.c
+
+cs_ldemo_SOURCES = cs_ldemo.c
+cs_ldemo_CPPFLAGS = $(AM_CPPFLAGS) -DCS_LONG
+
+if !CODE_COVERAGE_ENABLED
+MATRICES = \
+	$(top_srcdir)/Matrix/ash219 \
+	$(top_srcdir)/Matrix/bcsstk01 \
+	$(top_srcdir)/Matrix/bcsstk16 \
+	$(top_srcdir)/Matrix/c4 \
+	$(top_srcdir)/Matrix/c_mbeacxc \
+	$(top_srcdir)/Matrix/c_west0067 \
+	$(top_srcdir)/Matrix/fs_183_1 \
+	$(top_srcdir)/Matrix/lp_afiro \
+	$(top_srcdir)/Matrix/mbeacxc \
+	$(top_srcdir)/Matrix/mhd1280b \
+	$(top_srcdir)/Matrix/neumann \
+	$(top_srcdir)/Matrix/qc324 \
+	$(top_srcdir)/Matrix/t1 \
+	$(top_srcdir)/Matrix/t2 \
+	$(top_srcdir)/Matrix/t3 \
+	$(top_srcdir)/Matrix/t4 \
+	$(top_srcdir)/Matrix/west0067 \
+	$(top_srcdir)/Matrix/young1c
+
+check-local: $(check_PROGRAMS) $(MATRICES)
+	-./cs_demo1 < $(top_srcdir)/Matrix/t1
+	-./cs_demo2 < $(top_srcdir)/Matrix/t1
+	-./cs_demo2 < $(top_srcdir)/Matrix/fs_183_1
+	-./cs_demo2 < $(top_srcdir)/Matrix/west0067
+	-./cs_demo2 < $(top_srcdir)/Matrix/lp_afiro
+	-./cs_demo2 < $(top_srcdir)/Matrix/ash219
+	-./cs_demo2 < $(top_srcdir)/Matrix/mbeacxc
+	-./cs_demo2 < $(top_srcdir)/Matrix/bcsstk01
+	-./cs_demo3 < $(top_srcdir)/Matrix/bcsstk01
+	-./cs_demo2 < $(top_srcdir)/Matrix/bcsstk16
+	-./cs_demo3 < $(top_srcdir)/Matrix/bcsstk16
+	-./cs_di_demo1 < $(top_srcdir)/Matrix/t1
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/t1
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/fs_183_1
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/west0067
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/lp_afiro
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/ash219
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/mbeacxc
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/bcsstk01
+	-./cs_di_demo3 < $(top_srcdir)/Matrix/bcsstk01
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/bcsstk16
+	-./cs_di_demo3 < $(top_srcdir)/Matrix/bcsstk16
+	-./cs_dl_demo1 < $(top_srcdir)/Matrix/t1
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/t1
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/fs_183_1
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/west0067
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/lp_afiro
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/ash219
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/mbeacxc
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/bcsstk01
+	-./cs_dl_demo3 < $(top_srcdir)/Matrix/bcsstk01
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/bcsstk16
+	-./cs_dl_demo3 < $(top_srcdir)/Matrix/bcsstk16
+	-./cs_ci_demo1 < $(top_srcdir)/Matrix/t2
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/t2
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/t3
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/t4
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/c_west0067
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/c_mbeacxc
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/young1c
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/qc324
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/neumann
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/c4
+	-./cs_ci_demo3 < $(top_srcdir)/Matrix/c4
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/mhd1280b
+	-./cs_ci_demo3 < $(top_srcdir)/Matrix/mhd1280b
+	-./cs_cl_demo1 < $(top_srcdir)/Matrix/t2
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/t2
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/t3
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/t4
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/c_west0067
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/c_mbeacxc
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/young1c
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/qc324
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/neumann
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/c4
+	-./cs_cl_demo3 < $(top_srcdir)/Matrix/c4
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/mhd1280b
+	-./cs_cl_demo3 < $(top_srcdir)/Matrix/mhd1280b
+	-./cs_idemo < $(top_srcdir)/Matrix/t2
+	-./cs_ldemo < $(top_srcdir)/Matrix/t2
+
+else
+MATRICES = \
+	$(top_srcdir)/Tcov/czero \
+	$(top_srcdir)/Tcov/nil \
+	$(top_srcdir)/Tcov/zero \
+	$(top_srcdir)/Matrix/ash219 \
+	$(top_srcdir)/Matrix/bcsstk01 \
+	$(top_srcdir)/Matrix/bcsstk16 \
+	$(top_srcdir)/Matrix/c4 \
+	$(top_srcdir)/Matrix/c_ibm32a \
+	$(top_srcdir)/Matrix/c_ibm32b \
+	$(top_srcdir)/Matrix/c_mbeacxc \
+	$(top_srcdir)/Matrix/c_west0067 \
+	$(top_srcdir)/Matrix/fs_183_1 \
+	$(top_srcdir)/Matrix/ibm32a \
+	$(top_srcdir)/Matrix/ibm32b \
+	$(top_srcdir)/Matrix/lp_afiro \
+	$(top_srcdir)/Matrix/mbeacxc \
+	$(top_srcdir)/Matrix/mhd1280b \
+	$(top_srcdir)/Matrix/neumann \
+	$(top_srcdir)/Matrix/qc324 \
+	$(top_srcdir)/Matrix/t1 \
+	$(top_srcdir)/Matrix/t2 \
+	$(top_srcdir)/Matrix/t3 \
+	$(top_srcdir)/Matrix/t4 \
+	$(top_srcdir)/Matrix/west0067 \
+	$(top_srcdir)/Matrix/young1c
+
+check-local: $(check_PROGRAMS) $(MATRICES)
+	-./cs_idemo < $(top_srcdir)/Matrix/t2
+	-./cs_ldemo < $(top_srcdir)/Matrix/t2
+	-./cs_di_demo1 < $(top_srcdir)/Matrix/t1
+	-./cs_di_demo1 < $(top_srcdir)/Tcov/nil
+	-./cs_di_demo1 < $(top_srcdir)/Tcov/zero
+	-./cs_di_demo2 < $(top_srcdir)/Tcov/nil
+	-./cs_di_demo2 < $(top_srcdir)/Tcov/zero
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/t1
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/bcsstk01
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/fs_183_1
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/west0067
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/lp_afiro
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/ash219
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/mbeacxc
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/bcsstk16
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/ibm32a
+	-./cs_di_demo2 < $(top_srcdir)/Matrix/ibm32b
+	-./cs_di_demo3 < $(top_srcdir)/Tcov/nil
+	-./cs_di_demo3 < $(top_srcdir)/Matrix/bcsstk01
+	-./cs_di_demo3 < $(top_srcdir)/Matrix/bcsstk16
+	-./cs_dl_demo1 < $(top_srcdir)/Matrix/t1
+	-./cs_dl_demo1 < $(top_srcdir)/Tcov/nil
+	-./cs_dl_demo1 < $(top_srcdir)/Tcov/zero
+	-./cs_dl_demo2 < $(top_srcdir)/Tcov/nil
+	-./cs_dl_demo2 < $(top_srcdir)/Tcov/zero
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/t1
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/bcsstk01
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/fs_183_1
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/west0067
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/lp_afiro
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/ash219
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/mbeacxc
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/bcsstk16
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/ibm32a
+	-./cs_dl_demo2 < $(top_srcdir)/Matrix/ibm32b
+	-./cs_dl_demo3 < $(top_srcdir)/Tcov/nil
+	-./cs_dl_demo3 < $(top_srcdir)/Matrix/bcsstk01
+	-./cs_dl_demo3 < $(top_srcdir)/Matrix/bcsstk16
+	-./cs_ci_demo1 < $(top_srcdir)/Matrix/t2
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/t2
+	-./cs_ci_demo1 < $(top_srcdir)/Tcov/czero
+	-./cs_ci_demo2 < $(top_srcdir)/Tcov/czero
+	-./cs_ci_demo1 < $(top_srcdir)/Matrix/t3
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/t3
+	-./cs_ci_demo1 < $(top_srcdir)/Matrix/t4
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/t4
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/c_west0067
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/c_mbeacxc
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/c_ibm32a
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/c_ibm32b
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/young1c
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/qc324
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/neumann
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/c4
+	-./cs_ci_demo3 < $(top_srcdir)/Matrix/c4
+	-./cs_ci_demo2 < $(top_srcdir)/Matrix/mhd1280b
+	-./cs_ci_demo3 < $(top_srcdir)/Matrix/mhd1280b
+	-./cs_cl_demo1 < $(top_srcdir)/Matrix/t2
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/t2
+	-./cs_cl_demo1 < $(top_srcdir)/Tcov/czero
+	-./cs_cl_demo2 < $(top_srcdir)/Tcov/czero
+	-./cs_cl_demo1 < $(top_srcdir)/Matrix/t3
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/t3
+	-./cs_cl_demo1 < $(top_srcdir)/Matrix/t4
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/t4
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/c_west0067
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/c_mbeacxc
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/c_ibm32a
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/c_ibm32b
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/young1c
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/qc324
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/neumann
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/c4
+	-./cs_cl_demo3 < $(top_srcdir)/Matrix/c4
+	-./cs_cl_demo2 < $(top_srcdir)/Matrix/mhd1280b
+	-./cs_cl_demo3 < $(top_srcdir)/Matrix/mhd1280b
+endif
diff --git a/Include/Makefile.am b/Include/Makefile.am
new file mode 100644
--- /dev/null
+++ b/Include/Makefile.am
@@ -0,0 +1 @@
+pkginclude_HEADERS = cs.h
diff --git a/Makefile.am b/Makefile.am
new file mode 100644
--- /dev/null
+++ b/Makefile.am
@@ -0,0 +1,9 @@
+ACLOCAL_AMFLAGS = -I m4
+
+SUBDIRS = Include Source Matrix Demo Tcov
+EXTRA_DIST = README.txt Doc/ChangeLog Doc/License.txt Doc/lesser.txt
+
+pkgconfig_DATA = cxsparse.pc
+
+CODE_COVERAGE_BRANCH_COVERAGE = 1
+@CODE_COVERAGE_RULES@
diff --git a/Matrix/Makefile.am b/Matrix/Makefile.am
new file mode 100644
--- /dev/null
+++ b/Matrix/Makefile.am
@@ -0,0 +1,23 @@
+dist_noinst_DATA = \
+	ash219 \
+	bcsstk01 \
+	bcsstk16 \
+	c4 \
+	c_ibm32a \
+	c_ibm32b \
+	c_mbeacxc \
+	c_west0067 \
+	fs_183_1 \
+	ibm32a \
+	ibm32b \
+	lp_afiro \
+	mbeacxc \
+	mhd1280b \
+	neumann \
+	qc324 \
+	t1 \
+	t2 \
+	t3 \
+	t4 \
+	west0067 \
+	young1c
diff --git a/Source/Makefile.am b/Source/Makefile.am
new file mode 100644
--- /dev/null
+++ b/Source/Makefile.am
@@ -0,0 +1,101 @@
+AM_CPPFLAGS = -I$(top_srcdir)/Include $(SUITESPARSECONFIG_CFLAGS)
+
+CS_SOURCE = \
+	cs_add.c \
+	cs_amd.c \
+	cs_chol.c \
+	cs_cholsol.c \
+	cs_compress.c \
+	cs_counts.c \
+	cs_cumsum.c \
+	cs_dfs.c \
+	cs_dmperm.c \
+	cs_droptol.c \
+	cs_dropzeros.c \
+	cs_dupl.c \
+	cs_entry.c \
+	cs_ereach.c \
+	cs_etree.c \
+	cs_fkeep.c \
+	cs_gaxpy.c \
+	cs_happly.c \
+	cs_house.c \
+	cs_ipvec.c \
+	cs_leaf.c \
+	cs_load.c \
+	cs_lsolve.c \
+	cs_ltsolve.c \
+	cs_lu.c \
+	cs_lusol.c \
+	cs_maxtrans.c \
+	cs_multiply.c \
+	cs_norm.c \
+	cs_permute.c \
+	cs_pinv.c \
+	cs_post.c \
+	cs_print.c \
+	cs_pvec.c \
+	cs_qr.c \
+	cs_qrsol.c \
+	cs_randperm.c \
+	cs_reach.c \
+	cs_scatter.c \
+	cs_scc.c \
+	cs_schol.c \
+	cs_spsolve.c \
+	cs_sqr.c \
+	cs_symperm.c \
+	cs_tdfs.c \
+	cs_transpose.c \
+	cs_updown.c \
+	cs_usolve.c \
+	cs_util.c \
+	cs_utsolve.c
+
+if !CODE_COVERAGE_ENABLED
+CS_SOURCE += \
+	cs_malloc.c
+else
+CS_SOURCE += \
+	$(top_srcdir)/Tcov/cstcov_malloc_test.c \
+	$(top_srcdir)/Tcov/cstcov_malloc_test.h
+endif
+
+lib_LTLIBRARIES = \
+	libcxsparse.la
+
+noinst_LTLIBRARIES = \
+	libcxsparse_di.la \
+	libcxsparse_dl.la \
+	libcxsparse_ci.la \
+	libcxsparse_cl.la \
+	libcs_convert.la
+
+libcxsparse_di_la_SOURCES = $(CS_SOURCE)
+libcxsparse_di_la_LIBADD = -lm $(CODE_COVERAGE_LIBS)
+libcxsparse_di_la_CPPFLAGS = $(AM_CPPFLAGS) $(CODE_COVERAGE_CPPFLAGS)
+libcxsparse_di_la_CFLAGS = $(AM_CFLAGS) $(CODE_COVERAGE_CFLAGS)
+
+libcxsparse_dl_la_SOURCES = $(CS_SOURCE)
+libcxsparse_dl_la_LIBADD = -lm $(CODE_COVERAGE_LIBS)
+libcxsparse_dl_la_CPPFLAGS = $(AM_CPPFLAGS) $(CODE_COVERAGE_CPPFLAGS) -DCS_LONG
+libcxsparse_dl_la_CFLAGS = $(AM_CFLAGS) $(CODE_COVERAGE_CFLAGS)
+
+libcxsparse_ci_la_SOURCES = $(CS_SOURCE)
+libcxsparse_ci_la_LIBADD = -lm $(CODE_COVERAGE_LIBS)
+libcxsparse_ci_la_CPPFLAGS = $(AM_CPPFLAGS) $(CODE_COVERAGE_CPPFLAGS) -DCS_COMPLEX
+libcxsparse_ci_la_CFLAGS = $(AM_CFLAGS) $(CODE_COVERAGE_CFLAGS)
+
+libcxsparse_cl_la_SOURCES = $(CS_SOURCE)
+libcxsparse_cl_la_LIBADD = -lm $(CODE_COVERAGE_LIBS)
+libcxsparse_cl_la_CPPFLAGS = $(AM_CPPFLAGS) $(CODE_COVERAGE_CPPFLAGS) -DCS_LONG -DCS_COMPLEX
+libcxsparse_cl_la_CFLAGS = $(AM_CFLAGS) $(CODE_COVERAGE_CFLAGS)
+
+libcs_convert_la_SOURCES = cs_convert.c
+libcs_convert_la_LIBADD = $(CODE_COVERAGE_LIBS)
+libcs_convert_la_CPPFLAGS = $(AM_CPPFLAGS) $(CODE_COVERAGE_CPPFLAGS)
+libcs_convert_la_CFLAGS = $(AM_CFLAGS) $(CODE_COVERAGE_CFLAGS)
+
+libcxsparse_la_SOURCES =
+libcxsparse_la_LIBADD = libcxsparse_di.la libcxsparse_dl.la libcxsparse_ci.la libcxsparse_cl.la libcs_convert.la
+libcxsparse_la_LDFLAGS = -no-undefined -release $(PACKAGE_VERSION)
diff --git a/Tcov/Makefile.am b/Tcov/Makefile.am
new file mode 100644
--- /dev/null
+++ b/Tcov/Makefile.am
@@ -0,0 +1,78 @@
+AM_CPPFLAGS = -I$(top_srcdir)/Include -I$(top_srcdir)/Demo $(SUITESPARSECONFIG_CFLAGS)
+LDADD = $(top_builddir)/Source/libcxsparse.la
+
+EXTRA_DIST = \
+	README.txt
+
+dist_noinst_DATA = \
+	czero \
+	nil \
+	zero
+
+if CODE_COVERAGE_ENABLED
+check_PROGRAMS = \
+	cstcov_di_test \
+	cstcov_dl_test \
+	cstcov_ci_test \
+	cstcov_cl_test
+
+cstcov_di_test_SOURCES = cstcov_test.c
+cstcov_di_test_LDADD = $(top_builddir)/Demo/libcs_di_demo.la $(LDADD)
+
+cstcov_dl_test_SOURCES = cstcov_test.c
+cstcov_dl_test_LDADD = $(top_builddir)/Demo/libcs_dl_demo.la $(LDADD)
+cstcov_dl_test_CPPFLAGS = $(AM_CPPFLAGS) -DCS_LONG
+
+cstcov_ci_test_SOURCES = cstcov_test.c
+cstcov_ci_test_LDADD = $(top_builddir)/Demo/libcs_ci_demo.la $(LDADD)
+cstcov_ci_test_CPPFLAGS = $(AM_CPPFLAGS) -DCS_COMPLEX
+
+cstcov_cl_test_SOURCES = cstcov_test.c
+cstcov_cl_test_LDADD = $(top_builddir)/Demo/libcs_cl_demo.la $(LDADD)
+cstcov_cl_test_CPPFLAGS = $(AM_CPPFLAGS) -DCS_LONG -DCS_COMPLEX
+
+MATRICES = \
+	$(top_srcdir)/Tcov/nil \
+	$(top_srcdir)/Tcov/zero \
+	$(top_srcdir)/Matrix/ash219 \
+	$(top_srcdir)/Matrix/bcsstk01 \
+	$(top_srcdir)/Matrix/fs_183_1 \
+	$(top_srcdir)/Matrix/lp_afiro \
+	$(top_srcdir)/Matrix/mhd1280b \
+	$(top_srcdir)/Matrix/neumann \
+	$(top_srcdir)/Matrix/qc324 \
+	$(top_srcdir)/Matrix/t1 \
+	$(top_srcdir)/Matrix/t2 \
+	$(top_srcdir)/Matrix/west0067 \
+	$(top_srcdir)/Matrix/young1c
+
+check-local: $(check_PROGRAMS) $(MATRICES)
+	-./cstcov_di_test $(top_srcdir)/Tcov/nil > test_di_nil.out
+	-./cstcov_di_test $(top_srcdir)/Tcov/zero > test_di_zero.out
+	-./cstcov_di_test $(top_srcdir)/Matrix/t1 > test_di_t1.out
+	-./cstcov_di_test $(top_srcdir)/Matrix/bcsstk01 > test_di_k1.out
+	-./cstcov_di_test $(top_srcdir)/Matrix/fs_183_1 > test_di_fs.out
+	-./cstcov_di_test $(top_srcdir)/Matrix/west0067 > test_di_we.out
+	-./cstcov_di_test $(top_srcdir)/Matrix/ash219 > test_di_ash.out
+	-./cstcov_di_test $(top_srcdir)/Matrix/lp_afiro > test_di_afiro.out
+	-./cstcov_dl_test $(top_srcdir)/Tcov/nil > test_dl_nil.out
+	-./cstcov_dl_test $(top_srcdir)/Tcov/zero > test_dl_zero.out
+	-./cstcov_dl_test $(top_srcdir)/Matrix/t1 > test_dl_t1.out
+	-./cstcov_dl_test $(top_srcdir)/Matrix/bcsstk01 > test_dl_k1.out
+	-./cstcov_dl_test $(top_srcdir)/Matrix/fs_183_1 > test_dl_fs.out
+	-./cstcov_dl_test $(top_srcdir)/Matrix/west0067 > test_dl_we.out
+	-./cstcov_dl_test $(top_srcdir)/Matrix/ash219 > test_dl_ash.out
+	-./cstcov_dl_test $(top_srcdir)/Matrix/lp_afiro > test_dl_afiro.out
+	-./cstcov_ci_test $(top_srcdir)/Matrix/t2 > test_ci_t2.out
+	-./cstcov_ci_test $(top_srcdir)/Matrix/young1c > test_ci_young1c.out
+	-./cstcov_ci_test $(top_srcdir)/Matrix/qc324 > test_ci_qc324.out
+	-./cstcov_ci_test $(top_srcdir)/Matrix/neumann > test_ci_neumann.out
+	-./cstcov_ci_test $(top_srcdir)/Matrix/mhd1280b > test_ci_mhd1280b.out
+	-./cstcov_cl_test $(top_srcdir)/Matrix/t2 > test_cl_t2.out
+	-./cstcov_cl_test $(top_srcdir)/Matrix/young1c > test_cl_young1c.out
+	-./cstcov_cl_test $(top_srcdir)/Matrix/qc324 > test_cl_qc324.out
+	-./cstcov_cl_test $(top_srcdir)/Matrix/neumann > test_cl_neumann.out
+	-./cstcov_cl_test $(top_srcdir)/Matrix/mhd1280b > test_cl_mhd1280b.out
+endif
+
+CLEANFILES = *.out
diff --git a/configure.ac b/configure.ac
new file mode 100644
--- /dev/null
+++ b/configure.ac
@@ -0,0 +1,50 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT([CXSparse],[3.2.0],[DrTimothyAldenDavis@gmail.com],[cxsparse],[http://www.suitesparse.com/])
+AC_CONFIG_SRCDIR([Source/cs_print.c])
+AC_CONFIG_AUX_DIR([build-aux])
+AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE([foreign])
+AM_SILENT_RULES([yes])
+LT_INIT([disable-static])
+
+# Checks for programs.
+PKG_PROG_PKG_CONFIG
+AC_PROG_INSTALL
+AC_PROG_CC
+AX_CODE_COVERAGE
+
+AS_VAR_SET([LIBS_SAVED], [$LIBS])
+
+# Checks for libraries.
+AC_CHECK_LIB([m], [sqrt])
+
+PKG_CHECK_MODULES([SUITESPARSECONFIG], [suitesparseconfig])
+
+# Checks for header files.
+AC_CHECK_HEADERS([limits.h stdlib.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_SIZE_T
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([sqrt])
+
+AS_VAR_SET([LIBS], [$LIBS_SAVED])
+
+PKG_INSTALLDIR
+AC_CONFIG_FILES([
+	cxsparse.pc
+	Makefile
+	Demo/Makefile
+	Include/Makefile
+	Matrix/Makefile
+	Source/Makefile
+	Tcov/Makefile
+])
+AC_OUTPUT
diff --git a/cxsparse.pc.in b/cxsparse.pc.in
new file mode 100644
--- /dev/null
+++ b/cxsparse.pc.in
@@ -0,0 +1,13 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: @PACKAGE_NAME@
+Description: An extended version of CSparse
+Version: @PACKAGE_VERSION@
+URL: @PACKAGE_URL@
+Requires.private: suitesparseconfig
+Libs: -L${libdir} -lcxsparse
+Libs.private: -lm
+Cflags: -I${includedir}/cxsparse
diff --git a/m4/ax_code_coverage.m4 b/m4/ax_code_coverage.m4
new file mode 100644
--- /dev/null
+++ b/m4/ax_code_coverage.m4
@@ -0,0 +1,264 @@
+# ===========================================================================
+#     https://www.gnu.org/software/autoconf-archive/ax_code_coverage.html
+# ===========================================================================
+#
+# SYNOPSIS
+#
+#   AX_CODE_COVERAGE()
+#
+# DESCRIPTION
+#
+#   Defines CODE_COVERAGE_CPPFLAGS, CODE_COVERAGE_CFLAGS,
+#   CODE_COVERAGE_CXXFLAGS and CODE_COVERAGE_LIBS which should be included
+#   in the CPPFLAGS, CFLAGS CXXFLAGS and LIBS/LIBADD variables of every
+#   build target (program or library) which should be built with code
+#   coverage support. Also defines CODE_COVERAGE_RULES which should be
+#   substituted in your Makefile; and $enable_code_coverage which can be
+#   used in subsequent configure output. CODE_COVERAGE_ENABLED is defined
+#   and substituted, and corresponds to the value of the
+#   --enable-code-coverage option, which defaults to being disabled.
+#
+#   Test also for gcov program and create GCOV variable that could be
+#   substituted.
+#
+#   Note that all optimization flags in CFLAGS must be disabled when code
+#   coverage is enabled.
+#
+#   Usage example:
+#
+#   configure.ac:
+#
+#     AX_CODE_COVERAGE
+#
+#   Makefile.am:
+#
+#     @CODE_COVERAGE_RULES@
+#     my_program_LIBS = ... $(CODE_COVERAGE_LIBS) ...
+#     my_program_CPPFLAGS = ... $(CODE_COVERAGE_CPPFLAGS) ...
+#     my_program_CFLAGS = ... $(CODE_COVERAGE_CFLAGS) ...
+#     my_program_CXXFLAGS = ... $(CODE_COVERAGE_CXXFLAGS) ...
+#
+#   This results in a "check-code-coverage" rule being added to any
+#   Makefile.am which includes "@CODE_COVERAGE_RULES@" (assuming the module
+#   has been configured with --enable-code-coverage). Running `make
+#   check-code-coverage` in that directory will run the module's test suite
+#   (`make check`) and build a code coverage report detailing the code which
+#   was touched, then print the URI for the report.
+#
+#   In earlier versions of this macro, CODE_COVERAGE_LDFLAGS was defined
+#   instead of CODE_COVERAGE_LIBS. They are both still defined, but use of
+#   CODE_COVERAGE_LIBS is preferred for clarity; CODE_COVERAGE_LDFLAGS is
+#   deprecated. They have the same value.
+#
+#   This code was derived from Makefile.decl in GLib, originally licenced
+#   under LGPLv2.1+.
+#
+# LICENSE
+#
+#   Copyright (c) 2012, 2016 Philip Withnall
+#   Copyright (c) 2012 Xan Lopez
+#   Copyright (c) 2012 Christian Persch
+#   Copyright (c) 2012 Paolo Borelli
+#   Copyright (c) 2012 Dan Winship
+#   Copyright (c) 2015 Bastien ROUCARIES
+#
+#   This library is free software; you can redistribute it and/or modify it
+#   under the terms of the GNU Lesser General Public License as published by
+#   the Free Software Foundation; either version 2.1 of the License, or (at
+#   your option) any later version.
+#
+#   This library is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
+#   General Public License for more details.
+#
+#   You should have received a copy of the GNU Lesser General Public License
+#   along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#serial 24
+
+AC_DEFUN([AX_CODE_COVERAGE],[
+	dnl Check for --enable-code-coverage
+	AC_REQUIRE([AC_PROG_SED])
+
+	# allow to override gcov location
+	AC_ARG_WITH([gcov],
+	  [AS_HELP_STRING([--with-gcov[=GCOV]], [use given GCOV for coverage (GCOV=gcov).])],
+	  [_AX_CODE_COVERAGE_GCOV_PROG_WITH=$with_gcov],
+	  [_AX_CODE_COVERAGE_GCOV_PROG_WITH=gcov])
+
+	AC_MSG_CHECKING([whether to build with code coverage support])
+	AC_ARG_ENABLE([code-coverage],
+	  AS_HELP_STRING([--enable-code-coverage],
+	  [Whether to enable code coverage support]),,
+	  enable_code_coverage=no)
+
+	AM_CONDITIONAL([CODE_COVERAGE_ENABLED], [test x$enable_code_coverage = xyes])
+	AC_SUBST([CODE_COVERAGE_ENABLED], [$enable_code_coverage])
+	AC_MSG_RESULT($enable_code_coverage)
+
+	AS_IF([ test "$enable_code_coverage" = "yes" ], [
+		# check for gcov
+		AC_CHECK_TOOL([GCOV],
+		  [$_AX_CODE_COVERAGE_GCOV_PROG_WITH],
+		  [:])
+		AS_IF([test "X$GCOV" = "X:"],
+		  [AC_MSG_ERROR([gcov is needed to do coverage])])
+		AC_SUBST([GCOV])
+
+		dnl Check if gcc is being used
+		AS_IF([ test "$GCC" = "no" ], [
+			AC_MSG_ERROR([not compiling with gcc, which is required for gcov code coverage])
+		])
+
+		AC_CHECK_PROG([LCOV], [lcov], [lcov])
+		AC_CHECK_PROG([GENHTML], [genhtml], [genhtml])
+
+		AS_IF([ test -z "$LCOV" ], [
+			AC_MSG_ERROR([To enable code coverage reporting you must have lcov installed])
+		])
+
+		AS_IF([ test -z "$GENHTML" ], [
+			AC_MSG_ERROR([Could not find genhtml from the lcov package])
+		])
+
+		dnl Build the code coverage flags
+		dnl Define CODE_COVERAGE_LDFLAGS for backwards compatibility
+		CODE_COVERAGE_CPPFLAGS="-DNDEBUG"
+		CODE_COVERAGE_CFLAGS="-O0 -g -fprofile-arcs -ftest-coverage"
+		CODE_COVERAGE_CXXFLAGS="-O0 -g -fprofile-arcs -ftest-coverage"
+		CODE_COVERAGE_LIBS="-lgcov"
+		CODE_COVERAGE_LDFLAGS="$CODE_COVERAGE_LIBS"
+
+		AC_SUBST([CODE_COVERAGE_CPPFLAGS])
+		AC_SUBST([CODE_COVERAGE_CFLAGS])
+		AC_SUBST([CODE_COVERAGE_CXXFLAGS])
+		AC_SUBST([CODE_COVERAGE_LIBS])
+		AC_SUBST([CODE_COVERAGE_LDFLAGS])
+
+		[CODE_COVERAGE_RULES_CHECK='
+	-$(A''M_V_at)$(MAKE) $(AM_MAKEFLAGS) -k check
+	$(A''M_V_at)$(MAKE) $(AM_MAKEFLAGS) code-coverage-capture
+']
+		[CODE_COVERAGE_RULES_CAPTURE='
+	$(code_coverage_v_lcov_cap)$(LCOV) $(code_coverage_quiet) $(addprefix --directory ,$(CODE_COVERAGE_DIRECTORY)) --capture --output-file "$(CODE_COVERAGE_OUTPUT_FILE).tmp" --test-name "$(call code_coverage_sanitize,$(PACKAGE_NAME)-$(PACKAGE_VERSION))" --no-checksum --compat-libtool $(CODE_COVERAGE_LCOV_SHOPTS) $(CODE_COVERAGE_LCOV_OPTIONS)
+	$(code_coverage_v_lcov_ign)$(LCOV) $(code_coverage_quiet) $(addprefix --directory ,$(CODE_COVERAGE_DIRECTORY)) --remove "$(CODE_COVERAGE_OUTPUT_FILE).tmp" "/tmp/*" $(CODE_COVERAGE_IGNORE_PATTERN) --output-file "$(CODE_COVERAGE_OUTPUT_FILE)" $(CODE_COVERAGE_LCOV_SHOPTS) $(CODE_COVERAGE_LCOV_RMOPTS)
+	-@rm -f $(CODE_COVERAGE_OUTPUT_FILE).tmp
+	$(code_coverage_v_genhtml)LANG=C $(GENHTML) $(code_coverage_quiet) $(addprefix --prefix ,$(CODE_COVERAGE_DIRECTORY)) --output-directory "$(CODE_COVERAGE_OUTPUT_DIRECTORY)" --title "$(PACKAGE_NAME)-$(PACKAGE_VERSION) Code Coverage" --legend --show-details "$(CODE_COVERAGE_OUTPUT_FILE)" $(CODE_COVERAGE_GENHTML_OPTIONS)
+	@echo "file://$(abs_builddir)/$(CODE_COVERAGE_OUTPUT_DIRECTORY)/index.html"
+']
+		[CODE_COVERAGE_RULES_CLEAN='
+clean: code-coverage-clean
+distclean: code-coverage-clean
+code-coverage-clean:
+	-$(LCOV) --directory $(top_builddir) -z
+	-rm -rf $(CODE_COVERAGE_OUTPUT_FILE) $(CODE_COVERAGE_OUTPUT_FILE).tmp $(CODE_COVERAGE_OUTPUT_DIRECTORY)
+	-find . \( -name "*.gcda" -o -name "*.gcno" -o -name "*.gcov" \) -delete
+']
+	], [
+		[CODE_COVERAGE_RULES_CHECK='
+	@echo "Need to reconfigure with --enable-code-coverage"
+']
+		CODE_COVERAGE_RULES_CAPTURE="$CODE_COVERAGE_RULES_CHECK"
+		CODE_COVERAGE_RULES_CLEAN=''
+	])
+
+[CODE_COVERAGE_RULES='
+# Code coverage
+#
+# Optional:
+#  - CODE_COVERAGE_DIRECTORY: Top-level directory for code coverage reporting.
+#    Multiple directories may be specified, separated by whitespace.
+#    (Default: $(top_builddir))
+#  - CODE_COVERAGE_OUTPUT_FILE: Filename and path for the .info file generated
+#    by lcov for code coverage. (Default:
+#    $(PACKAGE_NAME)-$(PACKAGE_VERSION)-coverage.info)
+#  - CODE_COVERAGE_OUTPUT_DIRECTORY: Directory for generated code coverage
+#    reports to be created. (Default:
+#    $(PACKAGE_NAME)-$(PACKAGE_VERSION)-coverage)
+#  - CODE_COVERAGE_BRANCH_COVERAGE: Set to 1 to enforce branch coverage,
+#    set to 0 to disable it and leave empty to stay with the default.
+#    (Default: empty)
+#  - CODE_COVERAGE_LCOV_SHOPTS_DEFAULT: Extra options shared between both lcov
+#    instances. (Default: based on $CODE_COVERAGE_BRANCH_COVERAGE)
+#  - CODE_COVERAGE_LCOV_SHOPTS: Extra options to shared between both lcov
+#    instances. (Default: $CODE_COVERAGE_LCOV_SHOPTS_DEFAULT)
+#  - CODE_COVERAGE_LCOV_OPTIONS_GCOVPATH: --gcov-tool pathtogcov
+#  - CODE_COVERAGE_LCOV_OPTIONS_DEFAULT: Extra options to pass to the
+#    collecting lcov instance. (Default: $CODE_COVERAGE_LCOV_OPTIONS_GCOVPATH)
+#  - CODE_COVERAGE_LCOV_OPTIONS: Extra options to pass to the collecting lcov
+#    instance. (Default: $CODE_COVERAGE_LCOV_OPTIONS_DEFAULT)
+#  - CODE_COVERAGE_LCOV_RMOPTS_DEFAULT: Extra options to pass to the filtering
+#    lcov instance. (Default: empty)
+#  - CODE_COVERAGE_LCOV_RMOPTS: Extra options to pass to the filtering lcov
+#    instance. (Default: $CODE_COVERAGE_LCOV_RMOPTS_DEFAULT)
+#  - CODE_COVERAGE_GENHTML_OPTIONS_DEFAULT: Extra options to pass to the
+#    genhtml instance. (Default: based on $CODE_COVERAGE_BRANCH_COVERAGE)
+#  - CODE_COVERAGE_GENHTML_OPTIONS: Extra options to pass to the genhtml
+#    instance. (Default: $CODE_COVERAGE_GENHTML_OPTIONS_DEFAULT)
+#  - CODE_COVERAGE_IGNORE_PATTERN: Extra glob pattern of files to ignore
+#
+# The generated report will be titled using the $(PACKAGE_NAME) and
+# $(PACKAGE_VERSION). In order to add the current git hash to the title,
+# use the git-version-gen script, available online.
+
+# Optional variables
+CODE_COVERAGE_DIRECTORY ?= $(top_builddir)
+CODE_COVERAGE_OUTPUT_FILE ?= $(PACKAGE_NAME)-$(PACKAGE_VERSION)-coverage.info
+CODE_COVERAGE_OUTPUT_DIRECTORY ?= $(PACKAGE_NAME)-$(PACKAGE_VERSION)-coverage
+CODE_COVERAGE_BRANCH_COVERAGE ?=
+CODE_COVERAGE_LCOV_SHOPTS_DEFAULT ?= $(if $(CODE_COVERAGE_BRANCH_COVERAGE),\
+--rc lcov_branch_coverage=$(CODE_COVERAGE_BRANCH_COVERAGE))
+CODE_COVERAGE_LCOV_SHOPTS ?= $(CODE_COVERAGE_LCOV_SHOPTS_DEFAULT)
+CODE_COVERAGE_LCOV_OPTIONS_GCOVPATH ?= --gcov-tool "$(GCOV)"
+CODE_COVERAGE_LCOV_OPTIONS_DEFAULT ?= $(CODE_COVERAGE_LCOV_OPTIONS_GCOVPATH)
+CODE_COVERAGE_LCOV_OPTIONS ?= $(CODE_COVERAGE_LCOV_OPTIONS_DEFAULT)
+CODE_COVERAGE_LCOV_RMOPTS_DEFAULT ?=
+CODE_COVERAGE_LCOV_RMOPTS ?= $(CODE_COVERAGE_LCOV_RMOPTS_DEFAULT)
+CODE_COVERAGE_GENHTML_OPTIONS_DEFAULT ?=\
+$(if $(CODE_COVERAGE_BRANCH_COVERAGE),\
+--rc genhtml_branch_coverage=$(CODE_COVERAGE_BRANCH_COVERAGE))
+CODE_COVERAGE_GENHTML_OPTIONS ?= $(CODE_COVERAGE_GENHTML_OPTIONS_DEFAULT)
+CODE_COVERAGE_IGNORE_PATTERN ?=
+
+code_coverage_v_lcov_cap = $(code_coverage_v_lcov_cap_$(V))
+code_coverage_v_lcov_cap_ = $(code_coverage_v_lcov_cap_$(AM_DEFAULT_VERBOSITY))
+code_coverage_v_lcov_cap_0 = @echo "  LCOV   --capture"\
+ $(CODE_COVERAGE_OUTPUT_FILE);
+code_coverage_v_lcov_ign = $(code_coverage_v_lcov_ign_$(V))
+code_coverage_v_lcov_ign_ = $(code_coverage_v_lcov_ign_$(AM_DEFAULT_VERBOSITY))
+code_coverage_v_lcov_ign_0 = @echo "  LCOV   --remove /tmp/*"\
+ $(CODE_COVERAGE_IGNORE_PATTERN);
+code_coverage_v_genhtml = $(code_coverage_v_genhtml_$(V))
+code_coverage_v_genhtml_ = $(code_coverage_v_genhtml_$(AM_DEFAULT_VERBOSITY))
+code_coverage_v_genhtml_0 = @echo "  GEN   " $(CODE_COVERAGE_OUTPUT_DIRECTORY);
+code_coverage_quiet = $(code_coverage_quiet_$(V))
+code_coverage_quiet_ = $(code_coverage_quiet_$(AM_DEFAULT_VERBOSITY))
+code_coverage_quiet_0 = --quiet
+
+# sanitizes the test-name: replaces with underscores: dashes and dots
+code_coverage_sanitize = $(subst -,_,$(subst .,_,$(1)))
+
+# Use recursive makes in order to ignore errors during check
+check-code-coverage:'"$CODE_COVERAGE_RULES_CHECK"'
+
+# Capture code coverage data
+code-coverage-capture: code-coverage-capture-hook'"$CODE_COVERAGE_RULES_CAPTURE"'
+
+# Hook rule executed before code-coverage-capture, overridable by the user
+code-coverage-capture-hook:
+
+'"$CODE_COVERAGE_RULES_CLEAN"'
+
+GITIGNOREFILES ?=
+GITIGNOREFILES += $(CODE_COVERAGE_OUTPUT_FILE) $(CODE_COVERAGE_OUTPUT_DIRECTORY)
+
+A''M_DISTCHECK_CONFIGURE_FLAGS ?=
+A''M_DISTCHECK_CONFIGURE_FLAGS += --disable-code-coverage
+
+.PHONY: check-code-coverage code-coverage-capture code-coverage-capture-hook code-coverage-clean
+']
+
+	AC_SUBST([CODE_COVERAGE_RULES])
+	m4_ifdef([_AM_SUBST_NOTMAKE], [_AM_SUBST_NOTMAKE([CODE_COVERAGE_RULES])])
+])
