--- monkey-bubble-0.4.0/src/net/message-handler.c	2004-12-24 09:35:30.000000000 -0200
+++ monkey-bubble-0.4.0_patched/src/net/message-handler.c	2012-09-16 13:23:34.000000000 -0300
@@ -28,7 +28,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <arpa/inet.h>
-#include <glib/gthread.h>
+#include <glib.h>
 #include <sys/time.h>
 #include <time.h>
 
--- monkey-bubble-0.4.0/src/net/simple-server.c	2006-03-21 17:17:40.000000000 -0300
+++ monkey-bubble-0.4.0_patched/src/net/simple-server.c	2012-09-16 13:23:55.000000000 -0300
@@ -31,7 +31,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <arpa/inet.h>
-#include <glib/gthread.h>
+#include <glib.h>
 #include <sys/time.h>
 #include <time.h>
 
--- monkey-bubble-0.4.0/src/ui/main.c	2004-12-03 05:05:59.000000000 -0200
+++ monkey-bubble-0.4.0_patched/src/ui/main.c	2012-09-16 13:24:07.000000000 -0300
@@ -15,7 +15,7 @@
 #include <gtk/gtk.h>
 #include <gst/gst.h>
 #include <bonobo/bonobo-i18n.h>
-#include <glib/gthread.h>
+#include <glib.h>
 #include <libgnomeui/gnome-ui-init.h>
 
 #include <math.h>
