diff -pruN old/src/glib-utils.c new/src/glib-utils.c
--- old/src/glib-utils.c	2011-04-26 17:29:35.000000000 +0300
+++ new/src/glib-utils.c	2021-01-14 00:29:26.236713125 +0200
@@ -542,21 +542,6 @@ get_time_string (time_t time)
 }
 
 
-GPtrArray *
-g_ptr_array_copy (GPtrArray *array)
-{
-	GPtrArray *new_array;
-	
-	if (array == NULL)
-		return NULL;
-		
-	new_array = g_ptr_array_sized_new (array->len);
-	memcpy (new_array->pdata, array->pdata, array->len * sizeof (gpointer)); 
-	new_array->len = array->len;
-	
-	return new_array;
-}
-
 
 void
 g_ptr_array_free_full (GPtrArray *array,
diff -pruN old/src/glib-utils.h new/src/glib-utils.h
--- old/src/glib-utils.h	2011-04-26 17:29:35.000000000 +0300
+++ new/src/glib-utils.h	2021-01-14 00:30:57.756705920 +0200
@@ -59,7 +59,6 @@ const char *        get_last_field
 						  int          last_field);
 int                 n_fields                     (char       **str_array);
 char *              get_time_string              (time_t       time);
-GPtrArray *         g_ptr_array_copy             (GPtrArray   *array);
 void                g_ptr_array_free_full        (GPtrArray   *array,
                        				  GFunc        func,
                        				  gpointer     user_data);
