diff -Naur SuperTuxKart-1.4-src.orig/CMakeLists.txt SuperTuxKart-1.4-src/CMakeLists.txt
--- SuperTuxKart-1.4-src.orig/CMakeLists.txt	2022-10-31 22:00:41.000000000 +0100
+++ SuperTuxKart-1.4-src/CMakeLists.txt	2025-07-28 15:43:38.715368259 +0200
@@ -7,7 +7,7 @@
 add_definitions( -DSUPERTUXKART_VERSION="${PROJECT_VERSION}" )
 
 if(NOT (CMAKE_MAJOR_VERSION VERSION_LESS 3))
-  cmake_policy(SET CMP0043 OLD)
+  cmake_policy(SET CMP0043 NEW)
 endif()
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake")
diff -Naur SuperTuxKart-1.4-src.orig/lib/wiiuse/CMakeLists.txt SuperTuxKart-1.4-src/lib/wiiuse/CMakeLists.txt
--- SuperTuxKart-1.4-src.orig/lib/wiiuse/CMakeLists.txt	2022-10-31 22:00:48.000000000 +0100
+++ SuperTuxKart-1.4-src/lib/wiiuse/CMakeLists.txt	2025-07-28 15:45:38.170368259 +0200
@@ -7,7 +7,7 @@
 # Added for STK, silences CMake warning
 # See https://github.com/supertuxkart/stk-code/commit/b0ff15873ee7fa8901672672b47def9039a5534b#diff-a3e272598233e89e4b577a434cc2a89d
 if(NOT (CMAKE_MAJOR_VERSION VERSION_LESS 3))
-  cmake_policy(SET CMP0048 OLD)
+  cmake_policy(SET CMP0048 NEW)
 endif()
 
 # Set package properties
