From c46e1ce457529cc575543eb9186702216913fccd Mon Sep 17 00:00:00 2001
From: CodingJellyfish <dumaosen_main01@outlook.com>
Date: Wed, 19 Feb 2025 17:20:15 +0800
Subject: [PATCH] Fix #5301

---
 lib/mcpp/system.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/lib/mcpp/system.c b/lib/mcpp/system.c
index a5b0d85c742..9b8ffeec6c3 100644
--- a/lib/mcpp/system.c
+++ b/lib/mcpp/system.c
@@ -3435,7 +3435,7 @@ static int  open_file(
     if (! fullname)                 /* Non-existent or directory    */
         return  FALSE;
     if (standard && included( fullname))        /* Once included    */
-        goto  true;
+        goto  case_true;
         
     if ((max_open != 0 && max_open <= include_nest)
                             /* Exceed the known limit of open files */
@@ -3462,12 +3462,12 @@ static int  open_file(
         if ((fp = fopen( fullname, "r")) == NULL) {
             file->fp = fopen( cur_fullname, "r");
             fseek( file->fp, file->pos, SEEK_SET);
-            goto  false;
+            goto  case_false;
         }
         if (max_open == 0)      /* Remember the limit of the system */
             max_open = include_nest;
     } else if (fp == NULL)                  /* No read permission   */ 
-        goto  false;
+        goto  case_false;
     /* Truncate buffer of the includer to save memory   */
     len = (int) (file->bptr - file->buffer);
     if (len) {
@@ -3514,9 +3514,9 @@ static int  open_file(
     if (mkdep && ((mkdep & MD_SYSHEADER) || ! infile->sys_header))
         put_depend( fullname);          /* Output dependency line   */
 
-true:
+case_true:
     return  TRUE;
-false:
+case_false:
     free( fullname);
     return  FALSE;
 }
