From 3a3953f38c3555e87f3608d0291dbfccf34e9775 Mon Sep 17 00:00:00 2001
From: Deve <deveee@gmail.com>
Date: Wed, 21 Nov 2018 21:07:55 +0100
Subject: [PATCH] Fix for system glew

---
 src/graphics/gl_headers.hpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/graphics/gl_headers.hpp b/src/graphics/gl_headers.hpp
index fc9867cb59..290805cafe 100644
--- a/src/graphics/gl_headers.hpp
+++ b/src/graphics/gl_headers.hpp
@@ -25,6 +25,9 @@
 extern "C" {
 #if !defined(USE_GLES2)
 #   include <GL/glew.h>
+// This is a workaround for mesa drivers that now use __gl_glext_h_ instead of
+// __glext_h_ in header file which is not defined in current glew version
+#   define __gl_glext_h_
 #endif
 }
 #include <cinttypes>
