diff -Naur supertuxkart-0.6.1a.orig/src/file_manager.cpp supertuxkart-0.6.1a/src/file_manager.cpp
--- supertuxkart-0.6.1a.orig/src/file_manager.cpp	2009-01-29 01:30:09.000000000 +0000
+++ supertuxkart-0.6.1a/src/file_manager.cpp	2009-06-10 13:12:03.000000000 +0000
@@ -212,7 +212,12 @@
 //-----------------------------------------------------------------------------
 std::string FileManager::getTranslationDir() const
 {
+#if !defined(TRANSLATION_DIR) && !defined(__APPLE__) && !defined(WIN32)
+#define TRANSLATION_DIR "/usr/share/locale"
+    return TRANSLATION_DIR;
+#else
     return m_root_dir+"/data/po";
+#endif
 }   // getTranslationDir
 
 //-----------------------------------------------------------------------------
