--- supertuxkart-0.7.3/src/io/file_manager.cpp	2011-11-01 22:04:13.000000000 -0200
+++ supertuxkart-0.7.3_patched/src/io/file_manager.cpp	2011-12-09 16:54:27.000000000 -0200
@@ -442,7 +442,12 @@
  */
 std::string FileManager::getTranslationDir() const
 {
+#if !defined(TRANSLATION_DIR) && !defined(__APPLE__) && !defined(WIN32)
+#define TRANSLATION_DIR "/usr/share/locale"
+    return TRANSLATION_DIR;
+#else
     return m_root_dir+"/data/po";
+#endif
 }   // getTranslationDir
 
 //-----------------------------------------------------------------------------
