--- xfe-1.32.1_orig/src/xfeutils.cpp	2009-11-09 14:17:17.000000000 -0200
+++ xfe-1.32.1/src/xfeutils.cpp	2010-02-17 00:39:22.000000000 -0200
@@ -512,7 +512,7 @@
 
 
 // Replacement of the stat function
-inline FXint statrep(const FXchar* filename, struct stat* buf)
+FXint statrep(const FXchar* filename, struct stat* buf)
 {
 #if defined(linux)
 
@@ -546,7 +546,7 @@
 
 
 // Replacement of the lstat function
-inline FXint lstatrep(const FXchar* filename, struct stat* buf)
+FXint lstatrep(const FXchar* filename, struct stat* buf)
 {
 #if defined(linux)
 
