From f0bcd186190fe6e67b3f0eaaceb7a99aa6a98865 Mon Sep 17 00:00:00 2001
From: Steven Morgan <stevmorg@cisco.com>
Date: Thu, 5 Jan 2017 12:30:35 -0500
Subject: [PATCH] bb111711 - fix zlib version check - patch by Daniel J. Luke.

---
 m4/reorganization/libs/libz.m4 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/m4/reorganization/libs/libz.m4 b/m4/reorganization/libs/libz.m4
index b5c74147b..f7b67ca10 100644
--- a/m4/reorganization/libs/libz.m4
+++ b/m4/reorganization/libs/libz.m4
@@ -29,9 +29,9 @@ then
     AC_MSG_ERROR([Please install zlib and zlib-devel packages])
 else
 
-    vuln=`grep "ZLIB_VERSION \"1.2.0" $ZLIB_HOME/include/zlib.h`
+    vuln=`grep "ZLIB_VERSION \"1.2.0\"" $ZLIB_HOME/include/zlib.h`
     if test -z "$vuln"; then
-	vuln=`grep "ZLIB_VERSION \"1.2.1" $ZLIB_HOME/include/zlib.h`
+	vuln=`grep "ZLIB_VERSION \"1.2.1\"" $ZLIB_HOME/include/zlib.h`
     fi
 
     if test -n "$vuln"; then
-- 
2.11.0

