diff -Naur clamav-0.97.4.orig/libclamav/c++/llvm/lib/ExecutionEngine/JIT/Intercept.cpp clamav-0.97.4/libclamav/c++/llvm/lib/ExecutionEngine/JIT/Intercept.cpp
--- clamav-0.97.4.orig/libclamav/c++/llvm/lib/ExecutionEngine/JIT/Intercept.cpp	2012-03-06 17:31:31.000000000 +0100
+++ clamav-0.97.4/libclamav/c++/llvm/lib/ExecutionEngine/JIT/Intercept.cpp	2012-04-02 15:21:58.172989384 +0200
@@ -15,6 +15,7 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include "unistd.h"
 #include "JIT.h"
 #include "llvm/Support/ErrorHandling.h"
 #include "llvm/System/DynamicLibrary.h"
