diff -Naur glbsp-2.24-source/Makefile.unx glbsp-2.24-source.patched/Makefile.unx
--- glbsp-2.24-source/Makefile.unx	2007-07-25 23:14:34.000000000 -0400
+++ glbsp-2.24-source.patched/Makefile.unx	2015-11-26 07:20:52.012999000 -0500
@@ -94,9 +94,8 @@
 
 # ----- GUI PROGRAM ---------------------------------------------
 
-FLTK_FLAGS=-I$(FLTK_DIR) -I$(FLTK_DIR)/zlib
-FLTK_LIBS=-L$(FLTK_DIR)/lib -lfltk_images -lpng -lz -ljpeg \
-          -lfltk -lX11 -lXext
+FLTK_FLAGS=$(shell fltk-config --cflags)
+FLTK_LIBS=$(shell fltk-config --use-images --ldflags)
 
 ## may need: -L/usr/X11R6/lib
 
diff -Naur glbsp-2.24-source/nodeview/Makefile.unx glbsp-2.24-source.patched/nodeview/Makefile.unx
--- glbsp-2.24-source/nodeview/Makefile.unx	2007-05-21 09:17:03.000000000 -0400
+++ glbsp-2.24-source.patched/nodeview/Makefile.unx	2015-11-26 07:23:19.660999000 -0500
@@ -7,8 +7,8 @@
 BIN=nodeview
 SRC_DIR=nodeview
 
-FLTK_CFLAGS=-I/usr/local/lib
-FLTK_LIBS=-lfltk_images -lfltk -lX11 -lXext -lpng -ljpeg
+FLTK_CFLAGS=$(shell fltk-config --cflags)
+FLTK_LIBS=$(shell fltk-config --use-images --ldflags)
 
 CC=gcc
 CXX=g++
