diff -ru langdrill-0.3-orig/src/Makefile langdrill-0.3/src/Makefile
--- langdrill-0.3-orig/src/Makefile	2006-03-14 23:46:31.000000000 -0800
+++ langdrill-0.3/src/Makefile	2006-03-14 23:48:17.000000000 -0800
@@ -16,7 +16,7 @@
 DEF_FLAGS = -DTARGET='"$(TARGET)"' \
 	-DDRILL_DEF='"$(DRILL_DEF)"' \
 	-DDRILL_DIR='"$(DRILL_DIR)"' \
-	-I /usr/include/gtk-2.0/
+	`pkg-config --cflags gtk+-2.0`
 
 ###### Test for Debian GNU/Linux
 # If your system is not Debian GNU/Linux please modify the values after
@@ -37,7 +37,7 @@
 endif
 
 CC = g++
-LFLAGS = `vdk-config-2 --libs` 
+LFLAGS = `vdk-config-2 --libs` `pkg-config --libs gtk+-2.0`
 
 
 ####### Implicit rules
