--- langdrill-0.3.orig/src/main.cc
+++ langdrill-0.3/src/main.cc
@@ -962,17 +962,17 @@
 MyLangForm::AboutClicked( VDKObject*)
 {
   Application()->
-    MessageBox( " Let's talk about Language Drills",
-                " Brought to you by\n"
-                "   Ionutz Borcoman\n"
-                "   <borco@usa.net>\n"
-                "   <borco@borco-ei.eng.hokudai.ac.jp>\n"
-                " \n"
-                " Nothing would have been possible without\n"
-                "   Mario Motta's VDK\n"
-                "   <mmotta@guest.net>"
-                "\n",
-                MB_OK | MB_ICONINFORMATION ) ;
+    VDKMessageBox( " Let's talk about Language Drills",
+                   " Brought to you by\n"
+                   "   Ionutz Borcoman\n"
+                   "   <borco@usa.net>\n"
+                   "   <borco@borco-ei.eng.hokudai.ac.jp>\n"
+                   " \n"
+                   " Nothing would have been possible without\n"
+                   "   Mario Motta's VDK\n"
+                   "   <mmotta@guest.net>"
+                   "\n",
+                   VDK_OK | VDK_ICONINFORMATION ) ;
   return true;
 }
 
@@ -985,12 +985,12 @@
   if( _timerRunning==true
       || _timerCheck->Checked==true ){
     bool answ = ( Application()->
-                  MessageBox("Language Drill Message",
-                             " Are you sure you want to close \n"
-                             " the Language Drills ?\n"
-                             " The Timer seems to be active !",
-                             MB_YESNO)
-                  == IDYES );
+                  VDKMessageBox("Language Drill Message",
+                                " Are you sure you want to close \n"
+                                " the Language Drills ?\n"
+                                " The Timer seems to be active !",
+                                VDK_YESNO)
+                  == VDK_IDYES );
     if( answ == false ){
       return true;
     } else {
@@ -1052,8 +1052,9 @@
     gtk_signal_emit_stop_by_name( GTK_OBJECT(_simpleQuizzText->WrappedWidget()),
                                   "key_press_event");
     QuizzAnswerClicked( obj );
+    return true;
   }
-  return true;
+  return false;
 }
 
 
