--- langdrill-0.3.orig/src/config.hh
+++ langdrill-0.3/src/config.hh
@@ -31,7 +31,7 @@
 
 #include <stdio.h>
 #include <string.h>
-#include <fstream.h>
+#include <fstream>
 #include <stdlib.h>
 #include <vdk/dlist.h>
 
--- langdrill-0.3.orig/src/main.hh
+++ langdrill-0.3/src/main.hh
@@ -29,7 +29,7 @@
 #ifndef __main_hh__
 #define __main_hh__
 
-#include <fstream.h>
+#include <fstream>
 #include <unistd.h>
 #include <time.h>
 
--- langdrill-0.3.orig/src/config.cc
+++ langdrill-0.3/src/config.cc
@@ -347,7 +347,7 @@
   char message[MAX_CHARS_FOR_MESSAGE];
   iconv_t direct = (iconv_t)-1, reverse = (iconv_t)-1;
 
-  ifstream in(_name);
+  std::ifstream in(_name);
   while(1){
     in.getline(buff,MAX_CHARS_FOR_STRING);
     lineNr++;
