--- langdrill-0.3.orig/README
+++ langdrill-0.3/README
@@ -1,9 +1,9 @@
 This is langrill, a language drill program. It is used to test your 
-vocabulary in foreign languages. It has some Japanese support included.
+vocabulary in foreign languages. It has Japanese support included.
 
-http://borco-ei.eng.hokudai.ac.jp/~borco/vdk/langdrill.html
+http://storm.prohosting.com/borco/programming/langdrill.html
 
-Mail suggestions and bug reports to borco@borco-ei.eng.hokudai.ac.jp.
+Mail suggestions and bug reports to Ionutz Borcoman <borco@go.ro>.
 
 This program is inspired from JavaDrill. However no code from JavaDrill 
 was used. JavaDrill can be found at:
