From: Markus Koschany <apo@debian.org>
Date: Fri, 19 Jan 2018 10:39:45 +0100
Subject: link against only needed libraries

---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index c0b467b..aec5e4e 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -37,7 +37,7 @@ else
 endif
 
 CC = g++
-LFLAGS = `vdk-config-2 --libs` `pkg-config --libs gtk+-2.0`
+LFLAGS = -Wl,--as-needed `vdk-config-2 --libs` `pkg-config --libs gtk+-2.0`
 
 
 ####### Implicit rules
