diff -ur langdrill-0.3/src/Makefile langdrill-0.3/src/Makefile
--- langdrill-0.3/src/Makefile	2018-03-19 10:15:09.855912688 -0700
+++ langdrill-0.3/src/Makefile	2018-03-19 10:13:38.535093214 -0700
@@ -16,7 +16,9 @@
 DEF_FLAGS = -DTARGET='"$(TARGET)"' \
 	-DDRILL_DEF='"$(DRILL_DEF)"' \
 	-DDRILL_DIR='"$(DRILL_DIR)"' \
-	`pkg-config --cflags gtk+-2.0`
+	`pkg-config --cflags gtk+-2.0` \
+	`vdk-config-2 --cflags` \
+	-Wall
 
 ###### Test for Debian GNU/Linux
 # If your system is not Debian GNU/Linux please modify the values after
@@ -27,16 +29,15 @@
 	DRILL_DIR=$(DESTDIR)/usr/share/langdrill
 	BIN_DIR=$(DESTDIR)/usr/bin
 	MAN_DIR=$(DESTDIR)/usr/share/man/man1
-	CFLAGS = -Wall `vdk-config-2 --cflags` $(DEF_FLAGS)
 else
 	MY_MESSAGE_OF_THE_DAY=">>" $(TARGET) compilled for NON-Debian GNU/Linux system. "<<"
 	DRILL_DIR=$(DESTDIR)/usr/local/share/langdrill
 	BIN_DIR=$(DESTDIR)/usr/local/bin
 	MAN_DIR=$(DESTDIR)/usr/local/man/man1
-	CFLAGS = -Wall -g `vdk-config-2 --cflags` $(DEF_FLAGS)
 endif
 
 CC = g++
+CFLAGS = -g
 LFLAGS = -Wl,--as-needed `vdk-config-2 --libs` `pkg-config --libs gtk+-2.0`
 
 
@@ -45,7 +46,7 @@
 .SUFFIXES: .cc
 
 .cc.o: $(SRC)
-	$(CC) $(CFLAGS) -c $*.cc -o $@
+	$(CC) $(CFLAGS) $(DEF_FLAGS) -c $*.cc -o $@
 
 ####### Build rules
 
@@ -62,16 +63,16 @@
 
 $(TARGET): tarclean $(OBJ)
 	@echo $(MY_MESSAGE_OF_THE_DAY)
-	$(CC) $(CFLAGS) -o $(TARGET) $(OBJ) $(LFLAGS)
+	$(CC) $(CFLAGS) $(DEF_FLAGS) -o $(TARGET) $(OBJ) $(LFLAGS)
 
 run:	tarclean $(OBJ)
 	@echo $(MY_MESSAGE_OF_THE_DAY)
-	$(CC) $(CFLAGS) -o $(TARGET) $(OBJ) $(LFLAGS)
+	$(CC) $(CFLAGS) $(DEF_FLAGS) -o $(TARGET) $(OBJ) $(LFLAGS)
 	./$(TARGET)
 
 build: clean tarclean $(OBJ)
 	@echo $(MY_MESSAGE_OF_THE_DAY)
-	$(CC) $(CFLAGS) -o $(TARGET) $(OBJ) $(LFLAGS)
+	$(CC) $(CFLAGS) $(DEF_FLAGS) -o $(TARGET) $(OBJ) $(LFLAGS)
 
 install: all
 	install -d $(BIN_DIR) $(DRILL_DIR) $(MAN_DIR)
