diff -Naur xbmc-21.0-Omega.orig/xbmc/platform/posix/filesystem/SMBWSDiscoveryListener.cpp xbmc-21.0-Omega/xbmc/platform/posix/filesystem/SMBWSDiscoveryListener.cpp
--- xbmc-21.0-Omega.orig/xbmc/platform/posix/filesystem/SMBWSDiscoveryListener.cpp	2024-04-06 04:29:22.000000000 +0200
+++ xbmc-21.0-Omega/xbmc/platform/posix/filesystem/SMBWSDiscoveryListener.cpp	2024-05-16 19:52:18.646667000 +0200
@@ -17,6 +17,7 @@
 
 #include "platform/posix/filesystem/SMBWSDiscovery.h"
 
+#include <algorithm>
 #include <array>
 #include <chrono>
 #include <mutex>
