Fixes #645 & #646 (GCC 4.7 compatibility)

--- src/libs/math/v3_t.h.orig	2012-03-30 18:09:00.697957753 -0500
+++ src/libs/math/v3_t.h	2012-03-30 18:09:15.421679640 -0500
@@ -28,6 +28,7 @@
 #ifndef _LINALG_V3_T_H_
 #define _LINALG_V3_T_H_
 
+#include <cmath>
 
 template<class T> class v3t;
 #ifndef _MSC_VER
--- src/libs/portability/portability.h.orig	2012-03-30 18:03:35.047110686 -0500
+++ src/libs/portability/portability.h	2012-03-30 18:10:19.171475549 -0500
@@ -22,10 +22,13 @@
 
 #include <cstdlib>
 #include <cstring>
+
 #ifdef _MSC_VER
 #include <direct.h>
 #include <process.h>
 #define _WINSOCKAPI_   /* Prevent inclusion of winsock.h in windows.h */
+#else
+#include <unistd.h>
 #endif
 
 #ifdef _MSC_VER
