--- gazebo/util/LogPlay.cc      2017-06-09 03:51:31.000000000 +0100
+++ gazebo/util/LogPlay.cc.new  2018-02-12 16:52:06.382230361 +0000
@@ -118,8 +118,8 @@
   {
     gzerr << "Unable to load file[" << _logFile << "]. "
       << "Check the Gazebo server log file for more information.\n";
-    const char *errorStr1 = this->dataPtr->xmlDoc.GetErrorStr1();
-    const char *errorStr2 = this->dataPtr->xmlDoc.GetErrorStr2();
+    const char *errorStr1 = this->dataPtr->xmlDoc.ErrorStr();
+    const char *errorStr2 = this->dataPtr->xmlDoc.ErrorStr();
     if (errorStr1)
       gzlog << "Log Error 1:\n" << errorStr1 << std::endl;
     if (errorStr2)
