diff -Naur xu4-20130612_svn/src/event_sdl.cpp xu4-20130612_svn.patched/src/event_sdl.cpp
--- xu4-20130612_svn/src/event_sdl.cpp	2013-06-12 19:34:25.000000000 -0400
+++ xu4-20130612_svn.patched/src/event_sdl.cpp	2021-10-12 03:38:36.048487922 -0400
@@ -43,7 +43,7 @@
     case U4_META + 'q': /* Cmd+q */
     case U4_META + 'x': /* Cmd+x */
 #endif
-    case U4_ALT + 'x': /* Alt+x */
+    // case U4_ALT + 'x': /* Alt+x */
 #if defined(WIN32)
     case U4_ALT + U4_FKEY + 3:
 #endif
diff -Naur xu4-20130612_svn/src/game.cpp xu4-20130612_svn.patched/src/game.cpp
--- xu4-20130612_svn/src/game.cpp	2013-06-12 19:34:24.000000000 -0400
+++ xu4-20130612_svn.patched/src/game.cpp	2021-10-12 03:33:55.794513398 -0400
@@ -1205,7 +1205,7 @@
             screenMessage("Quit & Save...\n%d moves\n", c->saveGame->moves);
             if (c->location->context & CTX_CAN_SAVE_GAME) {        
                 gameSave();
-                screenMessage("Press Alt-x to quit\n");
+                screenMessage("Close window to quit\n");
             }
             else screenMessage("%cNot here!%c\n", FG_GREY, FG_WHITE);
             
