diff -Naur xu4-20111026_svn/src/settings.cpp xu4-20111026_svn.patched//src/settings.cpp
--- xu4-20111026_svn/src/settings.cpp	2011-06-23 13:36:06.000000000 -0400
+++ xu4-20111026_svn.patched//src/settings.cpp	2011-10-27 17:27:02.000000000 -0400
@@ -81,12 +81,6 @@
  * Initialize the settings.
  */
 void Settings::init(const bool useProfile, const string profileName) {
-	if (useProfile) {
-		userPath = "./profiles/";
-		userPath += profileName.c_str();
-		userPath += "/";
-	} else {
-
 #if defined(MACOSX)
             FSRef folder;
             OSErr err = FSFindFolder(kUserDomain, kApplicationSupportFolderType, kCreateFolder, &folder);
@@ -150,7 +144,12 @@
         userPath = "./";
 #endif
 
+	if (useProfile) {
+		userPath += "profiles/";
+		userPath += profileName.c_str();
+		userPath += "/";
 	}
+
     FileSystem::createDirectory(userPath);
 
     filename = userPath + SETTINGS_BASE_FILENAME;
