--- xflr5-gui/xdirect/XDirectStyleDlg.cpp	2018-02-28 23:07:05.640183963 -0500
+++ xflr5-gui/xdirect/XDirectStyleDlg.cpp.new	2018-02-28 23:06:54.047183259 -0500
@@ -25,7 +25,7 @@
 #include "viewwidgets/oppointwidget.h"
 #include <QGridLayout>
 #include <QLabel>
-
+#include <QKeyEvent>
 
 
 XDirectStyleDlg::XDirectStyleDlg(QWidget *pParent) : QDialog(pParent)
--- xflr5-gui/xdirect/analysis/XFoilAnalysisDlg.cpp	2018-02-28 23:10:02.922194720 -0500
+++ xflr5-gui/xdirect/analysis/XFoilAnalysisDlg.cpp.new	2018-02-17 07:56:55.430707766 -0500
@@ -37,8 +37,7 @@
 #include "graphwidget.h"
 #include <XFoil.h>
 #include <xdirect/analysis/XFoilTask.h>
-
-
+#include <QShowEvent>
 
 void *XFoilAnalysisDlg::s_pXDirect;
 QPoint XFoilAnalysisDlg::s_Position;
