--- citeproc-hs.cabal~	2013-01-23 15:06:35.000000000 +0200
+++ citeproc-hs.cabal	2013-02-25 19:39:32.000000000 +0200
@@ -125,10 +125,12 @@
 
     if flag(hexpat)
        build-depends: hexpat >= 0.20.2
+       extensions:    CPP
        exposed-modules:  Text.CSL.Pickle.Hexpat
        cpp-options:   -DUSE_HEXPAT
     else
        build-depends: xml
+       extensions:    CPP
        exposed-modules:  Text.CSL.Pickle.Xml
 
     if flag(embed_data_files)
