diff -ur monodevelop.old/main/monodevelop.desktop monodevelop/main/monodevelop.desktop
--- monodevelop.old/main/monodevelop.desktop	2017-04-12 10:50:03.000000000 -0300
+++ monodevelop/main/monodevelop.desktop	2017-07-09 19:04:01.306846766 -0300
@@ -6,7 +6,7 @@
 GenericName[ja]=統合開発環境
 Comment=Develop .NET applications in an Integrated Development Environment
 Comment[ja]=.NETアプリケーション開発を行うための統合開発環境
-Exec=monodevelop %F
+Exec=env MONO_TLS_PROVIDER=legacy monodevelop %F
 TryExec=monodevelop
 Icon=monodevelop
 StartupNotify=true
