--- mps-youtube-0.2.5/mps-youtube.desktop.orig	2015-06-02 04:05:05.775959790 +0300
+++ mps-youtube-0.2.5/mps-youtube.desktop	2015-06-02 04:40:39.404285765 +0300
@@ -2,10 +2,8 @@
 Name=mps-youtube
 GenericName=Music Player
 Keywords=Audio;Song;Podcast;Playlist;youtube.com;
-Exec=mpsyt %U
+Exec=mpsyt
 Terminal=true
 Icon=terminal
 Type=Application
-Categories=AudioVideo;Audio;Player;
-StartupNotify=true
-NoDisplay=true
+Categories=GTK;AudioVideo;Audio;Player;
