From 08296d7cc2697696f2e6098e619f316baaebc626 Mon Sep 17 00:00:00 2001
From: CycoZA <peter@cerebus.co.za>
Date: Sun, 5 May 2024 16:29:52 +0200
Subject: [PATCH] Fix the compiling under Linux/Mac

---
 src/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index 74971d1..d0b5cff 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -19,7 +19,7 @@ UNIX    = -DUNIX
 #CFLAGS  = -g -Wall $(USEZ) $(DOS) $(UNIX) $(ARCH)
 CFLAGS  = -Wall $(USEZ) $(DOS) $(UNIX) -O3 -fomit-frame-pointer $(ARCH)
 
-L       =  $(ZLIB)
+L       =  $(ZLIB) -lm
 CC      = gcc
 DESTDIR	= /usr/local/bin
 MANDIR	= /usr/local/man/man1
@@ -30,7 +30,7 @@ DOCDIR	= /usr/local/doc/atasm
 
 all:    prog
 
-OBJS    = asm.o symbol.o parser.o setparse.o state.o dimage.o inc_path.o crc32.o atasm_err.o state2.o
+OBJS    = asm.o symbol.o parser.o setparse.o state.o dimage.o inc_path.o crc32.o atasm_err.o state2.o parse_float.o setfloatparse.o
 
 asm.o: ops.h directive.h symbol.h inc_path.h atasm_err.h
 atasm_err.o: atasm_err.h symbol.h
-- 
2.35.8

