diff --git a/src/setfloatparse.c b/src/setfloatparse.c
index 3bf78b8..61c1569 100644
--- a/src/setfloatparse.c
+++ b/src/setfloatparse.c
@@ -41,7 +41,7 @@ float fnums[64];
 extern int get_dotname(char* src, char* dst);
 extern symbol* validate_symbol(char* str);
 
-char* parse_string;
+static char* parse_string;
 /*=========================================================================*
   function floatlex()
 
diff --git a/src/setparse.c b/src/setparse.c
index 07904c5..fb4fc66 100644
--- a/src/setparse.c
+++ b/src/setparse.c
@@ -36,7 +36,7 @@ int yyparse();
 extern int rval;
 int vnum, nums[64];
 
-char *parse_string;
+static char *parse_string;
 /*=========================================================================*
   function yylex()
 
