diff -Naur Voice2MIDI-0.3.10.orig/SConstruct Voice2MIDI-0.3.10/SConstruct
--- Voice2MIDI-0.3.10.orig/SConstruct	2009-06-29 16:46:18.000000000 +0200
+++ Voice2MIDI-0.3.10/SConstruct	2012-09-16 09:50:30.637847765 +0200
@@ -164,6 +164,8 @@
 extralinkFlags = []
 if sys.platform == 'darwin' :
 	extralinkFlags=['-dynamic','-bind_at_load']
+if sys.platform == 'linux2' :
+	extralinkFlags=['-lGL', '-lpthread']
 
 programs = []
 for main in mainSources.items() :
