diff -Naur koules1.4/sound.c koules1.4.patched/sound.c
--- koules1.4/sound.c	1998-03-04 13:59:19.000000000 -0500
+++ koules1.4.patched/sound.c	2014-05-02 03:29:02.000000000 -0400
@@ -40,7 +40,7 @@
 init_sound ()
 {
   int             i, fd[2];
-  char           *argv[4];
+  char           *argv[5];
   char            filename[512];
 
   signal (SIGCHLD, SIG_IGN);
@@ -63,11 +63,12 @@
       dup2 (fd[0], STDIN_FILENO);
       close (fd[0]);
       sprintf (filename, SOUNDSERVER);
-      argv[0] = filename;
-      argv[1] = unixSoundPath;
-      argv[2] = unixSoundDev;
-      argv[3] = NULL;
-      execvp (filename, argv);
+      argv[0] = "aoss";
+      argv[1] = filename;
+      argv[2] = unixSoundPath;
+      argv[3] = unixSoundDev;
+      argv[4] = NULL;
+      execvp ("aoss", argv);
       fprintf (stderr, "Couldn't Execute Sound Server %s!\n", filename);
       exit (0);
     };
