diff -Naur koules1.4/xlib/init.c koules1.4.patched/xlib/init.c
--- koules1.4/xlib/init.c	1998-03-05 12:03:01.000000000 -0500
+++ koules1.4.patched/xlib/init.c	2009-04-15 06:12:07.000000000 -0400
@@ -88,11 +88,11 @@
   return (pixmap);
 }
 
+static int      bpp;
 #ifdef MITSHM
 extern int      XShmQueryExtension (Display * dpy);
 static int      haderror;
 static int      (*origerrorhandler) (Display *, XErrorEvent *);
-static int      bpp;
 
 
 static int
