diff -Naur koules1.4/xkoules.6 koules1.4.patched/xkoules.6
--- koules1.4/xkoules.6	1998-03-04 13:59:19.000000000 -0500
+++ koules1.4.patched/xkoules.6	2009-04-15 06:30:32.000000000 -0400
@@ -69,6 +69,9 @@
 for large display(640x480)
 default size
 .TP
+.B \-E 
+for extra large display(900x600)
+.TP
 .B \-m 
 for monochrome displays
 .TP
diff -Naur koules1.4/xlib/init.c koules1.4.patched/xlib/init.c
--- koules1.4/xlib/init.c	1998-03-05 12:03:01.000000000 -0500
+++ koules1.4.patched/xlib/init.c	2009-04-15 06:29:13.000000000 -0400
@@ -775,6 +775,7 @@
 		  " -m for monochrome displays\n"
 		  " -s for small display(320x250)\n"
 		  " -l for large display(640x480)\n"
+		  " -E for extra large display(900x600)\n"
 		  " -p use private colormap\n"
 		  " -y Synchronize with X(for debugging)\n"
 		  " -f nofade(for debugging)\n"
