diff -Naur koules1.4/xlib/init.c koules1.4.patched/xlib/init.c
--- koules1.4/xlib/init.c	1998-03-05 12:03:01.000000000 -0500
+++ koules1.4.patched/xlib/init.c	2023-01-07 23:23:21.963726308 -0500
@@ -334,7 +334,7 @@
   printf ("Autoprobing hardware\n");
   printf ("Initializing joystick driver\n");
 #ifdef JOYSTICK
-  joystickdevice[0] = open ("/dev/js0", O_RDONLY);
+  joystickdevice[0] = open ("/dev/input/js0", O_RDONLY);
   if (joystickdevice[0] < 0)
     {
       perror ("Joystick driver");
@@ -343,7 +343,7 @@
     }
   else
     printf ("Joystick 1 initialized\n");
-  joystickdevice[1] = open ("/dev/js1", O_RDONLY);
+  joystickdevice[1] = open ("/dev/input/js1", O_RDONLY);
   if (joystickdevice[1] < 0)
     {
       perror ("Joystick driver");
