diff -Naur koules1.4/Iconfig koules1.4.patched/Iconfig
--- koules1.4/Iconfig	1998-03-04 15:29:05.000000000 -0500
+++ koules1.4.patched/Iconfig	2009-04-15 05:54:44.000000000 -0400
@@ -29,13 +29,14 @@
 /* linux joystick support
    Now compiles w/o joystick toolkit. For using joystick support is
    joystick toolikit required...of course */
-/*#define JOYSTICK*/
+#define JOYSTICK
 /*for fast 386 based assembler routines
   recomended for linux*/
 /*#define I386ASSEMBLY*/
+#define LinuxArchitecture
 /* directories*/
 KOULESDIR		=/usr/bin/X11
-SOUNDDIR		=/usr/local/lib/koules
+SOUNDDIR		=/usr/share/koules
 MANDIR			=/usr/local/man/man6
 
 /*You need some extra libraryes for BSD sockets compatibility?*/
@@ -51,7 +52,6 @@
 SUBDIRS = xlib
 
 #if defined(LinuxArchitecture)
-    SYSDEFS = -Wall -fomit-frame-pointer -O6  -ffast-math
 #if !defined(NAS_SOUND)&&!defined(RSOUND)
 #define SOUND
     SOUNDSERVER = koules.sndsrv.linux
diff -Naur koules1.4/Imakefile koules1.4.patched/Imakefile
--- koules1.4/Imakefile	1998-03-04 13:59:19.000000000 -0500
+++ koules1.4.patched/Imakefile	2009-04-15 05:35:39.000000000 -0400
@@ -56,7 +56,7 @@
 
 DEFINES = -DONLYANSI -DSOUND $(JOYSTICK1) $(NET) $(MITSHM1) $(HAVEUSLEEP1) $(SYSDEFS) $(JOYSTICK) -Ixlib\
 	$(NAMEDEF) $(ASMDEF)\
-	-DSOUNDSERVER=\"$(SOUNDDIR)/$(SOUNDSERVER)\" \
+	-DSOUNDSERVER=\"/usr/libexec/$(SOUNDSERVER)\" \
 	-DSOUNDDIR=\"$(SOUNDDIR)\" \
 	-DSOUNDDEV=\"$(SOUNDDEV)\"
 #endif
