diff -Naur koules1.4/koules.tcl koules1.4.patched/koules.tcl
--- koules1.4/koules.tcl	1998-03-04 13:59:19.000000000 -0500
+++ koules1.4.patched/koules.tcl	2009-04-15 08:00:38.000000000 -0400
@@ -3,9 +3,9 @@
 # This is simple user (anti)friendly dialog for starting koules
 # Only reason why I did it is that I wanted to try tcl/tk
 #files - configure here if paths are differ
-set xkoules "/usr/bin/X11/xkoules"
-set koulessvga "/usr/local/bin/koules.svga"
-set koulessound "/usr/local/lib/koules/creator1.raw"
+set xkoules "/usr/games/koules"
+set koulessvga "@@@NOSUCHFILE@@@"
+set koulessound "/usr/share/koules/creator1.raw"
 set config "~/.xkoules.opt"
 set revision 1.1
 # Set default
