diff -Naur youtube-viewer-3.7.4/bin/gtk3-youtube-viewer youtube-viewer-3.7.4.patched/bin/gtk3-youtube-viewer
--- youtube-viewer-3.7.4/bin/gtk3-youtube-viewer	2020-02-11 22:35:13.000000000 -0500
+++ youtube-viewer-3.7.4.patched/bin/gtk3-youtube-viewer	2020-03-05 17:02:07.819892140 -0500
@@ -638,11 +638,11 @@
             undef $history{lc($line)};
         }
 
-        require List::Util;
+        require List::MoreUtils;
 
         # Keep only the most recent non-duplicated entries
-        @history        = reverse(List::Util::uniq(reverse(@history)));
-        @search_history = List::Util::uniq(@search_history);
+        @history        = reverse(List::MoreUtils::uniq(reverse(@history)));
+        @search_history = List::MoreUtils::uniq(@search_history);
 
         # Set entry completion
         $completion = Gtk3::EntryCompletion->new;
