diff -urp a/src/libutil/TimestampedBuffer.cpp b/src/libutil/TimestampedBuffer.cpp
--- a/src/libutil/TimestampedBuffer.cpp	2014-06-01 19:00:10.000000000 -0500
+++ b/src/libutil/TimestampedBuffer.cpp	2016-02-29 21:01:56.693239943 -0600
@@ -32,6 +32,7 @@
 
 #include <cstdlib>
 #include <cstring>
+#include <math.h>
 
 #define DLL_PI        (3.141592653589793238)
 #define DLL_SQRT2     (1.414213562373095049)
@@ -179,7 +180,6 @@ bool TimestampedBuffer::setWrapValue(ffa
     m_wrap_at=w;
     return true;
 }
-#include <math.h>
 
 /**
  * \brief return the effective rate
