diff --git a/vendor/github.com/aead/poly1305/poly1305_amd64.go b/vendor/github.com/aead/poly1305/poly1305_amd64.go
index cd1a717..41b630f 100644
--- a/vendor/github.com/aead/poly1305/poly1305_amd64.go
+++ b/vendor/github.com/aead/poly1305/poly1305_amd64.go
@@ -7,13 +7,11 @@
 package poly1305
 
 import (
+	"golang.org/x/sys/cpu"
 	"io"
 )
 
-var useAVX2 = supportsAVX2()
-
-//go:noescape
-func supportsAVX2() bool
+var useAVX2 = cpu.X86.HasAVX2
 
 //go:noescape
 func initialize(state *[7]uint64, key *[32]byte)
