diff -Naur thttpd-2.25b.orig/libhttpd.c thttpd-2.25b/libhttpd.c
--- thttpd-2.25b.orig/libhttpd.c	2007-10-14 10:09:11.000000000 +0000
+++ thttpd-2.25b/libhttpd.c	2007-10-14 10:21:37.000000000 +0000
@@ -3279,7 +3279,7 @@
 	cp += strspn( cp, " \t" );
 	status = atoi( cp );
 	}
-    if ( ( cp = strstr( headers, "Location:" ) ) != (char*) 0 &&
+    else if ( ( cp = strstr( headers, "Location:" ) ) != (char*) 0 &&
 	 cp < br &&
 	 ( cp == headers || *(cp-1) == '\012' ) )
 	status = 302;
