diff -Nur ion-3-20090110.orig/system.mk ion-3-20090110/system.mk
--- ion-3-20090110.orig/system.mk	2009-01-09 16:20:27.000000000 -0600
+++ ion-3-20090110/system.mk	2009-09-28 09:47:40.062672371 -0500
@@ -8,7 +8,7 @@
 
 # Installation path prefix. Unless you know what you're doing, the default
 # of /usr/local is likely the correct choice.
-PREFIX=/usr/local
+PREFIX=/usr
 
 # Unless you are creating a package conforming to some OS's standards, you
 # probably do not want to modify the following directories:
@@ -16,17 +16,17 @@
 # Main binaries
 BINDIR=$(PREFIX)/bin
 # Configuration .lua files
-ETCDIR=$(PREFIX)/etc/ion3
+ETCDIR=/etc/ion3
 # Some .lua files and ion-* shell scripts
 SHAREDIR=$(PREFIX)/share/ion3
 # Manual pages
-MANDIR=$(PREFIX)/share/man
+MANDIR=$(PREFIX)/man
 # Some documents
-DOCDIR=$(PREFIX)/share/doc/ion3
+DOCDIR=$(PREFIX)/doc/ion3-@VERSION@
 # Nothing at the moment
 INCDIR=$(PREFIX)/include/ion3
 # Nothing at the moment
-LIBDIR=$(PREFIX)/lib
+LIBDIR=$(PREFIX)/lib@LIBDIRSUFFIX@
 # Modules
 MODULEDIR=$(LIBDIR)/ion3/mod
 # Compiled Lua source code
@@ -62,8 +62,8 @@
 
 # If you have installed Lua 5.1 from the official tarball without changing
 # paths, this should do it.
-LUA_DIR=/usr/local
-LUA_LIBS = -L$(LUA_DIR)/lib -llua
+LUA_DIR=/usr
+LUA_LIBS = -L$(LUA_DIR)/lib@LIBDIRSUFFIX@ -llua
 LUA_INCLUDES = -I$(LUA_DIR)/include
 LUA=$(LUA_DIR)/bin/lua
 LUAC=$(LUA_DIR)/bin/luac
@@ -81,11 +81,11 @@
 ##
 
 # Paths
-X11_PREFIX=/usr/X11R6
+X11_PREFIX=/usr
 # SunOS/Solaris
 #X11_PREFIX=/usr/openwin
 
-X11_LIBS=-L$(X11_PREFIX)/lib -lX11 -lXext
+X11_LIBS=-L$(X11_PREFIX)/lib@LIBDIRSUFFIX@/X11 -lX11 -lXext
 X11_INCLUDES=-I$(X11_PREFIX)/include
 
 # XFree86 libraries up to 4.3.0 have a bug that can cause a segfault.
@@ -157,7 +157,7 @@
 	-Wtrigraphs -Wformat -Wchar-subscripts \
 	-Wparentheses -pedantic -Wuninitialized
 
-CFLAGS=-Os $(WARN) $(DEFINES) $(INCLUDES) $(EXTRA_INCLUDES)
+CFLAGS= $(OPTS) $(WARN) $(DEFINES) $(INCLUDES) $(EXTRA_INCLUDES)
 LDFLAGS=$(LIBS) $(EXTRA_LIBS)
 EXPORT_DYNAMIC=-Xlinker --export-dynamic
 
