diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/CLAMWidgets.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/CLAMWidgets.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/CLAMWidgets.hxx	2006-10-03 14:32:09.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/CLAMWidgets.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -1,5 +1,5 @@
 #include <QtDesigner/QDesignerContainerExtension>
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 #include <QtDesigner/QDesignerCustomWidgetCollectionInterface>
 #include <QtCore/QtPlugin>
 
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/ChordRankingPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/ChordRankingPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/ChordRankingPlugin.hxx	2007-04-16 12:10:57.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/ChordRankingPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef ChordRankingPlugin_hxx
 #define ChordRankingPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class ChordRankingPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/ControlSurfaceWidgetPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/ControlSurfaceWidgetPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/ControlSurfaceWidgetPlugin.hxx	2007-04-20 04:46:36.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/ControlSurfaceWidgetPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef ControlSurfaceWidgetPlugin_hxx
 #define ControlSurfaceWidgetPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class ControlSurfaceWidgetPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/KeySpacePlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/KeySpacePlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/KeySpacePlugin.hxx	2007-04-16 12:10:57.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/KeySpacePlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef KeySpacePlugin_hxx
 #define KeySpacePlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class KeySpacePlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/LPModelViewPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/LPModelViewPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/LPModelViewPlugin.hxx	2007-04-16 12:10:57.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/LPModelViewPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef LPModelViewPlugin_hxx
 #define LPModelViewPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class LPModelViewPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/MelCepstrumViewPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/MelCepstrumViewPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/MelCepstrumViewPlugin.hxx	2007-04-16 12:10:57.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/MelCepstrumViewPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef MelCepstrumViewPlugin_hxx
 #define MelCepstrumViewPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class MelCepstrumViewPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/MelSpectrumViewPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/MelSpectrumViewPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/MelSpectrumViewPlugin.hxx	2007-04-16 12:10:57.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/MelSpectrumViewPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef MelSpectrumViewPlugin_hxx
 #define MelSpectrumViewPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class MelSpectrumViewPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/OscilloscopePlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/OscilloscopePlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/OscilloscopePlugin.hxx	2007-04-16 12:10:57.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/OscilloscopePlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef OscilloscopePlugin_hxx
 #define OscilloscopePlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class OscilloscopePlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/PeakViewPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/PeakViewPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/PeakViewPlugin.hxx	2007-04-16 12:10:57.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/PeakViewPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef PeakViewPlugin_hxx
 #define PeakViewPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class PeakViewPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/PixmapWidgetsPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/PixmapWidgetsPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/PixmapWidgetsPlugin.hxx	2007-04-16 12:10:57.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/PixmapWidgetsPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef PixmapSliderPlugin_hxx
 #define PixmapSliderPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class PixmapSliderPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/PolarChromaPeaksPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/PolarChromaPeaksPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/PolarChromaPeaksPlugin.hxx	2007-04-16 12:10:57.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/PolarChromaPeaksPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef PolarChromaPeaksPlugin_hxx
 #define PolarChromaPeaksPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class PolarChromaPeaksPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/ProgressControlWidgetPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/ProgressControlWidgetPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/ProgressControlWidgetPlugin.hxx	2008-07-08 11:36:21.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/ProgressControlWidgetPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef ProgressControlWidgetPlugin_hxx
 #define ProgressControlWidgetPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class ProgressControlWidgetPlugin : public QObject,
                                     public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/QFirstPersonPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/QFirstPersonPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/QFirstPersonPlugin.hxx	2008-03-11 21:24:12.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/QFirstPersonPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef QFirstPersonPlugin_hxx
 #define QFirstPersonPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class QFirstPersonPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/QSynthKnobPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/QSynthKnobPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/QSynthKnobPlugin.hxx	2007-04-16 12:10:57.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/QSynthKnobPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef QSynthKnobPlugin_hxx
 #define QSynthKnobPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class QSynthKnobPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/SegmentationViewPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/SegmentationViewPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/SegmentationViewPlugin.hxx	2008-07-28 05:50:33.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/SegmentationViewPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef SegmentationViewPlugin_hxx
 #define SegmentationViewPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class SegmentationViewPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/SpectrogramPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/SpectrogramPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/SpectrogramPlugin.hxx	2007-08-13 23:08:23.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/SpectrogramPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef SpectrogramPlugin_hxx
 #define SpectrogramPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class SpectrogramPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/SpectrumViewPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/SpectrumViewPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/SpectrumViewPlugin.hxx	2007-04-16 12:10:57.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/SpectrumViewPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef SpectrumViewPlugin_hxx
 #define SpectrumViewPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class SpectrumViewPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/TonnetzPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/TonnetzPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/TonnetzPlugin.hxx	2007-04-16 12:10:57.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/TonnetzPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef TonnetzPlugin_hxx
 #define TonnetzPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class TonnetzPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
diff -Naur NetworkEditor-1.3.0/src/clamWidgetsPlugin/VumeterPlugin.hxx NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/VumeterPlugin.hxx
--- NetworkEditor-1.3.0/src/clamWidgetsPlugin/VumeterPlugin.hxx	2007-04-16 12:10:57.000000000 -0400
+++ NetworkEditor-1.3.0.patched/src/clamWidgetsPlugin/VumeterPlugin.hxx	2009-12-29 03:04:24.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef VumeterPlugin_hxx
 #define VumeterPlugin_hxx
 
-#include <QtDesigner/QDesignerCustomWidgetInterface>
+#include <QDesignerCustomWidgetInterface>
 
 class VumeterPlugin : public QObject,
                              public QDesignerCustomWidgetInterface
