--- a/src/ntpd.c.orig	2015-03-25 02:18:56.000000000 +0100
+++ b/src/ntpd.c	2015-03-25 14:55:29.964709538 +0100
@@ -100,9 +100,8 @@ usage(void)
 {
 	extern char *__progname;
 
-	if (strcmp(__progname, "ntpctl") == 0)
-		fprintf(stderr,
-		    "usage: ntpctl -s all | peers | Sensors | status\n");
+	if (strcmp(__progname, "openntpctl") == 0)
+		fprintf(stderr, "usage: %s -s all | peers | Sensors | status\n", __progname);
 	else
 		fprintf(stderr, "usage: %s [-dnSsv] [-f file] [-p file]\n",
 		    __progname);
@@ -131,7 +130,7 @@ main(int argc, char *argv[])
 	struct passwd		*pw;
 	extern char		*__progname;
 
-	if (strcmp(__progname, "ntpctl") == 0) {
+	if (strcmp(__progname, "openntpctl") == 0) {
 		ctl_main(argc, argv);
 		/* NOTREACHED */
 	}
@@ -598,7 +597,7 @@ ctl_main(int argc, char *argv[])
 		/* NOTREACHED */
 
 	if ((fd = socket(AF_UNIX, SOCK_STREAM, 0)) == -1)
-		err(1, "ntpctl: socket");
+		err(1, "openntpctl: socket");
 
 	bzero(&sa, sizeof(sa));
 	sa.sun_family = AF_UNIX;
@@ -643,7 +642,7 @@ ctl_main(int argc, char *argv[])
 		if ((n = imsg_read(ibuf_ctl)) == -1)
 			err(1, "ibuf_ctl: imsg_read error");
 		if (n == 0)
-			errx(1, "ntpctl: pipe closed");
+			errx(1, "openntpctl: pipe closed");
 
 		while (!done) {
 			if ((n = imsg_get(ibuf_ctl, &imsg)) == -1)
