diff -Nur openntpd-6.8p1.orig/src/ntpd.c openntpd-6.8p1/src/ntpd.c
--- openntpd-6.8p1.orig/src/ntpd.c	2020-12-07 02:20:35.000000000 -0600
+++ openntpd-6.8p1/src/ntpd.c	2021-07-18 21:33:55.169706930 -0500
@@ -113,9 +113,8 @@
 {
 	extern char *__progname;
 
-	if (strcmp(__progname, "ntpctl") == 0)
-		fprintf(stderr,
-		    "usage: ntpctl -s all | peers | Sensors | status\n");
+	if (strcmp(__progname, "openntpctl") == 0)
+		fprintf(stderr, "usage: %s -s all | peers | Sensors | status\n", __progname);
 	else
 		fprintf(stderr, "usage: %s [-dnv] [-f file] [-p file]\n",
 		    __progname);
@@ -171,7 +170,7 @@
 
 	__progname = get_progname(argv[0]);
 
-	if (strcmp(__progname, "ntpctl") == 0) {
+	if (strcmp(__progname, "openntpctl") == 0) {
 		ctl_main(argc, argv);
 		/* NOTREACHED */
 	}
@@ -736,7 +735,7 @@
 		/* NOTREACHED */
 
 	if ((fd = socket(AF_UNIX, SOCK_STREAM, 0)) == -1)
-		err(1, "ntpctl: socket");
+		err(1, "openntpctl: socket");
 
 	memset(&sa, 0, sizeof(sa));
 	sa.sun_family = AF_UNIX;
@@ -784,7 +783,7 @@
 		if ((n = imsg_read(ibuf_ctl)) == -1 && errno != EAGAIN)
 			err(1, "ibuf_ctl: imsg_read error");
 		if (n == 0)
-			errx(1, "ntpctl: pipe closed");
+			errx(1, "openntpctl: pipe closed");
 
 		while (!done) {
 			if ((n = imsg_get(ibuf_ctl, &imsg)) == -1)
