Index: wally-2.3.2/src/main.cpp
===================================================================
--- wally-2.3.2.orig/src/main.cpp
+++ wally-2.3.2/src/main.cpp
@@ -17,18 +17,6 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-#if (!defined(QT_SHARED)) && (!defined(QT_DLL))
-  #include <QtPlugin>
-
-  Q_IMPORT_PLUGIN(qgif)
-  Q_IMPORT_PLUGIN(qico)
-  Q_IMPORT_PLUGIN(qjpeg)
-  Q_IMPORT_PLUGIN(qmng)
-  Q_IMPORT_PLUGIN(qsvg)
-  Q_IMPORT_PLUGIN(qtiff)
-  Q_IMPORT_PLUGIN(qsqlite)
-#endif
-
 #include <QTimer>
 #include <QMessageBox>
 
