diff -Naur wally-2.3.0.orig/wallyplugin/CMakeLists.txt wally-2.3.0/wallyplugin/CMakeLists.txt
--- wally-2.3.0.orig/wallyplugin/CMakeLists.txt	2009-10-21 06:59:24.000000000 +0000
+++ wally-2.3.0/wallyplugin/CMakeLists.txt	2009-11-14 07:17:20.000000000 +0000
@@ -27,5 +27,3 @@
 install(FILES plasma-wallpaper-wallyplugin.desktop DESTINATION ${SERVICES_INSTALL_DIR})
 install(FILES wallyplugin.png DESTINATION ${ICON_INSTALL_DIR}/oxygen/16x16/apps)
 
-install(CODE "execute_process(COMMAND kbuildsycoca4 OUTPUT_QUIET ERROR_QUIET)")
-install(CODE "execute_process(COMMAND kdialog --title \"WallyPlugin\" --msgbox \"WallyPlugin will be loaded next time KDE is started. Please select it in Appearance Settings.\")")
