Index: git/doc/fr/index.docbook
===================================================================
--- git.orig/doc/fr/index.docbook	2012-04-01 23:00:28.000000000 +0200
+++ git/doc/fr/index.docbook	2012-04-01 23:16:07.000000000 +0200
@@ -34,8 +34,12 @@
 > <surname
 >Pedersen</surname
 > </author>
-      
-      &traducteurXavierBesnard; 
+
+      <othercredit role="translator">
+		  <firstname>Xavier</firstname>
+		  <surname>Besnard</surname>
+		  <affiliation><address><email></email></address></affiliation><contrib>Übersetzung</contrib>
+      </othercredit>
     </authorgroup>
     
     <copyright>
Index: git/doc/nl/index.docbook
===================================================================
--- git.orig/doc/nl/index.docbook	2012-04-01 23:00:28.000000000 +0200
+++ git/doc/nl/index.docbook	2012-04-01 23:15:56.000000000 +0200
@@ -42,7 +42,24 @@
 >Turkia</surname
 > </author>
       
-      &ged.vertaald;&Freek.de.Kruijf; 
+      <othercredit role="translator">
+		  <firstname>Ged</firstname>
+		  <surname>Vertaald</surname>
+<affiliation
+><address
+><email
+></email
+></address
+></affiliation
+><contrib
+>Übersetzung</contrib
+>
+	  </othercredit> 
+      <othercredit role="translator">
+		  <firstname>Freek.de</firstname>
+		  <surname>Kruijf</surname>
+		  <affiliation><address><email></email></address></affiliation><contrib>Übersetzung</contrib>
+      </othercredit>
     </authorgroup>
     
     <copyright>
