diff -ur a/libmakemkv/src/libmkv.cpp b/libmakemkv/src/libmkv.cpp
--- a/libmakemkv/src/libmkv.cpp	2016-01-15 06:39:10.000000000 -0500
+++ b/libmakemkv/src/libmkv.cpp	2016-01-31 17:14:32.761225345 -0500
@@ -18,6 +18,9 @@
     Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
 */
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS 1
+#endif
 #include <libmkv/libmkv.h>
 #include <libmkv/internal.h>
 #include <libmkv/ebmlwrite.h>
@@ -27,9 +30,6 @@
 #include <lgpl/world.h>
 #include <vector>
 
-#ifndef __STDC_FORMAT_MACROS
-#define __STDC_FORMAT_MACROS 1
-#endif
 #include <inttypes.h>
 
 #define TIMECODE_SCALE              1000000
