commit 2e3b7a715793bbb9fe857b37ffaad44b076e3e12
Author: orbea <orbea@fredslev.dk>
Date:   Sat Oct 20 19:54:17 2018 -0700

    Makefile: Support a custom binary name.

diff --git a/Makefile b/Makefile
index 6b96f38..707ab5e 100644
--- a/Makefile
+++ b/Makefile
@@ -3,6 +3,7 @@ PREFIX =	$(PREFIX)
 else
 PREFIX =	/usr
 endif
+BINNAM =	pdksh
 BINDIR = 	$(DESTDIR)/bin
 MANDIR =	$(DESTDIR)$(PREFIX)/man
 TMP ?= /tmp
@@ -33,14 +34,14 @@ all: $(PROG)
 
 check test:
 	/usr/bin/perl tests/th -s tests -p ./ksh -T $(TMP) \
-		-C pdksh,sh,ksh,posix,posix-upu
+		-C $(BINNAM),sh,ksh,posix,posix-upu
 
 install:
 	install -m755 -d $(BINDIR)
-	install -m755 --strip --no-target-directory ksh $(BINDIR)/pdksh
+	install -m755 --strip --no-target-directory ksh $(BINDIR)/$(BINNAM)
 	install -m755 -d $(MANDIR)/man1
-	install -m644 --no-target-directory ksh.1 $(MANDIR)/man1/pdksh.1
-	install -m644 --no-target-directory sh.1 $(MANDIR)/man1/pdksh-sh.1
+	install -m644 --no-target-directory ksh.1 $(MANDIR)/man1/$(BINNAM).1
+	install -m644 --no-target-directory sh.1 $(MANDIR)/man1/$(BINNAM)-sh.1
 
 clean:
 	rm -f $(OBJS) $(PROG)
