--- /tmp/SBo/clamav-gui-1.4.4/src/clamdconfspinboxoption.cpp	2026-08-11 10:24:39.000000000 +0100
+++ 0/patched/clamdconfspinboxoption.cpp.new	2026-08-19 21:30:02.829467445 +0100
@@ -12,9 +12,9 @@
     QString StringSpinMin;
     QString StringSpinMax = values.at(1);
     QString StringSpinValue = values.at(2);
-    options.size() > 0?StringSpinMin = values.at(0):StringSpinMin = 0;
-    options.size() > 1?StringSpinMax = values.at(1):StringSpinMax = 0;
-    options.size() > 2?StringSpinValue = values.at(2):StringSpinValue = 0;
+    options.size() > 0?StringSpinMin = values.at(0):StringSpinMin = "0";
+    options.size() > 1?StringSpinMax = values.at(1):StringSpinMax = "0";
+    options.size() > 2?StringSpinValue = values.at(2):StringSpinValue = "0";
     int spinvalue;
     int spinmin;
     int spinmax;
