--- gtk+extra-2.1.2_orig/gtkextra/gtkcolorcombo.c	2004-09-16 14:48:05.000000000 -0300
+++ gtk+extra-2.1.2/gtkextra/gtkcolorcombo.c	2010-11-01 00:22:36.000000000 -0200
@@ -146,7 +146,7 @@
                                       GTK_CLASS_TYPE(object_class),
                                       GTK_SIGNAL_OFFSET(GtkColorComboClass,
                                       changed),
-                                      gtkextra_VOID__INT_BOXED,
+                                      gtkextra_VOID__BOXED_BOXED,
                                       GTK_TYPE_NONE,  
                                       2, GTK_TYPE_INT, GDK_TYPE_COLOR);
 
