--- gtk+extra-2.1.2_orig/gtkextra/Makefile.in	2010-03-20 21:05:56.000000000 -0300
+++ gtk+extra-2.1.2/gtkextra/Makefile.in	2010-10-31 23:56:00.000000000 -0200
@@ -1028,13 +1028,14 @@
 $(libgtkextra_x11_2_0_la_OBJECTS) $(libgtkextra_linux_fb_2_0_la_OBJECTS) $(libgtkextra_win32_2_0_la_OBJECTS): ${gtkextra_built_public_sources} ${gtkextra_built_private_headers}
 
 $(srcdir)/gtkextra-marshal.h: gtkextra-marshal.list
-        ( @GLIB_GENMARSHAL@ --prefix=gtkextra gtkextra-marshal.list --header > gtkextra-marshal.tmp \
-        && mv gtkextra-marshal.tmp gtkextra-marshal.h ) \
-        || ( rm -f gtkextra-marshal.tmp && exit 1 )
+	( glib-genmarshal --prefix=gtkextra gtkextra-marshal.list --header > gtkextra-marshal.tmp \
+		&& mv gtkextra-marshal.tmp gtkextra-marshal.h ) \
+		|| ( rm -f gtkextra-marshal.tmp && exit 1 )
+
 $(srcdir)/gtkextra-marshal.c: gtkextra-marshal.h
-        ( @GLIB_GENMARSHAL@ --prefix=gtkextra gtkextra-marshal.list --body > gtkextra-marshal.tmp \
-        && mv gtkextra-marshal.tmp gtkextra-marshal.c ) \
-        || ( rm -f gtkextra-marshal.tmp && exit 1 )
+	( glib-genmarshal --prefix=gtkextra gtkextra-marshal.list --body > gtkextra-marshal.tmp \
+		&& mv gtkextra-marshal.tmp gtkextra-marshal.c ) \
+		|| ( rm -f gtkextra-marshal.tmp && exit 1 )
 
 $(srcdir)/gtkextratypebuiltins.h: $(gtkextra_public_h_sources)
 	( cd $(srcdir) && glib-mkenums \
