diff -Naur xfce4-xkb-plugin-0.5.3.3.orig//panel-plugin/xkb-settings-dialog.c xfce4-xkb-plugin-0.5.3.3//panel-plugin/xkb-settings-dialog.c
--- xfce4-xkb-plugin-0.5.3.3.orig//panel-plugin/xkb-settings-dialog.c	2010-10-04 21:14:05.000000000 +0300
+++ xfce4-xkb-plugin-0.5.3.3//panel-plugin/xkb-settings-dialog.c	2010-10-04 21:13:49.000000000 +0300
@@ -31,8 +31,11 @@
 #include <glib.h>
 #include <gtk/gtk.h>
 
+#undef XFCE_DISABLE_DEPRECATED
+
 #include <libxfce4panel/xfce-panel-plugin.h>
 #include <libxfcegui4/libxfcegui4.h>
+#include <libxfcegui4/xfce_framebox.h>
 
 #include "xfce4-xkb-plugin.h"
 #include "xfce4-xkb-plugin-private.h"
@@ -678,7 +681,7 @@
 xfce_xkb_about (XfcePanelPlugin *plugin)
 {
     GtkWidget *about;
-    const gchar* authors[2] = {
+    const gchar* authors[] = {
         "Alexander Iliev <sasoiliev@mamul.org>", 
         "Gauvain Pocentek <gauvainpocentek@gmail.com>", 
         NULL
@@ -825,15 +828,17 @@
 xkb_settings_update_from_ui (t_xkb *xkb)
 {
     gchar *layouts, *variants, *kbdmodel, *toggle_option, 
-          *compose_key_position, *tmp;
+          *compose_key_position;
     t_xkb_kbd_config *kbd_config = xkb->settings->kbd_config;
     gboolean is_default;
     gint i = 0;
 
     model = GTK_TREE_MODEL (xkb->combo_store);
-    gtk_combo_box_get_active_iter (GTK_COMBO_BOX (xkb->kbd_model_combo), &iter);
-    gtk_tree_model_get (model, &iter, NOM, &kbdmodel, -1);
-    kbd_config->model = kbdmodel;
+    if (gtk_combo_box_get_active_iter (GTK_COMBO_BOX (xkb->kbd_model_combo), &iter))
+    {
+        gtk_tree_model_get (model, &iter, NOM, &kbdmodel, -1);
+        kbd_config->model = kbdmodel;
+    }
 
     model = GTK_TREE_MODEL (xkb->toggle_options_store);
     if (gtk_combo_box_get_active_iter (GTK_COMBO_BOX (xkb->toggle_options_combo), &iter))
