--- notion-3-2015061300/system-autodetect.mk.orig	2016-01-28 07:43:36.217463685 +0200
+++ notion-3-2015061300/system-autodetect.mk	2016-01-28 07:47:27.529477603 +0200
@@ -12,7 +12,7 @@
 # Installation path prefix. Unless you know what you're doing, the default
 # of /usr/local is likely the correct choice.
 #DIST: PREFIX=/usr/local
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 
 # Unless you are creating a package conforming to some OS's standards, you
 # probably do not want to modify the following directories:
@@ -22,9 +22,9 @@
 # Some .lua files and ion-* shell scripts
 SHAREDIR=$(PREFIX)/share/notion
 # Manual pages
-MANDIR=$(PREFIX)/share/man
+MANDIR=$(PREFIX)/man
 # Some documents
-DOCDIR=$(PREFIX)/share/doc/notion
+DOCDIR=$(PREFIX)/doc/notion
 # Nothing at the moment
 LIBDIR=$(PREFIX)/lib
 # Modules
@@ -40,7 +40,7 @@
 
 # Configuration .lua files. Overrideable, as config files are usually
 # not placed under $(PREFIX).
-ETCDIR ?= $(PREFIX)/etc/notion
+ETCDIR ?= /etc/notion
 
 # Force all include files to be installed to /usr even if the
 # PREFIX is unset. No header files are installed at the moment
