diff -Naur irrlicht-svn_3843.orig/source/Irrlicht/Makefile irrlicht-svn_3843/source/Irrlicht/Makefile
--- irrlicht-svn_3843.orig/source/Irrlicht/Makefile	2011-05-18 21:32:00.000000000 +0000
+++ irrlicht-svn_3843/source/Irrlicht/Makefile	2012-07-14 23:23:53.693933309 +0000
@@ -82,7 +82,6 @@
 LIB_PATH = ../../lib/$(SYSTEM)
 INSTALL_DIR = /usr/local/lib
 sharedlib install: SHARED_LIB = libIrrlicht.so
-staticlib sharedlib: LDFLAGS += --no-export-all-symbols --add-stdcall-alias
 sharedlib: LDFLAGS += -L/usr/X11R6/lib$(LIBSELECT) -lGL -lXxf86vm
 staticlib sharedlib: CXXINCS += -I/usr/X11R6/include
 
