diff -Naur pycxx-6.2.3.orig/setup.py pycxx-6.2.3/setup.py
--- pycxx-6.2.3.orig/setup.py	2010-05-02 21:10:27.000000000 +0200
+++ pycxx-6.2.3/setup.py	2011-08-24 17:57:18.650467125 +0200
@@ -13,7 +13,7 @@
 
     def finalize_options (self):
 	if not self.install_data or (len(self.install_data) < 8) :
-	    self.install_data = "$base/share/python$py_version_short"
+	    self.install_data = "$base/share"
         install.finalize_options (self)
 
     def run (self):
