--- fakechroot-2.20.1.orig/configure.ac	2019-03-22 21:02:42.000000000 +0000
+++ fakechroot-2.20.1/configure.ac	2023-02-09 00:46:10.059938547 +0000
@@ -1,10 +1,10 @@
-AC_PREREQ(2.64)
-AC_INIT([fakechroot], [2.20.1], [dexter@debian.org], [fakechroot], [https://github.com/dex4er/fakechroot])
+AC_PREREQ([2.71])
+AC_INIT([fakechroot],[2.20.1],[dexter@debian.org],[fakechroot],[https://github.com/dex4er/fakechroot])
 
 AC_CONFIG_SRCDIR([src/libfakechroot.c])
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_MACRO_DIR([m4])
-AC_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 
 AM_INIT_AUTOMAKE([1.10 foreign])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
@@ -19,8 +19,8 @@ AM_PROG_AR
 AC_PROG_CC
 AC_PROG_MAKE_SET
 AC_PROG_LN_S
-AM_PROG_LIBTOOL
-AM_PROG_NM
+LT_INIT
+LT_PATH_NM
 LT_INIT
 
 AC_CANONICAL_HOST
@@ -77,7 +77,11 @@ AC_CHECK_LIB([dl], [dlsym])
 
 # Checks for header files.
 AC_HEADER_DIRENT
-AC_HEADER_STDC
+# Autoupdate added the next two lines to ensure that your configure
+# script's behavior did not change.  They are probably safe to remove.
+AC_CHECK_INCLUDES_DEFAULT
+AC_PROG_EGREP
+
 AC_CHECK_HEADERS(m4_normalize([
     alloca.h
     dirent.h
