diff -Naur OpenCOLLADA-1.6.62.orig/COLLADABaseUtils/include/COLLADABUPcreCompiledPattern.h OpenCOLLADA-1.6.62/COLLADABaseUtils/include/COLLADABUPcreCompiledPattern.h
--- OpenCOLLADA-1.6.62.orig/COLLADABaseUtils/include/COLLADABUPcreCompiledPattern.h	2018-01-13 01:23:52.000000000 +0100
+++ OpenCOLLADA-1.6.62/COLLADABaseUtils/include/COLLADABUPcreCompiledPattern.h	2018-07-08 08:40:38.764948000 +0200
@@ -13,8 +13,7 @@
 
 #include "COLLADABUPrerequisites.h"
 
-struct real_pcre;
-typedef struct real_pcre pcre;
+#include "pcre.h"
 
 
 namespace COLLADABU
