diff -Naur OpenCOLLADA-1.6.62.orig/COLLADABaseUtils/include/COLLADABUPcreCompiledPattern.h OpenCOLLADA-1.6.62/COLLADABaseUtils/include/COLLADABUPcreCompiledPattern.h
--- OpenCOLLADA-1.6.62.orig/COLLADABaseUtils/include/COLLADABUPcreCompiledPattern.h	2018-01-13 01:23:52.000000000 +0100
+++ OpenCOLLADA-1.6.62/COLLADABaseUtils/include/COLLADABUPcreCompiledPattern.h	2018-05-03 14:35:29.723601000 +0200
@@ -12,9 +12,7 @@
 #define __COLLADABU_PCRECOMPILEDPATTERN_H__
 
 #include "COLLADABUPrerequisites.h"
-
-struct real_pcre;
-typedef struct real_pcre pcre;
+#include "pcre.h"
 
 
 namespace COLLADABU
