--- calibre-2.28.0/setup/extensions.py.orig	2015-05-15 05:29:00.000000000 +0200
+++ calibre-2.28.0/setup/extensions.py	2015-05-27 10:20:43.236708204 +0200
@@ -532,7 +532,7 @@
             INCLUDEPATH += {freetype}
             DESTDIR = {destdir}
             CONFIG -= create_cmake  # Prevent qmake from generating a cmake build file which it puts in the calibre src directory
-            QMAKE_LIBS_PRIVATE += {glib} {fontconfig}
+            QMAKE_LIBS_PRIVATE += {glib} {fontconfig} -lfreetype
             ''').format(
                 headers=' '.join(headers), sources=' '.join(sources), others=' '.join(others), destdir=self.d(
                     target), glib=glib_flags, fontconfig=fontconfig_flags, freetype=' '.join(ft_inc_dirs))
