diff -up libclaw-1.7.4/claw-config.in.libdir libclaw-1.7.4/claw-config.in
--- libclaw-1.7.4/claw-config.in.libdir	2013-07-03 13:34:38.000000000 -0600
+++ libclaw-1.7.4/claw-config.in	2016-12-29 16:58:52.140591117 -0700
@@ -140,7 +140,7 @@ while test $# -gt 0; do
       echo -I$CLAW_INCLUDE_DIRECTORY
       ;;
      --libs)
-      libdirs="-L${exec_prefix}/lib "
+      libdirs="-L${exec_prefix}/@CLAW_INSTALLDIR_LIB@ "
 
       echo -n "$libdirs"
       echo $(print_libraries)
diff -up libclaw-1.7.4/CMakeLists.txt.libdir libclaw-1.7.4/CMakeLists.txt
--- libclaw-1.7.4/CMakeLists.txt.libdir	2016-12-29 16:50:24.124507641 -0700
+++ libclaw-1.7.4/CMakeLists.txt	2016-12-29 16:50:24.137508590 -0700
@@ -30,7 +30,7 @@ endif(SVN_ENTRIES)
 set( CLAW_CODE_DIR claw/code/ )
 set( CLAW_INCLUDE_DIRECTORY "${CMAKE_INSTALL_PREFIX}/include/" )
 set( CLAW_INSTALLDIR_SOURCE include/claw/ )
-set( CLAW_INSTALLDIR_LIB lib/ )
+set( CLAW_INSTALLDIR_LIB "lib${LIB_SUFFIX}/" )
 set( CLAW_INSTALLDIR_DOC share/doc/libclaw${CLAW_VERSION_MAJOR}/ )
 set( CLAW_TRANSLATIONS_INSTALL_DIR "share/locale" )
 set( CLAW_EXECUTABLE_DIR bin/ )
diff -up libclaw-1.7.4/cmake-module/CMakeLists.txt.libdir libclaw-1.7.4/cmake-module/CMakeLists.txt
--- libclaw-1.7.4/cmake-module/CMakeLists.txt.libdir	2013-07-03 13:34:38.000000000 -0600
+++ libclaw-1.7.4/cmake-module/CMakeLists.txt	2016-12-29 16:50:24.148509393 -0700
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 2.6)
 project(claw:cmake-modules)
 
 if( NOT CLAW_CMAKE_MODULES_INSTALL_PATH )
-  set( CLAW_CMAKE_MODULES_INSTALL_PATH "share/cmake/libclaw" )
+  set( CLAW_CMAKE_MODULES_INSTALL_PATH "lib${LIB_SUFFIX}/cmake/libclaw" )
 endif( NOT CLAW_CMAKE_MODULES_INSTALL_PATH )
 
 set( CLAW_CMAKE_MODULES
