--- pygame-1.9.6/buildconfig/config_unix.py	2019-09-11 20:29:30.060515857 +0200
+++ pygame-1.9.6/buildconfig/config_unix.py.orig	2019-09-11 20:29:54.659518227 +0200
@@ -254,14 +254,14 @@
     for d in DEPS:
         d.configure(incdirs, libdirs)
 
-    for d in DEPS[1:]:
-        if not d.found:
-            if "-auto" not in sys.argv and not confirm("""
-Warning, some of the pygame dependencies were not found. Pygame can still
-compile and install, but games that depend on those missing dependencies
-will not run. Would you like to continue the configuration?"""):
-                raise SystemExit("Missing dependencies")
-            break
+#    for d in DEPS[1:]:
+#        if not d.found:
+#            if "-auto" not in sys.argv and not confirm("""
+#Warning, some of the pygame dependencies were not found. Pygame can still
+#compile and install, but games that depend on those missing dependencies
+#will not run. Would you like to continue the configuration?"""):
+#                raise SystemExit("Missing dependencies")
+#            break
 
     return DEPS
 
