--- pygame-2.0.3/setup.orig	2022-02-20 22:52:08.449183353 +0200
+++ pygame-2.0.3/setup.py	2022-02-20 22:52:57.916188118 +0200
@@ -75,7 +75,7 @@
 
 
 # A (bit hacky) fix for https://github.com/pygame/pygame/issues/2613
-# This is due to the fact that distutils uses command line args to 
+# This is due to the fact that distutils uses command line args to
 # export PyInit_* functions on windows, but those functions are already exported
 # and that is why compiler gives warnings
 from distutils.command.build_ext import build_ext
@@ -95,6 +95,9 @@
                 the_system = 'Ubuntu'
             elif distro[0].lower() == 'debian':
                 the_system = 'Debian'
+            elif distro[0].lower() == 'slackware':
+                the_system = 'Slackware'
+
 
     help_urls = {
         'Linux': 'https://www.pygame.org/wiki/Compilation',
