Author: Andreas Beckmann <anbe@debian.org>
Description: avoid strings.h collision

--- a/cuneiform_src/Kern/hhh/tigerh/h/strings.h
+++ b/cuneiform_src/Kern/hhh/tigerh/h/strings.h
@@ -75,6 +75,8 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  *                                                                          *
  ***************************************************************************/
 
+#include_next <strings.h>
+#if 0
 # ifndef __STRINGS_H_INCLUDE
 # define __STRINGS_H_INCLUDE
 
@@ -154,3 +156,4 @@ extern STRING *pStringsListEnd;
 extern STRING *pStringsUpList;
 extern STRING *pStringsDownList;
 # endif
+#endif
