--- a/doc/hdapsd.man
+++ b/doc/hdapsd.man
@@ -10,7 +10,7 @@ hdapsd \- park the drive in case of an emergency
 .TP
 \fB\-f\fR \fB\-\-force\fR
 Force unloading heads, even if kernel thinks different (on pre ATA7 drives).
-This only works when adding devices by hand (-d).
+This only works when adding devices by hand (\-d).
 .TP
 \fB\-s\fR \fB\-\-sensitivity=\fR\fI<sensitivity>\fR
 How sensitive hdapsd should be to movements.
