diff --git a/htmldoc/htmlsep.cxx b/htmldoc/htmlsep.cxx
index f2d57bff..d9b2ca30 100644
--- a/htmldoc/htmlsep.cxx
+++ b/htmldoc/htmlsep.cxx
@@ -517,7 +517,7 @@ write_doc(FILE   **out,			// I - Output file
     if (t->markup >= MARKUP_H1 && t->markup < (MARKUP_H1 + TocLevels) &&
         htmlGetVariable(t, (uchar *)"_HD_OMIT_TOC") == NULL)
     {
-      if (heading >= 0)
+      if (*heading >= 0)
         write_footer(out, *heading);
 
       (*heading) ++;
