From 48b4c991164c1aff4699f30ec0df9df660be75b8 Mon Sep 17 00:00:00 2001
From: jgeboski <jgeboski@gmail.com>
Date: Thu, 22 Mar 2012 18:16:23 -0400
Subject: [PATCH 1/2] AudioInput: compilation fix for builds using celt >=
 0.11.0

---
 src/mumble/AudioInput.cpp |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/src/mumble/AudioInput.cpp b/src/mumble/AudioInput.cpp
index 93b7b8e..b541aea 100644
--- a/src/mumble/AudioInput.cpp
+++ b/src/mumble/AudioInput.cpp
@@ -823,7 +823,12 @@ void AudioInput::encodeAudioFrame() {
 
 		cCodec->celt_encoder_ctl(ceEncoder, CELT_SET_PREDICTION(0));
 
-		cCodec->celt_encoder_ctl(ceEncoder,CELT_SET_VBR_RATE(iAudioQuality));
+#ifdef CELT_SET_VBR_RATE
+		cCodec->celt_encoder_ctl(ceEncoder, CELT_SET_VBR_RATE(iAudioQuality));
+#else
+		cCodec->celt_encoder_ctl(ceEncoder, CELT_SET_BITRATE(iAudioQuality));
+#endif
+
 		len = cCodec->encode(ceEncoder, psSource, buffer, qMin(iAudioQuality / 800, 127));
 		iBitrate = len * 100 * 8;
 	} else {
-- 
1.7.4.4

