From 38bdb16b33436a11bd506a80af9191cfc6f98bb3 Mon Sep 17 00:00:00 2001
From: jgeboski <jgeboski@gmail.com>
Date: Thu, 22 Mar 2012 18:17:56 -0400
Subject: [PATCH 2/2] MainWindow: directly set the window icon

---
 src/mumble/MainWindow.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/mumble/MainWindow.cpp b/src/mumble/MainWindow.cpp
index ee71d6e..ce88421 100644
--- a/src/mumble/MainWindow.cpp
+++ b/src/mumble/MainWindow.cpp
@@ -160,6 +160,10 @@ MainWindow::MainWindow(QWidget *p) : QMainWindow(p) {
 	// current open document (i.e. you can copy the open document anywhere
 	// simply by dragging this icon).
 	qApp->setWindowIcon(qiIcon);
+
+	// Set the icon on the MainWindow directly. This fixes the icon not
+	// being set on the MainWindow in certain environments (Ex: GTK+).
+	setWindowIcon(qiIcon);
 #endif
 
 #ifdef Q_OS_WIN
-- 
1.7.4.4

