--- makefile.orig	2005-10-03 00:46:56.000000000 +0200
+++ makefile	2009-01-10 20:57:03.000000000 +0100
@@ -19,7 +19,16 @@
 ### -DLEARN_DIR=\"/var/local/lib\"
 ### -DQCAPSONLY
 
-DEFINES = -DGNUFUN 
+DEFINES = -DGNUFUN \
+-DPBOOK_FILE=\"pbook.phalanx\" \
+-DSBOOK_FILE=\"sbook.phalanx\"  \
+-DPBOOK_DIR=\"/usr/share/games/phalanx\" \
+-DSBOOK_DIR=\"/usr/share/games/phalanx\" \
+-DLEARN_DIR=\"/var/games/phalanx\" \
+-DLEARN_FILE=\"learn.phalanx\" \
+-DECO_FILE=\"eco.phalanx\" \
+-DECO_DIR=\"/usr/share/games/phalanx\" 
+
 LDFLAGS =
 
 OBJ = .o/phalanx.o .o/bcreate.o .o/search.o .o/io.o .o/data.o \
