--- gnome-vfs-2.8.2/schemas/desktop_gnome_url_handlers.schemas.in.browser_default	2004-10-18 21:38:08.605885937 -0400
+++ gnome-vfs-2.8.2/schemas/desktop_gnome_url_handlers.schemas.in	2004-10-18 21:38:20.470253994 -0400
@@ -153,7 +153,7 @@
       <applyto>/desktop/gnome/url-handlers/http/command</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>epiphany %s</default>
+      <default>gio open %s</default>
       <locale name="C">
         <short>The handler for "http" URLs</short>
         <long>The command used to handle "http" URLs, if enabled.</long>
@@ -187,7 +187,7 @@
       <applyto>/desktop/gnome/url-handlers/https/command</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>epiphany %s</default>
+      <default>gio open %s</default>
       <locale name="C">
         <short>The handler for "https" URLs</short>
         <long>The command used to handle "https" URLs, if enabled.</long>
