--- config/master.conf.orig	2016-05-08 14:29:36.954240236 -0500
+++ config/master.conf	2016-05-08 14:33:04.116828055 -0500
@@ -24,8 +24,8 @@
 # Set the appropriate ClamD user and group accounts for your system.
 # If you do not want the script to set user and group permissions on
 # files and directories, comment the next two variables.
-#clam_user="clamav"
-#clam_group="clamav"
+clam_user="clamav"
+clam_group="clamav"
 
 # If you do not want the script to change the file mode of all signature
 # database files in the ClamAV working directory to 0644 (-rw-r--r--):
@@ -55,7 +55,7 @@
 reload_dbs="yes"
 
 # Top level working directory, script will attempt to create them.
-work_dir="/var/lib/clamav-unofficial-sigs"   #Top level working directory
+work_dir="/var/lib/clamav/clamav-unofficial-sigs"   #Top level working directory
 
 # Log update information to '$log_file_path/$log_file_name'.
 logging_enabled="yes"
@@ -357,7 +357,7 @@
 max_sleep_time="600"   # Default maximum is 600 seconds (10 minutes).
 
 # Command to do a full clamd service stop/start
-#clamd_restart_opt="service clamd restart"
+clamd_restart_opt="/etc/rc.d/rc.clamav restart"
 
 # Custom Command to fo a full clamd reload, this defaults to "clamdscan --reload" when not set
 #clamd_reload_opt="clamdscan --reload"
@@ -459,15 +459,15 @@
 
 # Custom man install settings, these are detected and only used if you want to override
 # the automatic detection and generation of the values when not set, this is mainly to aid package maintainers
-#man_dir="" #default: /usr/share/man/man8
-#man_filename="" #default: clamav-unofficial-sigs.8
+man_dir="/usr/man/man8" #default: /usr/share/man/man8
+man_filename="clamav-unofficial-sigs.8" #default: clamav-unofficial-sigs.8
 
 # Provided two variables that package and port maintainers can use in order to 
 # prevent the script from removing itself with the '-r' flag
 # If the script was installed via a package manager like yum, apt, pkg, etc. 
 # The script will instead provide feedback to the user about how to uninstall the package.
-#pkg_mgr="" #the package manager name
-#pkg_rm="" #the package manager command to remove the script
+pkg_mgr="pkgtool" #the package manager name
+pkg_rm="removepkg clamav-unofficial-sigs" #the package manager command to remove the script
 
 # Custom full working directory paths, these are detected and only used if you want to override
 # the automatic detection and generation of the values when not set, this is mainly to aid package maintainers
