--- os.slackware.conf.orig	2017-03-27 19:20:18.394890692 -0500
+++ os.slackware.conf	2017-03-27 19:33:55.428193026 -0500
@@ -23,13 +23,12 @@
 clam_user="clamav"
 clam_group="clamav"
 
-#clam_dbs="/var/lib/clamav"
-clam_dbs="/usr/local/share/clamav"
+clam_dbs="/var/lib/clamav"
 
 clamd_pid="/var/run/clamav/clamd.pid"
 
-clamd_restart_opt="service clamd restart"
+clamd_restart_opt="/etc/rc.d/rc.clamav restart"
 
-#clamd_socket="/var/run/clamav/clamd.socket"
+clamd_socket="/var/run/clamav/clamd.socket"
 
 # https://eXtremeSHOK.com ######################################################
