--- config/os.slackware.conf.orig	2016-05-03 21:29:02.518557766 -0500
+++ config/os.slackware.conf	2016-05-03 21:31:20.083647753 -0500
@@ -27,13 +27,13 @@
 clam_user="clamav"
 clam_group="clamav"
 
-#clam_dbs="/var/lib/clamav"
-clam_dbs="/usr/local/share/clamav"
+clam_dbs="/var/lib/clamav"
+#clam_dbs="/usr/local/share/clamav"
 
 clamd_pid="/var/run/clamav/clamd.pid"
 
-clamd_restart_opt="service clamd restart"
+clamd_restart_opt="/etc/rc.d/rc.clamav restart"
 
-#clamd_socket="/var/run/clamav/clamd.socket"
+clamd_socket="/var/run/clamav/clamd.socket"
 
 # https://eXtremeSHOK.com ######################################################
