--- ext/toolchain/commands1.py.orig	2017-03-04 00:05:27.000000000 +1000
+++ ext/toolchain/commands1.py	2020-02-12 09:12:23.304443034 +1000
@@ -967,16 +967,17 @@
 		if sys.version_info < (2, 4):
 			raise Exception("Python 2.4 or greater required.")
 
-		p = subprocess.Popen(
-			["git", "log", "--pretty=format:%h", "-n", "1"],
-			stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-
-		stdout, stderr = p.communicate()
-
-		if p.returncode != 0:
-			raise Exception('Could not get revision, git error: ' + str(p.returncode))
-	
-		return stdout.strip()
+#		p = subprocess.Popen(
+#			["git", "log", "--pretty=format:%h", "-n", "1"],
+#			stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+#
+#		stdout, stderr = p.communicate()
+#
+#		if p.returncode != 0:
+#			raise Exception('Could not get revision, git error: ' + str(p.returncode))
+#	
+#		return stdout.strip()
+                return "1.8.8"
 
 	def getGitBranchName(self):
 		if sys.version_info < (2, 4):
--- src/gui/src/AboutDialog.cpp.orig	2017-03-04 00:05:27.000000000 +1000
+++ src/gui/src/AboutDialog.cpp	2020-02-12 09:47:52.648113058 +1000
@@ -29,7 +29,7 @@
 
 	m_versionChecker.setApp(synergyApp);
 	QString version = m_versionChecker.getVersion();
-	version = version + '-' + VERSION_STAGE +  '-' + VERSION_REVISION;
+	version = version + '-' + VERSION_STAGE;
 	m_pLabelSynergyVersion->setText(version);
 
 	QString buildDateString = QString::fromLocal8Bit(__DATE__).simplified();
