Author: Goedson Teixeira Paixao <goedson@debian.org>
Description: Removes reference to non-existent menu.h file
Forwarded: https://savannah.gnu.org/patch/?7050

Index: b/src/menu.c
===================================================================
--- a/src/menu.c	2009-12-19 17:11:48.000000000 -0200
+++ b/src/menu.c	2009-12-19 17:12:10.000000000 -0200
@@ -34,7 +34,6 @@
 #include "image.h"
 #include "image_processing.h"
 #include "tool_palette.h"
-#include "menu.h"
 #include "print.h"
 #include "file.h"
 #include "text.h"
