--- trunk/bleachbit/Unix.py	2009/02/08 20:08:26	274
+++ trunk/bleachbit/Unix.py	2009/05/08 02:27:54	346
@@ -447,7 +447,9 @@
     regex = '-[0-9]{8}$'
     globpaths = ( '/var/log/*-*', '/var/log/*/*-*' )
     for path in FileUtilities.globex(globpaths, regex):
-        yield path
+        whitelist_re = '^/var/log/(removed_)?(packages|scripts)'
+        if None == re.match(whitelist_re, path): # for Slackware, Launchpad #367575
+            yield path
 
 
 def wine_to_linux_path(wineprefix, windows_pathname):
