Implement and install a launcher to start the GUI.

Index: bleachbit-0.2.1/bin/bleachbit
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ bleachbit-0.2.1/bin/bleachbit	2009-01-29 22:10:20.641192983 +0100
@@ -0,0 +1,8 @@
+#!/usr/bin/env python
+
+import bleachbit.GUI
+import gtk
+
+if __name__ == '__main__':
+    gui = bleachbit.GUI.GUI()
+    gtk.main()
Index: bleachbit-0.2.1/setup.py
===================================================================
--- bleachbit-0.2.1.orig/setup.py	2009-01-29 22:10:13.625194708 +0100
+++ bleachbit-0.2.1/setup.py	2009-01-29 22:10:20.641192983 +0100
@@ -25,6 +25,7 @@
 data_files = []
 data_files.append(('/usr/share/applications', ['./bleachbit.desktop']))
 data_files.append(('/usr/share/pixmaps/', ['./bleachbit.png']))
+data_files.append(('/usr/bin', ['bin/bleachbit']))
 
 setup(name='bleachbit',
       version='0.4.1',
