diff -Naur ois-v1-3.orig/demos/FFConsoleDemo.cpp ois-v1-3/demos/FFConsoleDemo.cpp
--- ois-v1-3.orig/demos/FFConsoleDemo.cpp	2010-04-01 03:38:04.000000000 +0000
+++ ois-v1-3/demos/FFConsoleDemo.cpp	2012-07-10 02:40:07.339818611 +0000
@@ -3,6 +3,7 @@
 #include <math.h>
 #include <cstdlib>
 #include <iostream>
+#include <unistd.h>
 #include <iomanip>
 #include <ios>
 #include <sstream>
diff -Naur ois-v1-3.orig/demos/OISConsole.cpp ois-v1-3/demos/OISConsole.cpp
--- ois-v1-3.orig/demos/OISConsole.cpp	2010-08-15 22:51:34.000000000 +0000
+++ ois-v1-3/demos/OISConsole.cpp	2012-07-10 02:38:39.624815796 +0000
@@ -10,6 +10,7 @@
 #include "OISForceFeedback.h"
 
 #include <iostream>
+#include <unistd.h>
 #include <vector>
 #include <sstream>
 
@@ -77,7 +78,7 @@
 	}
 	bool keyReleased( const KeyEvent &arg ) {
 		if( arg.key == KC_ESCAPE || arg.key == KC_Q )
-			appRunning = false;
+			appRunning = false;
 		std::cout << "KeyReleased {" << ((Keyboard*)(arg.device))->getAsString(arg.key) << "}\n";
 		return true;
 	}
diff -Naur ois-v1-3.orig/src/linux/LinuxForceFeedback.cpp ois-v1-3/src/linux/LinuxForceFeedback.cpp
--- ois-v1-3.orig/src/linux/LinuxForceFeedback.cpp	2010-04-01 03:06:55.000000000 +0000
+++ ois-v1-3/src/linux/LinuxForceFeedback.cpp	2012-07-10 02:33:03.871465001 +0000
@@ -33,7 +33,8 @@
 #define OIS_LINUX_JOYFF_DEBUG 1
 
 #ifdef OIS_LINUX_JOYFF_DEBUG
-# include <iostream>
+#include <iostream>
+#include <unistd.h>
   using namespace std;
 #endif
 
diff -Naur ois-v1-3.orig/src/linux/LinuxJoyStickEvents.cpp ois-v1-3/src/linux/LinuxJoyStickEvents.cpp
--- ois-v1-3.orig/src/linux/LinuxJoyStickEvents.cpp	2010-07-26 01:28:41.000000000 +0000
+++ ois-v1-3/src/linux/LinuxJoyStickEvents.cpp	2012-07-10 02:27:07.663922138 +0000
@@ -31,12 +31,13 @@
 #include "OISException.h"
 
 #include <fcntl.h>        //Needed to Open a file descriptor
-#include <cassert>	
+#include <cassert>
 #include <linux/input.h>
 
 
 #include <sstream>
-# include <iostream>
+#include <iostream>
+#include <unistd.h>
 using namespace std;
 
 using namespace OIS;
