diff -rNu nedit-5.5.orig/source/nedit.c nedit-5.5/source/nedit.c
--- nedit-5.5.orig/source/nedit.c	2014-04-15 21:37:41.893066434 +0200
+++ nedit-5.5/source/nedit.c	2014-04-15 21:41:30.262069783 +0200
@@ -389,6 +389,9 @@
 	    "-bw", "-title", NULL};
     unsigned char* invalidBindings = NULL;
     
+    /* workaround for lesstif */
+    setenv("XLIB_SKIP_ARGB_VISUALS", "1", 1);
+
     /* Save the command which was used to invoke nedit for restart command */
     ArgV0 = argv[0];
 
diff -rNu nedit-5.5.orig/util/check_lin_tif.c nedit-5.5/util/check_lin_tif.c
--- nedit-5.5.orig/util/check_lin_tif.c	2014-04-15 21:37:41.973066435 +0200
+++ nedit-5.5/util/check_lin_tif.c	2014-04-15 21:38:39.407067277 +0200
@@ -87,6 +87,7 @@
     "0.93.12",
     "0.93.18",
     "0.93.94",
+    "0.95.2",
     NULL
 };
 
