diff -u -r nedit-5.6/util/check_lin_tif.c nedit-5.6-patched/util/check_lin_tif.c
--- nedit-5.6/util/check_lin_tif.c	2014-12-28 13:56:21.000000000 +0000
+++ nedit-5.6-patched/util/check_lin_tif.c	2015-08-07 11:18:29.404307905 +0100
@@ -103,28 +103,11 @@
     if (stability == MotifUnknown)
     {
         /* This version is neither known-good nor known-bad */
-        fprintf(stderr, 
-                "ERROR:  Untested Motif Version:\n\t%s\n",
-                XmVERSION_STRING);
-
-        fprintf(stderr, 
-            "You are attempting to build NEdit with a version of Motif that\n"
-            "has not been verified to work well with NEdit.  This could be fine,\n"
-            "but it could also lead to crashes and instability.  Historically, \n"
-            "older versions of Motif have quite often been more stable\n"
-            "than newer versions when used with NEdit, so don't assume newer\n"
-            "is better.\n");
+        /* Yeah, whatevs, it's the only version we have, mister snottypants */
 
 #ifdef BUILD_UNTESTED_NEDIT
         {
-            char buf[2];
-            fprintf(stderr,
-                "\n========================== WARNING ===========================\n"
-                "You have chosen to build NEdit with an untested version of Motif.\n"
-                "Please report your success or failure with this version to:\n"
-                "\tdevelop@nedit.org\n"
-                "\nHIT ENTER TO CONTINUE\n");
-            fgets(buf, 2, stdin);
+            /* Bollocks. */
             return EXIT_SUCCESS;
         }
 #else
