diff -ruN subtitlecomposer-0.5.3-old/src/config/appconfig.cpp subtitlecomposer-0.5.3/src/config/appconfig.cpp
--- subtitlecomposer-0.5.3-old/src/config/appconfig.cpp	2009-02-25 23:46:09.000000000 +0100
+++ subtitlecomposer-0.5.3/src/config/appconfig.cpp	2010-05-31 19:38:02.402837345 +0200
@@ -120,7 +120,7 @@
  	return m_groups.contains( name ) ? m_groups[name] : 0;
 }
 
-const AppConfigGroup* const AppConfig::group( const QString& name ) const
+const AppConfigGroup* AppConfig::group( const QString& name ) const
 {
 	return m_groups.contains( name ) ? m_groups[name] : 0;
 }
diff -ruN subtitlecomposer-0.5.3-old/src/config/appconfiggroupwidget.cpp subtitlecomposer-0.5.3/src/config/appconfiggroupwidget.cpp
--- subtitlecomposer-0.5.3-old/src/config/appconfiggroupwidget.cpp	2009-02-25 23:46:09.000000000 +0100
+++ subtitlecomposer-0.5.3/src/config/appconfiggroupwidget.cpp	2010-05-31 19:35:25.526185415 +0200
@@ -39,7 +39,7 @@
 	delete m_config;
 }
 
-const AppConfigGroup* const AppConfigGroupWidget::config()
+const AppConfigGroup* AppConfigGroupWidget::config()
 {
 	return m_config;
 }
diff -ruN subtitlecomposer-0.5.3-old/src/config/appconfiggroupwidget.h subtitlecomposer-0.5.3/src/config/appconfiggroupwidget.h
--- subtitlecomposer-0.5.3-old/src/config/appconfiggroupwidget.h	2009-02-25 23:46:09.000000000 +0100
+++ subtitlecomposer-0.5.3/src/config/appconfiggroupwidget.h	2010-05-31 19:35:20.630352728 +0200
@@ -43,7 +43,7 @@
 			explicit AppConfigGroupWidget( AppConfigGroup* configGroup, QWidget* parent=0 );
 			virtual ~AppConfigGroupWidget();
 
-			const AppConfigGroup* const config();
+			const AppConfigGroup* config();
 
 		signals:
 
diff -ruN subtitlecomposer-0.5.3-old/src/config/appconfig.h subtitlecomposer-0.5.3/src/config/appconfig.h
--- subtitlecomposer-0.5.3-old/src/config/appconfig.h	2009-02-25 23:46:09.000000000 +0100
+++ subtitlecomposer-0.5.3/src/config/appconfig.h	2010-05-31 19:37:54.703670415 +0200
@@ -58,7 +58,7 @@
 			void writeTo( KSharedConfig* config ) const;
 
 			AppConfigGroup* group( const QString& name );
-			const AppConfigGroup* const group( const QString& name ) const;
+			const AppConfigGroup* group( const QString& name ) const;
 
 			void setGroup( AppConfigGroup* group ); /// ownership is transferred to this object
 			AppConfigGroup* removeGroup( const QString& name ); /// ownership is transferred to the caller
diff -ruN subtitlecomposer-0.5.3-old/src/core/sstring.cpp subtitlecomposer-0.5.3/src/core/sstring.cpp
--- subtitlecomposer-0.5.3-old/src/core/sstring.cpp	2009-05-24 23:56:30.000000000 +0200
+++ subtitlecomposer-0.5.3/src/core/sstring.cpp	2010-05-31 19:03:39.173876937 +0200
@@ -1112,7 +1112,7 @@
 	append( str );
 }
 
-SStringList::SStringList( const SStringList::SStringList& list ):
+SStringList::SStringList( const SStringList& list ):
 	QList<SString>( list )
 {
 }
diff -ruN subtitlecomposer-0.5.3-old/src/core/subtitleactions.cpp subtitlecomposer-0.5.3/src/core/subtitleactions.cpp
--- subtitlecomposer-0.5.3-old/src/core/subtitleactions.cpp	2009-02-25 23:46:11.000000000 +0100
+++ subtitlecomposer-0.5.3/src/core/subtitleactions.cpp	2010-05-31 19:35:47.497016112 +0200
@@ -136,7 +136,7 @@
 
 	SubtitleLine* line;
 	int insertOffset = 0;
-	int lineIndex;
+	int lineIndex = 0;
 	while ( ! m_lines.isEmpty() )
 	{
 		line = m_lines.takeFirst();
@@ -242,7 +242,7 @@
 	emit m_subtitle.linesAboutToBeInserted( m_firstIndex, m_lastIndex );
 
 	int insertOffset = 0;
-	int lineIndex;
+	int lineIndex = 0;
 	while ( ! m_lines.isEmpty() )
 	{
 		SubtitleLine* line = m_lines.takeFirst();
diff -ruN subtitlecomposer-0.5.3-old/src/core/subtitle.cpp subtitlecomposer-0.5.3/src/core/subtitle.cpp
--- subtitlecomposer-0.5.3-old/src/core/subtitle.cpp	2009-05-24 10:08:53.000000000 +0200
+++ subtitlecomposer-0.5.3/src/core/subtitle.cpp	2010-05-31 19:36:33.366179618 +0200
@@ -216,7 +216,7 @@
 	}
 }
 
-FormatData* const Subtitle::formatData() const
+FormatData* Subtitle::formatData() const
 {
 	return m_formatData;
 }
@@ -1201,7 +1201,7 @@
 		return;
 
 	int splitIndex = -1; // the index of the first line to move (or copy) to dstSub
-	bool splitsLine; // splitTime falls in within a line's time
+	bool splitsLine = false; // splitTime falls in within a line's time
 
 	QList<SubtitleLine*> lines;
 	for ( SubtitleIterator it( *this, Range::full() ); it.current(); ++it )
diff -ruN subtitlecomposer-0.5.3-old/src/core/subtitle.h subtitlecomposer-0.5.3/src/core/subtitle.h
--- subtitlecomposer-0.5.3-old/src/core/subtitle.h	2009-03-05 03:56:25.000000000 +0100
+++ subtitlecomposer-0.5.3/src/core/subtitle.h	2010-05-31 19:36:28.420346163 +0200
@@ -183,7 +183,7 @@
 
 		private:
 
-			FormatData* const formatData() const;
+			FormatData* formatData() const;
 			void setFormatData( const FormatData* formatData );
 
 			void beginCompositeAction( const QString& title, bool immediateExecution=true, bool delaySignals=true );
diff -ruN subtitlecomposer-0.5.3-old/src/core/subtitleline.cpp subtitlecomposer-0.5.3/src/core/subtitleline.cpp
--- subtitlecomposer-0.5.3-old/src/core/subtitleline.cpp	2009-03-03 05:16:07.000000000 +0100
+++ subtitlecomposer-0.5.3/src/core/subtitleline.cpp	2010-05-31 19:35:54.000000000 +0200
@@ -289,7 +289,7 @@
 }
 
 
-FormatData* const SubtitleLine::formatData() const
+FormatData* SubtitleLine::formatData() const
 {
 	return m_formatData;
 }
diff -ruN subtitlecomposer-0.5.3-old/src/core/subtitleline.h subtitlecomposer-0.5.3/src/core/subtitleline.h
--- subtitlecomposer-0.5.3-old/src/core/subtitleline.h	2009-03-03 05:16:07.000000000 +0100
+++ subtitlecomposer-0.5.3/src/core/subtitleline.h	2010-05-31 19:36:02.793682498 +0200
@@ -247,7 +247,7 @@
 
 		private:
 
-			FormatData* const formatData() const;
+			FormatData* formatData() const;
 			void setFormatData( const FormatData* formatData );
 
 			void processAction( Action* action );
diff -ruN subtitlecomposer-0.5.3-old/src/player/gstreamer/gstreamerbackend.cpp subtitlecomposer-0.5.3/src/player/gstreamer/gstreamerbackend.cpp
--- subtitlecomposer-0.5.3-old/src/player/gstreamer/gstreamerbackend.cpp	2009-02-25 23:46:17.000000000 +0100
+++ subtitlecomposer-0.5.3/src/player/gstreamer/gstreamerbackend.cpp	2010-05-31 19:37:23.237008638 +0200
@@ -596,7 +596,7 @@
 	params = g_object_class_list_properties( G_OBJECT_GET_CLASS( GST_ELEMENT( object ) ), &length );
 	for ( guint index = 0; index < length; ++index )
 	{
-		gchar* strValue;
+		gchar* strValue = 0;
 
 		if ( params[index]->flags & G_PARAM_READABLE )
 		{
diff -ruN subtitlecomposer-0.5.3-old/src/player/gstreamer/gstreamerbackend.h subtitlecomposer-0.5.3/src/player/gstreamer/gstreamerbackend.h
--- subtitlecomposer-0.5.3-old/src/player/gstreamer/gstreamerbackend.h	2009-02-25 23:46:17.000000000 +0100
+++ subtitlecomposer-0.5.3/src/player/gstreamer/gstreamerbackend.h	2010-05-31 19:34:31.977024143 +0200
@@ -47,7 +47,7 @@
 			GStreamerBackend( Player* player );
 			virtual ~GStreamerBackend();
 
-			const GStreamerConfig* const config() { return static_cast<const GStreamerConfig* const>( PlayerBackend::config() ); }
+			const GStreamerConfig* config() { return static_cast<const GStreamerConfig* const>( PlayerBackend::config() ); }
 
 			virtual AppConfigGroupWidget* newAppConfigGroupWidget( QWidget* parent );
 
diff -ruN subtitlecomposer-0.5.3-old/src/player/mplayer/mplayerbackend.h subtitlecomposer-0.5.3/src/player/mplayer/mplayerbackend.h
--- subtitlecomposer-0.5.3-old/src/player/mplayer/mplayerbackend.h	2009-02-25 23:46:17.000000000 +0100
+++ subtitlecomposer-0.5.3/src/player/mplayer/mplayerbackend.h	2010-05-31 19:34:38.849523428 +0200
@@ -45,7 +45,7 @@
 			MPlayerBackend( Player* player );
 			virtual ~MPlayerBackend();
 
-			const MPlayerConfig* const config() { return static_cast<const MPlayerConfig* const>( PlayerBackend::config() ); }
+			const MPlayerConfig* config() { return static_cast<const MPlayerConfig* const>( PlayerBackend::config() ); }
 
 			virtual AppConfigGroupWidget* newAppConfigGroupWidget( QWidget* parent );
 
diff -ruN subtitlecomposer-0.5.3-old/src/player/phonon/phononbackend.h subtitlecomposer-0.5.3/src/player/phonon/phononbackend.h
--- subtitlecomposer-0.5.3-old/src/player/phonon/phononbackend.h	2009-02-25 23:46:16.000000000 +0100
+++ subtitlecomposer-0.5.3/src/player/phonon/phononbackend.h	2010-05-31 19:34:44.607023515 +0200
@@ -53,7 +53,7 @@
 			PhononBackend( Player* player );
 			virtual ~PhononBackend();
 
-			const PhononConfig* const config() { return static_cast<const PhononConfig* const>( PlayerBackend::config() ); }
+			const PhononConfig* config() { return static_cast<const PhononConfig* const>( PlayerBackend::config() ); }
 
 			virtual AppConfigGroupWidget* newAppConfigGroupWidget( QWidget* parent );
 
diff -ruN subtitlecomposer-0.5.3-old/src/player/playerbackend.h subtitlecomposer-0.5.3/src/player/playerbackend.h
--- subtitlecomposer-0.5.3-old/src/player/playerbackend.h	2009-02-25 23:46:17.000000000 +0100
+++ subtitlecomposer-0.5.3/src/player/playerbackend.h	2010-05-31 19:34:18.996192330 +0200
@@ -48,7 +48,7 @@
 			PlayerBackend( Player* player, AppConfigGroup* config );
 			virtual ~PlayerBackend();
 
-			const AppConfigGroup* const config() const { return m_config; }
+			const AppConfigGroup* config() const { return m_config; }
 
 			/// If possible (i.e., configs are compatible), copies the config object into
 			/// the player backend config. Ownership of config object it's not transferred.
diff -ruN subtitlecomposer-0.5.3-old/src/player/player.cpp subtitlecomposer-0.5.3/src/player/player.cpp
--- subtitlecomposer-0.5.3-old/src/player/player.cpp	2009-02-25 23:46:17.000000000 +0100
+++ subtitlecomposer-0.5.3/src/player/player.cpp	2010-05-31 19:38:38.306165999 +0200
@@ -82,7 +82,7 @@
 {
 }
 
-Player* const Player::instance()
+Player* Player::instance()
 {
 	static Player player;
 
diff -ruN subtitlecomposer-0.5.3-old/src/player/player.h subtitlecomposer-0.5.3/src/player/player.h
--- subtitlecomposer-0.5.3-old/src/player/player.h	2009-02-25 23:46:17.000000000 +0100
+++ subtitlecomposer-0.5.3/src/player/player.h	2010-05-31 19:38:31.893666505 +0200
@@ -53,7 +53,7 @@
 				Ready // same as Stopped or Finished
 			} State;
 
-			static Player* const instance();
+			static Player* instance();
 
 			inline VideoWidget* videoWidget();
 
diff -ruN subtitlecomposer-0.5.3-old/src/player/xine/xinebackend.h subtitlecomposer-0.5.3/src/player/xine/xinebackend.h
--- subtitlecomposer-0.5.3-old/src/player/xine/xinebackend.h	2009-02-25 23:46:16.000000000 +0100
+++ subtitlecomposer-0.5.3/src/player/xine/xinebackend.h	2010-05-31 19:34:49.336188698 +0200
@@ -51,7 +51,7 @@
 			XineBackend( Player* player );
 			virtual ~XineBackend();
 
-			const XineConfig* const config() { return static_cast<const XineConfig* const>( PlayerBackend::config() ); }
+			const XineConfig* config() { return static_cast<const XineConfig* const>( PlayerBackend::config() ); }
 
 			virtual AppConfigGroupWidget* newAppConfigGroupWidget( QWidget* parent );
 
diff -ruN subtitlecomposer-0.5.3-old/src/widgets/simplerichtextedit.cpp subtitlecomposer-0.5.3/src/widgets/simplerichtextedit.cpp
--- subtitlecomposer-0.5.3-old/src/widgets/simplerichtextedit.cpp	2009-02-25 23:46:10.000000000 +0100
+++ subtitlecomposer-0.5.3/src/widgets/simplerichtextedit.cpp	2010-05-31 19:42:18.002811365 +0200
@@ -176,7 +176,7 @@
 		delete m_insertUnicodeControlCharMenu->parent();
 }
 
-KAction* const SimpleRichTextEdit::action( int action ) const
+KAction* SimpleRichTextEdit::action( int action ) const
 {
 	return action >= 0 && action < ActionCount ? m_actions[action] : 0;
 }
diff -ruN subtitlecomposer-0.5.3-old/src/widgets/simplerichtextedit.h subtitlecomposer-0.5.3/src/widgets/simplerichtextedit.h
--- subtitlecomposer-0.5.3-old/src/widgets/simplerichtextedit.h	2009-02-25 23:46:10.000000000 +0100
+++ subtitlecomposer-0.5.3/src/widgets/simplerichtextedit.h	2010-05-31 19:42:11.763644715 +0200
@@ -60,7 +60,7 @@
 		bool fontBold();
 		bool fontStrikeOut();
 
-		virtual KAction* const action( int action ) const;
+		virtual KAction* action( int action ) const;
 
 		virtual bool event( QEvent* event );
 
