--- xss-lock-0.3.0/CMakeLists.txt	2013-11-05 02:10:10.000000000 +0900
+++ xss-lock-0.3.0/CMakeLists.txt.new	2021-04-08 08:03:28.349195750 +0900
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 2.8.12)
 project(xss-lock C)
 set(PROJECT_VERSION 0.3.0)

--- xss-lock-0.3.0/src/CMakeLists.txt	2013-11-05 02:10:10.000000000 +0900
+++ xss-lock-0.3.0/src/CMakeLists.txt.new	2021-04-08 08:25:15.923235654 +0900
@@ -17,7 +17,7 @@
     xss-lock.c
     xcb_utils.c
     xcb_utils.h
-    config.h
+    config.h.in
 )

 target_link_libraries(xss-lock ${GLIB2_LIBRARIES} ${XCB_LIBRARIES})
