diff --git a/data/com.system76.PowerDaemon.conf b/data/com.system76.PowerDaemon.conf
index 335981c..1705b0c 100644
--- a/data/com.system76.PowerDaemon.conf
+++ b/data/com.system76.PowerDaemon.conf
@@ -6,17 +6,9 @@
         <allow send_destination="com.system76.PowerDaemon"/>
         <allow receive_sender="com.system76.PowerDaemon"/>
     </policy>
-    <policy group="sudo">
-        <allow send_destination="com.system76.PowerDaemon"/>
-        <allow receive_sender="com.system76.PowerDaemon"/>
-    </policy>
     <policy user="root">
         <allow own="com.system76.PowerDaemon"/>
         <allow send_destination="com.system76.PowerDaemon"/>
         <allow receive_sender="com.system76.PowerDaemon"/>
     </policy>
-    <policy user="gnome-initial-setup">
-        <allow send_destination="com.system76.PowerDaemon"/>
-        <allow receive_sender="com.system76.PowerDaemon"/>
-    </policy>
 </busconfig>
