diff -Naur ldns-1.6.17.orig/packaging/ldns-config.in ldns-1.6.17/packaging/ldns-config.in
--- ldns-1.6.17.orig/packaging/ldns-config.in	2014-01-10 21:04:41.000000000 +0000
+++ ldns-1.6.17/packaging/ldns-config.in	2016-08-15 02:32:32.794747069 +0000
@@ -9,7 +9,7 @@
 LIBS="@LIBS@ @LIBSSL_LIBS@"
 LIBDIR="@libdir@"
 INCLUDEDIR="@includedir@"
-LIBVERSION="@LIBLDNS_CURRENT@.@LIBLDNS_REVISION@.@LIBLDNS_AGE@"
+LIBVERSION="@PACKAGE_VERSION@"
 
 for arg in $@
 do
