--- configure.ac.orig	2016-06-24 00:15:55.000000000 +1000
+++ configure.ac	2016-08-08 16:53:47.075986217 +1000
@@ -232,8 +232,6 @@
 		 man/nbd-server.1.sh
 		 man/nbd-trdump.1.sh
 		 man/nbdtab.5.sh
-		 systemd/Makefile
-		 systemd/nbd@.service.sh
 		])
 AC_OUTPUT
 
--- Makefile.am.orig	2016-06-24 00:15:55.000000000 +1000
+++ Makefile.am	2016-08-08 17:03:26.635959597 +1000
@@ -1,4 +1,4 @@
-SUBDIRS = . man doc tests systemd gznbd
+SUBDIRS = . man doc tests gznbd
 bin_PROGRAMS = nbd-server nbd-trdump
 sbin_PROGRAMS = @NBD_CLIENT_NAME@
 EXTRA_PROGRAMS = nbd-client make-integrityhuge
--- /dev/null	2016-08-08 16:48:51.867999776 +1000
+++ support/genver.sh	2016-08-08 17:07:50.087947496 +1000
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+echo "3.14"
