--- spacezero/Makefile.original	2014-04-07 20:07:25.673407307 +0300
+++ spacezero/Makefile	2014-04-07 20:08:30.901591842 +0300
@@ -9,11 +9,11 @@
 RECORDFILE=$(CONFIGDIR)/record
 
 # The binary executable will be copied in the next directory.
-INSTALL_DIR=/usr/local/games
+INSTALL_DIR=/usr/bin
 
 # All data files will be copied in the next directory.
 # tar.gz 
-INSTALL_DATA_DIR=/usr/local/share/spacezero
+INSTALL_DATA_DIR=/usr/share/games/SpaceZero
 
 # DEB pkg 
 #INSTALL_DIR=/usr/games
@@ -27,7 +27,7 @@
 GTKFLAGS12=`gtk-config --libs`
 
 CFLAGS=  -Wall -O3 --pedantic -I./include -DDATADIR=\"$(DATADIR)\" -DINSTALL_DATA_DIR=\"$(INSTALL_DATA_DIR)\"
-LDFLAGS= -lm -lpthread $(SOUNDFLAGS) $(GTKLIBS) 
+LDFLAGS= -lm -lpthread -lX11 -lXext -ldl $(SOUNDFLAGS) $(GTKLIBS) 
 
 # DEB pkg
 #CFLAGS += `dpkg-buildflags --get CFLAGS`
