diff -Naur hxtools-20211204/smm/extract_d3pkg hxtools-20211204.patched/smm/extract_d3pkg
--- hxtools-20211204/smm/extract_d3pkg	2021-12-04 08:56:44.000000000 -0500
+++ hxtools-20211204.patched/smm/extract_d3pkg	2022-01-27 15:49:49.273028431 -0500
@@ -10,8 +10,8 @@
 
 use Getopt::Long;
 use strict;
-do (($0 =~ m{^(.*)/})[0] || ".")."/shared.pm" ||
-	die "Could not load shared.pm: $!\n";
+do "/usr/share/hxtools/extract_dfqshared.pm" ||
+	die "Could not load extract_dfqshared.pm: $!\n";
 
 my $ex_out_dir      = ".";
 my $ex_archive_file = "-";
diff -Naur hxtools-20211204/smm/extract_dxhog hxtools-20211204.patched/smm/extract_dxhog
--- hxtools-20211204/smm/extract_dxhog	2021-12-04 08:56:44.000000000 -0500
+++ hxtools-20211204.patched/smm/extract_dxhog	2022-01-27 15:50:02.081027267 -0500
@@ -10,8 +10,8 @@
 
 use Getopt::Long;
 use strict;
-do (($0 =~ m{^(.*)/})[0] || ".")."/shared.pm" ||
-	die "Could not load shared.pm: $!\n";
+do "/usr/share/hxtools/extract_dfqshared.pm" ||
+   die "Could not load extract_dfqshared.pm: $!\n";
 
 my $ex_out_dir      = ".";
 my $ex_archive_file = "-";
diff -Naur hxtools-20211204/smm/extract_f3pod hxtools-20211204.patched/smm/extract_f3pod
--- hxtools-20211204/smm/extract_f3pod	2021-12-04 08:56:44.000000000 -0500
+++ hxtools-20211204.patched/smm/extract_f3pod	2022-01-27 15:50:12.877026285 -0500
@@ -10,8 +10,8 @@
 
 use Getopt::Long;
 use strict;
-do (($0 =~ m{^(.*)/})[0] || ".")."/shared.pm" ||
-	die "Could not load shared.pm: $!\n";
+do "/usr/share/hxtools/extract_dfqshared.pm" ||
+   die "Could not load extract_dfqshared.pm: $!\n";
 
 my $ex_out_dir      = ".";
 my $ex_archive_file = "-";
diff -Naur hxtools-20211204/smm/extract_qupak hxtools-20211204.patched/smm/extract_qupak
--- hxtools-20211204/smm/extract_qupak	2021-12-04 08:56:44.000000000 -0500
+++ hxtools-20211204.patched/smm/extract_qupak	2022-01-27 15:50:19.046025725 -0500
@@ -10,8 +10,8 @@
 
 use Getopt::Long;
 use strict;
-do (($0 =~ m{^(.*)/})[0] || ".")."/shared.pm" ||
-	die "Could not load shared.pm: $!\n";
+do "/usr/share/hxtools/extract_dfqshared.pm" ||
+   die "Could not load extract_dfqshared.pm: $!\n";
 
 my $ex_out_dir      = ".";
 my $ex_archive_file = "-";
