diff -Nur xmlstarlet-1.0.1.orig/configure xmlstarlet-1.0.1/configure
--- xmlstarlet-1.0.1.orig/configure	2005-03-13 19:32:18.000000000 -0600
+++ xmlstarlet-1.0.1/configure	2009-08-15 11:40:10.052257819 -0500
@@ -1095,7 +1095,7 @@
   withval="$with_libxml_prefix"
   LIBXML_PREFIX=$withval
         LIBXML_CFLAGS="-I$withval/include/libxml2 -I$withval/include/libxml"
-        LIBXML_LIBS="$withval/lib/libxml2.a -lz -lm -lpthread -ldl"
+        LIBXML_LIBS="$withval/lib64/libxml2.a -lz -lm -lpthread -ldl"
 
 fi
 
@@ -1144,7 +1144,7 @@
   withval="$with_libxslt_prefix"
   LIBXSLT_PREFIX=$withval
         LIBXSLT_CFLAGS="-I$withval/include/libxslt -I$withval/include/libexslt"
-        LIBXSLT_LIBS="$withval/lib/libxslt.a $withval/lib/libexslt.a -lz -lm -lpthread"
+        LIBXSLT_LIBS="$withval/lib64/libxslt.a $withval/lib64/libexslt.a -lz -lm -lpthread"
 
 fi
 
@@ -1543,7 +1543,7 @@
   echo "configure: failed program was:" >&5
   cat conftest.$ac_ext >&5
   rm -rf conftest*
-  CPP=/lib/cpp
+  CPP=/lib64/cpp
 fi
 rm -f conftest*
 fi
@@ -2159,7 +2159,7 @@
 
 if test "x$LIBXML_LIBS" = "x"
 then
-	LIBXML_LIBS="${LIBXML_PREFIX}/lib/libxml2.a -lz -lm -lpthread -ldl"
+	LIBXML_LIBS="${LIBXML_PREFIX}/lib64/libxml2.a -lz -lm -lpthread -ldl"
 fi
 
 if test "x$LIBXSLT_CFLAGS" = "x"
@@ -2169,7 +2169,7 @@
 
 if test "x$LIBXSLT_LIBS" = "x"
 then
-	LIBXSLT_LIBS="${LIBXSLT_PREFIX}/lib/libxslt.a ${LIBXSLT_PREFIX}/lib/libexslt.a -lz -lm -lpthread"
+	LIBXSLT_LIBS="${LIBXSLT_PREFIX}/lib64/libxslt.a ${LIBXSLT_PREFIX}/lib64/libexslt.a -lz -lm -lpthread"
 fi
 
 if test "x$LIBICONV_CFLAGS" = "x"
