diff -ur aiksaurus-1.2.1.orig/base/aiksaurus.t.cpp aiksaurus-1.2.1/base/aiksaurus.t.cpp
--- aiksaurus-1.2.1.orig/base/aiksaurus.t.cpp	2003-06-27 02:44:38.000000000 +0100
+++ aiksaurus-1.2.1/base/aiksaurus.t.cpp	2009-08-30 11:29:05.000000000 +0100
@@ -24,6 +24,7 @@
 #include "config.h"
 #include <iostream>
 #include <string>
+#include <cstdlib>
 using namespace std;
 using AiksaurusImpl::AsciiCompare;
 
diff -ur aiksaurus-1.2.1.orig/gtk/src/Display.h aiksaurus-1.2.1/gtk/src/Display.h
--- aiksaurus-1.2.1.orig/gtk/src/Display.h	2002-07-11 11:09:37.000000000 +0100
+++ aiksaurus-1.2.1/gtk/src/Display.h	2009-08-30 11:28:17.000000000 +0100
@@ -5,6 +5,7 @@
 #include <gtk/gtk.h>
 #include <vector>
 #include <string>
+#include <strings.h>
 #include <new>
 #include "Exception.h"
 
diff -ur aiksaurus-1.2.1.orig/gtk/src/gtkAiksaur.t.cpp aiksaurus-1.2.1/gtk/src/gtkAiksaur.t.cpp
--- aiksaurus-1.2.1.orig/gtk/src/gtkAiksaur.t.cpp	2002-01-12 02:02:12.000000000 +0000
+++ aiksaurus-1.2.1/gtk/src/gtkAiksaur.t.cpp	2009-08-30 11:29:11.000000000 +0100
@@ -23,6 +23,7 @@
 #include <gtk/gtk.h>
 #include <iostream>
 #include <string>
+#include <cstdlib>
 using namespace std;
 
 int main(int argc, char** argv)
