--- repology-1.10.0/repology.d	2026-06-08 02:22:06.000000000 +0300
+++ repology.d	2026-09-19 00:07:47.918386827 +0300
@@ -50,7 +50,7 @@
     Options options;
     JSONValue json;
     string[] pkgs;
-    string uri = "https://repology.org/api/v1/project";
+    string uri = "https://repology.amdmi3.ru/api/v1/project";
     string[string] queryParts = null;
 
     version (OpenBSD) options.repo = "openbsd";
