diff -Naur ghc-8.10.4.orig/docs/users_guide/ghc_config.py.in ghc-8.10.4/docs/users_guide/ghc_config.py.in
--- ghc-8.10.4.orig/docs/users_guide/ghc_config.py.in	2020-12-02 15:16:57.000000000 +0100
+++ ghc-8.10.4/docs/users_guide/ghc_config.py.in	2025-06-17 22:42:02.429581000 +0200
@@ -1,6 +1,6 @@
 extlinks = {
-    'ghc-ticket': ('https://gitlab.haskell.org/ghc/ghc/issues/%s', '#'),
-    'ghc-wiki': ('https://gitlab.haskell.org/ghc/ghc/wikis/%s', '#'),
+    'ghc-ticket': ('https://gitlab.haskell.org/ghc/ghc/issues/%s', '%s'),
+    'ghc-wiki': ('https://gitlab.haskell.org/ghc/ghc/wikis/%s', '%s'),
 }
 
 libs_base_uri = '../libraries'
