diff -Nur tvtime-1.0.2-old/src/videoinput.c tvtime-1.0.2/src/videoinput.c
--- tvtime-1.0.2-old/src/videoinput.c	2007-07-08 14:29:13.539465861 +0300
+++ tvtime-1.0.2/src/videoinput.c	2007-07-08 14:42:32.776963292 +0300
@@ -377,6 +377,7 @@
         wait_for_frame_v4l2( vidin );
  
         cur_buf.type = vidin->capbuffers[ 0 ].vidbuf.type;
+        cur_buf.memory = vidin->capbuffers[ 0 ].vidbuf.memory;
         if( ioctl( vidin->grab_fd, VIDIOC_DQBUF, &cur_buf ) < 0 ) {
             /* some drivers return EIO when there is no signal */
             if( errno != EIO ) {
