diff -u -r tuxpaint-0.9.22/Makefile tuxpaint-0.9.22-patched/Makefile
--- tuxpaint-0.9.22/Makefile	2014-08-19 08:28:33.000000000 +0100
+++ tuxpaint-0.9.22-patched/Makefile	2015-08-04 14:35:11.258065084 +0100
@@ -138,14 +138,14 @@
 
 # Magic Tool plug-ins
 INCLUDE_PREFIX:=$(DESTDIR)$(PREFIX)/include
-MAGIC_PREFIX:=$(DESTDIR)$(LIBDIR)/lib/tuxpaint/plugins
+MAGIC_PREFIX:=$(DESTDIR)$(PREFIX)/lib$(LIBDIRSUFFIX)/tuxpaint/plugins
 
 
 # Docs and man page:
-DOC_PREFIX:=$(DESTDIR)$(PREFIX)/share/doc/tuxpaint
-DEVDOC_PREFIX:=$(DESTDIR)$(PREFIX)/share/doc/tuxpaint-dev
-MAN_PREFIX:=$(DESTDIR)$(PREFIX)/share/man
-DEVMAN_PREFIX:=$(DESTDIR)$(PREFIX)/share/man
+DOC_PREFIX:=$(DESTDIR)$(PREFIX)/doc/tuxpaint-$(VER_VERSION)
+DEVDOC_PREFIX:=$(DESTDIR)$(PREFIX)/doc/tuxpaint-$(VER_VERSION)/tuxpaint-dev
+MAN_PREFIX:=$(DESTDIR)$(PREFIX)/man
+DEVMAN_PREFIX:=$(DESTDIR)$(PREFIX)/man
 
 # BASH tab-completion file:
 COMPLETIONDIR:=$(DESTDIR)/etc/bash_completion.d
@@ -163,10 +163,10 @@
 
 # Icons and launchers:
 ICON_PREFIX:=$(DESTDIR)$(PREFIX)/share/pixmaps
-X11_ICON_PREFIX:=$(DESTDIR)$(PREFIX)/X11R6/include/X11/pixmaps
+X11_ICON_PREFIX:=$(DESTDIR)$(PREFIX)/share/pixmaps
 GNOME_PREFIX:=$(shell gnome-config --prefix 2> /dev/null)
-KDE_PREFIX:=$(shell kde-config --install apps --expandvars 2> /dev/null)
-KDE_ICON_PREFIX:=$(shell kde-config --install icon --expandvars 2> /dev/null)
+KDE_PREFIX:=/usr/share/applications
+KDE_ICON_PREFIX:=$(shell kde4-config --install icon --expandvars 2> /dev/null)
 
 # Maemo flag
 MAEMOFLAG:=
@@ -435,7 +435,7 @@
 windows_ARCH_INSTALL:=
 osx_ARCH_INSTALL:=
 beos_ARCH_INSTALL:=install-haiku
-linux_ARCH_INSTALL:=install-gnome install-kde install-kde-icons
+linux_ARCH_INSTALL:=install-kde install-kde-icons
 ARCH_INSTALL:=$($(OS)_ARCH_INSTALL)
 
 # "make install" installs all of the various parts
@@ -494,7 +494,8 @@
 	@cp src/tp_magic_api.h $(INCLUDE_PREFIX)/tuxpaint
 	@chmod a+r,g-w,o-w $(INCLUDE_PREFIX)/tuxpaint/tp_magic_api.h
 	@install -d $(DEVDOC_PREFIX)
-	@cp -R magic/docs/* $(DEVDOC_PREFIX)
+	@cp magic/docs/*.txt $(DEVDOC_PREFIX)
+	@cp magic/magic-docs/*.txt $(DEVDOC_PREFIX)
 	@chmod a=rX,g=rX,u=rwX $(DEVDOC_PREFIX)
 
 # Installs the various parts for the MinGW/MSYS development/testing environment.
@@ -573,7 +574,7 @@
 	  rm /usr/share/pixmaps/tuxpaint.png; \
 	fi
 	-if [ "x$(KDE_PREFIX)" != "x" ]; then \
-	  rm $(KDE_PREFIX)/Graphics/tuxpaint.desktop; \
+	  rm $(KDE_PREFIX)/tuxpaint.desktop; \
 	fi
 	-rm $(ICON_PREFIX)/tuxpaint.png
 	-rm $(X11_ICON_PREFIX)/tuxpaint.xpm
@@ -761,9 +762,9 @@
 	@echo
 	@echo "...Installing launcher icon into KDE..."
 	@if [ "x$(KDE_PREFIX)" != "x" ]; then \
-	  install -d $(DESTDIR)$(KDE_PREFIX)/Graphics; \
-	  cp src/tuxpaint.desktop $(DESTDIR)$(KDE_PREFIX)/Graphics/; \
-	  chmod 644 $(DESTDIR)$(KDE_PREFIX)/Graphics/tuxpaint.desktop; \
+	  install -d $(DESTDIR)$(KDE_PREFIX)/; \
+	  cp src/tuxpaint.desktop $(DESTDIR)$(KDE_PREFIX)/; \
+	  chmod 644 $(DESTDIR)$(KDE_PREFIX)/tuxpaint.desktop; \
 	fi
 
 .PHONY: install-kde-icons
@@ -944,8 +945,7 @@
 	@echo
 	@echo "...Installing documentation..."
 	@install -d $(DOC_PREFIX)
-	@cp -R docs/* $(DOC_PREFIX)
-	@cp -R magic/magic-docs $(DOC_PREFIX)
+	@cp docs/*.txt $(DOC_PREFIX)
 	@chmod -R a=rX,g=rX,u=rwX $(DOC_PREFIX)
 
 
@@ -959,21 +959,21 @@
 	@# tuxpaint.1
 	@cp src/manpage/tuxpaint.1 $(MAN_PREFIX)/man1
 	@gzip -f $(MAN_PREFIX)/man1/tuxpaint.1
-	@chmod a+rx,g-w,o-w $(MAN_PREFIX)/man1/tuxpaint.1.gz
+	@chmod 644 $(MAN_PREFIX)/man1/tuxpaint.1.gz
 	@# pl/man1 directory...
 	@install -d $(MAN_PREFIX)/pl/man1/
 	@# tuxpaint-pl.1
 	@cp src/manpage/tuxpaint-pl.1 $(MAN_PREFIX)/pl/man1/tuxpaint.1
 	@gzip -f $(MAN_PREFIX)/pl/man1/tuxpaint.1
-	@chmod a+rx,g-w,o-w $(MAN_PREFIX)/pl/man1/tuxpaint.1.gz
+	@chmod 644 $(MAN_PREFIX)/pl/man1/tuxpaint.1.gz
 	@# tuxpaint-import.1
 	@cp src/manpage/tuxpaint-import.1 $(MAN_PREFIX)/man1/
 	@gzip -f $(MAN_PREFIX)/man1/tuxpaint-import.1
-	@chmod a+rx,g-w,o-w $(MAN_PREFIX)/man1/tuxpaint-import.1.gz
+	@chmod 644 $(MAN_PREFIX)/man1/tuxpaint-import.1.gz
 	@# tp-magic-config.1
 	@cp src/manpage/tp-magic-config.1 $(MAN_PREFIX)/man1/
 	@gzip -f $(MAN_PREFIX)/man1/tp-magic-config.1
-	@chmod a+rx,g-w,o-w $(MAN_PREFIX)/man1/tp-magic-config.1.gz
+	@chmod 644 $(MAN_PREFIX)/man1/tp-magic-config.1.gz
 
 
 
