diff -Naur a/src/AudioFilename.cpp b/src/AudioFilename.cpp
--- a/src/AudioFilename.cpp	2014-07-25 01:35:40.000000000 +0000
+++ b/src/AudioFilename.cpp	2022-02-04 05:04:13.086368320 +0000
@@ -240,7 +240,7 @@
       }
     else {
       input_decompose.assign(&input[i],input.extent(i));
-      input_decompose = decompose(input_decompose,DecomposeCompat);
+      input_decompose = FXString::decompose(input_decompose,DecomposeCompat);
       for (j=0;j<input_decompose.length();j=input_decompose.inc(j)){
         len = codec->utf2mb(&c,1,&input_decompose[j],input_decompose.extent(j));
         if (len>0 && c!=0x1A) {
@@ -256,7 +256,7 @@
 static FXString convert_and_decompose(const FXString & input) {
   register FXint i=0;
   FXString result;
-  FXString in = decompose(input,DecomposeCanonical);
+  FXString in = FXString::decompose(input,DecomposeCanonical);
   for (i=0;i<in.length();i=in.inc(i)){
     if (Ascii::isAscii(in[i]) && Ascii::isPrint(in[i]) ) {
       result+=in[i];
@@ -273,7 +273,7 @@
   result = filter(input,forbidden,opts);
 
   /// Make sure it is properly composed. Should we do this?
-  result = compose(result,DecomposeCompat);
+  result = FXString::compose(result,DecomposeCompat);
 
   /// convert to given codec.
   if (dynamic_cast<FXUTF8Codec*>(codec)==NULL)
@@ -290,7 +290,7 @@
   result = filter(input,forbidden,opts);
 
   /// Make sure it is properly composed. Should we do this?
-  result = compose(result,DecomposeCompat);
+  result = FXString::compose(result,DecomposeCompat);
 
   /// convert to given codec.
   result = convert_and_decompose(result);
