--- vboxadd.sh	2012-04-26 18:56:57.000000000 +0200
+++ vboxadd.sh	2012-05-06 11:55:44.161897154 +0200
@@ -28,10 +28,6 @@
 # Description:    VirtualBox Linux Additions kernel modules
 ### END INIT INFO
 
-. /var/lib/VBoxGuestAdditions/config
-export BUILD_TYPE
-export USERNAME
-
 PATH=$PATH:/bin:/sbin:/usr/sbin
 PACKAGE=VBoxGuestAdditions
 BUILDVBOXGUEST=`/bin/ls /usr/src/vboxguest*/vboxguest/build_in_tmp 2>/dev/null|cut -d' ' -f1`
@@ -500,31 +496,13 @@
 # setup_script
 setup()
 {
-    setup_modules
-    mod_succ="$?"
-    extra_setup
-    if [ "$mod_succ" -eq "0" ]; then
-        if running_vboxguest || running_vboxadd; then
-            printf "You should restart your guest to make sure the new modules are actually used\n\n"
-        else
-            start
-        fi
-    fi
+    echo "Not implemented! Please use the virtualbox-kernel-addons.SlackBuild available at SlackBuilds.org instead."
 }
 
 # cleanup_script
 cleanup()
 {
-    # Delete old versions of VBox modules.
-    cleanup_modules
-    depmod
-
-    # Remove old module sources
-    rm -rf /usr/src/vboxadd-* /usr/src/vboxguest-* /usr/src/vboxvfs-* /usr/src/vboxsf-* /usr/src/vboxvideo-*
-
-    # Remove other files
-    rm /sbin/mount.vboxsf 2>/dev/null
-    rm /etc/udev/rules.d/60-vboxadd.rules 2>/dev/null
+    echo "Not implemented! Please use removepkg or pkgtool to remove virtualbox-addons and/or virtualbox-kernel-addons instead."
 }
 
 dmnstatus()
