--- ./source/MaterialXGenShader/TypeDesc.h
+++ ./source/MaterialXGenShader/fix-TypeDesc.h
@@ -12,6 +12,7 @@
 #include <MaterialXGenShader/Export.h>
 #include <MaterialXCore/Value.h>
 #include <MaterialXCore/Document.h>
+#include <cstdint>
 
 #include <string_view>
 
--- ./source/MaterialXRender/Types.h
+++ ./source/MaterialXRender/fix-Types.h
@@ -12,6 +12,7 @@
 #include <MaterialXRender/Export.h>
 
 #include <MaterialXCore/Types.h>
+#include <cstdint>
 
 MATERIALX_NAMESPACE_BEGIN
 
--- ./source/MaterialXRender/Mesh.h
+++ ./source/MaterialXRender/fix-Mesh.h
@@ -11,6 +11,7 @@
 
 #include <MaterialXCore/Types.h>
 #include <MaterialXRender/Export.h>
+#include <cstdint>
 
 MATERIALX_NAMESPACE_BEGIN
 
