--- ./python/setup.py.in	2023-09-08 21:34:30.000000000 +0200
+++ ./python/setup.py.in	2023-12-29 20:55:34.110939801 +0100
@@ -1,4 +1,4 @@
-from setuptools import setup
+from setuptools import setup, find_packages
 import os
 
 os.chdir(os.path.dirname(os.path.abspath(__file__)))
@@ -13,6 +13,6 @@
 setup(name='MaterialX',
       url='www.materialx.org',
       version='${MATERIALX_MAJOR_VERSION}.${MATERIALX_MINOR_VERSION}.${MATERIALX_BUILD_VERSION}',
-      packages=['MaterialX'],
+      packages=find_packages(['MaterialX']),
       package_data={'MaterialX' : getRecursivePackageData('MaterialX')},
       zip_safe = False)
