Index: chromium-0.9.12/src/Makefile
===================================================================
--- chromium-0.9.12.orig/src/Makefile	2007-04-06 22:52:03.000000000 +0200
+++ chromium-0.9.12/src/Makefile	2007-04-06 22:53:10.000000000 +0200
@@ -10,8 +10,8 @@
 
 CC	=	gcc
 CXX	=	g++
-CFLAGS	=	-pipe $(PKG_CFLAGS) $(AL_CFLAGS) $(SDL_CFLAGS) $(SMPEG_CFLAGS) -O2 -DOLD_OPENAL -DAUDIO_OPENAL -D_REENTRANT
-CXXFLAGS=	-pipe $(PKG_CFLAGS) $(AL_CFLAGS) $(SDL_CFLAGS) $(SMPEG_CFLAGS) -O2 -DOLD_OPENAL -DAUDIO_OPENAL -D_REENTRANT
+CFLAGS	=	-pipe $(PKG_CFLAGS) $(AL_CFLAGS) $(SDL_CFLAGS) $(SMPEG_CFLAGS) -Wall -W -g -O2 -DOLD_OPENAL -DAUDIO_OPENAL -D_REENTRANT
+CXXFLAGS=	-pipe $(PKG_CFLAGS) $(AL_CFLAGS) $(SDL_CFLAGS) $(SMPEG_CFLAGS) -Wall -W -g -O2 -DOLD_OPENAL -DAUDIO_OPENAL -D_REENTRANT
 INCPATH	=	-I../support/include -I../support/glpng/include -I/usr/X11R6/include
 LINK	=	g++
 LFLAGS	=	
