Index: chromium-0.9.12/src/HeroAircraft.cpp
===================================================================
--- chromium-0.9.12.orig/src/HeroAircraft.cpp	2007-04-06 22:27:23.000000000 +0200
+++ chromium-0.9.12/src/HeroAircraft.cpp	2007-04-06 22:28:08.000000000 +0200
@@ -663,12 +663,12 @@
 			}
 			game->explosions->addExplo(Explosions::PowerBurst, pwrUp->pos);
 			delUp = pwrUp;
-			pwrUp = powerUps->getNext();
+			pwrUp = pwrUp->next;
 			powerUps->remove(delUp);
 		}
 		else
 		{
-			pwrUp = powerUps->getNext();
+			pwrUp = pwrUp->next;
 		}
 	}
 }
