Index: chromium-0.9.12/src/MainSDL_Event.cpp
===================================================================
--- chromium-0.9.12.orig/src/MainSDL_Event.cpp	2007-04-06 22:28:46.000000000 +0200
+++ chromium-0.9.12/src/MainSDL_Event.cpp	2007-04-06 22:29:28.000000000 +0200
@@ -71,6 +71,8 @@
 		case SDL_JOYBUTTONUP:
 			joystickButtonUp(event);
 			break;
+		case SDL_QUIT:
+			return true;
 		default:
 			break;
 	}
