diff -Naur wordgrinder-0.3.3.orig/pmfile wordgrinder-0.3.3/pmfile
--- wordgrinder-0.3.3.orig/pmfile	2009-12-13 14:44:53.000000000 +0100
+++ wordgrinder-0.3.3/pmfile	2012-08-11 07:54:34.340087801 +0200
@@ -19,7 +19,7 @@
 -- Where do you want WordGrinder installed? By default, it goes into your
 -- home directory.
 
-PREFIX = HOME
+PREFIX = "/usr"
 -- PREFIX = "/usr/local"
 
 -- What build flags do you want to use? (Not including -g or -Os, which are
@@ -45,8 +45,8 @@
 }
 	
 CINCLUDES = {
-	"/usr/include/ncursesw",
-	"/usr/include/lua5.1",
+	"/usr/include/ncurses",
+	"/usr/include",
 	"src/c"
 }
 
@@ -215,7 +215,7 @@
 	CLIBRARIES = {
 		PARENT,
 		'ncursesw',
-		'lua5.1'
+		'lua'
 	}
 }
 
@@ -300,7 +300,7 @@
 
 wordgrinder_release_exe = group {
 	wordgrinder_curses_exe,
-	CBUILDFLAGS = {PARENT, '-Os'},
+	CBUILDFLAGS = {PARENT, @CFLAGS@},
 	LUABUILDFLAGS = {PARENT, '-s'},
 	CDEFINES = {PARENT, 'NDEBUG'},
 	CLINKFLAGS = {PARENT, '-s'},
