diff -ur bbdock-0.2.8.orig/src/Icon.cc bbdock-0.2.8/src/Icon.cc
--- bbdock-0.2.8.orig/src/Icon.cc	2006-09-19 14:46:21.000000000 +0100
+++ bbdock-0.2.8/src/Icon.cc	2009-08-09 06:19:50.000000000 +0100
@@ -26,6 +26,7 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
+#include <stdlib.h>
 #include <string>
 #include <vector>
 
diff -ur bbdock-0.2.8.orig/src/WindowManager.cc bbdock-0.2.8/src/WindowManager.cc
--- bbdock-0.2.8.orig/src/WindowManager.cc	2006-09-19 14:48:23.000000000 +0100
+++ bbdock-0.2.8/src/WindowManager.cc	2009-08-09 06:20:39.000000000 +0100
@@ -32,7 +32,7 @@
 #endif // _GNU_SOURCE
 
 #include <iostream>
-#include <string>
+#include <string.h>
 
 #include "WindowManager.hh"
 
