--- kadu-0.12.3/plugins/jabber_protocol/3rdparty/CMakeLists.txt.orig	2013-11-17 07:23:54.186097640 +0700
+++ kadu-0.12.3/plugins/jabber_protocol/3rdparty/CMakeLists.txt	2013-11-17 07:24:32.686563005 +0700
@@ -26,7 +26,7 @@
 					get_filename_component (_basename ${_current_MOC} NAME_WE)
 					set (_header ${_abs_FILE})
 					set (_moc ${_abs_PATH}/${_current_MOC})
-					QT4_CREATE_MOC_COMMAND (${_header} ${_moc} "${_moc_INCS}" "")
+					QT4_CREATE_MOC_COMMAND (${_header} ${_moc} "${_moc_INCS}" "" "${moc_target}")
 					MACRO_ADD_FILE_DEPENDENCIES (${_abs_FILE} ${_moc})
 				endforeach (_current_MOC_INC)
 			endif (_match)
