diff -Naur dockapps/gui.c dockapps.patched/gui.c
--- dockapps/gui.c	2017-02-25 03:58:36.637396483 -0500
+++ dockapps.patched/gui.c	2017-02-25 04:04:34.568392242 -0500
@@ -173,7 +173,6 @@
 		history_items = g_list_remove_link(history_items, list_node);
 		gtk_container_remove(GTK_CONTAINER(hist_item->menu),
 				hist_item->menu_item);
-		gtk_widget_destroy(hist_item->menu_item);
 		g_free(hist_item->content);
 		g_free(hist_item);
 		g_list_free_1(list_node);
diff -Naur dockapps/history.c dockapps.patched/history.c
--- dockapps/history.c	2014-11-23 09:59:59.000000000 -0500
+++ dockapps.patched/history.c	2017-02-25 04:12:31.276386594 -0500
@@ -362,7 +362,6 @@
 		hist_item = (HISTORY_ITEM *)list_node->data;
 		gtk_container_remove(GTK_CONTAINER(hist_item->menu),
 				hist_item->menu_item);
-		gtk_widget_destroy(hist_item->menu_item);
 		g_free(hist_item->content);
 		g_free(hist_item);
 		list_node = g_list_previous(list_node);
