diff -Naur texstudio2.3.orig/qcodeedit/lib/document/qdocumentcommand.cpp texstudio2.3/qcodeedit/lib/document/qdocumentcommand.cpp
--- texstudio2.3.orig/qcodeedit/lib/document/qdocumentcommand.cpp	2012-02-16 13:16:25.000000000 +0000
+++ texstudio2.3/qcodeedit/lib/document/qdocumentcommand.cpp	2012-07-10 01:48:50.123060180 +0000
@@ -22,6 +22,8 @@
 
 #include "qdocument_p.h"
 
+#include "stdint.h"
+
 /*!
 	\ingroup document
 	@{
diff -Naur texstudio2.3.orig/qtsingleapplication/qtlocalpeer.cpp texstudio2.3/qtsingleapplication/qtlocalpeer.cpp
--- texstudio2.3.orig/qtsingleapplication/qtlocalpeer.cpp	2012-02-16 13:16:25.000000000 +0000
+++ texstudio2.3/qtsingleapplication/qtlocalpeer.cpp	2012-07-10 01:40:24.169046420 +0000
@@ -48,9 +48,9 @@
 #include "qtlocalpeer.h"
 #include <QtCore/QCoreApplication>
 #include <QtCore/QTime>
-#if QT_VERSION >= 0x040600
+#if QT_VERSION >= 0x040600
 #include <QtCore/QProcessEnvironment>
-#endif
+#endif
 
 #if defined(Q_OS_WIN)
 #include <QtCore/QLibrary>
@@ -60,6 +60,7 @@
 #endif
 #if defined(Q_OS_UNIX)
 #include <time.h>
+#include "unistd.h"
 #endif
 
 namespace QtLP_Private {
