From c826330f3434fe684b830b271fa3f7a0bc57fee4 Mon Sep 17 00:00:00 2001
From: Andriy Grytsenko <andrej@rep.kiev.ua>
Date: Thu, 1 Nov 2012 01:05:22 +0200
Subject: [PATCH 01/11] Update for NEWS file: changed '1.0.2' into '1.1.0';
 added missing entries.

---
 NEWS | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/NEWS b/NEWS
index a3bdffa..c2ff56a 100644
--- a/NEWS
+++ b/NEWS
@@ -1,4 +1,4 @@
-Changes on 1.0.2 since 1.0.1:
+Changes on 1.1.0 since 1.0.1:
 
 * Corrected underscores (mnemonics) on every menu and dialog.
 
@@ -23,6 +23,11 @@ Changes on 1.0.2 since 1.0.1:
 * The items which cannot act (such as "Copy" when no items selected) now
     are disabled in menu and toolbar.
 
+* Experimental implementation of option --one-screen to apply --desktop
+    option only for one (current) X-Screen.
+
+* Implemented side pane hiding, bound to 'F9' key press.
+
 
 Changes on 1.0.1 since 1.0.0:
 
-- 
1.8.0.1

