--- yaze.sh	2015-04-26 15:15:29.000000000 +0100
+++ yaze.sh.new	2022-05-28 20:33:43.386759212 +0100
@@ -1,9 +1,9 @@
 #!/bin/sh
 
-S_YAZEFILES=/usr/local/lib/yaze
-S_CPMDSKS=/usr/local/lib/yaze/disks
-S_DOCFILES=/usr/local/lib/yaze/doc
-S_DOCFILES_html=/usr/local/lib/yaze/doc_html
+S_YAZEFILES=/usr/lib/yaze-ag
+S_CPMDSKS=/usr/share/yaze-ag/disks
+S_DOCFILES=/usr/doc/yaze-ag-2.51.3/doc
+S_DOCFILES_html=/usr/doc/yaze-ag-2.51.3/doc_html
 
 if [ ! -f .yazerc ]
 then
