diff -Naur a/os/linux/config.mk b/os/linux/config.mk
--- a/os/linux/config.mk	2010-12-17 20:44:47.000000000 +0900
+++ b/os/linux/config.mk	2011-01-27 04:52:05.753313553 +0900
@@ -1,18 +1,18 @@
 # Support ATE function
-HAS_ATE=n
+HAS_ATE=y
 
 # Support QA ATE function
-HAS_QA_SUPPORT=n
+HAS_QA_SUPPORT=y
 
 # Support XLINK mode
-HAS_XLINK=n
+HAS_XLINK=y
 
 
 # Support Wpa_Supplicant
-HAS_WPA_SUPPLICANT=n
+HAS_WPA_SUPPLICANT=y
 
 # Support Native WpaSupplicant for Network Maganger
-HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=n
+HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=y
 
 #Support Net interface block while Tx-Sw queue full
 HAS_BLOCK_NET_IF=n

