diff -Naur a/Makefile b/Makefile
--- a/Makefile	2010-12-17 20:44:47.000000000 +0900
+++ b/Makefile	2011-01-27 04:39:12.732361523 +0900
@@ -1,8 +1,7 @@
 RT28xx_MODE = STA
 
 TARGET = LINUX
-
-CHIPSET = 3562
+MODE = STA
 
 #OS ABL - YES or NO
 OSABL = NO
@@ -139,10 +138,10 @@
 
 ifeq ($(PLATFORM),PC)
 # Linux 2.6
-LINUX_SRC = /lib/modules/$(shell uname -r)/build
+LINUX_SRC = /lib/modules/$(KERNEL)/build
 # Linux 2.4 Change to your local setting
 #LINUX_SRC = /usr/src/linux-2.4
-LINUX_SRC_MODULE = /lib/modules/$(shell uname -r)/kernel/drivers/net/wireless/
+LINUX_SRC_MODULE = /lib/modules/$(KERNEL)/kernel/drivers/net/wireless
 CROSS_COMPILE = 
 endif
 
@@ -239,7 +238,7 @@
 CROSS_COMPILE = /opt/montavista/pro/devkit/arm/v5t_le_mvl5/bin/arm_v5t_le-
 endif
 
-export OSABL RT28xx_DIR RT28xx_MODE LINUX_SRC CROSS_COMPILE CROSS_COMPILE_INCLUDE PLATFORM RELEASE CHIPSET RTMP_SRC_DIR LINUX_SRC_MODULE TARGET
+export OSABL RT28xx_DIR RT28xx_MODE CROSS_COMPILE CROSS_COMPILE_INCLUDE PLATFORM RELEASE LINUX_SRC RTMP_SRC_DIR LINUX_SRC_MODULE TARGET
 
 # The targets that may be used.
 PHONY += all build_tools test UCOS THREADX LINUX release prerelease clean uninstall install libwapi osabl
