diff -Nur ink-0.4.1.orig/Makefile ink-0.4.1/Makefile
--- ink-0.4.1.orig/Makefile	2006-09-06 09:06:22.000000000 -0500
+++ ink-0.4.1/Makefile	2008-01-06 13:52:42.531413083 -0600
@@ -1,12 +1,13 @@
-CFLAGS = -Wall -O2 -L$(PWD)/../../../libinklevel/cvs/libinklevel \
+CFLAGS += -Wall -O2 -L$(PWD)/../../../libinklevel/cvs/libinklevel \
          -I$(PWD)/../../../libinklevel/cvs/libinklevel -linklevel
-DESTDIR = /usr/local/
+DESTDIR= 
+PREFIX = /usr/local/
 
 all: ink 
 
 install: all
-	mkdir -p $(DESTDIR)/bin
-	install -m 755 -o root -g root ink $(DESTDIR)/bin/ink
+	mkdir -p $(DESTDIR)/$(PREFIX)/bin
+	install -m 755 -o root -g root ink $(DESTDIR)/$(PREFIX)/bin/ink
 
 clean:
 	rm -fr ink *.o *~
