diff -Naur libxosd-xft-1.1.0/man/osd-echo.1 libxosd-xft-1.1.0.patched/man/osd-echo.1
--- libxosd-xft-1.1.0/man/osd-echo.1	2021-07-05 06:33:07.000000000 -0400
+++ libxosd-xft-1.1.0.patched/man/osd-echo.1	2026-10-05 18:47:55.490278846 -0400
@@ -55,6 +55,10 @@
 \f[R]
 .fi
 .TP
+-d \f[I]milliseconds\f[R], --delay-in-millis=\f[I]milliseconds\f[R]
+Display message for this long, before exiting. Default is
+1000, or one second.
+.TP
 -t \f[I]ALIGN\f[R], --text-align=\f[I]ALIGN\f[R]
 Text alignment within the OSD window
 .IP
diff -Naur libxosd-xft-1.1.0/src/osd-cat.c libxosd-xft-1.1.0.patched/src/osd-cat.c
--- libxosd-xft-1.1.0/src/osd-cat.c	2021-07-05 06:33:07.000000000 -0400
+++ libxosd-xft-1.1.0.patched/src/osd-cat.c	2026-10-05 18:46:43.554279121 -0400
@@ -253,6 +253,7 @@
               "                             Where:\n"
               "                                 valign: one of top,middle,bottom or none\n"
               "                                 halign: one of left,center,right or none\n"
+              "  -d, --delay-in-millis=<ms> Display message for this long before exiting.\n"
               "  -t, --text-align=<align>   Text alignment within the OSD window(default: %s)\n"
               "                             <align> Format: <halign/valign>\n"
               "                                 valign: one of top,middle,bottom or none\n"
diff -Naur libxosd-xft-1.1.0/src/osd-echo.c libxosd-xft-1.1.0.patched/src/osd-echo.c
--- libxosd-xft-1.1.0/src/osd-echo.c	2021-07-05 06:33:07.000000000 -0400
+++ libxosd-xft-1.1.0.patched/src/osd-echo.c	2026-10-05 18:47:02.962279046 -0400
@@ -279,6 +279,7 @@
               "                              Where:\n"
               "                                 valign: one of top,middle,bottom or none\n"
               "                                 halign: one of left,center,right or none\n"
+              "  -d, --delay-in-millis=<ms> Display message for this long before exiting.\n"
               "  -t, --text-align=<align>    Text alignment within the OSD window(default: %s)\n"
               "                              <align> Format: <halign/valign>\n"
               "                                 valign: one of top,middle,bottom or none\n"
