--- Makefile.in	2013-10-08 07:33:03.000000000 -0500
+++ Makefile.in.fixed	2014-07-13 15:33:30.471310610 -0500
@@ -820,13 +820,6 @@
 	  done \
 	  fi
 
-	if test -d $(HOME)/$(PACKAGE); then \
-	  echo Working directory exists: $(HOME)/$(PACKAGE) - not copying; \
-	  else \
-	  echo Copying $(PACKAGE) working directory to $(HOME)/$(PACKAGE); \
-	  cp -r $(srcdir)/$(PACKAGE) $(HOME); \
-	  fi
-
 dist-hook:
 	if test -d pixmaps; then \
 	  mkdir $(distdir)/pixmaps; \
