--- direwolf-1.1/Makefile.linux	2014-11-21 21:36:04.000000000 -0600
+++ Makefile.linux	2015-05-14 19:30:26.500837838 -0500
@@ -4,7 +4,26 @@
 
 all : direwolf decode_aprs text2tt tt2text ll2utm utm2ll aclients log2gpx
 
-CC = gcc
+CC ?= gcc
+INSTALL ?= sudo install
+INSTALL_PROGRAM = $(INSTALL) -m 755
+INSTALL_DATA = $(INSTALL) -m 644
+INSTALL_DIR = $(INSTALL) -d
+
+prefix ?= /usr/local
+exec_prefix = $(prefix)
+bindir ?= $(exec_prefix)/bin
+docdir ?= $(prefix)/share/doc/direwolf
+datadir ?= $(prefix)/share/direwolf
+appdir ?= /usr/share/applications
+sysconfdir ?= ~
+LINK_IN_HOME ?= yes
+ 
+ifeq ($(LINK_IN_HOME),yes)
+       HOME_DESKTOP = ln -f -s /usr/share/applications/direwolf.desktop ~/Desktop/direwolf.desktop
+else
+       HOME_DESKTOP = 
+endif
 
 #
 # The DSP filters can be sped up considerably with the SSE 
@@ -78,15 +97,17 @@
 # cause compatibility issues for those with older computers.
 #
 
-arch := $(shell echo | gcc -E -dM - | grep __i386__)
+arch ?= $(shell echo | gcc -E -dM - | grep __i386__)
 
-ifneq ($(arch),)
+ifndef CFLAGS
+ ifneq ($(arch),)
 # You might see improvement with -march fine tuned to your hardware.
 # Probably should keep pentium3 if you will be redistributing binaries
 # to other people.
-CFLAGS := -O3 -march=pentium3 -pthread -Iutm
-else
-CFLAGS := -O3 -pthread -Iutm
+        CFLAGS := -O3 -march=pentium3 -pthread -Iutm
+ else
+        CFLAGS := -O3 -pthread -Iutm
+ endif
 endif
 
 
@@ -150,27 +171,31 @@
 # TODO: Review file locations.
 
 install : direwolf decode_aprs tocalls.txt symbols-new.txt symbolsX.txt dw-icon.png direwolf.desktop
-	install direwolf /usr/local/bin
-	install decode_aprs /usr/local/bin
-	install text2tt /usr/local/bin
-	install tt2text /usr/local/bin
-	install ll2utm /usr/local/bin
-	install utm2ll /usr/local/bin
-	install aclients /usr/local/bin
-	install log2gpx /usr/local/bin
-	install -D --mode=644 tocalls.txt /usr/share/direwolf/tocalls.txt
-	install -D --mode=644 symbols-new.txt /usr/share/direwolf/symbols-new.txt
-	install -D --mode=644 symbolsX.txt /usr/share/direwolf/symbolsX.txt
-	install -D --mode=644 dw-icon.png /usr/share/direwolf/dw-icon.png
-	install -D --mode=644 direwolf.desktop /usr/share/applications/direwolf.desktop
-	install -D --mode=644 CHANGES.txt /usr/local/share/doc/direwolf/CHANGES.txt
-	install -D --mode=644 LICENSE-dire-wolf.txt /usr/local/share/doc/direwolf/LICENSE-dire-wolf.txt
-	install -D --mode=644 LICENSE-other.txt /usr/local/share/doc/direwolf/LICENSE-other.txt
-	install -D --mode=644 User-Guide.pdf /usr/local/share/doc/direwolf/User-Guide.pdf
-	install -D --mode=644 Raspberry-Pi-APRS.pdf /usr/local/share/doc/direwolf/Raspberry-Pi-APRS.pdf
-	install -D --mode=644 Raspberry-Pi-APRS-Tracker.pdf /usr/local/share/doc/direwolf/Raspberry-Pi-APRS-Tracker.pdf
-	install -D --mode=644 APRStt-Implementation-Notes.pdf /usr/local/share/doc/direwolf/APRStt-Implementation-Notes.pdf
-	install -D --mode=644 Quick-Start-Guide-Windows.pdf /usr/local/share/doc/direwolf/Quick-Start-Guide-Windows.pdf
+	$(INSTALL_DIR) $(DESTDIR)$(bindir)/
+	$(INSTALL_PROGRAM) direwolf $(DESTDIR)$(bindir)/
+	$(INSTALL_PROGRAM) decode_aprs $(DESTDIR)$(bindir)/
+	$(INSTALL_PROGRAM) text2tt $(DESTDIR)$(bindir)/
+	$(INSTALL_PROGRAM) tt2text $(DESTDIR)$(bindir)/
+	$(INSTALL_PROGRAM) ll2utm $(DESTDIR)$(bindir)/
+	$(INSTALL_PROGRAM) utm2ll $(DESTDIR)$(bindir)/
+	$(INSTALL_PROGRAM) aclients $(DESTDIR)$(bindir)/
+	$(INSTALL_PROGRAM) log2gpx $(DESTDIR)$(bindir)/
+	$(INSTALL_DIR) $(DESTDIR)$(datadir)
+	$(INSTALL_DATA) tocalls.txt $(DESTDIR)$(datadir)/
+	$(INSTALL_DATA) symbols-new.txt $(DESTDIR)$(datadir)/
+	$(INSTALL_DATA) symbolsX.txt $(DESTDIR)$(datadir)/
+	$(INSTALL_DATA) dw-icon.png $(DESTDIR)$(datadir)/
+	$(INSTALL_DIR) $(DESTDIR)$(appdir)
+	$(INSTALL_DATA) direwolf.desktop $(DESTDIR)$(appdir)/
+	$(INSTALL_DIR) $(DESTDIR)$(docdir)
+	$(INSTALL_DATA) CHANGES.txt $(DESTDIR)$(docdir)/
+	$(INSTALL_DATA) LICENSE-dire-wolf.txt $(DESTDIR)$(docdir)/
+	$(INSTALL_DATA) LICENSE-other.txt $(DESTDIR)$(docdir)/
+	$(INSTALL_DATA) User-Guide.pdf $(DESTDIR)$(docdir)/
+	$(INSTALL_DATA) Raspberry-Pi-APRS.pdf $(DESTDIR)$(docdir)/
+	$(INSTALL_DATA) Raspberry-Pi-APRS-Tracker.pdf $(DESTDIR)$(docdir)/
+	$(INSTALL_DATA) APRStt-Implementation-Notes.pdf $(DESTDIR)$(docdir)/
+	$(INSTALL_DATA) Quick-Start-Guide-Windows.pdf $(DESTDIR)$(docdir)/
 
 
 # The Raspberry Pi has ~/Desktop but Ubuntu does not.
@@ -183,7 +208,8 @@
 	ln -f -s /usr/share/applications/direwolf.desktop ~/Desktop/direwolf.desktop
 
 install-conf : direwolf.conf
-	cp direwolf.conf ~
+	$(INSTALL_DIR) $(DESTDIR)$(sysconfdir)
+	$(INSTALL_DATA) direwolf.conf $(DESTDIR)$(sysconfdir)/
 
 
 # Separate application to decode raw data.
