--- moodbar-0.1.2-old/analyzer/main.c	2007-06-21 18:43:57.000000000 -0400
+++ moodbar-0.1.2/analyzer/main.c	2007-06-21 18:40:46.000000000 -0400
@@ -262,6 +262,9 @@
   GOptionContext *ctx;
   GError *err = NULL;
 
+  /* Must be called before other glib stuff is called */
+  if (!g_thread_supported ()) g_thread_init(NULL);
+
   ctx = g_option_context_new ("[INFILE] - Run moodbar analyzer");
   g_option_context_add_group (ctx, gst_init_get_option_group ());
   g_option_context_add_main_entries (ctx, entries, NULL);
