--- fcitx-autostart.orig	2017-03-26 17:58:45.586105103 +0200
+++ fcitx-autostart	2017-03-26 17:59:09.693105345 +0200
@@ -1,4 +1,26 @@
-#!/bin/sh
+#!/bin/bash
+
+START="no"
+
+if [ "$XIM" = "fcitx" ]; then
+  START="yes"
+fi
+
+if [ "$XIM_PROGRAM" = "fcitx" ]; then
+  START="yes"
+fi
+
+if [ "$GTK_IM_MODULE" = "fcitx" ]; then
+  START="yes"
+fi
+
+if [ "$QT_IM_MOFULE" = "fcitx" ]; then
+  START="yes"
+fi
+
+if [ "$START" = "no" ]; then
+  exit 0
+fi
 
 # sleep for a little while to avoid duplicate startup
 sleep 2
