diff -uN id3v2-0.1.11/id3v2.cpp id3v2-0.1.11-patched/id3v2.cpp
--- id3v2-0.1.11/id3v2.cpp	2004-05-04 19:30:15.000000000 +0100
+++ id3v2-0.1.11-patched/id3v2.cpp	2008-02-16 19:11:27.000000000 +0000
@@ -423,7 +423,7 @@
         {
           // check if there is a total track number and if we only have 
           // the track number for this file.  In this case combine them.
-          char *currentTrackNum, *newTrackNum;
+          char *currentTrackNum, *newTrackNum = NULL;
 
           if (pFrame != NULL) 
           {
