--- a/kernel/nvidia/os-interface.c	2026-06-23 01:57:59.000000000 -0700
+++ b/kernel/nvidia/os-interface.c	2026-08-21 11:53:54.492305209 -0700
@@ -729,8 +729,7 @@
 void NV_API_CALL os_get_current_process_name(char *buf, NvU32 len)
 {
     task_lock(current);
-    strncpy(buf, current->comm, len - 1);
-    buf[len - 1] = '\0';
+    strscpy(buf, current->comm, len - 1);
     task_unlock(current);
 }
 
--- a/kernel/nvidia/linux_nvswitch.c	2026-06-23 01:57:20.000000000 -0700
+++ b/kernel/nvidia/linux_nvswitch.c	2026-08-21 11:54:19.265222719 -0700
@@ -2037,8 +2037,7 @@
         return -NVL_ERR_GENERIC;
     }
 
-    strncpy(regkey_val, regkey_val_start, regkey_val_len);
-    regkey_val[regkey_val_len] = '\0';
+    strscpy(regkey_val, regkey_val_start, regkey_val_len);
 
     if (nvswitch_os_strtouint(regkey_val, data) != 0)
     {
@@ -2392,7 +2391,8 @@
     NvLength length
 )
 {
-    return strncpy(dest, src, length);
+    strscpy(dest, src, length);
+    return dest;
 }
 
 int
--- a/kernel/nvidia-uvm/uvm_pmm_gpu.c	2026-06-23 01:57:44.000000000 -0700
+++ b/kernel/nvidia-uvm/uvm_pmm_gpu.c	2026-08-21 13:09:08.341270780 -0700
@@ -2799,7 +2799,7 @@
             size_t size;
             size_t align;
             if (level == 0) {
-                strncpy(chunk_split_cache[level].name, "uvm_gpu_chunk_t", sizeof(chunk_split_cache[level].name) - 1);
+                strscpy(chunk_split_cache[level].name, "uvm_gpu_chunk_t", sizeof(chunk_split_cache[level].name) - 1);
                 size = sizeof(uvm_gpu_chunk_t);
                 align = __alignof__(uvm_gpu_chunk_t);
             } else {
--- a/kernel/nvidia-modeset/nvidia-modeset-linux.c	2026-06-23 01:00:11.000000000 -0700
+++ b/kernel/nvidia-modeset/nvidia-modeset-linux.c	2026-08-21 13:09:31.891192362 -0700
@@ -643,7 +643,8 @@
 
 char* nvkms_strncpy(char *dest, const char *src, size_t n)
 {
-    return strncpy(dest, src, n);
+    strscpy(dest, src, n);
+    return dest;
 }
 
 void nvkms_usleep(NvU64 usec)
