--- Makefile	2012-11-28 11:23:15.000000000 -0700
+++ Makefile.new	2013-02-08 10:21:31.839628119 -0700
@@ -61,15 +61,21 @@
 	install -m 755 -d $(DESTDIR)$(LIBDIR)
 	install -m 755 -d $(DESTDIR)$(MANDIR)/man1
 	install -m 755 -d $(DESTDIR)$(RESDIR)
+	install -m 755 -d $(DESTDIR)$(PREFIX)/share/applications
+	install -m 755 -d $(DESTDIR)$(PREFIX)/share/icons/hicolor/apps/16x16
+	install -m 755 -d $(DESTDIR)$(PREFIX)/share/icons/hicolor/apps/32x32
+	install -m 755 -d $(DESTDIR)$(PREFIX)/share/icons/hicolor/apps/48x48
+	install -m 755 -d $(DESTDIR)$(PREFIX)/share/icons/hicolor/apps/64x64
+	install -m 755 -d $(DESTDIR)$(PREFIX)/share/icons/hicolor/apps/128x128
 	install -m 755 xombrero $(DESTDIR)$(BINDIR)
 	install -m 644 ../xombrero.1 $(DESTDIR)$(MANDIR)/man1/xombrero.1
 	install -m 644 ../xombrero.css $(DESTDIR)$(RESDIR)
 	install -m 644 ../xombrero.desktop $(DESTDIR)$(PREFIX)/share/applications
-	install -m 644 ../xombreroicon16.png $(DESTDIR)$(RESDIR)
-	install -m 644 ../xombreroicon32.png $(DESTDIR)$(RESDIR)
-	install -m 644 ../xombreroicon48.png $(DESTDIR)$(RESDIR)
-	install -m 644 ../xombreroicon64.png $(DESTDIR)$(RESDIR)
-	install -m 644 ../xombreroicon128.png $(DESTDIR)$(RESDIR)
+	install -m 644 ../xombreroicon16.png $(DESTDIR)$(PREFIX)/share/icons/hicolor/apps/16x16/xombrero.png
+	install -m 644 ../xombreroicon32.png $(DESTDIR)$(PREFIX)/share/icons/hicolor/apps/32x32/xombrero.png
+	install -m 644 ../xombreroicon48.png $(DESTDIR)$(PREFIX)/share/icons/hicolor/apps/48x48/xombrero.png
+	install -m 644 ../xombreroicon64.png $(DESTDIR)$(PREFIX)/share/icons/hicolor/apps/64x64/xombrero.png
+	install -m 644 ../xombreroicon128.png $(DESTDIR)$(PREFIX)/share/icons/hicolor/apps/128x128/xombrero.png
 	install -m 644 ../tld-rules $(DESTDIR)$(RESDIR)
 	install -m 644 ../style.css $(DESTDIR)$(RESDIR)
 	install -m 644 ../hsts-preload $(DESTDIR)$(RESDIR)
