--- Makefile	2013-07-11 11:05:15.000000000 -0600
+++ Makefile.new	2013-09-04 22:10:27.708971431 -0600
@@ -25,6 +25,7 @@
 APPDIR?= $(PREFIX)/share/applications
 MANDIR?= $(PREFIX)/share/man
 RESDIR?= $(PREFIX)/share/xombrero
+ICODIR?= $(PREFIX)/share/icons/hicolor/apps
 
 SRCS= $(shell ls ../*.c)
 SRCS+= linux.c
@@ -63,15 +64,20 @@
 	install -m 755 -d $(DESTDIR)$(MANDIR)/man1
 	install -m 755 -d $(DESTDIR)$(RESDIR)
 	install -m 755 -d $(DESTDIR)$(APPDIR)
+	install -m 755 -d $(DESTDIR)$(ICODIR)/16x16
+	install -m 755 -d $(DESTDIR)$(ICODIR)/32x32
+	install -m 755 -d $(DESTDIR)$(ICODIR)/48x48
+	install -m 755 -d $(DESTDIR)$(ICODIR)/64x64
+	install -m 755 -d $(DESTDIR)$(ICODIR)/128x128
 	install -m 755 xombrero $(DESTDIR)$(BINDIR)
 	install -m 644 ../xombrero.1 $(DESTDIR)$(MANDIR)/man1/xombrero.1
 	install -m 644 ../xombrero.css $(DESTDIR)$(RESDIR)
 	install -m 644 ../xombrero.desktop $(DESTDIR)$(APPDIR)
-	install -m 644 ../xombreroicon16.png $(DESTDIR)$(RESDIR)
-	install -m 644 ../xombreroicon32.png $(DESTDIR)$(RESDIR)
-	install -m 644 ../xombreroicon48.png $(DESTDIR)$(RESDIR)
-	install -m 644 ../xombreroicon64.png $(DESTDIR)$(RESDIR)
-	install -m 644 ../xombreroicon128.png $(DESTDIR)$(RESDIR)
+	install -m 644 ../xombreroicon16.png $(DESTDIR)$(ICODIR)/16x16
+	install -m 644 ../xombreroicon32.png $(DESTDIR)$(ICODIR)/32x32
+	install -m 644 ../xombreroicon48.png $(DESTDIR)$(ICODIR)/48x48
+	install -m 644 ../xombreroicon64.png $(DESTDIR)$(ICODIR)/64x64
+	install -m 644 ../xombreroicon128.png $(DESTDIR)$(ICODIR)/128x128
 	install -m 644 ../tld-rules $(DESTDIR)$(RESDIR)
 	install -m 644 ../style.css $(DESTDIR)$(RESDIR)
 	install -m 644 ../hsts-preload $(DESTDIR)$(RESDIR)
