--- xombrero.h	2012-11-28 11:23:15.000000000 -0700
+++ xombrero.h.new	2013-02-08 12:49:32.000610834 -0700
@@ -689,7 +689,7 @@
 #define XT_DS_ALLOW_INSECURE_CONTENT	(TRUE)
 #define XT_DS_ALLOW_INSECURE_SCRIPTS	(TRUE)
 #define XT_DS_WARN_CERT_CHANGES	(0)
-#define XT_DS_RESOURCE_DIR	("/usr/local/share/xombrero")
+#define XT_DS_RESOURCE_DIR	("/usr/share/xombrero")
 #define XT_DS_DO_NOT_TRACK	(0)
 #define XT_DS_PRELOAD_STRICT_TRANSPORT	(1)
 #define XT_DS_GNUTLS_PRIORITY_STRING	(NULL)
