--- /etc/inittab	2011-11-22 00:05:33.883260270 +0100
+++ /etc/inittab-heimdal	2011-11-21 23:45:38.922009963 +0100
@@ -49,12 +49,12 @@
 pg::powerokwait:/sbin/genpowerfail stop
 
 # These are the standard console login getties in multiuser mode:
-c1:12345:respawn:/sbin/agetty 38400 tty1 linux
-c2:12345:respawn:/sbin/agetty 38400 tty2 linux
-c3:12345:respawn:/sbin/agetty 38400 tty3 linux
-c4:12345:respawn:/sbin/agetty 38400 tty4 linux
-c5:12345:respawn:/sbin/agetty 38400 tty5 linux
-c6:12345:respawn:/sbin/agetty 38400 tty6 linux
+c1:12345:respawn:/sbin/agetty -l /usr/heimdal/bin/login 38400 tty1 linux
+c2:12345:respawn:/sbin/agetty -l /usr/heimdal/bin/login 38400 tty2 linux
+c3:12345:respawn:/sbin/agetty -l /usr/heimdal/bin/login 38400 tty3 linux
+c4:12345:respawn:/sbin/agetty -l /usr/heimdal/bin/login 38400 tty4 linux
+c5:12345:respawn:/sbin/agetty -l /usr/heimdal/bin/login 38400 tty5 linux
+c6:12345:respawn:/sbin/agetty -l /usr/heimdal/bin/login 38400 tty6 linux
 
 # Local serial lines:
 #s1:12345:respawn:/sbin/agetty -L ttyS0 9600 vt100
