diff -Nur hydra-5.4-src.orig/Makefile.am hydra-5.4-src/Makefile.am
--- hydra-5.4-src.orig/Makefile.am	2006-05-05 09:10:44.000000000 -0500
+++ hydra-5.4-src/Makefile.am	2009-07-05 11:23:02.983188212 -0500
@@ -2,7 +2,8 @@
 # Makefile for Hydra - (c) 2001-2006 by van Hauser / THC <vh@thc.org>
 #
 
-OPTS=-I. -Wall -O2
+EXTRA_CFLAGS=
+OPTS=-I. -Wall $(EXTRA_CFLAGS)
 # -ggdb -pedantic
 LIBS=-lm
 DIR=/bin
