--- projectlibre-1.9.0/projectlibre.sh.orig	2019-03-20 20:27:04.056920691 +0700
+++ projectlibre-1.9.0/projectlibre.sh	2019-03-20 20:27:29.407188487 +0700
@@ -70,7 +70,7 @@
 JAVA_EXE="java"
 JAVA_OPTS="-Xms128m -Xmx768m"
 #PROJECTLIBRE_HOME0="/usr/share/projectlibre"
-PROJECTLIBRE_HOME0=`(dirname $0)`
+PROJECTLIBRE_HOME0="/opt/ProjectLibre"
 PROJECTLIBRE_HOME="$PROJECTLIBRE_HOME0"
 LOG_LEVEL="DEBUG"
 LOG_FILE="/dev/null"
