--- projectlibre-1.5_beta4/projectlibre.sh.orig	2012-09-16 15:10:24.135176843 +0700
+++ projectlibre-1.5_beta4/projectlibre.sh	2012-09-16 15:11:14.134177925 +0700
@@ -52,7 +52,7 @@
 CONF_DIR="$HOME/.projectlibre"
 RUN_CONF="$CONF_DIR/run.conf"
 VERSION_FILE="$CONF_DIR/java-version"
-JAVA_INSTALL_DIR="/usr/java"
+JAVA_INSTALL_DIR="/usr/lib"
 JAVA_INSTALL_DIR_EXE="bin/java"
 JAVA_INSTALL_DIR2="/usr/lib/jvm"
 JAVA_INSTALL_DIR_EXE2="bin/java"
@@ -64,7 +64,7 @@
 JAVA_EXE="java"
 JAVA_OPTS="-Xms128m -Xmx768m"
 #OPENPROJ_HOME0="/usr/share/projectlibre"
-OPENPROJ_HOME0=`(dirname $0)`
+OPENPROJ_HOME0="/opt/ProjectLibre"
 OPENPROJ_HOME="$OPENPROJ_HOME0"
 LOG_LEVEL="DEBUG"
 LOG_FILE="/dev/null"
