diff -rup lcdtest-1.18.orig/src/lcdtest.c lcdtest-1.18/src/lcdtest.c
--- lcdtest-1.18.orig/src/lcdtest.c	2010-01-26 09:26:30.000000000 +0300
+++ lcdtest-1.18/src/lcdtest.c	2011-02-01 16:31:37.000000000 +0300
@@ -742,7 +742,8 @@ int enumerate_sdl_modes (int *width, int
 }
 
 
-const char font_path [] = "/usr/share/fonts/liberation/LiberationMono-Regular.ttf";
+//const char font_path [] = "/usr/share/fonts/liberation/LiberationMono-Regular.ttf";
+const char font_path [] = "/usr/share/fonts/TTF/LiberationMono-Regular.ttf";
 
 int main (int argc, char *argv [])
 {
