--- src/lib/corelib/tools/tools.pri.orig	2018-03-27 13:40:50.000000000 +0200
+++ src/lib/corelib/tools/tools.pri	2018-04-21 22:01:02.214730669 +0200
@@ -126,6 +126,7 @@
         $$PWD/installoptions.h \
         $$PWD/qbspluginmanager.h \
         $$PWD/setupprojectparameters.h \
+        $$PWD/projectgeneratormanager.h \
         $$PWD/toolchains.h \
         $$PWD/version.h
     tools_headers.path = $${QBS_INSTALL_PREFIX}/include/qbs/tools
