diff --git a/kmodsrc/ksplice.c b/kmodsrc/ksplice.c
index 6f6805c..63b0398 100644
--- a/kmodsrc/ksplice.c
+++ b/kmodsrc/ksplice.c
@@ -4098,6 +4098,8 @@ static const struct kernel_symbol *find_symbol(const char *name,
 	return NULL;
 }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,29)
+/* a06f6211ef9b1785922f9d0e8766d63ac4e66de1 was after 2.6.28 */
 static inline int within_module_core(unsigned long addr, struct module *mod)
 {
         return (unsigned long)mod->module_core <= addr &&
@@ -4109,6 +4111,7 @@ static inline int within_module_init(unsigned long addr, struct module *mod)
         return (unsigned long)mod->module_init <= addr &&
                addr < (unsigned long)mod->module_init + mod->init_size;
 }
+#endif
 
 static struct module *__module_address(unsigned long addr)
 {
