--- conf.h	2020-09-12 15:40:06.962795268 -0300
+++ b/conf.h	2020-09-12 15:54:57.712812014 -0300
@@ -14,7 +14,7 @@
 	VERBOSE          = false,  /* write dump of input to stdout, debug message to stderr */
 	TABSTOP          = 8,      /* hardware tabstop */
 	LAZY_DRAW        = true,   /* don't draw when input data size is larger than BUFSIZE */
-	BACKGROUND_DRAW  = false,  /* always draw even if vt is not active */
+	BACKGROUND_DRAW  = true,  /* always draw even if vt is not active */
 	VT_CONTROL       = true,   /* handle vt switching */
 	FORCE_TEXT_MODE  = false,  /* force KD_TEXT mode (not use KD_GRAPHICS mode) */
 	SUBSTITUTE_HALF  = 0x0020, /* used for missing glyph(single width): U+0020 (SPACE) */
