--- notepadqq.desktop	2022-06-10 15:08:20.381029816 +0100
+++ notepadqq.desktop.new	2022-06-19 15:08:52.798842702 +0100
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=Notepadqq
 Name[de_DE]=Notepadqq
 Name[fr]=Notepadqq
@@ -24,7 +23,7 @@
 StartupNotify=false
 MimeType=text/plain;text/html;text/x-php;text/x-c;text/x-shellscript;
 Icon=notepadqq
-Categories=Development;Utility;TextEditor;
+Categories=Qt;KDE;Utility;TextEditor;
 Actions=Window;Document;
 
 [Desktop Action Window]
@@ -77,7 +76,6 @@
 Name[zh_HK]=開啟新的視窗
 Name[zh_TW]=開啟新的視窗
 Exec=notepadqq --new-window
-OnlyShowIn=Unity;
 
 [Desktop Action Document]
 Name=Open a New Document
@@ -129,4 +127,3 @@
 Name[zh_HK]=開啟新的文件
 Name[zh_TW]=開啟新的文件
 Exec=notepadqq
-OnlyShowIn=Unity;
