--- a/plugins/lensfun/lensfun.c
+++ b/plugins/lensfun/lensfun.c
@@ -668,7 +668,7 @@ get_image(RSFilter *filter, const RSFilterRequest *request)
 			}
 
 			/* Start threads to apply phase 2, Vignetting and CA Correction */
-			if (effective_flags & (LF_MODIFY_VIGNETTING | LF_MODIFY_CCI)) 
+			if (effective_flags & LF_MODIFY_VIGNETTING)
 			{
 				/* Phase 2 is corrected inplace, so copy input first */
 				guint y_offset, y_per_thread, threaded_h;
