diff -Naur ocaml-4.12.0.orig/configure ocaml-4.12.0/configure
--- ocaml-4.12.0.orig/configure	2021-02-24 10:49:29.000000000 -0300
+++ ocaml-4.12.0/configure	2021-06-07 00:30:10.161272617 -0300
@@ -12706,7 +12706,7 @@
 -fno-builtin-memcmp";
       internal_cflags="$cc_warnings -fexcess-precision=standard" ;; #(
   gcc-*) :
-    common_cflags="-O2 -fno-strict-aliasing -fwrapv";
+    common_cflags="@CFLAGS@ -fno-strict-aliasing -fwrapv";
       internal_cflags="$cc_warnings -fno-common \
 -fexcess-precision=standard" ;; #(
   msvc-*) :
