diff -Naur ocaml-4.08.1.orig/configure ocaml-4.08.1/configure
--- ocaml-4.08.1.orig/configure	2019-08-05 19:32:44.000000000 +0200
+++ ocaml-4.08.1/configure	2019-09-06 10:35:25.454923040 +0200
@@ -12424,7 +12424,7 @@
 -fno-builtin-memcmp";
       internal_cflags="$gcc_warnings" ;; #(
   gcc-*) :
-    common_cflags="-O2 -fno-strict-aliasing -fwrapv";
+    common_cflags="@CFLAGS@ -fno-strict-aliasing -fwrapv";
       internal_cflags="$gcc_warnings" ;; #(
   msvc-*) :
     common_cflags="-nologo -O2 -Gy- -MD"
