diff -Naur ocaml-4.13.1.orig/configure ocaml-4.13.1/configure
--- ocaml-4.13.1.orig/configure	2021-09-30 17:40:11.000000000 +0200
+++ ocaml-4.13.1/configure	2022-02-15 16:03:05.417656000 +0100
@@ -12758,7 +12758,7 @@
     internal_cflags="-Wno-unused $cc_warnings \
 -fexcess-precision=standard"
         # TODO: see whether the code can be fixed to avoid -Wno-unused
-        common_cflags="-O2 -fno-strict-aliasing -fwrapv -mms-bitfields"
+        common_cflags="@CFLAGS@ -fno-strict-aliasing -fwrapv -mms-bitfields"
         internal_cppflags='-D__USE_MINGW_ANSI_STDIO=0 -DUNICODE -D_UNICODE'
         internal_cppflags="$internal_cppflags -DWINDOWS_UNICODE="
         internal_cppflags="${internal_cppflags}\$(WINDOWS_UNICODE)" ;; #(
