diff -u3 libcutefish-0.7/screen/outputmodel.cpp libcutefish-0.7-patched/screen/outputmodel.cpp
--- libcutefish-0.7/screen/outputmodel.cpp	2021-12-27 14:50:55.000000000 -0300
+++ libcutefish-0.7-patched/screen/outputmodel.cpp	2022-09-17 02:50:59.687552439 -0300
@@ -603,7 +603,7 @@
             return false;
         }
         m_config->setReplicationSource(output.ptr, nullptr);
-        output.ptr->setLogicalSize(QSizeF());
+        output.ptr->setExplicitLogicalSize(QSizeF());
         resetPosition(output);
     } else {
         const auto source = m_outputs[sourceIndex].ptr;
@@ -614,7 +614,7 @@
         m_config->setReplicationSource(output.ptr, source);
         output.posReset = output.ptr->pos();
         output.ptr->setPos(source->pos());
-        output.ptr->setLogicalSize(source->logicalSize());
+        output.ptr->setExplicitLogicalSize(source->explicitLogicalSize());
     }
 
     reposition();
