--- makefile	2023-11-17 22:37:21.787892000 +0000
+++ makefile.new	2023-11-17 23:05:02.368574413 +0000
@@ -23,8 +23,8 @@
 bbccon.o: bbccon.c bbccon.h
 	$(CXX) -Wno-array-bounds -Wno-unused-result -c -Os $< -o $@
 
-bbdata.o: ../../../BBCSDL/src/bbdata_x86_64.nas
-	nasm -f elf64 -s ../../../BBCSDL/src/bbdata_x86_64.nas -o bbdata.o
+bbdata.o: ../../src/bbdata_x86_32.nas
+	nasm -f elf32 -s ../../src/bbdata_x86_32.nas -o bbdata.o
 
 bbcbasic: $(OBJ)
 	$(CXX) $(OBJ) -L . -L/usr/include -ldl -lm -lrt -pthread \
