diff -Nur evince-2.30.1.orig//shell/ev-window.c evince-2.30.1/shell/ev-window.c
--- evince-2.30.1.orig//shell/ev-window.c	2010-04-26 09:58:15.000000000 -0500
+++ evince-2.30.1/shell/ev-window.c	2010-06-03 16:54:29.974562518 -0500
@@ -6107,7 +6107,7 @@
 	GFile *file;
 	gchar *path;
 
-	path = g_build_filename (ev_application_get_dot_dir (EV_APP, FALSE), "last_settings", NULL);
+	path = g_build_filename (ev_application_get_dot_dir (EV_APP, TRUE), "last_settings", NULL);
 	file = g_file_new_for_path (path);
 	g_free (path);
 
