diff -Naur gzdoom-g4.5.0/src/common/scripting/core/dynarrays.cpp gzdoom-g4.5.0.patched/src/common/scripting/core/dynarrays.cpp
--- gzdoom-g4.5.0/src/common/scripting/core/dynarrays.cpp	2020-10-31 05:07:33.000000000 -0400
+++ gzdoom-g4.5.0.patched/src/common/scripting/core/dynarrays.cpp	2020-11-08 14:45:48.654222783 -0500
@@ -89,7 +89,7 @@
 {
 	int oldSize = self->Size();
 	self->Insert(index, static_cast<typename T::value_type>(val));
-	if constexpr (fill) 
+	if (fill) 
 	{
 		for (unsigned i = oldSize; i < self->Size() - 1; i++) (*self)[i] = 0;
 	}
