diff -Naur gzdoom-g3.1.0/src/gitinfo.cpp gzdoom-g3.1.0.patched/src/gitinfo.cpp
--- gzdoom-g3.1.0/src/gitinfo.cpp	2017-05-31 06:11:33.000000000 -0400
+++ gzdoom-g3.1.0.patched/src/gitinfo.cpp	2017-06-10 16:21:51.138003050 -0400
@@ -39,27 +39,20 @@
 
 const char *GetGitDescription()
 {
-	return GIT_DESCRIPTION;
+	return VERSIONSTR;
 }
 
 const char *GetGitHash()
 {
-	return GIT_HASH;
+	return "release";
 }
 
 const char *GetGitTime()
 {
-	return GIT_TIME;
+	return __DATE__;
 }
 
 const char *GetVersionString()
 {
-	if (GetGitDescription()[0] == '\0')
-	{
-		return VERSIONSTR;
-	}
-	else
-	{
-		return GIT_DESCRIPTION;
-	}
+	return VERSIONSTR;
 }
