diff -Naur re-typeset-1.2.0/re-typeset-src/mainwindow.cpp re-typeset-1.2.0.patched/re-typeset-src/mainwindow.cpp
--- re-typeset-1.2.0/re-typeset-src/mainwindow.cpp	2016-10-13 15:51:19.000000000 -0400
+++ re-typeset-1.2.0.patched/re-typeset-src/mainwindow.cpp	2018-12-26 16:51:03.683711862 -0500
@@ -42,7 +42,6 @@
 	ui->pageSizeHeight->setSuffix( " px" );
 	ui->pageSizeWidth->setSuffix( " px" );
 
-	createLanguageMenu();
 	setWindowIcon( QIcon( QApplication::applicationDirPath() + "res/re-typeset.png" ) );
 
 	QString locale = QLocale::system().name();
@@ -337,38 +336,6 @@
 	}
 }
 
-void MainWindow::createLanguageMenu() {
-	QActionGroup* langGroup = new QActionGroup(ui->menuLanguage);
-	langGroup->setExclusive(true);
-	connect(langGroup, SIGNAL(triggered(QAction *)), this, SLOT(slotLanguageChanged(QAction *)));
-	QString defaultLocale = QLocale::system().name();       // e.g. "de_DE"	// format systems language
-	defaultLocale.truncate(defaultLocale.lastIndexOf('_')); // e.g. "de"
-	langPath_ = QApplication::applicationDirPath();
-	langPath_.append("/res");
-	QDir dir(langPath_);
-	QStringList fileNames = dir.entryList(QStringList("re-typeset_*.qm"));
-	if( fileNames.size() == 0 ) {
-		ui->menuLanguage->setDisabled( true );
-	} else {
-		for (int i = 0; i < fileNames.size(); ++i) {
-			QString locale;// get locale extracted by filename
-			locale = fileNames[i];                  // "TranslationExample_de.qm"
-			locale.truncate(locale.lastIndexOf('.'));   // "TranslationExample_de"
-			locale.remove(0, locale.indexOf('_') + 1);   // "de"
-			QString lang = QLocale::languageToString(QLocale(locale).language());
-			QIcon ico(QString("%1/%2.png").arg(langPath_).arg(locale));
-			QAction *action = new QAction(ico, lang, this);
-			action->setCheckable(true);
-			action->setData(locale);
-			ui->menuLanguage->addAction(action);
-			langGroup->addAction(action);
-			if (defaultLocale == locale) {// set default translators and language checked
-				action->setChecked(true);
-			}
-		}
-	}
-}
-
 void MainWindow::changeEvent(QEvent * event) {
 	if(0 != event) {
 		switch(event->type()) {
diff -Naur re-typeset-1.2.0/re-typeset-src/mainwindow.ui re-typeset-1.2.0.patched/re-typeset-src/mainwindow.ui
--- re-typeset-1.2.0/re-typeset-src/mainwindow.ui	2016-10-13 15:51:19.000000000 -0400
+++ re-typeset-1.2.0.patched/re-typeset-src/mainwindow.ui	2018-12-26 16:51:03.683711862 -0500
@@ -757,11 +757,6 @@
      <height>21</height>
     </rect>
    </property>
-   <widget class="QMenu" name="menuLanguage">
-    <property name="title">
-     <string>&amp;Language</string>
-    </property>
-   </widget>
    <widget class="QMenu" name="menuOptions">
     <property name="title">
      <string>Options</string>
