diff -Naur gtetrinet-0.8.0_20220828_6d816ee.orig/src/gtetrinet.c gtetrinet-0.8.0_20220828_6d816ee/src/gtetrinet.c
--- gtetrinet-0.8.0_20220828_6d816ee.orig/src/gtetrinet.c	2023-02-10 14:01:33.000000000 -0500
+++ gtetrinet-0.8.0_20220828_6d816ee/src/gtetrinet.c	2023-02-13 01:31:08.820702289 -0500
@@ -76,13 +76,13 @@
 GSettings* settings_keys;
 GSettings* settings_themes;
 
-static const struct poptOption options[] = {
-    {"connect", 'c', POPT_ARG_STRING, &option_connect, 0, ("Connect to server"), ("SERVER")},
-    {"nickname", 'n', POPT_ARG_STRING, &option_nick, 0, ("Set nickname to use"), ("NICKNAME")},
-    {"team", 't', POPT_ARG_STRING, &option_team, 0, ("Set team name"), ("TEAM")},
-    {"spectate", 's', POPT_ARG_NONE, &option_spec, 0, ("Connect as a spectator"), NULL},
-    {"password", 'p', POPT_ARG_STRING, &option_pass, 0, ("Spectator password"), ("PASSWORD")},
-    {NULL, 0, 0, NULL, 0, NULL, NULL}
+static const GOptionEntry options[] = {
+    {"connect", 'c', 0, G_OPTION_ARG_STRING, &option_connect, ("Connect to server"), ("SERVER")},
+    {"nickname", 'n', 0, G_OPTION_ARG_STRING, &option_nick, ("Set nickname to use"), ("NICKNAME")},
+    {"team", 't', 0, G_OPTION_ARG_STRING, &option_team, ("Set team name"), ("TEAM")},
+    {"spectate", 's', 0, G_OPTION_ARG_NONE, &option_spec, ("Connect as a spectator"), NULL},
+    {"password", 'p', 0, G_OPTION_ARG_STRING, &option_pass, ("Spectator password"), ("PASSWORD")},
+    {NULL, 0, 0, 0, NULL, NULL, NULL}
 };
 
 static int gtetrinet_poll_func(GPollFD *passed_fds,
@@ -134,7 +134,6 @@
                         argc, argv, GNOME_PARAM_POPT_TABLE, options,
                         GNOME_PARAM_NONE);
     */
-    GOptionEntry options[] = { {NULL}};
     if (!gtk_init_with_args(&argc,&argv,"gtetrinet",options,NULL,&err))
     {
         fprintf (stderr, "Failed to init GTK: %s\n", err->message);
