diff -Naur gtetrinet-0.8.0_20220828_6d816ee.orig/src/client.c gtetrinet-0.8.0_20220828_6d816ee/src/client.c
--- gtetrinet-0.8.0_20220828_6d816ee.orig/src/client.c	2023-02-10 14:01:33.000000000 -0500
+++ gtetrinet-0.8.0_20220828_6d816ee/src/client.c	2023-02-12 23:55:28.980242462 -0500
@@ -431,6 +431,8 @@
       {
         case G_IO_STATUS_EOF :
           g_warning ("End of file (server closed connection).");
+			 connected = 1;
+			 client_disconnect();
           return -1;
           break;
         
