diff -Naur tetrinetx-1.13.16+qirc-1.40c.debpatched/src/net.c tetrinetx-1.13.16+qirc-1.40c/src/net.c
--- tetrinetx-1.13.16+qirc-1.40c.debpatched/src/net.c	2023-02-12 22:55:25.362581596 -0500
+++ tetrinetx-1.13.16+qirc-1.40c/src/net.c	2023-02-12 22:57:05.941572131 -0500
@@ -370,6 +370,12 @@
 int answer(sock,ip,binary)
 int sock; unsigned long *ip; int binary;
 {
+  int new_sock,addrlen; struct sockaddr_in from;
+  addrlen=sizeof(struct sockaddr);
+  new_sock=accept(sock,(struct sockaddr *)&from,&addrlen);
+  if (new_sock<0) return -1;
+  *ip=from.sin_addr.s_addr;
+  *ip=ntohl(*ip);
   /* set up all the normal socket crap */
   // setsock(new_sock,(binary ? SOCK_BINARY : 0));
   return new_sock;
