From 5c80d88f6c2662d215e3c00ef0012756bece8d3b Mon Sep 17 00:00:00 2001
From: Manuel Wassermann <manuel.wassermann97@gmail.com>
Date: Wed, 30 Mar 2022 13:56:41 +0200
Subject: [PATCH 1/3] gduatasmartdialog: Replace deprecated GIO function

---
 src/disks/gduatasmartdialog.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/disks/gduatasmartdialog.c b/src/disks/gduatasmartdialog.c
index da3116c17..2deb6618c 100644
--- a/src/disks/gduatasmartdialog.c
+++ b/src/disks/gduatasmartdialog.c
@@ -1684,15 +1684,15 @@ gdu_ata_smart_dialog_show (GduWindow    *window,
 
   action = g_simple_action_new ("short", NULL);
   g_signal_connect (action, "activate", G_CALLBACK (on_selftest_short), data);
-  g_simple_action_group_insert (group, G_ACTION (action));
+  g_action_map_add_action (G_ACTION_MAP (group), G_ACTION (action));
 
   action = g_simple_action_new ("extended", NULL);
   g_signal_connect (action, "activate", G_CALLBACK (on_selftest_extended), data);
-  g_simple_action_group_insert (group, G_ACTION (action));
+  g_action_map_add_action (G_ACTION_MAP (group), G_ACTION (action));
 
   action = g_simple_action_new ("conveyance", NULL);
   g_signal_connect (action, "activate", G_CALLBACK (on_selftest_conveyance), data);
-  g_simple_action_group_insert (group, G_ACTION (action));
+  g_action_map_add_action (G_ACTION_MAP (group), G_ACTION (action));
 
   gtk_widget_insert_action_group (GTK_WIDGET (data->dialog), "test", G_ACTION_GROUP (group));
 
-- 
GitLab


From 1e42b8796e69d922382c91b5313cc8f86e871186 Mon Sep 17 00:00:00 2001
From: Manuel Wassermann <manuel.wassermann97@gmail.com>
Date: Wed, 30 Mar 2022 14:06:31 +0200
Subject: [PATCH 2/3] gduformatdiskdialog: Fix parameter of g_signal_connect ()
 and remove unused variable

---
 src/disks/gduformatdiskdialog.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/disks/gduformatdiskdialog.c b/src/disks/gduformatdiskdialog.c
index 44a0280a0..2a0311722 100644
--- a/src/disks/gduformatdiskdialog.c
+++ b/src/disks/gduformatdiskdialog.c
@@ -436,7 +436,6 @@ gdu_format_disk_dialog_show (GduWindow    *window,
                              UDisksObject *object)
 {
   FormatDiskData *data;
-  gint response;
 
   data = g_new0 (FormatDiskData, 1);
   data->window = g_object_ref (window);
@@ -470,6 +469,6 @@ gdu_format_disk_dialog_show (GduWindow    *window,
   gtk_widget_show_all (data->dialog);
   gtk_widget_grab_focus (data->type_combobox);
 
-  g_signal_connect (data->dialog, "response", dialog_response_cb, data);
+  g_signal_connect (data->dialog, "response", G_CALLBACK (dialog_response_cb), data);
   gtk_window_present (GTK_WINDOW (data->dialog));
 }
-- 
GitLab


From 465e2e45547ea586923f4be152f0caed112f8b25 Mon Sep 17 00:00:00 2001
From: Manuel Wassermann <manuel.wassermann97@gmail.com>
Date: Wed, 30 Mar 2022 14:09:27 +0200
Subject: [PATCH 3/3] gduapplication: Fix parameter of g_signal_connect ()

---
 src/disks/gduapplication.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/disks/gduapplication.c b/src/disks/gduapplication.c
index b97c38293..bca06e6bf 100644
--- a/src/disks/gduapplication.c
+++ b/src/disks/gduapplication.c
@@ -354,7 +354,7 @@ about_activated (GSimpleAction *action,
 
   gtk_window_set_transient_for (GTK_WINDOW (dialog), GTK_WINDOW (app->window));
   gtk_widget_show_all (dialog);
-  g_signal_connect (dialog, "response", on_about_dialog_response, NULL);
+  g_signal_connect (dialog, "response", G_CALLBACK (on_about_dialog_response), NULL);
   gtk_window_present (GTK_WINDOW (dialog));
 }
 
-- 
GitLab

