diff -Nur wmdrawer-0.10.5.orig/Makefile wmdrawer-0.10.5/Makefile
--- wmdrawer-0.10.5.orig/Makefile	2004-06-26 13:16:04.000000000 -0500
+++ wmdrawer-0.10.5/Makefile	2009-05-10 21:40:22.098379875 -0500
@@ -24,19 +24,19 @@
 
 ifdef USE_GDKPIXBUF
 IMGLIB_DEF = -DUSE_GDKPIXBUF
-CFLAGS = -O3 -Wall `gdk-pixbuf-config --cflags`
+CFLAGS = -Wall $(EXTRA_CFLAGS) `gdk-pixbuf-config --cflags`
 LDFLAGS = `gdk-pixbuf-config --libs` -lgdk_pixbuf_xlib
 endif
 
 ifdef USE_GDKPIXBUF2
 IMGLIB_DEF = -DUSE_GDKPIXBUF2
-CFLAGS = -O3 -Wall `pkg-config --cflags gdk-pixbuf-xlib-2.0` -I/usr/X11R6/include
-LDFLAGS = `pkg-config --libs gdk-pixbuf-xlib-2.0` -rdynamic -lX11 -lXi -lXext -L/usr/X11R6/lib
+CFLAGS = -Wall $(EXTRA_CFLAGS) `pkg-config --cflags gdk-pixbuf-xlib-2.0` -I/usr/include
+LDFLAGS = `pkg-config --libs gdk-pixbuf-xlib-2.0` -rdynamic -lX11 -lXi -lXext -L/usr/lib
 endif
 
 ifdef USE_IMLIB
 IMGLIB_DEF = -DUSE_IMLIB
-CFLAGS = -O3 -Wall `imlib-config --cflags`
+CFLAGS = -Wall $(EXTRA_CFLAGS)  `imlib-config --cflags`
 LDFLAGS = `imlib-config --libs`
 endif
 
