--- yabause-0.9.15/src/mpeg_card.c.orig	2017-04-14 21:15:44.908378882 +0700
+++ yabause-0.9.15/src/mpeg_card.c	2017-04-14 21:17:15.386497605 +0700
@@ -535,7 +535,7 @@
   yab_mpeg.audio.context->channels = 2;
   yab_mpeg.audio.context->sample_rate = 44100;
 
-  ret = av_image_alloc(out_buf,out_linesize, 320, 240,PIX_FMT_RGB32, 1);
+  ret = av_image_alloc(out_buf,out_linesize, 320, 240,AV_PIX_FMT_RGB32, 1);
 
   if(!ret)
     assert(0);
@@ -561,7 +561,7 @@
       //dest width, height, format
       out_width,
       out_height,
-      PIX_FMT_RGB32,
+      AV_PIX_FMT_RGB32,
       //flags
       SWS_BILINEAR,
       //source/dest filters
