diff -Naur yabause-0.9.15.orig/src/qt/ui/UICheatRaw.cpp yabause-0.9.15/src/qt/ui/UICheatRaw.cpp
--- yabause-0.9.15.orig/src/qt/ui/UICheatRaw.cpp	2016-08-19 17:31:13.000000000 +0200
+++ yabause-0.9.15/src/qt/ui/UICheatRaw.cpp	2022-03-04 20:54:59.817095000 +0100
@@ -20,8 +20,6 @@
 #include "UIHexInput.h"
 #include "../QtYabause.h"
 
-#include <QButtonGroup>
-
 UICheatRaw::UICheatRaw( QWidget* p )
 	: QDialog( p )
 {
diff -Naur yabause-0.9.15.orig/src/qt/ui/UICheatRaw.h yabause-0.9.15/src/qt/ui/UICheatRaw.h
--- yabause-0.9.15.orig/src/qt/ui/UICheatRaw.h	2016-08-19 17:31:13.000000000 +0200
+++ yabause-0.9.15/src/qt/ui/UICheatRaw.h	2022-03-04 20:55:40.401095000 +0100
@@ -20,8 +20,7 @@
 #define UICHEATRAW_H
 
 #include "ui_UICheatRaw.h"
-
-class QButtonGroup;
+#include <QButtonGroup>
 
 class UICheatRaw : public QDialog, public Ui::UICheatRaw
 {
diff -Naur yabause-0.9.15.orig/src/qt/ui/UICheats.cpp yabause-0.9.15/src/qt/ui/UICheats.cpp
--- yabause-0.9.15.orig/src/qt/ui/UICheats.cpp	2016-08-19 17:31:13.000000000 +0200
+++ yabause-0.9.15/src/qt/ui/UICheats.cpp	2022-03-04 20:56:02.818095000 +0100
@@ -20,6 +20,7 @@
 #include "UICheatAR.h"
 #include "UICheatRaw.h"
 #include "../CommonDialogs.h"
+#include <QButtonGroup>
 
 UICheats::UICheats( QWidget* p )
 	: QDialog( p )
diff -Naur yabause-0.9.15.orig/src/qt/ui/UIHexInput.h yabause-0.9.15/src/qt/ui/UIHexInput.h
--- yabause-0.9.15.orig/src/qt/ui/UIHexInput.h	2016-08-19 17:31:13.000000000 +0200
+++ yabause-0.9.15/src/qt/ui/UIHexInput.h	2022-03-04 20:56:27.297095000 +0100
@@ -21,6 +21,7 @@
 
 #include "ui_UIHexInput.h"
 #include "../QtYabause.h"
+#include <QValidator>
 
 class HexValidator : public QValidator
 {
