--- lirc-0.8.6/drivers/lirc_dev/lirc_dev.c.old	2009-08-31 12:57:55.000000000 -0400
+++ lirc-0.8.6/drivers/lirc_dev/lirc_dev.c	2010-03-07 01:40:17.000000000 -0500
@@ -32,7 +32,11 @@
 #error "**********************************************************"
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 #include <linux/autoconf.h>
+#else
+#include <generated/autoconf.h>
+#endif
 #include <linux/module.h>
 #include <linux/kernel.h>
 #include <linux/sched.h>
--- lirc-0.8.6/drivers/lirc_sir/lirc_sir.c.old	2009-07-09 18:24:23.000000000 -0400
+++ lirc-0.8.6/drivers/lirc_sir/lirc_sir.c	2010-03-07 01:40:17.000000000 -0500
@@ -45,7 +45,11 @@
 # include <config.h>
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 #include <linux/autoconf.h>
+#else
+#include <generated/autoconf.h>
+#endif
 #if !defined(CONFIG_SERIAL_MODULE)
 #if !defined(LIRC_ON_SA1100)
 #warning "******************************************"
--- lirc-0.8.6/drivers/lirc_serial/lirc_serial.c.old	2009-03-15 05:34:00.000000000 -0400
+++ lirc-0.8.6/drivers/lirc_serial/lirc_serial.c	2010-03-07 01:40:17.000000000 -0500
@@ -60,7 +60,11 @@
 #error "**********************************************************"
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 #include <linux/autoconf.h>
+#else
+#include <generated/autoconf.h>
+#endif
 
 #if defined(CONFIG_SERIAL) || defined(CONFIG_SERIAL_8250)
 #warning "******************************************"
--- lirc-0.8.6/drivers/lirc_bt829/lirc_bt829.c.old	2009-03-09 14:54:17.000000000 -0400
+++ lirc-0.8.6/drivers/lirc_bt829/lirc_bt829.c	2010-03-07 01:40:17.000000000 -0500
@@ -22,7 +22,11 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 4, 0)
 #error "This driver needs kernel version 2.4.0 or higher"
 #endif
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 #include <linux/autoconf.h>
+#else
+#include <generated/autoconf.h>
+#endif
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <linux/threads.h>
--- lirc-0.8.6/drivers/lirc_sasem/lirc_sasem.c.old	2009-02-28 05:27:10.000000000 -0500
+++ lirc-0.8.6/drivers/lirc_sasem/lirc_sasem.c	2010-03-07 01:40:17.000000000 -0500
@@ -41,7 +41,11 @@
 #error "*** Sorry, this driver requires kernel version 2.4.22 or higher"
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 #include <linux/autoconf.h>
+#else
+#include <generated/autoconf.h>
+#endif
 
 #include <linux/errno.h>
 #include <linux/init.h>
--- lirc-0.8.6/drivers/lirc_igorplugusb/lirc_igorplugusb.c.old	2009-08-02 05:54:10.000000000 -0400
+++ lirc-0.8.6/drivers/lirc_igorplugusb/lirc_igorplugusb.c	2010-03-07 01:40:17.000000000 -0500
@@ -47,7 +47,11 @@
 #error "*******************************************************"
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 #include <linux/autoconf.h>
+#else
+#include <generated/autoconf.h>
+#endif
 #include <linux/module.h>
 #include <linux/kernel.h>
 #include <linux/kmod.h>
--- lirc-0.8.6/drivers/lirc_imon/lirc_imon.c.old	2009-09-11 00:56:18.000000000 -0400
+++ lirc-0.8.6/drivers/lirc_imon/lirc_imon.c	2010-03-07 01:40:17.000000000 -0500
@@ -30,7 +30,11 @@
 #error "*** Sorry, this driver requires a 2.6 kernel"
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 #include <linux/autoconf.h>
+#else
+#include <generated/autoconf.h>
+#endif
 
 #include <linux/errno.h>
 #include <linux/init.h>
--- lirc-0.8.6/drivers/lirc_it87/lirc_it87.c.old	2009-06-01 08:21:31.000000000 -0400
+++ lirc-0.8.6/drivers/lirc_it87/lirc_it87.c	2010-03-07 01:40:17.000000000 -0500
@@ -36,7 +36,11 @@
 
 #include <linux/version.h>
 #include <linux/module.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 #include <linux/autoconf.h>
+#else
+#include <generated/autoconf.h>
+#endif
 #include <linux/sched.h>
 #include <linux/errno.h>
 #include <linux/signal.h>
--- lirc-0.8.6/drivers/lirc_streamzap/lirc_streamzap.c.old	2009-03-15 05:34:00.000000000 -0400
+++ lirc-0.8.6/drivers/lirc_streamzap/lirc_streamzap.c	2010-03-07 01:40:17.000000000 -0500
@@ -35,7 +35,11 @@
 #error "Sorry, this driver needs kernel version 2.4.0 or higher"
 #error "*******************************************************"
 #endif
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 #include <linux/autoconf.h>
+#else
+#include <generated/autoconf.h>
+#endif
 #include <linux/kernel.h>
 #include <linux/errno.h>
 #include <linux/init.h>
--- lirc-0.8.6/drivers/lirc_atiusb/lirc_atiusb.c.old	2009-03-10 20:21:46.000000000 -0400
+++ lirc-0.8.6/drivers/lirc_atiusb/lirc_atiusb.c	2010-03-07 01:40:17.000000000 -0500
@@ -43,7 +43,11 @@
 #error "*******************************************************"
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 #include <linux/autoconf.h>
+#else
+#include <generated/autoconf.h>
+#endif
 
 #include <linux/kernel.h>
 #include <linux/errno.h>
--- lirc-0.8.6/drivers/lirc_parallel/lirc_parallel.c.old	2009-03-08 15:22:28.000000000 -0400
+++ lirc-0.8.6/drivers/lirc_parallel/lirc_parallel.c	2010-03-07 01:40:17.000000000 -0500
@@ -34,7 +34,11 @@
 #error "**********************************************************"
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 #include <linux/autoconf.h>
+#else
+#include <generated/autoconf.h>
+#endif
 #ifdef CONFIG_SMP
 #error "--- Sorry, this driver is not SMP safe. ---"
 #endif
@@ -43,7 +47,11 @@
 #include <linux/sched.h>
 #include <linux/errno.h>
 #include <linux/signal.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 #include <linux/autoconf.h>
+#else
+#include <generated/autoconf.h>
+#endif
 #include <linux/fs.h>
 #include <linux/kernel.h>
 #include <linux/ioport.h>
--- lirc-0.8.6/drivers/lirc_wpc8769l/lirc_wpc8769l.c.old	2009-03-15 05:34:01.000000000 -0400
+++ lirc-0.8.6/drivers/lirc_wpc8769l/lirc_wpc8769l.c	2010-03-07 01:40:17.000000000 -0500
@@ -37,7 +37,11 @@
 #error "**********************************************************"
 #endif
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 #include <linux/autoconf.h>
+#else
+#include <generated/autoconf.h>
+#endif
 
 #include <linux/module.h>
 #include <linux/errno.h>
--- lirc-0.8.6/drivers/lirc_mceusb/lirc_mceusb.c.old	2009-09-02 10:04:02.000000000 -0400
+++ lirc-0.8.6/drivers/lirc_mceusb/lirc_mceusb.c	2010-03-07 01:40:17.000000000 -0500
@@ -52,7 +52,11 @@
 #error "Sorry, this driver needs kernel version 2.6.5 or higher"
 #error "*******************************************************"
 #endif
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 #include <linux/autoconf.h>
+#else
+#include <generated/autoconf.h>
+#endif
 #include <linux/kernel.h>
 #include <linux/errno.h>
 #include <linux/init.h>
