From 69877e30c86771d2a99d9820cbc8d96415bb3395 Mon Sep 17 00:00:00 2001
From: Robin Gareus <robin@gareus.org>
Date: Sat, 23 Apr 2022 02:32:54 +0200
Subject: [PATCH] Fix ffmpeg compatibility

---
 src/xjadeo/ffcompat.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/xjadeo/ffcompat.h b/src/xjadeo/ffcompat.h
index 6c71292..b60baad 100644
--- a/src/xjadeo/ffcompat.h
+++ b/src/xjadeo/ffcompat.h
@@ -75,8 +75,10 @@ static inline void av_packet_unref (AVPacket *pkt)
 #if LIBAVUTIL_VERSION_INT < AV_VERSION_INT(51, 63, 100)
 static inline int av_image_get_buffer_size (enum AVPixelFormat pix_fmt, int width, int height, int align)
 {
-	return avpicture_get_size (pix_fmt, width, height)
+	return avpicture_get_size (pix_fmt, width, height);
 }
+#else
+#include <libavutil/imgutils.h>
 #endif
 
 #if LIBAVFORMAT_VERSION_INT < AV_VERSION_INT(53, 2, 0)
