--- quickswitch-i3-2.1/quickswitch.original.py	2014-05-02 13:32:46.939145487 +0300
+++ quickswitch-i3-2.1/quickswitch.py	2014-05-02 13:41:20.155729397 +0300
@@ -30,8 +30,13 @@
 try:
     import i3
 except ImportError:
+    print("")
+    print("----------------------------------------------------")
     print("quickswitch requires i3-py.")
-    print("You can install it from the PyPI with ``pip install i3-py''.")
+    print("You can install it from http://slackbuilds.org site")
+    print("or from PyPI with ``pip install i3-py''.")
+    print("----------------------------------------------------")
+    print("")
     exit(1)
 
 
