From: John MacFarlane <jgm@berkeley.edu>
Date: Thu, 22 Jul 2010 19:11:14 -0700
Subject: [PATCH] Put utf8-string dependency in right place.

---
 texmath.cabal |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/texmath.cabal b/texmath.cabal
index f14d7dc..ad83d05 100644
--- a/texmath.cabal
+++ b/texmath.cabal
@@ -77,11 +77,10 @@ Executable testTeXMathML
 
 Executable texmath-cgi
     Main-is:             cgi/texmath-cgi.hs
-    Build-depends:       utf8-string
     Ghc-Options:         -Wall -fno-warn-unused-do-bind
     Ghc-Prof-Options:    -auto-all
     if flag(cgi)
       Buildable:         True
-      Build-depends:     cgi, json
+      Build-depends:     cgi, json, utf8-string
     else
       Buildable:         False
-- 
1.7.1

