diff -Naur byobu-2.82.orig/etc/byobu/socketdir byobu-2.82/etc/byobu/socketdir
--- byobu-2.82.orig/etc/byobu/socketdir	2010-05-15 22:47:59.000000000 +0200
+++ byobu-2.82/etc/byobu/socketdir	2010-07-20 09:48:36.235677637 +0200
@@ -7,4 +7,4 @@
 #  * the variable name is SOCKETDIR
 #  * there is no space around the "="
 #  * and that the path value is quoted
-SOCKETDIR="/var/run/screen"
+SOCKETDIR="~/.screen"
diff -Naur byobu-2.82.orig/usr/bin/byobu-config byobu-2.82/usr/bin/byobu-config
--- byobu-2.82.orig/usr/bin/byobu-config	2010-06-23 23:44:51.000000000 +0200
+++ byobu-2.82/usr/bin/byobu-config	2010-07-20 15:16:12.221678378 +0200
@@ -29,10 +29,11 @@
 HOME=os.getenv("HOME")
 USER=os.getenv("USER")
 PKG="byobu"
+VERSION="2.82"
 SHARE='/usr/share/'+PKG
 if not os.path.exists(SHARE):
     SHARE = "%s/.%s/%s" % (HOME, PKG, SHARE)
-DOC='/usr/share/doc/'+PKG
+DOC='/usr/doc/'+PKG+'-'+VERSION
 if not os.path.exists(DOC):
     DOC = "%s/.%s/%s" % (HOME, PKG, DOC)
 DEF_ESC="A"
diff -Naur byobu-2.82.orig/usr/bin/byobu-export byobu-2.82/usr/bin/byobu-export
--- byobu-2.82.orig/usr/bin/byobu-export	2010-06-24 17:18:56.000000000 +0200
+++ byobu-2.82/usr/bin/byobu-export	2010-07-20 15:14:28.705957556 +0200
@@ -20,6 +20,7 @@
 set -e
 
 PKG="byobu"
+VERSION="2.82"
 DIR=`mktemp -t -d $PKG.XXXXXXXX` || error "Could not create a temporary directory"
 [ -d "$TMP" ] || TMP=/tmp
 
@@ -91,7 +92,7 @@
 # Copy necessary scripts
 cp -a /etc/$PKG "$DIR/.$PKG/etc"
 cp -a /usr/share/$PKG "$DIR/.$PKG/usr/share"
-cp -a /usr/share/doc/$PKG "$DIR/.$PKG/usr/share/doc"
+cp -a /usr/doc/$PKG-$VERSION "$DIR/.$PKG/usr/share/doc"
 cp -a /usr/lib/$PKG "$DIR/.$PKG/usr/lib"
 cp -a /usr/bin/$PKG-* "$DIR/.$PKG/usr/bin"
 cp /etc/$PKG/statusrc "$DIR/.$PKG/status"
