diff -Nur gforth-0.7.3/Makefile.in gforth-0.7.3.new/Makefile.in
--- gforth-0.7.3/Makefile.in	2013-02-24 17:38:34.965892542 +0000
+++ gforth-0.7.3.new/Makefile.in	2013-02-25 13:36:38.227716339 +0000
@@ -60,7 +60,7 @@
 MASSAGE_EXE = @MASSAGE_EXE@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
 INSTALL_DATA = @INSTALL_DATA@
-INSTALL_INFO = @INSTALL_INFO@
+INSTALL_INFO = 
 LN_S	= @LN_S@
 M4      = @M4@
 GCC	= @CC@
