diff --color -Naur old/configure.in new/configure.in
--- old/configure.in    2025-11-29 09:42:17.338905456 -0500
+++ new/configure.in    2025-11-29 11:35:45.782499641 -0500
@@ -196,7 +196,7 @@
 LDFLAGS="$LDFLAGS -export-dynamic"
 dnl AC_TRY_LINK gives false positive on rs6000-ibm-aix4.2.1.0
 dnl AC_TRY_LINK(,,ac_export_dynamic=yes,ac_export_dynamic=no)
-AC_TRY_RUN(main(){exit(0);},ac_export_dynamic=yes,ac_export_dynamic=no,ac_export_dynamic=no)
+AC_TRY_RUN(int main(){return 0;},ac_export_dynamic=yes,ac_export_dynamic=no,ac_export_dynamic=no)
 test $ac_export_dynamic = yes|| LDFLAGS=$OLDLDFLAGS
 AC_MSG_RESULT($ac_export_dynamic)
 
@@ -771,7 +771,7 @@
 [int foo(int,int,int);
 main()
 {
-  exit(foo(0,0,0)!=16);
+  return foo(0,0,0) < 16;
 }
 int foo(int x, int y, int z)
 {
diff --color -Naur old/engine/signals.c new/engine/signals.c
--- old/engine/signals.c        2013-10-11 17:31:28.060992169 -0400
+++ new/engine/signals.c        2025-11-29 10:51:06.463659341 -0500
@@ -399,7 +399,7 @@
 #endif
   };
   int i;
-  void (*throw_handler)() = die_on_signal ? graceful_exit : signal_throw;
+  void (*throw_handler)(int) = die_on_signal ? graceful_exit : signal_throw;
 #ifdef SIGSTKSZ 
   stack_t sigstack;
   int sas_retval=-1;
diff --color -Naur old/engine/support.c new/engine/support.c
--- old/engine/support.c        2013-10-11 17:31:28.062992193 -0400
+++ new/engine/support.c        2025-11-29 10:51:06.463659341 -0500
@@ -79,7 +79,7 @@
 {
   char *s1,*s2;
   int s1_len, s2_len;
-  struct passwd *getpwnam (), *user_entry;
+  struct passwd *getpwnam (const char *), *user_entry;
 
   if (size<1 || from[0]!='~')
     return cstr(from, size, clear);
diff --color -Naur old/fflib.fs new/fflib.fs
--- old/fflib.fs        2013-10-11 17:31:28.065992229 -0400
+++ new/fflib.fs        2025-11-29 11:02:31.103618534 -0500
@@ -97,7 +97,7 @@
 c-function av-call-longlong av_call_longlong -- d
 \c #define av_call_ptr() (av_call(alist), prv)
 c-function av-call-ptr av_call_ptr -- a
-\c #define alloc_callback1(a_ip) alloc_callback(gforth_callback_ffcall, (Xt *)a_ip)
+\c #define alloc_callback1(a_ip) alloc_callback((callback_function_t) gforth_callback_ffcall, (Xt *)a_ip)
 c-function alloc-callback alloc_callback1 a -- a
 \c #define va_start_void1() va_start_void(gforth_clist)
 c-function va-start-void va_start_void1 -- void
diff --color -Naur old/libffi.fs new/libffi.fs
--- old/libffi.fs       2013-10-11 17:31:28.074992337 -0400
+++ new/libffi.fs       2025-11-29 11:06:12.294605350 -0500
@@ -96,7 +96,7 @@
 c-function ffi-call ffi_call1 a a a a -- void
 
 \c #define ffi_prep_closure1(a_ip, a_cif, a_closure) \
-\c              ffi_prep_closure((ffi_closure *)a_closure, (ffi_cif *)a_cif, gforth_callback_ffi, (void *)a_ip)
+\c              ffi_prep_closure_loc((ffi_closure *)a_closure, (ffi_cif *)a_cif, gforth_callback_ffi, (void *)a_ip, NULL)
 c-function ffi-prep-closure ffi_prep_closure1 a a a -- n
 
 \ !! use ud?
