diff -Naur Amaya11.3.1.orig/Amaya/amaya/Makefile.in Amaya11.3.1.new/Amaya/amaya/Makefile.in
--- Amaya11.3.1.orig/Amaya/amaya/Makefile.in	2012-01-14 18:53:00.000000000 +0100
+++ Amaya11.3.1.new/Amaya/amaya/Makefile.in	2012-01-14 18:55:12.000000000 +0100
@@ -21,7 +21,7 @@
 	  $(GL_LIBRARIES) $(GTK_GL_LIBRARIES) $(FREETYPE_LIBRARIES) $(XFT_LIBRARIES) \
 	  $(AMAYA_LIBWWW_LIBS) \
 	   @AMAYA_RAPTOR_LIBS@ \
-          $(WX_LIBRARIES)
+          $(WX_LIBRARIES) -lpng14 -ljpeg -lexpat -lwwwssl
 
 AMAYA	= @top_srcdir@/amaya
 
diff -Naur Amaya11.3.1.orig/Amaya/thotlib/Makefile.in Amaya11.3.1.new/Amaya/thotlib/Makefile.in
--- Amaya11.3.1.orig/Amaya/thotlib/Makefile.in	2012-01-14 18:53:00.000000000 +0100
+++ Amaya11.3.1.new/Amaya/thotlib/Makefile.in	2012-01-14 18:55:45.000000000 +0100
@@ -351,7 +351,7 @@
 
 ../bin/print : $(THOTK_OBJ) $(DISPLAY_OBJ) $(PRINT_OBJ) \
         ../amaya/libCSS.a
-	$(CXX) $(LDFLAGS) -o $@ $(THOTK_OBJ) $(DISPLAY_OBJ) $(PRINT_OBJ) ../amaya/libCSS.a $(LIBS)
+	$(CXX) $(LDFLAGS) -o $@ $(THOTK_OBJ) $(DISPLAY_OBJ) $(PRINT_OBJ) ../amaya/libCSS.a $(LIBS) -lpng14 -ljpeg
 	@$(ECHO) "====>" print is done
 
 CEXTRACT= ../bin/cextract
