--- Amaya11.4.4/Amaya/amaya/html2thot.c.old	2012-03-07 10:53:27.904494374 +0100
+++ Amaya11.4.4/Amaya/amaya/html2thot.c	2012-03-07 11:20:46.931113208 +0100
@@ -4349,7 +4349,7 @@
       WorkBuffer = FileBuffer;
     }
   // need to read a new set of characters
-  LastCharInWorkBuffer = gzread (infile, &FileBuffer[StartOfRead],
+  LastCharInWorkBuffer = gzread ((gzFile) infile, &FileBuffer[StartOfRead],
                                  INPUT_FILE_BUFFER_SIZE - StartOfRead);
   // add previous read characters not managed yet
   LastCharInWorkBuffer += StartOfRead;
--- Amaya11.4.4/Amaya/amaya/Xml2thot.c.old	2012-03-07 11:21:30.994445257 +0100
+++ Amaya11.4.4/Amaya/amaya/Xml2thot.c	2012-03-07 11:22:04.644444274 +0100
@@ -5847,6 +5847,6 @@
   while (!endOfFile && !XMLNotWellFormed && !XMLInvalidToken)
     {
       /* read the XML file */
-      res = gzread (infile, bufferRead, COPY_BUFFER_SIZE);
+      res = gzread ((gzFile) infile, bufferRead, COPY_BUFFER_SIZE);
       if (res < 0)
         return;
