--- guake/paths.py.orig	2022-02-06 15:09:08.000000000 -0500
+++ guake/paths.py	2022-02-06 20:42:24.476974008 -0500
@@ -47,37 +47,37 @@
 
 
 def get_default_data_dir():
-    d = os.path.join(get_data_files_dir(), "data")
+    d = '/usr/share/guake/data'
     log.debug("Using guake data directory: %s", d)
     return d
 
 
 def get_default_locale_dir():
-    d = os.path.join(get_data_files_dir(), "po")
-    log.debug("Using guake image directory: %s", d)
+    d = '/usr/share/locale'
+    log.debug("Using guake locale directory: %s", d)
     return d
 
 
 def get_default_image_dir():
-    d = os.path.join(get_default_data_dir(), 'pixmaps')
+    d = '/usr/share/guake/pixmaps'
     log.debug("Using guake image directory: %s", d)
     return d
 
 
 def get_default_glade_dir():
-    d = get_default_data_dir()
+    d = '/usr/share/guake'
     log.debug("Using guake glade directory: %s", d)
     return d
 
 
 def get_default_schema_dir():
-    d = get_default_data_dir()
+    d = '/usr/share/glib-2.0/schemas'
     log.debug("Using guake scheme directory: %s", d)
     return d
 
 
 def get_default_theme_dir():
-    d = os.path.join(get_default_data_dir(), 'theme')
+    d = '/usr/share/guake/theme'
     log.debug("Using guake theme directory: %s", d)
     return d
 
