diff -Nur kernel-open.orig/nvidia-uvm/uvm_va_range_device_p2p.c kernel-open/nvidia-uvm/uvm_va_range_device_p2p.c
--- kernel-open.orig/nvidia-uvm/uvm_va_range_device_p2p.c	2025-09-23 05:56:12.000000000 -0400
+++ kernel-open/nvidia-uvm/uvm_va_range_device_p2p.c	2025-10-20 21:11:44.667846111 -0400
@@ -360,7 +360,7 @@
         // a reference to them, so take one now if using DEVICE_COHERENT pages.
         if (gpu->parent->cdmm_enabled) {
             get_page(page);
-            get_dev_pagemap(page_to_pfn(page), NULL);
+            get_dev_pagemap(page_to_pfn(page));
         }
 #else
         // CDMM P2PDMA will never be enabled for this case
