diff -Naur openvas-manager-1.0.0-beta5.orig/src/CMakeLists.txt openvas-manager-1.0.0-beta5/src/CMakeLists.txt
--- openvas-manager-1.0.0-beta5.orig/src/CMakeLists.txt	2010-03-04 15:00:41.000000000 +0100
+++ openvas-manager-1.0.0-beta5/src/CMakeLists.txt	2010-03-09 14:27:37.161580945 +0100
@@ -128,7 +128,7 @@
 endif (OPENVAS_DATA_DIR)
 
 set_target_properties (openvasmd PROPERTIES LINK_FLAGS
-                       "${LIB_TEMP} -lsqlite3 -lgnutls -luuid ${GLIB_LDFLAGS} ${OPENVAS_LDFLAGS}")
+                       "${LIB_TEMP} -lsqlite3 -ldl -lgnutls -luuid ${GLIB_LDFLAGS} ${OPENVAS_LDFLAGS}")
 
 set_target_properties (openvasmd PROPERTIES COMPILE_FLAGS
                        "${HEADER_TEMP} ${OPENVAS_CFLAGS} ${GLIB_CFLAGS}")
diff -Naur openvas-manager-1.0.0-beta5.orig/src/tests/CMakeLists.txt openvas-manager-1.0.0-beta5/src/tests/CMakeLists.txt
--- openvas-manager-1.0.0-beta5.orig/src/tests/CMakeLists.txt	2010-03-04 15:00:41.000000000 +0100
+++ openvas-manager-1.0.0-beta5/src/tests/CMakeLists.txt	2010-03-09 14:28:06.409890850 +0100
@@ -45,7 +45,7 @@
   set (LIB_TEMP "")
 endif (OPENVAS_LIB_INSTALL_DIR)
 
-set (OVAS_LDFLAG "${LIB_TEMP} -lsqlite3 -lgnutls ${OPENVAS_LDFLAGS}")
+set (OVAS_LDFLAG "${LIB_TEMP} -lsqlite3 -ldl -lgnutls ${OPENVAS_LDFLAGS}")
 
 MARK_AS_ADVANCED (LIB_TEMP)
 
