diff -Naur MySQL-python-1.2.4.orig/_mysql.c MySQL-python-1.2.4/_mysql.c
--- MySQL-python-1.2.4.orig/_mysql.c	2013-01-01 17:18:48.000000000 +0100
+++ MySQL-python-1.2.4/_mysql.c	2014-11-07 20:25:20.483927425 +0100
@@ -26,6 +26,14 @@
 PERFORMANCE OF THIS SOFTWARE.
 */
 
+#if defined(MS_WINDOWS)
+#include <config-win.h>
+#else
+#include "my_config.h"
+#endif
+#include "mysql.h"
+#include "mysqld_error.h"
+#include "errmsg.h"
 #include "Python.h"
 #if PY_MAJOR_VERSION >= 3
 #define IS_PY3K
@@ -38,14 +46,6 @@
 #endif
 #include "pymemcompat.h"
 #include "structmember.h"
-#if defined(MS_WINDOWS)
-#include <config-win.h>
-#else
-#include "my_config.h"
-#endif
-#include "mysql.h"
-#include "mysqld_error.h"
-#include "errmsg.h"
 
 #if PY_VERSION_HEX < 0x02020000
 # define MyTuple_Resize(t,n,d) _PyTuple_Resize(t, n, d)
