Index: lastfm-1.5.1.31879/src/myvolumeslider.h
===================================================================
--- /dev/null
+++ lastfm-1.5.1.31879/src/myvolumeslider.h
@@ -0,0 +1 @@
+#include "volumeslider.h"
Index: lastfm-1.5.1.31879/src/playcontrols.ui
===================================================================
--- lastfm-1.5.1.31879.orig/src/playcontrols.ui
+++ lastfm-1.5.1.31879/src/playcontrols.ui
@@ -146,7 +146,7 @@
   <customwidget>
    <class>VolumeSlider</class>
    <extends>QFrame</extends>
-   <header>volumeslider.h</header>
+   <header>myvolumeslider.h</header>
    <container>1</container>
   </customwidget>
  </customwidgets>
