--- a/duckstation-d690732c7d553ab1dc1a8e59889c788d7397ec9d/CMakeLists.txt	2023-09-01 05:30:53.000000000 -0600
+++ b/duckstation-d690732c7d553ab1dc1a8e59889c788d7397ec9d/CMakeLists.txt	2026-07-08 17:22:51.120161967 -0600
@@ -97,7 +97,7 @@
     find_package(CURL REQUIRED)
   endif()
   if(BUILD_QT_FRONTEND)
-    find_package(Qt6 COMPONENTS Core Gui Widgets Network LinguistTools REQUIRED)
+    find_package(Qt6 COMPONENTS Core Gui GuiPrivate Widgets Network LinguistTools REQUIRED)
   endif()
 endif()
 
--- a/duckstation-d690732c7d553ab1dc1a8e59889c788d7397ec9d/src/duckstation-qt/CMakeLists.txt	2023-09-01 05:30:53.000000000 -0600
+++ b/duckstation-d690732c7d553ab1dc1a8e59889c788d7397ec9d/src/duckstation-qt/CMakeLists.txt	2026-07-08 17:23:05.733463135 -0600
@@ -166,7 +166,7 @@
 
 add_executable(duckstation-qt ${SRCS} ${QM_FILES})
 target_include_directories(duckstation-qt PRIVATE "${Qt6Gui_PRIVATE_INCLUDE_DIRS}" "${CMAKE_CURRENT_SOURCE_DIR}")
-target_link_libraries(duckstation-qt PRIVATE core common imgui minizip scmversion Qt6::Core Qt6::Gui Qt6::Widgets Qt6::Network)
+target_link_libraries(duckstation-qt PRIVATE core common imgui minizip scmversion Qt6::Core Qt6::Gui Qt6::GuiPrivate Qt6::Widgets Qt6::Network)
 
 if(WIN32)
   # We want a Windows subsystem application not console.
