--- a/docs/libdbusmenu-glib/reference/libdbusmenu-glib-docs.sgml
+++ b/docs/libdbusmenu-glib/reference/libdbusmenu-glib-docs.sgml
@@ -15,9 +15,6 @@
     <xi:include href="xml/menuitem-proxy.xml"/>
     <xi:include href="xml/menuitem.xml"/>
     <xi:include href="xml/client.xml"/>
-    <xi:include href="xml/client-menuitem.xml"/>
-    <xi:include href="xml/dbusmenu-client.xml"/>
-    <xi:include href="xml/dbusmenu-server.xml"/>
     <xi:include href="xml/types.xml"/>
 
   </chapter>
--- a/docs/libdbusmenu-gtk/reference/libdbusmenu-gtk-docs.sgml
+++ b/docs/libdbusmenu-gtk/reference/libdbusmenu-gtk-docs.sgml
@@ -15,7 +15,6 @@
     <xi:include href="xml/menuitem.xml"/>
     <xi:include href="xml/client.xml"/>
     <xi:include href="xml/parser.xml"/>
-    <xi:include href="xml/serializablemenuitem.xml"/>
 
   </chapter>
   <chapter id="object-tree">
