--- THE-4.0/INSTALL.orig	2022-12-28 14:20:28.000000000 -0800
+++ THE-4.0/INSTALL	2023-01-07 16:32:48.849598481 -0800
@@ -32,7 +32,7 @@
 
 % mkdir xcurses
 % cd xcurses
-% ../configure --with-curses=x11
+% ../configure --with-curses=pdcurses-x11
                --with-rexx=oorexx
                --with-rexxincdir=[where oorexx is installed]/api
                --with-rexxlibdir=[where oorexx is installed]/api
