--- THE-4.0/Makefile.in.orig	2022-12-28 21:50:18.000000000 -0800
+++ THE-4.0/Makefile.in	2023-01-07 17:58:35.946838735 -0800
@@ -176,9 +176,9 @@
 doc/html:
 	mkdir -p doc/html
 
-html:	manext doc/html doc/html/index.html
+html:  doc/html doc/html/index.html
 
-doc/html/index.html: $(sourcedir)/*.c $(APPENDIX) $(GLOSSARY) $(OVERVIEW)
+doc/html/index.html: manext $(sourcedir)/*.c $(APPENDIX) $(GLOSSARY) $(OVERVIEW)
 #	-rm -fr *$(HTML_EXT) *.man
 	cp $(imagedir)/the64.png .
 	rexx $(srcdir)/makehtml.rex src=$(srcdir) ext=$(HTML_EXT) ver=$(VERSION) dest=./doc/html doc=$(srcdocdir) \
@@ -197,9 +197,9 @@
 doc/helpviewer:
 	mkdir -p doc/helpviewer
 
-helpviewer: manext doc/helpviewer doc/helpviewer/index.html
+helpviewer: doc/helpviewer doc/helpviewer/index.html
 
-doc/helpviewer/index.html: $(sourcedir)/*.c $(APPENDIX) $(GLOSSARY) $(OVERVIEW)
+doc/helpviewer/index.html: manext $(sourcedir)/*.c $(APPENDIX) $(GLOSSARY) $(OVERVIEW)
 #	-rm -f *$(HTML_EXT) *.man comm/* commsos/* commset/* misc/*
 	cp $(imagedir)/the64.png .
 	rexx $(srcdir)/makehv.rex src=$(srcdir) ext=$(HTML_EXT) ver=$(VERSION) dest=./doc/helpviewer doc=$(srcdocdir) \
