--- redirector.conf.dist-	2014-04-13 16:18:26.105058254 +0300
+++ redirector.conf.dist	2014-04-13 16:21:55.893896100 +0300
@@ -14,7 +14,7 @@
 # change_log filename   # Log file of change contest.
 #                       # Comment for turn off
 #
-# make-cache /usr/local/rejik3/make-cache # path to make-cache binary
+# make-cache /var/lib/rejik/make-cache # path to make-cache binary
 #
 # work_ip IP            # Ip adress or network
 # work_ip IP/MASK       # or file with ip for
@@ -70,27 +70,26 @@
 #
 #############################################################
 
-error_log /usr/local/rejik3/redirector.err
-change_log /usr/local/rejik3/redirector.log
-make-cache /usr/local/rejik3/make-cache
-
-#allow_urls /usr/local/rejik3/banlists/allow_urls
+error_log /var/log/squid/rejik/redirector.err
+change_log /var/log/squid/rejik/redirector.log
+make-cache /var/lib/rejik/make-cache
+#allow_urls /var/lib/rejik/banlists/allow_urls
 #write_hostname_to_log on
 
 <BANNER>  
-ban_dir /usr/local/rejik3/banlists/banners
+ban_dir /var/lib/rejik/banlists/banners
 url http://127.0.0.1/ban/1x1.gif
 #log off
 
 <PORNO>  
-ban_dir  /usr/local/rejik3/banlists/porno
+ban_dir  /var/lib/rejik/banlists/porno
 url http://127.0.0.1/ban/porno.html
 
 <MP3>  
-ban_dir /usr/local/rejik3/banlists/mp3
+ban_dir /var/lib/rejik/banlists/mp3
 url http://127.0.0.1/ban/mp3.html
 
 <JS>
-ban_dir /usr/local/rejik3/banlists/js
+ban_dir /var/lib/rejik/banlists/js
 url http://127.0.0.1/ban/js.js
 #log off
