diff -Naur boto-2.2.2.orig/setup.py boto-2.2.2/setup.py
--- boto-2.2.2.orig/setup.py	2012-02-15 03:17:52.000000000 +0000
+++ boto-2.2.2/setup.py	2012-05-07 02:23:28.885052767 +0000
@@ -59,11 +59,7 @@
                   "boto.mturk", "boto.mturk.test", "boto.pyami",
                   "boto.pyami.installers", "boto.pyami.installers.ubuntu",
                   "boto.mashups", "boto.contrib", "boto.manage",
-                  "tests", "tests.autoscale", "tests.cloudfront",
-                  "tests.devpay", "tests.db", "tests.dynamodb",
-                  "tests.ec2", "tests.ec2.cloudwatch", "tests.ec2.elb",
-                  "tests.s3", "tests.sdb", "tests.sqs", "tests.sts",
-                  "tests.utils", "boto.services", "boto.cloudfront",
+                  "boto.services", "boto.cloudfront",
                   "boto.roboto", "boto.rds", "boto.vpc", "boto.fps",
                   "boto.fps.test", "boto.emr", "boto.emr.tests", "boto.sns",
                   "boto.ecs", "boto.iam", "boto.route53", "boto.ses",
