diff -urp libjingle-0.3.12.OLD/talk/base/logging.cc libjingle-0.3.12/talk/base/logging.cc
--- libjingle-0.3.12.OLD/talk/base/logging.cc	2008-09-16 09:55:22.000000000 -0400
+++ libjingle-0.3.12/talk/base/logging.cc	2009-03-02 14:32:09.000000000 -0500
@@ -34,6 +34,7 @@
 #include <iostream>
 #include <iomanip>
 #include <string.h>
+#include <cstdio> 
 
 #include "talk/base/logging.h"
 #include "talk/base/stream.h"
diff -urp libjingle-0.3.12.OLD/talk/base/stream.cc libjingle-0.3.12/talk/base/stream.cc
--- libjingle-0.3.12.OLD/talk/base/stream.cc	2008-09-16 09:55:22.000000000 -0400
+++ libjingle-0.3.12/talk/base/stream.cc	2009-03-02 14:43:48.000000000 -0500
@@ -30,6 +30,7 @@
 #include <errno.h>
 #include <string>
 #include <string.h>
+#include <cstdio>
 #include "talk/base/basictypes.h"
 #include "talk/base/common.h"
 #include "talk/base/stream.h"
