diff -Naur libjingle-0.3.12.orig/talk/base/sigslot.h libjingle-0.3.12/talk/base/sigslot.h
--- libjingle-0.3.12.orig/talk/base/sigslot.h	2008-09-16 15:55:22.000000000 +0200
+++ libjingle-0.3.12/talk/base/sigslot.h	2012-09-12 11:26:35.849989149 +0200
@@ -80,6 +80,7 @@
 
 #include <set>
 #include <list>
+#include <unistd.h>
 
 // On our copy of sigslot.h, we force single threading
 #define SIGSLOT_PURE_ISO
