--- vsx_platform.h.orig	2019-03-19 10:38:34.582591102 +1000
+++ vsx_platform.h	2019-03-19 11:30:54.854922482 +1000
@@ -66,7 +66,7 @@
         #define PLATFORM_NAME                                   "GNU / Linux"
         #define PLATFORM_FAMILY                                 PLATFORM_FAMILY_UNIX
         #define PLATFORM_SHARED_FILES                           vsx_string<>(get_exec_path().c_str()) + "/../share/vsxu/"
-        #define VSXU_PLUGIN_LOCATION                           vsx_string<>(get_exec_path().c_str()) + "/../lib/vsxu/plugins"
+        #define VSXU_PLUGIN_LOCATION                           vsx_string<>(get_exec_path().c_str()) + "/../lib64/vsxu/plugins"
         #define PLATFORM_DLL_SUFFIX                             ".so"
         #ifdef PLATFORM_SHARED_FILES_STL
           #define PLATFORM_SHARED_FILES_STLSTRING               get_exec_path() + std::string("/../share/vsxu/")
