--- sphinx-2.0.5-release/src/sphinxstd.h	2012-06-19 10:53:02.000000000 -0300
+++ sphinx-2.0.5-release_patched/src/sphinxstd.h	2013-04-13 22:10:17.828838165 -0300
@@ -95,11 +95,11 @@
 
 #if USE_ODBC
 // UnixODBC compatible DWORD
-#if defined(__alpha) || defined(__sparcv9) || defined(__LP64__) || (defined(__HOS_AIX__) && defined(_LP64)) || defined(__APPLE__)
+//#if defined(__alpha) || defined(__sparcv9) || defined(__LP64__) || (defined(__HOS_AIX__) && defined(_LP64)) || defined(__APPLE__)
 typedef unsigned int		DWORD;
-#else
-typedef unsigned long		DWORD;
-#endif
+//#else
+//typedef unsigned long		DWORD;
+//#endif
 #else
 // default DWORD
 typedef unsigned int		DWORD;
