--- sphinx-2.2.10-release/src/sphinxstd.h.orig	2016-03-03 22:46:11.233963956 +0700
+++ sphinx-2.2.10-release/src/sphinxstd.h	2016-03-03 22:46:23.830802814 +0700
@@ -127,7 +127,7 @@
 #if defined(__alpha) || defined(__sparcv9) || defined(__LP64__) || (defined(__HOS_AIX__) && defined(_LP64)) || defined(__APPLE__)
 typedef unsigned int		DWORD;
 #else
-typedef unsigned long		DWORD;
+typedef unsigned int		DWORD;
 #endif
 #else
 // default DWORD
