--- src/gui/rect2Darray.h.orig
+++ src/gui/rect2Darray.h
@@ -305,7 +305,7 @@ void RectArray<T>::myAppend ( const BBox
 		minY = std::min( minY, item->top() );
 		maxY = std::max( maxY, item->bottom() );
 	}
-	append( item );
+	this->append( item );
 }
 
 template <typename T>
@@ -461,7 +461,7 @@ void Rect2DArray<T>::myAppend( BBoxOfObj
 	if (toAppend) {
 		current = new RectArray<T>();
 		current->myAppend( bbox );
-		append( current );
+		this->append( current );
 	}
 }
 
--- src/kernel/cobject2xpdf.cc	2009/09/11 12:02:56	1.22
+++ src/kernel/cobject2xpdf.cc	2011/05/21 22:13:21	1.24
@@ -727,6 +727,13 @@
 {
 	/*assert (!"operation not permitted...");*//*THIS IS FORBIDDEN IN THE CALLER*/
 }
+template void simpleValueFromXpdfObj<pBool, bool&> (const Object&, bool& val);
+template void simpleValueFromXpdfObj<pInt, int&> (const Object&,  int& val);
+template void simpleValueFromXpdfObj<pReal, double&> (const Object&, double& val);
+template void simpleValueFromXpdfObj<pString, string&> (const Object&, string& val);
+template void simpleValueFromXpdfObj<pName, string&> (const Object&, string& val);
+template void simpleValueFromXpdfObj<pRef, IndiRef&> (const Object&, IndiRef& val);
+
 
 
 //
