diff -Nur opera-11.50-1074.x86_64.linux.orig/install opera-11.50-1074.x86_64.linux/install
--- opera-11.50-1074.x86_64.linux.orig/install	2011-06-27 07:01:11.000000000 -0500
+++ opera-11.50-1074.x86_64.linux/install	2011-07-06 01:12:20.290465769 -0500
@@ -8,16 +8,16 @@
 	then
 		"$@" GX - bin/uninstall-opera@@{SUFFIX}
 	fi
-	"$@" NF 7f5617779c8a0c3ab3aed07274b85274 lib/opera@@{SUFFIX}/gstreamer/plugins/libgstoperamatroska.so
-	"$@" NF 2873abacfe8fa4f8dc210175dbf8b71c lib/opera@@{SUFFIX}/gstreamer/plugins/libgstoperavp8.so
-	"$@" NF f04788b0e63090d706ee55587580d8e0 lib/opera@@{SUFFIX}/liboperagtk.so
-	"$@" NF 8c898c575c72634806c5086a63a5960f lib/opera@@{SUFFIX}/liboperakde4.so
-	"$@" NX a83538e9fea0ef7f6e341ba75d357234 lib/opera@@{SUFFIX}/opera@@{SUFFIX}
-	"$@" NX 3122e86518173f1795e022cf22e99a6a lib/opera@@{SUFFIX}/operaplugincleaner
-	"$@" NX bc7331102ef434f19e5a275de153af66 lib/opera@@{SUFFIX}/operapluginwrapper
-	"$@" NX b76cbde97922472317d1eb8cc79086cf lib/opera@@{SUFFIX}/operapluginwrapper-ia32-linux
-	"$@" NX 23e36e618845efd96bc466261f9278fe lib/opera@@{SUFFIX}/operapluginwrapper-native
-	"$@" NF 0a8389b127c477aa2c55ee5d234473ca lib/opera@@{SUFFIX}/plugins/README
+	"$@" NF 7f5617779c8a0c3ab3aed07274b85274 lib64/opera@@{SUFFIX}/gstreamer/plugins/libgstoperamatroska.so
+	"$@" NF 2873abacfe8fa4f8dc210175dbf8b71c lib64/opera@@{SUFFIX}/gstreamer/plugins/libgstoperavp8.so
+	"$@" NF f04788b0e63090d706ee55587580d8e0 lib64/opera@@{SUFFIX}/liboperagtk.so
+	"$@" NF 8c898c575c72634806c5086a63a5960f lib64/opera@@{SUFFIX}/liboperakde4.so
+	"$@" NX a83538e9fea0ef7f6e341ba75d357234 lib64/opera@@{SUFFIX}/opera@@{SUFFIX}
+	"$@" NX 3122e86518173f1795e022cf22e99a6a lib64/opera@@{SUFFIX}/operaplugincleaner
+	"$@" NX bc7331102ef434f19e5a275de153af66 lib64/opera@@{SUFFIX}/operapluginwrapper
+	"$@" NX b76cbde97922472317d1eb8cc79086cf lib64/opera@@{SUFFIX}/operapluginwrapper-ia32-linux
+	"$@" NX 23e36e618845efd96bc466261f9278fe lib64/opera@@{SUFFIX}/operapluginwrapper-native
+	"$@" NF 0a8389b127c477aa2c55ee5d234473ca lib64/opera@@{SUFFIX}/plugins/README
 	"$@" PF 7391965010108b488c3bc526c8c7015c share/applications/opera@@{SUFFIX}-browser.desktop
 	"$@" PF e78f45e2dc8098046f2a19e268031058 share/applications/opera@@{SUFFIX}-widget-installer.desktop
 	"$@" PF 4d84c2febef83fd835ebf0d9f1f5d7df share/applications/opera@@{SUFFIX}-widget-manager.desktop
@@ -873,7 +873,7 @@
 #!/bin/sh
 export OPERA_DIR=${OPERA_DIR:-@@{PREFIX}/share/opera@@{SUFFIX}}
 export OPERA_PERSONALDIR=${OPERA_PERSONALDIR:-$HOME/.opera@@{SUFFIX}}
-exec @@{PREFIX}/lib/opera@@{SUFFIX}/opera@@{SUFFIX} "$@"
+exec @@{PREFIX}/lib64/opera@@{SUFFIX}/opera@@{SUFFIX} "$@"
 EOF_generate_file
 			;;
 		opera-widget-manager)
diff -Nur opera-11.50-1074.x86_64.linux.orig/opera opera-11.50-1074.x86_64.linux/opera
--- opera-11.50-1074.x86_64.linux.orig/opera	2011-06-27 07:01:11.000000000 -0500
+++ opera-11.50-1074.x86_64.linux/opera	2011-07-06 01:11:55.216735389 -0500
@@ -1,4 +1,4 @@
 #!/bin/sh
 export OPERA_DIR=${OPERA_DIR:-"${0%/*}"/share/opera}
 export OPERA_PERSONALDIR=${OPERA_PERSONALDIR:-"${0%/*}"/profile}
-exec "${0%/*}"/lib/opera/opera "$@"
+exec "${0%/*}"/lib64/opera/opera "$@"
