--- usr/share/opera/ini/font.ini.bak	2008-06-30 11:55:03.000000000 -0700
+++ usr/share/opera/ini/font.ini	2008-07-24 12:45:29.000000000 -0700
@@ -7,7 +7,7 @@
 
 ; One can disable xft or core fonts if necessary
 ;engine:xft=blacklist
-;engine:x11=blacklist
+engine:x11=blacklist
 
 ; Foundries that are generally known to have bad implementations
 ;foundry:urw|abiword|ultimo=bad
@@ -30,9 +30,9 @@
 foundry:bitstream=excellent
 
 ; Known fonts that solves specific problems
-family:mincho|gothic=japanese good try-first
-family:kochi*=japanese good try-first
-family:song|song ti|fangsong*=chinese-s try-first
+family:WenQuanYi*=japanese excellent try-first
+family:WenQuanYi*=chinese-s excellent try-first
+family:WenQuanYi*=chinese-t excellent try-first
 family:baekmuk*=korean good try-first
 
 ; Fonts that we know should not be used for displaying text
