diff -Naur /tmp/SBo/gtk-qt-engine.orig/src/qt_theme_draw.c /tmp/SBo/gtk-qt-engine/src/qt_theme_draw.c
--- /tmp/SBo/gtk-qt-engine.orig/src/qt_theme_draw.c	2008-03-11 01:30:19.000000000 +0000
+++ /tmp/SBo/gtk-qt-engine/src/qt_theme_draw.c	2009-07-11 14:33:46.000000000 +0000
@@ -19,6 +19,7 @@
  ***************************************************************************/
 
 #include <math.h>
+#include <stdlib.h>
 #include <string.h>
 #include <gtk/gtkprogressbar.h>
 #include <gdk/gdk.h>
