diff -ur mstch-1.0.2.old/src/CMakeLists.txt mstch-1.0.2/src/CMakeLists.txt
--- mstch-1.0.2.old/src/CMakeLists.txt	2016-02-01 18:08:46.000000000 -0200
+++ mstch-1.0.2/src/CMakeLists.txt	2017-01-09 12:52:26.414929590 -0200
@@ -28,8 +28,8 @@
 
 install(
     TARGETS mstch EXPORT mstchTargets
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
 install(
     FILES "${PROJECT_SOURCE_DIR}/include/mstch/mstch.hpp"
@@ -55,10 +55,10 @@
     EXPORT mstchTargets
     FILE mstch-targets.cmake
     NAMESPACE mstch::
-    DESTINATION lib/cmake/mstch)
+    DESTINATION lib${LIB_SUFFIX}/cmake/mstch)
 
 install(FILES
     "${PROJECT_SOURCE_DIR}/cmake/mstch-config.cmake"
     "${CMAKE_CURRENT_BINARY_DIR}/mstch/mstch-config-version.cmake"
-    DESTINATION lib/cmake/mstch
+    DESTINATION lib${LIB_SUFFIX}/cmake/mstch
     COMPONENT Devel)
