--- devede/install.sh	2011-05-05 19:38:01.000000000 -0300
+++ devede_patched/install.sh	2011-07-11 20:23:09.000000000 -0300
@@ -64,9 +64,6 @@
 	install -m 755 -d "${DESTDIR}${2}/pixmaps"
 	install -m 644 "${scriptdir}/devede.svg" "${DESTDIR}${2}/pixmaps/"
 
-	install -m 755 -d "${DESTDIR}${2}/pixmaps/backgrounds"
-	install -m 644 "${scriptdir}/devede.svg" "${DESTDIR}${2}/pixmaps/"
-
 	install -m 755 -d "${DESTDIR}${2}/applications"
 	install -m 644 "${scriptdir}/devede.desktop"			\
 						"${DESTDIR}${2}/applications/"
