--- devede-3.16.9/install.sh	2010-07-02 19:29:29.000000000 -0300
+++ devede-3.16.9_patched/install.sh	2010-09-13 22:57:48.000000000 -0300
@@ -64,9 +64,6 @@
 	install -m 755 -d "${DESTDIR}${2}/pixmaps"
 	install -m 644 "${scriptdir}/devede.svg" "${DESTDIR}${2}/pixmaps/"
 
-        install -m 755 -d "${DESTDIR}${2}/pixmaps/backgrounds"
-        install -m 644 "${scriptdir}/devede.svg" "${DESTDIR}${2}/pixmaps/"
-
 	install -m 755 -d "${DESTDIR}${2}/applications"
 	install -m 644 "${scriptdir}/devede.desktop"			\
 						"${DESTDIR}${2}/applications/"
