diff -Nur fdm-1.6.orig/GNUmakefile fdm-1.6/GNUmakefile
--- fdm-1.6.orig/GNUmakefile	2008-12-22 10:20:05.000000000 -0600
+++ fdm-1.6/GNUmakefile	2009-01-16 23:57:52.922462755 -0600
@@ -10,9 +10,6 @@
 
 PREFIX?= /usr/local
 
-BIN_OWNER= bin
-BIN_GROUP= root
-
 CC= gcc
 
 INCDIRS= -I$(PREFIX)/include
@@ -27,8 +24,8 @@
 endif
 
 INSTALLDIR= install -d
-INSTALLBIN= install -g $(BIN_OWNER) -o $(BIN_GROUP) -m 555
-INSTALLMAN= install -g $(BIN_OWNER) -o $(BIN_GROUP) -m 444
+INSTALLBIN= install -m 0755
+INSTALLMAN= install -m 0644
 
 SRCS= fdm.c \
       attach.c buffer.c cleanup.c command.c connect.c io.c log.c netrc.c \
diff -Nur fdm-1.6.orig/Makefile fdm-1.6/Makefile
--- fdm-1.6.orig/Makefile	2008-12-22 10:20:05.000000000 -0600
+++ fdm-1.6/Makefile	2009-01-16 23:57:12.893462354 -0600
@@ -94,8 +94,8 @@
 
 PREFIX?= /usr/local
 INSTALLDIR= install -d
-INSTALLBIN= install -g bin -o root -m 555
-INSTALLMAN= install -g bin -o root -m 444
+INSTALLBIN= install -m 0755
+INSTALLMAN= install -m 0644
 
 LDFLAGS+= -L/usr/local/lib
 .ifdef PROFILE
