Author: Marcin Kolny <marcin.kolny@gmail.com>
Description: Fix warnings
Origin: https://git.gnome.org/browse/gstreamermm/commit/gstreamer/gstreamermm/check.cc?id=c5006faf05bc4b46767d30f42bcb6157836b7540
Last-update: 2016-09-25
--- a/gstreamer/gstreamermm/check.cc
+++ b/gstreamer/gstreamermm/check.cc
@@ -54,7 +54,7 @@
 
     element = ElementFactory::create_element (factory, factory);
 
-    ThrowIf (element, "Could not create a '" + factory + "' element");
+    ThrowIf (!!element, "Could not create a '" + factory + "' element");
     CheckRefCount (element->gobj(), factory, 1);
 
     return element;
