diff -Nur supertux-0.3.3.orig/src/addon/addon_manager.cpp supertux-0.3.3/src/addon/addon_manager.cpp
--- supertux-0.3.3.orig/src/addon/addon_manager.cpp	2010-03-01 06:40:58.000000000 -0600
+++ supertux-0.3.3/src/addon/addon_manager.cpp	2012-09-08 23:24:07.209851421 -0500
@@ -29,7 +29,6 @@
 #ifdef HAVE_LIBCURL
 #  include <curl/curl.h>
 #  include <curl/easy.h>
-#  include <curl/types.h>
 #endif
 
 #include "addon/addon.hpp"
