diff -Nur AdobeReader.orig/bin/acroread AdobeReader/bin/acroread
--- AdobeReader.orig/bin/acroread	2007-01-05 13:56:39.000000000 -0600
+++ AdobeReader/bin/acroread	2007-03-26 13:10:02.000000000 -0500
@@ -415,7 +415,7 @@
            return 1 
         fi
 
-   		echo $mfile| sed 's/libgtk-x11-\([0-9]*\).0.so.0.\([0-9]\)00.\([0-9]*\)\|\(.*\)/\1\2\3/g'
+		echo $mfile| sed 's/libgtk-x11-\([0-9]*\).0.so.0.\([0-9]\)000.\([0-9]*\)\|\(.*\)/\1\2\3/g'
         return 0
     fi
 
