diff -Naur apitrace-11.1.orig/frametrim/ft_dependecyobject.hpp apitrace-11.1/frametrim/ft_dependecyobject.hpp
--- apitrace-11.1.orig/frametrim/ft_dependecyobject.hpp	2022-05-20 16:14:48.000000000 +0200
+++ apitrace-11.1/frametrim/ft_dependecyobject.hpp	2025-11-29 18:39:08.114043000 +0100
@@ -28,6 +28,7 @@
 #pragma once
 
 #include "ft_tracecall.hpp"
+#include <cstdint>
 
 namespace frametrim {
 
diff -Naur apitrace-11.1.orig/retrace/metric_writer.cpp apitrace-11.1/retrace/metric_writer.cpp
--- apitrace-11.1.orig/retrace/metric_writer.cpp	2022-05-20 16:14:48.000000000 +0200
+++ apitrace-11.1/retrace/metric_writer.cpp	2025-11-29 18:39:45.330043000 +0100
@@ -24,6 +24,7 @@
  **************************************************************************/
 
 #include <iostream>
+#include <cstdint>
 
 #include "metric_writer.hpp"
 
