--- seq24/trunk/src/perfedit.cpp.old	2016-01-03 23:30:55.562890265 +0700
+++ seq24/trunk/src/perfedit.cpp	2016-01-03 23:32:24.958812214 +0700
@@ -119,7 +119,7 @@
     /* snap */
     m_button_snap = manage( new Button());
     m_button_snap->add( *manage( new Image(Gdk::Pixbuf::create_from_xpm_data( snap_xpm ))));
-    m_button_snap->signal_clicked().connect(  bind<Menu *>( mem_fun( *this, &perfedit::popup_menu), m_menu_snap  ));
+    m_button_snap->signal_clicked().connect(  bind<void>( mem_fun( *this, &perfedit::popup_menu), m_menu_snap  ));
     add_tooltip( m_button_snap, "Grid snap. (Fraction of Measure Length)" );
     m_entry_snap = manage( new Entry());
     m_entry_snap->set_size_request( 40, -1 );
@@ -152,7 +152,7 @@
     /* beats per measure */
     m_button_bpm = manage( new Button());
     m_button_bpm->add( *manage( new Image(Gdk::Pixbuf::create_from_xpm_data( down_xpm  ))));
-    m_button_bpm->signal_clicked().connect(  bind<Menu *>( mem_fun( *this, &perfedit::popup_menu), m_menu_bpm  ));
+    m_button_bpm->signal_clicked().connect(  bind<void>( mem_fun( *this, &perfedit::popup_menu), m_menu_bpm  ));
     add_tooltip( m_button_bpm, "Time Signature. Beats per Measure" );
     m_entry_bpm = manage( new Entry());
     m_entry_bpm->set_width_chars(2);
@@ -162,7 +162,7 @@
     /* beat width */
     m_button_bw = manage( new Button());
     m_button_bw->add( *manage( new Image(Gdk::Pixbuf::create_from_xpm_data( down_xpm  ))));
-    m_button_bw->signal_clicked().connect(  bind<Menu *>( mem_fun( *this, &perfedit::popup_menu), m_menu_bw  ));
+    m_button_bw->signal_clicked().connect(  bind<void>( mem_fun( *this, &perfedit::popup_menu), m_menu_bw  ));
     add_tooltip( m_button_bw, "Time Signature.  Length of Beat" );
     m_entry_bw = manage( new Entry());
     m_entry_bw->set_width_chars(2);
