diff -Naur opencomal-0.2.6/src/pdccmd.c opencomal-0.2.6.patched/src/pdccmd.c
--- opencomal-0.2.6/src/pdccmd.c	2002-09-22 13:36:49.000000000 -0400
+++ opencomal-0.2.6.patched/src/pdccmd.c	2017-12-10 16:51:45.087203071 -0500
@@ -22,6 +22,7 @@
 #include "pdcsqash.h"
 #include "pdcenv.h"
 
+#include <string.h>
 
 PRIVATE void cmd_list_horse(struct string *filename, long from, long to)
 {
diff -Naur opencomal-0.2.6/src/pdcenv.c opencomal-0.2.6.patched/src/pdcenv.c
--- opencomal-0.2.6/src/pdcenv.c	2002-09-17 12:20:06.000000000 -0400
+++ opencomal-0.2.6.patched/src/pdcenv.c	2017-12-10 16:52:01.592203099 -0500
@@ -16,6 +16,8 @@
 #include "pdcseg.h"
 #include "pdcid.h"
 
+#include <string.h>
+
 PUBLIC struct comal_env *env_new(char *name)
 {
 	struct comal_env *work = GETCORE(MISC_POOL, struct comal_env);
diff -Naur opencomal-0.2.6/src/pdcexp.c opencomal-0.2.6.patched/src/pdcexp.c
--- opencomal-0.2.6/src/pdcexp.c	2002-12-06 06:02:35.000000000 -0500
+++ opencomal-0.2.6.patched/src/pdcexp.c	2017-12-10 16:52:33.823203154 -0500
@@ -20,7 +20,7 @@
 #include "pdcstr.h"
 #include "pdcval.h"
 
-
+#include <string.h>
 #include <math.h>
 
 #ifdef HAS_ROUND
diff -Naur opencomal-0.2.6/src/pdcext.c opencomal-0.2.6.patched/src/pdcext.c
--- opencomal-0.2.6/src/pdcext.c	2002-12-05 02:17:02.000000000 -0500
+++ opencomal-0.2.6.patched/src/pdcext.c	2017-12-10 16:53:06.231203208 -0500
@@ -18,6 +18,8 @@
 #include "pdcval.h"
 #include "version.h"
 
+#include <string.h>
+
 struct inpfile_stkent {
 	struct inpfile_stkent *next;
 	FILE *inpfile;
diff -Naur opencomal-0.2.6/src/pdclist.c opencomal-0.2.6.patched/src/pdclist.c
--- opencomal-0.2.6/src/pdclist.c	2002-11-30 09:39:49.000000000 -0500
+++ opencomal-0.2.6.patched/src/pdclist.c	2017-12-10 16:52:20.328203131 -0500
@@ -15,6 +15,8 @@
 #include "pdcparss.h"
 #include "pdcmisc.h"
 
+#include <string.h>
+
 PUBLIC int show_exec = 0;
 
 PRIVATE void list_horse();
diff -Naur opencomal-0.2.6/src/pdcmisc.c opencomal-0.2.6.patched/src/pdcmisc.c
--- opencomal-0.2.6/src/pdcmisc.c	2002-12-06 05:58:13.000000000 -0500
+++ opencomal-0.2.6.patched/src/pdcmisc.c	2017-12-10 16:51:25.608203038 -0500
@@ -18,6 +18,7 @@
 #include "pdcexec.h"
 #include "pdclist.h"
 
+#include <string.h>
 #include <math.h>
 #include <fcntl.h>
 
diff -Naur opencomal-0.2.6/src/pdcsqash.c opencomal-0.2.6.patched/src/pdcsqash.c
--- opencomal-0.2.6/src/pdcsqash.c	2002-11-30 09:40:16.000000000 -0500
+++ opencomal-0.2.6.patched/src/pdcsqash.c	2017-12-10 16:52:52.257203185 -0500
@@ -17,6 +17,7 @@
 #include "pdcexec.h"
 #include <fcntl.h>
 #include <sys/stat.h>
+#include <string.h>
 
 PRIVATE void sqash_exp();
 PRIVATE void sqash_horse();
