Description: Fix FTBFS due to unknown std::cout
Author: Martin Pitt <martin.pitt@ubuntu.com>
Forwarded: No, file is gone in latest upstream

Index: libbitcoin-2.0/src/utility/mnemonic.cpp
===================================================================
--- libbitcoin-2.0.orig/src/utility/mnemonic.cpp
+++ libbitcoin-2.0/src/utility/mnemonic.cpp
@@ -21,6 +21,7 @@
 
 #include <algorithm>
 #include <sstream>
+#include <iostream>
 #include <bitcoin/utility/assert.hpp>
 
 namespace libbitcoin {
