diff -Naur rhythmbox-2.97.orig/configure rhythmbox-2.97/configure
--- rhythmbox-2.97.orig/configure	2012-06-03 11:07:09.000000000 +0200
+++ rhythmbox-2.97/configure	2012-09-24 19:13:42.643306600 +0200
@@ -15592,7 +15592,6 @@
 		  libpeas-1.0 >= \$LIBPEAS_REQS		\\
 		  libpeas-gtk-1.0 >= \$LIBPEAS_REQS	\\
 		  libxml-2.0 >= \$LIBXML2_REQS		\\
-		  tdb >= 1.2.6				\\
 		  json-glib-1.0\""; } >&5
   ($PKG_CONFIG --exists --print-errors "\
 		  gobject-introspection-1.0 >= $GOBJECT_INTROSPECTION_REQS \
@@ -15606,7 +15605,6 @@
 		  libpeas-1.0 >= $LIBPEAS_REQS		\
 		  libpeas-gtk-1.0 >= $LIBPEAS_REQS	\
 		  libxml-2.0 >= $LIBXML2_REQS		\
-		  tdb >= 1.2.6				\
 		  json-glib-1.0") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
@@ -15623,7 +15621,6 @@
 		  libpeas-1.0 >= $LIBPEAS_REQS		\
 		  libpeas-gtk-1.0 >= $LIBPEAS_REQS	\
 		  libxml-2.0 >= $LIBXML2_REQS		\
-		  tdb >= 1.2.6				\
 		  json-glib-1.0" 2>/dev/null`
 else
   pkg_failed=yes
@@ -15647,7 +15644,6 @@
 		  libpeas-1.0 >= \$LIBPEAS_REQS		\\
 		  libpeas-gtk-1.0 >= \$LIBPEAS_REQS	\\
 		  libxml-2.0 >= \$LIBXML2_REQS		\\
-		  tdb >= 1.2.6				\\
 		  json-glib-1.0\""; } >&5
   ($PKG_CONFIG --exists --print-errors "\
 		  gobject-introspection-1.0 >= $GOBJECT_INTROSPECTION_REQS \
@@ -15661,7 +15657,6 @@
 		  libpeas-1.0 >= $LIBPEAS_REQS		\
 		  libpeas-gtk-1.0 >= $LIBPEAS_REQS	\
 		  libxml-2.0 >= $LIBXML2_REQS		\
-		  tdb >= 1.2.6				\
 		  json-glib-1.0") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
@@ -15678,7 +15673,6 @@
 		  libpeas-1.0 >= $LIBPEAS_REQS		\
 		  libpeas-gtk-1.0 >= $LIBPEAS_REQS	\
 		  libxml-2.0 >= $LIBXML2_REQS		\
-		  tdb >= 1.2.6				\
 		  json-glib-1.0" 2>/dev/null`
 else
   pkg_failed=yes
@@ -15711,7 +15705,6 @@
 		  libpeas-1.0 >= $LIBPEAS_REQS		\
 		  libpeas-gtk-1.0 >= $LIBPEAS_REQS	\
 		  libxml-2.0 >= $LIBXML2_REQS		\
-		  tdb >= 1.2.6				\
 		  json-glib-1.0" 2>&1`
         else
 	        RHYTHMBOX_PKG_ERRORS=`$PKG_CONFIG --print-errors "\
@@ -15726,7 +15719,6 @@
 		  libpeas-1.0 >= $LIBPEAS_REQS		\
 		  libpeas-gtk-1.0 >= $LIBPEAS_REQS	\
 		  libxml-2.0 >= $LIBXML2_REQS		\
-		  tdb >= 1.2.6				\
 		  json-glib-1.0" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
@@ -15744,7 +15736,6 @@
 		  libpeas-1.0 >= $LIBPEAS_REQS		\
 		  libpeas-gtk-1.0 >= $LIBPEAS_REQS	\
 		  libxml-2.0 >= $LIBXML2_REQS		\
-		  tdb >= 1.2.6				\
 		  json-glib-1.0) were not met:
 
 $RHYTHMBOX_PKG_ERRORS
