diff -Naur mathics-0.8/setup.py mathics-0.8.patched/setup.py
--- mathics-0.8/setup.py	2015-05-30 20:15:49.000000000 -0700
+++ mathics-0.8.patched/setup.py	2015-06-27 18:41:36.237193840 -0700
@@ -61,8 +61,8 @@
     INSTALL_REQUIRES = ['cython>=0.15.1']
 
 # General Requirements
-INSTALL_REQUIRES += ['sympy==0.7.6', 'django >= 1.6, < 1.7', 'ply>=3.6',
-                     'mpmath>=0.19', 'argparse', 'python-dateutil', 'colorama',
+INSTALL_REQUIRES += ['sympy==0.7.6', 'django >= 1.6', 'ply>=3.6',
+                     'mpmath>=0.19', 'python-dateutil', 'colorama',
                      'interruptingcow']
 
 # if sys.platform == "darwin":
