diff -Naur ecasound-2.8.1/configure ecasound-2.8.1.patched//configure
--- ecasound-2.8.1/configure	2011-05-22 07:25:44.000000000 -0400
+++ ecasound-2.8.1.patched//configure	2011-06-22 02:22:38.000000000 -0400
@@ -6691,19 +6691,7 @@
 fi
 
 
-
-ECA_S_PYTHON_MODULES=NO
-for i in $pymoddirs;
-do
-  for j in lib-dynload;
-  do
-    if test -r "$i/$j"; then
-      ECA_S_PYTHON_MODULES=$i
-      break 2
-    fi
-  done
-done
-
+ECA_S_PYTHON_MODULES="$pymoddirs"
 if test -d $ECA_S_PYTHON_MODULES/site-packages -o "X$python_force_site_packages" = "Xyes"
 then
     ECA_S_PYTHON_MODULES=$ECA_S_PYTHON_MODULES/site-packages
