--- libdvdnav-4.1.1-orig/src/dvdread/ifo_print.h	2004-01-11 22:43:13.000000000 +0100
+++ libdvdnav-4.1.1/src/dvdread/ifo_print.h	2008-04-01 10:31:18.000000000 +0200
@@ -24,7 +24,7 @@
 #ifdef DVDNAV_COMPILE
 #  include "ifo_types.h"
 #else
-#  include <dvdnav/ifo_types.h> /*  Only for vm_cmd_t  */
+#  include <dvdread/ifo_types.h> /*  Only for vm_cmd_t  */
 #endif
 
 void ifo_print(dvd_reader_t *dvd, int title);
