--- plex/Third-Party/aes/aesopt.h.org	2015-02-03 21:32:46.000000000 +0100
+++ plex/Third-Party/aes/aesopt.h	2015-02-09 11:00:02.966550150 +0100
@@ -163,11 +163,7 @@
 
 /*  2. Intel AES AND VIA ACE SUPPORT */
 
-#if defined( __GNUC__ ) && defined( __i386__ ) \
-	|| defined(_WIN32) && defined(_M_IX86) \
-	&& !(defined(_WIN64) || defined(_WIN32_WCE) || defined(_MSC_VER) && (_MSC_VER <= 800))
-#  define VIA_ACE_POSSIBLE
-#endif
+#  undef VIA_ACE_POSSIBLE
 
 /*  Define this option if support for the Intel AESNI is required (not
     currently available with GCC). If AESNI is known to be present, then 
