diff -Nur xxkb.orig/Imakefile xxkb/Imakefile
--- xxkb.orig/Imakefile	2007-03-14 16:54:53.000000000 -0500
+++ xxkb/Imakefile	2009-05-17 15:43:25.028469025 -0500
@@ -3,7 +3,7 @@
 BASEDIR = $(PROJECTROOT)
 BINDIR = $(BASEDIR)/bin
 PIXMAPDIR = $(BASEDIR)/share/xxkb
-LOCAL_LIBRARIES = $(XONLYLIB) -lXpm -lXt
+LOCAL_LIBRARIES = $(XONLYLIB) -lXpm -lXt -lXext
 
 OBJS = xxkb.o resource.o wlist.o
 PIXMAPS = pixmaps/by15.xpm pixmaps/by48.xpm \
