diff -Naur pipe-viewer-0.1.8/bin/gtk-pipe-viewer pipe-viewer-0.1.8.patched/bin/gtk-pipe-viewer
--- pipe-viewer-0.1.8/bin/gtk-pipe-viewer	2022-01-21 21:34:05.000000000 -0500
+++ pipe-viewer-0.1.8.patched/bin/gtk-pipe-viewer	2022-01-30 11:34:17.005950305 -0500
@@ -187,6 +187,13 @@
                               fs    => q{--fullscreen},
                               arg   => q{--really-quiet --force-media-title=*TITLE* --no-ytdl *VIDEO*},
                              },
+                      mplayer => {
+                              cmd   => q{mplayer},
+                              srt   => q{-sub *SUB*},
+                              audio => q{-audiofile *AUDIO*},
+                              fs    => q{-fs},
+                              arg   => q{-prefer-ipv4 -really-quiet -title *TITLE* *VIDEO*},
+                             },
                      },
     video_player_selected => undef,    # autodetect it later
 
diff -Naur pipe-viewer-0.1.8/bin/pipe-viewer pipe-viewer-0.1.8.patched/bin/pipe-viewer
--- pipe-viewer-0.1.8/bin/pipe-viewer	2022-01-21 21:34:05.000000000 -0500
+++ pipe-viewer-0.1.8.patched/bin/pipe-viewer	2022-01-30 11:35:25.549944074 -0500
@@ -177,6 +177,14 @@
                               arg     => q{--really-quiet --force-media-title=*TITLE* --no-ytdl *VIDEO*},
                               novideo => q{--no-video},
                              },
+                      mplayer => {
+                              cmd     => q{mplayer},
+                              srt     => q{-sub *SUB*},
+                              audio   => q{-audiofile *AUDIO*},
+                              fs      => q{-fs},
+                              arg     => q{-prefer-ipv4 -really-quiet -title *TITLE* *VIDEO*},
+                              novideo => q{-novideo},
+                             },
                      },
 
     video_player_selected => (
