diff -Naur pipe-viewer-0.5.7/bin/gtk-pipe-viewer pipe-viewer-0.5.7.patched/bin/gtk-pipe-viewer
--- pipe-viewer-0.5.7/bin/gtk-pipe-viewer	2026-03-13 18:34:21.000000000 -0400
+++ pipe-viewer-0.5.7.patched/bin/gtk-pipe-viewer	2026-05-11 19:49:52.218893517 -0400
@@ -196,6 +196,13 @@
                               fs    => q{--fullscreen},
                               arg   => q{--really-quiet --force-media-title=*TITLE* --start=*START* --no-ytdl --no-terminal *VIDEO*},
                              },
+                      mpplayer => {
+                              cmd   => q{mpplayer},
+                              srt   => q{-sub *SUB*},
+                              audio => q{-audiofile *AUDIO*},
+                              fs    => q{-fs},
+                              arg   => q{-prefer-ipv4 -really-quiet -title *TITLE* *VIDEO*},
+                             },
                      },
     video_player_selected => undef,    # autodetect it later
 
diff -Naur pipe-viewer-0.5.7/bin/pipe-viewer pipe-viewer-0.5.7.patched/bin/pipe-viewer
--- pipe-viewer-0.5.7/bin/pipe-viewer	2026-03-13 18:34:21.000000000 -0400
+++ pipe-viewer-0.5.7.patched/bin/pipe-viewer	2026-05-11 19:51:16.362893196 -0400
@@ -205,6 +205,15 @@
                 arg     => q{--really-quiet --force-media-title=*TITLE* --start=*START* --no-ytdl *VIDEO*},
                 novideo => q{--no-video},
                },
+
+        mplayer => {
+                cmd     => q{mplayer},
+                srt     => q{-sub *SUB*},
+                audio   => q{-audiofile *AUDIO*},
+                fs      => q{-fs},
+                arg     => q{-prefer-ipv4 -really-quiet -title *TITLE* *VIDEO*},
+                novideo => q{-novideo},
+               },
     },
 
     video_player_selected => (
