diff -Naur pipe-viewer-0.5.2/bin/gtk-pipe-viewer pipe-viewer-0.5.2.patched/bin/gtk-pipe-viewer
--- pipe-viewer-0.5.2/bin/gtk-pipe-viewer	2024-08-11 13:11:46.000000000 -0400
+++ pipe-viewer-0.5.2.patched/bin/gtk-pipe-viewer	2024-08-14 02:15:31.745767652 -0400
@@ -202,6 +202,13 @@
                                  srt => q{--sub-file=*SUB*},
                                 },
                      },
+                      mplayer => {
+                              cmd   => q{mplayer},
+                              srt   => q{-sub *SUB*},
+                              audio => q{-audiofile *AUDIO*},
+                              fs    => q{-fs},
+                              arg   => q{-prefer-ipv4 -really-quiet -title *TITLE* *VIDEO*},
+                             },
     video_player_selected => undef,    # autodetect it later
 
     # GUI options
diff -Naur pipe-viewer-0.5.2/bin/pipe-viewer pipe-viewer-0.5.2.patched/bin/pipe-viewer
--- pipe-viewer-0.5.2/bin/pipe-viewer	2024-08-11 13:11:46.000000000 -0400
+++ pipe-viewer-0.5.2.patched/bin/pipe-viewer	2024-08-14 02:16:07.201764316 -0400
@@ -213,6 +213,16 @@
                    novideo => q{--no-video},
                    srt     => q{--sub-file=*SUB*},
                   },
+
+        mplayer => {
+                cmd     => q{mplayer},
+                srt     => q{-sub *SUB*},
+                audio   => q{-audiofile *AUDIO*},
+                fs      => q{-fs},
+                arg     => q{-prefer-ipv4 -really-quiet -title *TITLE* *VIDEO*},
+                novideo => q{-novideo},
+               },
+
     },
 
     video_player_selected => (
