--- src/acpustatus.cc.orig	2010-10-31 17:09:36.000000000 +0300
+++ src/acpustatus.cc	2013-08-26 00:38:16.868170497 +0400
@@ -323,7 +323,7 @@
     int fd;
     float cpufreq = 0;
 
-    sprintf(namebuf, "/sys/devices/system/cpu/cpu%d/cpufreq/scaling_cur_freq", cpu);
+    sprintf(namebuf, "/sys/devices/system/cpu/cpu%d/cpufreq/cpuinfo_cur_freq", cpu);
     fd = open(namebuf, O_RDONLY);
     if (fd != -1) {
         int len = read(fd, buf, sizeof(buf) - 1);
