diff -Nur icewm-1.3.7.orig/src/Makefile.in icewm-1.3.7/src/Makefile.in
--- icewm-1.3.7.orig/src/Makefile.in	2010-10-31 09:09:36.000000000 -0500
+++ icewm-1.3.7/src/Makefile.in	2012-05-11 10:29:55.356344991 -0500
@@ -37,7 +37,7 @@
 CXXFLAGS =      @CXXFLAGS@ $(DEBUG) $(DEFS) `pkg-config gdk-pixbuf-xlib-2.0 --cflags` \
 	        @CORE_CFLAGS@ @IMAGE_CFLAGS@ @AUDIO_CFLAGS@ # `fc-config --cflags`
 LFLAGS =	@LDFLAGS@ 
-LIBS =          @LIBS@ `pkg-config gdk-pixbuf-xlib-2.0 --libs` 
+LIBS =          @LIBS@ `pkg-config gdk-pixbuf-xlib-2.0 fontconfig --libs` 
 
 CORE_LIBS =     @CORE_LIBS@ # `fc-config --libs`
 IMAGE_LIBS =    @IMAGE_LIBS@
