diff -Nur icewm-1.3.12.orig/lib/keys.in icewm-1.3.12/lib/keys.in
--- icewm-1.3.12.orig/lib/keys.in	2015-09-13 16:29:28.000000000 -0500
+++ icewm-1.3.12/lib/keys.in	2016-01-03 16:28:55.332001195 -0600
@@ -20,8 +20,8 @@
 # advanced function keys by watching the output of the xev command whilest
 # pressing those keys and map those symbols by using xmodmap.
 
-key "XF86Standby"			/bin/sh -c "{ test -e /run/systemd/system && systemctl suspend; } ||:"
-key "XF86Sleep"				/bin/sh -c "{ test -e /run/systemd/system && systemctl suspend; } ||:"
+key "XF86Standby"			/bin/sh -c "{ pm-suspend; } ||:"
+key "XF86Sleep"				/bin/sh -c "{ pm-suspend; } ||:"
 key "XF86AudioLowerVolume"		amixer sset PCM 5%-
 key "XF86AudioRaiseVolume"		amixer sset PCM 5%+
 key "XF86AudioMute"			amixer sset PCM 0%
