diff -Nur icewm-1.4.2.orig/lib/keys.in icewm-1.4.2/lib/keys.in
--- icewm-1.4.2.orig/lib/keys.in	2017-07-30 03:53:35.524267864 -0500
+++ icewm-1.4.2/lib/keys.in	2019-07-21 00:54:04.355694881 -0500
@@ -20,8 +20,8 @@
 # advanced function keys by watching the output of the xev command whilest
 # pressing those keys and map those symbols by using xmodmap.
 
-key "XF86Standby"			/bin/sh -c "{ test -e /run/systemd/system && systemctl suspend; } ||:"
-key "XF86Sleep"				/bin/sh -c "{ test -e /run/systemd/system && systemctl suspend; } ||:"
+key "XF86Standby"			/bin/sh -c "{ /usr/bin/icewm-sleep; } ||:"
+key "XF86Sleep"				/bin/sh -c "{ /usr/bin/icewm-sleep; } ||:"
 key "XF86AudioLowerVolume"		amixer sset PCM 5%-
 key "XF86AudioRaiseVolume"		amixer sset PCM 5%+
 key "XF86AudioMute"			amixer sset PCM 0%
