diff -Nur icewm-1.5.5.orig/lib/keys.in icewm-1.5.5/lib/keys.in
--- icewm-1.5.5.orig/lib/keys.in	2019-06-06 17:42:00.019998596 -0500
+++ icewm-1.5.5/lib/keys.in	2019-07-27 17:42:43.139090310 -0500
@@ -13,8 +13,8 @@
 key "Alt+Ctrl+b"			xdg-open about:blank
 key "Alt+Ctrl+s"			xdg-open https://www.google.com
 
-key "Super+KP_Subtract"			amixer sset Master 5%-
-key "Super+KP_Add"			amixer sset Master 5%+
+key "Super+KP_Subtract"			pamixer -d 5
+key "Super+KP_Add"			pamixer -i 5
 
 # "Multimedia key" bindings for Xorg. Gather the keycodes of your
 # advanced function keys by watching the output of the xev command whilest
@@ -23,11 +23,11 @@
 # Note: some of them might have unwanted side effects through concurrency with
 # other listeners like systemd for the suspend key events
 #
-# key "XF86Standby"			/bin/sh -c "{ test -e /run/systemd/system && systemctl suspend; } ||:"
-# key "XF86Sleep"				/bin/sh -c "{ test -e /run/systemd/system && systemctl suspend; } ||:"
-key "XF86AudioLowerVolume"		amixer sset Master 5%-
-key "XF86AudioRaiseVolume"		amixer sset Master 5%+
-key "XF86AudioMute"			amixer sset Master toggle
+# key "XF86Standby"			/bin/sh -c "{ sudo /usr/sbin/pm-suspend; } ||:"
+# key "XF86Sleep"				/bin/sh -c "{ sudo /usr/sbin/pm-suspend; } ||:"
+key "XF86AudioLowerVolume"		pamixer -d 5
+key "XF86AudioRaiseVolume"		pamixer -i 5
+key "XF86AudioMute"			pamixer -t
 key "XF86HomePage"			xdg-open about:blank
 key "XF86Search"			xdg-open https://www.google.com
 key "XF86Eject"				eject
diff -Nur icewm-1.5.5.orig/src/default.h icewm-1.5.5/src/default.h
--- icewm-1.5.5.orig/src/default.h	2019-06-06 17:42:00.043999061 -0500
+++ icewm-1.5.5/src/default.h	2019-07-27 17:36:10.134553913 -0500
@@ -184,9 +184,9 @@
 XSV(const char *, logoutCancelCommand,          0)
 #if defined(__linux__)
 // use shell code since those are wrapped through shell in YWindowManager::execAfterFork
-XSV(const char *, shutdownCommand,              "test -e /run/systemd/system && systemctl poweroff")
-XSV(const char *, rebootCommand,                "test -e /run/systemd/system && systemctl reboot")
-XSV(const char *, suspendCommand,               "test -e /run/systemd/system && systemctl suspend")
+XSV(const char *, shutdownCommand,              "sudo /sbin/halt")
+XSV(const char *, rebootCommand,                "sudo /sbin/reboot")
+XSV(const char *, suspendCommand,               "sudo /usr/sbin/pm-suspend")
 #else
 XSV(const char *, shutdownCommand,              0)
 XSV(const char *, rebootCommand,                0)
