diff -Nur midisport-firmware-1.2.orig//configure midisport-firmware-1.2/configure
--- midisport-firmware-1.2.orig//configure	2006-03-05 15:01:49.000000000 -0600
+++ midisport-firmware-1.2/configure	2010-09-12 01:47:17.753971701 -0500
@@ -1737,7 +1737,7 @@
 
 echo "$as_me:$LINENO: checking for udev version" >&5
 echo $ECHO_N "checking for udev version... $ECHO_C" >&6
-udev_ver=`udevinfo -V | awk '{ print $3 }'`
+udev_ver=`udevadm -V`
 if test -n "$udev_ver"; then
   echo "$as_me:$LINENO: result: $udev_ver" >&5
 echo "${ECHO_T}$udev_ver" >&6
