--- tuxguitar	2009-04-04 05:45:46.863811994 -0500
+++ tuxguitar.new	2009-04-04 05:44:42.817812900 -0500
@@ -23,7 +23,7 @@
 LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/usr/local/lib
 LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${MOZILLA_FIVE_HOME}
 ##CLASSPATH
-CLASSPATH=$CLASSPATH:./tuxguitar.jar:lib/swt.jar:lib/itext.jar:share/
+CLASSPATH=$CLASSPATH:/usr/share/tuxguitar/tuxguitar.jar:/usr/share/tuxguitar/swt.jar:/usr/share/tuxguitar/itext.jar:/usr/share/tuxguitar/
 ##MAINCLASS
 MAINCLASS=org.herac.tuxguitar.gui.TGMain
 ##JVM ARGUMENTS
@@ -33,4 +33,4 @@
 export LD_LIBRARY_PATH
 export MOZILLA_FIVE_HOME
 ##LAUNCH
-${JAVA} ${VM_ARGS} -cp :${CLASSPATH} -Dtuxguitar.share.path="share/" -Djava.library.path="${LD_LIBRARY_PATH}" ${MAINCLASS} "$1" "$2"
+${JAVA} ${VM_ARGS} -cp :${CLASSPATH} -Dtuxguitar.share.path="/usr/share/tuxguitar/" -Djava.library.path="${LD_LIBRARY_PATH}" ${MAINCLASS} "$1" "$2"
