Index: ario-1.5.1/src/ario-profiles.h
===================================================================
--- ario-1.5.1.orig/src/ario-profiles.h
+++ ario-1.5.1/src/ario-profiles.h
@@ -20,7 +20,7 @@
 #ifndef __ARIO_PROFILES_H
 #define __ARIO_PROFILES_H
 
-#include <glib/gslist.h>
+#include <glib.h>
 #include "servers/ario-server.h"
 
 G_BEGIN_DECLS
Index: ario-1.5.1/src/ario-util.h
===================================================================
--- ario-1.5.1.orig/src/ario-util.h
+++ ario-1.5.1/src/ario-util.h
@@ -18,7 +18,7 @@
  */
 
 #include "servers/ario-server.h"
-#include "glib/gslist.h"
+#include <glib.h>
 #include "gdk/gdkpixbuf.h"
 
 /* Number of covers used to generate the drag & drop image */
Index: ario-1.5.1/src/plugins/ario-plugin-info.c
===================================================================
--- ario-1.5.1.orig/src/plugins/ario-plugin-info.c
+++ ario-1.5.1/src/plugins/ario-plugin-info.c
@@ -27,7 +27,7 @@
 
 #include <string.h>
 #include <glib/gi18n.h>
-#include <glib/gkeyfile.h>
+#include <glib.h>
 
 #include "plugins/ario-plugin-info-priv.h"
 #include "ario-debug.h"
