diff -rupN vbam-r1159.orig/src/gtk/window.cpp vbam-r1159/src/gtk/window.cpp
--- vbam-r1159.orig/src/gtk/window.cpp	2013-02-03 12:59:48.000000000 +0000
+++ vbam-r1159/src/gtk/window.cpp	2013-09-11 15:21:43.646331404 +0100
@@ -248,6 +248,7 @@ Window::Window(GtkWindow * _pstWindow, c
   m_poRecentChooserMenu->set_show_numbers();
   m_poRecentChooserMenu->set_show_tips();
   m_poRecentChooserMenu->set_local_only();
+  m_poRecentChooserMenu->set_show_icons(false);
   m_poRecentChooserMenu->add_filter(oRecentFilter);
   m_poRecentChooserMenu->signal_item_activated().connect(
                                                    sigc::mem_fun(*this, &Window::vOnRecentFile));
