--- postfix-3.0.3/postfix-install.orig	2015-01-29 23:15:30.000000000 +0100
+++ postfix-3.0.3/postfix-install	2015-10-17 03:57:42.269896095 +0200
@@ -248,7 +248,7 @@
     case "$junk" in
     *MAIL_VERSION*) 
 	case "$mail_version" in
-	"") mail_version="`bin/postconf -dhx mail_version`" || exit 1
+	"") mail_version="`bin/postconf -c conf -dhx mail_version`" || exit 1
 	esac
 	val=`echo "$junk" | sed 's/MAIL_VERSION$/'"$mail_version/g"` || exit 1
 	case "$val" in
@@ -810,7 +810,7 @@
 # the wrong place when Postfix is being upgraded.
 
 case "$mail_version" in
-"") mail_version="`bin/postconf -dhx mail_version`" || exit 1
+"") mail_version="`bin/postconf -c conf -dhx mail_version`" || exit 1
 esac
 
 # Undo MAIL_VERSION expansion at the end of a parameter value. If
--- postfix-3.0.3/Makefile.in.orig	2014-09-25 21:16:29.000000000 +0200
+++ postfix-3.0.3/Makefile.in	2015-10-17 04:01:27.170920308 +0200
@@ -20,7 +20,7 @@
 EXPAND	= sed -e "s;\$${LIB_PREFIX};$(LIB_PREFIX);" \
 	    -e "s;\$${LIB_SUFFIX};$(LIB_SUFFIX);"
 SHLIB_DIR_OVERRIDE = \
-	$${shlib_directory:-`$(SHLIB_ENV) bin/postconf -dhx shlib_directory`}
+	$${shlib_directory:-`$(SHLIB_ENV) bin/postconf -c conf -dhx shlib_directory`}
 
 default: update
 
