diff -Nur newsbeuter-2.5.a/src/ttrss_api.cpp newsbeuter-2.5.b/src/ttrss_api.cpp
--- newsbeuter-2.5.a/src/ttrss_api.cpp	2012-01-06 06:41:13.000000000 -0800
+++ newsbeuter-2.5.b/src/ttrss_api.cpp	2012-12-24 20:08:35.505897356 -0800
@@ -220,7 +220,7 @@
 		const char * link = json_object_get_string(json_object_object_get(item_obj, "link"));
 		const char * content = json_object_get_string(json_object_object_get(item_obj, "content"));
 		time_t updated = (time_t)json_object_get_int(json_object_object_get(item_obj, "updated"));
-		boolean unread = json_object_get_boolean(json_object_object_get(item_obj, "unread"));
+		bool unread = json_object_get_boolean(json_object_object_get(item_obj, "unread"));
 
 		rsspp::item item;
 
