diff -Nur newsbeuter-2.9.a/config.sh newsbeuter-2.9.b/config.sh
--- newsbeuter-2.9.a/config.sh	2015-02-19 02:56:59.000000000 -0800
+++ newsbeuter-2.9.b/config.sh	2017-08-19 07:09:13.234215744 -0700
@@ -110,7 +110,7 @@
 if [ `uname -s` = "Darwin" ]; then
 	check_custom "ncurses5.4" "ncurses5.4-config" || fail "ncurses5.4"
 elif [ `uname -s` != "OpenBSD" ]; then
-	check_custom "ncursesw5" "ncursesw5-config" ||  fail "ncursesw"
+	check_custom "ncursesw5" "ncursesw5-config" || check_custom "ncursesw6" "ncursesw6-config" || fail "ncursesw"
 fi
 check_ssl_implementation
 all_aboard_the_fail_boat
