--- VTKConfig.cmake	2014-01-07 20:30:39.472055732 -0500
+++ ../lib64-fix-good/VTKConfig.cmake	2014-01-07 20:44:02.498085679 -0500
@@ -10,7 +10,7 @@
 get_filename_component(VTK_INSTALL_PREFIX "${VTK_INSTALL_PREFIX}" PATH)
 get_filename_component(VTK_INSTALL_PREFIX "${VTK_INSTALL_PREFIX}" PATH)
 get_filename_component(VTK_INSTALL_PREFIX "${VTK_INSTALL_PREFIX}" PATH)
-set(VTK_MODULES_DIR "${VTK_INSTALL_PREFIX}/lib/cmake/vtk-6.0/Modules")
+set(VTK_MODULES_DIR "${VTK_INSTALL_PREFIX}/lib64/cmake/vtk-6.0/Modules")
 
 # The C and C++ flags added by VTK to the cmake-configured flags.
 SET(VTK_REQUIRED_C_FLAGS "")
@@ -25,7 +25,7 @@
 SET(VTK_BUILD_VERSION "0")
 
 # The location of the UseVTK.cmake file.
-SET(VTK_CMAKE_DIR "${VTK_INSTALL_PREFIX}/lib/cmake/vtk-6.0")
+SET(VTK_CMAKE_DIR "${VTK_INSTALL_PREFIX}/lib64/cmake/vtk-6.0")
 SET(VTK_USE_FILE "${VTK_CMAKE_DIR}/UseVTK.cmake")
 
 #-----------------------------------------------------------------------------
@@ -44,13 +44,13 @@
 endforeach()
 
 # Import VTK targets.
-set(VTK_CONFIG_TARGETS_FILE "${VTK_INSTALL_PREFIX}/lib/cmake/vtk-6.0/VTKTargets.cmake")
+set(VTK_CONFIG_TARGETS_FILE "${VTK_INSTALL_PREFIX}/lib64/cmake/vtk-6.0/VTKTargets.cmake")
 if(NOT TARGET vtkCommonCore)
   include("${VTK_CONFIG_TARGETS_FILE}")
 endif()
 
 # Load module interface macros.
-include("${VTK_INSTALL_PREFIX}/lib/cmake/vtk-6.0/vtkModuleAPI.cmake")
+include("${VTK_INSTALL_PREFIX}/lib64/cmake/vtk-6.0/vtkModuleAPI.cmake")
 
 # Compute set of requested modules.
 if(VTK_FIND_COMPONENTS)
