--- VTKTargets.cmake	2014-01-07 20:30:39.472055732 -0500
+++ ../lib64-fix-good/VTKTargets.cmake	2014-01-07 20:44:02.498085679 -0500
@@ -43,9 +43,9 @@
 # Use original install prefix when loaded through a
 # cross-prefix symbolic link such as /lib -> /usr/lib.
 get_filename_component(_realCurr "${_IMPORT_PREFIX}" REALPATH)
-get_filename_component(_realOrig "/usr/lib/cmake/vtk-6.0" REALPATH)
+get_filename_component(_realOrig "/usr/lib64/cmake/vtk-6.0" REALPATH)
 if(_realCurr STREQUAL _realOrig)
-  set(_IMPORT_PREFIX "/usr/lib/cmake/vtk-6.0")
+  set(_IMPORT_PREFIX "/usr/lib64/cmake/vtk-6.0")
 endif()
 unset(_realOrig)
 unset(_realCurr)
