From c20e9bb1fbf1175b76522e85d04ff7df96beaf51 Mon Sep 17 00:00:00 2001
From: orbea <orbea@riseup.net>
Date: Sun, 27 Oct 2019 11:07:18 -0700
Subject: [PATCH] Fix build issue with gettext-0.20.

*** error: gettext infrastructure mismatch: using a Makefile.in.in from gettext version 0.19 but the autoconf macros are from gettext version 0.20
---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index c6e1dcb..746c768 100644
--- a/configure.ac
+++ b/configure.ac
@@ -19,6 +19,7 @@ AM_GNU_GETTEXT([external])
 #For rhel6 but rhel6 doesn't have SDL2
 #AM_GNU_GETTEXT_VERSION(0.17)
 AM_GNU_GETTEXT_VERSION(0.19.8)
+AM_GNU_GETTEXT_REQUIRE_VERSION([0.19.8])
 
 dnl this is a c++ program
 dnl CFLAGS="-g -W -Wall -O2"
