diff -ur pinball-0.3.4/Makefile.am pinball-0.3.4/Makefile.am
--- pinball-0.3.4/Makefile.am	2018-03-24 16:40:31.000000000 -0700
+++ pinball-0.3.4/Makefile.am	2018-04-04 23:11:27.780673376 -0700
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS = gnu
 
-SUBDIRS = libltdl addon base data src test
+SUBDIRS = addon base data src test
 
 EXTRA_DIST = bootstrap pinball.spec clean pinball.desktop
 
@@ -10,4 +10,4 @@
 
 bin_SCRIPTS = pinball-config
 
-ACLOCAL_AMFLAGS = -I libltdl/m4
+ACLOCAL_AMFLAGS =
diff -ur pinball-0.3.4/configure.ac pinball-0.3.4/configure.ac
--- pinball-0.3.4/configure.ac	2018-03-24 16:40:31.000000000 -0700
+++ pinball-0.3.4/configure.ac	2018-04-04 23:07:50.882589216 -0700
@@ -4,10 +4,6 @@
 AC_INIT(pinball, 0.3.4.1)
 
 AC_CONFIG_HEADERS(pinconfig.h)
-LT_CONFIG_LTDL_DIR([libltdl])
-AC_CONFIG_AUX_DIR([libltdl/config])
-AC_CONFIG_MACRO_DIR([libltdl/m4])
-
 AC_CANONICAL_HOST
 AC_CANONICAL_TARGET
 
@@ -37,7 +33,6 @@
     [CXXFLAGS=$with_cxxflags; export CXXFLAGS])
 
 AC_LIBTOOL_DLOPEN
-LTDL_INIT
 
 dnl *******************************************
 dnl PATHS AND DIRS ****************************
diff -ur pinball-0.3.4/src/Makefile.am pinball-0.3.4/src/Makefile.am
--- pinball-0.3.4/src/Makefile.am	2018-03-24 16:40:31.000000000 -0700
+++ pinball-0.3.4/src/Makefile.am	2018-04-04 23:09:07.672325490 -0700
@@ -7,10 +7,10 @@
 
 bin_PROGRAMS = pinball
 
-AM_CPPFLAGS = -I../base -I../addon @INCLTDL@
+AM_CPPFLAGS = -I../base -I../addon
 
 pinball_LDADD = \
- libemilia_pin.a ../addon/libemilia_addon.a ../base/libemilia_base.a @LIBLTDL@
+ libemilia_pin.a ../addon/libemilia_addon.a ../base/libemilia_base.a -lltdl
 
 pinball_LDFLAGS = -export-dynamic
 
diff -ur pinball-0.3.4/test/Makefile.am pinball-0.3.4/test/Makefile.am
--- pinball-0.3.4/test/Makefile.am	2018-03-24 16:40:31.000000000 -0700
+++ pinball-0.3.4/test/Makefile.am	2018-04-04 23:09:36.731605043 -0700
@@ -5,8 +5,8 @@
 
 noinst_PROGRAMS = scale simple light texture load explode collision signal billboard font thread menu joy sound trans math misc varray unittest
 
-AM_CPPFLAGS = -I../base -I../addon -I../src @INCLTDL@
-LDADD = ../src/libemilia_pin.a ../addon/libemilia_addon.a ../base/libemilia_base.a @LIBLTDL@
+AM_CPPFLAGS = -I../base -I../addon -I../src
+LDADD = ../src/libemilia_pin.a ../addon/libemilia_addon.a ../base/libemilia_base.a -lltdl
 
 testlib_LTLIBRARIES = libModuleTest.la
 
