diff -Naur 0ad-0.27.0.orig/build/workspaces/update-workspaces.sh 0ad-0.27.0/build/workspaces/update-workspaces.sh
--- 0ad-0.27.0.orig/build/workspaces/update-workspaces.sh	2025-01-28 20:57:28.000000000 +0100
+++ 0ad-0.27.0/build/workspaces/update-workspaces.sh	2025-04-03 08:50:49.616288000 +0200
@@ -1,10 +1,5 @@
 #!/bin/sh
 
-if [ "$(id -u)" = "0" ]; then
-	echo "Running as root will mess up file permissions. Aborting ..." 1>&2
-	exit 1
-fi
-
 die()
 {
 	echo ERROR: "$*"
