diff -Nur pptpd-1.4.0.orig/plugins/patchlevel.h pptpd-1.4.0/plugins/patchlevel.h
--- pptpd-1.4.0.orig/plugins/patchlevel.h	2013-02-18 16:05:46.000000000 -0600
+++ pptpd-1.4.0/plugins/patchlevel.h	2016-01-10 23:24:22.095247474 -0600
@@ -1,5 +1,5 @@
 /* upstream patchlevel.h,v 1.60 2004/01/13 04:46:52 paulus Exp */
 /* $Id: patchlevel.h,v 1.4 2005/02/24 01:25:34 quozl Exp $ */
 
-#define VERSION		"2.4.3"
-#define DATE		"13 Jan 2004"
+#define VERSION		"2.4.7"
+#define DATE		"9 Nov 2014"
