diff -Nur pptpd-1.4.0.orig/plugins/Makefile pptpd-1.4.0/plugins/Makefile
--- pptpd-1.4.0.orig/plugins/Makefile	2013-02-18 16:05:46.000000000 -0600
+++ pptpd-1.4.0/plugins/Makefile	2016-01-10 23:23:06.816535697 -0600
@@ -3,7 +3,7 @@
 CFLAGS	= $(COPTS) -I.. -I../../include -fPIC
 LDFLAGS	= -shared
 LDADD   = -lutil
-INSTALL	= install -o root
+INSTALL	= install
 prefix  = /usr/local
 
 PLUGINS = pptpd-logwtmp.so
@@ -18,14 +18,14 @@
 %.so: %.c
 	$(CC) -o $@ $(LDFLAGS) $(CFLAGS) $^ $(LDADD)
 
-LIBDIR	?= $(DESTDIR)$(prefix)/lib/pptpd
+LIBDIR	?= $(prefix)/@baselibdir@/pptpd
 
 install: $(PLUGINS)
-	$(INSTALL) -d $(LIBDIR)
-	$(INSTALL) $? $(LIBDIR)
+	$(INSTALL) -d $(DESTDIR)/$(LIBDIR)
+	$(INSTALL) $? $(DESTDIR)/$(LIBDIR)
 
 uninstall:
-	rm -f $(LIBDIR)$(PLUGINS)
+	rm -f $(DESTDIR)/$(LIBDIR)$(PLUGINS)
 
 clean:
 	rm -f *.o *.so *.a
