diff -rupN pptpd-1.3.4/pptpctrl.c pptpd-1.3.4.new/pptpctrl.c
--- pptpd-1.3.4/pptpctrl.c	2006-12-08 13:01:40.000000000 +1300
+++ pptpd-1.3.4.new/pptpctrl.c	2012-09-14 11:16:38.646659399 +1200
@@ -759,8 +759,8 @@ static void launch_pppd(char **pppaddrs,
 			syslog(LOG_DEBUG, "CTRL (PPPD Launcher): remote address = %s", pppaddrs[1]);
 	}
 	
+	char pppInterfaceIPs[33];
 	if (*pppaddrs[0] || *pppaddrs[1]) {
-		char pppInterfaceIPs[33];
 		sprintf(pppInterfaceIPs, "%s:%s", pppaddrs[0], pppaddrs[1]);
 		pppd_argv[an++] = pppInterfaceIPs;
 	}
