diff -Naur hylafax-6.0.6.orig/libhylafax/FaxRecvInfo.c++ hylafax-6.0.6/libhylafax/FaxRecvInfo.c++
--- hylafax-6.0.6.orig/libhylafax/FaxRecvInfo.c++	2012-06-06 02:58:38.000000000 +0200
+++ hylafax-6.0.6/libhylafax/FaxRecvInfo.c++	2017-06-03 19:51:39.491707130 +0200
@@ -112,7 +112,7 @@
     if (cp == NULL || cp[1] != ',' || cp[2] != '"')
 	return (false);
     u_int i = 0;
-    while (cp+2 != '\0') {
+    while (*(cp+2) != '\0') {
 	callid[i] = cp+3;		// +1 for "/+1 for ,/+1 for "
 	if (*cp == '\"') break;
 	callid[i].resize(callid[i].next(0,'"'));
