diff -up ./m4/libpq.m4.pg10 ./m4/libpq.m4
--- ./m4/libpq.m4.pg10	2017-12-18 11:33:49.378910317 -0500
+++ ./m4/libpq.m4	2017-12-18 11:34:16.833013133 -0500
@@ -47,11 +47,11 @@ AC_ARG_ENABLE(psqltest, [  --disable-psq
     PG_LIBS="-L`$PG_CONFIG $psqlconf_args --libdir`"
 
     psql_major_version=`$PG_CONFIG $psql_args --version | \
-           sed 's/PostgreSQL \([[0-9]]*\).\([[0-9]]*\)\(.\([[0-9]]*\)\).*$/\1/'`
+           sed 's/PostgreSQL \([[0-9]]*\)\.\([[0-9]]*\)\(\.*\([[0-9]]*\)\).*$/\1/'`
     psql_minor_version=`$PG_CONFIG $psql_args --version | \
-           sed 's/PostgreSQL \([[0-9]]*\).\([[0-9]]*\)\(.\([[0-9]]*\)\).*$/\2/'`
+           sed 's/PostgreSQL \([[0-9]]*\)\.\([[0-9]]*\)\(\.*\([[0-9]]*\)\).*$/\2/'`
     psql_micro_version=`$PG_CONFIG $psql_config_args --version | \
-           sed 's/PostgreSQL \([[0-9]]*\).\([[0-9]]*\)\(.\([[0-9]]*\)\).*$/\4/'`
+           sed 's/PostgreSQL \([[0-9]]*\)\.\([[0-9]]*\)\(\.*\([[0-9]]*\)\).*$/\4/'`
     if test "x$psql_micro_version" = "x" ; then
       psql_micro_version=0
     fi
