--- a/src/compat/compat.h	2026-03-29 18:44:01.000000000 +0300
+++ b/src/compat/compat.h	2026-05-03 14:07:42.185698712 +0300
@@ -34,6 +34,8 @@
 #elif LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0) && LINUX_VERSION_CODE >= KERNEL_VERSION(5, 15, 0)
 #define ISUBUNTU2204
 #endif
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0) && LINUX_VERSION_CODE >= KERNEL_VERSION(5, 15, 0)
+#define ISSLACKWARE150
 #endif
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3, 10, 0)
@@ -1288,7 +1290,7 @@
 }
 #endif
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 91) && !defined(ISUBUNTU2004) && !defined(ISUBUNTU2204) && !defined(ISRHEL9)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 91) && !defined(ISUBUNTU2004) && !defined(ISUBUNTU2204) && !defined(ISRHEL9) && !defined(ISSLACKWARE150)
 #include <linux/timer.h>
 static inline int timer_delete(struct timer_list *timer)
 {
