diff -Nur slock-0.9.orig/config.mk slock-0.9/config.mk
--- slock-0.9.orig/config.mk	2008-07-29 13:22:46.000000000 -0500
+++ slock-0.9/config.mk	2009-07-06 00:50:51.209930578 -0500
@@ -15,7 +15,7 @@
 
 # flags
 CPPFLAGS = -DVERSION=\"${VERSION}\" -DHAVE_SHADOW_H
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
+CFLAGS = @SLACKCFLAGS@ -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
 LDFLAGS = -s ${LIBS}
 
 # On *BSD remove -DHAVE_SHADOW_H from CPPFLAGS and add -DHAVE_BSD_AUTH
