diff -Nur slock-1.1.orig/config.mk slock-1.1/config.mk
--- slock-1.1.orig/config.mk	2012-10-25 14:00:04.000000000 -0500
+++ slock-1.1/config.mk	2013-02-17 15:57:26.679384573 -0600
@@ -15,7 +15,7 @@
 
 # flags
 CPPFLAGS = -DVERSION=\"${VERSION}\" -DHAVE_SHADOW_H -DCOLOR1=\"black\" -DCOLOR2=\"\#005577\"
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
+CFLAGS = @SLACKCFLAGS@ -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
 LDFLAGS = -s ${LIBS}
 
 # On *BSD remove -DHAVE_SHADOW_H from CPPFLAGS and add -DHAVE_BSD_AUTH
