--- slock-1.2/config.mk.orig	2014-12-22 11:18:07.000000000 +0100
+++ slock-1.2/config.mk	2015-03-01 16:50:44.945646629 +0100
@@ -15,7 +15,7 @@
 
 # flags
 CPPFLAGS = -DVERSION=\"${VERSION}\" -DHAVE_SHADOW_H
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
+CFLAGS = @SLACKCFLAGS@ -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
 LDFLAGS = -s ${LIBS}
 
 # On *BSD remove -DHAVE_SHADOW_H from CPPFLAGS and add -DHAVE_BSD_AUTH
