diff -u -r exim-4.82.1-orig/src/version.sh exim-4.82.1/src/version.sh
--- exim-4.82.1-orig/src/version.sh	2014-05-26 22:45:04.000000000 +0200
+++ exim-4.82.1/src/version.sh	2014-05-28 16:14:14.952290924 +0200
@@ -1,4 +1,4 @@
 # automatically generated file - see ../scripts/reversion
-EXIM_RELEASE_VERSION="4.82"
-EXIM_VARIANT_VERSION="_1-5b7a7c0-XX"
+EXIM_RELEASE_VERSION="4.82.1"
+EXIM_VARIANT_VERSION=""
 EXIM_COMPILE_NUMBER="1"
