diff -Naur grisbi-0.6.0b1.orig/help/C/Makefile.in grisbi-0.6.0b1/help/C/Makefile.in
--- grisbi-0.6.0b1.orig/help/C/Makefile.in	2009-09-19 09:41:04.000000000 +0000
+++ grisbi-0.6.0b1/help/C/Makefile.in	2009-09-20 16:20:32.000000000 +0000
@@ -230,7 +230,7 @@
 	quickstart.html \
 	translation.html
 
-grisbi_helpdir = $(datadir)/doc/grisbi/C
+grisbi_helpdir = $(prefix)/doc/@PACKAGE@-@VERSION@/C
 grisbi_help_DATA = \
 	topic.dat     \
 	contents_motif.gif \
diff -Naur grisbi-0.6.0b1.orig/help/de/Makefile.in grisbi-0.6.0b1/help/de/Makefile.in
--- grisbi-0.6.0b1.orig/help/de/Makefile.in	2009-09-19 09:41:04.000000000 +0000
+++ grisbi-0.6.0b1/help/de/Makefile.in	2009-09-20 16:21:10.000000000 +0000
@@ -230,7 +230,7 @@
 	quickstart.html \
 	translation.html
 
-grisbi_helpdir = $(datadir)/doc/grisbi/de
+grisbi_helpdir = $(prefix)/doc/@PACKAGE@-@VERSION@/de
 grisbi_help_DATA = \
 	topic.dat     \
 	contents_motif.gif \
diff -Naur grisbi-0.6.0b1.orig/help/en/Makefile.in grisbi-0.6.0b1/help/en/Makefile.in
--- grisbi-0.6.0b1.orig/help/en/Makefile.in	2009-09-19 09:41:04.000000000 +0000
+++ grisbi-0.6.0b1/help/en/Makefile.in	2009-09-20 16:21:47.000000000 +0000
@@ -230,7 +230,7 @@
 	quickstart.html \
 	translation.html
 
-grisbi_helpdir = $(datadir)/doc/grisbi/en
+grisbi_helpdir = $(prefix)/doc/@PACKAGE@-@VERSION@/en
 grisbi_help_DATA = \
 	topic.dat     \
 	contents_motif.gif \
diff -Naur grisbi-0.6.0b1.orig/help/fr/Makefile.in grisbi-0.6.0b1/help/fr/Makefile.in
--- grisbi-0.6.0b1.orig/help/fr/Makefile.in	2009-09-19 09:41:04.000000000 +0000
+++ grisbi-0.6.0b1/help/fr/Makefile.in	2009-09-20 16:22:19.000000000 +0000
@@ -230,7 +230,7 @@
 	quickstart.html \
 	translation.html
 
-grisbi_helpdir = $(datadir)/doc/grisbi/fr
+grisbi_helpdir = $(prefix)/doc/@PACKAGE@-@VERSION@/fr
 grisbi_help_DATA = \
 	topic.dat     \
 	contents_motif.gif \
diff -Naur grisbi-0.6.0b1.orig/src/Makefile.am grisbi-0.6.0b1/src/Makefile.am
--- grisbi-0.6.0b1.orig/src/Makefile.am	2009-07-14 19:33:32.000000000 +0000
+++ grisbi-0.6.0b1/src/Makefile.am	2009-09-20 16:07:36.000000000 +0000
@@ -1,7 +1,7 @@
 # Process this file with automake to produce Makefile.in
 
 localedir = $(datadir)/locale
-INCLUDES = -I$(top_srcdir) -I$(includedir) -I../intl -DLOCALEDIR=\"$(localedir)\" -DPIXMAPS_DIR="\"$(datadir)/pixmaps/@PACKAGE@\"" -DPLUGINS_DIR="\"$(prefix)/lib/@PACKAGE@\"" -DLOGO_PATH="\"$(datadir)/pixmaps/@PACKAGE@/grisbi-logo.png\"" -DDATA_PATH="\"$(datadir)/@PACKAGE@\"" -DHELP_PATH="\"$(datadir)/doc/@PACKAGE@\"" -Wall
+INCLUDES = -I$(top_srcdir) -I$(includedir) -I../intl -DLOCALEDIR=\"$(localedir)\" -DPIXMAPS_DIR="\"$(datadir)/pixmaps/@PACKAGE@\"" -DPLUGINS_DIR="\"$(prefix)/lib/@PACKAGE@\"" -DLOGO_PATH="\"$(datadir)/pixmaps/@PACKAGE@/grisbi-logo.png\"" -DDATA_PATH="\"$(datadir)/@PACKAGE@\"" -DHELP_PATH="\"$(prefix)/doc/@PACKAGE@-@VERSION@\"" -Wall
 bin_PROGRAMS = grisbi
 
 # TODO: This manual setting of -Wl is a hack, I cannot get it from
diff -Naur grisbi-0.6.0b1.orig/src/Makefile.in grisbi-0.6.0b1/src/Makefile.in
--- grisbi-0.6.0b1.orig/src/Makefile.in	2009-09-19 09:41:05.000000000 +0000
+++ grisbi-0.6.0b1/src/Makefile.in	2009-09-20 16:06:26.000000000 +0000
@@ -379,7 +379,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-INCLUDES = -I$(top_srcdir) -I$(includedir) -I../intl -DLOCALEDIR=\"$(localedir)\" -DPIXMAPS_DIR="\"$(datadir)/pixmaps/@PACKAGE@\"" -DPLUGINS_DIR="\"$(prefix)/lib/@PACKAGE@\"" -DLOGO_PATH="\"$(datadir)/pixmaps/@PACKAGE@/grisbi-logo.png\"" -DDATA_PATH="\"$(datadir)/@PACKAGE@\"" -DHELP_PATH="\"$(datadir)/doc/@PACKAGE@\"" -Wall
+INCLUDES = -I$(top_srcdir) -I$(includedir) -I../intl -DLOCALEDIR=\"$(localedir)\" -DPIXMAPS_DIR="\"$(datadir)/pixmaps/@PACKAGE@\"" -DPLUGINS_DIR="\"$(prefix)/lib/@PACKAGE@\"" -DLOGO_PATH="\"$(datadir)/pixmaps/@PACKAGE@/grisbi-logo.png\"" -DDATA_PATH="\"$(datadir)/@PACKAGE@\"" -DHELP_PATH="\"$(prefix)/doc/@PACKAGE@-@VERSION@\"" -Wall
 
 # TODO: This manual setting of -Wl is a hack, I cannot get it from
 # configure right now because of no doc.
