diff -Naur qdvdauthor-2.1.0.orig/qdvdauthor/oneclickwizard.h qdvdauthor-2.1.0/qdvdauthor/oneclickwizard.h
--- qdvdauthor-2.1.0.orig/qdvdauthor/oneclickwizard.h	2010-01-09 15:01:14.000000000 +0100
+++ qdvdauthor-2.1.0/qdvdauthor/oneclickwizard.h	2012-09-13 13:21:08.191001640 +0200
@@ -12,6 +12,7 @@
 #ifndef ONECLICKWIZARD_H
 #define ONECLICKWIZARD_H
 
+#include <unistd.h>
 #include "dialogfiles.h"
 #include "plugins/interface.h"
 #include "uioneclickwizard.h"
diff -Naur qdvdauthor-2.1.0.orig/qdvdauthor/qplayer/engines/xineinfo.h qdvdauthor-2.1.0/qdvdauthor/qplayer/engines/xineinfo.h
--- qdvdauthor-2.1.0.orig/qdvdauthor/qplayer/engines/xineinfo.h	2010-01-09 15:01:14.000000000 +0100
+++ qdvdauthor-2.1.0/qdvdauthor/qplayer/engines/xineinfo.h	2012-09-13 13:24:14.570985447 +0200
@@ -12,6 +12,7 @@
 #ifndef XINEINFO_H
 #define XINEINFO_H
 
+#include <unistd.h>
 #include <qthread.h>
 #include <qstring.h>
 
diff -Naur qdvdauthor-2.1.0.orig/qdvdauthor/render_client.h qdvdauthor-2.1.0/qdvdauthor/render_client.h
--- qdvdauthor-2.1.0.orig/qdvdauthor/render_client.h	2010-01-09 15:01:14.000000000 +0100
+++ qdvdauthor-2.1.0/qdvdauthor/render_client.h	2012-09-13 13:20:28.254016622 +0200
@@ -12,6 +12,7 @@
 #ifndef RENDER_CLIENT_H
 #define RENDER_CLIENT_H
 
+#include <unistd.h>
 #include <qthread.h>
 #include <qsocket.h>
 #include <qpixmap.h>
