--- iotop-0.6.orig/iotop/data.py	2013-05-26 18:44:18.000000000 -0400
+++ iotop-0.6/iotop/data.py	2018-08-18 23:21:03.528110821 -0400
@@ -193,6 +193,7 @@
     result_dict = {}
     try:
         for line in open('/proc/%d/status' % pid):
+            if not line.strip(): continue            
             key, value = line.split(':\t', 1)
             result_dict[key] = value.strip()
     except IOError:
