diff -Naur VirtualBox-6.1.26.orig/configure VirtualBox-6.1.26/configure
--- VirtualBox-6.1.26.orig/configure	2021-07-28 18:13:45.000000000 +0200
+++ VirtualBox-6.1.26/configure	2021-08-20 18:43:20.615000000 +0200
@@ -1561,9 +1561,9 @@
         qt5_ver=`pkg-config Qt5Core --modversion 2>> $LOG`
         if [ $? -eq 0 ]; then
           echo "(Qt5 from pkg-config)" >> $LOG
-          FLGQT5=`pkg-config Qt5Core Qt5Gui --cflags`
+          FLGQT5=`pkg-config Qt5Core --cflags`
           # gcc 4.8 is able to compile with C++11 (see also VBOX_GCC_std in Config.kmk)
-          [ $(($cc_maj * 100 + $cc_min)) -ge 408 ] && FLGQT5="$FLGQT5 -std=c++11"
+          FLGQT5="$FLGQT5 -std=c++11"
           INCQT5=`strip_I "$FLGQT5"`
           LIBDIR5=`pkg-config Qt5Core --variable=libdir`
           LIBQT5=`pkg-config Qt5Core Qt5Gui --libs`
