--- Makefile.kmk.orig	2011-08-15 14:28:45.000000000 +0200
+++ Makefile.kmk	2011-09-16 12:21:33.049000016 +0200
@@ -19,6 +19,7 @@
 include $(KBUILD_PATH)/subheader.kmk
 if1of ($(KBUILD_TARGET), linux)
 
+ ifdef VBOX_VITH_VBOXDRV
  #
  # The driver.
  # Note! For Linux, this is just a compile test. Don't use the binary!
@@ -40,6 +41,7 @@
  	VBoxPci.c
  VBoxPci_LIBS          += \
  	$(PATH_STAGE_LIB)/SUPR0IdcClient$(VBOX_SUFF_LIB)
+ endif # VBOX_VITH_VBOXDRV
 
 
  #
