--- src/VBox/ExtPacks/VBoxDTrace/include/VBoxDTraceTypes.h.orig	2017-08-30 14:43:09.000632019 +0200
+++ src/VBox/ExtPacks/VBoxDTrace/include/VBoxDTraceTypes.h	2017-08-30 14:46:09.577874319 +0200
@@ -55,7 +55,7 @@
 typedef uint32_t                    id_t;
 typedef uint32_t                    zoneid_t;
 #endif
-#if !defined(NGREG) || !defined(RT_OS_LINUX)
+#if !defined(__NGREG) || !defined(RT_OS_LINUX)
 typedef RTCCINTREG                  greg_t;
 #else
 AssertCompileSize(greg_t, sizeof(RTCCINTREG));
