diff -Naur xtrlock-2.16/xtrlock.man xtrlock-2.16.patched/xtrlock.man
--- xtrlock-2.16/xtrlock.man	2025-01-03 12:34:41.000000000 -0500
+++ xtrlock-2.16.patched/xtrlock.man	2025-01-21 19:42:41.791013326 -0500
@@ -44,9 +44,9 @@
 \fB\-f\fR
 fork after locking is complete, and return success from the parent
 process
+.TP
 \fB\-F\fR
 force lock, e.g. if there is no password set.
-process
 .SH X RESOURCES, CONFIGURATION
 None.
 .SH BUGS
