From: Michael Casadevall <sonicmctails@gmail.com>
Subject: hardcode the proper plugins folder to /usr/lib/codeblocks/plugins
Forwarded: no

---
 src/sdk/configmanager.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- codeblocks.orig/src/sdk/configmanager.cpp
+++ codeblocks/src/sdk/configmanager.cpp
@@ -514,7 +514,7 @@ wxString ConfigManager::GetFolder(Search
 
         case sdPluginsGlobal:
 #ifndef CB_AUTOCONF
-            return ConfigManager::data_path_global + _T("/plugins");
+            return  _T("/usr/lib/codeblocks/plugins");
 #else
             return ConfigManager::plugin_path_global;
 #endif
