Index: codeblocks-10.05/src/plugins/contrib/help_plugin/defs.h
===================================================================
--- codeblocks-10.05.orig/src/plugins/contrib/help_plugin/defs.h	2010-05-22 10:29:35.000000000 +0000
+++ codeblocks-10.05/src/plugins/contrib/help_plugin/defs.h	2012-05-29 05:18:08.674124530 +0000
@@ -289,7 +289,7 @@
 
             if (i.second == false && overwrite)
             {
-                find(k)->second = t;
+                this->find(k)->second = t;
             }
 
             return i.first;
@@ -297,7 +297,7 @@
 
         int remove(const Key &k)
         {
-            return erase(k);
+            return this->erase(k);
         }
 };
 
Index: codeblocks-10.05/src/plugins/contrib/help_plugin/man2html.cpp
===================================================================
--- codeblocks-10.05.orig/src/plugins/contrib/help_plugin/man2html.cpp	2010-05-22 10:29:35.000000000 +0000
+++ codeblocks-10.05/src/plugins/contrib/help_plugin/man2html.cpp	2012-05-29 05:15:59.186120279 +0000
@@ -150,6 +150,7 @@
 # include <kdeversion.h>
 # define BYTEARRAY(x) x
 #endif
+#include <unistd.h>
 
 #ifdef __MINGW32__
     #include <io.h>
