diff -Naur flashcards-2.4/flashcards/configure.py flashcards-2.4.patched/flashcards/configure.py
--- flashcards-2.4/flashcards/configure.py	2012-02-06 03:32:51.000000000 -0500
+++ flashcards-2.4.patched/flashcards/configure.py	2020-01-17 23:03:15.495097878 -0500
@@ -29,6 +29,8 @@
     documents = os.path.join(home_path, 'Documents')
     ROOT = os.path.join(documents, 'flashcards')
     
+if not os.path.isdir(documents):
+    os.mkdir(documents)
 if not os.path.isdir(ROOT):
     os.mkdir(ROOT)
     
diff -Naur flashcards-2.4/setup.py flashcards-2.4.patched/setup.py
--- flashcards-2.4/setup.py	2012-02-06 03:45:22.000000000 -0500
+++ flashcards-2.4.patched/setup.py	2020-01-17 23:03:32.311096350 -0500
@@ -14,7 +14,7 @@
     author_email = "luisnaranjo733@hotmail.com",
     description = ("A simple command line flashcards utility, similar to physical flashcards."),
     license = "GNU GPL",
-    install_requires= ['argparse','configobj',],
+    install_requires= ['configobj',],
     entry_points = {
     'console_scripts': ['flashcards = flashcards.flashcards:main']
     },
