--- tkimg1.4/libtiff/configure.in	2010-07-06 15:48:46.000000000 +0200
+++ tkimg1.4.new/libtiff/configure.in	2011-02-28 23:24:18.471259943 +0100
@@ -67,7 +67,7 @@
 else
     TIFFHOST="$host"
 fi
-${srcdir}/../compat/libtiff/configure --disable-zlib --disable-pixarlog --disable-jpeg \
+${srcdir}/../compat/libtiff/configure --disable-zlib --disable-pixarlog --disable-jpeg --disable-jbig \
 	"--with-CC=$TIFFCC" $TIFFHOST
 mv -f tifftcl_confdefs.h confdefs.h
 
--- tkimg1.4/libtiff/configure	2010-07-07 10:09:46.000000000 +0200
+++ tkimg1.4.new/libtiff/configure	2011-02-28 23:25:53.529824614 +0100
@@ -5133,7 +5133,7 @@
 else
     TIFFHOST="$host"
 fi
-${srcdir}/../compat/libtiff/configure --disable-zlib --disable-pixarlog --disable-jpeg \
+${srcdir}/../compat/libtiff/configure --disable-zlib --disable-pixarlog --disable-jpeg --disable-jbig \
 	"--with-CC=$TIFFCC" $TIFFHOST
 mv -f tifftcl_confdefs.h confdefs.h
 
