--- trunk/src/callbacks.c	2012/03/23 01:30:32	1462
+++ trunk/src/callbacks.c	2012/03/30 22:41:23	1465
@@ -3999,7 +3999,7 @@
 	    lives_alarm_clear(alarm_handle);
 	    frames_written=mainw->jackd->frames_written;
 	    seek_pos=mainw->jackd->seek_pos;
-	    audio_ticks=mainw->pulsed->audio_ticks;
+	    audio_ticks=mainw->jackd->audio_ticks;
 	    
 	    jack_message.command=ASERVER_CMD_FILE_CLOSE;
 	    jack_message.data=NULL;