--- trunk/smogrify	2009/12/29 21:59:04	356
+++ trunk/smogrify	2010/01/08 02:10:47	368
@@ -3932,7 +3932,6 @@
 	system("sox -t .raw -r $arate -c $achans -$osigned -$osamps $oendian $audio_in -t .raw -r $nrate -c $nchans -$nsigned $nendian -$nsamps $audio_out stretch $stretch>/dev/null 2>&1");
     }
     else {
-	print("sox -t .raw -r $arate -c $achans -$osigned -$osamps $oendian $audio_in -t .raw -r $nrate -c $nchans -$nsigned $nendian -$nsamps $audio_out>/dev/null 2>&1");
 	system("sox -t .raw -r $arate -c $achans -$osigned -$osamps $oendian $audio_in -t .raw -r $nrate -c $nchans -$nsigned $nendian -$nsamps $audio_out>/dev/null 2>&1");
     }
 }
@@ -4302,7 +4301,7 @@
     $name=&mkname($i);
     my $nframes=0;
 
-    my $conv_cmd=$convert_command;
+    my $conv_command=$convert_command;
 
     if (!defined($file_ext)) {
 	$file_ext=&get_ext($file);
