diff -r 658ded2153f2 src/celeste/Utilities.h
--- a/src/celeste/Utilities.h	Sun Jan 01 13:23:39 2012 +0100
+++ b/src/celeste/Utilities.h	Sat Jan 07 19:38:25 2012 +0000
@@ -33,6 +33,8 @@
 #include <stdio.h>
 #ifdef _WINDOWS
 #include "direct.h"
+#else
+#include <unistd.h>
 #endif
 
 using namespace std; 
diff -r 658ded2153f2 src/foreign/zthread/include/zthread/Guard.h
--- a/src/foreign/zthread/include/zthread/Guard.h	Sun Jan 01 13:23:39 2012 +0100
+++ b/src/foreign/zthread/include/zthread/Guard.h	Sat Jan 07 19:38:25 2012 +0000
@@ -428,7 +428,7 @@
   template <class U, class V>
   Guard(Guard<U, V>& g) : LockHolder<LockType>(g) {
 
-    LockingPolicy::shareScope(*this, extract(g));
+    LockingPolicy::shareScope(*this, this->extract(g));
     
   }
 
@@ -458,7 +458,7 @@
   template <class U, class V>
   Guard(Guard<U, V>& g, LockType& lock) : LockHolder<LockType>(lock) {
 
-    LockingPolicy::transferScope(*this, extract(g));
+    LockingPolicy::transferScope(*this, this->extract(g));
 
   }
 
