diff -u serialmail-0.75.orig/error.h serialmail-0.75/error.h
--- serialmail-0.75.orig/error.h	1998-11-14 20:28:23.000000000 -0600
+++ serialmail-0.75/error.h	2004-01-29 11:15:32.000000000 -0600
@@ -1,7 +1,7 @@
 #ifndef ERROR_H
 #define ERROR_H
 
-extern int errno;
+#include <errno.h>
 
 extern int error_intr;
 extern int error_nomem;
diff -u serialmail-0.75.orig/leapsecs_read.c serialmail-0.75/leapsecs_read.c
--- serialmail-0.75.orig/leapsecs_read.c	1998-11-14 20:28:23.000000000 -0600
+++ serialmail-0.75/leapsecs_read.c	2004-01-29 11:20:09.000000000 -0600
@@ -2,7 +2,6 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <errno.h>
-extern int errno;
 #include "tai.h"
 #include "leapsecs.h"
 
