--- ./Makefile.orig	2010-11-22 11:47:26.000000000 +0200
+++ ./Makefile	2011-06-21 22:51:24.000000000 +0300
@@ -158,7 +158,7 @@
 				$(CPPFLAGS) $(CPPFLAGS.dmalloc) \
 				-DSYS_LIBDIR=$(SYS_LIBDIR)
 LDFLAGS.l2tpd=		-Wl,-E -L. -Lusl -lusl -ldl $(LIBS.dmalloc) -lc
-LDFLAGS.l2tpconfig=	-Lcli -lcli -lreadline $(LIBS.dmalloc) $(READLINE_LDFLAGS) -lc
+LDFLAGS.l2tpconfig=	-Lcli -lcli -lreadline -ltermcap $(LIBS.dmalloc) $(READLINE_LDFLAGS) -lc
 
 OPT_CFLAGS?=		-O
 
