diff -Nru /tmp/Kr0L01zv32/vbetool-0.7/Makefile.am /tmp/dmaQa97eLk/vbetool-0.7/Makefile.am
--- /tmp/Kr0L01zv32/vbetool-0.7/Makefile.am	2006-07-26 03:21:31.000000000 +0200
+++ /tmp/dmaQa97eLk/vbetool-0.7/Makefile.am	2006-11-27 03:37:31.000000000 +0100
@@ -23,4 +23,4 @@
 	$(RM) Makefile.in aclocal.m4 config.h.in stamp-h.in configure
 
 AM_CFLAGS = -g -Wall -pedantic -std=gnu99
-AM_LDFLAGS = 
+AM_LDFLAGS = -lz
diff -Nru /tmp/Kr0L01zv32/vbetool-0.7/Makefile.in /tmp/dmaQa97eLk/vbetool-0.7/Makefile.in
--- /tmp/Kr0L01zv32/vbetool-0.7/Makefile.in	2006-07-26 03:26:30.000000000 +0200
+++ /tmp/dmaQa97eLk/vbetool-0.7/Makefile.in	2006-11-27 03:37:55.000000000 +0100
@@ -182,7 +182,7 @@
 noinst_HEADERS = include/lrmi.h vbetool.h
 EXTRA_DIST = vbetool.1 LRMI
 AM_CFLAGS = -g -Wall -pedantic -std=gnu99
-AM_LDFLAGS = 
+AM_LDFLAGS = -lz
 all: all-recursive
 
 .SUFFIXES:
diff -Nru /tmp/Kr0L01zv32/vbetool-0.7/debian/changelog /tmp/dmaQa97eLk/vbetool-0.7/debian/changelog
--- /tmp/Kr0L01zv32/vbetool-0.7/debian/changelog	2006-07-26 03:29:16.000000000 +0200
+++ /tmp/dmaQa97eLk/vbetool-0.7/debian/changelog	2006-11-27 03:44:03.000000000 +0100
@@ -1,3 +1,11 @@
+vbetool (0.7-1.1) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Add zlib1g-dev to Build-Depends and -lz to LDFLAGS, as needed by newer
+    pciutils (closes: #396844).
+
+ -- Julien Cristau <julien.cristau@ens-lyon.org>  Mon, 27 Nov 2006 03:43:37 +0100
+
 vbetool (0.7-1) unstable; urgency=low
 
   * Add copyright information. Closes: #375159
diff -Nru /tmp/Kr0L01zv32/vbetool-0.7/debian/control /tmp/dmaQa97eLk/vbetool-0.7/debian/control
--- /tmp/Kr0L01zv32/vbetool-0.7/debian/control	2005-11-23 03:53:59.000000000 +0100
+++ /tmp/dmaQa97eLk/vbetool-0.7/debian/control	2006-11-27 03:43:23.000000000 +0100
@@ -2,7 +2,7 @@
 Section: utils
 Priority: optional
 Maintainer: Matthew Garrett <mjg59@srcf.ucam.org>
-Build-Depends: debhelper (>= 4.0.0), pciutils-dev
+Build-Depends: debhelper (>= 4.0.0), pciutils-dev, zlib1g-dev
 Standards-Version: 3.6.1
 
 Package: vbetool
