diff -Naur a/zcut12.c b/zcut12.c
--- a/zcut12.c	2001-07-24 02:20:32.000000000 -0400
+++ b/zcut12.c	2012-11-30 14:58:22.000000000 -0500
@@ -362,6 +362,7 @@
 	if (hiscore < 900)
 		error ("Could not find story header");
 
+	printf ("Z-Machine version %d\n", (int) GET_BYTE(header,0));
 	printf ("Story file release %d ", (int) GET_WORD(header,2));
 	printf ("serial %c%c%c%c%c%c.\n",
 		isprint (GET_BYTE(header,18)) ? GET_BYTE(header,18) : ' ',
