diff -Naur a/zcut12.c b/zcut12.c
--- a/zcut12.c	2001-07-24 02:20:32.000000000 -0400
+++ b/zcut12.c	2012-11-30 14:48:28.000000000 -0500
@@ -1556,13 +1556,17 @@
 			"- Atari 800/XL/XE disk images [.ATR]\n"
 			"- Commodore 64/128 disk images [.D64]\n"
 			"- Commodore Amiga ADF format (only Deadline so far) [.ADF]\n"
-			"- IBM PC bootable disks (MS-DOS version only)\n"
+			"- IBM PC bootable disks or disk images\n"
 			"- Oric disk images (after conversion through UnMFM) [.DSK]\n"
 			"- any file if the Z-code is stored in one piece\n"
 			"\n"
 			"Usage: zcut input-file-1 [input-file-2...] output-file\n"
-			"       zcut drive: output-file");
-
+#if defined( __MSDOS__) || defined(_WINNT_)
+			"       zcut drive: output-file"
+#else
+			"       zcut /dev/fd0 output-file"
+#endif
+			);
 		return EXIT_FAILURE;
 
 	}
