diff -Naur a/Makefile b/Makefile
--- a/Makefile	1992-10-31 21:52:32.000000000 -0500
+++ b/Makefile	2012-12-01 11:02:49.000000000 -0500
@@ -1,17 +1,9 @@
-# Makefile for Infocom Debugging Tool 
-# (works with Aztec 5.2a)
-
-CFLAGS = -DAMIGA -ms -so -wlc
-
 OBJS = 	alphabet.o check.o decode.o extern.o header.o macros.o main.o \
 	object.o recog.o stuffing.o tree.o vars.o vocab.o
 
-all: DebugTool
-
-DebugTool:	$(OBJS)
-	ln -o DebugTool $(OBJS) -lc
+all: zdebugtool
 
+zdebugtool:	$(OBJS)
+	$(CC) $(LDFLAGS) -o zdebugtool $(OBJS)
 
 $(OBJS):	frobnitz.h Makefile
-
-# (c) 1992 by Paul David Doherty
diff -Naur a/frobnitz.h b/frobnitz.h
--- a/frobnitz.h	1993-05-06 17:33:42.000000000 -0400
+++ b/frobnitz.h	2012-12-01 11:03:40.000000000 -0500
@@ -51,6 +51,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 
 /*****************************************************************/
