diff -Nur iniparser-2.17.orig//src/iniparser.h iniparser-2.17/src/iniparser.h
--- iniparser-2.17.orig//src/iniparser.h	2006-09-27 06:03:35.000000000 -0500
+++ iniparser-2.17/src/iniparser.h	2010-11-27 20:17:54.230042402 -0600
@@ -34,7 +34,7 @@
  */
 /* #include <unistd.h> */
 
-#include "dictionary.h"
+#include <iniparser/dictionary.h>
 
 /*-------------------------------------------------------------------------*/
 /**
