diff -up ugene-1.22.0/src/libs_3rdparty/samtools/samtools.pri.dont_hardcode_includes ugene-1.22.0/src/libs_3rdparty/samtools/samtools.pri
--- ugene-1.22.0/src/libs_3rdparty/samtools/samtools.pri.dont_hardcode_includes	2016-03-31 07:21:56.000000000 -0500
+++ ugene-1.22.0/src/libs_3rdparty/samtools/samtools.pri	2016-06-02 17:49:02.383069490 -0500
@@ -3,7 +3,7 @@ UGENE_RELATIVE_DESTDIR = ''
 
 TEMPLATE = lib
 CONFIG +=thread debug_and_release staticlib warn_off
-INCLUDEPATH += src src/samtools ../../include /usr/include
+INCLUDEPATH += src src/samtools ../../include
 win32 : INCLUDEPATH += src/samtools/win32
 DEFINES+="_FILE_OFFSET_BITS=64" _LARGEFILE64_SOURCE _USE_KNETFILE
 win32 : DEFINES += _USE_MATH_DEFINES "inline=__inline" "__func__=__FUNCTION__" "R_OK=4" "atoll=_atoi64" "alloca=_alloca"
diff -up ugene-1.22.0/src/plugins_3rdparty/variants/variants.pri.dont_hardcode_includes ugene-1.22.0/src/plugins_3rdparty/variants/variants.pri
--- ugene-1.22.0/src/plugins_3rdparty/variants/variants.pri.dont_hardcode_includes	2016-03-31 07:22:09.000000000 -0500
+++ ugene-1.22.0/src/plugins_3rdparty/variants/variants.pri	2016-06-02 20:26:35.265356432 -0500
@@ -20,7 +20,7 @@ use_bundled_zlib() {
 win32 : LIBS += -lwsock32
 
 
-INCLUDEPATH += src ../../include /usr/include
+INCLUDEPATH += src ../../include
 INCLUDEPATH += ../../libs_3rdparty/samtools/src/samtools 
 INCLUDEPATH += ../../libs_3rdparty/samtools/src
 INCLUDEPATH += ../../libs_3rdparty/samtools/src/samtools/bcftools
