diff -Naur warmux-11.04.orig/tools/list_games/main.cpp warmux-11.04/tools/list_games/main.cpp
--- warmux-11.04.orig/tools/list_games/main.cpp	2011-04-28 21:03:12.000000000 +0200
+++ warmux-11.04/tools/list_games/main.cpp	2012-04-01 10:03:27.752999997 +0200
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <unistd.h>
 #include <WARMUX_types.h>
 #include <WARMUX_network.h>
 #include <WARMUX_index_server.h>
