--- conf.c.orig	2008-06-12 00:34:23.000000000 -0400
+++ conf.c	2008-06-12 00:41:27.000000000 -0400
@@ -410,7 +410,7 @@
 									}
 									break;
 					case t_sep: 	/* gcc -Wall sucks */
-					case t_func:
+					case t_func: break;
 				}
 			}
 		}
@@ -481,7 +481,7 @@
 							break;
 				case t_listval: sprintf(rv, "%s", (char *)ci->v);
 				case t_sep: break;
-				case t_func:
+				case t_func: break;
 			}
 			
 			add = 1;
--- conf_scr.c.orig	2008-06-12 00:34:32.000000000 -0400
+++ conf_scr.c	2008-06-12 00:42:17.000000000 -0400
@@ -63,7 +63,7 @@
 			case t_listval:	strncpy(s, (char *)item->v, 32);
 							break;
 			case t_sep:		/* gcc -Wall sucks */
-			case t_func:
+			case t_func: break;
 		}
 
 		if (!item->unit) {
@@ -136,7 +136,7 @@
 						strncpy((char *)item->v, ll_get(item->list, tmp), 32);
 						break;
 		case t_sep:		/* gcc -Wall sucks */
-		case t_func:
+		case t_func: break;
 	}
 }	
 
--- wavemon.c.orig	2008-06-11 22:09:07.000000000 -0400
+++ wavemon.c	2008-06-11 22:09:15.000000000 -0400
@@ -170,7 +170,7 @@
 
 	if (conf.dump == 1) {
 	  dump_parameters(&conf);
-	  exit();
+	  exit(0);
 	}
 
 	/* initialize the ncurses interface */
