diff -Naur weston-10.0.0/libweston/meson.build weston-10.0.0-elogind/libweston/meson.build
--- weston-10.0.0/libweston/meson.build	2022-02-01 16:59:57.000000000 -0500
+++ weston-10.0.0-elogind/libweston/meson.build	2022-03-28 23:44:39.649373893 -0400
@@ -157,6 +157,9 @@
 systemd_dep = dependency('', required: false)
 if get_option('launcher-logind')
 	systemd_dep = dependency('libsystemd', version: '>= 209', required: false)
+	if not systemd_dep.found()
+		systemd_dep = dependency('libelogind', version: '>= 209', required: false)
+	endif
 	if systemd_dep.found()
 		config_h.set('HAVE_SYSTEMD_LOGIN_209', '1')
 	else
