The update will not work anyway.
Index: xvst-2.5/src/options.cpp
===================================================================
--- xvst-2.5.orig/src/options.cpp	2013-05-26 23:50:04.203059123 +0200
+++ xvst-2.5/src/options.cpp	2013-05-26 23:50:04.199059123 +0200
@@ -349,9 +349,9 @@
 
 	languageFile = LanguageManager::getDefaultUserLanguage(getLanguagesPath()); // "english_uk.language";
 
-	installAutomaticallyUpdates = true;
+	installAutomaticallyUpdates = false;
 	lastUpdate = QDate(2007, 01, 01);
-	checkForUpdatesOnStartup = true;
+	checkForUpdatesOnStartup = false;
 	checkForUpdatesEvery = 1;
 
 	displayBugReport = true;
