From d347ba63072ab13c0dbc1792d1a2b837af997a46 Mon Sep 17 00:00:00 2001
From: Martin Rotter <rotter@praktik.cz>
Date: Thu, 6 Jun 2019 11:35:25 +0200
Subject: [PATCH] Fix random class name

---
 src/miscellaneous/textfactory.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/miscellaneous/textfactory.cpp b/src/miscellaneous/textfactory.cpp
index 48597235..1ac6f049 100755
--- a/src/miscellaneous/textfactory.cpp
+++ b/src/miscellaneous/textfactory.cpp
@@ -151,8 +151,8 @@ quint64 TextFactory::initializeSecretEncryptionKey() {
 quint64 TextFactory::generateSecretEncryptionKey() {
   std::random_device rd;
   std::mt19937 mt(rd());
-  std::uniform_int<quint64> dist(std::numeric_limits<quint64>().min() + 1000000UL,
-                                 std::numeric_limits<quint64>().max());
+  std::uniform_int_distribution<quint64> dist(std::numeric_limits<quint64>().min() + 1000000UL,
+                                              std::numeric_limits<quint64>().max());
 
   return dist(mt);
 }
