diff -Naur zbar-0.23.93.orig/configure.ac zbar-0.23.93/configure.ac
--- zbar-0.23.93.orig/configure.ac	2024-01-09 10:13:01.000000000 +0100
+++ zbar-0.23.93/configure.ac	2024-11-24 11:37:03.410992000 +0100
@@ -675,8 +675,8 @@
        AS_IF([test "x$with_qt6" = "xno"],
              PKG_CHECK_MODULES([QT],
                                [Qt5Core >= 5 Qt5Gui >= 5 Qt5Widgets >= 5.0 Qt5X11Extras >= 5.0],
-                               [MOC=`pkg-config Qt5 --variable=moc`
-                                QT_VERSION=`$PKG_CONFIG Qt5 --modversion`
+                               [AC_CHECK_PROGS([MOC], [moc-qt5 moc])
+                                QT_VERSION=`$PKG_CONFIG Qt5Gui --modversion`;
                                 qt_pkgconfig_file="zbar-qt5.pc"
                                ],
                                [with_qt="no"]))
