diff -Naur meandmyshadow-0.3.orig/src/TreeStorageNode.cpp meandmyshadow-0.3/src/TreeStorageNode.cpp
--- meandmyshadow-0.3.orig/src/TreeStorageNode.cpp	2012-02-07 18:19:03.000000000 +0000
+++ meandmyshadow-0.3/src/TreeStorageNode.cpp	2012-07-11 02:17:49.188110565 +0000
@@ -17,6 +17,10 @@
 **
 ****************************************************************************/
 
+#ifndef WIN32
+#include <stdint.h>
+#include <unistd.h>
+#endif
 #include "TreeStorageNode.h"
 #include "MD5.h"
 #include <string.h>
