diff --color -Naur ./rasdaemon-0.8.0/configure.ac ./rasdaemon-0.8.0-patched/configure.ac
--- ./rasdaemon-0.8.0/configure.ac	2023-02-18 18:02:41.000000000 +0900
+++ ./rasdaemon-0.8.0-patched/configure.ac	2023-04-10 14:53:22.668527535 +0900
@@ -3,7 +3,7 @@
 AC_CANONICAL_TARGET
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADERS([config.h])
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([foreign])
 AC_PROG_CC
 AC_PROG_INSTALL
 LT_INIT
