From 65d2491083abbcd4e2ef53d488b1be38a3f5b34d Mon Sep 17 00:00:00 2001
From: Mike Kershaw / Dragorn <dragorn@kismetwireless.net>
Date: Tue, 15 Mar 2016 22:04:37 -0400
Subject: Hacky fix for GCC5

---
 spectool_gtk_widget.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/spectool_gtk_widget.h b/spectool_gtk_widget.h
index 1a77a2f..0caa7f7 100644
--- a/spectool_gtk_widget.h
+++ b/spectool_gtk_widget.h
@@ -196,16 +196,16 @@ void spectool_widget_link_channel(GtkWidget *widget, SpectoolChannelOpts *opts);
 gint spectool_widget_timeout(gpointer *data);
 
 /* Calculate the channel clicked in */
-inline int spectool_widget_find_chan_pt(SpectoolWidget *wwidget, int x, int y);
+extern inline int spectool_widget_find_chan_pt(SpectoolWidget *wwidget, int x, int y);
 
 void spectool_widget_context_channels(gpointer *aux);
 void spectool_widget_context_dbm(gpointer *aux);
 void spectool_widget_context_dbmlines(gpointer *aux);
 
 /* Color space conversion tools */
-inline void rgb_to_hsv(double r, double g, double b, 
+extern inline void rgb_to_hsv(double r, double g, double b, 
 					   double *h, double *s, double *v);
-inline void hsv_to_rgb(double *r, double *g, double *b, 
+extern inline void hsv_to_rgb(double *r, double *g, double *b, 
 					   double h, double s, double v);
 
 G_END_DECLS
-- 
2.14.5

