diff -Nur kernel.orig/nvidia-drm/nvidia-drm-helper.h kernel/nvidia-drm/nvidia-drm-helper.h
--- a/kernel-open/nvidia-drm/nvidia-drm-helper.h	2022-07-20 09:42:52.000000000 -0400
+++ b/kernel-open/nvidia-drm/nvidia-drm-helper.h	2022-09-07 20:36:36.725936810 -0400
@@ -109,6 +109,11 @@
  */
 #include <drm/drm_crtc.h>
 
+#include <linux/version.h>
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 0, 0))
+#include <drm/drm_framebuffer.h>
+#endif
+
 #if defined(drm_for_each_plane)
 #define nv_drm_for_each_plane(plane, dev) \
     drm_for_each_plane(plane, dev)
