diff -up qwt-6.1.2/qwtfunctions.pri.qt5 qwt-6.1.2/qwtfunctions.pri
--- qwt-6.1.2/qwtfunctions.pri.qt5	2014-12-11 08:13:13.511186492 -0600
+++ qwt-6.1.2/qwtfunctions.pri	2014-12-12 20:45:24.556261966 -0600
@@ -12,7 +12,12 @@
 defineReplace(qwtLibraryTarget) {
 
     unset(LIBRARY_NAME)
+
+    greaterThan(QT_MAJOR_VERSION, 4) {
+    LIBRARY_NAME = $${1}-qt$${QT_MAJOR_VERSION}
+    } else {
     LIBRARY_NAME = $$1
+    }
 
     mac:contains(QWT_CONFIG, QwtFramework) {
 
@@ -36,7 +41,11 @@ defineReplace(qwtLibraryTarget) {
 defineTest(qwtAddLibrary) {
 
     LIB_PATH = $$1
+    greaterThan(QT_MAJOR_VERSION, 4) {
+    LIB_NAME = $${2}-qt$${QT_MAJOR_VERSION}
+    } else {
     LIB_NAME = $$2
+    }
 
     mac:contains(QWT_CONFIG, QwtFramework) {
