--- a/I-Nex/i-nex/.src/Finfosys.class	2019-09-28 17:03:37.613473723 +0200
+++ b/I-Nex/i-nex/.src/Finfosys.class	2019-09-28 17:04:45.017472516 +0200
@@ -322,6 +322,7 @@
 Public Sub System()
   Dim hostname, arch, gcc, timezone, xorg_version, glx_version As String
   Dim SCREENFETCH_P As Process
+  Dim sLine2 As String
   SCREENFETCH_P = Exec ["" & ChkPrm.ChkExecDir("bash") & ""] For Write
   Write #SCREENFETCH_P, File.Load("Data/screenfetch-dev")
   SCREENFETCH_P.Wait
@@ -332,9 +333,9 @@
     Shell "timedatectl status | grep \"Time zone:\" | cut -d \":\" -f 2 | cut -d \"(\" -f 1" Wait To timezone
   Endif
   'Detect heap size
-  For Each sLine In Split(File.Load("/proc/self/maps"), "\n")
-      If InStr(sLine, "[heap]") Then
-        aScan = Scan(sLine, "*-* *")
+  For Each sLine2 In Split(File.Load("/proc/self/maps"), "\n")
+      If InStr(sLine2, "[heap]") Then
+        aScan = Scan(sLine2, "*-* *")
         Label176.Text = "Heap size is " & Format(Val("&H" & aScan[1] & "&") - Val("&H" & aScan[0] & "&"), "#,##0") & " bytes."
         Break
       Endif
