diff -rupN gnokii-0.6.29.orig/common/phones/pcsc.c gnokii-0.6.29new/common/phones/pcsc.c
--- gnokii-0.6.29.orig/common/phones/pcsc.c	2010-07-29 08:38:28.000000000 +1000
+++ gnokii-0.6.29/common/phones/pcsc.c	2010-07-29 08:51:29.000000000 +1000
@@ -199,8 +199,8 @@ static gn_error get_gn_error(PCSC_IOSTRU
 	case SCARD_W_UNPOWERED_CARD:
 	case SCARD_W_RESET_CARD:
 	case SCARD_W_REMOVED_CARD:
-	case SCARD_W_INSERTED_CARD:
-	 	return GN_ERR_SIMPROBLEM;
+	/* removed in pscc case SCARD_W_INSERTED_CARD:
+	 	return GN_ERR_SIMPROBLEM; */
 	case SCARD_E_NO_MEMORY:
 		return GN_ERR_MEMORYFULL;
 	case SCARD_F_WAITED_TOO_LONG:
