diff -Nur cairo-compmgr-0.3.0.orig//configure cairo-compmgr-0.3.0/configure
--- cairo-compmgr-0.3.0.orig//configure	2010-03-17 17:13:42.000000000 -0500
+++ cairo-compmgr-0.3.0/configure	2010-12-31 18:15:49.781205330 -0600
@@ -13373,7 +13373,7 @@
         cairo >= \$CAIRO_REQUIRED,
         pixman-1 >= \$PIXMAN_REQUIRED,
         gtk+-2.0 >= \$GTK_REQUIRED
-        vala-1.0 >= \$VALA_REQUIRED\""; } >&5
+        libvala-0.12 >= \$VALA_REQUIRED\""; } >&5
   ($PKG_CONFIG --exists --print-errors "xcomposite,
         xdamage,
         xext,
@@ -13382,7 +13382,7 @@
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
         gtk+-2.0 >= $GTK_REQUIRED
-        vala-1.0 >= $VALA_REQUIRED") 2>&5
+        libvala-0.12 >= $VALA_REQUIRED") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
@@ -13394,7 +13394,7 @@
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
         gtk+-2.0 >= $GTK_REQUIRED
-        vala-1.0 >= $VALA_REQUIRED" 2>/dev/null`
+        libvala-0.12 >= $VALA_REQUIRED" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -13415,7 +13415,7 @@
         cairo >= \$CAIRO_REQUIRED,
         pixman-1 >= \$PIXMAN_REQUIRED,
         gtk+-2.0 >= \$GTK_REQUIRED
-        vala-1.0 >= \$VALA_REQUIRED\""; } >&5
+        libvala-0.12 >= \$VALA_REQUIRED\""; } >&5
   ($PKG_CONFIG --exists --print-errors "xcomposite,
         xdamage,
         xext,
@@ -13424,7 +13424,7 @@
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
         gtk+-2.0 >= $GTK_REQUIRED
-        vala-1.0 >= $VALA_REQUIRED") 2>&5
+        libvala-0.12 >= $VALA_REQUIRED") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
@@ -13436,7 +13436,7 @@
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
         gtk+-2.0 >= $GTK_REQUIRED
-        vala-1.0 >= $VALA_REQUIRED" 2>/dev/null`
+        libvala-0.12 >= $VALA_REQUIRED" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -13463,7 +13463,7 @@
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
         gtk+-2.0 >= $GTK_REQUIRED
-        vala-1.0 >= $VALA_REQUIRED"`
+        libvala-0.12 >= $VALA_REQUIRED"`
         else
 	        CAIRO_COMPMGR_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "xcomposite,
         xdamage,
@@ -13473,7 +13473,7 @@
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
         gtk+-2.0 >= $GTK_REQUIRED
-        vala-1.0 >= $VALA_REQUIRED"`
+        libvala-0.12 >= $VALA_REQUIRED"`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$CAIRO_COMPMGR_PKG_ERRORS" >&5
@@ -13486,7 +13486,7 @@
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
         gtk+-2.0 >= $GTK_REQUIRED
-        vala-1.0 >= $VALA_REQUIRED) were not met:
+        libvala-0.12 >= $VALA_REQUIRED) were not met:
 
 $CAIRO_COMPMGR_PKG_ERRORS
 
