diff -u -r sudosh2-1.0.4-orig/src/Makefile.am sudosh2-1.0.4/src/Makefile.am
--- sudosh2-1.0.4-orig/src/Makefile.am	2010-07-09 21:37:01.000000000 +0100
+++ sudosh2-1.0.4/src/Makefile.am	2016-05-29 11:51:19.902051323 +0100
@@ -7,7 +7,7 @@
 sysconfdir = @sysconfdir@
 
 install-sudosh.conf:
-	test -z "$(sysconfdir)" || $(mkdir_p) "$(DESTDIR)$(sysconfdir)"
-	test -f $(sysconfdir)/sudosh.conf || $(INSTALL) -o 0 -g 0 -m 0444 '$(srcdir)/sudosh.conf' '$(sysconfdir)/sudosh.conf'
+	$(mkdir_p) "$(DESTDIR)$(sysconfdir)"
+	$(INSTALL) -o 0 -g 0 -m 0444 '$(srcdir)/sudosh.conf' '$(DESTDIR)$(sysconfdir)/sudosh.conf'
 
 install: install-am install-sudosh.conf
