--- xfce4-mailwatch-plugin-1.1.0.orig/libmailwatch-core/mailwatch-net-conn.c	2008-08-30 10:57:36.000000000 +0200
+++ xfce4-mailwatch-plugin-1.1.0/libmailwatch-core/mailwatch-net-conn.c	2013-06-10 14:18:36.209263440 +0200
@@ -621,9 +621,11 @@
                            net_conn->gt_creds);
     gnutls_transport_set_ptr(net_conn->gt_session,
                              (gnutls_transport_ptr_t)net_conn->fd);
+#if GNUTLS_VERSION_NUMBER < 0x020c00
     if(fcntl(net_conn->fd, F_GETFL) & O_NONBLOCK)
         gnutls_transport_set_lowat(net_conn->gt_session, 0);
-    
+#endif
+
     if(!xfce_mailwatch_net_conn_tls_handshake(net_conn, error)) {
 #if 0
         gnutls_bye(net_conn->gt_session, GNUTLS_SHUT_RDWR);
