--- xfce4-mailwatch-plugin-1.1.0.orig/libmailwatch-core/mailwatch-mailbox-mbox.c	2008-09-11 06:31:50.000000000 +0200
+++ xfce4-mailwatch-plugin-1.1.0/libmailwatch-core/mailwatch-mailbox-mbox.c	2013-08-12 13:38:07.613031827 +0200
@@ -347,6 +347,7 @@
             g_source_remove( mbox->check_id );
         mbox->check_id = g_timeout_add( mbox->interval * 1000, mbox_check_mail_timeout, mbox );
     }
+    mbox->interval = val;
 }
     
 static GtkContainer *
