Author: Enrico Tassi <gareuselesinge@debian.org>
Description: fix regex to get auth options from curl.h
Index: lua-curl-0.3.0/Makefile
===================================================================
--- lua-curl-0.3.0.orig/Makefile	2008-10-25 17:26:26.000000000 +0200
+++ lua-curl-0.3.0/Makefile	2012-05-30 21:57:54.000000000 +0200
@@ -119,8 +119,8 @@
 		> curl_netrcopt.h
 
 curl_authopt.h:$(HEADER)
-	$(H)cat $(HEADER) | grep "CURLAUTH_" | \
-		sed "s/#define *CURL/{\"/" | sed "s/ *\/\*.*\*\///" | \
+	$(H)cat $(HEADER) | grep "define *CURLAUTH_" | \
+		sed "s/#define *CURL/{\"/" | sed "s/ *\/\*.*//" | \
 		sed "s/ /\",/" | sed "s/$$/},/" > curl_authopt.h
 
 curl_ftpauthopt.h:$(HEADER)
