--- configure	2014-04-16 07:20:43.000000000 +0300
+++ ../../icedtea-2.4.7/configure	2014-05-17 08:33:36.000000000 +0300
@@ -9170,7 +9170,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: not specified" >&5
 $as_echo "not specified" >&6; }
     if test "x${enable_bootstrap}" = "xyes"; then
-      BOOTSTRAP_VMS="/usr/lib/jvm/java-gcj /usr/lib/jvm/gcj-jdk /usr/lib/jvm/cacao";
+      BOOTSTRAP_VMS="/usr/lib/jvm/java-gcj /usr/lib/jvm/gcj-jdk /usr/lib/jvm/cacao /usr/lib/jvm";
     fi
     ICEDTEA6_VMS="/usr/lib/jvm/icedtea-6 /usr/lib/jvm/icedtea6 /usr/lib/jvm/java-6-openjdk
     		  /usr/lib/jvm/java-1.6.0-openjdk.x86_64 /usr/lib64/jvm/java-1.6.0-openjdk
