--- Make-config.sh	2010-03-26 07:21:14.000000000 +0000
+++ Make-config.sh.new	2010-03-26 07:24:34.000000000 +0000
@@ -114,7 +114,7 @@
 ### Man pages (trick to get around .man extension)
 
 %.1: %.man
-        cp $< $@
+	cp $< $@
 CLEANFILES = *.1
 man_MANS = rbox.1 qhull.1
 
@@ -195,7 +195,10 @@
   geom.c \
   poly.c \
   qset.c \
-  mem.c
+  mem.c \
+  usermem.c \
+  userprintf.c \
+  rboxlib.c
 
 # how:
 libqhull_la_LDFLAGS = -version-info 4:0:0 -lm
@@ -233,7 +236,9 @@
   merge.h \
   libqhull.h  \
   qset.h \
-  user.h
+  user.h \
+  random.h \
+  qhull.h
 
 
 ### Example programs
@@ -246,10 +251,9 @@
 examples_DATA = \
   user_eg.c \
   user_eg2.c \
-  qhull_interface.cpp \
-  Makefile.txt \
+  Makefile \
   Make-config.sh \
-  MBorland
+  Mborland
 
 doc_DATA = Changes.txt \
     index.htm \
