diff -Naur xdemorse-3.0.1/files/xdemorse.desktop xdemorse-3.0.1.patched/files/xdemorse.desktop
--- xdemorse-3.0.1/files/xdemorse.desktop	2014-03-06 00:47:02.000000000 -0500
+++ xdemorse-3.0.1.patched/files/xdemorse.desktop	2020-01-17 11:03:18.745024631 -0500
@@ -4,6 +4,6 @@
 Comment=Decode Morse signals to Text
 Exec=xdemorse
 Type=Application
-Icon=xdemorse.xpm
-Categories=Audio
+Icon=/usr/share/pixmaps/xdemorse.xpm
+Categories=AudioVideo;Audio;HamRadio;
 Keywords=hamradio;amateur;radio;morse;code;gtk;signal
