--- a/src/runtime/x86-64-linux-os.h.ORIG 2017-08-31 11:40:51.000000000 +0200
+++ b/src/runtime/x86-64-linux-os.h 2017-08-31 18:56:10.447100144 +0200
@@ -1,7 +1,9 @@
 #ifndef _X86_64_LINUX_OS_H
 #define _X86_64_LINUX_OS_H

-typedef struct ucontext os_context_t;
+#include <sys/ucontext.h>
+
+typedef struct ucontext_t os_context_t;
 typedef long os_context_register_t;

 static inline os_context_t *arch_os_get_context(void **void_context)
