--- a/lib/MusicBrainz/DiscID.pm	2013-10-20 22:04:53.340046220 +0200
+++ b/lib/MusicBrainz/DiscID.pm	2013-10-20 22:06:21.580047858 +0200
@@ -156,7 +156,7 @@
 Construct a new DiscID object.
 
 As an optional argument the name of the device to read the ID from may 
-be given. If you donÔt specify a device here you can later read the ID with 
+be given. If you don't specify a device here you can later read the ID with 
 the read method.
 
 =item $discid->error_msg()
