diff -Naur unac-1.8.0.orig/unaccent-local-map.perl unac-1.8.0.orig.patched/unaccent-local-map.perl
--- unac-1.8.0.orig/unaccent-local-map.perl	2003-10-13 06:24:25.000000000 -0400
+++ unac-1.8.0.orig.patched/unaccent-local-map.perl	2021-08-20 14:13:53.224474419 -0400
@@ -23,5 +23,12 @@
 %::localmap = (
     "LATIN SMALL LETTER AE" => [ "LATIN SMALL LETTER A", "LATIN SMALL LETTER E" ],
     "LATIN CAPITAL LETTER AE" => [ "LATIN CAPITAL LETTER A", "LATIN CAPITAL LETTER E" ],
+    "LATIN SMALL LIGATURE OE" => [ "LATIN SMALL LETTER O", "LATIN SMALL LETTER E" ],
+    "LATIN CAPITAL LIGATURE OE" => [ "LATIN CAPITAL LETTER O", "LATIN CAPITAL LETTER E" ],
+    "LATIN SMALL LETTER SHARP S" => [ "LATIN SMALL LETTER S", "LATIN SMALL LETTER S" ],
+    "LATIN SMALL LETTER HV" => [ "LATIN SMALL LETTER H", "LATIN SMALL LETTER V" ],
 		 );
 
+# 20210820 bkw: Unicode standard 4.0.0 doesn't define these:
+#    "LATIN CAPITAL LETTER HV" => [ "LATIN CAPITAL LETTER H", "LATIN CAPITAL LETTER V" ],
+#    "LATIN CAPITAL LETTER SHARP S" => [ "LATIN CAPITAL LETTER S", "LATIN CAPITAL LETTER S" ],
