--- medit-0.10.4/config-dirs.h.in	2010-04-29 03:43:23.000000000 -0300
+++ medit-0.10.4_patched/config-dirs.h.in	2010-06-27 22:04:27.000000000 -0300
@@ -6,6 +6,6 @@
 #define MOO_DATA_DIR "@MOO_DATA_DIR_ABS@"
 #define MOO_LIB_DIR "@MOO_LIB_DIR_ABS@"
 #define MOO_LOCALE_DIR "@LOCALEDIR_ABS@"
-#define MOO_HELP_DIR "@MOO_HELP_DIR_ABS@"
+#define MOO_HELP_DIR "@MOO_HELP_DIR@"
 
 #endif /* CONFIG_DIRS_H */
