--- share/Makefile.orig	2007-02-11 09:56:05.000000000 -0500
+++ share/Makefile	2007-02-11 09:56:41.000000000 -0500
@@ -225,8 +225,8 @@
 	-if test -d templates; then rm -f tmpl.tar.gz; fi
 
 install-data-local: tmpl.tar.gz
-	tar zxvf tmpl.tar.gz -C $(datadir)/motor/
-	-find $(datadir)/motor/ -name CVS -type d -exec rm -rf {} \;
+	tar zxvf tmpl.tar.gz -C ${DESTDIR}/$(datadir)/motor/
+	-find ${DESTDIR}/$(datadir)/motor/ -name CVS -type d -exec rm -rf {} \;
 	if test -d templates; then rm -f tmpl.tar.gz; fi
 
 uninstall-local:
