From f5bab226a31bdeab53f5fe7abf48cf0311d75cc1 Mon Sep 17 00:00:00 2001
From: Zhu Qun-Ying <zhu.qunying@gmail.com>
Date: Fri, 7 Mar 2025 23:02:02 -0800
Subject: [PATCH 1/4] gentoo

---
 Makefile | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/Makefile b/Makefile
index c38bb22..46030e8 100644
--- a/Makefile
+++ b/Makefile
@@ -25,13 +25,13 @@ mintk = 8.4
 
 ### you need to localize the paths on these lines
 
-WISH = /usr/local/bin/wish8.4
+WISH = /usr/bin/wish
 #WISH = /usr/sww/tcl-8.4/bin/wish
 #WISH = /private/share/bin/wish8.4
 
 # the executable `tkman' is placed in BINDIR
 # this should be a directory that's in your bin PATH
-BINDIR = /usr/local/bin
+BINDIR = ${DESTDIR}/usr/bin
 #BINDIR = /private/share/bin
 
 
@@ -203,7 +203,7 @@ shortnames = 0
 # BY DEFAULT OFF
 glimpse = ""
 # Glimpse on, with Boolean scope set to entire file
-glimpse = "glimpse -W"
+#glimpse = "glimpse -W"
 # give the full path, if you'd like
 #glimpse = /usr/sww/bin/glimpse
 # variations (refer to the Glimpse manual page)
@@ -228,11 +228,11 @@ glimpse = "glimpse -W"
 #glimpseindex = "glimpseindex -o -B"
 # To shorten most index build times by incrementally building index when possible.
 # For me, this drops indexing time from ~25 minutes to ~3 minutes, depending
-glimpseindex = "glimpseindex -o -B -f"
+#glimpseindex = "glimpseindex -o -B -f"
 # -M 8 gives 8MB of main memory instead of default 2, which speeds up indexing, maybe
 #glimpseindex = "glimpseindex -o -B -f -M 8"
 # if you don't want give the power to index, disable glimpseindex here
-#glimpseindex = ""
+glimpseindex = ""
 #glimpseindex = "/usr/sww/bin/glimpseindex -o -B"
 # compressed manual pages are handled automatically; don't add -z to above
 
@@ -289,9 +289,9 @@ vdiff = diff
 rman = rman
 #rman = /usr/local/bin/rman
 
-apropos = {man -k}
+#apropos = {man -k}
 # Linux uses a shell script instead:
-#apropos = apropos
+apropos = apropos
 # if don't have apropos (maybe on Windoze)
 #apropos = ""
 
-- 
2.46.3

