From d7c3cd90d72b47ce766af7da1c8e555c103def91 Mon Sep 17 00:00:00 2001
From: Zhu Qun-Ying <zhu.qunying@gmail.com>
Date: Fri, 7 Mar 2025 23:03:18 -0800
Subject: [PATCH 3/4] openbsd-gui_tcl.patch

---
 gui.tcl | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gui.tcl b/gui.tcl
index 525f437..200ba53 100644
--- a/gui.tcl
+++ b/gui.tcl
@@ -746,6 +746,7 @@ proc manManTabSet {w} {
 	set repstr "ETRNIOASM          aaabcdeeeefghiijklmnnooprrsttuy"; set replen [string length $repstr]
 
 	set rm [expr {[winfo width $t]-2*[$t cget -padx]-10}]
+	if {$rm <= 150} { set rm 151 }
 	$t tag configure info -tabs [list 0.3i $rm right]
 	$t tag configure census -tabs [list [expr {$rm-150}] right $rm right]
 	$t tag configure high -tabs [list $rm right]
-- 
2.46.3

