From ed02c6538a758b92efb433f3d79746dd5a38c989 Mon Sep 17 00:00:00 2001
From: Zhu Qun-Ying <zhu.qunying@gmail.com>
Date: Sat, 8 Mar 2025 21:50:30 -0800
Subject: [PATCH 4/4] use unicode arrow characters

---
 gui.tcl | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gui.tcl b/gui.tcl
index 200ba53..349d644 100644
--- a/gui.tcl
+++ b/gui.tcl
@@ -480,7 +480,7 @@ if 0 {
 		searchboxNext search $t $wi 0
 	"
 	set manx(hitlist$t) {}
-	button $w.search.next -text "\xdf" -font guisymbol -command "
+	button $w.search.next -text "⇓" -command "
 		incr stat(page-regexp-next)
 		# just before start to page through hits, open all sections with hits.
 		# subsequently abide by user's outlining changes
@@ -491,13 +491,13 @@ if 0 {
 		} else {searchboxNext search $t $wi}
 		catch {$t see hit}
 	"
-	button $w.search.prev -text "\xdd" -font guisymbol -command "
+	button $w.search.prev -text "⇑" -command "
 		incr stat(page-regexp-prev)
 		searchboxPrev search $t $wi
 		manOutline $t 0 \$manx(hitlist$t); set manx(hitlist$t) {}
 		catch {$t see hit}
 	"
-	menubutton [set mb $w.search.tags] -text "\xdf" -font guisymbol -menu [set m $mb.m]; menu $m -tearoff no
+	menubutton [set mb $w.search.tags] -text "⇓" -menu [set m $mb.m]; menu $m -tearoff no
 	foreach {name val} $manx(searchtags) {
 		$m add command -label $name -command "set manx(search,string$w) \"TAG:$val\"; $w.search.t icursor end"
 #"$w.search.t insert insert TAG:$val"
-- 
2.46.3

