diff -rupN AlephOne-20140104/Source_Files/FFmpeg/SDL_ffmpeg.c AlephOne-20140104.new/Source_Files/FFmpeg/SDL_ffmpeg.c
--- AlephOne-20140104/Source_Files/FFmpeg/SDL_ffmpeg.c	2013-07-05 02:33:29.000000000 +0200
+++ AlephOne-20140104.new/Source_Files/FFmpeg/SDL_ffmpeg.c	2014-03-16 01:53:19.327213212 +0100
@@ -48,6 +48,11 @@ extern "C"
 }
 #endif
 
+/* set magic number for 1 second of 48khz 32bit audio (removed from upstream) */
+#ifndef AVCODEC_MAX_AUDIO_FRAME_SIZE
+#define AVCODEC_MAX_AUDIO_FRAME_SIZE 192000
+#endif
+
 #include "SDL_ffmpeg.h"
 
 #ifdef MSVC
diff -rupN AlephOne-20140104/Source_Files/FFmpeg/Movie.cpp AlephOne-20140104.new/Source_Files/FFmpeg/Movie.cpp
--- AlephOne-20140104/Source_Files/FFmpeg/Movie.cpp	2014-01-04 03:19:57.000000000 +0100
+++ AlephOne-20140104.new/Source_Files/FFmpeg/Movie.cpp	2014-03-16 01:43:32.030011019 +0100
@@ -491,7 +491,7 @@ bool Movie::Setup()
         
         // tuning options
         int aq = graphics_preferences->movie_export_audio_quality;
-        audio_stream->codec->global_quality = audio_stream->quality = FF_QP2LAMBDA * (aq / 10);
+        audio_stream->codec->global_quality = FF_QP2LAMBDA * (aq / 10);
         audio_stream->codec->flags |= CODEC_FLAG_QSCALE;
         
         // find correct sample format
