diff -Naur audiere/configure.in audiere.tpg/configure.in
--- audiere/configure.in	2004-11-07 23:41:37.000000000 +0000
+++ audiere.tpg/configure.in	2008-06-03 17:52:17.000000000 +0000
@@ -174,7 +174,7 @@
     AC_DEFINE(NO_OGG))
 AM_CONDITIONAL(HAVE_OGG, test "x$HAVE_OGG" = "xtrue")
 
-AC_CHECK_HEADER(speex.h,
+AC_CHECK_HEADER(speex/speex.h,
     HAVE_SPEEX=true
     LIBS="-lspeex $LIBS"
     EXTRA_LIBS="-lspeex $EXTRA_LIBS"
diff -Naur audiere/src/speexfile/speexfile.cpp audiere.tpg/src/speexfile/speexfile.cpp
--- audiere/src/speexfile/speexfile.cpp	2004-01-11 03:36:25.000000000 +0000
+++ audiere.tpg/src/speexfile/speexfile.cpp	2008-06-03 17:51:38.000000000 +0000
@@ -871,7 +871,7 @@
 void *speexfile::header_to_decoder ( SpeexHeader *header, int enh_enabled, int *frame_size, int *rate, int *nframes, int forceMode, int *channels, SpeexStereoState *stereo )
 {
     void *st;
-    SpeexMode *mode;
+    const SpeexMode *mode;
     SpeexCallback callback;
     int modeID;
 
diff -Naur audiere/src/speexfile/speexfile.h audiere.tpg/src/speexfile/speexfile.h
--- audiere/src/speexfile/speexfile.h	2004-01-11 03:32:14.000000000 +0000
+++ audiere.tpg/src/speexfile/speexfile.h	2008-06-03 17:51:06.000000000 +0000
@@ -13,10 +13,10 @@
 //#define SL_READER   // uncomment to provide own reader
 
 extern "C" {
-#include <speex.h>
-#include <speex_header.h>
-#include <speex_stereo.h>
-#include <speex_callbacks.h>
+#include <speex/speex.h>
+#include <speex/speex_header.h>
+#include <speex/speex_stereo.h>
+#include <speex/speex_callbacks.h>
 //#include <misc.h>
 #include <ogg/ogg.h>
 }
