diff -Naur audiere/src/audiere.h audiere.tpg/src/audiere.h
--- audiere/src/audiere.h	2004-11-28 01:41:59.000000000 +0000
+++ audiere.tpg/src/audiere.h	2008-06-03 17:39:43.000000000 +0000
@@ -29,6 +29,13 @@
 #include <vector>
 #include <string>
 
+/* Headers for gcc-4.3 */
+
+/* strlen, strchr, etc. */
+#include <cstring>
+/* getenv, atexit, etc. */
+#include <cstdlib>
+
 #ifdef _MSC_VER
 #pragma warning(disable : 4786)
 #endif
diff -Naur audiere/src/debug.h audiere.tpg/src/debug.h
--- audiere/src/debug.h	2004-04-09 09:33:50.000000000 +0000
+++ audiere.tpg/src/debug.h	2008-06-03 17:39:43.000000000 +0000
@@ -6,6 +6,9 @@
 #include <stdio.h>
 #include <string>
 
+/* Headers for gcc-4.3 */
+#include <cstring>
+#include <cstdlib>
 
 namespace audiere {
 
diff -Naur audiere/src/utility.cpp b/src/utility.cpp
index f49c028..28fefc8 100644
--- audiere/src/utility.cpp	2004-04-09 09:33:50.000000000 +0000
+++ audiere.tpg/src/utility.cpp	2008-06-03 17:39:43.000000000 +0000
@@ -2,7 +2,7 @@
 #pragma warning(disable : 4786)
 #endif
 
-
+#include <cstdio>
 #include <ctype.h>
 #include "utility.h"
 #include "internal.h"
