--- assimp--3.0.1270-sdk/CMakeLists.original.txt	2014-03-23 08:22:28.393687604 +0200
+++ assimp--3.0.1270-sdk/CMakeLists.txt	2014-03-23 08:23:05.041229447 +0200
@@ -32,13 +32,13 @@
 # source tree. During an out-of-source build, however, do not litter this
 # directory, since that is probably what the user wanted to avoid.
 IF ( CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR )
-	SET( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_HOME_DIRECTORY}/lib )
-	SET( CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_HOME_DIRECTORY}/lib )
+	SET( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_HOME_DIRECTORY}/lib64 )
+	SET( CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_HOME_DIRECTORY}/lib64 )
 	SET( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_HOME_DIRECTORY}/bin )
 ENDIF ( CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR )
 
 # Cache these to allow the user to override them manually.
-SET( LIB_INSTALL_DIR "lib" CACHE PATH
+SET( LIB_INSTALL_DIR "lib64" CACHE PATH
 	"Path the built library files are installed to." )
 SET( INCLUDE_INSTALL_DIR "include" CACHE PATH
 	"Path the header files are installed to." )
