diff -ur code.old/AMF/AMFImporter.hpp code/AMF/AMFImporter.hpp
--- code.old/AMF/AMFImporter.hpp	2025-06-14 16:33:11.800468946 +0900
+++ code/AMF/AMFImporter.hpp	2025-06-14 16:33:31.484791445 +0900
@@ -62,6 +62,7 @@
 
 // Header files, stdlib.
 #include <set>
+#include <cstdint>
 
 namespace Assimp {
 
diff -ur code.old/AMF/AMFImporter_Node.hpp code/AMF/AMFImporter_Node.hpp
--- code.old/AMF/AMFImporter_Node.hpp	2025-06-14 16:33:11.800468946 +0900
+++ code/AMF/AMFImporter_Node.hpp	2025-06-14 16:33:40.064933012 +0900
@@ -54,6 +54,7 @@
 #include <list>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 // Header files, Assimp.
 #include "assimp/types.h"
diff -ur code.old/Collada/ColladaHelper.h code/Collada/ColladaHelper.h
--- code.old/Collada/ColladaHelper.h	2025-06-14 16:33:11.803468995 +0900
+++ code/Collada/ColladaHelper.h	2025-06-14 16:34:51.575133548 +0900
@@ -51,6 +51,7 @@
 #include <assimp/light.h>
 #include <assimp/mesh.h>
 #include <assimp/material.h>
+#include <cstdint>
 
 struct aiMaterial;
 
diff -ur code.old/Common/IFF.h code/Common/IFF.h
--- code.old/Common/IFF.h	2025-06-14 16:33:11.804469012 +0900
+++ code/Common/IFF.h	2025-06-14 16:34:33.248822639 +0900
@@ -6,6 +6,7 @@
 #define AI_IFF_H_INCLUDED
 
 #include <assimp/ByteSwapper.h>
+#include <cstdint>
 
 namespace Assimp    {
 namespace IFF       {
diff -ur code.old/FBX/FBXDocument.h code/FBX/FBXDocument.h
--- code.old/FBX/FBXDocument.h	2025-06-14 16:33:11.805469028 +0900
+++ code/FBX/FBXDocument.h	2025-06-14 16:35:01.614304714 +0900
@@ -51,6 +51,7 @@
 #include <assimp/mesh.h>
 #include "FBXProperties.h"
 #include "FBXParser.h"
+#include <cstdint>
 
 #define _AI_CONCAT(a,b)  a ## b
 #define  AI_CONCAT(a,b)  _AI_CONCAT(a,b)
diff -ur code.old/FBX/FBXExportProperty.h code/FBX/FBXExportProperty.h
--- code.old/FBX/FBXExportProperty.h	2025-06-14 16:33:11.805469028 +0900
+++ code/FBX/FBXExportProperty.h	2025-06-14 16:34:55.867206655 +0900
@@ -54,6 +54,7 @@
 #include <vector>
 #include <ostream>
 #include <type_traits> // is_void
+#include <cstdint>
 
 namespace Assimp {
 namespace FBX {
diff -ur code.old/FBX/FBXUtil.h code/FBX/FBXUtil.h
--- code.old/FBX/FBXUtil.h	2025-06-14 16:33:11.806469044 +0900
+++ code/FBX/FBXUtil.h	2025-06-14 16:35:04.593355617 +0900
@@ -49,6 +49,7 @@
 #include "FBXCompileConfig.h"
 #include "FBXTokenizer.h"
 #include <stdint.h>
+#include <cstdint>
 
 namespace Assimp {
 namespace FBX {
diff -ur code.old/HMP/HMPFileData.h code/HMP/HMPFileData.h
--- code.old/HMP/HMPFileData.h	2025-06-14 16:33:11.806469044 +0900
+++ code/HMP/HMPFileData.h	2025-06-14 16:35:49.747132942 +0900
@@ -48,6 +48,7 @@
 
 #include <assimp/Compiler/pushpack1.h>
 #include <stdint.h>
+#include <cstdint>
 
 // to make it easier for us, we test the magic word against both "endianesses"
 #define AI_HMP_MAGIC_NUMBER_BE_4    AI_MAKE_MAGIC("HMP4")
diff -ur code.old/LWO/LWOLoader.h code/LWO/LWOLoader.h
--- code.old/LWO/LWOLoader.h	2025-06-14 16:33:11.810469109 +0900
+++ code/LWO/LWOLoader.h	2025-06-14 16:34:48.021073093 +0900
@@ -52,6 +52,7 @@
 #include <assimp/BaseImporter.h>
 
 #include <map>
+#include <cstdint>
 
 struct aiTexture;
 struct aiNode;
diff -ur code.old/MD2/MD2FileData.h code/MD2/MD2FileData.h
--- code.old/MD2/MD2FileData.h	2025-06-14 16:33:11.810469109 +0900
+++ code/MD2/MD2FileData.h	2025-06-14 16:35:43.018016448 +0900
@@ -49,6 +49,7 @@
 
 #include <assimp/types.h>
 #include <stdint.h>
+#include <cstdint>
 
 #include <assimp/Compiler/pushpack1.h>
 
diff -ur code.old/MD2/MD2Loader.h code/MD2/MD2Loader.h
--- code.old/MD2/MD2Loader.h	2025-06-14 16:33:11.810469109 +0900
+++ code/MD2/MD2Loader.h	2025-06-14 16:35:46.490076530 +0900
@@ -50,6 +50,7 @@
 #include <assimp/types.h>
 #include <assimp/ByteSwapper.h>
 #include "MD2FileData.h"
+#include <cstdint>
 
 struct aiNode;
 
diff -ur code.old/MDC/MDCFileData.h code/MDC/MDCFileData.h
--- code.old/MDC/MDCFileData.h	2025-06-14 16:33:11.811469126 +0900
+++ code/MDC/MDCFileData.h	2025-06-14 16:35:53.468197452 +0900
@@ -57,6 +57,7 @@
 
 #include <assimp/Compiler/pushpack1.h>
 #include <stdint.h>
+#include <cstdint>
 
 namespace Assimp {
 namespace MDC {
diff -ur code.old/MDL/MDLFileData.h code/MDL/MDLFileData.h
--- code.old/MDL/MDLFileData.h	2025-06-14 16:33:11.811469126 +0900
+++ code/MDL/MDLFileData.h	2025-06-14 16:35:59.158296221 +0900
@@ -60,6 +60,7 @@
 #include <assimp/ByteSwapper.h>
 #include <stdint.h>
 #include <vector>
+#include <cstdint>
 
 struct aiMaterial;
 
diff -ur code.old/MMD/MMDPmdParser.h code/MMD/MMDPmdParser.h
--- code.old/MMD/MMDPmdParser.h	2025-06-14 16:33:11.811469126 +0900
+++ code/MMD/MMDPmdParser.h	2025-06-14 16:35:08.105415692 +0900
@@ -47,6 +47,7 @@
 #include <iostream>
 #include <fstream>
 #include "MMDCpp14.h"
+#include <cstdint>
 
 namespace pmd
 {
diff -ur code.old/MMD/MMDPmxParser.h code/MMD/MMDPmxParser.h
--- code.old/MMD/MMDPmxParser.h	2025-06-14 16:33:11.811469126 +0900
+++ code/MMD/MMDPmxParser.h	2025-06-14 16:35:15.902549307 +0900
@@ -47,6 +47,7 @@
 #include <fstream>
 #include <memory>
 #include "MMDCpp14.h"
+#include <cstdint>
 
 namespace pmx
 {
diff -ur code.old/MMD/MMDVmdParser.h code/MMD/MMDVmdParser.h
--- code.old/MMD/MMDVmdParser.h	2025-06-14 16:33:11.811469126 +0900
+++ code/MMD/MMDVmdParser.h	2025-06-14 16:35:11.989482209 +0900
@@ -48,6 +48,7 @@
 #include <fstream>
 #include <ostream>
 #include "MMDCpp14.h"
+#include <cstdint>
 
 namespace vmd
 {
diff -ur code.old/NDO/NDOLoader.h code/NDO/NDOLoader.h
--- code.old/NDO/NDOLoader.h	2025-06-14 16:33:11.812469142 +0900
+++ code/NDO/NDOLoader.h	2025-06-14 16:34:39.594930069 +0900
@@ -49,6 +49,7 @@
 #include <stdint.h>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 
 struct aiImporterDesc;
diff -ur code.old/Ogre/OgreBinarySerializer.h code/Ogre/OgreBinarySerializer.h
--- code.old/Ogre/OgreBinarySerializer.h	2025-06-14 16:33:11.813469158 +0900
+++ code/Ogre/OgreBinarySerializer.h	2025-06-14 16:35:56.385248068 +0900
@@ -47,6 +47,7 @@
 
 #include "OgreStructs.h"
 #include <assimp/StreamReader.h>
+#include <cstdint>
 
 namespace Assimp
 {
diff -ur code.old/X3D/FIReader.hpp code/X3D/FIReader.hpp
--- code.old/X3D/FIReader.hpp	2025-06-14 16:33:11.817469223 +0900
+++ code/X3D/FIReader.hpp	2025-06-14 16:33:42.289969819 +0900
@@ -56,7 +56,7 @@
 #include <cwchar>
 #include <vector>
 //#include <stdio.h>
-//#include <cstdint>
+#include <cstdint>
 #ifdef ASSIMP_USE_HUNTER
 #  include <irrXML/irrXML.h>
 #else
diff -ur code.old/glTF/glTFAsset.h code/glTF/glTFAsset.h
--- code.old/glTF/glTFAsset.h	2025-06-14 16:33:11.818469240 +0900
+++ code/glTF/glTFAsset.h	2025-06-14 16:35:31.092810543 +0900
@@ -60,6 +60,7 @@
 #include <vector>
 #include <algorithm>
 #include <stdexcept>
+#include <cstdint>
 
 #define RAPIDJSON_HAS_STDSTRING 1
 #include <rapidjson/rapidjson.h>
diff -ur code.old/glTF/glTFCommon.h code/glTF/glTFCommon.h
--- code.old/glTF/glTFCommon.h	2025-06-14 16:33:11.819469256 +0900
+++ code/glTF/glTFCommon.h	2025-06-14 16:35:34.826874941 +0900
@@ -51,6 +51,7 @@
 #include <vector>
 #include <algorithm>
 #include <stdexcept>
+#include <cstdint>
 
 #define RAPIDJSON_HAS_STDSTRING 1
 #include <rapidjson/rapidjson.h>
diff -ur code.old/glTF2/glTF2Asset.h code/glTF2/glTF2Asset.h
--- code.old/glTF2/glTF2Asset.h	2025-06-14 16:33:11.819469256 +0900
+++ code/glTF2/glTF2Asset.h	2025-06-14 16:35:39.648958206 +0900
@@ -61,6 +61,7 @@
 #include <vector>
 #include <algorithm>
 #include <stdexcept>
+#include <cstdint>
 
 #define RAPIDJSON_HAS_STDSTRING 1
 #include <rapidjson/rapidjson.h>
