diff -Naur mumble-1.2.16/src/mumble/TextToSpeech_unix.cpp mumble-1.2.16-new/src/mumble/TextToSpeech_unix.cpp
--- mumble-1.2.16/src/mumble/TextToSpeech_unix.cpp	2016-05-05 07:56:03.000000000 -0700
+++ mumble-1.2.16-new/src/mumble/TextToSpeech_unix.cpp	2016-05-15 21:58:16.605311547 -0700
@@ -33,7 +33,7 @@
 #include "TextToSpeech.h"
 
 #ifdef USE_SPEECHD
-#include <libspeechd.h>
+#include </usr/include/speech-dispatcher/libspeechd.h>
 #endif
 
 #include "Global.h"
