diff -rupN tiled-qt-0.9.0.old/src/tiled/tiled.pro tiled-qt-0.9.0/src/tiled/tiled.pro
--- tiled-qt-0.9.0.old/src/tiled/tiled.pro	2013-01-27 14:35:01.000000000 -0800
+++ tiled-qt-0.9.0/src/tiled/tiled.pro	2013-02-05 20:09:03.592075258 -0800
@@ -278,7 +278,7 @@ desktopfile.path = $${PREFIX}/share/appl
 desktopfile.files += ../../docs/tiled.desktop
 INSTALLS += desktopfile
 
-manpage.path = $${PREFIX}/share/man/man1/
+manpage.path = $${PREFIX}/man/man1/
 manpage.files += ../../docs/tiled.1
 INSTALLS += manpage
 
diff -rupN tiled-qt-0.9.0.old/src/tmxrasterizer/tmxrasterizer.pro tiled-qt-0.9.0/src/tmxrasterizer/tmxrasterizer.pro
--- tiled-qt-0.9.0.old/src/tmxrasterizer/tmxrasterizer.pro	2013-01-27 14:35:01.000000000 -0800
+++ tiled-qt-0.9.0/src/tmxrasterizer/tmxrasterizer.pro	2013-02-05 20:16:54.903093428 -0800
@@ -39,6 +39,6 @@ SOURCES += main.cpp \
 
 HEADERS += tmxrasterizer.h
 
-manpage.path = $${PREFIX}/share/man/man1/
+manpage.path = $${PREFIX}/man/man1/
 manpage.files += ../../docs/tmxrasterizer.1
 INSTALLS += manpage
diff -rupN tiled-qt-0.9.0.old/src/tmxviewer/tmxviewer.pro tiled-qt-0.9.0/src/tmxviewer/tmxviewer.pro
--- tiled-qt-0.9.0.old/src/tmxviewer/tmxviewer.pro	2013-01-27 14:35:01.000000000 -0800
+++ tiled-qt-0.9.0/src/tmxviewer/tmxviewer.pro	2013-02-05 20:22:31.367106374 -0800
@@ -39,6 +39,6 @@ SOURCES += main.cpp \
 
 HEADERS += tmxviewer.h
 
-manpage.path = $${PREFIX}/share/man/man1/
+manpage.path = $${PREFIX}/man/man1/
 manpage.files += ../../docs/tmxviewer.1
 INSTALLS += manpage
