diff -Naur a/amule.desktop b/amule.desktop
--- a/amule.desktop	2015-12-05 12:17:44.000000000 +0800
+++ b/amule.desktop	2016-01-28 17:59:27.159722045 +0800
@@ -5,6 +5,6 @@
 Terminal=false
 Type=Application
 Categories=Network;P2P;
-Comment=A client for the eD2k network
-Comment[fr]=Un client pour le réseau eD2k
-Comment[tr]=eD2k ağı için istemci
+GenericName=A client for the eD2k network
+GenericName[fr]=Un client pour le réseau eD2k
+GenericName[tr]=eD2k ağı için istemci
diff -Naur a/amulegui.desktop b/amulegui.desktop
--- a/amulegui.desktop	2015-12-05 12:17:44.000000000 +0800
+++ b/amulegui.desktop	2016-01-28 18:00:09.484782640 +0800
@@ -5,8 +5,8 @@
 Terminal=false
 Type=Application
 Categories=Network;P2P;
-Comment=aMule remote control
-Comment[de]=aMule-Fernsteuerung
-Comment[fr]=Contrôle à distance d'aMule
-Comment[hu]=aMule távoli vezérlő
-Comment[tr]=aMule uzaktan kumandası
+GenericName=aMule remote control
+GenericName[de]=aMule-Fernsteuerung
+GenericName[fr]=Contrôle à distance d'aMule
+GenericName[hu]=aMule távoli vezérlő
+GenericName[tr]=aMule uzaktan kumandası
diff -Naur a/src/utils/aLinkCreator/alc.desktop b/src/utils/aLinkCreator/alc.desktop
--- a/src/utils/aLinkCreator/alc.desktop	2015-12-05 12:17:44.000000000 +0800
+++ b/src/utils/aLinkCreator/alc.desktop	2016-01-28 17:56:59.532032109 +0800
@@ -5,8 +5,8 @@
 Terminal=false
 Type=Application
 Categories=Network;P2P;
-Comment=aMule ed2k link creator
-Comment[de]=ed2k-Linkersteller
-Comment[fr]=Créateur de liens ed2k d'aMule
-Comment[hu]=ed2k hivatkozás készítő
-Comment[tr]=aMule için ed2k bağlantı yaratıcısı
+GenericName=aMule ed2k link creator
+GenericName[de]=ed2k-Linkersteller
+GenericName[fr]=Créateur de liens ed2k d'aMule
+GenericName[hu]=ed2k hivatkozás készítő
+GenericName[tr]=aMule için ed2k bağlantı yaratıcısı
diff -Naur a/src/utils/wxCas/wxcas.desktop b/src/utils/wxCas/wxcas.desktop
--- a/src/utils/wxCas/wxcas.desktop	2015-12-05 12:17:44.000000000 +0800
+++ b/src/utils/wxCas/wxcas.desktop	2016-01-28 17:58:46.434702658 +0800
@@ -5,8 +5,8 @@
 Terminal=false
 Type=Application
 Categories=Network;P2P;
-Comment=aMule online statistics
-Comment[de]=aMule-Onlinestatistik
-Comment[fr]=Statistiques d'aMule en ligne
-Comment[hu]=aMule online statisztikák
-Comment[tr]=Çevrim içi aMule istatistikleri
+GenericName=aMule online statistics
+GenericName[de]=aMule-Onlinestatistik
+GenericName[fr]=Statistiques d'aMule en ligne
+GenericName[hu]=aMule online statisztikák
+GenericName[tr]=Çevrim içi aMule istatistikleri

