diff -Naur CLAM-r15456.orig/scons/libs/clam_dependent_libs_checks.py CLAM-r15456/scons/libs/clam_dependent_libs_checks.py
--- CLAM-r15456.orig/scons/libs/clam_dependent_libs_checks.py	2015-11-26 05:11:44.000000000 -0500
+++ CLAM-r15456/scons/libs/clam_dependent_libs_checks.py	2015-11-26 18:30:34.574999000 -0500
@@ -91,10 +91,14 @@
 	return True
 
 def test_oggvorbis( env, conf ) :
-	if not conf.CheckPkgConfigFile(['vorbisfile', 'vorbisenc']) :
+	if not conf.CheckPkgConfigFile(['ogg']) :
 		return False
 	if not conf.CheckLibrarySample('libogg', 'c', None, libogg_test_code ) : return False
+	if not conf.CheckPkgConfigFile(['vorbis', 'vorbisenc']) :
+		return False
 	if not conf.CheckLibrarySample('libvorbis', 'c', None, libvorbis_test_code ) : return False
+	if not conf.CheckPkgConfigFile(['vorbisfile', 'vorbisenc']) :
+		return False
 	if not conf.CheckLibrarySample('libvorbisfile', 'c', None, libvorbisfile_test_code) : return False
 	return True
 
