diff -Naur CLAM-r15298.orig/src/core/ports/OutPort.hxx CLAM-r15298/src/core/ports/OutPort.hxx
--- CLAM-r15298.orig/src/core/ports/OutPort.hxx	2012-09-16 07:59:34.924000047 +0200
+++ CLAM-r15298/src/core/ports/OutPort.hxx	2012-09-16 08:08:17.459002302 +0200
@@ -23,7 +23,7 @@
 #define __OutPort_hxx__
 
 #include "WritingRegion.hxx"
-#include <typeinfo>
+#include "TypeInfo.hxx"
 #include <list>
 #include <string>
 #include "InPort.hxx"
