diff -Naur rakarrack-0.3.0/configure rakarrack-0.3.0.patched/configure
--- rakarrack-0.3.0/configure	2008-11-13 22:14:43.000000000 -0500
+++ rakarrack-0.3.0.patched/configure	2009-11-20 03:19:54.000000000 -0500
@@ -7761,7 +7761,7 @@
 fi
 
 DATADIR="$prefix/share/${PACKAGE}"
-HELPDIR="$prefix/share/doc/${PACKAGE}"
+HELPDIR="$docdir"
 
 
 cat >>confdefs.h <<_ACEOF
diff -Naur rakarrack-0.3.0/doc/Makefile.am rakarrack-0.3.0.patched/doc/Makefile.am
--- rakarrack-0.3.0/doc/Makefile.am	2008-11-05 08:45:13.000000000 -0500
+++ rakarrack-0.3.0.patched/doc/Makefile.am	2009-11-20 03:14:22.000000000 -0500
@@ -1,5 +1,5 @@
 SUBDIRS = help
-docdir = $(prefix)/share/doc/${PACKAGE}
+docdir = @docdir@
 doc_DATA= COPYING
 
 EXTRA_DIST = $(doc_DATA)
diff -Naur rakarrack-0.3.0/doc/Makefile.in rakarrack-0.3.0.patched/doc/Makefile.in
--- rakarrack-0.3.0/doc/Makefile.in	2008-11-13 22:14:43.000000000 -0500
+++ rakarrack-0.3.0.patched/doc/Makefile.in	2009-11-20 03:14:29.000000000 -0500
@@ -127,7 +127,7 @@
 build_alias = @build_alias@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(prefix)/share/doc/${PACKAGE}
+docdir = @docdir@
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host_alias = @host_alias@
diff -Naur rakarrack-0.3.0/doc/help/Makefile.am rakarrack-0.3.0.patched/doc/help/Makefile.am
--- rakarrack-0.3.0/doc/help/Makefile.am	2008-11-05 08:45:13.000000000 -0500
+++ rakarrack-0.3.0.patched/doc/help/Makefile.am	2009-11-20 03:13:17.000000000 -0500
@@ -1,5 +1,5 @@
 SUBDIRS = imagenes css
-docdir = $(prefix)/share/doc/${PACKAGE}/html
+docdir = @docdir@/html
 doc_DATA= help.html \
 	license.html \
 	help_es.html
diff -Naur rakarrack-0.3.0/doc/help/Makefile.in rakarrack-0.3.0.patched/doc/help/Makefile.in
--- rakarrack-0.3.0/doc/help/Makefile.in	2008-11-13 22:14:43.000000000 -0500
+++ rakarrack-0.3.0.patched/doc/help/Makefile.in	2009-11-20 03:13:29.000000000 -0500
@@ -127,7 +127,7 @@
 build_alias = @build_alias@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(prefix)/share/doc/${PACKAGE}/html
+docdir = @docdir@/html
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host_alias = @host_alias@
diff -Naur rakarrack-0.3.0/doc/help/css/Makefile.am rakarrack-0.3.0.patched/doc/help/css/Makefile.am
--- rakarrack-0.3.0/doc/help/css/Makefile.am	2008-11-05 08:45:13.000000000 -0500
+++ rakarrack-0.3.0.patched/doc/help/css/Makefile.am	2009-11-20 03:12:46.000000000 -0500
@@ -1,4 +1,4 @@
-docdir = $(prefix)/share/doc/${PACKAGE}/html/css
+docdir = @docdir@/html/css
 doc_DATA= kde-default.css
 	
 EXTRA_DIST = $(doc_DATA)
diff -Naur rakarrack-0.3.0/doc/help/css/Makefile.in rakarrack-0.3.0.patched/doc/help/css/Makefile.in
--- rakarrack-0.3.0/doc/help/css/Makefile.in	2008-11-13 22:14:43.000000000 -0500
+++ rakarrack-0.3.0.patched/doc/help/css/Makefile.in	2009-11-20 03:13:03.000000000 -0500
@@ -118,7 +118,7 @@
 build_alias = @build_alias@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(prefix)/share/doc/${PACKAGE}/html/css
+docdir = @docdir@/html/css
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host_alias = @host_alias@
diff -Naur rakarrack-0.3.0/doc/help/imagenes/Makefile.am rakarrack-0.3.0.patched/doc/help/imagenes/Makefile.am
--- rakarrack-0.3.0/doc/help/imagenes/Makefile.am	2008-11-05 08:45:14.000000000 -0500
+++ rakarrack-0.3.0.patched/doc/help/imagenes/Makefile.am	2009-11-20 03:13:45.000000000 -0500
@@ -1,4 +1,4 @@
-docdir = $(prefix)/share/doc/${PACKAGE}/html/imagenes
+docdir = @docdir@/html/imagenes
 doc_DATA= icono_rakarrack_128x128.png \
 	screen.png \
 	jack_conections.jpg \
diff -Naur rakarrack-0.3.0/doc/help/imagenes/Makefile.in rakarrack-0.3.0.patched/doc/help/imagenes/Makefile.in
--- rakarrack-0.3.0/doc/help/imagenes/Makefile.in	2008-11-13 22:14:43.000000000 -0500
+++ rakarrack-0.3.0.patched/doc/help/imagenes/Makefile.in	2009-11-20 03:13:54.000000000 -0500
@@ -118,7 +118,7 @@
 build_alias = @build_alias@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(prefix)/share/doc/${PACKAGE}/html/imagenes
+docdir = @docdir@/html/imagenes
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host_alias = @host_alias@
