--- cbpadmin.orig	2012-11-18 11:59:56.030942355 -0600
+++ cbpadmin	2012-11-18 12:05:00.078476122 -0600
@@ -22,7 +22,7 @@
 use warnings;
 
 
-use lib('/usr/local/lib/policyd-2.0','/usr/lib/policyd-2.0');
+use lib('/usr/lib@LIBDIRSUFFIX@/policyd-2.0');
 
 use Config::IniFiles;
 use Getopt::Long;
@@ -52,7 +52,7 @@
 
 # Set defaults
 my $cfg;
-$cfg->{'config_file'} = "/etc/cluebringer.conf";
+$cfg->{'config_file'} = "/etc/policyd2.conf";
 
 # Check if we must override
 if (defined($opts{'config'}) && $opts{'config'} ne "") {
