--- cbpolicyd.orig	2010-01-02 10:38:11.820144965 -0600
+++ cbpolicyd	2010-01-02 10:41:45.097397504 -0600
@@ -22,7 +22,7 @@
 use strict;
 use warnings;
 
-use lib('/usr/local/lib/policyd-2.0','/usr/lib/policyd-2.0');
+use lib('/usr/lib/policyd-2.0','/usr/lib64/policyd-2.0');
 
 package cbp;
 
@@ -53,13 +53,13 @@
 
 
 	# Set defaults
-	$cfg->{'config_file'} = "/etc/cluebringer.conf";
+	$cfg->{'config_file'} = "/etc/policyd2.conf";
 
 	$server->{'timeout'} = 120;
 	$server->{'background'} = "yes";
-	$server->{'pid_file'} = "/var/run/cbpolicyd.pid";
+	$server->{'pid_file'} = "/var/run/policyd2.pid";
 	$server->{'log_level'} = 2;
-	$server->{'log_file'} = "/var/log/cbpolicyd.log";
+	$server->{'log_file'} = "/var/log/policyd2.log";
 
 	$server->{'host'} = "*";
 	$server->{'port'} = 10031;
