--- cbpolicyd.orig	2012-11-18 12:00:04.180070111 -0600
+++ cbpolicyd	2012-11-18 12:06:26.079296269 -0600
@@ -22,7 +22,7 @@
 use strict;
 use warnings;
 
-use lib('/usr/local/lib/policyd-2.0','/usr/lib/policyd-2.0','/usr/lib64/policyd-2.0');
+use lib('/usr/lib@LIBDIRSUFFIX@/policyd-2.0');
 
 package cbp;
 
@@ -53,15 +53,15 @@
 
 
 	# Set defaults
-	$cfg->{'config_file'} = "/etc/cluebringer.conf";
+	$cfg->{'config_file'} = "/etc/policyd2.conf";
 
 	$server->{'timeout'} = 120;
 	$server->{'timeout_idle'} = 1015;
 	$server->{'timeout_busy'} = 115;
 	$server->{'background'} = "yes";
-	$server->{'pid_file'} = "/var/run/cbpolicyd.pid";
+	$server->{'pid_file'} = "/var/run/policyd2.pid";
 	$server->{'log_level'} = 2;
-	$server->{'log_file'} = "/var/log/cbpolicyd.log";
+	$server->{'log_file'} = "/var/log/policyd2.log";
 
 	$server->{'host'} = "*";
 	$server->{'port'} = 10031;
