--- httpd-policyd2.conf.orig	2009-01-06 00:53:09.000000000 -0600
+++ httpd-policyd2.conf.new	2009-12-25 15:25:50.593040339 -0600
@@ -8,9 +8,9 @@
 # as well as the authors' long time involvement in large scale mail
 # hosting industry.
 
-Alias /cluebringer /usr/share/cluebringer/webui
+Alias /policyd2 @DOCROOT@/policyd2-webui
 
-<Directory /usr/share/cluebringer/webui>
+<Directory @DOCROOT@/policyd2-webui>
     # Comment out the following 3 lines to make web ui accessible from anywhere
     Order Deny,Allow
     Deny from all
