--- cluebringer-httpd.conf.orig	2012-05-10 02:04:16.000000000 -0500
+++ cluebringer-httpd.conf	2013-11-25 15:08:29.228400024 -0600
@@ -8,11 +8,11 @@
 # as well as the authors' long time involvement in large scale mail
 # hosting industry.
 
-Alias /cluebringer /usr/share/cluebringer/webui
+Alias /policyd2 @DOCROOT@/policyd2-webui
 
-<Directory /usr/share/cluebringer/webui>
-    # Comment out the following 3 lines to make web ui accessible from anywhere
-    Order Deny,Allow
-    Deny from all
-    Allow from 127.0.0.1
+<Directory @DOCROOT@/policyd2-webui>
+    # Swap the commenting on the following two lines to make web ui 
+    # accessible from anywhere
+    Require ip 127.0.0.1
+    #Require all granted
 </Directory>
