--- policyd2.conf.orig	2012-11-18 12:00:16.298772997 -0600
+++ policyd2.conf.new	2012-11-18 12:10:42.361942813 -0600
@@ -20,11 +20,11 @@
 EOT
 
 # User to run this daemon as
-#user=
-#group=
+user=policyd
+group=policyd
 
 # Filename to store pid of parent process
-#pid_file=/var/run/cbpolicyd.pid
+pid_file=/var/run/policyd2.pid
 
 # Uncommenting the below option will prevent cbpolicyd going into the background
 #background=no
@@ -60,7 +60,7 @@
 #log_level=2
 
 # File to log to instead of stdout
-#log_file=/var/log/cbpolicyd.log
+log_file=/var/log/policyd2.log
 
 # Log destination for mail logs...
 # main		- Default. Log to policyd's main log mechanism, accepts NO args
@@ -102,7 +102,7 @@
 #host=*
 
 # Port to run on
-#port=10031
+port=10031
 
 # Timeout in communication with clients
 # Idle timeout in postfix defaults to 1015s (active connection)
