http://bugs.gentoo.org/321357

--- src/app/mainWindow.cpp
+++ src/app/mainWindow.cpp
@@ -222,7 +222,7 @@
 
 inline bool MainWindow::slotScanPath(const QString &path)
 {
-    return slotScanUrl(KUrl::KUrl(path));
+    return slotScanUrl(KUrl(path));
 }
 
 bool MainWindow::slotScanUrl(const KUrl &url)
--- src/part/radialMap/widget.cpp
+++ src/part/radialMap/widget.cpp
@@ -64,7 +64,7 @@
 KUrl
 RadialMap::Widget::url(File const * const file) const
 {
-    return KUrl::KUrl(file ? file->fullPath() : m_tree->fullPath());
+    return KUrl(file ? file->fullPath() : m_tree->fullPath());
 }
 
 void
