Author: Andreas Tille <tille@debian.org>
Last-Update: Sun, 22 Sep 2024 21:52:14 +0200
Description: Fix typo in Makefile.am

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -7,7 +7,7 @@ testprog_SOURCES = testprog.c
 display_info_SOURCES = display_info.c
 
 osd_cat_LDADD 	= libxosd/libxosd.la
-diplsy_info_LDADD = libxosd/libxosd.la
+display_info_LDADD = libxosd/libxosd.la
 testprog_LDADD 	= libxosd/libxosd.la
 
 include_HEADERS = xosd.h
