diff -uaN command-list.c command-list.c.new
--- command-list.c          2020-01-29 17:56:11.000000000 -0800
+++ command-list.c.new  2020-03-08 11:24:03.000000000 -0700
@@ -890,6 +890,8 @@
     // Allocate enough space for the whole file list, plus two extra files and
     // a NULL at the end.
     cfList = nvalloc((p->num_entries + 3) * sizeof(ConflictingFileInfo));
+    /* let's just return an empty list!  */
+    return cfList;

     for (i = 0; i < p->num_entries; i++) {
         PackageEntry *entry = &p->entries[i];

diff -uaN misc.c misc.c.new
--- misc.c          2020-01-29 17:56:11.000000000 -0800
+++ misc.c.new  2020-03-08 11:42:12.000000000 -0700
@@ -1153,13 +1153,12 @@
         }

         if (!install) {
-            ui_warn(op, "The %s module will not be installed. As a result, %s "
-                    "will not function with this installation of the NVIDIA "
-                    "driver.", optional_modules[i].module_name,
+            ui_warn(op, "The %s module will be installed with no gpu detect. As a result, %s "
+                    "will function with this installation of the NVIDIA "
+                    "driver, but only if the card is installed", optional_modules[i].module_name,
                     optional_modules[i].optional_module_dependee);
-
-            remove_kernel_module_from_package(p,
-                                              optional_modules[i].module_name);
+//            remove_kernel_module_from_package(p,
+//                                              optional_modules[i].module_name);
         }
     }
 }
@@ -2036,7 +2035,10 @@
                  "For further details, please see the appendix SUPPORTED "
                  "NVIDIA GRAPHICS CHIPS in the README available on the Linux "
                  "driver download page at www.nvidia.com.", p->version);
-        return FALSE;
+       /*        return FALSE;
+        * We're just going to keep the message but skip the result
+        * so we can run this in a VM.
+        */
     }

     if (!found_vga_device)
