--- hal.conf.orig	2007-05-19 19:20:28.000000000 -0500
+++ hal.conf	2007-05-19 19:23:09.000000000 -0500
@@ -47,6 +47,7 @@
   </policy>
 
   <!-- This will not work if pam_console support is not enabled -->
+  <!-- This build (from SlackBuilds.org) does not include pam_console support -->
   <policy at_console="true">
     <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
     <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
@@ -54,11 +55,19 @@
     <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
   </policy>
 
-  <!-- You can change this to a more suitable user, or make per-group -->
-  <policy user="0">
+  <!-- Allow members of 'power' group to shutdown/restart/suspend/hibernate -->
+  <policy group="power">
     <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
+  </policy>
+
+  <!-- Allow members of 'video' group to manipulate video settings -->
+  <policy group="video">
     <allow send_interface="org.freedesktop.Hal.Device.VideoAdapterPM"/>
     <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
+  </policy>
+
+  <!-- Allow members of the 'plugdev' group to mount volumes -->
+  <policy group="plugdev">
     <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
     <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
   </policy>
