diff --git a/script/rofi-sensible-terminal b/script/rofi-sensible-terminal
index bab92d80..067a218a 100755
--- a/script/rofi-sensible-terminal
+++ b/script/rofi-sensible-terminal
@@ -9,7 +9,7 @@
 # We welcome patches that add distribution-specific mechanisms to find the
 # preferred terminal emulator. On Debian, there is the x-terminal-emulator
 # symlink for example.
-for terminal in $TERMINAL x-terminal-emulator urxvt rxvt st terminology qterminal Eterm aterm uxterm xterm roxterm xfce4-terminal.wrapper mate-terminal lxterminal konsole alacritty kitty wezterm ghostty; do
+for terminal in $TERMINAL x-terminal-emulator urxvt rxvt st terminology qterminal Eterm aterm uxterm xterm roxterm xfce4-terminal mate-terminal lxterminal konsole alacritty kitty wezterm ghostty; do
 	if command -v $terminal >/dev/null 2>&1; then
 		exec $terminal "$@"
 	fi
