diff -Naur kegs.0.91/src/sound.c kegs.0.91.patched/src/sound.c
--- kegs.0.91/src/sound.c	2004-12-06 19:42:47.000000000 -0500
+++ kegs.0.91.patched/src/sound.c	2015-06-21 17:48:21.000000000 -0400
@@ -363,9 +363,9 @@
 
 	ret = read(read_fd, &tmp, 4);
 	if(ret != 4) {
-		printf("parent dying, could not get sample rate from child\n");
+		printf("parent could not get audio sample rate from child, disabling sound.\n");
 		printf("ret: %d, fd: %d errno:%d\n", ret, read_fd, errno);
-		exit(1);
+		g_audio_enable = 0;
 	}
 	close(read_fd);
 
