diff -Naur disk-manager-1.0.1~/configure disk-manager-1.0.1/configure
--- disk-manager-1.0.1~/configure	2009-01-18 13:05:41.000000000 -0600
+++ disk-manager-1.0.1/configure	2009-01-18 13:05:45.000000000 -0600
@@ -752,6 +752,7 @@
 INTLTOOL_MSGMERGE
 INTLTOOL_XGETTEXT
 ALL_LINGUAS
+PYTHON2_7
 PYTHON2_4
 PYTHON
 PYTHON2_5
@@ -6370,6 +6371,91 @@
 
 
   if test -z "$PYTHON"; then
+    # Extract the first word of "python2.7", so it can be a program name with args.
+set dummy python2.7; ac_word=$2
+{ echo "$as_me:$LINENO: checking for $ac_word" >&5
+echo $ECHO_N "checking for $ac_word... $ECHO_C" >&6; }
+if test "${ac_cv_path_PYTHON2_7+set}" = set; then
+  echo $ECHO_N "(cached) $ECHO_C" >&6
+else
+  case $PYTHON2_7 in
+  [\\/]* | ?:[\\/]*)
+  ac_cv_path_PYTHON2_7="$PYTHON2_7" # Let the user override the test with a path.
+  ;;
+  *)
+  as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
+for as_dir in $PATH
+do
+  IFS=$as_save_IFS
+  test -z "$as_dir" && as_dir=.
+  for ac_exec_ext in '' $ac_executable_extensions; do
+  if { test -f "$as_dir/$ac_word$ac_exec_ext" && $as_test_x "$as_dir/$ac_word$ac_exec_ext"; }; then
+    ac_cv_path_PYTHON2_7="$as_dir/$ac_word$ac_exec_ext"
+    echo "$as_me:$LINENO: found $as_dir/$ac_word$ac_exec_ext" >&5
+    break 2
+  fi
+done
+done
+IFS=$as_save_IFS
+
+  ;;
+esac
+fi
+PYTHON2_7=$ac_cv_path_PYTHON2_7
+if test -n "$PYTHON2_7"; then
+  { echo "$as_me:$LINENO: result: $PYTHON2_7" >&5
+echo "${ECHO_T}$PYTHON2_7" >&6; }
+else
+  { echo "$as_me:$LINENO: result: no" >&5
+echo "${ECHO_T}no" >&6; }
+fi
+
+
+  else
+    PYTHON2_7="$PYTHON"
+  fi
+
+  PYTHON="$PYTHON2_7"
+
+
+  if test -n "$PYTHON"; then
+
+
+            { echo "$as_me:$LINENO: checking whether $PYTHON version is >= 2.7.0" >&5
+echo $ECHO_N "checking whether $PYTHON version is >= 2.7.0... $ECHO_C" >&6; }
+      prog="import sys, string
+# split strings by '.' and convert to numeric.  Append some zeros
+# because we need at least 4 digits for the hex conversion.
+minver = map(int, string.split('2.7.0', '.')) + [0, 0, 0]
+minverhex = 0
+for i in xrange(0, 4): minverhex = (minverhex << 8) + minver[i]
+sys.exit(sys.hexversion < minverhex)"
+  if { echo "$as_me:$LINENO: $PYTHON -c "$prog"" >&5
+   ($PYTHON -c "$prog") >&5 2>&5
+   ac_status=$?
+   echo "$as_me:$LINENO: \$? = $ac_status" >&5
+   (exit $ac_status); }; then
+  { echo "$as_me:$LINENO: result: yes" >&5
+echo "${ECHO_T}yes" >&6; }
+else
+  { echo "$as_me:$LINENO: result: no" >&5
+echo "${ECHO_T}no" >&6; }; PYTHON=""
+fi
+
+
+
+  fi
+
+  if test -z "$PYTHON"; then
+
+        PYTHON="$_python_save"
+
+
+  _python_save="$PYTHON"
+
+
+
+  if test -z "$PYTHON"; then
     # Extract the first word of "python2.4", so it can be a program name with args.
 set dummy python2.4; ac_word=$2
 { echo "$as_me:$LINENO: checking for $ac_word" >&5
@@ -6608,6 +6694,7 @@
 
 
   fi
+  fi
 
 
 
diff -Naur disk-manager-1.0.1~/Makefile.in disk-manager-1.0.1/Makefile.in
--- disk-manager-1.0.1~/Makefile.in	2009-01-18 13:05:41.000000000 -0600
+++ disk-manager-1.0.1/Makefile.in	2009-01-18 13:05:45.000000000 -0600
@@ -581,6 +581,8 @@
 info-am:
 
 install-data-am:
+	$(mkinstalldirs) $(DESTDIR)$(datadir)/pixmaps
+	ln -sf ../icons/hicolor/scalable/apps/@PACKAGE@.svg $(DESTDIR)$(datadir)/pixmaps
 
 install-exec-am: install-binSCRIPTS install-sbinSCRIPTS
 
diff -Naur disk-manager-1.0.1~/data/disk-manager-check.desktop.in.in disk-manager-1.0.1/data/disk-manager-check.desktop.in.in
--- disk-manager-1.0.1~/data/disk-manager-check.desktop.in.in	2009-01-18 13:05:41.000000000 -0600
+++ disk-manager-1.0.1/data/disk-manager-check.desktop.in.in	2009-01-18 13:06:16.000000000 -0600
@@ -1,12 +1,9 @@
 [Desktop Entry]
 Version=1.0
-Encoding=UTF-8
-Name=Aucun nom
-_Name=Disk manager
+_Name=Disk Manager
 _Comment=Check for new partitions
 Icon=disk-manager
 Exec=@prefix@/bin/disk-manager --check
 Terminal=false
 Type=Application
-Categories=
-GenericName[fr_FR]=
+Categories=System;Filesystem;
diff -Naur disk-manager-1.0.1~/data/disk-manager.desktop.in.in disk-manager-1.0.1/data/disk-manager.desktop.in.in
--- disk-manager-1.0.1~/data/disk-manager.desktop.in.in	2009-01-18 13:05:41.000000000 -0600
+++ disk-manager-1.0.1/data/disk-manager.desktop.in.in	2009-01-18 13:05:45.000000000 -0600
@@ -1,12 +1,9 @@
 [Desktop Entry]
 Version=1.0
-Encoding=UTF-8
-Name=Aucun nom
 _Name=Disk Manager
 _Comment=Manage filesystem configuration
 Icon=disk-manager
 Exec=@prefix@@EXEC_PATH@/disk-manager-root
 Terminal=false
 Type=Application
-Categories=Application;System;Settings
-GenericName[fr_FR]=
+Categories=System;Filesystem;
