diff -Nur mp3splt-gtk-0.5.4.orig/mp3splt-gtk.desktop mp3splt-gtk-0.5.4/mp3splt-gtk.desktop
--- mp3splt-gtk-0.5.4.orig/mp3splt-gtk.desktop	2008-06-06 15:29:03.000000000 -0500
+++ mp3splt-gtk-0.5.4/mp3splt-gtk.desktop	2009-01-30 23:54:08.806698497 -0600
@@ -5,7 +5,7 @@
 GenericName=Sound Splitter
 Comment=Split mp3 and ogg files without decoding
 
-Icon=/usr/share/mp3splt-gtk/mp3splt-gtk_ico.png
+Icon=mp3splt-gtk_ico
 
 Type=Application
 Categories=Application;AudioVideo;Audio;AudioVideoEditing;
