*** mp3splt-gtk.desktop.orig	2008-09-27 11:16:10.000000000 +0200
--- mp3splt-gtk.desktop	2008-10-14 09:20:13.000000000 +0200
***************
*** 5,11 ****
  GenericName=Sound Splitter
  Comment=Split mp3 and ogg files without decoding
  
! Icon=/usr/share/mp3splt-gtk/mp3splt-gtk_ico.png
  
  Type=Application
  Categories=Application;AudioVideo;Audio;AudioVideoEditing;
--- 5,11 ----
  GenericName=Sound Splitter
  Comment=Split mp3 and ogg files without decoding
  
! Icon=mp3splt-gtk_ico.png
  
  Type=Application
  Categories=Application;AudioVideo;Audio;AudioVideoEditing;
