diff -Naur debootstrap.orig/Makefile debootstrap/Makefile
--- debootstrap.orig/Makefile	2009-01-30 12:12:06.000000000 +0100
+++ debootstrap/Makefile	2009-03-03 17:27:14.000000000 +0100
@@ -1,7 +1,8 @@
 # avoid dpkg-dev dependency; fish out the version with sed
 VERSION := $(shell sed 's/.*(\(.*\)).*/\1/; q' debian/changelog)
 
-ARCH := $(shell dpkg --print-architecture)
+# not needed
+#ARCH := $(shell dpkg --print-architecture)
 
 MAKEDEV := $(shell if [ -e /dev/MAKEDEV ]; then echo /dev/MAKEDEV; else echo /sbin/MAKEDEV; fi)
 
@@ -10,7 +11,7 @@
 	rm -f devices.tar.gz
 	rm -rf dev
 
-DSDIR=$(DESTDIR)/usr/share/debootstrap
+DSDIR=$(DESTDIR)/usr/lib/debootstrap
 install:
 	mkdir -p $(DSDIR)/scripts
 	mkdir -p $(DESTDIR)/usr/sbin
diff -Naur debootstrap.orig/debootstrap debootstrap/debootstrap
--- debootstrap.orig/debootstrap	2009-01-13 20:30:15.000000000 +0100
+++ debootstrap/debootstrap	2009-03-03 17:28:16.000000000 +0100
@@ -10,7 +10,7 @@
 	if [ -x /debootstrap/debootstrap ]; then
 		DEBOOTSTRAP_DIR=/debootstrap
 	else
-		DEBOOTSTRAP_DIR=/usr/share/debootstrap
+		DEBOOTSTRAP_DIR=/usr/lib/debootstrap
 	fi
 fi
 
