diff -Naur debootstrap.orig/Makefile debootstrap/Makefile
--- debootstrap.orig/Makefile	2009-01-30 12:12:06.000000000 +0100
+++ debootstrap/Makefile	2009-03-03 17:27:14.000000000 +0100
@@ -1,7 +1,8 @@
 # avoid dpkg-dev dependency; fish out the version with sed
 VERSION := $(shell sed 's/.*(\(.*\)).*/\1/; q' debian/changelog)
 
-ARCH := $(shell dpkg --print-architecture)
+# not needed
+#ARCH := $(shell dpkg --print-architecture)
 
 MAKEDEV := $(shell if [ -e /dev/MAKEDEV ]; then echo /dev/MAKEDEV; else echo /sbin/MAKEDEV; fi)
 
