diff -Naur cwiid-0.6.00.orig/lswm/Makefile.in cwiid-0.6.00/lswm/Makefile.in
--- cwiid-0.6.00.orig/lswm/Makefile.in	2007-07-28 18:47:03.000000000 +0000
+++ cwiid-0.6.00/lswm/Makefile.in	2012-09-19 01:36:44.624050597 +0000
@@ -8,7 +8,7 @@
 
 CFLAGS += -I@top_builddir@/libcwiid
 LDFLAGS += -L@top_builddir@/libcwiid
-LDLIBS += -lcwiid
+LDLIBS += -lcwiid -lbluetooth
 INST_DIR = @bindir@
 
 include $(COMMON)/include/app.mak
diff -Naur cwiid-0.6.00.orig/wmdemo/Makefile.in cwiid-0.6.00/wmdemo/Makefile.in
--- cwiid-0.6.00.orig/wmdemo/Makefile.in	2007-04-11 02:57:27.000000000 +0000
+++ cwiid-0.6.00/wmdemo/Makefile.in	2012-09-19 01:37:25.008051673 +0000
@@ -8,7 +8,7 @@
 
 CFLAGS += -I@top_builddir@/libcwiid
 LDFLAGS += -L@top_builddir@/libcwiid
-LDLIBS += -lcwiid
+LDLIBS += -lcwiid -lbluetooth
 INST_DIR = @bindir@
 
 include $(COMMON)/include/app.mak
diff -Naur cwiid-0.6.00.orig/wmgui/Makefile.in cwiid-0.6.00/wmgui/Makefile.in
--- cwiid-0.6.00.orig/wmgui/Makefile.in	2007-07-28 18:47:03.000000000 +0000
+++ cwiid-0.6.00/wmgui/Makefile.in	2012-09-19 01:31:32.977042284 +0000
@@ -8,7 +8,7 @@
 
 CFLAGS += @GTK_CFLAGS@ -I@top_builddir@/libcwiid
 LDFLAGS += -L@top_builddir@/libcwiid
-LDLIBS += @GTK_LIBS@ -lm -lcwiid
+LDLIBS += @GTK_LIBS@ -lm -lcwiid -lbluetooth
 INST_DIR = @bindir@
 
 include $(COMMON)/include/app.mak
diff -Naur cwiid-0.6.00.orig/wminput/Makefile.in cwiid-0.6.00/wminput/Makefile.in
--- cwiid-0.6.00.orig/wminput/Makefile.in	2007-07-28 18:47:03.000000000 +0000
+++ cwiid-0.6.00/wminput/Makefile.in	2012-09-19 01:38:42.031053733 +0000
@@ -14,7 +14,7 @@
 	-DWMINPUT_CONFIG_DIR=\"$(WMINPUT_CONFIG_DIR)\" \
 	-DCWIID_PLUGINS_DIR=\"$(CWIID_PLUGINS_DIR)\"
 
-LDLIBS += -lcwiid -ldl
+LDLIBS += -lcwiid -ldl -lbluetooth -lpthread
 LDFLAGS += -L@top_builddir@/libcwiid -rdynamic
 INST_DIR = @bindir@
 
