--- cwiid/configure.ac.orig	2021-10-25 12:06:48.965614546 +0700
+++ cwiid/configure.ac	2021-10-25 12:07:01.190613822 +0700
@@ -67,7 +67,7 @@
 fi
 AC_SUBST(PYTHON)
 if test $PYTHON; then
-	PYTHON_VERSION=[`$PYTHON -c 'import sys; print (sys.version[:3])'`]
+	PYTHON_VERSION=[`$PYTHON -c 'import sys; print (sys.version[:4])'`]
 	AC_SUBST(PYTHON_VERSION)
 	AC_DEFINE([HAVE_PYTHON],1,[Define to 1 if python support is enabled])
 fi
