--- tv-0.5.1/ncl-2.0/src/charactersblock.cpp.orig	2017-08-13 08:38:11.451581249 +0700
+++ tv-0.5.1/ncl-2.0/src/charactersblock.cpp	2017-08-13 08:38:25.267599533 +0700
@@ -8,6 +8,7 @@
 #include "discretematrix.h"
 #include "assumptionsblock.h"
 #include "charactersblock.h"
+#include <cstring>
 
 using namespace std;
 
--- tv-0.5.1/ncl-2.0/src/nexus.cpp.orig	2017-08-13 08:40:07.443734383 +0700
+++ tv-0.5.1/ncl-2.0/src/nexus.cpp	2017-08-13 08:40:17.625747774 +0700
@@ -2,6 +2,7 @@
 #include "xnexus.h"
 #include "nexustoken.h"
 #include "nexus.h"
+#include <cstring>
 
 #define NCL_NAME_AND_VERSION  "NCL version 2.01"
 #define NCL_COPYRIGHT         "Copyright (c) 2000 by Paul O. Lewis"
--- tv-0.5.1/ncl-2.0/src/nexustoken.cpp.orig	2017-08-13 08:41:41.569857692 +0700
+++ tv-0.5.1/ncl-2.0/src/nexustoken.cpp	2017-08-13 08:41:52.463871895 +0700
@@ -1,6 +1,7 @@
 #include "nexusdefs.h"
 #include "xnexus.h"
 #include "nexustoken.h"
+#include <cstring>
 
 using namespace std;
 
--- tv-0.5.1/ncl-2.0/src/nxsstring.cpp.orig	2017-08-13 08:42:36.847929712 +0700
+++ tv-0.5.1/ncl-2.0/src/nxsstring.cpp	2017-08-13 08:42:48.870945321 +0700
@@ -20,6 +20,7 @@
 #include <stdio.h>
 #include <assert.h>
 #include <math.h>
+#include <cstring>
 #include "nxsstring.h"
 
 nxsstring& nxsstring::operator+=( const double d )
--- tv-0.5.1/TreeLib/Parse.cpp.orig	2017-08-13 08:43:50.188024809 +0700
+++ tv-0.5.1/TreeLib/Parse.cpp	2017-08-13 08:44:00.407038015 +0700
@@ -23,6 +23,7 @@
  
 #include <ctype.h>
 #include "Parse.h"
+#include <cstring>
 
 // Return the next token in the string
 tokentype Parser::NextToken ()
--- tv-0.5.1/TreeLib/TreeLib.cpp.orig	2017-08-13 08:45:24.898146879 +0700
+++ tv-0.5.1/TreeLib/TreeLib.cpp	2017-08-13 08:45:33.518157938 +0700
@@ -23,6 +23,7 @@
 
 #include "TreeLib.h"
 #include "Parse.h"
+#include <cstdlib>
 
 #include <vector>
 
--- tv-0.5.1/TreeLib/treereader.cpp.orig        2017-08-13 08:47:09.041280171 +0700
+++ tv-0.5.1/TreeLib/treereader.cpp     2017-08-13 08:51:04.192577614 +0700
@@ -22,6 +22,8 @@
  // $Id: treereader.cpp,v 1.5 2003/09/10 12:58:16 rdmp1c Exp $
 
 #include "treereader.h"
+#include <cstring>
+#include <stdlib.h>
 
 #if __MWERKS__
        #include <string.h>
