diff -Naur gr-osmosdr-20170612.orig/lib/rfspace/rfspace_source_c.cc gr-osmosdr-20170612/lib/rfspace/rfspace_source_c.cc
--- gr-osmosdr-20170612.orig/lib/rfspace/rfspace_source_c.cc	2018-06-14 21:01:32.000000000 +0200
+++ gr-osmosdr-20170612/lib/rfspace/rfspace_source_c.cc	2020-05-17 15:11:25.958538000 +0200
@@ -1089,14 +1089,14 @@
     // Start the asynchronous receive operation. The handle_receive function
     // used as a callback will update the ec and rx_bytes variables.
     socket.async_receive( boost::asio::buffer(data, sizeof(data)),
-        boost::bind(handle_receive, _1, _2, &ec, &rx_bytes) );
+        boost::bind(handle_receive, boost::placeholders::_1, boost::placeholders::_2, &ec, &rx_bytes) );
 
     // Set a deadline for the asynchronous operation.
     timer.expires_from_now( boost::posix_time::milliseconds(10) );
 
     // Start an asynchronous wait on the timer. The handle_timer function
     // used as a callback will update the ec variable.
-    timer.async_wait( boost::bind(handle_timer, _1, &ec) );
+    timer.async_wait( boost::bind(handle_timer, boost::placeholders::_1, &ec) );
 
     // Reset the io_service in preparation for a subsequent run_one() invocation.
     ios.reset();
