diff -Naur idesk-0.7.5/src/DesktopConfig.cpp idesk-0.7.5.patched/src/DesktopConfig.cpp
--- idesk-0.7.5/src/DesktopConfig.cpp	2005-11-10 18:07:58.000000000 -0500
+++ idesk-0.7.5.patched/src/DesktopConfig.cpp	2014-08-15 20:50:47.000000000 -0400
@@ -236,11 +236,13 @@
 
     directory += "/.idesktop/";
 
+	 mkdir(directory.c_str(), 0755);
+
     fileCount = scandir(directory.c_str(), &files, 0, alphasort);
     
     if (fileCount == -1)
     {
-        cout << "Error: you have to create the .idesktop dir on your HOME!!\n";
+        cout << "Error: Can't find or create the .idesktop dir on your HOME!!\n";
         _exit (1);
     }
 
