diff -Naur libsmi-0.5.0.orig/lib/parser-smi.h libsmi-0.5.0/lib/parser-smi.h
--- libsmi-0.5.0.orig/lib/parser-smi.h	2013-06-04 14:34:00.000000000 +0200
+++ libsmi-0.5.0/lib/parser-smi.h	2025-07-03 06:48:23.823380000 +0200
@@ -24,7 +24,7 @@
 
 
 
-extern int smiparse();
+extern int smiparse(struct Parser *parserPtr);
 
 
 
diff -Naur libsmi-0.5.0.orig/lib/parser-yang.h libsmi-0.5.0/lib/parser-yang.h
--- libsmi-0.5.0.orig/lib/parser-yang.h	2013-06-04 14:34:00.000000000 +0200
+++ libsmi-0.5.0/lib/parser-yang.h	2025-07-03 06:48:54.799380000 +0200
@@ -13,7 +13,7 @@
 
 
 
-extern int yangparse();
+extern int yangparse(struct Parser *parserPtr);
 
 
 
diff -Naur libsmi-0.5.0.orig/lib/smi-data.c libsmi-0.5.0/lib/smi-data.c
--- libsmi-0.5.0.orig/lib/smi-data.c	2013-06-04 14:34:00.000000000 +0200
+++ libsmi-0.5.0/lib/smi-data.c	2025-07-03 06:49:46.413380000 +0200
@@ -39,7 +39,7 @@
 
 #ifdef BACKEND_SMI
 #include "scanner-smi.h"
-extern int smiparse();
+extern int smiparse(struct Parser *parserPtr);
 #endif
 
 #ifdef BACKEND_SMING
