diff -Nur libcsv-3.0.0.orig//Makefile libcsv-3.0.0/Makefile
--- libcsv-3.0.0.orig//Makefile	2008-07-26 20:48:28.000000000 -0500
+++ libcsv-3.0.0/Makefile	2010-05-11 00:01:39.900783556 -0500
@@ -17,6 +17,7 @@
 install_static: install_headers install_static_lib
 
 install_man: csv.3.gz
+	mkdir -p $(DESTDIR)$(MANDIR)/
 	cp -f $^ $(DESTDIR)$(MANDIR)/
 
 install_headers: csv.h
@@ -24,11 +25,13 @@
 	cp -f $^ $(DESTDIR)$(INCDIR)/libcsv/
 
 install_shared_lib: libcsv.so
+	mkdir -p $(DESTDIR)$(LIBDIR)
 	cp -f $< $(DESTDIR)$(LIBDIR)/$<.$(VERSION)
 	ln -sf $<.$(VERSION) $(DESTDIR)$(LIBDIR)/$<.3
 	ln -sf $<.3 $(DESTDIR)$(LIBDIR)/$<
 
 install_static_lib: libcsv.a
+	mkdir -p $(DESTDIR)$(LIBDIR)
 	cp -f $< $(DESTDIR)$(LIBDIR)/$<
 
 libcsv.o: libcsv.c csv.h
