diff -Naur lastfm-desktop-2.1.36.orig/app/fingerprinter/LAV_Source.cpp lastfm-desktop-2.1.36/app/fingerprinter/LAV_Source.cpp
--- lastfm-desktop-2.1.36.orig/app/fingerprinter/LAV_Source.cpp	2013-09-03 12:19:10.000000000 +0000
+++ lastfm-desktop-2.1.36/app/fingerprinter/LAV_Source.cpp	2014-01-01 02:15:23.422896266 +0000
@@ -34,6 +34,9 @@
 # include <libavresample/avresample.h>
 # include <libavutil/opt.h>
 #endif
+#ifndef AVCODEC_MAX_AUDIO_FRAME_SIZE
+#define AVCODEC_MAX_AUDIO_FRAME_SIZE 192000 // 1 second of 48khz 32bit audio
+#endif
 }
 
 #include <iostream>
