diff -Naur lastfm-desktop-2.1.36.orig/lib/unicorn/UnicornApplication.cpp lastfm-desktop-2.1.36/lib/unicorn/UnicornApplication.cpp
--- lastfm-desktop-2.1.36.orig/lib/unicorn/UnicornApplication.cpp	2013-09-03 12:19:10.000000000 +0000
+++ lastfm-desktop-2.1.36/lib/unicorn/UnicornApplication.cpp	2014-01-01 03:33:09.389688237 +0000
@@ -234,7 +234,7 @@
     t1->load( d.filePath( "lastfm_" + qmExt ) );
 
     QTranslator* t2 = new QTranslator( this );
-    t2->load( d.filePath( "qt_" + qmExt ) );
+    t2->load( QDir("/usr/lib@LIBDIRSUFFIX@/qt/translations").filePath( "qt_" + qmExt ) );
 
     installTranslator( t1 );
     installTranslator( t2 );
