--- a/Doxyfile.in	2005-01-05 14:27:05.000000000 +0100
+++ b/Doxyfile.in	2005-01-05 14:30:19.000000000 +0100
@@ -263,14 +263,17 @@
 
 INPUT                  = ../src \
                          ../include/id3/field.h \
-                         ../include/id3/frame.h \
                          ../include/id3/globals.h \
+                         ../include/id3/helpers.h \
+                         ../include/id3/id3lib_frame.h \
+                         ../include/id3/id3lib_streams.h \
+                         ../include/id3/id3lib_strings.h \
                          ../include/id3/misc_support.h \
                          ../include/id3/reader.h \
                          ../include/id3/readers.h \
                          ../include/id3/sized_types.h \
                          ../include/id3/tag.h \
-                         ../include/id3/tag.h \
+                         ../include/id3/utils.h \
                          ../include/id3/writer.h \
                          ../include/id3/writers.h
 
