--- ccrypt-1.11/emacs/ps-ccrypt.el.orig	2025-08-09 22:39:46.168206463 +0700
+++ ccrypt-1.11/emacs/ps-ccrypt.el	2025-08-09 22:40:09.539760630 +0700
@@ -486,7 +486,7 @@
   :type 'string
   :group 'ps-ccrypt)
 
-(defvar ps-ccrypt-temp-name-table (make-vector 31 nil))
+(defvar ps-ccrypt-temp-name-table (make-vector 31 0))
 
 (defun ps-ccrypt-make-temp-name (&optional local-copy)
   "This routine will return the name of a new file."
