diff -Naur csh-20110502.orig/csh.h csh-20110502.patched/csh.h
--- csh-20110502.orig/csh.h	2014-04-25 17:31:52.000000000 -0400
+++ csh-20110502.patched/csh.h	2014-04-25 17:34:08.000000000 -0400
@@ -36,12 +36,10 @@
  * Fundamental definitions which may vary from system to system.
  *
  *	BUFSIZ		The i/o buffering size; also limits word size
+ *             20140425 bkw: moved below the #include <stdio.h> since
+ *             we want to use the system's default BUFSIZ.
  *	MAILINTVL	How often to mailcheck; more often is more expensive
  */
-#ifndef BUFSIZ
-#define	BUFSIZ	1024		/* default buffer size */
-#endif				/* BUFSIZ */
-
 #ifndef MAXPATHLEN
 #define MAXPATHLEN BUFSIZ
 #endif
@@ -96,6 +94,11 @@
 #include <stdio.h>
 FILE *cshin, *cshout, *csherr;
 
+/* 20140425 bkw: moved here so stdio.h is what initially defines BUFSIZ. */
+#ifndef BUFSIZ
+#define	BUFSIZ	1024		/* default buffer size */
+#endif				/* BUFSIZ */
+
 #include <stdio_ext.h>
 #define fpurge __fpurge
 
