--- /tmp/SBo/ventoy-1.1.17/Ventoy2Disk.sh	2026-07-24 13:59:22.000000000 +0100
+++ 0/patched/Ventoy2Disk.sh.new	2026-07-24 20:43:07.595600506 +0100
@@ -31,34 +31,6 @@
 echo ''
 
 
-if ! [ -f ./boot/boot.img ]; then
-    if [ -d ./grub ]; then
-        echo "Don't run Ventoy2Disk.sh here, please download the released install package, and run the script in it."
-    else
-        echo "Please run under the correct directory!" 
-    fi
-    exit 1
-fi
-
-echo "############# Ventoy2Disk $* [$TOOLDIR] ################" >> ./log.txt
-date >> ./log.txt
-
-#decompress tool
-echo "decompress tools" >> ./log.txt
-cd ./tool/$TOOLDIR
-
-ls *.xz > /dev/null 2>&1
-if [ $? -eq 0 ]; then
-    [ -f ./xzcat ] && chmod +x ./xzcat
-
-    for file in $(ls *.xz); do
-        echo "decompress $file" >> ./log.txt
-        xzcat $file > ${file%.xz}
-        [ -f ./${file%.xz} ] && chmod +x ./${file%.xz}
-        [ -f ./$file ] && rm -f ./$file
-    done
-fi
-
 #use static linked mkexfatfs for musl-libc environment
 if [ -f mkexfatfs_static ]; then
     if ldd --version 2>&1 | grep -qi musl; then
@@ -66,10 +38,10 @@
         mv mkexfatfs_static mkexfatfs
     else
         if ./mkexfatfs -V > /dev/null 2>&1; then
-            echo "mkexfatfs can not run, check static version" >> ./log.txt
+            echo "mkexfatfs can not run, check static version" > /var/log/ventoy.log
         else
             if ./mkexfatfs_static -V > /dev/null 2>&1; then
-                echo "Use static version of mkexfatfs" >> ./log.txt
+                echo "Use static version of mkexfatfs" > /var/log/ventoy.log
                 mv mkexfatfs mkexfatfs_shared
                 mv mkexfatfs_static mkexfatfs
             fi
