diff -Naur libslack-0.6.orig/libslack-config.pod libslack-0.6/libslack-config.pod
--- libslack-0.6.orig/libslack-config.pod	2010-06-12 15:37:00.000000000 +0200
+++ libslack-0.6/libslack-config.pod	2013-01-02 20:36:02.802707880 +0100
@@ -32,13 +32,10 @@
    -h, --help      - Print this help and exit
    -v, --version   - Print the version of the currently installed libslack
    -L, --latest    - Print the latest version of libslack (uses wget)
-   -D, --download  - Download the latest version of libslack (uses wget)
-   -U, --upgrade   - Upgrade to the latest version of libslack (uses wget)
    -p, --prefix    - Print the prefix directory of the libslack installation
    -c, --cflags    - Print CFLAGS needed to compile clients of libslack
    -l, --libs      - Print LDFLAGS needed to link against libslack
    -l, --ldflags   - Identical to --libs
-   -u, --uninstall - Uninstall libslack
 
  Note: The dashes are optional for long option names
 
@@ -75,19 +72,6 @@
 Print the latest version of I<libslack> available at
 C<http://libslack.org/>.
 
-=item C<-D>, C<--download>
-
-Download the latest version of I<libslack> from
-C<http://libslack.org/>.
-
-=item C<-U>, C<--upgrade>
-
-Upgrade to the latest version of I<libslack> from
-C<http://libslack.org/>. This downloads the latest
-version, configures it, compiles it, uninstalls the currently installed
-version and then installs the new version wherever the current version
-was installed.
-
 =item C<-p>, C<--prefix>
 
 Print the prefix directory under which I<libslack> was installed.
@@ -100,10 +84,6 @@
 
 Print the linker flags needed to link code against I<libslack>.
 
-=item C<-u>, C<--uninstall>
-
-Uninstall I<libslack>.
-
 =back
 
 =head1 SEE ALSO
diff -Naur libslack-0.6.orig/libslack-config.t libslack-0.6/libslack-config.t
--- libslack-0.6.orig/libslack-config.t	2010-06-12 15:37:00.000000000 +0200
+++ libslack-0.6/libslack-config.t	2013-01-02 19:55:45.772706368 +0100
@@ -45,13 +45,10 @@
     -h, --help      - Print this help and exit
     -v, --version   - Print the version of the currently installed libslack
     -L, --latest    - Print the latest version of libslack (uses wget)
-    -D, --download  - Download the latest version of libslack (uses wget)
-    -U, --upgrade   - Upgrade to the latest version of libslack (uses wget)
     -p, --prefix    - Print the prefix directory of the libslack installation
     -c, --cflags    - Print CFLAGS needed to compile clients of libslack
     -l, --libs      - Print LDFLAGS needed to link against libslack
     -l, --ldflags   - Identical to --libs
-    -u, --uninstall - Uninstall libslack
 
 Note: the dashes are optional for long option names
 
@@ -102,37 +99,6 @@
 	'
 }
 
-download()
-{
-	latest="`latest 2>&1`"
-	test "$latest" = "No versions found at ${url}download" && die "$latest"
-	file="`echo $latest | sed 's/^.*\///'`"
-	test -f "$file" && die "The file $file already exists"
-	wget "$latest"
-}
-
-upgrade()
-{
-	latest="`latest 2>&1`"
-	test "$latest" = "No versions found at ${url}/download" && die "$latest"
-	file="`echo $latest | sed 's/^.*\///'`"
-	dir="`echo $file | sed 's/\.tar\.gz$//'`"
-	test -f "$file" || wget "$latest"
-	test -s "$file" || die "Failed to download $latest"
-	tar xzf "$file" || die "Failed to untar $file"
-	cd "$dir" || die "Failed to cd $dir"
-	./configure || die "Failed to configure $dir"
-	make || die "Failed to make $dir"
-	uninstall || die "Failed to uninstall current version"
-	make PREFIX="$prefix" install || die "Failed to install $dir into $prefix"
-	cd .. && rm -rf "$dir"
-}
-
-uninstall()
-{
-@@UNINSTALL@@
-}
-
 test $# -eq 0 && usage 1 1>&2
 
 while test $# -gt 0
@@ -141,13 +107,10 @@
 		-h|--help|help)           usage 0;;
 		-v|--version|version)     echo "$version";;
 		-L|--latest|latest)       latest;;
-		-D|--download|download)   download;;
-		-U|--upgrade|upgrade)     upgrade;;
 		-p|--prefix|prefix)       echo "$prefix";;
 		-c|--cflags|cflags)       echo "$cflags";;
 		-l|--libs|libs)           echo "$libs";;
 		-l|--ldflags|ldflags)     echo "$libs";;
-		-u|--uninstall|uninstall) uninstall;;
 		*) usage 1 >&2;;
 	esac
 	shift
