diff -ru a/lib/Test/TempDir/Tiny.pm b/lib/Test/TempDir/Tiny.pm
--- a/lib/Test/TempDir/Tiny.pm	2018-07-12 14:22:03.000000000 +0200
+++ b/lib/Test/TempDir/Tiny.pm	2018-09-13 21:05:04.417528503 +0200
@@ -18,7 +18,7 @@
     use File::Path 2.07 qw/remove_tree/;
 }
 use File::Spec::Functions qw/catdir/;
-use File::Temp 0.2308;
+use File::Temp 0.2304;
 
 my ( $ROOT_DIR, $TEST_DIR, %COUNTER );
 my ( $ORIGINAL_PID, $ORIGINAL_CWD, $TRIES, $DELAY, $SYSTEM_TEMP ) =
diff -ru a/Makefile.PL b/Makefile.PL
--- a/Makefile.PL	2018-07-12 14:22:03.000000000 +0200
+++ b/Makefile.PL	2018-09-13 21:04:51.652738922 +0200
@@ -23,7 +23,7 @@
     "Exporter" => "5.57",
     "File::Path" => "2.07",
     "File::Spec::Functions" => 0,
-    "File::Temp" => "0.2308",
+    "File::Temp" => "0.2304",
     "strict" => 0,
     "warnings" => 0
   },
@@ -50,7 +50,7 @@
   "File::Path" => "2.07",
   "File::Spec" => 0,
   "File::Spec::Functions" => 0,
-  "File::Temp" => "0.2308",
+  "File::Temp" => "0.2304",
   "Test::More" => 0,
   "strict" => 0,
   "warnings" => 0
diff -ru a/t/00-report-prereqs.dd b/t/00-report-prereqs.dd
--- a/t/00-report-prereqs.dd	2018-07-12 14:22:03.000000000 +0200
+++ b/t/00-report-prereqs.dd	2018-09-13 21:05:27.798806480 +0200
@@ -37,7 +37,7 @@
                                       'Exporter' => '5.57',
                                       'File::Path' => '2.07',
                                       'File::Spec::Functions' => '0',
-                                      'File::Temp' => '0.2308',
+                                      'File::Temp' => '0.2304',
                                       'perl' => '5.006002',
                                       'strict' => '0',
                                       'warnings' => '0'
@@ -58,4 +58,4 @@
                  }
      };
   $x;
- }
\ No newline at end of file
+ }
