--- hnefatafl130527.orig/config.mk	2013-05-31 16:39:25.000000000 -0500
+++ hnefatafl130527/config.mk	2013-07-12 01:45:58.685050150 -0500
@@ -93,7 +93,7 @@
 MACRO_AIM_HASH_MEM_TAB=		-DHT_AI_MINIMAX_HASH_MEM_TAB=32000000
 MACRO_AIM_HASH_MEM_COL=		-DHT_AI_MINIMAX_HASH_MEM_COL=48000000
 MACRO_UI_CLI=			-DHT_UI_CLI
-MACRO_UI_XLIB=			-DHT_UI_XLIB
+#MACRO_UI_XLIB=			-DHT_UI_XLIB
 MACROS=				-DHT_VERSION=$(VERSION)\
 				-DHT_FILE_RC_DEF=\"$(FILE_RC)\"\
 -DHT_FILE_RULE_DEF=\"$(PREFIX_DATA_RULES_CONF)/$(FILE_RULE_DEF)\"\
@@ -119,7 +119,7 @@
 # interface).
 ########################################################################
 LIB_C=		-lm
-LIB_XLIB=	-L$(PREFIX_X11_LIB) -lX11 -lXpm
+#LIB_XLIB=	-L$(PREFIX_X11_LIB) -lX11 -lXpm
 LIBS=		$(LIB_C) $(LIB_XLIB)
 
 ########################################################################
