diff -Naur jfsw.orig/jfaudiolib/Makefile.shared jfsw/jfaudiolib/Makefile.shared
--- jfsw.orig/jfaudiolib/Makefile.shared	2017-02-02 07:22:54.000000000 -0500
+++ jfsw/jfaudiolib/Makefile.shared	2018-07-12 15:25:01.996310856 -0400
@@ -30,8 +30,4 @@
   JFAUDIOLIB_LDFLAGS+= $(shell pkg-config --libs alsa)
   JFAUDIOLIB_LDFLAGS+=-lpthread
  endif
- ifeq (yes,$(shell pkg-config --exists fluidsynth && echo yes))
-  JFAUDIOLIB_HAVE_FLUIDSYNTH=1
-  JFAUDIOLIB_LDFLAGS+= $(shell pkg-config --libs fluidsynth)
- endif
 endif
