diff -Naur jfsw_src_20051009.orig/source/game.c jfsw_src_20051009/source/game.c
--- jfsw_src_20051009.orig/source/game.c	2005-10-09 15:28:24.000000000 -0400
+++ jfsw_src_20051009/source/game.c	2010-02-09 00:08:14.000000000 -0500
@@ -197,7 +197,7 @@
 TRUE, // fx on
 TRUE, // Music on
 TRUE, // talking
-TRUE, // ambient
+FALSE, // ambient
 FALSE, // Flip Stereo
 
 // Network game settings
diff -Naur jfsw_src_20051009.orig/source/menus.c jfsw_src_20051009/source/menus.c
--- jfsw_src_20051009.orig/source/menus.c	2005-10-09 15:28:24.000000000 -0400
+++ jfsw_src_20051009/source/menus.c	2010-02-09 00:26:35.000000000 -0500
@@ -196,7 +196,7 @@
     {DefInert(0, NULL), OPT_XSIDE,                            OPT_LINE(3), 0, m_defshade, 0, NULL, NULL, NULL}, 
     
     //{DefButton(btn_talking, 0, "Talking"), OPT_XS,            OPT_LINE(4), 1, m_defshade, 0, NULL, MNU_FxCheck, NULL},
-    {DefButton(btn_ambience, 0, "Ambience"), OPT_XS,          OPT_LINE(4), 1, m_defshade, 0, NULL, MNU_FxCheck, NULL},
+    {DefButton(btn_ambience, 0, "Ambience"), OPT_XS,          OPT_LINE(4), 1, m_defshade, 0, NULL, MNU_DisableButton, NULL},
     {DefButton(btn_flipstereo, 0, "Flip Stereo"), OPT_XS,     OPT_LINE(5), 1, m_defshade, 0, NULL, MNU_FxCheck, NULL},
     //{DefButton(btn_playcd, 0, "Play CD"), OPT_XS,         OPT_LINE(6), 1, m_defshade, 0, NULL, NULL, NULL},
     {DefNone}
@@ -2695,6 +2695,13 @@
     }
     
 BOOL
+MNU_DisableButton(MenuItem *item)
+    {
+    SET(item->flags, mf_disabled);
+    return (TRUE);
+	 }
+
+BOOL
 MNU_FxCheck(MenuItem *item)
     {
     if (FXDevice < 0 || !FxInitialized)
diff -Naur jfsw_src_20051009.orig/source/menus.h jfsw_src_20051009/source/menus.h
--- jfsw_src_20051009.orig/source/menus.h	2005-10-09 15:28:24.000000000 -0400
+++ jfsw_src_20051009/source/menus.h	2010-02-09 00:27:29.000000000 -0500
@@ -289,6 +289,7 @@
 
 BOOL MNU_MusicFxCheck(MenuItem_p item);
 BOOL MNU_MusicCheck(MenuItem_p item);
+BOOL MNU_DisableButton(MenuItem_p item);
 BOOL MNU_FxCheck(MenuItem_p item);
 BOOL MNU_MouseCheck(MenuItem_p item);
 BOOL MNU_BorderCheck(MenuItem_p item);
diff -Naur jfsw_src_20051009.orig/source/swconfig.c jfsw_src_20051009/source/swconfig.c
--- jfsw_src_20051009.orig/source/swconfig.c	2005-10-09 15:28:24.000000000 -0400
+++ jfsw_src_20051009/source/swconfig.c	2010-02-09 00:09:21.000000000 -0500
@@ -170,7 +170,7 @@
        
     dummy = -1;
     SCRIPT_GetNumber( scripthandle, "Options", "Ambient",&dummy);
-    if (dummy != -1) gs.Ambient = dummy;
+    gs.Ambient = 0;
        
     dummy = -1;
     SCRIPT_GetNumber( scripthandle, "Options", "FxOn",&dummy);
