diff -Naur jfsw.orig2/jfaudiolib/Makefile.shared jfsw/jfaudiolib/Makefile.shared
--- jfsw.orig2/jfaudiolib/Makefile.shared	2018-07-12 15:44:51.443280637 -0400
+++ jfsw/jfaudiolib/Makefile.shared	2018-07-12 15:43:41.666282410 -0400
@@ -16,15 +16,10 @@
   JFAUDIOLIB_HAVE_VORBIS=1
   JFAUDIOLIB_LDFLAGS+= $(shell pkg-config --libs vorbisfile)
  endif
- ifeq (yes,$(shell pkg-config --exists sdl2 && echo yes))
-  JFAUDIOLIB_HAVE_SDL=2
-  JFAUDIOLIB_LDFLAGS+= $(shell pkg-config --libs sdl2)
- else
   ifeq (yes,$(shell pkg-config --exists sdl && echo yes))
    JFAUDIOLIB_HAVE_SDL=1
    JFAUDIOLIB_LDFLAGS+= $(shell pkg-config --libs sdl)
   endif
- endif
  ifeq (yes,$(shell pkg-config --exists alsa && echo yes))
   JFAUDIOLIB_HAVE_ALSA=1
   JFAUDIOLIB_LDFLAGS+= $(shell pkg-config --libs alsa)
diff -Naur jfsw.orig2/jfbuild/Makefile.shared jfsw/jfbuild/Makefile.shared
--- jfsw.orig2/jfbuild/Makefile.shared	2017-02-04 02:45:28.000000000 -0500
+++ jfsw/jfbuild/Makefile.shared	2018-07-12 15:51:08.320271062 -0400
@@ -9,7 +9,7 @@
 GLLIBLIN=-lGL
 
 SDLCONFIG=$(shell which sdl-config)
-SDL2CONFIG=$(shell which sdl2-config)
+SDL2CONFIG=
 
 # Path to the included libsquish
 LIBSQUISH=libsquish
