diff -Naur jfsw-20210725_716e1e3/jfaudiolib/src/driver_fluidsynth.c jfsw-20210725_716e1e3.patched/jfaudiolib/src/driver_fluidsynth.c
--- jfsw-20210725_716e1e3/jfaudiolib/src/driver_fluidsynth.c	2021-10-28 17:13:18.000000000 -0400
+++ jfsw-20210725_716e1e3.patched/jfaudiolib/src/driver_fluidsynth.c	2021-10-29 02:12:45.665659393 -0400
@@ -371,6 +371,10 @@
         return FSynthErr_Error;
     }
 
+	 /* 20180712 bkw: don't try to use JACK (it doesn't work) */
+	 if(fluid_settings_str_equal(fluidsettings, "audio.driver", "jack"))
+		 fluid_settings_setstr(fluidsettings, "audio.driver", "alsa");
+
     fluidaudiodriver = new_fluid_audio_driver(fluidsettings, fluidsynth);
     if (!fluidsettings) {
         FluidSynthDrv_MIDI_Shutdown();
