diff -Naur jfsw.orig/jfaudiolib/src/driver_fluidsynth.c jfsw/jfaudiolib/src/driver_fluidsynth.c
--- jfsw.orig/jfaudiolib/src/driver_fluidsynth.c	2017-02-02 07:22:54.000000000 -0500
+++ jfsw/jfaudiolib/src/driver_fluidsynth.c	2018-07-12 05:08:37.119250512 -0400
@@ -342,6 +342,10 @@
         return FSynthErr_Error;
     }
     
+	 /* 20180712 bkw: don't try to use JACK (it doesn't work) */
+	 if(fluid_settings_str_equal(fluidsettings, "audio.driver", "jack"))
+		 fluid_settings_setstr(fluidsettings, "audio.driver", "alsa");
+
     fluidaudiodriver = new_fluid_audio_driver(fluidsettings, fluidsynth);
     if (!fluidsettings) {
         FluidSynthDrv_MIDI_Shutdown();
