diff -Naur jfsw.orig/jfaudiolib/Makefile.shared jfsw/jfaudiolib/Makefile.shared
--- jfsw.orig/jfaudiolib/Makefile.shared	2017-02-02 07:22:54.000000000 -0500
+++ jfsw/jfaudiolib/Makefile.shared	2018-07-12 15:31:55.593300348 -0400
@@ -22,7 +22,7 @@
  else
   ifeq (yes,$(shell pkg-config --exists sdl && echo yes))
    JFAUDIOLIB_HAVE_SDL=1
-   JFAUDIOLIB_LDFLAGS+= $(shell pkg-config --libs sdl1)
+   JFAUDIOLIB_LDFLAGS+= $(shell pkg-config --libs sdl)
   endif
  endif
  ifeq (yes,$(shell pkg-config --exists alsa && echo yes))
