diff -Naur jfsw.orig/jfaudiolib/src/driver_sdl.c jfsw/jfaudiolib/src/driver_sdl.c
--- jfsw.orig/jfaudiolib/src/driver_sdl.c	2017-02-02 07:22:54.000000000 -0500
+++ jfsw/jfaudiolib/src/driver_sdl.c	2018-07-12 17:08:14.907153516 -0400
@@ -375,6 +375,7 @@
 int SDLDrv_CD_Init(void)
 {
 #if (SDL_MAJOR_VERSION == 1)
+    CDstatus status;
     Uint32 inited;
     Uint32 err = 0;
     int i;
@@ -407,6 +408,8 @@
         return SDLErr_Error;
     }
     
+    status = SDL_CDStatus(CDRom);
+
     fprintf(stderr, "SDL_CD: numtracks: %d\n", CDRom->numtracks);
     for (i = 0; i < CDRom->numtracks; i++) {
         fprintf(stderr, "SDL_CD: track %d - %s, %dsec\n",
