Index: gtkglext-1.2.0/examples/Makefile.am
===================================================================
--- gtkglext-1.2.0.orig/examples/Makefile.am
+++ gtkglext-1.2.0/examples/Makefile.am
@@ -17,11 +17,7 @@ INCLUDES = \
 	-I$(top_builddir)/gdk		\
 	-I$(top_builddir)/gtk		\
 	$(GTKGLEXT_DEBUG_FLAGS)		\
-	$(EXAMPLES_DEP_CFLAGS)		\
-	-DG_DISABLE_DEPRECATED		\
-	-DGDK_DISABLE_DEPRECATED	\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
-	-DGTK_DISABLE_DEPRECATED
+	$(EXAMPLES_DEP_CFLAGS)
 
 DEPS = \
 	$(top_builddir)/gdk/$(gdkglext_targetlib)	\
Index: gtkglext-1.2.0/examples/Makefile.in
===================================================================
--- gtkglext-1.2.0.orig/examples/Makefile.in
+++ gtkglext-1.2.0/examples/Makefile.in
@@ -233,11 +233,7 @@ INCLUDES = \
 	-I$(top_builddir)/gdk		\
 	-I$(top_builddir)/gtk		\
 	$(GTKGLEXT_DEBUG_FLAGS)		\
-	$(EXAMPLES_DEP_CFLAGS)		\
-	-DG_DISABLE_DEPRECATED		\
-	-DGDK_DISABLE_DEPRECATED	\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
-	-DGTK_DISABLE_DEPRECATED
+	$(EXAMPLES_DEP_CFLAGS)
 
 
 DEPS = \
Index: gtkglext-1.2.0/gdk/Makefile.am
===================================================================
--- gtkglext-1.2.0.orig/gdk/Makefile.am
+++ gtkglext-1.2.0/gdk/Makefile.am
@@ -43,10 +43,7 @@ common_includes = \
 	-DGDK_GL_COMPILATION		\
 	-I$(top_srcdir)			\
 	$(GTKGLEXT_DEBUG_FLAGS)		\
-	$(GDKGLEXT_DEP_CFLAGS)		\
-	-DG_DISABLE_DEPRECATED		\
-	-DGDK_DISABLE_DEPRECATED	\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED
+	$(GDKGLEXT_DEP_CFLAGS)
 
 common_ldflags = \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) 	\
Index: gtkglext-1.2.0/gdk/Makefile.in
===================================================================
--- gtkglext-1.2.0.orig/gdk/Makefile.in
+++ gtkglext-1.2.0/gdk/Makefile.in
@@ -238,10 +238,7 @@ common_includes = \
 	-DGDK_GL_COMPILATION		\
 	-I$(top_srcdir)			\
 	$(GTKGLEXT_DEBUG_FLAGS)		\
-	$(GDKGLEXT_DEP_CFLAGS)		\
-	-DG_DISABLE_DEPRECATED		\
-	-DGDK_DISABLE_DEPRECATED	\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED
+	$(GDKGLEXT_DEP_CFLAGS)
 
 
 common_ldflags = \
Index: gtkglext-1.2.0/gdk/x11/Makefile.am
===================================================================
--- gtkglext-1.2.0.orig/gdk/x11/Makefile.am
+++ gtkglext-1.2.0/gdk/x11/Makefile.am
@@ -10,10 +10,7 @@ INCLUDES = \
 	-I$(top_builddir)/gdk		\
 	$(GTKGLEXT_DEBUG_FLAGS)		\
 	$(GDKGLEXT_DEP_CFLAGS)		\
-        $(GDKGLEXT_WIN_EXTRA_DEFS)	\
-	-DG_DISABLE_DEPRECATED		\
-	-DGDK_DISABLE_DEPRECATED	\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED
+        $(GDKGLEXT_WIN_EXTRA_DEFS)
 
 LDADDS = \
 	$(GDKGLEXT_DEP_LIBS)
Index: gtkglext-1.2.0/gdk/x11/Makefile.in
===================================================================
--- gtkglext-1.2.0.orig/gdk/x11/Makefile.in
+++ gtkglext-1.2.0/gdk/x11/Makefile.in
@@ -225,10 +225,7 @@ INCLUDES = \
 	-I$(top_builddir)/gdk		\
 	$(GTKGLEXT_DEBUG_FLAGS)		\
 	$(GDKGLEXT_DEP_CFLAGS)		\
-        $(GDKGLEXT_WIN_EXTRA_DEFS)	\
-	-DG_DISABLE_DEPRECATED		\
-	-DGDK_DISABLE_DEPRECATED	\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED
+        $(GDKGLEXT_WIN_EXTRA_DEFS)
 
 
 LDADDS = \
Index: gtkglext-1.2.0/gtk/Makefile.am
===================================================================
--- gtkglext-1.2.0.orig/gtk/Makefile.am
+++ gtkglext-1.2.0/gtk/Makefile.am
@@ -36,11 +36,7 @@ common_includes = \
 	-I$(top_srcdir)			\
 	-I$(top_builddir)/gdk		\
 	$(GTKGLEXT_DEBUG_FLAGS)		\
-	$(GTKGLEXT_DEP_CFLAGS)		\
-	-DG_DISABLE_DEPRECATED		\
-	-DGDK_DISABLE_DEPRECATED	\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
-	-DGTK_DISABLE_DEPRECATED
+	$(GTKGLEXT_DEP_CFLAGS)
 
 common_ldflags = \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) 	\
Index: gtkglext-1.2.0/gtk/Makefile.in
===================================================================
--- gtkglext-1.2.0.orig/gtk/Makefile.in
+++ gtkglext-1.2.0/gtk/Makefile.in
@@ -234,11 +234,7 @@ common_includes = \
 	-I$(top_srcdir)			\
 	-I$(top_builddir)/gdk		\
 	$(GTKGLEXT_DEBUG_FLAGS)		\
-	$(GTKGLEXT_DEP_CFLAGS)		\
-	-DG_DISABLE_DEPRECATED		\
-	-DGDK_DISABLE_DEPRECATED	\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
-	-DGTK_DISABLE_DEPRECATED
+	$(GTKGLEXT_DEP_CFLAGS)
 
 
 common_ldflags = \
