--- mxml-2.7.orig/Makefile.in	2012-06-14 20:05:40.319997357 -0700
+++ mxml-2.7/Makefile.in	2012-06-14 20:23:52.961000179 -0700
@@ -297,7 +297,7 @@
 
 libmxml.so.1.5:	$(LIBOBJS)
 	echo Creating $@...
-	$(DSO) $(DSOFLAGS) -o libmxml.so.1.5 $(LIBOBJS)
+	$(DSO) $(DSOFLAGS) -o libmxml.so.1.5 $(LIBOBJS) $(LIBS)
 	$(RM) libmxml.so libmxml.so.1
 	$(LN) libmxml.so.1.5 libmxml.so
 	$(LN) libmxml.so.1.5 libmxml.so.1
