diff -Naur mxml-2.6.orig/Makefile.in mxml-2.6/Makefile.in
--- mxml-2.6.orig/Makefile.in	2008-12-06 04:20:38.000000000 +0000
+++ mxml-2.6/Makefile.in	2009-06-02 07:30:21.000000000 +0000
@@ -256,7 +256,7 @@
 #
 
 libmxml.so.1.4:	$(LIBOBJS)
-	$(DSO) $(DSOFLAGS) -o libmxml.so.1.4 $(LIBOBJS)
+	$(DSO) $(DSOFLAGS) -o libmxml.so.1.4 $(LIBOBJS) $(LIBS)
 	$(RM) libmxml.so libmxml.so.1
 	$(LN) libmxml.so.1.4 libmxml.so
 	$(LN) libmxml.so.1.4 libmxml.so.1
