--- mxml-2.7.orig/configure	2012-06-14 20:05:40.322997627 -0700
+++ mxml-2.7/configure	2012-06-14 20:28:11.276000237 -0700
@@ -4715,8 +4715,8 @@
 echo "${ECHO_T}yes" >&6; }
 			LIBMXML="libmxml.so.1.5"
 			DSO="\$(CC)"
-			DSOFLAGS="$DSOFLAGS -Wl,-soname,libmxml.so.1,-rpath,\$(libdir) -shared \$(OPTIM)"
-                        LDFLAGS="$LDFLAGS -Wl,-rpath,\$(libdir)"
+			DSOFLAGS="$DSOFLAGS -Wl,-soname,libmxml.so.1 -shared \$(OPTIM)"
+                        LDFLAGS="$LDFLAGS"
 			;;
 
 		*BSD*)
