Index: trunk/configure
===================================================================
--- a/configure	2007-12-15 20:40:05.000000000 -0500
+++ b/configure	2007-12-15 20:40:37.000000000 -0500
@@ -4715,8 +4715,8 @@
 echo "${ECHO_T}yes" >&6; }
 			LIBMXML="libmxml.so.1.4"
 			DSO="\$(CC)"
-			DSOFLAGS="$DSOFLAGS -Wl,-soname,libmxml.so.1,-rpath,\$(libdir) -shared \$(OPTIM)"
-                        LDFLAGS="$LDFLAGS -Wl,-rpath,\$(libdir)"
+			DSOFLAGS="$DSOFLAGS -Wl,-soname,libmxml.so.1 -shared \$(OPTIM)"
+                        LDFLAGS="$LDFLAGS"
 			;;
 
 		*BSD*)
