diff -Naur a/k9copy.desktop b/k9copy.desktop
--- a/k9copy.desktop	2014-09-29 11:00:11.000000000 -0400
+++ b/k9copy.desktop	2020-12-25 20:56:36.228414115 -0500
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=k9copy
-Exec=k9copy -caption "%c" %i %u
+Exec=k9copy "%c" %i %u
 Comment=DVD9 to DVD5 backup tool
 Comment[fr]=Backup de DVD9 en DVD5
 Comment[tr]=DVD9 DVD5 yedekleme aracı
@@ -15,5 +15,5 @@
 Terminal=false
 Icon=k9copy
 X-DocPath=k9copy/index.html
-Categories=Qt;KDE;AudioVideo;DiscBurning;AudioVideoEditing;
+Categories=Qt;KDE;AudioVideo;DiscBurning;AudioVideoEditing;Video;
 X-KDE-StartupNotify=true
diff -Naur a/k9copy_assistant.desktop b/k9copy_assistant.desktop
--- a/k9copy_assistant.desktop	2014-09-29 11:00:12.000000000 -0400
+++ b/k9copy_assistant.desktop	2020-12-25 20:56:36.228414115 -0500
@@ -1,9 +1,7 @@
-#!/usr/bin/env xdg-open
 [Desktop Entry]
-Encoding=UTF-8
 Type=Application
 Name=k9copy assistant
-Exec=k9copy --assistant -caption "%c" %i %m  %u
+Exec=k9copy --assistant "%c" %i %m  %u
 Comment=DVD9 to DVD5 backup tool
 Comment[fr]=Backup de DVD9 en DVD5
 Comment[tr]=DVD9 DVD5 yedekleme aracı
@@ -14,9 +12,8 @@
 GenericName[tr]=DVD Vidyo Yedekleme Aracı
 GenericName[ru]=Копирование с видео-DVD
 GenericName[uk]=Копіювання відео-DVD
-MimeTypes=
 Terminal=false
 Icon=k9copy
 X-DocPath=k9copy/index.html
-Categories=Qt;KDE;AudioVideo;DiscBurning;
-X-KDE-StartupNotify=true
\ No newline at end of file
+Categories=Qt;KDE;AudioVideo;DiscBurning;Video;
+X-KDE-StartupNotify=true
diff -Naur a/k9copy_assistant_open.desktop b/k9copy_assistant_open.desktop
--- a/k9copy_assistant_open.desktop	2014-09-29 11:00:12.000000000 -0400
+++ b/k9copy_assistant_open.desktop	2020-12-25 20:57:16.993844890 -0500
@@ -11,4 +11,4 @@
 [Desktop Entry]
 X-KDE-Solid-Predicate=[[ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ] OR [ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ]]
 Type=Service
-Actions=open;
\ No newline at end of file
+Actions=open;
diff -Naur a/k9copy_open.desktop b/k9copy_open.desktop
--- a/k9copy_open.desktop	2014-09-29 11:00:12.000000000 -0400
+++ b/k9copy_open.desktop	2020-12-25 20:57:45.522142523 -0500
@@ -11,4 +11,4 @@
 [Desktop Entry]
 X-KDE-Solid-Predicate=[[ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ] OR [ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Data|VideoDvd' ]]
 Type=Service
-Actions=open;
\ No newline at end of file
+Actions=open;
