diff -Naur k9copy.orig/src/core/k9config.cpp k9copy/src/core/k9config.cpp
--- k9copy.orig/src/core/k9config.cpp	2014-09-29 11:00:12.000000000 -0400
+++ k9copy/src/core/k9config.cpp	2020-12-26 12:17:00.538310987 -0500
@@ -100,7 +100,7 @@
     m_prefAutoBurn=group.readEntry("autoburn",false);
     m_quickScan=group.readEntry("quickscan",true);
     m_prefSize=group.readEntry("dvdsize",4400);
-    m_useDvdAuthor=group.readEntry("useDvdAuthor",true);
+    m_useDvdAuthor=group.readEntry("useDvdAuthor",false);
     m_keepMenus=group.readEntry("keepMenus",true);
     m_prefDelTmpFiles=group.readEntry("deltmpfiles",false);
     m_prefReadAhead=group.readEntry("readahead",-1);
