diff -up k9copy/src/backup/k9dvdbackup.cpp.gcc7 k9copy/src/backup/k9dvdbackup.cpp
--- k9copy/src/backup/k9dvdbackup.cpp.gcc7	2014-09-29 18:55:40.000000000 +0200
+++ k9copy/src/backup/k9dvdbackup.cpp	2017-10-04 14:48:33.100692360 +0200
@@ -907,7 +907,7 @@ k9Vobu * k9DVDBackup::remapOffset(uint32
 
 
         if ((vobu1 !=NULL) && (vobu2!=NULL)) {
-            *_offset = abs(vobu1->newSector - vobu2->newSector)  | maskOffset1 ;
+            *_offset = abs((int32_t)(vobu1->newSector - vobu2->newSector))  | maskOffset1 ;
             *_offset |= maskOffset2;
             return vobu2;
         }
diff -up k9copy/src/backup/k9execcopy.cpp.gcc7 k9copy/src/backup/k9execcopy.cpp
--- k9copy/src/backup/k9execcopy.cpp.gcc7	2014-09-29 17:00:12.000000000 +0200
+++ k9copy/src/backup/k9execcopy.cpp	2017-10-04 14:51:04.899589523 +0200
@@ -306,12 +306,12 @@ void k9ExecCopy::createMkv(k9DVDTitle *_
 
 #if QT_VERSION >= 0x050000
     m_progressDialog=new QProgressDialog(k9Dialogs::getMainWidget() );
-    m_progressDialog->setCancelButton(false);
+    m_progressDialog->setCancelButton(NULL);
 
 #else
     m_progressDialog=new KProgressDialog(k9Dialogs::getMainWidget() );
 
-    m_progressDialog->showCancelButton(false);
+    m_progressDialog->showCancelButton(NULL);
 #endif
 #if QT_VERSION >= 0x050000
     m_progressDialog->setMaximum(100);
