--- Makefile.unix.mak.	2023-06-10 11:56:03.000000000 +0100
+++ Makefile.unix.mak	2023-06-21 21:10:15.499776974 +0100
@@ -52,7 +52,7 @@
 	@echo ---------------------------------------------------------------------------
 	rm -Rf "$(_FOLDER)"
 	mkdir -p "$(_FOLDER)"
-	(cd "$(_FOLDER)" && cmake -G "$(CMAKE_TYPE)" $(CMAKE_DEFINES) -DCMAKE_BUILD_TYPE=$(BUILD) $(if $(SANITIZER),-DSANITIZE_$(SANITIZER)=On) ../..) || $(if $(SANITIZER),rm -Rf "$(_FOLDER)",false)
+	(cd "$(_FOLDER)" && cmake -G "$(CMAKE_TYPE)" $(CMAKE_DEFINES) -DCMAKE_BUILD_TYPE=$(BUILD) $(if $(SANITIZER),-DSANITIZE_$(SANITIZER)=Off) ../..) || $(if $(SANITIZER),rm -Rf "$(_FOLDER)",false)
 
 .PHONY:buildall
 buildall:
