diff -ur sandbox-2.13.orig/libsandbox/Makefile.am sandbox-2.13/libsandbox/Makefile.am
--- sandbox-2.13.orig/libsandbox/Makefile.am	2018-02-19 05:50:18.000000000 -0000
+++ sandbox-2.13/libsandbox/Makefile.am	2018-06-03 12:58:04.270611236 -0000
@@ -20,7 +20,6 @@
 # and destructor will not be executed ...
 libsandbox_la_LDFLAGS = \
 	-no-undefined \
-	-nodefaultlibs \
 	-avoid-version \
 	$(LDFLAG_VER),libsandbox.map
 libsandbox_la_SOURCES = \
