--- xc5.5.1/Imakefile	1992-07-18 19:57:53.000000000 -0400
+++ patch/Imakefile	2024-06-14 07:26:46.131400787 -0400
@@ -31,7 +31,7 @@
 ## The {n,t,g,sq}roff command you use to generate ASCII docs
 ## from manuscript (ms) format.
 
-NROFF=nroff -ms
+NROFF=groff -T pdf -ms
 
 
 ## If you want to change any of the names, do so here
@@ -44,7 +44,7 @@
 ## XCONQLIB is where the period/map/scenario files will be installed.
 ## The default is amongst other X applications libraries.
 
-XCONQLIB=/tmp/judd/xconq/xconq/lib
+XCONQLIB=/usr/lib/xconq
 
 
 ## XAPPLRESDIR is where the application resource defaults are installed.
@@ -66,12 +66,12 @@
 ## Normally this is (should be) the standard X bin directory.  Modify
 ## if necessary.
 
-XCONQBIN=/usr/local/games/
+XCONQBIN=/usr/bin
 
 
 ## If your system uses different curses libs, change them here
 
-CURSESLIB=-lcurses -ltermlib
+CURSESLIB=-lcurses
 
 
 ## If your system needs strcasecmp, uncomment the definitions below.
@@ -116,7 +116,7 @@
 	$(XFONTDIR) $(CRMODE) $(BCOPY) $(SELECT) $(IS_UNIX) $(USESNF)
 
 ## Additional make suffixes supported by this file
-.SUFFIXES: $(SUFFIXES) .ms .doc .per
+.SUFFIXES: $(SUFFIXES) .ms .pdf .per
 
 
 ## ******* THE MAIN PROGRAM ***********************
@@ -152,13 +152,13 @@
 ## ******* DOCS ***********************
 
 
-.ms.doc:
+.ms.pdf:
 	$(NROFF) $< > $@
 
-docs: xconq.doc xconq2.doc custom.doc period.doc
+docs: xconq.pdf xconq2.pdf custom.pdf period.pdf
 
 clean::
-	$(RM) *.doc
+	$(RM) *.pdf
 
 
 ## ******* PERIODS ***********************
@@ -169,7 +169,7 @@
 
 .per.c:
 	$(RM) $@
-	per2c < $< > $@
+	./per2c < $< > $@
 
 standard.c: per2c lib/$(PERIOD).c
 	$(RM) $@
@@ -182,9 +182,9 @@
 ## ******* THE LIBRARY ***********************
 
 install:: ./lib
-	MakeDir($(XCONQLIB))
-	for i in `ls lib`; do (set -x; \	
-		$(INSTALL) -c $(INSTDATFLAGS) ./lib/$$i $(XCONQLIB)); done
+	MakeDir($(DESTDIR)$(XCONQLIB))
+	for i in `ls lib`; do (set -x; \
+		$(INSTALL) -c $(INSTDATFLAGS) ./lib/$$i $(DESTDIR)$(XCONQLIB)); done
 
 
 ## ******* FONTS ***********************
@@ -205,13 +205,13 @@
 ##ifdef SnfFonts
 OBJS=xconq.snf standard.snf
 ##else
-OBJS=xconq.pcf standard.pcf
+OBJS=xconq.pcf.gz standard.pcf.gz
 ##endif
 
 FontTarget(xconq)
 FontTarget(standard)
 MakeFonts()
-InstallFonts($(FONTDIR)/misc/xconq)
+InstallFonts($(FONTDIR)/misc)
 
 
 ## ******** MISCELLANEOUS ***********************
--- xc5.5.1/X11.c	1992-08-15 01:15:13.000000000 -0400
+++ patch/X11.c	2024-05-31 00:22:27.258096071 -0400
@@ -4,6 +4,8 @@
 /* This program may be used, copied, modified, and redistributed freely */
 /* for noncommercial purposes, so long as this notice remains intact.  */
 
+#include <stdlib.h>
+
 /* Interface implementations for the X11 version of xconq. */
 
 #include "config.h"
@@ -310,7 +312,7 @@
     already_been_here = TRUE;
     write_savefile("emergency.save.xconq");
     if (sig != 2)
-      abort(1);
+      abort();
     else exit(1);
   }
 }
--- xc5.5.1/curses.c	1992-07-18 19:57:59.000000000 -0400
+++ patch/curses.c	2024-06-02 00:21:45.142002331 -0400
@@ -500,8 +500,7 @@
 
 /* Beep the beeper! */
 
-beep(side)
-Side *side;
+beep(void)
 {
     putchar('\007');
 }
--- xc5.5.1/misc.h	1992-07-26 22:17:48.000000000 -0400
+++ patch/misc.h	2024-06-04 00:20:00.690228027 -0400
@@ -56,7 +56,6 @@
 #endif
 void recenter();
 #ifdef UNIX
-extern char *malloc();
 extern char *getenv();
 #endif UNIX
 extern int unwrap();
--- xc5.5.1/util.c	1992-07-26 22:17:25.000000000 -0400
+++ patch/util.c	2024-06-04 00:20:52.225579207 -0400
@@ -10,7 +10,7 @@
 #include "map.h"
 
 #ifndef DEBUG 
-char *procedure_executing[PROCSTACKSIZE], *routine_executing;
+char *procedure_executing[PROCSTACKSIZE];
 int procedure_stack_ptr = -1;
 #endif
 
