diff -Naur noteye/src/rogue.sh noteye.patched/src/rogue.sh
--- noteye/src/rogue.sh	2014-03-06 00:54:36.000000000 -0500
+++ noteye.patched/src/rogue.sh	2014-09-02 14:18:35.000000000 -0400
@@ -5,31 +5,31 @@
 
   if test -e rogue.save
   then
-    rogue rogue.save
+    ./rogue rogue.save
   else
-    rogue
+    ./rogue
   fi
 
-elif [ -n "`which rogue`" ]
+elif [ -x /usr/games/rogue ]
 then
 
   cd $NOTEYECONFIG
   if test -e rogue.save
   then
-    rogue rogue.save
+    /usr/games/rogue rogue.save
   else
-    rogue
+    /usr/games/rogue
   fi
 
 else
 
 echo Rogue not found on your system.
 echo
-echo Make sure that it can be run as \'$2\' or \'./$2\'.
+echo Make sure that it can be run as \'/usr/games/rogue\' or \'./$2\'.
 echo
-echo For example, in Ubuntu you can install Rogue with:
+echo For example, in Slackware you can install Rogue with:
 echo
-echo sudo apt-get install bsdgames-nonfree
+echo sbopkg -i rogue
 echo
 echo Otherwise, you can also run it manually from the prompt below.
 echo
