--- glyptodon.sh.old	2010-09-14 22:45:49.000000000 +0200
+++ glyptodon.sh	2010-09-22 15:52:25.000000000 +0200
@@ -196,11 +196,11 @@
 # Verifye system is compatiable and has the tools to manage POSIX capabilities
 if [ $CAPABILITIES ]
 then
-	if   ! grep -x "CONFIG_SECURITY_FILE_CAPABILITIES=y" /boot/config-`cat /proc/version | cut -d " " -f 3` &> /dev/null
-	then
-		writeAndLog "Error : Your kernel configuration does not allow file capabilities -> Abort script."
-		exit 1
-	fi
+#	if   ! grep -x "CONFIG_SECURITY_FILE_CAPABILITIES=y" /boot/config-`cat /proc/version | cut -d " " -f 3` &> /dev/null
+#	then
+#		writeAndLog "Error : Your kernel configuration does not allow file capabilities -> Abort script."
+#		exit 1
+#	fi
 	for  tool in $CAPABILITIES_TOOLS
 	do
 		if ! command -v "$tool" &>/dev/null
