diff -Nur gkrellm-volume.orig/Makefile gkrellm-volume/Makefile
--- gkrellm-volume.orig/Makefile	2004-08-19 15:07:27.000000000 -0500
+++ gkrellm-volume/Makefile	2009-11-30 00:48:22.746711831 -0600
@@ -14,7 +14,7 @@
 GTK_CFLAGS = `$(GTK_CONFIG) --cflags` 
 GTK_LIB = `$(GTK_CONFIG) --libs`
 
-FLAGS = -O2 -Wall -fPIC $(GTK_CFLAGS) $(GKRELLM_INCLUDE)
+FLAGS = -Wall -fPIC $(GTK_CFLAGS) $(GKRELLM_INCLUDE)
 LIBS = $(GTK_LIB)
 LFLAGS = -shared
 
@@ -33,6 +33,7 @@
 
 CC = gcc $(CFLAGS) $(FLAGS)
 
+MKDIR = mkdir -p
 INSTALL = install -c
 INSTALL_PROGRAM = $(INSTALL) -s
 
@@ -48,7 +49,8 @@
 
 install: 
 	(cd po && ${MAKE} install)
-	$(INSTALL_PROGRAM) volume.so $(PLUGIN_DIR)
+	$(MKDIR) $(DESTDIR)$(PLUGIN_DIR)
+	$(INSTALL_PROGRAM) volume.so $(DESTDIR)/$(PLUGIN_DIR)
 
 %.c.o: %.c
 
diff -Nur gkrellm-volume.orig/po/Makefile gkrellm-volume/po/Makefile
--- gkrellm-volume.orig/po/Makefile	2004-08-19 15:07:27.000000000 -0500
+++ gkrellm-volume/po/Makefile	2009-11-30 00:47:52.124675094 -0600
@@ -14,8 +14,8 @@
 install: 
 	$(MAKE) all
 	for f in *.mo ; do mkdir -p \
-		$(INSTALL_PREFIX)$(LOCALEDIR)/`basename $$f .mo`/LC_MESSAGES ; \
-		cp $$f $(INSTALL_PREFIX)$(LOCALEDIR)/`basename $$f .mo`/LC_MESSAGES/$(PACKAGE).mo ; done
+		$(DESTDIR)$(INSTALL_PREFIX)$(LOCALEDIR)/`basename $$f .mo`/LC_MESSAGES ; \
+		cp $$f $(DESTDIR)$(INSTALL_PREFIX)$(LOCALEDIR)/`basename $$f .mo`/LC_MESSAGES/$(PACKAGE).mo ; done
 
 %.mo: %.po
 	$(MSGFMT) -f -v -o $@ $<
