diff -Naur gkrellm-volume.orig/Makefile gkrellm-volume/Makefile
--- gkrellm-volume.orig/Makefile	2004-08-19 15:07:27.000000000 -0500
+++ gkrellm-volume/Makefile	2006-10-29 21:57:34.000000000 -0600
@@ -1,15 +1,15 @@
 # Makefile for  GKrellM volume plugin
 
 PACKAGE ?= gkrellm-volume
-LOCALEDIR ?= /usr/local/share/locale
+LOCALEDIR ?= /usr/share/locale
 
 FLAGS += -DPACKAGE="\"$(PACKAGE)\"" 
 export PACKAGE LOCALEDIR
 
 GTK_CONFIG = pkg-config gtk+-2.0
 
-PLUGIN_DIR ?= /usr/local/lib/gkrellm2/plugins
-GKRELLM_INCLUDE = -I/usr/local/include
+PLUGIN_DIR ?= /usr/lib/gkrellm2/plugins
+GKRELLM_INCLUDE = -I/usr/include
 
 GTK_CFLAGS = `$(GTK_CONFIG) --cflags` 
 GTK_LIB = `$(GTK_CONFIG) --libs`
@@ -33,6 +33,7 @@
 
 CC = gcc $(CFLAGS) $(FLAGS)
 
+MKDIR = mkdir -p
 INSTALL = install -c
 INSTALL_PROGRAM = $(INSTALL) -s
 
@@ -48,7 +49,8 @@
 
 install: 
 	(cd po && ${MAKE} install)
-	$(INSTALL_PROGRAM) volume.so $(PLUGIN_DIR)
+	$(MKDIR) $(DESTDIR)$(PLUGIN_DIR)
+	$(INSTALL_PROGRAM) volume.so $(DESTDIR)/$(PLUGIN_DIR)
 
 %.c.o: %.c
 
diff -Naur gkrellm-volume.orig/po/Makefile gkrellm-volume/po/Makefile
--- gkrellm-volume.orig/po/Makefile	2004-08-19 15:07:27.000000000 -0500
+++ gkrellm-volume/po/Makefile	2006-10-29 21:55:33.000000000 -0600
@@ -14,8 +14,8 @@
 install: 
 	$(MAKE) all
 	for f in *.mo ; do mkdir -p \
-		$(INSTALL_PREFIX)$(LOCALEDIR)/`basename $$f .mo`/LC_MESSAGES ; \
-		cp $$f $(INSTALL_PREFIX)$(LOCALEDIR)/`basename $$f .mo`/LC_MESSAGES/$(PACKAGE).mo ; done
+		$(DESTDIR)$(INSTALL_PREFIX)$(LOCALEDIR)/`basename $$f .mo`/LC_MESSAGES ; \
+		cp $$f $(DESTDIR)$(INSTALL_PREFIX)$(LOCALEDIR)/`basename $$f .mo`/LC_MESSAGES/$(PACKAGE).mo ; done
 
 %.mo: %.po
 	$(MSGFMT) -f -v -o $@ $<
