diff -Nur clamav-0.95.2.orig/etc/clamd.conf clamav-0.95.2/etc/clamd.conf
--- clamav-0.95.2.orig/etc/clamd.conf	2009-04-07 05:32:06.000000000 -0500
+++ clamav-0.95.2/etc/clamd.conf	2009-08-10 10:56:05.791514384 -0500
@@ -5,7 +5,7 @@
 
 
 # Comment or remove the line below.
-Example
+#Example
 
 # Uncomment this option to enable logging.
 # LogFile must be writable for the user running daemon.
@@ -40,12 +40,12 @@
 
 # Use system logger (can work together with LogFile).
 # Default: no
-#LogSyslog yes
+LogSyslog yes
 
 # Specify the type of syslog messages - please refer to 'man syslog'
 # for facility names.
 # Default: LOG_LOCAL6
-#LogFacility LOG_MAIL
+LogFacility LOG_MAIL
 
 # Enable verbose logging.
 # Default: no
@@ -54,7 +54,7 @@
 # This option allows you to save a process identifier of the listening
 # daemon (main thread).
 # Default: disabled
-#PidFile /var/run/clamd.pid
+PidFile /var/run/clamav/clamd.pid
 
 # Optional path to the global temporary directory.
 # Default: system specific (usually /tmp or /var/tmp).
@@ -69,11 +69,11 @@
 
 # Path to a local socket file the daemon will listen on.
 # Default: disabled (must be specified by a user)
-LocalSocket /tmp/clamd.socket
+LocalSocket /var/run/clamav/clamd.socket
 
 # Remove stale socket after unclean shutdown.
 # Default: yes
-#FixStaleSocket yes
+FixStaleSocket yes
 
 # TCP port address.
 # Default: no
@@ -167,14 +167,14 @@
 
 # Run as another user (clamd must be started by root for this option to work)
 # Default: don't drop privileges
-#User clamav
+User clamav
 
 # Initialize supplementary group access (clamd must be started by root).
 # Default: no
-#AllowSupplementaryGroups no
+AllowSupplementaryGroups yes
 
 # Stop daemon when libclamav reports out of memory condition.
-#ExitOnOOM yes
+ExitOnOOM yes
 
 # Don't fork into background.
 # Default: no
