--- Makefile	2007-03-14 15:59:20.000000000 -0500
+++ Makefile.new	2007-03-14 16:00:29.000000000 -0500
@@ -11,16 +11,16 @@
 
 # Standard pbzip2 compile
 pbzip2: pbzip2.cpp
-	$(CC) -O3 -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -o pbzip2 pbzip2.cpp -pthread -lpthread -lbz2
+	$(CC) -O3 -march=i486 -mtune=i686 -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -o pbzip2 pbzip2.cpp -pthread -lpthread -lbz2
 
 # Choose this if you want to compile in a static version of the libbz2 library
 pbzip2-static: libbz2.a pbzip2.cpp
-	$(CC) -O3 -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -o pbzip2 pbzip2.cpp -pthread -lpthread -I. -L. -lbz2
+	$(CC) -O3 -march=i486 -mtune=i686 -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -o pbzip2 pbzip2.cpp -pthread -lpthread -I. -L. -lbz2
 
 # Compatability mode for 32bit file sizes (less than 2GB) and systems
 # that have compilers that treat int as 64bit natively (ie: modern AIX)
 pbzip2-compat: pbzip2.cpp
-	$(CC) -O3 -o pbzip2 pbzip2.cpp -pthread -lpthread -lbz2
+	$(CC) -O3 -march=i486 -mtune=i686 -o pbzip2 pbzip2.cpp -pthread -lpthread -lbz2
 
 # Install the binary pbzip2 program and man page
 install: pbzip2
