diff -Naur openmw-openmw-0.50.0.orig/CMakeLists.txt openmw-openmw-0.50.0/CMakeLists.txt
--- openmw-openmw-0.50.0.orig/CMakeLists.txt	2025-10-28 18:14:40.000000000 +0100
+++ openmw-openmw-0.50.0/CMakeLists.txt	2025-11-17 08:59:12.051531000 +0100
@@ -282,10 +282,10 @@
 endif()
 
 if(NOT COLLADA_DOM_VERSION_MINOR)
-    set(COLLADA_DOM_VERSION_MINOR 5)
+    set(COLLADA_DOM_VERSION_MINOR 4)
 endif()
 
-find_package(collada_dom 2.5)
+find_package(collada_dom 2.4)
 
 option(OPENMW_USE_SYSTEM_ICU "Use system ICU library instead of internal. If disabled, requires autotools" ON)
 if(OPENMW_USE_SYSTEM_ICU)
