diff -Naur haskell-platform-2013.2.0.0.orig/scripts/build.sh haskell-platform-2013.2.0.0/scripts/build.sh
--- haskell-platform-2013.2.0.0.orig/scripts/build.sh	2013-05-04 13:56:20.000000000 -0700
+++ haskell-platform-2013.2.0.0/scripts/build.sh	2014-03-26 00:53:37.800346652 -0700
@@ -66,7 +66,7 @@
 
   [ -f Setup ] && rm Setup
 
-  GHC_PACKAGE_PATH="${ORIG_GHC_PACKAGE_PATH}" \
+  #GHC_PACKAGE_PATH="${ORIG_GHC_PACKAGE_PATH}" \
     tell ${GHC} --make Setup -o Setup -package "${CABAL_PKG_VER}" \
     || die "Compiling the Setup script failed"
   [ -x Setup ] || die "The Setup script does not exist or cannot be run"
