--- ./scripts/install.sh.orig	2010-12-01 09:49:32.169509005 +0800
+++ ./scripts/install.sh	2010-12-01 09:58:02.551384509 +0800
@@ -27,15 +27,16 @@
   ./Setup register ${VERBOSE} --gen-pkg-config="${PKG}.conf" \
     || die "Generating the registration information for the package ${PKG} failed"
 
-  if [ -f ${PKG}.conf ]; then
-    if test "${USER_INSTALL}" = "YES"; then
-      GHC_PKG_DB="--user"
-    else
-      GHC_PKG_DB="--global"
-    fi
-    ${GHC_PKG} update ${GHC_PKG_DB} "${PKG}.conf" \
-      || die "Registering the package ${PKG} failed"
-  fi
+# Uncomment to allow actually registering packages at building time
+#  if [ -f ${PKG}.conf ]; then
+#    if test "${USER_INSTALL}" = "YES"; then
+#      GHC_PKG_DB="--user"
+#    else
+#      GHC_PKG_DB="--global"
+#    fi
+#    ${GHC_PKG} update ${GHC_PKG_DB} "${PKG}.conf" \
+#      || die "Registering the package ${PKG} failed"
+#  fi
 
   cd ../..
 }
