diff -Naur OpenDUNE-0.8/src/video/video_sdl.c OpenDUNE-0.8.patched/src/video/video_sdl.c
--- OpenDUNE-0.8/src/video/video_sdl.c	2016-06-26 17:46:42.000000000 -0400
+++ OpenDUNE-0.8.patched/src/video/video_sdl.c	2017-05-31 13:27:35.163259001 -0400
@@ -362,7 +362,7 @@
 				*((__m128i *)gfx2) = mh;
 				gfx2 += 16;
 			}
-#elif defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__))
+#elif defined(__GNUC__) && defined(__MMX__)
 			/* MMX code */
 			for (x = SCREEN_WIDTH / 4; x > 0; x--) {
 				__m64 m;
