diff -Nur metakit/tcl/Makefile.in metakit.new/tcl/Makefile.in
--- metakit/tcl/Makefile.in	2015-06-18 20:37:38.000000000 +0200
+++ metakit.new/tcl/Makefile.in	2019-07-03 23:02:52.180453468 +0200
@@ -57,7 +57,7 @@
 
 PACKAGE_NAME	= @PACKAGE_NAME@
 PACKAGE_VERSION	= @PACKAGE_VERSION@
-CC		= @CC@
+CC		= g++
 CFLAGS_DEFAULT	= @CFLAGS_DEFAULT@
 CFLAGS_WARNING	= @CFLAGS_WARNING@
 EXEEXT		= @EXEEXT@
@@ -117,7 +117,7 @@
 	@echo "        xml2nroff sample.xml > sample.n"
 	@echo "        xml2html sample.xml > sample.html"
 
-install: all install-binaries install-libraries install-doc
+install: all install-binaries install-libraries
 
 install-binaries: binaries install-lib-binaries install-bin-binaries
 
