--- configure	2009-03-29 06:39:51.000000000 -0400
+++ configure	2009-04-19 17:38:50.000000000 -0400
@@ -19578,7 +19578,7 @@
 prog="
 import sys
 try:
-        import pysqlite2
+        import sqlite3 as sqlite
 except ImportError:
         sys.exit(1)
 except:
--- src/boards/python/admin/module_users.py
+++ src/boards/python/admin/module_users.py
@@ -26,7 +26,13 @@
 from gcompris import gcompris_gettext as _
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 import module
 import class_list
--- src/boards/python/admin/class_edit.py
+++ src/boards/python/admin/class_edit.py
@@ -23,7 +23,13 @@
 from gcompris import gcompris_gettext as _
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 import user_list
 
--- src/boards/python/admin/class_list.py
+++ src/boards/python/admin/class_list.py
@@ -26,7 +26,13 @@
 from gcompris import gcompris_gettext as _
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 import constants
 
--- src/boards/python/admin/module_boards.py
+++ src/boards/python/admin/module_boards.py
@@ -27,7 +27,13 @@
 import sys;
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 import module
 import board_list
--- src/boards/python/admin/user_edit.py
+++ src/boards/python/admin/user_edit.py
@@ -23,7 +23,13 @@
 from gcompris import gcompris_gettext as _
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 import user_list
 
--- src/boards/python/admin/group_list.py
+++ src/boards/python/admin/group_list.py
@@ -26,7 +26,13 @@
 from gcompris import gcompris_gettext as _
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 import group_user_list
 import group_edit
--- src/boards/python/admin/profile_list.py
+++ src/boards/python/admin/profile_list.py
@@ -26,7 +26,13 @@
 from gcompris import gcompris_gettext as _
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 import profile_group_list
 import profile_edit
--- src/boards/python/admin/log_list.py
+++ src/boards/python/admin/log_list.py
@@ -27,7 +27,13 @@
 from gcompris import gcompris_gettext as _
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 import constants
 
--- src/boards/python/admin/profile_group_list.py
+++ src/boards/python/admin/profile_group_list.py
@@ -28,7 +28,13 @@
 import constants
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 #import group_edit
 
--- src/boards/python/admin/profile_edit.py
+++ src/boards/python/admin/profile_edit.py
@@ -27,7 +27,13 @@
 import constants
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 # Group Management
 (
--- src/boards/python/admin/group_edit.py
+++ src/boards/python/admin/group_edit.py
@@ -27,7 +27,13 @@
 import constants
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 # User List Management
 (
--- src/boards/python/admin/module_groups.py
+++ src/boards/python/admin/module_groups.py
@@ -25,7 +25,13 @@
 from gcompris import gcompris_gettext as _
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 import module
 import group_list
--- src/boards/python/admin/module_reports.py
+++ src/boards/python/admin/module_reports.py
@@ -26,8 +26,14 @@
 from gcompris import gcompris_gettext as _
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
-
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
+  
 import module
 import log_list
 
--- src/boards/python/admin/module_profiles.py
+++ src/boards/python/admin/module_profiles.py
@@ -28,7 +28,13 @@
 import profile_list
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 class Profiles(module.Module):
   """Administrating GCompris Profiles"""
--- src/boards/python/admin/group_user_list.py
+++ src/boards/python/admin/group_user_list.py
@@ -28,7 +28,13 @@
 import constants
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 #import group_edit
 
--- src/boards/python/admin/board_list.py
+++ src/boards/python/admin/board_list.py
@@ -27,7 +27,13 @@
 from gcompris import gcompris_gettext as _
 
 # Database
-#from pysqlite2 import dbapi2 as sqlite
+#try:
+#  import sqlite3 as sqlite
+#except ImportError:
+#  try:
+#    from pysqlite2 import dbapi2 as sqlite
+#  except ImportError:
+#    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 # Board Management
 (
--- src/boards/python/admin/user_list.py
+++ src/boards/python/admin/user_list.py
@@ -25,7 +25,13 @@
 from gcompris import gcompris_gettext as _
 
 # Database
-from pysqlite2 import dbapi2 as sqlite
+try:
+  import sqlite3 as sqlite
+except ImportError:
+  try:
+    from pysqlite2 import dbapi2 as sqlite
+  except ImportError:
+    raise ImportError, "no module named sqlite3 or pysqlite2.dbapi2"
 
 import constants
 import user_edit
