--- xarchon-0.50/src/board.c.orig	2002-09-26 08:28:46.000000000 -0400
+++ xarchon-0.50/src/board.c	2002-09-26 08:29:02.000000000 -0400
@@ -27,7 +27,7 @@
 #define FLOOR_DARK     0                /* steppings for light and dark */
 #define FLOOR_LIGHT    7                /*   outside luminance cycle */
 
-#define FONT_NAME      "-misc-fixed-medium-*-normal-*-15-0-*-*-*-*-iso8859-1"
+#define FONT_NAME      "-misc-fixed-medium-*-normal-*-*-*-*-*-*-*-iso8859-1"
 
 #define V_CELL_YSIZE   40               /* revive cell height */
 
--- xarchon-0.50/src/field.c.orig	2002-09-26 08:22:45.000000000 -0400
+++ xarchon-0.50/src/field.c	2002-09-26 08:23:11.000000000 -0400
@@ -20,7 +20,7 @@
 /* defines                                                                  */
 /*--------------------------------------------------------------------------*/
 
-#define FONT_NAME      "-misc-fixed-medium-*-normal-*-40-0-*-*-*-*-iso8859-1"
+#define FONT_NAME      "-misc-fixed-medium-*-normal-*-*-*-*-*-*-*-iso8859-1"
 
 #define NUM_ROCKS      12
 #define ROCK_DELAY     2                /* rocks delay time in frames */
