diff -Naur xarchon-0.60.orig/src/qt/qthumaniface.cpp xarchon-0.60.joypatch/src/qt/qthumaniface.cpp
--- xarchon-0.60.orig/src/qt/qthumaniface.cpp	2002-01-09 13:34:33.000000000 -0500
+++ xarchon-0.60.joypatch/src/qt/qthumaniface.cpp	2016-08-11 08:07:19.978650355 -0400
@@ -133,7 +133,6 @@
         if (!human_joystick_init(non_keyboard)) {
 #endif
             toolkit_message_box("Joystick not available");
-            return false;
 #ifdef HAVE_LINUX_JOYSTICK_H
         }
 #endif
