diff -Naur xarchon-0.60/src/qt/qmymenubar.cpp xarchon-0.60.patched/src/qt/qmymenubar.cpp
--- xarchon-0.60/src/qt/qmymenubar.cpp	2002-01-09 13:34:33.000000000 -0500
+++ xarchon-0.60.patched/src/qt/qmymenubar.cpp	2021-10-15 16:02:15.359869566 -0400
@@ -46,7 +46,7 @@
 /*--------------------------------------------------------------------------*/
 
 int QMyMenuBar::insertItem(const QString &text, QPopupMenu *popup,
-                           int id = -1, int index = -1)
+                           int id, int index)
 {
    QMenuBar::insertItem(text, popup, id, index);
 }
@@ -58,7 +58,7 @@
 int QMyMenuBar::insertItem(const char *name, QPopupMenu *popup,
                            const QString &text,
                            const QObject *receiver, const char *member,
-                           int accel = 0, int id = -1, int index = -1)
+                           int accel, int id, int index)
 {
     int the_id = popup->insertItem(text, receiver, member, accel, id, index);
     QMyMenuBarItem *item = new QMyMenuBarItem(popup, the_id);
