diff -Naur xarchon-0.60/src/qt/qtguiwidget.cpp xarchon-0.60.patched/src/qt/qtguiwidget.cpp
--- xarchon-0.60/src/qt/qtguiwidget.cpp	2002-01-09 13:34:33.000000000 -0500
+++ xarchon-0.60.patched/src/qt/qtguiwidget.cpp	2016-10-17 21:34:55.885894979 -0400
@@ -31,8 +31,8 @@
 
     menubar->insertItem("gameArchon", popup, "New Game of Archon",
                         this, SLOT(gameArchon()));
-    menubar->insertItem("gameAdept", popup, "New Game of Adept",
-                        this, SLOT(gameAdept()));
+    // menubar->insertItem("gameAdept", popup, "New Game of Adept",
+                        // this, SLOT(gameAdept()));
     menubar->insertItem("gameUnpause", popup, "Un&pause",
                         this, SLOT(gameUnpause()), Qt::Key_Escape);
     menubar->insertItem("gameStop", popup, "&Stop",
@@ -121,7 +121,7 @@
 void QtGuiWidget::setSensitiveOptions(bool game_active)
 {
     menubar->setEnabled("gameArchon",      !game_active);
-    menubar->setEnabled("gameAdept",       !game_active);
+    // menubar->setEnabled("gameAdept",       !game_active);
     menubar->setEnabled("gameStop",         game_active);
     menubar->setEnabled("settingsPlayers", !game_active);
     menubar->setEnabled("settingsTheme",   !game_active);
