diff -Naur xarchon-0.60.orig/src/qt/qtguiwidget.cpp xarchon-0.60.menu/src/qt/qtguiwidget.cpp
--- xarchon-0.60.orig/src/qt/qtguiwidget.cpp	2002-01-09 13:34:33.000000000 -0500
+++ xarchon-0.60.menu/src/qt/qtguiwidget.cpp	2016-08-11 09:10:20.179782939 -0400
@@ -46,8 +46,6 @@
 
     menubar->insertItem("settingsPlayers", popup, "Configure &Players...",
                         this, SLOT(settingsPlayers()));
-    menubar->insertItem("settingsTheme", popup, "Select &Theme...",
-                        this, SLOT(settingsTheme()));
     menubar->insertItem("settingsSound", popup, "Toggle &Sound",
                         this, SLOT(settingsSound()), Qt::Key_F11);
     menubar->setChecked("settingsSound", TRUE);
