diff -Naur xarchon-0.60/configure.in xarchon-0.60.nogtk/configure.in
--- xarchon-0.60/configure.in	2002-01-09 13:34:33.000000000 -0500
+++ xarchon-0.60.nogtk/configure.in	2016-08-11 17:32:05.385838825 -0400
@@ -69,22 +69,6 @@
 TOOLKIT_LIBS=""
 TOOLKIT_LDADDS=""
 
-GTK_SUBDIR=""
-AM_PATH_GTK(1.2.0, have_gtk=yes, )
-if test "x$have_gtk" = xyes; then
-   AC_PATH_PROG(GLADE, glade, notfound, $PATH:$gtk_config_exec_prefix/bin)
-   if test "x$GLADE" = xnotfound; then
-       AC_MSG_WARN([Glade is missing;  the GTK+ interface will not be built])
-       have_gtk=""
-   else
-       GTK_SUBDIR="gtk"
-       TOOLKIT_LIBS="$TOOLKIT_LIBS gtk/libgtk.a"
-       TOOLKIT_LDADDS="$TOOLKIT_LDADDS $GTK_LIBS"
-       AC_DEFINE(HAVE_GTK)
-   fi
-fi
-AC_SUBST(GTK_SUBDIR)
-
 QT_SUBDIR=""
 XC_PATH_QT(whatever, have_qt=yes, )
 if test "x$have_qt" = xyes; then
@@ -104,11 +88,6 @@
 
 def_toolkit="none"
 
-AC_ARG_WITH(default-gtk,
-    [  --with-default-gtk      default to GTK+ toolkit ],
-    def_toolkit="gtk",
-)
-
 AC_ARG_WITH(default-qt,
     [  --with-default-qt       default to Qt toolkit ],
     def_toolkit="qt",
@@ -118,22 +97,11 @@
 
 case $def_toolkit in
     none)
-        if test "x$have_gtk" = xyes; then
-            DEFAULT_TOOLKIT="GTK+"
-        elif test "x$have_qt" = xyes; then
+        if test "x$have_qt" = xyes; then
             DEFAULT_TOOLKIT="Qt"
         else
             AC_MSG_ERROR([
-        Neither GTK+ nor Qt toolkits were found])
-        fi
-        ;;
-
-    gtk)
-        if test "x$have_gtk" = xyes; then
-            DEFAULT_TOOLKIT="GTK+"
-        else
-            AC_MSG_ERROR([
-        GTK+ was requested, but could not be found])
+        Qt3 toolkit not found])
         fi
         ;;
 
@@ -265,7 +233,6 @@
 AC_OUTPUT(
 Makefile
 src/Makefile
-src/gtk/Makefile
 src/qt/Makefile
 data/Makefile
 stamp.h)
diff -Naur xarchon-0.60/src/Makefile.am xarchon-0.60.nogtk/src/Makefile.am
--- xarchon-0.60/src/Makefile.am	2016-08-11 17:36:55.973849017 -0400
+++ xarchon-0.60.nogtk/src/Makefile.am	2016-08-11 17:37:05.196849341 -0400
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = @GTK_SUBDIR@ @QT_SUBDIR@
+SUBDIRS = @QT_SUBDIR@
 
 INCLUDES = @X_CFLAGS@ @ESD_CFLAGS@
 
