--- cmake/sscanf/CMakeLists.txt.orig	2026-01-01 22:38:07.474419897 +0200
+++ cmake/sscanf/CMakeLists.txt	2026-01-01 22:38:15.414419787 +0200
@@ -1,5 +1,5 @@
 
-CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5)
 PROJECT($CMAKE_PROJECT_NAME)
 add_definitions()
 include_directories()
--- cmake/rtclock/CMakeLists.txt.orig	2026-01-01 22:44:02.252414991 +0200
+++ cmake/rtclock/CMakeLists.txt	2026-01-01 22:45:15.250413981 +0200
@@ -1,5 +1,5 @@
 
-CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5)
 PROJECT($CMAKE_PROJECT_NAME)
 add_definitions()
 include_directories()
