diff -ur hydrogen-0.9.4.orig/libs/hydrogen/src/IO/portmidi_driver.cpp hydrogen-0.9.4/libs/hydrogen/src/IO/portmidi_driver.cpp
--- hydrogen-0.9.4.orig/libs/hydrogen/src/IO/portmidi_driver.cpp	2009-07-15 02:10:17.000000000 +0100
+++ hydrogen-0.9.4/libs/hydrogen/src/IO/portmidi_driver.cpp	2009-10-29 17:27:22.000000000 +0000
@@ -31,7 +31,7 @@
 #ifdef PORTMIDI_SUPPORT
 
 #include <porttime.h>
-#define TIME_PROC ((long (*)(void *)) Pt_Time)
+#define TIME_PROC ((PmTimestamp (*)(void *)) Pt_Time)
 #define TIME_START Pt_Start(1, 0, 0) /* timer started w/millisecond accuracy */
 
 #include <pthread.h>
