--- plugins/internetradio/decoder_thread.cpp.ffmpeg	2011-06-26 02:14:41.000000000 +0200
+++ plugins/internetradio/decoder_thread.cpp	2011-06-26 02:14:55.000000000 +0200
@@ -535,7 +535,7 @@
     m_av_audioStream = -1;
     for (unsigned int i = 0; i < m_av_pFormatCtx->nb_streams; i++) {
 //         if (m_av_pFormatCtx->streams[i]->codec->codec_type == CODEC_TYPE_AUDIO && m_av_audioStream < 0) {
-        if (m_av_pFormatCtx->streams[i]->codec->codec_type == CODEC_TYPE_AUDIO) { // take last stream
+        if (m_av_pFormatCtx->streams[i]->codec->codec_type == AVMEDIA_TYPE_AUDIO) { // take last stream
             m_av_audioStream = i;
         }
     }
