diff -Naur open-invaders-0.3/src/main.cc open-invaders-0.3.patched/src/main.cc
--- open-invaders-0.3/src/main.cc	2007-11-11 05:04:57.000000000 -0500
+++ open-invaders-0.3.patched/src/main.cc	2023-01-22 03:00:59.915438072 -0500
@@ -48,6 +48,11 @@
 
 int main(int argc, char *argv[])
 {
+	if(!getenv("HOME")) {
+		fprintf(stderr, "%s: HOME not set in environment, bailing.\n", argv[0]);
+		exit(1);
+	}
+
 	cout << PACKAGE_DATA_DIR << endl;
 	
 	// Parse parameters, perhaps include -windowed and -fullscreen?
