--- BasiliskII-20120218/src/Unix/configure.ac.orig	2012-02-24 10:18:59.744000107 +0100
+++ BasiliskII-20120218/src/Unix/configure.ac	2012-02-24 10:19:10.334991387 +0100
@@ -110,22 +110,6 @@
   x86_64* ) HAVE_X86_64=yes;;
 esac
 
-dnl Check if we should really be assuming x86_64 even if we detected HAVE_I386 above.
-if [[ "x$HAVE_I386" = "xyes" ]]; then
-  AC_TRY_RUN([
-    int main(void) {
-        #if defined(__x86_64__)
-                return 0;
-        #else
-                return 1;
-        #endif
-    }
-  ], [
-    HAVE_I386=no
-    HAVE_X86_64=yes
-  ])
-fi
-
 dnl Checks for programs.
 AC_PROG_CC
 AC_PROG_CC_C_O
