--- GCC/libiberty/CMakeLists.txt~	2007-12-19 06:32:15.000000000 -0600
+++ GCC/libiberty/CMakeLists.txt	2007-12-19 06:33:59.000000000 -0600
@@ -14,7 +14,7 @@
   physmem.c pex-common.c pex-one.c # pex-unix.c
   safe-ctype.c
   sort.c spaces.c splay-tree.c strerror.c
-  ternary.c unlink-if-ordinary.c xatexit.c xexit.c xmalloc.c
+  ternary.c unlink-if-ordinary.c xexit.c xmalloc.c
   xmemdup.c xstrdup.c xstrerror.c xstrndup.c
   )
