diff -Nur libixp-0.4.orig/mk/lib.mk libixp-0.4/mk/lib.mk
--- libixp-0.4.orig/mk/lib.mk	2007-11-05 08:36:22.000000000 -0600
+++ libixp-0.4/mk/lib.mk	2009-08-03 00:20:42.479105075 -0500
@@ -1,4 +1,4 @@
-PTARG = ${ROOT}/lib/${TARG}
+PTARG = ${ROOT}/@baselibdir@/${TARG}
 LIB = ${PTARG}.a
 OFILES = ${OBJ:=.o}
 
@@ -20,7 +20,7 @@
 
 ${LIB}: ${OFILES}
 	echo AR $$($(ROOT)/util/cleanname $(BASE)/$@)
-	mkdir ${ROOT}/lib 2>/dev/null || true
+	mkdir ${ROOT}/@baselibdir@ 2>/dev/null || true
 	${AR} $@ ${OFILES}
 
 include ${ROOT}/mk/common.mk
