diff -Naur libixp-0.5.orig/mk/lib.mk libixp-0.5/mk/lib.mk
--- libixp-0.5.orig/mk/lib.mk	2008-09-21 05:39:51.000000000 +0200
+++ libixp-0.5/mk/lib.mk	2011-11-15 09:59:26.333380697 +0100
@@ -1,4 +1,4 @@
-PTARG = $(ROOT)/lib/$(TARG)
+PTARG = $(ROOT)/@basedir@/$(TARG)
 LIB = $(PTARG).a
 OFILES = ${OBJ:=.o}
 
@@ -22,7 +22,7 @@
 
 $(LIB): $(OFILES)
 	echo AR $$($(CLEANNAME) $(BASE)/$@)
-	mkdir $(ROOT)/lib 2>/dev/null || true
+	mkdir $(ROOT)/@basedir@ 2>/dev/null || true
 	$(AR) $@ $(OFILES)
 
 SOMKSH=case "$(MAKESO)" in 1|[Yy][Ee][Ss]|[Tt][Rr][Uu][Ee]) echo $(ROOT)/mk/so.mk;; *) echo /dev/null;; esac
