From 209bb5d24d538a0fdd256c64ef103071b756ca72 Mon Sep 17 00:00:00 2001
From: Brenden Matthews <brenden@brndn.io>
Date: Sun, 26 Feb 2023 12:50:05 -0500
Subject: [PATCH] Require Lua 5.3 or newer

This fixes #1431.
---
 cmake/ConkyPlatformChecks.cmake | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/cmake/ConkyPlatformChecks.cmake b/cmake/ConkyPlatformChecks.cmake
index 2232c63d0..5e3715090 100644
--- a/cmake/ConkyPlatformChecks.cmake
+++ b/cmake/ConkyPlatformChecks.cmake
@@ -442,7 +442,7 @@ if(BUILD_WAYLAND)
   set(conky_includes ${conky_includes} ${PANGOFT2_INCLUDE_DIRS})
 endif(BUILD_WAYLAND)
 
-include(FindLua)
+find_package(Lua "5.3" REQUIRED)
 
 set(conky_libs ${conky_libs} ${LUA_LIBRARIES})
 set(conky_includes ${conky_includes} ${LUA_INCLUDE_DIR})
@@ -465,9 +465,9 @@ if(BUILD_X11)
     pkg_search_module(IMLIB2 REQUIRED imlib2 Imlib2)
     set(luaimlib2_libs ${IMLIB2_LIBS} ${IMLIB2_LDFLAGS} ${LUA_LIBRARIES})
     set(luaimlib2_includes
-        ${IMLIB2_INCLUDE_DIRS}
-        ${LUA_INCLUDE_DIR}
-        ${X11_INCLUDE_DIR})
+      ${IMLIB2_INCLUDE_DIRS}
+      ${LUA_INCLUDE_DIR}
+      ${X11_INCLUDE_DIR})
   endif(BUILD_LUA_IMLIB2)
 
   if(BUILD_LUA_RSVG)
