diff -Naur fontmatrix-0.6.0-Source.orig/cmake/modules/FindLIBPODOFO.cmake fontmatrix-0.6.0-Source/cmake/modules/FindLIBPODOFO.cmake
--- fontmatrix-0.6.0-Source.orig/cmake/modules/FindLIBPODOFO.cmake	2009-07-03 15:26:41.000000000 +0000
+++ fontmatrix-0.6.0-Source/cmake/modules/FindLIBPODOFO.cmake	2009-07-04 02:15:14.000000000 +0000
@@ -32,6 +32,7 @@
     "${LIBPODOFO_DIR}/src" 
     "${LIBPODOFO_DIR}"
     /usr/lib /usr/local/lib
+    /usr/lib64 /usr/local/lib64
 )
 
 IF(LIBPODOFO_INCLUDE_DIR AND LIBPODOFO_LIBRARY)
