diff -ur old.fontmatrix-0.9.99-Source/src/icushaper.cpp fontmatrix-0.9.99-Source/src/icushaper.cpp
--- old.fontmatrix-0.9.99-Source/src/icushaper.cpp	2011-10-03 08:02:54.369203955 +0200
+++ fontmatrix-0.9.99-Source/src/icushaper.cpp	2015-10-29 12:58:30.806886446 +0100
@@ -248,7 +248,7 @@
 
 }
 
-const void * IcuFontImpl::getFontTable ( LETag tableTag ) const
+const void * IcuFontImpl::getFontTable ( LETag tableTag, size_t &olength ) const
 {
 // 	qDebug()<< "IcuFontImpl::getFontTable" << OTF_tag_name( tableTag );
 	FT_Face face ( otf->face() );
@@ -263,13 +263,21 @@
 			FT_Load_Sfnt_Table ( face, tableTag, 0, bA, &length );
 			
 			regTables( tableTag,  bA );
+			olength = length;
 			return  (const void*) tables.value(tableTag);
 		}
 
 	}
+	olength = length;
 	return 0;
 }
 
+const void * IcuFontImpl::getFontTable ( LETag tableTag ) const
+{
+	size_t length(0);
+	return getFontTable( tableTag, length);
+}
+
 le_int32 IcuFontImpl::getUnitsPerEM() const
 {
 	return otf->face()->units_per_EM;
diff -ur old.fontmatrix-0.9.99-Source/src/icushaper.h fontmatrix-0.9.99-Source/src/icushaper.h
--- old.fontmatrix-0.9.99-Source/src/icushaper.h	2011-10-03 08:02:54.363203913 +0200
+++ fontmatrix-0.9.99-Source/src/icushaper.h	2015-10-29 12:55:57.983050904 +0100
@@ -27,7 +27,8 @@
 		~IcuFontImpl();
 		// implements pure virtual methods of LEFontInstance
 
-		const void* getFontTable(LETag   tableTag ) const;
+		const void* getFontTable(LETag   tableTag, size_t &length ) const;
+		const void* getFontTable(LETag   tableTag ) const;  // DEPRECATED
 		le_bool 	canDisplay (LEUnicode32 ch) const {return true;}
 		le_int32 	getUnitsPerEM () const;
 		LEGlyphID 	mapCharToGlyph (LEUnicode32 ch) const ;
