--- eispack/Makefile.gfortran	2014-05-22 12:51:00.000000000 -0400
+++ eispack/Makefile.gfortran.new	2017-04-12 12:23:00.115184305 -0400
@@ -1,14 +1,11 @@
 
 FC = gfortran
-FLG = -O -m64
+FLG = -O2
 AR = ar r
 RAN = ranlib
 
-DP =
-EISLIB = libeispack.a
-
-#DP = -fdefault-real-8
-#EISLIB = eispack_gDP.a
+DP = -fdefault-real-8
+EISLIB = eispack_gDP.a
 
 
 $(EISLIB): eispack.f
@@ -17,6 +14,6 @@
 	$(RAN) $(EISLIB)
 
 clean:
-	/bin/rm *.o *.a
+	rm *.o *.a
 
 
