--- bin/Makefile.gfortran	2017-02-12 14:25:41.000000000 -0500
+++ bin/Makefile.gfortran.new	2017-04-12 12:51:21.829538710 -0400
@@ -3,9 +3,7 @@
 # H.Youngren 2/8/17
 #
 
-#BINDIR = .
-BINDIR = /home/codes/bin/
-INSTALLCMD = install -s
+BINDIR = $(DESTDIR)/usr/bin/
 
 SRC = ../src
 
@@ -21,85 +19,24 @@
 second.o
 
 # libraries needed 
-PLTOBJ = ../plotlib/libPlt.a
-EIGOBJ = ../eispack/eispack.a
+PLTOBJ = ../plotlib/libPlt_gDP.a
+EIGOBJ = ../eispack/eispack_gDP.a
 ## EIGOBJ = eispack.o
 
-###================================================
-###  Default compilers and flags
-FC = f77
-FFLAGS = -O
-DP =
-DBG = 
-LFLG =
-
-PLTLIB = -lX11
-FTNLIB =
-#Subroutine SECOND source file
-SECOND = second_g77.f
-
-##--------------------------
-
-# Uncomment flags for desired machine...
-
-
-##--------------------------
-### Intel Fortran Compiler 
-#FC = ifort
-#OPT = -O
-##OPT = -O0 -g
-#ACHK =
-#PLTLIB = -L/usr/X11R6/lib -lX11
-#SECOND = second_ifc.f
-#FTNLIB = -Vaxlib
-#PLTOBJ = ../plotlib/libPlt.a
-#EIGOBJ = ../eispack/libeispack.a
-#DP = 
-##--------------------------
-### double-precision (also need double precision eispack and libPlt)
-#DP = -r8
-#PLTOBJ = ../plotlib/libPltDP.a
-#FFLAGS = $(OPT) $(ACHK) $(DP)
-
-
-##---------------------------------------
-##  Uncomment for gfortran
+# Compiler and flags
 FC = gfortran
-OPT = -O
-DP =
-ACHK =
-
-PLTLIB = -L/usr/X11R6/lib -lX11
 SECOND = second_g77.f
-# debug 
-#OPT = -O0 
-#DBG= -g
-#ACHK = -fbounds-check -finit-real=inf
-#ACHK = -fbounds-check -finit-real=inf -ffpe-trap=invalid,zero,denormal
-
-PLTOBJ = ../plotlib/libPlt_gSP.a
-EIGOBJ = ../eispack/libeispack.a
-
-##--------------------------
-# double precision (compile appropriate eispack and libPlt)
-#DP = -fdefault-real-8
-#PLTOBJ = ../plotlib/libPlt_gDP.a
-#EIGOBJ = ../eispack/libeispackDP.a
-##--------------------------
-
-# use -m32 for 32-bit binary, -m64 for 64-bit (check your system!)
-#FFLAGS = $(OPT) -m32 $(ACHK) $(DP)
-FFLAGS = $(OPT) -m64 $(ACHK) $(DP)
-# to make static executable
-#LFLG = -static
-##---------------------------------------
-
-
+ACHK = 
+PLTLIB=-lX11
+FTNLIB=
+DP = -fdefault-real-8
+FFLAGS = -O2 $(ACHK) $(DP)
 
 all:	 $(PROGS)
 
 install: $(PROGS)
-	$(INSTALLCMD) $(PROGS) $(BINDIR)
+	install -d -m 0755 $(BINDIR)	
+	install -m 0755 $(PROGS) $(BINDIR)
 
 clean: 
 	-/bin/rm $(PROGS)
@@ -110,7 +47,6 @@
 
 avl: $(AOBJS)
 	$(FC) -o avl $(AOBJS)  $(ALIBS) $(PLTOBJ) $(EIGOBJ) $(PLTLIB) $(FTNLIB) $(LFLG)
-	$(INSTALLCMD) avl $(BINDIR)
 
 dtest: dtest.o
 	$(FC) -o dtest dtest.o $(LFLG)
