--- eispack/Makefile	2013-10-06 19:33:49.000000000 -0400
+++ eispack/Makefile.eispack	2016-09-11 19:26:35.000000000 -0400
@@ -1,14 +1,10 @@
-
 FC = gfortran
-FLG = -O
+FLG = -O2
 AR = ar r
 RAN = ranlib
 
-DP =
-EISLIB = eispack_gSP.a
-
-#DP = -fdefault-real-8
-#EISLIB = eispack_gDP.a
+DP = -fdefault-real-8
+EISLIB = eispack_gDP.a
 
 
 $(EISLIB): eispack.f
@@ -17,6 +13,5 @@
 	$(RAN) $(EISLIB)
 
 clean:
-	/bin/rm *.o *.a
-
+	rm *.o *.a
 
