--- bin/Makefile	2013-10-06 19:34:54.000000000 -0400
+++ bin/Makefile.avl	2016-09-27 21:49:02.000000000 -0400
@@ -2,8 +2,7 @@
 # Makefile for AVL
 #
 
-BINDIR = /home/codes/bin/
-INSTALLCMD = install -s
+BINDIR = $(DESTDIR)/usr/bin/
 
 SRC = ../src
 
@@ -18,92 +17,27 @@
 matrix.o spline.o sgutil.o \
 second.o
 
-PLTOBJ = ../plotlib/libPlt.a
-EIGOBJ = ../eispack/eispack.a
-## EIGOBJ = eispack.o
-
-###================================================
-###  Default compilers and flags
-FC = f77
-FFLAGS = -O
-PLTLIB = -lX11
-FTNLIB =
+PLTOBJ = ../plotlib/libPlt_gDP.a
+EIGOBJ = ../eispack/eispack_gDP.a
 
-#Subroutine SECOND source file
-SECOND = second_g77.f
-
-##--------------------------
-
-# Uncomment flags for desired machine...
-
-##--------------------------
-### DEC Alpha with OSF and DEC f77/f90 compiler
-#FC = f77
-#FFLAGS  = -fast -O4 -tune host
-##--------------------------
-### SGI setup
-#FC = f77
-#FFLAGS  = -O2
-##--------------------------
-##  Uncomment for RS/6000
-#FFLAGS  = -O -qextname
-##--------------------------
-##  Uncomment for HP-9000
-#FFLAGS  = -O +ppu
-#FTNLIB  = -U77
-##--------------------------
-### Absoft Linux f77
-#FC = f77
-#FFLAGS  = -O -f -s -W -B108 -N34
-##--------------------------
-### f2c/gcc compiler driver 
-#FC = fort77
-#FFLAGS  =  -O2 -fomit-frame-pointer
-##--------------------------
-### GNU g77
-#FC = g77
-#FFLAGS = -O3
-##--------------------------
-### Intel Fortran Compiler 8.x
-FC = ifort
-FFLAGS = -O 
-PLTLIB = -L/usr/X11R6/lib -lX11
-SECOND = second_ifc.f
-#FTNLIB = -Vaxlib
-##--------------------------
-### double-precision option (also need double precision Eispack and libPlt)
-FFLAGS = -O -r8
-PLTOBJ = ../plotlib/libPltDP.a
-
-##---------------------------------------
-##  Uncomment for Mac OSX gfortran
 FC = gfortran
 
-ACHK = -fbounds-check -finit-real=inf -ffpe-trap=invalid,zero,denormal
-#ACHK =
-
+#ACHK = -fbounds-check -finit-real=inf -ffpe-trap=invalid,zero,denormal
+ACHK = 
+PLTLIB=-lX11
+FTNLIB=
 
-#FTNLIB = -Vaxlib
-FTNLIB = 
 SECOND = second_g77.f
 
-DP =
-PLTOBJ = ../plotlib/libPlt_gSP.a
-EIGOBJ = ../eispack/eispack_gSP.a
-
-#DP = -fdefault-real-8
-#PLTOBJ = ../plotlib/libPltDP.a
-#EIGOBJ = ../eispack/eispack_gDP.a
-
-FFLAGS = -O -m64 $(ACHK) $(DP)
-##---------------------------------------
-
+DP = -fdefault-real-8
 
+FFLAGS = -O2 $(ACHK) $(DP)
 
 all:	 $(PROGS)
 
 install: $(PROGS)
-	$(INSTALLCMD) $(PROGS) $(BINDIR)
+	install -d -m 0755 $(BINDIR)
+	install -m 0755 $(PROGS) $(BINDIR)
 
 clean: 
 	-/bin/rm $(PROGS)
@@ -114,7 +48,6 @@
 
 avl: $(AOBJS)
 	$(FC) -o avl $(AOBJS)  $(ALIBS) $(PLTOBJ) $(EIGOBJ) $(PLTLIB) $(FTNLIB)
-	$(INSTALLCMD) avl $(BINDIR)
 
 dtest: dtest.o
 	$(FC) -o dtest dtest.o
