--- test-framework.cabal.orig	2012-05-17 17:38:22.833999961 +0300
+++ test-framework.cabal	2012-05-17 17:41:57.088000279 +0300
@@ -21,6 +21,10 @@
         Description:    Build the tests
         Default:        False
 
+Flag Dynamic
+        Description:    Link executable dynamically
+        Default:        False
+
 
 Library
         Exposed-Modules:        Test.Framework
@@ -102,6 +106,8 @@
                 Cpp-Options:            -DTEST
 
                 Ghc-Options:            -Wall -threaded
+                if flag(Dynamic)
+                        Ghc-Options:            -dynamic
 
                 if impl(ghc)
                         Cpp-Options:            -DCOMPILER_GHC
