--- config.mk~	2007-10-06 08:28:56.000000000 +0200
+++ config.mk	2007-10-06 08:31:02.000000000 +0200
@@ -8,7 +8,7 @@
 # This is likely to be what you'll want for most systems:
 #
 CC=gcc
-CFLAGS=-O2 -Wall -fomit-frame-pointer -finline-functions
+CFLAGS=$(SLKCFLAGS) -Wall -fomit-frame-pointer -finline-functions
 #
 # If you're brave enough to try compiling zgv on a non-x86 system :-),
 # this might be a better bet:
@@ -37,7 +37,7 @@
 # MANDIR to directory for man page.
 # Usually it will be simpler to just set PREFIX.
 #
-PREFIX=/usr/local
+PREFIX=/usr
 
 # In theory it would be nice to put the info file and man page under
 # /usr/local/share. However, it's not clear if this is widely
@@ -52,14 +52,14 @@
 #
 #SHARE_INFIX=/share
 
-BINDIR=$(PREFIX)/bin
-INFODIR=$(PREFIX)$(SHARE_INFIX)/info
-MANDIR=$(PREFIX)$(SHARE_INFIX)/man/man1
+BINDIR=$(DESTDIR)/$(PREFIX)/bin
+INFODIR=$(DESTDIR)/$(PREFIX)$(SHARE_INFIX)/info
+MANDIR=$(DESTDIR)/$(PREFIX)$(SHARE_INFIX)/man/man1
 
 # Set the location/filename of the system-wide configuration file. You
 # may prefer to have this under /etc, for example.
 #
-RCFILE=$(PREFIX)/etc/zgv.conf
+RCFILE=$(DESTDIR)/etc/zgv.conf
 
 
 # Normally `make install' will update your `dir' file (in INFODIR),
@@ -73,7 +73,7 @@
 #
 # If you don't know what to do, leave it as-is.
 #
-#INFO_DIR_UPDATE=no
+INFO_DIR_UPDATE=no
 
 
 # ------------------- Format-related options ----------------------
@@ -91,7 +91,7 @@
 # colours will be recognised, which will mean many XPMs won't be
 # readable. This setting should be ok.
 #
-RGB_DB=/usr/X11R6/lib/X11/rgb.txt
+RGB_DB=/usr/share/X11/rgb.txt
 
 # Finally, an option for `make dvi' in the `doc' directory. You only need
 # worry about what this is set to if you plan to make a printed manual.
