Description: fix spelling errors in manpage.
Author: Joao Eriberto Mota Filho <eriberto@debian.org>
Last-Update: 2016-11-20
Index: packit-1.0/man/packit.8.in
===================================================================
--- packit-1.0.orig/man/packit.8.in
+++ packit-1.0/man/packit.8.in
@@ -82,7 +82,7 @@ Don't resolve host addresses or port num
 \fB-r\fR \fIfile\fR
 .br
 .ti +5
-Read packet data from tcpdump formated binary log \fIfile\fR. (example:
+Read packet data from tcpdump formatted binary log \fIfile\fR. (example:
 .br
 .ti +5
 a file created with -w)
@@ -281,7 +281,7 @@ The IP address of the machine you would
 \fB-o\fR \fItype of service\fR
 .br
 .ti +5
-TOS values are typically in hexidecimal format, however, packit 
+TOS values are typically in hexadecimal format, however, packit 
 .br
 .ti +5
 only accepts TOS values as integers. 
@@ -331,7 +331,7 @@ The default value is \fB128\fR.
 \fB-V\fR \fIIP protocol number\fR
 .br
 .ti +5
-Specify the IP protocol assocated with this packet (RAWIP only). 
+Specify the IP protocol associated with this packet (RAWIP only). 
 .br
 .ti +5
 The default value is \fB255\fR.
@@ -899,7 +899,7 @@ from 1-1024 with the SYN flag set and di
 .br
 
 Inject a broadcast ARP reply stating that 4.3.2.1 is at 5:4:3:2:1:0. Also, spoof the source 
-ethernet adddress for a little more authenticity and supply the payload in hex. 
+ethernet address for a little more authenticity and supply the payload in hex. 
 .br
 
 .ti +5
