Description: call to pcap-bpf.h instead of net/bpf.h.
Author: LaMont Jones <lamont@debian.org>
Last-Update: 2007-09-10
Index: packit-1.0/configure.in
===================================================================
--- packit-1.0.orig/configure.in
+++ packit-1.0/configure.in
@@ -130,7 +130,7 @@ AC_CHECK_TYPE(u_int32_t, unsigned int)
 AC_CHECK_TYPE(u_int64_t, unsigned long long)
 
 dnl Checks for header files.
-AC_CHECK_HEADERS(signal.h net/bpf.h sys/time.h)
+AC_CHECK_HEADERS(signal.h pcap-bpf.h sys/time.h)
 
 AC_ARG_WITH(libnet_includes,
     [  --with-libnet-includes=DIR    libnet include directory],
