--- packit-1.0.orig/man/packit.8.in	2003-10-03 03:57:51.000000000 -0600
+++ packit-1.0/man/packit.8.in	2016-12-16 18:20:31.066760261 -0700
@@ -503,13 +503,13 @@
 \fB-K\fR \fItype\fR
 .br
 .ti +5
-Specify the ICMP type. See docs/ICMP.txt for details on types.
+Specify the ICMP type. See the documentation in ICMP.txt.
 .br
 
 \fB-C\fR \fIcode\fR
 .br
 .ti +5
-Specify the ICMP code. See docs/ICMP.txt for details on codes.
+Specify the ICMP code.
 .br
 
 \fBECHO REQUEST / ECHO REPLY OPTIONS\fR
