Description: fix spelling errors in some files.
Author: Joao Eriberto Mota Filho
Last-Update: 2015-12-13
Index: packit-1.0/man/packit.8.in
===================================================================
--- packit-1.0.orig/man/packit.8.in
+++ packit-1.0/man/packit.8.in
@@ -665,10 +665,10 @@ Define the 32-bit original timestamp. Th
 by default.
 .br
 
-\fB-k\fR \fIrecieved timestamp\fR
+\fB-k\fR \fIreceived timestamp\fR
 .br
 .ti +5
-Define the 32-bit recieved timestamp. This value is 0  
+Define the 32-bit received timestamp. This value is 0  
 .br
 .ti +5
 by default.
Index: packit-1.0/src/print_injection.c
===================================================================
--- packit-1.0.orig/src/print_injection.c
+++ packit-1.0/src/print_injection.c
@@ -134,7 +134,7 @@ print_injection_details()
 		case ICMP_TSTAMPREPLY:
 		    fprintf(stdout, "ID: %d  Seqn: %d\n", i4hdr_o.id, i4hdr_o.seqn);
 	            fprintf(stdout, "\t     Original Timestamp: %d\n", i4hdr_o.otime);
-		    fprintf(stdout, "\t     Recieved Timestamp: %d\n", i4hdr_o.rtime);
+		    fprintf(stdout, "\t     Received Timestamp: %d\n", i4hdr_o.rtime);
 		    fprintf(stdout, "\t     Transmit Timestamp: %d", i4hdr_o.ttime);
 
                     break;
Index: packit-1.0/src/usage.c
===================================================================
--- packit-1.0.orig/src/usage.c
+++ packit-1.0/src/usage.c
@@ -107,7 +107,7 @@ print_usage()
     fprintf(stdout, "  -N id       ID number\n");
     fprintf(stdout, "  -Q seq      Sequence number\n");
     fprintf(stdout, "  -U ts       Original timestamp\n");
-    fprintf(stdout, "  -k ts       Recieved timestamp\n");
+    fprintf(stdout, "  -k ts       Received timestamp\n");
     fprintf(stdout, "  -z ts       Transmit timestamp\n");
     fprintf(stdout, "\n");
     
Index: packit-1.0/src/shape_ipv4_hdr.c
===================================================================
--- packit-1.0.orig/src/shape_ipv4_hdr.c
+++ packit-1.0/src/shape_ipv4_hdr.c
@@ -38,7 +38,7 @@ shape_ipv4_hdr(libnet_t *pkt_d)
     if(ip4hdr_o.s_addr == NULL)
     {
         if((ip4hdr_o.n_saddr = libnet_get_ipaddr4(pkt_d)) == -1)
-            fatal_error("Unable to retreive local IP address: %s", libnet_geterror(pkt_d));
+            fatal_error("Unable to retrieve local IP address: %s", libnet_geterror(pkt_d));
 
         ip4hdr_o.s_addr = libnet_addr2name4(ip4hdr_o.n_saddr, 1);
     }
