diff -Naur packit-1.0.orig/src/globals.h packit-1.0/src/globals.h
--- packit-1.0.orig/src/globals.h	2003-10-03 11:52:08.000000000 +0200
+++ packit-1.0/src/globals.h	2008-12-14 22:18:49.000000000 +0100
@@ -31,7 +31,7 @@
 #include <libnet.h>
 #include <pcap.h>
 #include <sys/types.h>
-#include <net/bpf.h>
+#include <pcap-bpf.h>
 #include <signal.h>
 #include <time.h>
 #include <stdio.h>
diff -Naur packit-1.0.orig/src/main.h packit-1.0/src/main.h
--- packit-1.0.orig/src/main.h	2003-09-27 16:22:52.000000000 +0200
+++ packit-1.0/src/main.h	2008-12-14 22:19:18.000000000 +0100
@@ -26,7 +26,7 @@
 #include <libnet.h>
 #include <pcap.h>
 #include <sys/types.h>
-#include <net/bpf.h>
+#include <pcap-bpf.h>
 #include <signal.h>
 #include <time.h>
 #include <stdio.h>
