From ea6d673d84b447b81b54c2cb825bdf8284952aea Mon Sep 17 00:00:00 2001
From: Andriy Grytsenko <andrej@rep.kiev.ua>
Date: Mon, 12 Nov 2012 23:25:39 +0200
Subject: [PATCH 05/22] [#3582493]Fixing test-fm-path two tests.

---
 src/tests/test-fm-path.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/tests/test-fm-path.c b/src/tests/test-fm-path.c
index a4a3a6c..72c340f 100644
--- a/src/tests/test-fm-path.c
+++ b/src/tests/test-fm-path.c
@@ -71,8 +71,9 @@ static void test_uri_parsing()
     TEST_PARSING(fm_path_new_for_uri, "file:/test/path",
         "/", "test", "path");
 
+    /* The 'test' in this format is a host part of URI */
     TEST_PARSING(fm_path_new_for_uri, "file://test/path",
-        "/", "test", "path");
+        "/", "path");
 
     TEST_PARSING(fm_path_new_for_uri, "http://test/path/",
         "http://test/", "path");
@@ -87,7 +88,11 @@ static void test_uri_parsing()
         "http://test/");
 
     TEST_PARSING(fm_path_new_for_uri, "http://wiki.lxde.org/zh/%E9%A6%96%E9%A0%81",
+#if 0
+        /* It should not do any break in URI since 1.0.1 */
         "http://wiki.lxde.org/", "zh", "首頁");
+#endif
+        "http://wiki.lxde.org/", "zh", "%E9%A6%96%E9%A0%81");
 
     TEST_PARSING(fm_path_new_for_uri, "mailto:test",
         "mailto:test");
-- 
1.8.0.1

