--- watchdog-5.4.orig/examples/repair.sh	2006-07-30 05:50:06.000000000 -0500
+++ watchdog-5.9/examples/repair.sh	2008-03-02 12:58:08.086346303 -0600
@@ -48,6 +48,9 @@
 	elif [ -x /etc/init.d/networking ]; then
 		# Debian
 		/etc/init.d/networking stop
+	elif [ -x /etc/rc.d/rc.inet1 ]; then
+		# Slackware
+		/etc/rc.d/rc.inet1 stop
 	else
 		ifconfig |
 		awk '/Link/ {print $1}' |
@@ -79,6 +82,9 @@
 elif [ -x /etc/rc.d/init.d/network ]; then
     # Redhat
     /etc/rc.d/init.d/network start
+elif [ -x /etc/rc.d/rc.inet1 ]; then
+    # Slackware
+    /etc/rc.d/rc.inet1 start
 else
     echo "Couldn't find network script to relaunch networking. Please edit $0" | logger -i -t repair -p daemon.info 
     exit $1
