# See: https://sourceforge.net/p/freeimage/discussion/36110/thread/605ef8e4
#
--- Source/LibWebP/src/dsp/dsp.yuv_mips_dsp_r2.c.orig	2015-02-20 03:34:38.000000000 +1000
+++ Source/LibWebP/src/dsp/dsp.yuv_mips_dsp_r2.c	2016-11-20 08:42:49.733624164 +1000
@@ -54,9 +54,9 @@
   "srl              %[temp5],   %[temp5],       24              \n\t"          \
   "srl              %[temp6],   %[temp6],       24              \n\t"          \
   "srl              %[temp7],   %[temp7],       24              \n\t"          \
-  "sb               %[temp5],   "#R"(%[dst])                    \n\t"          \
-  "sb               %[temp6],   "#G"(%[dst])                    \n\t"          \
-  "sb               %[temp7],   "#B"(%[dst])                    \n\t"          \
+  "sb               %[temp5],   " #R "(%[dst])                    \n\t"          \
+  "sb               %[temp6],   " #G "(%[dst])                    \n\t"          \
+  "sb               %[temp7],   " #B "(%[dst])                    \n\t"          \
 
 #define ASM_CLOBBER_LIST()                                                     \
   : [temp0]"=&r"(temp0), [temp1]"=&r"(temp1), [temp2]"=&r"(temp2),             \
--- Source/LibWebP/src/dsp/dsp.upsampling_mips_dsp_r2.c.orig	2015-02-20 03:34:36.000000000 +1000
+++ Source/LibWebP/src/dsp/dsp.upsampling_mips_dsp_r2.c	2016-11-20 08:39:33.163757201 +1000
@@ -34,15 +34,15 @@
     G = G - t2 + kGCst;                                                        \
     B = B + kBCst;                                                             \
     __asm__ volatile (                                                         \
-      "shll_s.w         %["#R"],      %["#R"],        9              \n\t"     \
-      "shll_s.w         %["#G"],      %["#G"],        9              \n\t"     \
-      "shll_s.w         %["#B"],      %["#B"],        9              \n\t"     \
-      "precrqu_s.qb.ph  %["#R"],      %["#R"],        $zero          \n\t"     \
-      "precrqu_s.qb.ph  %["#G"],      %["#G"],        $zero          \n\t"     \
-      "precrqu_s.qb.ph  %["#B"],      %["#B"],        $zero          \n\t"     \
-      "srl              %["#R"],      %["#R"],        24             \n\t"     \
-      "srl              %["#G"],      %["#G"],        24             \n\t"     \
-      "srl              %["#B"],      %["#B"],        24             \n\t"     \
+      "shll_s.w         %[" #R "],      %[" #R "],        9              \n\t"     \
+      "shll_s.w         %[" #G "],      %[" #G "],        9              \n\t"     \
+      "shll_s.w         %[" #B "],      %[" #B "],        9              \n\t"     \
+      "precrqu_s.qb.ph  %[" #R "],      %[" #R "],        $zero          \n\t"     \
+      "precrqu_s.qb.ph  %[" #G "],      %[" #G "],        $zero          \n\t"     \
+      "precrqu_s.qb.ph  %[" #B "],      %[" #B "],        $zero          \n\t"     \
+      "srl              %[" #R "],      %[" #R "],        24             \n\t"     \
+      "srl              %[" #G "],      %[" #G "],        24             \n\t"     \
+      "srl              %[" #B "],      %[" #B "],        24             \n\t"     \
       : [R]"+r"(R), [G]"+r"(G), [B]"+r"(B)                                     \
       :                                                                        \
     );                                                                         \
