diff -up FreeImage/Source/ZLib/gzlib.c.orig FreeImage/Source/ZLib/gzlib.c
--- FreeImage/Source/ZLib/gzlib.c.orig	2024-08-18 15:22:03.585726491 -0500
+++ FreeImage/Source/ZLib/gzlib.c	2024-08-18 15:22:29.764924485 -0500
@@ -4,6 +4,7 @@
  */
 
 #include "gzguts.h"
+#include <unistd.h>
 
 #if defined(_WIN32) && !defined(__BORLANDC__) && !defined(__MINGW32__)
 #  define LSEEK _lseeki64
diff -up FreeImage/Source/ZLib/gzread.c.orig FreeImage/Source/ZLib/gzread.c
--- FreeImage/Source/ZLib/gzread.c.orig	2024-08-18 15:22:41.952016662 -0500
+++ FreeImage/Source/ZLib/gzread.c	2024-08-18 15:23:11.521240308 -0500
@@ -4,6 +4,7 @@
  */
 
 #include "gzguts.h"
+#include <unistd.h>
 
 /* Local functions */
 local int gz_load OF((gz_statep, unsigned char *, unsigned, unsigned *));
diff -up FreeImage/Source/ZLib/gzwrite.c.orig FreeImage/Source/ZLib/gzwrite.c
--- FreeImage/Source/ZLib/gzwrite.c.orig	2024-08-18 15:23:24.238336499 -0500
+++ FreeImage/Source/ZLib/gzwrite.c	2024-08-18 15:23:41.145464386 -0500
@@ -4,6 +4,7 @@
  */
 
 #include "gzguts.h"
+#include <unistd.h>
 
 /* Local functions */
 local int gz_init OF((gz_statep));

diff -up FreeImage/Source/LibJXR/jxrgluelib/JXRGlueJxr.c.orig FreeImage/Source/LibJXR/jxrgluelib/JXRGlueJxr.c
--- FreeImage/Source/LibJXR/jxrgluelib/JXRGlueJxr.c.orig	2024-08-18 15:32:17.325371341 -0500
+++ FreeImage/Source/LibJXR/jxrgluelib/JXRGlueJxr.c	2024-08-18 15:32:44.343575960 -0500
@@ -28,7 +28,7 @@
 //*@@@---@@@@******************************************************************
 #include <limits.h>
 #include <JXRGlue.h>
-
+#include <wchar.h>
 
 static const char szHDPhotoFormat[] = "<dc:format>image/vnd.ms-photo</dc:format>";
 const U32 IFDEntryTypeSizes[] = { 0, 1, 1, 2, 4, 8, 1, 1, 2, 4, 8, 4, 8 };
