diff -Naur oblige_357_src/gui/q_bsp.cc oblige_357_src.patched//gui/q_bsp.cc
--- oblige_357_src/gui/q_bsp.cc	2010-03-07 05:24:42.000000000 -0500
+++ oblige_357_src.patched//gui/q_bsp.cc	2010-12-17 18:24:48.000000000 -0500
@@ -107,7 +107,7 @@
   // convert each newline into CR/LF pair
   while (*str)
   {
-    char *next = strchr(str, '\n');
+    char *next = (char *)strchr(str, '\n');
 
     Append(str, next ? (next - str) : strlen(str));
 
