diff -Naur oblige_357_src/gui/q_bsp.cc oblige_357_src.patched//gui/q_bsp.cc
--- oblige_357_src/gui/q_bsp.cc	2010-03-07 05:24:42.000000000 -0500
+++ oblige_357_src.patched//gui/q_bsp.cc	2010-12-17 18:24:48.000000000 -0500
@@ -107,7 +107,7 @@
   // convert each newline into CR/LF pair
   while (*str)
   {
-    char *next = strchr(str, '\n');
+    char *next = (char *)strchr(str, '\n');
 
     Append(str, next ? (next - str) : strlen(str));
 
diff -Naur oblige_357_src/gui/headers.h oblige_357_src.patched/gui/headers.h
--- oblige_357_src/gui/headers.h	2009-07-17 05:25:11.000000000 -0400
+++ oblige_357_src.patched/gui/headers.h	2012-09-20 01:45:29.000000000 -0400
@@ -29,6 +29,7 @@
 
 /* C library */
 
+#include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
