diff -Naur oblige_351_src.orig/gui/Makefile.unixy oblige_351_src/gui/Makefile.unixy
--- oblige_351_src.orig/gui/Makefile.unixy	2010-01-21 07:34:30.000000000 -0500
+++ oblige_351_src/gui/Makefile.unixy	2010-02-28 15:58:31.000000000 -0500
@@ -40,10 +40,10 @@
 FLTK_LIBS=-lfltk_images -lfltk -lX11 -lXext -lpng -ljpeg
 
 LUA_FLAGS=-I$(LUA_DIR)/src
-LUA_LIBS=$(LUA_DIR)/src/liblua.a
+LUA_LIBS=-llua
 
 GLBSP_FLAGS=-I$(GLBSP_DIR)/src
-GLBSP_LIBS=$(GLBSP_DIR)/libglbsp.a
+GLBSP_LIBS=-lglbsp
 
 CXXFLAGS=$(OPTIMISE) -Wall -DFHS_INSTALL -D$(OS) \
          $(FLTK_FLAGS) $(LUA_FLAGS) $(GLBSP_FLAGS)
diff -Naur oblige_351_src.orig/gui/hdr_lua.h oblige_351_src/gui/hdr_lua.h
--- oblige_351_src.orig/gui/hdr_lua.h	2009-07-17 05:25:10.000000000 -0400
+++ oblige_351_src/gui/hdr_lua.h	2010-02-28 15:58:02.000000000 -0500
@@ -23,9 +23,7 @@
 
 /* LUA Scripting Language */
 
-#include "lua.h"
-#include "lauxlib.h"
-#include "lualib.h"
+#include "lua.hpp"
 
 #endif // __OB_HDR_LUA__
 
diff -Naur oblige_351_src.orig/vis_viewer/Makefile oblige_351_src/vis_viewer/Makefile
--- oblige_351_src.orig/vis_viewer/Makefile	2009-10-01 05:01:34.000000000 -0400
+++ oblige_351_src/vis_viewer/Makefile	2010-02-28 15:59:43.000000000 -0500
@@ -23,15 +23,15 @@
 #--- Internal stuff from here -----------------------------------
 
 FLTK_FLAGS=-I$(FLTK_DIR)
-FLTK_LIBS=$(FLTK_DIR)/lib/libfltk_images.a \
-          $(FLTK_DIR)/lib/libfltk.a \
+FLTK_LIBS=-lfltk_images \
+          -lfltk \
           -lX11 -lXext -lpng -ljpeg
 
 LUA_FLAGS=-I$(LUA_DIR)/src
-LUA_LIBS=$(LUA_DIR)/src/liblua.a
+LUA_LIBS=-llua
 
 GLBSP_FLAGS=-I$(GLBSP_DIR)/src
-GLBSP_LIBS=$(GLBSP_DIR)/libglbsp.a
+GLBSP_LIBS=-lglbsp
 
 CXXFLAGS=$(OPTIMISE) -Wall -D$(OS) \
          $(FLTK_FLAGS) $(LUA_FLAGS) $(GLBSP_FLAGS)
