Index: gtk-engines-2.20.2/engines/clearlooks/src/animation.c
===================================================================
--- gtk-engines-2.20.2.orig/engines/clearlooks/src/animation.c
+++ gtk-engines-2.20.2/engines/clearlooks/src/animation.c
@@ -28,7 +28,7 @@
 #include "animation.h"
 
 #ifdef HAVE_WORKING_ANIMATION
-#include <glib/gtimer.h>
+#include <glib.h>
 
 struct _AnimationInfo {
 	GTimer *timer;
