--- a/data/flags/Makefile.am	2024-07-13 12:33:45.000000000 +0300
+++ b/data/flags/Makefile.am	2024-07-13 12:35:10.684982352 +0300
@@ -13,6 +13,7 @@
     br.png           \
     bt.png           \
     bw.png           \
+    by.png           \
     ca.png           \
     cd.png           \
     ch.png           \
@@ -78,6 +79,7 @@
     pt.png           \
     ro.png           \
     rs.png           \
+    ru.png           \
     se.png           \
     si.png           \
     sk.png           \
