diff -Naur cherokee-1.2.98.orig/admin/SystemStats.py cherokee-1.2.98/admin/SystemStats.py
--- cherokee-1.2.98.orig/admin/SystemStats.py	2011-05-03 18:01:41.000000000 +0900
+++ cherokee-1.2.98/admin/SystemStats.py	2011-08-26 13:05:08.128851462 +0900
@@ -39,7 +39,7 @@
     global _stats
 
     if not _stats:
-        if sys.platform == 'linux2':
+        if sys.platform == 'linux2' or sys.platform == 'linux3':
             _stats = System_stats__Linux()
         elif sys.platform == 'darwin':
             _stats = System_stats__Darwin()
diff -Naur cherokee-1.2.98.orig/admin/market/InstallUtil.py cherokee-1.2.98/admin/market/InstallUtil.py
--- cherokee-1.2.98.orig/admin/market/InstallUtil.py	2011-05-03 18:01:41.000000000 +0900
+++ cherokee-1.2.98/admin/market/InstallUtil.py	2011-08-26 13:05:06.596844132 +0900
@@ -56,7 +56,7 @@
         first_group = str(root_group)
 
     # Systems
-    if sys.platform == 'linux2':
+    if sys.platform == 'linux2' or sys.platform == 'linux3':
         if os.getuid() == 0:
             return root_group
         return first_group
@@ -72,7 +72,7 @@
 
 
 def current_UID_is_admin():
-    if sys.platform == 'linux2':
+    if sys.platform == 'linux2' or sys.platform == 'linux3':
         return os.getuid() == 0
     elif sys.platform == 'darwin':
         return os.getuid() == 0
diff -Naur cherokee-1.2.98.orig/admin/util.py cherokee-1.2.98/admin/util.py
--- cherokee-1.2.98.orig/admin/util.py	2011-03-31 17:30:04.000000000 +0900
+++ cherokee-1.2.98/admin/util.py	2011-08-26 13:05:04.622834682 +0900
@@ -341,7 +341,7 @@
 def os_get_document_root():
     if sys.platform == 'darwin':
         return "/Library/WebServer/Documents"
-    elif sys.platform == 'linux2':
+    elif sys.platform == 'linux2' or sys.platform == 'linux3':
         if os.path.exists ("/etc/redhat-release"):
             return '/var/www'
         elif os.path.exists ("/etc/fedora-release"):
