diff -Naur cherokee-1.2.101.orig/admin/configured.py.pre cherokee-1.2.101/admin/configured.py.pre
--- cherokee-1.2.101.orig/admin/configured.py.pre	2011-10-12 20:07:44.000000000 +0200
+++ cherokee-1.2.101/admin/configured.py.pre	2012-09-08 08:25:28.694838641 +0200
@@ -22,7 +22,7 @@
 CHEROKEE_DATADIR      = join (DATADIR, "cherokee")
 CHEROKEE_DEPSDIR      = join (DATADIR, "cherokee/deps")
 CHEROKEE_CONFDIR      = join (SYSCONFDIR, "cherokee")
-CHEROKEE_VAR_LOG      = join (LOCALSTATE, "log")
+CHEROKEE_VAR_LOG      = join (LOCALSTATE, "log/cherokee")
 CHEROKEE_VAR_RUN      = join (LOCALSTATE, "run")
 CHEROKEE_VAR_LIB      = join (LOCALSTATE, "lib/cherokee")
 CHEROKEE_RRD_DIR      = join (LOCALSTATE, "lib/cherokee/graphs")
diff -Naur cherokee-1.2.101.orig/cherokee.conf.sample.pre cherokee-1.2.101/cherokee.conf.sample.pre
--- cherokee-1.2.101.orig/cherokee.conf.sample.pre	2011-10-12 20:07:44.000000000 +0200
+++ cherokee-1.2.101/cherokee.conf.sample.pre	2012-09-08 08:25:17.512923553 +0200
@@ -27,11 +27,11 @@
 
 vserver!1!logger = combined
 vserver!1!logger!access!type = file
-vserver!1!logger!access!filename = %localstatedir%/log/cherokee.access
+vserver!1!logger!access!filename = %localstatedir%/log/cherokee/cherokee.access
 vserver!1!logger!access!buffsize = 16384
 
 vserver!1!error_writer!type = file
-vserver!1!error_writer!filename = %localstatedir%/log/cherokee.error
+vserver!1!error_writer!filename = %localstatedir%/log/cherokee/cherokee.error
 
 vserver!1!rule!1!match = default
 vserver!1!rule!1!handler = common
