diff -Nur kvm-88.orig/kvm/kernel/external-module-compat-comm.h kvm-88/kvm/kernel/external-module-compat-comm.h
--- kvm-88.orig/kvm/kernel/external-module-compat-comm.h	2009-07-17 20:47:08.000000000 -0300
+++ kvm-88/kvm/kernel/external-module-compat-comm.h	2009-07-17 20:55:21.000000000 -0300
@@ -845,7 +845,7 @@ 
 
 #include <linux/tracepoint.h>
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,30)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,31)
 
 struct tracepoint;

