diff -Nur kvm-88.orig/kvm/kernel/Makefile kvm-88/kvm/kernel/Makefile
--- kvm-88.orig/kvm/kernel/Makefile	2009-07-12 08:19:45.000000000 -0500
+++ kvm-88/kvm/kernel/Makefile	2009-10-05 16:11:50.668418494 -0500
@@ -49,11 +49,6 @@
 install:
 	mkdir -p $(DESTDIR)/$(INSTALLDIR)
 	cp $(ARCH_DIR)/*.ko $(DESTDIR)/$(INSTALLDIR)
-	for i in $(ORIGMODDIR)/drivers/kvm/*.ko \
-		 $(ORIGMODDIR)/arch/$(ARCH_DIR)/kvm/*.ko; do \
-		if [ -f "$$i" ]; then mv "$$i" "$$i.orig"; fi; \
-	done
-	/sbin/depmod -a $(DEPMOD_VERSION)
 
 tmpspec = .tmp.kvm-kmod.spec
 
