--- src/cairo-clock.c.orig	2018-04-07 10:47:48.240014958 +0200
+++ src/cairo-clock.c	2018-04-07 10:48:12.998016063 +0200
@@ -1423,8 +1423,8 @@
 		    g_iDefaultY >= gdk_screen_get_height (gdk_screen_get_default ()))
 			g_iDefaultY = 0;
 
-		if (g_iRefreshRate <= MIN_REFRESH_RATE ||
-		    g_iRefreshRate >= MAX_REFRESH_RATE)
+		if (g_iRefreshRate < MIN_REFRESH_RATE ||
+		    g_iRefreshRate > MAX_REFRESH_RATE)
 			g_iRefreshRate = 30;
 	}
 	else
