--- ./src/cairo-clock.c.orig	2007-08-20 06:22:45.000000000 +0200
+++ ./src/cairo-clock.c	2007-10-14 18:53:47.000000000 +0200
@@ -1455,11 +1455,16 @@
 
 	if (!gdk_screen_is_composited (gtk_widget_get_screen (g_pMainWindow)))
 	{
+		printf( "You are not running under a composited desktop-environment (e.g. compiz).\n" );
+		printf( "cairo-clock cannot work properly without one.\n" );
+		/*
+
 		gtk_window_set_icon_from_file (GTK_WINDOW (g_pErrorDialog),
 					       get_icon_filename (),
 					       NULL);
 		gtk_dialog_run (GTK_DIALOG (g_pErrorDialog));
 		exit (2);
+		*/
 	}
 
 	g_pPopUpMenu = glade_xml_get_widget (pGladeXml,
