diff -Nur mozplugger-1.14.3.orig//Makefile.in mozplugger-1.14.3/Makefile.in
--- mozplugger-1.14.3.orig//Makefile.in	2011-03-03 15:40:49.000000000 -0600
+++ mozplugger-1.14.3/Makefile.in	2011-05-08 14:10:21.550678141 -0500
@@ -152,17 +152,16 @@
 	make localinstall BROWSERDIR=.opera
 
 install:
-	-@$(MKDIR) @bindir@
-	-@$(MKDIR) $(prefix)$(libprefix)/mozilla/plugins
-	-@$(MKDIR) @mandir@/man7
-	-@$(MKDIR) @sysconfdir@
-	cp mozplugger-helper     @bindir@
-	cp mozplugger-controller @bindir@
-	cp mozplugger-linker     @bindir@
-	for target in ${PLUGINDIRS}; do cp mozplugger.so $${target}; done
-#	cp mozplugger.so $(prefix)$(libprefix)/mozilla/plugins/
-	cp $(srcdir)/mozpluggerrc  @sysconfdir@
-	cp $(srcdir)/mozplugger.7  @mandir@/man7/
+	-@$(MKDIR) $(root)/@bindir@
+	-@$(MKDIR) $(root)/@mandir@/man7
+	-@$(MKDIR) $(root)/@sysconfdir@
+	cp mozplugger-helper     $(root)/@bindir@
+	cp mozplugger-controller $(root)/@bindir@
+	cp mozplugger-linker     $(root)/@bindir@
+	for target in ${PLUGINDIRS}; do $(MKDIR) $(root)/$${target} ; cp mozplugger.so $(root)/$${target}; done
+#	cp mozplugger.so $(root)/$(prefix)$(libprefix)/mozilla/plugins/
+	cp $(srcdir)/mozpluggerrc  $(root)/@sysconfdir@
+	cp $(srcdir)/mozplugger.7  $(root)/@mandir@/man7/
 
 mozplugger.tar.gz: $(SOURCE_FILES)
 	@( DIR=`pwd`;\
