diff -Naur xroar-0.31.1/src/module.c xroar-0.31.1.patched/src/module.c
--- xroar-0.31.1/src/module.c	2013-12-04 02:38:24.000000000 -0500
+++ xroar-0.31.1.patched/src/module.c	2014-03-14 01:03:20.000000000 -0400
@@ -90,6 +90,9 @@
 #ifdef HAVE_SUN_AUDIO
 	&sound_sun_module,
 #endif
+#ifdef HAVE_ALSA_AUDIO
+	&sound_alsa_module,
+#endif
 #ifdef HAVE_OSS_AUDIO
 	&sound_oss_module,
 #endif
@@ -104,9 +107,6 @@
 #ifdef HAVE_SDL
 	&sound_sdl_module,
 #endif
-#ifdef HAVE_ALSA_AUDIO
-	&sound_alsa_module,
-#endif
 #ifdef HAVE_JACK_AUDIO
 	&sound_jack_module,
 #endif
