diff -Nru KKEdit.orig/KKEdit/app/Makefile.am KKEdit/KKEdit/app/Makefile.am
--- KKEdit.orig/KKEdit/app/Makefile.am	2013-05-24 15:21:21.000000000 +0200
+++ KKEdit/KKEdit/app/Makefile.am	2013-06-10 11:20:42.857232918 +0200
@@ -21,10 +21,12 @@
 #man1_MANS = ../resources/man/kkedit.1
 
 install-exec-hook:
-	cp ../resources/pixmaps/KKEdit.desktop /usr/share/applications/KKEdit.desktop
-	cp ../resources/pixmaps/KKEdit.png /usr/share/pixmaps
-	echo Icon=${prefix}/share/KKEdit/pixmaps/KKEdit.png >> /usr/share/applications/KKEdit.desktop
-	echo Exec=${prefix}/bin/kkedit >> /usr/share/applications/KKEdit.desktop
+	$(MKDIR_P) $(DESTDIR)${prefix}/share/applications
+	$(MKDIR_P) $(DESTDIR)${prefix}/share/pixmaps
+	cp ../resources/pixmaps/KKEdit.desktop $(DESTDIR)${prefix}/share/applications/KKEdit.desktop
+	cp ../resources/pixmaps/KKEdit.png $(DESTDIR)${prefix}/share/pixmaps
+	echo Icon=${prefix}/share/KKEdit/pixmaps/KKEdit.png >> $(DESTDIR)${prefix}/share/applications/KKEdit.desktop
+	echo Exec=${prefix}/bin/kkedit >> $(DESTDIR)${prefix}/share/applications/KKEdit.desktop
 
 uninstall-hook:
 	rm -f /usr/share/applications/KKEdit.desktop /usr/share/pixmaps/KKEdit.png
