--- Pd-0.42.5-extended/externals/pidip/configure.ac	2010-08-14 21:02:57.000000000 -0300
+++ Pd-0.42.5-extended_patched/externals/pidip/configure.ac	2011-04-04 01:59:43.000000000 -0300
@@ -148,8 +148,8 @@
 AC_CHECK_LIB(theora, theora_encode_init, , AC_MSG_ERROR(theora library needed! sorry...), -lm)
 AC_CHECK_LIB(quicktime, lqt_decode_video, , AC_MSG_ERROR(libquicktime not found) )
 
-qt_minor_version="`lqt-config --version | cut -f3 -d'.'`"
-qt_major_version="`lqt-config --version | cut -f1 -d'.'`"
+qt_minor_version="`pkg-config --modversion libquicktime | cut -f3 -d'.'`"
+qt_major_version="`pkg-config --modversion libquicktime | cut -f1 -d'.'`"
 echo "qt version minor : ${qt_minor_version} major : ${qt_major_version}"
 if test ${qt_minor_version} == "8";
 then
