diff -Nur id3-0.78.orig/charconv.cpp id3-0.78/charconv.cpp
--- id3-0.78.orig/charconv.cpp	2006-02-14 15:44:58.000000000 -0600
+++ id3-0.78/charconv.cpp	2009-07-29 00:23:24.555116104 -0500
@@ -5,6 +5,7 @@
 #include <cstdlib>
 #include <clocale>
 #include <climits>
+#include <string.h>
 #if defined(__STDC_ISO_10646__) || defined(__WIN32__)
 #  include <wchar.h>
 #  define fallback(call) (0)
diff -Nur id3-0.78.orig/getid3v2.cpp id3-0.78/getid3v2.cpp
--- id3-0.78.orig/getid3v2.cpp	2006-02-10 21:23:14.000000000 -0600
+++ id3-0.78/getid3v2.cpp	2009-07-29 00:23:38.549842736 -0500
@@ -1,5 +1,6 @@
 #include <vector>
 #include <cstdio>
+#include <string.h>
 #include "char_ucs.h"
 #include "id3v2.h"
 #include "getid3v2.h"
diff -Nur id3-0.78.orig/setfname.cpp id3-0.78/setfname.cpp
--- id3-0.78.orig/setfname.cpp	2006-03-19 05:13:36.000000000 -0600
+++ id3-0.78/setfname.cpp	2009-07-29 00:24:04.691291911 -0500
@@ -3,6 +3,7 @@
 #include <ctime>                                    // borland needs this
 #include <sys/stat.h>
 #include <utime.h>
+#include <string.h>
 #if defined(__WIN32__)
 #    include <io.h>
 #    define F_OK 0
