diff -Naur iaxclient-2.1beta3.orig/lib/codec_ffmpeg.c iaxclient-2.1beta3/lib/codec_ffmpeg.c
--- iaxclient-2.1beta3.orig/lib/codec_ffmpeg.c	2008-04-07 18:05:42.000000000 +0200
+++ iaxclient-2.1beta3/lib/codec_ffmpeg.c	2009-04-20 15:25:27.000000000 +0200
@@ -26,7 +26,7 @@
 #ifdef WIN32
 #include "libavcodec/avcodec.h"
 #else
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #endif
 
 struct slice_header_t
