diff -Naur mono-3.2.3.orig/data/config.in mono-3.2.3/data/config.in
--- mono-3.2.3.orig/data/config.in	2013-09-13 22:20:29.000000000 +0000
+++ mono-3.2.3/data/config.in	2014-01-19 22:31:49.545304766 +0000
@@ -26,6 +26,6 @@
 		<dllentry dll="__Internal" name="MoveMemory" target="mono_win32_compat_MoveMemory"/>
 		<dllentry dll="__Internal" name="ZeroMemory" target="mono_win32_compat_ZeroMemory"/>
 	</dllmap>
-	<dllmap dll="gdiplus" target="@prefix@/lib/libgdiplus@libsuffix@" />
-	<dllmap dll="gdiplus.dll" target="@prefix@/lib/libgdiplus@libsuffix@" />
+	<dllmap dll="gdiplus" target="@libdir@/libgdiplus@libsuffix@" />
+	<dllmap dll="gdiplus.dll" target="@libdir@/libgdiplus@libsuffix@" />
 </configuration>
