Description: Allow normal users to send ioctl requests
Author: Cyril Lacoux <clacoux@easter-eggs.com>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=672677
Last-Update: 2012-11-19

Index: broadcom-sta/amd64/src/wl/sys/wl_linux.c
===================================================================
--- broadcom-sta.orig/amd64/src/wl/sys/wl_linux.c
+++ broadcom-sta/amd64/src/wl/sys/wl_linux.c
@@ -1661,11 +1661,7 @@ wl_ioctl(struct net_device *dev, struct
 	}
 
 	WL_LOCK(wl);
-	if (!capable(CAP_NET_ADMIN)) {
-		bcmerror = BCME_EPERM;
-	} else {
-		bcmerror = wlc_ioctl(wl->wlc, ioc.cmd, buf, ioc.len, wlif->wlcif);
-	}
+	bcmerror = wlc_ioctl(wl->wlc, ioc.cmd, buf, ioc.len, wlif->wlcif);
 	WL_UNLOCK(wl);
 
 done1:
