Description: Remove __TIME__ and __DATE__ macros from code
Author: Cyril Lacoux <clacoux@easter-eggs.com>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=752864
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=754434
Last-Update: 2014-08-13

Index: broadcom-sta/amd64/src/wl/sys/wl_linux.c
===================================================================
--- broadcom-sta.orig/amd64/src/wl/sys/wl_linux.c
+++ broadcom-sta/amd64/src/wl/sys/wl_linux.c
@@ -726,7 +726,7 @@ wl_attach(uint16 vendor, uint16 device,
 		WL_ALL_PASSIVE_ENAB(wl) ?  ", Passive Mode" : "", EPI_VERSION_STR);
 
 #ifdef BCMDBG
-	printf(" (Compiled in " SRCBASE " at " __TIME__ " on " __DATE__ ")");
+	printf(" (Compiled in " SRCBASE);
 #endif 
 	printf("\n");
 
@@ -2051,8 +2051,7 @@ wl_osl_pcie_rc(struct wl_info *wl, uint
 void
 wl_dump_ver(wl_info_t *wl, struct bcmstrbuf *b)
 {
-	bcm_bprintf(b, "wl%d: %s %s version %s\n", wl->pub->unit,
-		__DATE__, __TIME__, EPI_VERSION_STR);
+	bcm_bprintf(b, "wl%d: version %s\n", wl->pub->unit, EPI_VERSION_STR);
 }
 
 #if defined(BCMDBG)
