diff -Nur xfce4-smartbookmark-plugin-0.4.2.orig/src/smartbookmark.desktop.in.in xfce4-smartbookmark-plugin-0.4.2/src/smartbookmark.desktop.in.in
--- xfce4-smartbookmark-plugin-0.4.2.orig/src/smartbookmark.desktop.in.in	2006-08-12 12:41:44.000000000 -0500
+++ xfce4-smartbookmark-plugin-0.4.2/src/smartbookmark.desktop.in.in	2011-10-27 21:17:39.692214866 -0500
@@ -4,5 +4,6 @@
 _Name=SmartBookmark
 _Comment=Query websites from the Xfce panel
 Icon=gtk-preferences
-X-XFCE-Module=@INTERNAL_PLUGIN_PATH@/libsmartbookmark.so
+X-XFCE-Module=smartbookmark
+X-XFCE-Module-Path=@INTERNAL_PLUGIN_PATH@
 X-XFCE-Unique=false
