https://anonhg.netbsd.org/pkgsrc/rev/b7ba46cae228

diff -Naur gst-ffmpeg-0.10.13.orig/ext/libpostproc/gstpostproc.c gst-ffmpeg-0.10.13/ext/libpostproc/gstpostproc.c
--- gst-ffmpeg-0.10.13.orig/ext/libpostproc/gstpostproc.c	2011-07-12 16:35:27.000000000 +0200
+++ gst-ffmpeg-0.10.13/ext/libpostproc/gstpostproc.c	2019-11-17 20:04:03.283000000 +0100
@@ -299,7 +299,7 @@
     ppflags = (mmx_flags & ORC_TARGET_MMX_MMX ? PP_CPU_CAPS_MMX : 0)
         | (mmx_flags & ORC_TARGET_MMX_MMXEXT ? PP_CPU_CAPS_MMX2 : 0)
         | (mmx_flags & ORC_TARGET_MMX_3DNOW ? PP_CPU_CAPS_3DNOW : 0)
-        | (altivec_flags & ORC_TARGET_ALTIVEC_ALTIVEC ? PP_CPU_CAPS_ALTIVEC :
+        | (altivec_flags & ORC_TARGET_POWERPC_ALTIVEC ? PP_CPU_CAPS_ALTIVEC :
         0);
 #else
     mmx_flags = 0;
diff -Naur gst-ffmpeg-0.10.13.orig/ext/libswscale/gstffmpegscale.c gst-ffmpeg-0.10.13/ext/libswscale/gstffmpegscale.c
--- gst-ffmpeg-0.10.13.orig/ext/libswscale/gstffmpegscale.c	2011-11-02 14:04:05.000000000 +0100
+++ gst-ffmpeg-0.10.13/ext/libswscale/gstffmpegscale.c	2019-11-17 20:05:18.947000000 +0100
@@ -638,7 +638,7 @@
   swsflags = (mmx_flags & ORC_TARGET_MMX_MMX ? SWS_CPU_CAPS_MMX : 0)
       | (mmx_flags & ORC_TARGET_MMX_MMXEXT ? SWS_CPU_CAPS_MMX2 : 0)
       | (mmx_flags & ORC_TARGET_MMX_3DNOW ? SWS_CPU_CAPS_3DNOW : 0)
-      | (altivec_flags & ORC_TARGET_ALTIVEC_ALTIVEC ? SWS_CPU_CAPS_ALTIVEC : 0);
+      | (altivec_flags & ORC_TARGET_POWERPC_ALTIVEC ? SWS_CPU_CAPS_ALTIVEC : 0);
 #else
   mmx_flags = 0;
   altivec_flags = 0;
