Author: Thorsten Alteholz <debian@alteholz.de>
Index: opencaster-3.2.2/tools/mpe2sec/mpe.c
===================================================================
--- opencaster-3.2.2.orig/tools/mpe2sec/mpe.c	2013-11-06 15:13:24.000000000 +0100
+++ opencaster-3.2.2/tools/mpe2sec/mpe.c	2013-11-06 15:13:48.000000000 +0100
@@ -12,8 +12,6 @@
 #include <sys/socket.h>
 #include <net/if.h>
 
-#include <net/if_tun.h>
-
 #include "sectioncrc.h"
 
 /* pre 2.4.6 compatibility */
