Author: Thorsten Alteholz <debian@alteholz.de>
Description: choose correct path for Debian installation
Index: opencaster/tools/oc2sec/oc-update.sh
===================================================================
--- opencaster.orig/tools/oc2sec/oc-update.sh	2013-08-20 12:31:50.000000000 +0200
+++ opencaster/tools/oc2sec/oc-update.sh	2013-08-23 14:34:13.000000000 +0200
@@ -74,7 +74,7 @@
 TEMP_DIR_SEC=`/bin/mktemp -d`
 
 #Generate the modules from the directory, the modules are stored into a tmp directory TEMP_DIR_MOD
-/usr/local/bin/file2mod.py $OCDIR $TEMP_DIR_MOD $CAROUSEL_ID $CAROUSEL_ID $ASSOCIATION_TAG $MODULE_VERSION $BLOCK_SIZE $UPDATE_FLAG $COMPRESS_MODE
+/usr/bin/file2mod.py $OCDIR $TEMP_DIR_MOD $CAROUSEL_ID $CAROUSEL_ID $ASSOCIATION_TAG $MODULE_VERSION $BLOCK_SIZE $UPDATE_FLAG $COMPRESS_MODE
 
 #Compress all modules
 if [ "$COMPRESS_MODE" = "1" ]
@@ -82,7 +82,7 @@
 	for file in $TEMP_DIR_MOD/*.mod
 	do
 		/usr/bin/du --apparent-size --block-size 1 $file > $file.size
-		/usr/local/bin/zpipe < $file > $file.z 
+		/usr/bin/zpipe < $file > $file.z
 		/bin/mv $file.z $file
 	done
 fi
@@ -95,7 +95,7 @@
 		if [ ! -f $file.solo ]
 		then
 			/usr/bin/du --apparent-size --block-size 1 $file > $file.size
-			/usr/local/bin/zpipe < $file > $file.z 
+			/usr/bin/zpipe < $file > $file.z
 			/bin/mv $file.z $file
 		else
 			/bin/rm $file.solo
@@ -104,7 +104,7 @@
 fi
 
 #Generate sections from modules, the sections are stored into a tmp directory TEMP_DIR_SEC
-/usr/local/bin/mod2sec.py $TEMP_DIR_MOD $TEMP_DIR_SEC
+/usr/bin/mod2sec.py $TEMP_DIR_MOD $TEMP_DIR_SEC
 
 # Check if it is necessary to pad every sections or not, unluckly we have found some decoders having buggy section filtering that needed this
 if [ "$PAD_ON" = "1" ]
@@ -134,7 +134,7 @@
 			/bin/cat $TEMP_DIR_SEC/0001_000000.sec >> $TEMP_DIR_SEC/temp_sec # SGW is 1 section of module id 1
 		fi
 	done
-	/usr/local/bin/sec2ts $PID < $TEMP_DIR_SEC/temp_sec > $OCDIR.ts
+	/usr/bin/sec2ts $PID < $TEMP_DIR_SEC/temp_sec > $OCDIR.ts
 fi
 
 # Delete temp files
