Author: Thorsten Alteholz <debian@alteholz.de>
Description: add tscrypt tool
Index: opencaster-3.2.1/tools/Makefile
===================================================================
--- opencaster-3.2.1.orig/tools/Makefile	2013-09-23 19:18:44.000000000 +0200
+++ opencaster-3.2.1/tools/Makefile	2013-09-23 19:19:08.000000000 +0200
@@ -10,7 +10,8 @@
 	tsoutputswitch tsccc txt2pes ts2sec \
 	pesdata2ts tsmask tsorts tsororts tspcrrestamp \
 	eitsecfilter eitsecmapper m2ts2cbrts \
-	ts2m2ts tsdiscont eitsecactualtoanother 
+	ts2m2ts tsdiscont eitsecactualtoanother \
+	tscrypt
 
 subdirs:        $(SUBDIRS)
 $(SUBDIRS):
