From b0da67ea324fd0b9880071bad69d6b4886c9019b Mon Sep 17 00:00:00 2001
From: Tokenicrat <tokenicrat@tuta.io>
Date: Sat, 14 Jun 2025 10:20:22 +0800
Subject: [PATCH] fix: Rename del_timer_sync to timer_delete_sync

---
 hid-uclogic-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hid-uclogic-core.c b/hid-uclogic-core.c
index 72680dd..9e9897c 100644
--- a/hid-uclogic-core.c
+++ b/hid-uclogic-core.c
@@ -491,7 +491,7 @@ static void uclogic_remove(struct hid_device *hdev)
 {
 	struct uclogic_drvdata *drvdata = hid_get_drvdata(hdev);
 
-	del_timer_sync(&drvdata->inrange_timer);
+	timer_delete_sync(&drvdata->inrange_timer);
 	hid_hw_stop(hdev);
 	kfree(drvdata->desc_ptr);
 	uclogic_params_cleanup(&drvdata->params);
