diff -up cpulimit-0.2/tests/busy.c.orig cpulimit-0.2/tests/busy.c
--- cpulimit-0.2/tests/busy.c.orig	2026-04-26 17:33:57.463960735 -0500
+++ cpulimit-0.2/tests/busy.c	2026-04-26 17:34:06.051737462 -0500
@@ -3,9 +3,10 @@
 #include <pthread.h>
 #include <unistd.h>
 
-void *loop()
+void *loop(void *arg)
 {
 	while(1);
+	return NULL;
 }
 
 int main(int argc, char **argv) {
@@ -23,7 +24,6 @@ int main(int argc, char **argv) {
 			exit(1);
 		}
 	}
-	loop();
+	loop(NULL);
 	return 0;
 }
-
