--- a/artix/Main.qml	2020-06-23 13:43:28.000000000 -0700
+++ b/artix/Main.qml	2022-08-16 21:06:03.751406362 -0700
@@ -44,6 +44,15 @@
         color: "transparent"
         //visible: primaryScreen
 
+        Clock {
+            id: clock
+            anchors.margins: 5
+            anchors.top: parent.top; anchors.right: parent.right
+
+            color: "white"
+            timeFont.family: "Oxygen"
+        }
+
         Image {
             id: rectangle
             anchors.centerIn: parent
