--- a/artix/Main.qml	2022-08-16 21:06:03.751406362 -0700
+++ b/artix/Main.qml	2022-08-16 21:17:32.840380440 -0700
@@ -66,13 +66,6 @@
                 anchors.centerIn: parent
                 spacing: 12
 
-                Image {
-                    mipmap: true
-                    source: "artix-banner.svg"
-                    width: parent.width
-                    fillMode: Image.PreserveAspectFit
-                }
-
                 Text {
                     anchors.horizontalCenter: parent.horizontalCenter
                     color: "#cc000000"
