diff -Naur jigdo-0.8.1/doc/Makefile.in jigdo-0.8.1.patched/doc/Makefile.in
--- jigdo-0.8.1/doc/Makefile.in	2021-10-31 14:09:53.000000000 -0400
+++ jigdo-0.8.1.patched/doc/Makefile.in	2022-03-01 12:34:10.348574996 -0500
@@ -15,12 +15,9 @@
 .SUFFIXES: .1 .sgml .html
 
 .sgml.1:
-		mkdir -p tmp && cp "$<" tmp \
-		    && cd tmp && $(DOCBOOK2MAN) "`echo $< | sed 's%.*/%%'`"
-		sed -e 's/   */ /g' <"`ls tmp/*.1`" >"$@"
-		rm -rf tmp
-#		sed -e 's%\\%\\\\%g;' <"$<" | tr -s '[:space:]' ' ' \
-#		    | $(DOCBOOKTOMAN) - >"$@"
+		$(DOCBOOK2MAN) $<
+		mv $(subst .1,.en.1,$@) $@
+
 .sgml.html:
 		rm -f "$@"
 		$(DOCBOOK2HTML) "$<" --nochunks
@@ -32,8 +29,7 @@
 .PHONY:		all clean distclean mostlyclean maintainer-clean \
 		dep depend doc check
 all:		jigdo-file.1 jigdo-file.html jigdo-lite.1 jigdo-lite.html \
-		jigdo-mirror.1 jigdo-mirror.html \
-		debian-jigdo-mini-howto.html
+		jigdo-mirror.1 jigdo-mirror.html
 doc mostlyclean dep depend check: ;
 
 clean:		Makefile
