diff -Naur jigdo-0.7.3.orig/src/mkimage.cc jigdo-0.7.3/src/mkimage.cc
--- jigdo-0.7.3.orig/src/mkimage.cc	2005-07-09 21:14:46.000000000 +0200
+++ jigdo-0.7.3/src/mkimage.cc	2017-02-17 14:47:12.884488507 +0100
@@ -285,27 +285,27 @@
 //______________________________________________________________________
 
 namespace {
-  const int SIZE_WIDTH = 12;
+  const int size_width = 12;
 }
 
 ostream& JigdoDesc::ImageInfo::put(ostream& s) const {
-  s << "image-info  " << setw(SIZE_WIDTH) << size() << "              "
+  s << "image-info  " << setw(size_width) << size() << "              "
     << md5() << ' ' << blockLength() << '\n';
   return s;
 }
 ostream& JigdoDesc::UnmatchedData::put(ostream& s) const {
-  s << "in-template " << setw(SIZE_WIDTH) << offset() << ' '
-    << setw(SIZE_WIDTH) << size() << '\n';
+  s << "in-template " << setw(size_width) << offset() << ' '
+    << setw(size_width) << size() << '\n';
   return s;
 }
 ostream& JigdoDesc::MatchedFile::put(ostream& s) const {
-  s << "need-file   " << setw(SIZE_WIDTH) << offset() << ' '
-    << setw(SIZE_WIDTH) << size() << ' ' << md5() << ' ' << rsync() << '\n';
+  s << "need-file   " << setw(size_width) << offset() << ' '
+    << setw(size_width) << size() << ' ' << md5() << ' ' << rsync() << '\n';
   return s;
 }
 ostream& JigdoDesc::WrittenFile::put(ostream& s) const {
-  s << "have-file   " << setw(SIZE_WIDTH) << offset() << ' '
-    << setw(SIZE_WIDTH) << size() << ' ' << md5() << ' ' << rsync() << '\n';
+  s << "have-file   " << setw(size_width) << offset() << ' '
+    << setw(size_width) << size() << ' ' << md5() << ' ' << rsync() << '\n';
   return s;
 }
 
