diff -u -r JSONObject-1.0.2/JSONObject.hpp JSONObject-1.0.2-patched/JSONObject.hpp
--- JSONObject-1.0.2/JSONObject.hpp	2013-03-15 10:07:13.000000000 +0000
+++ JSONObject-1.0.2-patched/JSONObject.hpp	2015-08-21 15:49:05.151811279 +0100
@@ -15,6 +15,7 @@
 #include <string>
 #include <sstream>
 #include <vector>
+#include <iostream>
 
 #include <boost/property_tree/ptree.hpp>
 #include <boost/regex.hpp>
