Index: gtkhtml-3.18.3/a11y/text.c
===================================================================
--- gtkhtml-3.18.3.orig/a11y/text.c
+++ gtkhtml-3.18.3/a11y/text.c
@@ -27,7 +27,7 @@
 #include <atk/atkhypertext.h>
 #include <atk/atkhyperlink.h>
 #include <glib/gi18n-lib.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <pango/pango.h>
 #include "gtkhtml.h"
 #include "htmlengine.h"
Index: gtkhtml-3.18.3/gtkhtml/htmlobject.c
===================================================================
--- gtkhtml-3.18.3.orig/gtkhtml/htmlobject.c
+++ gtkhtml-3.18.3/gtkhtml/htmlobject.c
@@ -23,8 +23,7 @@
 
 #include <config.h>
 #include <string.h>
-#include <glib/gdataset.h>
-#include <glib/gquark.h>
+#include <glib.h>
 
 #include "htmlclue.h"
 #include "htmlclueflow.h"
Index: gtkhtml-3.18.3/gtkhtml/test-stress.c
===================================================================
--- gtkhtml-3.18.3.orig/gtkhtml/test-stress.c
+++ gtkhtml-3.18.3/gtkhtml/test-stress.c
@@ -1,7 +1,7 @@
 #include <string.h>
 #include <stdio.h>
 #include <stdlib.h>
-#include <glib/gstring.h>
+#include <glib.h>
 #include <gtk/gtkmain.h>
 #include <gtk/gtkscrolledwindow.h>
 #include <gtk/gtkwindow.h>
Index: gtkhtml-3.18.3/gtkhtml/test-suite.c
===================================================================
--- gtkhtml-3.18.3.orig/gtkhtml/test-suite.c
+++ gtkhtml-3.18.3/gtkhtml/test-suite.c
@@ -1,6 +1,6 @@
 #include <string.h>
 #include <stdio.h>
-#include <glib/gstring.h>
+#include <glib.h>
 #include <gtk/gtkmain.h>
 #include <gtk/gtkscrolledwindow.h>
 #include <gtk/gtkwindow.h>
