diff -Nur gtkhtml-3.18.3.orig/a11y/Makefile.am gtkhtml-3.18.3/a11y/Makefile.am
--- gtkhtml-3.18.3.orig/a11y/Makefile.am	2008-04-04 04:06:50.000000000 -0500
+++ gtkhtml-3.18.3/a11y/Makefile.am	2009-06-03 10:27:22.879317264 -0500
@@ -2,7 +2,7 @@
 
 INCLUDES = -I$(top_srcdir)/gtkhtml $(GTKHTML_CFLAGS) $(I18N_LIB_CFLAGS)
 
-AM_CFLAGS = $(GAIL_CFLAGS) -DGTK_DISABLE_DEPRECATED=1 -DGDK_DISABLE_DEPRECATED=1 -DG_DISABLE_DEPRECATED=1 -DGNOME_DISABLE_DEPRECATED=1
+AM_CFLAGS = $(GAIL_CFLAGS) -DGDK_DISABLE_DEPRECATED=1 -DGNOME_DISABLE_DEPRECATED=1
 
 noinst_LTLIBRARIES = libgtkhtml-a11y.la
 
diff -Nur gtkhtml-3.18.3.orig/a11y/Makefile.in gtkhtml-3.18.3/a11y/Makefile.in
--- gtkhtml-3.18.3.orig/a11y/Makefile.in	2008-06-29 00:41:02.000000000 -0500
+++ gtkhtml-3.18.3/a11y/Makefile.in	2009-06-03 10:28:28.053709547 -0500
@@ -244,7 +244,7 @@
 top_srcdir = @top_srcdir@
 END = 
 INCLUDES = -I$(top_srcdir)/gtkhtml $(GTKHTML_CFLAGS) $(I18N_LIB_CFLAGS)
-AM_CFLAGS = $(GAIL_CFLAGS) -DGTK_DISABLE_DEPRECATED=1 -DGDK_DISABLE_DEPRECATED=1 -DG_DISABLE_DEPRECATED=1 -DGNOME_DISABLE_DEPRECATED=1
+AM_CFLAGS = $(GAIL_CFLAGS) -DGDK_DISABLE_DEPRECATED=1 -DGNOME_DISABLE_DEPRECATED=1
 noinst_LTLIBRARIES = libgtkhtml-a11y.la
 libgtkhtml_a11y_la_SOURCES = \
 	cell.h						\
diff -Nur gtkhtml-3.18.3.orig/components/html-editor/Makefile.am gtkhtml-3.18.3/components/html-editor/Makefile.am
--- gtkhtml-3.18.3.orig/components/html-editor/Makefile.am	2008-04-04 04:06:35.000000000 -0500
+++ gtkhtml-3.18.3/components/html-editor/Makefile.am	2009-06-03 10:29:16.240612316 -0500
@@ -28,7 +28,6 @@
 	-DGNOMELOCALEDIR=\""$(prefix)/$(DATADIRNAME)/locale"\"\
 	-DGLADE_DATADIR=\"$(gladedir)\"			\
 	-DGDK_DISABLE_DEPRECATED=1			\
-	-DG_DISABLE_DEPRECATED=1			\
 	-DPREFIX=\""$(prefix)"\"			\
 	-DSYSCONFDIR=\""$(sysconfdir)"\"		\
 	-DDATADIR=\""$(datadir)"\"			\
diff -Nur gtkhtml-3.18.3.orig/components/html-editor/Makefile.in gtkhtml-3.18.3/components/html-editor/Makefile.in
--- gtkhtml-3.18.3.orig/components/html-editor/Makefile.in	2008-06-29 00:41:03.000000000 -0500
+++ gtkhtml-3.18.3/components/html-editor/Makefile.in	2009-06-03 10:29:41.423922363 -0500
@@ -328,7 +328,6 @@
 	-DGNOMELOCALEDIR=\""$(prefix)/$(DATADIRNAME)/locale"\"\
 	-DGLADE_DATADIR=\"$(gladedir)\"			\
 	-DGDK_DISABLE_DEPRECATED=1			\
-	-DG_DISABLE_DEPRECATED=1			\
 	-DPREFIX=\""$(prefix)"\"			\
 	-DSYSCONFDIR=\""$(sysconfdir)"\"		\
 	-DDATADIR=\""$(datadir)"\"			\
diff -Nur gtkhtml-3.18.3.orig/gtkhtml/Makefile.am gtkhtml-3.18.3/gtkhtml/Makefile.am
--- gtkhtml-3.18.3.orig/gtkhtml/Makefile.am	2008-04-04 04:06:49.000000000 -0500
+++ gtkhtml-3.18.3/gtkhtml/Makefile.am	2009-06-03 10:30:32.202760845 -0500
@@ -20,7 +20,6 @@
 
 AM_CFLAGS = 						\
 	-DGDK_DISABLE_DEPRECATED=1			\
-	-DG_DISABLE_DEPRECATED=1			\
 	-DGNOME_DISABLE_DEPRECATED=1			\
 	-DPREFIX=\""$(prefix)"\"			\
 	-DLIBDIR=\""$(datadir)"\"			\
@@ -30,7 +29,6 @@
         -DGTKHTML_DATADIR=\""$(gtkhtml_data)"\"		\
 	-DGLADE_DATADIR=\"$(gladedir)\"			\
         -DGNOMELOCALEDIR=\""$(prefix)/$(DATADIRNAME)/locale"\"
-#	-DGTK_DISABLE_DEPRECATED=1
 
 lib_LTLIBRARIES = libgtkhtml-3.14.la
 
diff -Nur gtkhtml-3.18.3.orig/gtkhtml/Makefile.in gtkhtml-3.18.3/gtkhtml/Makefile.in
--- gtkhtml-3.18.3.orig/gtkhtml/Makefile.in	2008-06-29 00:41:03.000000000 -0500
+++ gtkhtml-3.18.3/gtkhtml/Makefile.in	2009-06-03 10:30:53.710234099 -0500
@@ -384,7 +384,6 @@
 
 AM_CFLAGS = \
 	-DGDK_DISABLE_DEPRECATED=1			\
-	-DG_DISABLE_DEPRECATED=1			\
 	-DGNOME_DISABLE_DEPRECATED=1			\
 	-DPREFIX=\""$(prefix)"\"			\
 	-DLIBDIR=\""$(datadir)"\"			\
@@ -395,7 +394,6 @@
 	-DGLADE_DATADIR=\"$(gladedir)\"			\
         -DGNOMELOCALEDIR=\""$(prefix)/$(DATADIRNAME)/locale"\"
 
-#	-DGTK_DISABLE_DEPRECATED=1
 lib_LTLIBRARIES = libgtkhtml-3.14.la
 libgtkhtmldir = $(includedir)/libgtkhtml-@GTKHTML_API_VERSION@/gtkhtml
 
