--- scorched/src/common/weapons/AccessoryStore.cpp.orig	2017-07-07 22:27:57.777261322 +0700
+++ scorched/src/common/weapons/AccessoryStore.cpp	2017-07-07 22:39:34.368313611 +0700
@@ -156,7 +156,7 @@
 	Accessory *parent, XMLNode *currentNode)
 {
 	XMLNode *typeNode = 0;
-	if (!currentNode->getNamedParameter("type", typeNode)) return false;
+	if (!currentNode->getNamedParameter("type", typeNode)) return 0;
 
 	AccessoryPart *accessoryPart = 
 		AccessoryMetaRegistration::getNewAccessory(typeNode->getContent(), this);
