diff -Naur zathura-0.3.3/zathura/zathura.c zathura-0.3.3.patched/zathura/zathura.c
--- zathura-0.3.3/zathura/zathura.c	2015-04-13 16:54:30.000000000 -0400
+++ zathura-0.3.3.patched/zathura/zathura.c	2015-05-11 14:45:31.000000000 -0400
@@ -3,6 +3,7 @@
 #define _DEFAULT_SOURCE
 #define _XOPEN_SOURCE 700
 
+#include <dlfcn.h>
 #include <errno.h>
 #include <stdlib.h>
 #include <unistd.h>
@@ -116,6 +117,7 @@
   }
 
   /* load plugins */
+  dlopen("libjpeg.so", RTLD_NOW | RTLD_GLOBAL); /* dirty hack! */
   zathura_plugin_manager_load(zathura->plugins.manager);
 
   /* configuration */
