diff -Naur zathura-0.5.1/data/bash-completion.in zathura-0.5.1.patched/data/bash-completion.in
--- zathura-0.5.1/data/bash-completion.in	2022-09-08 18:17:27.000000000 -0400
+++ zathura-0.5.1.patched/data/bash-completion.in	2022-11-14 14:00:24.545471924 -0500
@@ -4,6 +4,7 @@
 
   local EXTS=""
   for PLUGIN in @PLUGINDIR@/lib*.so; do
+    PLUGIN="$( realpath "$PLUGIN" )"
     case ${PLUGIN##*/} in
       libpdf-poppler.so)
         EXTS="$EXTS|pdf|PDF"
diff -Naur zathura-0.5.1/data/fish-completion.in zathura-0.5.1.patched/data/fish-completion.in
--- zathura-0.5.1/data/fish-completion.in	2022-09-08 18:17:27.000000000 -0400
+++ zathura-0.5.1.patched/data/fish-completion.in	2022-11-14 14:02:01.958086580 -0500
@@ -1,6 +1,7 @@
 # Complete custom suffix based on libraries installed
 function __fish_complete_zathura
     for plugin in @PLUGINDIR@/*.so
+        set plugin (realpath $plugin)
         switch (basename $plugin)
             case libpdf-poppler.so
                 __fish_complete_suffix .pdf
diff -Naur zathura-0.5.1/data/zsh-completion.in zathura-0.5.1.patched/data/zsh-completion.in
--- zathura-0.5.1/data/zsh-completion.in	2022-09-08 18:17:27.000000000 -0400
+++ zathura-0.5.1.patched/data/zsh-completion.in	2022-11-14 14:00:17.978363071 -0500
@@ -27,6 +27,7 @@
 local PLUGIN
 local -a exts
 for PLUGIN in @PLUGINDIR@/lib*.so; do
+  PLUGIN="$( realpath "$PLUGIN" )"
   case ${PLUGIN##*/} in
     libpdf-poppler.so)
       exts+=( pdf PDF )
