Description: Link libm for ceil
Author: Sebastian Ramacher <sramacher@debian.org>
Last-Update: 2014-11-09
Bug-Debian: https://bugs.debian.org/768674

--- ffmpeg2theora-0.29.0~git+20140316.orig/SConstruct
+++ ffmpeg2theora-0.29.0~git+20140316/SConstruct
@@ -221,6 +221,8 @@ if not env.GetOption('clean'):
       env.Append(LIBS=['m'])
   elif env['static']:
       env.Append(LIBS=['m', 'dl'])
+  else:
+      env.Append(LIBS=['m'])
 
 
 env = conf.Finish()
