--- midori-0.2.9~/katze/katze-throbber.c    2010-10-31 17:20:48.000000000 -0500
+++ midori-0.2.9/katze/katze-throbber.c 2010-12-30 10:35:08.950402793 -0600
@@ -855,6 +855,7 @@
 katze_throbber_expose_event (GtkWidget*      widget,
                              GdkEventExpose* event)
 {
+    gint ax, ay;
     KatzeThrobber* throbber = KATZE_THROBBER (widget);
 
     if (G_UNLIKELY (!throbber->width || !throbber->height))
@@ -867,7 +868,6 @@
     if (!throbber->animated && (throbber->static_pixbuf
         || throbber->static_icon_name || throbber->static_stock_id))
     {
-        gint ax, ay;
         cairo_t* cr;
 
         if (G_UNLIKELY (!throbber->static_pixbuf && throbber->static_icon_name))
