diff -Naur an-0.95.orig/README an-0.95/README
--- an-0.95.orig/README	2001-03-17 08:58:48.000000000 -0500
+++ an-0.95/README	2010-01-26 22:56:44.000000000 -0500
@@ -3,7 +3,7 @@
 
 Generates anagrams for a phrase supplied by the user, the words used in the
 anagram are taken from a specified dictionary which should contain one
-word per line (default:/usr/dict/words).
+word per line (default:/usr/share/dict/words).
 
 Read INSTALL for instructions on installing.
 
diff -Naur an-0.95.orig/an.6 an-0.95/an.6
--- an-0.95.orig/an.6	2000-11-13 19:37:48.000000000 -0500
+++ an-0.95/an.6	2010-01-26 22:56:59.000000000 -0500
@@ -11,7 +11,7 @@
 .SH DESCRIPTION
 .B an
 finds all anagrams which can be made from the letters in \fBPHRASE\fP, 
-using words in the specified dictionary (default=/usr/dict/words).
+using words in the specified dictionary (default=/usr/share/dict/words).
 .SH OPTIONS
 .TP
 .I \-w, --words
