From d529db70aa51a8e4f2b89339d7de9a81122414e2 Mon Sep 17 00:00:00 2001
From: Andriy Grytsenko <andrej@rep.kiev.ua>
Date: Thu, 4 Sep 2014 21:49:08 +0300
Subject: [PATCH 2/2] Remove call to libtoolize from autogen.sh - it never
 used.

---
 autogen.sh | 1 -
 1 file changed, 1 deletion(-)

diff --git a/autogen.sh b/autogen.sh
index a282893..7fd9bd6 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -26,7 +26,6 @@ set -x
 
 ${ACLOCAL:-aclocal$AM_VERSION} ${ACLOCAL_ARG}
 ${AUTOHEADER:-autoheader$AC_VERSION} --force
-AUTOMAKE=$AUTOMAKE libtoolize -c --automake --force
 AUTOMAKE=$AUTOMAKE intltoolize -c --automake --force
 $AUTOMAKE --add-missing --copy --include-deps
 ${AUTOCONF:-autoconf$AC_VERSION}
-- 
2.1.0

