--- ./index_ori.py	2009-11-24 17:06:09.000000000 +0100
+++ ./index.py	2010-02-01 23:33:13.000000000 +0100
@@ -30,7 +30,18 @@
 import sys
 import traceback
 from thread import get_ident
-import sqlite3  as sqlite
+#import sqlite3  as sqlite
+
+#modif mcc
+try:
+    import sqlite3  as sqlite
+except ImportError:
+    try:
+        from pysqlite2 import dbapi2 as sqlite
+    except ImportError:
+        raise ImportError, "no module sqlite3 or pysqlite2"
+
+
 sqlite.enable_shared_cache(True)
 #sqlite.threadsafety = 0
 
