diff -Naur pcmanfm-20101129_0f075cf.orig/src/main-win.c pcmanfm-20101129_0f075cf/src/main-win.c
--- pcmanfm-20101129_0f075cf.orig/src/main-win.c	2010-11-29 15:11:45.000000000 +0100
+++ pcmanfm-20101129_0f075cf/src/main-win.c	2010-12-03 11:05:46.000000000 +0100
@@ -550,19 +550,6 @@
     /* the location bar */
     self->location = fm_path_entry_new();
     g_signal_connect(self->location, "activate", on_location_activate, self);
-    if(geteuid() == 0) /* if we're using root, give the location entry a different color */
-    {
-        GtkStyle* style = gtk_rc_get_style_by_paths(
-                            gtk_settings_get_for_screen(gtk_widget_get_screen(self->location)),
-                            "gtk-tooltip", NULL, G_TYPE_NONE);
-        if(style)
-        {
-            gtk_widget_modify_base(self->location, GTK_STATE_NORMAL, &style->bg[GTK_STATE_NORMAL]);
-            gtk_widget_modify_fg(self->location, GTK_STATE_NORMAL, &style->fg[GTK_STATE_NORMAL]);
-            gtk_entry_set_icon_from_stock(GTK_ENTRY(self->location), GTK_ENTRY_ICON_PRIMARY, GTK_STOCK_DIALOG_WARNING);
-        }
-        gtk_entry_set_icon_tooltip_text(GTK_ENTRY(self->location), GTK_ENTRY_ICON_PRIMARY, _("You are in super user mode"));
-    }
 
     toolitem = gtk_tool_item_new();
     gtk_container_add( GTK_CONTAINER(toolitem), self->location );
