From 942fd81a00e93fa576f05c6f9a9f52864586f0a0 Mon Sep 17 00:00:00 2001
From: Hong Jen Yee (PCMan) <pcman.tw@gmail.com>
Date: Mon, 5 Dec 2011 23:39:24 +0800
Subject: [PATCH] Apply patch #3438582 to fix bug #3325415 - window resize problem.

---
 src/main-win.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/main-win.c b/src/main-win.c
index c5af013..3d80372 100644
--- a/src/main-win.c
+++ b/src/main-win.c
@@ -1326,7 +1326,7 @@ gboolean on_button_press_event(GtkWidget* w, GdkEventButton* evt)
         act = gtk_ui_manager_get_action(win->ui, "/Next2");
         gtk_action_activate(act);
     }
-    return TRUE;
+    return GTK_WIDGET_CLASS(fm_main_win_parent_class)->button_press_event(w, evt);
 }
 
 static void on_reload(GtkAction* act, FmMainWin* win)
-- 
1.7.4.1

