diff -Naur PokerTH-0.9.3-src.orig/src/third_party/qtsingleapplication/qtlocalpeer.cpp PokerTH-0.9.3-src/src/third_party/qtsingleapplication/qtlocalpeer.cpp
--- PokerTH-0.9.3-src.orig/src/third_party/qtsingleapplication/qtlocalpeer.cpp	2012-02-11 17:57:38.000000000 +0100
+++ PokerTH-0.9.3-src/src/third_party/qtsingleapplication/qtlocalpeer.cpp	2012-04-01 08:49:19.961007697 +0200
@@ -57,6 +57,7 @@
 #endif
 #if defined(Q_OS_UNIX)
 #include <time.h>
+#include <unistd.h>
 #endif
 
 namespace QtLP_Private {
