--- milkytracker-0.90.85/src/compression/DecompressorGZIP.cpp	2009-04-17 22:34:16.000000000 +0200
+++ milkytracker-trunk/src/compression/DecompressorGZIP.cpp	2012-02-20 20:24:49.161937553 +0100
@@ -61,7 +61,7 @@
 	int len = 0;
 	pp_uint8 *buf;
 	
-	if ((gz_input_file = (void **)gzopen (fileName.getStrBuffer(), "r")) == NULL)
+	if ((gz_input_file = (gzFile*)gzopen (fileName.getStrBuffer(), "r")) == NULL)
 		return false;
 	
 	if ((buf = new pp_uint8[0x10000]) == NULL)
@@ -71,7 +71,7 @@
 	
     while (true) 
 	{
-        len = gzread (gz_input_file, buf, 0x10000);
+        len = gzread (*gz_input_file, buf, 0x10000);
 		
         if (len < 0) 
 		{
@@ -84,7 +84,7 @@
 		fOut.write(buf, 1, len);
     }
 	
-    if (gzclose (gz_input_file) != Z_OK)
+    if (gzclose (*gz_input_file) != Z_OK)
 	{
 	    delete[] buf;
         return false;
