diff --git a/jollygood/conf/configure.ac b/jollygood/conf/configure.ac
index 8357435..b1d7d15 100644
--- a/jollygood/conf/configure.ac
+++ b/jollygood/conf/configure.ac
@@ -89,9 +89,9 @@ if ! expr x"$host" : 'x.*-mingw*' > /dev/null &&
 	AC_SEARCH_LIBS(pthread_create, [pthread])
 	AC_CHECK_FUNCS(pthread_create sem_init, [],
 		AC_MSG_ERROR([*** pthreads not found!]))
-	AC_CHECK_FUNCS(sem_timedwait sem_timedwait_monotonic sem_clockwait
-		sem_clockwait_np pthread_condattr_setclock pthread_setname_np
-		pthread_set_name_np pthread_getaffinity_np
+	AC_CHECK_FUNCS(sem_timedwait sem_timedwait_monotonic sem_clockwait \
+		sem_clockwait_np pthread_condattr_setclock pthread_setname_np \
+		pthread_set_name_np pthread_getaffinity_np \
 		pthread_setaffinity_np pthread_cond_timedwait_relative_np)
 	AC_CHECK_HEADERS(pthread.h sched.h)
 	AC_CHECK_HEADERS(pthread_np.h, [], [], [
