diff -rupN SDL2_image-2.0.0.orig/IMG_png.c SDL2_image-2.0.0/IMG_png.c
--- SDL2_image-2.0.0.orig/IMG_png.c	2013-08-10 10:38:36.000000000 -0700
+++ SDL2_image-2.0.0/IMG_png.c	2014-01-13 09:33:32.569432897 -0800
@@ -68,8 +68,8 @@
 #include <png.h>
 
 /* Check for the older version of libpng */
-#if (PNG_LIBPNG_VER_MAJOR == 1) 
-#if (PNG_LIBPNG_VER_MINOR < 4)
+#if (PNG_LIBPNG_VER_MAJOR == 1)
+#if (PNG_LIBPNG_VER_MINOR < 5)
 #define LIBPNG_VERSION_12
 typedef png_bytep png_const_bytep;
 #endif
