--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -263,17 +263,6 @@ set(REQUIRED_LIBS ${REQUIRED_LIBS}
   ${QUAZIP_LIBRARY}
 )
 
-#Find Python
-if(PYTHON_USE_PYTHON3)
-    find_package(PythonLibs 3.3 REQUIRED)
-else(PYTHON_USE_PYTHON3)
-    find_package(PythonLibs 2.6 REQUIRED)
-endif(PYTHON_USE_PYTHON3)
-include_directories(${PYTHON_INCLUDE_DIR})
-set(REQUIRED_LIBS ${REQUIRED_LIBS}
-    ${PYTHON_LIBRARIES}
-)
-
 #Find PythonQt
 find_package(PythonQt REQUIRED)
 include_directories(${PYTHONQT_INCLUDE_DIR})
@@ -293,6 +282,17 @@ else(PYTHONQT_QTALL_FOUND)
     message(STATUS "Using PythonQt_QtBindings (old) for Qt bindings.")
 endif(PYTHONQT_QTALL_FOUND)
 
+#Find Python
+if(PYTHON_USE_PYTHON3)
+    find_package(PythonLibs 3.3 REQUIRED)
+else(PYTHON_USE_PYTHON3)
+    find_package(PythonLibs 2.6 REQUIRED)
+endif(PYTHON_USE_PYTHON3)
+include_directories(${PYTHON_INCLUDE_DIR})
+set(REQUIRED_LIBS ${REQUIRED_LIBS}
+    ${PYTHON_LIBRARIES}
+)
+
 #Platform specific libs
 if(WIN32)
     set(REQUIRED_LIBS ${REQUIRED_LIBS}
