diff -Naur vinterm-0.5.0/Makefile vinterm-0.5.0.patched/Makefile
--- vinterm-0.5.0/Makefile	2013-08-19 12:34:06.000000000 -0400
+++ vinterm-0.5.0.patched/Makefile	2017-03-04 03:15:33.786462685 -0500
@@ -86,14 +86,22 @@
 	@echo compiling terminfo data
 	@tic -s data/vinterm.info
 	@echo installing icons
-	@cp -f icon/icon_16.png  /usr/share/icons/hicolor/16x16/apps/vinterm.png
-	@cp -f icon/icon_22.png  /usr/share/icons/hicolor/22x22/apps/vinterm.png
-	@cp -f icon/icon_32.png  /usr/share/icons/hicolor/32x32/apps/vinterm.png
-	@cp -f icon/icon_48.png  /usr/share/icons/hicolor/48x48/apps/vinterm.png
-	@cp -f icon/icon_64.png  /usr/share/icons/hicolor/64x64/apps/vinterm.png
-	@cp -f icon/icon_128.png /usr/share/icons/hicolor/128x128/apps/vinterm.png
-	@cp -f icon/icon.svg     /usr/share/icons/hicolor/scalable/apps/vinterm.svg
-	@cp -f data/vinterm.desktop /usr/share/applications
+	@mkdir -p ${DESTDIR}/usr/share/icons/hicolor/16x16/apps
+	@mkdir -p ${DESTDIR}/usr/share/icons/hicolor/22x22/apps
+	@mkdir -p ${DESTDIR}/usr/share/icons/hicolor/32x32/apps
+	@mkdir -p ${DESTDIR}/usr/share/icons/hicolor/48x48/apps
+	@mkdir -p ${DESTDIR}/usr/share/icons/hicolor/64x64/apps
+	@mkdir -p ${DESTDIR}/usr/share/icons/hicolor/128x128/apps
+	@mkdir -p ${DESTDIR}/usr/share/icons/hicolor/scalable/apps
+	@mkdir -p ${DESTDIR}/usr/share/applications
+	@cp -f icon/icon_16.png  ${DESTDIR}/usr/share/icons/hicolor/16x16/apps/vinterm.png
+	@cp -f icon/icon_22.png  ${DESTDIR}/usr/share/icons/hicolor/22x22/apps/vinterm.png
+	@cp -f icon/icon_32.png  ${DESTDIR}/usr/share/icons/hicolor/32x32/apps/vinterm.png
+	@cp -f icon/icon_48.png  ${DESTDIR}/usr/share/icons/hicolor/48x48/apps/vinterm.png
+	@cp -f icon/icon_64.png  ${DESTDIR}/usr/share/icons/hicolor/64x64/apps/vinterm.png
+	@cp -f icon/icon_128.png ${DESTDIR}/usr/share/icons/hicolor/128x128/apps/vinterm.png
+	@cp -f icon/icon.svg     ${DESTDIR}/usr/share/icons/hicolor/scalable/apps/vinterm.svg
+	@cp -f data/vinterm.desktop ${DESTDIR}/usr/share/applications
 
 # TODO - check data
 uninstall:
