--- a/src/util.py.orig	2016-07-21 16:16:45.785925215 -0600
+++ b/src/util.py	2016-07-21 17:33:12.716614138 -0600
@@ -727,7 +727,7 @@
     return network == orig_network
 
 
-def get_iptables_version(exe="/sbin/iptables"):
+def get_iptables_version(exe="/usr/sbin/iptables"):
     '''Return iptables version'''
     (rc, out) = cmd([exe, '-V'])
     if rc != 0:
@@ -737,7 +737,7 @@
 
 
 # must be root, so don't report coverage in unit tests
-def get_netfilter_capabilities(exe="/sbin/iptables", do_checks=True):
+def get_netfilter_capabilities(exe="/usr/sbin/iptables", do_checks=True):
     '''Return capabilities set for netfilter to support new features. Callers
        must be root.'''
     def test_cap(exe, chain, rule):
