--- ufw-0.33.orig/conf/ufw.conf	2012-08-18 05:12:49.000000000 +0800
+++ ufw-0.33/conf/ufw.conf	2013-07-16 04:18:44.365425697 +0800
@@ -3,7 +3,7 @@
 
 # Set to yes to start on boot. If setting this remotely, be sure to add a rule
 # to allow your remote connection before starting ufw. Eg: 'ufw allow 22/tcp'
-ENABLED=no
+ENABLED=yes
 
 # Please use the 'ufw' command to set the loglevel. Eg: 'ufw logging medium'.
 # See 'man ufw' for details.
