--- orpheus-1.6/configure.old	2006-11-25 16:56:53.000000000 +0100
+++ orpheus-1.6/configure	2006-11-25 16:57:30.000000000 +0100
@@ -4219,7 +4219,7 @@
 fi
 echo "$as_me:$LINENO: result: $ac_cv_lib_ghttp_ghttp_request_new" >&5
 echo "${ECHO_T}$ac_cv_lib_ghttp_ghttp_request_new" >&6
-if test $ac_cv_lib_ghttp_ghttp_request_new = yes; then
+if test $ac_cv_lib_ghttp_ghttp_request_new = nolibghttp; then
   cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBGHTTP 1
 _ACEOF
