Author: Philipp Benner <mail@philipp-benner.de>
Description: Avoid dependency from csh

--- wise-2.4.1.orig/src/makefile
+++ wise-2.4.1/src/makefile
@@ -109,7 +109,7 @@
 bin : 
 	mkdir bin
 	cp models/pswdb models/psw models/genewisedb models/estwisedb models/estwise models/genewise models/dba models/dnal models/promoterwise network/scanwise_server models/scanwise ./bin
-	csh welcome.csh
+	./welcome.csh
 
 libs : 
 	(cd base   ; $(MAKE) CC="$(CC)" CFLAGS="$(CFLAGS)" libwisebase.a )
--- wise-2.4.1.orig/src/welcome.csh
+++ wise-2.4.1/src/welcome.csh
@@ -1,4 +1,4 @@
-#!/bin/tcsh -f
+#!/bin/sh
 
 echo ""
 echo "Welcome to Wise2.4"
