Description: Fix FTBFS with ld --as-needed.
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/wise/+bug/832935

--- wise-2.4.1.orig/src/models/makefile
+++ wise-2.4.1/src/models/makefile
@@ -215,7 +215,7 @@
 	$(CC)  $(CFLAGS) $(INCFLAGS)  -o scanwisep_wiseserver.o -DSCAN_WISESERVER -I../network -I../socket -I../external/mott scanwisep.c 
 
 scanwise : scanwisep_wiseserver.o seqaligndisplay.o proteinsw.o sw_wrap.o abc.o pba.o hsp2aln_sw.o
-	$(CC) -o scanwise scanwisep_wiseserver.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o hsp2aln_sw.o   ../network/net_hspscan.o ../network/client_multihspscan.o  $(LDFLAGS) -L../external/mott -L../socket -lmott -ldyna_glib -ldyna -lwisesocket -lwisebase  $(EXTRALIBS) -lpthread 
+	$(CC) -o scanwise scanwisep_wiseserver.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o hsp2aln_sw.o   ../network/net_hspscan.o ../network/client_multihspscan.o  $(LDFLAGS) -L../external/mott -L../socket -lmott -ldyna_glib -ldyna -lwisesocket -lwisebase  $(EXTRALIBS) -lpthread $(LDFLAGS)
 
 scanwisep_compress.o : scanwisep.c
 	$(CC)  $(CFLAGS) -DSCAN_COMPRESS -I../dnaindex -I../network -I../socket -I../external/mott scanwisep.c 
@@ -275,7 +275,7 @@
 	$(CC) -o makerandomdb makerandomdb.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 genewise : genewise.o $(NEWG) 
-	$(CC) -o genewise genewise.o $(NEWG) $(LDFLAGS) -lhmmer -ldyna_glib -ldyna_glib -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -o genewise genewise.o $(NEWG) $(LDFLAGS) -lhmmer -ldyna_glib -ldyna_glib -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) $(LDFLAGS)
 
 cdnawise : cdnawise.o $(NEWG) cdnawise10.o 
 	$(CC) -g -o cdnawise cdnawise.o cdnawise10.o $(NEWG) $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
@@ -300,16 +300,16 @@
 
 
 genewisedb : genewisedb.o $(NEWG) 
-	$(CC) -g -o genewisedb genewisedb.o  $(NEWG)  $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) 
+	$(CC) -g -o genewisedb genewisedb.o  $(NEWG)  $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)  $(LDFLAGS)
 
 estwisedb : estwisedb.o $(NEWG) 
-	$(CC) -g -o estwisedb estwisedb.o $(NEWG)  $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o estwisedb estwisedb.o $(NEWG)  $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) $(LDFLAGS)
 
 edb : estwisedb.o $(NEWG) 
 	$(CC) -g -o estwisedb estwisedb.o $(NEWG)  $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) -lefence
 
 estwise : estwise.o $(NEWG) 
-	$(CC) -g -o estwise estwise.o  $(NEWG) $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o estwise estwise.o  $(NEWG) $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) $(LDFLAGS)
 
 
 amplimer_resolver : amplimer_resolver.o dnaalign.o dnamatcher.o 
@@ -331,7 +331,7 @@
 	$(CC) -g -o statwise statwise.o statwise10.o syexonmodel.o genestats.o pwmdna.o geneutil.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 pseudowise : pseudowise.o pseudowise7.o $(NEWG) 
-	$(CC) -g -o pseudowise pseudowise.o pseudowise7.o $(NEWG) $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
+	$(CC) -g -o pseudowise pseudowise.o pseudowise7.o $(NEWG) $(LDFLAGS) -lhmmer -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) $(LDFLAGS)
 
 clonewise : clonewise.o clonewisedp.o localclonewisedp.o mapstruct.o
 	$(CC) -g -o clonewise clonewise.o clonewisedp.o localclonewisedp.o mapstruct.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
