diff --git a/m4/ax_blas.m4 b/m4/ax_blas.m4
--- a/m4/ax_blas.m4
+++ b/m4/ax_blas.m4
@@ -116,13 +116,10 @@
 
 # BLAS in ATLAS library? (http://math-atlas.sourceforge.net/)
 if test $ax_blas_ok = no; then
-	AC_CHECK_LIB(atlas, ATL_xerbla,
-		[AC_CHECK_LIB(f77blas, $sgemm,
-		[AC_CHECK_LIB(cblas, cblas_dgemm,
-			[ax_blas_ok=yes
-			 BLAS_LIBS="-lcblas -lf77blas -latlas"],
-			[], [-lf77blas -latlas])],
-			[], [-latlas])])
+	AC_CHECK_LIB(tatlas, $sgemm,
+		[ax_blas_ok=yes; BLAS_LIBS="-ltatlas"],
+		[AC_CHECK_LIB(satlas, $sgemm,
+			[ax_blas_ok=yes; BLAS_LIBS="-lsatlas"])])
 fi
 
 # BLAS in PhiPACK libraries? (requires generic BLAS lib, too)
