--- qcad-2.0.5.0-1-community.src/dxflib/src/dl_writer_ascii.cpp	2005-11-22 09:46:57.000000000 -0200
+++ qcad-2.0.5.0-1-community.src_patched/dxflib/src/dl_writer_ascii.cpp	2010-05-04 22:54:14.000000000 -0300
@@ -29,6 +29,7 @@
 #pragma once
 #endif // _MSC_VER > 1000
 
+#include <string.h>
 #include <stdio.h>
 
 #include "dl_writer_ascii.h"
--- qcad-2.0.5.0-1-community.src/qcadlib/src/information/rs_information.cpp	2005-11-22 09:52:39.000000000 -0200
+++ qcad-2.0.5.0-1-community.src_patched/qcadlib/src/information/rs_information.cpp	2010-05-04 23:31:33.000000000 -0300
@@ -24,6 +24,7 @@
 **
 **********************************************************************/
 
+#include <algorithm>
 #include "rs_information.h"
 
 #include "rs_constructionline.h"
--- qcad-2.0.5.0-1-community.src/qcadactions/src/rs_actionzoompan.cpp	2005-11-22 09:51:46.000000000 -0200
+++ qcad-2.0.5.0-1-community.src_patched/qcadactions/src/rs_actionzoompan.cpp	2010-05-04 23:34:19.000000000 -0300
@@ -24,6 +24,7 @@
 **
 **********************************************************************/
 
+#include <algorithm>
 #include "rs_actionzoompan.h"
 #include "rs_snapper.h"
 #include "rs_point.h"
