--- qcad-2.0.5.0-1-community.src/qcad/src/qc_applicationwindow.cpp	2005-11-22 09:49:33.000000000 -0200
+++ qcad-2.0.5.0-1-community.src_patched/qcad/src/qc_applicationwindow.cpp	2010-05-09 13:48:20.000000000 -0300
@@ -2943,7 +2943,7 @@
                         RS_SYSTEM->getAppDir().latin1());
         RS_DEBUG->print("QC_ApplicationWindow::slotHelpManual(): appdir: %s",
                         RS_SYSTEM->getAppDir().latin1());
-        assistant = new QAssistantClient(RS_SYSTEM->getAppDir()+"/bin", this);
+        assistant = new QAssistantClient("@QTDIR@/bin", this);
 		connect(assistant, SIGNAL(error(const QString&)), 
 			this, SLOT(slotError(const QString&)));
         QStringList args;
