From 214e27993fbbadad4241b6bf4e1c5f23eea7a70d Mon Sep 17 00:00:00 2001
From: Andrew Mustun <andrew@qcad.org>
Date: Mon, 29 May 2017 11:31:50 +0200
Subject: [PATCH] back port to qt 4

---
 src/core/RBlockReferenceEntity.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/RBlockReferenceEntity.cpp b/src/core/RBlockReferenceEntity.cpp
index 3141aa649..9032faeb3 100644
--- a/src/core/RBlockReferenceEntity.cpp
+++ b/src/core/RBlockReferenceEntity.cpp
@@ -417,7 +417,7 @@ void RBlockReferenceEntity::exportEntity(RExporter& e, bool preview, bool forceS
                 if (entity->getLayerId()==layer0Id) {
                     QStack<REntity*> blockRefStack = e.getBlockRefViewportStack();
                     bool skip = false;
-                    for (int i=blockRefStack.length()-1; i>=0; i--) {
+                    for (int i=blockRefStack.size()-1; i>=0; i--) {
                         REntity* ent = blockRefStack.at(i);
                         RBlockReferenceEntity* blockRef = dynamic_cast<RBlockReferenceEntity*>(ent);
                         if (blockRef==NULL) {
