--- qcad-2.0.5.0-1-community.src/qcadlib/src/engine/rs_entity.cpp	2005-11-22 09:52:32.000000000 -0200
+++ qcad-2.0.5.0-1-community.src_patched/qcadlib/src/engine/rs_entity.cpp	2010-05-04 23:23:53.000000000 -0300
@@ -861,7 +861,7 @@
         os << " layer: NULL ";
     } else {
         os << " layer: " << e.layer->getName().latin1() << " ";
-        os << " layer address: " << (int)(e.layer) << " ";
+        os << " layer address: " << (e.layer) << " ";
     }
 
     os << e.pen << "\n";
--- qcad-2.0.5.0-1-community.src/qcadlib/src/engine/rs_layer.cpp	2005-11-22 09:52:38.000000000 -0200
+++ qcad-2.0.5.0-1-community.src_patched/qcadlib/src/engine/rs_layer.cpp	2010-05-04 23:27:39.000000000 -0300
@@ -55,7 +55,7 @@
     os << " name: " << l.getName().latin1()
     << " pen: " << l.getPen()
 	<< " frozen: " << (int)l.isFrozen()
-	<< " address: " << (int)(&l)
+	<< " address: " << (&l)
     << std::endl;
     return os;
 }
