diff -Nur pdksh-5.2.14.orig/Makefile.in pdksh-5.2.14/Makefile.in
--- pdksh-5.2.14.orig/Makefile.in	1999-07-13 12:06:53.000000000 -0500
+++ pdksh-5.2.14/Makefile.in	2009-09-26 19:09:46.183509569 -0500
@@ -80,8 +80,8 @@
 	$(CC) -c $(CPPFLAGS) $(DEFS) -I. -I$(srcdir) $(CFLAGS) $<
 
 install: installdirs all
-	$(INSTALL_PROGRAM) $(SHELL_PROG)$(exe_suffix) $(bindir)/`echo $(SHELL_PROG)|sed '$(transform)'`$(exe_suffix)
-	-$(INSTALL_DATA) $(SHELL_PROG).1 $(mandir)/`echo $(SHELL_PROG)|sed '$(transform)'`.$(manext)
+	$(INSTALL_PROGRAM) $(SHELL_PROG)$(exe_suffix) $(DESTDIR)/$(bindir)/`echo $(SHELL_PROG)|sed '$(transform)'`$(exe_suffix)
+	-$(INSTALL_DATA) $(SHELL_PROG).1 $(DESTDIR)/$(mandir)/`echo $(SHELL_PROG)|sed '$(transform)'`.$(manext)
 	-@prog=$(bindir)/`echo $(SHELL_PROG)|sed '$(transform)'`$(exe_suffix);\
 	    test -f /etc/shells \
 	    && (grep "^$$prog\$$" /etc/shells > /dev/null \
@@ -90,7 +90,7 @@
       you should add it if you want to set your shell to $(SHELL_PROG)")
 
 installdirs:
-	$(srcdir)/mkinstalldirs $(bindir) $(mandir)
+	$(srcdir)/mkinstalldirs $(DESTDIR)/$(bindir) $(DESTDIR)/$(mandir)
 
 uninstall:
 	rm -f $(bindir)/`echo $(SHELL_PROG)|sed '$(transform)'`$(exe_suffix)
