support for `ulimit -v'
(from PLD)

Index: pdksh-5.2.14/c_ulimit.c
===================================================================
--- pdksh-5.2.14.orig/c_ulimit.c	2008-04-15 20:50:48.000000000 +0200
+++ pdksh-5.2.14/c_ulimit.c	2008-04-15 20:55:52.000000000 +0200
@@ -38,6 +38,10 @@
 # define KSH_RLIM_INFINITY ((rlim_t) 1 << (sizeof(rlim_t) * 8 - 1) - 1)
 #endif /* RLIM_INFINITY */
 
+#if !defined(RLIMIT_VMEM) && defined(RLIMIT_AS)
+#	define RLIMIT_VMEM RLIMIT_AS
+#endif
+
 int
 c_ulimit(wp)
 	char **wp;
