Quote '-' chars in man page to make lintian happy.
Index: pdksh-5.2.14/ksh.Man
===================================================================
--- pdksh-5.2.14.orig/ksh.Man	2009-07-18 15:18:29.000000000 +0200
+++ pdksh-5.2.14/ksh.Man	2009-07-18 15:18:30.000000000 +0200
@@ -445,7 +445,7 @@
 the \fB&&\fP and \fB||\fP operators.
 This means that in the statement
 .ce
-\fB[[ -r foo && $(< foo) = b*r ]]\fP
+\fB[[ \-r foo && $(< foo) = b*r ]]\fP
 the \fB$(< foo)\fP is evaluated if and only if the file \fBfoo\fP exists
 and is readable.
 .nr PD \n(P2
@@ -2092,7 +2092,7 @@
 See Shell Startup above for a description of what this
 means.
 T}
--r	restricted	T{
+\-r	restricted	T{
 Enable restricted mode \(em this option can only be used when the shell is
 invoked.  See Shell Startup above for a description of what this
 means.
@@ -2198,7 +2198,7 @@
 These options can also be used upon invocation of the shell.  The current
 set of options (with single letter names) can be found in the
 parameter \fB\-\fP.
-\fBset -o\fP with no option name will list all the options and whether each
+\fBset \-o\fP with no option name will list all the options and whether each
 is on or off; \fBset +o\fP will print the long names of all options that
 are currently on.
 .sp
@@ -2303,7 +2303,7 @@
 As a non-standard extension, if the option starts with a \fB!\fP, the test
 is negated; the test always fails if option doesn't exist (thus
 .RS
-\fB[ -o \fP\fIfoo\fP \fB-o -o !\fP\fIfoo\fP \fB]\fP
+\fB[ \-o \fP\fIfoo\fP \fB\-o \-o !\fP\fIfoo\fP \fB]\fP
 .RE
 returns true if and only if option \fIfoo\fP exists).
 T}
@@ -2380,7 +2380,7 @@
 (where \fIn\fP is a file descriptor number),
 the \fBtest\fP command will attempt to fake it for all tests that
 operate on files (except the \fB-e\fP test).
-I.e., \fB[ -w /dev/fd/2 ]\fP tests if file descriptor 2 is writable.
+I.e., \fB[ \-w /dev/fd/2 ]\fP tests if file descriptor 2 is writable.
 .sp
 Note that some special rules are applied (courtesy of POSIX) if the
 number of arguments to \fBtest\fP or \fB[\fP \&... \fB]\fP is less than
@@ -2410,7 +2410,7 @@
 .nf
     0.00s real     0.00s user     0.00s system
 .fi
-unless the -p option is given (only possible if \fIpipeline\fP is a simple
+unless the \-p option is given (only possible if \fIpipeline\fP is a simple
 command), in which case the output is slightly longer:
 .nf
     real   0.00
