Fix gcc warnings.
Index: pdksh-5.2.14/eval.c
===================================================================
--- pdksh-5.2.14.orig/eval.c	2008-04-15 21:11:04.000000000 +0200
+++ pdksh-5.2.14/eval.c	2008-04-15 21:15:24.000000000 +0200
@@ -201,6 +201,8 @@
 	word = (f&DOBLANK) ? IFS_WS : IFS_WORD;
 	st_head.next = (SubType *) 0;
 	st = &st_head;
+	x.split = 0;
+	x.str   = NULL;
 
 	while (1) {
 		Xcheck(ds, dp);
@@ -280,7 +282,7 @@
 			  {
 				char *varname = ++sp; /* skip the { or x (}) */
 				int stype;
-				int slen;
+				int slen = 0;
 
 				sp = strchr(sp, '\0') + 1; /* skip variable */
 				type = varsub(&x, varname, sp, &stype, &slen);
