diff -Naur minidjvu-0.8.svn.2010.05.06.orig/include/minidjvu/alg/smooth.h minidjvu-0.8.svn.2010.05.06/include/minidjvu/alg/smooth.h
--- minidjvu-0.8.svn.2010.05.06.orig/include/minidjvu/alg/smooth.h	2010-08-02 17:15:58.000000000 +0400
+++ minidjvu-0.8.svn.2010.05.06/include/minidjvu/alg/smooth.h	2015-12-02 03:12:43.000000000 +0300
@@ -4,7 +4,7 @@
 
 
 /*
- * `smooth' is applied to a bitmap even before it's splitted.
+ * `smooth' is applied to a bitmap even before it's split.
  * 
  * Right now, the algorithm flips pixels which are surrounded
  * by at least 3 of 4 neighboring pixels of another color.
diff -Naur minidjvu-0.8.svn.2010.05.06.orig/Makefile.in minidjvu-0.8.svn.2010.05.06/Makefile.in
--- minidjvu-0.8.svn.2010.05.06.orig/Makefile.in	2010-08-02 17:15:58.000000000 +0400
+++ minidjvu-0.8.svn.2010.05.06/Makefile.in	2015-12-02 02:18:36.000000000 +0300
@@ -8,7 +8,7 @@
 manpage:=@PACKAGE_NAME@
 
 DOCDIR:=doc
-MANDIR:=$(prefix)/share/man/man1
+MANDIR:=$(prefix)/man/man1
 INCLUDEDIR:=$(prefix)/include
 SUBDIRS:=src \
          po \
@@ -71,7 +71,7 @@
 	${GZIP} -nf -9 ${DESTDIR}$(MANDIR)/$(manpage).1
 	for dir in $(DOCDIR)/*/; do \
 	  if [ -f $${dir}$(manpage).1 ]; then \
-	    LOCMANDIR=${DESTDIR}$(prefix)/share/man$${dir#$(DOCDIR)}/man1 &&\
+	    LOCMANDIR=${DESTDIR}$(prefix)/man$${dir#$(DOCDIR)}/man1 &&\
 	    ${INSTALL} -d $${LOCMANDIR} &&\
 	    ${INSTALL_DATA} $${dir}$(manpage).1 $${LOCMANDIR} && \
 	    ${GZIP} -nf -9 $${LOCMANDIR}/$(manpage).1; \
@@ -91,7 +91,7 @@
 	${RM} ${DESTDIR}$(MANDIR)/$(manpage).1.gz
 	for dir in $(DOCDIR)/*/; do \
 	  if [ -f $${dir}$(manpage).1 ]; then \
-	    LOCMANDIR=${DESTDIR}$(prefix)/share/man$${dir#$(DOCDIR)}/man1 &&\
+	    LOCMANDIR=${DESTDIR}$(prefix)/man$${dir#$(DOCDIR)}/man1 &&\
 	    ${RM} $${LOCMANDIR}/$(manpage).1.gz; \
 	  fi; \
 	done
diff -Naur minidjvu-0.8.svn.2010.05.06.orig/po/ru.po minidjvu-0.8.svn.2010.05.06/po/ru.po
--- minidjvu-0.8.svn.2010.05.06.orig/po/ru.po	2010-08-02 17:15:58.000000000 +0400
+++ minidjvu-0.8.svn.2010.05.06/po/ru.po	2015-12-02 03:12:43.000000000 +0300
@@ -239,7 +239,7 @@
 
 #: minidjvu.c:361
 #, c-format
-msgid "the splitted image has %d pieces\n"
+msgid "the split image has %d pieces\n"
 msgstr "после разбиения картинка состоит из %d участков\n"
 
 #: minidjvu.c:366
diff -Naur minidjvu-0.8.svn.2010.05.06.orig/tools/minidjvu.c minidjvu-0.8.svn.2010.05.06/tools/minidjvu.c
--- minidjvu-0.8.svn.2010.05.06.orig/tools/minidjvu.c	2010-08-02 17:15:58.000000000 +0400
+++ minidjvu-0.8.svn.2010.05.06/tools/minidjvu.c	2015-12-02 03:12:43.000000000 +0300
@@ -360,7 +360,7 @@
     mdjvu_bitmap_destroy(bitmap);
     if (verbose)
     {
-        printf(_("the splitted image has %d pieces\n"),
+        printf(_("the split image has %d pieces\n"),
                 mdjvu_image_get_blit_count(image));
     }
     if (clean)
