--- nSnake-2.0.0/Makefile.original	2014-04-20 10:32:07.056427339 +0300
+++ nSnake-2.0.0/Makefile	2014-04-20 17:48:36.912009088 +0300
@@ -48,15 +48,15 @@
 PREFIX  = $(DESTDIR)/usr
 
 EXEC_PREFIX = $(PREFIX)
-DATAROOTDIR = $(PREFIX)/share
-MANROOT     = $(DATAROOTDIR)/man
+DATAROOTDIR = $(PREFIX)/share/games
+MANROOT     = $(PREFIX)/man
 
 MANNUMBER   = 6
 
-BINDIR      = $(EXEC_PREFIX)/games
+BINDIR      = $(EXEC_PREFIX)/bin
 MANDIR      = $(MANROOT)/man$(MANNUMBER)
 
-MANFILE     = $(PACKAGE).$(MANNUMBER)
+# MANFILE     = $(PACKAGE).$(MANNUMBER)
 MANPAGE     = doc/man/$(MANFILE)
 
 # Build info
@@ -121,10 +121,10 @@
 	# Installing...
 	$(MUTE)install -d -m 755 $(BINDIR)
 	$(MUTE)install -m 755 bin/$(EXE) $(BINDIR)
-	-$(MUTE)cat $(MANPAGE) | sed -e "s|DATE|$(DATE)|g" -e "s|VERSION|$(VERSION)|g" >$(MANFILE)
-	$(MUTE)install -d $(MANDIR)
-	$(MUTE)install $(MANFILE) $(MANDIR)
-	$(MUTE)rm -f $(MANFILE)
+	# -$(MUTE)cat $(MANPAGE) | sed -e "s|DATE|$(DATE)|g" -e "s|VERSION|$(VERSION)|g" > $(MANFILE) 
+	# $(MUTE)install -d $(MANDIR)
+	# $(MUTE)install $(MANFILE) $(MANDIR)
+	# $(MUTE)rm -f $(MANFILE)
 	# $(PACKAGE) successfuly installed!
 
 uninstall:
