--- wmget/dockapp/da_run.c.orig	2016-02-14 00:57:46.773867780 +0700
+++ wmget/dockapp/da_run.c	2016-02-14 01:18:46.455679579 +0700
@@ -157,10 +157,8 @@
 static long da_timer_msec_remaining (void)
 {
     struct timeval right_now;
-    int rv;
+    gettimeofday(&right_now, NULL);
     
-    rv = gettimeofday (&right_now, 0);
-
     return
         (da_timer_next_timeout.tv_sec - right_now.tv_sec) * 1000L
         + (da_timer_next_timeout.tv_usec - right_now.tv_usec) / 1000L;
