--- dockapp/da_run.c	2003-02-09 01:09:26.000000000 -0200
+++ dockapp/da_run.c.new	2014-06-24 01:29:29.000000000 -0300
@@ -157,9 +157,9 @@
 static long da_timer_msec_remaining (void)
 {
     struct timeval right_now;
-    int rv;
+/*    int rv;
     
-    rv = gettimeofday (&right_now, 0);
+    rv = gettimeofday (&right_now, 0); */
 
     return
         (da_timer_next_timeout.tv_sec - right_now.tv_sec) * 1000L
