diff --git a/src/lib/libumem/malloc.c b/src/lib/libumem/malloc.c
index 7eec207..5a4d763 100644
--- a/src/lib/libumem/malloc.c
+++ b/src/lib/libumem/malloc.c
@@ -453,7 +453,7 @@ static void __attribute__((constructor)) umem_malloc_init_hook(void)
        }
 }
 
-void (*__malloc_initialize_hook)(void) = umem_malloc_init_hook;
+void (* __volatile __malloc_initialize_hook)(void) = umem_malloc_init_hook;
 
 #else
 void __attribute__((constructor))
