--- a/third-party/proxygen/bundled_proxygen-prefix/src/bundled_proxygen/proxygen/lib/services/WorkerThread.cpp	2022-06-16 13:33:00.766000000 +0200
+++ b/third-party/proxygen/bundled_proxygen-prefix/src/bundled_proxygen/proxygen/lib/services/WorkerThread.cpp	2022-06-16 13:37:39.111000000 +0200
@@ -17,7 +17,7 @@
 #include <glog/logging.h>
 #include <signal.h>
 
-#if !FOLLY_MOBILE && __has_include(<liburing.h>)
+#if !FOLLY_MOBILE && false
 
 DEFINE_int32(pwt_io_uring_capacity, -1, "io_uring backend capacity");
 DEFINE_int32(pwt_io_uring_max_submit, 128, "io_uring backend max submit");
--- a/third-party/folly/bundled_folly-prefix/src/bundled_folly/folly/experimental/io/IoUringBackend.h	2022-06-16 13:38:45.523000000 +0200
+++ b/third-party/folly/bundled_folly-prefix/src/bundled_folly/folly/experimental/io/IoUringBackend.h	2022-06-16 13:43:08.748000000 +0200
@@ -40,11 +40,7 @@
 #include <poll.h>
 #endif
 
-#if __has_include(<liburing.h>)
-#include <liburing.h>
-#endif
-
-#if __has_include(<liburing.h>)
+#if false
 
 namespace folly {
 
--- a/third-party/folly/bundled_folly-prefix/src/bundled_folly/folly/experimental/io/IoUringBackend.cpp	2022-06-16 14:12:06.191000000 +0200
+++ b/third-party/folly/bundled_folly-prefix/src/bundled_folly/folly/experimental/io/IoUringBackend.cpp	2022-06-16 14:09:28.887000000 +0200
@@ -31,7 +31,7 @@
 #include <sys/timerfd.h>
 #endif
 
-#if __has_include(<liburing.h>)
+#if false
 
 extern "C" FOLLY_ATTR_WEAK void eb_poll_loop_pre_hook(uint64_t* call_time);
 extern "C" FOLLY_ATTR_WEAK void eb_poll_loop_post_hook(
