--- bombono-dvd-1.2.2/libs/boost-logging/boost/logging/detail/tss/tss_impl_pthread.hpp  2013-03-08 00:50:45.000000000 +0330
+++ new/libs/boost-logging/boost/logging/detail/tss/tss_impl_pthread.hpp        2015-11-26 20:06:48.693423531 +0330
@@ -39,7 +39,7 @@
     slots = static_cast<tss_slots*>( pthread_getspecific(tss_data_native_key()));
     if (slots == 0)
     {
-        std::auto_ptr<tss_slots> temp( new_object_ensure_delete<tss_slots>() );
+        std::unique_ptr<tss_slots> temp( new_object_ensure_delete<tss_slots>() );
         // pre-allocate a few elems, so that we'll be fast
         temp->resize(BOOST_LOG_TSS_SLOTS_SIZE);
 

