--- bombono-dvd-1.2.2/src/mgui/project/menu-actions.cpp 2013-03-08 00:50:45.000000000 +0330
+++ new/src/mgui/project/menu-actions.cpp       2015-11-20 19:17:23.516912549 +0330
@@ -187,7 +187,7 @@
         txt.MediaItem().ClearLink();
 }
 
-void UpdateMenuRegionObject(Comp::Object* obj, const Point& menu_sz, RectListRgn& lst)
+void UpdateMenuRegionObject(Comp::Object* obj, Point menu_sz, RectListRgn& lst)
 {
     MenuRegionEraserVis vis(obj, menu_sz, lst);
     //PrimaryRegionEraserVis vis(obj, false);
@@ -198,7 +198,7 @@
 {
     CanvasBuf& cb = mp.thRgn.GetCanvasBuf();
     ForeachLinked(mp.Owner(), bb::bind(&UpdateMenuRegionObject, 
-                                       _1, boost::cref(cb.Size()), 
+                                       _1, cb.Size(), 
                                        boost::ref(cb.RenderList())));
 }
 

