--- bombono-dvd-1.2.2/src/mgui/stream_utils.h   2013-03-08 00:50:45.000000000 +0330
+++ new/src/mgui/stream_utils.h 2015-11-26 20:37:42.780113771 +0330
@@ -42,8 +42,8 @@
                 stream(std::iostream& strm): io_strm(strm) {}
 
  std::iostream& strm() { return io_strm; }
-                operator void*() const { return (void*)io_strm; }
-          bool  operator!() const { return !operator void*(); }
+                operator bool() const { return static_cast<bool> (io_strm); }
+          bool  operator!() const { return !operator bool(); }
         stream& seekg(streamoff off, ios_base::seekdir dir)
                 { io_strm.seekg(off, dir); return *this; }
         stream& seekp(streamoff off, ios_base::seekdir dir)
