diff -Naur -x '.hg*' a/irchelper/irchelper.c b/irchelper/irchelper.c
--- a/irchelper/irchelper.c	2018-07-28 12:09:23.714988017 +0100
+++ b/irchelper/irchelper.c	2018-08-07 19:29:44.194109009 +0100
@@ -719,7 +719,7 @@
 				nickserv_do_identify(authentication, connection->proto_data, connection, nickpassword);
 			}
 			else
-				nickserv_identify(connection->proto_data, connection, nickpassword);
+				nickserv_msg_identify("identify", connection->proto_data, connection, nickpassword);
 
 			g_strfreev(userparts);
 		}
