Index: src/plugins/clientinfo/clientinfo.cpp
===================================================================
--- src/plugins/clientinfo/clientinfo.cpp	(revision 1887)
+++ src/plugins/clientinfo/clientinfo.cpp	(working copy)
@@ -2,6 +2,8 @@
 
 #include <QDir>
 #include <QProcess>
+#include <QFile>
+#include <QIODevice>
 #include <QFileInfo>
 #include <QTextStream>
 
@@ -434,8 +436,23 @@
 			}
 		}
 
+// Slackware detect
 		if (osver.isEmpty())
 		{
+			QFile slackwarever("/etc/slackware-version");
+			if (!slackwarever.open(QIODevice::ReadOnly | QIODevice::Text));
+			QTextStream slackin(&slackwarever);
+			while (!slackin.atEnd())
+			{
+				QString slackline = slackin.readLine();
+				QString slackprocess_line(slackline);
+				osver.append(slackprocess_line);
+			}
+			slackwarever.close();
+		}
+
+		if (osver.isEmpty())
+		{
 			utsname buf;
 			if (uname(&buf) != -1)
 			{
