diff --git a/src/gui/piano_roll.cpp b/src/gui/piano_roll.cpp
index e94c022..eec83fe 100644
--- a/src/gui/piano_roll.cpp
+++ b/src/gui/piano_roll.cpp
@@ -599,7 +599,7 @@ pianoRoll::pianoRoll() :
 	if( engine::mainWindow()->workspace() )
 	{
 		engine::mainWindow()->workspace()->addSubWindow( this );
-		parentWidget()->setMinimumWidth( tb_layout->minimumSize().width()+10 );
+		parentWidget()->setMinimumSize( tb_layout->minimumSize().width()+10, 200 );
 		parentWidget()->resize( tb_layout->minimumSize().width()+10,
 						INITIAL_PIANOROLL_HEIGHT );
 		parentWidget()->hide();
