diff -ur spdlog-0.11.0.old/CMakeLists.txt spdlog-0.11.0/CMakeLists.txt
--- spdlog-0.11.0.old/CMakeLists.txt	2016-09-17 20:28:41.000000000 -0300
+++ spdlog-0.11.0/CMakeLists.txt	2017-01-09 16:01:16.795518326 -0200
@@ -36,9 +36,9 @@
 # * https://github.com/forexample/package-example
 set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
 
-set(config_install_dir "lib/cmake/${PROJECT_NAME}")
+set(config_install_dir "lib${LIB_SUFFIX}/cmake/${PROJECT_NAME}")
 set(include_install_dir "include")
-set(pkgconfig_install_dir "lib/pkgconfig")
+set(pkgconfig_install_dir "lib${LIB_SUFFIX}/pkgconfig")
 
 set(version_config "${generated_dir}/${PROJECT_NAME}ConfigVersion.cmake")
 set(project_config "${generated_dir}/${PROJECT_NAME}Config.cmake")
