Index: /trunk/xbmc/cores/DllLoader/exports/wrapper.c
===================================================================
--- xbmc/cores/DllLoader/exports/wrapper.c	(revision 23049)
+++ xbmc/cores/DllLoader/exports/wrapper.c	(revision 25862)
@@ -77,5 +77,5 @@
 int dll_fseek64(FILE *stream, off64_t offset, int origin);
 long dll_ftell(FILE *stream);
-off64_t ftell64(FILE *stream);
+off64_t dll_ftell64(FILE *stream);
 void dll_rewind(FILE* stream);
 int dll_fgetpos(FILE* stream, fpos_t* pos);
