diff -Naur xbmc-10.00.orig/xbmc/lib/libsquish/Makefile.in xbmc-10.00/xbmc/lib/libsquish/Makefile.in
--- xbmc-10.00.orig/xbmc/lib/libsquish/Makefile.in	2010-05-18 07:29:17.000000000 +0000
+++ xbmc-10.00/xbmc/lib/libsquish/Makefile.in	2010-10-29 03:17:26.000000000 +0000
@@ -8,7 +8,7 @@
 ifeq ($(ARCH),arm)
 CXXFLAGS+=-I.
 else
-CXXFLAGS+=-I. -DSQUISH_USE_SSE=2 -msse2
+CXXFLAGS+=-I. -DSQUISH_USE_SSE=1 -msse
 endif
 endif
 LIB=libsquish-@ARCH@.a
