diff -Naur xbmc-11.0-Eden_beta1.orig/lib/libsquish/Makefile.in xbmc-11.0-Eden_beta1/lib/libsquish/Makefile.in
--- xbmc-11.0-Eden_beta1.orig/lib/libsquish/Makefile.in	2011-12-23 03:12:21.000000000 +0000
+++ xbmc-11.0-Eden_beta1/lib/libsquish/Makefile.in	2012-01-02 20:14:20.792407216 +0000
@@ -15,7 +15,7 @@
 ifeq ($(findstring powerpc,$(ARCH)),powerpc)
   CXXFLAGS+=-DSQUISH_USE_ALTIVEC=1 -maltivec
 else ifeq ($(findstring x86,$(ARCH)), x86)
-  CXXFLAGS+=-DSQUISH_USE_SSE=2 -msse2
+  CXXFLAGS+=-DSQUISH_USE_SSE=1 -msse
 endif
 
 LIB=libsquish.a
@@ -30,7 +30,7 @@
 ifeq ($(findstring osx,$(ARCH)),osx)
 	g++ -m32 -DSQUISH_USE_SSE=2 -msse2 -I. $(SRCS) -dynamiclib -install_name `pwd`/libsquish-native.so -o $@
 else
-	g++ -DSQUISH_USE_SSE=2 -msse2 -I. $(SRCS) -shared -fPIC -Wl,-soname,`pwd`/libsquish-native.so -o $@
+	g++ -DSQUISH_USE_SSE=1 -msse -I. $(SRCS) -shared -fPIC -Wl,-soname,`pwd`/libsquish-native.so -o $@
 endif
 endif
 
