diff -Naur xbmc-12.2.orig/xbmc/filesystem/SMBDirectory.cpp xbmc-12.2/xbmc/filesystem/SMBDirectory.cpp
--- xbmc-12.2.orig/xbmc/filesystem/SMBDirectory.cpp	2013-05-03 05:57:41.000000000 +0000
+++ xbmc-12.2/xbmc/filesystem/SMBDirectory.cpp	2013-11-12 00:30:28.143499083 +0000
@@ -44,7 +44,7 @@
 #include "threads/SingleLock.h"
 #include "PasswordManager.h"
 
-#include <libsmbclient.h>
+#include <samba-4.0/libsmbclient.h>
 
 #if defined(TARGET_DARWIN)
 #define XBMC_SMB_MOUNT_PATH "Library/Application Support/XBMC/Mounts/"
diff -Naur xbmc-12.2.orig/xbmc/filesystem/SmbFile.cpp xbmc-12.2/xbmc/filesystem/SmbFile.cpp
--- xbmc-12.2.orig/xbmc/filesystem/SmbFile.cpp	2013-05-03 05:57:41.000000000 +0000
+++ xbmc-12.2/xbmc/filesystem/SmbFile.cpp	2013-11-12 00:30:59.577499442 +0000
@@ -27,7 +27,7 @@
 #include "PasswordManager.h"
 #include "SMBDirectory.h"
 #include "Util.h"
-#include <libsmbclient.h>
+#include <samba-4.0/libsmbclient.h>
 #include "settings/AdvancedSettings.h"
 #include "settings/GUISettings.h"
 #include "threads/SingleLock.h"
