diff -Naur xbmc-11.0-Eden_beta1.orig/Makefile.include.in xbmc-11.0-Eden_beta1/Makefile.include.in
--- xbmc-11.0-Eden_beta1.orig/Makefile.include.in	2011-12-23 03:12:21.000000000 +0000
+++ xbmc-11.0-Eden_beta1/Makefile.include.in	2012-01-02 22:07:34.114048145 +0000
@@ -32,7 +32,6 @@
 CXXFLAGS+=@CXXFLAGS@
 CFLAGS+=@CFLAGS@
 LDFLAGS+=@LDFLAGS@
-INCLUDES+=$(sort @INCLUDES@)
 INCLUDES+=-I@abs_top_srcdir@/lib
 INCLUDES+=-I@abs_top_srcdir@/xbmc
 ifneq (@USE_EXTERNAL_FFMPEG@,1)
@@ -40,6 +39,7 @@
 endif
 INCLUDES+=-I@abs_top_srcdir@/xbmc/linux
 INCLUDES+=-I@abs_top_srcdir@/xbmc/cores/dvdplayer
+INCLUDES+=$(sort @INCLUDES@)
 DEFINES+= \
 	@ARCH_DEFINES@ \
 	-D_FILE_DEFINED \
