diff -Naur xbmc-11.0.orig/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamRTMP.cpp xbmc-11.0/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamRTMP.cpp
--- xbmc-11.0.orig/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamRTMP.cpp	2012-03-21 22:07:50.000000000 +0000
+++ xbmc-11.0/xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamRTMP.cpp	2012-03-31 21:44:15.320639375 +0000
@@ -22,9 +22,7 @@
 #if (defined HAVE_CONFIG_H) && (!defined WIN32)
   #include "config.h"
 #endif
-#ifdef _WIN32
 #include "system.h" // just for HAS_LIBRTMP
-#endif
 
 #ifdef HAS_LIBRTMP
 #include "settings/AdvancedSettings.h"
