diff -Naur xbmc-Gotham_alpha9.orig/xbmc/filesystem/SMBDirectory.cpp xbmc-Gotham_alpha9/xbmc/filesystem/SMBDirectory.cpp
--- xbmc-Gotham_alpha9.orig/xbmc/filesystem/SMBDirectory.cpp	2013-10-31 16:48:45.000000000 +0000
+++ xbmc-Gotham_alpha9/xbmc/filesystem/SMBDirectory.cpp	2013-11-28 06:04:24.156764793 +0000
@@ -43,7 +43,7 @@
 #include "threads/SingleLock.h"
 #include "PasswordManager.h"
 
-#include <libsmbclient.h>
+#include <samba-4.0/libsmbclient.h>
 
 #if defined(TARGET_DARWIN)
 #define XBMC_SMB_MOUNT_PATH "Library/Application Support/XBMC/Mounts/"
diff -Naur xbmc-Gotham_alpha9.orig/xbmc/filesystem/SmbFile.cpp xbmc-Gotham_alpha9/xbmc/filesystem/SmbFile.cpp
--- xbmc-Gotham_alpha9.orig/xbmc/filesystem/SmbFile.cpp	2013-10-31 16:48:45.000000000 +0000
+++ xbmc-Gotham_alpha9/xbmc/filesystem/SmbFile.cpp	2013-11-28 06:05:11.648764834 +0000
@@ -26,7 +26,7 @@
 #include "SmbFile.h"
 #include "PasswordManager.h"
 #include "SMBDirectory.h"
-#include <libsmbclient.h>
+#include <samba-4.0/libsmbclient.h>
 #include "settings/AdvancedSettings.h"
 #include "settings/Settings.h"
 #include "threads/SingleLock.h"
