diff -Naur xbmc-9.11.orig/xbmc/FileSystem/Makefile xbmc-9.11/xbmc/FileSystem/Makefile
--- xbmc-9.11.orig/xbmc/FileSystem/Makefile	2009-10-10 13:00:16.000000000 +0000
+++ xbmc-9.11/xbmc/FileSystem/Makefile	2009-12-31 16:37:12.000000000 +0000
@@ -1,4 +1,4 @@
-INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include
+INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/lib64/dbus-1.0/include -I/usr/lib64/glib-2.0/include
 INCLUDES+=-I../lib/libcdio/libcdio/include
 
 CXXFLAGS+=-D__STDC_FORMAT_MACROS \
diff -Naur xbmc-9.11.orig/xbmc/Makefile xbmc-9.11/xbmc/Makefile
--- xbmc-9.11.orig/xbmc/Makefile	2009-10-26 12:04:25.000000000 +0000
+++ xbmc-9.11/xbmc/Makefile	2009-12-31 16:37:12.000000000 +0000
@@ -1,4 +1,4 @@
-INCLUDES+=-I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -I/usr/include/freetype2 -Icores/dvdplayer -Icores/ffmpeg
+INCLUDES+=-I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -I/usr/include/freetype2 -Icores/dvdplayer -Icores/ffmpeg -I/usr/lib64/dbus-1.0/include -I/usr/lib64/glib-2.0/include
 
 INCLUDES+=-Ilib/libUPnP/Platinum/Source/Core \
           -Ilib/libUPnP/Platinum/Source/Platinum \
diff -Naur xbmc-9.11.orig/xbmc/linux/Makefile.in xbmc-9.11/xbmc/linux/Makefile.in
--- xbmc-9.11.orig/xbmc/linux/Makefile.in	2009-10-12 10:36:28.000000000 +0000
+++ xbmc-9.11/xbmc/linux/Makefile.in	2009-12-31 16:37:12.000000000 +0000
@@ -1,6 +1,6 @@
 ARCH=@ARCH@
 
-INCLUDES=-I. -I.. -I../../guilib -I../utils -I../cores -I../FileSystem
+INCLUDES=-I. -I.. -I../../guilib -I../utils -I../cores -I../FileSystem -I/usr/lib64/dbus-1.0/include -I/usr/lib64/glib-2.0/include
 ifeq ($(findstring osx,$(ARCH)), osx)
 	INCLUDES+=-I../osx
 endif
diff -Naur xbmc-9.11.orig/xbmc/utils/Makefile xbmc-9.11/xbmc/utils/Makefile
--- xbmc-9.11.orig/xbmc/utils/Makefile	2009-10-28 01:59:14.000000000 +0000
+++ xbmc-9.11/xbmc/utils/Makefile	2009-12-31 16:37:12.000000000 +0000
@@ -1,4 +1,4 @@
-INCLUDES=-I. -I.. -I../../ -I../linux -I../cores -I../../guilib
+INCLUDES=-I. -I.. -I../../ -I../linux -I../cores -I../../guilib -I/usr/lib64/dbus-1.0/include -I/usr/lib64/glib-2.0/include
 
 SRCS=AlarmClock.cpp \
      Archive.cpp \
