--- avant-window-navigator-0.4.0.orig/configure	2010-04-11 15:29:20.000000000 +0100
+++ avant-window-navigator-0.4.0/configure	2010-04-12 17:24:42.000000000 +0100
@@ -13993,7 +13993,7 @@ fi
 
 
 LIBRARY_MODULES="glib-2.0 >= $MIN_GLIB_VERSION gthread-2.0 gobject-2.0 desktop-agnostic >= $MIN_LDA_VERSION gtk+-2.0 >= $MIN_GTK_VERSION gdk-2.0 >= $MIN_GTK_VERSION dbus-glib-1"
-DOCK_MODULES="x11 xproto xcomposite xrender xext"
+DOCK_MODULES="x11 xproto xcomposite xrender xext desktop-agnostic gtk+-2.0 dbus-glib-1"
 TASKMANAGER_MODULES="libwnck-1.0 >= $MIN_WNCK_VERSION x11 libgtop-2.0 xext"
 
 
--- avant-window-navigator-0.4.0.orig/tests/Makefile.in	2010-04-11 15:29:23.000000000 +0100
+++ avant-window-navigator-0.4.0/tests/Makefile.in	2010-04-12 21:24:56.000000000 +0100
@@ -179,7 +179,7 @@ LD = @LD@
 LDA_BINDIR = @LDA_BINDIR@
 LDA_DEFSDIR = @LDA_DEFSDIR@
 LDA_VAPIDIR = @LDA_VAPIDIR@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ -lglib-2.0 -lgtk-x11-2.0 -lgdk-x11-2.0 -lgdk_pixbuf-2.0 -lcairo -lgobject-2.0
 LIBAPPLETDIR = @LIBAPPLETDIR@
 LIBDIR = @LIBDIR@
 LIBOBJS = @LIBOBJS@
