diff -Naur xgalaga++-0.8.orig/config.cc xgalaga++-0.8/config.cc
--- xgalaga++-0.8.orig/config.cc	2008-05-21 02:14:32.000000000 +0000
+++ xgalaga++-0.8/config.cc	2009-04-02 01:57:34.000000000 +0000
@@ -4,6 +4,7 @@
 #include <pwd.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <stdlib.h>
 
 
 namespace {
diff -Naur xgalaga++-0.8.orig/explosion.cc xgalaga++-0.8/explosion.cc
--- xgalaga++-0.8.orig/explosion.cc	2004-03-07 04:52:37.000000000 +0000
+++ xgalaga++-0.8/explosion.cc	2009-04-02 01:24:06.000000000 +0000
@@ -1,5 +1,6 @@
 #include "explosion.h"
 #include "config.h"
+#include <cstdlib>
 
 
 /*
diff -Naur xgalaga++-0.8.orig/gfxinterface.cc xgalaga++-0.8/gfxinterface.cc
--- xgalaga++-0.8.orig/gfxinterface.cc	2008-05-21 02:10:31.000000000 +0000
+++ xgalaga++-0.8/gfxinterface.cc	2009-04-02 01:24:06.000000000 +0000
@@ -3,6 +3,7 @@
 #include <cmath>
 #include <stdexcept>
 #include <vector>
+#include <cstdlib>
 
 
 
diff -Naur xgalaga++-0.8.orig/menu.cc xgalaga++-0.8/menu.cc
--- xgalaga++-0.8.orig/menu.cc	2008-05-21 23:02:26.000000000 +0000
+++ xgalaga++-0.8/menu.cc	2009-04-02 01:24:06.000000000 +0000
@@ -13,6 +13,7 @@
 #include <iomanip>
 #include <stdexcept>
 #include <sstream>
+#include <cstring>
 
 
 StartMenu * StartMenu::singleton_ (0);
