diff -Naur xgalaga++-0.8.orig/managers.cc xgalaga++-0.8/managers.cc
--- xgalaga++-0.8.orig/managers.cc	2008-05-21 07:37:59.000000000 +0000
+++ xgalaga++-0.8/managers.cc	2009-04-02 01:24:06.000000000 +0000
@@ -218,9 +218,9 @@
 
 void AliensManager::Move() {
 	base_cruise_.x += base_cruise_speed_;
-	if (base_cruise_speed_ < 0 && base_cruise_.x <= g_alien_width / 2 ||
-	    base_cruise_speed_ > 0 && base_cruise_.x >= X11::Inst().WindowWidth() -
-	    g_alien_width / 2 - (max_convoy_size_ - 1) * g_aliens_hspacing) {
+	if ((base_cruise_speed_ < 0 && base_cruise_.x <= g_alien_width / 2) ||
+	    (base_cruise_speed_ > 0 && base_cruise_.x >= X11::Inst().WindowWidth() -
+	    g_alien_width / 2 - (max_convoy_size_ - 1) * g_aliens_hspacing)) {
 		base_cruise_speed_ *= -1;
 	}
 
diff -Naur xgalaga++-0.8.orig/menu.cc xgalaga++-0.8/menu.cc
--- xgalaga++-0.8.orig/menu.cc	2008-05-21 23:02:26.000000000 +0000
+++ xgalaga++-0.8/menu.cc	2009-04-02 01:24:06.000000000 +0000
@@ -225,8 +226,8 @@
 	     (look_for_player || pos.y < last_window_size_.y - LineHeight());
 	     ++it) {
 		++rank;
-		if (rank > scroll_scores_ &&
-		    !look_for_player || look_for_player && it->Name() == player_name) {
+		if ((rank > scroll_scores_ &&
+		    !look_for_player) || (look_for_player && it->Name() == player_name)) {
 			// Construct this line
 			ost.str("");
 			if (it->Value() != last_score) {
