diff -Naur xgalaga++-0.8.orig/xgalaga++.desktop xgalaga++-0.8/xgalaga++.desktop
--- xgalaga++-0.8.orig/xgalaga++.desktop        1970-01-01 00:00:00.000000000 +0000
+++ xgalaga++-0.8/xgalaga++.desktop     2009-04-02 01:24:06.000000000 +0000
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Version=1.0
+Type=Application
+Name=XGalaga++
+Comment=A classic shoot 'em up arcade game inspired by XGalaga
+TryExec=xgalaga++
+Exec=xgalaga++
+Icon=/usr/share/pixmaps/xgalaga++.xpm
+Categories=Game;ArcadeGame;
