Index: cairo-compmgr-0.3.0/configure.ac
===================================================================
--- cairo-compmgr-0.3.0.orig/configure.ac
+++ cairo-compmgr-0.3.0/configure.ac
@@ -81,6 +81,7 @@ PKG_CHECK_MODULES(CAIRO_COMPMGR,
         xdamage,
         xext,
         xi,
+        x11,
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
Index: cairo-compmgr-0.3.0/configure
===================================================================
--- cairo-compmgr-0.3.0.orig/configure
+++ cairo-compmgr-0.3.0/configure
@@ -13369,6 +13369,7 @@ if test -n "$PKG_CONFIG"; then
         xdamage,
         xext,
         xi,
+        x11,
         sm,
         cairo >= \$CAIRO_REQUIRED,
         pixman-1 >= \$PIXMAN_REQUIRED,
@@ -13378,6 +13379,7 @@ if test -n "$PKG_CONFIG"; then
         xdamage,
         xext,
         xi,
+        x11,
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
@@ -13390,6 +13392,7 @@ if test -n "$PKG_CONFIG"; then
         xdamage,
         xext,
         xi,
+        x11,
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
@@ -13411,6 +13414,7 @@ if test -n "$PKG_CONFIG"; then
         xdamage,
         xext,
         xi,
+        x11,
         sm,
         cairo >= \$CAIRO_REQUIRED,
         pixman-1 >= \$PIXMAN_REQUIRED,
@@ -13420,6 +13424,7 @@ if test -n "$PKG_CONFIG"; then
         xdamage,
         xext,
         xi,
+        x11,
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
@@ -13432,6 +13437,7 @@ if test -n "$PKG_CONFIG"; then
         xdamage,
         xext,
         xi,
+        x11,
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
@@ -13459,6 +13465,7 @@ fi
         xdamage,
         xext,
         xi,
+        x11,
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
@@ -13469,6 +13476,7 @@ fi
         xdamage,
         xext,
         xi,
+        x11,
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
@@ -13482,6 +13490,7 @@ fi
         xdamage,
         xext,
         xi,
+        x11,
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
