Index: cairo-compmgr-0.3.0/configure
===================================================================
--- cairo-compmgr-0.3.0.orig/configure
+++ cairo-compmgr-0.3.0/configure
@@ -13372,6 +13372,7 @@ if test -n "$PKG_CONFIG"; then
         sm,
         cairo >= \$CAIRO_REQUIRED,
         pixman-1 >= \$PIXMAN_REQUIRED,
+        gmodule-2.0,
         gtk+-2.0 >= \$GTK_REQUIRED
         vala-1.0 >= \$VALA_REQUIRED\""; } >&5
   ($PKG_CONFIG --exists --print-errors "xcomposite,
@@ -13381,6 +13382,7 @@ if test -n "$PKG_CONFIG"; then
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
+        gmodule-2.0,
         gtk+-2.0 >= $GTK_REQUIRED
         vala-1.0 >= $VALA_REQUIRED") 2>&5
   ac_status=$?
@@ -13393,6 +13395,7 @@ if test -n "$PKG_CONFIG"; then
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
+        gmodule-2.0,
         gtk+-2.0 >= $GTK_REQUIRED
         vala-1.0 >= $VALA_REQUIRED" 2>/dev/null`
 else
@@ -13414,6 +13417,7 @@ if test -n "$PKG_CONFIG"; then
         sm,
         cairo >= \$CAIRO_REQUIRED,
         pixman-1 >= \$PIXMAN_REQUIRED,
+        gmodule-2.0,
         gtk+-2.0 >= \$GTK_REQUIRED
         vala-1.0 >= \$VALA_REQUIRED\""; } >&5
   ($PKG_CONFIG --exists --print-errors "xcomposite,
@@ -13423,6 +13427,7 @@ if test -n "$PKG_CONFIG"; then
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
+        gmodule-2.0,
         gtk+-2.0 >= $GTK_REQUIRED
         vala-1.0 >= $VALA_REQUIRED") 2>&5
   ac_status=$?
@@ -13435,6 +13440,7 @@ if test -n "$PKG_CONFIG"; then
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
+        gmodule-2.0,
         gtk+-2.0 >= $GTK_REQUIRED
         vala-1.0 >= $VALA_REQUIRED" 2>/dev/null`
 else
@@ -13462,6 +13468,7 @@ fi
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
+        gmodule-2.0,
         gtk+-2.0 >= $GTK_REQUIRED
         vala-1.0 >= $VALA_REQUIRED"`
         else
@@ -13472,6 +13479,7 @@ fi
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
+        gmodule-2.0,
         gtk+-2.0 >= $GTK_REQUIRED
         vala-1.0 >= $VALA_REQUIRED"`
         fi
@@ -13485,6 +13493,7 @@ fi
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
+        gmodule-2.0,
         gtk+-2.0 >= $GTK_REQUIRED
         vala-1.0 >= $VALA_REQUIRED) were not met:
 
Index: cairo-compmgr-0.3.0/configure.ac
===================================================================
--- cairo-compmgr-0.3.0.orig/configure.ac
+++ cairo-compmgr-0.3.0/configure.ac
@@ -84,6 +84,7 @@ PKG_CHECK_MODULES(CAIRO_COMPMGR,
         sm,
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
+        gmodule-2.0,
         gtk+-2.0 >= $GTK_REQUIRED
         vala-1.0 >= $VALA_REQUIRED])
 AC_SUBST(CAIRO_COMPMGR_CFLAGS)
