--- Makefile.orig	2013-07-27 15:23:14.285754141 +0200
+++ Makefile	2013-07-27 15:23:44.974756110 +0200
@@ -2,7 +2,7 @@
 INSTALLPROG=install
 INSTALLROOT=
 INSTALLARGS=
-PREFIX=/usr/local
+PREFIX=${DESTDIR}/usr
 BINDIR=bin
 BINPERM=0755
 # Might use: MANDIR=share/man
@@ -32,12 +32,14 @@ all: $(SCRIPTNAME)
 install: all install-bin install-man
 
 install-bin: $(SCRIPTNAME)
+	mkdir -m755 -p $(PREFIX)/$(BINDIR)
 	$(INSTALLPROG) -m $(BINPERM) $(INSTALLARGS) $(SCRIPTNAME) $(INSTALLROOT)$(PREFIX)/$(BINDIR)/./
 
 # making the man-page is dependent upon files not distributed, so they're
 # regenerated, so we don't list it as a dependency here -- instead we
 # assume that the maintainer created it for us (as a tarball depenency)
 install-man:
+	mkdir -m755 -p $(PREFIX)/$(MANDIR)/$(MANSECTDIR)
 	$(INSTALLPROG) -m $(MANPERM) $(INSTALLARGS) $(MANPAGE) $(INSTALLROOT)$(PREFIX)/$(MANDIR)/$(MANSECTDIR)/./
 
 bin $(SCRIPTNAME): $(SCRIPTDIST)
