From a44f5fcf0779b84c75bf3c6c99f1c659c1da0d14 Mon Sep 17 00:00:00 2001
From: Robby Workman <rworkman@slackware.com>
Date: Tue, 27 May 2014 22:34:24 -0500
Subject: [PATCH 4/4] Install icon to $datadir/icons/hicolor/scalable/apps

---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0515bd6..1362c89 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -200,9 +200,9 @@ ENDIF()
 
 # If it begins with a '/'
 IF( IS_ABSOLUTE ${DATAROOTDIR} )
-   SET( brewtarget_ICONS_LOCATION "${DATAROOTDIR}/icons/${CMAKE_PROJECT_NAME}" )
+   SET( brewtarget_ICONS_LOCATION "${DATAROOTDIR}/icons/hicolor/scalable/apps/" )
 ELSE()
-   SET( brewtarget_ICONS_LOCATION "/${DATAROOTDIR}/icons/${CMAKE_PROJECT_NAME}" )
+   SET( brewtarget_ICONS_LOCATION "/${DATAROOTDIR}/icons/hicolor/scalable/apps/" )
 ENDIF()
 
 CONFIGURE_FILE( brewtarget.desktop.in brewtarget.desktop )
@@ -488,7 +488,7 @@ IF( NOT ${BUILD_DESIGNER_PLUGINS} )
    IF( UNIX AND NOT APPLE )
       # Install the icons.
       INSTALL( FILES ${brewtarget_ICONS}
-               DESTINATION "${DATAROOTDIR}/icons/${CMAKE_PROJECT_NAME}" )
+               DESTINATION "${DATAROOTDIR}/icons/hicolor/scalable/apps/" )
 
       # Install the .desktop file.
       INSTALL( FILES ${brewtarget_DESKTOP}
-- 
1.8.4

