diff -Naur flightgear-2020.1.2/CMakeLists.txt flightgear-2020.1.2.new/CMakeLists.txt
--- flightgear-2020.1.2/CMakeLists.txt	2020-05-24 11:14:48.000000000 -0400
+++ flightgear-2020.1.2.new/CMakeLists.txt	2020-06-01 09:51:41.076758051 -0400
@@ -83,13 +83,13 @@
 set(CMAKE_MINSIZEREL_POSTFIX     ""  CACHE STRING "add a postfix, usually empty on windows")
 
 # read 'version' file into a variable (stripping any newlines or spaces)
-file(READ version versionFile)
+file(READ flightgear-version versionFile)
 if (NOT versionFile)
     message(FATAL_ERROR "Unable to determine FlightGear version. Version file is missing.")
 endif()
 string(STRIP "${versionFile}" FLIGHTGEAR_VERSION)
 # add a dependency on the version file
-set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS version)
+set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS flightgear-version)
 
 # FlightGear packaging (to build a source tarball)
 include( ConfigureCPack )
diff -Naur flightgear-2020.1.2/flightgear-version flightgear-2020.1.2.new/flightgear-version
--- flightgear-2020.1.2/flightgear-version	1969-12-31 19:00:00.000000000 -0500
+++ flightgear-2020.1.2.new/flightgear-version	2020-05-24 11:14:48.000000000 -0400
@@ -0,0 +1 @@
+2020.1.2
diff -Naur flightgear-2020.1.2/version flightgear-2020.1.2.new/version
--- flightgear-2020.1.2/version	2020-05-24 11:14:48.000000000 -0400
+++ flightgear-2020.1.2.new/version	1969-12-31 19:00:00.000000000 -0500
@@ -1 +0,0 @@
-2020.1.2
