diff -Naur flightgear-2020.1.1/CMakeLists.txt flightgear-2020.1.1.new/CMakeLists.txt
--- flightgear-2020.1.1/CMakeLists.txt	2020-05-14 14:13:11.836478728 -0400
+++ flightgear-2020.1.1.new/CMakeLists.txt	2020-05-14 14:29:27.692261758 -0400
@@ -83,7 +83,7 @@
 set(CMAKE_MINSIZEREL_POSTFIX     ""  CACHE STRING "add a postfix, usually empty on windows")
 
 # read 'version' file into a variable (stripping any newlines or spaces)
-file(READ version versionFile)
+file(READ flightgear-version versionFile)
 if (NOT versionFile)
     message(FATAL_ERROR "Unable to determine FlightGear version. Version file is missing.")
 endif()
diff -Naur flightgear-2020.1.1/flightgear-version flightgear-2020.1.1.new/flightgear-version
--- flightgear-2020.1.1/flightgear-version	1969-12-31 19:00:00.000000000 -0500
+++ flightgear-2020.1.1.new/flightgear-version	2020-05-10 16:32:27.000000000 -0400
@@ -0,0 +1 @@
+2020.1.1
diff -Naur flightgear-2020.1.1/version flightgear-2020.1.1.new/version
--- flightgear-2020.1.1/version	2020-05-10 16:32:27.000000000 -0400
+++ flightgear-2020.1.1.new/version	1969-12-31 19:00:00.000000000 -0500
@@ -1 +0,0 @@
-2020.1.1
