diff -Naur flightgear-2018.3.2/src/Environment/realwx_ctrl.cxx flightgear-2018.3.2.new/src/Environment/realwx_ctrl.cxx
--- flightgear-2018.3.2/src/Environment/realwx_ctrl.cxx	2019-01-29 08:17:50.000000000 -0500
+++ flightgear-2018.3.2.new/src/Environment/realwx_ctrl.cxx	2019-02-05 07:11:40.532121269 -0500
@@ -426,7 +426,7 @@
 )
 {
   static const std::string NOAA_BASE_URL =
-    "http://tgftp.nws.noaa.gov/data/observations/metar/stations/";
+    "https://tgftp.nws.noaa.gov/data/observations/metar/stations/";
   class NoaaMetarGetRequest:
     public simgear::HTTP::MemoryRequest
   {
diff -Naur flightgear-2018.3.2/src/Main/metar_main.cxx flightgear-2018.3.2.new/src/Main/metar_main.cxx
--- flightgear-2018.3.2/src/Main/metar_main.cxx	2019-01-29 08:17:50.000000000 -0500
+++ flightgear-2018.3.2.new/src/Main/metar_main.cxx	2019-02-05 07:11:40.533121277 -0500
@@ -539,7 +539,7 @@
 			try
 			{
               static const std::string NOAA_BASE_URL =
-                "http://tgftp.nws.noaa.gov/data/observations/metar/stations/";
+                "https://tgftp.nws.noaa.gov/data/observations/metar/stations/";
                 HTTP::MemoryRequest* mr = new HTTP::MemoryRequest
                 (
                     NOAA_BASE_URL
diff -Naur flightgear-2018.3.2/utils/fgcom/utils/build_fgcom_server.sh flightgear-2018.3.2.new/utils/fgcom/utils/build_fgcom_server.sh
--- flightgear-2018.3.2/utils/fgcom/utils/build_fgcom_server.sh	2019-01-29 08:17:50.000000000 -0500
+++ flightgear-2018.3.2.new/utils/fgcom/utils/build_fgcom_server.sh	2019-02-05 07:11:40.533121277 -0500
@@ -6,7 +6,7 @@
 DAHDI_SRC="http://downloads.asterisk.org/pub/telephony/dahdi-linux-complete/dahdi-linux-complete-current.tar.gz"
 LOGSEP="###########################################"
 GETMETAR_SCRIPT="#!/bin/bash
-#curl http://tgftp.nws.noaa.gov/data/observations/metar/stations/$1.TXT
+#curl https://tgftp.nws.noaa.gov/data/observations/metar/stations/$1.TXT
 echo \"Hello World !\""
 
 ROOT=$PWD
diff -Naur flightgear-2018.3.2/utils/metarproxy/README flightgear-2018.3.2.new/utils/metarproxy/README
--- flightgear-2018.3.2/utils/metarproxy/README	2019-01-29 08:17:50.000000000 -0500
+++ flightgear-2018.3.2.new/utils/metarproxy/README	2019-02-05 07:10:25.154570061 -0500
@@ -266,13 +266,13 @@
 =====================
 Download addresses for the last 24 hours:
 
-  http://tgftp.nws.noaa.gov/data/observations/metar/cycles/
+  https://tgftp.nws.noaa.gov/data/observations/metar/cycles/
   ftp://tgftp.nws.noaa.gov/data/observations/metar/cycles/
 
 Addresses for the most recent METAR data strings of particular
 stations:
 
-      http://tgftp.nws.noaa.gov/pub/data/observations/metar/stations/
+      https://tgftp.nws.noaa.gov/pub/data/observations/metar/stations/
       ftp://tgftp.nws.noaa.gov/data/observations/metar/stations/
 
 
diff -Naur flightgear-2018.3.2/utils/metarproxy/metarproxy flightgear-2018.3.2.new/utils/metarproxy/metarproxy
--- flightgear-2018.3.2/utils/metarproxy/metarproxy	2019-01-29 08:17:50.000000000 -0500
+++ flightgear-2018.3.2.new/utils/metarproxy/metarproxy	2019-02-05 07:11:40.534121284 -0500
@@ -81,7 +81,7 @@
        \$ http_proxy= metarproxy --record LOXL
 
 Sources:
-       http://tgftp.nws.noaa.gov/data/observations/metar/{stations,cycles}/
+       https://tgftp.nws.noaa.gov/data/observations/metar/{stations,cycles}/
        ftp://tgftp.nws.noaa.gov/data/observations/metar/{stations,cycles}/
 EOF
 
@@ -399,7 +399,7 @@
 			s/\s+$//;
 			&log($DEBUG, $_);
 
-			if (m|^GET\s+http://tgftp.nws.noaa.gov/.*/([A-Z][A-Z0-9]{3}).TXT\s+HTTP/|) {
+			if (m|^GET\s+https://tgftp.nws.noaa.gov/.*/([A-Z][A-Z0-9]{3}).TXT\s+HTTP/|) {
 				$icao = $1;
 			} elsif (/X-Time: (\d+)/) {
 				$epoch = $1;
