--- IO/Image/vtkSEPReader.h.orig	2023-02-15 14:03:53.000000000 +1000
+++ IO/Image/vtkSEPReader.h	2023-06-11 17:30:00.164000000 +1000
@@ -26,6 +26,7 @@
 #include "vtkNew.h" // for ivars
 
 #include <array>  // for std::array
+#include <cstdint>
 #include <string> // for std::string
 
 namespace details
--- ThirdParty/libproj/vtklibproj/src/proj_json_streaming_writer.hpp.orig	2023-02-15 14:03:53.000000000 +1000
+++ ThirdParty/libproj/vtklibproj/src/proj_json_streaming_writer.hpp	2023-06-11 17:42:36.366000000 +1000
@@ -32,6 +32,7 @@
 /*! @cond Doxygen_Suppress */
 
 #include <vector>
+#include <cstdint>
 #include <string>
 
 #define CPL_DLL
