--- ../Common/Core/vtkGenericDataArrayLookupHelper.h.orig	2021-07-03 01:31:21.000000000 +1000
+++ ../Common/Core/vtkGenericDataArrayLookupHelper.h	2021-08-22 22:38:52.682000000 +1000
@@ -22,6 +22,7 @@
 #ifndef vtkGenericDataArrayLookupHelper_h
 #define vtkGenericDataArrayLookupHelper_h
 
+#include <limits>
 #include "vtkIdList.h"
 #include <algorithm>
 #include <cmath>
--- ../Common/DataModel/vtkPiecewiseFunction.h.orig	2021-07-03 01:31:21.000000000 +1000
+++ ../Common/DataModel/vtkPiecewiseFunction.h	2021-08-22 22:40:08.253000000 +1000
@@ -37,6 +37,7 @@
 #ifndef vtkPiecewiseFunction_h
 #define vtkPiecewiseFunction_h
 
+#include <limits>
 #include "vtkCommonDataModelModule.h" // For export macro
 #include "vtkDataObject.h"
 
--- ../Rendering/Core/vtkColorTransferFunction.h.orig	2021-07-03 01:31:21.000000000 +1000
+++ ../Rendering/Core/vtkColorTransferFunction.h	2021-08-22 22:40:42.364000000 +1000
@@ -36,6 +36,7 @@
 #ifndef vtkColorTransferFunction_h
 #define vtkColorTransferFunction_h
 
+#include <limits>
 #include "vtkRenderingCoreModule.h" // For export macro
 #include "vtkScalarsToColors.h"
 
--- ../Filters/HyperTree/vtkHyperTreeGridThreshold.h.orig	2021-07-03 01:31:21.000000000 +1000
+++ ../Filters/HyperTree/vtkHyperTreeGridThreshold.h	2021-08-22 22:46:16.135000000 +1000
@@ -40,6 +40,7 @@
 #ifndef vtkHyperTreeGridThreshold_h
 #define vtkHyperTreeGridThreshold_h
 
+#include <limits>
 #include "vtkFiltersHyperTreeModule.h" // For export macro
 #include "vtkHyperTreeGridAlgorithm.h"
 
