--- Makefile.in.orig	2014-08-10 08:46:19.947934827 +0700
+++ Makefile.in	2014-08-10 08:46:38.961957767 +0700
@@ -465,14 +465,14 @@
 	mv $(DOC) doc
 
 install:	sqlite libsqlite.la sqlite.h
-	$(INSTALL) -d $(DESTDIR)$(exec_prefix)/lib
-	$(LTINSTALL) libsqlite.la $(DESTDIR)$(exec_prefix)/lib
+	$(INSTALL) -d $(DESTDIR)$(exec_prefix)/lib64
+	$(LTINSTALL) libsqlite.la $(DESTDIR)$(exec_prefix)/lib64
 	$(INSTALL) -d $(DESTDIR)$(exec_prefix)/bin
 	$(LTINSTALL) sqlite $(DESTDIR)$(exec_prefix)/bin
 	$(INSTALL) -d $(DESTDIR)$(prefix)/include
 	$(INSTALL) -m 0644 sqlite.h $(DESTDIR)$(prefix)/include
-	$(INSTALL) -d $(DESTDIR)$(exec_prefix)/lib/pkgconfig; 
-	$(INSTALL) -m 0644 sqlite.pc $(DESTDIR)$(exec_prefix)/lib/pkgconfig; 
+	$(INSTALL) -d $(DESTDIR)$(exec_prefix)/lib64/pkgconfig; 
+	$(INSTALL) -m 0644 sqlite.pc $(DESTDIR)$(exec_prefix)/lib64/pkgconfig; 
 
 clean:	
 	rm -f *.lo *.la *.o sqlite@TARGET_EXEEXT@ libsqlite.la sqlite.h opcodes.*
