diff -Naur cicb-5.1.3b4+git20251030/CMakeLists.txt cicb-5.1.3b4+git20251030.patched/CMakeLists.txt
--- cicb-5.1.3b4+git20251030/CMakeLists.txt	2026-07-31 03:53:23.494194874 -0400
+++ cicb-5.1.3b4+git20251030.patched/CMakeLists.txt	2026-07-31 04:04:26.797192344 -0400
@@ -33,13 +33,6 @@
     message(STATUS "SSL support disabled via ENABLE_SSL=OFF")
 endif()
 
-if(ENABLE_SSL AND OPENSSL_FOUND)
-	set(DEFAULT_SSL_PORT 7326 CACHE STRING "Default SSL port")
-	target_include_directories(icb PRIVATE ${OPENSSL_INCLUDE_DIR})
-	target_link_libraries(icb PRIVATE ${OPENSSL_SSL_LIBRARY} ${OPENSSL_CRYPTO_LIBRARY})
-	target_compile_definitions(icb PRIVATE HAVE_OPENSSL)
-endif()
-
 # Define the custom command to generate the C file from the text file
 add_custom_command(
     OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/tcl_init.string
@@ -80,7 +73,14 @@
 		DEFAULT_HOST=\"${DEFAULT_HOST}\"
 )
 
-target_link_libraries(icb ${TCL_LIBRARY} ${READLINE_LIBRARY})
+if(ENABLE_SSL AND OPENSSL_FOUND)
+	set(DEFAULT_SSL_PORT 7326 CACHE STRING "Default SSL port")
+	target_include_directories(icb PRIVATE ${OPENSSL_INCLUDE_DIR})
+	target_link_libraries(icb ${OPENSSL_SSL_LIBRARY} ${OPENSSL_CRYPTO_LIBRARY} ${TCL_LIBRARY} ${READLINE_LIBRARY})
+	target_compile_definitions(icb PRIVATE HAVE_OPENSSL)
+else()
+	target_link_libraries(icb ${TCL_LIBRARY} ${READLINE_LIBRARY})
+endif()
 
 #find_library(HISTORY_LIBRARY history)
 #if(HISTORY_LIBRARY)
