diff -ur libgraphqlparser-0.6.0.old/CMakeLists.txt libgraphqlparser-0.6.0/CMakeLists.txt
--- libgraphqlparser-0.6.0.old/CMakeLists.txt	2016-12-22 18:39:29.000000000 -0200
+++ libgraphqlparser-0.6.0/CMakeLists.txt	2017-02-21 22:51:26.509154832 -0300
@@ -99,7 +99,7 @@
   syntaxdefs.h
   DESTINATION include/graphqlparser)
 INSTALL(TARGETS graphqlparser
-  LIBRARY DESTINATION lib)
+	LIBRARY DESTINATION lib${LIB_SUFFIX})
 
 if (UNIX)
   # generate pkgconfig file
@@ -108,7 +108,7 @@
     # generate .pc and install
     configure_file("libgraphqlparser.pc.in" "libgraphqlparser.pc" @ONLY)
     install(FILES       "${CMAKE_CURRENT_BINARY_DIR}/libgraphqlparser.pc"
-            DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
+	    DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/pkgconfig")
   endif()
 endif()
 
diff -ur libgraphqlparser-0.6.0.old/libgraphqlparser.pc.in libgraphqlparser-0.6.0/libgraphqlparser.pc.in
--- libgraphqlparser-0.6.0.old/libgraphqlparser.pc.in	2017-02-21 23:00:17.928182153 -0300
+++ libgraphqlparser-0.6.0/libgraphqlparser.pc.in	2016-12-22 18:39:29.000000000 -0200
@@ -1,6 +1,6 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=${exec_prefix}/lib@LIB_SUFFIX@
 includedir=${prefix}/include/graphqlparser
 
 Name: @CMAKE_PROJECT_NAME@
